# METABRIC: Associate Epcam+ inflammatory with survival

Associate the signature with outcome in MET: Load in MET right now

```{r}
#load("../data/METABRIC/GE_19215x1992_sym_MADscaled.RData")
# RNA data here
METrsem=read.delim("../data/METABRIC/data_mRNA_median_all_sample_Zscores.txt", sep="\t")
#METb=read.delim("../data/METABRIC/data_mRNA_median_all_sample_Zscores.txt", sep="\t")
#METrsem=read.delim("../data/MET/BRCA.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt", sep="\t")
rownames(METrsem)=METrsem[ ,1]
METrsem=METrsem[ ,-c(1:2)]
colnames(METrsem)=gsub("\\.", "-", colnames(METrsem))

# load the clinical information
METclin=read.delim("../data/METABRIC/brca_metabric_clinical_data.tsv", sep="\t")
METclin$Sample.ID=gsub("-", "", METclin$Sample.ID)
M2=read.delim("../data/METABRIC/ali_2016_supplement_data.txt", sep="\t")

METclin=merge(METclin, M2, by.x="Sample.ID", by.y="inputsample", all.x=T)

boxplot(METclin$Relapse.Free.Status..Months.~METclin$Chemotherapy+METclin$X3.Gene.classifier.subtype)
```


Edit the clinical data to make sure data is censored at 60 months (5 years) for easier comparisons


```{r}
#m1=match(colnames(METssgsea), METclin$Patient.ID)
METclin$Overall.Survival..Months.=as.numeric(METclin$Overall.Survival..Months.)
METclin$Relapse.Free.Status..Months.=as.numeric(METclin$Relapse.Free.Status..Months.)
METclin$Overall.Survival.Status[which(METclin$Overall.Survival..Months.>=60)]="0:LIVING"
METclin$Overall.Survival..Months.[which(METclin$Overall.Survival..Months.>=60)]=60
METclin$Pam50...Claudin.low.subtype[which(METclin$Pam50...Claudin.low.subtype=="NC")]=NA
METclin$Relapse.Free.Status[which(METclin$Relapse.Free.Status..Months.>=60)]="0:Not Recurred"
METclin$Relapse.Free.Status..Months.[which(METclin$Relapse.Free.Status..Months.>=60)]=60

```

## Associating CD74 with phenotype and outcome

The Thorsson data has pre-calculated scores for:

* immune subtypes
* leukocyte fractins
* proportion of data with coding mutations
* TCR shannon index

Z-Scale CD74 scores prior to analysis:

```{r 5dmet, fig.cap="CD74 assoc with patient data"}
#METssgsea=scale(METrsem[match("CD74", rownames(METrsem)), ])
m1=match(colnames(METrsem), METclin$Patient.ID)

tinfo=data.frame(pam=METclin$Pam50...Claudin.low.subtype[m1], OSM=as.numeric(METclin$Overall.Survival..Months.[m1]),
                 pam2=METclin$Pam50...Claudin.low.subtype[m1],
                 OSS=METclin$Overall.Survival.Status[m1], cd74=t(METrsem["CD74",  ]),
                 DFS=METclin$Relapse.Free.Status[m1],
                 DFM=as.numeric(METclin$Relapse.Free.Status..Months.[m1]),
                 Stage=METclin$Tumor.Stage[m1],
                 Size=METclin$Tumor.Size[m1],
                 CD8=METclin$tcellscd8[m1],
                 cytolytic=METclin$cyt_mb[m1],
                 cellularity=METclin$Cellularity[m1],
                 codingMut=METclin$Mutation.Count[m1])
tinfo$pam2[which(tinfo$pam2=="claudin-low")]="Basal"

par(mfrow=c(2,3))
boxplot(tinfo$CD74~tinfo$pam, ylab="gene exp", main="PAM50")
boxplot(tinfo$CD74~tinfo$cellularity, ylab="gene exp", main="Immune sbtype")
ax=cor.test(tinfo$CD74, log10(tinfo$codingMut+1), method="spearman")
smoothScatter(tinfo$CD74, log10(tinfo$codingMut+1), xlab="gene exp", ylab="log10 mut")
text(-3,0.5, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))
ax=cor.test(tinfo$CD74, tinfo$CD8, method="spearman")
smoothScatter(tinfo$CD74, (tinfo$CD8), xlab="gene exp", ylab="CD8 frac")
text(-3,0.2, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))

ax=cor.test(tinfo$CD74, tinfo$cytolytic, method="spearman")
smoothScatter(tinfo$CD74, (tinfo$cytolytic), xlab="gene exp", ylab="cytMB")
text(-3,10, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))

hist(tinfo$CD74)

#write.csv(tinfo, file="nature-tables/5d-MET_data.csv")
```

We can also check if there is an association with survival:

```{r}
axD=Surv(tinfo$DFM, ifelse(tinfo$DFS=="1:Recurred", 1, ifelse(tinfo$DFS=="0:Not Recurred", 0, NA)))

axO=Surv(tinfo$OSM, ifelse(tinfo$OSS=="1:DECEASED", 1, ifelse(tinfo$OSS=="0:LIVING", 0, NA)))

SurvOSS=coxph(axO~tinfo$pam2+factor(tinfo$Stage)+tinfo$CD74)
SurvDFS=coxph(axD~tinfo$pam2+tinfo$Stage+tinfo$CD74)

ao=summary(SurvOSS)
ao
as=summary(SurvDFS)
as
```
Note that CD74 is associated with survival: The harzards ratio is `r ao$coefficients[5,2]` (95CI:`r ao$conf.int[ 5,3:4]`), p value `r ao$logtest[3]`. 

This means that this is associated with survival regardless of subtype. We can look at these values in forestplots

```{r 5emet, fig.cap="forest plot CD74", eval=F}

data1=data.frame(X1=c("","PAM50", "", "", "", "", "Stage", "","","", "CD74"),
                X2=c("","Basal", "Her2", "LumA", "LumB", "Normal" ,"1", "2", "3", "4", ""),
                X3=c("NSamp",  table(tinfo$pam), table(tinfo$Stage), length(na.omit(tinfo$CD74))),
                X4=c("HR", "", round(summary(SurvOSS)$coefficients[ 1:4,2],2), "",
                     round(summary(SurvOSS)$coefficients[ 5:8,2],2)))

mdata=data.frame(summary(SurvOSS)$conf.int[ ,c(1,3:4)])
mdata=rbind(NA,NA, mdata[1:4, ], NA, mdata[5:8, ])

#pdf("figure-outputs/Figure5_CD74_forestplot.pdf", width=5, height=5)

forestplot(data1, mdata, xlog=T,
           boxsize=0.5)

#write.csv(cbind(data1, mdata), file="nature-tables/5e_forest_plot.csv")

DT::datatable(cbind(data1, mdata), rownames=F, class='cell-border stripe',
          extensions="Buttons", options=list(dom="Bfrtip", buttons=c('csv', 'excel')))

```





We can look below what the association with subtype is in KM curves:

```{r Ext5e-osmet, fig.cap="OS CD74 by subtype", fig.height=8}
par(mfrow=c(2,2))

Xind=c("Basal", "Her2", "LumA", "LumB")

for (i in Xind){
  l1=which(tinfo$pam==i)
  ax=Surv(tinfo$OSM[l1], ifelse(tinfo$OSS[l1]=="1:DECEASED", 1, ifelse(tinfo$OSS[l1]=="0:LIVING", 0, NA)))
  METvalCut=cut(tinfo$CD74[l1], quantile(tinfo$CD74[l1], c(0, 0.33, 0.67, 1), na.rm=T), c("L","M", "P"))
  ax1=summary(coxph(ax~METvalCut+factor(tinfo$Stage[l1])))
  ax2=plot(survfit(ax~METvalCut), main=paste(i, "CD74 gene exp"), col=brewer.pal(3, "Blues"),lwd=2, xlab="Time (months)", ylab="OS", mark.time=T)
  text(20, 0.2, sprintf("HR:%s (%s-%s) p=%s", round(ax1$coefficients[2,2 ],2), 
                     round(ax1$conf.int[2,3 ],2), round(ax1$conf.int[2, 4],2),round(ax1$logtest[3],2)))
}
```


```{r Ext5e-met, fig.cap="DFS CD74 by subtype"}

par(mfrow=c(2,2))

for (i in Xind){
   l1=which(tinfo$pam==i)
  ax=Surv(tinfo$DFM[l1], ifelse(tinfo$DFS[l1]=="1:Recurred", 1, ifelse(tinfo$DFS[l1]=="0:Not Recurred", 0, NA)))
  METvalCut=cut(tinfo$CD74[l1], quantile(tinfo$CD74[l1], c(0, 0.33, 0.67, 1)), c("L","M", "P"))
  ax1=summary(coxph(ax~METvalCut+tinfo$Stage[l1]))
  ax2=plot(survfit(ax~METvalCut), main=paste(i, "CD74 gene exp"), col=brewer.pal(3, "Blues"),lwd=2, xlab="Time (months)", ylab="DFS", mark.time=T)
  text(20, 0.2, sprintf("HR:%s (%s-%s) p=%s", round(ax1$coefficients[2,2 ],2), 
                     round(ax1$conf.int[2,3 ],2), round(ax1$conf.int[2, 4],2),round(ax1$logtest[3],2)))
}
```

## Associating ADMATS10 with phenotype and outcome

Do the same thing as above with the ADAMTS10 scores

### All samples

Z-Scale ADAMTS10 scores prior to analysis:

```{r ADAMTS10met, fig.cap="ADAMTS10 assoc with patient data"}
#METssgsea=scale(METrsem[match("ADAMTS10", rownames(METrsem)), ])
tinfo$ADAMTS10=t(METrsem[match("ADAMTS10", rownames(METrsem)), ])

#pdf("figure-outputs/Figure5_ADAMTS10_output.pdf", height=7, width=9)

par(mfrow=c(2,3))

boxplot(tinfo$ADAMTS10~tinfo$pam, ylab="gene exp", main="PAM50")
boxplot(tinfo$ADAMTS10~tinfo$cellularity, ylab="gene exp", main="Immune sbtype")
ax=cor.test(tinfo$ADAMTS10, log10(tinfo$codingMut+1), method="spearman")
smoothScatter(tinfo$ADAMTS10, log10(tinfo$codingMut+1), xlab="gene exp", ylab="log10 mut")
text(-3,0.5, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))
ax=cor.test(tinfo$ADAMTS10, tinfo$CD8, method="spearman")
smoothScatter(tinfo$ADAMTS10, (tinfo$CD8), xlab="gene exp", ylab="leuk frac")
text(-3,0.2, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))
ax=cor.test(tinfo$ADAMTS10, tinfo$cytolytic, method="spearman")
smoothScatter(tinfo$ADAMTS10, tinfo$cytolytic, xlab="gene exp", ylab="cytolytic score")
text(4,1, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))
hist(tinfo$ADAMTS10)

DT::datatable(tinfo, rownames=F, class='cell-border stripe',
          extensions="Buttons", options=list(dom="Bfrtip", buttons=c('csv', 'excel')))

```

We can also check if there is an association with survival:

```{r adamst-survmet}
tinfo$pam2=tinfo$pam
tinfo$pam2[which(tinfo$pam2=="claudin-low")]="Basal"

axD=Surv(tinfo$DFM, ifelse(tinfo$DFS=="1:Recurred", 1, ifelse(tinfo$DFS=="0:Not Recurred", 0, NA)))

axO=Surv(tinfo$OSM, ifelse(tinfo$OSS=="1:DECEASED", 1, ifelse(tinfo$OSS=="0:LIVING", 0, NA)))

SurvOSS=coxph(axO~tinfo$pam2+tinfo$Stage+tinfo$ADAMTS10)
SurvDFS=coxph(axD~tinfo$pam2+tinfo$Stage+tinfo$ADAMTS10)

ao=summary(SurvOSS)
ao
as=summary(SurvDFS)
as
```
Note that ADAMTS10 is associated with survival: The harzards ratio is `r ao$coefficients[5,2]` (95CI:`r ao$conf.int[ 5,3:4]`), p value `r ao$logtest[3]`. 

This means that this is associated with survival regardless of subtype. We can look at these values in forestplots

```{r ADAMTS10-forestplotmet, fig.cap="forest plot ADAMTS10", eval=F}

data1=data.frame(X1=c("","PAM50", "", "", "", "", "Stage", "","","", "ADAMTS10"),
                X2=c("","Basal", "Her2", "LumA", "LumB", "Normal" ,"1", "2", "3", "4", ""),
                X3=c("NSamp",  table(tinfo$pam), table(tinfo$Stage), length(na.omit(tinfo$CD74))),
                X4=c("HR", "", round(summary(SurvOSS)$coefficients[ 1:4,2],2), "",
                     round(summary(SurvOSS)$coefficients[ 5:8,2],2)))

mdata=data.frame(summary(SurvOSS)$conf.int[ ,c(1,3:4)])
mdata=rbind(NA,NA, mdata[1:4, ], NA, mdata[5:8, ])

#pdf("figure-outputs/Figure5_CD74_forestplot.pdf", width=5, height=5)

forestplot(data1, mdata, xlog=T,
           boxsize=0.5)

#dev.off()
#write.csv(cbind(data1, mdata), file="nature-tables/5e_forest_plot_ADAMTS10.csv")

DT::datatable(cbind(data1, mdata), rownames=F, class='cell-border stripe',
          extensions="Buttons", options=list(dom="Bfrtip", buttons=c('csv', 'excel')))


```


We can look below what the association with subtype is in KM curves:

```{r Ext5e-osADmet, fig.cap="OS CD74 by subtype", fig.height=8}
par(mfrow=c(2,2))

Xind=c("Basal", "Her2", "LumA", "LumB")

for (i in Xind){
  l1=which(tinfo$pam==i)
  ax=Surv(tinfo$OSM[l1], ifelse(tinfo$OSS[l1]=="1:DECEASED", 1, ifelse(tinfo$OSS[l1]=="0:LIVING", 0, NA)))
  METvalCut=cut(tinfo$ADAMTS10[l1], quantile(tinfo$ADAMTS10[l1], c(0, 0.33, 0.67, 1)), c("L","M", "P"))
  ax1=summary(coxph(ax~METvalCut+tinfo$Stage[l1]))
  ax2=plot(survfit(ax~METvalCut), main=paste(i, "ADAMTS10 gene exp"), col=brewer.pal(3, "Blues"),lwd=2, xlab="Time (months)", ylab="OS", mark.time=T)
  text(20, 0.2, sprintf("HR:%s (%s-%s) p=%s", round(ax1$coefficients[2,2 ],2), 
                     round(ax1$conf.int[2,3 ],2), round(ax1$conf.int[2, 4],2),round(ax1$logtest[3],2)))
}
```

```{r Ext5e-dfsADmet, fig.cap="DFS ADAMTS10 by subtype", fig.height=8}

par(mfrow=c(2,2))

for (i in Xind){
   l1=which(tinfo$pam==i)
  ax=Surv(tinfo$DFM[l1], ifelse(tinfo$DFS[l1]=="1:Recurred", 1, ifelse(tinfo$DFS[l1]=="0:Not Recurred", 0, NA)))
  METvalCut=cut(tinfo$ADAMTS10[l1], quantile(tinfo$ADAMTS10[l1], c(0, 0.33, 0.67, 1)), c("L","M", "P"))
  ax1=summary(coxph(ax~METvalCut+tinfo$Stage[l1]))
  ax2=plot(survfit(ax~METvalCut), main=paste(i, "ADAMTS10 gene exp"), col=brewer.pal(3, "Blues"),lwd=2, xlab="Time (months)", ylab="DFS", mark.time=T)
  text(20, 0.2, sprintf("HR:%s (%s-%s) p=%s", round(ax1$coefficients[2,2 ],2), 
                     round(ax1$conf.int[2,3 ],2), round(ax1$conf.int[2, 4],2),round(ax1$logtest[3],2)))
}
```

### Redone using only ER cases

Redo survival and limit only to ER+ cases:

```{r}
## make a change here
tinfoER=tinfo[which(tinfo$pam%in%c("LumA", "LumB")),  ]
tinfoER$pam=factor(tinfoER$pam)

axD=Surv(tinfoER$DFM, ifelse(tinfoER$DFS=="1:Recurred", 1, ifelse(tinfoER$DFS=="0:Not Recurred", 0, NA)))

axO=Surv(tinfoER$OSM, ifelse(tinfoER$OSS=="1:DECEASED", 1, ifelse(tinfoER$OSS=="0:LIVING", 0, NA)))

SurvOSS=coxph(axO~tinfoER$pam+tinfoER$Stage+tinfoER$ADAMTS10)
SurvDFS=coxph(axD~tinfoER$pam+tinfoER$Stage+tinfoER$ADAMTS10)

ao=summary(SurvOSS)
ao
as=summary(SurvDFS)
as

```


```{r}
#pdf("figure-outputs/NEwpanelB.pdf", height=7, width=9)

par(mfrow=c(2,3))

boxplot(tinfoER$ADAMTS10~tinfoER$pam, ylab="gene exp", main="PAM50")
boxplot(tinfoER$ADAMTS10~tinfoER$cellularity, ylab="gene exp", main="cellularity")
ax=cor.test(tinfoER$ADAMTS10, log10(tinfoER$codingMut+1), method="spearman")
smoothScatter(tinfoER$ADAMTS10, log10(tinfoER$codingMut+1), xlab="gene exp", ylab="log10 mut")
text(-3,0.5, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))
ax=cor.test(tinfoER$ADAMTS10, tinfoER$CD8, method="spearman")
smoothScatter(tinfoER$ADAMTS10, (tinfoER$CD8), xlab="gene exp", ylab="leuk frac")
text(-3,0.2, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))
ax=cor.test(tinfoER$ADAMTS10, tinfoER$cytolytic, method="spearman")
smoothScatter(tinfoER$ADAMTS10, tinfoER$cytolytic, xlab="gene exp", ylab="cytolytic score")
text(-3,8, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 3)))
hist(tinfoER$ADAMTS10)


DT::datatable(tinfoER, rownames=F, class='cell-border stripe',
          extensions="Buttons", options=list(dom="Bfrtip", buttons=c('csv', 'excel')))

```

Also check the forest plots

```{r Ext5fmet, fig.cap="ADAMTS10 expression survival analysis, ER only ", eval=F}

data1=data.frame(X1=c("","PAM50", "",  "Stage", "","","", "ADAMTS10"),
                X2=c("","LumA", "LumB" ,"1", "2", "3", "4", ""),
                X3=c("NSamp",  table(tinfoER$pam), table(tinfoER$Stage), length(na.omit(tinfoER$ADAMTS10))),
                X4=c("HR", "", round(summary(SurvOSS)$coefficients[ 1,2],2), "",
                     round(summary(SurvOSS)$coefficients[ 2:5,2],2)))

mdata=data.frame(summary(SurvOSS)$conf.int[ ,c(1,3:4)])
mdata=rbind(NA,NA, mdata[1, ], NA, mdata[2:5, ])

#pdf("figure-outputs/Figure5_CD74_forestplot.pdf", width=5, height=5)

forestplot(data1, mdata, xlog=T,
           boxsize=0.5)

DT::datatable(cbind(data1, mdata), rownames=F, class='cell-border stripe',
          extensions="Buttons", options=list(dom="Bfrtip", buttons=c('csv', 'excel')))

```


## Signature: Lum cases non-inflammatory: growing vs stable

Here, use the signature which was applied before to see the difference between the non-inflammatory growing vs stable samples.


```{r, results='hide'}
#GlistGrowing=rownames(vstLumRes)[which(vstLumRes$log2FoldChange>1.5 & vstLumRes$baseMean>100 & vstLumRes$padj<0.05 )]
#Glistnon=rownames(vstLumRes)[which(vstLumRes$log2FoldChange<(-1.5) & vstLumRes$baseMean>100 & vstLumRes$padj<0.05 )]

#HumGeneList=SymHum2Rat$HGNC.symbol[match(GlistGrowing, SymHum2Rat$RGD.symbol)]
#HumGeneList2=Rat2Hum$HGNC.symbol[match(GlistGrowing, Rat2Hum$RGD.symbol)]

METssgsea=gsva(data.matrix(METrsem), list(grow=na.omit(HumGeneList2), 
                                                stab=na.omit(Rat2Hum$HGNC.symbol[match(Glistnon, Rat2Hum$RGD.symbol)]),  Onco=Oncoprint[[1]], Mamma=Oncoprint[[2]]), method="ssgsea", ssgsea.norm=F)

METssgsea=t(scale(t(METssgsea)))

par(mfrow=c(1,2))
hist(METssgsea[1, ], main="stable")
hist(METssgsea[2, ], main="growing")
plot(METssgsea[1, ], METssgsea[2, ], xlab="stable", ylab="growing")


write.csv("nature-tables/Supplemental_table5met.csv")
```

The Growing signature genes are `r Glistnon`
and stable are `r GlistGrowing`.

Below are the survival curves for the stable signature:

```{r, fig.height=8}
#pdf("figure-outputs/Fig5_survival_analysis_Lum_samples_baseMean_greater100.pdf", height=8, width = 8)

par(mfrow=c(2,2))

Xind=c("Basal", "Her2", "LumA", "LumB")

for (i in Xind){
  l1=which(METclin$Pam50...Claudin.low.subtype==i)
  lx2=match(METclin$Patient.ID[l1], colnames(METrsem))
  ax=Surv(METclin$Overall.Survival..Months.[l1], ifelse(METclin$Overall.Survival.Status[l1]=="1:DECEASED", 1, ifelse(METclin$Overall.Survival.Status[l1]=="0:LIVING", 0, NA)))
   axb=coxph(ax~METssgsea[1, lx2]+METclin$Tumor.Stage[l1])
  axc=summary(axb)
  METvalCut=cut(METssgsea[1, lx2], quantile(METssgsea[1, lx2], c(0, 0.33, 0.67, 1), na.rm=T), c("L","M", "P"))
  ax2=plot(survfit(ax~METvalCut), main=paste(i, "lum stable"), col=brewer.pal(3, "Blues"),lwd=2, xlab="Time (months)", ylab="OS", mark.time=T)
   text(10, 0, sprintf("univ. cts. var. HR=%s (%s-%s), p=%s", round(axc$coefficients[1,2], 2),
                      round(axc$conf.int[1,3],2), round(axc$conf.int[1,4],2),round(axc$logtest[3],2 )))
   legend("topleft",paste(c("L", "M", "H"), summary(METvalCut)), col=brewer.pal(3, "Blues"), lwd=1,
          pch=19)
  
}


for (i in Xind){
  l1=which(METclin$Pam50...Claudin.low.subtype==i)
  lx2=match(METclin$Patient.ID[l1], colnames(METrsem))
  ax=Surv(METclin$Relapse.Free.Status..Months.[l1], ifelse(METclin$Relapse.Free.Status[l1]=="1:Recurred", 1, ifelse(METclin$Relapse.Free.Status[l1]=="0:Not Recurred", 0, NA)))
  axb=coxph(ax~METssgsea[1, lx2]+METclin$Tumor.Stage[l1])
  axc=summary(axb)
  
  METvalCut=cut(METssgsea[1, lx2], quantile(METssgsea[1, lx2], c(0, 0.33, 0.67, 1), na.rm=T), c("L","M", "P"))
  ax2=plot(survfit(ax~METvalCut), main=paste(i, "lum stable"), col=brewer.pal(3, "Blues"),lwd=2, xlab="Time (months)", ylab="DFS", mark.time=T)
  text(10, 0, sprintf("univ. cts. var. HR=%s (%s-%s), p=%s", round(axc$coefficients[1,2], 2),
                      round(axc$conf.int[1,3],2), round(axc$conf.int[1,4],2),round(axc$logtest[3],2 )))
  legend("topleft",paste(c("L", "M", "H"), summary(METvalCut)), col=brewer.pal(3, "Blues"), lwd=1,
          pch=19)
}


#DT::datatable(as.data.frame(test), rownames=F, class='cell-border stripe',
#         extensions="Buttons", options=list(dom="Bfrtip", buttons=c('csv', 'excel')))

```

The OS curves for the growing signature

```{r 5imet, fig.cap="OS: growing signature", fig.height=8}
par(mfrow=c(2,2))

Xind=c("Basal", "Her2", "LumA", "LumB")

for (i in Xind){
  l1=which(METclin$Pam50...Claudin.low.subtype==i)
   lx2=match(METclin$Patient.ID[l1], colnames(METrsem))
  ax=Surv(METclin$Overall.Survival..Months.[l1], ifelse(METclin$Overall.Survival.Status[l1]=="1:DECEASED", 1, ifelse(METclin$Overall.Survival.Status[l1]=="0:LIVING", 0, NA)))
   axb=coxph(ax~METssgsea[2, lx2]+METclin$Tumor.Stage[l1])
  axc=summary(axb)
  
  METvalCut=cut(METssgsea[2, lx2], quantile(METssgsea[2, lx2], c(0, 0.33, 0.67, 1), na.rm=T), c("L","M", "P"))
  ax2=plot(survfit(ax~METvalCut), main=paste(i, "lum growing"), col=brewer.pal(3, "Blues"),lwd=2, xlab="Time", ylab="OS", mark.time=T)
   text(10, 0, sprintf("univ. cts. var. HR=%s (%s-%s), p=%s", round(axc$coefficients[1,2], 2),
                      round(axc$conf.int[1,3],2), round(axc$conf.int[1,4],2),round(axc$logtest[3],2 )))
   legend("topleft",paste(c("L", "M", "H"), summary(METvalCut)), col=brewer.pal(3, "Blues"), lwd=1,
          pch=19)
  
}
```

Followed by the DFS plots for this signature

```{r 5i-dfsmet, fig.cap="DFS forgrowing signature", fig.height=8}
par(mfrow=c(2,2))

for (i in Xind){
  l1=which(METclin$Pam50...Claudin.low.subtype==i)
   lx2=match(METclin$Patient.ID[l1], colnames(METrsem))
  ax=Surv(METclin$Relapse.Free.Status..Months.[l1], ifelse(METclin$Relapse.Free.Status[l1]=="1:Recurred", 1, ifelse(METclin$Relapse.Free.Status[l1]=="0:Not Recurred", 0, NA)))
   axb=coxph(ax~METssgsea[2, lx2]+METclin$Tumor.Stage[l1])
   
  axc=summary(axb)
  axc
  METvalCut=cut(METssgsea[2, lx2], quantile(METssgsea[2, lx2], c(0, 0.33, 0.67, 1), na.rm=T), c("L","M", "P"))
  ax2=plot(survfit(ax~METvalCut), main=paste(i, "lum growing"), col=brewer.pal(3, "Blues"),lwd=2, xlab="Time (months)", ylab="DFS", mark.time=T)
    text(10, 0, sprintf("univ. cts. var. HR=%s (%s-%s), p=%s", round(axc$coefficients[1,2], 2),
                      round(axc$conf.int[1,3],2), round(axc$conf.int[1,4],2),round(axc$logtest[3],2 )))
    legend("topleft",paste(c("L", "M", "H"), summary(METvalCut)), col=brewer.pal(3, "Blues"), lwd=1,
          pch=19)
  
}

 l1=which(METclin$Pam50...Claudin.low.subtype=="LumA")
 lx2=match(METclin$Patient.ID[l1], colnames(METssgsea))
  ax=Surv(METclin$Relapse.Free.Status..Months.[l1], ifelse(METclin$Relapse.Free.Status[l1]=="1:Recurred", 1, ifelse(METclin$Relapse.Free.Status[l1]=="0:Not Recurred", 0, NA)))
   axb=coxph(ax~METssgsea[2, lx2]+METclin$Tumor.Stage[l1])
   summary(axb)

   hist(METssgsea[2, lx2])
   
```

The gene lists are 

Stable:

`r GlistGrowing` 

Growing:

`r Glistnon`


We can see whether this signature associates with leukocyte content and TCR diversity in the MET cohort:

```{r Ext5i-growingmet, fig.cap="luminal signature associated with clinical variables", fig.height=8}

m1=which(METclin$Pam50...Claudin.low.subtype=="LumA")
l1=match(METclin$Patient.ID[m1], colnames(METssgsea))

#lx3=match(METclin$Patient.ID[l1], ThorssData$MET.Participant.Barcode)

#newinfoTab=data.frame(pat=METclin$Patient.ID[l1], METval=METssgsea[2, lx2],
#                      leuk=as.numeric(ThorssData$Leukocyte.Fraction[lx3]),
#                      str=as.numeric(ThorssData$Stromal.Fraction[lx3]),
#                      TCRshann=as.numeric(ThorssData$TCR.Shannon[lx3]),
#                      immSub=ThorssData$Immune.Subtype[lx3])

newinfoTab=data.frame(pat=METclin$Patient.ID[m1],
                      pam=METclin$Pam50...Claudin.low.subtype[m1], 
                      METstab=METssgsea[2, l1],
                      METgrow=METssgsea[1, l1],
                      METcomb=METssgsea[1, l1]-METssgsea[2, l1],
                      OSM=METclin$Overall.Survival..Months.[m1],
                      Mamma=METssgsea[4, l1],
                      Onco=METssgsea[3, l1],
                 OSS=METclin$Overall.Survival.Status[m1],
                 DFS=METclin$Relapse.Free.Status[m1],
                 DFM=METclin$Relapse.Free.Status..Months.[m1],
                 Stage=METclin$Tumor.Stage[m1],
                 Size=METclin$Tumor.Size[m1],
                 CD8=METclin$tcellscd8[m1],
                 cytolytic=METclin$cyt_mb[m1],
                 cellularity=METclin$Cellularity[m1],
                 codingMut=METclin$Mutation.Count[m1])

#pdf("figure-outputs/NewpanelC.pdf", height=8, width=9)
par(mfrow=c(2,2))
ax=cor.test(newinfoTab$METcomb, newinfoTab$CD8, method="spearman")
smoothScatter(newinfoTab$METcomb, newinfoTab$CD8)
text(0.6,0.6, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 2)))
ax=cor.test(newinfoTab$METcomb, newinfoTab$cytolytic, method="spearman")
smoothScatter(newinfoTab$METcomb, newinfoTab$cytolytic)
text(0.6,0.6, sprintf("cor=%s, p=%s", round(ax$estimate, 2), round(ax$p.value, 2)))

#dev.off()

DT::datatable(newinfoTab, rownames=T, class='cell-border stripe',
          extensions="Buttons", options=list(dom="Bfrtip", buttons=c('csv', 'excel'), Xscroll=T))
```

## Comparison to oncotype and mammaprint 

Finally, see how these compare to oncotype or mammaprint.

We can do this two ways:

### Overall survival

1. Run survival ~ Oncotype/Mamma/Growing + PAM50 + Stage

```{r}
SurvOS=Surv(newinfoTab$OSM, ifelse(newinfoTab$OSS=="0:LIVING", 0, 1))
SurvDFS=Surv(newinfoTab$DFM, ifelse(newinfoTab$DFS=="0:Not Recurred", 0,ifelse(newinfoTab$DFS=="1:Recurred", 1, NA) ))

model1=coxph(SurvOS~METgrow+Stage, data=newinfoTab)
summary(model1)
model2=coxph(SurvOS~METgrow+Onco+Stage, data=newinfoTab)
summary(model2)
model3=coxph(SurvOS~METgrow+Mamma+Stage, data=newinfoTab)
summary(model3)
model3=coxph(SurvOS~METgrow+Mamma+Onco+Stage, data=newinfoTab)
summary(model3)
```

2. Also try the growing signature:

```{r}
model1=coxph(SurvOS~METstab+Stage, data=newinfoTab)
summary(model1)
model2=coxph(SurvOS~METstab+Onco+Stage, data=newinfoTab)
summary(model2)
model3=coxph(SurvOS~METstab+Mamma+Stage, data=newinfoTab)
summary(model3)
model3=coxph(SurvOS~METstab+Mamma+Onco+Stage, data=newinfoTab)
summary(model3)
```

3. Combined signature:

```{r}
model1=coxph(SurvOS~METcomb+Stage, data=newinfoTab)
summary(model1)
model2=coxph(SurvOS~METcomb+Onco+Stage, data=newinfoTab)
summary(model2)
model3=coxph(SurvOS~METcomb+Mamma+Stage, data=newinfoTab)
summary(model3)
model3=coxph(SurvOS~METcomb+Mamma+Onco+Stage, data=newinfoTab)
summary(model3)
```

### MET: LumA high vs low signatures

We can pull out the samples here which have high vs low signature (in the lumA group) and perform differential gene expression analysis on these cases: Need to rerun this section usng limma

```{r MET-limma}
newinfoTab$METvalcut=cut(newinfoTab$METcomb,  quantile(newinfoTab$METcomb, c(0, 0.33, 0.67, 1), na.rm=T ), c("L", "M", "O"))

colDataMET=newinfoTab[which(newinfoTab$METvalcut!="M"), ]

#head(colDataMET)
#METdeseq=DESeqDataSetFromMatrix(METrsem[ ,match(colDataMET$pat, colnames(METrsem))], colDataMET, design=~METvalcut)
METlimma=lmFit(METrsem[ ,match(colDataMET$pat, colnames(METrsem))], design = model.matrix( ~ 0 + factor(colDataMET$METvalcut)))
METlimma <- eBayes(METlimma)
Output=topTable(METlimma , coef=2, number=nrow(METrsem))
Output2=topTable(METlimma , coef=2, number=nrow(METrsem), p.value=0.05, lfc=0.5)
#results <- decideTests(METlimma)

#write.csv(METdeseqRes, file="nature-tables/5J_MET.csv")
A1=as.data.frame(Output)

DT::datatable(A1, rownames=T, class='cell-border stripe',
          extensions="Buttons", options=list(dom="Bfrtip", buttons=c('csv', 'excel'), Xscroll=T))
write.csv(A1, file="nature-tables/MET_stable_growing.csv")
```

We can plot the differential genes here in a volcano plot:

```{r MET-volcano, fig.cap="yet another volcano plot"}
#pdf("figure-outputs/Fig5J_MET_deg.pdf", height=8, width=8)
with(Output, plot(logFC, -log10(adj.P.Val), pch=20, main="Volcano plot: Low exp (-) vs high (+)", cex=1.0, xlab=bquote(~Log[2]~fold~change), ylab=bquote(~-log[10]~Q~value)))
with(subset(Output, adj.P.Val<0.05 & abs(logFC)>0.5), points(logFC, -log10(adj.P.Val), pch=20, col="red", cex=0.5))
text(Output2$logFC+0.02, -log10(Output2$adj.P.Val)+0.05, rownames(Output2), col="red")
#with(subset(METdeseqRes, padj<0.05 & abs(log2FoldChange)>2), text(log2FoldChange+0.05, -log10(padj)+0.05, rownames(METdeseqRes), pch=20, col="red", cex=0.75))
#dev.off()
```


```{r heatmap-MET, fig.cap="because someone will ask for it"}
#vstMET=vst(assay(METdeseq))

ColSideCols=c("skyblue", "blue")[factor(colDataMET$METvalcut)]

ax1=METrsem[ match(rownames(Output2), rownames(METrsem)),  match(colDataMET$pat, colnames(METrsem))]

#pdf("figure-outputs/Figure5J_XX_heatmap_LumA_MET.pdf", height=15, width=10)

heatmap.2(data.matrix(ax1), scale="row", trace="none", ColSideColors = ColSideCols, col=RdBu[11:1], main="LumA MET")
```


Run gsea here

```{r MET-gsea, fig.cap="pointless MET hairball"}

## why doesnt this work

load("../anntotations/ListofGeneSets2.RData")

hits=rownames(Output2)
fcTab=Output$logFC

names(fcTab)=rownames(Output)
  gscaMET=GSCA(listOfGeneSetCollections=ListGSC,geneList=fcTab, hits = hits)
   gscaMET<- preprocess( gscaMET, species="Hs", initialIDs="SYMBOL",
                    keepMultipleMappings=TRUE, duplicateRemoverMethod="max",
                    orderAbsValue=FALSE)
   gscaMET <- analyze( gscaMET, 
                 para=list(pValueCutoff=0.05, pAdjustMethod="BH",
                           nPermutations=100, minGeneSetSize=5,
                           exponent=1), 
                           doGSOA = F)
   
save( gscaMET, file="figure-outputs/MET_hairball.RData")
   
TermsA=sapply(strsplit(rownames(gscaMET@result$GSEA.results$ProcessNetworks), "_"), function(x) x[2])
TermsA[which(is.na(TermsA))]=substr(rownames(gscaMET@result$GSEA.results$ProcessNetworks)[which(is.na(TermsA))], 2, 50)

## check whether this runs:
gscaMET@result$GSEA.results$ProcessNetworks$Gene.Set.Term=TermsA
viewEnrichMap(gscaMET, gscs=c("ProcessNetworks"),
              allSig = TRUE, gsNameType="term" )

write.csv(gscaMET@result$GSEA.results$ProcessNetworks, file="nature-tables/5k_MET_lumA_GSEA.csv")
```

