# Associating expression with Mutations

Also plot the expression of the corresponding genes here, if the epithelial RNA data is present. Note, with this early batch, the DN population was not sorted by FACS nor sequenced.


```{r}
# Filter out the sample names and find their RNA-sample ID
Xnames=unique(MeltCosmic$L1)
midx=Cdata$TumorID[match(Xnames, sapply(strsplit(Cdata$WGS, ".final.bam"), function(x) x[1]))]
# Filter out all gene expression data which is in the characterisation cohort
allTPMchar=allTPMFinal[ , which(infoTableFinal$Cohort!="Progression")]
infoTableFinalchar=infoTableFinal[ which(infoTableFinal$Cohort!="Progression"),]
nVSDchar=normalizedTableVSD[ , which(infoTableFinal$Cohort!="Progression")]

cnames2=gsub("_CD45","", colnames(allTPMchar))
cnames2=gsub("_Ep", "", cnames2)

lxMatch=which(cnames2 %in% midx)
lxMatch2=which(midx %in% cnames2)
```

The following samples have matching WGX data: `r colnames(allTPMchar)[lxMatch]` from samples `r midx[lxMatch2]`

## Expression of genes in common

```{r}
RNames=c("Ccn2", sapply(strsplit(rownames(Wtemp), " "), function(x) x[1]))

TtempNames=sapply(strsplit(colnames(Ttemp), " "), function(x) x[1])

ExprVSD=nVSDchar[na.omit(match(RNames, rownames(nVSDchar))), na.omit(match(TtempNames, colnames(nVSDchar)))]

par(oma=c(2,2,2,5))
heatmap.2(t(ExprVSD), trace="none", scale="none", Rowv = NA, Colv = NA, col=RdBu[11:1],
          main="gene expression, vst transformed")
heatmap.2(t(ExprVSD), trace="none", scale="col", Rowv = NA, Colv = NA, col=RdBu[11:1],
          main="gene expression, col scaled")

```

Compare RNA mut freq & expression values

```{r}
par(oma=c(2,2,2,2))
heatmap.2(t(ExprVSD), trace="none", scale="none", Rowv = NA, Colv = NA, col=RdBu[11:1],
          main="gene expression, vst transformed")
heatmap.2(t(ExprVSD), trace="none", scale="col", Rowv = NA, Colv = NA, col=RdBu[11:1],
          main="gene expression, col scaled")

# calculate the correlation between mutation and expression

mvals=rep(NA, nrow(ExprVSD))
names(mvals)=rownames(ExprVSD)
mvalsP=mvals

head(Ttemp)

# x1=match(colnames(ExprVSD), TtempNames)
# for (i in 1:length(mvals)){
#   ax1=glm(sign(Ttemp[i, x1])~ExprVSD[i, ], family =binomial(link="logit"))
#   mvals[i]=coefficients(ax1)[2]
#   mvalsP[i]=summary(ax1)$coefficients[2,4]
# }
# 
# barplot(mvalsP,las=2, ylab = "P value of association bw mutation and expression")
```

## Expression of genes with key mutations

We can still look at expression of genes of interest.
Eg. C2N may be enriched in kalikrein systen, connective tissue degradation
NMU12_LA in endothelium_leukocyte interactions, IL6 signalling
NMU12 RL in NOTCH signalling?

```{r}
# calculate z-scores for the process network pathways
PNzscore=scale(SumTable2)
PNsub=which(abs(PNzscore[Xnames[lxMatch2], ])>2, arr.ind=T)
pways=colnames(SumTable2)[unique(PNsub[ ,2])]

library(GSVA)
load("../anntotations/Metacore_extracted_Process_networks_nov2020.RData")

# perform GSVA for process networks based on epithelial samples first
# Convert teh Gene symbols to Human
mxd=SymHum2Rat$HGNC.symbol[match(rownames(nVSDchar), SymHum2Rat$RGD.symbol)]
mxd[which(is.na(mxd))]=toupper(rownames(nVSDchar)[which(is.na(mxd))])

tVar=nVSDchar[ ,grep("Ep", colnames(allTPMchar))]
rownames(tVar)=mxd

GSVAoutput1=gsva(tVar, gset.idx.list=PathwayMapAllComp, method="ssgsea", kcdf="Poisson", ssgsea.norm=T, verbose=F, min.sz=1)
# match the rownames
msets=match(gsub(" ", "", colnames(SumTable2)), gsub(" ", "", rownames(GSVAoutput1)))

gsvaVal=GSVAoutput1[msets, ]

par(oma=c(3, 1,1,3))
heatmap.2(t(gsvaVal[ lx, ]), scale="none", trace="none", col=RdBu[11:1], main="Process Networks: CD45 population", ColSideColors = cCols[lx], Colv = NA)


## plot heatmaps for the above pathways
pdf("outputs/1M_Char_cohort_pways_of_interest-Ep.pdf", height=7, width=6)
par(oma=c(3, 1,1,3))
for (i in pways){
  a1=PathwayMapAllComp[[i]]
  t2=nVSDchar[ na.omit(match(a1, mxd)) ,grep("Ep", colnames(allTPMchar))]
  rm=rowSds(t2)
  heatmap.2(t2[which(rm>1),], scale = "row", col=RdBu[11:1], trace="none", main=paste(i, "Epi"))
}
dev.off()


pdf("outputs/1M_Char_cohort_pways_of_interest-CD45.pdf", height=7, width=6)
par(oma=c(3, 1,1,3))
for (i in pways){
  a1=PathwayMapAllComp[[i]]
  t2=nVSDchar[ na.omit(match(a1, mxd)) ,grep("CD45", colnames(allTPMchar))]
   rm=rowSds(t2)
  heatmap.2(t2[which(rm>1), ], scale = "row", col=RdBu[11:1], trace="none", main=paste(i, "CD"))
}
dev.off()
```

## Expression of common mutations?

Run the following to plot this for both the cd45 fraction as well as the epithelial fraction:


```{r}

fgenes=colnames(tempz2)
#pdf("outputs/1Ja_gene-expression-common-mutations.pdf", height=8, width=5.6)

t2=nVSDchar[ na.omit(match(fgenes,rownames(nVSDchar))) ,grep("CD45", colnames(nVSDchar))]
OutputplotFun(t2, "row", "Scaled CD", "wgs")
OutputplotFun(t2, "none", "non-Scaled CD", "wgs")
  
t2=nVSDchar[ na.omit(match(fgenes,rownames(nVSDchar))) ,grep("Ep", colnames(nVSDchar))]
OutputplotFun(t2, "row", "Scaled Ep", "wgs")
OutputplotFun(t2, "none", "non-Scaled Ep", "wgs")

#dev.off()

```

### Look at cancer specific pathways


```{r}

colCols=rep("red", ncol(nVSDchar))
colCols[grep("CD45", colnames(nVSDchar))]="green"
colCols[match(names(which(km4$cluster==2)), colnames(nVSDchar))]="blue"

Cancer_pways=gsva(nVSDchar, PathwayListRat,
                    method="ssgsea", kcdf="Poisson", ssgsea.norm=T, verbose=F, min.sz=1)

pdf("outputs/1N_enriched_Cancer_signalling_pathways.pdf", width=6, height=6)
heatmap.2(Cancer_pways[-5, ], scale = "none", col=RdBu[11:1], trace="none", main="ssGSEA enrichment scores unscaled",
          ColSideColors = colCols)
x2=grep("CD45", colnames(nVSDchar))
heatmap.2(Cancer_pways[-5, -x2], scale = "none", col=RdBu[11:1], trace="none", main="ssGSEA enrichment scores unscaled",
          ColSideColors = colCols[-x2])
dev.off()

```


## Expression of immune related genes in the CD45 population

Initially do a search for the mutations shown before, also can do immune specific mutations

```{r, eval=F}
xa=colnames(nVSDchar)[grep("CD45", colnames(nVSDchar))]
lx2=ColC
names(lx2)=gsub("Ep", "CD45", names(ColC))
m1=match(xa, names(lx2))
CD45PAM=lx2[m1]
CD45PAMcol=ifelse(CD45PAM=="Lum", "blue", ifelse(CD45PAM=="Basal", "red", "grey"))
# correlation plots based on all genes

pdf("~/Dropbox (Partners HealthCare)/Carlos' leukocyte project/Manuscript/figure_panels_generated_R/Figure1/test.pdf", height=6, width=6)

Ax1=cor(nVSDchar[, xa])
heatmap.2(Ax1, ColSideColors = CD45PAMcol, trace="none", scale="none", col=RdBu[11:1],
          main="All genes")

# correlation based on RatCosmic genes
Ax2=cor(nVSDchar[na.omit(match(RatCosmic, rownames(nVSDchar))), xa])
heatmap.2(Ax2, ColSideColors = CD45PAMcol, trace="none", scale="none", col=RdBu[11:1],
          main="immune related genes")

plotPCA(vsd[, which(infoTableFinal$Fraction=="CD45" & infoTableFinal$Cohort!="Progression")],
        "Treatment")+geom_label(aes(label = name)) 

dev.off()

# correlation plot based on immune related genes


# x1=which(colnames(rGeneOut2)%in%RatAllImm)
# 
# temp=rGeneOut2[ ,x1]
# #a1=heatmap(temp, col=c("white", 1:9), scale="none", main="all mutations in cosmic")
# a2=temp[a1$rowInd, a1$colInd]
# hotspotC=hotspotLocExtract(a2)

MatchSampleCD45=paste(RNADNAsamples$Rat_ID[Midx], RNADNAsamples$Location.x[Midx],"CD45",  sep="_")

mCDsamples=match(MatchSampleCD45, colnames(assay(vsd)))

#tempVSDcd45=matrix(NA, nrow=nrow(a2), ncol=ncol(a2))

idx1=match(colnames(a2),rownames(assay(vsd)))
tempVSDcd45=t(assay(vsd)[ na.omit(idx1),na.omit(mCDsamples)])
rownames(tempVSDcd45)=rownames(a2)[which(!is.na(mCDsamples))]
a1=setdiff(colnames(a2), colnames(tempVSDcd45))
mat2=matrix(NA, ncol=length(a1), nrow=nrow(tempVSDcd45))
colnames(mat2)=a1
tempVSDcd45=cbind(tempVSDcd45, mat2)
mat3=matrix(NA, nrow=length(which(is.na(mCDsamples))), ncol=ncol(tempVSDcd45))
rownames(mat3)=rownames(a2)[which(is.na(mCDsamples))]
tempVSDcd45=rbind(tempVSDcd45, mat3)
tempVSDcd45=tempVSDcd45[match(rownames(a2), rownames(tempVSDcd45)), match(colnames(a2), colnames(tempVSDcd45))]
#tempVSDb=matrix(NA, ncol=ncol(a2), nrow=nrow(a2))
#tempVSDb[match(rownames(tempVSD), rownames(a2)), ]

#tempVSD=scale(tempVSD)
rownames(tempVSDcd45)=paste(rownames(a2), "CD45", sep="-")
#colnames(tempVSDcd45)=colnames(a2)
tempVSDcd45scale=scale(tempVSDcd45)


par(mar = c(4, 8, 4, 2),  xpd = TRUE)
image(t(tempVSDcd45scale), col=RdBu[11:1], xaxt='n', yaxt='n')
axis(1, at=seq(0, 1, length=ncol(a2)), colnames(a2), las=2, cex=0.7)
axis(2, at=seq(0, 1, length=nrow(a2)), rownames(a2), las=2, cex=0.7)
text(na.omit(hotspot1$Y-1)/(ncol(a2)-1), na.omit(hotspot1$X-1)/(nrow(a2)-1) , "*")
legend("top", inset = c(-0.5, -0.15),    legend = c("Nonsense","frameshift", "missense","splice", "hotspot human"),  pch = c(19, 19, 19,19, 8),  col = c(1:4, "black"), horiz = T)

## rm the non expressing samples

tempVSD3b=tempVSDcd45[-which(is.na(mCDsamples)), ]
MutB=a2[-which(is.na(mCDsamples)), ]

List2=which(MutB>0, arr.ind=T)
hotspotB=hotspotLocExtract(MutB)

par(mar = c(4, 8, 4, 2),  xpd = TRUE)
image(t(tempVSD3b), col=RdBu[11:1], xaxt='n', yaxt='n')
axis(1, at=seq(0, 1, length=ncol(tempVSD3b)), colnames(tempVSD3b), las=2, cex=0.7)
axis(2, at=seq(0, 1, length=nrow(tempVSD3b)), rownames(tempVSD3b), las=2, cex=0.7)
text(na.omit(List2[ ,2]-1)/(ncol(tempVSD3b)-1), na.omit(List2[ ,1]-1)/(nrow(tempVSD3b)-1) , "*")

image(t(MutB), col=c("white", 1:4), xaxt='n', yaxt='n')
axis(1, at=seq(0, 1, length=ncol(MutB)), colnames(MutB), las=2, cex=0.7)
axis(2, at=seq(0, 1, length=nrow(MutB)), rownames(MutB), las=2, cex=0.7)
text(na.omit(hotspotB$Y-1)/(ncol(MutB)-1), na.omit(hotspotB$X-1)/(nrow(MutB)-1) , "*")
```

Immune specific mutations

```{r, eval=F}
# x1=which(colnames(rGeneOut2)%in%RatAllImm)
# 
# temp=rGeneOut2[ ,x1]
# #a1=heatmap(temp, col=c("white", 1:9), scale="none", main="all mutations in cosmic")
# a2=temp[a1$rowInd, a1$colInd]
# hotspotC=hotspotLocExtract(a2)
# load("WGS_mutations_coding.RData")
# MeltWGSCoding=melt(AllMutWGScoding)
idx1=which(MeltWGSCodingB$Gene_Symbol%in%unique(unlist(Exp2RatImm[c(10:14)]))) # change this to RatAllImm
#sigNames=unlist(Exp2RatImm[c(1:5, 8:9, 15:17)])
sigNames=unlist(Exp2RatImm[c(10:14)])

rGeneOutImm=acast(MeltWGSCodingB[ idx1,c("Gene_Symbol", "L1", "Variant_Classification")], L1~Gene_Symbol, 
               value.var="Variant_Classification", fun.aggregate=function(x) paste(x, collapse = ", "))

#mxl=unique(as.vector(rGeneOut))
rGeneOut2I=rGeneOutImm
rGeneOut2I[grep("Nonsense",rGeneOut2I)]=3
rGeneOut2I[grep("Frame_Shift",rGeneOut2I)]=2
rGeneOut2I[grep("Missense_", rGeneOut2I)]=1
rGeneOut2I[grep("Splice_", rGeneOut2I)]=4
rGeneOut2I[which(rGeneOut2I=="")]=0

rGeneOut2I=apply(rGeneOut2I, 2, as.numeric)
rownames(rGeneOut2I)=rownames(rGeneOutImm)

GeneList=colSums(sign(rGeneOut2I), na.rm=T)

top30=names(sort(GeneList, decreasing=T)[1:length(GeneList)])
# do a plot of the top 30 most common mutations
temp=rGeneOut2I[ ,top30]
tempx=temp[grep("^N|C", rownames(temp)), ]
a1a=heatmap(tempx, col=c("white", 1:9), scale="none", main="all mutations in cosmic")
tempy=temp[grep("D", rownames(temp)), ]
a1b=heatmap(tempy, col=c("white", 1:9), scale="none", main="all mutations in cosmic")
a2=rbind(tempx[a1a$rowInd, a1a$colInd], tempy[a1b$rowInd, a1a$colInd])

l1=as.numeric(RNADNAsamples$Treatment[match(rownames(a2), sapply(strsplit(as.character(RNADNAsamples$WGS), ".final.bam"), function(x) x[1]))])
colSC=palette()[l1+4]

x1=order(l1)

RowSC=sigNames[match(colnames(a2), sigNames)]
RowSC=substr(names(RowSC), 1, 4)

a2=a2[x1, order(RowSC) ]

#hotspot1=hotspotLocExtract(a2)


par(mar = c(4, 8, 4, 2),  xpd = TRUE)
image(t(a2), col=c("white", 1:4), xaxt='n', yaxt='n')
axis(1, at=seq(0, 1, length=ncol(a2)), colnames(a2), las=2, cex=0.7)
axis(2, at=seq(0, 1, length=nrow(a2)), rownames(a2), las=2, cex=0.7)
#text(na.omit(hotspot1$Y-1)/(ncol(a2)-1), na.omit(hotspot1$X-1)/(nrow(a2)-1) , "*")
legend("top", inset = c(-0.5, -0.15),    legend = c("missense","frameshift","Nonsense", "splice", "hotspot human"),  pch = c(19, 19, 19,19, 8),  col = c(1:4, "black"), horiz = T)



# use specifically the cases where GeneList > 3

idx2=names(GeneList)[which(GeneList>2)]


MatchSampleCD45=paste(RNADNAsamples$Rat_ID[Midx],  RNADNAsamples$Location.x[Midx], "CD45",sep="_")

mCDsamples=match(MatchSampleCD45, colnames(assay(vsd)))

#tempVSDcd45=matrix(NA, nrow=nrow(a2), ncol=ncol(a2))

idx1=match(idx2,rownames(assay(vsd)))
tempVSDcd45b=t(assay(vsd)[ na.omit(idx1),na.omit(mCDsamples)])
rownames(tempVSDcd45b)=rownames(a2)[which(!is.na(mCDsamples))]
a1=setdiff(rownames(a2), rownames(tempVSDcd45b))
mat2=matrix(NA, nrow=length(a1), ncol=ncol(tempVSDcd45b))
rownames(mat2)=a1
tempVSDcd45b=rbind(tempVSDcd45b, mat2)

tempVSDcd45b=tempVSDcd45b[match(rownames(a2), rownames(tempVSDcd45b)), ]

#tempVSD=scale(tempVSD)
#rownames(tempVSDcd45b)=paste(rownames(a2), "CD45", sep="-")
#colnames(tempVSDcd45)=colnames(a2)
tempVSDcd45scaleb=scale(tempVSDcd45b)

l2=rGeneOut2I[ match(rownames(tempVSDcd45b), rownames(rGeneOut2I)), match(colnames(tempVSDcd45b), colnames(rGeneOut2I))]

hotspot2=which(l2>0, arr.ind = T)

par(mar = c(4, 8, 4, 2),  xpd = TRUE)
image(t(tempVSDcd45scaleb), col=RdBu[11:1], xaxt='n', yaxt='n')
axis(1, at=seq(0, 1, length=ncol(tempVSDcd45scaleb)), colnames(tempVSDcd45scaleb), las=2, cex=0.7)
axis(2, at=seq(0, 1, length=nrow(tempVSDcd45scaleb)), rownames(tempVSDcd45scaleb), las=2, cex=0.7)
text(na.omit(hotspot2[ ,2]-1)/(ncol(tempVSDcd45scaleb)-1), na.omit(hotspot2[ ,1]-1)/(nrow(tempVSDcd45scaleb)-1) , "*")
legend("top", inset = c(-0.5, -0.15),    legend = c("Nonsense","frameshift", "missense","splice", "hotspot human"),  pch = c(19, 19, 19,19, 8),  col = c(1:4, "black"), horiz = T)



```

#### Combine the two fractions together

```{r, eval=F}
# getidx=rbind(mEpsamples, mCDsamples)
# colnames(getidx)=rownames(a2)
tempV2=rbind(tempVSD, tempVSDcd45)
tempV2scale=scale((tempV2))

index=rep(c(1:16), each=2)
index[seq(2, length(index), by=2)]=index[seq(2, length(index), by=2)]+16

tempV2N=tempV2scale[index, ]


cases=rep(rownames(a2), each=2)

MutB=a2[match(cases, rownames(a2)) ,]
hotspotB=hotspotLocExtract(MutB)
List2=which(MutB>0, arr.ind=T)


par(mar = c(4, 8, 4, 2),  xpd = TRUE)
image(t(tempV2N), col=RdBu[11:1], xaxt='n', yaxt='n')
axis(1, at=seq(0, 1, length=ncol(tempV2scale)), colnames(tempV2N), las=2, cex=0.7)
axis(2, at=seq(0, 1, length=nrow(tempV2scale)), rownames(tempV2N), las=2, cex=0.7)
text(na.omit(List2[ ,2]-1)/(ncol(tempV2scale)-1), na.omit(List2[ ,1]-1)/(nrow(tempV2scale)-1) , "*")


```

## Cancer Specific Pathways

```{r, eval=F}
CancerMeltCoding=MeltWGSCoding[which(MeltWGSCoding$Gene_Symbol%in%AllCancerPathwayGenes & MeltWGSCoding$Variant_Classification%in%searchThese), ]


rGeneOut=acast(CancerMeltCoding[ ,c("Gene_Symbol", "L1", "Variant_Classification")], L1~Gene_Symbol, 
               value.var="Variant_Classification", fun.aggregate=function(x) paste(x, collapse = ", "))
rGeneOut[grep( "Missense",rGeneOut) ]="1"
rGeneOut[grep( "Nonsense",rGeneOut) ]="2"
rGeneOut[grep( "Frame",rGeneOut) ]="3"
rGeneOut[grep( "Nonstop",rGeneOut) ]="4"
rGeneOut[which( rGeneOut=="", arr.ind=T) ]="0"
rGeneOut2=apply(rGeneOut, 2, as.numeric)
rownames(rGeneOut2)=rownames(rGeneOut)

GenePathway=names(AllCancerPathwayGenes)[match(colnames(rGeneOut2), AllCancerPathwayGenes)]
GenePathway=substr(GenePathway, 1, 3)

rowSC=palette()[factor(GenePathway)]
rowSC2=rep(NA, length(rowSC))
rowSC2[which(colnames(rGeneOut2)%in%unlist(AllCancerPathwayTS))]="red"
rowSC2[which(colnames(rGeneOut2)%in%unlist(AllCancerPathwayOG))]="green"

l1=as.numeric(RNADNAsamples$Treatment[match(rownames(rGeneOut2), sapply(strsplit(as.character(RNADNAsamples$WGS), ".final.bam"), function(x) x[1]))])
colSC=palette()[l1+4]

x1=order(rowSC, rowSC2)
x2=order(l1)


#a1=heatmap.2(rGeneOut2[-grep("CD45", rownames(rGeneOut2)), x1], Colv = NA, scale="none", trace="none", col=c("white", 1:4), ColSideColors = rowSC[x1])

a2=rGeneOut2[, x1]
a2=a2[x2, ]

# rsMatrix=acast(CancerMeltCoding[ ,c("GENE", "L1", "ID")], L1~GENE,value.var="ID", fun.aggregate = function(x) length(which(x!=""))) #grep("H", x)
# rsMatrixSearch=rsMatrix[match(rownames(a2), rownames(rsMatrix)), match( colnames(a2), colnames(rsMatrix)) ]
# 
# Listout=which(rsMatrixSearch>0, arr.ind = T)

heatmap.plus(a2, Colv = NA, Rowv=NA, scale="none", trace="none", col=c("white", 1:4), ColSideColors = cbind(rowSC[x1],rowSC2[x1]), RowSideColors=cbind(colSC[x2],colSC[x2]))

SSumm1=sapply(1:nrow(rGeneOut2), function(x) stack(by(sign(rGeneOut2[x, ]), GenePathway, sum, na.rm=T))$values)
colnames(SSumm1)=rownames(rGeneOut2)
rownames(SSumm1)=levels(factor(GenePathway))

Nlist=sapply(PathwayListRat, length)

AllMat3=(SSumm1)/Nlist[c(11, 2:4, 6, 8:10)]
AllMat3[which(AllMat3>0, arr.ind = T)]=AllMat3[which(AllMat3>0, arr.ind = T)]+0.1

heatmap.2(t(AllMat3[ ,x2]), col=brewer.pal(9, "Blues"), trace="none", scale="none", RowSideColors =colSC[x2], Rowv = NA)
```

Also separate out the TS and OG: change the first two lines of this to list this informaiton

```{r, eval=F}
names(AllCancerPathwayOG)=c(names(PathwayListRat), "Breast")
AllCancerPathwayTSlist=unlist(AllCancerPathwayOG)

CancerMeltCoding=MeltWGSCoding[which(MeltWGSCoding$Gene_Symbol%in%AllCancerPathwayTSlist & MeltWGSCoding$Variant_Classification%in%searchThese), ]

rGeneOut=acast(CancerMeltCoding[ ,c("Gene_Symbol", "L1", "Variant_Classification")], L1~Gene_Symbol, 
               value.var="Variant_Classification", fun.aggregate=function(x) paste(x, collapse = ", "))
rGeneOut[grep( "Missense",rGeneOut) ]="1"
rGeneOut[grep( "Nonsense",rGeneOut) ]="2"
rGeneOut[grep( "Frame",rGeneOut) ]="3"
rGeneOut[grep( "Nonstop",rGeneOut) ]="4"
rGeneOut[which( rGeneOut=="", arr.ind=T) ]="0"
rGeneOut2=apply(rGeneOut, 2, as.numeric)
rownames(rGeneOut2)=rownames(rGeneOut)

GenePathway=names(AllCancerPathwayTSlist)[match(colnames(rGeneOut2), AllCancerPathwayTSlist)]
GenePathway=substr(GenePathway, 1, 3)

rowSC=palette()[factor(GenePathway)]
l1=as.numeric(RNADNAsamples$Treatment[match(rownames(rGeneOut2), 
                                                         sapply(strsplit(as.character(RNADNAsamples$WGS), ".final.bam"), function(x) x[1]))])
colSC=palette()[l1+4]

x1=order(rowSC)
x2=order(l1)
#a1=heatmap.2(rGeneOut2[-grep("CD45", rownames(rGeneOut2)), x1], Colv = NA, scale="none", trace="none", col=c("white", 1:4), ColSideColors = rowSC[x1])

a2=rGeneOut2[, x1]
a2=a2[x2, ]

# rsMatrix=acast(CancerMeltCoding[ ,c("GENE", "L1", "ID")], L1~GENE,value.var="ID", fun.aggregate = function(x) length(which(x!=""))) #grep("H", x)
# rsMatrixSearch=rsMatrix[match(rownames(a2), rownames(rsMatrix)), match( colnames(a2), colnames(rsMatrix)) ]
# 
# Listout=which(rsMatrixSearch>0, arr.ind = T)

heatmap.plus(a2, Colv = NA, Rowv=NA, scale="none", trace="none", col=c("white", 1:3), ColSideColors = cbind(rowSC[x1],rowSC[x1]), RowSideColors=cbind(colSC[x2],colSC[x2]))
```


### Compute co-occurence matrices

Use all genes in WGS coding

Firstly by genomic location: pull out the top mutations occurring in at least 5 samples: this gives approximately 7000 loci.


```{r, eval=F}

## run the analysis

x1a=table(MeltWGSCodingB$Start_Position)
checkThese=names(x1a)[which(x1a>5)]
matrixSumP=matrix(NA, ncol=length(checkThese), nrow=length(checkThese))
matrixSumOR=matrix(NA, ncol=length(checkThese), nrow=length(checkThese))

allSamp=unique(MeltWGSCodingB$L1)

for (i in 662:length(checkThese)){
  xa=which(MeltWGSCodingB$Start_Position==checkThese[i])
  T1=MeltWGSCodingB$L1[xa]
  tt1=allSamp%in%T1
  if (sum(tt1)<length(allSamp)){
    for (j in 1:length(checkThese)){
    xb=which(MeltWGSCodingB$Start_Position==checkThese[j])
    T2=MeltWGSCodingB$L1[xb]
    tt2=allSamp%in%T2
      if (sum(tt2)<length(allSamp)){
        xn=fisher.test(table(tt1, tt2))
        matrixSumP[i,j]=xn$p.value
        matrixSumOR[i,j]=xn$estimate
  }}
  }
}

matrixSumP_padj=matrix(p.adjust(matrixSumP, "fdr"), nrow=nrow(matrixSumP)) 
rownames(matrixSumP_padj)=paste(MeltWGSCodingB$Gene_Symbol[match(checkThese, MeltWGSCodingB$Start_Position)], checkThese)
colnames(matrixSumP_padj)=rownames(matrixSumP_padj)
rownames(matrixSumOR)=rownames(matrixSumP_padj)
colnames(matrixSumOR)=colnames(matrixSumP_padj)

save(matrixSumOR, matrixSumP_padj, file="rslt/WXS_co-occurrence_rslt.RData")
```


```{r, eval=F}
load("rslt/WXS_co-occurrence_rslt.RData")

# image the data
# figure out the rows and columns which ahve p<0.1 off the diagonal
olfidx=c(grep("Olr", colnames(matrixSumP_padj)), grep("Vom", colnames(matrixSumP_padj)), grep("AABR", colnames(matrixSumP_padj)), grep("RT1", colnames(matrixSumP_padj))) 

matrixP_rm=matrixSumP_padj[-olfidx, -olfidx ]
keepTheseCols=which(matrixP_rm<0.05, arr.ind=T)
mx1Sum=table(keepTheseCols[ ,2])
length(which(mx1Sum>1))
colidx=names(mx1Sum)[which(mx1Sum>1)]


# Note: most of these are olfactory receptors
matP2=matrixP_rm[as.numeric(colidx), as.numeric(colidx)]
matP2=-log10(matP2)
hmres=heatmap.2(matP2, trace="none", col=brewer.pal(9, "Blues"))

Idx2=cutree(as.hclust(hmres$rowDendrogram), k=10)

# Figure out the oncogenes in this list
checkGenes=MeltWGSCodingB$Gene_Symbol[match(checkThese, MeltWGSCodingB$Start_Position)]
mx1=unique(checkGenes[which(checkGenes%in% c(RatCosmic, AllCancerPathwayGenes))])

midxN=lapply(mx1, function(x) grep(x, colnames(matrixP_rm)))
matNew=matrixP_rm[ , unlist(midxN)]
Idx4=which(matNew<0.05, arr.ind=T)
matNew2=matNew[unique(Idx4[ ,1]), ]
matP2=-log10(matNew2)
matP2[which(matP2<1, arr.ind = T)]=0
par(mar=c(10,2,1,10))
hmres=heatmap.2(matP2, trace="none", col=brewer.pal(9, "Blues"), scale="none")

# 
getThese=c("Tcf", "Ros1", "Notch4", "Cebpa", "Prf1", "Ptprb")
x1=lapply(getThese, function(x) grep(x, colnames(matP2)))
matP3=matP2[, unlist(x1)]
matP3=matP3[-which(rowSums(matP3)==0), ]
hmres=heatmap.2(matP3, trace="none", col=brewer.pal(9, "Blues"), scale="none")

matP4=matrixSumOR[ rownames(matP3),colnames(matP3)]
matP4[which(matP4=="Inf", arr.ind = T)]=100
matP4[which(matP4>100, arr.ind = T)]=100
matP4[which(matP3==0, arr.ind=T)]=1
matP4[which(matP4==0, arr.ind=T)]=0.01
hmres=heatmap.2(log2(matP4), trace="none", col=brewer.pal(9, "RdBu"), scale="none")

```