--- 
title: "Rat DCIS study"
author: "Anne Trinh"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
description: "This is a summary of the code required in the study: 'insert study name' "
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F, cache.lazy=F, fig.height =5, fig.width = 7, cache = T, 
                      warning=F, message=F)
```

This is a summary of the code required in the study: 'insert study name' 

# Prerequisites

## Packages and Software

The following packages are required to conduct the analyses described below.

In house scripts are deposited in the rscript folder.

```{r, warning=F, message=F, echo=T, results='hide'}
library(AnnotationHub)
library(biomaRt)
library(Biostrings)
library(colorspace)
library(DESeq2)
library(dplyr)
library(DT)
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(GenVisR)
library(GenomicFeatures)
library(ggplot2)
library(gplots)
library(GSEABase)
library(GSVA)
library(heatmap.plus)
library(HTSanalyzeR2)
library(kableExtra)
library(limma)
library(matrixStats)
library(pamr)
library(reshape2)
library(RColorBrewer)
library(scales)
library(spatstat)
library(tcR)
library(vcfR)
library(xlsx)
library(writexl)

DiffCols=hue_pal()(8)
palette(brewer.pal(9, "Set1"))
RdBu=brewer.pal(11, "RdBu")
SetCols=brewer.pal(12, "Set3")

source("../rscript/cnFreq_fn.R") #modified version of GenVisR
source("../rscript/merge_contig.R")
source("../rscript/gseaCode.R")
source("../rscript/ContingencyTable.R")
source("../rscript/PvalueHeatMap.R")
source("../rscript/BootstrapShannonIdx.R")
source("../rscript/CreateRnor87db.R")
source("../rscript/FindRatAAHomolog.R")

firstup <- function(x) {
  substr(x, 1, 1) <- toupper(substr(x, 1, 1))
  x
}

ColMerge=matrix(c("#FFC82F", "#FFEDBC", "#73FDFE","#D2FFFF", "#FF41FF", "#FECAFF", "#5D5D5D", "#BEBEBE"), ncol=2, byrow = T)
rownames(ColMerge)=c("LY","PDL1", "PDL1+LY","Vehicle")

Hsedb<-EnsDb.Hsapiens.v86
```

## External software

The following external software was utilised:  

|Software| Function |
|--------|-----------------------------------------------------------------------|
|[bwa](http://bio-bwa.sourceforge.net/)| Alignment of WGS data to reference|
|[GATK4](https://software.broadinstitute.org/gatk/)| Mutation calling, done by NYGC. Mutation calling from RNA (Haplotype caller) |
|[strelka](https://github.com/Illumina/strelka)| Mutation calling, done by NYGC |
|[BICseq](http://compbio.med.harvard.edu/BIC-seq/) | CNV calling |
|[GEM3](https://sourceforge.net/projects/gemlibrary/files/gem-library/)| create mappability files for CNV calling|
|[STAR](https://github.com/alexdobin/STAR) | Alignment of RNAseq data|
|[RSEM](https://github.com/deweylab/RSEM)|Calculate RSEM, TPM, FPKM from RNAseq data|
|[TRUST4](https://github.com/liulab-dfci/TRUST4) | assignment of T and B cell clonotypes from RNA-seq data |
|[Oncotator](https://gatkforums.broadinstitute.org/gatk/discussion/4154/howto-install-and-run-oncotator-for-the-first-time) | Annotation of genetic variants|
|[QuPath](https://qupath.github.io/)| Tool for cell segmentation and extraction of features from IF images |
|[samtools, bcftools](http://www.htslib.org/)| querying and dispalying information from bam files, extracting allelic depth at specific genomic locations |
|[CIBERSORT](https://cibersort.stanford.edu/) | inferring immune composition from RNA|
|[lumpy](https://github.com/arq5x/lumpy-sv) | structural variants|
|[PAM50](https://genome.unc.edu/pubsup/breastGEO/PAM50.zip) | code from parker et al 2009 to infer PAM50 subtypes|


## Annotations

### Genomic properties

Information on chromosome sizes, cytobands and centromere locations were obtained from the [UCSC genome browser](http://hgdownload.soe.ucsc.edu/goldenPath/rn6/database/).

The following annotation data for the hg19/b37/GRCh37 genome is required:

|Data Type| Download link|
|--------|-----------------|
|ref. genome| http://hgdownload.soe.ucsc.edu/goldenPath/rn6/bigZips/rn6.fa.gz|
|refSeq annot|http://hgdownload.soe.ucsc.edu/goldenPath/rn6/bigZips/genes/rn6.refGene.gtf.gz|
|refSeq annot|http://hgdownload.soe.ucsc.edu/goldenPath/rn6/bigZips/genes/rn6.ncbiRefSeq.gtf.gz|
|gff3 file| for TRUST4 (ftp://ftp.ensembl.org/pub/release-100/gff3/rattus_norvegicus/Rattus_norvegicus.Rnor_6.0.100.gff3.gz)|
|gene lengths| Extracted from GRCh37.75.gtf file from [ensembl](ftp://ftp.ensembl.org/pub/grch37/current/gtf/homo_sapiens/)|
|hg19cytoBand | [ucsc server](http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/cytoBand.txt.gz) of all cytoband locations  |
|chromosome sizes | [uscs genome browser](http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/chromInfo.txt.gz) |
|[biomart](https://bioconductor.org/packages/release/bioc/html/biomaRt.html)| conversion between gene symbol, ensbl and entrez was faciliated using biomart package |

Below is the summary of chromosome sizes and centromere locations:

```{r}
cytoInfo=read.delim("../anntotations/cytoBand_Rat2.txt", header=F, stringsAsFactors = F)
colnames(cytoInfo)=c("chrom", "chromStart", "chromEnd", "name", "gieStain")
GRcytoInfo=GRanges(seqnames=cytoInfo$V1, ranges=IRanges(start = cytoInfo$V2, end=cytoInfo$V3), cytoband=cytoInfo$V4)

head(cytoInfo)
```

Create a TxDb object from a gtf file and save information: <span style="color:red">Not sure which version to use??</span>

Below is an example of the gene annotation files

```{r, cache=T, message=F, eval=F}
Rn6TxDb=makeTxDbFromGFF("../dontUpload/rn6_refGene.gtf",
                        organism="Rattus norvegicus")
Rn6TxDb2<-genes(Rn6TxDb)
```

```{r, eval=F}
txRn6=makeTxDbFromGFF("../anntotations/rn6_refGene.gtf", format="gtf")
txRn6=genes(txRn6)
txRn6=sort(txRn6)
txRn6$gene_width=width(txRn6)
save(txRn6, file="../anntotations/txRn6_refGene.RData")
```

```{r}
load("../anntotations/txRn6_refGene.RData")
head(txRn6)
```

### Gene name homologs between organisms

Biomart was used to convert between rat, mouse and human gene symbols and ensembl ids. Below is an example of the human gene names mapped to the rat homolog

<span style="color:red">
Figure out what is required here for later analysis
</span>
```{r, eval=F}
## save here:
## human to rat mapping of genes
## rat ENSEMBL vs symbol conversion

library(biomaRt)
#library(refGenome)
TS=read.delim("anntotations//hg38_allsymbols.txt")
TS=as.character(TS[ ,1])
human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl")
rat = useMart("ensembl", dataset = "rnorvegicus_gene_ensembl")

SymHum2Rat = getLDS(attributes = c("hgnc_symbol"), filters = "hgnc_symbol", values = TS , mart = human, attributesL = c("rgd_symbol", "ensembl_gene_id"), martL = rat, uniqueRows=T)

## need to run this after evaluating the gene names 
#Rat2Hum = getLDS(attributes = c("rgd_symbol"), filters = "rgd_symbol", values = rownames(allrsemFinal) , mart = rat, attributesL = c("hgnc_symbol"), martL = human, uniqueRows=T)

# Rat2Mouse= getLDS(attributes = c("mmusculus_homolog_associated_gene_name"), filters = "rgd_symbol", values = rownames(allrsemFinal) , mart = rat, attributesL = c("mgi_symbol"), martL = mouse, uniqueRows=T)

Rat2Mouse= getLDS(attributes = c("rgd_symbol"), filters = "rgd_symbol", values = rownames(allrsemFinal) , mart = rat, attributesL = c("mgi_symbol"), martL = mouse, uniqueRows=T)

Mouse2Hum = getLDS(attributes = c("mgi_symbol"), filters = "mgi_symbol", values = rownames(allrsemFinal) , mart = mouse, attributesL = c("hgnc_symbol"), martL = human, uniqueRows=T)

save(SymHum2Rat, Rat2Hum,Mouse2Hum,Rat2Mouse,  file="../anntotations/Rat_biomart_gene_annotations2.RData")
```

```{r}
load("../anntotations/Rat_biomart_gene_annotations.RData")
head(SymHum2Rat)
```

### Gene signatures and data-bases

Gene sets/signatures were obtained from the following sources:

|Source | Description      |
|-------|------------------------|
|[IEDB](https://www.iedb.org/)| database of immune epitopes |
|[MsigDB](http://software.broadinstitute.org/gsea/msigdb/index.jsp) | c2, c5, hallmark set of curated pathway gene sets |
|[Metacore]()| Process Networks and Pathway Maps data bases |
|[COSMIC](https://cancer.sanger.ac.uk/census) |database of concensus oncogenes|
|[ImmPort](https://www.immport.org/home)| List of immune related genes |
|[InnateDB](https://www.innatedb.com/) | List of genes associated with innate immune system|
|[Rosenthal 2019](https://dx.doi.org/10.1038/s41586-019-1032-7) | genes associated with MHC-I presentation |
|[Thorsson 2018](http://dx.doi.org/10.1016/j.immuni.2018.03.023) | Immune gene signatures curated from studies by Wolf, Calabro, Teschendorff, Beck, Chang |
|[Pardoll](https://www.ncbi.nlm.nih.gov/pubmed/22437870), [Wykes](https://www.ncbi.nlm.nih.gov/pubmed/28990586) | Immune checkpoint genes| 
|[gil del alcazar 2017](https://cancerdiscovery.aacrjournals.org/content/early/2017/09/19/2159-8290.CD-17-0222) | Supplementary table 5: list of activation, dysfunction gene signatures |
|[Bailey 2018](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6029450/)| List of 10 most comon tumor pathways |
|[Chang 2018](https://pubmed.ncbi.nlm.nih.gov/29247016/)| Common mutation locations in cancer |

The PAM50 signature was implemented using the scripts provided in supplementary from [Parker et al 2010](https://genome.unc.edu/pubsup/breastGEO/PAM50.zip). These gene lists can be found in the annotations folder


```{r}
# These gene signatures are saved in the annotations folder and loaded below:
#########################
## cell specific markers
#######################
GeneList=read.csv("../anntotations/cell_type_markers.csv")
cn=colnames(GeneList)
GeneList=lapply(1:ncol(GeneList), function(x) setdiff(unique(GeneList[ ,x]), ""))
names(GeneList)=cn
## map all the names to rat names
GeneListRat=lapply(GeneList, function(x) SymHum2Rat$RGD.symbol[SymHum2Rat$HGNC.symbol%in%x])

######################
## cancer genes
######################

## cosmic cancer genes
AllCosmic=read.csv("../anntotations/Census_COSMIC_Feb2020.csv")
RatCosmic=SymHum2Rat$RGD.symbol[match(AllCosmic$Gene.Symbol, SymHum2Rat$HGNC.symbol)]
tx=AllCosmic$Gene.Symbol[which(is.na(RatCosmic)|RatCosmic=="")]
tx=tolower(tx)
tx=firstup(tx)
RatCosmic[which(is.na(RatCosmic)|RatCosmic=="")]=tx
RatBreastCosmic=RatCosmic[grep("breast", AllCosmic$Tumour.Types.Somatic.)]

## Bailey List of 10 most common tumor pathways
PathwayList=read.csv("../anntotations/cancer_pathways_annot_Bailey_cell2018_modified.csv")
cn=colnames(PathwayList)
PathwayListA=lapply(seq(1, length(cn), by=2), function(x) setdiff(unique(PathwayList[ ,x]), ""))
names(PathwayListA)=cn[seq(1, length(cn), by=2)]
PathwaySign=lapply(seq(2, length(cn), by=2), function(x) PathwayList[ which(PathwayList[ ,x]!=""),x])
names(PathwaySign)=cn[seq(1, length(cn), by=2)]

PathwayListRata=lapply(PathwayListA, function(x) SymHum2Rat$RGD.symbol[match(x, SymHum2Rat$HGNC.symbol)])
PathwayListRatb=lapply(PathwayListA, function(x) Rat2Hum$RGD.symbol[match(x, Rat2Hum$HGNC.symbol)])
PathwayListRat=lapply(1:length(PathwayListRata), function(x) ifelse(is.na(PathwayListRata[[x]]),
                        PathwayListRatb[[x]], PathwayListRata[[x]]))
names(PathwayListRat)=names(PathwayListA)

AllCancerPathwayGenes=na.omit(unlist(PathwayListRat))

############################
## List of immunesignatures
############################
## read in all the files
Exp2=read.csv("../anntotations/Supplementary Table 5.csv")
Exp2List=lapply(1:ncol(Exp2), function(x) setdiff(unique(Exp2[ ,x]), ""))
names(Exp2List)=colnames(Exp2)

List2=read.csv("../anntotations/Thorsson_signatures.csv")
List2b=lapply(1:ncol(List2), function(x) setdiff(unique(List2[ ,x]), ""))
names(List2b)=colnames(List2)
Exp2List=c(Exp2List, List2b)

ImmSuppAPC=read.delim("../anntotations/immune_Suppression.csv", header=T, stringsAsFactors = F, sep=",")
ImmSuppAPC=lapply(1:ncol(ImmSuppAPC), function(x) setdiff(unique(ImmSuppAPC[ ,x]), ""))
names(ImmSuppAPC)=c("Inh", "Act", "Both")
Exp2List=c(Exp2List, ImmSuppAPC)

MHCPres=read.delim("../anntotations/MHCloss.csv", header=F, stringsAsFactors = F, sep=",")
MHCPres=as.character(MHCPres[ ,1])
MHCPres2Rat=unique(SymHum2Rat$RGD.symbol[SymHum2Rat$HGNC.symbol%in%MHCPres])

## change the names to Rat-specific 
Exp2RatImm=lapply(Exp2List, function(x) unique(SymHum2Rat$RGD.symbol[SymHum2Rat$HGNC.symbol%in%x]))

## All immune genes
AllImmGenes1=read.csv("../anntotations/ImmPort_Set.csv")
AllImmGenes2=read.csv("../anntotations/innatedb_curated_genes.csv")
AllImmGenes=unique(c(as.character(AllImmGenes1$Symbol), as.character(AllImmGenes2$Gene.Symbol[which(AllImmGenes2$Species==9606)])))
RatAllImm=na.omit(unique(c(SymHum2Rat$RGD.symbol[match(AllImmGenes, SymHum2Rat$HGNC.symbol)], as.character(AllImmGenes2$Gene.Symbol[which(AllImmGenes2$Species!=9606)]))))

## CIBERSORT specific rat genes
lm22rat=read.delim("../anntotations/LM22_to_rnorvegicus_1.txt", sep="\t")
```

#### Human gene homologs

Below, lists of common mutations in cancer are loaded and the "homolog" in rat is determined using an in-house script. The steps involved are:

* determine the amino acid context in human (find 5 a.a. prior and after)
* find the region with most amino acid homology in rat (2 or less differences)
* check whether the amino acid of interest is present in rat

An example of the output is shown

```{r mut-hum2rat}
######################
## Annotation for Mutation Locations
######################
BaileyList=read.csv("../anntotations/list_mutations_bailey.csv", stringsAsFactors = F)
BList=FindRatAAHomolog(BaileyList$Gene, substr(BaileyList$Mutation, 3, 3),  substr(BaileyList$Mutation, 4, nchar(as.character(BaileyList$Mutation))-1), substr(BaileyList$Mutation, nchar(as.character(BaileyList$Mutation)), nchar(as.character(BaileyList$Mutation))))

ChangList=read.delim("../anntotations/hotspots_chang_et_al_2017_cancer_discovery.txt", sep="\t", stringsAsFactors = F)
ChangList=ChangList[which(ChangList$Type=="single residue"), ]
ChangList$AA1=substr(ChangList$Residue, 1, 1)
ChangList$Loc=substr(ChangList$Residue,2, nchar(ChangList$Residue))
AA2list=sapply(ChangList$Variants, function(x) strsplit(x, ":[0-9]+[\\|]*"))
AAun=unlist(AA2list)
C2List=tibble(ChangList[ ,c("Gene", "AA1", "Loc")])
C2List=C2List %>% slice(rep(1:n(), times=sapply(AA2list, length)))
C2List$AA2=AAun
C2List=C2List[-which(C2List$AA2=="sp"| C2List$AA1=="*"), ]
#C2ListB=C2List[-which(is.na(C2List$Loc)), ]

ChangList2=FindRatAAHomolog(C2List$Gene, C2List$AA1,  C2List$Loc, C2List$AA2)

head(ChangList2)
```

#### GSEA compendiums

For pathway analysis, the c2 (pathway), Hallmark and c5 (Gene Ontology). In addition, metacore pathways (pathway maps and process networks) were obtained and loaded below. This gives a list of 7 different data-sets to interrogate
<span style="color:red">
CHECK: When GSVA is run, which input is required??
</span>

```{r gsea-load}
PathInc2=getGmt(con="../anntotations/c2.cp.v7.0.symbols.gmt", geneIdType=SymbolIdentifier(),
                collectionType=BroadCollection(category="c2"))

c2entrez=mapIdentifiers(PathInc2, EntrezIdentifier('org.Hs.eg.db'))
c2ListHs=geneIds(c2entrez)

PathInH=getGmt(con="../anntotations/h.all.v7.1.symbols.gmt", geneIdType=SymbolIdentifier(),
                collectionType=BroadCollection(category="h"))

cHentrez=mapIdentifiers(PathInH, EntrezIdentifier('org.Hs.eg.db'))
cHListHs=geneIds(cHentrez)

#####################################################
# also load in the process networks and pathway maps 
#################

# load("../anntotations/Metacore_extracted_Process_networks_nov2020.RData.RData")
# 
# ProcessNetworksAllComp2=list()
# for (i in 1:(length(ProcessNetworksAllComp)-1)){
# ProcessNetworksAllComp2[[i]]=GeneSet(ProcessNetworksAllComp[[i]], setName=names(ProcessNetworksAllComp)[i], geneIdType=SymbolIdentifier())
# }
# 
# ProcessPathway=GeneSetCollection(ProcessNetworksAllComp2)
# ProcessPathway=mapIdentifiers(ProcessPathway, EntrezIdentifier('org.Hs.eg.db'))
# ProcessPathwayList=geneIds(ProcessPathway)

load("../anntotations/ListofGeneSets2.RData")
```

<!--chapter:end:index.Rmd-->

# Cohort summary

Below we assess summary statistics on clinico-pathological features of this data set. This includes information on:

* treatment
* tumor size
* growth rates (mm/week)
* number of tumors per rat 

```{r}
Cdata=xlsx::read.xlsx("../metadata/Carlos_samples_list_master_updated_2020-10-05.xlsx", sheetIndex=1)

scroll_box(kable(Cdata, format="html"),
         height="300px", width="100%")
```

## Calculating growth rates

In this section, we estimate the growth rates of the samples: Below is a plot of the tumor size per week for each recorded tumor, color-coded according to treatment. Time is measured at the first time point at which a tumor is palpated. Spontaneous large tumors are assumed to have a tumor size of 0 or 1 one week prior to palpating.

```{r}
GrowthRaw=read.csv("../metadata/growth_rates_0915.csv")
colnames(GrowthRaw)[-1]=substr(colnames(GrowthRaw)[-1], 2, 10)
colnames(GrowthRaw)=gsub("\\.", "-", colnames(GrowthRaw))

CTreat=Cdata$Treatment[match(colnames(GrowthRaw), gsub("_", "", Cdata$TumorID))]
CTreat=ColMerge[match(CTreat, rownames(ColMerge)) ,1]
Cgrowth=Cdata$Tumor.growth.status[match(colnames(GrowthRaw), gsub("_", "", Cdata$TumorID))]

#pdf(sprintf("rslt/Clinicopath/summary_growth_plots_%s.pdf", Sys.Date()), width=8, height=8)

par(xpd=T)
plot(NA, xlim=c(0, 27), ylim=c(0, 50), ylab="Tumor Size (mm)", xlab="Time (weeks)", main="Tumor size over time")

for (i in 2:ncol(GrowthRaw)){
  x1=which(!is.na(GrowthRaw[, i]))
  lines(GrowthRaw[x1, 1], GrowthRaw[x1, i], col=CTreat[i])
}
legend("bottom", inset = c(-0.2, -0.2), rownames(ColMerge), col=ColMerge[ ,1], lwd=2, horiz = T)

ax1=colnames(GrowthRaw)[-1][which(CTreat[-1]=="Unknown")]
bx1=Cdata$TumorID[!gsub("_", "", Cdata$TumorID)%in%colnames(GrowthRaw)]
bx1=bx1[which(bx1%in%Cdata$TumorID[Cdata$Tumor.growth.status%in%c("stable", "growing")])]

#Plot the above into quadrants based on different treatments:
par(mfrow=c(2,2))
xn=levels(factor(names(CTreat)))
x2=CTreat[xn]
for (j in 1:4){
  indx=which(CTreat==x2[j])
  indx=setdiff(indx, 1)
  plot(NA, xlim=c(0, 27), ylim=c(0, 50), ylab="Tumor Size (mm)", xlab="Time (weeks)", main=xn[j])
for (i in indx){
  x1=which(!is.na(GrowthRaw[, i]))
  lines(GrowthRaw[x1, 1], GrowthRaw[x1, i], col=CTreat[i])
  text(GrowthRaw[x1[length(x1)], 1], GrowthRaw[x1[length(x1)], i], colnames(GrowthRaw)[i], cex=0.6)
}
}

#dev.off()
```

These tumors doesn't have growth rate data: `r bx1`.

We can then compute the growth rate for the above samples by considering the change in size over a given period of time using a linear regression model. Below is the histogram of growth rates:

```{r, fig.height=5}
GR=sapply(2:ncol(GrowthRaw), function(x) lm(GrowthRaw[, x]~GrowthRaw[, 1])$coefficients[2])
names(GR)=colnames(GrowthRaw)[-1]

d1=data.frame(growthrate=GR, treatment=names(CTreat[-1]), growth=Cgrowth[-1], color=CTreat[-1])

Nweeks=sapply(2:ncol(GrowthRaw), function(x) max(which(!is.na(GrowthRaw[ , x]))))
names(Nweeks)=names(GR)

d1$Nweeks=Nweeks[match(rownames(d1), names(Nweeks))]
d1$Time.Tum2Sac=Cdata$Time.Tumor2Sac[match(rownames(d1), gsub("_", "", Cdata$TumorID))]
d1$Time.NMU2Sac=Cdata$Time.NMU2Sac[match(rownames(d1), gsub("_", "", Cdata$TumorID))]
d1$tum.size=Cdata$Tumor.dia..sac..mm.[match(rownames(d1), gsub("_", "", Cdata$TumorID))]
d1$growthrate_cutoff2=ifelse(d1$growthrate>=2, "growing", "stable")

#pdf(sprintf("rslt/Clinicopath/summary_growth_rates_%s.pdf", Sys.Date()), width=8, height=6)
hist(GR, breaks=30, main="distribution of growth rates (mm/week)")
```

Based on the above distribution, a cut-off of 2mm/week may be an optimal cut-off to separate growing and stable tumors. Below are growth rates of tumors under different treatments:

```{r}
ggplot(d1, aes(x=treatment, y=growthrate, col=treatment))+geom_boxplot()+geom_jitter()+
  scale_color_manual(values=ColMerge[ ,1])
```

We can calculate the p.values below, using a  t.test. The growth rates comparing the treatment to the controls are:

```{r}
print('LY samples')
wilcox.test(d1$growthrate[d1$treatment=="LY"], d1$growthrate[d1$treatment=="Vehicle"])
print('PDL1 samples')
wilcox.test(d1$growthrate[d1$treatment=="PDL1"], d1$growthrate[d1$treatment=="Vehicle"])
print('PDL1+LY samples')
wilcox.test(d1$growthrate[d1$treatment=="PDL1+LY"], d1$growthrate[d1$treatment=="Vehicle"])
```

This shows a smaller growth-rate in PDL1 single and double treated cases compared to the vehicles.

Overall the distribution of growing vs stable tumors is shown below:

```{r}
table(ifelse(d1$growthrate>=2, "grow", "stable"))
```

We can replot the previous graphs according to growth, and color code according to whether it is a fast or slow growing tumor

```{r}
#pdf(sprintf("rslt/Clinicopath/summary_growth_plots_with shadings%s.pdf", Sys.Date()), width=8, height=8)

par(xpd=T)

#Plot the above into quadrants based on different treatments:
par(mfrow=c(2,2))
for (j in 1:length(xn)){
  indx=which(names(CTreat)==xn[j])
  indx=setdiff(indx, 1)
#   plot(NA, xlim=c(0, 27), ylim=c(0, 50), ylab="Tumor Size (mm)", xlab="Time (weeks)", main=paste("Original", xn[j]))
# for (i in indx){
#   x1=which(!is.na(GrowthRaw[, i]))
#   lines(GrowthRaw[x1, 1], GrowthRaw[x1, i], col=ifelse(d1$growth[i-1]=="growing", ColMerge[j, 1], ifelse(d1$growth[i-1]=="stable", ColMerge[j, 2], "black")), type="o", pch=19)
#   text(GrowthRaw[x1[length(x1)], 1], GrowthRaw[x1[length(x1)], i], colnames(GrowthRaw)[i], cex=0.6)
# }
  plot(NA, xlim=c(0, 27), ylim=c(0, 50), ylab="Tumor Size (mm)", xlab="Time (weeks)", main=paste("Redone", xn[j]))
for (i in indx){
  x1=which(!is.na(GrowthRaw[, i]))
  lines(GrowthRaw[x1, 1], GrowthRaw[x1, i], col=ifelse(d1$growthrate_cutoff2[i-1]=="growing", ColMerge[j, 1], ifelse(d1$growthrate_cutoff2[i-1]=="stable", ColMerge[j, 2], "black")), type="o", pch=19)
  text(GrowthRaw[x1[length(x1)], 1], GrowthRaw[x1[length(x1)], i], colnames(GrowthRaw)[i], cex=0.6)
}
  
}

#dev.off()

```

As a sanity check, compare these growth rates with differences in tumour size at different time points:

* comparing the growth rate according to classifications (growing, stable)
* tumor size at time of sacrifice
* rate of tumor development from the time of NMU injection

```{r, fig.height=4,warning='hide'}
par(mfrow=c(1,3))
# boxplot(d1$growthrate~d1$growth, main="growth rate, original growth", las=2, ylab="rate (mm/week)")
boxplot(d1$growthrate~d1$growthrate_cutoff2, main="growth rate, new growth", las=2,  ylab="tumor growth rate (mm/week)", xlab="")
x1=wilcox.test(d1$growthrate~d1$growthrate_cutoff2)$p.value
text(1.5, 20, paste("wilcox p =", round(x1, 3)))

# boxplot(d1$tum.size~d1$growth, main="tum size at sac, original growth", las=2, ylab="size @ sac (mm)")
boxplot(d1$tum.size~d1$growthrate_cutoff2, main="tum size at sac, new growth", las=2, ylab="Tumor diameter (mm)", xlab="")
x1=wilcox.test(d1$tum.size~d1$growthrate_cutoff2)$p.value
text(1.5, 40, paste("wilcox p =", round(x1, 3)))
# boxplot(d1$tum.size/d1$Time.NMU2Sac~d1$growth, main="rate NMU2SAC, original growth", las=2, ylab="rate (mm/week)")
boxplot(d1$tum.size/d1$Time.NMU2Sac~d1$growthrate_cutoff2, main="rate NMU2SAC", las=2, ylab="growth rate from NMU injection (mm/wk)", xlab="")
x1=wilcox.test(d1$tum.size/d1$Time.NMU2Sac~d1$growthrate_cutoff2)$p.value
text(1.5, 0.6, paste("wilcox p =", round(x1, 3)))

# boxplot(d1$growthrate~d1$treatment, main="growth rate, original growth", las=2, ylab="rate (mm/week)")
#boxplot(d1$growthrate~d1$growthrate_cutoff2, main="growth rate, new growth", las=2)

```

Is there an association with treatment? Calculate below using chi-squared test:

```{r, fig.height=3, fig.width=5}
## Old data

# print('new data outcome')
 a3=chisq.test(table(factor(d1$treatment), d1$growthrate_cutoff2))
 a3

#chisq.test(table(factor(d1$treatment[d1$growth%in%c("growing", "stable")]), factor(d1$growth[d1$growth%in%c("growing", "stable")])))
#a4

#pdf(sprintf("rslt/summary_growth_plots_contigency%s.pdf", Sys.Date()), width=6, height=4)

 ContTable(table(factor(d1$treatment), d1$growthrate_cutoff2), "new rates", T, "growth", "treatment")

#ContTable(table(factor(d1$treatment[d1$growth%in%c("growing", "stable")]), factor(d1$growth[d1$growth%in%c("growing", "stable")])), "old rates", T, "growth", "treatment")

#dev.off()

```

Overall, it appears that there is an association between growth rate and treatment

```{r}
# Replace the Cdata information with new growth information
Cdata$Growth2=d1$growthrate_cutoff2[match(gsub("_", "", Cdata$TumorID), rownames(d1))]
Cdata$GrowthRate=d1$growthrate[match(gsub("_", "", Cdata$TumorID), rownames(d1))]

#write.xlsx(Cdata, file=sprintf("metadata/Carlos_samples_list_master_updated_%s.xlsx", Sys.Date()))
```

## FACS data

The immune (CD45) fractions from a number of samples were collected, and assessed using FACs. The major cell types detected are:

Leukocytes:

* Tregs 
* CD8 T cells
* Thelper cells
* B cells
* NK T cells
* gamma delta T cells 

Myeloid cells:

* Macrophages M1
* Macrophages M2
* Dendritic cells
* Monocytes
* Neutrophils

We can look at the:

* types of cells
* distributions

Note that in a number of samples the leukocyte population could not be inferred with confidence, and proportions are normalised to the myeloid population

```{r}
Fdata=read.csv("../data/carlos_facs_tumors.csv", stringsAsFactors = F)
Fdata[ ,2:ncol(Fdata)]=Fdata[ ,2:ncol(Fdata)]/100
m1=substr(colnames(Fdata), 2, 5)
colnames(Fdata)=m1
colnames(Fdata)[1]="type"

scroll_box(kable(Fdata, format="html"),
         height="300px", width="100%")

t2=Fdata[-which(Fdata[ ,1]==""),]
#rownames(t2)=Fdata[-which(Fdata[ ,1]=="") ,1]
t2=t2[-c(1:2, 14), ]
t2melt=melt(t2)

ggplot(t2melt, aes(x=variable, y=value, fill=type))+geom_bar(stat="identity")+xlab("sample")+ylab("proportion")+ggtitle('all samples')

```

We can look solely at the myeloid population (and normalise to this total), and color according to growth

```{r}
t3=t2[7:11, ]
t3[, 2:ncol(t3)]=t(t(t3[, 2:ncol(t3)])/colSums(t3[, 2:ncol(t3)]))
t3melt=melt(t3)

t3melt$growth=d1$growthrate_cutoff2[match(t3melt$variable, rownames(d1))]

ggplot(t3melt, aes(x=variable, y=value, fill=type))+geom_bar(stat="identity")+xlab("sample")+ylab("proportion")+ggtitle('myeloid specific')
```
Similarly, we can look at the leukocyte population. Note that the Treg population in some of these samples is very high.

```{r}
t3=t2[1:6, ]
t3[, 2:ncol(t3)]=t(t(t3[, 2:ncol(t3)])/colSums(t3[, 2:ncol(t3)]))
t3melt=melt(t3)

t3melt$growth=d1$growthrate_cutoff2[match(t3melt$variable, rownames(d1))]

ggplot(t3melt, aes(x=variable, y=value, fill=type))+geom_bar(stat="identity")+xlab("sample")+ylab("proportion")+ggtitle('leukocyte specific')
```

<!--chapter:end:01a-cohort-summary.Rmd-->

## Summary Table

Firstly, look at the total number of samples:

```{r}
#recur variables here
avtum=table(Cdata$Rat_ID)
Cchar=Cdata[Cdata$Cohort!="Progression", ]
Cprog=Cdata[Cdata$Cohort=="Progression", ]
```


Feature | Levels | N
--------|---------|-----
Total No Tumors |All | `r nrow(Cdata)`
- | Characterisation| `r length(which(Cdata$Cohort=="Characterisation"))`
- | Prevention | `r length(which(Cdata$Cohort=="Prevention"))`
- | Progression | `r length(which(Cdata$Cohort=="Progression"))`
Total No Rats | All| `r length(unique(Cdata$Rat_ID))`
- | Characterisation| `r length(unique(Cdata$Rat_ID[which(Cdata$Cohort=="Characterisation")]))`
- | Prevention | `r length(unique(Cdata$Rat_ID[which(Cdata$Cohort=="Prevention")]))`
- | Progression | `r length(unique(Cdata$Rat_ID[which(Cdata$Cohort=="Progression")]))`
No tumours per rat | | `r round(mean(avtum),2) ` ( `r quantile(avtum, c(0.1, 0.9))`)

### Compare the characterisation vs progression cohort

Feature | Levels | Characterisation | Progression
--------|---------|-------- |------
Total No tumours | | `r nrow(Cchar)` | `r nrow(Cprog)`
Treatments| Vehicle | `r length(which(Cdata$Treatment=="Vehicle"))`| `r length(which(Cprog$Treatment=="Vehicle"))`
- | Untreated  | `r length(which(Cchar$Treatment=="Untreated"))`|
-| LY | | `r length(which(Cprog$Treatment=="LY"))`
-| PDL1| | `r length(which(Cprog$Treatment=="PDL1"))`
-| PDL1+LY| | `r length(which(Cprog$Treatment=="PDL1+LY"))`
- | NA | `r length(which(is.na(Cchar$Treatment)))` |`r length(which(is.na(Cprog$Treatment)))`
Histology |  diff. adenocarcinomas | `r length(which(Cchar$Tumor.Histology=="well differentiated mammary solid adenocarcinoma"))`|`r length(which(Cprog$Tumor.Histology=="well differentiated mammary solid adenocarcinoma"))`
- | mucinous carcinoma | `r length(which(Cchar$Tumor.Histology=="mucinous carcinoma"))`|`r length(which(Cprog$Tumor.Histology=="mucinous carcinoma"))`
- | Fibroadenoma | `r length(which(Cchar$Tumor.Histology=="fibroadenoma"))`|`r length(which(Cprog$Tumor.Histology=="fibroadenoma"))`
- | NA |`r length(which(is.na(Cchar$Tumor.Histology)))` | `r length(which(is.na(Cprog$Tumor.Histology)))`
Age Injection | 32-36 | `r length(which(Cchar$Age.at.injection=="32-36")) `|`r length(which(Cprog$Age.at.injection=="32-36")) `
 - | 35 | `r length(which(Cchar$Age.at.injection=="35")) ` | `r length(which(Cprog$Age.at.injection=="35")) `
  - | 49 | `r length(which(Cchar$Age.at.injection=="49")) `|`r length(which(Cprog$Age.at.injection=="49")) `
 - | NA | `r length(which(is.na(Cchar$Age.at.injection))) `|`r length(which(is.na(Cprog$Age.at.injection))) `
Time (days) | NMU 2 Sac| `r round(mean(Cchar$Time.NMU2Sac, na.rm=T),2) ` (`r quantile(Cchar$Time.NMU2Sac, c(0.1, 0.9), na.rm=T)`) | `r round(mean(Cprog$Time.NMU2Sac, na.rm=T), 2) ` (`r quantile(Cprog$Time.NMU2Sac, c(0.1, 0.9), na.rm=T)`)
 - | Cases with NA| `r length(which(is.na(Cchar$Time.NMU2Sac)))` | `r length(which(is.na(Cprog$Time.NMU2Sac)))` 
 - | NMU 2 Tumor | | `r round(mean(Cprog$Time.NMU2Tumor, na.rm=T), 2)` (`r  quantile(Cprog$Time.NMU2Tumor, c(0.1, 0.9), na.rm=T)`)
 - | Cases with NA| `r length(which(is.na(Cchar$Time.NMU2Tumor)))` | `r length(which(is.na(Cprog$Time.NMU2Tumor)))` 
 - | Tum Spec Surv|  | `r round(mean(Cprog$Time.Tumor2Sac, na.rm=T),2)` (`r quantile(Cprog$Time.Tumor2Sac, c(0.1, 0.9), na.rm=T)`)
 - | Cases with NA| `r length(which(is.na(Cchar$Time.Tumor2Sac)))` | `r length(which(is.na(Cprog$Time.Tumor2Sac)))`
Growth Rate/Size (mm) | overall size @ sac | `r round(mean(Cchar$Tumor.dia..sac..mm.,na.rm=T), 2)`  (`r quantile(Cchar$Tumor.dia..sac..mm., c(0.1,  0.9), na.rm=T)`) | `r round(mean(Cprog$Tumor.dia..sac..mm.,na.rm=T), 2)` (`r quantile(Cprog$Tumor.dia..sac..mm., c(0.1, 0.9), na.rm=T)`)
- | Growing No. | | `r length(grep("growing", Cprog$Growth2))`
- | Growing size @ sac |  |  `r round(mean(Cprog$Tumor.dia..sac..mm.[grep("growing", Cprog$Growth2)]),2)` `r quantile(Cprog$Tumor.dia..sac..mm.[grep("growing", Cprog$Growth2)], c(0.1, 0.5, 0.9), na.rm=T)`
- | Stable ||`r length(grep("stable", Cprog$Growth2))`
- | Stable size @ sac | | `r round(mean(Cprog$Tumor.dia..sac..mm.[grep("stable", Cprog$Growth2)]),2)` (`r quantile(Cprog$Tumor.dia..sac..mm.[grep("stable", Cprog$Growth2)], c(0.1, 0.5, 0.9), na.rm=T)`)
- | NA |`r length(which(is.na(Cchar$Growth2)))` | `r length(which(is.na(Cprog$Growth2)))`
Spatial Pattern| | |
- | Infiltrating| `r length(which(Cchar$InfiltratingVsRestricted=="Infiltrating"))` |  `r length(which(Cprog$InfiltratingVsRestricted=="Infiltrating"))`
- | Restricted |  `r length(which(Cchar$InfiltratingVsRestricted=="restricted"))` |  `r length(which(Cprog$InfiltratingVsRestricted=="restricted"))`
- | NA| `r length(which(is.na(Cchar$InfiltratingVsRestricted)))` |  `r length(which(is.na(Cprog$InfiltratingVsRestricted)))`
RNA samples | any fraction | `r length(which(!is.na(Cchar$RNA.seqSampleID)))`| `r length(which(!is.na(Cprog$RNA.seqSampleID)))`
- | Ep|`r length(which(!is.na(Cchar$FqFile.Ep)))`|`r length(which(!is.na(Cprog$FqFile.Ep)))`
- | DN|`r length(which(!is.na(Cchar$FqFile.DN)))`|`r length(which(!is.na(Cprog$FqFile.DN)))`
- | CD45| `r length(which(!is.na(Cchar$FqFile.CD45)))`| `r length(which(!is.na(Cprog$FqFile.CD45)))`
Imaging Data | No tumors | | `r length(which(!is.na(Cprog$Image.Name)))`
- | No tumors with RNA | |`r length(which(!is.na(Cprog$Image.Name)& !is.na(Cprog$RNA.seqSampleID)))`
FACS data| Comprehensive|  | `r length(which(colnames(Fdata)%in%gsub("_", "", Cprog$TumorID)))`
- | EpCAM/CD45 | `r length(which(!is.na(Cchar$X..CD45..by.FACS.)))` |  `r length(which(!is.na(Cprog$X..CD45..by.FACS.)))` 

<span style="color:red">
TO INCLUDE: 
- trichrome data?
</span>

### Summary of the RNA data

Below is a table of the samples with RNA information

```{r}
CRchar=Cchar[which(Cchar$AnyRNAdata=="yes"), ]
CRprog=Cprog[which(Cprog$AnyRNAdata=="yes"), ]
```

Feature | Levels | Characterisation | Progression
--------|---------|-------- |------
RNA samples | any fraction | `r length(which(!is.na(CRchar$RNA.seqSampleID)))`| `r length(which(!is.na(CRprog$RNA.seqSampleID)))`
- | Ep|`r length(which(!is.na(CRchar$FqFile.Ep)))`|`r length(which(!is.na(CRprog$FqFile.Ep)))`
- | DN|`r length(which(!is.na(CRchar$FqFile.DN)))`|`r length(which(!is.na(CRprog$FqFile.DN)))`
- | CD45| `r length(which(!is.na(CRchar$FqFile.CD45)))`| `r length(which(!is.na(CRprog$FqFile.CD45)))`
Treatments Char/Prev  | Vehicle | `r length(which(CRchar$Treatment=="Vehicle"))`| `r length(which(CRprog$Treatment=="Vehicle"))`
- | Untreat (char) | `r length(which(CRchar$Treatment=="Untreated"))` |
-| LY | | `r length(which(CRprog$Treatment=="LY"))`
-| PDL1| | `r length(which(CRprog$Treatment=="PDL1"))`
-| PDL1+LY| | `r length(which(CRprog$Treatment=="PDL1+LY"))`
- | NA | `r length(which(is.na(CRchar$Treatment) ))` |`r length(which(is.na(CRprog$Treatment)))`
Time | NMU 2 Sac| `r round(mean(CRchar$Time.NMU2Sac, na.rm=T),2) ` (`r quantile(CRchar$Time.NMU2Sac, c(0.1, 0.9), na.rm=T)`) | `r round(mean(CRprog$Time.NMU2Sac, na.rm=T), 2) ` (`r quantile(CRprog$Time.NMU2Sac, c(0.1, 0.9), na.rm=T)`)
Growth Rate/Size (mm) | overall size @ sac | `r round(mean(CRchar$Tumor.dia..sac..mm.,na.rm=T), 2)`  (`r quantile(CRchar$Tumor.dia..sac..mm., c(0.1,  0.9), na.rm=T)`) | `r round(mean(CRprog$Tumor.dia..sac..mm.,na.rm=T), 2)` (`r quantile(CRprog$Tumor.dia..sac..mm., c(0.1, 0.9), na.rm=T)`)
- | Growing No. | | `r length(grep("growing", CRprog$Growth2))`
- | Growing size @ sac |  |  `r round(mean(CRprog$Tumor.dia..sac..mm.[grep("growing", CRprog$Growth2)]),2)` `r quantile(CRprog$Tumor.dia..sac..mm.[grep("growing", CRprog$Growth2)], c(0.1, 0.5, 0.9), na.rm=T)`
- | Stable ||`r length(grep("stable", CRprog$Growth2))`
- | Stable size @ sac | | `r round(mean(CRprog$Tumor.dia..sac..mm.[grep("stable", CRprog$Growth2)]),2)` (`r quantile(CRprog$Tumor.dia..sac..mm.[grep("stable", CRprog$Growth2)], c(0.1, 0.5, 0.9), na.rm=T)`)
- | NA |`r length(which(is.na(CRchar$Growth2)))` | `r length(which(is.na(CRprog$Growth2)))`
Growth and Treatment: comparing small/stable vs large/growing | | 
Vehicle | N s/l |  | `r length(which(CRprog$Treatment=="Vehicle" & CRprog$Growth2=="stable"))`, `r length(which(CRprog$Treatment=="Vehicle" & CRprog$Growth2=="growing"))`
LY | N s/l |  | `r length(which(CRprog$Treatment=="LY" & CRprog$Growth2=="stable"))`, `r length(which(CRprog$Treatment=="LY" & CRprog$Growth2=="growing"))`
PDL1 and Treatment | N s/l |  | `r length(which(CRprog$Treatment=="PDL1" & CRprog$Growth2=="stable"))`, `r length(which(CRprog$Treatment=="PDL1" & CRprog$Growth2=="growing"))`
PDL1+LY | N s/l |  | `r length(which(CRprog$Treatment=="PDL1+LY" & CRprog$Growth2=="stable"))`, `r length(which(CRprog$Treatment=="PDL1+LY" & CRprog$Growth2=="growing"))`



<!--chapter:end:01b-cohortSummary.Rmd-->


# Whole-slide imaging


```{r, eval=F}
# Delete this part when finished

library(ggplot2)
library(reshape2)
library(gplots)
library(RColorBrewer)
library(spatstat)
source("rscript/PvalueHeatMap.R")
```


In this section, we will be looking at the composition and spatial distribution of cells in whole slide images. These sections have previously been assessed using an  <span style="color:red">external script. Save this somewhere </span>

The following markers have been used:

* EpCAM (tumor cells)
* SMA (fibroblasts or myeopithelial cells)
* CD8 (T cells)

Note that in some images a double positive EpCAM+/SMA+ population exists. Some CD8 cells have Epcam+ or SMA+ staining, however, we consider all of these to be simply CD8+


```{r load-wsi}
WSIpath="../data/WSI-data/locationData/"
WSIfiles=dir(WSIpath, "*.csv")
WSIsummary=lapply(WSIfiles, function(x) read.csv(paste(WSIpath, x, sep="")))
names(WSIsummary)=sapply(strsplit(WSIfiles, "_"), function(x) x[1])

## knn data
WSIpath="../data/WSI-data/knn-values/"
WSIfiles=dir(WSIpath, "*.csv")
WSIknn=lapply(WSIfiles, function(x) read.csv(paste(WSIpath, x, sep="")))
names(WSIknn)=sapply(strsplit(WSIfiles, "_"), function(x) x[1])

## knn data
WSIpath="../data/WSI-data/interactingFraction/"
WSIfiles=dir(WSIpath, "*.csv")
WSIIF=lapply(WSIfiles, function(x) read.csv(paste(WSIpath, x, sep="")))
WSIIF=lapply(WSIIF, function(x) {colnames(x)<-c("RN","Grid", "NearestNeighbor", "IF", "Reference"); x})
names(WSIIF)=sapply(strsplit(WSIfiles, "_"), function(x) x[1])

# MH data
WSIpath="../data/WSI-data/mh-values/"
WSIfiles=dir(WSIpath, "*.csv")
WSIMH=lapply(WSIfiles, function(x) read.csv(paste(WSIpath, x, sep="")))
names(WSIMH)=sapply(strsplit(WSIfiles, "_"), function(x) x[1])

# MH set-up summary
WSIpath="../data/WSI-data/MHset-upSummary/"
WSIfiles=dir(WSIpath, "*.csv")
WSIMHsetup=lapply(WSIfiles, function(x) read.csv(paste(WSIpath, x, sep="")))
names(WSIMHsetup)=sapply(strsplit(WSIfiles, "_"), function(x) x[1])

save(WSIsummary, WSIknn, WSIIF, WSIMH, WSIMHsetup, file=sprintf("outputs/WSI_raw_data%s.RData", Sys.Date()))
```

## Associate the frequencies with other data types

Note there are 47 samples with imaging data. 33 of these samples have FACS data, manual counts and TIMER scores

Correlate the following information:

 * "CD8.WSI": CD8 total counts 
 * "CD8Frac.WSI": CD8 fraction (normalised by cell count)
 * "CD8_EPorSMARatio.WSI": CD8/EP+SMA ratio (any EpCAM or SMA + cell)
 * "CD8_AnySMARatio.WSI": CD8/Any EPcam+ cell
 * "CD8_EPRatio.WSI": CD8 to EpCAM+SMA- ratio
 * "CD8normTumSize": normalised CD8 counts per mm of tumor size at sac
 * "CD8.EpBoundingBox": approx area per CD8 cell (density)

<span style="color:red">
UPDATE THE CIBERSORT INFORMATION. </span>

Below are heatmaps which show the correlation between two variables (red is correlated and blue is anti-correlated), and the p.value is indicated in the middle of the square.

It appears that CD8 whole-slide imaging associates well with:

* FACS data (both CD8 and CD45)
* Manual scoring (Fig 4) of CD8 cells
* Some CD8 gene signature scores (mainly in EPC, TIMER)

```{r}
SummaryData=read.csv("../metadata/new_SummaryData.csv", row.names = 1)

## CD summary
NAidx=sapply(1:ncol(SummaryData), function(x) length(which(!is.na(SummaryData[, x]))))
names(NAidx)=colnames(SummaryData)

colTestCD8=c("CD8.EpBoundingBox", "CD8.TumSize", "CD45.FACS", "CD8.FACS", "OverallCD8_Fig4c.Manual", "TIL_Fig4d.Manual", "T.cell.CD8._TIMER", "T.cell.CD8._CIBERSORT","T.cell.CD8._CIBERSORT.ABS", "T.cell.CD8._EPIC")

# colTestCD8=c("CD8.EpDomTiles","CD8.EpBoundingBox", "CD8.TumSize", "CD45.FACS", "CD8.FACS", "OverallCD8_Fig4c.Manual", "TIL_Fig4d.Manual", )
indx1=c("CD8.WSI", "CD8Frac.WSI", "CD8_EPorSMARatio.WSI","CD8_AnySMARatio.WSI", "CD8_EPRatio.WSI","CD8normTumSize", "CD8.EpBoundingBox", "log2CD8_EPorSMARatio.WSI", "log2CD8_EPRatio.WSI")

CDsummary=matrix(NA, nrow=length(indx1), ncol=length(colTestCD8))
rownames(CDsummary)=paste(indx1, " N=", NAidx[match(indx1, names(NAidx))], sep="")
colnames(CDsummary)=paste(colTestCD8, " N=", NAidx[match(colTestCD8, names(NAidx))], sep="")
CDsummaryP=CDsummary
for (i in 1:length(indx1)){
  CDsummary[i, ]=sapply(colTestCD8, function(x) cor(SummaryData[, indx1[i]], SummaryData[,x], use="complete"))
  CDsummaryP[i, ]=sapply(colTestCD8, function(x) cor.test(SummaryData[, indx1[i]], SummaryData[,x], use="complete")$p.value)
}

## Do the same with EPCAM/FACS data

colTestFACS=grep("FACS", colnames(SummaryData), value = T)
colTestWSI=grep("WSI", colnames(SummaryData), value = T)
FACSsummary=matrix(NA, nrow=length(colTestFACS), ncol=length(colTestWSI))
rownames(FACSsummary)=paste(colTestFACS, " N=", NAidx[match(colTestFACS, names(NAidx))], sep="")
colnames(FACSsummary)=paste(colTestWSI, " N=", NAidx[match(colTestWSI, names(NAidx))], sep="")
FACSsummaryP=FACSsummary


for (i in 1:length(colTestFACS)){
  FACSsummary[i, ]=sapply(colTestWSI, function(x) cor(SummaryData[, colTestFACS[i]], SummaryData[,x], use="complete"))
  FACSsummaryP[i, ]=sapply(colTestWSI, function(x) cor.test(SummaryData[, colTestFACS[i]], SummaryData[,x], use="complete")$p.value)
}

#pdf(sprintf("rslt/WSI-analysis/WSI_vs_FACS_correlationCoeff_w_pvalues_%s.pdf", Sys.Date()), height=8, width=7)
PvalHM(CDsummary, round(CDsummaryP,2), "CD8 WSI correlation with Pvalues")
PvalHM(FACSsummary, round(FACSsummaryP, 2), "WSI vs FACS w Pvalues")
#devoff()

#write.csv(SummaryData, file="metadata/new_SummaryData.csv")
```

## Cellular composition

```{r, cache=T}
WSIvals=sapply(WSIsummary, function(x) table(factor(x$Class2, levels=c("CD8", "EpCAM", "EpCAM: SMA", "SMA", "Unclass"))))
WSIvalFracs=t(t(WSIvals)/colSums(WSIvals))

lxmatch=match(colnames(WSIvalFracs), gsub("_", "", Cdata$TumorID))

Cdata$CD8Fraction=NA
Cdata$CD8Fraction[na.omit(lxmatch)]=WSIvalFracs[1, ]

Cdata$EpCAMFraction=NA
Cdata$EpCAMFraction[na.omit(lxmatch)]=WSIvalFracs[2, ]

Cdata$DPFraction=NA
Cdata$DPFraction[na.omit(lxmatch)]=WSIvalFracs[3, ]

Cdata$SMAFraction=NA
Cdata$SMAFraction[na.omit(lxmatch)]=WSIvalFracs[4, ]

Cdata$UnclassFraction=NA
Cdata$UnclassFraction[na.omit(lxmatch)]=WSIvalFracs[5, ]

```

Here, we look at the raw distributions of the different cell types and see if there are associations with:

* tumor size
* growth rate
* growth rate (categorical)
* treatment
* stromal restricted or infiltrating 

Below are the total cell counts:

```{r, fig.height=4}
WSIvals=WSIvals[ , order(WSIvals[1, ])]
ordV=colnames(WSIvals)

WSIvalFracs=WSIvalFracs[ , order(WSIvalFracs[1, ])]
ordV2=colnames(WSIvalFracs)


## match with meta data
# get the sample Names
CdataTID=gsub("_", "", Cdata$TumorID)
l1=match(colnames(WSIvals), CdataTID)

Treat=Cdata$Treatment[l1]
Growth=Cdata$Growth2[l1]
WSImelt=melt(WSIvals)
WSImelt$Var2=factor(WSImelt$Var2, levels=ordV)
WSImelt$treatment=Treat[match(WSImelt$Var2, colnames(WSIvals))]
WSImelt$growth=Growth[match(WSImelt$Var2, colnames(WSIvals))]

l2=match(colnames(WSIvalFracs), CdataTID)
Treat=factor(Cdata$Treatment[l2])
Growth=Cdata$Growth2[l2]
WSIfracMelt=melt(WSIvalFracs)
WSIfracMelt$Var2=factor(WSIfracMelt$Var2, levels=ordV2)
WSIfracMelt$treatment=Treat[match(WSIfracMelt$Var2, colnames(WSIvalFracs))]
WSIfracMelt$growth=Growth[match(WSIfracMelt$Var2, colnames(WSIvalFracs))]


#pdf(sprintf("rslt/WSI-analysis/summary_distributions_norm_cell_count_%s.pdf", Sys.Date()), width=9, height=5)
ggplot(WSImelt, aes(x=Var2, y=value, fill=Var1))+geom_bar(stat="identity")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ggtitle("ordered by CD8")
ggplot(WSImelt, aes(x=Var2, y=value, fill=Var1))+geom_bar(stat="identity")+facet_grid(~treatment, space="free_x", scale="free_x")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ggtitle("ordered by treatment")
ggplot(WSImelt, aes(x=Var2, y=value, fill=Var1))+geom_bar(stat="identity")+facet_grid(~growth, space="free_x", scale="free_x")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ggtitle("ordered by growth")
```

Here, the same data is shown and normalised according to total cell count:

```{r, fig.hright=5}
ggplot(WSIfracMelt, aes(x=Var2, y=value, fill=Var1))+geom_bar(stat="identity")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ggtitle("ordered by CD8")
ggplot(WSIfracMelt, aes(x=Var2, y=value, fill=Var1))+geom_bar(stat="identity")+facet_grid(~treatment, space="free_x", scale="free_x")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ggtitle("ordered by treatment")
ggplot(WSIfracMelt, aes(x=Var2, y=value, fill=Var1))+geom_bar(stat="identity")+facet_grid(~growth, space="free_x", scale="free_x")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ggtitle("ordered by growth")
#devoff()
```

## Associate composition with other covariates

We can collapse the above data into boxplots to see if there is an association with treatments, performed using non-parametric wilcoxon test. 

Compared to the vehicle, LY treated samples have a lower "stromal" fraction (unclassed DAPI+ cells) and PDL1 treated samples are more likely to have a EpCAM+SMA+ double positive fraction

```{r, warning=F, message=F}
## boxplots here for each cell type stratified by treatment or growth
tsamples=colnames(Cdata)[grep("Fraction", colnames(Cdata))]
#pdf(sprintf("rslt/WSI-analysis/cell-distribution-by-treatment-or-growth_norm_cell_count_%s.pdf", Sys.Date()), width=12, height=8)

par(mfrow=c(2,3))
for (i in tsamples){
  t1=sapply(levels(Treat)[1:3], function(x) wilcox.test(Cdata[Cdata$Treatment==x, i], Cdata[Cdata$Treatment=="Vehicle", i])$p.value)
  boxplot(Cdata[, i]~Cdata$Treatment, main=sprintf("Treatment: %s", i), xlab="", ylab=i)
  legend("topleft", paste(levels(Treat)[1:3], round(t1, 2)))
}
```

When comparing these fractions to growth, there is no association:

```{r, warning='hide'}
par(mfrow=c(2,3))
for (i in tsamples){
  t1=wilcox.test(Cdata[Cdata$Growth2=="growing", i], Cdata[Cdata$Growth2=="stable", i])$p.value
  boxplot(Cdata[, i]~Cdata$Growth2, main=sprintf("Growth: %s p=%s",i, round(t1,2)), xlab="", ylab=i)
# legend("topleft", paste(levels(Cdata$Growth2), round(t1, 2)))
}
```

Nor is there any association with whether a sample is "immune infiltrated" or not (by manual inspection)

```{r, warning='hide'}
par(mfrow=c(2,3))
for (i in tsamples){
  t1=wilcox.test(Cdata[Cdata$InfiltratingVsRestricted=="Infiltrating", i], Cdata[Cdata$InfiltratingVsRestricted=="restricted", i])$p.value
  boxplot(Cdata[, i]~Cdata$InfiltratingVsRestricted, main=sprintf("inf/rest: %s p=%s",i, round(t1,2)), xlab="", ylab=i)
# legend("topleft", paste(levels(Cdata$Growth2), round(t1, 2)))
}
```

We can also use linear regression models to assess:

* correlation between growth rate and cell fractions (no associations)
* correlation between final tumor size and cell fractions (no association)

```{r}

par(mfrow=c(2,3))
for (i in tsamples){
  a1=cor.test(Cdata[ ,i],Cdata$Tumor.dia..sac..mm./Cdata$Time.NMU2Sac, use="complete")
  #a1=cor.test(WSIvals[i, ],TumSizeNorm)
  plot( Cdata$Tumor.dia..sac..mm./Cdata$Time.NMU2Sac, Cdata[ ,i],xlab="growth rate", ylab=i, main=sprintf("rho=%s, p=%s, %s",round(a1$estimate,2), round(a1$p.value,2), i))
}


par(mfrow=c(2,3))
for (i in tsamples){
  a1=cor.test(Cdata[ ,i],Cdata$Tumor.dia..sac..mm., use="complete")
  #a1=cor.test(WSIvals[i, ],TumSizeNorm)
  plot( Cdata$Tumor.dia..sac..mm.,Cdata[ ,i], xlab="tumor size @ sac", ylab=i, main=sprintf("rho=%s, p=%s, %s",round(a1$estimate,2), round(a1$p.value,2), i))
}

#devoff()

```

## Estimate tumor size

Using WSI data, we can estimate a tumor size for each tissue sample and compare to the final tumor sizes. This will be based on the distribution of EpCAM+ cells.
This estimate is also used to normalise CD8 counts earlier on.

```{r}
#estimate the tumor areas
Tarea=lapply(WSIsummary, function(x) ripras(x$Centroid.X.µm[x$Class2=="EpCAM"], x$Centroid.Y.µm[x$Class2=="EpCAM"], "convex"))
TareaSum=sapply(Tarea, area)

# Tumor diameter
Tdiameter=Cdata$Tumor.dia..sac..mm.[match(colnames(WSIvals), gsub("_", "", Cdata$TumorID))]

#par(mfrow=c(1,2))
#plot(TareaSum, WSIvals[2, ], ylab="EpCAM cell count", xlab="Epcam cells bounding area")
plot(TareaSum, Tdiameter, ylab="Tumor Diameter", xlab="Epcam cells bounding area")

cor.test(TareaSum, Tdiameter, use="complete")

## append to Cdata
Cdata$TumorAreaWSI=NA
lxmatch=match(colnames(WSIvalFracs), gsub("_", "", Cdata$TumorID)) # probably don't need this
Cdata$TumorAreaWSI[(lxmatch)]=TareaSum #[-which(is.na(lxmatch))]
## Normalisation strategies:
```


## Correlations between different subpopulations

Look for correlates between different subpopulations: Naturally, we would expect a negative correlation since this should sum to 1. Below are heatmaps showing correlations between different cell types, and significant associations are linearly shown.

Note the following negative correlations:

* epcam and SMA
* SMA+ and Unclass 

```{r}
Ax1=cor(t(WSIvalFracs))

#pdf(sprintf("rslt/WSI-analysis/summary-associations-between-subgroups_%s.pdf", Sys.Date()), height=8, width=8)
par(oma=c(4, 0,0, 4))
heatmap.2(Ax1, col=brewer.pal(9, "RdBu"), trace="none",scale="none")

par(mfrow=c(2,2))

a1=cor.test(WSIvalFracs[2, ], WSIvalFracs[4, ])
plot(WSIvalFracs[2, ], WSIvalFracs[4, ], xlab="Epcam+", ylab="SMA+", main=sprintf("rho=%s,p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))

a1=cor.test(WSIvalFracs[4, ], WSIvalFracs[5, ])
plot(WSIvalFracs[4, ], WSIvalFracs[5, ], xlab="SMA+", ylab="Unclass", main=sprintf("rho=%s,p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))

a1=cor.test(WSIvalFracs[3, ], WSIvalFracs[4, ])
plot(WSIvalFracs[3, ], WSIvalFracs[4, ], xlab="Epcam+SMA+", ylab="SMA", main=sprintf("rho=%s,p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))

a1=cor.test(WSIvalFracs[2, ], WSIvalFracs[5, ])
plot(WSIvalFracs[2, ], WSIvalFracs[5, ], xlab="Epcam+", ylab="nclass", main=sprintf("rho=%s,p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))

#devoff()
# correlations
Ax1

ctest=matrix(NA, nrow=5, ncol=5)
for (i in 1:5){
  ctest[i, ]=sapply(1:5, function(x) cor.test(WSIvalFracs[i, ], WSIvalFracs[x, ])$p.value)
}

ctest
```

## Associations between CD8 counts with other clinical variables

Below we assess whether any of the CD8-variables described in section 3.1 is associated with 

* treatment
* growth
* spatial pattern

```{r}
indx1=c(indx1, "log2CD8_EPorSMARatio.WSI", "log2CD8_EPRatio.WSI")

a1=matrix(NA, ncol=length(indx1), nrow=5)

for (i in 1:length(indx1)){
a1[1, i]=wilcox.test(SummaryData[SummaryData$Growth%in%c("stable", "growing") ,indx1[i] ]~ SummaryData$Growth[SummaryData$Growth%in%c("stable", "growing")])$p.value 
a1[2, i]=wilcox.test(SummaryData[SummaryData$SpatialManual%in%c("Infiltrating", "restricted") ,indx1[i] ]~ SummaryData$SpatialManual[SummaryData$SpatialManual%in%c("Infiltrating", "restricted")])$p.value
a1[3, i]=wilcox.test(SummaryData[SummaryData$Treatment%in%c("PDL1", "Vehicle") ,indx1[i] ]~ SummaryData$Treatment[SummaryData$Treatment%in%c("PDL1", "Vehicle")])$p.value
a1[4, i]=wilcox.test(SummaryData[SummaryData$Treatment%in%c("PDL1+LY", "Vehicle") ,indx1[i] ]~ SummaryData$Treatment[SummaryData$Treatment%in%c("PDL1+LY", "Vehicle")])$p.value
a1[5, i]=wilcox.test(SummaryData[SummaryData$Treatment%in%c("LY", "Vehicle") ,indx1[i] ]~ SummaryData$Treatment[SummaryData$Treatment%in%c("LY", "Vehicle")])$p.value
}

colnames(a1)=indx1
rownames(a1)=c("growth", "spatial", "pdl1", "pdl1+ly", "ly")

par(oma=c(5, 0,0,4))
heatmap.2(-log10(a1), col=brewer.pal(9, "Blues"), trace="none", scale="none")
```

Note that CD8 normalised by tumor size is associated with growth (but this could a reflection of the size of the tumor), and there is a borderline difference once normalised by epithelial content. In addition the CD8 total count is associated with pdl1+ly treatment.

Note that p=0.05 is designated by a value of 1.3


```{r, eval=F}
## save this data to file to generate the summary data file
## Check if the other metrics used correlate with growth status or treatment

#####
# COMMENT OUT IF NOT NEEDED
#####


## add FACS data
# lx=match(rownames(SummaryData), colnames(Fdata))
# SummaryData$CD8.FACS=NA
# SummaryData$CD8.FACS[which(!is.na(lx))]=unlist(t(Fdata[ 6, na.omit(lx)]))
# lx=match(rownames(SummaryData), gsub("_", "", Cdata$TumorID))
# SummaryData$CD45.FACS=Cdata$X..CD45..by.FACS.[lx]
# SummaryData$TumSize=Cdata$Tumor.dia..sac..mm.[lx]
# SummaryData$CD8normTumSize=SummaryData$CD8.WSI/SummaryData$TumSize
# SummaryData$CD8.EpBoundingBox=TareaSum[match(rownames(SummaryData), names(TareaSum))]/SummaryData$CD8.WSI
# SummaryData$Treatment=factor(Cdata$Treatment[lx])
# SummaryData$Growth=Cdata$Growth2[lx]
# SummaryData$SpatialManual=Cdata$InfiltratingVsRestricted[lx]
# SummaryData$log2CD8_EPorSMARatio.WSI=log2(SummaryData$CD8_EPorSMARatio.WSI+0.001)
# SummaryData$log2CD8_EPRatio.WSI=log2(SummaryData$CD8_EPRatio.WSI+0.001)

#####
# END COMMENT section
#####

indx1=c(indx1, "log2CD8_EPorSMARatio.WSI", "log2CD8_EPRatio.WSI")

#pdf(sprintf("rslt/WSI-analysis/WSI_CD8_metrics_vs_growth_%s.pdf", Sys.Date()), height=8, width=8)
for (i in 1:length(indx1)){
  a1=wilcox.test(SummaryData[SummaryData$Growth%in%c("stable", "growing") ,indx1[i] ]~ SummaryData$Growth[SummaryData$Growth%in%c("stable", "growing")])
p<-ggplot(SummaryData, aes_string(y=(indx1[i]), x="Growth"))+geom_boxplot(outlier.shape=NA)+geom_jitter(position=position_jitter(width=.1, height=0))+ggtitle(paste(indx1[i], " stab vs grow p=", round(a1$p.value,2),sep=""))
print(p)
}
#devoff()

#pdf(sprintf("rslt/WSI-analysis/WSI_CD8_metrics_vs_treatment_%s.pdf", Sys.Date()), height=8, width=8)
for (i in 1:length(indx1)){
a1=wilcox.test(SummaryData[SummaryData$Treatment%in%c("Vehicle", "PDL1"),indx1[i] ]~ SummaryData$Treatment[SummaryData$Treatment%in%c("Vehicle", "PDL1")])
a2=wilcox.test(SummaryData[SummaryData$Treatment%in%c("Vehicle", "LY"),indx1[i] ]~ SummaryData$Treatment[SummaryData$Treatment%in%c("Vehicle", "LY")])
a3=wilcox.test(SummaryData[SummaryData$Treatment%in%c("Vehicle", "PDL1+LY"),indx1[i] ]~ SummaryData$Treatment[SummaryData$Treatment%in%c("Vehicle", "PDL1+LY")])
p<-ggplot(SummaryData, aes_string(y=(indx1[i]), x="Treatment"))+geom_boxplot(outlier.shape=NA)+geom_jitter(position=position_jitter(width=.1, height=0))+ggtitle(paste(indx1[i], " PDL1p=", round(a1$p.value,2), " LYp=", round(a2$p.value,2), " P+Lp=", round(a3$p.value,2), sep=""))
print(p)
}
#devoff()
```



<!--chapter:end:02-wsi.Rmd-->

# Spatial statistics

Below, we use three different metrics to compare spatial distributions:

* k-nearest neighbour distances
* the interacting fraction
* morisita-horn distances

These are compared to manual inspection of the result

## knn-Distances:

```{r}
#load("outputs/WSI_raw_data2020-11-05.RData")
```
The k-nearest neighbour distances looks at the average distance from a given cell type of class A to a cell type of class B. In this section, the reference class A is the CD8 T cell, and we will look at the mean distance to SMA, Epcam, double positive and unclassified cells in each image.

To account for potential fluctuations due to misclassified cells, or isolated single cells, k values of 1, 3, 5 will be used. I.e. for each cell, we will compute the mean distance from each Cd8Tcell to its 1, 3, and 5 nearest neighbours.

### Comparison to manual classification, treatment, growth

Overall, we see that the differences in infiltrating vs restricted are similar. We see statistical differences (using anova followed by Tukey's test) between: 

* epcam and SMA-epcam in both cases (higher distances to EpCAM on average)
* SMA-Epcam to SMA (CD8s are closer to SMA+)
* Unclass to Epcam-SMA (CD8s closer to unclass)

In the infiltrating case: 

* Unclass to Epcam (CD8s closer to unclass, borderline significant)

In the restricted cases, we see:

* Unclass to SMA (higher distance to unclass in the restricted case)
* SMA to epcam (CD8s are closer to the SMA)

This last result is consistent with what we expect for a CD8+ cell which is stroma-restricted.

```{r, fig.height=5}
knnMelt=melt(WSIknn, measure.vars=c("KS.pval"))

knnTemp1=knnMelt[knnMelt$CellType=="CD8" & knnMelt$NearestCellType%in%c("EpCAM", "SMA", "EpCAM: SMA", "Unclass"), ]
knnTemp1$SpatialManual=Cdata$InfiltratingVsRestricted[match(knnTemp1$L1, gsub("_", "", Cdata$TumorID))]

# compute p values?
ptest=sapply(levels(knnTemp1$knn), function(x) wilcox.test(knnTemp1$MeanDistance[which(knnTemp1$knn==x & knnTemp1$NearestCellType=="EpCAM" & knnTemp1$SpatialManual=="Infiltrating")], knnTemp1$MeanDistance[which(knnTemp1$knn==x & knnTemp1$NearestCellType=="SMA" & knnTemp1$SpatialManual=="Infiltrating")])$p.value)

fit1=aov(MeanDistance~NearestCellType+knn, data=knnTemp1[knnTemp1$SpatialManual=="Infiltrating", ])
#summary(fit1)
fit2=aov(MeanDistance~NearestCellType+knn, data=knnTemp1[knnTemp1$SpatialManual=="restricted", ])
#summary(fit2)

fit1b=aov(MedianDistance~NearestCellType+knn, data=knnTemp1[knnTemp1$SpatialManual=="Infiltrating", ])
#summary(fit1b)
fit2b=aov(MedianDistance~NearestCellType+knn, data=knnTemp1[knnTemp1$SpatialManual=="restricted", ])
#summary(fit2b)

#pdf(sprintf("rslt/WSI-analysis/knn_distances_vs_manualspatial_%s.pdf", Sys.Date()), height=7, width=12)

p<-ggplot(knnTemp1, aes(x=NearestCellType, y=MeanDistance, fill=SpatialManual))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=SpatialManual))+facet_grid(~knn+SpatialManual)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c", "#377eb8"),na.value="black")+ggtitle("CD8-CelltypeX spatial distributions: mean values")

print(p)
par(mfrow=c(1,2), oma=c(0, 0, 2, 0))
plot(TukeyHSD(fit1, "NearestCellType"), las=2)
title("Infiltrating Mean Distance from CD8", line=3.3)
plot(TukeyHSD(fit2, "NearestCellType"), las=2)
title("Restricted Mean Distance from CD8", line=3.3)

# p<-ggplot(knnTemp1, aes(x=NearestCellType, y=MedianDistance, fill=SpatialManual))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=SpatialManual))+facet_grid(~knn+SpatialManual)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c", "#377eb8"),na.value="black")+ggtitle("CD8-CelltypeX spatial distributions: median values")
# print(p)
# 
# par(mfrow=c(1,2), oma=c(0, 0, 2, 0))
# plot(TukeyHSD(fit1b, "NearestCellType"), las=2)
# title("Infiltrating Median Distance from CD8", line=3.3)
# plot(TukeyHSD(fit2b, "NearestCellType"), las=2)
# title("Restricted Median Distance from CD8", line=3.3)

#devoff()

```

### Associations with outcome

We can also see if there is an association between these distances with growth and treatment

Treatment:

* CD8 cells in PDL1 sample are further away to SMA+ cells and EpCAM+   (compared to vehicle or double agent)
* CD8 cells in LY treated samples are further away from unclassified cells  (compared to any of the other treatments)

```{r}

knnTemp1$Treatment=Cdata$Treatment[match(knnTemp1$L1, gsub("_", "", Cdata$TumorID))]

fit1=aov(MeanDistance~Treatment+knn, data=knnTemp1[knnTemp1$NearestCellType=="EpCAM", ])
fit2=aov(MeanDistance~Treatment+knn, data=knnTemp1[knnTemp1$NearestCellType=="SMA", ])
fit3=aov(MeanDistance~Treatment+knn, data=knnTemp1[knnTemp1$NearestCellType=="EpCAM: SMA", ])
fit4=aov(MeanDistance~Treatment+knn, data=knnTemp1[knnTemp1$NearestCellType=="Unclass", ])

knnTemp1$Growth=Cdata$Growth2[match(knnTemp1$L1, gsub("_", "", Cdata$TumorID))]

fit1b=aov(MeanDistance~Growth+knn, data=knnTemp1[knnTemp1$NearestCellType=="EpCAM", ])
fit2b=aov(MeanDistance~Growth+knn, data=knnTemp1[knnTemp1$NearestCellType=="SMA", ])
fit3b=aov(MeanDistance~Growth+knn, data=knnTemp1[knnTemp1$NearestCellType=="EpCAM: SMA", ])
fit4b=aov(MeanDistance~Growth+knn, data=knnTemp1[knnTemp1$NearestCellType=="Unclass", ])

#pdf(sprintf("rslt/WSI-analysis/knn_distances_vs_treatment_growth_%s.pdf", Sys.Date()), height=7, width=12)

# p<-ggplot(knnTemp1, aes(x=NearestCellType, y=MeanDistance, fill=Treatment))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=Treatment))+facet_grid(~knn+Treatment)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c", "#4daf4a","#377eb8",  "#984ea3"),na.value="black")
# print(p)
p<-ggplot(knnTemp1, aes(fill=Treatment, y=MeanDistance, x=Treatment))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=Treatment))+facet_grid(~knn+NearestCellType)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c", "#4daf4a","#377eb8",  "#984ea3"),na.value="black")
print(p)
par(mfrow=c(2,2), oma=c(0, 0, 2, 0))
plot(TukeyHSD(fit1, "Treatment"), las=2)
title("EpCAM Mean Distance", line=3.3)
plot(TukeyHSD(fit2, "Treatment"), las=2)
title("SMA Mean Distance", line=3.3)
plot(TukeyHSD(fit3, "Treatment"), las=2)
title("EpCAM:SMA Mean Distance", line=3.3)
plot(TukeyHSD(fit4, "Treatment"), las=2)
title("Unclass Distance", line=3.3)
```

### Growth

All 95% confidence lines cross 0, but it appears that stable cases have a closer unclass-CD8 interaction distance compared to growing.

```{r}
# p<-ggplot(knnTemp1, aes(x=NearestCellType, y=MeanDistance, fill=Growth))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=Growth))+facet_grid(~knn+Growth)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c","#ff7f00", "#4daf4a","#377eb8",  "#984ea3", "purple"),na.value="black")
# print(p)
p<-ggplot(knnTemp1, aes(fill=Growth, y=MeanDistance, x=Growth))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=Growth))+facet_grid(~knn+NearestCellType)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c","#ff7f00", "#4daf4a","#377eb8",  "#984ea3", "purple"),na.value="black")
print(p)
par(mfrow=c(2,2), oma=c(0, 0, 2, 0))
plot(TukeyHSD(fit1b, "Growth"), las=2)
title("EpCAM Mean Distance", line=3.3)
plot(TukeyHSD(fit2b, "Growth"), las=2)
title("SMA Mean Distance", line=3.3)
plot(TukeyHSD(fit3b, "Growth"), las=2)
title("EpCAM:SMA Mean Distance", line=3.3)
plot(TukeyHSD(fit4b, "Growth"), las=2)
title("Unclass Distance", line=3.3)

#devoff()

```


Based on the above distributions, knn1, knn3, and knn5 analyses give similar results. In the following section, we will make comparisons using knn3 results.

### Association between distance and content

The spatial differences could be influenced by CD8 content. Here, we test if the distances from CD8Tcells to celltype B could be influenced by this. 

We see that there is a correlation between CD8 fraction and the SMA proportion (both SMA and double positive).
But we don't see an association with epcam+ cells or unclassified stromal cells


```{r, fig.height=6}
A1=knnTemp1[knnTemp1$knn=="knn3" & knnTemp1$CellType=="CD8",  ]
A1$CD8frac=WSIvalFracs[1, match(A1$L1, colnames(WSIvals))]

#pdf(sprintf("rslt/WSI-analysis/check_knn3-averageDistance_association_withCD8fraction_%s.pdf", Sys.Date()), height=8, width=8)
par(mfrow=c(2,2))
for (i in unique(A1$NearestCellType)){
a1=cor.test(A1$MeanDistance[A1$NearestCellType==i], A1$CD8frac[A1$NearestCellType==i], use="complete")
plot(A1$MeanDistance[A1$NearestCellType==i], A1$CD8frac[A1$NearestCellType==i], xlab="3nn mean distance", ylab="CD8 fraction", main=sprintf("%s, cor=%s p=%s", i, round(a1$estimate,2), round(a1$p.value,2)), log="x")
}

```

CD8-Epcam distances are looked at in greater detail below. Here, we look at whether there is an association with CD8 fraction in a growing and stable tumors.

In growing or spatially restricted tumors, there is an association between the CD8-epcam distances with the CD8 fraction, but not in infiltrating or stable tumours.

```{r, fig.height=4}
knnTempSumm=knnTemp1[which(knnTemp1$knn=="knn3"), ]
knnreshape=acast(knnTempSumm[ ,c("NearestCellType", "MeanDistance", "L1")], L1~NearestCellType, value.var="MeanDistance" )
knnreshape=data.frame(knnreshape)

knnreshape$EpMIN=ifelse(knnreshape$EpCAM..SMA<knnreshape$EpCAM, knnreshape$EpCAM..SMA, knnreshape$EpCAM)
knnreshape$EpMIN[which(is.na(knnreshape$EpMIN))]=knnreshape$EpCAM[which(is.na(knnreshape$EpMIN))]
knnreshape$EpStrRatio1=knnreshape$EpCAM/rowSums(knnreshape[ ,c("EpCAM..SMA", "SMA")], na.rm=T)
knnreshape$EpStrRatio2=knnreshape$EpCAM/(knnreshape$SMA)
knnreshape$EpStrRatio3=rowSums(knnreshape[ ,c("EpCAM..SMA", "EpCAM")], na.rm=T)/rowSums(knnreshape[ ,c("EpCAM..SMA", "SMA")], na.rm=T)
knnreshape$EpStrRatio4=rowSums(knnreshape[ ,c("EpCAM..SMA", "EpCAM")], na.rm=T)/(knnreshape$SMA)
knnreshape$Treatment=factor(knnTempSumm$Treatment[match(rownames(knnreshape), knnTempSumm$L1)])
knnreshape$Growth=factor(knnTempSumm$Growth[match(rownames(knnreshape), knnTempSumm$L1)])
knnreshape$Infil=factor(knnTempSumm$SpatialManual[match(rownames(knnreshape), knnTempSumm$L1)])
knnreshape$CD8frac=SummaryData$CD8Frac.WSI[match(rownames(knnreshape), rownames(SummaryData))]


par(mfrow=c(1,2))
a1x=cor.test(knnreshape$CD8frac[knnreshape$Infil=="Infiltrating"], knnreshape$EpCAM[knnreshape$Infil=="Infiltrating"])
a1y=cor.test(knnreshape$CD8frac[knnreshape$Infil=="restricted"], knnreshape$EpCAM[knnreshape$Infil=="restricted"])
plot(knnreshape$CD8frac, knnreshape$EpCAM, col=knnreshape$Infil, pch=19, xlab="CD8 fraction", ylab="CD8-EpCAM knn3", main="spatial scoring")
legend("topright", c(paste("infil p=", round(a1x$p.value,2)), paste("restrict p=", round(a1y$p.value,2))), lwd=2, col=c(1,2))

a1x=cor.test(knnreshape$CD8frac[knnreshape$Growth=="growing"], knnreshape$EpCAM[knnreshape$Growth=="growing"])
a1y=cor.test(knnreshape$CD8frac[knnreshape$Growth=="stable"], knnreshape$EpCAM[knnreshape$Growth=="stable"])

plot(knnreshape$CD8frac, knnreshape$EpCAM, col=knnreshape$Growth, pch=19, xlab="CD8 fraction", ylab="CD8-EpCAM knn3", main="tumor growth")
legend("topright", c(paste("growing p=", round(a1x$p.value,2)), paste("stable p=", round(a1y$p.value,2))), lwd=2, col=c(1,2))
```

Do any of the below metrics correlate with manual scoring:

* We can assess the distances from CD8 cells which come automatically: 
    * Ep-distance
    * SMA distance 
    * Ep:SMA distance
    * SMA distance.
* EpMIN: distance to any Epcam expressing cell (includes EP+SMA+)
* Ratios between Epcam and SMA: There are different values depending on whether the double positive fraction is used or not
    * (Ratio 1): Ep/any SMA
    * (Ratio 2): EP/SMA
    * (Ratio 3): Any Ep / Any SMA
    * (Ratio 4): Any Ep/ SMA
    
Whilst the manual scorings associate strongly with CD8-EPcam distances, there is no association with any other cell type.

```{r, fig.height=7}

#pdf(sprintf("rslt/WSI-analysis/knn_metrics_to_manualScoring_%s.pdf", Sys.Date()), height=12, width=12)
par(mfrow=c(3,3))
for (i in 1:9){
  a1=wilcox.test(knnreshape[ ,i]~knnreshape$Infil)
  boxplot(knnreshape[ ,i]~knnreshape$Infil, main=paste(colnames(knnreshape)[i]," p=", round(a1$p.value,2), sep=""), ylab="knn distance" )
}
# dev.off()
```

## The interacting fraction

The interacting fraction uses the knn-distances and determines the proportion of CD8 cells which are within a proximity of r um from celltype B.

### Comparison to manual & select optimal r

Below are plots of the proportion of CD8 cells within an "interacting distance" as we increase r. This looks at both the interacting fraction of CD8 cells with Epcam+ and SMA+ cells. Lines are color coded according to the manual spatial-infiltration annotation. 

We notice from the line plots for each single sample that the restricted samples generally have low interacting fractins with EpCAM and SMA compared to the infiltrating samples. In addition, there is a statistical difference in EpCAM measurements compared to SMA.

```{r, fig.height=4}
IFmelt=melt(WSIIF, measure.vars=c("IF"))

IFmelt$SpatialManual=Cdata$InfiltratingVsRestricted[match(IFmelt$L1, gsub("_", "", Cdata$TumorID))]
IFmelt$Treatment=Cdata$Treatment[match(IFmelt$L1, gsub("_", "", Cdata$TumorID))]
IFmelt$Growth=Cdata$Growth2[match(IFmelt$L1, gsub("_", "", Cdata$TumorID))]
IFmelt$Growth2=IFmelt$Growth
IFmelt$Growth2[grep("no data", IFmelt$Growth2)]="no data"

IFmelt$Dist=(substr(IFmelt$Grid, 6, 7))
IFmelt$knn=substr(IFmelt$Grid, 1, 4)

IFTempSumm=IFmelt[IFmelt$Grid=="knn3-15" & IFmelt$Reference=="CD8", ]
IFreshape=acast(IFTempSumm[ ,c("NearestNeighbor", "value", "L1")], L1~NearestNeighbor, value.var="value" )
IFreshape=data.frame(IFreshape)

IFreshape$EpMIN=ifelse(IFreshape$EpCAM..SMA<IFreshape$EpCAM, IFreshape$EpCAM..SMA, IFreshape$EpCAM)
IFreshape$EpMIN[which(is.na(IFreshape$EpMIN))]=IFreshape$EpCAM[which(is.na(IFreshape$EpMIN))]
IFreshape$EpStrRatio1=IFreshape$EpCAM/rowSums(IFreshape[ ,c("EpCAM..SMA", "SMA")], na.rm=T)
IFreshape$EpStrRatio2=IFreshape$EpCAM/(IFreshape$SMA)
IFreshape$EpStrRatio3=rowSums(IFreshape[ ,c("EpCAM..SMA", "EpCAM")], na.rm=T)/rowSums(IFreshape[ ,c("EpCAM..SMA", "SMA")], na.rm=T)
IFreshape$EpStrRatio4=rowSums(IFreshape[ ,c("EpCAM..SMA", "EpCAM")], na.rm=T)/(IFreshape$SMA)


IFreshape$Treatment=factor(IFTempSumm$Treatment[match(rownames(IFreshape), IFTempSumm$L1)])
IFreshape$Growth=factor(IFTempSumm$Growth[match(rownames(IFreshape), IFTempSumm$L1)])
IFreshape$Infil=factor(IFTempSumm$SpatialManual[match(rownames(IFreshape), IFTempSumm$L1)])
IFreshape$CD8frac=SummaryData$CD8Frac.WSI[match(rownames(IFreshape), rownames(SummaryData))]
IFreshape$TumSize=SummaryData$TumSize[match(rownames(IFreshape), rownames(SummaryData))]

####
# line plots to see the best separation between infiltrating and restricted
###

#pdf(sprintf("rslt/WSI-analysis/interacting_fraction_compared_manual_%s.pdf", Sys.Date()), height=8, width=14)

IFmelt2=IFmelt[IFmelt$NearestNeighbor=="EpCAM" & IFmelt$Reference=="CD8" , ]
IFmelt2$label=IFmelt2$L1
IFmelt2$label[which(IFmelt2$Dist!=30)]=NA
p<-ggplot(IFmelt2, aes(x=Dist, y=value, col=SpatialManual, group=L1, label=label))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-EpCAM interacting fraction: spatial manual")+geom_label()
print(p)

#IFmelt2$Dist=as.numeric(IFmelt2$Dist)

p<-ggplot(IFmelt2, aes(x=Dist, y=value, col=SpatialManual, label=label))+geom_boxplot()+ylab("Interacting Fraction")+xlab("Distance in microns")+facet_grid(~knn)+ggtitle("CD8-EpCAM knn3")+stat_smooth()
print(p)

IFmelt2=IFmelt[IFmelt$NearestNeighbor=="SMA" & IFmelt$Reference=="CD8" , ]
IFmelt2$label=IFmelt2$L1
IFmelt2$label[which(IFmelt2$Dist!=30)]=NA

p<-ggplot(IFmelt2, aes(x=Dist, y=value, col=SpatialManual, group=L1, label=label))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-SMA interacting fraction: spatial manual")+geom_label()
print(p)


p<-ggplot(IFmelt2, aes(x=Dist, y=value, col=SpatialManual, label=label))+geom_boxplot()+ylab("Interacting Fraction")+xlab("Distance in microns")+facet_grid(~knn)+ggtitle("CD8-SMA knn3 ")
print(p)

# ggplot(IFmelt2, aes(x=Dist, y=value, col=SpatialManual, linetype=SpatialManual))+geom_point()+stat_smooth()+ggtitle("CD8-EpCAM interacting fraction: spatial manual")+ylab("Interacting Fraction")

# ## Do a dot plot
 # p<-ggplot(IFmelt2, aes(x=Dist, y=value, col=SpatialManual, label=label))+geom_boxplot()+ylab("Interacting Fraction")+xlab("Distance in microns")+facet_grid(~knn)+ggtitle("CD8-SMA knn3 ")
 # print(p)
```


Using the boxplots as a guide, we can determine optimal "interacting distances" at which to perform downstream analysis. The best separation between restricted and infiltrating for EpCAM appears at:

* 1-nn: 10-15 um
* 3-nn: 15 um
* 5-nn: 20 um

The interacting fraction does not distinguish SMA fractions (all restricted boxplots overlap with the infiltrating boxplots)


The following plots use 3NN analysis with an interacting distance of 15um. We can firstly check if there is an association between different "interacting fraction" types and manual scoring, similar to what was performed for knn-analysis. Only CD8-EpCAM interacting distances is associated with manual scoring. All other metrics are not significant.

```{r, fig.height=6}
## reshape and do a correlation plot like for the knn analysis
par(mfrow=c(3,3))
for (i in 1:9){
  a1=wilcox.test(IFreshape[ ,i]~IFreshape$Infil)
  boxplot(IFreshape[ ,i]~IFreshape$Infil, main=paste(colnames(IFreshape)[i]," p=", round(a1$p.value,2), sep="" ), xlab="IF: knn3, 15um")
}
```

Again, association between CD8 content and interacting fraction was observed ONLY in the growing samples or restricted cases. 

```{r, fig.height=4}
par(mfrow=c(1,2))
a1x=cor.test(IFreshape$CD8frac[IFreshape$Infil=="Infiltrating"], IFreshape$EpCAM[IFreshape$Infil=="Infiltrating"])
a1y=cor.test(IFreshape$CD8frac[IFreshape$Infil=="restricted"], IFreshape$EpCAM[IFreshape$Infil=="restricted"])
plot(IFreshape$CD8frac, IFreshape$EpCAM, col=IFreshape$Infil, pch=19, xlab="CD8 fraction", ylab="CD8-EpCAM IF (knn3-15um)", main="spatial scoring")
legend("topright", c(paste("infil p=", round(a1x$p.value,2)), paste("restrict p=", round(a1y$p.value,2))), lwd=2, col=c(1,2))

a1x=cor.test(IFreshape$CD8frac[IFreshape$Growth=="growing"], IFreshape$EpCAM[IFreshape$Growth=="growing"])
a1y=cor.test(IFreshape$CD8frac[IFreshape$Growth=="stable"], IFreshape$EpCAM[IFreshape$Growth=="stable"])

plot(IFreshape$CD8frac, IFreshape$EpCAM, col=IFreshape$Growth, pch=19, xlab="CD8 fraction", ylab="CD8-EpCAM IF (knn3-15um)", main="tumor growth")
legend("topright", c(paste("growing p=", round(a1x$p.value,2)), paste("stable p=", round(a1y$p.value,2))), lwd=2, col=c(1,2))
```

### Growth

Here, we check if there is an association between the spatial pattern and tumor growth.

```{r if-growth}

######
# compare these metrics with growth
####

#pdf(sprintf("rslt/WSI-analysis/interacting_fraction_vs_treatment_growth_%s.pdf", Sys.Date()), height=7, width=12)
ctypes=unique(IFmelt$NearestNeighbor)
t2=IFmelt[which(IFmelt$Reference=="CD8" & IFmelt$knn=="knn3" & IFmelt$Dist==15) , ]

pval2=sapply(ctypes, function(x) wilcox.test(t2$value[t2$NearestNeighbor==x & t2$Growth2=="growing" ], t2$value[t2$NearestNeighbor==x & t2$Growth2=="stable" ])$p.value)

ann_text=data.frame(Glabel=round(pval2,2), NearestNeighbor=(ctypes ), Growth2="stable", value=0.7)

p<-ggplot(IFmelt[IFmelt$Reference=="CD8" & IFmelt$knn=="knn3" & IFmelt$Dist==15 , ], aes(x=Growth2, y=value, col=Growth2))+facet_grid(~NearestNeighbor)+geom_boxplot()+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("Interacting fraction, knn3, dist=15")
p+geom_text(data=ann_text, mapping=aes(x=2, y=0.75, label=Glabel))


# p<-ggplot(IFmelt[IFmelt$NearestNeighbor=="EpCAM" & IFmelt$Reference=="CD8" , ], aes(x=Dist, y=value, col=Growth2, group=L1))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-EpCAM interacting fraction: growth")
# print(p)
# p<-ggplot(IFmelt[IFmelt$NearestNeighbor=="SMA" & IFmelt$Reference=="CD8" , ], aes(x=Dist, y=value, col=Growth2, group=L1))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-SMA interacting fraction: growth")
# print(p)
# p<-ggplot(IFmelt[IFmelt$NearestNeighbor=="EpCAM: SMA" & IFmelt$Reference=="CD8" , ], aes(x=Dist, y=value, col=Growth2, group=L1))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-EpCAM:SMA interacting fraction: growth")
# print(p)
# p<-ggplot(IFmelt[IFmelt$NearestNeighbor=="Unclass" & IFmelt$Reference=="CD8" , ], aes(x=Dist, y=value, col=Growth2, group=L1))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-Unclass interacting fraction; spatial manual")
# print(p)
```

None of the above metrics associate with growth. (P value by wilcox test shown)

### Treatment

Similarly, compare the distances with treatment: 

```{r}

ctypes=unique(IFmelt$NearestNeighbor)
t2=IFmelt[which(IFmelt$Reference=="CD8" & IFmelt$knn=="knn3" & IFmelt$Dist==15) , ]
TreatV=sort(unique(IFmelt$Treatment))

pval2=matrix(NA, nrow=3, ncol=5)
colnames(pval2)=ctypes
rownames(pval2)=TreatV[1:3]

for (i in 1:3){
pval2[i, ]=sapply(ctypes, function(x) wilcox.test(t2$value[t2$NearestNeighbor==x & t2$Treatment==TreatV[i]], t2$value[t2$NearestNeighbor==x & t2$Treatment=="Vehicle"])$p.value)
}

pmelt=melt(pval2)
colnames(pmelt)=c("Treatment", "NearestNeighbor", "label")
pmelt$label=round(pmelt$label, 2)
pmelt$value=0.8

p<-ggplot(IFmelt[IFmelt$Reference=="CD8" & IFmelt$knn=="knn3" & IFmelt$Dist==15 , ], aes(x=Treatment, y=value, col=Treatment))+facet_grid(~NearestNeighbor)+geom_boxplot()+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("Interacting fraction, knn3, dist=15")
p+geom_text(data=pmelt, mapping=aes(x=Treatment, y=0.75, label=label, col=Treatment))
print(p)
# p<-ggplot(IFmelt[IFmelt$NearestNeighbor=="EpCAM" & IFmelt$Reference=="CD8" , ], aes(x=Dist, y=value, col=Treatment, group=L1))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-EpCAM interacting fraction: Treatment")
# print(p)
# p<-ggplot(IFmelt[IFmelt$NearestNeighbor=="SMA" & IFmelt$Reference=="CD8" , ], aes(x=Dist, y=value, col=Treatment, group=L1))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-SMA interacting fraction: Treatment")
# print(p)
# p<-ggplot(IFmelt[IFmelt$NearestNeighbor=="EpCAM: SMA" & IFmelt$Reference=="CD8" , ], aes(x=Dist, y=value, col=Treatment, group=L1))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-EpCAM:SMA interacting fraction: Treatment")
# print(p)
# p<-ggplot(IFmelt[IFmelt$NearestNeighbor=="Unclass" & IFmelt$Reference=="CD8" , ], aes(x=Dist, y=value, col=Treatment, group=L1))+facet_grid(~knn)+geom_line(aes(group=L1))+ylab("Interacting Fraction")+xlab("Distance in microns")+ggtitle("CD8-Unclass interacting fraction; spatial manual")
# print(p)
#devoff()
```

There appears to be a difference in CD8-unclass interactions in LY treated samples (LY, PDL1+LY), but not in the other cases

## M-H distances

The M-H distance (or Morisita Horn index) can be considered as a correlation coefficient in spatial distribution between cell type A and cell type B. To calculate this metric, the whole slide image is divided into grids of size 50 to 500um. Within each grid, the total number of cells A and B are determined. 

The M-H index is thus determined as:

$$ \frac{2\sum_{i=1}^n a_ib_i}{(D_a+D_b)AB} $$
where $a_i$ and $b_i$ are the number of cells in grid $i$, $$A$$ and $$B$$ the total number of cells, and $$D_x$$ is the Simpson's index.

### Comparison to Manual Scoring

Similar to the interacting fraction, we plot the MH index for increasing values of gridsize to determine an optimal metric to compare spatial patterns. Ideally, we would pick a metric has the following properties:

* good separation of the different values
* a reasonable number of cells within each grid (avoid too small grids which give counts of 0)
* avoid plateauing of MH values because the grid size is too large

```{r, fig.height=4}

MHmeltsumm=melt(WSIMHsetup, id.vars=c("gridsize", "Ntiles"))

MHmelt=melt(WSIMH, measure.vars="MH.mean")
MHmelt$SpatialManual=Cdata$InfiltratingVsRestricted[match(MHmelt$L1, gsub("_", "", Cdata$TumorID))]
MHmeltsumm$SpatialManual=Cdata$InfiltratingVsRestricted[match(MHmeltsumm$L1, gsub("_", "", Cdata$TumorID))]
MHmelt$Treatment=factor(Cdata$Treatment[match(MHmelt$L1, gsub("_", "", Cdata$TumorID))])
MHmelt$Growth=factor(Cdata$Growth2[match(MHmelt$L1, gsub("_", "", Cdata$TumorID))])
MHmelt$Growth2=MHmelt$Growth
#MHmelt$Growth2[grep("no data", MHmelt$Growth2)]="no data"

A1=MHmelt[MHmelt$Var2=="CD8", ]
A1$label=A1$L1
A1$label[which(A1$gridsize!=500)]=NA

A2=MHmelt[MHmelt$Var2=="CD8" & MHmelt$gridsize==250 & MHmelt$Var1%in%c("EpCAM", "SMA"), ]
A3=MHmelt[MHmelt$Var2=="CD8" & MHmelt$gridsize==250, ]
A3$CD8frac=WSIvalFracs[ 1, match(A3$L1, colnames(WSIvals))]

MHTempSumm=A3 #MHmelt[which(MHmelt$gridsize==300 & MHmelt$Var2=="CD8"), ]
MHreshape=acast(A3[ ,c("Var1", "value", "L1")], L1~Var1, value.var="value" )
MHreshape=data.frame(MHreshape)
MHreshape$EpMIN=ifelse(MHreshape$EpCAM..SMA<MHreshape$EpCAM, MHreshape$EpCAM..SMA, MHreshape$EpCAM)
MHreshape$EpMIN[which(is.na(MHreshape$EpMIN))]=MHreshape$EpCAM[which(is.na(MHreshape$EpMIN))]
MHreshape$EpStrRatio1=MHreshape$EpCAM/rowSums(MHreshape[ ,c("EpCAM..SMA", "SMA")], na.rm=T)

MHreshape$EpStrRatio2=MHreshape$EpCAM/(MHreshape$SMA)
MHreshape$EpStrRatio3=rowSums(MHreshape[ ,c("EpCAM..SMA", "EpCAM")], na.rm=T)/rowSums(MHreshape[ ,c("EpCAM..SMA", "SMA")], na.rm=T)
MHreshape$EpStrRatio4=rowSums(MHreshape[ ,c("EpCAM..SMA", "EpCAM")], na.rm=T)/(MHreshape$SMA)
MHreshape$Treatment=factor(A3$Treatment[match(rownames(MHreshape), A3$L1)])
MHreshape$Growth=factor(A3$Growth[match(rownames(MHreshape), A3$L1)])
MHreshape$Infil=factor(A3$SpatialManual[match(rownames(MHreshape), A3$L1)])

#pdf(sprintf("rslt/WSI-analysis/MHplots_compare_spatial_manual_%s.pdf", Sys.Date()), height=7, width=12)

p<-ggplot(A1, aes(x=gridsize, y=value, col=SpatialManual, label=label))+facet_grid(~Var1)+geom_line(aes(group=L1))+xlab("grid size")+ylab("Morisita Horn index")+ggtitle("MH index with CD8: spatial manual")+geom_label()
print(p)

ggplot(A1, aes(x=factor(gridsize), y=value, col=SpatialManual, label=label))+facet_grid(~Var1)+geom_boxplot()+xlab("grid size")+ylab("Morisita Horn index")+ggtitle("MH index with CD8: spatial manual")

ggplot(MHmeltsumm, aes(x=factor(gridsize), y=value, col=SpatialManual))+facet_wrap(~variable, scale="free_y")+geom_boxplot()+xlab("grid size")+ylab("Morisita Horn index")+ggtitle("expected number of cells in each grid size")+scale_y_log10()
```


With increasing grid size, optimal differences between infiltrating and restricted appear at the following sizes:

* epCAML 100um+
* epcam:SMA most significant at 350+
* SMA: 150+
* Unclass:200+

We probably want to use a metric/gridsize of 250 um as the expected/mean number of cells in each grid is 10 here.
Other notes:

* double positive cells (EpCAM+SMA+) appear in predominantly the restricted cases?
* higher SMA- stromal cells in restricted


Below, we check whether these metrics can differentiate between infiltrating and restricted tumors:

```{r, fig.height=6}
# a1=t.test(A2$value[A2$SpatialManual=="Infiltrating" & A2$Var1=="EpCAM"], A2$value[A2$SpatialManual=="Infiltrating" & A2$Var1=="SMA"], paired = T)
# a2=t.test(A2$value[which(A2$SpatialManual=="restricted" & A2$Var1=="EpCAM" & !A2$L1%in%c("15RD", "5LB"))], A2$value[which(A2$SpatialManual=="restricted" & A2$Var1=="SMA")], paired = T)
# p<-ggplot(A2, aes(x=Var1, y=value, col=SpatialManual))+facet_grid(~SpatialManual)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index @ 300um")+ggtitle(sprintf("MH index with CD8: Infiltrating p=%s, restricted p=%s", round(a1$p.value, 2), round(a2$p.value, 2)))+geom_errorbar(aes(ymin=MH.lower, ymax=MH.upper), width=.2)+geom_point()
# print(p)
# par(mfrow=c(2,2))
# for (i in unique(A3$Var1)){
# a1=cor(A3$value[A3$Var1==i], A3$CD8frac[A3$Var1==i], use="complete")
# plot(A3$value[A3$Var1==i], A3$CD8frac[A3$Var1==i], xlab="MH-index", ylab="CD8 fraction", main=sprintf("%s, cor=%s", i, round(a1,2)), col=A3$Growth2, pch=19)
# }
# mtext("association bw CD8frac and MH scores", 3, -2, outer = T)


par(mfrow=c(3,3))
for (i in 1:9){
  a1=wilcox.test(MHreshape[ ,i]~MHreshape$Infil)
  boxplot(MHreshape[ ,i]~MHreshape$Infil, main=paste(colnames(MHreshape)[i]," p=", round(a1$p.value,2), sep="" ), xlab="MH: knn3, 15um", ylab=colnames(MHreshape)[i])
}


#devoff()


#pdf(sprintf("rslt/WSI-analysis/MHplots_spatial_growth_treatment_zoom_300um_%s.pdf", Sys.Date()), height=7, width=12)
```

The above analysis shows that the MH-index shows differences in mixing between CD8 cells and mixing with most cell types. However MH-CD8-to-Epcam to MH-CD8-to-stroma ratios do not support this difference.

### Growth

Below are the MH indices with increasing grid-size for individual samples. In general, there is a subset of stable samples which have very high intermixing

```{r}

p<-ggplot(A1, aes(x=gridsize, y=value, col=Growth2))+facet_grid(~Var1)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index")+ggtitle("MH index with CD8: growth")
print(p)

pvals=sapply(unique(A3$Var1), function(x) wilcox.test(A3$value[A3$Var1==x & A3$Growth2=="growing"], 
                                                      A3$value[A3$Var1==x & A3$Growth2=="stable"])$p.value)

ann_text=data.frame(label=round(pvals,2), Var1=unique(A3$Var1), Growth2="growing")

ggplot(A3, aes(x=Growth2, y=value, col=Growth2))+facet_grid(~Var1)+geom_boxplot()+ylab("grid size")+ylab("Morisita Horn index")+ggtitle("MH index with CD8: growth")+geom_text(data=ann_text, mapping = aes(x=2, y=0.75, label=label))
```

Although the MH values for growing vs stable are not different, we can compare the mixing in epcam vs stroma in matched samples:

```{r}
A2t=A2[-which(A2$L1%in%c("15RD", "5LB")), ]

 a1=t.test(A2t$value[A2t$Growth=="stable" & A2t$Var1=="EpCAM"], A2t$value[A2t$Growth=="stable" & A2t$Var1=="SMA"], paired=T)
 a2=t.test(A2t$value[A2t$Growth=="growing" & A2t$Var1=="EpCAM"], A2t$value[A2t$Growth=="growing" & A2t$Var1=="SMA"], paired=T)
# 
ggplot(A2, aes(x=Var1, y=value, col=Growth2))+facet_grid(~Growth2)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index @ 250um")+ggtitle(sprintf("MH index with CD8: stable p=%s, growing p=%s", round(a1$p.value, 2), round(a2$p.value, 2)))+geom_errorbar(aes(ymin=MH.lower, ymax=MH.upper), width=.2)+geom_point()
# print(p)
```

### Treatment

```{r}

pval2=matrix(NA, nrow=3, ncol=4)
colnames(pval2)=unique(A3$Var1)
rownames(pval2)=TreatV[1:3]

for (i in 1:3){
pval2[i, ]=sapply(unique(A3$Var1), function(x) wilcox.test(A3$value[A3$Var1==x & A3$Treatment==TreatV[i]], A3$value[A3$Var1==x & A3$Treatment=="Vehicle"])$p.value)
}

pmelt=melt(pval2)
colnames(pmelt)=c("Treatment", "Var1", "label")
pmelt$label=round(pmelt$label, 2)
pmelt$value=0.8

p<-ggplot(A1, aes(x=gridsize, y=value, col=Treatment))+facet_grid(~Var1)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index")+ggtitle("MH index with CD8: Treatment")
print(p)

ggplot(A3, aes(x=Treatment, y=value, col=Treatment))+facet_grid(~Var1)+geom_boxplot()+ylab("grid size")+ylab("Morisita Horn index")+ggtitle("MH index with CD8: Treatment")+geom_text(data=pmelt, mapping=aes(x=Treatment, y=0.8, label=label))
```

There is no difference between the different spatial metrics compared to the vehicle, however, we can compare for a given treatment if there is a difference between the epcam and the stromal interaction scores. It appears that there is a difference only in the control, where SMA mixing is higher than EPcam mixing:

```{r}
## pairwise comparison: Epcam vs stroma

## calculate the pairwise p values here
A2t=A2[-which(A2$L1%in%c("15RD", "5LB")), ]

a1=t.test(A2t$value[A2t$Treatment=="Vehicle" & A2t$Var1=="EpCAM"], A2t$value[A2t$Treatment=="Vehicle"& A2t$Var1=="SMA"], paired=T)
a2=t.test(A2t$value[A2t$Treatment=="PDL1" & A2t$Var1=="EpCAM"], A2t$value[A2t$Treatment=="PDL1" & A2t$Var1=="SMA"], paired=T)
a3=t.test(A2t$value[A2t$Treatment=="PDL1+LY" & A2t$Var1=="EpCAM"], A2t$value[A2t$Treatment=="PDL1+LY" & A2t$Var1=="SMA"], paired=T)
a4=t.test(A2t$value[A2t$Treatment=="LY" & A2t$Var1=="EpCAM"], A2t$value[A2t$Treatment=="LY" & A2t$Var1=="SMA"], paired=T)

p<-ggplot(A2, aes(x=Var1, y=value, col=Treatment))+facet_grid(~Treatment)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index @ 250um")+ggtitle(sprintf("MH index with CD8: Cntl p=%s, PDL1 p=%s LY p=%s P+L p=%s", round(a1$p.value, 2), round(a2$p.value, 2), round(a3$p.value,2), round(a4$p.value, 2)))+geom_errorbar(aes(ymin=MH.lower, ymax=MH.upper), width=.2)+geom_point()
print(p)

#devoff()

```



```{r,eval=F}


#pdf(sprintf("rslt/WSI-analysis/MH_gridsize300_epstromal_ratios_%s.pdf", Sys.Date()), height=7, width=12)
par(mfrow=c(1,2))
hist(log2(MHreshape$EpStrRatio1), main="ratio Epcam:any SMA")
hist(log2(MHreshape$EpStrRatio2), main="ratio Epcam:SMA", breaks=30)
a1=wilcox.test(MHreshape$EpStrRatio1~MHreshape$Infil)
ggplot(MHreshape, aes(x=Infil,y=log2(EpStrRatio1), col=Growth))+geom_boxplot()+ggtitle(sprintf("ratio Epcam:any SMA p = %s", round(a1$p.value, 2)))
ggplot(MHreshape, aes(x=Infil,y=log2(EpStrRatio1), col=Growth))+geom_point()+ggtitle(sprintf("ratio Epcam:any SMA p = %s", round(a1$p.value, 2)))
a1=wilcox.test(MHreshape$EpStrRatio2~MHreshape$Infil)
ggplot(MHreshape, aes(x=Infil,y=log2(EpStrRatio2), col=Growth))+geom_boxplot()+ggtitle(sprintf("ratio Epcam:any SMA p = %s", round(a1$p.value, 2)))
ggplot(MHreshape, aes(x=Infil,y=log2(EpStrRatio2), col=Growth))+geom_point()+ggtitle(sprintf("ratio Epcam:any SMA p = %s", round(a1$p.value, 2)))
plot(log2(MHreshape$EpStrRatio2), log2(MHreshape$EpStrRatio1), col=MHreshape$Infil, xlab="any SMA", ylab="ep:sma ratio")
#devoff()

lx1=match(rownames(knnreshape), gsub("_", "", Cdata$TumorID))
Cdata$MHEpstrRatio=NA
Cdata$MHEpstrRatio[lx1]=knnreshape$EpStrRatio1
Cdata$MHEpAntstrRatio=NA
Cdata$MHEpAntstrRatio[lx1]=knnreshape$EpStrRatio2
Cdata$MHEpcam=NA
Cdata$MHEpcam[lx1]=knnreshape$EpCAM
Cdata$MHSMA=NA
Cdata$MHSMA[lx1]=knnreshape$SMA
```

## Comparison between metrics

After assessing optimal parameters for each metric, in this section we assess which metric could be the best for spatial analysis.

Below is a table of the different metrics and their values for each sample

```{r}
write.xlsx(Cdata, file=sprintf("../metadata/Carlos_samples_list_master_updated_%s.xlsx", Sys.Date()))
#write.table(SummaryData, file="../metadata/WSI_compared_other_metrics.csv")

##
# Combine the data from above into one file
#
df.Spatial=cbind(knnreshape[ , 1:9], IFreshape[ ,1:10], MHreshape[ , 1:9] )
colnames(df.Spatial)=paste(rep(c("knn", "IF", "MH"), times=c(9, 10, 9)), colnames(df.Spatial),sep=".")

df.Spatial=cbind(df.Spatial, knnreshape[ ,10:13])
df.Spatial$GrowthRate=Cdata$GrowthRate[match(rownames(df.Spatial), gsub("_", "", Cdata$TumorID))]
df.Spatial$TumSize=Cdata$Tumor.dia..sac..mm.[match(rownames(df.Spatial), gsub("_", "", Cdata$TumorID))]
write.csv(df.Spatial, file="../data/WSI-data/summary_scores_spatial.csv")
df.Spatial$knn.EpCAMcut=cut(df.Spatial$knn.EpCAM, c(-1, median(df.Spatial$knn.EpCAM, na.rm = T), 1.2), c("low", "high"))
df.Spatial$MH.EpCAMcut=cut(df.Spatial$MH.EpCAM, c(-1, median(df.Spatial$MH.EpCAM, na.rm = T), 1.2), c("low", "high"))
df.Spatial$IF.EpCAMcut=cut(df.Spatial$IF.EpCAM, c(-1, median(df.Spatial$IF.EpCAM, na.rm = T), 1.2), c("low", "high"))
df.Spatial$CD8Fraccut=cut(df.Spatial$CD8frac, c(-1, median(df.Spatial$CD8frac, na.rm = T), 1.2), c("low", "high"))


df.Spatial[which(df.Spatial=="Inf", arr.ind = T)]=NA

testSig1=sapply(c(1:28, 32), function(x) wilcox.test(df.Spatial[ ,x] ~ df.Spatial$Growth)$p.value)
knnreshapeB=knnreshape
knnreshapeB[which(knnreshapeB=="Inf", arr.ind=T)]=NA
testSig1=sapply(c(1:28, 32), function(x) wilcox.test(df.Spatial[ ,x] ~ knnreshapeB$Infil)$p.value)
testSig2=sapply(c(1:28, 32), function(x) cor.test(df.Spatial[ ,x] , df.Spatial$GrowthRate, use="complete", method="spearman")$p.value)
testSig3=sapply(c(1:28, 32), function(x) cor.test(df.Spatial[ ,x] , df.Spatial$TumSize, use="complete", method="spearman")$p.value)

#datatable(df.Spatial[ , c("knn.EpCAM", "IF.EpCAM", "MH.EpCAM", "Treatment", "Growth", #"Infil", "CD8frac", "TumSize")], options = list(
#  searching = T,
#  pageLength = 5,
#  lengthMenu = c(5, 10, 15, 20)
#))


scroll_box(kable(df.Spatial[ , c("knn.EpCAM", "IF.EpCAM", "MH.EpCAM", "Treatment", "Growth", "Infil", "CD8frac", "TumSize")], format="html"),
         height="300px", width="100%")

```

```{r}
df.Spatial=read.csv("../data/WSI-data/summary_scores_spatial.csv", row.names = 1)
```

Firstly, we can compare the different metrics to determine how similar or different they are:

```{r, fig.height=6}
#pdf(sprintf("rslt/WSI-analysis/similarity_between_spatial_methods_%s.pdf", Sys.Date()), height=7, width=12)

par(mfrow=c(2,2))
a1=cor.test(df.Spatial$knn.EpCAM, df.Spatial$MH.EpCAM)
plot(df.Spatial$knn.EpCAM, df.Spatial$MH.EpCAM, xlab="knn", ylab="MH",
     main=sprintf("cor:%s, p:%s", round(a1$estimate,2), round(a1$p.value, 2)))
a1=cor.test(df.Spatial$IF.EpCAM, df.Spatial$MH.EpCAM)
plot(df.Spatial$MH.EpCAM, df.Spatial$IF.EpCAM,  xlab="MH", ylab="IF",
      main=sprintf("cor:%s, p:%s", round(a1$estimate,2), round(a1$p.value, 2)))
a1=cor.test(df.Spatial$knn.EpCAM, df.Spatial$MH.EpCAM)
plot( df.Spatial$IF.EpCAM,df.Spatial$knn.EpCAM,  xlab="IF", ylab="knn",
       main=sprintf("cor:%s, p:%s", round(a1$estimate,2), round(a1$p.value, 2)))

#devoff()
```

Do any of these metrics associate with growth or treatments?

```{r, fig.height=6}
par(mfrow=c(3,3))

cNames=c("knn.EpCAM", "IF.EpCAM", "MH.EpCAM")

for (i in cNames){
t1=wilcox.test(df.Spatial[, i]~df.Spatial$Growth)$p.value
boxplot(df.Spatial[, i]~df.Spatial$Growth, ylab=i, xlab="Growth", main=sprintf("p:%s", round(t1,2)))

t2=wilcox.test(df.Spatial[df.Spatial$Treatment%in%c("PDL1+LY", "Vehicle"), i]~df.Spatial$Treatment[df.Spatial$Treatment%in%c("PDL1+LY", "Vehicle")])$p.value
t1=wilcox.test(df.Spatial[df.Spatial$Treatment%in%c("PDL1", "Vehicle"), i]~df.Spatial$Treatment[df.Spatial$Treatment%in%c("PDL1", "Vehicle")])$p.value
t3=wilcox.test(df.Spatial[df.Spatial$Treatment%in%c("LY", "Vehicle"), i]~df.Spatial$Treatment[df.Spatial$Treatment%in%c("LY", "Vehicle")])$p.value

boxplot(df.Spatial[, i]~df.Spatial$Treatment, ylab=i, xlab="Treatment",
         main=sprintf("LY:%s PDL1:%s, P+L:%s", round(t3, 2), round(t1, 2), round(t2, 2)))

t1=wilcox.test(df.Spatial[df.Spatial$Infil%in%c("Infiltrating", "restricted"), i]~df.Spatial$Infil[df.Spatial$Infil%in%c("Infiltrating", "restricted")])$p.value
boxplot(df.Spatial[, i]~df.Spatial$Infil, ylab=i, xlab="Infil", main=sprintf("p:%s", round(t1, 2)))
}
```
We can also compare this to raw tumor size or growth rate data as well:

```{r}
par(mfrow=c(2,3))

cNames=c("knn.EpCAM", "IF.EpCAM", "MH.EpCAM")

for (i in cNames){
t1=cor.test(df.Spatial[, i],df.Spatial$GrowthRate, use="complete")
plot(df.Spatial[, i]~df.Spatial$GrowthRate, ylab=i, xlab="Growth Rate", main=sprintf("cor:%s p:%s", round(t1$estimate, 2), round(t1$p.value,2)))

t1=cor.test(df.Spatial[ ,i],df.Spatial$TumSize, use="complete")
plot(df.Spatial[, i]~as.numeric(as.character(df.Spatial$TumSize)), ylab=i, xlab="Tumor size", main=sprintf("cor:%s p:%s", round(t1$estimate, 2), round(t1$p.value, 2)))
}

```


```{r, eval=F}
cNames=c("IFEpAntstrRatio", "MHEpAntstrRatio", "MHEpstrRatio", "IFEpstrRatio", "MHEpcam", "IFEpcam", "MHSMA", "IFSMA",
         "knnSMADist", "knnEpDist", "knnEpAntstrRatio", "knnEpstrRatio")

ttestTable=rep(NA, length(cNames))

#pdf(sprintf("rslt/WSI-analysis/compare_Spatial_Manual_vs_Metric_%s.pdf", Sys.Date()), height=7, width=8)

par(mfrow=c(3,4))
for (i in 1:length(cNames)){
  xa=which(is.na(Cdata[ ,cNames[i]]) | is.infinite(Cdata[ ,cNames[i]]) | is.nan(Cdata[ ,cNames[i]]))
  if (length(xa)>0){
  ttestTable[i]=t.test( Cdata[ -xa,cNames[i]]~Cdata$InfiltratingVsRestricted[-xa])$p.value
  } else {
    ttestTable[i]=t.test( Cdata[ ,cNames[i]]~Cdata$InfiltratingVsRestricted)$p.value
  }
  boxplot(Cdata[ ,cNames[i]]~Cdata$InfiltratingVsRestricted, main=sprintf("%s p:%s", cNames[i], round(ttestTable[i],2)))
}

#devoff()

```


```{r,eval=F}
Cdata$IFEpstrRatio[which(Cdata$IFEpstrRatio=="Inf")]=NA
Cdata$MHEpstrRatio[which(Cdata$MHEpstrRatio=="Inf")]=NA
Cdata$MHEpstrRatio[which(Cdata$MHEpstrRatio=="Inf")]=NA

Cdata$IFEpAntstrRatio[which(Cdata$IFEpAntstrRatio=="Inf")]=NA
Cdata$MHEpAntstrRatio[which(Cdata$MHEpAntstrRatio=="Inf")]=NA
Cdata$MHEpAntstrRatio[which(Cdata$MHEpAntstrRatio=="Inf")]=NA

par(mfrow=c(2,3))

#pdf(sprintf("rslt/WSI-analysis/Compare_knn_IF_MH_index_%s.pdf", Sys.Date()), height=7, width=12)
a1=cor.test(log2(Cdata$IFEpstrRatio+0.01), log2(Cdata$MHEpstrRatio+0.01), use="complete", na.rm=T)
plot(log2(Cdata$IFEpstrRatio), log2(Cdata$MHEpstrRatio), col=Cdata$InfiltratingVsRestricted, xlab="IF", ylab="MH",ylim=c(-3, 3),
      main=sprintf("ep to SMA fraction cor=%s p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))
a1=cor.test(log2(Cdata$IFEpstrRatio+0.01), log2(Cdata$knnEpstrRatio), use="complete", na.rm=T)
plot(log2(Cdata$IFEpstrRatio), log2(Cdata$knnEpstrRatio), col=Cdata$InfiltratingVsRestricted, xlab="IF", ylab="knn", 
      main=sprintf("ep to SMA fraction cor=%s p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))
a1=cor.test(log2(Cdata$IFEpstrRatio+0.01), log2(Cdata$knnEpstrRatio), use="complete", na.rm=T)
plot(log2(Cdata$MHEpstrRatio), log2(Cdata$knnEpstrRatio), col=Cdata$InfiltratingVsRestricted, xlab="MH", ylab="knn", xlim=c(-4, 4),
      main=sprintf("ep to SMA fraction cor=%s p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))

a1=cor.test(log2(Cdata$IFEpAntstrRatio+0.01), log2(Cdata$MHEpAntstrRatio+0.01), use="complete", na.rm=T)
plot(log2(Cdata$IFEpAntstrRatio), log2(Cdata$MHEpAntstrRatio), col=Cdata$InfiltratingVsRestricted, xlab="IF", ylab="MH",ylim=c(-3, 3),      main=sprintf("ep to SMA fraction cor=%s p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))
a1=cor.test(log2(Cdata$IFEpAntstrRatio+0.01), log2(Cdata$knnEpAntstrRatio), use="complete", na.rm=T)
plot(log2(Cdata$IFEpAntstrRatio), log2(Cdata$knnEpAntstrRatio), col=Cdata$InfiltratingVsRestricted, xlab="IF", ylab="knn", 
      main=sprintf("ep to SMA fraction cor=%s p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))
a1=cor.test(log2(Cdata$IFEpAntstrRatio+0.01), log2(Cdata$knnEpAntstrRatio), use="complete", na.rm=T)
plot(log2(Cdata$MHEpAntstrRatio), log2(Cdata$knnEpAntstrRatio), col=Cdata$InfiltratingVsRestricted, xlab="MH", ylab="knn", xlim=c(-4, 4),     main=sprintf("ep to SMA fraction cor=%s p=%s", round(a1$estimate, 2), round(a1$p.value, 2)))

#devoff()
```


## Other cell types

We noted that the proportion of Unclassified cells seemed to be different between the treatments. Assess here whether the MH index for this cell type is associated with growth or treatment here:

```{r}
varsearch="Unclass"

#MHmeltsumm=melt(WSIMHsetup, id.vars=c("gridsize", "Ntiles"))

MHmelt=melt(WSIMH, measure.vars="MH.mean")
MHmelt$SpatialManual=Cdata$InfiltratingVsRestricted[match(MHmelt$L1, gsub("_", "", Cdata$TumorID))]
#MHmeltsumm$SpatialManual=Cdata$InfiltratingVsRestricted[match(MHmeltsumm$L1, gsub("_", "", Cdata$TumorID))]
MHmelt$Treatment=Cdata$Treatment[match(MHmelt$L1, gsub("_", "", Cdata$TumorID))]
MHmelt$Growth2=Cdata$Growth2[match(MHmelt$L1, gsub("_", "", Cdata$TumorID))]
#MHmelt$Growth2=Cdata$Tumor.growth.status[match(MHmelt$L1, gsub("_", "", Cdata$TumorID))]
#MHmelt$Growth2[grep("no data", MHmelt$Growth2)]="no data"

A1=MHmelt[MHmelt$Var2==varsearch| MHmelt$Var1==varsearch, ]
A1$label=A1$L1
A1$label[which(A1$gridsize!=500)]=NA
A1$Var1=ifelse(A1$Var1==varsearch, as.character(A1$Var2), as.character(A1$Var1))

A2=MHmelt[(MHmelt$Var2==varsearch|MHmelt$Var1==varsearch) & MHmelt$gridsize==250 & MHmelt$Var1%in%c("EpCAM", "SMA"), ]
A3=MHmelt[(MHmelt$Var2==varsearch|MHmelt$Var1==varsearch) & MHmelt$gridsize==250, ]
A3$CD8frac=WSIvalFracs[ 1, match(A3$L1, colnames(WSIvals))]

A3$Var1=ifelse(A3$Var1==varsearch, as.character(A3$Var2), as.character(A3$Var1))

MHTempSumm=A3 #MHmelt[which(MHmelt$gridsize==300 & MHmelt$Var2==varsearch), ]
MHreshape=acast(A3[ ,c("Var1", "value", "L1")], L1~Var1, value.var="value" )
MHreshape=data.frame(MHreshape)

MHreshape$Treatment=A3$Treatment[match(rownames(MHreshape), A3$L1)]
MHreshape$Growth=A3$Growth2[match(rownames(MHreshape), A3$L1)]
MHreshape$Infil=A3$SpatialManual[match(rownames(MHreshape), A3$L1)]

colnames(MHreshape)=paste(varsearch, colnames(MHreshape), sep=".")
write.csv(MHreshape, file=sprintf("outputs/%s_MH_comparisons_gridsize250.csv", varsearch))

# pdf(sprintf("rslt/WSI-analysis/MHplots_spatial_growth_treatment_%s_celltype_%s.pdf", varsearch, Sys.Date()), height=7, width=12)
# 
 p<-ggplot(A1, aes(x=gridsize, y=value, col=Growth2))+facet_grid(~Var1)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index")+ggtitle(sprintf("MH index with %s: growth", varsearch))
 print(p)
 
pvals=sapply(unique(A3$Var1), function(x) wilcox.test(A3$value[A3$Var1==x & A3$Growth2=="growing"],A3$value[A3$Var1==x & A3$Growth2=="stable"])$p.value)

ann_text=data.frame(Growth2="stable", y=0.8, label=round(pvals,2), Var1=unique(A3$Var1))
 
ggplot(A3, aes(x=Growth2, y=value, col=Growth2))+facet_grid(~Var1)+geom_boxplot()+ylab("grid size")+ylab("Morisita Horn index")+ggtitle(sprintf("MH index with %s: growth", varsearch))+
  geom_text(data=ann_text, mapping = aes(x=Growth2 , y=0.75, label=label))
 

pval2=matrix(NA, nrow=3, ncol=4)
colnames(pval2)=unique(A3$Var1)
rownames(pval2)=TreatV[1:3]

for (i in 1:3){
pval2[i, ]=sapply(unique(A3$Var1), function(x) wilcox.test(A3$value[A3$Var1==x & A3$Treatment==TreatV[i]], A3$value[A3$Var1==x & A3$Treatment=="Vehicle"])$p.value)
}

pmelt=melt(pval2)
colnames(pmelt)=c("Treatment", "Var1", "label")
pmelt$label=round(pmelt$label, 2)
pmelt$value=0.8

p<-ggplot(A1, aes(x=gridsize, y=value, col=Treatment))+facet_grid(~Var1)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index")+ggtitle(sprintf("MH index with %s: growth", varsearch))
print(p)

ggplot(A3, aes(x=Treatment, y=value, col=Treatment))+facet_grid(~Var1)+geom_boxplot()+ylab("grid size")+ylab("Morisita Horn index")+ggtitle(sprintf("MH index with %s: growth", varsearch))+geom_text(data=pmelt, mapping=aes(x=Treatment, y=0.8, label=label))


#  p<-ggplot(A2, aes(x=Var1, y=value, col=Growth2))+facet_grid(~Growth2)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index @ 250um")+ggtitle(sprintf("MH index with CD8: stable p=%s, growing p=%s", round(a1$p.value, 2), round(a2$p.value, 2)))+geom_errorbar(aes(ymin=MH.lower, ymax=MH.upper), width=.2)+geom_point()
# # print(p)
# 
#  p<-ggplot(A1, aes(x=gridsize, y=value, col=Treatment))+facet_grid(~Var1)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index")+ggtitle("MH index with CD8: Treatment")
#  print(p)
# 
# ggplot(A3, aes(x=Treatment, y=value, col=Growth2))+facet_grid(~Var1)+geom_boxplot()+ylab("grid size")+ylab("Morisita Horn index")+ggtitle("MH index with CD8: Treatment")
# 
# ## pairwise comparison: Epcam vs stroma
# 
# ## calculate the pairwise p values here
# 
# a1=t.test(A2t$value[A2t$Treatment=="Vehicle" & A2t$Var1=="EpCAM"], A2t$value[A2t$Treatment=="Vehicle"& A2t$Var1=="SMA"], paired=T)
# a2=t.test(A2t$value[A2t$Treatment=="PDL1" & A2t$Var1=="EpCAM"], A2t$value[A2t$Treatment=="PDL1" & A2t$Var1=="SMA"], paired=T)
# a3=t.test(A2t$value[A2t$Treatment=="PDL1+LY" & A2t$Var1=="EpCAM"], A2t$value[A2t$Treatment=="PDL1+LY" & A2t$Var1=="SMA"], paired=T)
# a4=t.test(A2t$value[A2t$Treatment=="LY" & A2t$Var1=="EpCAM"], A2t$value[A2t$Treatment=="LY" & A2t$Var1=="SMA"], paired=T)
# 
# p<-ggplot(A2, aes(x=Var1, y=value, col=Treatment))+facet_grid(~Treatment)+geom_line(aes(group=L1))+ylab("grid size")+ylab("Morisita Horn index @ 250um")+ggtitle(sprintf("MH index with CD8: Cntl p=%s, PDL1 p=%s LY p=%s P+L p=%s", round(a1$p.value, 2), round(a2$p.value, 2), round(a3$p.value,2), round(a4$p.value, 2)))+geom_errorbar(aes(ymin=MH.lower, ymax=MH.upper), width=.2)+geom_point()
# print(p)

#dev.off()
```



```{r, eval=F}
#This is epcam- sma- stroma and based on images is restricted:

CelltypeStudy="EpCAM"

knnTemp1=knnMelt[knnMelt$CellType==CelltypeStudy & knnMelt$NearestCellType%in%c("CD8","EpCAM", "SMA", "EpCAM: SMA", "Unclass"), ]
knnTemp1$SpatialManual=Cdata$InfiltratingVsRestricted[match(knnTemp1$L1, gsub("_", "", Cdata$TumorID))]

# compute p values?
ptest=sapply(levels(knnTemp1$knn), function(x) wilcox.test(knnTemp1$MeanDistance[which(knnTemp1$knn==x & knnTemp1$NearestCellType=="EpCAM" & knnTemp1$SpatialManual=="Infiltrating")], knnTemp1$MeanDistance[which(knnTemp1$knn==x & knnTemp1$NearestCellType=="SMA" & knnTemp1$SpatialManual=="Infiltrating")])$p.value)

fit1=aov(MeanDistance~NearestCellType+knn, data=knnTemp1[knnTemp1$SpatialManual=="Infiltrating", ])
summary(fit1)
fit2=aov(MeanDistance~NearestCellType+knn, data=knnTemp1[knnTemp1$SpatialManual=="restricted", ])
summary(fit2)

fit1b=aov(MedianDistance~NearestCellType+knn, data=knnTemp1[knnTemp1$SpatialManual=="Infiltrating", ])
summary(fit1b)
fit2b=aov(MedianDistance~NearestCellType+knn, data=knnTemp1[knnTemp1$SpatialManual=="restricted", ])
summary(fit2b)

#pdf(sprintf("rslt/WSI-analysis/Unclass_cells_knn_distances_vs_manualspatial_%s.pdf", Sys.Date()), height=7, width=12)

p<-ggplot(knnTemp1, aes(x=NearestCellType, y=MeanDistance, fill=SpatialManual))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=SpatialManual))+facet_grid(~knn+SpatialManual)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c", "#377eb8"),na.value="black")

print(p)
par(mfrow=c(1,2), oma=c(0, 0, 2, 0))
plot(TukeyHSD(fit1, "NearestCellType"))
title("Infiltrating Mean Distance", line=3.3)
plot(TukeyHSD(fit2, "NearestCellType"))
title("Restricted Mean Distance", line=3.3)

p<-ggplot(knnTemp1, aes(x=NearestCellType, y=MedianDistance, fill=SpatialManual))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=SpatialManual))+facet_grid(~knn+SpatialManual)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c", "#377eb8"),na.value="black")
print(p)

par(mfrow=c(1,2), oma=c(0, 0, 2, 0))
plot(TukeyHSD(fit1b, "NearestCellType"))
title("Infiltrating Median Distance", line=3.3)
plot(TukeyHSD(fit2b, "NearestCellType"))
title("Restricted Median Distance", line=3.3)

knnTemp1$Treatment=Cdata$Treatment[match(knnTemp1$L1, gsub("_", "", Cdata$TumorID))]

fit1=aov(MeanDistance~Treatment+knn, data=knnTemp1[knnTemp1$NearestCellType=="EpCAM", ])
fit2=aov(MeanDistance~Treatment+knn, data=knnTemp1[knnTemp1$NearestCellType=="SMA", ])
fit3=aov(MeanDistance~Treatment+knn, data=knnTemp1[knnTemp1$NearestCellType=="EpCAM: SMA", ])

knnTemp1$Growth=Cdata$Growth2[match(knnTemp1$L1, gsub("_", "", Cdata$TumorID))]

fit1b=aov(MeanDistance~Growth+knn, data=knnTemp1[knnTemp1$NearestCellType=="EpCAM", ])
fit2b=aov(MeanDistance~Growth+knn, data=knnTemp1[knnTemp1$NearestCellType=="SMA", ])
fit3b=aov(MeanDistance~Growth+knn, data=knnTemp1[knnTemp1$NearestCellType=="EpCAM: SMA", ])


p<-ggplot(knnTemp1, aes(x=NearestCellType, y=MeanDistance, fill=Treatment))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=Treatment))+facet_grid(~knn+Treatment)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c", "#4daf4a","#377eb8",  "#984ea3"),na.value="black")
print(p)

p<-ggplot(knnTemp1, aes(fill=Treatment, y=MeanDistance, x=Treatment))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=Treatment))+facet_grid(~knn+NearestCellType)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c", "#4daf4a","#377eb8",  "#984ea3"),na.value="black")
print(p)

par(mfrow=c(1,3), oma=c(0, 0, 2, 0))
plot(TukeyHSD(fit1, "Treatment"), las=2)
title("EpCAM Mean Distance", line=3.3)
plot(TukeyHSD(fit2, "Treatment"), las=2)
title("SMA Mean Distance", line=3.3)
plot(TukeyHSD(fit3, "Treatment"), las=2)
title("EpCAM:SMA Mean Distance", line=3.3)

p<-ggplot(knnTemp1, aes(x=NearestCellType, y=MeanDistance, fill=Growth))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=Growth))+facet_grid(~knn+Growth)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c","#ff7f00", "#4daf4a","#377eb8",  "#984ea3", "purple"),na.value="black")
print(p)

p<-ggplot(knnTemp1, aes(fill=Growth, y=MeanDistance, x=Growth))+geom_boxplot(outlier.shape=NA)+geom_jitter(aes(col=Growth))+facet_grid(~knn+NearestCellType)+scale_y_continuous(trans='log10')+theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_color_manual(values = c("#e41a1c","#ff7f00", "#4daf4a","#377eb8",  "#984ea3", "purple"),na.value="black")
print(p)

par(mfrow=c(1,3), oma=c(0, 0, 2, 0))
plot(TukeyHSD(fit1b, "Growth"), las=2)
title("EpCAM Mean Distance", line=3.3)
plot(TukeyHSD(fit2b, "Growth"), las=2)
title("SMA Mean Distance", line=3.3)
plot(TukeyHSD(fit3b, "Growth"), las=2)
title("EpCAM:SMA Mean Distance", line=3.3)

#dev.off()

```

<!--chapter:end:02b-wsi.Rmd-->

# Expression data

This file looks at loading and pre-processing data for:

* differential gene expression analysis
* PAM50 subtyping
* uploading into CIBERSORT/TIMER

## Running alignment

Samples were mapped in star using the following parameters. Note that the first two batches of samples run had shorter read lengths (~75 bp) whereas batch 3 had lengths of ~150bp

```{r, eval=F, echo=T}
## Not run here
STAR \
     --readFilesCommand zcat \
     --genomeDir /n/scratch2/at268/rn6_v2 \
     --sjdbGTFfile /n/scratch2/at268/rn6_v2/rn6.refGene.gtf \
     --runThreadN 10 \
     --runMode alignReads \
     --genomeLoad NoSharedMemory\
     --outSAMattributes	NH HI AS nM NM\
     --outSAMstrandField intronMotif\
     --outFilterMultimapNmax 20\
     --alignSJoverhangMin 8\
     --readFilesIn $1 $2 \
     --alignSJDBoverhangMin 1\
     --outFilterMismatchNmax 999\
     --outFilterMismatchNoverLmax 0.1\
     --alignIntronMin 20\
     --alignIntronMax 1000000\
     --alignMatesGapMax	1000000\
     --outFilterType BySJout\
     --outFilterScoreMinOverLread 0.33 \
     --outFilterMatchNminOverLread 0.33 \
     --limitSjdbInsertNsj 1200000 \
     --outFilterIntronMotifs None \
     --alignSoftClipAtReferenceEnds Yes\
     --outSAMattrRGline	ID:$4 SM:$4 \
     --chimSegmentMin 15 \
     --chimJunctionOverhangMin 15\
     --limitBAMsortRAM 0\
     --outSAMtype BAM SortedByCoordinate\
     --outSAMunmapped Within \
     --quantMode GeneCounts transcriptomeSAM \
     --quantTranscriptomeBan IndelSoftclipSingleend \
     --outFileNamePrefix $3 \
     --twopassMode Basic
```

In addition, RSEM was run to obtain TPM, rsem and FPKM counts.
Note: [rn6.refGene.gtf.gz](http://hgdownload.soe.ucsc.edu/goldenPath/rn6/bigZips/genes/rn6.ncbiRefSeq.gtf.gz) was used to generate the RSEM library!(This is the may version)

```{r}
# Upload infoTable

infoTable=read.csv("../metadata/AllRNA_samples_sept.csv")

####
# uncomment to update these parameters
####
infoTable$TumorID=paste(infoTable$Rat_ID, infoTable$Location, sep="_")
a1=match(infoTable$TumorID, Cdata$TumorID)
```

## RNA Initial QC


```{r rna-preprocess, cache=T}
BatchNo="april"

rsemFiles=dir("../data/RNA_expression/rsem/", ".results")
allrsem=matrix(NA, nrow=17455, ncol=length(rsemFiles)) #31038
allTPM=matrix(NA, nrow=17455, ncol=length(rsemFiles))
allFPKM=matrix(NA, nrow=17455, ncol=length(rsemFiles))
for (i in 1:length(rsemFiles)){
  a1=read.delim(file.path("../data/RNA_expression/rsem/", rsemFiles[i]))
  allrsem[ ,i]=a1$expected_count
  allTPM[ ,i]=a1$TPM
  allFPKM[ ,i]=a1$FPKM
}

cNames=unlist(strsplit(rsemFiles, ".genes.results"))
cNames=unlist(strsplit(cNames, "_0.33_v2"))
sAnnot=match(cNames, infoTable$starSampleName)
colnames(allrsem)=paste(infoTable$Rat_ID[sAnnot],infoTable$Location[sAnnot], infoTable$Fraction[sAnnot],  sep="_")
rownames(allrsem)=a1$gene_id

colnames(allTPM)=colnames(allrsem)
rownames(allTPM)=rownames(allrsem)

colnames(allFPKM)=colnames(allrsem)
rownames(allFPKM)=rownames(allrsem)

starFiles=dir("../data/RNA_expression/star_april/", ".tab")
allstar=matrix(NA, nrow=17455, ncol=length(starFiles))
allmapp=matrix(NA, nrow=4, ncol=length(starFiles))


for (i in 1:length(starFiles)){
  a1=read.delim(file.path("../data/RNA_expression/star_april/", starFiles[i]), header=F)
  allstar[ ,i]=a1[ -c(1:4),2]
  allmapp[ ,i]=a1[ c(1:4),2]
}

cNames=unlist(strsplit(starFiles, "ReadsPerGene.out.tab"))
cNames=unlist(strsplit(cNames, "_0.33_v2"))
sAnnot=match(cNames, infoTable$starSampleName)
colnames(allstar)=paste(infoTable$Rat_ID[sAnnot], infoTable$Location[sAnnot],infoTable$Fraction[sAnnot], sep="_")
rownames(allstar)=a1[-c(1:4) ,1]

colnames(allmapp)=colnames(allstar)
allmapp=rbind(allmapp, colSums(allstar))

rownames(allmapp)=c(as.character(a1[c(1:4), 1]), "UniqueReads")


## in all cases, remove the files 
id=match(cNames, infoTable$starSampleName)
infoTable=infoTable[ na.omit(id), ]
id2=which(is.na(id))

if (length(id2)>0){

allmapp=allmapp[ , -grep("NA_NA", colnames(allmapp))]

allTPM=allTPM[ , -id2]
allFPKM=allFPKM[ , -id2]
allrsem=allrsem[ , -id2]
allstar=allstar[, -id2]
}
```


Default output from R showing the number of unique reads compared to multimapped, unmapped etc. This is shown for each batch.
Note that batch 3 has differences (high percentage of unmapped) compared to the other batches, possibly due to DNA contamination.

Below we check for three measures:

- mapped million reads (ideally, 10M+ reads)
- Gene Sparsity: This is a measurement of the number of genes which have non-zero values. Ideally, would be greater than 10K, but values which are too high may also suggest contamination from DNA (unexpressed genes are also counted)
- Varability: standard deviation of the transcriptomic counts. If this value is too low, would suggest that high DNA contamination, non-representative transcriptome.


```{r}
# number of mapped reads
UnMappedNorm=t(allmapp)/colSums(allmapp)

mUnMap=melt(UnMappedNorm)
mUnMap$Batch=infoTable$Batch[match(mUnMap$Var1, infoTable$SampleID)]

# how many genes represented
Sparsity=colSums(sign(allstar))
# check how skewed the data is

cSDs=colSds(allstar)

TVals=data.frame(MappedReadsM=allmapp[5, ]/1E6, GeneSparsityK=Sparsity/1E3,Batch=infoTable$Batch, GeneVariabilityCounts=cSDs, Type=as.character(infoTable$Fraction), names=colnames(allstar))

mTV=melt(TVals, measure.vars = c("MappedReadsM", "GeneSparsityK", "GeneVariabilityCounts"))

#pdf(sprintf("../rslt/DESeq/GE_preprocessing_%s_%s.pdf", BatchNo,Sys.Date()), height=5, width=8)

ggplot(mUnMap, aes(x=Var1, y=value, fill=Var2))+geom_bar(stat="identity")+facet_grid(~Batch, space="free", scale="free")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ylab("proportion of reads")+ggtitle("mapping summary")

ggplot(mTV, aes(x=names, y=value,fill=Type))+geom_bar(stat="identity")+
  facet_grid(variable~Batch, space="free_x", scale="free")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ggtitle("#Mapped Reads, #Unique Gebes, #Variability")

par(mfrow=c(2,2))

plot(density(TVals$MappedReadsM), main="mapped reads")
x1=mean(TVals$MappedReadsM)
sdv=sd(TVals$MappedReadsM)
abline(v=c(x1, x1-1.5*sdv, x1+1.5*sdv), col="grey", lty=2)
text( c(x1, x1-1.5*sdv, x1+1.5*sdv), 0.07, c(x1, x1-2*sdv, x1+2*sdv), las=2, cex = 0.75, srt=90)

plot(density(TVals$GeneSparsityK), main="gene sparsity")
x1=mean(TVals$GeneSparsityK)
sdv=sd(TVals$GeneSparsityK)
abline(v=c(x1, x1-1.5*sdv, x1+1.5*sdv), col="grey", lty=2)
text( c(x1, x1-1.5*sdv, x1+1.5*sdv), 0.07, c(x1, x1-2*sdv, x1+2*sdv), las=2, cex = 0.75, srt=90)

plot(density(TVals$GeneVariabilityCounts), main="Gene Variability")
x1=mean(TVals$GeneVariabilityCounts)
sdv=sd(TVals$GeneVariabilityCounts)
abline(v=c(x1, x1-1.5*sdv, x1+1.5*sdv), col="grey", lty=2)
text( c(x1, x1-1.5*sdv, x1+1.5*sdv), 0.00015, c(x1, x1-2*sdv, x1+2*sdv), las=2, cex = 0.75, srt=90)


#dev.off()
```


Samples to remove from analysis:

The thresholds indicated below are based on the above density plots, and removes cases which are <1.5 SD of the mean

* low total number of mapped reads (under 1.5M)
* sparsity: less than 8K genes
* variability : threshold under 500


```{r}
rmSamples=which(TVals$MappedReadsM<1.5 | TVals$GeneSparsityK<8 | TVals$GeneVariabilityCounts<500)
allstarFinal=allstar[ ,-rmSamples]
allrsemFinal=allrsem[ ,-rmSamples]
allTPMFinal=allTPM[ ,-rmSamples]
allFPKMFinal=allFPKM[ ,-rmSamples]
infoTableFinal=infoTable[-rmSamples, ]
```

The omitted samples are:

```{r}
kable(TVals[rmSamples, ])
```


We are left with `r ncol(allstarFinal)` samples. 

There are `r table(infoTableFinal$Fraction)` samples in the `r unique(infoTableFinal$Fraction)` fractions.

There are `r table(infoTableFinal$Batch)` samples from batches 1, 2 and 3 respectively.

## Normalisation

Run through DESEq and normalise the library. Using all samples, we run the model:

expression ~ Celltype + factor (Batch)

and keep the genes which have a total count of at least half the number of samples. ie.
$ sum(gene_i) > N_{samples}/2 $


```{r deseq-1, cache=T, eval=T}
# remove rows where counts are low?
rownames(infoTableFinal)=infoTableFinal$SampleID
infoTableFinal$Batch=factor(infoTableFinal$Batch)
dds=DESeqDataSetFromMatrix(allstarFinal, infoTableFinal, design=~Fraction+factor(Batch)) ## change class
keep=rowSums(counts(dds))>(ncol(dds)/2)
dds=dds[keep, ]
dds=DESeq(dds)

vsd <- varianceStabilizingTransformation(dds)
normalizedTableVSD <- assay(vsd)

# infoTableFinal$Treatment=RNADNAsamples$Treatment[match(infoTableFinal$FqFile, RNADNAsamples$FqFile.CD45)]
# idx=match(infoTableFinal$FqFile, RNADNAsamples$FqFile.Ep)
# infoTableFinal$Treatment[-which(is.na(idx))]=RNADNAsamples$Treatment[na.omit(idx)]
# idx=match(infoTableFinal$FqFile, RNADNAsamples$FqFile.DN)
# infoTableFinal$Treatment[-which(is.na(idx))]=RNADNAsamples$Treatment[na.omit(idx)]

save(dds, vsd,allstarFinal, allrsemFinal,allTPMFinal, normalizedTableVSD,infoTableFinal,  file=sprintf("../dds_normalised_data_newstar_RNAseq%s_%s.RData", BatchNo, Sys.Date()))
```

### preliminary visualisation (to remove outliers)

Below are PCA plots based on:

* Batch
* CellType

```{r vsd-transform,eval=T, cache=T, fig.height=4}
vsd2 <- vst(dds)

#pdf(sprintf("../rslt/DESeq/PCA_preliminary_%s_%s.pdf",BatchNo, Sys.Date() ), width=8, height=6)

#All samples
plotPCA(vsd2, "Batch")+ggtitle("Batch")
plotPCA(vsd2, "Fraction")+ggtitle("Fraction")
plotPCA(vsd2, c("Fraction"))+geom_label(aes(label = name)) 

```

Batches in general separate out well, however, some samples appear to be outliers in comparison to the main group. We look in closer detail the CD45, DN and EpCAM populations.

In the CD45 population, narrow down to only immune related genes to see if there is a difference.

```{r}
# plotPCA(vsd2, c("Treatment"))+ggtitle("Treatment")
# plotPCA(vsd2, c("Growth"))+ggtitle("Growth")
# plotPCA(vsd2, "TumSize")+ggtitle("tumor size")
# plotPCA(vsd2, "CD8Frac")+ggtitle("CD8 Fraction")
# plotPCA(vsd2, "SpatialManual")+ggtitle("CD8 distribution")
# plotPCA(vsd2, "IFEpAntstrRatio")+ggtitle("CD8 distribution")
# plotPCA(vsd2, "MHSMA")+ggtitle("CD8 distribution")
# plotPCA(vsd2, "MHEpCAM")+ggtitle("CD8 distribution")


#Cd45 specific
# plotPCA(vsd2[, grep("CD45", colnames(vsd2))], c("Treatment"))+ggtitle("CD45 only Treatment")+geom_label(aes(label = name))
plotPCA(vsd2[rownames(assay(vsd2))%in%RatAllImm , grep("CD45", colnames(vsd2))], c("Growth"))+ggtitle("CD45 only Treatment: immune only genes")+geom_label(aes(label = name))
# plotPCA(vsd2[rownames(assay(vsd2))%in%RatAllImm  , grep("CD45", colnames(vsd2))], c("Growth"))+ggtitle("CD45 only Growth")
plotPCA(vsd2[ , grep("Ep", colnames(vsd2))], "Growth")+geom_label(aes(label = name))+ggtitle("Ep only Growth")
plotPCA(vsd2[ , grep("DN", colnames(vsd2))], "Growth")+geom_label(aes(label = name))+ggtitle("DN only Growth")

# vsdLimma=vsd2
# assay(vsdLimma)<- limma::removeBatchEffect(assay(vsdLimma), vsd$Batch)
# plotPCA(vsdLimma, c("Fraction", "Batch"))
# plotPCA(vsdLimma, c("Fraction"))
# plotPCA(vsdLimma, c("Batch"))
# plotPCA(vsdLimma, c("Growth", "Fraction"))

#dev.off()
```

Based on the above plots, we remove the following outliers and re-run the normalisation:

* 2R_D_DN
* 4L_B_CD45

```{r rm-ill-samples, cache=T}
rmThese=c("2R_D_DN", "4L_B_CD45")

## Note based on the above plots, sample "2R_D_DN" is misclassified as a cd45 sample. Need to remove this sample and re-run the preprocessing:
allstarFinal=allstarFinal[ ,-match(rmThese, colnames(allstarFinal))]
infoTableFinal=infoTableFinal[-match(rmThese, rownames(infoTableFinal)), ]
allrsemFinal=allrsemFinal[ ,-match(rmThese, colnames(allrsemFinal))]
allTPMFinal=allTPMFinal[ ,-match(rmThese, colnames(allTPMFinal))]

dds=DESeqDataSetFromMatrix(allstarFinal, infoTableFinal, design=~Fraction+factor(Batch)) ## change class
keep=rowSums(counts(dds))>(ncol(dds)/2)
dds=dds[keep, ]
dds=DESeq(dds)

vsd <- varianceStabilizingTransformation(dds)
normalizedTableVSD <- assay(vsd)

infoTableFinal$MHEpCAM=df.Spatial$MH.EpCAM[match(gsub("_", "", infoTableFinal$TumorID), rownames(df.Spatial))]
infoTableFinal$IFEpCAM=df.Spatial$IF.EpCAM[match(gsub("_", "", infoTableFinal$TumorID), rownames(df.Spatial))]
infoTableFinal$knnEpCAM=df.Spatial$knn.EpCAM[match(gsub("_", "", infoTableFinal$TumorID), rownames(df.Spatial))]
infoTableFinal$CD8Frac=df.Spatial$CD8frac[match(gsub("_", "", infoTableFinal$TumorID),rownames(df.Spatial))]

# infoTableFinal$Treatment=RNADNAsamples$Treatment[match(infoTableFinal$FqFile, RNADNAsamples$FqFile.CD45)]
# idx=match(infoTableFinal$FqFile, RNADNAsamples$FqFile.Ep)
# infoTableFinal$Treatment[-which(is.na(idx))]=RNADNAsamples$Treatment[na.omit(idx)]
# idx=match(infoTableFinal$FqFile, RNADNAsamples$FqFile.DN)
# infoTableFinal$Treatment[-which(is.na(idx))]=RNADNAsamples$Treatment[na.omit(idx)]

save(dds, vsd,allstarFinal, allrsemFinal,allTPMFinal, normalizedTableVSD,infoTableFinal,  file=sprintf("outputs/dds_normalised_data_newstar_RNAseq%s_%s_rm_outliers.RData", BatchNo, Sys.Date()))

```


## Processing files for external software

We also process these files for external software:

Software | RNA-data
---------|---------------
PAM50 | rsem data after quantile normalised, genes converted to human
TIMER | TPM values, retain rat ids
xcell | rsem data, genes converted ti human
cibersort | tpm?? convert to rat genes and use lm22 rat


```{r, cache=T}
#load("dds_normalised_data_newstar_RNAseqmay_2020-09-16.RData")

p50_annot=read.delim("../rscript/PAM50/bioclassifier_R/pam50_annotation.txt")

# remove counts
allRSEMrm=allrsemFinal[-which(rowSums(allrsemFinal)<ncol(allrsemFinal)),]

#allRSEMrm=allRSEMrm[ ,which(colnames(allRSEMrm) %in% infoTableFinal$SampleID[infoTableFinal$Cohort!="Progression"])]

# find the quantile: we can either
# 1. include all genes
data.quantileAll <- apply(allRSEMrm, 2, function(x){quantile(x, 0.75)});
# 2. only consider the expressed genes
data.quantileExpressed <-  apply(allRSEMrm, 2, function(x){quantile(x[x>0], 0.75)});

# normalise data
allRSEMnorm1 <- t(t(allRSEMrm) / data.quantileAll)
allRSEMnorm2 <- t(t(allRSEMrm) / data.quantileExpressed)

# compare the colsums of these samples after normalisation
a1=colSums(allRSEMnorm1)
b1=colSums(allRSEMnorm2)

# par(mfrow=c(1, 2))
# hist(a1, main="all genes")
# hist(b1, main="expressed genes")

# outliers
which(a1>5E5)
which(b1>5E4)

# log transform and set infinite values to NA

allRSEMnorm1log=log2(allRSEMnorm1)
allRSEMnorm2log=log2(allRSEMnorm2)

allRSEMnorm1log[which(allRSEMnorm1log=="-Inf", arr.ind=T)]=NA
allRSEMnorm2log[which(allRSEMnorm2log=="-Inf", arr.ind=T)]=NA

# Determine the number of NAs in each row

r1NA=sapply(1:nrow(allRSEMnorm1log), function(x) length(which(is.na(allRSEMnorm1log[x, ]))))
r2NA=sapply(1:nrow(allRSEMnorm2log), function(x) length(which(is.na(allRSEMnorm2log[x, ]))))

r1idx=which(r1NA > ncol(allRSEMnorm1log)*0.4)
r2idx=which(r2NA > ncol(allRSEMnorm2log)*0.4)

# match ids

lx1=match(p50_annot$GeneName, toupper(rownames(allRSEMnorm1log)))

r1idx=setdiff(r1idx, na.omit(lx1))
r2idx=setdiff(r2idx, na.omit(lx1))

# remove the genes with NAs

allRSEMnorm1log=allRSEMnorm1log[-r1idx, ]
allRSEMnorm2log=allRSEMnorm2log[-r2idx, ]

# change the names to human gene names
mmatcha=SymHum2Rat$HGNC.symbol[match(rownames(allRSEMnorm1log), SymHum2Rat$RGD.symbol)]
mmatchb=Rat2Hum$HGNC.symbol[match(rownames(allRSEMnorm1log), Rat2Hum$RGD.symbol)]
mmc=ifelse(is.na(mmatcha)==T, mmatchb, mmatcha)

mmatch2a=SymHum2Rat$HGNC.symbol[match(rownames(allRSEMnorm2log), SymHum2Rat$RGD.symbol)]
mmatch2b=Rat2Hum$HGNC.symbol[match(rownames(allRSEMnorm2log), Rat2Hum$RGD.symbol)]
mmc2=ifelse(is.na(mmatch2a)==T, mmatch2b, mmatch2a)

rownames(allRSEMnorm1log)=ifelse(is.na(mmc), rownames(allRSEMnorm1log), mmc)
rownames(allRSEMnorm2log)=ifelse(is.na(mmc2), rownames(allRSEMnorm2log), mmc2)

# save for non-CD45 samples by batch columns
write.table(allRSEMnorm1log[ ,which(colnames(allRSEMnorm1log)%in%infoTableFinal$SampleID[infoTableFinal$Cohort!="Progression" & infoTableFinal$Fraction!="CD45"])], sep="\t", file=sprintf("../output4external/RSEM_for_PAM50_allgenes_noCD45_characterisationOnly%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)
write.table(allRSEMnorm1log[ ,which(colnames(allRSEMnorm1log)%in%infoTableFinal$SampleID[infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction!="CD45"])], sep="\t", file=sprintf("../output4external/RSEM_for_PAM50_allgenes_noCD45_progression%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)
write.table(allRSEMnorm1log[ ,which(colnames(allRSEMnorm1log)%in%infoTableFinal$SampleID[infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction=="Ep"])], sep="\t", file=sprintf("../output4external/RSEM_for_PAM50_allgenes_epOnly_progression%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)
write.table(allRSEMnorm1log[ ,which(colnames(allRSEMnorm1log)%in%infoTableFinal$SampleID[infoTableFinal$Fraction!="CD45"])], sep="\t", file=sprintf("../output4external/RSEM_for_PAM50_allgenes_noCD45_%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)

write.table(allRSEMnorm2log[ ,which(colnames(allRSEMnorm2log)%in%infoTableFinal$SampleID[infoTableFinal$Cohort!="Progression" & infoTableFinal$Fraction!="CD45"])], sep="\t", file=sprintf("../output4external/RSEM_for_PAM50_allExpressedgenes_noCD45_characterisationOnly%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)
write.table(allRSEMnorm2log[ ,which(colnames(allRSEMnorm2log)%in%infoTableFinal$SampleID[infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction!="CD45"])], sep="\t", file=sprintf("../output4external/RSEM_for_PAM50_allExpressedgenes_noCD45_progression%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)
write.table(allRSEMnorm2log[ ,which(colnames(allRSEMnorm2log)%in%infoTableFinal$SampleID[infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction=="Ep"])], sep="\t", file=sprintf("../output4external/RSEM_for_PAM50_allExpressedgenes_epOnly_progression%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)
write.table(allRSEMnorm2log[ ,which(colnames(allRSEMnorm2log)%in%infoTableFinal$SampleID[infoTableFinal$Fraction!="CD45"])], sep="\t", file=sprintf("../output4external/RSEM_for_PAM50_allExpressedgenes_noCD45_%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)

# save specifically for EP samples
#write.table(allRSEMnorm1log[ ,grep("Ep", colnames(allRSEMnorm1log))], sep="\t", file=sprintf("../output4external/EPonly_RSEM_for_PAM50_allgenes_%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)
#write.table(allRSEMnorm2log[ ,grep("Ep", colnames(allRSEMnorm2log))], sep="\t", file=sprintf("../output4external/EPonly_RSEM_for_PAM50_allExpressedgenes_%s_%s.txt", BatchNo, Sys.Date()), col.names = NA)
```


```{r data4immune, cache=T, eval=T}
Rnames=rownames(allrsemFinal)
mNames1=SymHum2Rat$HGNC.symbol[match(Rnames, SymHum2Rat$RGD.symbol)]
mNames2=Rat2Hum$HGNC.symbol[match(Rnames, Rat2Hum$RGD.symbol)]

HumNameFinal=ifelse(is.na(mNames1), mNames2, mNames1)
x1=which(is.na(HumNameFinal)==T)

## save rsem for xcell
allrsemSave=allrsemFinal[-x1, ]
rownames(allrsemSave)=na.omit(HumNameFinal)

write.table(allrsemSave, sep="\t", file=sprintf("../output4external/RSEM_for_xcell%s__%s.txt",BatchNo, Sys.Date()), col.names = NA)

## save row names for cibersort
alltpmSave=allTPMFinal[-x1, ]
rownames(alltpmSave)=na.omit(HumNameFinal)
write.table(alltpmSave, sep="\t", file=sprintf("../output4external/TPM_for_cibersort%s_%s.txt",BatchNo, Sys.Date()), col.names = NA)
```

Save the mouse names for TIMER cistrome: check that this is actually required for TIMER

```{r, cache=T, eval=F}
allTPMCD45=allTPMFinal[ , grep("CD45", colnames(allTPMFinal))]
allTPMCD45_mus_ensmbl=allTPMCD45

# Finf Rat2Mouse

midx=Rat2Mouse$MGI.symbol[match(rownames(allTPMCD45),Rat2Mouse$RGD.symbol)]
lx1=rownames(allTPMCD45)[which(is.na(midx))]
# midx2=SymHum2Rat$HGNC.symbol[match(rownames(allTPMCD45), SymHum2Rat$RGD.symbol)]
# midx2=Rat2Hum$Gene.stable.ID[match(rownames(allTPMCD45), Rat2Hum$RGD.symbol)]
# midx2=SymHum2Rat$Gene.stable.ID[match(rownames(allTPMCD45), SymHum2Rat$MGI.symbol)]
# midxAll=ifelse(is.na(midx)==T,midx2, midx)


rownames(allTPMCD45_mus_ensmbl)=midx
t3=sort(unique(midx[which(duplicated(midx)==T)]))
t3=na.omit(t3)

for (i in t3){
  x1=which(rownames(allTPMCD45_mus_ensmbl)==i)
  allTPMCD45_mus_ensmbl[x1[1], ]=colSums(allTPMCD45_mus_ensmbl[x1, ])
}

allTPMCD45_mus_ensmbl=allTPMCD45_mus_ensmbl[-which(duplicated(rownames(allTPMCD45_mus_ensmbl))), ]

allTPMCD45_mus_ensmbl=allTPMCD45_mus_ensmbl[-which(is.na(rownames(allTPMCD45_mus_ensmbl))), ]
write.csv(allTPMCD45_mus_ensmbl, file=sprintf("../output4external/CD45_TPM_mm_mgi_names%s_%s.csv", BatchNo, Sys.Date()))

write.csv(allTPMCD45, file=sprintf("../output4external/CD45_TPM_rgd_names%s_%s.csv", BatchNo, Sys.Date()))
```




<!--chapter:end:03-rna-preprocessing.Rmd-->

# RNA data: preliminary plots

Prior to doing any comparative analysis, we will look at the following plots to get an overview of the data.

```{r}
# Load any required files:
load("../anntotations/Rat_biomart_gene_annotations2.RData")
#load("../dds_normalised_data_newstar_RNAseqapril_2020-11-07_rm_outliers.RData")

```

## PCA plots

We can check the new PCA plots, and overlay parameters of interest including treatment, growth, tumor size, CD8 fraction, spatial distribution.

```{r vsd-corrected,eval=T, cache=T}
vsd2 <- vst(dds)
vsd2$MHEpCAMcut=cut(infoTableFinal$MHEpCAM, c(-1, median(infoTableFinal$MHEpCAM, na.rm = T), 1.1), c("low", "high"))
vsd2$knnEpCAMcut=cut(infoTableFinal$knnEpCAM, c(-1, median(infoTableFinal$knnEpCAM, na.rm = T), 1E7), c("low", "high"))
vsd2$IFEpCAMcut=cut(infoTableFinal$IFEpCAM, c(-1, median(infoTableFinal$IFEpCAM, na.rm = T), 1.1), c("low", "high"))
vsd2$CD8FracCut=cut(infoTableFinal$CD8Frac, c(-1, median(infoTableFinal$CD8Frac, na.rm = T), 1.1), c("low", "high"))

# vsd2$MHEpAntstrRatio[which(vsd2$MHEpAntstrRatio>4)]=4
# vsd2$CD8FracCut=ifelse(vsd2$CD8Frac<0.05, "<0.05", ifelse(vsd2$CD8Frac<0.1, "<0.1", "0.1+"))
# vsd2$IFcut=ifelse(log2(vsd2$IFEpAntstrRatio)>0, "inf", "res")
# vsd2$MHcut=ifelse(log2(vsd2$MHEpAntstrRatio)>0, "inf", "res")

#vsd2$IFEpcam=Cdata$IFEpcam[match(vsd2$TumorID, Cdata$TumorID)]
  
#pdf(sprintf("rslt/DESeq/PCA_rm_outliers_%s.pdf",Sys.Date() ), width=8, height=6)

#All samples
#plotPCA(vsd2, "Batch")
#plotPCA(vsd2, "Fraction")
#plotPCA(vsd2, "Cohort")
#plotPCA(vsd2, c("Fraction"))+geom_label(aes(label = name)) 
plotPCA(vsd2, c("Treatment"))+ggtitle("Treatment")
plotPCA(vsd2, c("Growth"))+ggtitle("Growth")
plotPCA(vsd2, "TumSize")+ggtitle("tumor size")
#plotPCA(vsd2, "CD8Frac")+ggtitle("CD8 Fraction")
plotPCA(vsd2, "CD8FracCut")+ggtitle("CD8 Fraction")
plotPCA(vsd2, "SpatialManual")+ggtitle("CD8 distribution")
plotPCA(vsd2, "MHEpCAMcut")+ggtitle("MH EpCAM")
plotPCA(vsd2, "IFEpCAMcut")+ggtitle("IF SMA")
plotPCA(vsd2, "knnEpCAMcut")+ggtitle("knn EpCAM")

#Treatment
# plotPCA(vsd2[, grep("CD45", colnames(vsd2))], c("Treatment"))+ggtitle("CD45 only Treatment")+geom_label(aes(label = name))
# plotPCA(vsd2[rownames(assay(vsd2))%in%RatAllImm , grep("CD45", colnames(vsd2))], c("Treatment"))+ggtitle("CD45 only Treatment: immune only genes")+geom_label(aes(label = name))
# plotPCA(vsd2[, grep("Ep", colnames(vsd2))], c("Treatment"))+ggtitle("Ep only Treatment")
# plotPCA(vsd2[, grep("DN", colnames(vsd2))], c("Treatment"))+ggtitle("DN only Treatment")
# # growth
# plotPCA(vsd2[rownames(assay(vsd2))%in%RatAllImm  , grep("CD45", colnames(vsd2))], c("Growth"))+ggtitle("CD45 only Growth")
# plotPCA(vsd2[ , grep("Ep", colnames(vsd2))], "Growth")+ggtitle("Ep only Growth")
# plotPCA(vsd2[ , grep("DN", colnames(vsd2))], "Growth")+ggtitle("DN only Growth")
# # CD8 content
# plotPCA(vsd2[rownames(assay(vsd2))%in%RatAllImm  , grep("CD45", colnames(vsd2))], c("CD8FracCut"))+ggtitle("CD45 only CD8 Fraction")
# plotPCA(vsd2[ , grep("Ep", colnames(vsd2))], "CD8FracCut")+ggtitle("Ep only CD8 Fraction")
# plotPCA(vsd2[ , grep("DN", colnames(vsd2))], "CD8FracCut")+ggtitle("DN only CD8 Fraction")
# # CD8 infiltrating
# plotPCA(vsd2[rownames(assay(vsd2))%in%RatAllImm  , grep("CD45", colnames(vsd2))], c("SpatialManual"))+ggtitle("CD45 only CD8 Infiltrating")
# plotPCA(vsd2[ , grep("Ep", colnames(vsd2))], "SpatialManual")+ggtitle("Ep only CD8 Infiltrating")
# plotPCA(vsd2[ , grep("DN", colnames(vsd2))], "SpatialManual")+ggtitle("DN only CD8 Infiltrating")
```


In addition, we can look at the CD45 population and the distributions based on CD8 content and spatial infiltration

```{r}
# CD8 restricted/infil
plotPCA(vsd2[ , grep("CD45", colnames(vsd2))], "CD8FracCut")+ggtitle("CD8 content")
plotPCA(vsd2[ , grep("CD45", colnames(vsd2))], "IFEpCAMcut")+ggtitle("CD8-EpCAM Infiltrating Fraction")
plotPCA(vsd2[ , grep("CD45", colnames(vsd2))], "knnEpCAMcut")+ggtitle("CD8-EpCAM nearest neighbor distances")
plotPCA(vsd2[ , grep("CD45", colnames(vsd2))], "MHEpCAMcut")+ggtitle("CD8-EpCAM MH index")
```


## Expression patterns by cell type

Below, we check whether the different fractions are expressing expected markers 

The cell types are:

* Red: Cd45
* Epcam: green 
* DN:blue

Reference Genes:

* purple: immune
* blue: epithelial
* green: stroma
* orange: myoepithelial
* red: endothelial

Below, we see that the CD45 cells separate from the Ep/DN populations have high expression of immune related genes including CD3, CD4, IFNG.

However, the DN/Ep fractions are more intermixed. The DN fraction has expression of keratins, as well as fibroblast markers (Acta1), and myeoepithelial markers (Tp63)

```{r, fig.height=6}
Agenes=unlist(GeneListRat)

# Use the original desseq data,a nd then the limma data
x1=match(Agenes, rownames(assay(vsd2)))
RatExpr2=assay(vsd2)[na.omit(x1), ]

RowSideCol=names(Agenes)[which(!is.na(x1))]
RowSideCol=substr(RowSideCol, 1, 3)

ColSideCol=sapply(strsplit(colnames(RatExpr2), "_"), function(x) x[length(x)])

#pdf("rslt/DESeq/celltype_markers_RNA.pdf", height = 6, width=10)

a1=heatmap.2(RatExpr2, col=RdBu[11:1], trace="none", RowSideColors = brewer.pal(6, "Set1")[factor(RowSideCol)], scale="row",
             ColSideColors =brewer.pal(3, "Set1")[factor(ColSideCol)] )

a1=heatmap.2(RatExpr2[, -grep("CD45", colnames(RatExpr2))], col=RdBu[11:1], trace="none", RowSideColors = brewer.pal(6, "Set1")[factor(RowSideCol)], scale="row",
             ColSideColors =brewer.pal(3, "Set1")[factor(ColSideCol[-grep("CD45", colnames(RatExpr2))])] )



# plot the heatmap col-side bar
# ColBars=rbind(factor(infoTable$Fraction[a1$colInd]), factor(infoTable$Batch[a1$colInd]))
# image(t(ColBars), col=c(1:3), main="column index", yaxt="none", xaxt="none")
# axis(2, at=c(0, 1), labels=c("cell type", "batch"), las=2)
# par(oma=c(3, 0,0,0))
# a1=heatmap.2(RatExpr2[ ,grep("2[R|N]", colnames(RatExpr2))], col=RdBu[11:1], trace="none", RowSideColors = brewer.pal(6, "Set1")[factor(RowSideCol)], scale="row")

#dev.off()
```


<!--chapter:end:04-rna-prelim.Rmd-->

# DESeq analysis

This document sets up DESeq runs to compare:

* DN vs Ep samples
* growing vs stable samples (all 3 fractions)
* treatments (all 3 fractions)
* spatial patterns (all 3 fractions)

The outputs of these analyses will be used for Gene Set Enrichment Analysis,
using MSigDB databases (c2, c5, hallmark), alongside pathways from Metacore (Process Networks, Pathway Maps)

## DN vs Ep

In section 6.2, we have noticed that some DN samples had expression of epithelial markers. Here, we perform a differential gene expression analysis to find genes which are different between these two fractions. 

Below is a summary of the number of differential genes, using p value cut off of 0.05 and log2 fold change of 1.5 and base expression of 100+.

```{r ep-v-dn, cache=T}
resDNep=results(dds, contrast=c("Fraction", "DN", "Ep"))
summary(resDNep)

print('significant differential genes')

resDNeprslt2=resDNep[which(resDNep$padj<0.05 & abs(resDNep$log2FoldChange)>1.5 &
                             resDNep$baseMean>100), ]
resDNeprslt2$CellMarker=ifelse(rownames(resDNeprslt2)%in%unlist(GeneListRat), 1, 0)
resDNeprslt2=resDNeprslt2[order(resDNeprslt2$CellMarker, abs(resDNeprslt2$log2FoldChange), decreasing = T), ]


scroll_box(kable(resDNeprslt2, format="html"),
         height="300px", width="100%")

HighExprGenes=rownames(resDNeprslt2)[which(resDNeprslt2$baseMean>100 & resDNeprslt2$log2FoldChange<0) ]

colSide=factor(infoTableFinal$Fraction[ which(infoTableFinal$Fraction!="CD45")])
colSide2=factor(infoTableFinal$Growth[ which(infoTableFinal$Fraction!="CD45")])
colSide3=factor(infoTableFinal$Treatment[ which(infoTableFinal$Fraction!="CD45")])
colSide4=cut(infoTableFinal$CD8Frac[ which(infoTableFinal$Fraction!="CD45")], c(-1, 0.025, 0.05, 0.1, 1), brewer.pal(4, "Greens"))
colSide5=cut(infoTableFinal$IFEpCAM[which(infoTableFinal$Fraction!="CD45")], c(-1, 0.05, 0.1, 0.15, 1), brewer.pal(4, "Greens"))

TableCols=rbind(palette()[colSide], palette()[colSide2], palette()[colSide3], as.character(colSide4), as.character(colSide5))
rownames(TableCols)=c("Fraction", "growth", "treatment", "cd8 fraction", "cd8 int. fraction")
colnames(TableCols)=colnames(assay(vsd2))[which(infoTableFinal$Fraction!="CD45")]

#pdf(sprintf("rslt/DESeq/difference_between_DN_ep_samples_%s.pdf", Sys.Date()), height=7, width=8)

t1=assay(vsd2)[rownames(vsd2)%in%unlist(GeneListRat), ]

heatmap.2(t1[ ,-grep("CD45", colnames(t1))], trace="none", col=RdBu[11:1], ColSideColors = palette()[colSide], scale="row", main="cell type specific markers")

t2=assay(vsd2)[which(rownames(vsd2)%in%RatCosmic & rownames(vsd2)%in%rownames(resDNeprslt2)), ]
heatmap.2(t2[ ,-grep("CD45", colnames(t2))], trace="none", col=RdBu[11:1], ColSideColors = palette()[colSide], scale="row", main="DEG cosmic genes")

t2=assay(vsd2)[which( rownames(vsd2)%in%HighExprGenes), ]
a1=heatmap.2(t2[ ,-grep("CD45", colnames(t2))], trace="none", col=RdBu[11:1], ColSideColors = palette()[colSide], scale="row", main="Enriched in Ep genes")
#heatmap.2(t2[ ,-grep("CD45", colnames(t2))], trace="none", col=RdBu[11:1], ColSideColors = palette()[colSide2], scale="row", main="Enriched in Ep genes")

TableCols2=TableCols[ ,a1$colInd]

colOutput=melt(TableCols2)
colOutput$Var2=factor(colOutput$Var2, levels=unique(colOutput$Var2))

heatmap.plus::heatmap.plus(t2[ ,-grep("CD45", colnames(t2))], trace="none", col=RdBu[11:1], ColSideColors = t(TableCols), scale="row", main="Enriched in Ep genes")

write.csv(resDNeprslt2, file=sprintf("outputs/DESeq/difference_between_DN_Ep(ref)_samples_%s.csv", Sys.Date()))
```

Note that there seems to be two main groups of Ep cells:

The first one is enriched for canonical tumour progression pathways: eg. expression of wnt, fox1a etc.
The second one doesn't seem to have as high expression of these genes, but have high immune related signalling pathways.

We can append this information into the overall data file and determine whether there is an association with growth or treatment

```{r}
a2=heatmap.2(t2[ ,grep("Ep", colnames(t2))], trace="none", col=RdBu[11:1], scale="row", main="Enriched in Ep genes")
ac=as.hclust(a2$colDendrogram)
ad=cutree(ac, 3)
#table(ad)
head(ad)

infoTableFinal$EpCAMsubtype=NA
infoTableFinal$EpCAMsubtype[match(names(ad), rownames(infoTableFinal))]=ad

table(infoTableFinal$EpCAMsubtype, infoTableFinal$Growth)
table(infoTableFinal$EpCAMsubtype, infoTableFinal$Treatment)
table(infoTableFinal$EpCAMsubtype, infoTableFinal$SpatialManual)

```

## No. samples in comparisons

Separate out the tables into CD45, DN and Ep fractions. Compare

1. Differences based on treatment (and treatment vs control)
2. Specific treatments vs control
3. Growing vs stable (overall)
4. Growing vs stable (in each subgroup)

Below, we check the number of samples in each subgroup:

```{r, fig.height=5}
infoTableFinal$Treatment=factor(infoTableFinal$Treatment)#[which(is.na(infoTableFinal$Treatment))]="PDL1"
infoTableFinal$SpatialManual=factor(infoTableFinal$SpatialManual)
infoTableFinal$treatA=factor(ifelse(infoTableFinal$Treatment=="Vehicle", "control", "imm"))
infoTableFinal$Comp4=factor(paste(infoTableFinal$Growth, infoTableFinal$treatA, sep=""))
infoTableFinal$MHcut=factor(ifelse(infoTableFinal$MHEpCAM>=median(infoTableFinal$MHEpCAM, na.rm = T), "inf", "res"))
infoTableFinal$IFcut=factor(ifelse(infoTableFinal$IFEpCAM>=median(infoTableFinal$IFEpCAM, na.rm = T), "inf", "res"))
infoTableFinal$knncut=factor(ifelse(infoTableFinal$knnEpCAM<=median(infoTableFinal$knnEpCAM, na.rm = T), "inf", "res"))
infoTableFinal$CD8FracCut=factor(ifelse(infoTableFinal$CD8Frac>=median(infoTableFinal$CD8Frac, na.rm = T), "high", "low"))
# infoTableFinal$MHcutComp4=factor(paste(infoTableFinal$MHcut, infoTableFinal$treatA, sep=""))
# infoTableFinal$IFcutComp4=factor(paste(infoTableFinal$IFcut, infoTableFinal$treatA, sep=""))
infoTableFinal$Growth=factor(infoTableFinal$Growth)

# print('number of samples for each treatment and growth rate')
# table(infoTableFinal$Fraction, infoTableFinal$Treatment, infoTableFinal$Growth)

a1=table(infoTableFinal$Fraction, infoTableFinal$Treatment, infoTableFinal$Growth)
par(mfrow=c(2,2))
ContTable(t(a1[ , , 1]), "growing", F, ylabL="fraction", xlabL="treatment")
ContTable(t(a1[ , , 2]), "stable", F, ylabL="fraction", xlabL="treatment")

a1=table(infoTableFinal$Fraction, infoTableFinal$Treatment, infoTableFinal$MHcut)
ContTable(t(a1[ , , 1]), "infiltrating (MH)", F, ylabL="fraction", xlabL="treatment")
ContTable(t(a1[ , , 2]), "restricted (MH)", F, ylabL="fraction", xlabL="treatment")

a1
```

Note that based on the above tables, there some comparisons which are slightly imbalanced:

* There are few "stable" samples for comparisons
* LY samples are overwhelmingly "infiltrating"


## Set-up cell-type specifoc the comparisons

Setup the following comparisons for each cell type:

expression ~ growing + treatment + batch (exclude 1)  (start with vehicle vs some treatment)

4 categories only: immunotherapy vs non-immuno

Separate out the tables into CD45, DN and Ep fractions. Compare:

1. Treatment (EpddsTreat)
2. Comp4: Growth + any immunotherapy treatment (Epdds)
3. Comp8: Growth + immunotherapy (EpddsTreatG)
4. Growth alone: (EpddsGrowth)
5. MH index: (EpddsStrMH)
6. Interacting Fraction (EpddsStrIF) 
7. knn values (EpddsStrknn)
8. CD8 content (EpddsCD8)
9. Manual scoring spatial (EpddsSpatMan)
10. MH index + CD8 frac (Epddscd8MH)

We remove genes which have 0 counts in more than half the samples, and genes which have a row sum less than $10^{(log10(mean(rowsums))-log10(sd(rowsums)))}$.


```{r deseq-assoc-runs, cache=T}
# find epithelial samples
epidx=as.character(infoTableFinal$SampleID[which(infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction=="Ep" & !is.na(infoTableFinal$Growth))])
Epdds=DESeqDataSetFromMatrix(allstarFinal[ ,epidx], infoTableFinal[epidx, ], design=~Comp4+factor(Batch)) ## change class
a1x=rowSums(counts(Epdds))
a1b=apply(counts(Epdds), 1, function(c) sum(c!=0))
 # par(mfrow=c(1,2))
 # hist(log10(a1x+1), main="log10 total counts")
 # hist((a1b+1), main="Non-zero entries")
sd1vals=mean(log10(a1x+1))-sd(log10(a1x+1))
keep=which(rowSums(counts(Epdds))>10^sd1vals)
keep2=which(apply(counts(Epdds), 1, function(c) sum(c!=0))> (ncol(Epdds)/2))
Epdds=Epdds[intersect(keep, keep2), ]
Epdds=DESeq(Epdds)

EpddsTreat=Epdds
design(EpddsTreat)=~Treatment+factor(Batch)
EpddsTreat=DESeq(EpddsTreat)

EpddsTreatG=Epdds
EpddsTreatG$Comp8=factor(paste(EpddsTreatG$Growth, EpddsTreatG$Treatment, sep="."))
design(EpddsTreatG)=~Comp8+factor(Batch)
EpddsTreatG=DESeq(EpddsTreatG)

EpddsGrowth=Epdds
design(EpddsGrowth)=~Growth+factor(Batch)  ## error is here!
EpddsGrowth=DESeq(EpddsGrowth)

EpddsStrMH=Epdds[ ,which(!is.na(Epdds$MHcut))]
design(EpddsStrMH)=~MHcut+factor(Batch)
EpddsStrMH=DESeq(EpddsStrMH)

EpddsStrIF=Epdds[ ,which(!is.na(Epdds$IFcut))]
design(EpddsStrIF)=~IFcut+factor(Batch)
EpddsStrIF=DESeq(EpddsStrIF)

EpddsStrknn=Epdds[ ,which(!is.na(Epdds$knncut))]
design(EpddsStrknn)=~knncut+factor(Batch)
EpddsStrknn=DESeq(EpddsStrknn)

Epddscd8=Epdds[ ,which(!is.na(Epdds$CD8FracCut))]
design(Epddscd8)=~CD8FracCut+factor(Batch)
Epddscd8=DESeq(Epddscd8)

# EpddsStrMHComp4=Epdds[ ,which(!is.na(Epdds$MHcut))]
# EpddsStrMHComp4$MHcutComp4<-droplevels(EpddsStrMHComp4$MHcutComp4)
# design(EpddsStrMHComp4)=~MHcutComp4+factor(Batch)
# EpddsStrMHComp4=DESeq(EpddsStrMHComp4)
# 
# EpddsStrIFComp4=Epdds[ ,which(!is.na(Epdds$IFcut))]
# EpddsStrIFComp4$IFcutComp4<-droplevels(EpddsStrIFComp4$IFcutComp4)
# design(EpddsStrIFComp4)=~IFcutComp4+factor(Batch)
# EpddsStrIFComp4=DESeq(EpddsStrIFComp4)

EpddsSpatMan=Epdds[ , which(!is.na(Epdds$SpatialManual))]
design(EpddsSpatMan)=~SpatialManual+factor(Batch)
EpddsSpatMan=DESeq(EpddsSpatMan)


Epddscd8MH=Epdds[ ,which(!is.na(Epdds$MHcut))]
Epddscd8MH$cd8MH=factor(paste(Epddscd8MH$CD8FracCut, Epddscd8MH$MHcut, sep="."))
design(Epddscd8MH)=~cd8MH+factor(Batch)
Epddscd8MH=DESeq(Epddscd8MH)

#R1=results(EpddsTreatG, contrast = list(c("Treatment_LY_vs_Vehicle", "Growthgrowing")))
## CD45 samples

cdidx=as.character(infoTableFinal$SampleID[which(infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction=="CD45" & !is.na(infoTableFinal$Growth))])
CDdds=DESeqDataSetFromMatrix(allstarFinal[ ,cdidx], infoTableFinal[cdidx, ], design=~Comp4+factor(Batch)) ## change class
a1x=rowSums(counts(CDdds))
a1b=apply(counts(CDdds), 1, function(c) sum(c!=0))
# par(mfrow=c(1,2))
# hist(log10(a1x+1), main="log10 total counts")
# hist((a1b+1), main="Non-zero entries")
sd1vals=mean(log10(a1x+1))-sd(log10(a1x+1))
keep=which(rowSums(counts(CDdds))>10^sd1vals)
keep2=which(apply(counts(CDdds), 1, function(c) sum(c!=0))> (ncol(CDdds)/2))
CDdds=CDdds[intersect(keep, keep2), ]
CDdds=DESeq(CDdds)

CDddsTreat=CDdds
design(CDddsTreat)=~Treatment+factor(Batch)
CDddsTreat=DESeq(CDddsTreat)

CDddsTreatG=CDdds
CDddsTreatG$Comp8=factor(paste(CDddsTreatG$Growth, CDddsTreatG$Treatment, sep="."))
design(CDddsTreatG)=~Comp8+factor(Batch)
CDddsTreatG=DESeq(CDddsTreatG)

CDddsGrowth=CDdds
design(CDddsGrowth)=~Growth+factor(Batch)
CDddsGrowth=DESeq(CDddsGrowth)

CDddsStrMH=CDdds[ ,which(!is.na(CDdds$MHcut))]
design(CDddsStrMH)=~MHcut+factor(Batch)
CDddsStrMH=DESeq(CDddsStrMH)

CDddsSpatMan=CDdds[ , which(!is.na(CDdds$SpatialManual))]
design(CDddsSpatMan)=~SpatialManual+factor(Batch)
CDddsSpatMan=DESeq(CDddsSpatMan)

CDddsStrIF=CDdds[ ,which(!is.na(CDdds$IFcut))]
design(CDddsStrIF)=~IFcut+factor(Batch)
CDddsStrIF=DESeq(CDddsStrIF)

CDddsStrknn=CDdds[ ,which(!is.na(CDdds$knncut))]
design(CDddsStrknn)=~knncut+factor(Batch)
CDddsStrknn=DESeq(CDddsStrknn)

CDddscd8=CDdds[ ,which(!is.na(CDdds$CD8FracCut))]
design(CDddscd8)=~CD8FracCut+factor(Batch)
CDddscd8=DESeq(CDddscd8)

CDddscd8MH=CDdds[ ,which(!is.na(CDdds$MHcut))]
CDddscd8MH$cd8MH=factor(paste(CDddscd8MH$CD8FracCut, CDddscd8MH$MHcut, sep="."))
design(CDddscd8MH)=~cd8MH+factor(Batch)
CDddscd8MH=DESeq(CDddscd8MH)
## DN samples

DNidx=as.character(infoTableFinal$SampleID[which(infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction=="DN" & !is.na(infoTableFinal$Growth))])
DNdds=DESeqDataSetFromMatrix(allstarFinal[ ,DNidx], infoTableFinal[DNidx, ], design=~Comp4+factor(Batch)) ## change class
a1x=rowSums(counts(DNdds))
a1b=apply(counts(DNdds), 1, function(c) sum(c!=0))
# par(mfrow=c(1,2))
# hist(log10(a1x+1), main="log10 total counts")
# hist((a1b+1), main="Non-zero entries")
sd1vals=median(log10(a1x+1))-sd(log10(a1x+1))
keep=which(rowSums(counts(DNdds))>10^sd1vals)
keep2=which(apply(counts(DNdds), 1, function(c) sum(c!=0))> (ncol(DNdds)/2))
keep=which(rowSums(counts(DNdds))>(ncol(DNdds)))
keep2=which(apply(counts(DNdds), 1, function(c) sum(c!=0))> (ncol(DNdds)/2))
DNdds=DNdds[intersect(keep, keep2), ]
DNdds=DESeq(DNdds)

DNddsTreat=DNdds
design(DNddsTreat)=~Treatment+factor(Batch)
DNddsTreat=DESeq(DNddsTreat)

DNddsTreatG=DNdds
DNddsTreatG$Comp8=factor(paste(DNddsTreatG$Growth, DNddsTreatG$Treatment, sep="."))
design(DNddsTreatG)=~Comp8+factor(Batch)
DNddsTreatG=DESeq(DNddsTreatG)

DNddsGrowth=DNdds
design(DNddsGrowth)=~Growth+factor(Batch)
DNddsGrowth=DESeq(DNddsGrowth)

DNddsStrMH=DNdds[ ,which(!is.na(DNdds$MHcut))]
design(DNddsStrMH)=~MHcut+factor(Batch)
DNddsStrMH=DESeq(DNddsStrMH)

# DNddsStrMHComp4=DNdds[ ,which(!is.na(DNdds$MHcut))]
# DNddsStrMHComp4$MHcutComp4=droplevels(DNddsStrMHComp4$MHcutComp4)
# 
# design(DNddsStrMHComp4)=~MHcutComp4+factor(Batch)
# DNddsStrMHComp4=DESeq(DNddsStrMHComp4)

DNddsStrIF=DNdds[ ,which(!is.na(DNdds$IFcut))]
design(DNddsStrIF)=~IFcut+factor(Batch)
DNddsStrIF=DESeq(DNddsStrIF)

# DNddsStrIFComp4=DNdds[ ,which(!is.na(DNdds$IFcut))]
# DNddsStrIFComp4$IFcutComp4=droplevels(DNddsStrIFComp4$IFcutComp4)
# design(DNddsStrIFComp4)=~IFcutComp4+factor(Batch)
# DNddsStrIFComp4=DESeq(DNddsStrIFComp4)

DNddsSpatMan=DNdds[ , which(!is.na(DNdds$SpatialManual))]
design(DNddsSpatMan)=~SpatialManual+factor(Batch)
DNddsSpatMan=DESeq(DNddsSpatMan)

DNddsStrknn=DNdds[ ,which(!is.na(DNdds$knncut))]
design(DNddsStrknn)=~knncut+factor(Batch)
DNddsStrknn=DESeq(DNddsStrknn)

DNddscd8=DNdds[ ,which(!is.na(DNdds$CD8FracCut))]
design(DNddscd8)=~CD8FracCut+factor(Batch)
DNddscd8=DESeq(DNddscd8)


DNddscd8MH=DNdds[ ,which(!is.na(DNdds$MHcut))]
DNddscd8MH$cd8MH=factor(paste(DNddscd8MH$CD8FracCut, DNddscd8MH$MHcut, sep="."))
design(DNddscd8MH)=~cd8MH+factor(Batch)
DNddscd8MH=DESeq(DNddscd8MH)

save(Epdds, EpddsTreat,CDdds, CDddsTreat, DNdds, DNddsTreat,
     EpddsTreatG, CDddsTreatG, DNddsTreatG, EpddsGrowth, CDddsGrowth, DNddsGrowth, 
     DNddsStrknn, DNddsStrIF, DNddscd8, DNddscd8MH, DNddsStrMH,
     CDddsStrknn, CDddsStrIF, CDddscd8, CDddscd8MH, CDddsStrMH,
     EpddsStrknn, EpddsStrIF, Epddscd8, Epddscd8MH, EpddsStrMH,
     EpddsSpatMan, CDddsSpatMan, DNddsSpatMan,
     file=sprintf("outputs/subfraction_analysis_%s.RData", Sys.Date()))

# Alist=results(EpddsSpatMan, c("SpatialManual", "Infiltrating", "restricted"))
# Clist=results(CDddsSpatMan, c("SpatialManual", "Infiltrating", "restricted"))
# Dlist=results(DNddsSpatMan, c("SpatialManual", "Infiltrating", "restricted"))

#load("rslt/DESeq/subfraction_analysis_2020-09-29.RData")
#write.csv(a2, sprintf("rslt/DESeq/Spatial_Manual_comparison_%s.csv", Sys.Date()))
```

These comparisons are saved in the temporary outputfile `r sprintf("outputs/subfraction_analysis_%s.RData", Sys.Date())`. In each comparison, there are 
`r nrow(Epdds); nrow(CDdds); nrow(DNdds)` genes compared.


```{r deseq-spatManual, cache=T, eval=F}
FractionTest="CD45"
# infoTableFinal$MHEpCAMcut=cut(infoTableFinal$MHEpCAM, c(-1, median(infoTableFinal$MHEpCAM, na.rm = T), 1.1), c("low", "high"))
 infoTableFinal$knnEpCAMcut=cut(infoTableFinal$knnEpCAM, c(-1, median(infoTableFinal$knnEpCAM, na.rm = T), 1E7), c("low", "high"))
# infoTableFinal$IFEpCAMcut=cut(infoTableFinal$IFEpCAM, c(-1, median(infoTableFinal$IFEpCAM, na.rm = T), 1.1), c("low", "high"))
 infoTableFinal$CD8FracCut=cut(infoTableFinal$CD8Frac, c(-1, median(infoTableFinal$CD8Frac, na.rm = T), 1.1), c("low", "high"))

EpSpatialddsList=list()

epidx=as.character(infoTableFinal$SampleID[which(infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction==FractionTest & !is.na(infoTableFinal$MHcut))])
EpddsMH=DESeqDataSetFromMatrix(allstarFinal[ ,epidx], infoTableFinal[epidx, ], design=~MHcut) ## change class
EpSpatialddsList[[1]]=DESeq(EpddsMH)
EpddsIF=EpddsMH
design(EpddsIF)=~IFcut
EpSpatialddsList[[2]]=DESeq(EpddsIF)
Epddsknn=EpddsMH
design(Epddsknn)=~knnEpCAMcut
EpSpatialddsList[[3]]=DESeq(Epddsknn)
Epddscd8=EpddsMH
design(Epddscd8)=~CD8FracCut
EpSpatialddsList[[4]]=DESeq(Epddscd8)
epidx=as.character(infoTableFinal$SampleID[which(infoTableFinal$Cohort=="Progression" & infoTableFinal$Fraction==FractionTest & !is.na(infoTableFinal$SpatialManual))])
EpddsSM=DESeqDataSetFromMatrix(allstarFinal[ ,epidx], infoTableFinal[epidx, ], design=~SpatialManual) ## change class
EpSpatialddsList[[5]]=DESeq(EpddsSM)


#run the summaries?
Rlist=lapply(EpSpatialddsList, results)
RlistSum=lapply(Rlist, summary)
DiffGenes1=lapply(Rlist, function(x) x$log2FoldChange[which(x$padj<0.1 & abs(x$log2FoldChange)>1.5 & x$baseMean>50)])
DiffGenes2=lapply(Rlist, function(x) rownames(x)[which(x$padj<0.1 & abs(x$log2FoldChange)>1.5 & x$baseMean>50)])
DiffGenes1[[5]]=DiffGenes1[[5]]*(-1)
DiffGenes1[[3]]=DiffGenes1[[3]]*(-1)

#DiffGenes2=lapply(1:length(Rlist), function(x) {names(DiffGenes[[x]])<-rownames(Rlist[[x]])[which(Rlist[[x]]$padj<0.05 & abs(Rlist[[x]]$log2FoldChange)>1.5 & Rlist[[x]]$baseMean>50)]; DiffGenes[[x]]})
names(DiffGenes1)=c("MH", "IF", "knn", "CD8", "SpatMan")
names(DiffGenes2)=c("MH", "IF", "knn", "CD8", "SpatMan")
lxa=melt(DiffGenes1)
lxb=melt(DiffGenes2)
lx=cbind(lxa, lxb)
lx=lx[ ,-4]
colnames(lx)=c("value", "L1", "Gene")
lxTable=acast(lx, Gene~L1)
lxTable[which(is.na(lxTable), arr.ind = T)]=0

lxTableRS=rowSums(abs(sign(lxTable)))
idx=which(lxTableRS>0)

#pdf(sprintf("rslt/DESeq/Spatial_DEG_common_%s_%s.pdf", FractionTest, Sys.Date()), width = 14, height=6)
heatmap.2(t(lxTable[idx, ]), trace="none",scale="none", col=RdBu[11:1], main=sprintf("diffGenes fold change: %s Blue:low/rest, Red:high/infil ", FractionTest))
par(mfrow=c(1,2))
venn(DiffGenes2[1:3])
venn(DiffGenes2[c(1, 4:5)])
intersect(DiffGenes1[[1]], DiffGenes1[[2]])
intersect(intersect(DiffGenes2[[1]], DiffGenes2[[4]]), DiffGenes2[[5]])
#dev.off()

genelist1=rownames(Rlist[[1]])
midx=lapply(Rlist, function(x) match(rownames(x), genelist1))

df.All.save=cbind(Rlist[[1]][midx[[1]], c(1,2, 5:6) ], Rlist[[2]][midx[[2]], c(2, 5:6)], 
                  Rlist[[3]][midx[[3]], c(2, 5:6)],Rlist[[4]][midx[[4]], c(2, 5:6)],
                  Rlist[[5]][midx[[5]], c(2, 5:6)])
colnames(df.All.save)[2:ncol(df.All.save)]=paste(rep(names(DiffGenes1), each=3), colnames(df.All.save)[2:ncol(df.All.save)])

write.csv(df.All.save, file=sprintf("outputs/DESeq/DEG_spatial_manual_%s_%s.csv", FractionTest, Sys.Date()))

TestInput=read.csv(sprintf("outputs/DESeq/DEG_spatial_manual_%s_%s.csv", FractionTest, Sys.Date()), row.names = 1)
a1=match(lm22rat$Gene.symbol, rownames(TestInput))
TestInput$LM22gene=""
TestInput$LM22gene[na.omit(a1)]=1
write.csv(TestInput, "outputs/DEG_spatial_manual_Ep_2020-10-14_annot_Imm_genes.csv")
```

## PCA plots

Below, we look at PCA plots with information on treatment, growth, spatial patterns overlaid. 

These are separated based on cell type

### EpCAM

```{r}
#pdf(sprintf("rslt/DESeq/Sample_specific_PCA_%s.pdf", Sys.Date()), height=9, width=9)
Epdds$cd8MH=paste(Epdds$CD8FracCut, Epdds$MHcut)

vstEp=vst(Epdds, blind=T)
plotPCA(vstEp, intgroup=c("Treatment"))+geom_text(aes(label=colnames(vstEp)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("Ep population: treatment")
plotPCA(vstEp, intgroup=c("Treatment", "Growth"))+geom_text(aes(label=colnames(vstEp)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("Ep population: treatment + growth")
plotPCA(vstEp, intgroup=c("Comp4"))+geom_text(aes(label=colnames(vstEp)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("Ep treatment (immune vs control) + growth")
plotPCA(vstEp, intgroup=c("Growth"))+geom_text(aes(label=Epdds$TumSize),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("Ep population: growth: kinetics vs final tum size")
plotPCA(vstEp, intgroup=c("MHcut"))+geom_text(aes(label=Epdds$MHEpCAM),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("Ep population: MH ep to stroma index")
plotPCA(vstEp, intgroup=c("IFcut"))+geom_text(aes(label=Epdds$IFEpCAM),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("Ep population: IF ep to stroma index")
plotPCA(vstEp, intgroup=c("cd8MH"))+geom_text(aes(label=Epdds$cd8MH),vjust=2,check_overlap = TRUE,size = 4)+ggtitle("CD45 population: cd8 + MH")
```

### CD45

```{r}
CDdds$cd8MH=paste(CDdds$CD8FracCut, CDdds$MHcut)

vstCD=vst(CDdds, blind=T)
plotPCA(vstCD, intgroup=c("Treatment"))+geom_text(aes(label=colnames(vstCD)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("CD45 population: treatment")
plotPCA(vstCD, intgroup=c("Treatment", "Growth"))+geom_text(aes(label=colnames(vstCD)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("CD45 population: treatment + growth")
plotPCA(vstCD, intgroup=c("Comp4"))+geom_text(aes(label=colnames(vstCD)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("CD45 treatment (immune vs control) + growth")
plotPCA(vstCD, intgroup=c("Growth"))+geom_text(aes(label=CDdds$TumSize),vjust=2,check_overlap = TRUE,size = 4)+ggtitle("CD45 population: growth: kinetics vs final tum size")
plotPCA(vstCD, intgroup=c("MHcut"))+geom_text(aes(label=CDdds$MHEpCAM),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("CD45 treatment MH value")
plotPCA(vstCD, intgroup=c("IFcut"))+geom_text(aes(label=CDdds$IFEpCAM),vjust=2,check_overlap = TRUE,size = 4)+ggtitle("CD45 population: IF value")
plotPCA(vstCD, intgroup=c("cd8MH"))+geom_text(aes(label=CDdds$cd8MH),vjust=2,check_overlap = TRUE,size = 4)+ggtitle("CD45 population: cd8 + MH")
```

### DN

```{r}
DNdds$cd8MH=paste(DNdds$CD8FracCut, DNdds$MHcut)

vstDN=vst(DNdds, blind=T)
plotPCA(vstDN, intgroup=c("Treatment"))+geom_text(aes(label=colnames(vstDN)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("DN population: treatment ")
plotPCA(vstDN, intgroup=c("Treatment", "Growth"))+geom_text(aes(label=colnames(vstDN)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("DN population: treatment + growth")
plotPCA(vstDN, intgroup=c("Comp4"))+geom_text(aes(label=colnames(vstDN)),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("DN treatment (immune vs control) + growth")
plotPCA(vstDN, intgroup=c("Growth"))+geom_text(aes(label=DNdds$TumSize),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("DN population: growth: kinetics vs final tum size")
plotPCA(vstDN, intgroup=c("MHcut"))+geom_text(aes(label=DNdds$MHEpCAM),vjust=1,check_overlap = TRUE,size = 4)+ggtitle("DN treatment (immune vs control) + growth")
plotPCA(vstDN, intgroup=c("IFcut"))+geom_text(aes(label=DNdds$IFEpCAM),vjust=2,check_overlap = TRUE,size = 4)+ggtitle("DN population: growth: kinetics vs final tum size")
plotPCA(vstDN, intgroup=c("cd8MH"))+geom_text(aes(label=DNdds$cd8MH),vjust=2,check_overlap = TRUE,size = 4)+ggtitle("CD45 population: cd8 + MH")
#dev.off()

```


## Collating results and running GSEA

In this section, we combine all the comparisons together for downstream analysis, including

* number of differential genes per comparisons
* number of samples per comparison
* write the output to file (outputs/DESeq/Ep_growth_treat.xlsx)

For GSEA we

* convert rat to human symbols
* run overlap analysis if there are more than 3 hits (outputs/DESeq/HyperGeo_Ep_growth_treat.xlsx)
* run gene set enrichment analysis on all samples (outputs/DESeq/GSEA_Ep_growth_treat.xlsx)
* all outputs saved to outputs/all_differential_comparisons.RData)

```{r}
#load("rslt/DESeq/Epithelial_fraction.RData")

table(Epdds$Comp4)

xsearch=c("stablecontrol","growingimm",  "stableimm", "stableimm")
ysearch=c("growingcontrol","growingcontrol",  "stablecontrol", "growingimm")

## for treatment control: big differences here which are not replicated

EpComp4=list()
EpCompSig=list()
UpDn1=matrix(NA, nrow=2, ncol=21)

for (i in 1:4){
Eres1=results(Epdds, contrast=c("Comp4", xsearch[i], ysearch[i]))
EpComp4[[i]]=as.data.frame(cbind(Gene=rownames(Eres1), Eres1))
EpCompSig[[i]]=sort(rownames(Eres1)[which(Eres1$padj<0.1)])
UpDn1[ ,i]=c(length(which(Eres1$padj<0.1 & Eres1$log2FoldChange<0)),
             length(which(Eres1$padj<0.1 & Eres1$log2FoldChange>0)))
}
# for control

cList=c("PDL1+LY", "LY", "PDL1")
for (i in 1:3){
Eres5=results(EpddsTreat, contrast=c("Treatment", cList[i], "Vehicle"))
EpComp4[[i+4]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[i+4]]=g1
UpDn1[ ,i+4]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}

xsearchA=c("stable.Vehicle","stable.LY",  "stable.PDL1+LY", "stable.PDL1")
ysearchB=c("growing.Vehicle","growing.LY",  "growing.PDL1+LY", "growing.PDL1")


for (i in 1:4){
Eres5=results(EpddsTreatG, contrast=c("Comp8", xsearchA[i], ysearchB[i]))
EpComp4[[i+7]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[i+7]]=g1
UpDn1[ ,i+7]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}


Eres5=results(EpddsGrowth, contrast=c("Growth", "stable", "growing"))
EpComp4[[12]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[12]]=g1
UpDn1[ ,12]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))



Eres5=results(EpddsStrMH, contrast=c("MHcut", "res", "inf"))
EpComp4[[13]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[13]]=g1
UpDn1[ ,13]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(EpddsStrIF, contrast=c("IFcut", "res", "inf"))
EpComp4[[14]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[14]]=g1
UpDn1[ ,14]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(EpddsStrknn, contrast=c("knncut", "res", "inf"))
EpComp4[[15]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[15]]=g1
UpDn1[ ,15]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(Epddscd8, contrast=c("CD8FracCut", "low", "high"))
EpComp4[[16]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[16]]=g1
UpDn1[ ,16]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(EpddsSpatMan, contrast=c("SpatialManual", "Infiltrating", "restricted"))
EpComp4[[17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[17]]=g1
UpDn1[ ,17]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

xsearchC=c("low.inf","high.inf",  "low.inf", "low.res")
ysearchD=c("low.res","high.res", "high.inf", "high.res")

for (i in 1:4){
Eres5=results(Epddscd8MH, contrast=c("cd8MH", xsearchC[i], ysearchD[i]))
EpComp4[[i+17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
EpCompSig[[i+17]]=g1
UpDn1[ ,i+17]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}

#boxplot(assay(Epdds)["Snca", ]~Epdds$treatA+Epdds$Growth)

rownames(UpDn1)=c("down", "up")
colnames(UpDn1)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")
names(EpComp4)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")
names(EpCompSig)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")

## write to a xls
write_xlsx(EpComp4, path=sprintf("outputs/DESeq/Ep_growth_treat.xlsx"))
```


```{r run-gsea, cache=T}
#load("anntotations/Rat_biomart_gene_annotations.RData")

epGenes=rownames(Epdds)
cdGenes=rownames(CDdds)
dnGenes=rownames(DNdds)

l1=SymHum2Rat$HGNC.symbol[match(epGenes, SymHum2Rat$RGD.symbol)]
l2=Rat2Hum$HGNC.symbol[match(epGenes, Rat2Hum$RGD.symbol)]
l3=Mouse2Hum$HGNC.symbol[match(epGenes, Mouse2Hum$MGI.symbol)]
epGenesConv=ifelse(is.na(l1)==F, l1, ifelse(is.na(l2)==F, l2, ifelse(is.na(l3)==F, l3, epGenes)))

l1=SymHum2Rat$HGNC.symbol[match(cdGenes, SymHum2Rat$RGD.symbol)]
l2=Rat2Hum$HGNC.symbol[match(cdGenes, Rat2Hum$RGD.symbol)]
l3=Mouse2Hum$HGNC.symbol[match(cdGenes, Mouse2Hum$MGI.symbol)]
cdGenesConv=ifelse(is.na(l1)==F, l1, ifelse(is.na(l2)==F, l2, ifelse(is.na(l3)==F, l3, cdGenes)))

l1=SymHum2Rat$HGNC.symbol[match(dnGenes, SymHum2Rat$RGD.symbol)]
l2=Rat2Hum$HGNC.symbol[match(dnGenes, Rat2Hum$RGD.symbol)]
l3=Mouse2Hum$HGNC.symbol[match(dnGenes, Mouse2Hum$MGI.symbol)]
dnGenesConv=ifelse(is.na(l1)==F, l1, ifelse(is.na(l2)==F, l2, ifelse(is.na(l3)==F, l3, dnGenes)))

```


```{r}
load("../anntotations/ListofGeneSets2.RData")
```

```{r ep-samples-gsea, cache=T, results='hide', warning=F, message=F}
#save(ListGSC, file="anntotations/ListofGeneSets2.RData")
OutputMatrix=lapply( ListGSC, function(x) matrix(NA, ncol=length(EpComp4)*2, nrow=length(x)))
ismr3 <- lapply(1:length(OutputMatrix), function(x){ row.names(OutputMatrix[[x]])<-names(ListGSC[[x]]); OutputMatrix[[x]]})
ismrB<-ismr3 

for (i in 1:length(EpComp4)){

hits=EpCompSig[[i]]
hits=epGenesConv[match(hits, rownames(Epdds))]
fcTab=EpComp4[[i]]$log2FoldChange
names(fcTab)=epGenesConv


if (length(hits)>2){
  gsca=GSCA(listOfGeneSetCollections=ListGSC,geneList=fcTab, hits = hits)
  gsca1 <- preprocess(gsca, species="Hs", initialIDs="SYMBOL",
                    keepMultipleMappings=TRUE, duplicateRemoverMethod="max",
                    orderAbsValue=FALSE)
  gsca2 <- analyze(gsca1, 
                 para=list(pValueCutoff=0.05, pAdjustMethod="BH",
                           nPermutations=100, minGeneSetSize=5,
                           exponent=1), 
                           doGSOA = T)
}else{
  gsca=GSCA(listOfGeneSetCollections=ListGSC, geneList=fcTab)
  gsca1 <- preprocess(gsca, species="Hs", initialIDs="SYMBOL",
                    keepMultipleMappings=TRUE, duplicateRemoverMethod="max",
                    orderAbsValue=FALSE)
  gsca2 <- analyze(gsca1, 
                 para=list(pValueCutoff=0.05, pAdjustMethod="BH",
                           nPermutations=100, minGeneSetSize=5,
                           exponent=1), 
                           doGSOA = F)
}

for (j in 1:length(ListGSC)){
  x1=gsca2@result$GSEA.results[[j]]
  mx1=match(rownames(x1), rownames(ismr3[[j]]))
  ismr3[[j]][ mx1,(2*i-1):(2*i)]=data.matrix(x1[, c(1, 3)])
}

if (length(hits)>2){
for (j in 1:length(ListGSC)){
  x1=gsca2@result$HyperGeo.results[[j]]
  mx1=match(rownames(x1), rownames(ismrB[[j]]))
  ismrB[[j]][ mx1,(2*i-1):(2*i)]=data.matrix(x1[, c(6, 7)])
}}


}

Epismr3 <- lapply(ismr3, function(x){ colnames(x)<- paste(rep(names(EpComp4), each=2), c("NES", "padj")); x})
names(Epismr3)=names(ListGSC)
Epismr3<-lapply(Epismr3, function(x) as.data.frame(cbind(GeneSet=rownames(x), x)))
write_xlsx(Epismr3, path=sprintf("outputs/DESeq/GSEA_Ep_growth_treat_%s.xlsx", Sys.Date()))

EpismrHG <- lapply( ismrB, function(x){ colnames(x)<- paste(rep(names(EpComp4), each=2), c("pval", "padj")); x})
names(EpismrHG)=names(ListGSC)
EpismrHG<-lapply(EpismrHG, function(x) as.data.frame(cbind(GeneSet=rownames(x), x)))
write_xlsx(EpismrHG, path=sprintf("outputs/DESeq/HyperGeo_Ep_growth_treat_%s.xlsx", Sys.Date()))
```


```{r cd45-samples-gsea, results='hide', warning=F, message=F, cache=T}
#load("rslt/DESeq/Epithelial_fraction.RData")

table(CDdds$Comp4)

xsearch=c("stablecontrol","growingimm",  "stableimm", "stableimm")
ysearch=c("growingcontrol","growingcontrol",  "stablecontrol", "growingimm")

## for treatment control: big differences here which are not replicated

CDComp4=list()
CDCompSig=list()
CDUpDn1=matrix(NA, nrow=2, ncol=21)

for (i in 1:4){
Eres1=results(CDdds, contrast=c("Comp4", xsearch[i], ysearch[i]))
CDComp4[[i]]=as.data.frame(cbind(Gene=rownames(Eres1), Eres1))
CDCompSig[[i]]=sort(rownames(Eres1)[which(Eres1$padj<0.1)])
CDUpDn1[ ,i]=c(length(which(Eres1$padj<0.1 & Eres1$log2FoldChange<0)),
             length(which(Eres1$padj<0.1 & Eres1$log2FoldChange>0)))
}
# for control

cList=c("PDL1+LY", "LY", "PDL1")
for (i in 1:3){
Eres5=results(CDddsTreat, contrast=c("Treatment", cList[i], "Vehicle"))
CDComp4[[i+4]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[i+4]]=g1
CDUpDn1[ ,i+4]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}

xsearchA=c("stable.Vehicle","stable.LY",  "stable.PDL1+LY", "stable.PDL1")
ysearchB=c("growing.Vehicle","growing.LY",  "growing.PDL1+LY", "growing.PDL1")


for (i in 1:4){
Eres5=results(CDddsTreatG, contrast=c("Comp8", xsearchA[i], ysearchB[i]))
CDComp4[[i+7]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[i+7]]=g1
CDUpDn1[ ,i+7]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}

#boxplot(assay(CDdds)["Snca", ]~CDdds$treatA+CDdds$Growth)

Eres5=results(CDddsGrowth, contrast=c("Growth", "stable", "growing"))
CDComp4[[12]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[12]]=g1
CDUpDn1[ ,12]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))


Eres5=results(CDddsStrMH, contrast=c("MHcut", "res", "inf"))
CDComp4[[13]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[13]]=g1
CDUpDn1[ ,13]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))


Eres5=results(CDddsStrIF, contrast=c("IFcut", "res", "inf"))
CDComp4[[14]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[14]]=g1
CDUpDn1[ ,14]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(CDddsStrknn, contrast=c("knncut", "res", "inf"))
CDComp4[[15]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[15]]=g1
CDUpDn1[ ,15]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(CDddscd8, contrast=c("CD8FracCut", "low", "high"))
CDComp4[[16]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[16]]=g1
CDUpDn1[ ,16]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(CDddsSpatMan, contrast=c("SpatialManual", "Infiltrating", "restricted"))
CDComp4[[17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[17]]=g1
CDUpDn1[ ,17]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

xsearchC=c("low.inf","high.inf",  "low.inf", "low.res")
ysearchD=c("low.res","high.res", "high.inf", "high.res")

for (i in 1:4){
Eres5=results(CDddscd8MH, contrast=c("cd8MH", xsearchC[i], ysearchD[i]))
CDComp4[[i+17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
CDCompSig[[i+17]]=g1
CDUpDn1[ ,i+17]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}


rownames(CDUpDn1)=c("down", "up")
colnames(CDUpDn1)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")
names(CDComp4)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")
names(CDCompSig)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")

## write to a xls
write_xlsx(CDComp4, path=sprintf("outputs/DESeq/CD_growth_treat_%s.xlsx", Sys.Date()))

OutputMatrix=lapply( ListGSC, function(x) matrix(NA, ncol=length(CDComp4)*2, nrow=length(x)))
CDismr3 <- lapply(1:length(OutputMatrix), function(x){ row.names(OutputMatrix[[x]])<-names(ListGSC[[x]]); OutputMatrix[[x]]})
CDismrB=CDismr3
for (i in 1:length(CDComp4)){

hits=CDCompSig[[i]]
hits=cdGenesConv[match(hits, rownames(CDdds))]
fcTab=CDComp4[[i]]$log2FoldChange
names(fcTab)=cdGenesConv


if (length(hits)>3){
  gsca=GSCA(listOfGeneSetCollections=ListGSC,geneList=fcTab, hits = hits)
gsca1 <- preprocess(gsca, species="Hs", initialIDs="SYMBOL",
                    keepMultipleMappings=TRUE, duplicateRemoverMethod="max",
                    orderAbsValue=FALSE)
  gsca2 <- analyze(gsca1, 
                 para=list(pValueCutoff=0.05, pAdjustMethod="BH",
                           nPermutations=100, minGeneSetSize=5,
                           exponent=1), 
                           doGSOA = T)
}else{
  gsca=GSCA(listOfGeneSetCollections=ListGSC,geneList=fcTab)
gsca1 <- preprocess(gsca, species="Hs", initialIDs="SYMBOL",
                    keepMultipleMappings=TRUE, duplicateRemoverMethod="max",
                    orderAbsValue=FALSE)
  gsca2 <- analyze(gsca1, 
                 para=list(pValueCutoff=0.05, pAdjustMethod="BH",
                           nPermutations=100, minGeneSetSize=5,
                           exponent=1), 
                           doGSOA = F)
}

for (j in 1:length(ListGSC)){
  x1=gsca2@result$GSEA.results[[j]]
  mx1=match(rownames(x1), rownames(CDismr3[[j]]))
  CDismr3[[j]][ mx1,(2*i-1):(2*i)]=data.matrix(x1[, c(1, 3)])
}

if (length(hits)>3){
for (j in 1:length(ListGSC)){
  x1=gsca2@result$HyperGeo.results[[j]]
  mx1=match(rownames(x1), rownames(CDismrB[[j]]))
  CDismrB[[j]][ mx1,(2*i-1):(2*i)]=data.matrix(x1[, c(6, 7)])
}}

}

CDismr3 <- lapply(CDismr3, function(x){ colnames(x)<- paste(rep(names(CDComp4), each=2), c("NES", "padj")); x})
names(CDismr3)=names(ListGSC)
CDismr3<-lapply(CDismr3, function(x) as.data.frame(cbind(GeneSet=rownames(x), x)))
write_xlsx(CDismr3, path=sprintf("outputs/DESeq/GSEA_CD_growth_treat.xlsx"))


CDismrB <- lapply(CDismrB, function(x){ colnames(x)<- paste(rep(names(CDComp4), each=2), c("pval", "padj")); x})
names(CDismrB)=names(ListGSC)
CDismrB<-lapply(CDismrB, function(x) as.data.frame(cbind(GeneSet=rownames(x), x)))
write_xlsx(CDismrB, path=sprintf("outputs/DESeq/HyperGeom_CD_growth_treat.xlsx"))
```


```{r dn-samples-gsea, cache=T, results='hide', warning=F, message=F}
#load("rslt/DESeq/Epithelial_fraction.RData")

table(DNdds$Comp4)

xsearch=c("stablecontrol","growingimm",  "stableimm", "stableimm")
ysearch=c("growingcontrol","growingcontrol",  "stablecontrol", "growingimm")

## for treatment control: big differences here which are not replicated

DNComp4=list()
DNCompSig=list()
DNUpDn1=matrix(NA, nrow=2, ncol=21)

for (i in 1:4){
Eres1=results(DNdds, contrast=c("Comp4", xsearch[i], ysearch[i]))
DNComp4[[i]]=as.data.frame(cbind(Gene=rownames(Eres1), Eres1))
DNCompSig[[i]]=sort(rownames(Eres1)[which(Eres1$padj<0.1)])
DNUpDn1[ ,i]=c(length(which(Eres1$padj<0.1 & Eres1$log2FoldChange<0)),
             length(which(Eres1$padj<0.1 & Eres1$log2FoldChange>0)))
}
# for control

cList=c("PDL1+LY", "LY", "PDL1")
for (i in 1:3){
Eres5=results(DNddsTreat, contrast=c("Treatment", cList[i], "Vehicle"))
DNComp4[[i+4]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[i+4]]=g1
DNUpDn1[ ,i+4]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}

xsearchA=c("stable.Vehicle","stable.LY",  "stable.PDL1+LY", "stable.PDL1")
ysearchB=c("growing.Vehicle","growing.LY",  "growing.PDL1+LY", "growing.PDL1")


for (i in 1:4){
Eres5=results(DNddsTreatG, contrast=c("Comp8", xsearchA[i], ysearchB[i]))
DNComp4[[i+7]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[i+7]]=g1
DNUpDn1[ ,i+7]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}

boxplot(assay(DNdds)["Snca", ]~DNdds$treatA+DNdds$Growth)

Eres5=results(DNddsGrowth, contrast=c("Growth", "stable", "growing"))
DNComp4[[12]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[12]]=g1
DNUpDn1[ ,12]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))


Eres5=results(DNddsStrMH, contrast=c("MHcut", "res", "inf"))
DNComp4[[13]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[13]]=g1
DNUpDn1[ ,13]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))


Eres5=results(DNddsStrIF, contrast=c("IFcut", "res", "inf"))
DNComp4[[14]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[14]]=g1
DNUpDn1[ ,14]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(DNddsStrknn, contrast=c("knncut", "res", "inf"))
DNComp4[[15]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[15]]=g1
DNUpDn1[ ,15]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(DNddscd8, contrast=c("CD8FracCut", "low", "high"))
DNComp4[[16]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[16]]=g1
DNUpDn1[ ,16]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

Eres5=results(DNddsSpatMan, contrast=c("SpatialManual", "Infiltrating", "restricted"))
DNComp4[[17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[17]]=g1
DNUpDn1[ ,17]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))

xsearchC=c("low.inf","high.inf",  "low.inf", "low.res")
ysearchD=c("low.res","high.res", "high.inf", "high.res")

for (i in 1:4){
Eres5=results(DNddscd8MH, contrast=c("cd8MH", xsearchC[i], ysearchD[i]))
DNComp4[[i+17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))
g1=sort(rownames(Eres5)[which(Eres5$padj<0.1)])
DNCompSig[[i+17]]=g1
DNUpDn1[ ,i+17]=c(length(which(Eres5$padj<0.1 & Eres5$log2FoldChange<0)),
             length(which(Eres5$padj<0.1 & Eres5$log2FoldChange>0)))
}

rownames(DNUpDn1)=c("down", "up")
colnames(DNUpDn1)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")

  #c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"),"stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF")


names(DNComp4)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")

  #c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"),"stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF")
names(DNCompSig)=c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"), "stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF", "res_vs_inf.knn", "lo_vs_hi_cd8", "inf_vs_res.SpatMan", "low.inf_vs_low.res.cd8.mh",  "hi.inf_vs_hi.res.cd8.mh",  "low.inf_vs_hi.inf.cd8.mh",  "low.res_vs_hi.res.cd8.mh")

#c(paste(xsearch, "vs", ysearch, sep="_"), paste(cList, "_vs_Vehicle", sep=""), paste(xsearchA, "vs", ysearchB, sep="_"),"stable_vs_growing", "res_vs_inf.MH", "res_vs_inf.IF")

## write to a xls
write_xlsx(DNComp4, path=sprintf("outputs/DN_growth_treat.xlsx"))

OutputMatrix=lapply( ListGSC, function(x) matrix(NA, ncol=length(DNComp4)*2, nrow=length(x)))
DNismr3 <- lapply(1:length(OutputMatrix), function(x){ row.names(OutputMatrix[[x]])<-names(ListGSC[[x]]); OutputMatrix[[x]]})
DNismrB=DNismr3

for (i in 1:length(DNComp4)){

hits=DNCompSig[[i]]
hits=dnGenesConv[match(hits, rownames(DNdds))]
fcTab=DNComp4[[i]]$log2FoldChange
names(fcTab)=dnGenesConv

gsca=GSCA(listOfGeneSetCollections=ListGSC,geneList=fcTab, hits = hits)
gsca1 <- preprocess(gsca, species="Hs", initialIDs="SYMBOL",
                    keepMultipleMappings=TRUE, duplicateRemoverMethod="max",
                    orderAbsValue=FALSE)
if (length(hits)>0){
  gsca2 <- analyze(gsca1, 
                 para=list(pValueCutoff=0.05, pAdjustMethod="BH",
                           nPermutations=100, minGeneSetSize=5,
                           exponent=1), 
                           doGSOA = T)
}else{
  gsca2 <- analyze(gsca1, 
                 para=list(pValueCutoff=0.05, pAdjustMethod="BH",
                           nPermutations=100, minGeneSetSize=5,
                           exponent=1), 
                           doGSOA = F)
}

for (j in 1:length(ListGSC)){
  x1=gsca2@result$GSEA.results[[j]]
  mx1=match(rownames(x1), rownames(DNismr3[[j]]))
  DNismr3[[j]][ mx1,(2*i-1):(2*i)]=data.matrix(x1[, c(1, 3)])
}


if (length(hits)>2){
for (j in 1:length(ListGSC)){
  x1=gsca2@result$HyperGeo.results[[j]]
  mx1=match(rownames(x1), rownames(DNismrB[[j]]))
  DNismrB[[j]][ mx1,(2*i-1):(2*i)]=data.matrix(x1[, c(6, 7)])
}}


}

DNismr3 <- lapply(DNismr3, function(x){ colnames(x)<- paste(rep(names(DNComp4), each=2), c("NES", "padj")); x})
names(DNismr3)=names(ListGSC)
DNismr3<-lapply(DNismr3, function(x) as.data.frame(cbind(GeneSet=rownames(x), x)))
write_xlsx(DNismr3, path=sprintf("outputs/GSEA_DN_growth_treat.xlsx"))

DNismrB <- lapply(DNismrB, function(x){ colnames(x)<- paste(rep(names(DNComp4), each=2), c("pval", "padj")); x})
names(DNismrB)=names(ListGSC)
DNismrB<-lapply(DNismrB, function(x) as.data.frame(cbind(GeneSet=rownames(x), x)))
write_xlsx(DNismrB, path=sprintf("outputs/DESeq/HyperGeom_DN_growth_treat.xlsx"))


save(DNComp4, EpComp4, CDComp4, DNismr3,Epismr3, CDismr3,DNismrB, EpismrHG, CDismrB,
     DNCompSig, EpCompSig, CDCompSig, UpDn1, CDUpDn1, DNUpDn1, file="outputs/all_differential_comparisons.RData")
```


```{r, eval=F}

load("anntotations/Rat_biomart_gene_annotations.RData")

PathInH=getGmt(con="anntotations/h.all.v7.1.symbols.gmt", geneIdType=SymbolIdentifier(),
                collectionType=BroadCollection(category="h"))

HallmarkNames=unique(unlist(geneIds(PathInH)))

## change the HallmarkNames to rat

RatHallmark=SymHum2Rat$RGD.symbol[match(HallmarkNames, SymHum2Rat$HGNC.symbol)]
RatHallmark2=Rat2Hum$RGD.symbol[match(HallmarkNames, Rat2Hum$HGNC.symbol)]

RatUn=na.omit(unique(c(RatHallmark, RatHallmark2)))

midx=sapply(EpCompSig, function(x) intersect(x, RatUn))

mUnique2=unique(unlist(midx))

ColsideCols=DiffCols[EpddsTreatG$Comp8_v2]
ColsideColsB=hue_pal()(4)[EpddsTreatG$Treatment]
ColsideColsC=hue_pal()(2)[EpddsTreatG$Growth]

colTable=rbind(ColsideCols, ColsideColsB, ColsideColsC)
rownames(colTable)=c("Treat+growth", "Treatment", "growth")

x1=match(c("2N__Ep", "NMU12_RL_Ep", "10L_D_Ep", "3N_B_Ep"), colnames(vstEp))

heatmap.plus(assay(vstEp)[mUnique2, ], trace="none", scale="none", col=RdBu[11:1], ColSideColors = t(colTable), main=sprintf("Hallmark genes %s", SampleFrac))


```

<!--chapter:end:05-deseqruns.Rmd-->

# DESeq analysis

## Summary of comparisons

```{r}
#load("rslt/DESeq/all_differential_comparisons.RData")
#load("outputs/subfraction_analysis_2020-11-07.RData")
```

Quick plot of the differences in number of differential genes:

```{r}
Allchanges=rbind(UpDn1, CDUpDn1, DNUpDn1)
Allchanges=cbind(Allchanges, Frac=c("Ep", "Ep", "CD", "CD", "DN", "DN"), dir=rownames(Allchanges))

AllchangesB=melt(as.data.frame(Allchanges), id.vars = c("Frac", "dir"))

ggplot(AllchangesB, aes(x=variable, y=as.numeric(value), fill=dir))+geom_bar(stat="identity",position = "dodge")+facet_grid(Frac~., scale="free_y", space="free_y")+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ylab("No. differential genes")+ggtitle("number of differential genes")+scale_y_log10()
## plot the number of samples in each comparison

Nsamp=matrix(NA, nrow=6, ncol=ncol(Allchanges))
ta=table(Epdds$Comp4)
tb=table(CDdds$Comp4)
tc=table(DNdds$Comp4)

for (i in 1:4){
  Nsamp[1:2, i]=c(ta[xsearch[i]], ta[ysearch[i]])
  Nsamp[3:4, i]=c(tb[xsearch[i]], tb[ysearch[i]])
  Nsamp[5:6, i]=c(tc[xsearch[i]], tc[ysearch[i]])
}

ta=table(Epdds$Treatment)
tb=table(CDdds$Treatment)
tc=table(DNdds$Treatment)

for (i in 1:3){
  Nsamp[1:2, i+4]=c(ta[cList[i]], ta["Vehicle"])
  Nsamp[3:4, i+4]=c(tb[cList[i]], tb["Vehicle"])
  Nsamp[5:6, i+4]=c(tc[cList[i]], tc["Vehicle"])
}

ta=table(EpddsTreatG$Comp8)
tb=table(CDddsTreatG$Comp8)
tc=table(DNddsTreatG$Comp8)

for (i in 1:4){
  Nsamp[1:2, i+7]=c(ta[xsearchA[i]], ta[ysearchB[i]])
  Nsamp[3:4, i+7]=c(tb[xsearchA[i]], tb[ysearchB[i]])
  Nsamp[5:6, i+7]=c(tc[xsearchA[i]], tc[ysearchB[i]])
}

Nsamp[ ,12]=c(table(EpddsGrowth$Growth), table(CDddsGrowth$Growth), table(DNddsGrowth$Growth))
Nsamp[ ,13]=c(table(EpddsStrMH$MHcut), table(CDddsStrMH$MHcut), table(DNddsStrMH$MHcut))
Nsamp[ ,14]=c(table(EpddsStrIF$IFcut), table(CDddsStrIF$IFcut), table(DNddsStrIF$IFcut))
Nsamp[ ,15]=c(table(EpddsStrknn$knncut), table(CDddsStrknn$knncut), table(DNddsStrknn$knncut))
Nsamp[ ,16]=c(table(Epddscd8$CD8FracCut), table(CDddscd8$CD8FracCut), table(DNddscd8$CD8FracCut))
Nsamp[ ,17]=c(table(EpddsSpatMan$SpatialManual), table(CDddsSpatMan$SpatialManual), table(DNddsSpatMan$SpatialManual))

ta=table(Epddscd8MH$cd8MH)
tb=table(CDddscd8MH$cd8MH)
tc=table(DNddscd8MH$cd8MH)


for (i in 1:4){
  Nsamp[1:2, i+17]=c(ta[xsearchC[i]], ta[ysearchD[i]])
  Nsamp[3:4, i+17]=c(tb[xsearchC[i]], tb[ysearchD[i]])
  Nsamp[5:6, i+17]=c(tc[xsearchC[i]], tc[ysearchD[i]])
}

colnames(Nsamp)=colnames(Allchanges)
Nsamp[ , 22:23]=Allchanges[ ,22:23]


Nsamp2=melt(as.data.frame(Nsamp), id.vars=c("Frac", "dir"))

ggplot(Nsamp2, aes(x=variable, y=as.numeric(value), fill=dir))+geom_bar(stat="identity",position = "dodge")+facet_grid(Frac~., scale="free_y", space="free")+theme(axis.text.x = element_text(angle = 90, hjust = 1)) + scale_fill_discrete(name = "Comparison", labels = c("numerator", "reference"))+ylab("No. samples in comparison")+ggtitle("number of samples in comparison")

#dev.off()
```

Notes:

Comparison Type | Main cell type impacted
----------------|-----------------------
Treatment | CD45+, PDL1+LY in the DN, LY may affect Ep
growth | Ep and CD45
growth in control | CD and DN
Spatial Patterns | CD45, Ep
Spatial, high cd8 | all cases

## Summary of comaprisons

Pearson correlation:

* Pearson correlation of all the samples
* Unlist all genes, and pick out genes which are impacted in at least 1 comparison, and base expression >100
* columns are colored by: Spatial Distribution (MH),  Growth, Treatment, CD8 content

Differential genes:

* top 100 variable genes
* DEG in at least 2 samples
* differential genes in each comparison shown

Gene sets:

* c2, c5, Process networks, metacore, hallmark

### EPCAM samples

Note that there are a subset of samples different from the other based on CD8, growth and MH index (6RB, 3NB, 10LD, 8LD, 11ND) but these parameters don't group the other samples well


There is a small subset of samples which have high expression of CCls and based on the top variable genes, there seems to be three types of epithelial cases. 

```{r}
SampleFrac="Ep"

allEpGenes=unlist(EpCompSig)
allEpTab=table(allEpGenes)
EpGenesList=names(allEpTab)[which(allEpTab>1)]
#EpGenesList=EpGenesList[-grep("LOC", EpGenesList)]
#EpGenesList=EpGenesList[-grep("RGD", EpGenesList)]
# also, remove all samples which have a base mean less than 10
x1=EpComp4[[1]]$baseMean[match(EpGenesList, EpComp4[[1]]$Gene)]
x2=which(x1>100)
vstEp=vst(Epdds, blind=T)

#EpddsTreatG$Comp8_v2=factor(paste(EpddsTreatG$Treatment, EpddsTreatG$Growth))

ColsideCols=hue_pal()(2)[Epdds$CD8FracCut]
ColsideColsB=hue_pal()(4)[Epdds$Treatment]
ColsideColsC=hue_pal()(2)[Epdds$Growth]
ColsideColsD=hue_pal()(2)[Epdds$MHcut]

colTable=rbind(ColsideCols, ColsideColsB, ColsideColsC, ColsideColsD)
rownames(colTable)=c("CD8", "Treatment", "growth", "MH")

#pdf(sprintf("rslt/DESeq/DEseq_comparisons_%s.pdf", SampleFrac), width=7, height=10)

sampleDist=dist(t(assay(vstEp)))
sampleDistMatrix <- as.matrix(sampleDist)
heatmap.plus(sampleDistMatrix, trace="none", scale="none", col=brewer.pal(9, "Blues")[9:1], ColSideColors = t(colTable), main=sprintf("Similarity between %s samples", SampleFrac))

# Take the top 150 variable genes 
# compute the sds for each genes
sds = rowSds(assay(vstEp))
means=rowMeans(assay(vstEp))
CVs=sds/means
CVs=CVs[-which(means<7)]
sds2=sort(CVs, decreasing = T)

heatmap.plus(assay(vstEp)[names(sds2)[1:100], ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable), main=sprintf("top 100 variable genes all %s samples", SampleFrac))


heatmap.plus(assay(vstEp)[EpGenesList[x2], ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable),
          main=sprintf("DEG at least 2 %s samples", SampleFrac))

# Make a table for these genes
FCvalues=matrix(NA, nrow=length(x2), ncol=length(EpComp4))

for (i in 1:length(EpComp4)){
  FCvalues[ ,i]=EpComp4[[i]]$log2FoldChange[match(EpGenesList[x2], EpComp4[[i]]$Gene)]
}

colnames(FCvalues)=names(EpComp4)
rownames(FCvalues)=EpGenesList[x2]


# for each list pick the top 50 diff genes FC>2 and baseman > 50
glist=NA
for (i in 1:length(EpComp4)){
  glist=c(glist, EpComp4[[i]]$Gene[which(EpComp4[[i]]$padj<0.1 & abs(EpComp4[[i]]$log2FoldChange)>2 & EpComp4[[i]]$baseMean>25)])
}

ColsideCols=SetCols[EpddsTreatG$Comp8]

glistmod=unique(glist)
glistmod=glistmod[-grep("LOC", glistmod)]
glistmod=na.omit(glistmod[-grep("RGD", glistmod)])

#heatmap.plus(assay(vstEp)[glistmod, ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable),  main=sprintf("All DEG %s samples", SampleFrac))
```

We can also view this in the form of what is highly expressed in each comparison. Below is a heatmap of the fold changes:

```{r}
# distance of the samples to one another
par(oma=c(1,0,0,0))
heatmap.2(FCvalues, trace="none", scale="row", col=RdBu[11:1], 
          main=sprintf("All comparisons %s, log2FC for each gene", SampleFrac))
#dev.off()
```

Below, we visualise the enrichment scores using different gene sets and 

```{r}
#Epismr3=ismr4

#pdf(sprintf("rslt/DESeq/GSEA_signature_comparisons_%s.pdf", SampleFrac), width=10, height=12)

for (i in 1:length(Epismr3)){
  x1=Epismr3[[i]]
  # find padj<0.1
  idx2=lapply(seq(3, ncol(x1), by=2), function(x) which(as.numeric(as.character(x1[ ,x]))<0.01))
  unList=table(unlist(idx2))
  #pickThese=names(unList)#[which(unList>1)]
  pickThese2=names(unList)[which(unList>1)]
  
  if (length(pickThese2)>100){
    temp=x1[as.numeric(pickThese2), seq(2, ncol(x1), by=2)]
    temp2=sapply(temp, function(x) as.numeric(as.character(x)))
    temp2=rowSds(abs(temp2))
    names(temp2)=pickThese2
    temp2=sort(temp2, decreasing = T)
    pickThese2=names(temp2)[1:100]
  }
  
  x23=x1[as.numeric(pickThese2), seq(2, ncol(x1), by=2)]
  
  x234=sapply(x23, function(x) as.numeric(as.character(x)))
  rownames(x234)=rownames(x23)
  
  par(oma=c(2,0,0,2))
  heatmap.2(x234, col=RdBu[11:1], scale="none", trace="none", main=sprintf("%s pathways %s fraction", 
                                                                           names(Epismr3)[i], SampleFrac))
}

#dev.off()

# boxplot(assay(vstEp)["Tap1", ]~EpddsTreatG$Treatment)
# boxplot(assay(vstEp)["Cd74", ]~EpddsTreatG$Comp4)
# boxplot(assay(vstEp)["Cd8a", ]~EpddsTreatG$Treatment)
# boxplot(assay(vstEp)["Ctla4", ]~EpddsTreatG$Treatment)
# boxplot(assay(vstEp)["Tgfbr1", ]~EpddsTreatG$Comp8)
```

### CD45 comparisons

```{r}
SampleFrac="CD"

allCDGenes=unlist(CDCompSig)
allCDTab=table(allCDGenes)
CDGenesList=names(allCDTab)[which(allCDTab>1)]
#CDGenesList=CDGenesList[-grCD("LOC", CDGenesList)]
#CDGenesList=CDGenesList[-grCD("RGD", CDGenesList)]
# also, remove all samples which have a base mean less than 10
x1=CDComp4[[1]]$baseMean[match(CDGenesList, CDComp4[[1]]$Gene)]
x2=which(x1>25)
vstCD=vst(CDdds, blind=T)

CDddsTreatG$Comp8_v2=factor(paste(CDddsTreatG$Treatment, CDddsTreatG$Growth))

ColsideCols=DiffCols[CDddsTreatG$Comp8_v2]
ColsideColsB=hue_pal()(4)[CDddsTreatG$Treatment]
ColsideColsC=hue_pal()(2)[CDddsTreatG$Growth]

colTable=rbind(ColsideCols, ColsideColsB, ColsideColsC)
rownames(colTable)=c("Treat+growth", "Treatment", "growth")



#pdf(sprintf("rslt/DESeq/DEseq_comparisons_%s.pdf", SampleFrac), width=7, height=10)

sampleDist=dist(t(assay(vstCD)))
sampleDistMatrix <- as.matrix(sampleDist)
heatmap.plus(sampleDistMatrix, trace="none", scale="none", col=brewer.pal(11, "Blues")[11:1], ColSideColors = t(colTable), main=sprintf("Similarity between %s samples", SampleFrac))

# Take the top 150 variable genes 
# compute the sds for each genes
sds = rowSds(assay(vstCD))
means=rowMeans(assay(vstCD))
CVs=sds/means
CVs=CVs[-which(means<7)]
sds2=sort(CVs, decreasing = T)

heatmap.plus(assay(vstCD)[names(sds2)[1:200], ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable), main=sprintf("top 200 variable genes all %s samples", SampleFrac))


heatmap.plus(assay(vstCD)[CDGenesList[x2], ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable),
          main=sprintf("DEG at least 2 %s samples", SampleFrac))

# Make a table for these genes
FCvalues=matrix(NA, nrow=length(x2), ncol=length(CDComp4))

for (i in 1:length(CDComp4)){
  FCvalues[ ,i]=CDComp4[[i]]$log2FoldChange[match(CDGenesList[x2], CDComp4[[i]]$Gene)]
}

colnames(FCvalues)=names(CDComp4)
rownames(FCvalues)=CDGenesList[x2]


# for each list pick the top 50 diff genes FC>2 and baseman > 50
glist=NA
for (i in 1:length(CDComp4)){
  glist=c(glist, CDComp4[[i]]$Gene[which(CDComp4[[i]]$padj<0.1 & abs(CDComp4[[i]]$log2FoldChange)>2 & CDComp4[[i]]$baseMean>25)])
}

ColsideCols=SetCols[CDddsTreatG$Comp8]

glistmod=unique(glist)
#glistmod=glistmod[-grep("LOC", glistmod)]
#glistmod=na.omit(glistmod[-grCD("RGD", glistmod)])

heatmap.plus(assay(vstCD)[na.omit(glistmod), ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable),
          main=sprintf("All DEG %s samples", SampleFrac))

# distance of the samples to one another
par(oma=c(2,0,0,0))
heatmap.2(FCvalues, trace="none", scale="row", col=RdBu[11:1], 
          main=sprintf("All comparisons %s, log2FC for each gene", SampleFrac))
#dev.off()

#CDismr3=ismr4

#pdf(sprintf("rslt/DESeq/GSEA_signature_comparisons_%s.pdf", SampleFrac), width=10, height=12)

for (i in 1:length(CDismr3)){
  x1=CDismr3[[i]]
  # find padj<0.1
  idx2=lapply(seq(3, ncol(x1), by=2), function(x) which(as.numeric(as.character(x1[ ,x]))<0.01))
  unList=table(unlist(idx2))
  #pickThese=names(unList)#[which(unList>1)]
  pickThese2=names(unList)[which(unList>1)]
  
  if (length(pickThese2)>100){
    temp=x1[as.numeric(pickThese2), seq(2, ncol(x1), by=2)]
    temp2=sapply(temp, function(x) as.numeric(as.character(x)))
    temp2=rowSds(abs(temp2))
    names(temp2)=pickThese2
    temp2=sort(temp2, decreasing = T)
    pickThese2=names(temp2)[1:100]
  }
  
  x23=x1[as.numeric(pickThese2), seq(2, ncol(x1), by=2)]
  
  x234=sapply(x23, function(x) as.numeric(as.character(x)))
  rownames(x234)=rownames(x23)
  
  par(oma=c(2,0,0,2))
  heatmap.2(x234, col=RdBu[11:1], scale="none", trace="none", main=sprintf("%s pathways %s fraction", 
                                                                           names(CDismr3)[i], SampleFrac))
}

#dev.off()


```

### DN samples

```{r}
SampleFrac="DN"

allDNGenes=unlist(DNCompSig)
allDNTab=table(allDNGenes)
DNGenesList=names(allDNTab)[which(allDNTab>1)]
#DNGenesList=DNGenesList[-grep("LOC", DNGenesList)]
#DNGenesList=DNGenesList[-grep("RGD", DNGenesList)]
# also, remove all samples which have a base mean less than 10
x1=DNComp4[[1]]$baseMean[match(DNGenesList, DNComp4[[1]]$Gene)]
x2=which(x1>25)
vstDN=vst(DNdds, blind=T)

DNddsTreatG$Comp8_v2=factor(paste(DNddsTreatG$Treatment, DNddsTreatG$Growth))

ColsideCols=DiffCols[DNddsTreatG$Comp8_v2]
ColsideColsB=hue_pal()(4)[DNddsTreatG$Treatment]
ColsideColsC=hue_pal()(2)[DNddsTreatG$Growth]

colTable=rbind(ColsideCols, ColsideColsB, ColsideColsC)
rownames(colTable)=c("Treat+growth", "Treatment", "growth")



#pdf(sprintf("rslt/DESeq/DEseq_comparisons_%s.pdf", SampleFrac), width=7, height=10)

sampleDist=dist(t(assay(vstDN)))
sampleDistMatrix <- as.matrix(sampleDist)
heatmap.plus(sampleDistMatrix, trace="none", scale="none", col=brewer.pal(11, "Blues")[11:1], ColSideColors = t(colTable), main=sprintf("Similarity between %s samples", SampleFrac))

# Take the top 150 variable genes 
# compute the sds for each genes
sds = rowSds(assay(vstDN))
means=rowMeans(assay(vstDN))
CVs=sds/means
CVs=CVs[-which(means<7)]
sds2=sort(CVs, decreasing = T)

heatmap.plus(assay(vstDN)[names(sds2)[1:200], ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable), main=sprintf("top 200 variable genes all %s samples", SampleFrac))


heatmap.plus(assay(vstDN)[DNGenesList[x2], ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable),
          main=sprintf("DEG at least 2 %s samples", SampleFrac))

# Make a table for these genes
FCvalues=matrix(NA, nrow=length(x2), ncol=length(DNComp4))

for (i in 1:length(DNComp4)){
  FCvalues[ ,i]=DNComp4[[i]]$log2FoldChange[match(DNGenesList[x2], DNComp4[[i]]$Gene)]
}

colnames(FCvalues)=names(DNComp4)
rownames(FCvalues)=DNGenesList[x2]


# for each list pick the top 50 diff genes FC>2 and baseman > 50
glist=NA
for (i in 1:length(DNComp4)){
  glist=c(glist, DNComp4[[i]]$Gene[which(DNComp4[[i]]$padj<0.1 & abs(DNComp4[[i]]$log2FoldChange)>2 & DNComp4[[i]]$baseMean>25)])
}

ColsideCols=SetCols[DNddsTreatG$Comp8]

glistmod=na.omit(unique(glist))
#glistmod=glistmod[-grep("LOC", glistmod)]
#glistmod=na.omit(glistmod[-grep("RGD", glistmod)])

heatmap.plus(assay(vstDN)[glistmod, ], trace="none", scale="row", col=RdBu[11:1], ColSideColors = t(colTable),
          main=sprintf("All DEG %s samples", SampleFrac))

# distance of the samples to one another
par(oma=c(2,0,0,0))
heatmap.2(FCvalues, trace="none", scale="row", col=RdBu[11:1], 
          main=sprintf("All comparisons %s, log2FC for each gene", SampleFrac))
#dev.off()

#DNismr3=ismr4

#pdf(sprintf("rslt/DESeq/GSEA_signature_comparisons_%s.pdf", SampleFrac), width=10, height=12)

for (i in 1:length(DNismr3)){
  x1=DNismr3[[i]]
  # find padj<0.1
  idx2=lapply(seq(3, ncol(x1), by=2), function(x) which(as.numeric(as.character(x1[ ,x]))<0.01))
  unList=table(unlist(idx2))
  #pickThese=names(unList)#[which(unList>1)]
  pickThese2=names(unList)[which(unList>1)]
  
  if (length(pickThese2)>100){
    temp=x1[as.numeric(pickThese2), seq(2, ncol(x1), by=2)]
    temp2=sapply(temp, function(x) as.numeric(as.character(x)))
    temp2=rowSds(abs(temp2))
    names(temp2)=pickThese2
    temp2=sort(temp2, decreasing = T)
    pickThese2=names(temp2)[1:100]
  }
  
  x23=x1[as.numeric(pickThese2), seq(2, ncol(x1), by=2)]
  
  x234=sapply(x23, function(x) as.numeric(as.character(x)))
  rownames(x234)=rownames(x23)
  
  par(oma=c(2,0,0,2))
  heatmap.2(x234, col=RdBu[11:1], scale="none", trace="none", main=sprintf("%s pathways %s fraction", 
                                                                           names(DNismr3)[i], SampleFrac))
}

#dev.off()

```

## Growing vs stable emphasis

Below, we focus specifically on the growing vs stable comparison in greater depth.
Here, we look at the 3 different fractions in greater depth and look at volcano plots of DEG and heatmaps of differential genes. Samples are colored according to whether they are growing or stable

### DN fraction

```{r}

#pdf(sprintf("rslt/DESeq/volcano_plots_differences_stable_vs_growing_%s.pdf", Sys.Date()), height=7, width=8)

DNa1=DNComp4$stable_vs_growing
DNa=DNComp4$stable_vs_growing[ which(DNComp4$stable_vs_growing$padj<0.1 & DNComp4$stable_vs_growing$baseMean>50 & abs(DNComp4$stable_vs_growing$log2FoldChange)>1.5), ]
with(DNa1, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot: DN Stable (+) vs Growing (-)", cex=1.0, xlab=bquote(~Log[2]~fold~change), ylab=bquote(~-log[10]~Q~value)))
with(subset(DNa1, padj<0.1 & abs(log2FoldChange)>1.5), points(log2FoldChange, -log10(padj), pch=20, col="red", cex=0.5))
with(subset(DNa1, padj<0.1 & abs(log2FoldChange)>1.5), text(log2FoldChange+0.05, -log10(padj)+0.05, Gene, pch=20, col="red", cex=0.75))


heatmap.2(assay(vstDN)[ DNa$Gene, ], scale="row", trace="none", ColSideColors = as.character(as.numeric(vstDN$Growth)), col=RdBu[11:1], main="DN genes")
```

### CD fraction

```{r}
CDa1=CDComp4$stable_vs_growing
CDa=CDComp4$stable_vs_growing[ which(CDComp4$stable_vs_growing$padj<0.05 & CDComp4$stable_vs_growing$baseMean>100 & abs(CDComp4$stable_vs_growing$log2FoldChange)>2), ]
with(CDa1, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot: CD Stable (+) vs Growing (-)", cex=1.0, xlab=bquote(~Log[2]~fold~change), ylab=bquote(~-log[10]~Q~value)))
with(subset(CDa1, padj<0.1 & abs(log2FoldChange)>1.5), points(log2FoldChange, -log10(padj), pch=20, col="red", cex=0.5))
with(subset(CDa1, padj<0.1 & abs(log2FoldChange)>1.5), text(log2FoldChange+0.05, -log10(padj)+0.05, Gene, pch=20, col="red", cex=0.75))

heatmap.2(assay(vstCD)[ CDa$Gene, ], scale="row", trace="none", ColSideColors = as.character(as.numeric(vstCD$Growth)), col=RdBu[11:1], main="CD genes")
```

### Ep Fraction

```{r}
Epa1=EpComp4$stable_vs_growing
Epa=EpComp4$stable_vs_growing[ which(EpComp4$stable_vs_growing$padj<0.1 & EpComp4$stable_vs_growing$baseMean>50 & abs(EpComp4$stable_vs_growing$log2FoldChange)>1.5), ]
with(Epa1, plot(log2FoldChange, -log10(padj), pch=20, main="Volcano plot: Ep Stable vs Growing", cex=1.0, xlab=bquote(~Log[2]~fold~change), ylab=bquote(~-log[10]~Q~value)))
with(subset(Epa1, padj<0.1 & abs(log2FoldChange)>1.5), points(log2FoldChange, -log10(padj), pch=20, col="red", cex=0.5))
with(subset(Epa1, padj<0.1 & abs(log2FoldChange)>1.5), text(log2FoldChange+0.05, -log10(padj)+0.1, Gene, pch=20, col="red", cex=0.75))

heatmap.2(assay(vstEp)[ Epa$Gene, ], scale="row", trace="none", ColSideColors = as.character(as.numeric(vstEp$Growth)), col=RdBu[11:1], main="Ep genes")


#dev.off()

```

### Specific pathways

The above plots suggest that there could be a different in growing and stable based on :

* inflammation
* Kras signalling
* MHC presentation
* checkpoint proteins

We can pull out the genes in these sets and visualise the relative expression in a heatmap in the DN, CD45 and Ep samples:

Note that red is growing and green is stable

```{r}
# library(org.Hs.eg.db)
# load("../anntotations/ListofGeneSets2.RData")

SetNamesc2=names(PathInc2) #names(ListGSC$c2List)
x1=grep("MHC", SetNamesc2)
GeneNames=unique(unlist(geneIds(PathInc2[x1[c(1, 4,5)]])))

#test1=mapIds(org.Hs.eg.db, GeneNames, 'SYMBOL','ENTREZID')
RatGeneNamesMHC=na.omit(unique(SymHum2Rat$RGD.symbol[match(GeneNames, SymHum2Rat$HGNC.symbol)]))

#RatGeneNamesKras=mapIds(org.Hs.eg.db, ListGSC$Hallmark$HALLMARK_KRAS_SIGNALING_UP, 'SYMBOL','ENTREZID')
GeneNames=unlist(geneIds(PathInH["HALLMARK_KRAS_SIGNALING_UP"]))
RatGeneNamesKras=na.omit(unique(SymHum2Rat$RGD.symbol[match(GeneNames, SymHum2Rat$HGNC.symbol)]))

GeneNames=unlist(geneIds(PathInH["HALLMARK_INFLAMMATORY_RESPONSE"]))
RatGeneNamesInf=na.omit(unique(SymHum2Rat$RGD.symbol[match(GeneNames, SymHum2Rat$HGNC.symbol)]))

CheckpointProt=unlist(ImmSuppAPC)
RatGeneNamesCheckpoint=na.omit(unique(SymHum2Rat$RGD.symbol[match(CheckpointProt, SymHum2Rat$HGNC.symbol)]))

# RatCP=read.delim("../anntotations/immune_Suppression.csv", sep="\t", stringsAsFactors = F)
# RatCP=na.omit(unique(SymHum2Rat$RGD.symbol[match(unique(unlist(RatCP)), SymHum2Rat$HGNC.symbol)]))
# 
# 
# RatMHC=read.csv("../anntotations/MHCloss.csv", header = F, stringsAsFactors = F)[ ,1]
# RatMHCb=na.omit(unique(SymHum2Rat$RGD.symbol[match(RatMHC, SymHum2Rat$HGNC.symbol)]))
# read in checkpoint proteins
```

#### DN samples:

Note the distribution of growing/stable in DN is: `r table(DNdds$Growth)`

```{r}
ColsideColsC=hue_pal()(2)[DNdds$Growth]

heatmap.2(assay(vstDN)[na.omit(match(RatGeneNamesMHC, rownames(vstDN))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="MHC expression")

heatmap.2(assay(vstDN)[na.omit(match(RatGeneNamesKras, rownames(vstDN))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="Kras expression")

heatmap.2(assay(vstDN)[na.omit(match(RatGeneNamesInf, rownames(vstDN))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="Inflammatory response")

heatmap.2(assay(vstDN)[na.omit(match(RatGeneNamesCheckpoint, rownames(vstDN))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="checkpoint proteins")

heatmap.2(assay(vstDN)[na.omit(match(MHCPres2Rat, rownames(vstDN))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="MHC proteins")
```

#### CD45 samples:

Note the distribution of growing/stable in CD is: `r table(CDdds$Growth)`


```{r}
ColsideColsC=hue_pal()(2)[CDdds$Growth]

heatmap.2(assay(vstCD)[na.omit(match(RatGeneNamesMHC, rownames(vstCD))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="MHC expression")

heatmap.2(assay(vstCD)[na.omit(match(RatGeneNamesKras, rownames(vstCD))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="Kras expression")

heatmap.2(assay(vstCD)[na.omit(match(RatGeneNamesInf, rownames(vstCD))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="Inflammatory response")

heatmap.2(assay(vstCD)[na.omit(match(RatGeneNamesCheckpoint, rownames(vstCD))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="checkpoint proteins")

heatmap.2(assay(vstCD)[na.omit(match(MHCPres2Rat, rownames(vstCD))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="MHC proteins")
```

#### Epithelial samples:

```{r}
ColsideColsC=hue_pal()(2)[Epdds$Growth]

heatmap.2(assay(vstEp)[na.omit(match(RatGeneNamesMHC, rownames(vstEp))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="MHC expression")

heatmap.2(assay(vstEp)[na.omit(match(RatGeneNamesKras, rownames(vstEp))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="Kras expression")

heatmap.2(assay(vstEp)[na.omit(match(RatGeneNamesInf, rownames(vstEp))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="Inflammatory response")

heatmap.2(assay(vstEp)[na.omit(match(RatGeneNamesCheckpoint, rownames(vstEp))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="checkpoint proteins")

heatmap.2(assay(vstEp)[na.omit(match(MHCPres2Rat, rownames(vstEp))), ], col=RdBu[11:1], trace="none", scale="row", ColSideColors = ColsideColsC, main="MHC proteins")
```



```{r, eval=F}
### comapre DN samples: PAM50 Lum vs Basal/Normal?

DNdds$PAM50_v2="ERplus"
DNdds$PAM50_v2[which(DNdds$PAM50%in%c("Basal", "Normal"))]="ERneg"
DNdds$PAM50_v2=factor(DNdds$PAM50_v2)
design(DNdds)=~PAM50_v2+factor(Batch)
DNddsP50=DESeq(DNdds)

DNresP50=results(DNddsP50)

x1=rownames(DNresP50)[which(abs(DNresP50$log2FoldChange)>2 & DNresP50$padj<0.05)]
```

<!--chapter:end:06-deseqoutputs.Rmd-->

# ER/Pgr Subtyping

In this section, we will look at whether there are differences in rat subtype using the following approaches:

* Creating our own subtyping approach using specific markers of interest (e.g. using PAM clustering)
    * Using known gene list 
    * Using random genes (top 50, 100, 200 etc variable genes)
* PAM50 subtyping using the methods published by [Parker et al](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667820/). This method uses rsem scores to classify samples.

As the DN samples do have keratin expression, we will also try to determine the subtypes in these samples too.

In the below analyses, we will conduct this subtyping as follows:

* progression specific epithelial cohort
* characterisation specific epithelial cohort
* dn specific cohort (progression)

## In-house classifier

The selected gene list includes three lists: A. is a mroe comprehensive list containing epithelial, mesenchymal markers and proliferation markers, B. has a more narrow list of subtype specific markers 

a. 'Ar', 'Cd24', 'Cdh1', 'Foxa1', 'Gata3', 'Krt8', 'Krt18', 'Krt5', 'Vim', 'Erbb2', 'Esr1', 'Pgr', 'Mki67', 'Pcna'
b. 'Ar', 'Foxa1', 'Gata3', 'Erbb2', 'Esr1', 'Pgr'

We will be using variance stabilised counts in this section.

Firstly, we will look at the progression cohort. Note that the overlap between the two gene-sets are similar, and the heatmap is row-scaled. 

```{r}
Markers=c('Ar', 'Cd24', 'Cdh1', 'Foxa1', 'Gata3', 'Krt8', 'Krt18', 'Krt5', 'Vim', 'Erbb2', 'Esr1', 'Pgr', 'Mki67', 'Pcna')
Markers2=c('Ar', 'Foxa1', 'Gata3', 'Erbb2', 'Esr1', 'Pgr')

#pdf(sprintf("rslt/signatureAnalysis/new_classifer_PAM_%s.pdf", Sys.Date()), height=6, width=7)
# Progression cohort
Ax1=assay(vstEp)[ match(Markers, rownames(assay(vstEp))),]
Pam1=pam(t(Ax1), 2)
plot(Pam1, main="extended list")
ColA=Pam1$clustering
ColA=ifelse(ColA==1, "Basal", "Lum")

Ax2=assay(vstEp)[ match(Markers2, rownames(assay(vstEp))),]
Pam2=pam(t(Ax2), 2)
plot(Pam2, main="narrow list")
ColB=Pam2$clustering
ColB=ifelse(ColB==1, "Basal", "Lum")

table(ColA, ColB)

CTable=cbind(ColA, ColB)
CTable=gsub("Basal", "red", CTable)
CTable=gsub("Lum", "blue", CTable)

heatmap.plus(Ax1, col=brewer.pal(11, "RdBu")[11:1], ColSideColors =CTable,scale="row", main="Progression cohort vsd marker list1", trace="none")
```

We can perform a similar analysis to the characterisation cohort:

```{r}
c2=infoTableFinal$SampleID[which(infoTableFinal$Fraction=="Ep" & infoTableFinal$Cohort=="Characterisation")]
Ax1=assay(vsd)[ match(Markers, rownames(assay(vsd))), match(c2, colnames(vsd))]
Pam3=pam(t(Ax1), 2)
#plot(Pam3, "extended list")
ColC=Pam3$clustering
ColC=ifelse(ColC==1, "Basal", "Lum")


Ax2=assay(vsd)[ match(Markers2, rownames(assay(vsd))), match(c2, colnames(vsd))]
Pam4=pam(t(Ax1), 2)
#plot(Pam4, "narrow list")
ColD=Pam4$clustering
ColD=ifelse(ColD==1, "Basal", "Lum")

table(ColC, ColD)

CTable=cbind(ColC, ColD)
CTable=gsub("Basal", "red", CTable)
CTable=gsub("Lum", "blue", CTable)

heatmap.plus(Ax1, col=brewer.pal(11, "RdBu")[11:1], ColSideColors =CTable,scale="row", main="Char cohort vsd marker list1", trace="none")
```

DN samples:

```{r}
Ax1=assay(vstDN)[ match(Markers, rownames(assay(vstDN))),]
Pam5=pam(t(Ax1), 2)
plot(Pam5, main="extended list DN")
ColE=Pam5$clustering
ColE=ifelse(ColE==1, "Basal", "Lum")

Ax2=assay(vstDN)[ match(Markers2, rownames(assay(vstDN))),]
Pam6=pam(t(Ax2), 2)
plot(Pam6, main="narrow list")
ColF=Pam6$clustering
ColF=ifelse(ColF==1, "Basal", "Lum")

table(ColE, ColF)

CTable=cbind(ColE, ColF)
CTable=gsub("Basal", "red", CTable)
CTable=gsub("Lum", "blue", CTable)

heatmap.plus(Ax1, col=brewer.pal(11, "RdBu")[11:1], ColSideColors =CTable,scale="row", main="DN samples vsd marker list1", trace="none")

```

We can collate these different scores and assess whether there are similarities with PgR and ER staining:

```{r}
## match to infoTable here
infoTableFinal$Subtype_extendedList=NA
infoTableFinal$Subtype_compressedList=NA

id=match(names(ColA), infoTableFinal$Sample)
infoTableFinal$Subtype_extendedList[id]=ColA
infoTableFinal$Subtype_compressedList[id]=ColB

id2=match(names(ColE), infoTableFinal$Sample)
infoTableFinal$Subtype_extendedList[id2]=ColE
infoTableFinal$Subtype_compressedList[id2]=ColF

id3=match(names(ColC), infoTableFinal$Sample)
infoTableFinal$Subtype_extendedList[id3]=ColC
infoTableFinal$Subtype_compressedList[id3]=ColD


# Contingency Tables
par(mfrow=c(2,2))
a1=table(infoTableFinal$Subtype_compressedList, Cdata$PgR..IF.[match(infoTableFinal$TumorID, Cdata$TumorID)])
ContTable(a1[ ,-2], "Pgr Expr", T, "Pgr IF", "compressed subtype")
a1=table(infoTableFinal$Subtype_extendedList, Cdata$PgR..IF.[match(infoTableFinal$TumorID, Cdata$TumorID)])
ContTable(a1[ ,-2], "Pgr Expr", T, "Pgr IF", "Extended subtype")

# remove all the DN samples
x1=which(infoTableFinal$Fraction=="Ep")
a1=table(infoTableFinal$Subtype_compressedList[x1], Cdata$PgR..IF.[match(infoTableFinal$TumorID[x1], Cdata$TumorID)])
ContTable(a1[ ,-2], "Pgr Expr", T, "Pgr IF", "compressed subtype")
a1=table(infoTableFinal$Subtype_extendedList[x1], Cdata$PgR..IF.[match(infoTableFinal$TumorID[x1], Cdata$TumorID)])
ContTable(a1[ ,-2], "Pgr Expr", T, "Pgr IF", "Extended subtype")

# Contingency Tables
par(mfrow=c(2,2))
a1=table(infoTableFinal$Subtype_compressedList, Cdata$ER..IF.[match(infoTableFinal$TumorID, Cdata$TumorID)])
ContTable(a1[ ,-2], "ER Expr", T, "ER IF", "compressed subtype")
a1=table(infoTableFinal$Subtype_extendedList, Cdata$ER..IF.[match(infoTableFinal$TumorID, Cdata$TumorID)])
ContTable(a1[ ,-2], "ER Expr", T, "ER IF", "Extended subtype")

# remove all the DN samples
x1=which(infoTableFinal$Fraction=="Ep")
a1=table(infoTableFinal$Subtype_compressedList[x1], Cdata$ER..IF.[match(infoTableFinal$TumorID[x1], Cdata$TumorID)])
ContTable(a1[ ,-2], "ER Expr", T, "ER IF", "compressed subtype no DN")
a1=table(infoTableFinal$Subtype_extendedList[x1], Cdata$ER..IF.[match(infoTableFinal$TumorID[x1], Cdata$TumorID)])
ContTable(a1[ ,-2], "ER Expr", T, "ER IF", "Extended subtype no DN")

```


Finally, look at whether any of the above samples have similar DN and Ep calls: it appears that this call is random, and the subtype does not dictate growth rate

```{r}
idx1=substr(names(ColA), 1, nchar(names(ColA))-3)
idx2=substr(names(ColE), 1, nchar(names(ColE))-3)

a1=data.frame(nid=idx1, ep=ColA)
a2=data.frame(nid=idx2, dn=ColE)
a3=merge(a1, a2, by.x="nid", by.y="nid", all.x=T, all.y=T)
a3$growth=infoTableFinal$Growth[match(a3$nid, infoTableFinal$TumorID)]
print('extended list')
table(a3$ep, a3$dn, a3$growth)

a1=data.frame(nid=idx1, ep=ColB)
a2=data.frame(nid=idx2, dn=ColF)
a3=merge(a1, a2, by.x="nid", by.y="nid", all.x=T, all.y=T)
a3$growth=infoTableFinal$Growth[match(a3$nid, infoTableFinal$TumorID)]
print('narrow list')
table(a3$ep, a3$dn, a3$growth)
```

## PAM50 (Parker) scoring

Here, we have used rsem counts in PAM50 subtyping.

```{r}

ExpressedaprilChar=read.delim("../output4external/RSEM_for_PAM50_allExpressedgenes_noCD45_characterisationOnlyapril_2020-11-03_pam50scores.txt")
ExpressedaprilProg=read.delim("../output4external/RSEM_for_PAM50_allExpressedgenes_noCD45_progressionapril_2020-11-03_pam50scores.txt")
ExpressedaprilProg2=read.delim("../output4external/RSEM_for_PAM50_allExpressedgenes_epOnly_progressionapril_2020-11-03_pam50scores.txt")

Pall=rbind(ExpressedaprilChar, ExpressedaprilProg)
Pall$Call[match(ExpressedaprilProg2$X, Pall$X)]=ExpressedaprilProg2$Call

# add to pamcalls

Pall$PamCallLong=infoTableFinal$Subtype_extendedList[match(Pall$X, infoTableFinal$SampleID)]
Pall$PamCallshort=infoTableFinal$Subtype_compressedList[match(Pall$X, infoTableFinal$SampleID)]


```

Plot all samples together and compare the confidence and scores in different subtypes:
There is overlap between the LumB and Her2 scores, similarly Normal and LumA scores are similar

```{r}
t2=data.matrix(Pall[ ,2:6])
rownames(t2)=Pall$X
heatmap.2(t2, col=RdBu[11:1], trace="none")
```

We can pull out the main markers of interest: ER, PGR, Erbb2 and Ki67 (or PCNA). 
We also compare these expression distribution of these markers to CD45, DN and Ep overall:

Notes:

* Pgr highest in luminal samples
* Esr1 dynamic range is lower
* Ar is higher in basal samples
* Pcna is higher in Basal

```{r}
midx=match(ExpressedaprilChar$X, colnames(allTPMFinal))
midx3=match(Pall$X[grep("DN",Pall$X)], colnames(allTPMFinal))
midx2=match(ExpressedaprilProg2$X, colnames(allTPMFinal))

summ1=table(ExpressedaprilChar$Call)
summ2=table(ExpressedaprilProg2$Call)
summ3=table(Pall$Call[grep("DN",Pall$X)])

gList=c("Esr1", "Pgr", "Erbb2", "Ar", "Pcna", "Mki67")

par(mfrow=c(2,3))

for (i in gList){
boxplot(allTPMFinal[i, midx ]~ExpressedaprilChar$Call, main="Char",  names=paste(names(summ1)," N=",summ1, sep=""), las=2, ylab=i, xlab="")
boxplot(allTPMFinal[i, midx2 ]~ExpressedaprilProg2$Call, main="Prog", names=paste(names(summ2)," N=",summ2, sep=""), las=2,ylab=i, xlab="")
boxplot(allTPMFinal[i, midx3 ]~Pall$Call[grep("DN",Pall$X)], main="DN",  names=paste(names(summ3)," N=",summ3, sep=""), ylab=i, las=2, xlab="")
}

summ4=table(infoTableFinal$Fraction)

for (i in gList){
boxplot(allTPMFinal[i, ]~infoTableFinal$Fraction, main=i,  names=paste(names(summ4)," N=",summ4, sep=""), las=2, ylab=i, xlab="")
}

```

heatmap based on the previous markers:

```{r}
Ax1=assay(vsd)[ match(Markers, rownames(assay(vsd))),na.omit(midx)]
Ax1b=allRSEMnorm2log[na.omit(match(toupper(Markers), rownames(allRSEMnorm2log))), na.omit(midx)]

Ax2=assay(vsd)[ match(Markers, rownames(assay(vsd))),na.omit(midx2)]
Ax2b=allRSEMnorm2log[na.omit(match(toupper(Markers), rownames(allRSEMnorm2log))), na.omit(midx2)]

Ax3=assay(vsd)[match(Markers, rownames(assay(vsd))),na.omit(midx3)]
Ax3b=allRSEMnorm2log[na.omit(match(toupper(Markers), rownames(allRSEMnorm2log))), na.omit(midx3)]

BLcall=as.character(Pall$Call)
BLcall[grep("Lum", BLcall)]="blue"
BLcall[grep("Her2", BLcall)]="magenta"
BLcall[grep("Normal", BLcall)]="green"
BLcall[grep("Basal", BLcall)]="red"

heatmap.2(Ax1, col=brewer.pal(11, "RdBu")[11:1], ColSideColors = BLcall[which(Pall$X%in%colnames(Ax1)) ],scale="row", main="vsd marker list1", trace="none")
heatmap.2(Ax1b, col=brewer.pal(11, "RdBu")[11:1], ColSideColors = BLcall[which(Pall$X%in%colnames(Ax1)) ], trace="none", scale="row", main="RSEM marker1")

heatmap.2(Ax2, col=brewer.pal(11, "RdBu")[11:1], ColSideColors =BLcall[which(Pall$X%in%colnames(Ax2))], scale="row", main="vsd marker list2", trace="none")
heatmap.2(Ax2b, col=brewer.pal(11, "RdBu")[11:1], ColSideColors = BLcall[which(Pall$X%in%colnames(Ax2))], trace="none", scale="row", main="RSEM marker2")

heatmap.2(Ax3, col=brewer.pal(11, "RdBu")[11:1], ColSideColors = BLcall[which(Pall$X%in%colnames(Ax3))], scale="row", main="vsd marker list3", trace="none")
heatmap.2(Ax3b, col=brewer.pal(11, "RdBu")[11:1], ColSideColors = BLcall[which(Pall$X%in%colnames(Ax3))], trace="none", scale="row", main="RSEM marker3")

```

Based on the above calls, we can condense the calls:

* Her2, LumB are sumilar and can be combined with LumA
* Normal and Basal are most likely most similar

```{r}
Pall$Call2=Pall$Call
Pall$Call2[which(Pall$Call2=="Her2")]="LumB"

Pall$Call3="Basal"
Pall$Call3[grep("Lum", Pall$Call2)]="Lum"

BLcall=ifelse(Pall$Call3=="Basal", "blue", "red")

midx4=match(Pall$X, colnames(allTPMFinal))
Ax4=assay(vsd)[match(Markers, rownames(assay(vsd))),na.omit(midx4)]
Ax4b=allRSEMnorm2log[na.omit(match(toupper(Markers), rownames(allRSEMnorm2log))), na.omit(midx4)]

heatmap.2(Ax4, col=brewer.pal(11, "RdBu")[11:1], ColSideColors = BLcall, scale="row", main="vsd marker list3", trace="none")
heatmap.2(Ax4b, col=brewer.pal(11, "RdBu")[11:1], ColSideColors = BLcall, trace="none", scale="row", main="RSEM marker3")

```


## Clustering based on random genes

In this section we randomly select the most variable 50, 100, 200, 500 genes: we use both the standard deviation and coefficient of variation (sd normalised by the mean) to select these most variable genes.

Below is a correlation plot which visualises the concordance between the different methods.

```{r}
ProgCoh=assay(vsd)[ ,which(infoTableFinal$Fraction=="Ep" & infoTableFinal$Cohort=="Progression")]
CharCoh=assay(vsd)[ ,which(infoTableFinal$Fraction=="Ep" & infoTableFinal$Cohort=="Characterisation")]

GeneVar1=sapply(1:nrow(ProgCoh), function(x) sd(ProgCoh[x, ]))
GeneVar2=sapply(1:nrow(ProgCoh), function(x) sd(ProgCoh[x, ])/mean(ProgCoh[x, ]))
names(GeneVar2)=rownames(ProgCoh)
names(GeneVar1)=rownames(ProgCoh)
OrdGene1=sort(GeneVar1, decreasing = T)
OrdGene2=sort(GeneVar2, decreasing = T)

xsearch=c(50, 100, 200, 500)

CallsTableProg=matrix(NA, ncol=ncol(ProgCoh), nrow=11)
colnames(CallsTableProg)=colnames(ProgCoh)
CallsTableProg[1, ]=Pall$Call3[match(colnames(CallsTableProg), Pall$X)]
#CallsTableProg[2, ]=Pall$Call2[match(colnames(CallsTableProg), Pall$X)]
CallsTableProg[2, ]=Pall$PamCallLong[match(colnames(CallsTableProg), Pall$X)]
CallsTableProg[3, ]=Pall$PamCallshort[match(colnames(CallsTableProg), Pall$X)]

count=4

par(mfrow=c(2,2))

for (i in xsearch){
 # Int1=venn(list(SD=names(OrdGene1)[1:i], CoV=names(OrdGene2)[1:i]))
  ax1=pam(t(ProgCoh[names(OrdGene1)[1:i],  ]), 2)
  bx1=pam(t(ProgCoh[names(OrdGene2)[1:i],  ]), 2)
  CallsTableProg[count, ]=ax1$clustering
  CallsTableProg[count+1, ]=bx1$clustering
  count=count+2
}

ConcordanceMat=matrix(NA, nrow=nrow(CallsTableProg), ncol=nrow(CallsTableProg))
for (i in 1:nrow(CallsTableProg)){
  for (j in 1:nrow(CallsTableProg)){
    a1=table(CallsTableProg[i, ], CallsTableProg[j, ])
    ConcordanceMat[i, j]=sum(diag(a1))/sum(a1)
  }
}

colnames(ConcordanceMat)=c("P50", "PamLong","PamShort", paste(rep(c("sd", "cov"), 4), rep(xsearch, each=2), sep="."))
rownames(ConcordanceMat)=colnames(ConcordanceMat)
ConcordanceMat2=ConcordanceMat
ConcordanceMat2[which(ConcordanceMat2<0.5, arr.ind=T)]=1-ConcordanceMat2[which(ConcordanceMat2<0.5, arr.ind=T)]

# Example heatmap
Ax1=assay(vstEp)[ match(Markers, rownames(assay(vstEp))),]

TabOutput=cbind(ifelse(CallsTableProg[4, ]==1, "blue", "red"),ifelse(CallsTableProg[5, ]==1, "blue", "red"))
x1=match(colnames(Ax1), colnames(CallsTableProg))

heatmap.plus(Ax1, col=brewer.pal(11, "RdBu")[11:1], ColSideColors =(TabOutput[ x1,]),scale="row", main="Progression cohort vsd marker list1", trace="none")

```

## Consolidating the different results

Below is a concordance map of the similarity between the different metrics calculated as

$$ \frac{N_{b,b}+N_{l,l}}{N_{total}} $$

where $N_{b,b}$ is the number of cases which are basal in two comparisons and $N_{l,l}$ the number of cases which are luminal in both comparisons.

Note that the values extend from 0.5 to 1, which indicates that any two methods overlap more than just by chance. 

There is good overlap between the in-house specific list characterisation, and some overlap between pam50 calling and in-house short list. 

```{r}
heatmap.2(ConcordanceMat2, col=RdBu[11:1], trace="none",scale="none")

heatmap.2(ConcordanceMat2[1:7, 1:7], col=RdBu[11:1], trace="none",scale="none")
```

Below is a summary of the average expression values of the classes defined by the different methods:

* The targeted lists show differences in esr1, pgr, erbb2, pcna
* Ar has differing directions for the pam classifications
* Random genes selected show differences in pgr, erbb2, Ar and perhaps Mki67

```{r}
par(mfrow=c(2,3))

compv=c("PAM50", "pamlong", "pamshort", "top50cov", "top50sd")

for (j in gList){
for (i in 1:5){
  boxplot(ProgCoh[j, ]~CallsTableProg[i, ], main=compv[i], ylab=j, xlab="")
}}
mtext("ESR1", side=2, outer=T, line=-2)

```

We can summarise the outputs for each sample below

```{r}
CallsTableProg2=CallsTableProg[1:5, ]
rownames(CallsTableProg2)=c("PAM50", "pamlong", "pamshort", "top50cov", "top50sd")

scroll_box(kable(t(CallsTableProg2), format="html"),
         height="300px", width="100%")


```

# Signature analysis

In this section, we use a combination of gene-set enrichment analysis, heatmap visualisation and XX to investigate differences in 

- PAM50 expression
- Major signalling pathways
- MHC expression signatures

## MHC signature analysis

Look if there is an association between MHC class I and class II and checkpoint proteins with growth in the different fractions. The summary appears:

* class I: association with stable in ep samples
* class II: low expression in DN stable sampels
* MHC presentation: higher in stable samples Ep (and also the characterisation cohort in general), may also be the case in DN samples

### MHC-I

```{r}
classI <- c("RT1-A1", "RT1-A2", "RT1-A3", "RT1-Cl", "RT1-M2", "RT1-M3-1", "RT1-M4", "RT1-M5", "RT1-N1", "RT1-N2", "RT1-N3", "RT1-O1", "RT1-S2", "RT1-S3")
classII <- c("RT1-Ba", "RT1-Bb", "RT1-Da", "RT1-Db1", "RT1-Db2", "RT1-DMa", "RT1-DMb", "RT1-DOa", "RT1-DOb", "RT1-Ha")

#pdf(sprintf("rslt/signatureAnalysis/MHC_presentation_%s.pdf", Sys.Date()), height=6, width=7)

MHCclassSumm2=assay(vsd)[na.omit(match(classI, rownames(assay(vsd)))), ]

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="CD45")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="CD45")])], scale="row", main="CD: MHC class I")

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="Ep")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="Ep")])], scale="row", main="Ep: MHC class I")

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="DN")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="DN")])], scale="row", main="DN: MHC class I")
```

### MHC-II

```{r}
MHCclassSumm2=assay(vsd)[na.omit(match(classII, rownames(assay(vsd)))), ]

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="CD45")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="CD45")])], scale="row", main="CD: MHC class II")

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="Ep")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="Ep")])], scale="row", main="Ep: MHC class II")

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="DN")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="DN")])], scale="row", main="DN: MHC class II")
```

### MHC presentation proteins

```{r}
MHCclassSumm2=assay(vsd)[na.omit(match(MHCPres2Rat[-1], rownames(assay(vsd2)))), ]

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="CD45")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="CD45")])], scale="row", main="CD45: MHC presentation")

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="Ep")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="Ep")])], scale="row", main="Ep: MHC presentation")

heatmap.2(MHCclassSumm2[ ,which(infoTableFinal$Fraction=="DN")], col=RdBu[11:1], trace="none", ColSideColors = palette()[factor(infoTableFinal$Growth[which(infoTableFinal$Fraction=="DN")])], scale="row", main="DN: MHC presentation")

#dev.off()
```


<!--chapter:end:07-signature-analysis.Rmd-->

# Cancer pathways

## ssGSEA

Figure out the signalling pathway expression (e.g using ssGSEA)

```{r}
gsList=list()
for (i in 1:length(PathwayListRat)){
gsList[[i]]=GeneSet(na.omit(PathwayListRat[[i]]), setName = names(PathwayListRat)[i], geneIdType=SymbolIdentifier())
}

CancerRatSig=GeneSetCollection(gsList)
# 
Cancer_counts=gsva(assay(dds)[ ,-grep("CD45", colnames(assay(dds)))], CancerRatSig, mx.diff=T, method="ssgsea", verbose=F, parallel.sz=1, kcdf="Poisson", ssgsea.norm=T)

Cancer_counts2=gsva(allTPMFinal[ ,-grep("CD45", colnames(assay(dds)))], CancerRatSig, mx.diff=T, method="ssgsea", verbose=F, parallel.sz=1, kcdf="Poisson", ssgsea.norm=T)

# doesn't seem to matter that we convert to TPM or not. Also try the TPMs from the RSEM package
#Imm_counts2=gsva(allTPMFinal,ImmRatSig, mx.diff=T, method="ssgsea", verbose=F, parallel.sz=1, kcdf="Poisson", ssgsea.norm=T)

#pdf(sprintf("rslt/signatureAnalysis/Main_cancer_Pathways_%s.pdf", Sys.Date()), width=6, height=7)
# palette()[as.numeric(infoTableFinal$Batch[match(colnames(Cancer_counts), infoTableFinal$SampleID)])],   
heatmap.plus(Cancer_counts2, col=brewer.pal(11, "RdBu")[11:1], trace="none", ColSideColors =cbind(palette()[as.numeric(infoTableFinal$Treatment[match(colnames(Cancer_counts), infoTableFinal$SampleID)])+4],                             palette()[as.numeric(infoTableFinal$Fraction[match(colnames(Cancer_counts), infoTableFinal$SampleID)])],
             palette()[as.numeric(infoTableFinal$Growth[match(colnames(Cancer_counts), infoTableFinal$SampleID)])]), scale="row", symkey=T)


CCount=Cancer_counts2[ , grep("Ep", colnames(Cancer_counts2))]

heatmap.plus(CCount, col=brewer.pal(11, "RdBu")[11:1], trace="none", ColSideColors =cbind(palette()[as.numeric(infoTableFinal$Treatment[match(colnames(CCount), infoTableFinal$SampleID)])+4],                             palette()[as.numeric(infoTableFinal$Fraction[match(colnames(CCount), infoTableFinal$SampleID)])],
             palette()[as.numeric(infoTableFinal$Growth[match(colnames(CCount), infoTableFinal$SampleID)])]), scale="row", symkey=T,
             main="Ep only samples")

CCount=Cancer_counts2[ , grep("DN", colnames(Cancer_counts2))]

heatmap.plus(CCount, col=brewer.pal(11, "RdBu")[11:1], trace="none", ColSideColors =cbind(palette()[as.numeric(infoTableFinal$Treatment[match(colnames(CCount), infoTableFinal$SampleID)])+4],                             palette()[as.numeric(infoTableFinal$Fraction[match(colnames(CCount), infoTableFinal$SampleID)])],
             palette()[as.numeric(infoTableFinal$Growth[match(colnames(CCount), infoTableFinal$SampleID)])]), scale="row", symkey=T,
             main="DN only samples")

cancerCountsMelt=melt(Cancer_counts)
cancerCountsMelt$Fraction=infoTableFinal$Fraction[match(cancerCountsMelt$Var2, infoTableFinal$SampleID)]
cancerCountsMelt$Treatment=infoTableFinal$Treatment[match(cancerCountsMelt$Var2, infoTableFinal$SampleID)]

ggplot(data=cancerCountsMelt, aes(x=Treatment, y=value, col=Fraction))+geom_boxplot()+facet_wrap(~Var1, scales="free_y")


#dev.off()

```


### regression model for covariates

Linear model: Create a linear model to explain the above:

Score ~ Fraction + Treatment + CD8Fraction + Spatial Score + GrowthRate (+Tumor Size?) + PAM50

```{r, eval=F}
mx1=match(colnames(Cancer_counts2), infoTableFinal$SampleID)

infoTabTemp=infoTableFinal[mx1, ]
infoTabTemp$Fraction=factor(infoTabTemp$Fraction)
levels(infoTabTemp$Treatment)=c("Vehicle", "LY", "PDL1", "PDL1+LY")
#infoTabTemp$PAM50_v2="Basal"
#infoTabTemp$PAM50_v2[grep("Lum", infoTabTemp$PAM50)]="Lum"

## First do a score based on DN vs Ep

a1=glm(infoTableFinal$Fraction[mx1]~0+t(Cancer_counts2), family="binomial")
summary(a1)


MatrixSumm=matrix(NA, nrow=nrow(Cancer_counts2), ncol=11)
PValsSumm=MatrixSumm


for (i in 1:nrow(Cancer_counts2)){
  a1=glm(Cancer_counts2[i, ]~Fraction+Treatment+CD8Frac+MHcut+GrowthRate+TumSize+SpatialManual+PAM50_v2, data=infoTabTemp)
  MatrixSumm[i, ]=coefficients(a1)
  PValsSumm[i, ]=summary(a1)$coefficients[ ,4]
}

colnames(PValsSumm)=names(coefficients(a1))
rownames(PValsSumm)=rownames(Cancer_counts2)
colnames(MatrixSumm)=colnames(PValsSumm)
rownames(MatrixSumm)=rownames(PValsSumm)

MatrixSumm2=MatrixSumm
MatrixSumm2[which(PValsSumm>0.1, arr.ind = T)]=0

#pdf(sprintf("rslt/signatureAnalysis/cancer_pathway_variables_%s.pdf", Sys.Date()), height=6, width=6)
par(oma=c(4, 0,0,0))

heatmap.2(MatrixSumm2[ ,-1], col=RdBu[11:1], trace="none", main="Contributors to signalling pathway: All variables")

MatrixSumm=matrix(NA, nrow=nrow(Cancer_counts2), ncol=9)
PValsSumm=MatrixSumm

for (i in 1:nrow(Cancer_counts2)){
  a1=glm(Cancer_counts2[i, infoTabTemp$Fraction=="DN"]~Treatment+CD8Frac+MHcut+GrowthRate+TumSize+PAM50_v2, data=infoTabTemp[infoTabTemp$Fraction=="DN", ])
  MatrixSumm[i, ]=coefficients(a1)
  PValsSumm[i, ]=summary(a1)$coefficients[ ,4]
}

colnames(PValsSumm)=names(coefficients(a1))
rownames(PValsSumm)=rownames(Cancer_counts2)
colnames(MatrixSumm)=colnames(PValsSumm)
rownames(MatrixSumm)=rownames(PValsSumm)

MatrixSumm2=MatrixSumm
MatrixSumm2[which(PValsSumm>0.1, arr.ind = T)]=0

heatmap.2(MatrixSumm2[ ,-1], col=RdBu[11:1], trace="none", main="Contributors to signalling pathway: DN")

MatrixSumm=matrix(NA, nrow=nrow(Cancer_counts2), ncol=9)
PValsSumm=MatrixSumm

for (i in 1:nrow(Cancer_counts2)){
  a1=glm(Cancer_counts2[i, infoTabTemp$Fraction=="Ep"]~Treatment+CD8Frac+MHcut+GrowthRate+TumSize+PAM50_v2, data=infoTabTemp[infoTabTemp$Fraction=="Ep", ])
  MatrixSumm[i, ]=coefficients(a1)
  PValsSumm[i, ]=summary(a1)$coefficients[ ,4]
}

colnames(PValsSumm)=names(coefficients(a1))
rownames(PValsSumm)=rownames(Cancer_counts2)
colnames(MatrixSumm)=colnames(PValsSumm)
rownames(MatrixSumm)=rownames(PValsSumm)

MatrixSumm2=MatrixSumm
MatrixSumm2[which(PValsSumm>0.1, arr.ind = T)]=0

heatmap.2(MatrixSumm2[ ,-1], col=RdBu[11:1], trace="none", main="Contributors to signalling pathway: Ep")

#dev.off()
```

Associations: 
* notch signalling with LY treatment
* TGFB with PDL1+LY treatment
* Ras and Wnt signalling lower CD8Fraction
* TP53 with CD8 fraction

### TSG vs Oncogenes

Do the same scoring, taking into account oncogenic vs tumor suppressor activity?
Usually, there is no difference. The pathway is increased either way

```{r}
gsListUP=list()
for (i in 1:length(PathwayListRat)){
gsListUP[[i]]=GeneSet(na.omit(PathwayListRat[[i]][which(PathwaySign[[i]]=="OG")]), setName = paste(names(PathwayListRat)[i], "OG", sep="_"), geneIdType=SymbolIdentifier())
}
gsListDOWN=list()
for (i in 1:length(PathwayListRat)){
gsListDOWN[[i]]=GeneSet(na.omit(PathwayListRat[[i]][which(PathwaySign[[i]]=="TSG")]), setName = paste(names(PathwayListRat)[i], "TS", sep="_"), geneIdType=SymbolIdentifier())
}
AllGS=c(gsListUP, gsListDOWN)

CancerRatSig_Dir=GeneSetCollection(AllGS)
# 
Cancer_counts_dir=gsva(assay(dds)[ ,-grep("CD45", colnames(assay(dds)))], CancerRatSig_Dir, mx.diff=T, method="ssgsea", verbose=F, parallel.sz=1, kcdf="Poisson", ssgsea.norm=T)
# doesn't seem to matter that we convert to TPM or not. Also try the TPMs from the RSEM package
#Imm_counts2=gsva(allTPMFinal,ImmRatSig, mx.diff=T, method="ssgsea", verbose=F, parallel.sz=1, kcdf="Poisson", ssgsea.norm=T)

#pdf(sprintf("rslt/signatureAnalysis/Main_cancer_Pathways_w_direction_%s.pdf", Sys.Date()), width=6, height=7)


heatmap.plus(Cancer_counts_dir, col=brewer.pal(11, "RdBu")[11:1], trace="none", ColSideColors =cbind(palette()[as.numeric(infoTableFinal$Treatment[match(colnames(Cancer_counts), infoTableFinal$SampleID)])+4], palette()[as.numeric(infoTableFinal$Batch[match(colnames(Cancer_counts), infoTableFinal$SampleID)])], palette()[as.numeric(infoTableFinal$Fraction[match(colnames(Cancer_counts), infoTableFinal$SampleID)])]), scale="row")

cancerCountsMelt_dir=melt(Cancer_counts_dir)
cancerCountsMelt_dir$Var1=as.character(cancerCountsMelt_dir$Var1)
cancerCountsMelt_dir$Fraction=infoTableFinal$Fraction[match(cancerCountsMelt_dir$Var2, infoTableFinal$SampleID)]
cancerCountsMelt_dir$Treatment=infoTableFinal$Treatment[match(cancerCountsMelt_dir$Var2, infoTableFinal$SampleID)]
cancerCountsMelt_dir$Dir=substr(cancerCountsMelt_dir$Var1,  nchar(cancerCountsMelt_dir$Var1)-1, nchar(cancerCountsMelt_dir$Var1))

ggplot(data=cancerCountsMelt_dir, aes(x=Treatment, y=value, col=Fraction))+geom_boxplot()+facet_wrap(~Var1, scales="free_y")

#dev.off()

```

Do a regression analysis for TSG vs ONCO signatures in this pathway


```{r, eval=F}
mx1=match(colnames(Cancer_counts_dir), infoTableFinal$SampleID)

infoTabTemp=infoTableFinal[mx1, ]
infoTabTemp$Fraction=factor(infoTabTemp$Fraction)
levels(infoTabTemp$Treatment)=c("Vehicle", "LY", "PDL1", "PDL1+LY")

## First do a score based on DN vs Ep

a1=glm(infoTableFinal$Fraction[mx1]~0+t(Cancer_counts_dir), family="binomial")
summary(a1)


MatrixSumm=matrix(NA, nrow=nrow(Cancer_counts_dir), ncol=10)
PValsSumm=MatrixSumm


for (i in 1:nrow(Cancer_counts_dir)){
  a1=glm(Cancer_counts_dir[i, ]~Fraction+Treatment+CD8Frac+MHcut+GrowthRate+TumSize+SpatialManual, data=infoTabTemp)
  MatrixSumm[i, ]=coefficients(a1)
  PValsSumm[i, ]=summary(a1)$coefficients[ ,4]
}

colnames(PValsSumm)=names(coefficients(a1))
rownames(PValsSumm)=rownames(Cancer_counts_dir)
colnames(MatrixSumm)=colnames(PValsSumm)
rownames(MatrixSumm)=rownames(PValsSumm)

MatrixSumm2=MatrixSumm
MatrixSumm2[which(PValsSumm>0.1, arr.ind = T)]=0

#pdf(sprintf("rslt/signatureAnalysis/cancer_pathway_variables_withdirection_%s.pdf", Sys.Date()), height=6, width=6)
par(oma=c(4, 0,0,0))

heatmap.2(MatrixSumm2[ ,-1], col=RdBu[11:1], trace="none", main="Contributors to signalling pathway: All variables")

MatrixSumm=matrix(NA, nrow=nrow(Cancer_counts_dir), ncol=8)
PValsSumm=MatrixSumm

for (i in 1:nrow(Cancer_counts_dir)){
  a1=glm(Cancer_counts_dir[i, infoTabTemp$Fraction=="DN"]~Treatment+CD8Frac+MHcut+GrowthRate+TumSize, data=infoTabTemp[infoTabTemp$Fraction=="DN", ])
  MatrixSumm[i, ]=coefficients(a1)
  PValsSumm[i, ]=summary(a1)$coefficients[ ,4]
}

colnames(PValsSumm)=names(coefficients(a1))
rownames(PValsSumm)=rownames(Cancer_counts_dir)
colnames(MatrixSumm)=colnames(PValsSumm)
rownames(MatrixSumm)=rownames(PValsSumm)

MatrixSumm2=MatrixSumm
MatrixSumm2[which(PValsSumm>0.1, arr.ind = T)]=0

heatmap.2(MatrixSumm2[ ,-1], col=RdBu[11:1], trace="none", main="Contributors to signalling pathway: DN")

MatrixSumm=matrix(NA, nrow=nrow(Cancer_counts_dir), ncol=8)
PValsSumm=MatrixSumm

for (i in 1:nrow(Cancer_counts_dir)){
  a1=glm(Cancer_counts_dir[i, infoTabTemp$Fraction=="Ep"]~Treatment+CD8Frac+MHcut+GrowthRate+TumSize, data=infoTabTemp[infoTabTemp$Fraction=="Ep", ])
  MatrixSumm[i, ]=coefficients(a1)
  PValsSumm[i, ]=summary(a1)$coefficients[ ,4]
}

colnames(PValsSumm)=names(coefficients(a1))
rownames(PValsSumm)=rownames(Cancer_counts_dir)
colnames(MatrixSumm)=colnames(PValsSumm)
rownames(MatrixSumm)=rownames(PValsSumm)

MatrixSumm2=MatrixSumm
MatrixSumm2[which(PValsSumm>0.1, arr.ind = T)]=0

heatmap.2(MatrixSumm2[ ,-1], col=RdBu[11:1], trace="none", main="Contributors to signalling pathway: Ep")

#dev.off()

```

Looking at up vs down genes: are they differentially expressed?

```{r, eval=F}
AllSearch=lapply(AllGS, function(x) geneIds(x))
AllSearchUn=unlist(AllSearch)

DiffGenes=AllSearchUn[which(AllSearchUn%in%rownames(resDNeprslt2))]
TypeGenes=sapply(AllSearch, function(x) which(DiffGenes%in%x))

heatmap.2(assay(vsd)[ match(DiffGenes, rownames(assay(vsd))), -grep("CD45", colnames(assay(vsd)))],
          trace="none", col=RdBu[11:1], ColSideColors = palette()[colSide], scale="row")
```


<!--chapter:end:07b-cancer-pathways.Rmd-->

# Immune estimation

In this section, we will look at a number of methods for immune estimation:

1. ssGSEA using known signatures: This includes the Thorsson signatures, as well as signatures for activation and dysfunction as described by Tirosh and in the supplementary tables from Gil del Alcazar.
2. Deconvolution methods (CIBERSORT, TIMER etc)

## ssGSEA

Below, we use the GSVA package and TPM counts to assess variability in the different signatures. We will compare this to:

* growth
* CD8 counts 
* spatial patterns (infiltrating vs restricted, MH index)

```{r}
table(infoTableFinal$MHcut)

gsList=list()
for (i in 1:length(Exp2RatImm)){
gsList[[i]]=GeneSet(Exp2RatImm[[i]], setName = names(Exp2RatImm)[i], geneIdType=SymbolIdentifier())
}

ImmRatSig=GeneSetCollection(gsList)

Imm_counts=gsva(allTPMFinal[ , grep("CD45", colnames(allTPMFinal))],ImmRatSig, mx.diff=T, method="ssgsea", verbose=F, parallel.sz=1, kcdf="Poisson", ssgsea.norm=T)

growthDiff=sapply(1:nrow(Imm_counts), function(x) wilcox.test(Imm_counts[x,  ]~infoTableFinal$Growth[match(colnames(Imm_counts), infoTableFinal$SampleID)])$p.value)
names(growthDiff)=rownames(Imm_counts)

ax1=which(growthDiff<0.05)

SpatDiff=sapply(1:nrow(Imm_counts), function(x) wilcox.test(Imm_counts[x,  ]~infoTableFinal$MHcut[match(colnames(Imm_counts), infoTableFinal$SampleID)])$p.value)
names(SpatDiff)=rownames(Imm_counts)

bx1=which(SpatDiff<0.05)

CD8Diff=sapply(1:nrow(Imm_counts), function(x) wilcox.test(Imm_counts[x,  ]~infoTableFinal$CD8FracCut[match(colnames(Imm_counts), infoTableFinal$SampleID)])$p.value)
names(SpatDiff)=rownames(Imm_counts)

cx1=which(SpatDiff<0.05)

ColTable=cbind(palette()[as.numeric(infoTableFinal$Treatment[match(colnames(Imm_counts), infoTableFinal$SampleID)])+4],                             palette()[factor(infoTableFinal$MHcut[match(colnames(Imm_counts), infoTableFinal$SampleID)])],
palette()[factor(infoTableFinal$Growth[match(colnames(Imm_counts), infoTableFinal$SampleID)])],
palette()[factor(infoTableFinal$CD8FracCut[match(colnames(Imm_counts), infoTableFinal$SampleID)])])
colnames(ColTable)=c("Treatment", "Spatial", "Growth", "CD8")

#pdf(sprintf("rslt/signatureAnalysis/Immune_Pathways_%s.pdf", Sys.Date()), width=6, height=7)
# palette()[as.numeric(infoTableFinal$Batch[match(colnames(Cancer_counts), infoTableFinal$SampleID)])],   
heatmap.plus(Imm_counts, col=brewer.pal(11, "RdBu")[11:1], trace="none", ColSideColors =ColTable, scale="row", symkey=T, main="All CD45 samples")

xidx=which(infoTableFinal$Cohort[grep("CD45", colnames(allTPMFinal))]=="Progression")
xidx

heatmap.plus(Imm_counts[ ,xidx], col=brewer.pal(11, "RdBu")[11:1], trace="none", ColSideColors =ColTable[ xidx, ], scale="row", symkey=T, main="All Progression CD45 samples")
```

Based on wilcoxon tests for differences between growth, cd8 content, spatial arrangement, we can also pull out the significantly different signatures and plot these below:

```{r}
heatmap.plus(Imm_counts[unique(c(ax1, bx1, cx1)), ], col=brewer.pal(11, "RdBu")[11:1], trace="none", ColSideColors =ColTable, scale="row", symkey=T,
             main="significant differences")

AllTables=cbind(ax1, bx1, cx1)
colnames(AllTables)=c("growth", "cd8", "mh-idx")
#rownames(AllTables)=rownames(Imm_counts)

dim(AllTables)
length(rownames(Imm_counts))

heatmap.2(-log10(AllTables+0.001),col=brewer.pal(9, "Blues"), scale="none", trace="none", main="p values of differences" )
```

## Deconvolution methods

Deconvolution was performed using the TIMER website, which lists the following results

TIMER
XCELL
CIBERSORT
EPIC

For TIDE, we need to:
* compute log(RPKM+1) values
* Determine the values of the reference/or compute a sample average
* Subtract the two
* We can convert from Rat gene names to Human and exclude all the non mapping gnes.


TPM counts were used for this analysis (using mouse gene names)

```{r}
output1=read.csv("../data/RNA_expression/CD45_rgd_10-02-20-estimation_matrix.csv")
colnames(output1)=gsub("X", "", colnames(output1))
```

<!--chapter:end:08-immune-estimation.Rmd-->

