<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 RNA data: pre-processing | Rat DCIS study</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘insert study name’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 RNA data: pre-processing | Rat DCIS study" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘insert study name’" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 RNA data: pre-processing | Rat DCIS study" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘insert study name’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2020-11-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="expression-data.html"/>
<link rel="next" href="rna-data-preliminary-plots.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.1</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#facs-data"><i class="fa fa-check"></i><b>2.2</b> FACS data</a></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#summary-table"><i class="fa fa-check"></i><b>2.3</b> Summary Table</a><ul>
<li class="chapter" data-level="2.3.1" data-path="intro.html"><a href="intro.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>2.3.1</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="2.3.2" data-path="intro.html"><a href="intro.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>2.3.2</b> Summary of the RNA data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>3</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="3.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>3.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="3.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>3.2</b> Cellular composition</a></li>
<li class="chapter" data-level="3.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>3.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="3.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>3.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="3.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>3.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="3.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>3.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>4</b> Spatial statistics</a><ul>
<li class="chapter" data-level="4.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>4.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="4.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification-treatment-growth"><i class="fa fa-check"></i><b>4.1.1</b> Comparison to manual classification, treatment, growth</a></li>
<li class="chapter" data-level="4.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome"><i class="fa fa-check"></i><b>4.1.2</b> Associations with outcome</a></li>
<li class="chapter" data-level="4.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>4.1.3</b> Growth</a></li>
<li class="chapter" data-level="4.1.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#association-between-distance-and-content"><i class="fa fa-check"></i><b>4.1.4</b> Association between distance and content</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>4.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="4.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>4.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="4.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>4.2.2</b> Growth</a></li>
<li class="chapter" data-level="4.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>4.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>4.3</b> M-H distances</a><ul>
<li class="chapter" data-level="4.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>4.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="4.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>4.3.2</b> Growth</a></li>
<li class="chapter" data-level="4.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>4.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>4.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="4.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#other-cell-types"><i class="fa fa-check"></i><b>4.5</b> Other cell types</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>5</b> Expression data</a><ul>
<li class="chapter" data-level="5.1" data-path="expression-data.html"><a href="expression-data.html#load-and-qc-data"><i class="fa fa-check"></i><b>5.1</b> Load and QC data</a></li>
<li class="chapter" data-level="5.2" data-path="expression-data.html"><a href="expression-data.html#loading-in-data"><i class="fa fa-check"></i><b>5.2</b> Loading in data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-data-pre-processing.html"><a href="rna-data-pre-processing.html"><i class="fa fa-check"></i><b>6</b> RNA data: pre-processing</a><ul>
<li class="chapter" data-level="6.0.1" data-path="rna-data-pre-processing.html"><a href="rna-data-pre-processing.html#initial-qc"><i class="fa fa-check"></i><b>6.0.1</b> Initial QC</a></li>
<li class="chapter" data-level="6.1" data-path="rna-data-pre-processing.html"><a href="rna-data-pre-processing.html#normalisation"><i class="fa fa-check"></i><b>6.1</b> Normalisation</a><ul>
<li class="chapter" data-level="6.1.1" data-path="rna-data-pre-processing.html"><a href="rna-data-pre-processing.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.1.1</b> preliminary visualisation (to remove outliers)</a></li>
<li class="chapter" data-level="6.1.2" data-path="rna-data-pre-processing.html"><a href="rna-data-pre-processing.html#processing-files-for-pam50-calling"><i class="fa fa-check"></i><b>6.1.2</b> Processing files for PAM50 calling</a></li>
<li class="chapter" data-level="6.1.3" data-path="rna-data-pre-processing.html"><a href="rna-data-pre-processing.html#processing-data-for-xcell-and-cibersort"><i class="fa fa-check"></i><b>6.1.3</b> Processing data for xcell and cibersort</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#gene-annotations"><i class="fa fa-check"></i><b>7.1</b> Gene Annotations</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#visualise-the-data"><i class="fa fa-check"></i><b>7.1.1</b> Visualise the data</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://test.com" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Rat DCIS study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rna-data-pre-processing" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> RNA data: pre-processing</h1>
<p>RNA data is available as counts, as well as TPM, rsem and FPKM counts using the RSEM package
Note: <a href="http://hgdownload.soe.ucsc.edu/goldenPath/rn6/bigZips/genes/rn6.ncbiRefSeq.gtf.gz">rn6.refGene.gtf.gz</a> was used to generate the RSEM library!(This is the may version)</p>
<p>Note: april version is being used</p>
<div id="initial-qc" class="section level3">
<h3><span class="header-section-number">6.0.1</span> Initial QC</h3>
<p>Default output from R showing the number of unique reads compared to multimapped, unmapped etc. This is shown for each batch.
Note that batch 3 has differences (high percentage of unmapped) compared to the other batches, possibly due to DNA contamination.</p>
<p>Below we check for three measures:
- mapped million reads (ideally, 10M+ reads)
- Gene Sparsity: This is a measurement of the number of genes which have non-zero values. Ideally, would be greater than 10K, but values which are too high may also suggest contamination from DNA (unexpressed genes are also counted)
- Varability: standard deviation of the transcriptomic counts. If this value is too low, would suggest that high DNA contamination, non-representative transcriptome.</p>
<p>plot the variability of the above measures</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-65-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-65-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-65-3.png" width="672" /></p>
<p>Samples to remove from analysis. The thresholds are based on the above values, and removes cases which are &lt;1.5 SD of the mean</p>
<ul>
<li>low total number of mapped reads (under 4M)</li>
<li>sparsity: less than 11K genes</li>
<li>variability : threshold under 500</li>
</ul>
<p>April counts:
* low total number of mapped reads (under 1.5M)
* sparsity: less than 8K genes
* variability : threshold under 500</p>
<p>Do we need to set thresholds for high Gene Sparsity??</p>
<pre><code>## [1] &quot;6R_C_CD45&quot;    &quot;10L_B_DN&quot;     &quot;NMU1_LL_Ep&quot;   &quot;NMU5_LA_Ep&quot;   &quot;Control1__Ep&quot;
## [6] &quot;NMU13_RAU_Ep&quot;</code></pre>
<p>The omitted samples are:</p>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
MappedReadsM
</th>
<th style="text-align:right;">
GeneSparsityK
</th>
<th style="text-align:right;">
Batch
</th>
<th style="text-align:right;">
GeneVariabilityCounts
</th>
<th style="text-align:left;">
Type
</th>
<th style="text-align:left;">
names
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
6R_C_CD45
</td>
<td style="text-align:right;">
2.287110
</td>
<td style="text-align:right;">
5.585
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1063.38925
</td>
<td style="text-align:left;">
CD45
</td>
<td style="text-align:left;">
6R_C_CD45
</td>
</tr>
<tr>
<td style="text-align:left;">
10L_B_DN
</td>
<td style="text-align:right;">
0.864838
</td>
<td style="text-align:right;">
3.308
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
559.30659
</td>
<td style="text-align:left;">
DN
</td>
<td style="text-align:left;">
10L_B_DN
</td>
</tr>
<tr>
<td style="text-align:left;">
NMU1_LL_Ep
</td>
<td style="text-align:right;">
1.032440
</td>
<td style="text-align:right;">
16.708
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
69.82645
</td>
<td style="text-align:left;">
Ep
</td>
<td style="text-align:left;">
NMU1_LL_Ep
</td>
</tr>
<tr>
<td style="text-align:left;">
NMU5_LA_Ep
</td>
<td style="text-align:right;">
0.500429
</td>
<td style="text-align:right;">
15.707
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
35.56084
</td>
<td style="text-align:left;">
Ep
</td>
<td style="text-align:left;">
NMU5_LA_Ep
</td>
</tr>
<tr>
<td style="text-align:left;">
Control1__Ep
</td>
<td style="text-align:right;">
8.224593
</td>
<td style="text-align:right;">
5.800
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3164.00515
</td>
<td style="text-align:left;">
Ep
</td>
<td style="text-align:left;">
Control1__Ep
</td>
</tr>
<tr>
<td style="text-align:left;">
NMU13_RAU_Ep
</td>
<td style="text-align:right;">
3.049733
</td>
<td style="text-align:right;">
7.242
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1069.99912
</td>
<td style="text-align:left;">
Ep
</td>
<td style="text-align:left;">
NMU13_RAU_Ep
</td>
</tr>
</tbody>
</table>
<p>We are left with 17455 samples. There are samples in the fractions.
There are 20, 49, 41 samples from batches 1, 2 and 3 respectively.</p>
</div>
<div id="normalisation" class="section level2">
<h2><span class="header-section-number">6.1</span> Normalisation</h2>
<p>Run through DESEq and normalise the library. Using all samples, we run the model:</p>
<p>expression ~ Celltype + factor (Batch)</p>
<p>and keep the genes which have a total count of at least half the number of samples. ie.
sum(gene_i) &gt; nsamples/2</p>
<div id="preliminary-visualisation-to-remove-outliers" class="section level3">
<h3><span class="header-section-number">6.1.1</span> preliminary visualisation (to remove outliers)</h3>
<p><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-4.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-5.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-6.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-7.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-8.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-9.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-10.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-11.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-12.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-13.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-14.png" width="672" /></p>
<p>Based on the above plots, we remove the following outliers and re-run the normalisation:
* 2R_D_DN
* 4L_B_CD45</p>
</div>
<div id="processing-files-for-pam50-calling" class="section level3">
<h3><span class="header-section-number">6.1.2</span> Processing files for PAM50 calling</h3>
<p>Redo, this time taking into account batch effects</p>
<p>Also remove the following samples:
* 2R D DN
The first sample appears more CD45 like, the second gives weird</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-68-1.png" width="672" /></p>
<pre><code>## named integer(0)</code></pre>
<pre><code>## named integer(0)</code></pre>
</div>
<div id="processing-data-for-xcell-and-cibersort" class="section level3">
<h3><span class="header-section-number">6.1.3</span> Processing data for xcell and cibersort</h3>
<p>Save the mouse names for TIMER cistrome: check that this is actually required for TIMER</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="expression-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="rna-data-preliminary-plots.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
