<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 15 Associate Epcam+ inflammatory with survival | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 15 Associate Epcam+ inflammatory with survival | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 15 Associate Epcam+ inflammatory with survival | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-02-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="epcam-inflammatory-vs-non-inflammatory-samples.html"/>
<link rel="next" href="deseq-analysis-characterisation-cohort-big-vs-small.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.2</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.2.1" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.2.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="associate-epcam-inflammatory-with-survival" class="section level1">
<h1><span class="header-section-number">Chapter 15</span> Associate Epcam+ inflammatory with survival</h1>
<p>Associate the signature with outcome in TCGA: Load in TCGA right now</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="associate-epcam-inflammatory-with-survival.html#cb269-1"></a>TCGArsem=<span class="kw">read.delim</span>(<span class="st">&quot;../data/TCGA/BRCA.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb269-2"><a href="associate-epcam-inflammatory-with-survival.html#cb269-2"></a><span class="kw">rownames</span>(TCGArsem)=TCGArsem[ ,<span class="dv">1</span>]</span>
<span id="cb269-3"><a href="associate-epcam-inflammatory-with-survival.html#cb269-3"></a>TCGArsem=TCGArsem[, <span class="kw">grep</span>(<span class="st">&quot;01A&quot;</span>, <span class="kw">colnames</span>(TCGArsem))]</span>
<span id="cb269-4"><a href="associate-epcam-inflammatory-with-survival.html#cb269-4"></a><span class="kw">colnames</span>(TCGArsem)=<span class="kw">substr</span>(<span class="kw">colnames</span>(TCGArsem), <span class="dv">1</span>, <span class="dv">12</span>)</span>
<span id="cb269-5"><a href="associate-epcam-inflammatory-with-survival.html#cb269-5"></a><span class="kw">colnames</span>(TCGArsem)=<span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="kw">colnames</span>(TCGArsem))</span>
<span id="cb269-6"><a href="associate-epcam-inflammatory-with-survival.html#cb269-6"></a>TCGArsem=TCGArsem[<span class="op">-</span><span class="dv">1</span>, ]</span>
<span id="cb269-7"><a href="associate-epcam-inflammatory-with-survival.html#cb269-7"></a></span>
<span id="cb269-8"><a href="associate-epcam-inflammatory-with-survival.html#cb269-8"></a>TCGArsem2=<span class="kw">apply</span>(TCGArsem, <span class="dv">2</span>, as.numeric)</span>
<span id="cb269-9"><a href="associate-epcam-inflammatory-with-survival.html#cb269-9"></a><span class="co">#TCGArsem2=t(TCGArsem2)</span></span>
<span id="cb269-10"><a href="associate-epcam-inflammatory-with-survival.html#cb269-10"></a><span class="co">#colnames(TCGArsem2)=colnames(TCGArsem)</span></span>
<span id="cb269-11"><a href="associate-epcam-inflammatory-with-survival.html#cb269-11"></a><span class="co">#TCGArsem2=data.frame(TCGArsem2)</span></span>
<span id="cb269-12"><a href="associate-epcam-inflammatory-with-survival.html#cb269-12"></a><span class="kw">rownames</span>(TCGArsem2)=<span class="kw">rownames</span>(TCGArsem)</span>
<span id="cb269-13"><a href="associate-epcam-inflammatory-with-survival.html#cb269-13"></a><span class="kw">rownames</span>(TCGArsem2)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(TCGArsem2), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb269-14"><a href="associate-epcam-inflammatory-with-survival.html#cb269-14"></a>TCGArsem2=TCGArsem2[<span class="op">-</span><span class="kw">which</span>(<span class="kw">rownames</span>(TCGArsem2)<span class="op">==</span><span class="st">&quot;?&quot;</span>), ]</span>
<span id="cb269-15"><a href="associate-epcam-inflammatory-with-survival.html#cb269-15"></a></span>
<span id="cb269-16"><a href="associate-epcam-inflammatory-with-survival.html#cb269-16"></a><span class="kw">load</span>(<span class="st">&quot;../data/TCGA/BrClin_clinical_Nov2017.RData&quot;</span>)</span>
<span id="cb269-17"><a href="associate-epcam-inflammatory-with-survival.html#cb269-17"></a></span>
<span id="cb269-18"><a href="associate-epcam-inflammatory-with-survival.html#cb269-18"></a>ax1=<span class="kw">match</span>(BrClin<span class="op">$</span>Patient.ID, <span class="kw">colnames</span>(TCGArsem))</span>
<span id="cb269-19"><a href="associate-epcam-inflammatory-with-survival.html#cb269-19"></a></span>
<span id="cb269-20"><a href="associate-epcam-inflammatory-with-survival.html#cb269-20"></a>BrClin=BrClin[<span class="op">-</span><span class="kw">which</span>(<span class="kw">is.na</span>(ax1)), ]</span>
<span id="cb269-21"><a href="associate-epcam-inflammatory-with-survival.html#cb269-21"></a>TCGArsem=TCGArsem[ , <span class="kw">na.omit</span>(ax1)]</span>
<span id="cb269-22"><a href="associate-epcam-inflammatory-with-survival.html#cb269-22"></a></span>
<span id="cb269-23"><a href="associate-epcam-inflammatory-with-survival.html#cb269-23"></a><span class="co">## load in the inflammation subtype information</span></span>
<span id="cb269-24"><a href="associate-epcam-inflammatory-with-survival.html#cb269-24"></a>ThorssData=<span class="kw">read.xlsx</span>(<span class="st">&quot;../data/TCGA/Thorsson2018_table1.xlsx&quot;</span>,<span class="dv">1</span>)</span></code></pre></div>
<p>Edit the clinical data to make sure data is censored at 60 months (5 years) and that stage is given an integer value (no 2A, 2b etc.) for easier comparisons</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="associate-epcam-inflammatory-with-survival.html#cb270-1"></a><span class="co">#m1=match(colnames(TCGAssgsea), BrClin$Patient.ID)</span></span>
<span id="cb270-2"><a href="associate-epcam-inflammatory-with-survival.html#cb270-2"></a>BrClin<span class="op">$</span>Stage=(<span class="kw">substr</span>(BrClin<span class="op">$</span>American.Joint.Committee.on.Cancer.Tumor.Stage.Code, <span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb270-3"><a href="associate-epcam-inflammatory-with-survival.html#cb270-3"></a>BrClin<span class="op">$</span>Stage[<span class="kw">which</span>(BrClin<span class="op">$</span>Stage<span class="op">==</span><span class="st">&quot;&quot;</span><span class="op">|</span>BrClin<span class="op">$</span>Stage<span class="op">==</span><span class="st">&quot;X&quot;</span>)]=<span class="ot">NA</span></span>
<span id="cb270-4"><a href="associate-epcam-inflammatory-with-survival.html#cb270-4"></a></span>
<span id="cb270-5"><a href="associate-epcam-inflammatory-with-survival.html#cb270-5"></a>BrClin<span class="op">$</span>Overall.Survival..Months.[<span class="kw">which</span>(BrClin<span class="op">$</span>Overall.Survival..Months.<span class="op">&gt;=</span><span class="dv">60</span>)]=<span class="dv">60</span></span>
<span id="cb270-6"><a href="associate-epcam-inflammatory-with-survival.html#cb270-6"></a>BrClin<span class="op">$</span>Overall.Survival.Status[<span class="kw">which</span>(BrClin<span class="op">$</span>Overall.Survival..Months.<span class="op">&gt;=</span><span class="dv">60</span>)]=<span class="st">&quot;LIVING&quot;</span></span>
<span id="cb270-7"><a href="associate-epcam-inflammatory-with-survival.html#cb270-7"></a></span>
<span id="cb270-8"><a href="associate-epcam-inflammatory-with-survival.html#cb270-8"></a>BrClin<span class="op">$</span>Disease.Free..Months.[<span class="kw">which</span>(BrClin<span class="op">$</span>Disease.Free..Months.<span class="op">&gt;=</span><span class="dv">60</span>)]=<span class="dv">60</span></span>
<span id="cb270-9"><a href="associate-epcam-inflammatory-with-survival.html#cb270-9"></a>BrClin<span class="op">$</span>Disease.Free.Status[<span class="kw">which</span>(BrClin<span class="op">$</span>Disease.Free..Months.<span class="op">&gt;=</span><span class="dv">60</span>)]=<span class="st">&quot;Disease Free&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="associate-epcam-inflammatory-with-survival.html#cb271-1"></a><span class="co">## Load the old file</span></span>
<span id="cb271-2"><a href="associate-epcam-inflammatory-with-survival.html#cb271-2"></a>Nickold=<span class="kw">read.csv</span>(<span class="st">&quot;outputs/EpCAM_IFNg_low_growing_vs_stable.filtered.csv&quot;</span>, <span class="dt">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb271-3"><a href="associate-epcam-inflammatory-with-survival.html#cb271-3"></a></span>
<span id="cb271-4"><a href="associate-epcam-inflammatory-with-survival.html#cb271-4"></a>GlistGrowing=<span class="kw">rownames</span>(Nickold)[<span class="kw">which</span>(<span class="kw">sign</span>(Nickold<span class="op">$</span>log2FoldChange)<span class="op">&lt;</span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>Nickold<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb271-5"><a href="associate-epcam-inflammatory-with-survival.html#cb271-5"></a>Glistnon=<span class="kw">rownames</span>(Nickold)[<span class="kw">which</span>(<span class="kw">sign</span>(Nickold<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>Nickold<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb271-6"><a href="associate-epcam-inflammatory-with-survival.html#cb271-6"></a></span>
<span id="cb271-7"><a href="associate-epcam-inflammatory-with-survival.html#cb271-7"></a><span class="co">#alist=results(vstLumRes)</span></span></code></pre></div>
<div id="associating-cd74-with-phenotype-and-outcome" class="section level2">
<h2><span class="header-section-number">15.1</span> Associating CD74 with phenotype and outcome</h2>
<p>The Thorsson data has pre-calculated scores for:</p>
<ul>
<li>immune subtypes</li>
<li>leukocyte fractins</li>
<li>proportion of data with coding mutations</li>
<li>TCR shannon index</li>
</ul>
<p>Z-Scale CD74 scores prior to analysis:</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="associate-epcam-inflammatory-with-survival.html#cb272-1"></a>TCGAssgsea=<span class="kw">scale</span>(TCGArsem2[<span class="kw">match</span>(<span class="st">&quot;CD74&quot;</span>, <span class="kw">rownames</span>(TCGArsem2)), ])</span>
<span id="cb272-2"><a href="associate-epcam-inflammatory-with-survival.html#cb272-2"></a>m1=<span class="kw">match</span>(<span class="kw">colnames</span>(TCGArsem2), BrClin<span class="op">$</span>Patient.ID)</span>
<span id="cb272-3"><a href="associate-epcam-inflammatory-with-survival.html#cb272-3"></a></span>
<span id="cb272-4"><a href="associate-epcam-inflammatory-with-survival.html#cb272-4"></a>tinfo=<span class="kw">data.frame</span>(<span class="dt">pam=</span>BrClin<span class="op">$</span>PAM50[m1], <span class="dt">OSM=</span>BrClin<span class="op">$</span>Overall.Survival..Months.[m1],</span>
<span id="cb272-5"><a href="associate-epcam-inflammatory-with-survival.html#cb272-5"></a>                 <span class="dt">OSS=</span>BrClin<span class="op">$</span>Overall.Survival.Status[m1], <span class="dt">cd74=</span>TCGAssgsea,</span>
<span id="cb272-6"><a href="associate-epcam-inflammatory-with-survival.html#cb272-6"></a>                 <span class="dt">DFS=</span>BrClin<span class="op">$</span>Disease.Free.Status[m1],</span>
<span id="cb272-7"><a href="associate-epcam-inflammatory-with-survival.html#cb272-7"></a>                 <span class="dt">DFM=</span>BrClin<span class="op">$</span>Disease.Free..Months.[m1],</span>
<span id="cb272-8"><a href="associate-epcam-inflammatory-with-survival.html#cb272-8"></a>                 <span class="dt">Stage=</span>BrClin<span class="op">$</span>Stage[m1])</span>
<span id="cb272-9"><a href="associate-epcam-inflammatory-with-survival.html#cb272-9"></a></span>
<span id="cb272-10"><a href="associate-epcam-inflammatory-with-survival.html#cb272-10"></a>ThorssData=ThorssData[<span class="kw">which</span>(ThorssData<span class="op">$</span>TCGA.Study<span class="op">==</span><span class="st">&quot;BRCA&quot;</span>), ]</span>
<span id="cb272-11"><a href="associate-epcam-inflammatory-with-survival.html#cb272-11"></a></span>
<span id="cb272-12"><a href="associate-epcam-inflammatory-with-survival.html#cb272-12"></a>n1=<span class="kw">match</span>(<span class="kw">rownames</span>(tinfo), ThorssData<span class="op">$</span>TCGA.Participant.Barcode)</span>
<span id="cb272-13"><a href="associate-epcam-inflammatory-with-survival.html#cb272-13"></a>tinfo<span class="op">$</span>immSub=ThorssData<span class="op">$</span>Immune.Subtype[n1]</span>
<span id="cb272-14"><a href="associate-epcam-inflammatory-with-survival.html#cb272-14"></a>tinfo<span class="op">$</span>leukFrac=<span class="kw">as.numeric</span>(ThorssData<span class="op">$</span>Leukocyte.Fraction[n1])</span>
<span id="cb272-15"><a href="associate-epcam-inflammatory-with-survival.html#cb272-15"></a>tinfo<span class="op">$</span>codingMut=<span class="kw">as.numeric</span>(ThorssData<span class="op">$</span>Nonsilent.Mutation.Rate[n1])</span>
<span id="cb272-16"><a href="associate-epcam-inflammatory-with-survival.html#cb272-16"></a>tinfo<span class="op">$</span>TCRshann=<span class="kw">as.numeric</span>(ThorssData<span class="op">$</span>TCR.Shannon[n1])</span>
<span id="cb272-17"><a href="associate-epcam-inflammatory-with-survival.html#cb272-17"></a></span>
<span id="cb272-18"><a href="associate-epcam-inflammatory-with-survival.html#cb272-18"></a></span>
<span id="cb272-19"><a href="associate-epcam-inflammatory-with-survival.html#cb272-19"></a><span class="co">#pdf(&quot;figure-outputs/Figure5_CD74_output.pdf&quot;, height=7, width=9)</span></span>
<span id="cb272-20"><a href="associate-epcam-inflammatory-with-survival.html#cb272-20"></a></span>
<span id="cb272-21"><a href="associate-epcam-inflammatory-with-survival.html#cb272-21"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb272-22"><a href="associate-epcam-inflammatory-with-survival.html#cb272-22"></a><span class="kw">boxplot</span>(tinfo<span class="op">$</span>cd74<span class="op">~</span>tinfo<span class="op">$</span>pam, <span class="dt">ylab=</span><span class="st">&quot;gene exp&quot;</span>, <span class="dt">main=</span><span class="st">&quot;PAM50&quot;</span>)</span>
<span id="cb272-23"><a href="associate-epcam-inflammatory-with-survival.html#cb272-23"></a><span class="kw">boxplot</span>(tinfo<span class="op">$</span>cd74<span class="op">~</span>tinfo<span class="op">$</span>immSub, <span class="dt">ylab=</span><span class="st">&quot;gene exp&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Immune sbtype&quot;</span>)</span>
<span id="cb272-24"><a href="associate-epcam-inflammatory-with-survival.html#cb272-24"></a>ax=<span class="kw">cor.test</span>(tinfo<span class="op">$</span>cd74, <span class="kw">log10</span>(tinfo<span class="op">$</span>codingMut<span class="op">+</span><span class="dv">1</span>), <span class="dt">method=</span><span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb272-25"><a href="associate-epcam-inflammatory-with-survival.html#cb272-25"></a><span class="kw">smoothScatter</span>(tinfo<span class="op">$</span>cd74, <span class="kw">log10</span>(tinfo<span class="op">$</span>codingMut<span class="op">+</span><span class="dv">1</span>), <span class="dt">xlab=</span><span class="st">&quot;gene exp&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;log10 mut&quot;</span>)</span>
<span id="cb272-26"><a href="associate-epcam-inflammatory-with-survival.html#cb272-26"></a><span class="kw">text</span>(<span class="dv">4</span>,<span class="fl">0.5</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(ax<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(ax<span class="op">$</span>p.value, <span class="dv">2</span>)))</span>
<span id="cb272-27"><a href="associate-epcam-inflammatory-with-survival.html#cb272-27"></a>ax=<span class="kw">cor.test</span>(tinfo<span class="op">$</span>cd74, tinfo<span class="op">$</span>leukFrac, <span class="dt">method=</span><span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb272-28"><a href="associate-epcam-inflammatory-with-survival.html#cb272-28"></a><span class="kw">smoothScatter</span>(tinfo<span class="op">$</span>cd74, (tinfo<span class="op">$</span>leukFrac), <span class="dt">xlab=</span><span class="st">&quot;gene exp&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;leuk frac&quot;</span>)</span>
<span id="cb272-29"><a href="associate-epcam-inflammatory-with-survival.html#cb272-29"></a><span class="kw">text</span>(<span class="dv">4</span>,<span class="fl">0.2</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(ax<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(ax<span class="op">$</span>p.value, <span class="dv">2</span>)))</span>
<span id="cb272-30"><a href="associate-epcam-inflammatory-with-survival.html#cb272-30"></a>ax=<span class="kw">cor.test</span>(tinfo<span class="op">$</span>cd74, tinfo<span class="op">$</span>TCRshann, <span class="dt">method=</span><span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb272-31"><a href="associate-epcam-inflammatory-with-survival.html#cb272-31"></a><span class="kw">smoothScatter</span>(tinfo<span class="op">$</span>cd74, tinfo<span class="op">$</span>TCRshann, <span class="dt">xlab=</span><span class="st">&quot;gene exp&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;TCR diversity&quot;</span>)</span>
<span id="cb272-32"><a href="associate-epcam-inflammatory-with-survival.html#cb272-32"></a><span class="kw">text</span>(<span class="dv">4</span>,<span class="dv">1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(ax<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(ax<span class="op">$</span>p.value, <span class="dv">2</span>)))</span>
<span id="cb272-33"><a href="associate-epcam-inflammatory-with-survival.html#cb272-33"></a><span class="kw">hist</span>(tinfo<span class="op">$</span>cd74)</span></code></pre></div>
<div class="figure"><span id="fig:5d"></span>
<img src="RatDCIS-bookdown_files/figure-html/5d-1.png" alt="CD74 assoc with patient data" width="672" />
<p class="caption">
Figure 15.1: CD74 assoc with patient data
</p>
</div>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="associate-epcam-inflammatory-with-survival.html#cb273-1"></a></span>
<span id="cb273-2"><a href="associate-epcam-inflammatory-with-survival.html#cb273-2"></a><span class="co">#dev.off()</span></span>
<span id="cb273-3"><a href="associate-epcam-inflammatory-with-survival.html#cb273-3"></a></span>
<span id="cb273-4"><a href="associate-epcam-inflammatory-with-survival.html#cb273-4"></a><span class="kw">write.csv</span>(tinfo, <span class="dt">file=</span><span class="st">&quot;nature-tables/5d-TCGA_data.csv&quot;</span>)</span></code></pre></div>
<p>We can also check if there is an association with survival:</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="associate-epcam-inflammatory-with-survival.html#cb274-1"></a></span>
<span id="cb274-2"><a href="associate-epcam-inflammatory-with-survival.html#cb274-2"></a>axD=<span class="kw">Surv</span>(tinfo<span class="op">$</span>DFM, <span class="kw">ifelse</span>(tinfo<span class="op">$</span>DFS<span class="op">==</span><span class="st">&quot;Recurred/Progressed&quot;</span>, <span class="dv">1</span>, <span class="kw">ifelse</span>(tinfo<span class="op">$</span>DFS<span class="op">==</span><span class="st">&quot;DiseaseFree&quot;</span>, <span class="dv">0</span>, <span class="ot">NA</span>)))</span>
<span id="cb274-3"><a href="associate-epcam-inflammatory-with-survival.html#cb274-3"></a></span>
<span id="cb274-4"><a href="associate-epcam-inflammatory-with-survival.html#cb274-4"></a>axO=<span class="kw">Surv</span>(tinfo<span class="op">$</span>OSM, <span class="kw">ifelse</span>(tinfo<span class="op">$</span>OSS<span class="op">==</span><span class="st">&quot;DECEASED&quot;</span>, <span class="dv">1</span>, <span class="kw">ifelse</span>(tinfo<span class="op">$</span>OSS<span class="op">==</span><span class="st">&quot;LIVING&quot;</span>, <span class="dv">0</span>, <span class="ot">NA</span>)))</span>
<span id="cb274-5"><a href="associate-epcam-inflammatory-with-survival.html#cb274-5"></a></span>
<span id="cb274-6"><a href="associate-epcam-inflammatory-with-survival.html#cb274-6"></a>SurvOSS=<span class="kw">coxph</span>(axO<span class="op">~</span>tinfo<span class="op">$</span>pam<span class="op">+</span>tinfo<span class="op">$</span>Stage<span class="op">+</span>tinfo<span class="op">$</span>cd74)</span>
<span id="cb274-7"><a href="associate-epcam-inflammatory-with-survival.html#cb274-7"></a>SurvDFS=<span class="kw">coxph</span>(axD<span class="op">~</span>tinfo<span class="op">$</span>pam<span class="op">+</span>tinfo<span class="op">$</span>Stage<span class="op">+</span>tinfo<span class="op">$</span>cd74)</span>
<span id="cb274-8"><a href="associate-epcam-inflammatory-with-survival.html#cb274-8"></a></span>
<span id="cb274-9"><a href="associate-epcam-inflammatory-with-survival.html#cb274-9"></a>ao=<span class="kw">summary</span>(SurvOSS)</span>
<span id="cb274-10"><a href="associate-epcam-inflammatory-with-survival.html#cb274-10"></a>ao</span>
<span id="cb274-11"><a href="associate-epcam-inflammatory-with-survival.html#cb274-11"></a><span class="co">## Call:</span></span>
<span id="cb274-12"><a href="associate-epcam-inflammatory-with-survival.html#cb274-12"></a><span class="co">## coxph(formula = axO ~ tinfo$pam + tinfo$Stage + tinfo$cd74)</span></span>
<span id="cb274-13"><a href="associate-epcam-inflammatory-with-survival.html#cb274-13"></a><span class="co">## </span></span>
<span id="cb274-14"><a href="associate-epcam-inflammatory-with-survival.html#cb274-14"></a><span class="co">##   n= 804, number of events= 78 </span></span>
<span id="cb274-15"><a href="associate-epcam-inflammatory-with-survival.html#cb274-15"></a><span class="co">##    (276 observations deleted due to missingness)</span></span>
<span id="cb274-16"><a href="associate-epcam-inflammatory-with-survival.html#cb274-16"></a><span class="co">## </span></span>
<span id="cb274-17"><a href="associate-epcam-inflammatory-with-survival.html#cb274-17"></a><span class="co">##                    coef exp(coef) se(coef)      z Pr(&gt;|z|)    </span></span>
<span id="cb274-18"><a href="associate-epcam-inflammatory-with-survival.html#cb274-18"></a><span class="co">## tinfo$pamHer2    0.3709    1.4491   0.3780  0.981   0.3265    </span></span>
<span id="cb274-19"><a href="associate-epcam-inflammatory-with-survival.html#cb274-19"></a><span class="co">## tinfo$pamLumA   -0.8365    0.4332   0.3091 -2.707   0.0068 ** </span></span>
<span id="cb274-20"><a href="associate-epcam-inflammatory-with-survival.html#cb274-20"></a><span class="co">## tinfo$pamLumB   -0.3137    0.7308   0.3409 -0.920   0.3574    </span></span>
<span id="cb274-21"><a href="associate-epcam-inflammatory-with-survival.html#cb274-21"></a><span class="co">## tinfo$pamNormal -0.2546    0.7752   0.6282 -0.405   0.6853    </span></span>
<span id="cb274-22"><a href="associate-epcam-inflammatory-with-survival.html#cb274-22"></a><span class="co">## tinfo$Stage2     0.6162    1.8519   0.3309  1.862   0.0626 .  </span></span>
<span id="cb274-23"><a href="associate-epcam-inflammatory-with-survival.html#cb274-23"></a><span class="co">## tinfo$Stage3     0.6636    1.9417   0.4315  1.538   0.1241    </span></span>
<span id="cb274-24"><a href="associate-epcam-inflammatory-with-survival.html#cb274-24"></a><span class="co">## tinfo$Stage4     1.9960    7.3594   0.4255  4.691 2.72e-06 ***</span></span>
<span id="cb274-25"><a href="associate-epcam-inflammatory-with-survival.html#cb274-25"></a><span class="co">## tinfo$cd74      -0.3021    0.7392   0.1530 -1.974   0.0484 *  </span></span>
<span id="cb274-26"><a href="associate-epcam-inflammatory-with-survival.html#cb274-26"></a><span class="co">## ---</span></span>
<span id="cb274-27"><a href="associate-epcam-inflammatory-with-survival.html#cb274-27"></a><span class="co">## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb274-28"><a href="associate-epcam-inflammatory-with-survival.html#cb274-28"></a><span class="co">## </span></span>
<span id="cb274-29"><a href="associate-epcam-inflammatory-with-survival.html#cb274-29"></a><span class="co">##                 exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb274-30"><a href="associate-epcam-inflammatory-with-survival.html#cb274-30"></a><span class="co">## tinfo$pamHer2      1.4491     0.6901    0.6907    3.0400</span></span>
<span id="cb274-31"><a href="associate-epcam-inflammatory-with-survival.html#cb274-31"></a><span class="co">## tinfo$pamLumA      0.4332     2.3083    0.2364    0.7939</span></span>
<span id="cb274-32"><a href="associate-epcam-inflammatory-with-survival.html#cb274-32"></a><span class="co">## tinfo$pamLumB      0.7308     1.3685    0.3747    1.4253</span></span>
<span id="cb274-33"><a href="associate-epcam-inflammatory-with-survival.html#cb274-33"></a><span class="co">## tinfo$pamNormal    0.7752     1.2899    0.2263    2.6557</span></span>
<span id="cb274-34"><a href="associate-epcam-inflammatory-with-survival.html#cb274-34"></a><span class="co">## tinfo$Stage2       1.8519     0.5400    0.9681    3.5425</span></span>
<span id="cb274-35"><a href="associate-epcam-inflammatory-with-survival.html#cb274-35"></a><span class="co">## tinfo$Stage3       1.9417     0.5150    0.8335    4.5234</span></span>
<span id="cb274-36"><a href="associate-epcam-inflammatory-with-survival.html#cb274-36"></a><span class="co">## tinfo$Stage4       7.3594     0.1359    3.1963   16.9448</span></span>
<span id="cb274-37"><a href="associate-epcam-inflammatory-with-survival.html#cb274-37"></a><span class="co">## tinfo$cd74         0.7392     1.3528    0.5477    0.9978</span></span>
<span id="cb274-38"><a href="associate-epcam-inflammatory-with-survival.html#cb274-38"></a><span class="co">## </span></span>
<span id="cb274-39"><a href="associate-epcam-inflammatory-with-survival.html#cb274-39"></a><span class="co">## Concordance= 0.715  (se = 0.031 )</span></span>
<span id="cb274-40"><a href="associate-epcam-inflammatory-with-survival.html#cb274-40"></a><span class="co">## Likelihood ratio test= 42.32  on 8 df,   p=1e-06</span></span>
<span id="cb274-41"><a href="associate-epcam-inflammatory-with-survival.html#cb274-41"></a><span class="co">## Wald test            = 47.41  on 8 df,   p=1e-07</span></span>
<span id="cb274-42"><a href="associate-epcam-inflammatory-with-survival.html#cb274-42"></a><span class="co">## Score (logrank) test = 56.19  on 8 df,   p=3e-09</span></span>
<span id="cb274-43"><a href="associate-epcam-inflammatory-with-survival.html#cb274-43"></a>as=<span class="kw">summary</span>(SurvDFS)</span>
<span id="cb274-44"><a href="associate-epcam-inflammatory-with-survival.html#cb274-44"></a>as</span>
<span id="cb274-45"><a href="associate-epcam-inflammatory-with-survival.html#cb274-45"></a><span class="co">## Call:</span></span>
<span id="cb274-46"><a href="associate-epcam-inflammatory-with-survival.html#cb274-46"></a><span class="co">## coxph(formula = axD ~ tinfo$pam + tinfo$Stage + tinfo$cd74)</span></span>
<span id="cb274-47"><a href="associate-epcam-inflammatory-with-survival.html#cb274-47"></a><span class="co">## </span></span>
<span id="cb274-48"><a href="associate-epcam-inflammatory-with-survival.html#cb274-48"></a><span class="co">##   n= 587, number of events= 69 </span></span>
<span id="cb274-49"><a href="associate-epcam-inflammatory-with-survival.html#cb274-49"></a><span class="co">##    (493 observations deleted due to missingness)</span></span>
<span id="cb274-50"><a href="associate-epcam-inflammatory-with-survival.html#cb274-50"></a><span class="co">## </span></span>
<span id="cb274-51"><a href="associate-epcam-inflammatory-with-survival.html#cb274-51"></a><span class="co">##                    coef exp(coef) se(coef)      z Pr(&gt;|z|)    </span></span>
<span id="cb274-52"><a href="associate-epcam-inflammatory-with-survival.html#cb274-52"></a><span class="co">## tinfo$pamHer2   -0.4626    0.6296   0.4860 -0.952  0.34117    </span></span>
<span id="cb274-53"><a href="associate-epcam-inflammatory-with-survival.html#cb274-53"></a><span class="co">## tinfo$pamLumA   -1.0004    0.3677   0.3158 -3.168  0.00153 ** </span></span>
<span id="cb274-54"><a href="associate-epcam-inflammatory-with-survival.html#cb274-54"></a><span class="co">## tinfo$pamLumB   -0.8377    0.4327   0.3701 -2.263  0.02361 *  </span></span>
<span id="cb274-55"><a href="associate-epcam-inflammatory-with-survival.html#cb274-55"></a><span class="co">## tinfo$pamNormal -0.8810    0.4143   0.7608 -1.158  0.24681    </span></span>
<span id="cb274-56"><a href="associate-epcam-inflammatory-with-survival.html#cb274-56"></a><span class="co">## tinfo$Stage2     0.5470    1.7281   0.3389  1.614  0.10648    </span></span>
<span id="cb274-57"><a href="associate-epcam-inflammatory-with-survival.html#cb274-57"></a><span class="co">## tinfo$Stage3     1.0539    2.8687   0.4440  2.373  0.01762 *  </span></span>
<span id="cb274-58"><a href="associate-epcam-inflammatory-with-survival.html#cb274-58"></a><span class="co">## tinfo$Stage4     2.6058   13.5422   0.4836  5.388 7.13e-08 ***</span></span>
<span id="cb274-59"><a href="associate-epcam-inflammatory-with-survival.html#cb274-59"></a><span class="co">## tinfo$cd74      -0.1134    0.8928   0.1456 -0.779  0.43600    </span></span>
<span id="cb274-60"><a href="associate-epcam-inflammatory-with-survival.html#cb274-60"></a><span class="co">## ---</span></span>
<span id="cb274-61"><a href="associate-epcam-inflammatory-with-survival.html#cb274-61"></a><span class="co">## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb274-62"><a href="associate-epcam-inflammatory-with-survival.html#cb274-62"></a><span class="co">## </span></span>
<span id="cb274-63"><a href="associate-epcam-inflammatory-with-survival.html#cb274-63"></a><span class="co">##                 exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb274-64"><a href="associate-epcam-inflammatory-with-survival.html#cb274-64"></a><span class="co">## tinfo$pamHer2      0.6296    1.58824   0.24287    1.6323</span></span>
<span id="cb274-65"><a href="associate-epcam-inflammatory-with-survival.html#cb274-65"></a><span class="co">## tinfo$pamLumA      0.3677    2.71950   0.19802    0.6828</span></span>
<span id="cb274-66"><a href="associate-epcam-inflammatory-with-survival.html#cb274-66"></a><span class="co">## tinfo$pamLumB      0.4327    2.31113   0.20948    0.8937</span></span>
<span id="cb274-67"><a href="associate-epcam-inflammatory-with-survival.html#cb274-67"></a><span class="co">## tinfo$pamNormal    0.4143    2.41342   0.09329    1.8404</span></span>
<span id="cb274-68"><a href="associate-epcam-inflammatory-with-survival.html#cb274-68"></a><span class="co">## tinfo$Stage2       1.7281    0.57866   0.88943    3.3577</span></span>
<span id="cb274-69"><a href="associate-epcam-inflammatory-with-survival.html#cb274-69"></a><span class="co">## tinfo$Stage3       2.8687    0.34859   1.20151    6.8493</span></span>
<span id="cb274-70"><a href="associate-epcam-inflammatory-with-survival.html#cb274-70"></a><span class="co">## tinfo$Stage4      13.5422    0.07384   5.24827   34.9432</span></span>
<span id="cb274-71"><a href="associate-epcam-inflammatory-with-survival.html#cb274-71"></a><span class="co">## tinfo$cd74         0.8928    1.12006   0.67121    1.1876</span></span>
<span id="cb274-72"><a href="associate-epcam-inflammatory-with-survival.html#cb274-72"></a><span class="co">## </span></span>
<span id="cb274-73"><a href="associate-epcam-inflammatory-with-survival.html#cb274-73"></a><span class="co">## Concordance= 0.69  (se = 0.039 )</span></span>
<span id="cb274-74"><a href="associate-epcam-inflammatory-with-survival.html#cb274-74"></a><span class="co">## Likelihood ratio test= 32.15  on 8 df,   p=9e-05</span></span>
<span id="cb274-75"><a href="associate-epcam-inflammatory-with-survival.html#cb274-75"></a><span class="co">## Wald test            = 39.03  on 8 df,   p=5e-06</span></span>
<span id="cb274-76"><a href="associate-epcam-inflammatory-with-survival.html#cb274-76"></a><span class="co">## Score (logrank) test = 50.73  on 8 df,   p=3e-08</span></span></code></pre></div>
<p>Note that CD74 is associated with survival: The harzards ratio is 1.851886 (95CI:0.9680967, 3.5424989), p value 1.176905110^{-6}.</p>
<p>This means that this is associated with survival regardless of subtype. We can look at these values in forestplots</p>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="associate-epcam-inflammatory-with-survival.html#cb275-1"></a></span>
<span id="cb275-2"><a href="associate-epcam-inflammatory-with-survival.html#cb275-2"></a>data1=<span class="kw">data.frame</span>(<span class="dt">X1=</span><span class="kw">c</span>(<span class="st">&quot;&quot;</span>,<span class="st">&quot;PAM50&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;Stage&quot;</span>, <span class="st">&quot;&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="st">&quot;CD74&quot;</span>),</span>
<span id="cb275-3"><a href="associate-epcam-inflammatory-with-survival.html#cb275-3"></a>                <span class="dt">X2=</span><span class="kw">c</span>(<span class="st">&quot;&quot;</span>,<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Her2&quot;</span>, <span class="st">&quot;LumA&quot;</span>, <span class="st">&quot;LumB&quot;</span>, <span class="st">&quot;Normal&quot;</span> ,<span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>, <span class="st">&quot;4&quot;</span>, <span class="st">&quot;&quot;</span>),</span>
<span id="cb275-4"><a href="associate-epcam-inflammatory-with-survival.html#cb275-4"></a>                <span class="dt">X3=</span><span class="kw">c</span>(<span class="st">&quot;NSamp&quot;</span>,  <span class="kw">table</span>(tinfo<span class="op">$</span>pam), <span class="kw">table</span>(tinfo<span class="op">$</span>Stage), <span class="kw">length</span>(<span class="kw">na.omit</span>(tinfo<span class="op">$</span>cd74))),</span>
<span id="cb275-5"><a href="associate-epcam-inflammatory-with-survival.html#cb275-5"></a>                <span class="dt">X4=</span><span class="kw">c</span>(<span class="st">&quot;HR&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">round</span>(<span class="kw">summary</span>(SurvOSS)<span class="op">$</span>coefficients[ <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>,<span class="dv">2</span>],<span class="dv">2</span>), <span class="st">&quot;&quot;</span>,</span>
<span id="cb275-6"><a href="associate-epcam-inflammatory-with-survival.html#cb275-6"></a>                     <span class="kw">round</span>(<span class="kw">summary</span>(SurvOSS)<span class="op">$</span>coefficients[ <span class="dv">5</span><span class="op">:</span><span class="dv">8</span>,<span class="dv">2</span>],<span class="dv">2</span>)))</span>
<span id="cb275-7"><a href="associate-epcam-inflammatory-with-survival.html#cb275-7"></a></span>
<span id="cb275-8"><a href="associate-epcam-inflammatory-with-survival.html#cb275-8"></a>mdata=<span class="kw">data.frame</span>(<span class="kw">summary</span>(SurvOSS)<span class="op">$</span>conf.int[ ,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)])</span>
<span id="cb275-9"><a href="associate-epcam-inflammatory-with-survival.html#cb275-9"></a>mdata=<span class="kw">rbind</span>(<span class="ot">NA</span>,<span class="ot">NA</span>, mdata[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, ], <span class="ot">NA</span>, mdata[<span class="dv">5</span><span class="op">:</span><span class="dv">8</span>, ])</span>
<span id="cb275-10"><a href="associate-epcam-inflammatory-with-survival.html#cb275-10"></a></span>
<span id="cb275-11"><a href="associate-epcam-inflammatory-with-survival.html#cb275-11"></a><span class="co">#pdf(&quot;figure-outputs/Figure5_CD74_forestplot.pdf&quot;, width=5, height=5)</span></span>
<span id="cb275-12"><a href="associate-epcam-inflammatory-with-survival.html#cb275-12"></a></span>
<span id="cb275-13"><a href="associate-epcam-inflammatory-with-survival.html#cb275-13"></a><span class="kw">forestplot</span>(data1, mdata, <span class="dt">xlog=</span>T,</span>
<span id="cb275-14"><a href="associate-epcam-inflammatory-with-survival.html#cb275-14"></a>           <span class="dt">boxsize=</span><span class="fl">0.5</span>)</span></code></pre></div>
<div class="figure"><span id="fig:5e"></span>
<img src="RatDCIS-bookdown_files/figure-html/5e-1.png" alt="forest plot CD74" width="672" />
<p class="caption">
Figure 15.2: forest plot CD74
</p>
</div>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="associate-epcam-inflammatory-with-survival.html#cb276-1"></a></span>
<span id="cb276-2"><a href="associate-epcam-inflammatory-with-survival.html#cb276-2"></a><span class="co">#dev.off()</span></span>
<span id="cb276-3"><a href="associate-epcam-inflammatory-with-survival.html#cb276-3"></a></span>
<span id="cb276-4"><a href="associate-epcam-inflammatory-with-survival.html#cb276-4"></a><span class="kw">write.csv</span>(<span class="kw">cbind</span>(data1, mdata), <span class="dt">file=</span><span class="st">&quot;nature-tables/5e_forest_plot.csv&quot;</span>)</span></code></pre></div>
<p>We can look below what the association with subtype is in KM curves:</p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="associate-epcam-inflammatory-with-survival.html#cb277-1"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb277-2"><a href="associate-epcam-inflammatory-with-survival.html#cb277-2"></a></span>
<span id="cb277-3"><a href="associate-epcam-inflammatory-with-survival.html#cb277-3"></a>Xind=<span class="kw">c</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Her2&quot;</span>, <span class="st">&quot;LumA&quot;</span>, <span class="st">&quot;LumB&quot;</span>)</span>
<span id="cb277-4"><a href="associate-epcam-inflammatory-with-survival.html#cb277-4"></a></span>
<span id="cb277-5"><a href="associate-epcam-inflammatory-with-survival.html#cb277-5"></a><span class="cf">for</span> (i <span class="cf">in</span> Xind){</span>
<span id="cb277-6"><a href="associate-epcam-inflammatory-with-survival.html#cb277-6"></a>  l1=<span class="kw">which</span>(tinfo<span class="op">$</span>pam<span class="op">==</span>i)</span>
<span id="cb277-7"><a href="associate-epcam-inflammatory-with-survival.html#cb277-7"></a>  ax=<span class="kw">Surv</span>(tinfo<span class="op">$</span>OSM[l1], <span class="kw">ifelse</span>(tinfo<span class="op">$</span>OSS[l1]<span class="op">==</span><span class="st">&quot;DECEASED&quot;</span>, <span class="dv">1</span>, <span class="kw">ifelse</span>(tinfo<span class="op">$</span>OSS[l1]<span class="op">==</span><span class="st">&quot;LIVING&quot;</span>, <span class="dv">0</span>, <span class="ot">NA</span>)))</span>
<span id="cb277-8"><a href="associate-epcam-inflammatory-with-survival.html#cb277-8"></a>  TCGAvalCut=<span class="kw">cut</span>(tinfo<span class="op">$</span>cd74[l1], <span class="kw">quantile</span>(tinfo<span class="op">$</span>cd74[l1], <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>, <span class="fl">0.67</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;M&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb277-9"><a href="associate-epcam-inflammatory-with-survival.html#cb277-9"></a>  ax1=<span class="kw">summary</span>(<span class="kw">coxph</span>(ax<span class="op">~</span>TCGAvalCut<span class="op">+</span>tinfo<span class="op">$</span>Stage[l1]))</span>
<span id="cb277-10"><a href="associate-epcam-inflammatory-with-survival.html#cb277-10"></a>  ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ax<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;CD74 gene exp&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb277-11"><a href="associate-epcam-inflammatory-with-survival.html#cb277-11"></a>  <span class="kw">text</span>(<span class="dv">20</span>, <span class="fl">0.2</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR:%s (%s-%s) p=%s&quot;</span>, <span class="kw">round</span>(ax1<span class="op">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span> ],<span class="dv">2</span>), </span>
<span id="cb277-12"><a href="associate-epcam-inflammatory-with-survival.html#cb277-12"></a>                     <span class="kw">round</span>(ax1<span class="op">$</span>conf.int[<span class="dv">2</span>,<span class="dv">3</span> ],<span class="dv">2</span>), <span class="kw">round</span>(ax1<span class="op">$</span>conf.int[<span class="dv">2</span>, <span class="dv">4</span>],<span class="dv">2</span>),<span class="kw">round</span>(ax1<span class="op">$</span>logtest[<span class="dv">3</span>],<span class="dv">2</span>)))</span>
<span id="cb277-13"><a href="associate-epcam-inflammatory-with-survival.html#cb277-13"></a>}</span></code></pre></div>
<div class="figure"><span id="fig:Ext5e-os"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext5e-os-1.png" alt="OS CD74 by subtype" width="672" />
<p class="caption">
Figure 15.3: OS CD74 by subtype
</p>
</div>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="associate-epcam-inflammatory-with-survival.html#cb278-1"></a></span>
<span id="cb278-2"><a href="associate-epcam-inflammatory-with-survival.html#cb278-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb278-3"><a href="associate-epcam-inflammatory-with-survival.html#cb278-3"></a></span>
<span id="cb278-4"><a href="associate-epcam-inflammatory-with-survival.html#cb278-4"></a><span class="cf">for</span> (i <span class="cf">in</span> Xind){</span>
<span id="cb278-5"><a href="associate-epcam-inflammatory-with-survival.html#cb278-5"></a>   l1=<span class="kw">which</span>(tinfo<span class="op">$</span>pam<span class="op">==</span>i)</span>
<span id="cb278-6"><a href="associate-epcam-inflammatory-with-survival.html#cb278-6"></a>  ax=<span class="kw">Surv</span>(tinfo<span class="op">$</span>DFM[l1], <span class="kw">ifelse</span>(tinfo<span class="op">$</span>DFS[l1]<span class="op">==</span><span class="st">&quot;Recurred/Progressed&quot;</span>, <span class="dv">1</span>, <span class="kw">ifelse</span>(tinfo<span class="op">$</span>DFS[l1]<span class="op">==</span><span class="st">&quot;DiseaseFree&quot;</span>, <span class="dv">0</span>, <span class="ot">NA</span>)))</span>
<span id="cb278-7"><a href="associate-epcam-inflammatory-with-survival.html#cb278-7"></a>  TCGAvalCut=<span class="kw">cut</span>(tinfo<span class="op">$</span>cd74[l1], <span class="kw">quantile</span>(tinfo<span class="op">$</span>cd74[l1], <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>, <span class="fl">0.67</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;M&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb278-8"><a href="associate-epcam-inflammatory-with-survival.html#cb278-8"></a>  ax1=<span class="kw">summary</span>(<span class="kw">coxph</span>(ax<span class="op">~</span>TCGAvalCut<span class="op">+</span>tinfo<span class="op">$</span>Stage[l1]))</span>
<span id="cb278-9"><a href="associate-epcam-inflammatory-with-survival.html#cb278-9"></a>  ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ax<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;CD74 gene exp&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;DFS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb278-10"><a href="associate-epcam-inflammatory-with-survival.html#cb278-10"></a>  <span class="kw">text</span>(<span class="dv">20</span>, <span class="fl">0.2</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR:%s (%s-%s) p=%s&quot;</span>, <span class="kw">round</span>(ax1<span class="op">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span> ],<span class="dv">2</span>), </span>
<span id="cb278-11"><a href="associate-epcam-inflammatory-with-survival.html#cb278-11"></a>                     <span class="kw">round</span>(ax1<span class="op">$</span>conf.int[<span class="dv">2</span>,<span class="dv">3</span> ],<span class="dv">2</span>), <span class="kw">round</span>(ax1<span class="op">$</span>conf.int[<span class="dv">2</span>, <span class="dv">4</span>],<span class="dv">2</span>),<span class="kw">round</span>(ax1<span class="op">$</span>logtest[<span class="dv">3</span>],<span class="dv">2</span>)))</span>
<span id="cb278-12"><a href="associate-epcam-inflammatory-with-survival.html#cb278-12"></a>}</span></code></pre></div>
<div class="figure"><span id="fig:Ext5e-dfs"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext5e-dfs-1.png" alt="DFS CD74 by subtype" width="672" />
<p class="caption">
Figure 15.4: DFS CD74 by subtype
</p>
</div>
</div>
<div id="signature-lum-cases-non-inflammatory-growing-vs-stable" class="section level2">
<h2><span class="header-section-number">15.2</span> Signature: Lum cases non-inflammatory: growing vs stable</h2>
<p>Here, use the signature which was applied before to see the difference between the non-inflammatory growing vs stable samples</p>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="associate-epcam-inflammatory-with-survival.html#cb279-1"></a>GlistGrowing=<span class="kw">rownames</span>(vstLumRes)[<span class="kw">which</span>(vstLumRes<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="fl">1.5</span> <span class="op">&amp;</span><span class="st"> </span>vstLumRes<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span> <span class="op">&amp;</span><span class="st"> </span>vstLumRes<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> )]</span>
<span id="cb279-2"><a href="associate-epcam-inflammatory-with-survival.html#cb279-2"></a>Glistnon=<span class="kw">rownames</span>(vstLumRes)[<span class="kw">which</span>(vstLumRes<span class="op">$</span>log2FoldChange<span class="op">&lt;</span>(<span class="op">-</span><span class="fl">1.5</span>) <span class="op">&amp;</span><span class="st"> </span>vstLumRes<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span> <span class="op">&amp;</span><span class="st"> </span>vstLumRes<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> )]</span>
<span id="cb279-3"><a href="associate-epcam-inflammatory-with-survival.html#cb279-3"></a></span>
<span id="cb279-4"><a href="associate-epcam-inflammatory-with-survival.html#cb279-4"></a>HumGeneList=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(GlistGrowing, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb279-5"><a href="associate-epcam-inflammatory-with-survival.html#cb279-5"></a>HumGeneList2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(GlistGrowing, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb279-6"><a href="associate-epcam-inflammatory-with-survival.html#cb279-6"></a></span>
<span id="cb279-7"><a href="associate-epcam-inflammatory-with-survival.html#cb279-7"></a>TCGAssgsea=<span class="kw">gsva</span>((TCGArsem2), <span class="kw">list</span>(<span class="dt">grow=</span><span class="kw">na.omit</span>(HumGeneList2), </span>
<span id="cb279-8"><a href="associate-epcam-inflammatory-with-survival.html#cb279-8"></a>                                                <span class="dt">stab=</span>Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(Glistnon, Rat2Hum<span class="op">$</span>RGD.symbol)]), <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb279-9"><a href="associate-epcam-inflammatory-with-survival.html#cb279-9"></a></span>
<span id="cb279-10"><a href="associate-epcam-inflammatory-with-survival.html#cb279-10"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb279-11"><a href="associate-epcam-inflammatory-with-survival.html#cb279-11"></a><span class="kw">hist</span>(TCGAssgsea[<span class="dv">1</span>, ], <span class="dt">main=</span><span class="st">&quot;stable&quot;</span>)</span>
<span id="cb279-12"><a href="associate-epcam-inflammatory-with-survival.html#cb279-12"></a><span class="kw">hist</span>(TCGAssgsea[<span class="dv">2</span>, ], <span class="dt">main=</span><span class="st">&quot;growing&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-102-1.png" width="672" /></p>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="associate-epcam-inflammatory-with-survival.html#cb280-1"></a><span class="kw">plot</span>(TCGAssgsea[<span class="dv">1</span>, ], TCGAssgsea[<span class="dv">2</span>, ], <span class="dt">xlab=</span><span class="st">&quot;stable&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;growing&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-102-2.png" width="672" /></p>
<p>Below are the survival curves:</p>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="associate-epcam-inflammatory-with-survival.html#cb281-1"></a><span class="co">#pdf(&quot;figure-outputs/Fig5_survival_analysis_Lum_samples_baseMean_greater100.pdf&quot;, height=8, width = 8)</span></span>
<span id="cb281-2"><a href="associate-epcam-inflammatory-with-survival.html#cb281-2"></a></span>
<span id="cb281-3"><a href="associate-epcam-inflammatory-with-survival.html#cb281-3"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb281-4"><a href="associate-epcam-inflammatory-with-survival.html#cb281-4"></a></span>
<span id="cb281-5"><a href="associate-epcam-inflammatory-with-survival.html#cb281-5"></a>Xind=<span class="kw">c</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Her2&quot;</span>, <span class="st">&quot;LumA&quot;</span>, <span class="st">&quot;LumB&quot;</span>)</span>
<span id="cb281-6"><a href="associate-epcam-inflammatory-with-survival.html#cb281-6"></a></span>
<span id="cb281-7"><a href="associate-epcam-inflammatory-with-survival.html#cb281-7"></a><span class="cf">for</span> (i <span class="cf">in</span> Xind){</span>
<span id="cb281-8"><a href="associate-epcam-inflammatory-with-survival.html#cb281-8"></a>  l1=<span class="kw">which</span>(BrClin<span class="op">$</span>PAM50<span class="op">==</span>i)</span>
<span id="cb281-9"><a href="associate-epcam-inflammatory-with-survival.html#cb281-9"></a>  lx2=<span class="kw">match</span>(BrClin<span class="op">$</span>Patient.ID[l1], <span class="kw">colnames</span>(TCGAssgsea))</span>
<span id="cb281-10"><a href="associate-epcam-inflammatory-with-survival.html#cb281-10"></a>  ax=<span class="kw">Surv</span>(BrClin<span class="op">$</span>Overall.Survival..Months.[l1], <span class="kw">ifelse</span>(BrClin<span class="op">$</span>Overall.Survival.Status[l1]<span class="op">==</span><span class="st">&quot;DECEASED&quot;</span>, <span class="dv">1</span>, <span class="kw">ifelse</span>(BrClin<span class="op">$</span>Overall.Survival.Status[l1]<span class="op">==</span><span class="st">&quot;LIVING&quot;</span>, <span class="dv">0</span>, <span class="ot">NA</span>)))</span>
<span id="cb281-11"><a href="associate-epcam-inflammatory-with-survival.html#cb281-11"></a>   axb=<span class="kw">coxph</span>(ax<span class="op">~</span>TCGAssgsea[<span class="dv">1</span>, lx2]<span class="op">+</span>BrClin<span class="op">$</span>Stage[l1])</span>
<span id="cb281-12"><a href="associate-epcam-inflammatory-with-survival.html#cb281-12"></a>  axc=<span class="kw">summary</span>(axb)</span>
<span id="cb281-13"><a href="associate-epcam-inflammatory-with-survival.html#cb281-13"></a>  TCGAvalCut=<span class="kw">cut</span>(TCGAssgsea[<span class="dv">1</span>, lx2], <span class="kw">quantile</span>(TCGAssgsea[<span class="dv">1</span>, lx2], <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>, <span class="fl">0.67</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;M&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb281-14"><a href="associate-epcam-inflammatory-with-survival.html#cb281-14"></a>  ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ax<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum stable&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb281-15"><a href="associate-epcam-inflammatory-with-survival.html#cb281-15"></a>   <span class="kw">text</span>(<span class="dv">10</span>, <span class="dv">0</span>, <span class="kw">sprintf</span>(<span class="st">&quot;univ. cts. var. HR=%s (%s-%s), p=%s&quot;</span>, <span class="kw">round</span>(axc<span class="op">$</span>coefficients[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>),</span>
<span id="cb281-16"><a href="associate-epcam-inflammatory-with-survival.html#cb281-16"></a>                      <span class="kw">round</span>(axc<span class="op">$</span>conf.int[<span class="dv">1</span>,<span class="dv">3</span>],<span class="dv">2</span>), <span class="kw">round</span>(axc<span class="op">$</span>conf.int[<span class="dv">1</span>,<span class="dv">4</span>],<span class="dv">2</span>),<span class="kw">round</span>(axc<span class="op">$</span>logtest[<span class="dv">3</span>],<span class="dv">2</span> )))</span>
<span id="cb281-17"><a href="associate-epcam-inflammatory-with-survival.html#cb281-17"></a>  </span>
<span id="cb281-18"><a href="associate-epcam-inflammatory-with-survival.html#cb281-18"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-103-1.png" width="672" /></p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="associate-epcam-inflammatory-with-survival.html#cb282-1"></a></span>
<span id="cb282-2"><a href="associate-epcam-inflammatory-with-survival.html#cb282-2"></a></span>
<span id="cb282-3"><a href="associate-epcam-inflammatory-with-survival.html#cb282-3"></a><span class="cf">for</span> (i <span class="cf">in</span> Xind){</span>
<span id="cb282-4"><a href="associate-epcam-inflammatory-with-survival.html#cb282-4"></a>  l1=<span class="kw">which</span>(BrClin<span class="op">$</span>PAM50<span class="op">==</span>i)</span>
<span id="cb282-5"><a href="associate-epcam-inflammatory-with-survival.html#cb282-5"></a>  lx2=<span class="kw">match</span>(BrClin<span class="op">$</span>Patient.ID[l1], <span class="kw">colnames</span>(TCGAssgsea))</span>
<span id="cb282-6"><a href="associate-epcam-inflammatory-with-survival.html#cb282-6"></a>  ax=<span class="kw">Surv</span>(BrClin<span class="op">$</span>Disease.Free..Months.[l1], <span class="kw">ifelse</span>(BrClin<span class="op">$</span>Disease.Free.Status[l1]<span class="op">==</span><span class="st">&quot;Recurred/Progressed&quot;</span>, <span class="dv">1</span>, <span class="kw">ifelse</span>(BrClin<span class="op">$</span>Disease.Free.Status[l1]<span class="op">==</span><span class="st">&quot;DiseaseFree&quot;</span>, <span class="dv">0</span>, <span class="ot">NA</span>)))</span>
<span id="cb282-7"><a href="associate-epcam-inflammatory-with-survival.html#cb282-7"></a>  axb=<span class="kw">coxph</span>(ax<span class="op">~</span>TCGAssgsea[<span class="dv">1</span>, lx2]<span class="op">+</span>BrClin<span class="op">$</span>Stage[l1])</span>
<span id="cb282-8"><a href="associate-epcam-inflammatory-with-survival.html#cb282-8"></a>  axc=<span class="kw">summary</span>(axb)</span>
<span id="cb282-9"><a href="associate-epcam-inflammatory-with-survival.html#cb282-9"></a>  </span>
<span id="cb282-10"><a href="associate-epcam-inflammatory-with-survival.html#cb282-10"></a>  TCGAvalCut=<span class="kw">cut</span>(TCGAssgsea[<span class="dv">1</span>, lx2], <span class="kw">quantile</span>(TCGAssgsea[<span class="dv">1</span>, lx2], <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>, <span class="fl">0.67</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;M&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb282-11"><a href="associate-epcam-inflammatory-with-survival.html#cb282-11"></a>  ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ax<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum stable&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;DFS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb282-12"><a href="associate-epcam-inflammatory-with-survival.html#cb282-12"></a>  <span class="kw">text</span>(<span class="dv">10</span>, <span class="dv">0</span>, <span class="kw">sprintf</span>(<span class="st">&quot;univ. cts. var. HR=%s (%s-%s), p=%s&quot;</span>, <span class="kw">round</span>(axc<span class="op">$</span>coefficients[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>),</span>
<span id="cb282-13"><a href="associate-epcam-inflammatory-with-survival.html#cb282-13"></a>                      <span class="kw">round</span>(axc<span class="op">$</span>conf.int[<span class="dv">1</span>,<span class="dv">3</span>],<span class="dv">2</span>), <span class="kw">round</span>(axc<span class="op">$</span>conf.int[<span class="dv">1</span>,<span class="dv">4</span>],<span class="dv">2</span>),<span class="kw">round</span>(axc<span class="op">$</span>logtest[<span class="dv">3</span>],<span class="dv">2</span> )))</span>
<span id="cb282-14"><a href="associate-epcam-inflammatory-with-survival.html#cb282-14"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-103-2.png" width="672" /></p>
<p>Below are firstly the OS curves for the growing signature</p>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="associate-epcam-inflammatory-with-survival.html#cb283-1"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb283-2"><a href="associate-epcam-inflammatory-with-survival.html#cb283-2"></a></span>
<span id="cb283-3"><a href="associate-epcam-inflammatory-with-survival.html#cb283-3"></a>Xind=<span class="kw">c</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Her2&quot;</span>, <span class="st">&quot;LumA&quot;</span>, <span class="st">&quot;LumB&quot;</span>)</span>
<span id="cb283-4"><a href="associate-epcam-inflammatory-with-survival.html#cb283-4"></a></span>
<span id="cb283-5"><a href="associate-epcam-inflammatory-with-survival.html#cb283-5"></a><span class="cf">for</span> (i <span class="cf">in</span> Xind){</span>
<span id="cb283-6"><a href="associate-epcam-inflammatory-with-survival.html#cb283-6"></a>  l1=<span class="kw">which</span>(BrClin<span class="op">$</span>PAM50<span class="op">==</span>i)</span>
<span id="cb283-7"><a href="associate-epcam-inflammatory-with-survival.html#cb283-7"></a>   lx2=<span class="kw">match</span>(BrClin<span class="op">$</span>Patient.ID[l1], <span class="kw">colnames</span>(TCGAssgsea))</span>
<span id="cb283-8"><a href="associate-epcam-inflammatory-with-survival.html#cb283-8"></a>  ax=<span class="kw">Surv</span>(BrClin<span class="op">$</span>Overall.Survival..Months.[l1], <span class="kw">ifelse</span>(BrClin<span class="op">$</span>Overall.Survival.Status[l1]<span class="op">==</span><span class="st">&quot;DECEASED&quot;</span>, <span class="dv">1</span>, <span class="kw">ifelse</span>(BrClin<span class="op">$</span>Overall.Survival.Status[l1]<span class="op">==</span><span class="st">&quot;LIVING&quot;</span>, <span class="dv">0</span>, <span class="ot">NA</span>)))</span>
<span id="cb283-9"><a href="associate-epcam-inflammatory-with-survival.html#cb283-9"></a>   axb=<span class="kw">coxph</span>(ax<span class="op">~</span>TCGAssgsea[<span class="dv">2</span>, lx2]<span class="op">+</span>BrClin<span class="op">$</span>Stage[l1])</span>
<span id="cb283-10"><a href="associate-epcam-inflammatory-with-survival.html#cb283-10"></a>  axc=<span class="kw">summary</span>(axb)</span>
<span id="cb283-11"><a href="associate-epcam-inflammatory-with-survival.html#cb283-11"></a>  </span>
<span id="cb283-12"><a href="associate-epcam-inflammatory-with-survival.html#cb283-12"></a>  TCGAvalCut=<span class="kw">cut</span>(TCGAssgsea[<span class="dv">2</span>, lx2], <span class="kw">quantile</span>(TCGAssgsea[<span class="dv">2</span>, lx2], <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>, <span class="fl">0.67</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;M&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb283-13"><a href="associate-epcam-inflammatory-with-survival.html#cb283-13"></a>  ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ax<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum growing&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb283-14"><a href="associate-epcam-inflammatory-with-survival.html#cb283-14"></a>   <span class="kw">text</span>(<span class="dv">10</span>, <span class="dv">0</span>, <span class="kw">sprintf</span>(<span class="st">&quot;univ. cts. var. HR=%s (%s-%s), p=%s&quot;</span>, <span class="kw">round</span>(axc<span class="op">$</span>coefficients[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>),</span>
<span id="cb283-15"><a href="associate-epcam-inflammatory-with-survival.html#cb283-15"></a>                      <span class="kw">round</span>(axc<span class="op">$</span>conf.int[<span class="dv">1</span>,<span class="dv">3</span>],<span class="dv">2</span>), <span class="kw">round</span>(axc<span class="op">$</span>conf.int[<span class="dv">1</span>,<span class="dv">4</span>],<span class="dv">2</span>),<span class="kw">round</span>(axc<span class="op">$</span>logtest[<span class="dv">3</span>],<span class="dv">2</span> )))</span>
<span id="cb283-16"><a href="associate-epcam-inflammatory-with-survival.html#cb283-16"></a>  </span>
<span id="cb283-17"><a href="associate-epcam-inflammatory-with-survival.html#cb283-17"></a>}</span></code></pre></div>
<div class="figure"><span id="fig:5i"></span>
<img src="RatDCIS-bookdown_files/figure-html/5i-1.png" alt="OS: growing signature" width="672" />
<p class="caption">
Figure 15.5: OS: growing signature
</p>
</div>
<p>Followed by the DFS plots for this signature</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="associate-epcam-inflammatory-with-survival.html#cb284-1"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb284-2"><a href="associate-epcam-inflammatory-with-survival.html#cb284-2"></a></span>
<span id="cb284-3"><a href="associate-epcam-inflammatory-with-survival.html#cb284-3"></a><span class="cf">for</span> (i <span class="cf">in</span> Xind){</span>
<span id="cb284-4"><a href="associate-epcam-inflammatory-with-survival.html#cb284-4"></a>  l1=<span class="kw">which</span>(BrClin<span class="op">$</span>PAM50<span class="op">==</span>i)</span>
<span id="cb284-5"><a href="associate-epcam-inflammatory-with-survival.html#cb284-5"></a>   lx2=<span class="kw">match</span>(BrClin<span class="op">$</span>Patient.ID[l1], <span class="kw">colnames</span>(TCGAssgsea))</span>
<span id="cb284-6"><a href="associate-epcam-inflammatory-with-survival.html#cb284-6"></a>  ax=<span class="kw">Surv</span>(BrClin<span class="op">$</span>Disease.Free..Months.[l1], <span class="kw">ifelse</span>(BrClin<span class="op">$</span>Disease.Free.Status[l1]<span class="op">==</span><span class="st">&quot;Recurred/Progressed&quot;</span>, <span class="dv">1</span>, <span class="kw">ifelse</span>(BrClin<span class="op">$</span>Disease.Free.Status[l1]<span class="op">==</span><span class="st">&quot;DiseaseFree&quot;</span>, <span class="dv">0</span>, <span class="ot">NA</span>)))</span>
<span id="cb284-7"><a href="associate-epcam-inflammatory-with-survival.html#cb284-7"></a>   axb=<span class="kw">coxph</span>(ax<span class="op">~</span>TCGAssgsea[<span class="dv">2</span>, lx2]<span class="op">+</span>BrClin<span class="op">$</span>Stage[l1])</span>
<span id="cb284-8"><a href="associate-epcam-inflammatory-with-survival.html#cb284-8"></a>  axc=<span class="kw">summary</span>(axb)</span>
<span id="cb284-9"><a href="associate-epcam-inflammatory-with-survival.html#cb284-9"></a>  </span>
<span id="cb284-10"><a href="associate-epcam-inflammatory-with-survival.html#cb284-10"></a>  TCGAvalCut=<span class="kw">cut</span>(TCGAssgsea[<span class="dv">2</span>, lx2], <span class="kw">quantile</span>(TCGAssgsea[<span class="dv">2</span>, lx2], <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>, <span class="fl">0.67</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;M&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb284-11"><a href="associate-epcam-inflammatory-with-survival.html#cb284-11"></a>  ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ax<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum growing&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;DFS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb284-12"><a href="associate-epcam-inflammatory-with-survival.html#cb284-12"></a>    <span class="kw">text</span>(<span class="dv">10</span>, <span class="dv">0</span>, <span class="kw">sprintf</span>(<span class="st">&quot;univ. cts. var. HR=%s (%s-%s), p=%s&quot;</span>, <span class="kw">round</span>(axc<span class="op">$</span>coefficients[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>),</span>
<span id="cb284-13"><a href="associate-epcam-inflammatory-with-survival.html#cb284-13"></a>                      <span class="kw">round</span>(axc<span class="op">$</span>conf.int[<span class="dv">1</span>,<span class="dv">3</span>],<span class="dv">2</span>), <span class="kw">round</span>(axc<span class="op">$</span>conf.int[<span class="dv">1</span>,<span class="dv">4</span>],<span class="dv">2</span>),<span class="kw">round</span>(axc<span class="op">$</span>logtest[<span class="dv">3</span>],<span class="dv">2</span> )))</span>
<span id="cb284-14"><a href="associate-epcam-inflammatory-with-survival.html#cb284-14"></a>  </span>
<span id="cb284-15"><a href="associate-epcam-inflammatory-with-survival.html#cb284-15"></a>}</span></code></pre></div>
<div class="figure"><span id="fig:5i-dfs"></span>
<img src="RatDCIS-bookdown_files/figure-html/5i-dfs-1.png" alt="DFS forgrowing signature" width="672" />
<p class="caption">
Figure 15.6: DFS forgrowing signature
</p>
</div>
<p>The gene lists are</p>
<p>Stable:</p>
<p>Fgfr2, Mx1, Endou, Grhl3, Slpi, Gbp2, Plg, Aldh1a3, Upk3a</p>
<p>Growing:</p>
<p>Setd7, Hspa1l, Hbb, Kifc2, Kmt5c, Pik3c2b, Lmntd2, Nsmf, Nrbp2, LOC100134871, Hsp90aa1, Coro6, Chst8, Slc4a3, Fhod1, Rpl8, Spp1, Catsperg, Col5a1, LOC100911498, Noxa1, F8, Rps19, Rps6, Irs1, Leng8, Plekhh1, Lmbr1l, Ikbke, Pnisr, Adamts10, Zfp692, Trim41, Sema6d, Mgp, Cdc42bpg, Slc16a13, Dmpk</p>
<p>We can see whether this signature associates with leukocyte content and TCR diversity in the TCGA cohort:</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="associate-epcam-inflammatory-with-survival.html#cb285-1"></a></span>
<span id="cb285-2"><a href="associate-epcam-inflammatory-with-survival.html#cb285-2"></a>l1=<span class="kw">which</span>(BrClin<span class="op">$</span>PAM50<span class="op">==</span><span class="st">&quot;LumA&quot;</span>)</span>
<span id="cb285-3"><a href="associate-epcam-inflammatory-with-survival.html#cb285-3"></a>lx2=<span class="kw">match</span>(BrClin<span class="op">$</span>Patient.ID[l1], <span class="kw">colnames</span>(TCGAssgsea))</span>
<span id="cb285-4"><a href="associate-epcam-inflammatory-with-survival.html#cb285-4"></a></span>
<span id="cb285-5"><a href="associate-epcam-inflammatory-with-survival.html#cb285-5"></a>lx3=<span class="kw">match</span>(BrClin<span class="op">$</span>Patient.ID[l1], ThorssData<span class="op">$</span>TCGA.Participant.Barcode)</span>
<span id="cb285-6"><a href="associate-epcam-inflammatory-with-survival.html#cb285-6"></a></span>
<span id="cb285-7"><a href="associate-epcam-inflammatory-with-survival.html#cb285-7"></a>newinfoTab=<span class="kw">data.frame</span>(<span class="dt">pat=</span>BrClin<span class="op">$</span>Patient.ID[l1], <span class="dt">TCGAval=</span>TCGAssgsea[<span class="dv">2</span>, lx2],</span>
<span id="cb285-8"><a href="associate-epcam-inflammatory-with-survival.html#cb285-8"></a>                      <span class="dt">leuk=</span><span class="kw">as.numeric</span>(ThorssData<span class="op">$</span>Leukocyte.Fraction[lx3]),</span>
<span id="cb285-9"><a href="associate-epcam-inflammatory-with-survival.html#cb285-9"></a>                      <span class="dt">TCRshann=</span><span class="kw">as.numeric</span>(ThorssData<span class="op">$</span>TCR.Shannon[lx3]),</span>
<span id="cb285-10"><a href="associate-epcam-inflammatory-with-survival.html#cb285-10"></a>                      <span class="dt">immSub=</span>ThorssData<span class="op">$</span>Immune.Subtype[lx3])</span>
<span id="cb285-11"><a href="associate-epcam-inflammatory-with-survival.html#cb285-11"></a></span>
<span id="cb285-12"><a href="associate-epcam-inflammatory-with-survival.html#cb285-12"></a></span>
<span id="cb285-13"><a href="associate-epcam-inflammatory-with-survival.html#cb285-13"></a><span class="co">#pdf(&quot;figure-outputs/Fig5_assoc_growing_stable_signature.pdf&quot;, height=6, width=9)</span></span>
<span id="cb285-14"><a href="associate-epcam-inflammatory-with-survival.html#cb285-14"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb285-15"><a href="associate-epcam-inflammatory-with-survival.html#cb285-15"></a>ax=<span class="kw">cor.test</span>(newinfoTab<span class="op">$</span>TCGA, newinfoTab<span class="op">$</span>leuk, <span class="dt">method=</span><span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb285-16"><a href="associate-epcam-inflammatory-with-survival.html#cb285-16"></a><span class="kw">smoothScatter</span>(newinfoTab<span class="op">$</span>TCGA, newinfoTab<span class="op">$</span>leuk)</span>
<span id="cb285-17"><a href="associate-epcam-inflammatory-with-survival.html#cb285-17"></a><span class="kw">text</span>(<span class="fl">0.6</span>,<span class="fl">0.6</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(ax<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(ax<span class="op">$</span>p.value, <span class="dv">2</span>)))</span>
<span id="cb285-18"><a href="associate-epcam-inflammatory-with-survival.html#cb285-18"></a>ax=<span class="kw">cor.test</span>(newinfoTab<span class="op">$</span>TCGA, newinfoTab<span class="op">$</span>TCRshann, <span class="dt">method=</span><span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb285-19"><a href="associate-epcam-inflammatory-with-survival.html#cb285-19"></a><span class="kw">smoothScatter</span>(newinfoTab<span class="op">$</span>TCGA, newinfoTab<span class="op">$</span>TCRshann)</span>
<span id="cb285-20"><a href="associate-epcam-inflammatory-with-survival.html#cb285-20"></a><span class="kw">text</span>(<span class="fl">0.6</span>,<span class="dv">5</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(ax<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(ax<span class="op">$</span>p.value, <span class="dv">2</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:5h"></span>
<img src="RatDCIS-bookdown_files/figure-html/5h-1.png" alt="luminal signature associated with clinical variables" width="672" />
<p class="caption">
Figure 15.7: luminal signature associated with clinical variables
</p>
</div>
<p>As well as whether it associates with any of the previously identified immune subtypes:</p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="associate-epcam-inflammatory-with-survival.html#cb286-1"></a><span class="kw">ggplot</span>(newinfoTab, <span class="kw">aes</span>(<span class="dt">x=</span>immSub, <span class="dt">y=</span>TCGAval))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/thorrs-subtypes-1.png" width="672" /></p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="associate-epcam-inflammatory-with-survival.html#cb287-1"></a><span class="kw">write.csv</span>(newinfoTab, <span class="dt">file=</span><span class="st">&quot;nature-tables/5h_lum_growing_vs_stable.csv&quot;</span>)</span></code></pre></div>
<div id="tcga-luma-high-vs-low-signatures" class="section level3">
<h3><span class="header-section-number">15.2.1</span> TCGA: LumA high vs low signatures</h3>
<p>We can pull out the samples here which have high vs low signature (in the lumA group) and perform differential gene expression analysis on these cases:</p>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="associate-epcam-inflammatory-with-survival.html#cb288-1"></a>newinfoTab<span class="op">$</span>TCGAvalcut=<span class="kw">cut</span>(newinfoTab<span class="op">$</span>TCGAval,  <span class="kw">quantile</span>(newinfoTab<span class="op">$</span>TCGAval, <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>, <span class="fl">0.67</span>, <span class="dv">1</span>) ), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;M&quot;</span>, <span class="st">&quot;O&quot;</span>))</span>
<span id="cb288-2"><a href="associate-epcam-inflammatory-with-survival.html#cb288-2"></a></span>
<span id="cb288-3"><a href="associate-epcam-inflammatory-with-survival.html#cb288-3"></a>colDataTCGA=newinfoTab[<span class="kw">which</span>(newinfoTab<span class="op">$</span>TCGAvalcut<span class="op">!=</span><span class="st">&quot;M&quot;</span>), ]</span>
<span id="cb288-4"><a href="associate-epcam-inflammatory-with-survival.html#cb288-4"></a></span>
<span id="cb288-5"><a href="associate-epcam-inflammatory-with-survival.html#cb288-5"></a><span class="kw">head</span>(colDataTCGA)</span>
<span id="cb288-6"><a href="associate-epcam-inflammatory-with-survival.html#cb288-6"></a><span class="co">##            pat   TCGAval      leuk TCRshann immSub TCGAvalcut</span></span>
<span id="cb288-7"><a href="associate-epcam-inflammatory-with-survival.html#cb288-7"></a><span class="co">## 1 TCGA-A1-A0SD 0.4276658 0.1307793 2.966943     C2          L</span></span>
<span id="cb288-8"><a href="associate-epcam-inflammatory-with-survival.html#cb288-8"></a><span class="co">## 2 TCGA-A1-A0SE 0.4281833 0.1157228 3.404928     C1          L</span></span>
<span id="cb288-9"><a href="associate-epcam-inflammatory-with-survival.html#cb288-9"></a><span class="co">## 3 TCGA-A1-A0SF 0.4348571 0.2502148 4.145625     C1          L</span></span>
<span id="cb288-10"><a href="associate-epcam-inflammatory-with-survival.html#cb288-10"></a><span class="co">## 4 TCGA-A1-A0SH 0.4311287 0.1707323 2.441015     C6          L</span></span>
<span id="cb288-11"><a href="associate-epcam-inflammatory-with-survival.html#cb288-11"></a><span class="co">## 5 TCGA-A1-A0SJ 0.4293603 0.1281564 2.978508     C1          L</span></span>
<span id="cb288-12"><a href="associate-epcam-inflammatory-with-survival.html#cb288-12"></a><span class="co">## 7 TCGA-A1-A0SQ 0.4376568 0.1028228 2.025326     C3          L</span></span>
<span id="cb288-13"><a href="associate-epcam-inflammatory-with-survival.html#cb288-13"></a></span>
<span id="cb288-14"><a href="associate-epcam-inflammatory-with-survival.html#cb288-14"></a>TCGAdeseq=<span class="kw">DESeqDataSetFromMatrix</span>(<span class="kw">round</span>(TCGArsem2[ ,<span class="kw">match</span>(colDataTCGA<span class="op">$</span>pat, <span class="kw">colnames</span>(TCGArsem2))]), colDataTCGA, <span class="dt">design=</span><span class="op">~</span>TCGAvalcut)</span>
<span id="cb288-15"><a href="associate-epcam-inflammatory-with-survival.html#cb288-15"></a></span>
<span id="cb288-16"><a href="associate-epcam-inflammatory-with-survival.html#cb288-16"></a>TCGAdeseq=<span class="kw">DESeq</span>(TCGAdeseq)</span>
<span id="cb288-17"><a href="associate-epcam-inflammatory-with-survival.html#cb288-17"></a></span>
<span id="cb288-18"><a href="associate-epcam-inflammatory-with-survival.html#cb288-18"></a>TCGAdeseqRes=<span class="kw">results</span>(TCGAdeseq)</span>
<span id="cb288-19"><a href="associate-epcam-inflammatory-with-survival.html#cb288-19"></a></span>
<span id="cb288-20"><a href="associate-epcam-inflammatory-with-survival.html#cb288-20"></a><span class="kw">write.csv</span>(TCGAdeseqRes, <span class="dt">file=</span><span class="st">&quot;nature-tables/5J_TCGA.csv&quot;</span>)</span>
<span id="cb288-21"><a href="associate-epcam-inflammatory-with-survival.html#cb288-21"></a></span>
<span id="cb288-22"><a href="associate-epcam-inflammatory-with-survival.html#cb288-22"></a>TCGAdeseqRes2=TCGAdeseqRes[<span class="kw">which</span>(TCGAdeseqRes<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(TCGAdeseqRes<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span> </span>
<span id="cb288-23"><a href="associate-epcam-inflammatory-with-survival.html#cb288-23"></a>                                 <span class="op">&amp;</span><span class="st"> </span>TCGAdeseqRes<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>), ]</span></code></pre></div>
<p>We can plot the differential genes here in a volcano plot:</p>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="associate-epcam-inflammatory-with-survival.html#cb289-1"></a><span class="kw">pdf</span>(<span class="st">&quot;figure-outputs/Fig5J_TCGA_deg.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">8</span>, <span class="dt">width=</span><span class="dv">8</span>)</span>
<span id="cb289-2"><a href="associate-epcam-inflammatory-with-survival.html#cb289-2"></a><span class="kw">with</span>(TCGAdeseqRes, <span class="kw">plot</span>(log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">main=</span><span class="st">&quot;Volcano plot: Low exp (+) vs high (-)&quot;</span>, <span class="dt">cex=</span><span class="fl">1.0</span>, <span class="dt">xlab=</span><span class="kw">bquote</span>(<span class="op">~</span>Log[<span class="dv">2</span>]<span class="op">~</span>fold<span class="op">~</span>change), <span class="dt">ylab=</span><span class="kw">bquote</span>(<span class="op">~-</span>log[<span class="dv">10</span>]<span class="op">~</span>Q<span class="op">~</span>value)))</span>
<span id="cb289-3"><a href="associate-epcam-inflammatory-with-survival.html#cb289-3"></a><span class="kw">with</span>(<span class="kw">subset</span>(TCGAdeseqRes, padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), <span class="kw">points</span>(log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">cex=</span><span class="fl">0.5</span>))</span>
<span id="cb289-4"><a href="associate-epcam-inflammatory-with-survival.html#cb289-4"></a><span class="kw">with</span>(<span class="kw">subset</span>(TCGAdeseqRes, padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), <span class="kw">text</span>(log2FoldChange<span class="fl">+0.05</span>, <span class="op">-</span><span class="kw">log10</span>(padj)<span class="op">+</span><span class="fl">0.05</span>, <span class="kw">rownames</span>(TCGAdeseqRes2), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">cex=</span><span class="fl">0.75</span>))</span>
<span id="cb289-5"><a href="associate-epcam-inflammatory-with-survival.html#cb289-5"></a><span class="kw">dev.off</span>()</span>
<span id="cb289-6"><a href="associate-epcam-inflammatory-with-survival.html#cb289-6"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb289-7"><a href="associate-epcam-inflammatory-with-survival.html#cb289-7"></a><span class="co">##                 2</span></span></code></pre></div>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="associate-epcam-inflammatory-with-survival.html#cb290-1"></a>vstTCGA=<span class="kw">vst</span>(<span class="kw">assay</span>(TCGAdeseq))</span>
<span id="cb290-2"><a href="associate-epcam-inflammatory-with-survival.html#cb290-2"></a></span>
<span id="cb290-3"><a href="associate-epcam-inflammatory-with-survival.html#cb290-3"></a>ColSideCols=<span class="kw">c</span>(<span class="st">&quot;skyblue&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(TCGAdeseq<span class="op">$</span>TCGAvalcut)]</span>
<span id="cb290-4"><a href="associate-epcam-inflammatory-with-survival.html#cb290-4"></a></span>
<span id="cb290-5"><a href="associate-epcam-inflammatory-with-survival.html#cb290-5"></a>ax1=vstTCGA[ <span class="kw">match</span>(<span class="kw">rownames</span>(TCGAdeseqRes2), <span class="kw">rownames</span>(vstTCGA)),  ]</span>
<span id="cb290-6"><a href="associate-epcam-inflammatory-with-survival.html#cb290-6"></a>ax1[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb290-7"><a href="associate-epcam-inflammatory-with-survival.html#cb290-7"></a><span class="co">##          TCGA-A1-A0SD TCGA-A1-A0SE TCGA-A1-A0SF TCGA-A1-A0SH TCGA-A1-A0SJ</span></span>
<span id="cb290-8"><a href="associate-epcam-inflammatory-with-survival.html#cb290-8"></a><span class="co">## ARHGAP33     7.384462     7.604834     8.112232     6.986648     8.694451</span></span>
<span id="cb290-9"><a href="associate-epcam-inflammatory-with-survival.html#cb290-9"></a><span class="co">## ATG16L2      8.875778     7.670891     8.619986     7.414996     9.107129</span></span>
<span id="cb290-10"><a href="associate-epcam-inflammatory-with-survival.html#cb290-10"></a><span class="co">## ATHL1        8.714318    10.375118     8.319105     8.276105     7.447886</span></span>
<span id="cb290-11"><a href="associate-epcam-inflammatory-with-survival.html#cb290-11"></a><span class="co">## C1orf95      7.008673     6.021883     6.083560     5.901986     9.197793</span></span>
<span id="cb290-12"><a href="associate-epcam-inflammatory-with-survival.html#cb290-12"></a><span class="co">## CARTPT       8.789410     5.223711     5.223711     5.223711     5.223711</span></span>
<span id="cb290-13"><a href="associate-epcam-inflammatory-with-survival.html#cb290-13"></a></span>
<span id="cb290-14"><a href="associate-epcam-inflammatory-with-survival.html#cb290-14"></a><span class="kw">pdf</span>(<span class="st">&quot;figure-outputs/Figure5J_XX_heatmap_LumA_TCGA.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">15</span>, <span class="dt">width=</span><span class="dv">10</span>)</span>
<span id="cb290-15"><a href="associate-epcam-inflammatory-with-survival.html#cb290-15"></a></span>
<span id="cb290-16"><a href="associate-epcam-inflammatory-with-survival.html#cb290-16"></a><span class="kw">heatmap.2</span>(vstTCGA[ <span class="kw">rownames</span>(TCGAdeseqRes2),  ], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSideCols, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;LumA TCGA&quot;</span>)</span>
<span id="cb290-17"><a href="associate-epcam-inflammatory-with-survival.html#cb290-17"></a></span>
<span id="cb290-18"><a href="associate-epcam-inflammatory-with-survival.html#cb290-18"></a><span class="kw">dev.off</span>()</span>
<span id="cb290-19"><a href="associate-epcam-inflammatory-with-survival.html#cb290-19"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb290-20"><a href="associate-epcam-inflammatory-with-survival.html#cb290-20"></a><span class="co">##                 2</span></span>
<span id="cb290-21"><a href="associate-epcam-inflammatory-with-survival.html#cb290-21"></a></span>
<span id="cb290-22"><a href="associate-epcam-inflammatory-with-survival.html#cb290-22"></a></span>
<span id="cb290-23"><a href="associate-epcam-inflammatory-with-survival.html#cb290-23"></a><span class="kw">boxplot</span>(vstTCGA[<span class="st">&quot;CXCL13&quot;</span>, ]<span class="op">~</span>TCGAdeseq<span class="op">$</span>TCGAvalcut)</span></code></pre></div>
<div class="figure"><span id="fig:heatmap-tcga"></span>
<img src="RatDCIS-bookdown_files/figure-html/heatmap-tcga-1.png" alt="because someone will ask for it" width="672" />
<p class="caption">
Figure 15.8: because someone will ask for it
</p>
</div>
<p>Run gsea here</p>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="associate-epcam-inflammatory-with-survival.html#cb291-1"></a></span>
<span id="cb291-2"><a href="associate-epcam-inflammatory-with-survival.html#cb291-2"></a>hits=<span class="kw">rownames</span>(TCGAdeseqRes2)</span>
<span id="cb291-3"><a href="associate-epcam-inflammatory-with-survival.html#cb291-3"></a>fcTab=TCGAdeseqRes<span class="op">$</span>log2FoldChange</span>
<span id="cb291-4"><a href="associate-epcam-inflammatory-with-survival.html#cb291-4"></a></span>
<span id="cb291-5"><a href="associate-epcam-inflammatory-with-survival.html#cb291-5"></a><span class="kw">names</span>(fcTab)=<span class="kw">rownames</span>(TCGAdeseq)</span>
<span id="cb291-6"><a href="associate-epcam-inflammatory-with-survival.html#cb291-6"></a>  gscaTCGA=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb291-7"><a href="associate-epcam-inflammatory-with-survival.html#cb291-7"></a>   gscaTCGA&lt;-<span class="st"> </span><span class="kw">preprocess</span>( gscaTCGA, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb291-8"><a href="associate-epcam-inflammatory-with-survival.html#cb291-8"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb291-9"><a href="associate-epcam-inflammatory-with-survival.html#cb291-9"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb291-10"><a href="associate-epcam-inflammatory-with-survival.html#cb291-10"></a><span class="co">## -Preprocessing for input gene list and hit list ...</span></span>
<span id="cb291-11"><a href="associate-epcam-inflammatory-with-survival.html#cb291-11"></a><span class="co">## --Removing genes without values in geneList ...</span></span>
<span id="cb291-12"><a href="associate-epcam-inflammatory-with-survival.html#cb291-12"></a><span class="co">## --Removing duplicated genes ...</span></span>
<span id="cb291-13"><a href="associate-epcam-inflammatory-with-survival.html#cb291-13"></a><span class="co">## --Converting annotations ...</span></span>
<span id="cb291-14"><a href="associate-epcam-inflammatory-with-survival.html#cb291-14"></a><span class="co">## -- 2818 genes (out of 20055) could not be mapped to any identifier, and were removed from the data. </span></span>
<span id="cb291-15"><a href="associate-epcam-inflammatory-with-survival.html#cb291-15"></a><span class="co">## -- 2 genes (out of 29) could not be mapped to any identifier, and were removed from the data. </span></span>
<span id="cb291-16"><a href="associate-epcam-inflammatory-with-survival.html#cb291-16"></a><span class="co">## --Ordering Gene List decreasingly ...</span></span>
<span id="cb291-17"><a href="associate-epcam-inflammatory-with-survival.html#cb291-17"></a><span class="co">## -Preprocessing complete!</span></span>
<span id="cb291-18"><a href="associate-epcam-inflammatory-with-survival.html#cb291-18"></a>   gscaTCGA &lt;-<span class="st"> </span><span class="kw">analyze</span>( gscaTCGA, </span>
<span id="cb291-19"><a href="associate-epcam-inflammatory-with-survival.html#cb291-19"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb291-20"><a href="associate-epcam-inflammatory-with-survival.html#cb291-20"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb291-21"><a href="associate-epcam-inflammatory-with-survival.html#cb291-21"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb291-22"><a href="associate-epcam-inflammatory-with-survival.html#cb291-22"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb291-23"><a href="associate-epcam-inflammatory-with-survival.html#cb291-23"></a><span class="co">## --21 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named c2List!</span></span>
<span id="cb291-24"><a href="associate-epcam-inflammatory-with-survival.html#cb291-24"></a><span class="co">## --257 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named c5BP!</span></span>
<span id="cb291-25"><a href="associate-epcam-inflammatory-with-survival.html#cb291-25"></a><span class="co">## --73 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named c5MF!</span></span>
<span id="cb291-26"><a href="associate-epcam-inflammatory-with-survival.html#cb291-26"></a><span class="co">## --60 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named c5CC!</span></span>
<span id="cb291-27"><a href="associate-epcam-inflammatory-with-survival.html#cb291-27"></a><span class="co">## --301 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named MetPathway!</span></span>
<span id="cb291-28"><a href="associate-epcam-inflammatory-with-survival.html#cb291-28"></a><span class="co">## -Performing gene set enrichment analysis using HTSanalyzeR2... </span></span>
<span id="cb291-29"><a href="associate-epcam-inflammatory-with-survival.html#cb291-29"></a><span class="co">## --Calculating the permutations ... </span></span>
<span id="cb291-30"><a href="associate-epcam-inflammatory-with-survival.html#cb291-30"></a><span class="co">## -Gene set enrichment analysis using HTSanalyzeR2 complete </span></span>
<span id="cb291-31"><a href="associate-epcam-inflammatory-with-survival.html#cb291-31"></a><span class="co">## ==============================================</span></span>
<span id="cb291-32"><a href="associate-epcam-inflammatory-with-survival.html#cb291-32"></a>   </span>
<span id="cb291-33"><a href="associate-epcam-inflammatory-with-survival.html#cb291-33"></a><span class="kw">save</span>( gscaTCGA, <span class="dt">file=</span><span class="st">&quot;figure-outputs/TCGA_hairball.RData&quot;</span>)</span>
<span id="cb291-34"><a href="associate-epcam-inflammatory-with-survival.html#cb291-34"></a>   </span>
<span id="cb291-35"><a href="associate-epcam-inflammatory-with-survival.html#cb291-35"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(gscaTCGA<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb291-36"><a href="associate-epcam-inflammatory-with-survival.html#cb291-36"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(gscaTCGA<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb291-37"><a href="associate-epcam-inflammatory-with-survival.html#cb291-37"></a></span>
<span id="cb291-38"><a href="associate-epcam-inflammatory-with-survival.html#cb291-38"></a><span class="co">## check whether this runs:</span></span>
<span id="cb291-39"><a href="associate-epcam-inflammatory-with-survival.html#cb291-39"></a>gscaTCGA<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks<span class="op">$</span>Gene.Set.Term=TermsA</span>
<span id="cb291-40"><a href="associate-epcam-inflammatory-with-survival.html#cb291-40"></a><span class="kw">viewEnrichMap</span>(gscaTCGA, <span class="dt">gscs=</span><span class="kw">c</span>(<span class="st">&quot;ProcessNetworks&quot;</span>),</span>
<span id="cb291-41"><a href="associate-epcam-inflammatory-with-survival.html#cb291-41"></a>              <span class="dt">allSig =</span> <span class="ot">TRUE</span>, <span class="dt">gsNameType=</span><span class="st">&quot;term&quot;</span> )</span></code></pre></div>
<div class="figure"><span id="fig:TCGA-gsea"></span>
<div id="htmlwidget-afe823d819dde418631a" style="width:672px;height:480px;" class="forceGraph html-widget"></div>
<script type="application/json" data-for="htmlwidget-afe823d819dde418631a">{"x":{"nodes":{"id":["ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NBlood coagulation","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NApoptosis_Anti-Apoptosis mediated by external signals via PI3K_AKT"],"size":[5.21354166666667,8.046875,10.0833333333333,10.7916666666667,11.1458333333333,11.765625,11.765625,12.9166666666667,13.2708333333333,13.7135416666667,14.0677083333333,14.0677083333333,14.6875,15.0416666666667,15.1302083333333,15.3072916666667,15.3958333333333,15.5729166666667,16.28125,16.3697916666667,16.4583333333333,16.6354166666667,17.078125,17.2552083333333,17.34375,17.875,19.2916666666667,19.2916666666667,19.8229166666667,20,3],"color":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,null],"scheme":["neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","neg","pos"],"label":["Melatonin signaling","Androgen receptor signaling cross-talk","Blood coagulation","IgE signaling","IFN-gamma signaling","Interferon signaling","BCR pathway","Platelet aggregation","T helper cell differentiation","Chemotaxis","Wnt","Innate inflammatory response","NK cell cytotoxicity","Glycoconjugates","Histamine signaling","Ossification and bone remodeling","TCR signaling","Platelet-endothelium-leucocyte interactions","Ubiquitin-proteasomal proteolysis","G1-S Growth factor regulation","Jak-STAT Pathway","Potassium transport","Antigen presentation","Lymphocyte proliferation","Leucocyte chemotaxis","Cell-matrix interactions","Blood vessel morphogenesis","Phagosome in antigen presentation","Positive regulation cell proliferation","EMT",null],"label_id":["NNeurophysiological process_Melatonin signaling","NSignal transduction_Androgen receptor signaling cross-talk","NBlood coagulation","NInflammation_IgE signaling","NInflammation_IFN-gamma signaling","NInflammation_Interferon signaling","NImmune response_BCR pathway","NCell adhesion_Platelet aggregation","NImmune response_T helper cell differentiation","NChemotaxis","NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","NInflammation_Innate inflammatory response","NInflammation_NK cell cytotoxicity","NCell adhesion_Glycoconjugates","NInflammation_Histamine signaling","NDevelopment_Ossification and bone remodeling","NImmune response_TCR signaling","NCell adhesion_Platelet-endothelium-leucocyte interactions","NProteolysis_Ubiquitin-proteasomal proteolysis","NCell cycle_G1-S Growth factor regulation","NInflammation_Jak-STAT Pathway","NTransport_Potassium transport","NImmune response_Antigen presentation","NProliferation_Lymphocyte proliferation","NCell adhesion_Leucocyte chemotaxis","NCell adhesion_Cell-matrix interactions","NDevelopment_Blood vessel morphogenesis","NImmune response_Phagosome in antigen presentation","NProliferation_Positive regulation cell proliferation","NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","NApoptosis_Anti-Apoptosis mediated by external signals via PI3K_AKT"],"label_term":["Melatonin signaling","Androgen receptor signaling cross-talk","Blood coagulation","IgE signaling","IFN-gamma signaling","Interferon signaling","BCR pathway","Platelet aggregation","T helper cell differentiation","Chemotaxis","Wnt","Innate inflammatory response","NK cell cytotoxicity","Glycoconjugates","Histamine signaling","Ossification and bone remodeling","TCR signaling","Platelet-endothelium-leucocyte interactions","Ubiquitin-proteasomal proteolysis","G1-S Growth factor regulation","Jak-STAT Pathway","Potassium transport","Antigen presentation","Lymphocyte proliferation","Leucocyte chemotaxis","Cell-matrix interactions","Blood vessel morphogenesis","Phagosome in antigen presentation","Positive regulation cell proliferation","EMT",null]},"links":{"source":["ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation"],"target":["ProcessNetworks.NSignal transduction_Androgen receptor signaling cross-talk","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NBlood coagulation","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NInflammation_IgE signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NInflammation_Interferon signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCell adhesion_Platelet aggregation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NChemotaxis","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NInflammation_Innate inflammatory response","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NInflammation_NK cell cytotoxicity","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NDevelopment_Ossification and bone remodeling","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NImmune response_TCR signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCell cycle_G1-S Growth factor regulation","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NInflammation_Jak-STAT Pathway","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NImmune response_Antigen presentation","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NProliferation_Lymphocyte proliferation","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCell adhesion_Leucocyte chemotaxis","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NDevelopment_Blood vessel morphogenesis","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NImmune response_Phagosome in antigen presentation","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NProliferation_Positive regulation cell proliferation","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","ProcessNetworks.NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition"],"weight":[0.0379746835443038,0.036697247706422,0.0263157894736842,0.0163934426229508,0.0703125,0.0143884892086331,0.00689655172413793,0.0344827586206897,0.0273972602739726,0.0129032258064516,0.08,0.0122699386503067,0.0114942528735632,0.0170454545454545,0.00546448087431694,0.0163934426229508,0.0331491712707182,0.00966183574879227,0.00966183574879227,0.018957345971564,0.0187793427230047,0.00735294117647059,0.0902255639097744,0.0797101449275362,0.04,0.0909090909090909,0.0628930817610063,0.0745341614906832,0.0532544378698225,0.0520231213872832,0.0340909090909091,0.0384615384615385,0.0211640211640212,0.0777777777777778,0.0594594594594595,0.0706521739130435,0.0364583333333333,0.00975609756097561,0.155555555555556,0.077319587628866,0.023696682464455,0.0792079207920792,0.0186046511627907,0.00446428571428571,0.100456621004566,0.0343347639484979,0.122727272727273,0.136986301369863,0.0120481927710843,0.00584795321637427,0.00561797752808989,0.0971428571428571,0.0155440414507772,0.0255102040816327,0.00985221674876847,0.0148514851485149,0.00934579439252336,0.00462962962962963,0.00921658986175115,0.0138248847926267,0.2,0.0131578947368421,0.0220264317180617,0.00420168067226891,0.0168776371308017,0.004149377593361,0.0376569037656904,0.0232558139534884,0.00763358778625954,0.00746268656716418,0.0225563909774436,0.146496815286624,0.0446927374301676,0.307692307692308,0.149425287356322,0.221556886227545,0.0717948717948718,0.0492610837438424,0.196629213483146,0.164021164021164,0.037037037037037,0.300578034682081,0.0133928571428571,0.288135593220339,0.031390134529148,0.154589371980676,0.0958904109589041,0.0646551724137931,0.202898550724638,0.0869565217391304,0.0625,0.157446808510638,0.0859375,0.124497991967871,0.136904761904762,0.123529411764706,0.0625,0.124324324324324,0.0812182741116751,0.0382775119617225,0.085,0.131313131313131,0.0458715596330275,0.12807881773399,0.0358744394618834,0.115384615384615,0.0684931506849315,0.0125523012552301,0.135514018691589,0.0844444444444444,0.0635593220338983,0.165898617511521,0.109170305676856,0.0116731517509728,0.0534351145038168,0.0823529411764706,0.0722433460076046,0.0923076923076923,0.0531914893617021,0.00476190476190476,0.102564102564103,0.0576923076923077,0.00448430493273543,0.0418604651162791,0.132352941176471,0.0352422907488987,0.0776255707762557,0.00421940928270042,0.0669642857142857,0.0343347639484979,0.0593220338983051,0.146118721461187,0.075,0.111111111111111,0.044,0.0037593984962406,0.044280442804428,0.0179856115107914,0.0509090909090909,0.0505415162454874,0.0820512820512821,0.149732620320856,0.0628019323671498,0.0566037735849057,0.108910891089109,0.137931034482759,0.0633484162895928,0.16256157635468,0.00421940928270042,0.264550264550265,0.025531914893617,0.152073732718894,0.0590717299578059,0.0445344129554656,0.171171171171171,0.0920502092050209,0.0679245283018868,0.164609053497942,0.0905660377358491,0.127906976744186,0.0363636363636364,0.0401785714285714,0.0723981900452489,0.062780269058296,0.0655021834061135,0.0643776824034335,0.152777777777778,0.0203252032520325,0.0723404255319149,0.104347826086957,0.0691056910569106,0.03125,0.00757575757575758,0.0037037037037037,0.0748031496062992,0.132231404958678,0.037037037037037,0.049645390070922,0.112781954887218,0.0942028985507246,0.0704225352112676,0.0486725663716814,0.0432900432900433,0.181372549019608,0.164319248826291,0.0120481927710843,0.134529147982063,0.0240963855421687,0.267326732673267,0.032,0.00757575757575758,0.0897959183673469,0.175438596491228,0.07421875,0.276497695852535,0.0651340996168582,0.00353356890459364,0.048951048951049,0.0600706713780919,0.0588235294117647,0.096085409252669,0.00819672131147541,0.0884955752212389,0.0454545454545455,0.0619834710743802,0.0840336134453782,0.015625,0.0785123966942149,0.0647773279352227,0.0501930501930502,0.0963855421686747,0.0256410256410256,0.0804597701149425,0.257777777777778,0.0176056338028169,0.0664335664335664,0.0445205479452055,0.0687285223367698,0.0538720538720539,0.0288065843621399,0.0321285140562249,0.0275590551181102,0.0607287449392713,0.0731707317073171,0.0474308300395257,0.0308880308880309,0.00364963503649635,0.0494296577946768,0.0108695652173913,0.0324909747292419,0.0512820512820513,0.0280701754385965,0.0880281690140845,0.0334448160535117,0.0465116279069767,0.100694444444444,0.0619834710743802,0.0116279069767442,0.18552036199095,0.0193050193050193,0.104166666666667,0.0269230769230769,0.078125,0.0259259259259259,0.0518518518518519,0.0833333333333333,0.062962962962963,0.00687285223367698,0.0474576271186441,0.099644128113879,0.0260586319218241,0.0673400673400673,0.0634920634920635,0.125523012552301,0.0037037037037037,0.177489177489178,0.0339622641509434,0.109803921568627,0.126984126984127,0.182926829268293,0.195918367346939,0.0769230769230769,0.0604026845637584,0.0896551724137931,0.0697674418604651,0.0693069306930693,0.0705882352941176,0.0110294117647059,0.0697674418604651,0.0692307692307692,0.0323741007194245,0.0510948905109489,0.00340136054421769,0.0569395017793594,0.155038759689922,0.134328358208955,0.0457516339869281,0.0702341137123746,0.0516129032258065,0.0546623794212219,0.00363636363636364,0.209606986899563,0.0488721804511278,0.103448275862069,0.0664206642066421,0.0533807829181495,0.150579150579151,0.12406015037594,0.0735785953177258,0.122377622377622,0.0721311475409836,0.107744107744108,0.0108695652173913,0.0293040293040293,0.00347222222222222,0.0622710622710623,0.0210526315789474,0.00336700336700337,0.0101010101010101,0.00333333333333333,0.0165562913907285,0.0385852090032154,0.00623052959501558,0.0411392405063291,0.114478114478114,0.0444444444444444,0.0211267605633803,0.123552123552124,0.0774907749077491,0.115671641791045,0.218623481781377,0.139622641509434,0.0198675496688742,0.08,0.148936170212766,0.0855263157894737,0.14878892733564,0.0772058823529412,0.0888888888888889,0.0379310344827586,0.0484429065743945,0.0410958904109589,0.0839160839160839,0.065359477124183,0.0415335463258786,0.0573248407643312,0.117056856187291,0.00668896321070234,0.00332225913621262,0.136029411764706,0.00647249190938511,0.00645161290322581,0.00602409638554217,0.128378378378378,0.00591715976331361,0.00588235294117647,0.074468085106383,0.0508474576271186,0.168539325842697,0.0433333333333333,0.00314465408805031,0.139455782312925,0.0841423948220065,0.204946996466431,0.190972222222222,0.0332225913621262,0.247011952191235,0.0644067796610169,0.00628930817610063,0.0666666666666667,0.0212765957446809,0.128712871287129,0.075,0.00311526479750779,0.0666666666666667,0.0354838709677419,0.00306748466257669,0.03003003003003,0.279850746268657,0.0115942028985507,0.0384615384615385,0.106164383561644,0.0849056603773585,0.0815047021943574,0.117834394904459,0.113564668769716,0.0153846153846154,0.0484848484848485,0.123376623376623,0.0864197530864197,0.0535714285714286,0.0144092219020173,0.0202898550724638,0.0141643059490085,0.0344827586206897,0.0454545454545455,0.119760479041916,0.153374233128834,0.0476190476190476,0.0804597701149425,0.161094224924012]},"options":{"title":"Enrichment Map of GSEA on ProcessNetworks","legendTitle":"-Log10(Adjusted p-values)","type":"GSEA","nodeScheme":"dual","label":{"text":"term"},"colorScaler":"log10","nPermutations":100,"colorDomain":{"neg":[0,0.05],"pos":[null,null]}}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 15.9: pointless TCGA hairball
</p>
</div>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="associate-epcam-inflammatory-with-survival.html#cb292-1"></a></span>
<span id="cb292-2"><a href="associate-epcam-inflammatory-with-survival.html#cb292-2"></a><span class="kw">write.csv</span>(gscaTCGA<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks, <span class="dt">file=</span><span class="st">&quot;nature-tables/5k_TCGA_lumA_GSEA.csv&quot;</span>)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="epcam-inflammatory-vs-non-inflammatory-samples.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="deseq-analysis-characterisation-cohort-big-vs-small.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
