<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 19 BCR clonotype analysis | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 19 BCR clonotype analysis | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 19 BCR clonotype analysis | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-03-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="immune-estimation.html"/>
<link rel="next" href="whole-genome-sequencing-mutations.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.3.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.3.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>24.3</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bcr-clonotype-analysis" class="section level1">
<h1><span class="header-section-number">Chapter 19</span> BCR clonotype analysis</h1>
<div id="processing-summary" class="section level2">
<h2><span class="header-section-number">19.1</span> Processing summary</h2>
<p>Fastq files have previously been run through TRUST4 using the Rn6 BCR annotations using the following code (not run here). Note that the bcr/tcr .fa files were assembled using the ensembl annotation (these annotations are not present in the UCSC genomic files). Annotation files for the rn6 BCR regions are available at <a href="http://www.imgt.org//download/V-QUEST/IMGT_V-QUEST_reference_directory/Rattus_norvegicus/">IMGT</a>, but TCR regions are not available</p>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="bcr-clonotype-analysis.html#cb377-1"></a><span class="co"># obtain the gene names for rat:</span></span>
<span id="cb377-2"><a href="bcr-clonotype-analysis.html#cb377-2"></a><span class="kw">library</span>(biomaRt)</span>
<span id="cb377-3"><a href="bcr-clonotype-analysis.html#cb377-3"></a>human =<span class="st"> </span><span class="kw">useEnsembl</span>(<span class="st">&quot;ensembl&quot;</span>, <span class="dt">mirror=</span><span class="st">&quot;useast&quot;</span>, <span class="dt">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span>)</span>
<span id="cb377-4"><a href="bcr-clonotype-analysis.html#cb377-4"></a>rat =<span class="st"> </span><span class="kw">useEnsembl</span>(<span class="st">&quot;ensembl&quot;</span>, <span class="dt">mirror=</span><span class="st">&quot;useast&quot;</span>, <span class="dt">data=</span><span class="st">&quot;rnorvegicus_gene_ensembl&quot;</span>)</span>
<span id="cb377-5"><a href="bcr-clonotype-analysis.html#cb377-5"></a>TS =<span class="st"> </span>human_igg_trv_list</span>
<span id="cb377-6"><a href="bcr-clonotype-analysis.html#cb377-6"></a>Hum2RatProt =<span class="st"> </span><span class="kw">getLDS</span>(<span class="dt">attributes =</span> <span class="kw">c</span>(<span class="st">&quot;hgnc_symbol&quot;</span>, <span class="st">&quot;ensembl_transcript_id&quot;</span>), <span class="dt">filters =</span> <span class="st">&quot;hgnc_symbol&quot;</span>,  <span class="dt">values =</span> TS , <span class="dt">mart =</span> human, <span class="dt">attributesL =</span> <span class="kw">c</span>(<span class="st">&quot;rgd_symbol&quot;</span>, <span class="st">&quot;ensembl_transcript_id”, “ensemble_gene_id&quot;</span>), <span class="dt">martL =</span> rat, <span class="dt">uniqueRows=</span>T)</span>
<span id="cb377-7"><a href="bcr-clonotype-analysis.html#cb377-7"></a><span class="kw">write.table</span>(<span class="kw">unique</span>(Hum2RatProt<span class="op">$</span>ensemble_gene_id), <span class="st">&quot;ensbl_rat_bcr.txt&quot;</span>, <span class="dt">quote=</span>F, <span class="dt">row.names=</span>F,<span class="dt">colnames=</span>F)</span>
<span id="cb377-8"><a href="bcr-clonotype-analysis.html#cb377-8"></a></span>
<span id="cb377-9"><a href="bcr-clonotype-analysis.html#cb377-9"></a><span class="co"># obtain the tcr/bcr.fa file</span></span>
<span id="cb377-10"><a href="bcr-clonotype-analysis.html#cb377-10"></a><span class="co">#(Note that BuildDatabaseFa.pl needs to be edited on lines 95, 102, 114 and 121 to use transcript_id or gene_id instead)</span></span>
<span id="cb377-11"><a href="bcr-clonotype-analysis.html#cb377-11"></a></span>
<span id="cb377-12"><a href="bcr-clonotype-analysis.html#cb377-12"></a>perl BuildDatabaseFa.pl Rattus_norvegicus.Rnor_<span class="dv">6</span>.<span class="fl">0.</span>dna_sm.toplevel.fa Rattus_norvegicus.Rnor_<span class="dv">6</span>.<span class="dv">0</span>.<span class="fl">99.</span>gtf ensbl_rat_bcr.txt <span class="op">&gt;</span><span class="st"> </span>bcrtcr_rat_ens.fa</span>
<span id="cb377-13"><a href="bcr-clonotype-analysis.html#cb377-13"></a></span>
<span id="cb377-14"><a href="bcr-clonotype-analysis.html#cb377-14"></a><span class="co"># Obtain the reference annotation file</span></span>
<span id="cb377-15"><a href="bcr-clonotype-analysis.html#cb377-15"></a>perl BuildImgtAnnot.pl Rattus_norvegicus <span class="op">&gt;</span><span class="st"> </span>Rnor_IMGT<span class="op">+</span>C.fa</span>
<span id="cb377-16"><a href="bcr-clonotype-analysis.html#cb377-16"></a></span>
<span id="cb377-17"><a href="bcr-clonotype-analysis.html#cb377-17"></a><span class="co"># Run TRUST (below is single end)</span></span>
<span id="cb377-18"><a href="bcr-clonotype-analysis.html#cb377-18"></a>.<span class="op">/</span>run<span class="op">-</span>trust4 <span class="op">-</span>u <span class="dv">20170125</span>_NMU7_Tumor_RLU_CD45_CGDA5146_S1_R1_<span class="fl">001.</span>fastq.gz <span class="op">-</span>f bcrtcr_rat_ens.fa <span class="op">--</span>ref Rnor_IMGT<span class="op">+</span>C.fa</span></code></pre></div>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="bcr-clonotype-analysis.html#cb378-1"></a><span class="co">#Annotation file: Load in an annotation file indicating all the samples, batch effects etc </span></span>
<span id="cb378-2"><a href="bcr-clonotype-analysis.html#cb378-2"></a></span>
<span id="cb378-3"><a href="bcr-clonotype-analysis.html#cb378-3"></a></span>
<span id="cb378-4"><a href="bcr-clonotype-analysis.html#cb378-4"></a>TRUST4path=<span class="st">&quot;../data/TRUST4/&quot;</span></span>
<span id="cb378-5"><a href="bcr-clonotype-analysis.html#cb378-5"></a>TRUST4files=<span class="kw">dir</span>(TRUST4path,<span class="st">&quot;*.tsv&quot;</span>)</span>
<span id="cb378-6"><a href="bcr-clonotype-analysis.html#cb378-6"></a><span class="co">#FileNames=sapply(strsplit(TRUST4files, &quot;_report.tsv&quot;), function(x) x[1])</span></span>
<span id="cb378-7"><a href="bcr-clonotype-analysis.html#cb378-7"></a>matchidx=<span class="kw">match</span>(TRUST4files, infoTableFinal<span class="op">$</span>TRUSTName)</span>
<span id="cb378-8"><a href="bcr-clonotype-analysis.html#cb378-8"></a></span>
<span id="cb378-9"><a href="bcr-clonotype-analysis.html#cb378-9"></a><span class="co">#RepNames=paste(infoTableFinal$Rat_ID, tempAnnot$Location, tempAnnot$Fraction, sep=&quot;_&quot;)[matchidx]</span></span>
<span id="cb378-10"><a href="bcr-clonotype-analysis.html#cb378-10"></a><span class="co"># load in all the files and save the list to file</span></span>
<span id="cb378-11"><a href="bcr-clonotype-analysis.html#cb378-11"></a>RatTrust=<span class="kw">lapply</span>(TRUST4files, <span class="cf">function</span>(x) <span class="kw">read.delim</span>(<span class="kw">paste</span>(TRUST4path, x, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>))</span>
<span id="cb378-12"><a href="bcr-clonotype-analysis.html#cb378-12"></a><span class="kw">names</span>(RatTrust)=infoTableFinal<span class="op">$</span>SampleID[matchidx]</span>
<span id="cb378-13"><a href="bcr-clonotype-analysis.html#cb378-13"></a></span>
<span id="cb378-14"><a href="bcr-clonotype-analysis.html#cb378-14"></a></span>
<span id="cb378-15"><a href="bcr-clonotype-analysis.html#cb378-15"></a>TRUST4path=<span class="st">&quot;../data/TRUST4/matchedNormal/&quot;</span></span>
<span id="cb378-16"><a href="bcr-clonotype-analysis.html#cb378-16"></a>TRUST4files=<span class="kw">dir</span>(TRUST4path,<span class="st">&quot;*.tsv&quot;</span>)</span>
<span id="cb378-17"><a href="bcr-clonotype-analysis.html#cb378-17"></a><span class="co">#FileNames=sapply(strsplit(TRUST4files, &quot;_report.tsv&quot;), function(x) x[1])</span></span>
<span id="cb378-18"><a href="bcr-clonotype-analysis.html#cb378-18"></a>FileNames=<span class="kw">paste</span>(<span class="kw">substr</span>(TRUST4files, <span class="dv">10</span>, <span class="dv">12</span>), <span class="st">&quot;matchedNMUCD45&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb378-19"><a href="bcr-clonotype-analysis.html#cb378-19"></a>RatTrustNormal=<span class="kw">lapply</span>(TRUST4files, <span class="cf">function</span>(x) <span class="kw">read.delim</span>(<span class="kw">paste</span>(TRUST4path, x, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>))</span>
<span id="cb378-20"><a href="bcr-clonotype-analysis.html#cb378-20"></a><span class="kw">names</span>(RatTrustNormal)=FileNames</span>
<span id="cb378-21"><a href="bcr-clonotype-analysis.html#cb378-21"></a></span>
<span id="cb378-22"><a href="bcr-clonotype-analysis.html#cb378-22"></a>RatTrust=<span class="kw">c</span>(RatTrust, RatTrustNormal)</span>
<span id="cb378-23"><a href="bcr-clonotype-analysis.html#cb378-23"></a></span>
<span id="cb378-24"><a href="bcr-clonotype-analysis.html#cb378-24"></a><span class="kw">save</span>(RatTrust, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/Rat_TRUST4_%s.RData&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
</div>
<div id="summary-stats" class="section level2">
<h2><span class="header-section-number">19.2</span> Summary Stats</h2>
<p>Firstly, look at the number of BCR regions which have been identified by TRUST4. In green are the CD45 populations, which as expected appear to have a higher number of clonotypes compared to the epithelial and the double-negative populations (in read and orange).</p>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="bcr-clonotype-analysis.html#cb379-1"></a>NSamples=<span class="kw">sapply</span>(RatTrust, nrow)</span>
<span id="cb379-2"><a href="bcr-clonotype-analysis.html#cb379-2"></a>NSamplesCt2=<span class="kw">sapply</span>(RatTrust, <span class="cf">function</span>(x) <span class="kw">length</span>(<span class="kw">which</span>(x<span class="op">$</span>X.count<span class="op">&gt;=</span><span class="dv">2</span>)))</span>
<span id="cb379-3"><a href="bcr-clonotype-analysis.html#cb379-3"></a>NSamplesCDR3comp=<span class="kw">sapply</span>(RatTrust, <span class="cf">function</span>(x) <span class="kw">length</span>(<span class="kw">which</span>(x<span class="op">$</span>CDR3aa<span class="op">!=</span><span class="st">&quot;partial&quot;</span>)))</span>
<span id="cb379-4"><a href="bcr-clonotype-analysis.html#cb379-4"></a>NSamplesCDR3compCt2=<span class="kw">sapply</span>(RatTrust, <span class="cf">function</span>(x) <span class="kw">length</span>(<span class="kw">which</span>(x<span class="op">$</span>CDR3aa<span class="op">!=</span><span class="st">&quot;partial&quot;</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>X.count<span class="op">&gt;=</span><span class="dv">2</span>)))</span>
<span id="cb379-5"><a href="bcr-clonotype-analysis.html#cb379-5"></a></span>
<span id="cb379-6"><a href="bcr-clonotype-analysis.html#cb379-6"></a>TableOut=<span class="kw">cbind</span>(NSamples, NSamplesCt2, NSamplesCDR3comp, NSamplesCDR3compCt2)</span>
<span id="cb379-7"><a href="bcr-clonotype-analysis.html#cb379-7"></a></span>
<span id="cb379-8"><a href="bcr-clonotype-analysis.html#cb379-8"></a>matchidx=<span class="kw">match</span>(<span class="kw">rownames</span>(TableOut), infoTableFinal<span class="op">$</span>SampleID)</span>
<span id="cb379-9"><a href="bcr-clonotype-analysis.html#cb379-9"></a>ColOut=<span class="kw">factor</span>(<span class="kw">paste</span>(infoTableFinal<span class="op">$</span>Fraction[matchidx], infoTableFinal<span class="op">$</span>Batch[matchidx]))</span>
<span id="cb379-10"><a href="bcr-clonotype-analysis.html#cb379-10"></a><span class="co">#palette(c(&quot;#e5f5e0&quot;,&quot;#a1d99b&quot;, &quot;#31a354&quot;,  &quot;#fd8d3c&quot;,&quot;#fa9fb5&quot;, &quot;#f03b20&quot;, &quot;#005824&quot;))</span></span>
<span id="cb379-11"><a href="bcr-clonotype-analysis.html#cb379-11"></a></span>
<span id="cb379-12"><a href="bcr-clonotype-analysis.html#cb379-12"></a><span class="co">#pdf(sprintf(&quot;rslt/TRUST4/BCR_summary_QC_%s.pdf&quot;, Sys.Date()), height=5, width = 7)</span></span>
<span id="cb379-13"><a href="bcr-clonotype-analysis.html#cb379-13"></a></span>
<span id="cb379-14"><a href="bcr-clonotype-analysis.html#cb379-14"></a><span class="co"># plot the clonotype diversity, color-code according to both batch and the sample used</span></span>
<span id="cb379-15"><a href="bcr-clonotype-analysis.html#cb379-15"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb379-16"><a href="bcr-clonotype-analysis.html#cb379-16"></a><span class="kw">barplot</span>(NSamplesCt2<span class="op">+</span><span class="dv">1</span>, <span class="dt">las=</span><span class="dv">2</span>,<span class="dt">log =</span><span class="st">&#39;y&#39;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of BCR clonotypes&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Overview of clonotype diversity (2+ reads)&quot;</span>, <span class="dt">col=</span>ColOut, <span class="dt">cex.names =</span> <span class="fl">0.75</span>)</span>
<span id="cb379-17"><a href="bcr-clonotype-analysis.html#cb379-17"></a><span class="kw">legend</span>(<span class="st">&quot;top&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.03</span>),  <span class="dt">legend =</span> <span class="kw">levels</span>(ColOut),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>), <span class="dt">horiz =</span> T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-125-1.png" width="672" /></p>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="bcr-clonotype-analysis.html#cb380-1"></a></span>
<span id="cb380-2"><a href="bcr-clonotype-analysis.html#cb380-2"></a><span class="co"># plot the number of full vs partial clonotypes</span></span>
<span id="cb380-3"><a href="bcr-clonotype-analysis.html#cb380-3"></a>PartialvsFull=<span class="kw">rbind</span>(NSamplesCDR3compCt2,NSamplesCt2<span class="op">-</span>NSamplesCDR3compCt2)</span>
<span id="cb380-4"><a href="bcr-clonotype-analysis.html#cb380-4"></a><span class="kw">rownames</span>(PartialvsFull)=<span class="kw">c</span>(<span class="st">&quot;full&quot;</span>, <span class="st">&quot;partial&quot;</span>)</span>
<span id="cb380-5"><a href="bcr-clonotype-analysis.html#cb380-5"></a></span>
<span id="cb380-6"><a href="bcr-clonotype-analysis.html#cb380-6"></a></span>
<span id="cb380-7"><a href="bcr-clonotype-analysis.html#cb380-7"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb380-8"><a href="bcr-clonotype-analysis.html#cb380-8"></a><span class="kw">barplot</span>(PartialvsFull<span class="op">+</span><span class="dv">1</span>, <span class="dt">las=</span><span class="dv">2</span>,<span class="dt">log =</span><span class="st">&#39;y&#39;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of BCR clonotypes&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Proportion of full vs partial CDR3a&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>), <span class="dt">cex.names =</span> <span class="fl">0.75</span>)</span>
<span id="cb380-9"><a href="bcr-clonotype-analysis.html#cb380-9"></a><span class="kw">legend</span>(<span class="st">&quot;bottom&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">-0.5</span>),  <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;full&quot;</span>, <span class="st">&quot;partial&quot;</span>),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>), <span class="dt">horiz =</span> T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-125-2.png" width="672" /></p>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="bcr-clonotype-analysis.html#cb381-1"></a></span>
<span id="cb381-2"><a href="bcr-clonotype-analysis.html#cb381-2"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb381-3"><a href="bcr-clonotype-analysis.html#cb381-3"></a><span class="kw">barplot</span>(<span class="kw">t</span>(TableOut[ <span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction[matchidx]<span class="op">==</span><span class="st">&quot;CD45&quot;</span>),]<span class="op">+</span><span class="dv">1</span>), <span class="dt">las=</span><span class="dv">2</span>,<span class="dt">log =</span><span class="st">&#39;y&#39;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of BCR clonotypes&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Clonotype diversity CD45+ samples&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">cex.names =</span> <span class="fl">0.75</span>, <span class="dt">beside =</span> T)</span>
<span id="cb381-4"><a href="bcr-clonotype-analysis.html#cb381-4"></a><span class="kw">legend</span>(<span class="st">&quot;bottom&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">-0.5</span>),  <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;# clonotypes&quot;</span>, <span class="st">&quot;clonotypes &gt;2reads&quot;</span>, <span class="st">&quot;complete CDR3&quot;</span>, <span class="st">&quot;both&quot;</span>),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">horiz =</span> T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-125-3.png" width="672" /></p>
<div class="sourceCode" id="cb382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb382-1"><a href="bcr-clonotype-analysis.html#cb382-1"></a></span>
<span id="cb382-2"><a href="bcr-clonotype-analysis.html#cb382-2"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb382-3"><a href="bcr-clonotype-analysis.html#cb382-3"></a><span class="kw">barplot</span>(NSamplesCDR3compCt2<span class="op">+</span><span class="dv">1</span>, <span class="dt">las=</span><span class="dv">2</span>,<span class="dt">log =</span><span class="st">&#39;y&#39;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of BCR clonotypes&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Filtered Clonotypes based on batch&quot;</span>, <span class="dt">col=</span>ColOut, <span class="dt">cex.names =</span> <span class="fl">0.75</span>, <span class="dt">beside =</span> T)</span>
<span id="cb382-4"><a href="bcr-clonotype-analysis.html#cb382-4"></a><span class="kw">legend</span>(<span class="st">&quot;top&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.03</span>),  <span class="dt">legend =</span> <span class="kw">levels</span>(ColOut),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>), <span class="dt">horiz =</span> T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-125-4.png" width="672" /></p>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="bcr-clonotype-analysis.html#cb383-1"></a></span>
<span id="cb383-2"><a href="bcr-clonotype-analysis.html#cb383-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p>We will refine the above plot to contain only the CD45 population and assess:</p>
<ul>
<li>number of clonotypes</li>
<li>number of clonotypes with at least 2 reads</li>
<li>number of clonotypes which have a complete CDR3a region</li>
</ul>
<p>Note: although this does not look like a big drop, the data is plotted on a log-scale. For example, in the control 4 (first column) a quarter of the clonotypes have at least 2 counts and about half have a complete CDR3 sequence.</p>
<p>For the following analyses, the clonotypes are filtered to <span style="color:red"> only clonotypes with at least 2 supporting reads and a complete VDJ read </span>. When applying this restriction, we have the following distribution of BCR regions. Almost all the epithelial cases do not have supporting reads, and few of the DN cases have supporting reads too.</p>
<div class="sourceCode" id="cb384"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb384-1"><a href="bcr-clonotype-analysis.html#cb384-1"></a>RatTrustB=<span class="kw">lapply</span>(RatTrust, <span class="cf">function</span>(x) x[<span class="kw">which</span>(x<span class="op">$</span>CDR3aa<span class="op">!=</span><span class="st">&quot;partial&quot;</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>X.count<span class="op">&gt;=</span><span class="dv">2</span>), ])</span>
<span id="cb384-2"><a href="bcr-clonotype-analysis.html#cb384-2"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(RatTrustB)){</span>
<span id="cb384-3"><a href="bcr-clonotype-analysis.html#cb384-3"></a>  RatTrustB[[i]]<span class="op">$</span>frequency=RatTrustB[[i]]<span class="op">$</span>X.count<span class="op">/</span><span class="kw">sum</span>(RatTrustB[[i]]<span class="op">$</span>X.count)</span>
<span id="cb384-4"><a href="bcr-clonotype-analysis.html#cb384-4"></a>}</span></code></pre></div>
</div>
<div id="diversity-metrics" class="section level2">
<h2><span class="header-section-number">19.3</span> Diversity metrics</h2>
<p>We will assess BCR diversity using the following metrics:</p>
<p>** Shannon index **</p>
<p>The shannon index is computed by:</p>
<ul>
<li>filtering through reads of interest</li>
<li>recalculate the fraction such that the new list sums to 1</li>
<li>compute entropy as the sum of log(freq_x)*freq_x amongst all populations x</li>
<li>compute the maximum entropy expected for that case -log(1/N), where N is the number of populations present</li>
<li>to determine confidence intervals, bootstrap the population (500 times) and compute the expected entropy</li>
</ul>
<p>** Gini index **</p>
<p>The Gini index can be considered as an inverse of the Shannon index</p>
<p>** Top Clonotypes **</p>
<p>We will see the proportion of the BCR repertoire which is computed using the top 10 frequent clones.
This will give an idea of whether there is a clonal expansion</p>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="bcr-clonotype-analysis.html#cb385-1"></a><span class="co"># compute values here</span></span>
<span id="cb385-2"><a href="bcr-clonotype-analysis.html#cb385-2"></a>Div1=<span class="kw">sapply</span>(RatTrustB, <span class="cf">function</span>(x) <span class="op">-</span><span class="kw">sum</span>(x<span class="op">$</span>frequency<span class="op">*</span><span class="kw">log</span>(x<span class="op">$</span>frequency), <span class="dt">na.rm=</span>T))</span>
<span id="cb385-3"><a href="bcr-clonotype-analysis.html#cb385-3"></a>EDiv=<span class="kw">sapply</span>(NSamplesCDR3compCt2, <span class="cf">function</span>(x) <span class="op">-</span><span class="kw">log</span>(<span class="dv">1</span><span class="op">/</span>x))</span>
<span id="cb385-4"><a href="bcr-clonotype-analysis.html#cb385-4"></a>NormDiv=Div1<span class="op">/</span>EDiv</span>
<span id="cb385-5"><a href="bcr-clonotype-analysis.html#cb385-5"></a></span>
<span id="cb385-6"><a href="bcr-clonotype-analysis.html#cb385-6"></a><span class="co">## Shannon index</span></span>
<span id="cb385-7"><a href="bcr-clonotype-analysis.html#cb385-7"></a><span class="co"># bootstrap rslt</span></span>
<span id="cb385-8"><a href="bcr-clonotype-analysis.html#cb385-8"></a>BSrslt=<span class="kw">sapply</span>(RatTrustB, <span class="cf">function</span>(x) <span class="kw">tryCatch</span>(<span class="kw">BootstrapShannonIdx</span>(x[ ,<span class="dv">1</span>], <span class="dv">1000</span>), <span class="dt">error=</span><span class="cf">function</span>(e) <span class="kw">c</span>(<span class="ot">NA</span>, <span class="ot">NA</span>)))</span>
<span id="cb385-9"><a href="bcr-clonotype-analysis.html#cb385-9"></a><span class="co"># divide the CI by the maximum possible diversity</span></span>
<span id="cb385-10"><a href="bcr-clonotype-analysis.html#cb385-10"></a></span>
<span id="cb385-11"><a href="bcr-clonotype-analysis.html#cb385-11"></a>BSCI=<span class="kw">t</span>(BSrslt)<span class="op">/</span>EDiv</span>
<span id="cb385-12"><a href="bcr-clonotype-analysis.html#cb385-12"></a></span>
<span id="cb385-13"><a href="bcr-clonotype-analysis.html#cb385-13"></a>df=<span class="kw">data.frame</span>(<span class="dt">sample=</span><span class="kw">rownames</span>(BSCI), <span class="dt">Val=</span>NormDiv, <span class="dt">Lower=</span>BSCI[ , <span class="dv">1</span>], <span class="dt">Upper=</span>BSCI[ ,<span class="dv">2</span> ], <span class="dt">Type=</span>infoTableFinal<span class="op">$</span>Fraction[matchidx], <span class="dt">Batch=</span>infoTableFinal<span class="op">$</span>Batch[matchidx])</span>
<span id="cb385-14"><a href="bcr-clonotype-analysis.html#cb385-14"></a></span>
<span id="cb385-15"><a href="bcr-clonotype-analysis.html#cb385-15"></a><span class="co">## Gini index</span></span>
<span id="cb385-16"><a href="bcr-clonotype-analysis.html#cb385-16"></a></span>
<span id="cb385-17"><a href="bcr-clonotype-analysis.html#cb385-17"></a>Gini=<span class="kw">sapply</span>(RatTrustB, <span class="cf">function</span>(x) <span class="kw">gini</span>(x<span class="op">$</span>frequency))</span>
<span id="cb385-18"><a href="bcr-clonotype-analysis.html#cb385-18"></a></span>
<span id="cb385-19"><a href="bcr-clonotype-analysis.html#cb385-19"></a>PGini=<span class="kw">sapply</span>(RatTrustB, <span class="cf">function</span>(x) <span class="kw">tryCatch</span>(<span class="kw">PermuteGini</span>(x<span class="op">$</span>X.count, <span class="dv">1000</span>), <span class="dt">error=</span><span class="cf">function</span>(e) <span class="kw">c</span>(<span class="ot">NA</span>, <span class="ot">NA</span>)))</span>
<span id="cb385-20"><a href="bcr-clonotype-analysis.html#cb385-20"></a></span>
<span id="cb385-21"><a href="bcr-clonotype-analysis.html#cb385-21"></a>df<span class="op">$</span>Calc=<span class="st">&quot;shannon&quot;</span></span>
<span id="cb385-22"><a href="bcr-clonotype-analysis.html#cb385-22"></a></span>
<span id="cb385-23"><a href="bcr-clonotype-analysis.html#cb385-23"></a>df2=<span class="kw">data.frame</span>(<span class="dt">sample=</span>df<span class="op">$</span>sample, <span class="dt">Val=</span>Gini, <span class="dt">Lower=</span>PGini[<span class="dv">1</span>, ], <span class="dt">Upper=</span>PGini[<span class="dv">2</span>, ], <span class="dt">Type=</span>df<span class="op">$</span>Type, <span class="dt">Batch=</span>df<span class="op">$</span>Batch,</span>
<span id="cb385-24"><a href="bcr-clonotype-analysis.html#cb385-24"></a>               <span class="dt">Calc=</span><span class="st">&quot;Gini&quot;</span>)</span>
<span id="cb385-25"><a href="bcr-clonotype-analysis.html#cb385-25"></a></span>
<span id="cb385-26"><a href="bcr-clonotype-analysis.html#cb385-26"></a>dfAll=<span class="kw">rbind</span>(df, df2)</span>
<span id="cb385-27"><a href="bcr-clonotype-analysis.html#cb385-27"></a></span>
<span id="cb385-28"><a href="bcr-clonotype-analysis.html#cb385-28"></a><span class="co">## Top Clonotypes</span></span>
<span id="cb385-29"><a href="bcr-clonotype-analysis.html#cb385-29"></a></span>
<span id="cb385-30"><a href="bcr-clonotype-analysis.html#cb385-30"></a>TopClones=<span class="kw">sapply</span>(RatTrustB, <span class="cf">function</span>(x) <span class="kw">sum</span>(x<span class="op">$</span>frequency[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>], <span class="dt">na.rm =</span> T))</span>
<span id="cb385-31"><a href="bcr-clonotype-analysis.html#cb385-31"></a>TopClones[<span class="kw">which</span>(NSamplesCDR3compCt2<span class="op">==</span><span class="dv">0</span>)]=<span class="dv">0</span></span>
<span id="cb385-32"><a href="bcr-clonotype-analysis.html#cb385-32"></a>df2<span class="op">$</span>Val=TopClones</span>
<span id="cb385-33"><a href="bcr-clonotype-analysis.html#cb385-33"></a>df2<span class="op">$</span>Calc=<span class="st">&quot;Top10&quot;</span></span>
<span id="cb385-34"><a href="bcr-clonotype-analysis.html#cb385-34"></a>df2<span class="op">$</span>Lower=TopClones</span>
<span id="cb385-35"><a href="bcr-clonotype-analysis.html#cb385-35"></a>df2<span class="op">$</span>Upper=TopClones</span>
<span id="cb385-36"><a href="bcr-clonotype-analysis.html#cb385-36"></a></span>
<span id="cb385-37"><a href="bcr-clonotype-analysis.html#cb385-37"></a>dfAll=<span class="kw">rbind</span>(dfAll, df2)</span>
<span id="cb385-38"><a href="bcr-clonotype-analysis.html#cb385-38"></a></span>
<span id="cb385-39"><a href="bcr-clonotype-analysis.html#cb385-39"></a>dfAll<span class="op">$</span>Batch=<span class="kw">as.character</span>(dfAll<span class="op">$</span>Batch)</span>
<span id="cb385-40"><a href="bcr-clonotype-analysis.html#cb385-40"></a>dfAll<span class="op">$</span>Batch[<span class="kw">grep</span>(<span class="st">&quot;matchedNMU&quot;</span>, <span class="kw">rownames</span>(dfAll))]=<span class="st">&quot;match&quot;</span></span>
<span id="cb385-41"><a href="bcr-clonotype-analysis.html#cb385-41"></a>dfAll<span class="op">$</span>Type[<span class="kw">grep</span>(<span class="st">&quot;matchedNMU&quot;</span>, <span class="kw">rownames</span>(dfAll))]=<span class="st">&quot;CD45&quot;</span></span>
<span id="cb385-42"><a href="bcr-clonotype-analysis.html#cb385-42"></a><span class="co">#ggplot(df[df$Type==&quot;CD45&quot;, ], aes(x=sample, y=Val, fill=Type))+geom_bar(stat=&quot;identity&quot;)+facet_grid(~Batch, scales=&quot;free_x&quot;, space=&quot;free&quot;)+geom_errorbar( aes(ymin=Lower, ymax=Upper), width=.2)+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ggtitle(&quot;Shannon idx of BCR diversity&quot;)</span></span></code></pre></div>
<p>The following plot demonstrates the relationship between the above metrics</p>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb386-1"><a href="bcr-clonotype-analysis.html#cb386-1"></a><span class="co">#pdf(sprintf(&quot;rslt/TRUST4/summary_Scores_diversity_%s.pdf&quot;, Sys.Date()), width=7, height=5)</span></span>
<span id="cb386-2"><a href="bcr-clonotype-analysis.html#cb386-2"></a></span>
<span id="cb386-3"><a href="bcr-clonotype-analysis.html#cb386-3"></a><span class="kw">ggplot</span>(dfAll[dfAll<span class="op">$</span>Type<span class="op">==</span><span class="st">&quot;CD45&quot;</span>, ], <span class="kw">aes</span>(<span class="dt">x=</span>sample, <span class="dt">y=</span>Val, <span class="dt">fill=</span>Type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(Calc<span class="op">~</span>Batch, <span class="dt">scales=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">geom_errorbar</span>( <span class="kw">aes</span>(<span class="dt">ymin=</span>Lower, <span class="dt">ymax=</span>Upper), <span class="dt">width=</span>.<span class="dv">2</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;BCR diversity across samples&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;index value&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-128-1.png" width="672" /></p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="bcr-clonotype-analysis.html#cb387-1"></a></span>
<span id="cb387-2"><a href="bcr-clonotype-analysis.html#cb387-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p>Notice that some samples have very few clones. An example is 5RB which is represented by a single clonotype, accounting for the absence of a Gini or shannon index</p>
</div>
<div id="compare-the-characterisation-cohort" class="section level2">
<h2><span class="header-section-number">19.4</span> Compare the characterisation cohort</h2>
<p>In the characterisation cohort, we have 3 cases which have CD45 samples in both the tumor and a matched NMU treated mammary gland.</p>
<p>Is there a difference in the clonotypes between these samples?</p>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="bcr-clonotype-analysis.html#cb388-1"></a>CharTemp=dfAll[<span class="kw">which</span>(dfAll<span class="op">$</span>Type<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>dfAll<span class="op">$</span>Batch<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;match&quot;</span>)), ]</span>
<span id="cb388-2"><a href="bcr-clonotype-analysis.html#cb388-2"></a>CharTemp<span class="op">$</span>Batch[<span class="kw">grep</span>(<span class="st">&quot;Control&quot;</span>, CharTemp<span class="op">$</span>sample)]=<span class="st">&quot;Normal&quot;</span></span>
<span id="cb388-3"><a href="bcr-clonotype-analysis.html#cb388-3"></a>Nclonotypes=TableOut[<span class="kw">which</span>(<span class="kw">rownames</span>(TableOut)<span class="op">%in%</span>CharTemp<span class="op">$</span>sample), ]</span>
<span id="cb388-4"><a href="bcr-clonotype-analysis.html#cb388-4"></a>Nclonotypes2=<span class="kw">melt</span>(Nclonotypes)</span>
<span id="cb388-5"><a href="bcr-clonotype-analysis.html#cb388-5"></a>Nclonotypes2<span class="op">$</span>Case=<span class="st">&quot;tumor&quot;</span></span>
<span id="cb388-6"><a href="bcr-clonotype-analysis.html#cb388-6"></a>Nclonotypes2<span class="op">$</span>Case[<span class="kw">grep</span>(<span class="st">&quot;match&quot;</span>, Nclonotypes2<span class="op">$</span>Var1)]=<span class="st">&quot;mammary&quot;</span></span>
<span id="cb388-7"><a href="bcr-clonotype-analysis.html#cb388-7"></a>Nclonotypes2<span class="op">$</span>Case[<span class="kw">grep</span>(<span class="st">&quot;Control&quot;</span>, Nclonotypes2<span class="op">$</span>Var1)]=<span class="st">&quot;control&quot;</span></span>
<span id="cb388-8"><a href="bcr-clonotype-analysis.html#cb388-8"></a></span>
<span id="cb388-9"><a href="bcr-clonotype-analysis.html#cb388-9"></a><span class="co">#pdf(sprintf(&quot;rslt/TRUST4/compare_mammary_vs_tumor_%s.pdf&quot;, Sys.Date()), width=8, height=6)</span></span>
<span id="cb388-10"><a href="bcr-clonotype-analysis.html#cb388-10"></a>a1=<span class="kw">sapply</span>(<span class="kw">unique</span>(CharTemp<span class="op">$</span>Calc), <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(CharTemp<span class="op">$</span>Val[CharTemp<span class="op">$</span>Calc<span class="op">==</span>x <span class="op">&amp;</span><span class="st"> </span>CharTemp<span class="op">$</span>Batch<span class="op">==</span><span class="dv">1</span>], CharTemp<span class="op">$</span>Val[CharTemp<span class="op">$</span>Calc<span class="op">==</span>x <span class="op">&amp;</span><span class="st"> </span>CharTemp<span class="op">$</span>Batch<span class="op">==</span><span class="st">&quot;match&quot;</span>])<span class="op">$</span>p.value)</span>
<span id="cb388-11"><a href="bcr-clonotype-analysis.html#cb388-11"></a>p=<span class="kw">ggplot</span>(CharTemp, <span class="kw">aes</span>(<span class="dt">x=</span>Batch, <span class="dt">y=</span>Val))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>Calc)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;wilcox pval: Gini %s, Shannon %s, Top10 = %s&quot;</span>, <span class="kw">round</span>(a1[<span class="dv">2</span>],<span class="dv">2</span>), <span class="kw">round</span>(a1[<span class="dv">1</span>],<span class="dv">2</span>), <span class="kw">round</span>(a1[<span class="dv">3</span>],<span class="dv">2</span>)))<span class="op">+</span><span class="kw">geom_jitter</span>(<span class="dt">col=</span><span class="st">&quot;grey&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()</span>
<span id="cb388-12"><a href="bcr-clonotype-analysis.html#cb388-12"></a><span class="kw">print</span>(p)</span></code></pre></div>
<div class="figure"><span id="fig:2h"></span>
<img src="RatDCIS-bookdown_files/figure-html/2h-1.png" alt="bcr clonotypes in tumor and normal mammary glands" width="672" />
<p class="caption">
Figure 19.1: bcr clonotypes in tumor and normal mammary glands
</p>
</div>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="bcr-clonotype-analysis.html#cb389-1"></a></span>
<span id="cb389-2"><a href="bcr-clonotype-analysis.html#cb389-2"></a><span class="kw">write.csv</span>(CharTemp, <span class="dt">file=</span><span class="st">&quot;nature-tables/2h.csv&quot;</span>)</span></code></pre></div>
<p>Next, we check whether this could be attributed to the total number of clones. Here is a plot which assesses whether these are similar or different</p>
<div class="sourceCode" id="cb390"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb390-1"><a href="bcr-clonotype-analysis.html#cb390-1"></a></span>
<span id="cb390-2"><a href="bcr-clonotype-analysis.html#cb390-2"></a>a1=<span class="kw">sapply</span>(<span class="kw">unique</span>(Nclonotypes2<span class="op">$</span>Var2), <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(Nclonotypes2<span class="op">$</span>value[Nclonotypes2<span class="op">$</span>Var2<span class="op">==</span>x <span class="op">&amp;</span><span class="st"> </span>Nclonotypes2<span class="op">$</span>Case<span class="op">==</span><span class="st">&quot;tumor&quot;</span>], </span>
<span id="cb390-3"><a href="bcr-clonotype-analysis.html#cb390-3"></a>                                                             Nclonotypes2<span class="op">$</span>value[Nclonotypes2<span class="op">$</span>Var2<span class="op">==</span>x <span class="op">&amp;</span><span class="st"> </span>Nclonotypes2<span class="op">$</span>Case<span class="op">==</span><span class="st">&quot;mammary&quot;</span>])<span class="op">$</span>p.value)</span>
<span id="cb390-4"><a href="bcr-clonotype-analysis.html#cb390-4"></a></span>
<span id="cb390-5"><a href="bcr-clonotype-analysis.html#cb390-5"></a>ann_text2 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">lab=</span><span class="kw">paste</span>(<span class="st">&quot;p=&quot;</span>, <span class="kw">round</span>(a1,<span class="dv">4</span>),<span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">Var2=</span><span class="kw">unique</span>(Nclonotypes2<span class="op">$</span>Var2), <span class="dt">Case=</span><span class="dv">1</span>, <span class="dt">value=</span><span class="dv">2</span>)</span>
<span id="cb390-6"><a href="bcr-clonotype-analysis.html#cb390-6"></a>p=<span class="kw">ggplot</span>(Nclonotypes2, <span class="kw">aes</span>(<span class="dt">x=</span>Case, <span class="dt">y=</span>value))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>Var2)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Nclonotype with different filters&quot;</span>)<span class="op">+</span><span class="kw">geom_jitter</span>(<span class="dt">col=</span><span class="st">&quot;grey&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span></span>
<span id="cb390-7"><a href="bcr-clonotype-analysis.html#cb390-7"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">data=</span>ann_text2, <span class="kw">aes</span>(<span class="dt">label=</span>lab))</span>
<span id="cb390-8"><a href="bcr-clonotype-analysis.html#cb390-8"></a><span class="kw">print</span>(p)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/2h-nClones-1.png" width="672" /></p>
<p>What about association with size?</p>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="bcr-clonotype-analysis.html#cb391-1"></a>CharTemp2=CharTemp[<span class="kw">which</span>(CharTemp<span class="op">$</span>Batch<span class="op">==</span><span class="st">&quot;1&quot;</span>), ]</span>
<span id="cb391-2"><a href="bcr-clonotype-analysis.html#cb391-2"></a>midx=Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">match</span>(<span class="kw">substr</span>(<span class="kw">as.character</span>(CharTemp2<span class="op">$</span>sample), <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">as.character</span>(CharTemp2<span class="op">$</span>sample))<span class="op">-</span><span class="dv">5</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb391-3"><a href="bcr-clonotype-analysis.html#cb391-3"></a>CharTemp2<span class="op">$</span>size=midx</span>
<span id="cb391-4"><a href="bcr-clonotype-analysis.html#cb391-4"></a></span>
<span id="cb391-5"><a href="bcr-clonotype-analysis.html#cb391-5"></a></span>
<span id="cb391-6"><a href="bcr-clonotype-analysis.html#cb391-6"></a><span class="kw">ggplot</span>(CharTemp2, <span class="kw">aes</span>(<span class="dt">x=</span>size, <span class="dt">y=</span>Val))<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>Calc)<span class="op">+</span><span class="kw">theme_bw</span>()</span></code></pre></div>
<div class="figure"><span id="fig:Ext2e"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext2e-1.png" alt="clonotype assoc with size" width="672" />
<p class="caption">
Figure 19.2: clonotype assoc with size
</p>
</div>
<div class="sourceCode" id="cb392"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb392-1"><a href="bcr-clonotype-analysis.html#cb392-1"></a></span>
<span id="cb392-2"><a href="bcr-clonotype-analysis.html#cb392-2"></a><span class="kw">write.csv</span>(CharTemp2, <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext2e.csv&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="bcr-clonotype-analysis.html#cb393-1"></a>ProgTemp=dfAll[<span class="kw">which</span>(dfAll<span class="op">$</span>Type<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>dfAll<span class="op">$</span>sample<span class="op">%in%</span>infoTableFinal<span class="op">$</span>SampleID[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>)]), ]</span>
<span id="cb393-2"><a href="bcr-clonotype-analysis.html#cb393-2"></a>a1=<span class="kw">match</span>(ProgTemp<span class="op">$</span>sample, infoTable)</span>
<span id="cb393-3"><a href="bcr-clonotype-analysis.html#cb393-3"></a>ProgTemp<span class="op">$</span>Growth=infoTableFinal<span class="op">$</span>Growth[a1]</span>
<span id="cb393-4"><a href="bcr-clonotype-analysis.html#cb393-4"></a>ProgTemp<span class="op">$</span>NewID=infoTableFinal<span class="op">$</span>TumorIDnew[a1]</span>
<span id="cb393-5"><a href="bcr-clonotype-analysis.html#cb393-5"></a>ProgTemp<span class="op">$</span>Treatment=infoTableFinal<span class="op">$</span>Treatment[a1]</span>
<span id="cb393-6"><a href="bcr-clonotype-analysis.html#cb393-6"></a></span>
<span id="cb393-7"><a href="bcr-clonotype-analysis.html#cb393-7"></a>Nclonotypes=TableOut[<span class="kw">which</span>(<span class="kw">rownames</span>(TableOut)<span class="op">%in%</span>ProgTemp<span class="op">$</span>sample), ]</span>
<span id="cb393-8"><a href="bcr-clonotype-analysis.html#cb393-8"></a>Nclonotypes2=<span class="kw">melt</span>(Nclonotypes)</span>
<span id="cb393-9"><a href="bcr-clonotype-analysis.html#cb393-9"></a><span class="co"># Nclonotypes2$Case=&quot;tumor&quot;</span></span>
<span id="cb393-10"><a href="bcr-clonotype-analysis.html#cb393-10"></a><span class="co"># Nclonotypes2$Case[grep(&quot;match&quot;, Nclonotypes2$Var1)]=&quot;mammary&quot;</span></span>
<span id="cb393-11"><a href="bcr-clonotype-analysis.html#cb393-11"></a><span class="co"># Nclonotypes2$Case[grep(&quot;Control&quot;, Nclonotypes2$Var1)]=&quot;control&quot;</span></span>
<span id="cb393-12"><a href="bcr-clonotype-analysis.html#cb393-12"></a></span>
<span id="cb393-13"><a href="bcr-clonotype-analysis.html#cb393-13"></a></span>
<span id="cb393-14"><a href="bcr-clonotype-analysis.html#cb393-14"></a><span class="kw">head</span>(Nclonotypes2)</span>
<span id="cb393-15"><a href="bcr-clonotype-analysis.html#cb393-15"></a><span class="co">#pdf(sprintf(&quot;rslt/TRUST4/compare_mammary_vs_tumor_%s.pdf&quot;, Sys.Date()), width=8, height=6)</span></span>
<span id="cb393-16"><a href="bcr-clonotype-analysis.html#cb393-16"></a><span class="co">#a1=sapply(unique(ProgTemp$Calc), function(x) wilcox.test(ProgTemp$Val[ProgTemp$Calc==x &amp; ProgTemp$Batch==1], ProgTemp$Val[ProgTemp$Calc==x &amp; ProgTemp$Batch==&quot;match&quot;])$p.value)</span></span>
<span id="cb393-17"><a href="bcr-clonotype-analysis.html#cb393-17"></a></span>
<span id="cb393-18"><a href="bcr-clonotype-analysis.html#cb393-18"></a>p=<span class="kw">ggplot</span>(ProgTemp, <span class="kw">aes</span>(<span class="dt">x=</span>Growth, <span class="dt">y=</span>Val))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>Calc)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">sprintf</span>(<span class="st">&quot;wilcox pval: Gini %s, Shannon %s, Top10 = %s&quot;</span>, <span class="kw">round</span>(a1[<span class="dv">2</span>],<span class="dv">2</span>), <span class="kw">round</span>(a1[<span class="dv">1</span>],<span class="dv">2</span>), <span class="kw">round</span>(a1[<span class="dv">3</span>],<span class="dv">2</span>)))<span class="op">+</span><span class="kw">geom_jitter</span>(<span class="dt">col=</span><span class="st">&quot;grey&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()</span>
<span id="cb393-19"><a href="bcr-clonotype-analysis.html#cb393-19"></a><span class="kw">print</span>(p)</span>
<span id="cb393-20"><a href="bcr-clonotype-analysis.html#cb393-20"></a></span>
<span id="cb393-21"><a href="bcr-clonotype-analysis.html#cb393-21"></a></span>
<span id="cb393-22"><a href="bcr-clonotype-analysis.html#cb393-22"></a><span class="kw">write.csv</span>(ProgTemp, <span class="dt">file=</span><span class="st">&quot;nature-tables/4i_clonotypes.csv&quot;</span>)</span>
<span id="cb393-23"><a href="bcr-clonotype-analysis.html#cb393-23"></a></span>
<span id="cb393-24"><a href="bcr-clonotype-analysis.html#cb393-24"></a></span>
<span id="cb393-25"><a href="bcr-clonotype-analysis.html#cb393-25"></a><span class="co"># a1=sapply(unique(Nclonotypes2$Var2), function(x) wilcox.test(Nclonotypes2$value[Nclonotypes2$Var2==x &amp; Nclonotypes2$Case==&quot;tumor&quot;], Nclonotypes2$value[Nclonotypes2$Var2==x &amp; Nclonotypes2$Case==&quot;mammary&quot;])$p.value)</span></span>
<span id="cb393-26"><a href="bcr-clonotype-analysis.html#cb393-26"></a><span class="co"># </span></span>
<span id="cb393-27"><a href="bcr-clonotype-analysis.html#cb393-27"></a><span class="co"># ann_text2 &lt;- data.frame(lab=paste(&quot;p=&quot;, round(a1,4),sep=&quot;&quot;), Var2=unique(Nclonotypes2$Var2), Case=1, value=2)</span></span>
<span id="cb393-28"><a href="bcr-clonotype-analysis.html#cb393-28"></a><span class="co"># p=ggplot(Nclonotypes2, aes(x=Case, y=value))+geom_boxplot()+facet_grid(~Var2)+ggtitle(&quot;Nclonotype with different filters&quot;)+geom_jitter(col=&quot;grey&quot;)+theme_bw()+</span></span>
<span id="cb393-29"><a href="bcr-clonotype-analysis.html#cb393-29"></a><span class="co">#   geom_text(data=ann_text2, aes(label=lab))</span></span>
<span id="cb393-30"><a href="bcr-clonotype-analysis.html#cb393-30"></a><span class="co"># print(p)</span></span></code></pre></div>
</div>
<div id="associate-with-clinicopathological-data-progression" class="section level2">
<h2><span class="header-section-number">19.5</span> Associate with clinicopathological data (progression)</h2>
<p>We will associate the following metrics:</p>
<ul>
<li>total unique clones</li>
<li>diversity</li>
</ul>
<p>with the following immune data:</p>
<ul>
<li>CD45 fraction (from FACS)</li>
<li>CD8 fraction (from WSI)</li>
<li>MH-mixing indices</li>
<li>Tumor Size</li>
<li>Growth</li>
<li>Treatment</li>
</ul>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="bcr-clonotype-analysis.html#cb394-1"></a>matchidx=<span class="kw">match</span>(<span class="kw">rownames</span>(TableOut), infoTableFinal<span class="op">$</span>SampleID)</span>
<span id="cb394-2"><a href="bcr-clonotype-analysis.html#cb394-2"></a></span>
<span id="cb394-3"><a href="bcr-clonotype-analysis.html#cb394-3"></a>TrustTab=infoTableFinal[matchidx, ]</span></code></pre></div>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="bcr-clonotype-analysis.html#cb395-1"></a>TrustTab<span class="op">$</span>TrustNclonotypes=TableOut[<span class="kw">match</span>(TrustTab<span class="op">$</span>SampleID, <span class="kw">rownames</span>(TableOut)) ,<span class="dv">2</span>]</span>
<span id="cb395-2"><a href="bcr-clonotype-analysis.html#cb395-2"></a>TrustTab<span class="op">$</span>TrustShannon=dfAll<span class="op">$</span>Val[<span class="kw">match</span>(TrustTab<span class="op">$</span>SampleID, dfAll<span class="op">$</span>sample) ]</span>
<span id="cb395-3"><a href="bcr-clonotype-analysis.html#cb395-3"></a>TrustTab<span class="op">$</span>TrustTop10=df2<span class="op">$</span>Val[<span class="kw">match</span>(TrustTab<span class="op">$</span>SampleID, df2<span class="op">$</span>sample)]</span>
<span id="cb395-4"><a href="bcr-clonotype-analysis.html#cb395-4"></a>TrustTab<span class="op">$</span>TrustGini=dfAll<span class="op">$</span>Val[<span class="kw">which</span>(dfAll<span class="op">$</span>Calc<span class="op">==</span><span class="st">&quot;Gini&quot;</span>)[<span class="kw">match</span>(TrustTab<span class="op">$</span>SampleID, dfAll<span class="op">$</span>sample)] ]</span>
<span id="cb395-5"><a href="bcr-clonotype-analysis.html#cb395-5"></a>TrustTab<span class="op">$</span>CD45FACS=<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(Cdata<span class="op">$</span>CD45.Frac.FACS[<span class="kw">match</span>(TrustTab<span class="op">$</span>TumorID, Cdata<span class="op">$</span>TumorID)]))</span>
<span id="cb395-6"><a href="bcr-clonotype-analysis.html#cb395-6"></a>TrustTab<span class="op">$</span>UnclassFrac=<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(Cdata<span class="op">$</span>UnclassFraction[<span class="kw">match</span>(TrustTab<span class="op">$</span>TumorID, Cdata<span class="op">$</span>TumorID)]))</span>
<span id="cb395-7"><a href="bcr-clonotype-analysis.html#cb395-7"></a>CD45F=<span class="kw">which</span>(TrustTab<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span>)</span>
<span id="cb395-8"><a href="bcr-clonotype-analysis.html#cb395-8"></a><span class="co">#TrustTab=TrustTab[TrustTab$Fraction==&quot;CD45&quot;, ]</span></span>
<span id="cb395-9"><a href="bcr-clonotype-analysis.html#cb395-9"></a></span>
<span id="cb395-10"><a href="bcr-clonotype-analysis.html#cb395-10"></a>paramSearch=<span class="kw">c</span>(<span class="st">&quot;CD8Frac&quot;</span>, <span class="st">&quot;TumSize&quot;</span>, <span class="st">&quot;CD45FACS&quot;</span>, <span class="st">&quot;MHEpCAM&quot;</span>, <span class="st">&quot;MHSMA&quot;</span>, <span class="st">&quot;IFEpCAM&quot;</span>, <span class="st">&quot;IFSMA&quot;</span>, <span class="st">&quot;UnclassFrac&quot;</span>)</span>
<span id="cb395-11"><a href="bcr-clonotype-analysis.html#cb395-11"></a>FeatTrust=<span class="kw">c</span>(<span class="st">&quot;TrustNclonotypes&quot;</span>, <span class="st">&quot;TrustShannon&quot;</span>, <span class="st">&quot;TrustGini&quot;</span>, <span class="st">&quot;TrustTop10&quot;</span>)</span>
<span id="cb395-12"><a href="bcr-clonotype-analysis.html#cb395-12"></a></span>
<span id="cb395-13"><a href="bcr-clonotype-analysis.html#cb395-13"></a>Nclon=<span class="kw">sapply</span>(paramSearch, <span class="cf">function</span>(x) <span class="kw">cor.test</span>(TrustTab[CD45F ,x], TrustTab[ CD45F,<span class="kw">c</span>(<span class="st">&quot;TrustNclonotypes&quot;</span>)], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb395-14"><a href="bcr-clonotype-analysis.html#cb395-14"></a>SDiv=<span class="kw">sapply</span>(paramSearch, <span class="cf">function</span>(x) <span class="kw">cor.test</span>(TrustTab[CD45F ,x], TrustTab[CD45F ,<span class="kw">c</span>(<span class="st">&quot;TrustShannon&quot;</span>)], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb395-15"><a href="bcr-clonotype-analysis.html#cb395-15"></a>GDiv=<span class="kw">sapply</span>(paramSearch, <span class="cf">function</span>(x) <span class="kw">cor.test</span>(TrustTab[ CD45F,x], TrustTab[CD45F ,<span class="kw">c</span>(<span class="st">&quot;TrustGini&quot;</span>)], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb395-16"><a href="bcr-clonotype-analysis.html#cb395-16"></a>TDiv=<span class="kw">sapply</span>(paramSearch, <span class="cf">function</span>(x) <span class="kw">cor.test</span>(TrustTab[CD45F ,x], TrustTab[CD45F ,<span class="kw">c</span>(<span class="st">&quot;TrustTop10&quot;</span>)], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb395-17"><a href="bcr-clonotype-analysis.html#cb395-17"></a></span>
<span id="cb395-18"><a href="bcr-clonotype-analysis.html#cb395-18"></a>AllSmerge=<span class="kw">rbind</span>(Nclon, SDiv, GDiv, TDiv)</span>
<span id="cb395-19"><a href="bcr-clonotype-analysis.html#cb395-19"></a></span>
<span id="cb395-20"><a href="bcr-clonotype-analysis.html#cb395-20"></a><span class="co">#pdf(sprintf(&quot;rslt/TRUST4/association_trustmetrics_clinicopathological_%s.pdf&quot;, Sys.Date()), width=10, height=8)</span></span>
<span id="cb395-21"><a href="bcr-clonotype-analysis.html#cb395-21"></a></span>
<span id="cb395-22"><a href="bcr-clonotype-analysis.html#cb395-22"></a><span class="kw">heatmap.2</span>(AllSmerge, <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">main=</span><span class="st">&quot;p values of associations&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:trust-progression-1"></span>
<img src="RatDCIS-bookdown_files/figure-html/trust-progression-1.png" alt="4i: trust result" width="672" />
<p class="caption">
Figure 19.3: 4i: trust result
</p>
</div>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="bcr-clonotype-analysis.html#cb396-1"></a></span>
<span id="cb396-2"><a href="bcr-clonotype-analysis.html#cb396-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb396-3"><a href="bcr-clonotype-analysis.html#cb396-3"></a>idx=<span class="kw">which</span>(AllSmerge<span class="op">&lt;</span><span class="fl">0.1</span>, <span class="dt">arr.ind=</span>T)</span>
<span id="cb396-4"><a href="bcr-clonotype-analysis.html#cb396-4"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(idx)){</span>
<span id="cb396-5"><a href="bcr-clonotype-analysis.html#cb396-5"></a>  a1=<span class="kw">cor.test</span>(TrustTab[CD45F , FeatTrust[idx[i, <span class="dv">1</span>]]]  ,TrustTab[CD45F , paramSearch[idx[i, <span class="dv">2</span>]]], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb396-6"><a href="bcr-clonotype-analysis.html#cb396-6"></a>  <span class="kw">plot</span>(TrustTab[CD45F , FeatTrust[idx[i, <span class="dv">1</span>]]]  ,TrustTab[CD45F , paramSearch[idx[i, <span class="dv">2</span>]]], <span class="dt">xlab=</span>FeatTrust[idx[i, <span class="dv">1</span>]], <span class="dt">ylab=</span>paramSearch[idx[i, <span class="dv">2</span>]],</span>
<span id="cb396-7"><a href="bcr-clonotype-analysis.html#cb396-7"></a>       <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;cor:%s p:%s&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate,<span class="dv">2</span>), <span class="kw">round</span>(a1<span class="op">$</span>p.value,<span class="dv">2</span>)))</span>
<span id="cb396-8"><a href="bcr-clonotype-analysis.html#cb396-8"></a>  <span class="kw">text</span>(TrustTab[CD45F , FeatTrust[idx[i, <span class="dv">1</span>]]]  ,TrustTab[CD45F , paramSearch[idx[i, <span class="dv">2</span>]]], <span class="kw">rownames</span>(TrustTab)[CD45F], <span class="dt">cex=</span><span class="fl">0.75</span>)</span>
<span id="cb396-9"><a href="bcr-clonotype-analysis.html#cb396-9"></a>}</span></code></pre></div>
<div class="figure"><span id="fig:trust-progression-2"></span>
<img src="RatDCIS-bookdown_files/figure-html/trust-progression-2.png" alt="4i: trust result" width="672" />
<p class="caption">
Figure 19.4: 4i: trust result
</p>
</div>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="bcr-clonotype-analysis.html#cb397-1"></a></span>
<span id="cb397-2"><a href="bcr-clonotype-analysis.html#cb397-2"></a><span class="co">#pdf(&quot;~/Desktop/4L-TRUST-rslt-associate-outcome-progression.pdf&quot;, height=8, width=8)</span></span>
<span id="cb397-3"><a href="bcr-clonotype-analysis.html#cb397-3"></a></span>
<span id="cb397-4"><a href="bcr-clonotype-analysis.html#cb397-4"></a>CD45F2=<span class="kw">which</span>(TrustTab<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>TrustTab<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>)</span>
<span id="cb397-5"><a href="bcr-clonotype-analysis.html#cb397-5"></a></span>
<span id="cb397-6"><a href="bcr-clonotype-analysis.html#cb397-6"></a>yvals=<span class="kw">c</span>(<span class="st">&quot;TrustNclonotypes&quot;</span>, <span class="st">&quot;TrustShannon&quot;</span>, <span class="st">&quot;TrustGini&quot;</span>, <span class="st">&quot;TrustTop10&quot;</span>)</span>
<span id="cb397-7"><a href="bcr-clonotype-analysis.html#cb397-7"></a><span class="co">## associate with growth and treatment type</span></span>
<span id="cb397-8"><a href="bcr-clonotype-analysis.html#cb397-8"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<div class="figure"><span id="fig:trust-progression-3"></span>
<img src="RatDCIS-bookdown_files/figure-html/trust-progression-3.png" alt="4i: trust result" width="672" />
<p class="caption">
Figure 19.5: 4i: trust result
</p>
</div>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="bcr-clonotype-analysis.html#cb398-1"></a>CD45F2a=<span class="kw">which</span>(TrustTab<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>TrustTab<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span> <span class="op">&amp;</span><span class="st"> </span></span>
<span id="cb398-2"><a href="bcr-clonotype-analysis.html#cb398-2"></a><span class="st">                </span>TrustTab<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>, <span class="st">&quot;PDL1&quot;</span>))</span>
<span id="cb398-3"><a href="bcr-clonotype-analysis.html#cb398-3"></a>a1=<span class="kw">sapply</span>(yvals, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(TrustTab[CD45F2a ,x]<span class="op">~</span>TrustTab[CD45F2a, <span class="st">&quot;Treatment&quot;</span>])<span class="op">$</span>p.value)</span>
<span id="cb398-4"><a href="bcr-clonotype-analysis.html#cb398-4"></a>CD45F2a=<span class="kw">which</span>(TrustTab<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>TrustTab<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span> <span class="op">&amp;</span><span class="st"> </span></span>
<span id="cb398-5"><a href="bcr-clonotype-analysis.html#cb398-5"></a><span class="st">                </span>TrustTab<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>, <span class="st">&quot;LY&quot;</span>))</span>
<span id="cb398-6"><a href="bcr-clonotype-analysis.html#cb398-6"></a>b1=<span class="kw">sapply</span>(yvals, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(TrustTab[CD45F2a ,x]<span class="op">~</span>TrustTab[CD45F2a, <span class="st">&quot;Treatment&quot;</span>])<span class="op">$</span>p.value)</span>
<span id="cb398-7"><a href="bcr-clonotype-analysis.html#cb398-7"></a>CD45F2a=<span class="kw">which</span>(TrustTab<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>TrustTab<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span> <span class="op">&amp;</span><span class="st"> </span></span>
<span id="cb398-8"><a href="bcr-clonotype-analysis.html#cb398-8"></a><span class="st">                </span>TrustTab<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>, <span class="st">&quot;PDL1+LY&quot;</span>))</span>
<span id="cb398-9"><a href="bcr-clonotype-analysis.html#cb398-9"></a>c1=<span class="kw">sapply</span>(yvals, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(TrustTab[CD45F2a ,x]<span class="op">~</span>TrustTab[CD45F2a, <span class="st">&quot;Treatment&quot;</span>])<span class="op">$</span>p.value)</span>
<span id="cb398-10"><a href="bcr-clonotype-analysis.html#cb398-10"></a></span>
<span id="cb398-11"><a href="bcr-clonotype-analysis.html#cb398-11"></a></span>
<span id="cb398-12"><a href="bcr-clonotype-analysis.html#cb398-12"></a>TrustTabmelt=<span class="kw">melt</span>(TrustTab[CD45F2, ], <span class="dt">measure.vars=</span><span class="kw">c</span>(<span class="st">&quot;TrustNclonotypes&quot;</span>, </span>
<span id="cb398-13"><a href="bcr-clonotype-analysis.html#cb398-13"></a>                                                     <span class="st">&quot;TrustShannon&quot;</span>, <span class="st">&quot;TrustGini&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="bcr-clonotype-analysis.html#cb399-1"></a>tablex=TrustTab[CD45F2, <span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;Growth&quot;</span>, <span class="st">&quot;TrustNclonotypes&quot;</span>, </span>
<span id="cb399-2"><a href="bcr-clonotype-analysis.html#cb399-2"></a>                          <span class="st">&quot;TrustShannon&quot;</span>, <span class="st">&quot;TrustTop10&quot;</span>, <span class="st">&quot;TrustGini&quot;</span>)]</span>
<span id="cb399-3"><a href="bcr-clonotype-analysis.html#cb399-3"></a>tablexM=<span class="kw">melt</span>(tablex)</span>
<span id="cb399-4"><a href="bcr-clonotype-analysis.html#cb399-4"></a></span>
<span id="cb399-5"><a href="bcr-clonotype-analysis.html#cb399-5"></a><span class="kw">ggplot</span>(tablexM, <span class="kw">aes</span>(<span class="dt">x=</span>Treatment, <span class="dt">y=</span>value, <span class="dt">col=</span>Treatment))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>variable, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">scale_colour_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(ColMerge[ ,<span class="dv">1</span>], <span class="st">&quot;black&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()</span></code></pre></div>
<div class="figure"><span id="fig:4i"></span>
<img src="RatDCIS-bookdown_files/figure-html/4i-1.png" alt="trust treatment" width="672" />
<p class="caption">
Figure 19.6: trust treatment
</p>
</div>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="bcr-clonotype-analysis.html#cb400-1"></a></span>
<span id="cb400-2"><a href="bcr-clonotype-analysis.html#cb400-2"></a><span class="kw">write.csv</span>(tablexM, <span class="dt">file=</span><span class="st">&quot;nature-tables/4i_trust-treatment.csv&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb401-1"><a href="bcr-clonotype-analysis.html#cb401-1"></a><span class="kw">print</span>(<span class="st">&#39;P values PDL1&#39;</span>)</span>
<span id="cb401-2"><a href="bcr-clonotype-analysis.html#cb401-2"></a><span class="co">## [1] &quot;P values PDL1&quot;</span></span>
<span id="cb401-3"><a href="bcr-clonotype-analysis.html#cb401-3"></a>a1</span>
<span id="cb401-4"><a href="bcr-clonotype-analysis.html#cb401-4"></a><span class="co">## TrustNclonotypes     TrustShannon        TrustGini       TrustTop10 </span></span>
<span id="cb401-5"><a href="bcr-clonotype-analysis.html#cb401-5"></a><span class="co">##       0.07565166       0.76182641       0.14827564       0.75673324</span></span>
<span id="cb401-6"><a href="bcr-clonotype-analysis.html#cb401-6"></a><span class="kw">print</span>(<span class="st">&#39;P values LY&#39;</span>)</span>
<span id="cb401-7"><a href="bcr-clonotype-analysis.html#cb401-7"></a><span class="co">## [1] &quot;P values LY&quot;</span></span>
<span id="cb401-8"><a href="bcr-clonotype-analysis.html#cb401-8"></a>b1</span>
<span id="cb401-9"><a href="bcr-clonotype-analysis.html#cb401-9"></a><span class="co">## TrustNclonotypes     TrustShannon        TrustGini       TrustTop10 </span></span>
<span id="cb401-10"><a href="bcr-clonotype-analysis.html#cb401-10"></a><span class="co">##       0.06908261       0.73961333       0.23880663       0.73419223</span></span>
<span id="cb401-11"><a href="bcr-clonotype-analysis.html#cb401-11"></a><span class="kw">print</span>(<span class="st">&#39;P values PDL1+LY&#39;</span>)</span>
<span id="cb401-12"><a href="bcr-clonotype-analysis.html#cb401-12"></a><span class="co">## [1] &quot;P values PDL1+LY&quot;</span></span>
<span id="cb401-13"><a href="bcr-clonotype-analysis.html#cb401-13"></a>c1</span>
<span id="cb401-14"><a href="bcr-clonotype-analysis.html#cb401-14"></a><span class="co">## TrustNclonotypes     TrustShannon        TrustGini       TrustTop10 </span></span>
<span id="cb401-15"><a href="bcr-clonotype-analysis.html#cb401-15"></a><span class="co">##        0.5623927        0.3281335        0.2388066        0.9380125</span></span></code></pre></div>
<div class="sourceCode" id="cb402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb402-1"><a href="bcr-clonotype-analysis.html#cb402-1"></a><span class="kw">pdf</span>(<span class="st">&quot;figure-outputs/4I_BCR_clonotypes.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">8</span>, <span class="dt">width=</span><span class="dv">5</span>)</span>
<span id="cb402-2"><a href="bcr-clonotype-analysis.html#cb402-2"></a></span>
<span id="cb402-3"><a href="bcr-clonotype-analysis.html#cb402-3"></a><span class="kw">ggplot</span>(tablexM, <span class="kw">aes</span>(<span class="dt">x=</span>Growth, <span class="dt">y=</span>value, <span class="dt">col=</span>Growth))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>variable, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">scale_colour_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(ColSizeb, <span class="st">&quot;black&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()</span>
<span id="cb402-4"><a href="bcr-clonotype-analysis.html#cb402-4"></a><span class="kw">dev.off</span>()</span>
<span id="cb402-5"><a href="bcr-clonotype-analysis.html#cb402-5"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb402-6"><a href="bcr-clonotype-analysis.html#cb402-6"></a><span class="co">##                 2</span></span>
<span id="cb402-7"><a href="bcr-clonotype-analysis.html#cb402-7"></a><span class="co">#par(mfrow=c(2,2))</span></span>
<span id="cb402-8"><a href="bcr-clonotype-analysis.html#cb402-8"></a></span>
<span id="cb402-9"><a href="bcr-clonotype-analysis.html#cb402-9"></a>a1=<span class="kw">sapply</span>(yvals, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(TrustTab[CD45F2 ,x]<span class="op">~</span>TrustTab[CD45F2, <span class="st">&quot;Growth&quot;</span>])<span class="op">$</span>p.value)</span>
<span id="cb402-10"><a href="bcr-clonotype-analysis.html#cb402-10"></a><span class="co">#boxplot(TrustTab[CD45F2 ,c(&quot;TrustNclonotypes&quot;)]~TrustTab[CD45F2, &quot;Growth&quot;], main=sprintf(&quot;Growth: BCR # clonotypes %s&quot;, round(a1[1], 2)), xlab=&quot;growth&quot;, ylab=&quot;no clonotypes&quot;, col=ColSizeb)</span></span>
<span id="cb402-11"><a href="bcr-clonotype-analysis.html#cb402-11"></a><span class="co">#boxplot(TrustTab[CD45F2 ,c(&quot;TrustShannon&quot;)]~TrustTab[CD45F2, &quot;Growth&quot;], main=sprintf(&quot;Growth: BCR # shan %s&quot;, round(a1[2], 2)), xlab=&quot;growth&quot;, ylab=&quot;shannon&quot;, col=ColSizeb)</span></span>
<span id="cb402-12"><a href="bcr-clonotype-analysis.html#cb402-12"></a><span class="co">#boxplot(TrustTab[CD45F2 ,c(&quot;TrustGini&quot;)]~TrustTab[CD45F2, &quot;Growth&quot;], main=sprintf(&quot;Growth: Gini %s&quot;, round(a1[3], 2)),         ylab=&quot;gini&quot;, xlab=&quot;growth&quot;, col=ColSizeb)</span></span>
<span id="cb402-13"><a href="bcr-clonotype-analysis.html#cb402-13"></a><span class="co">#boxplot(TrustTab[CD45F2 ,c(&quot;TrustTop10&quot;)]~TrustTab[CD45F2, &quot;Growth&quot;],  main=sprintf(&quot;Growth: Top10&quot;, round(a1[4], 2)),ylab=&quot;top10 clonotypes&quot;, xlab=&quot;growth&quot;, col=ColSizeb)</span></span>
<span id="cb402-14"><a href="bcr-clonotype-analysis.html#cb402-14"></a></span>
<span id="cb402-15"><a href="bcr-clonotype-analysis.html#cb402-15"></a></span>
<span id="cb402-16"><a href="bcr-clonotype-analysis.html#cb402-16"></a><span class="kw">print</span>(<span class="st">&#39;p values here&#39;</span>)</span>
<span id="cb402-17"><a href="bcr-clonotype-analysis.html#cb402-17"></a><span class="co">## [1] &quot;p values here&quot;</span></span>
<span id="cb402-18"><a href="bcr-clonotype-analysis.html#cb402-18"></a>a1</span>
<span id="cb402-19"><a href="bcr-clonotype-analysis.html#cb402-19"></a><span class="co">## TrustNclonotypes     TrustShannon        TrustGini       TrustTop10 </span></span>
<span id="cb402-20"><a href="bcr-clonotype-analysis.html#cb402-20"></a><span class="co">##       0.07072366       0.17037895       0.02795396       0.91880063</span></span></code></pre></div>
</div>
<div id="associate-with-signature-scores" class="section level2">
<h2><span class="header-section-number">19.6</span> Associate with signature scores</h2>
<p>Compare to B cell counts and activation status based on RNA-seq</p>
<ul>
<li>B cell signature enrichments scores (from RNA data)</li>
</ul>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="bcr-clonotype-analysis.html#cb403-1"></a>bidx=<span class="kw">grep</span>(<span class="st">&quot;B cell&quot;</span>, output1<span class="op">$</span>cell_type)</span>
<span id="cb403-2"><a href="bcr-clonotype-analysis.html#cb403-2"></a>idx2=<span class="kw">match</span>(TrustTab<span class="op">$</span>SampleID, <span class="kw">colnames</span>(output1))</span>
<span id="cb403-3"><a href="bcr-clonotype-analysis.html#cb403-3"></a></span>
<span id="cb403-4"><a href="bcr-clonotype-analysis.html#cb403-4"></a>Bsig=output1[bidx, <span class="kw">na.omit</span>(idx2)]</span>
<span id="cb403-5"><a href="bcr-clonotype-analysis.html#cb403-5"></a><span class="kw">rownames</span>(Bsig)=output1<span class="op">$</span>cell_type[bidx]</span>
<span id="cb403-6"><a href="bcr-clonotype-analysis.html#cb403-6"></a></span>
<span id="cb403-7"><a href="bcr-clonotype-analysis.html#cb403-7"></a></span>
<span id="cb403-8"><a href="bcr-clonotype-analysis.html#cb403-8"></a>Bassoc=<span class="kw">sapply</span>(FeatTrust, <span class="cf">function</span>(y) <span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(Bsig), <span class="cf">function</span>(x) <span class="kw">cor</span>( <span class="kw">t</span>(Bsig[x, ]),(TrustTab[CD45F, y]), <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)))</span>
<span id="cb403-9"><a href="bcr-clonotype-analysis.html#cb403-9"></a>Bassoc2=<span class="kw">sapply</span>(FeatTrust, <span class="cf">function</span>(y) <span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(Bsig), <span class="cf">function</span>(x) <span class="kw">cor.test</span>( <span class="kw">t</span>(Bsig[x, ]),(TrustTab[CD45F, y]), <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)<span class="op">$</span>p.value))</span>
<span id="cb403-10"><a href="bcr-clonotype-analysis.html#cb403-10"></a><span class="kw">rownames</span>(Bassoc)=<span class="kw">rownames</span>(Bsig)</span>
<span id="cb403-11"><a href="bcr-clonotype-analysis.html#cb403-11"></a></span>
<span id="cb403-12"><a href="bcr-clonotype-analysis.html#cb403-12"></a>BassocM=Bassoc</span>
<span id="cb403-13"><a href="bcr-clonotype-analysis.html#cb403-13"></a>BassocM[<span class="kw">which</span>(Bassoc2<span class="op">&lt;</span><span class="fl">0.1</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">0</span></span>
<span id="cb403-14"><a href="bcr-clonotype-analysis.html#cb403-14"></a></span>
<span id="cb403-15"><a href="bcr-clonotype-analysis.html#cb403-15"></a></span>
<span id="cb403-16"><a href="bcr-clonotype-analysis.html#cb403-16"></a><span class="co">#pdf(sprintf(&quot;rslt/TRUST4/association_Bcell_signatures_%s.pdf&quot;, Sys.Date()), width=7, height=6)</span></span>
<span id="cb403-17"><a href="bcr-clonotype-analysis.html#cb403-17"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">5</span>))</span>
<span id="cb403-18"><a href="bcr-clonotype-analysis.html#cb403-18"></a><span class="kw">heatmap.2</span>(Bassoc, <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;correlation&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-132-1.png" width="672" /></p>
<div class="sourceCode" id="cb404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb404-1"><a href="bcr-clonotype-analysis.html#cb404-1"></a><span class="kw">heatmap.2</span>(BassocM, <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;significant associations p&lt;0.1 only&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-132-2.png" width="672" /></p>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="bcr-clonotype-analysis.html#cb405-1"></a><span class="co">#dev.off()</span></span></code></pre></div>
<ul>
<li>B cell frequencies based on FACS:</li>
</ul>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="bcr-clonotype-analysis.html#cb406-1"></a>lx1=<span class="kw">match</span>(<span class="kw">colnames</span>(Fdata), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">unlist</span>(<span class="kw">strsplit</span>( <span class="kw">rownames</span>(TrustTab), <span class="st">&quot;_CD45&quot;</span>))))</span>
<span id="cb406-2"><a href="bcr-clonotype-analysis.html#cb406-2"></a></span>
<span id="cb406-3"><a href="bcr-clonotype-analysis.html#cb406-3"></a><span class="co">#pdf(sprintf(&quot;rslt/TRUST4/BCR_association_FACS_%s.pdf&quot;, Sys.Date()), width=8, height=8)</span></span>
<span id="cb406-4"><a href="bcr-clonotype-analysis.html#cb406-4"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb406-5"><a href="bcr-clonotype-analysis.html#cb406-5"></a>a1=<span class="kw">cor.test</span>(<span class="kw">as.numeric</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ])), TrustTab<span class="op">$</span>TrustNclonotypes[lx1], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb406-6"><a href="bcr-clonotype-analysis.html#cb406-6"></a><span class="kw">plot</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ]), TrustTab<span class="op">$</span>TrustNclonotypes[lx1], <span class="dt">xlab=</span><span class="st">&quot;B cell FACS fraction&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;# clonotypes&quot;</span>, <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;cor:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="st">&quot; p:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>p.value,<span class="dv">2</span>) ,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb406-7"><a href="bcr-clonotype-analysis.html#cb406-7"></a><span class="kw">text</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ]), TrustTab<span class="op">$</span>TrustNclonotypes[lx1], TrustTab<span class="op">$</span>TumorID[lx1])</span>
<span id="cb406-8"><a href="bcr-clonotype-analysis.html#cb406-8"></a>a1=<span class="kw">cor.test</span>(<span class="kw">as.numeric</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ])), TrustTab<span class="op">$</span>TrustShannon[lx1], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb406-9"><a href="bcr-clonotype-analysis.html#cb406-9"></a><span class="kw">plot</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ]), TrustTab<span class="op">$</span>TrustShannon[lx1], <span class="dt">xlab=</span><span class="st">&quot;B cell FACS fraction&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Shannon Index&quot;</span>,  <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;cor:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="st">&quot; p:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>p.value,<span class="dv">2</span>),<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb406-10"><a href="bcr-clonotype-analysis.html#cb406-10"></a><span class="kw">text</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ]), TrustTab<span class="op">$</span>TrustShannon[lx1], TrustTab<span class="op">$</span>TumorID[lx1])</span>
<span id="cb406-11"><a href="bcr-clonotype-analysis.html#cb406-11"></a>a1=<span class="kw">cor.test</span>(<span class="kw">as.numeric</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ])), TrustTab<span class="op">$</span>TrustGini[lx1], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb406-12"><a href="bcr-clonotype-analysis.html#cb406-12"></a><span class="kw">plot</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ]), TrustTab<span class="op">$</span>TrustGini[lx1], <span class="dt">xlab=</span><span class="st">&quot;B cell FACS fraction&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Gini Index&quot;</span>,  <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;cor:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="st">&quot; p:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>p.value,<span class="dv">2</span>),<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb406-13"><a href="bcr-clonotype-analysis.html#cb406-13"></a><span class="kw">text</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ]), TrustTab<span class="op">$</span>TrustGini[lx1], TrustTab<span class="op">$</span>TumorID[lx1])</span>
<span id="cb406-14"><a href="bcr-clonotype-analysis.html#cb406-14"></a>a1=<span class="kw">cor.test</span>(<span class="kw">as.numeric</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ])), TrustTab<span class="op">$</span>TrustTop10[lx1], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb406-15"><a href="bcr-clonotype-analysis.html#cb406-15"></a><span class="kw">plot</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ]), TrustTab<span class="op">$</span>TrustTop10[lx1], <span class="dt">xlab=</span><span class="st">&quot;B cell FACS fraction&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Top10&quot;</span>,  <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;cor:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="st">&quot; p:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>p.value,<span class="dv">2</span>),<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb406-16"><a href="bcr-clonotype-analysis.html#cb406-16"></a><span class="kw">text</span>(<span class="kw">t</span>(Fdata[ <span class="dv">9</span>, ]), TrustTab<span class="op">$</span>TrustTop10[lx1], TrustTab<span class="op">$</span>TumorID[lx1])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-133-1.png" width="672" /></p>
<div class="sourceCode" id="cb407"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb407-1"><a href="bcr-clonotype-analysis.html#cb407-1"></a><span class="co">#dev.off()</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="immune-estimation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="whole-genome-sequencing-mutations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
