<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Cohort characteristics | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Cohort characteristics | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Cohort characteristics | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="prerequisites.html"/>
<link rel="next" href="cohort-summary-table.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redo-the-survival-plots-based-on-luminal-cases-only"><i class="fa fa-check"></i><b>15.4.3</b> Redo the survival plots based on Luminal cases only</a></li>
<li class="chapter" data-level="15.4.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.4</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>16</b> METABRIC: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="16.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="16.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="16.2.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#all-samples-1"><i class="fa fa-check"></i><b>16.2.1</b> All samples</a></li>
<li class="chapter" data-level="16.2.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases-1"><i class="fa fa-check"></i><b>16.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.4" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint-1"><i class="fa fa-check"></i><b>16.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="16.4.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#overall-survival-1"><i class="fa fa-check"></i><b>16.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="16.4.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#met-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>16.4.2</b> MET: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>21</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="21.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>21.1</b> Data</a><ul>
<li class="chapter" data-level="21.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>21.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>21.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="21.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>21.3</b> Plots</a><ul>
<li class="chapter" data-level="21.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>21.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="21.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>21.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>21.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="21.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>21.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>21.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="21.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>21.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>21.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>22</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>22.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>22.2</b> Load files</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>22.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="22.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>22.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="22.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>22.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>23</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>23.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>23.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>23.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>24</b> Trichrome staining</a><ul>
<li class="chapter" data-level="24.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>24.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="24.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>24.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="24.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>24.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="24.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>24.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>25</b> CNV calling</a><ul>
<li class="chapter" data-level="25.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>25.1</b> Sample data</a></li>
<li class="chapter" data-level="25.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>25.2</b> Data Summary</a><ul>
<li class="chapter" data-level="25.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>25.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="25.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>25.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="25.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>25.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>25.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="25.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>25.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>26</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cohort-characteristics" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Cohort characteristics</h1>
<p>Below we assess summary statistics on clinico-pathological features of this data set. This includes information on:</p>
<ul>
<li>treatment</li>
<li>tumor size</li>
<li>growth rates (mm/week)</li>
<li>number of tumors per rat</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="cohort-characteristics.html#cb10-1"></a>Cdata=xlsx<span class="op">::</span><span class="kw">read.xlsx</span>(<span class="st">&quot;../metadata/Extended Data Table1.xlsx&quot;</span>, <span class="dt">sheetIndex=</span><span class="dv">1</span>)</span>
<span id="cb10-2"><a href="cohort-characteristics.html#cb10-2"></a></span>
<span id="cb10-3"><a href="cohort-characteristics.html#cb10-3"></a>Cdata<span class="op">$</span>NewID[<span class="kw">which</span>(<span class="kw">duplicated</span>(Cdata<span class="op">$</span>NewID)<span class="op">==</span>T)]</span>
<span id="cb10-4"><a href="cohort-characteristics.html#cb10-4"></a><span class="co">## [1] NA NA NA</span></span>
<span id="cb10-5"><a href="cohort-characteristics.html#cb10-5"></a></span>
<span id="cb10-6"><a href="cohort-characteristics.html#cb10-6"></a>DT<span class="op">::</span><span class="kw">datatable</span>(Cdata[ ,<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>],<span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb10-7"><a href="cohort-characteristics.html#cb10-7"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div id="htmlwidget-5591ec28bfe2ba73fbaf" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5591ec28bfe2ba73fbaf">{"x":{"filter":"none","extensions":["Buttons"],"data":[["P24T1","P24T2","P24T3","P25T1","P25T2","P26T1","P26T2","P26T3","P27T1","P28T1","P29T1","P29T2","P30T1","P30T2","P41T1","P42T1","P42T2","P42T3","P42T4","P31T1","P31T2","P31T3","P32T1","P32T2","P33T1","P34T1","P34T2","P34T3","P35T1","P36T1","P36T2","P36T3","P37T1","P38T1","P38T2","P39T1","P39T2","P39T3","P40T1","P1T1","P2T1","P2T2","P3T2","P4T1","P5T1","P5T2","P5T3","P6T1","P6T2","P7T1","P7T2","P8T1","P8T2","P9T1","P10T1","P10T2","P11T1","P11T2","P12T1","P12T2","P12T3","P12T4","P13T1","P14T1","P15T1","P15T2","P16T1","P17T1","P17T2","P17T3","P18T1","P18T2","P19T1","P19T2","P19T3","P19T4","P20T1","P20T2","P20T3","P21T1","P22T1","P22T2","P23T1","P23T2","C10T1","C10T2","C10T3","C10T4","C10T5","C10T6","C10T7","C10T8","C11T1","C12T1","C13T1","C14T1","C14T2","C15T1","C15T2","C16T1","C16T2","C17T1","C17T2","C17T3","C17T4","C18T1",null,null,null,null,"C18T2","C19T1","C1T1","C1T2","C1T3","C2T1","C2T2","C32T1","C32T2","C33T1","C33T2","C34T1","C34T2","C34T3","C3T1","C3T2","C3T3","C3T4","C4T1","C4T2","C4T3","C4T4","C4T5","C5T1","C6T1","C7T1","C8T1","C9T1","C9T2","C20T1","C21T1","C22T1","C22T2","C22T3","C23T1","C24T1","C25T1","C26T1","C27T1","C28T1","C29T1","C30T2","C30T3","C31T1"],["10L_B","10L_C","10L_D","10N_A","10N_D","10R_BL","10R_BU","10R_C","11L_B","11N_D","11R_C","11R_D","12L_C","12L_D","13N_A","14L_A","14L_B","14L_C","14L_D","14N_B","14N_C","14N_D","14R_B","14R_D","15L_B","15N_C","15N_C-D","15N_D","15R_D","16L_A","16L_C","16L_D","16N_D","16R_C","16R_D","17N_A","17N_B","17N_D","17R_B","1L_A","1N_A","1N_D","1R_B","2N_","2R_A","2R_C","2R_D","3L_A","3L_D","3N_B","3N_D","3R_B","3R_C","4L_B","4N_C","4N_D","4R_B","4R_C","5L_A","5L_B","5L_C","5L_D","5N_A","5R_B","6L_DL","6L_DU","6N_D","6R_B","6R_C","6R_D","7L_C","7L_D","7N_A","7N_B","7N_C","7N_D","7R_A","7R_B","7R_D","8L_D","8R_CL","8R_CU","9L_A","9L_B","NMU13_LAL","NMU13_LAU","NMU13_LLL","NMU13_LLU","NMU13_RAL","NMU13_RAU","NMU13_RLL","NMU13_RLU","NMU14_","NMU2_RL","NMU3_LA","NMU4_RA","NMU4_RL","NMU5_LA","NMU5_LL","NMU6_LA","NMU6_RL","NMU7_LA","NMU7_LL","NMU7_RL lower x2","NMU7_RLU","NMU8_RA","Control1_","Control2_","Control3_","Control4_","NMU8_RL","NMU9_LL","C1R_","C1N1","C1N2","C2N_","C2N2_","C4N1_","C4N2_","C5N1","C5N2","C3N_","C3N2_","C3N3_","C2R_","C2R2_","C2R3_","C2R4_","C4P_","C4P2","C4P3","C4P4","C4P5","C5R_","NMU1_LL","NMU10_LL","NMU11_LA","NMU12_LA","NMU12_RL","D11N_","D14R_","D16N_","D16N2_","D16N3_","D16P_","D17N_","D18N_","D2N_","D3N_","D3R_","D4R_","D9N1_","D9N2_","D9R_"],["10L","10L","10L","10N","10N","10R","10R","10R","11L","11N","11R","11R","12L","12L","13N","14L","14L","14L","14L","14N","14N","14N","14R","14R","15L","15N","15N","15N","15R","16L","16L","16L","16N","16R","16R","17N","17N","17N","17R","1L","1N","1N","1R","2N","2R","2R","2R","3L","3L","3N","3N","3R","3R","4L","4N","4N","4R","4R","5L","5L","5L","5L","5N","5R","6L","6L","6N","6R","6R","6R","7L","7L","7N","7N","7N","7N","7R","7R","7R","8L","8R","8R","9L","9L","NMU13","NMU13","NMU13","NMU13","NMU13","NMU13","NMU13","NMU13","NMU14","NMU2","NMU3","NMU4","NMU4","NMU5","NMU5","NMU6","NMU6","NMU7","NMU7","NMU7","NMU7","NMU8","Control1","Control2","Control3","Control4","NMU8","NMU9","C1R","C1N","C1N","C2N","C2N","C4N","C4N","C5N","C5N","C3N","C3N","C3N","C2R","C2R","C2R","C2R","C4P","C4P","C4P","C4P","C4P","C5R","NMU1","NMU10","NMU11","NMU12","NMU12","D11N","D14R","D16N","D16N2","D16N3","D16P","D17N","D18N","D2N","D3N","D3R","D4R","D9N1","D9N2","D9R"],["B","C","D","A","D","BL","BU","C","B","D","C","D","C","D","A","A","B","C","D","B","C","D","B","D","B","C","C-D","D","D","A","C","D","D","C","D","A","B","D","B","A","A","D","B","D","A","C","D","A","D","B","D","B","C","B","C","D","B","C","A","B","C","D","A","B","DL","DU","D","B","C","D","C","D","A","B","C","D","A","B","D","D","CL","CU","A","B","LAL","LAU","LLL","LLU","RAL","RAU","RLL","RLU",null,"RL","LA","RA","RL","LA","LL","LA","RL","LA","LL","RL lower x2","RLU","RA",null,null,null,null,"RL","LL",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"LL","LL","LA","LA","RL",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Progression","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation",null,null,null,null,"Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation","Characterisation"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","cohort-2","cohort-2","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3",null,null,null,null,"penetrance-3","penetrance-3","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","cohort-2","penetrance-3","cohort-2","penetrance-3","penetrance-3","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs"],["PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","PDL1+LY","PDL1+LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Untreated","Untreated","Untreated","Untreated","Vehicle","Vehicle","Vehicle",null,null,"Vehicle",null,null,null,null,null,null,null,null,"Vehicle","Vehicle","Vehicle",null,"Vehicle",null,null,null,null,"Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","35","35","35","35","35","35","35","35","35","49","49","35","35","35","35","35","35","35","35","35","35","35","49","49","35","35","35","35","32-36",null,null,"32-36",null,null,null,null,null,null,null,null,"32-36","32-36","32-36",null,"32-36",null,null,null,null,"32-36","49","35","49","35","35","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36","32-36"],["2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2018-02-14","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2016-10-11","2016-10-11","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2016-10-11","2016-10-11","2017-01-06","2017-01-06","2017-01-06","2017-01-06","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2016-10-11","2017-01-06","2016-10-11","2017-01-06","2017-01-06","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26","2017-05-26"],["2018-03-29","2018-04-12","2018-03-29","2018-11-08","2018-04-05","2018-04-02","2018-04-02","2018-05-23","2018-04-19","2018-05-03","2018-04-26","2018-04-26","2018-05-14","2018-05-14","2018-06-22","2018-03-29","2018-07-26","2018-07-12","2018-08-16","2018-05-03","2018-04-05","2018-04-19","2018-05-07","2018-04-02","2018-04-19","2018-05-24","2018-05-17","2018-05-03","2018-05-03","2018-07-24","2018-06-26","2018-05-15","2018-06-21","2018-06-21","2018-06-21","2018-07-05","2018-05-31","2018-08-02","2018-05-31","2018-08-02","2018-09-20","2018-08-02","2018-08-02","2018-04-02","2018-03-29","2018-03-29","2018-03-29","2018-05-03","2018-04-05","2018-05-03","2018-05-03","2018-04-02","2018-04-02","2018-04-09","2018-04-09","2018-04-09","2018-04-09","2018-04-09","2018-10-03","2018-10-17","2018-10-03","2018-06-06","2018-06-07","2018-06-07","2018-03-29","2018-03-29","2018-04-02","2018-03-29","2018-04-19","2018-04-20","2018-06-14","2018-04-05","2018-06-14","2018-06-14","2018-06-14","2018-05-03","2018-09-06","2018-09-06","2018-04-26","2018-05-03","2018-06-20","2018-05-14","2018-08-29","2018-08-15",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"2017-07-24",null,null,"2017-08-09",null,null,null,null,null,null,null,null,"2017-08-02","2017-08-15","2017-08-15",null,"2017-07-07",null,null,null,null,"2017-08-09",null,null,null,null,null,"2017-08-15","2017-08-02","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-09","2017-08-21","2017-08-09","2017-08-09","2017-08-09","2017-08-09","2017-08-21","2017-08-09"],["2018-05-04","2018-05-04","2018-05-04","2018-11-15","2018-11-15","2018-05-23","2018-05-23","2018-05-23","2018-06-06","2018-06-21","2018-06-06","2018-06-06","2018-06-21","2018-06-21","2018-07-27","2018-10-10","2018-10-10","2018-10-10","2018-10-10","2018-05-04","2018-05-04","2018-05-04","2018-05-23","2018-05-23","2018-05-04","2018-06-06","2018-06-06","2018-06-06","2018-06-06","2018-07-27","2018-07-27","2018-07-27","2018-10-04","2018-08-16","2018-08-16","2018-08-02","2018-08-02","2018-08-02","2018-06-21","2018-11-29","2018-10-25","2018-10-25","2018-10-13","2018-05-02","2018-05-09","2018-05-09","2018-05-09","2018-05-09","2018-05-09","2018-06-21","2018-06-21","2018-05-09","2018-05-09","2018-05-21","2018-05-21","2018-05-21","2018-05-21","2018-05-21","2018-10-24","2018-10-24","2018-10-24","2018-10-24","2018-06-29","2018-06-29","2018-06-29","2018-06-29","2018-04-10","2018-05-04","2018-05-04","2018-05-04","2018-10-10","2018-10-10","2018-07-27","2018-07-27","2018-07-27","2018-07-27","2018-10-04","2018-10-04","2018-10-04","2018-06-21","2018-06-21","2018-06-21","2018-10-10","2018-10-10","2017-06-15","2017-06-15","2017-06-15","2017-06-15","2017-06-15","2017-06-15","2017-06-15","2017-06-15","2017-06-15","2017-01-07","2017-01-10","2017-03-01","2017-03-01","2017-03-01","2017-03-01","2017-03-02","2017-03-02","2017-03-02","2017-03-02","2017-03-02","2017-03-02","2017-03-06","2017-01-07","2017-01-10","2017-01-10","2017-01-10","2017-03-06","2017-03-06","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-01-07","2017-03-08","2017-04-26","2017-04-26","2017-03-08","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21","2017-08-21"],["growing","growing","stable","no data (big)","stable","growing","no data","no data (tiny)","growing","stable","growing","growing","no data (tiny)","stable","growing","stable","growing","stable","growing","growing","stable","stable","growing","stable","growing","growing","growing","stable","growing","no data","stable","stable","stable","growing","stable","growing","stable","stable","growing","stable","stable","growing","stable","growing","growing","growing","growing","growing","growing","stable","stable","growing","growing","growing","growing","growing","growing","growing","no data","no data","growing","stable","growing","growing","stable","growing","growing","stable","growing","stable","growing","stable","stable","growing","growing","stable","stable","growing","stable","stable","growing","growing","growing","growing","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data",null,null,"no data",null,null,null,null,null,null,null,null,"no data","no data","no data",null,"no data",null,null,null,null,"no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data","no data"],[43,57,43,267,50,47,47,98,64,78,71,71,89,89,128,43,162,148,183,78,50,64,82,47,64,99,92,78,78,160,132,90,127,127,127,141,106,169,106,169,218,169,169,47,43,43,43,78,50,78,78,47,47,54,54,54,54,54,231,245,231,112,113,113,43,43,47,43,64,65,120,50,120,120,120,78,204,204,71,78,126,89,196,182,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[36,22,36,7,224,51,51,0,48,49,41,41,38,38,35,195,76,90,55,1,29,15,16,51,15,13,20,34,34,3,31,73,105,56,56,28,63,0,21,119,35,84,72,30,41,41,41,6,34,49,49,37,37,42,42,42,42,42,21,7,21,140,22,22,92,92,8,36,15,14,118,188,43,43,43,85,28,28,161,49,1,38,42,56,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[79,79,79,274,274,98,98,98,112,127,112,112,127,127,163,238,238,238,238,79,79,79,98,98,79,112,112,112,112,163,163,163,232,183,183,169,169,169,127,288,253,253,241,77,84,84,84,84,84,127,127,84,84,96,96,96,96,96,252,252,252,252,135,135,135,135,55,79,79,79,238,238,163,163,163,163,232,232,232,127,127,127,238,238,160,160,160,160,160,160,160,160,160,88,91,54,54,54,54,55,55,55,55,55,55,59,88,91,91,91,59,59,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87,88,61,197,110,61,87,87,87,87,87,87,87,87,87,87,87,87,87,87,87],[35,25,10,12,15,40,12,4,25,3,19,25,4,6,35,1,40,2,40,23,6,5,35,1,10,30,22,3,17,5,7,24,25,25,3,28,4,4,35,17,10,29,15,47,20,15,20,12,30,6,3,35,32,35,15,20,40,15,6,4,22,15,26,35,0,32,40,18,40,6,40,1,30,31,40,0,8,20,20,6,12,27,45,35,8,14,5,4,20,8,7,20,6,10,10,4,1.5,3,4,1,6,2,1,2,4,1,null,null,null,null,5,5,6,35,5,19,8,3,3,7,6,5,2,6,21,10,3,3,25,1,2,2,2,9,10,4,10,12,6,16,13,9,10,17,10,30,30,10,10,20,40,20,20,17],["growing","growing","stable",null,"stable","growing",null,null,"growing","stable","growing","growing",null,"stable","growing","stable","stable","stable","growing","growing","stable","stable","growing","stable","growing","growing","growing","stable","growing",null,"stable","stable","stable","growing","stable","growing",null,"stable","growing","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","stable","stable","growing","growing","growing","growing","growing","growing","growing","stable","growing","growing","stable","growing","growing","stable","growing","growing","growing","growing","stable","growing","stable","growing","growing","growing","stable","stable","growing","stable","stable","growing","growing","growing","growing",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","mucinous carcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","fibroadenoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","fibroadenoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","mucinous carcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","fibroadenoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","mucinous carcinoma","well differentiated mammary solid adenocarcinoma",null,"well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","fibroadenoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","mucinous carcinoma","well differentiated mammary solid adenocarcinoma","fibroadenoma","fibroadenoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","mucinous carcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma","well differentiated mammary solid adenocarcinoma",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["15.2","51.4","49.8",null,null,"14.5","31.7",null,"37.3","70.3","23.8",null,null,"11.2","18.4",null,"12.1",null,"9.74","19.6","36.9","25.3","6.16",null,"45.5","14.8","7.25",null,"27.4",null,"61.7","40.5",null,"9.83",null,"20.4",null,"6.88","4.35",null,null,null,null,"21.8","70.7","73.5",null,"12.5","26.2","24.4",null,"29.7","39.4","1.64","4.86",null,null,null,null,null,null,null,"31","8.93",null,"12.3","6.85","61.1","27.2","80.8","4.66",null,"23.7","30.4","13.2",null,null,null,null,"34.6","31.8","9.79","13","15.8",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["75.7","35.6","39",null,null,"19.4","40.2",null,"46.9","13.2","32.5",null,null,"52","36.3",null,"32.6",null,"52.8","19.9","50.1","49","38.8",null,"32.7","39.9","65.1",null,"42",null,"23.1","27.1",null,"47.7",null,"29.3",null,"46.3","18.4",null,null,null,null,"61.6","21.9","19.2",null,"80.9","69.6","44.5",null,"65.2","40.2","11.5","41.3",null,null,null,null,null,null,null,"24.1","20.2",null,"34.1","41","31.1","63.7","11.1","72.3",null,"33.2","49.4","40",null,null,null,null,"43.4","32.3","29.7","21","15.7",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>NewID<\/th>\n      <th>TumorID<\/th>\n      <th>Rat_ID<\/th>\n      <th>Location<\/th>\n      <th>Cohort<\/th>\n      <th>Char.Cohort.Batch<\/th>\n      <th>Treatment<\/th>\n      <th>Age.at.injection.days.<\/th>\n      <th>Date.of.NMU.injection<\/th>\n      <th>Date.when.tumor.was.first.palpated<\/th>\n      <th>Date.Sacrificed<\/th>\n      <th>Tumor.growth.status<\/th>\n      <th>Time.NMU2Tumor.days<\/th>\n      <th>Time.Tumor2Sac.days<\/th>\n      <th>Time.NMU2Sac.days<\/th>\n      <th>Tumor.diameter.sac.mm<\/th>\n      <th>Tumor.Growth<\/th>\n      <th>Tumor.Histology<\/th>\n      <th>CD45.Frac.FACS<\/th>\n      <th>DN.Frac.FACS<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[12,13,14,15]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="size-information" class="section level2">
<h2><span class="header-section-number">2.1</span> Size information</h2>
<p>We have two cohorts, the characterisation and progression cohorts. Below is a plot of the size distribution in these two cohorts:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="cohort-characteristics.html#cb11-1"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb11-2"><a href="cohort-characteristics.html#cb11-2"></a>X1a=<span class="kw">median</span>(Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)], <span class="dt">na.rm =</span> T)</span>
<span id="cb11-3"><a href="cohort-characteristics.html#cb11-3"></a><span class="kw">hist</span>(Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">!=</span><span class="st">&quot;Characterisation&quot;</span>)], <span class="dt">breaks=</span><span class="dv">15</span>,</span>
<span id="cb11-4"><a href="cohort-characteristics.html#cb11-4"></a>     <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;characterisation sizes Median %s&quot;</span>, X1a), </span>
<span id="cb11-5"><a href="cohort-characteristics.html#cb11-5"></a>     <span class="dt">xlab=</span><span class="st">&quot;size at sac&quot;</span>)</span>
<span id="cb11-6"><a href="cohort-characteristics.html#cb11-6"></a></span>
<span id="cb11-7"><a href="cohort-characteristics.html#cb11-7"></a>X2a=<span class="kw">median</span>(Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>)], <span class="dt">na.rm=</span>T)</span>
<span id="cb11-8"><a href="cohort-characteristics.html#cb11-8"></a><span class="kw">hist</span>(Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>)], <span class="dt">breaks=</span><span class="dv">15</span>,</span>
<span id="cb11-9"><a href="cohort-characteristics.html#cb11-9"></a>     <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;progression sizes Median %s&quot;</span>, X2a),</span>
<span id="cb11-10"><a href="cohort-characteristics.html#cb11-10"></a>     <span class="dt">xlab=</span><span class="st">&quot;size at sac&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>Note that in the characterisation cohort, samples are selected for SAC at different time points:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="cohort-characteristics.html#cb12-1"></a></span>
<span id="cb12-2"><a href="cohort-characteristics.html#cb12-2"></a><span class="co">#pdf(&quot;~/Desktop/S1B_size_oveR_time.pdf&quot;, width=6, height=5)</span></span>
<span id="cb12-3"><a href="cohort-characteristics.html#cb12-3"></a></span>
<span id="cb12-4"><a href="cohort-characteristics.html#cb12-4"></a>am1=<span class="kw">lm</span>(Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)]<span class="op">~</span></span>
<span id="cb12-5"><a href="cohort-characteristics.html#cb12-5"></a><span class="st">         </span>Cdata<span class="op">$</span>Time.NMU2Sac[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)])</span>
<span id="cb12-6"><a href="cohort-characteristics.html#cb12-6"></a></span>
<span id="cb12-7"><a href="cohort-characteristics.html#cb12-7"></a><span class="kw">plot</span>(<span class="kw">jitter</span>(Cdata<span class="op">$</span>Time.NMU2Sac[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)]),</span>
<span id="cb12-8"><a href="cohort-characteristics.html#cb12-8"></a>     <span class="kw">jitter</span>(Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)]),</span>
<span id="cb12-9"><a href="cohort-characteristics.html#cb12-9"></a>     <span class="dt">xlab=</span><span class="st">&quot;Time from NMU to sac (days)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Tumor diameter (mm)&quot;</span>, </span>
<span id="cb12-10"><a href="cohort-characteristics.html#cb12-10"></a>     <span class="dt">col=</span><span class="kw">factor</span>(Cdata<span class="op">$</span>Char.Cohort.Batch[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)]), <span class="dt">pch=</span><span class="dv">19</span>, <span class="dt">main=</span><span class="st">&quot;characterisation: size over time&quot;</span>)</span>
<span id="cb12-11"><a href="cohort-characteristics.html#cb12-11"></a><span class="kw">text</span>(<span class="kw">jitter</span>(Cdata<span class="op">$</span>Time.NMU2Sac[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)])<span class="op">+</span><span class="dv">5</span>,</span>
<span id="cb12-12"><a href="cohort-characteristics.html#cb12-12"></a>     <span class="kw">jitter</span>(Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)])<span class="op">+</span><span class="dv">1</span>,</span>
<span id="cb12-13"><a href="cohort-characteristics.html#cb12-13"></a>     Cdata<span class="op">$</span>NewID[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)], <span class="dt">cex=</span><span class="fl">0.7</span>)</span>
<span id="cb12-14"><a href="cohort-characteristics.html#cb12-14"></a><span class="kw">abline</span>(am1)</span>
<span id="cb12-15"><a href="cohort-characteristics.html#cb12-15"></a><span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="kw">levels</span>(<span class="kw">factor</span>(Cdata<span class="op">$</span>Char.Cohort.Batch)), <span class="dt">col=</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">lwd=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Ext1c"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext1c-1.png" alt="growth of tumors over time" width="672" />
<p class="caption">
Figure 2.1: growth of tumors over time
</p>
</div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="cohort-characteristics.html#cb13-1"></a><span class="co">#dev.off()</span></span>
<span id="cb13-2"><a href="cohort-characteristics.html#cb13-2"></a></span>
<span id="cb13-3"><a href="cohort-characteristics.html#cb13-3"></a>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="kw">cbind</span>(Cdata<span class="op">$</span>Time.NMU2Sac[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)], </span>
<span id="cb13-4"><a href="cohort-characteristics.html#cb13-4"></a>                Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)],</span>
<span id="cb13-5"><a href="cohort-characteristics.html#cb13-5"></a>                Cdata<span class="op">$</span>Char.Cohort.Batch[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)]), <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>, <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>))) </span></code></pre></div>
<div class="figure"><span id="fig:Ext1c"></span>
<div id="htmlwidget-ee5c37be05500fb83b5a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ee5c37be05500fb83b5a">{"x":{"filter":"none","extensions":["Buttons"],"data":[["160","160","160","160","160","160","160","160","160","88","91","54","54","54","54","55","55","55","55","55","55","59","59","59","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87","88","61","197","110","61","87","87","87","87","87","87","87","87","87","87","87","87","87","87","87"],["8","14","5","4","20","8","7","20","6","10","10","4","1.5","3","4","1","6","2","1","2","4","1","5","5","6","35","5","19","8","3","3","7","6","5","2","6","21","10","3","3","25","1","2","2","2","9","10","4","10","12","6","16","13","9","10","17","10","30","30","10","10","20","40","20","20","17"],["penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","cohort-2","cohort-2","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","cohort-2","penetrance-3","cohort-2","penetrance-3","penetrance-3","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n      <th>V2<\/th>\n      <th>V3<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 2.1: growth of tumors over time
</p>
</div>
<p>Plot of number of tumors per rat</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="cohort-characteristics.html#cb14-1"></a>Nx1=<span class="kw">table</span>(Cdata<span class="op">$</span>Rat_ID[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Characterisation&quot;</span>)])</span>
<span id="cb14-2"><a href="cohort-characteristics.html#cb14-2"></a>X2=Cdata<span class="op">$</span>Char.Cohort.Batch[<span class="kw">match</span>(<span class="kw">names</span>(Nx1), Cdata<span class="op">$</span>Rat_ID)]</span>
<span id="cb14-3"><a href="cohort-characteristics.html#cb14-3"></a>t2temp=<span class="kw">data.frame</span>(<span class="dt">count=</span><span class="kw">as.numeric</span>(Nx1), <span class="dt">cohort=</span>X2)</span>
<span id="cb14-4"><a href="cohort-characteristics.html#cb14-4"></a><span class="kw">beeswarm</span>(<span class="kw">jitter</span>(<span class="kw">as.vector</span>(Nx1))<span class="op">~</span>X2, <span class="dt">col=</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">pch=</span><span class="dv">19</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Ext1b"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext1b-1.png" alt="Number of tumors per rat in each cohort" width="672" />
<p class="caption">
Figure 2.2: Number of tumors per rat in each cohort
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="cohort-characteristics.html#cb15-1"></a>DT<span class="op">::</span><span class="kw">datatable</span>(t2temp, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb15-2"><a href="cohort-characteristics.html#cb15-2"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:Ext1b"></span>
<div id="htmlwidget-b895b6b279dc258552d1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b895b6b279dc258552d1">{"x":{"filter":"none","extensions":["Buttons"],"data":[[2,1,2,4,3,2,5,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,8,1,1,1,2,2,2,4,2,1],["latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","latency-4","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","additional-wgs","cohort-2","penetrance-3","cohort-2","penetrance-3","penetrance-3","penetrance-3","cohort-2","cohort-2","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3","penetrance-3"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>count<\/th>\n      <th>cohort<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 2.2: Number of tumors per rat in each cohort
</p>
</div>
</div>
<div id="calculating-growth-rates" class="section level2">
<h2><span class="header-section-number">2.2</span> Calculating growth rates</h2>
<p>In this section, we estimate the growth rates of the samples: Below is a plot of the tumor size per week for each recorded tumor, color-coded according to treatment. Time is measured at the first time point at which a tumor is palpated. Spontaneous large tumors are assumed to have a tumor size of 0 or 1 one week prior to palpating.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="cohort-characteristics.html#cb16-1"></a>GrowthRaw=<span class="kw">read.csv</span>(<span class="st">&quot;../metadata/growth_rates_0915.csv&quot;</span>)</span>
<span id="cb16-2"><a href="cohort-characteristics.html#cb16-2"></a><span class="kw">colnames</span>(GrowthRaw)[<span class="op">-</span><span class="dv">1</span>]=<span class="kw">substr</span>(<span class="kw">colnames</span>(GrowthRaw)[<span class="op">-</span><span class="dv">1</span>], <span class="dv">2</span>, <span class="dv">10</span>)</span>
<span id="cb16-3"><a href="cohort-characteristics.html#cb16-3"></a><span class="kw">colnames</span>(GrowthRaw)=<span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="kw">colnames</span>(GrowthRaw))</span>
<span id="cb16-4"><a href="cohort-characteristics.html#cb16-4"></a></span>
<span id="cb16-5"><a href="cohort-characteristics.html#cb16-5"></a>CTreat=Cdata<span class="op">$</span>Treatment[<span class="kw">match</span>(<span class="kw">colnames</span>(GrowthRaw), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb16-6"><a href="cohort-characteristics.html#cb16-6"></a>CTreat=ColMerge[<span class="kw">match</span>(CTreat, <span class="kw">rownames</span>(ColMerge)) ,<span class="dv">1</span>]</span>
<span id="cb16-7"><a href="cohort-characteristics.html#cb16-7"></a>Cgrowth=Cdata<span class="op">$</span>Tumor.growth.status[<span class="kw">match</span>(<span class="kw">colnames</span>(GrowthRaw), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb16-8"><a href="cohort-characteristics.html#cb16-8"></a></span>
<span id="cb16-9"><a href="cohort-characteristics.html#cb16-9"></a><span class="co">#pdf(sprintf(&quot;rslt/Clinicopath/summary_growth_plots_%s.pdf&quot;, Sys.Date()), width=8, height=8)</span></span>
<span id="cb16-10"><a href="cohort-characteristics.html#cb16-10"></a></span>
<span id="cb16-11"><a href="cohort-characteristics.html#cb16-11"></a><span class="kw">par</span>(<span class="dt">xpd=</span>T)</span>
<span id="cb16-12"><a href="cohort-characteristics.html#cb16-12"></a><span class="kw">plot</span>(<span class="ot">NA</span>, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">27</span>), <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">50</span>), <span class="dt">ylab=</span><span class="st">&quot;Tumor Size (mm)&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (weeks)&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Tumor size over time&quot;</span>)</span>
<span id="cb16-13"><a href="cohort-characteristics.html#cb16-13"></a></span>
<span id="cb16-14"><a href="cohort-characteristics.html#cb16-14"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(GrowthRaw)){</span>
<span id="cb16-15"><a href="cohort-characteristics.html#cb16-15"></a>  x1=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(GrowthRaw[, i]))</span>
<span id="cb16-16"><a href="cohort-characteristics.html#cb16-16"></a>  <span class="kw">lines</span>(GrowthRaw[x1, <span class="dv">1</span>], GrowthRaw[x1, i], <span class="dt">col=</span>CTreat[i])</span>
<span id="cb16-17"><a href="cohort-characteristics.html#cb16-17"></a>}</span>
<span id="cb16-18"><a href="cohort-characteristics.html#cb16-18"></a><span class="kw">legend</span>(<span class="st">&quot;bottom&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.2</span>, <span class="fl">-0.2</span>), <span class="kw">rownames</span>(ColMerge), <span class="dt">col=</span>ColMerge[ ,<span class="dv">1</span>], <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">horiz =</span> T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="cohort-characteristics.html#cb17-1"></a></span>
<span id="cb17-2"><a href="cohort-characteristics.html#cb17-2"></a>ax1=<span class="kw">colnames</span>(GrowthRaw)[<span class="op">-</span><span class="dv">1</span>][<span class="kw">which</span>(CTreat[<span class="op">-</span><span class="dv">1</span>]<span class="op">==</span><span class="st">&quot;Unknown&quot;</span>)]</span>
<span id="cb17-3"><a href="cohort-characteristics.html#cb17-3"></a>bx1=Cdata<span class="op">$</span>TumorID[<span class="op">!</span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID)<span class="op">%in%</span><span class="kw">colnames</span>(GrowthRaw)]</span>
<span id="cb17-4"><a href="cohort-characteristics.html#cb17-4"></a>bx1=bx1[<span class="kw">which</span>(bx1<span class="op">%in%</span>Cdata<span class="op">$</span>TumorID[Cdata<span class="op">$</span>Tumor.growth.status<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;stable&quot;</span>, <span class="st">&quot;growing&quot;</span>)])]</span></code></pre></div>
<p>We can also separate the above plot into separate treatments:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="cohort-characteristics.html#cb18-1"></a><span class="co">#Plot the above into quadrants based on different treatments:</span></span>
<span id="cb18-2"><a href="cohort-characteristics.html#cb18-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb18-3"><a href="cohort-characteristics.html#cb18-3"></a>xn=<span class="kw">levels</span>(<span class="kw">factor</span>(<span class="kw">names</span>(CTreat)))</span>
<span id="cb18-4"><a href="cohort-characteristics.html#cb18-4"></a>x2=CTreat[xn]</span>
<span id="cb18-5"><a href="cohort-characteristics.html#cb18-5"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>){</span>
<span id="cb18-6"><a href="cohort-characteristics.html#cb18-6"></a>  indx=<span class="kw">which</span>(CTreat<span class="op">==</span>x2[j])</span>
<span id="cb18-7"><a href="cohort-characteristics.html#cb18-7"></a>  indx=<span class="kw">setdiff</span>(indx, <span class="dv">1</span>)</span>
<span id="cb18-8"><a href="cohort-characteristics.html#cb18-8"></a>  <span class="kw">plot</span>(<span class="ot">NA</span>, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">27</span>), <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">50</span>), <span class="dt">ylab=</span><span class="st">&quot;Tumor Size (mm)&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (weeks)&quot;</span>, <span class="dt">main=</span>xn[j])</span>
<span id="cb18-9"><a href="cohort-characteristics.html#cb18-9"></a><span class="cf">for</span> (i <span class="cf">in</span> indx){</span>
<span id="cb18-10"><a href="cohort-characteristics.html#cb18-10"></a>  x1=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(GrowthRaw[, i]))</span>
<span id="cb18-11"><a href="cohort-characteristics.html#cb18-11"></a>  <span class="kw">lines</span>(GrowthRaw[x1, <span class="dv">1</span>], GrowthRaw[x1, i], <span class="dt">col=</span>CTreat[i])</span>
<span id="cb18-12"><a href="cohort-characteristics.html#cb18-12"></a>  <span class="kw">text</span>(GrowthRaw[x1[<span class="kw">length</span>(x1)], <span class="dv">1</span>], GrowthRaw[x1[<span class="kw">length</span>(x1)], i], <span class="kw">colnames</span>(GrowthRaw)[i], <span class="dt">cex=</span><span class="fl">0.6</span>)</span>
<span id="cb18-13"><a href="cohort-characteristics.html#cb18-13"></a>}</span>
<span id="cb18-14"><a href="cohort-characteristics.html#cb18-14"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="cohort-characteristics.html#cb19-1"></a></span>
<span id="cb19-2"><a href="cohort-characteristics.html#cb19-2"></a><span class="co">#dev.off()</span></span>
<span id="cb19-3"><a href="cohort-characteristics.html#cb19-3"></a></span>
<span id="cb19-4"><a href="cohort-characteristics.html#cb19-4"></a><span class="kw">colnames</span>(GrowthRaw)=Cdata<span class="op">$</span>NewID[<span class="kw">match</span>(<span class="kw">colnames</span>(GrowthRaw), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb19-5"><a href="cohort-characteristics.html#cb19-5"></a>TreatType=Cdata<span class="op">$</span>Treatment[<span class="kw">match</span>(<span class="kw">colnames</span>(GrowthRaw),Cdata<span class="op">$</span>NewID)]</span>
<span id="cb19-6"><a href="cohort-characteristics.html#cb19-6"></a></span>
<span id="cb19-7"><a href="cohort-characteristics.html#cb19-7"></a>GrowthRaw2=<span class="kw">rbind</span>(TreatType, GrowthRaw)</span>
<span id="cb19-8"><a href="cohort-characteristics.html#cb19-8"></a><span class="kw">colnames</span>(GrowthRaw2)[<span class="dv">1</span>]=<span class="st">&quot;Time(weeks)&quot;</span></span>
<span id="cb19-9"><a href="cohort-characteristics.html#cb19-9"></a></span>
<span id="cb19-10"><a href="cohort-characteristics.html#cb19-10"></a>DT<span class="op">::</span><span class="kw">datatable</span>(GrowthRaw2, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb19-11"><a href="cohort-characteristics.html#cb19-11"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div id="htmlwidget-6d2ca3960f22862af73c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6d2ca3960f22862af73c">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Untreated","0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27"],["Vehicle","4","8","10","18","20","20",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","0","15",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","0","20",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","4","9","24","40","47",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","3","12","20","22","15","14",null,"6",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","1","3","3","3","3","3",null,"0",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","7","14","22","26","30",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","0",null,null,null,"12",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","5","12","20","25","30","32",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","0","4","10","20","32","35",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","5",null,"6",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","0","4",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","15",null,"22",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","2",null,"4","3","2",null,"2","5","10","10",null,"10",null,null,"15","17","15",null,"15",null,null,null,null,null,null,null,null,null],["Vehicle","4",null,"25","35",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","10",null,"22","26",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","10","20",null,"35",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","5","7",null,"16","16","9",null,"4","4",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","7","15","20","28",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","0",null,"22","26","31","35",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","0",null,"28","34","36","40",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","7","11","15",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","6","12","15","15",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","6","15","16","20",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","10","17","22","29","35",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","10","40",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","2","5","3",null,"10",null,null,null,"40",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","2","2","3","3","2","0","0","0","0","0","0",null,"0","0",null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","0",null,null,null,null,null,null,null,null,null,null,null,"22","32",null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","5","10","15","18","20","18",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","0",null,null,null,"35","40",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","0",null,null,null,"6","6",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","2","3","4","3","1","1","0","0","0","0",null,"0","0","0",null,"0","0","0","0","0",null,null,"0","0",null,null,null,null],["LY","3","2","3","1","1","1","1",null,"1","5","7","8","8","10","15",null,"15",null,null,"20","20",null,null,null,null,null,null,null],["LY","3",null,null,"5","8",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","3",null,null,"12","20",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","2","2","5","4","2","1",null,"0","0","0",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","0",null,null,null,"5","7",null,"15","18","28",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","0",null,null,null,null,null,null,null,"5","17",null,"25","30",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","0",null,null,null,null,null,null,null,"12","30",null,"40","40",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","3","6","9","8","7","7",null,"6",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","2","5","20","25","26","27",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","0","12",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","5","10","15","23","30","35",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","0",null,"5","8","8","10",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","0",null,null,"16","25","25",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","6","10","14","18","25","30","35","40",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","2","2","2","0","0","0","0","0","0","0","0",null,null,"0","1","1",null,"2","2","2","3","5",null,"7","10","15",null,"15"],["PDL1","3","2","0","0","7","19","22","25",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","3","5","6","8","12","20","25",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","0","3","4","4","6","11","19",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","2","2","0","0","1","1",null,"3",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","2","3","3","4",null,"6",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","3","3","3",null,"2","2","2","3",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","12",null,"12","15","20","23","25",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","5","8","16",null,"17","21","22","23","25",null,"20",null,null,"20","25",null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","6","10","12","12","8","6","4","3","2","2","2",null,"1","1","1",null,"1","1","1","1","1",null,null,"1","1",null,null,null],["PDL1+LY","5","8","8",null,"10",null,null,"10","10",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","2","2","5","4",null,"2",null,null,"2","2",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","15",null,null,"30","40",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","3","2","1","1","2","2","1","1","1",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","0",null,null,null,null,"20","25","30","35",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","3","4","5","7","7","7","8","6",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","0",null,null,"2","5","4","5","5",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","0",null,null,null,"13","17","20","23",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","3","5","10",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","4","8","17","17","17",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","4","5","4","4","4","3",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","0",null,"5","5","12",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","0",null,null,"12","20","30",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","3","4","5","7",null,"10","10","10",null,"20","24",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","7","10","10",null,"12",null,null,"13",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","4","6","10",null,"10",null,null,"12","15","15",null,"29",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","2","2","4",null,"4",null,null,"9","11","10",null,"12",null,null,null,"15","17",null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","6","6","7",null,"10",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","7","10","15",null,"20",null,null,"30",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","0",null,"20",null,"32",null,null,"42",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1+LY","0",null,null,null,null,null,null,"5",null,"5","7",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>Time(weeks)<\/th>\n      <th>P5T1<\/th>\n      <th>P5T2<\/th>\n      <th>P5T3<\/th>\n      <th>P4T1<\/th>\n      <th>P7T1<\/th>\n      <th>P7T2<\/th>\n      <th>P6T2<\/th>\n      <th>P6T1<\/th>\n      <th>P8T2<\/th>\n      <th>P8T1<\/th>\n      <th>P12T1<\/th>\n      <th>P12T2<\/th>\n      <th>P12T3<\/th>\n      <th>P12T4<\/th>\n      <th>P14T1<\/th>\n      <th>P13T1<\/th>\n      <th>P40T1<\/th>\n      <th>P39T3<\/th>\n      <th>P39T1<\/th>\n      <th>P9T1<\/th>\n      <th>P11T1<\/th>\n      <th>P11T2<\/th>\n      <th>P10T1<\/th>\n      <th>P10T2<\/th>\n      <th>P41T1<\/th>\n      <th>P16T1<\/th>\n      <th>P18T1<\/th>\n      <th>P15T1<\/th>\n      <th>P15T2<\/th>\n      <th>P17T1<\/th>\n      <th>P17T2<\/th>\n      <th>P17T3<\/th>\n      <th>P18T2<\/th>\n      <th>P20T3<\/th>\n      <th>P20T1<\/th>\n      <th>P20T2<\/th>\n      <th>P19T4<\/th>\n      <th>P19T2<\/th>\n      <th>P19T1<\/th>\n      <th>P19T3<\/th>\n      <th>P21T1<\/th>\n      <th>P22T2<\/th>\n      <th>P22T1<\/th>\n      <th>P24T1<\/th>\n      <th>P24T3<\/th>\n      <th>P24T2<\/th>\n      <th>P26T1<\/th>\n      <th>P25T2<\/th>\n      <th>P27T1<\/th>\n      <th>P29T2<\/th>\n      <th>P29T1<\/th>\n      <th>P28T1<\/th>\n      <th>P30T2<\/th>\n      <th>P38T2<\/th>\n      <th>P38T1<\/th>\n      <th>P37T1<\/th>\n      <th>P42T1<\/th>\n      <th>P42T2<\/th>\n      <th>P42T3<\/th>\n      <th>P42T4<\/th>\n      <th>P32T2<\/th>\n      <th>P32T1<\/th>\n      <th>P31T2<\/th>\n      <th>P31T3<\/th>\n      <th>P31T1<\/th>\n      <th>P33T1<\/th>\n      <th>P35T1<\/th>\n      <th>P34T3<\/th>\n      <th>P34T2<\/th>\n      <th>P34T1<\/th>\n      <th>P36T3<\/th>\n      <th>P3T2<\/th>\n      <th>P2T2<\/th>\n      <th>P1T1<\/th>\n      <th>P2T1<\/th>\n      <th>P23T2<\/th>\n      <th>P23T1<\/th>\n      <th>P36T2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Notably, there are a few samples which grow but then regress. These are plotted below</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="cohort-characteristics.html#cb20-1"></a>UturnSamp=<span class="kw">c</span>(<span class="st">&quot;6RB&quot;</span>,<span class="st">&quot;11ND&quot;</span>, <span class="st">&quot;8LD&quot;</span>, <span class="st">&quot;10LD&quot;</span>, <span class="st">&quot;3NB&quot;</span> )</span>
<span id="cb20-2"><a href="cohort-characteristics.html#cb20-2"></a></span>
<span id="cb20-3"><a href="cohort-characteristics.html#cb20-3"></a>tmp=Cdata<span class="op">$</span>NewID[<span class="kw">match</span>(UturnSamp, <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb20-4"><a href="cohort-characteristics.html#cb20-4"></a>mx=<span class="kw">match</span>(tmp, <span class="kw">colnames</span>(GrowthRaw))</span>
<span id="cb20-5"><a href="cohort-characteristics.html#cb20-5"></a></span>
<span id="cb20-6"><a href="cohort-characteristics.html#cb20-6"></a><span class="co">#pdf(&quot;~/Desktop/S5-inflammatory-samples-growth.pdf&quot;, height=5, width = 5)</span></span>
<span id="cb20-7"><a href="cohort-characteristics.html#cb20-7"></a></span>
<span id="cb20-8"><a href="cohort-characteristics.html#cb20-8"></a><span class="kw">plot</span>(<span class="ot">NA</span>, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>), <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">30</span>), <span class="dt">ylab=</span><span class="st">&quot;Tumor Size (mm)&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (weeks)&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Inflammatory  samples&quot;</span>)</span>
<span id="cb20-9"><a href="cohort-characteristics.html#cb20-9"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(mx)){</span>
<span id="cb20-10"><a href="cohort-characteristics.html#cb20-10"></a>    x1=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(GrowthRaw[, mx[i]]))</span>
<span id="cb20-11"><a href="cohort-characteristics.html#cb20-11"></a> <span class="kw">lines</span>(GrowthRaw[x1, <span class="dv">1</span>], GrowthRaw[x1, mx[i]], <span class="dt">col=</span>CTreat[mx[i]])</span>
<span id="cb20-12"><a href="cohort-characteristics.html#cb20-12"></a>  <span class="kw">text</span>(GrowthRaw[x1[<span class="kw">length</span>(x1)], <span class="dv">1</span>], GrowthRaw[x1[<span class="kw">length</span>(x1)], mx[i]], </span>
<span id="cb20-13"><a href="cohort-characteristics.html#cb20-13"></a>       <span class="kw">colnames</span>(GrowthRaw)[mx[i]], <span class="dt">cex=</span><span class="fl">0.6</span>)</span>
<span id="cb20-14"><a href="cohort-characteristics.html#cb20-14"></a></span>
<span id="cb20-15"><a href="cohort-characteristics.html#cb20-15"></a>}</span></code></pre></div>
<div class="figure"><span id="fig:5xta"></span>
<img src="RatDCIS-bookdown_files/figure-html/5xta-1.png" alt="inflammatory growth profiles" width="672" />
<p class="caption">
Figure 2.3: inflammatory growth profiles
</p>
</div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="cohort-characteristics.html#cb21-1"></a></span>
<span id="cb21-2"><a href="cohort-characteristics.html#cb21-2"></a>GrowthRaw3=GrowthRaw2[ ,<span class="kw">c</span>(<span class="dv">1</span>, mx)]</span>
<span id="cb21-3"><a href="cohort-characteristics.html#cb21-3"></a></span>
<span id="cb21-4"><a href="cohort-characteristics.html#cb21-4"></a>DT<span class="op">::</span><span class="kw">datatable</span>(GrowthRaw3, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb21-5"><a href="cohort-characteristics.html#cb21-5"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:5xta"></span>
<div id="htmlwidget-1d71cba84d39d9ba566f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1d71cba84d39d9ba566f">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Untreated","0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27"],["LY","5","10","15","18","20","18",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","2","2","0","0","1","1",null,"3",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["LY","3","6","9","8","7","7",null,"6",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["PDL1","0",null,"5","8","8","10",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Vehicle","3","12","20","22","15","14",null,"6",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>Time(weeks)<\/th>\n      <th>P17T1<\/th>\n      <th>P28T1<\/th>\n      <th>P21T1<\/th>\n      <th>P24T3<\/th>\n      <th>P7T1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 2.3: inflammatory growth profiles
</p>
</div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="cohort-characteristics.html#cb22-1"></a><span class="co">#write.csv(GrowthRaw3, file=&quot;nature-tables/Ext5a.csv&quot;)</span></span></code></pre></div>
<p>These tumors doesn’t have growth rate data: 17N_B.</p>
<p>We can then compute the growth rate for the above samples by considering the change in size over a given period of time using a linear regression model. Below is the histogram of growth rates, and we see a separation at approximately 2mm/week:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="cohort-characteristics.html#cb23-1"></a>GR=<span class="kw">sapply</span>(<span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(GrowthRaw), <span class="cf">function</span>(x) <span class="kw">lm</span>(GrowthRaw[, x]<span class="op">~</span>GrowthRaw[, <span class="dv">1</span>])<span class="op">$</span>coefficients[<span class="dv">2</span>])</span>
<span id="cb23-2"><a href="cohort-characteristics.html#cb23-2"></a><span class="kw">names</span>(GR)=<span class="kw">colnames</span>(GrowthRaw)[<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb23-3"><a href="cohort-characteristics.html#cb23-3"></a></span>
<span id="cb23-4"><a href="cohort-characteristics.html#cb23-4"></a>d1=<span class="kw">data.frame</span>(<span class="dt">growthrate=</span>GR, <span class="dt">treatment=</span><span class="kw">names</span>(CTreat[<span class="op">-</span><span class="dv">1</span>]), <span class="dt">growth=</span>Cgrowth[<span class="op">-</span><span class="dv">1</span>], <span class="dt">color=</span>CTreat[<span class="op">-</span><span class="dv">1</span>],</span>
<span id="cb23-5"><a href="cohort-characteristics.html#cb23-5"></a>              <span class="dt">sample=</span><span class="kw">names</span>(GR))</span>
<span id="cb23-6"><a href="cohort-characteristics.html#cb23-6"></a></span>
<span id="cb23-7"><a href="cohort-characteristics.html#cb23-7"></a>d1<span class="op">$</span>treatment=<span class="kw">factor</span>(d1<span class="op">$</span>treatment, <span class="dt">levels=</span><span class="kw">c</span>( <span class="st">&quot;PDL1+LY&quot;</span>, <span class="st">&quot;PDL1&quot;</span>, <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb23-8"><a href="cohort-characteristics.html#cb23-8"></a></span>
<span id="cb23-9"><a href="cohort-characteristics.html#cb23-9"></a>Nweeks=<span class="kw">sapply</span>(<span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(GrowthRaw), <span class="cf">function</span>(x) <span class="kw">max</span>(<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(GrowthRaw[ , x]))))</span>
<span id="cb23-10"><a href="cohort-characteristics.html#cb23-10"></a><span class="kw">names</span>(Nweeks)=<span class="kw">names</span>(GR)</span>
<span id="cb23-11"><a href="cohort-characteristics.html#cb23-11"></a></span>
<span id="cb23-12"><a href="cohort-characteristics.html#cb23-12"></a>d1<span class="op">$</span>Nweeks=Nweeks[<span class="kw">match</span>(d1<span class="op">$</span>sample, <span class="kw">names</span>(Nweeks))]</span>
<span id="cb23-13"><a href="cohort-characteristics.html#cb23-13"></a>d1<span class="op">$</span>Time.Tum2Sac=Cdata<span class="op">$</span>Time.Tumor2Sac[<span class="kw">match</span>(d1<span class="op">$</span>sample, Cdata<span class="op">$</span>NewID)]</span>
<span id="cb23-14"><a href="cohort-characteristics.html#cb23-14"></a>d1<span class="op">$</span>Time.NMU2Sac=Cdata<span class="op">$</span>Time.NMU2Sac[<span class="kw">match</span>(d1<span class="op">$</span>sample, Cdata<span class="op">$</span>NewID)]</span>
<span id="cb23-15"><a href="cohort-characteristics.html#cb23-15"></a>d1<span class="op">$</span>tum.size=Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">match</span>(d1<span class="op">$</span>sample, Cdata<span class="op">$</span>NewID)]</span>
<span id="cb23-16"><a href="cohort-characteristics.html#cb23-16"></a>d1<span class="op">$</span>growthrate_cutoff2=<span class="kw">ifelse</span>(d1<span class="op">$</span>growthrate<span class="op">&gt;=</span><span class="dv">2</span>, <span class="st">&quot;growing&quot;</span>, <span class="st">&quot;stable&quot;</span>)</span>
<span id="cb23-17"><a href="cohort-characteristics.html#cb23-17"></a></span>
<span id="cb23-18"><a href="cohort-characteristics.html#cb23-18"></a><span class="co">#pdf(sprintf(&quot;rslt/Clinicopath/summary_growth_rates_%s.pdf&quot;, Sys.Date()), width=8, height=6)</span></span>
<span id="cb23-19"><a href="cohort-characteristics.html#cb23-19"></a><span class="kw">ggplot</span>(d1, <span class="kw">aes</span>(<span class="dt">x=</span>growthrate, <span class="dt">fill=</span>treatment))<span class="op">+</span><span class="kw">geom_histogram</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">geom_vline</span>(<span class="kw">aes</span>(<span class="dt">xintercept=</span><span class="dv">2</span>), <span class="dt">colour=</span><span class="st">&quot;grey45&quot;</span>, <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>)<span class="op">+</span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>ColMerge[ ,<span class="dv">1</span>])</span></code></pre></div>
<div class="figure"><span id="fig:3d"></span>
<img src="RatDCIS-bookdown_files/figure-html/3d-1.png" alt="Histogram of growth rates" width="672" />
<p class="caption">
Figure 2.4: Histogram of growth rates
</p>
</div>
<p>Based on the above distribution, a cut-off of 2mm/week may be an optimal cut-off to separate growing and stable tumors. Below are growth rates of tumors under different treatments:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="cohort-characteristics.html#cb24-1"></a>pv1=<span class="kw">sapply</span>(<span class="kw">levels</span>(d1<span class="op">$</span>treatment)[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], <span class="cf">function</span>(x)</span>
<span id="cb24-2"><a href="cohort-characteristics.html#cb24-2"></a>  <span class="kw">wilcox.test</span>(d1<span class="op">$</span>growthrate[<span class="kw">which</span>(d1<span class="op">$</span>treatment<span class="op">==</span>x)], d1<span class="op">$</span>growthrate[<span class="kw">which</span>(d1<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span>)])<span class="op">$</span>p.value)</span>
<span id="cb24-3"><a href="cohort-characteristics.html#cb24-3"></a></span>
<span id="cb24-4"><a href="cohort-characteristics.html#cb24-4"></a>d1<span class="op">$</span>treatment=<span class="kw">factor</span>(d1<span class="op">$</span>treatment, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>, <span class="st">&quot;PDL1&quot;</span>, <span class="st">&quot;LY&quot;</span>,<span class="st">&quot;PDL1+LY&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="cohort-characteristics.html#cb25-1"></a><span class="kw">ggplot</span>(d1, <span class="kw">aes</span>(<span class="dt">x=</span>treatment, <span class="dt">y=</span>growthrate, <span class="dt">col=</span>treatment))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_jitter</span>()<span class="op">+</span></span>
<span id="cb25-2"><a href="cohort-characteristics.html#cb25-2"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>ColMerge[ ,<span class="dv">1</span>])<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="kw">paste</span>(<span class="kw">names</span>(pv1), <span class="kw">round</span>(pv1, <span class="dv">2</span>)), <span class="dt">collapse=</span><span class="st">&quot;;&quot;</span>))</span></code></pre></div>
<div class="figure"><span id="fig:3c"></span>
<img src="RatDCIS-bookdown_files/figure-html/3c-1.png" alt="Growth rate with respect to treatment" width="672" />
<p class="caption">
Figure 2.5: Growth rate with respect to treatment
</p>
</div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="cohort-characteristics.html#cb26-1"></a></span>
<span id="cb26-2"><a href="cohort-characteristics.html#cb26-2"></a>DT<span class="op">::</span><span class="kw">datatable</span>(d1, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb26-3"><a href="cohort-characteristics.html#cb26-3"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:3c"></span>
<div id="htmlwidget-aedc7c7d9192e0f67b56" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-aedc7c7d9192e0f67b56">{"x":{"filter":"none","extensions":["Buttons"],"data":[[3.54285714285714,15,20,11.7,0.0245901639344268,-0.151639344262295,5.8,3,5.54285714285714,7.68571428571428,0.500000000000001,4,3.5,0.929976851851852,10.3571428571429,5.42857142857143,8.21428571428571,-0.384615384615384,6.8,6.87837837837838,7.81081081081081,4,3,4.3,6.2,30,4.775,-0.237471952131638,2.19745222929936,2.8,8.21428571428571,1.28571428571429,-0.123523596441022,0.996699669966997,1.11538461538462,3.96153846153846,-0.404166666666667,2.88785046728972,2.35555555555556,3.44444444444444,0.241803278688525,5.51428571428571,12,6.22857142857142,1.97297297297297,5.35714285714286,4.97619047619048,0.393053320756811,3.78571428571429,3.64285714285714,2.67857142857143,0.10655737704918,0.77027027027027,-0.102739726027397,2.45714285714286,1.07457627118644,-0.404411764705883,0.50625,-0.111111111111111,5.96153846153846,-0.166666666666667,4.3298969072165,0.55952380952381,0.762162162162162,3.31506849315069,3.5,3.5,-0.228571428571429,2.68571428571429,5.71428571428571,1.91517857142857,0.759740259740259,1.67748917748918,0.944033790918691,1.05714285714286,3.25974025974026,5.85046728971962,0.647540983606557],["Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY"],["growing","growing","growing","growing","stable","stable","growing","growing","growing","growing","no data","no data","growing","stable","growing","growing","growing","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","growing","stable","growing","stable","stable","stable","stable","growing","stable","growing","stable","growing","stable","growing","growing","growing","stable","growing","growing","stable","growing","growing","growing","stable","stable","stable","growing","stable","stable","growing","stable","growing","stable","growing","stable","stable","growing","growing","growing","stable","growing","growing","stable","stable","growing","stable","stable","growing","growing","stable"],["#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#5D5D5D","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#FFC82F","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#73FDFE","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF","#FF41FF"],["P5T1","P5T2","P5T3","P4T1","P7T1","P7T2","P6T2","P6T1","P8T2","P8T1","P12T1","P12T2","P12T3","P12T4","P14T1","P13T1","P40T1","P39T3","P39T1","P9T1","P11T1","P11T2","P10T1","P10T2","P41T1","P16T1","P18T1","P15T1","P15T2","P17T1","P17T2","P17T3","P18T2","P20T3","P20T1","P20T2","P19T4","P19T2","P19T1","P19T3","P21T1","P22T2","P22T1","P24T1","P24T3","P24T2","P26T1","P25T2","P27T1","P29T2","P29T1","P28T1","P30T2","P38T2","P38T1","P37T1","P42T1","P42T2","P42T3","P42T4","P32T2","P32T1","P31T2","P31T3","P31T1","P33T1","P35T1","P34T3","P34T2","P34T1","P36T3","P3T2","P2T2","P1T1","P2T1","P23T2","P23T1","P36T2"],[6,2,2,5,8,8,5,5,6,6,3,2,3,19,4,4,4,9,4,6,6,3,4,4,5,2,9,14,14,6,6,6,24,21,5,5,10,10,13,13,8,6,2,6,6,6,8,28,8,7,7,8,6,8,7,15,25,9,10,5,9,9,8,8,8,3,5,6,5,6,11,8,12,17,5,8,8,11],[41,41,41,30,49,49,34,6,37,37,21,7,21,140,22,22,21,0,28,42,42,42,42,42,35,8,118,92,92,36,15,14,188,161,28,28,85,43,43,43,49,38,1,36,36,22,51,224,48,41,41,49,38,56,56,105,195,76,90,55,51,16,29,15,1,15,34,34,20,13,73,72,84,119,35,56,42,31],[84,84,84,77,127,127,84,84,84,84,252,252,252,252,135,135,127,169,169,96,96,96,96,96,163,55,238,135,135,79,79,79,238,232,232,232,163,163,163,163,127,127,127,79,79,79,98,274,112,112,112,127,127,183,183,232,238,238,238,238,98,98,79,79,79,79,112,112,112,112,163,241,253,288,253,238,238,163],[20,15,20,47,6,3,30,12,32,35,6,4,22,15,35,26,35,4,28,35,40,15,15,20,35,40,40,0,32,18,40,6,1,20,8,20,0,31,30,40,6,27,12,35,10,25,40,15,25,25,19,3,6,3,25,25,1,40,2,40,1,35,6,5,23,10,17,3,22,30,24,15,29,17,10,35,45,7],["growing","growing","growing","growing","stable","stable","growing","growing","growing","growing","stable","growing","growing","stable","growing","growing","growing","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","growing","growing","growing","stable","stable","stable","stable","growing","stable","growing","growing","growing","stable","growing","growing","growing","stable","growing","growing","stable","growing","growing","growing","stable","stable","stable","growing","stable","stable","stable","stable","growing","stable","growing","stable","stable","growing","growing","growing","stable","growing","growing","stable","stable","stable","stable","stable","growing","growing","stable"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>growthrate<\/th>\n      <th>treatment<\/th>\n      <th>growth<\/th>\n      <th>color<\/th>\n      <th>sample<\/th>\n      <th>Nweeks<\/th>\n      <th>Time.Tum2Sac<\/th>\n      <th>Time.NMU2Sac<\/th>\n      <th>tum.size<\/th>\n      <th>growthrate_cutoff2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[0,5,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 2.5: Growth rate with respect to treatment
</p>
</div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="cohort-characteristics.html#cb27-1"></a><span class="co">#write.csv(d1, file=&quot;nature-tables/Fig3c-d.csv&quot;)</span></span></code></pre></div>
<p>We can calculate the p.values below, using a wilcox test. The growth rates comparing the treatment to the controls are:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="cohort-characteristics.html#cb28-1"></a><span class="kw">print</span>(<span class="st">&#39;LY samples&#39;</span>)</span>
<span id="cb28-2"><a href="cohort-characteristics.html#cb28-2"></a><span class="co">## [1] &quot;LY samples&quot;</span></span>
<span id="cb28-3"><a href="cohort-characteristics.html#cb28-3"></a><span class="kw">wilcox.test</span>(d1<span class="op">$</span>growthrate[d1<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;LY&quot;</span>], d1<span class="op">$</span>growthrate[d1<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span>])</span>
<span id="cb28-4"><a href="cohort-characteristics.html#cb28-4"></a><span class="co">## </span></span>
<span id="cb28-5"><a href="cohort-characteristics.html#cb28-5"></a><span class="co">##  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb28-6"><a href="cohort-characteristics.html#cb28-6"></a><span class="co">## </span></span>
<span id="cb28-7"><a href="cohort-characteristics.html#cb28-7"></a><span class="co">## data:  d1$growthrate[d1$treatment == &quot;LY&quot;] and d1$growthrate[d1$treatment == &quot;Vehicle&quot;]</span></span>
<span id="cb28-8"><a href="cohort-characteristics.html#cb28-8"></a><span class="co">## W = 150.5, p-value = 0.06847</span></span>
<span id="cb28-9"><a href="cohort-characteristics.html#cb28-9"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb28-10"><a href="cohort-characteristics.html#cb28-10"></a><span class="kw">print</span>(<span class="st">&#39;PDL1 samples&#39;</span>)</span>
<span id="cb28-11"><a href="cohort-characteristics.html#cb28-11"></a><span class="co">## [1] &quot;PDL1 samples&quot;</span></span>
<span id="cb28-12"><a href="cohort-characteristics.html#cb28-12"></a><span class="kw">wilcox.test</span>(d1<span class="op">$</span>growthrate[d1<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;PDL1&quot;</span>], d1<span class="op">$</span>growthrate[d1<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span>])</span>
<span id="cb28-13"><a href="cohort-characteristics.html#cb28-13"></a><span class="co">## </span></span>
<span id="cb28-14"><a href="cohort-characteristics.html#cb28-14"></a><span class="co">##  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb28-15"><a href="cohort-characteristics.html#cb28-15"></a><span class="co">## </span></span>
<span id="cb28-16"><a href="cohort-characteristics.html#cb28-16"></a><span class="co">## data:  d1$growthrate[d1$treatment == &quot;PDL1&quot;] and d1$growthrate[d1$treatment == &quot;Vehicle&quot;]</span></span>
<span id="cb28-17"><a href="cohort-characteristics.html#cb28-17"></a><span class="co">## W = 90, p-value = 0.02673</span></span>
<span id="cb28-18"><a href="cohort-characteristics.html#cb28-18"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb28-19"><a href="cohort-characteristics.html#cb28-19"></a><span class="kw">print</span>(<span class="st">&#39;PDL1+LY samples&#39;</span>)</span>
<span id="cb28-20"><a href="cohort-characteristics.html#cb28-20"></a><span class="co">## [1] &quot;PDL1+LY samples&quot;</span></span>
<span id="cb28-21"><a href="cohort-characteristics.html#cb28-21"></a><span class="kw">wilcox.test</span>(d1<span class="op">$</span>growthrate[d1<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;PDL1+LY&quot;</span>], d1<span class="op">$</span>growthrate[d1<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span>])</span>
<span id="cb28-22"><a href="cohort-characteristics.html#cb28-22"></a><span class="co">## </span></span>
<span id="cb28-23"><a href="cohort-characteristics.html#cb28-23"></a><span class="co">##  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb28-24"><a href="cohort-characteristics.html#cb28-24"></a><span class="co">## </span></span>
<span id="cb28-25"><a href="cohort-characteristics.html#cb28-25"></a><span class="co">## data:  d1$growthrate[d1$treatment == &quot;PDL1+LY&quot;] and d1$growthrate[d1$treatment == &quot;Vehicle&quot;]</span></span>
<span id="cb28-26"><a href="cohort-characteristics.html#cb28-26"></a><span class="co">## W = 133.5, p-value = 0.002644</span></span>
<span id="cb28-27"><a href="cohort-characteristics.html#cb28-27"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span></code></pre></div>
<p>This shows a smaller growth-rate in PDL1 single and double treated cases compared to the vehicles.</p>
<p>Overall the distribution of growing vs stable tumors is shown below:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="cohort-characteristics.html#cb29-1"></a><span class="kw">table</span>(<span class="kw">ifelse</span>(d1<span class="op">$</span>growthrate<span class="op">&gt;=</span><span class="dv">2</span>, <span class="st">&quot;grow&quot;</span>, <span class="st">&quot;stable&quot;</span>))</span>
<span id="cb29-2"><a href="cohort-characteristics.html#cb29-2"></a><span class="co">## </span></span>
<span id="cb29-3"><a href="cohort-characteristics.html#cb29-3"></a><span class="co">##   grow stable </span></span>
<span id="cb29-4"><a href="cohort-characteristics.html#cb29-4"></a><span class="co">##     47     31</span></span>
<span id="cb29-5"><a href="cohort-characteristics.html#cb29-5"></a><span class="kw">head</span>(d1)</span>
<span id="cb29-6"><a href="cohort-characteristics.html#cb29-6"></a><span class="co">##       growthrate treatment  growth   color sample Nweeks Time.Tum2Sac Time.NMU2Sac tum.size growthrate_cutoff2</span></span>
<span id="cb29-7"><a href="cohort-characteristics.html#cb29-7"></a><span class="co">## P5T1  3.54285714   Vehicle growing #5D5D5D   P5T1      6           41           84       20            growing</span></span>
<span id="cb29-8"><a href="cohort-characteristics.html#cb29-8"></a><span class="co">## P5T2 15.00000000   Vehicle growing #5D5D5D   P5T2      2           41           84       15            growing</span></span>
<span id="cb29-9"><a href="cohort-characteristics.html#cb29-9"></a><span class="co">## P5T3 20.00000000   Vehicle growing #5D5D5D   P5T3      2           41           84       20            growing</span></span>
<span id="cb29-10"><a href="cohort-characteristics.html#cb29-10"></a><span class="co">## P4T1 11.70000000   Vehicle growing #5D5D5D   P4T1      5           30           77       47            growing</span></span>
<span id="cb29-11"><a href="cohort-characteristics.html#cb29-11"></a><span class="co">## P7T1  0.02459016   Vehicle  stable #5D5D5D   P7T1      8           49          127        6             stable</span></span>
<span id="cb29-12"><a href="cohort-characteristics.html#cb29-12"></a><span class="co">## P7T2 -0.15163934   Vehicle  stable #5D5D5D   P7T2      8           49          127        3             stable</span></span></code></pre></div>
<p>We can replot the previous graphs according to growth, and color code according to whether it is a fast or slow growing tumor</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="cohort-characteristics.html#cb30-1"></a><span class="kw">par</span>(<span class="dt">xpd=</span>T)</span>
<span id="cb30-2"><a href="cohort-characteristics.html#cb30-2"></a></span>
<span id="cb30-3"><a href="cohort-characteristics.html#cb30-3"></a><span class="co">#Plot the above into quadrants based on different treatments:</span></span>
<span id="cb30-4"><a href="cohort-characteristics.html#cb30-4"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb30-5"><a href="cohort-characteristics.html#cb30-5"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(xn)){</span>
<span id="cb30-6"><a href="cohort-characteristics.html#cb30-6"></a>  indx=<span class="kw">which</span>(<span class="kw">names</span>(CTreat)<span class="op">==</span>xn[j])</span>
<span id="cb30-7"><a href="cohort-characteristics.html#cb30-7"></a>  indx=<span class="kw">setdiff</span>(indx, <span class="dv">1</span>)</span>
<span id="cb30-8"><a href="cohort-characteristics.html#cb30-8"></a>  <span class="kw">plot</span>(<span class="ot">NA</span>, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">27</span>), <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">50</span>), <span class="dt">ylab=</span><span class="st">&quot;Tumor Size (mm)&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (weeks)&quot;</span>, <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;Redone&quot;</span>, xn[j]))</span>
<span id="cb30-9"><a href="cohort-characteristics.html#cb30-9"></a><span class="cf">for</span> (i <span class="cf">in</span> indx){</span>
<span id="cb30-10"><a href="cohort-characteristics.html#cb30-10"></a>  x1=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(GrowthRaw[, i]))</span>
<span id="cb30-11"><a href="cohort-characteristics.html#cb30-11"></a>  <span class="kw">lines</span>(GrowthRaw[x1, <span class="dv">1</span>], GrowthRaw[x1, i], <span class="dt">col=</span><span class="kw">ifelse</span>(d1<span class="op">$</span>growthrate_cutoff2[i<span class="dv">-1</span>]<span class="op">==</span><span class="st">&quot;growing&quot;</span>, ColMerge[j, <span class="dv">1</span>], <span class="kw">ifelse</span>(d1<span class="op">$</span>growthrate_cutoff2[i<span class="dv">-1</span>]<span class="op">==</span><span class="st">&quot;stable&quot;</span>, ColMerge[j, <span class="dv">2</span>], <span class="st">&quot;black&quot;</span>)), <span class="dt">type=</span><span class="st">&quot;o&quot;</span>, <span class="dt">pch=</span><span class="dv">19</span>)</span>
<span id="cb30-12"><a href="cohort-characteristics.html#cb30-12"></a>  <span class="kw">text</span>(GrowthRaw[x1[<span class="kw">length</span>(x1)], <span class="dv">1</span>], GrowthRaw[x1[<span class="kw">length</span>(x1)], i], <span class="kw">colnames</span>(GrowthRaw)[i], <span class="dt">cex=</span><span class="fl">0.6</span>)</span>
<span id="cb30-13"><a href="cohort-characteristics.html#cb30-13"></a>}</span>
<span id="cb30-14"><a href="cohort-characteristics.html#cb30-14"></a>  </span>
<span id="cb30-15"><a href="cohort-characteristics.html#cb30-15"></a>}</span></code></pre></div>
<div class="figure"><span id="fig:3b"></span>
<img src="RatDCIS-bookdown_files/figure-html/3b-1.png" alt="Overview of growth rates of immunotherapy treated tumors with fast and slow growing indicated" width="672" />
<p class="caption">
Figure 2.6: Overview of growth rates of immunotherapy treated tumors with fast and slow growing indicated
</p>
</div>
<p>As a sanity check, compare these growth rates with differences in tumour size at different time points:</p>
<ul>
<li>comparing the growth rate according to classifications (growing, stable)</li>
<li>tumor size at time of sacrifice</li>
<li>rate of tumor development from the time of NMU injection</li>
</ul>
<p>For all comparisons, wilcox rank sum test to assess differences was used</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="cohort-characteristics.html#cb31-1"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb31-2"><a href="cohort-characteristics.html#cb31-2"></a><span class="kw">boxplot</span>(d1<span class="op">$</span>growthrate<span class="op">~</span>d1<span class="op">$</span>growthrate_cutoff2, <span class="dt">main=</span><span class="st">&quot;growth rate, new growth&quot;</span>, <span class="dt">las=</span><span class="dv">2</span>,  <span class="dt">ylab=</span><span class="st">&quot;tumor growth rate (mm/week)&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb31-3"><a href="cohort-characteristics.html#cb31-3"></a>x1=<span class="kw">wilcox.test</span>(d1<span class="op">$</span>growthrate<span class="op">~</span>d1<span class="op">$</span>growthrate_cutoff2)<span class="op">$</span>p.value</span>
<span id="cb31-4"><a href="cohort-characteristics.html#cb31-4"></a><span class="co">## Warning in wilcox.test.default(x = c(3.54285714285714, 15, 20, 11.7, 5.8, : cannot compute exact p-value with ties</span></span>
<span id="cb31-5"><a href="cohort-characteristics.html#cb31-5"></a><span class="kw">text</span>(<span class="fl">1.5</span>, <span class="dv">20</span>, <span class="kw">paste</span>(<span class="st">&quot;wilcox p =&quot;</span>, <span class="kw">round</span>(x1, <span class="dv">3</span>)))</span>
<span id="cb31-6"><a href="cohort-characteristics.html#cb31-6"></a></span>
<span id="cb31-7"><a href="cohort-characteristics.html#cb31-7"></a><span class="kw">boxplot</span>(d1<span class="op">$</span>tum.size<span class="op">~</span>d1<span class="op">$</span>growthrate_cutoff2, <span class="dt">main=</span><span class="st">&quot;tum size at sac, new growth&quot;</span>, <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">ylab=</span><span class="st">&quot;Tumor diameter (mm)&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb31-8"><a href="cohort-characteristics.html#cb31-8"></a>x1=<span class="kw">wilcox.test</span>(d1<span class="op">$</span>tum.size<span class="op">~</span>d1<span class="op">$</span>growthrate_cutoff2)<span class="op">$</span>p.value</span>
<span id="cb31-9"><a href="cohort-characteristics.html#cb31-9"></a><span class="co">## Warning in wilcox.test.default(x = c(20, 15, 20, 47, 30, 12, 32, 35, 4, : cannot compute exact p-value with ties</span></span>
<span id="cb31-10"><a href="cohort-characteristics.html#cb31-10"></a><span class="kw">text</span>(<span class="fl">1.5</span>, <span class="dv">40</span>, <span class="kw">paste</span>(<span class="st">&quot;wilcox p =&quot;</span>, <span class="kw">round</span>(x1, <span class="dv">3</span>)))</span>
<span id="cb31-11"><a href="cohort-characteristics.html#cb31-11"></a></span>
<span id="cb31-12"><a href="cohort-characteristics.html#cb31-12"></a><span class="kw">boxplot</span>(d1<span class="op">$</span>tum.size<span class="op">/</span>d1<span class="op">$</span>Time.NMU2Sac<span class="op">~</span>d1<span class="op">$</span>growthrate_cutoff2, <span class="dt">main=</span><span class="st">&quot;rate NMU2SAC&quot;</span>, <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">ylab=</span><span class="st">&quot;growth rate from NMU injection (mm/wk)&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb31-13"><a href="cohort-characteristics.html#cb31-13"></a>x1=<span class="kw">wilcox.test</span>(d1<span class="op">$</span>tum.size<span class="op">/</span>d1<span class="op">$</span>Time.NMU2Sac<span class="op">~</span>d1<span class="op">$</span>growthrate_cutoff2)<span class="op">$</span>p.value</span>
<span id="cb31-14"><a href="cohort-characteristics.html#cb31-14"></a><span class="co">## Warning in wilcox.test.default(x = c(0.238095238095238, 0.178571428571429, : cannot compute exact p-value with ties</span></span>
<span id="cb31-15"><a href="cohort-characteristics.html#cb31-15"></a><span class="kw">text</span>(<span class="fl">1.5</span>, <span class="fl">0.6</span>, <span class="kw">paste</span>(<span class="st">&quot;wilcox p =&quot;</span>, <span class="kw">round</span>(x1, <span class="dv">3</span>)))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p>Is there an association with treatment? Calculate below using chi-squared test:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="cohort-characteristics.html#cb32-1"></a><span class="co"># print(&#39;new data outcome&#39;)</span></span>
<span id="cb32-2"><a href="cohort-characteristics.html#cb32-2"></a> a3=<span class="kw">chisq.test</span>(<span class="kw">table</span>(<span class="kw">factor</span>(d1<span class="op">$</span>treatment), d1<span class="op">$</span>growthrate_cutoff2))</span>
<span id="cb32-3"><a href="cohort-characteristics.html#cb32-3"></a>a3</span>
<span id="cb32-4"><a href="cohort-characteristics.html#cb32-4"></a><span class="co">## </span></span>
<span id="cb32-5"><a href="cohort-characteristics.html#cb32-5"></a><span class="co">##  Pearson&#39;s Chi-squared test</span></span>
<span id="cb32-6"><a href="cohort-characteristics.html#cb32-6"></a><span class="co">## </span></span>
<span id="cb32-7"><a href="cohort-characteristics.html#cb32-7"></a><span class="co">## data:  table(factor(d1$treatment), d1$growthrate_cutoff2)</span></span>
<span id="cb32-8"><a href="cohort-characteristics.html#cb32-8"></a><span class="co">## X-squared = 7.7366, df = 3, p-value = 0.05178</span></span>
<span id="cb32-9"><a href="cohort-characteristics.html#cb32-9"></a> </span>
<span id="cb32-10"><a href="cohort-characteristics.html#cb32-10"></a> <span class="kw">ContTable</span>(<span class="kw">table</span>((d1<span class="op">$</span>treatment), d1<span class="op">$</span>growthrate_cutoff2), <span class="st">&quot;new rates&quot;</span>, T, <span class="st">&quot;growth&quot;</span>, <span class="st">&quot;treatment&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:3e"></span>
<img src="RatDCIS-bookdown_files/figure-html/3e-1.png" alt="Contingency tables" width="672" />
<p class="caption">
Figure 2.7: Contingency tables
</p>
</div>
<p>We can also compare these values by assessing any I/O vs the control, as well as specific treatment arms</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="cohort-characteristics.html#cb33-1"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb33-2"><a href="cohort-characteristics.html#cb33-2"></a>  </span>
<span id="cb33-3"><a href="cohort-characteristics.html#cb33-3"></a><span class="kw">ContTable</span>(<span class="kw">table</span>((d1<span class="op">$</span>treatment<span class="op">!=</span><span class="st">&quot;Vehicle&quot;</span>), d1<span class="op">$</span>growthrate_cutoff2), <span class="st">&quot;new rates&quot;</span>, T, <span class="st">&quot;growth&quot;</span>, <span class="st">&quot;treatment&quot;</span>)</span>
<span id="cb33-4"><a href="cohort-characteristics.html#cb33-4"></a></span>
<span id="cb33-5"><a href="cohort-characteristics.html#cb33-5"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">levels</span>(d1<span class="op">$</span>treatment)[<span class="dv">2</span><span class="op">:</span><span class="dv">4</span>]){</span>
<span id="cb33-6"><a href="cohort-characteristics.html#cb33-6"></a>  <span class="kw">ContTable</span>(<span class="kw">table</span>(<span class="kw">factor</span>(d1<span class="op">$</span>treatment[d1<span class="op">$</span>treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>, i)]), </span>
<span id="cb33-7"><a href="cohort-characteristics.html#cb33-7"></a>                  d1<span class="op">$</span>growthrate_cutoff2[d1<span class="op">$</span>treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>, i)]), <span class="st">&quot;new rates&quot;</span>, T, <span class="st">&quot;growth&quot;</span>, <span class="st">&quot;treatment&quot;</span>)</span>
<span id="cb33-8"><a href="cohort-characteristics.html#cb33-8"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>Overall, it appears that there is an association between growth rate and treatment</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="cohort-characteristics.html#cb34-1"></a><span class="co"># Replace the Cdata information with new growth information</span></span>
<span id="cb34-2"><a href="cohort-characteristics.html#cb34-2"></a>Cdata<span class="op">$</span>Growth2=d1<span class="op">$</span>growthrate_cutoff2[<span class="kw">match</span>(Cdata<span class="op">$</span>NewID, d1<span class="op">$</span>sample)]</span>
<span id="cb34-3"><a href="cohort-characteristics.html#cb34-3"></a>Cdata<span class="op">$</span>GrowthRate=d1<span class="op">$</span>growthrate[<span class="kw">match</span>(Cdata<span class="op">$</span>NewID, d1<span class="op">$</span>sample)]</span></code></pre></div>
</div>
<div id="km-curves-for-survival" class="section level2">
<h2><span class="header-section-number">2.3</span> KM curves for survival</h2>
<p>Rats were given a treatment once tumors were first palpated. Below, we report a cox proportional model for time from which tumors were first detectd to date of sacrifice. Since subtype information is not available for all samples (only the ones that we have managed to profile, exclude this data)</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="cohort-characteristics.html#cb35-1"></a>Pdat=Cdata[<span class="kw">which</span>(Cdata<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>), ]</span>
<span id="cb35-2"><a href="cohort-characteristics.html#cb35-2"></a>ev=<span class="kw">rep</span>(<span class="dv">1</span>,<span class="kw">nrow</span>(Pdat) )</span>
<span id="cb35-3"><a href="cohort-characteristics.html#cb35-3"></a>ev[<span class="kw">which</span>(Pdat<span class="op">$</span>Time.Tumor2Sac.days<span class="op">&gt;</span><span class="dv">110</span>)]=<span class="dv">0</span></span>
<span id="cb35-4"><a href="cohort-characteristics.html#cb35-4"></a>days2=<span class="kw">ifelse</span>(Pdat<span class="op">$</span>Time.Tumor2Sac.days<span class="op">&gt;</span><span class="dv">100</span>, <span class="dv">110</span>, Pdat<span class="op">$</span>Time.Tumor2Sac.days)</span>
<span id="cb35-5"><a href="cohort-characteristics.html#cb35-5"></a>ssurv=<span class="kw">Surv</span>(days2, ev)</span>
<span id="cb35-6"><a href="cohort-characteristics.html#cb35-6"></a>Pdat<span class="op">$</span>Treatment=<span class="kw">factor</span>(Pdat<span class="op">$</span>Treatment, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>,  <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;PDL1&quot;</span>,<span class="st">&quot;PDL1+LY&quot;</span>))</span>
<span id="cb35-7"><a href="cohort-characteristics.html#cb35-7"></a>ax=<span class="kw">coxph</span>(ssurv<span class="op">~</span>Pdat<span class="op">$</span>Treatment)</span>
<span id="cb35-8"><a href="cohort-characteristics.html#cb35-8"></a><span class="kw">summary</span>(ax)</span>
<span id="cb35-9"><a href="cohort-characteristics.html#cb35-9"></a><span class="co">## Call:</span></span>
<span id="cb35-10"><a href="cohort-characteristics.html#cb35-10"></a><span class="co">## coxph(formula = ssurv ~ Pdat$Treatment)</span></span>
<span id="cb35-11"><a href="cohort-characteristics.html#cb35-11"></a><span class="co">## </span></span>
<span id="cb35-12"><a href="cohort-characteristics.html#cb35-12"></a><span class="co">##   n= 84, number of events= 77 </span></span>
<span id="cb35-13"><a href="cohort-characteristics.html#cb35-13"></a><span class="co">## </span></span>
<span id="cb35-14"><a href="cohort-characteristics.html#cb35-14"></a><span class="co">##                          coef exp(coef) se(coef)      z Pr(&gt;|z|)  </span></span>
<span id="cb35-15"><a href="cohort-characteristics.html#cb35-15"></a><span class="co">## Pdat$TreatmentLY      -0.7103    0.4915   0.3364 -2.112   0.0347 *</span></span>
<span id="cb35-16"><a href="cohort-characteristics.html#cb35-16"></a><span class="co">## Pdat$TreatmentPDL1    -0.4506    0.6373   0.3237 -1.392   0.1639  </span></span>
<span id="cb35-17"><a href="cohort-characteristics.html#cb35-17"></a><span class="co">## Pdat$TreatmentPDL1+LY -0.4650    0.6281   0.3045 -1.527   0.1267  </span></span>
<span id="cb35-18"><a href="cohort-characteristics.html#cb35-18"></a><span class="co">## ---</span></span>
<span id="cb35-19"><a href="cohort-characteristics.html#cb35-19"></a><span class="co">## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb35-20"><a href="cohort-characteristics.html#cb35-20"></a><span class="co">## </span></span>
<span id="cb35-21"><a href="cohort-characteristics.html#cb35-21"></a><span class="co">##                       exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb35-22"><a href="cohort-characteristics.html#cb35-22"></a><span class="co">## Pdat$TreatmentLY         0.4915      2.035    0.2542    0.9503</span></span>
<span id="cb35-23"><a href="cohort-characteristics.html#cb35-23"></a><span class="co">## Pdat$TreatmentPDL1       0.6373      1.569    0.3379    1.2017</span></span>
<span id="cb35-24"><a href="cohort-characteristics.html#cb35-24"></a><span class="co">## Pdat$TreatmentPDL1+LY    0.6281      1.592    0.3458    1.1409</span></span>
<span id="cb35-25"><a href="cohort-characteristics.html#cb35-25"></a><span class="co">## </span></span>
<span id="cb35-26"><a href="cohort-characteristics.html#cb35-26"></a><span class="co">## Concordance= 0.55  (se = 0.037 )</span></span>
<span id="cb35-27"><a href="cohort-characteristics.html#cb35-27"></a><span class="co">## Likelihood ratio test= 4.93  on 3 df,   p=0.2</span></span>
<span id="cb35-28"><a href="cohort-characteristics.html#cb35-28"></a><span class="co">## Wald test            = 5.07  on 3 df,   p=0.2</span></span>
<span id="cb35-29"><a href="cohort-characteristics.html#cb35-29"></a><span class="co">## Score (logrank) test = 5.19  on 3 df,   p=0.2</span></span>
<span id="cb35-30"><a href="cohort-characteristics.html#cb35-30"></a><span class="kw">plot</span>(<span class="kw">survfit</span>(ssurv<span class="op">~</span>Pdat<span class="op">$</span>Treatment), <span class="dt">col=</span>ColMerge[<span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>) ,<span class="dv">1</span>], <span class="dt">ylab=</span><span class="st">&quot;overall survival&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;time (days)&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb35-31"><a href="cohort-characteristics.html#cb35-31"></a><span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;0.49(0.25-0.95)&quot;</span>, <span class="st">&quot;0.64(0.33-1.21)&quot;</span>, <span class="st">&quot;0.63(0.35-1.14)&quot;</span>), <span class="dt">lwd=</span><span class="dv">1</span> , <span class="dt">col=</span>ColMerge[ <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">1</span>])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p>Can also use the growth rate data to evaluate the time point at which a tumor increases by 30% in size or more</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="cohort-characteristics.html#cb36-1"></a>cutoff=GrowthRaw[<span class="dv">1</span>, ]<span class="op">*</span><span class="fl">1.3</span></span>
<span id="cb36-2"><a href="cohort-characteristics.html#cb36-2"></a>Stime=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(cutoff), <span class="cf">function</span>(x) <span class="kw">which</span>(<span class="kw">as.numeric</span>(GrowthRaw[ ,x])<span class="op">&gt;</span><span class="kw">as.numeric</span>(cutoff[x]))[<span class="dv">1</span>])</span>
<span id="cb36-3"><a href="cohort-characteristics.html#cb36-3"></a>Sevent=<span class="kw">ifelse</span>(Stime<span class="op">&lt;</span><span class="dv">15</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb36-4"><a href="cohort-characteristics.html#cb36-4"></a>Sevent[<span class="kw">which</span>(<span class="kw">is.na</span>(Stime))]=<span class="dv">0</span></span>
<span id="cb36-5"><a href="cohort-characteristics.html#cb36-5"></a>Stime[<span class="kw">which</span>(<span class="kw">is.na</span>(Stime))]=<span class="dv">15</span></span>
<span id="cb36-6"><a href="cohort-characteristics.html#cb36-6"></a>Stime[<span class="kw">which</span>(Stime<span class="op">&gt;</span><span class="dv">15</span>)]=<span class="dv">15</span></span>
<span id="cb36-7"><a href="cohort-characteristics.html#cb36-7"></a></span>
<span id="cb36-8"><a href="cohort-characteristics.html#cb36-8"></a>midx=<span class="kw">match</span>(<span class="kw">colnames</span>(GrowthRaw), Pdat<span class="op">$</span>NewID)</span>
<span id="cb36-9"><a href="cohort-characteristics.html#cb36-9"></a></span>
<span id="cb36-10"><a href="cohort-characteristics.html#cb36-10"></a>ss=<span class="kw">Surv</span>(Stime[<span class="op">-</span><span class="dv">1</span>]<span class="op">*</span><span class="dv">7</span>, Sevent[<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb36-11"><a href="cohort-characteristics.html#cb36-11"></a><span class="kw">summary</span>(ax)</span>
<span id="cb36-12"><a href="cohort-characteristics.html#cb36-12"></a><span class="co">## Call:</span></span>
<span id="cb36-13"><a href="cohort-characteristics.html#cb36-13"></a><span class="co">## coxph(formula = ssurv ~ Pdat$Treatment)</span></span>
<span id="cb36-14"><a href="cohort-characteristics.html#cb36-14"></a><span class="co">## </span></span>
<span id="cb36-15"><a href="cohort-characteristics.html#cb36-15"></a><span class="co">##   n= 84, number of events= 77 </span></span>
<span id="cb36-16"><a href="cohort-characteristics.html#cb36-16"></a><span class="co">## </span></span>
<span id="cb36-17"><a href="cohort-characteristics.html#cb36-17"></a><span class="co">##                          coef exp(coef) se(coef)      z Pr(&gt;|z|)  </span></span>
<span id="cb36-18"><a href="cohort-characteristics.html#cb36-18"></a><span class="co">## Pdat$TreatmentLY      -0.7103    0.4915   0.3364 -2.112   0.0347 *</span></span>
<span id="cb36-19"><a href="cohort-characteristics.html#cb36-19"></a><span class="co">## Pdat$TreatmentPDL1    -0.4506    0.6373   0.3237 -1.392   0.1639  </span></span>
<span id="cb36-20"><a href="cohort-characteristics.html#cb36-20"></a><span class="co">## Pdat$TreatmentPDL1+LY -0.4650    0.6281   0.3045 -1.527   0.1267  </span></span>
<span id="cb36-21"><a href="cohort-characteristics.html#cb36-21"></a><span class="co">## ---</span></span>
<span id="cb36-22"><a href="cohort-characteristics.html#cb36-22"></a><span class="co">## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb36-23"><a href="cohort-characteristics.html#cb36-23"></a><span class="co">## </span></span>
<span id="cb36-24"><a href="cohort-characteristics.html#cb36-24"></a><span class="co">##                       exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb36-25"><a href="cohort-characteristics.html#cb36-25"></a><span class="co">## Pdat$TreatmentLY         0.4915      2.035    0.2542    0.9503</span></span>
<span id="cb36-26"><a href="cohort-characteristics.html#cb36-26"></a><span class="co">## Pdat$TreatmentPDL1       0.6373      1.569    0.3379    1.2017</span></span>
<span id="cb36-27"><a href="cohort-characteristics.html#cb36-27"></a><span class="co">## Pdat$TreatmentPDL1+LY    0.6281      1.592    0.3458    1.1409</span></span>
<span id="cb36-28"><a href="cohort-characteristics.html#cb36-28"></a><span class="co">## </span></span>
<span id="cb36-29"><a href="cohort-characteristics.html#cb36-29"></a><span class="co">## Concordance= 0.55  (se = 0.037 )</span></span>
<span id="cb36-30"><a href="cohort-characteristics.html#cb36-30"></a><span class="co">## Likelihood ratio test= 4.93  on 3 df,   p=0.2</span></span>
<span id="cb36-31"><a href="cohort-characteristics.html#cb36-31"></a><span class="co">## Wald test            = 5.07  on 3 df,   p=0.2</span></span>
<span id="cb36-32"><a href="cohort-characteristics.html#cb36-32"></a><span class="co">## Score (logrank) test = 5.19  on 3 df,   p=0.2</span></span>
<span id="cb36-33"><a href="cohort-characteristics.html#cb36-33"></a>ax=<span class="kw">coxph</span>(ss<span class="op">~</span>Pdat<span class="op">$</span>Treatment[<span class="kw">na.omit</span>(midx)])</span>
<span id="cb36-34"><a href="cohort-characteristics.html#cb36-34"></a><span class="kw">plot</span>(<span class="kw">survfit</span>(ss<span class="op">~</span>Pdat<span class="op">$</span>Treatment[<span class="kw">na.omit</span>(midx)]), <span class="dt">col=</span>ColMerge[<span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>) ,<span class="dv">1</span>], <span class="dt">ylab=</span><span class="st">&quot;DFS&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;time (days)&quot;</span>, <span class="dt">mark.time =</span> T)</span>
<span id="cb36-35"><a href="cohort-characteristics.html#cb36-35"></a><span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;0.51(0.27-0.95)&quot;</span>, <span class="st">&quot;0.45(0.22-0.93)&quot;</span>, <span class="st">&quot;0.53(0.29-0.97)&quot;</span>), <span class="dt">lwd=</span><span class="dv">1</span> , <span class="dt">col=</span>ColMerge[ <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">1</span>])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="facs-data-dncd45epcam" class="section level2">
<h2><span class="header-section-number">2.4</span> FACS data (DN/CD45/EpCAM)</h2>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="cohort-characteristics.html#cb37-1"></a>Ltab1=Cdata[ ,<span class="kw">c</span>(<span class="st">&quot;Tumor.Growth&quot;</span>, <span class="st">&quot;DN.Frac.FACS&quot;</span>, <span class="st">&quot;CD45.Frac.FACS&quot;</span>, <span class="st">&quot;EpCAM.Frac.FACS&quot;</span>, <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;GrowthRate&quot;</span>)]</span>
<span id="cb37-2"><a href="cohort-characteristics.html#cb37-2"></a>Ltab2=<span class="kw">melt</span>(Ltab1, <span class="dt">measure.vars =</span> <span class="kw">c</span>(<span class="st">&quot;DN.Frac.FACS&quot;</span>, <span class="st">&quot;CD45.Frac.FACS&quot;</span>, <span class="st">&quot;EpCAM.Frac.FACS&quot;</span>))</span>
<span id="cb37-3"><a href="cohort-characteristics.html#cb37-3"></a>Ltab2=Ltab2[<span class="op">-</span><span class="kw">which</span>(<span class="kw">is.na</span>(Ltab2<span class="op">$</span>value)<span class="op">|</span><span class="kw">is.na</span>(Ltab2<span class="op">$</span>Tumor.Growth)), ]</span>
<span id="cb37-4"><a href="cohort-characteristics.html#cb37-4"></a>Ltab2<span class="op">$</span>Tumor.Growth=<span class="kw">factor</span>(Ltab2<span class="op">$</span>Tumor.Growth)</span>
<span id="cb37-5"><a href="cohort-characteristics.html#cb37-5"></a>Ltab2<span class="op">$</span>Treatment=<span class="kw">factor</span>(Ltab2<span class="op">$</span>Treatment)</span>
<span id="cb37-6"><a href="cohort-characteristics.html#cb37-6"></a>Ltab2<span class="op">$</span>value=<span class="kw">as.numeric</span>(Ltab2<span class="op">$</span>value)</span>
<span id="cb37-7"><a href="cohort-characteristics.html#cb37-7"></a></span>
<span id="cb37-8"><a href="cohort-characteristics.html#cb37-8"></a><span class="kw">ggplot</span>(Ltab2, <span class="kw">aes</span>(<span class="dt">x=</span>Tumor.Growth, <span class="dt">y=</span>value, <span class="dt">col=</span>Tumor.Growth))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>variable)<span class="op">+</span></span>
<span id="cb37-9"><a href="cohort-characteristics.html#cb37-9"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(ColSizeb, <span class="st">&quot;black&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()</span></code></pre></div>
<div class="figure"><span id="fig:Ext3c"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext3c-1.png" alt="Ext3c: FACS data" width="672" />
<p class="caption">
Figure 2.8: Ext3c: FACS data
</p>
</div>
<p>Assess any significance below:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="cohort-characteristics.html#cb38-1"></a><span class="kw">wilcox.test</span>(<span class="kw">as.numeric</span>(Ltab1<span class="op">$</span>CD45.Frac.FACS)<span class="op">~</span>Ltab1<span class="op">$</span>Tumor.Growth)</span>
<span id="cb38-2"><a href="cohort-characteristics.html#cb38-2"></a><span class="co">## </span></span>
<span id="cb38-3"><a href="cohort-characteristics.html#cb38-3"></a><span class="co">##  Wilcoxon rank sum exact test</span></span>
<span id="cb38-4"><a href="cohort-characteristics.html#cb38-4"></a><span class="co">## </span></span>
<span id="cb38-5"><a href="cohort-characteristics.html#cb38-5"></a><span class="co">## data:  as.numeric(Ltab1$CD45.Frac.FACS) by Ltab1$Tumor.Growth</span></span>
<span id="cb38-6"><a href="cohort-characteristics.html#cb38-6"></a><span class="co">## W = 144, p-value = 0.04588</span></span>
<span id="cb38-7"><a href="cohort-characteristics.html#cb38-7"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb38-8"><a href="cohort-characteristics.html#cb38-8"></a><span class="kw">wilcox.test</span>(<span class="kw">as.numeric</span>(Ltab1<span class="op">$</span>DN.Frac.FACS)<span class="op">~</span>Ltab1<span class="op">$</span>Tumor.Growth)</span>
<span id="cb38-9"><a href="cohort-characteristics.html#cb38-9"></a><span class="co">## </span></span>
<span id="cb38-10"><a href="cohort-characteristics.html#cb38-10"></a><span class="co">##  Wilcoxon rank sum exact test</span></span>
<span id="cb38-11"><a href="cohort-characteristics.html#cb38-11"></a><span class="co">## </span></span>
<span id="cb38-12"><a href="cohort-characteristics.html#cb38-12"></a><span class="co">## data:  as.numeric(Ltab1$DN.Frac.FACS) by Ltab1$Tumor.Growth</span></span>
<span id="cb38-13"><a href="cohort-characteristics.html#cb38-13"></a><span class="co">## W = 242, p-value = 0.8695</span></span>
<span id="cb38-14"><a href="cohort-characteristics.html#cb38-14"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb38-15"><a href="cohort-characteristics.html#cb38-15"></a><span class="kw">wilcox.test</span>(<span class="kw">as.numeric</span>(Ltab1<span class="op">$</span>EpCAM.Frac.FACS)<span class="op">~</span>Ltab1<span class="op">$</span>Tumor.Growth)</span>
<span id="cb38-16"><a href="cohort-characteristics.html#cb38-16"></a><span class="co">## </span></span>
<span id="cb38-17"><a href="cohort-characteristics.html#cb38-17"></a><span class="co">##  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb38-18"><a href="cohort-characteristics.html#cb38-18"></a><span class="co">## </span></span>
<span id="cb38-19"><a href="cohort-characteristics.html#cb38-19"></a><span class="co">## data:  as.numeric(Ltab1$EpCAM.Frac.FACS) by Ltab1$Tumor.Growth</span></span>
<span id="cb38-20"><a href="cohort-characteristics.html#cb38-20"></a><span class="co">## W = 307, p-value = 0.1074</span></span>
<span id="cb38-21"><a href="cohort-characteristics.html#cb38-21"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span></code></pre></div>
<p>Also check if it correlates with raw growth rates:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="cohort-characteristics.html#cb39-1"></a>corVals=<span class="kw">sapply</span>(<span class="dv">2</span><span class="op">:</span><span class="dv">4</span>, <span class="cf">function</span>(x) <span class="kw">cor.test</span>(Ltab1<span class="op">$</span>GrowthRate, <span class="kw">as.numeric</span>(Ltab1[ ,x]), <span class="dt">method =</span> <span class="st">&quot;spearman&quot;</span>))</span>
<span id="cb39-2"><a href="cohort-characteristics.html#cb39-2"></a></span>
<span id="cb39-3"><a href="cohort-characteristics.html#cb39-3"></a><span class="kw">ggplot</span>(Ltab2, <span class="kw">aes</span>(<span class="dt">x=</span>value, <span class="dt">y=</span>GrowthRate))<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>variable)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">geom_smooth</span>(<span class="dt">method=</span>lm)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="cohort-characteristics.html#cb40-1"></a></span>
<span id="cb40-2"><a href="cohort-characteristics.html#cb40-2"></a>corVals[<span class="dv">3</span><span class="op">:</span><span class="dv">4</span>, ]</span>
<span id="cb40-3"><a href="cohort-characteristics.html#cb40-3"></a><span class="co">##          [,1]        [,2]      [,3]     </span></span>
<span id="cb40-4"><a href="cohort-characteristics.html#cb40-4"></a><span class="co">## p.value  0.7841102   0.1692151 0.2583497</span></span>
<span id="cb40-5"><a href="cohort-characteristics.html#cb40-5"></a><span class="co">## estimate -0.03932305 -0.195484 0.1612327</span></span></code></pre></div>
<p>We can also plot by treatment:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="cohort-characteristics.html#cb41-1"></a><span class="kw">ggplot</span>(Ltab2, <span class="kw">aes</span>(<span class="dt">x=</span>Treatment, <span class="dt">y=</span>value, <span class="dt">col=</span>Treatment))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>variable)<span class="op">+</span></span>
<span id="cb41-2"><a href="cohort-characteristics.html#cb41-2"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(ColMerge[ ,<span class="dv">1</span>]))<span class="op">+</span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/Ext3c-treat-1.png" width="672" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="cohort-characteristics.html#cb42-1"></a>DT<span class="op">::</span><span class="kw">datatable</span>(Ltab2, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb42-2"><a href="cohort-characteristics.html#cb42-2"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div id="htmlwidget-58fd56e46d13c4e354e1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-58fd56e46d13c4e354e1">{"x":{"filter":"none","extensions":["Buttons"],"data":[["growing","growing","stable","growing","growing","stable","growing","stable","growing","stable","growing","growing","stable","stable","growing","growing","growing","growing","growing","stable","stable","growing","growing","stable","growing","growing","growing","growing","growing","growing","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","growing","growing","growing","growing","stable","growing","growing","growing","growing","growing","growing","stable","growing","growing","stable","growing","stable","growing","stable","growing","growing","stable","stable","growing","growing","growing","growing","growing","stable","stable","growing","growing","stable","growing","growing","growing","growing","growing","growing","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","growing","growing","growing","growing","stable","growing","growing","growing","growing","growing","growing","stable","growing","growing","stable","growing","stable","growing","stable","growing","growing","stable","stable","growing","growing","growing","growing","growing","stable","stable","growing","growing","stable","growing","growing","growing","growing","growing","growing","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","growing","growing","growing","growing","stable","growing","growing","growing","growing"],["PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","PDL1+LY","PDL1+LY","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","PDL1+LY","PDL1+LY","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","PDL1+LY","PDL1+LY"],[6.22857142857142,5.35714285714286,1.97297297297297,4.97619047619048,3.78571428571429,0.10655737704918,2.67857142857143,0.77027027027027,6.2,0.50625,5.96153846153846,3.31506849315069,0.55952380952381,0.762162162162162,4.3298969072165,3.5,5.71428571428571,2.68571428571429,3.5,0.647540983606557,1.91517857142857,2.45714285714286,6.8,-0.384615384615384,8.21428571428571,11.7,3.54285714285714,15,3,5.8,0.0245901639344268,7.68571428571428,5.54285714285714,6.87837837837838,3,5.42857142857143,10.3571428571429,2.19745222929936,30,2.8,8.21428571428571,1.28571428571429,4.775,2.35555555555556,2.88785046728972,3.44444444444444,0.241803278688525,12,5.51428571428571,5.85046728971962,3.25974025974026,6.22857142857142,5.35714285714286,1.97297297297297,4.97619047619048,3.78571428571429,0.10655737704918,2.67857142857143,0.77027027027027,6.2,0.50625,5.96153846153846,3.31506849315069,0.55952380952381,0.762162162162162,4.3298969072165,3.5,5.71428571428571,2.68571428571429,3.5,0.647540983606557,1.91517857142857,2.45714285714286,6.8,-0.384615384615384,8.21428571428571,11.7,3.54285714285714,15,3,5.8,0.0245901639344268,7.68571428571428,5.54285714285714,6.87837837837838,3,5.42857142857143,10.3571428571429,2.19745222929936,30,2.8,8.21428571428571,1.28571428571429,4.775,2.35555555555556,2.88785046728972,3.44444444444444,0.241803278688525,12,5.51428571428571,5.85046728971962,3.25974025974026,6.22857142857142,5.35714285714286,1.97297297297297,4.97619047619048,3.78571428571429,0.10655737704918,2.67857142857143,0.77027027027027,6.2,0.50625,5.96153846153846,3.31506849315069,0.55952380952381,0.762162162162162,4.3298969072165,3.5,5.71428571428571,2.68571428571429,3.5,0.647540983606557,1.91517857142857,2.45714285714286,6.8,-0.384615384615384,8.21428571428571,11.7,3.54285714285714,15,3,5.8,0.0245901639344268,7.68571428571428,5.54285714285714,6.87837837837838,3,5.42857142857143,10.3571428571429,2.19745222929936,30,2.8,8.21428571428571,1.28571428571429,4.775,2.35555555555556,2.88785046728972,3.44444444444444,0.241803278688525,12,5.51428571428571,5.85046728971962,3.25974025974026],["DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","DN.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","CD45.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS","EpCAM.Frac.FACS"],[75.7,35.6,39,19.4,46.9,13.2,32.5,52,36.3,32.6,52.8,19.9,50.1,49,38.8,32.7,39.9,65.1,42,23.1,27.1,47.7,29.3,46.3,18.4,61.6,21.9,19.2,80.9,69.6,44.5,65.2,40.2,11.5,41.3,24.1,20.2,34.1,41,31.1,63.7,11.1,72.3,33.2,49.4,40,43.4,32.3,29.7,21,15.7,15.2,51.4,49.8,14.5,37.3,70.3,23.8,11.2,18.4,12.1,9.74,19.6,36.9,25.3,6.16,45.5,14.8,7.25,27.4,61.7,40.5,9.83,20.4,6.88,4.35,21.8,70.7,73.5,12.5,26.2,24.4,29.7,39.4,1.64,4.86,31,8.93,12.3,6.85,61.1,27.2,80.8,4.66,23.7,30.4,13.2,34.6,31.8,9.79,13,15.8,8.59,12,13.5,66.6,11.6,10.4,42.3,32.9,37.7,30,38,59.7,11.3,24.8,50.4,18.7,44,26.3,29.8,13.7,30.7,36.3,50,44.8,67.1,16.6,6.23,5.23,4.21,2.55,25.3,3.17,17,81.6,51.2,42.2,65.3,46.6,54.1,6.12,11.9,6.61,20.7,40,14.9,43.5,16.5,29.8,55.7,61.1,57.4]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>Tumor.Growth<\/th>\n      <th>Treatment<\/th>\n      <th>GrowthRate<\/th>\n      <th>variable<\/th>\n      <th>value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[2,4]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="cohort-characteristics.html#cb43-1"></a><span class="co">#write.csv(Ltab2, file=&quot;nature-tables/Ext3c.csv&quot;)</span></span></code></pre></div>
</div>
<div id="facs-data" class="section level2">
<h2><span class="header-section-number">2.5</span> FACS data</h2>
<p>The immune (CD45) fractions from a number of samples were collected, and assessed using FACs. The major cell types detected are:</p>
<p>Leukocytes:</p>
<ul>
<li>Tregs</li>
<li>CD8 T cells</li>
<li>Thelper cells</li>
<li>B cells</li>
<li>NK T cells</li>
<li>gamma delta T cells</li>
</ul>
<p>Myeloid cells:</p>
<ul>
<li>Macrophages M1</li>
<li>Macrophages M2</li>
<li>Dendritic cells</li>
<li>Monocytes</li>
<li>Neutrophils</li>
</ul>
<p>We can look at the:</p>
<ul>
<li>types of cells</li>
<li>distributions</li>
</ul>
<p>Note that in a number of samples the leukocyte population could not be inferred with confidence, and proportions are normalised to the myeloid population</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="cohort-characteristics.html#cb44-1"></a>Fdata=<span class="kw">read.csv</span>(<span class="st">&quot;../data/carlos_facs_tumors.csv&quot;</span>, <span class="dt">stringsAsFactors =</span> F)</span>
<span id="cb44-2"><a href="cohort-characteristics.html#cb44-2"></a>Fdata[ ,<span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(Fdata)]=Fdata[ ,<span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(Fdata)]<span class="op">/</span><span class="dv">100</span></span>
<span id="cb44-3"><a href="cohort-characteristics.html#cb44-3"></a>m1=<span class="kw">substr</span>(<span class="kw">colnames</span>(Fdata), <span class="dv">2</span>, <span class="dv">5</span>)</span>
<span id="cb44-4"><a href="cohort-characteristics.html#cb44-4"></a><span class="kw">colnames</span>(Fdata)=m1</span>
<span id="cb44-5"><a href="cohort-characteristics.html#cb44-5"></a><span class="kw">colnames</span>(Fdata)[<span class="dv">1</span>]=<span class="st">&quot;type&quot;</span></span>
<span id="cb44-6"><a href="cohort-characteristics.html#cb44-6"></a></span>
<span id="cb44-7"><a href="cohort-characteristics.html#cb44-7"></a><span class="kw">scroll_box</span>(<span class="kw">kable</span>(Fdata, <span class="dt">format=</span><span class="st">&quot;html&quot;</span>),</span>
<span id="cb44-8"><a href="cohort-characteristics.html#cb44-8"></a>         <span class="dt">height=</span><span class="st">&quot;300px&quot;</span>, <span class="dt">width=</span><span class="st">&quot;100%&quot;</span>)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:300px; overflow-x: scroll; width:100%; ">
<table>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
type
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
11ND
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
8LD
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
8RCU
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
12LD
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
6RB
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
11RD
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
14ND
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
14NC
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
3NB
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
3RC
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
10LC
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
10RB
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
11LB
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
11RC
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
15LB
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
15NC
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
15RD
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
16LD
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
2RA
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
2RC
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
5NA
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
6RD
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
8RCL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Leukocytes
</td>
<td style="text-align:right;">
913.8400000
</td>
<td style="text-align:right;">
379.5600000
</td>
<td style="text-align:right;">
829.5700000
</td>
<td style="text-align:right;">
298.8300000
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
291.4300000
</td>
<td style="text-align:right;">
576.7400000
</td>
<td style="text-align:right;">
1118.8200000
</td>
<td style="text-align:right;">
616.3800000
</td>
<td style="text-align:right;">
414.7500000
</td>
<td style="text-align:right;">
899.6000000
</td>
<td style="text-align:right;">
1243.7900000
</td>
<td style="text-align:right;">
204.8900000
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
leukocytes
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Th
</td>
<td style="text-align:right;">
0.1823733
</td>
<td style="text-align:right;">
0.1223259
</td>
<td style="text-align:right;">
0.1239799
</td>
<td style="text-align:right;">
0.0542784
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.1503277
</td>
<td style="text-align:right;">
0.1370462
</td>
<td style="text-align:right;">
0.0964141
</td>
<td style="text-align:right;">
0.0980077
</td>
<td style="text-align:right;">
0.0832309
</td>
<td style="text-align:right;">
0.1489996
</td>
<td style="text-align:right;">
0.0910604
</td>
<td style="text-align:right;">
0.0454390
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Tregs
</td>
<td style="text-align:right;">
0.1020748
</td>
<td style="text-align:right;">
0.3822584
</td>
<td style="text-align:right;">
0.0570175
</td>
<td style="text-align:right;">
0.0703075
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.0703085
</td>
<td style="text-align:right;">
0.0438673
</td>
<td style="text-align:right;">
0.1722261
</td>
<td style="text-align:right;">
0.1392972
</td>
<td style="text-align:right;">
0.1607233
</td>
<td style="text-align:right;">
0.1387394
</td>
<td style="text-align:right;">
0.0263389
</td>
<td style="text-align:right;">
0.0292352
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
CD8 T cells
</td>
<td style="text-align:right;">
0.0558632
</td>
<td style="text-align:right;">
0.0329329
</td>
<td style="text-align:right;">
0.0248804
</td>
<td style="text-align:right;">
0.0239936
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.0681124
</td>
<td style="text-align:right;">
0.0403648
</td>
<td style="text-align:right;">
0.0628966
</td>
<td style="text-align:right;">
0.0438204
</td>
<td style="text-align:right;">
0.0201326
</td>
<td style="text-align:right;">
0.0435749
</td>
<td style="text-align:right;">
0.0088681
</td>
<td style="text-align:right;">
0.0188394
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
gd T cells
</td>
<td style="text-align:right;">
0.0057231
</td>
<td style="text-align:right;">
0.0029508
</td>
<td style="text-align:right;">
0.0048338
</td>
<td style="text-align:right;">
0.0051869
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.0052843
</td>
<td style="text-align:right;">
0.0034851
</td>
<td style="text-align:right;">
0.0054343
</td>
<td style="text-align:right;">
0.0099614
</td>
<td style="text-align:right;">
0.0108981
</td>
<td style="text-align:right;">
0.0039573
</td>
<td style="text-align:right;">
0.0023155
</td>
<td style="text-align:right;">
0.0078091
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
NK cells
</td>
<td style="text-align:right;">
0.0081852
</td>
<td style="text-align:right;">
0.0023975
</td>
<td style="text-align:right;">
0.0086430
</td>
<td style="text-align:right;">
0.0082321
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.0024706
</td>
<td style="text-align:right;">
0.0033984
</td>
<td style="text-align:right;">
0.0057471
</td>
<td style="text-align:right;">
0.0130277
</td>
<td style="text-align:right;">
0.0073056
</td>
<td style="text-align:right;">
0.0078924
</td>
<td style="text-align:right;">
0.0047355
</td>
<td style="text-align:right;">
0.0009273
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
B cells
</td>
<td style="text-align:right;">
0.0320297
</td>
<td style="text-align:right;">
0.0259774
</td>
<td style="text-align:right;">
0.1150958
</td>
<td style="text-align:right;">
0.3691731
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.0552105
</td>
<td style="text-align:right;">
0.1627076
</td>
<td style="text-align:right;">
0.1690978
</td>
<td style="text-align:right;">
0.1648821
</td>
<td style="text-align:right;">
0.0852803
</td>
<td style="text-align:right;">
0.1415963
</td>
<td style="text-align:right;">
0.1131783
</td>
<td style="text-align:right;">
0.5891942
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
DC
</td>
<td style="text-align:right;">
0.0847253
</td>
<td style="text-align:right;">
0.0099781
</td>
<td style="text-align:right;">
0.0240649
</td>
<td style="text-align:right;">
0.0937623
</td>
<td style="text-align:right;">
0.0796767
</td>
<td style="text-align:right;">
0.0904331
</td>
<td style="text-align:right;">
0.0210829
</td>
<td style="text-align:right;">
0.0866281
</td>
<td style="text-align:right;">
0.1246231
</td>
<td style="text-align:right;">
0.2439219
</td>
<td style="text-align:right;">
0.0338069
</td>
<td style="text-align:right;">
0.0087373
</td>
<td style="text-align:right;">
0.0787654
</td>
<td style="text-align:right;">
0.0441001
</td>
<td style="text-align:right;">
0.0782997
</td>
<td style="text-align:right;">
0.1614173
</td>
<td style="text-align:right;">
0.0582119
</td>
<td style="text-align:right;">
0.0529557
</td>
<td style="text-align:right;">
0.0371143
</td>
<td style="text-align:right;">
0.0034515
</td>
<td style="text-align:right;">
0.0605741
</td>
<td style="text-align:right;">
0.0224583
</td>
<td style="text-align:right;">
0.0656124
</td>
</tr>
<tr>
<td style="text-align:left;">
Monocytes
</td>
<td style="text-align:right;">
0.0869640
</td>
<td style="text-align:right;">
0.0992694
</td>
<td style="text-align:right;">
0.1052141
</td>
<td style="text-align:right;">
0.1075509
</td>
<td style="text-align:right;">
0.0750577
</td>
<td style="text-align:right;">
0.0896695
</td>
<td style="text-align:right;">
0.1671775
</td>
<td style="text-align:right;">
0.0227902
</td>
<td style="text-align:right;">
0.0658291
</td>
<td style="text-align:right;">
0.0406536
</td>
<td style="text-align:right;">
0.0388811
</td>
<td style="text-align:right;">
0.0573525
</td>
<td style="text-align:right;">
0.0550409
</td>
<td style="text-align:right;">
0.0607867
</td>
<td style="text-align:right;">
0.0630813
</td>
<td style="text-align:right;">
0.0921618
</td>
<td style="text-align:right;">
0.1345743
</td>
<td style="text-align:right;">
0.2219148
</td>
<td style="text-align:right;">
0.0818758
</td>
<td style="text-align:right;">
0.0326953
</td>
<td style="text-align:right;">
0.1337320
</td>
<td style="text-align:right;">
0.0871336
</td>
<td style="text-align:right;">
0.0882122
</td>
</tr>
<tr>
<td style="text-align:left;">
Neutrophils
</td>
<td style="text-align:right;">
0.1195109
</td>
<td style="text-align:right;">
0.0713534
</td>
<td style="text-align:right;">
0.0989646
</td>
<td style="text-align:right;">
0.0961261
</td>
<td style="text-align:right;">
0.1039261
</td>
<td style="text-align:right;">
0.0484346
</td>
<td style="text-align:right;">
0.0821956
</td>
<td style="text-align:right;">
0.0795769
</td>
<td style="text-align:right;">
0.1165829
</td>
<td style="text-align:right;">
0.0390594
</td>
<td style="text-align:right;">
0.2022707
</td>
<td style="text-align:right;">
0.0442465
</td>
<td style="text-align:right;">
0.1378131
</td>
<td style="text-align:right;">
0.0369487
</td>
<td style="text-align:right;">
0.0839361
</td>
<td style="text-align:right;">
0.0629921
</td>
<td style="text-align:right;">
0.1058862
</td>
<td style="text-align:right;">
0.0959791
</td>
<td style="text-align:right;">
0.1177103
</td>
<td style="text-align:right;">
0.0213367
</td>
<td style="text-align:right;">
0.1875633
</td>
<td style="text-align:right;">
0.0681735
</td>
<td style="text-align:right;">
0.0702109
</td>
</tr>
<tr>
<td style="text-align:left;">
MHCII-hi Macro
</td>
<td style="text-align:right;">
0.0079215
</td>
<td style="text-align:right;">
0.0188760
</td>
<td style="text-align:right;">
0.0327395
</td>
<td style="text-align:right;">
0.0978332
</td>
<td style="text-align:right;">
0.0254042
</td>
<td style="text-align:right;">
0.1052689
</td>
<td style="text-align:right;">
0.1015139
</td>
<td style="text-align:right;">
0.0328633
</td>
<td style="text-align:right;">
0.0341709
</td>
<td style="text-align:right;">
0.0462336
</td>
<td style="text-align:right;">
0.0260688
</td>
<td style="text-align:right;">
0.0224593
</td>
<td style="text-align:right;">
0.0856493
</td>
<td style="text-align:right;">
0.0309893
</td>
<td style="text-align:right;">
0.0660404
</td>
<td style="text-align:right;">
0.0758769
</td>
<td style="text-align:right;">
0.0283632
</td>
<td style="text-align:right;">
0.1103736
</td>
<td style="text-align:right;">
0.0456621
</td>
<td style="text-align:right;">
0.0460621
</td>
<td style="text-align:right;">
0.0757107
</td>
<td style="text-align:right;">
0.1725741
</td>
<td style="text-align:right;">
0.0463399
</td>
</tr>
<tr>
<td style="text-align:left;">
MHCII-lo Macro
</td>
<td style="text-align:right;">
0.0451180
</td>
<td style="text-align:right;">
0.2975240
</td>
<td style="text-align:right;">
0.2654603
</td>
<td style="text-align:right;">
0.1099146
</td>
<td style="text-align:right;">
0.0542725
</td>
<td style="text-align:right;">
0.0976328
</td>
<td style="text-align:right;">
0.1608619
</td>
<td style="text-align:right;">
0.0690003
</td>
<td style="text-align:right;">
0.0894472
</td>
<td style="text-align:right;">
0.0278996
</td>
<td style="text-align:right;">
0.0253711
</td>
<td style="text-align:right;">
0.2559019
</td>
<td style="text-align:right;">
0.0511094
</td>
<td style="text-align:right;">
0.0536353
</td>
<td style="text-align:right;">
0.0862846
</td>
<td style="text-align:right;">
0.0606657
</td>
<td style="text-align:right;">
0.0647108
</td>
<td style="text-align:right;">
0.1009692
</td>
<td style="text-align:right;">
0.2576354
</td>
<td style="text-align:right;">
0.2502040
</td>
<td style="text-align:right;">
0.0721728
</td>
<td style="text-align:right;">
0.1834358
</td>
<td style="text-align:right;">
0.0866607
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Total
</td>
<td style="text-align:right;">
0.7304889
</td>
<td style="text-align:right;">
1.0658438
</td>
<td style="text-align:right;">
0.8608938
</td>
<td style="text-align:right;">
1.0363587
</td>
<td style="text-align:right;">
0.3383372
</td>
<td style="text-align:right;">
0.4314389
</td>
<td style="text-align:right;">
0.8845458
</td>
<td style="text-align:right;">
0.6817281
</td>
<td style="text-align:right;">
0.9424693
</td>
<td style="text-align:right;">
0.8667644
</td>
<td style="text-align:right;">
0.6939694
</td>
<td style="text-align:right;">
0.8734574
</td>
<td style="text-align:right;">
0.6548747
</td>
<td style="text-align:right;">
0.9179043
</td>
<td style="text-align:right;">
0.3776421
</td>
<td style="text-align:right;">
0.4531138
</td>
<td style="text-align:right;">
0.3917464
</td>
<td style="text-align:right;">
0.5821924
</td>
<td style="text-align:right;">
0.5399978
</td>
<td style="text-align:right;">
0.3537496
</td>
<td style="text-align:right;">
0.5297529
</td>
<td style="text-align:right;">
0.5337753
</td>
<td style="text-align:right;">
0.3570360
</td>
</tr>
<tr>
<td style="text-align:left;">
Lin-
</td>
<td style="text-align:right;">
0.2695111
</td>
<td style="text-align:right;">
-0.0658438
</td>
<td style="text-align:right;">
0.1391062
</td>
<td style="text-align:right;">
-0.0363587
</td>
<td style="text-align:right;">
0.6616628
</td>
<td style="text-align:right;">
0.5685611
</td>
<td style="text-align:right;">
0.1154542
</td>
<td style="text-align:right;">
0.3182719
</td>
<td style="text-align:right;">
0.0575307
</td>
<td style="text-align:right;">
0.1332356
</td>
<td style="text-align:right;">
0.3060306
</td>
<td style="text-align:right;">
0.1265426
</td>
<td style="text-align:right;">
0.3451253
</td>
<td style="text-align:right;">
0.0820957
</td>
<td style="text-align:right;">
0.6223579
</td>
<td style="text-align:right;">
0.5468862
</td>
<td style="text-align:right;">
0.6082536
</td>
<td style="text-align:right;">
0.4178076
</td>
<td style="text-align:right;">
0.4600022
</td>
<td style="text-align:right;">
0.6462504
</td>
<td style="text-align:right;">
0.4702471
</td>
<td style="text-align:right;">
0.4662247
</td>
<td style="text-align:right;">
0.6429640
</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="cohort-characteristics.html#cb45-1"></a></span>
<span id="cb45-2"><a href="cohort-characteristics.html#cb45-2"></a>t2=Fdata[<span class="op">-</span><span class="kw">which</span>(Fdata[ ,<span class="dv">1</span>]<span class="op">==</span><span class="st">&quot;&quot;</span>),]</span>
<span id="cb45-3"><a href="cohort-characteristics.html#cb45-3"></a><span class="co">#rownames(t2)=Fdata[-which(Fdata[ ,1]==&quot;&quot;) ,1]</span></span>
<span id="cb45-4"><a href="cohort-characteristics.html#cb45-4"></a>t2=t2[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, <span class="dv">14</span>), ]</span>
<span id="cb45-5"><a href="cohort-characteristics.html#cb45-5"></a>t2melt=<span class="kw">melt</span>(t2)</span>
<span id="cb45-6"><a href="cohort-characteristics.html#cb45-6"></a></span>
<span id="cb45-7"><a href="cohort-characteristics.html#cb45-7"></a><span class="kw">ggplot</span>(t2melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;sample&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;proportion&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&#39;all samples&#39;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p>We can look solely at the myeloid population (and normalise to this total), and color according to growth</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="cohort-characteristics.html#cb46-1"></a>t3=t2[<span class="dv">7</span><span class="op">:</span><span class="dv">11</span>, ]</span>
<span id="cb46-2"><a href="cohort-characteristics.html#cb46-2"></a>t3[, <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(t3)]=<span class="kw">t</span>(<span class="kw">t</span>(t3[, <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(t3)])<span class="op">/</span><span class="kw">colSums</span>(t3[, <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(t3)]))</span>
<span id="cb46-3"><a href="cohort-characteristics.html#cb46-3"></a>t3melt=<span class="kw">melt</span>(t3)</span>
<span id="cb46-4"><a href="cohort-characteristics.html#cb46-4"></a></span>
<span id="cb46-5"><a href="cohort-characteristics.html#cb46-5"></a>t3melt<span class="op">$</span>growth=d1<span class="op">$</span>growthrate_cutoff2[<span class="kw">match</span>(t3melt<span class="op">$</span>variable, d1<span class="op">$</span>sample)]</span>
<span id="cb46-6"><a href="cohort-characteristics.html#cb46-6"></a></span>
<span id="cb46-7"><a href="cohort-characteristics.html#cb46-7"></a><span class="kw">ggplot</span>(t3melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;sample&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;proportion&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&#39;myeloid specific&#39;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<p>Similarly, we can look at the leukocyte population. Note that the Treg population in some of these samples is very high.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="cohort-characteristics.html#cb47-1"></a>t3=t2[<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>, ]</span>
<span id="cb47-2"><a href="cohort-characteristics.html#cb47-2"></a>t3[, <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(t3)]=<span class="kw">t</span>(<span class="kw">t</span>(t3[, <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(t3)])<span class="op">/</span><span class="kw">colSums</span>(t3[, <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(t3)]))</span>
<span id="cb47-3"><a href="cohort-characteristics.html#cb47-3"></a>t3melt=<span class="kw">melt</span>(t3)</span>
<span id="cb47-4"><a href="cohort-characteristics.html#cb47-4"></a></span>
<span id="cb47-5"><a href="cohort-characteristics.html#cb47-5"></a>t3melt<span class="op">$</span>growth=d1<span class="op">$</span>growthrate_cutoff2[<span class="kw">match</span>(t3melt<span class="op">$</span>variable, d1<span class="op">$</span>sample)]</span>
<span id="cb47-6"><a href="cohort-characteristics.html#cb47-6"></a></span>
<span id="cb47-7"><a href="cohort-characteristics.html#cb47-7"></a><span class="kw">ggplot</span>(t3melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;sample&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;proportion&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&#39;leukocyte specific&#39;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="prerequisites.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cohort-summary-table.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
