<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Collating results and running GSEA | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Collating results and running GSEA | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Collating results and running GSEA | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-03-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="deseq-analysis-progressionimmunotherapy-cohort.html"/>
<link rel="next" href="erpgr-subtyping.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.3.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.3.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>24.3</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="collating-results-and-running-gsea" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Collating results and running GSEA</h1>
<p>In this section, we combine all the comparisons together for downstream analysis, including</p>
<ul>
<li>number of differential genes per comparisons</li>
<li>number of samples per comparison</li>
<li>write the output to file (outputs/DESeq/Ep_growth_treat.xlsx)</li>
</ul>
<p>For GSEA we</p>
<ul>
<li>convert rat to human symbols</li>
<li>run overlap analysis if there are more than 3 hits (outputs/DESeq/HyperGeo_Ep_growth_treat.xlsx)</li>
<li>run gene set enrichment analysis on all samples (outputs/DESeq/GSEA_Ep_growth_treat.xlsx)</li>
<li>all outputs saved to outputs/all_differential_comparisons.RData)</li>
</ul>
<p>An example of the code is below:</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="collating-results-and-running-gsea.html#cb159-1"></a>hits=EpCompSig[[i]] <span class="co">## list of significant genes</span></span>
<span id="cb159-2"><a href="collating-results-and-running-gsea.html#cb159-2"></a>hits=epGenesConv[<span class="kw">match</span>(hits, <span class="kw">rownames</span>(Epdds))]</span>
<span id="cb159-3"><a href="collating-results-and-running-gsea.html#cb159-3"></a>fcTab=EpComp4[[i]]<span class="op">$</span>log2FoldChange</span>
<span id="cb159-4"><a href="collating-results-and-running-gsea.html#cb159-4"></a><span class="kw">names</span>(fcTab)=epGenesConv</span>
<span id="cb159-5"><a href="collating-results-and-running-gsea.html#cb159-5"></a></span>
<span id="cb159-6"><a href="collating-results-and-running-gsea.html#cb159-6"></a></span>
<span id="cb159-7"><a href="collating-results-and-running-gsea.html#cb159-7"></a>  gsca=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb159-8"><a href="collating-results-and-running-gsea.html#cb159-8"></a>  gsca1 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gsca, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb159-9"><a href="collating-results-and-running-gsea.html#cb159-9"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb159-10"><a href="collating-results-and-running-gsea.html#cb159-10"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb159-11"><a href="collating-results-and-running-gsea.html#cb159-11"></a>  gsca2 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gsca1, </span>
<span id="cb159-12"><a href="collating-results-and-running-gsea.html#cb159-12"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb159-13"><a href="collating-results-and-running-gsea.html#cb159-13"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb159-14"><a href="collating-results-and-running-gsea.html#cb159-14"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb159-15"><a href="collating-results-and-running-gsea.html#cb159-15"></a>                           <span class="dt">doGSOA =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="collating-results-and-running-gsea.html#cb160-1"></a><span class="co">#load(&quot;rslt/DESeq/Epithelial_fraction.RData&quot;)</span></span>
<span id="cb160-2"><a href="collating-results-and-running-gsea.html#cb160-2"></a></span>
<span id="cb160-3"><a href="collating-results-and-running-gsea.html#cb160-3"></a><span class="co">#table(Epdds$Comp4)</span></span>
<span id="cb160-4"><a href="collating-results-and-running-gsea.html#cb160-4"></a></span>
<span id="cb160-5"><a href="collating-results-and-running-gsea.html#cb160-5"></a><span class="co">#xsearch=c(&quot;stablecontrol&quot;,&quot;growingimm&quot;,  &quot;stableimm&quot;, &quot;stableimm&quot;)</span></span>
<span id="cb160-6"><a href="collating-results-and-running-gsea.html#cb160-6"></a><span class="co">#ysearch=c(&quot;growingcontrol&quot;,&quot;growingcontrol&quot;,  &quot;stablecontrol&quot;, &quot;growingimm&quot;)</span></span>
<span id="cb160-7"><a href="collating-results-and-running-gsea.html#cb160-7"></a></span>
<span id="cb160-8"><a href="collating-results-and-running-gsea.html#cb160-8"></a><span class="co">## for treatment control: big differences here which are not replicated</span></span>
<span id="cb160-9"><a href="collating-results-and-running-gsea.html#cb160-9"></a></span>
<span id="cb160-10"><a href="collating-results-and-running-gsea.html#cb160-10"></a>EpComp4=<span class="kw">list</span>()</span>
<span id="cb160-11"><a href="collating-results-and-running-gsea.html#cb160-11"></a>EpCompSig=<span class="kw">list</span>()</span>
<span id="cb160-12"><a href="collating-results-and-running-gsea.html#cb160-12"></a>UpDn1=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">2</span>, <span class="dt">ncol=</span><span class="dv">7</span>)</span>
<span id="cb160-13"><a href="collating-results-and-running-gsea.html#cb160-13"></a></span>
<span id="cb160-14"><a href="collating-results-and-running-gsea.html#cb160-14"></a><span class="co">#for (i in 1:2){</span></span>
<span id="cb160-15"><a href="collating-results-and-running-gsea.html#cb160-15"></a>Eres1=<span class="kw">results</span>(Epdds, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;treatA&quot;</span>, <span class="st">&quot;imm&quot;</span>, <span class="st">&quot;control&quot;</span>))</span>
<span id="cb160-16"><a href="collating-results-and-running-gsea.html#cb160-16"></a>EpComp4[[<span class="dv">1</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres1), Eres1))</span>
<span id="cb160-17"><a href="collating-results-and-running-gsea.html#cb160-17"></a>EpCompSig[[<span class="dv">1</span>]]=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres1)[<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb160-18"><a href="collating-results-and-running-gsea.html#cb160-18"></a>UpDn1[ ,<span class="dv">1</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres1<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb160-19"><a href="collating-results-and-running-gsea.html#cb160-19"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres1<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb160-20"><a href="collating-results-and-running-gsea.html#cb160-20"></a><span class="co">#}</span></span>
<span id="cb160-21"><a href="collating-results-and-running-gsea.html#cb160-21"></a><span class="co"># for control</span></span>
<span id="cb160-22"><a href="collating-results-and-running-gsea.html#cb160-22"></a></span>
<span id="cb160-23"><a href="collating-results-and-running-gsea.html#cb160-23"></a>cList=<span class="kw">c</span>(<span class="st">&quot;PDL1+LY&quot;</span>, <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;PDL1&quot;</span>)</span>
<span id="cb160-24"><a href="collating-results-and-running-gsea.html#cb160-24"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>){</span>
<span id="cb160-25"><a href="collating-results-and-running-gsea.html#cb160-25"></a>Eres5=<span class="kw">results</span>(EpddsTreat, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, cList[i], <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb160-26"><a href="collating-results-and-running-gsea.html#cb160-26"></a>EpComp4[[i<span class="op">+</span><span class="dv">1</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb160-27"><a href="collating-results-and-running-gsea.html#cb160-27"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb160-28"><a href="collating-results-and-running-gsea.html#cb160-28"></a>EpCompSig[[i<span class="op">+</span><span class="dv">1</span>]]=g1</span>
<span id="cb160-29"><a href="collating-results-and-running-gsea.html#cb160-29"></a>UpDn1[ ,i<span class="op">+</span><span class="dv">1</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb160-30"><a href="collating-results-and-running-gsea.html#cb160-30"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb160-31"><a href="collating-results-and-running-gsea.html#cb160-31"></a>}</span>
<span id="cb160-32"><a href="collating-results-and-running-gsea.html#cb160-32"></a></span>
<span id="cb160-33"><a href="collating-results-and-running-gsea.html#cb160-33"></a><span class="co"># xsearchA=c(&quot;stable.Vehicle&quot;,&quot;stable.LY&quot;,  &quot;stable.PDL1+LY&quot;, &quot;stable.PDL1&quot;)</span></span>
<span id="cb160-34"><a href="collating-results-and-running-gsea.html#cb160-34"></a><span class="co"># ysearchB=c(&quot;growing.Vehicle&quot;,&quot;growing.LY&quot;,  &quot;growing.PDL1+LY&quot;, &quot;growing.PDL1&quot;)</span></span>
<span id="cb160-35"><a href="collating-results-and-running-gsea.html#cb160-35"></a><span class="co"># </span></span>
<span id="cb160-36"><a href="collating-results-and-running-gsea.html#cb160-36"></a><span class="co"># </span></span>
<span id="cb160-37"><a href="collating-results-and-running-gsea.html#cb160-37"></a><span class="co"># for (i in 1:4){</span></span>
<span id="cb160-38"><a href="collating-results-and-running-gsea.html#cb160-38"></a><span class="co"># Eres5=results(EpddsTreatG, contrast=c(&quot;Comp8&quot;, xsearchA[i], ysearchB[i]))</span></span>
<span id="cb160-39"><a href="collating-results-and-running-gsea.html#cb160-39"></a><span class="co"># EpComp4[[i+7]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb160-40"><a href="collating-results-and-running-gsea.html#cb160-40"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb160-41"><a href="collating-results-and-running-gsea.html#cb160-41"></a><span class="co"># EpCompSig[[i+7]]=g1</span></span>
<span id="cb160-42"><a href="collating-results-and-running-gsea.html#cb160-42"></a><span class="co"># UpDn1[ ,i+7]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb160-43"><a href="collating-results-and-running-gsea.html#cb160-43"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb160-44"><a href="collating-results-and-running-gsea.html#cb160-44"></a><span class="co"># }</span></span>
<span id="cb160-45"><a href="collating-results-and-running-gsea.html#cb160-45"></a></span>
<span id="cb160-46"><a href="collating-results-and-running-gsea.html#cb160-46"></a></span>
<span id="cb160-47"><a href="collating-results-and-running-gsea.html#cb160-47"></a></span>
<span id="cb160-48"><a href="collating-results-and-running-gsea.html#cb160-48"></a>Eres5=<span class="kw">results</span>(EpddsGrowth, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;Growth&quot;</span>, <span class="st">&quot;stable&quot;</span>, <span class="st">&quot;growing&quot;</span>))</span>
<span id="cb160-49"><a href="collating-results-and-running-gsea.html#cb160-49"></a>EpComp4[[<span class="dv">5</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb160-50"><a href="collating-results-and-running-gsea.html#cb160-50"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb160-51"><a href="collating-results-and-running-gsea.html#cb160-51"></a>EpCompSig[[<span class="dv">5</span>]]=g1</span>
<span id="cb160-52"><a href="collating-results-and-running-gsea.html#cb160-52"></a>UpDn1[ ,<span class="dv">5</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb160-53"><a href="collating-results-and-running-gsea.html#cb160-53"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb160-54"><a href="collating-results-and-running-gsea.html#cb160-54"></a></span>
<span id="cb160-55"><a href="collating-results-and-running-gsea.html#cb160-55"></a></span>
<span id="cb160-56"><a href="collating-results-and-running-gsea.html#cb160-56"></a>Eres5=<span class="kw">results</span>(EpddsStrMH, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;MHcut&quot;</span>, <span class="st">&quot;res&quot;</span>, <span class="st">&quot;inf&quot;</span>))</span>
<span id="cb160-57"><a href="collating-results-and-running-gsea.html#cb160-57"></a>EpComp4[[<span class="dv">6</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb160-58"><a href="collating-results-and-running-gsea.html#cb160-58"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb160-59"><a href="collating-results-and-running-gsea.html#cb160-59"></a>EpCompSig[[<span class="dv">6</span>]]=g1</span>
<span id="cb160-60"><a href="collating-results-and-running-gsea.html#cb160-60"></a>UpDn1[ ,<span class="dv">6</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb160-61"><a href="collating-results-and-running-gsea.html#cb160-61"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb160-62"><a href="collating-results-and-running-gsea.html#cb160-62"></a></span>
<span id="cb160-63"><a href="collating-results-and-running-gsea.html#cb160-63"></a><span class="co"># Eres5=results(EpddsStrIF, contrast=c(&quot;IFcut&quot;, &quot;res&quot;, &quot;inf&quot;))</span></span>
<span id="cb160-64"><a href="collating-results-and-running-gsea.html#cb160-64"></a><span class="co"># EpComp4[[14]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb160-65"><a href="collating-results-and-running-gsea.html#cb160-65"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb160-66"><a href="collating-results-and-running-gsea.html#cb160-66"></a><span class="co"># EpCompSig[[14]]=g1</span></span>
<span id="cb160-67"><a href="collating-results-and-running-gsea.html#cb160-67"></a><span class="co"># UpDn1[ ,14]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb160-68"><a href="collating-results-and-running-gsea.html#cb160-68"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb160-69"><a href="collating-results-and-running-gsea.html#cb160-69"></a><span class="co"># </span></span>
<span id="cb160-70"><a href="collating-results-and-running-gsea.html#cb160-70"></a><span class="co"># Eres5=results(EpddsStrknn, contrast=c(&quot;knncut&quot;, &quot;res&quot;, &quot;inf&quot;))</span></span>
<span id="cb160-71"><a href="collating-results-and-running-gsea.html#cb160-71"></a><span class="co"># EpComp4[[15]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb160-72"><a href="collating-results-and-running-gsea.html#cb160-72"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb160-73"><a href="collating-results-and-running-gsea.html#cb160-73"></a><span class="co"># EpCompSig[[15]]=g1</span></span>
<span id="cb160-74"><a href="collating-results-and-running-gsea.html#cb160-74"></a><span class="co"># UpDn1[ ,15]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb160-75"><a href="collating-results-and-running-gsea.html#cb160-75"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb160-76"><a href="collating-results-and-running-gsea.html#cb160-76"></a></span>
<span id="cb160-77"><a href="collating-results-and-running-gsea.html#cb160-77"></a>Eres5=<span class="kw">results</span>(Epddscd8, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;CD8FracCut&quot;</span>, <span class="st">&quot;low&quot;</span>, <span class="st">&quot;high&quot;</span>))</span>
<span id="cb160-78"><a href="collating-results-and-running-gsea.html#cb160-78"></a>EpComp4[[<span class="dv">7</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb160-79"><a href="collating-results-and-running-gsea.html#cb160-79"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb160-80"><a href="collating-results-and-running-gsea.html#cb160-80"></a>EpCompSig[[<span class="dv">7</span>]]=g1</span>
<span id="cb160-81"><a href="collating-results-and-running-gsea.html#cb160-81"></a>UpDn1[ ,<span class="dv">7</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb160-82"><a href="collating-results-and-running-gsea.html#cb160-82"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb160-83"><a href="collating-results-and-running-gsea.html#cb160-83"></a></span>
<span id="cb160-84"><a href="collating-results-and-running-gsea.html#cb160-84"></a><span class="co"># Eres5=results(EpddsSpatMan, contrast=c(&quot;SpatialManual&quot;, &quot;Infiltrating&quot;, &quot;restricted&quot;))</span></span>
<span id="cb160-85"><a href="collating-results-and-running-gsea.html#cb160-85"></a><span class="co"># EpComp4[[17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb160-86"><a href="collating-results-and-running-gsea.html#cb160-86"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb160-87"><a href="collating-results-and-running-gsea.html#cb160-87"></a><span class="co"># EpCompSig[[17]]=g1</span></span>
<span id="cb160-88"><a href="collating-results-and-running-gsea.html#cb160-88"></a><span class="co"># UpDn1[ ,17]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb160-89"><a href="collating-results-and-running-gsea.html#cb160-89"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb160-90"><a href="collating-results-and-running-gsea.html#cb160-90"></a><span class="co"># </span></span>
<span id="cb160-91"><a href="collating-results-and-running-gsea.html#cb160-91"></a><span class="co"># xsearchC=c(&quot;low.inf&quot;,&quot;high.inf&quot;,  &quot;low.inf&quot;, &quot;low.res&quot;)</span></span>
<span id="cb160-92"><a href="collating-results-and-running-gsea.html#cb160-92"></a><span class="co"># ysearchD=c(&quot;low.res&quot;,&quot;high.res&quot;, &quot;high.inf&quot;, &quot;high.res&quot;)</span></span>
<span id="cb160-93"><a href="collating-results-and-running-gsea.html#cb160-93"></a><span class="co"># </span></span>
<span id="cb160-94"><a href="collating-results-and-running-gsea.html#cb160-94"></a><span class="co"># for (i in 1:4){</span></span>
<span id="cb160-95"><a href="collating-results-and-running-gsea.html#cb160-95"></a><span class="co"># Eres5=results(Epddscd8MH, contrast=c(&quot;cd8MH&quot;, xsearchC[i], ysearchD[i]))</span></span>
<span id="cb160-96"><a href="collating-results-and-running-gsea.html#cb160-96"></a><span class="co"># EpComp4[[i+17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb160-97"><a href="collating-results-and-running-gsea.html#cb160-97"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb160-98"><a href="collating-results-and-running-gsea.html#cb160-98"></a><span class="co"># EpCompSig[[i+17]]=g1</span></span>
<span id="cb160-99"><a href="collating-results-and-running-gsea.html#cb160-99"></a><span class="co"># UpDn1[ ,i+17]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb160-100"><a href="collating-results-and-running-gsea.html#cb160-100"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb160-101"><a href="collating-results-and-running-gsea.html#cb160-101"></a><span class="co"># }</span></span>
<span id="cb160-102"><a href="collating-results-and-running-gsea.html#cb160-102"></a><span class="co"># </span></span>
<span id="cb160-103"><a href="collating-results-and-running-gsea.html#cb160-103"></a><span class="co">#boxplot(assay(Epdds)[&quot;Snca&quot;, ]~Epdds$treatA+Epdds$Growth)</span></span>
<span id="cb160-104"><a href="collating-results-and-running-gsea.html#cb160-104"></a></span>
<span id="cb160-105"><a href="collating-results-and-running-gsea.html#cb160-105"></a><span class="kw">rownames</span>(UpDn1)=<span class="kw">c</span>(<span class="st">&quot;down&quot;</span>, <span class="st">&quot;up&quot;</span>)</span>
<span id="cb160-106"><a href="collating-results-and-running-gsea.html#cb160-106"></a><span class="kw">colnames</span>(UpDn1)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb160-107"><a href="collating-results-and-running-gsea.html#cb160-107"></a><span class="kw">names</span>(EpComp4)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb160-108"><a href="collating-results-and-running-gsea.html#cb160-108"></a><span class="kw">names</span>(EpCompSig)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb160-109"><a href="collating-results-and-running-gsea.html#cb160-109"></a></span>
<span id="cb160-110"><a href="collating-results-and-running-gsea.html#cb160-110"></a><span class="co">## write to a xls</span></span>
<span id="cb160-111"><a href="collating-results-and-running-gsea.html#cb160-111"></a><span class="kw">write_xlsx</span>(EpComp4, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/DESeq/DGEA_Ep_growth_treat_%s.xlsx&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="collating-results-and-running-gsea.html#cb161-1"></a><span class="co">#load(&quot;anntotations/Rat_biomart_gene_annotations.RData&quot;)</span></span>
<span id="cb161-2"><a href="collating-results-and-running-gsea.html#cb161-2"></a></span>
<span id="cb161-3"><a href="collating-results-and-running-gsea.html#cb161-3"></a>epGenes=<span class="kw">rownames</span>(Epdds)</span>
<span id="cb161-4"><a href="collating-results-and-running-gsea.html#cb161-4"></a>cdGenes=<span class="kw">rownames</span>(CDdds)</span>
<span id="cb161-5"><a href="collating-results-and-running-gsea.html#cb161-5"></a>dnGenes=<span class="kw">rownames</span>(DNdds)</span>
<span id="cb161-6"><a href="collating-results-and-running-gsea.html#cb161-6"></a></span>
<span id="cb161-7"><a href="collating-results-and-running-gsea.html#cb161-7"></a>l1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(epGenes, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb161-8"><a href="collating-results-and-running-gsea.html#cb161-8"></a>l2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(epGenes, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb161-9"><a href="collating-results-and-running-gsea.html#cb161-9"></a>l3=Mouse2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(epGenes, Mouse2Hum<span class="op">$</span>MGI.symbol)]</span>
<span id="cb161-10"><a href="collating-results-and-running-gsea.html#cb161-10"></a>epGenesConv=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(l1)<span class="op">==</span>F, l1, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l2)<span class="op">==</span>F, l2, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l3)<span class="op">==</span>F, l3, epGenes)))</span>
<span id="cb161-11"><a href="collating-results-and-running-gsea.html#cb161-11"></a></span>
<span id="cb161-12"><a href="collating-results-and-running-gsea.html#cb161-12"></a>l1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(cdGenes, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb161-13"><a href="collating-results-and-running-gsea.html#cb161-13"></a>l2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(cdGenes, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb161-14"><a href="collating-results-and-running-gsea.html#cb161-14"></a>l3=Mouse2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(cdGenes, Mouse2Hum<span class="op">$</span>MGI.symbol)]</span>
<span id="cb161-15"><a href="collating-results-and-running-gsea.html#cb161-15"></a>cdGenesConv=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(l1)<span class="op">==</span>F, l1, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l2)<span class="op">==</span>F, l2, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l3)<span class="op">==</span>F, l3, cdGenes)))</span>
<span id="cb161-16"><a href="collating-results-and-running-gsea.html#cb161-16"></a></span>
<span id="cb161-17"><a href="collating-results-and-running-gsea.html#cb161-17"></a>l1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(dnGenes, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb161-18"><a href="collating-results-and-running-gsea.html#cb161-18"></a>l2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(dnGenes, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb161-19"><a href="collating-results-and-running-gsea.html#cb161-19"></a>l3=Mouse2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(dnGenes, Mouse2Hum<span class="op">$</span>MGI.symbol)]</span>
<span id="cb161-20"><a href="collating-results-and-running-gsea.html#cb161-20"></a>dnGenesConv=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(l1)<span class="op">==</span>F, l1, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l2)<span class="op">==</span>F, l2, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l3)<span class="op">==</span>F, l3, dnGenes)))</span></code></pre></div>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="collating-results-and-running-gsea.html#cb162-1"></a><span class="kw">load</span>(<span class="st">&quot;../anntotations/ListofGeneSets2.RData&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="collating-results-and-running-gsea.html#cb163-1"></a><span class="co">#save(ListGSC, file=&quot;anntotations/ListofGeneSets2.RData&quot;)</span></span>
<span id="cb163-2"><a href="collating-results-and-running-gsea.html#cb163-2"></a>OutputMatrix=<span class="kw">lapply</span>( ListGSC, <span class="cf">function</span>(x) <span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">ncol=</span><span class="kw">length</span>(EpComp4)<span class="op">*</span><span class="dv">2</span>, <span class="dt">nrow=</span><span class="kw">length</span>(x)))</span>
<span id="cb163-3"><a href="collating-results-and-running-gsea.html#cb163-3"></a>ismr3 &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(OutputMatrix), <span class="cf">function</span>(x){ <span class="kw">row.names</span>(OutputMatrix[[x]])&lt;-<span class="kw">names</span>(ListGSC[[x]]); OutputMatrix[[x]]})</span>
<span id="cb163-4"><a href="collating-results-and-running-gsea.html#cb163-4"></a>ismrB&lt;-ismr3 </span>
<span id="cb163-5"><a href="collating-results-and-running-gsea.html#cb163-5"></a></span>
<span id="cb163-6"><a href="collating-results-and-running-gsea.html#cb163-6"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(EpComp4)){</span>
<span id="cb163-7"><a href="collating-results-and-running-gsea.html#cb163-7"></a></span>
<span id="cb163-8"><a href="collating-results-and-running-gsea.html#cb163-8"></a>hits=EpCompSig[[i]]</span>
<span id="cb163-9"><a href="collating-results-and-running-gsea.html#cb163-9"></a>hits=epGenesConv[<span class="kw">match</span>(hits, <span class="kw">rownames</span>(Epdds))]</span>
<span id="cb163-10"><a href="collating-results-and-running-gsea.html#cb163-10"></a>fcTab=EpComp4[[i]]<span class="op">$</span>log2FoldChange</span>
<span id="cb163-11"><a href="collating-results-and-running-gsea.html#cb163-11"></a><span class="kw">names</span>(fcTab)=epGenesConv</span>
<span id="cb163-12"><a href="collating-results-and-running-gsea.html#cb163-12"></a></span>
<span id="cb163-13"><a href="collating-results-and-running-gsea.html#cb163-13"></a></span>
<span id="cb163-14"><a href="collating-results-and-running-gsea.html#cb163-14"></a><span class="cf">if</span> (<span class="kw">length</span>(hits)<span class="op">&gt;</span><span class="dv">2</span>){</span>
<span id="cb163-15"><a href="collating-results-and-running-gsea.html#cb163-15"></a>  gsca=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb163-16"><a href="collating-results-and-running-gsea.html#cb163-16"></a>  gsca1 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gsca, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb163-17"><a href="collating-results-and-running-gsea.html#cb163-17"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb163-18"><a href="collating-results-and-running-gsea.html#cb163-18"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb163-19"><a href="collating-results-and-running-gsea.html#cb163-19"></a>  gsca2 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gsca1, </span>
<span id="cb163-20"><a href="collating-results-and-running-gsea.html#cb163-20"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb163-21"><a href="collating-results-and-running-gsea.html#cb163-21"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb163-22"><a href="collating-results-and-running-gsea.html#cb163-22"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb163-23"><a href="collating-results-and-running-gsea.html#cb163-23"></a>                           <span class="dt">doGSOA =</span> T)</span>
<span id="cb163-24"><a href="collating-results-and-running-gsea.html#cb163-24"></a>}<span class="cf">else</span>{</span>
<span id="cb163-25"><a href="collating-results-and-running-gsea.html#cb163-25"></a>  gsca=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC, <span class="dt">geneList=</span>fcTab)</span>
<span id="cb163-26"><a href="collating-results-and-running-gsea.html#cb163-26"></a>  gsca1 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gsca, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb163-27"><a href="collating-results-and-running-gsea.html#cb163-27"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb163-28"><a href="collating-results-and-running-gsea.html#cb163-28"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb163-29"><a href="collating-results-and-running-gsea.html#cb163-29"></a>  gsca2 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gsca1, </span>
<span id="cb163-30"><a href="collating-results-and-running-gsea.html#cb163-30"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb163-31"><a href="collating-results-and-running-gsea.html#cb163-31"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb163-32"><a href="collating-results-and-running-gsea.html#cb163-32"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb163-33"><a href="collating-results-and-running-gsea.html#cb163-33"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb163-34"><a href="collating-results-and-running-gsea.html#cb163-34"></a>}</span>
<span id="cb163-35"><a href="collating-results-and-running-gsea.html#cb163-35"></a></span>
<span id="cb163-36"><a href="collating-results-and-running-gsea.html#cb163-36"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(ListGSC)){</span>
<span id="cb163-37"><a href="collating-results-and-running-gsea.html#cb163-37"></a>  x1=gsca2<span class="op">@</span>result<span class="op">$</span>GSEA.results[[j]]</span>
<span id="cb163-38"><a href="collating-results-and-running-gsea.html#cb163-38"></a>  mx1=<span class="kw">match</span>(<span class="kw">rownames</span>(x1), <span class="kw">rownames</span>(ismr3[[j]]))</span>
<span id="cb163-39"><a href="collating-results-and-running-gsea.html#cb163-39"></a>  ismr3[[j]][ mx1,(<span class="dv">2</span><span class="op">*</span>i<span class="dv">-1</span>)<span class="op">:</span>(<span class="dv">2</span><span class="op">*</span>i)]=<span class="kw">data.matrix</span>(x1[, <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)])</span>
<span id="cb163-40"><a href="collating-results-and-running-gsea.html#cb163-40"></a>}</span>
<span id="cb163-41"><a href="collating-results-and-running-gsea.html#cb163-41"></a></span>
<span id="cb163-42"><a href="collating-results-and-running-gsea.html#cb163-42"></a><span class="cf">if</span> (<span class="kw">length</span>(hits)<span class="op">&gt;</span><span class="dv">2</span>){</span>
<span id="cb163-43"><a href="collating-results-and-running-gsea.html#cb163-43"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(ListGSC)){</span>
<span id="cb163-44"><a href="collating-results-and-running-gsea.html#cb163-44"></a>  x1=gsca2<span class="op">@</span>result<span class="op">$</span>HyperGeo.results[[j]]</span>
<span id="cb163-45"><a href="collating-results-and-running-gsea.html#cb163-45"></a>  mx1=<span class="kw">match</span>(<span class="kw">rownames</span>(x1), <span class="kw">rownames</span>(ismrB[[j]]))</span>
<span id="cb163-46"><a href="collating-results-and-running-gsea.html#cb163-46"></a>  ismrB[[j]][ mx1,(<span class="dv">2</span><span class="op">*</span>i<span class="dv">-1</span>)<span class="op">:</span>(<span class="dv">2</span><span class="op">*</span>i)]=<span class="kw">data.matrix</span>(x1[, <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">7</span>)])</span>
<span id="cb163-47"><a href="collating-results-and-running-gsea.html#cb163-47"></a>}}</span>
<span id="cb163-48"><a href="collating-results-and-running-gsea.html#cb163-48"></a></span>
<span id="cb163-49"><a href="collating-results-and-running-gsea.html#cb163-49"></a></span>
<span id="cb163-50"><a href="collating-results-and-running-gsea.html#cb163-50"></a>}</span>
<span id="cb163-51"><a href="collating-results-and-running-gsea.html#cb163-51"></a></span>
<span id="cb163-52"><a href="collating-results-and-running-gsea.html#cb163-52"></a>Epismr3 &lt;-<span class="st"> </span><span class="kw">lapply</span>(ismr3, <span class="cf">function</span>(x){ <span class="kw">colnames</span>(x)&lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">rep</span>(<span class="kw">names</span>(EpComp4), <span class="dt">each=</span><span class="dv">2</span>), <span class="kw">c</span>(<span class="st">&quot;NES&quot;</span>, <span class="st">&quot;padj&quot;</span>)); x})</span>
<span id="cb163-53"><a href="collating-results-and-running-gsea.html#cb163-53"></a><span class="kw">names</span>(Epismr3)=<span class="kw">names</span>(ListGSC)</span>
<span id="cb163-54"><a href="collating-results-and-running-gsea.html#cb163-54"></a>Epismr3&lt;-<span class="kw">lapply</span>(Epismr3, <span class="cf">function</span>(x) <span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">GeneSet=</span><span class="kw">rownames</span>(x), x)))</span>
<span id="cb163-55"><a href="collating-results-and-running-gsea.html#cb163-55"></a><span class="kw">write_xlsx</span>(Epismr3, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/DESeq/GSEA_Ep_growth_treat_%s.xlsx&quot;</span>, <span class="kw">Sys.Date</span>()))</span>
<span id="cb163-56"><a href="collating-results-and-running-gsea.html#cb163-56"></a></span>
<span id="cb163-57"><a href="collating-results-and-running-gsea.html#cb163-57"></a>EpismrHG &lt;-<span class="st"> </span><span class="kw">lapply</span>( ismrB, <span class="cf">function</span>(x){ <span class="kw">colnames</span>(x)&lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">rep</span>(<span class="kw">names</span>(EpComp4), <span class="dt">each=</span><span class="dv">2</span>), <span class="kw">c</span>(<span class="st">&quot;pval&quot;</span>, <span class="st">&quot;padj&quot;</span>)); x})</span>
<span id="cb163-58"><a href="collating-results-and-running-gsea.html#cb163-58"></a><span class="kw">names</span>(EpismrHG)=<span class="kw">names</span>(ListGSC)</span>
<span id="cb163-59"><a href="collating-results-and-running-gsea.html#cb163-59"></a>EpismrHG&lt;-<span class="kw">lapply</span>(EpismrHG, <span class="cf">function</span>(x) <span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">GeneSet=</span><span class="kw">rownames</span>(x), x)))</span>
<span id="cb163-60"><a href="collating-results-and-running-gsea.html#cb163-60"></a><span class="kw">write_xlsx</span>(EpismrHG, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/DESeq/HyperGeo_Ep_growth_treat_%s.xlsx&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="collating-results-and-running-gsea.html#cb164-1"></a><span class="co">#load(&quot;rslt/DESeq/Epithelial_fraction.RData&quot;)</span></span>
<span id="cb164-2"><a href="collating-results-and-running-gsea.html#cb164-2"></a></span>
<span id="cb164-3"><a href="collating-results-and-running-gsea.html#cb164-3"></a><span class="co">## for treatment control: big differences here which are not replicated</span></span>
<span id="cb164-4"><a href="collating-results-and-running-gsea.html#cb164-4"></a></span>
<span id="cb164-5"><a href="collating-results-and-running-gsea.html#cb164-5"></a>CDComp4=<span class="kw">list</span>()</span>
<span id="cb164-6"><a href="collating-results-and-running-gsea.html#cb164-6"></a>CDCompSig=<span class="kw">list</span>()</span>
<span id="cb164-7"><a href="collating-results-and-running-gsea.html#cb164-7"></a>CDUpDn1=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">2</span>, <span class="dt">ncol=</span><span class="dv">7</span>)</span>
<span id="cb164-8"><a href="collating-results-and-running-gsea.html#cb164-8"></a></span>
<span id="cb164-9"><a href="collating-results-and-running-gsea.html#cb164-9"></a><span class="co">#for (i in 1:4){</span></span>
<span id="cb164-10"><a href="collating-results-and-running-gsea.html#cb164-10"></a>Eres1=<span class="kw">results</span>(CDdds, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;treatA&quot;</span>, <span class="st">&quot;imm&quot;</span>, <span class="st">&quot;control&quot;</span>))</span>
<span id="cb164-11"><a href="collating-results-and-running-gsea.html#cb164-11"></a>CDComp4[[<span class="dv">1</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres1), Eres1))</span>
<span id="cb164-12"><a href="collating-results-and-running-gsea.html#cb164-12"></a>CDCompSig[[<span class="dv">1</span>]]=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres1)[<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb164-13"><a href="collating-results-and-running-gsea.html#cb164-13"></a>CDUpDn1[ ,<span class="dv">1</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres1<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb164-14"><a href="collating-results-and-running-gsea.html#cb164-14"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres1<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb164-15"><a href="collating-results-and-running-gsea.html#cb164-15"></a><span class="co">#}</span></span>
<span id="cb164-16"><a href="collating-results-and-running-gsea.html#cb164-16"></a><span class="co"># for control</span></span>
<span id="cb164-17"><a href="collating-results-and-running-gsea.html#cb164-17"></a></span>
<span id="cb164-18"><a href="collating-results-and-running-gsea.html#cb164-18"></a>cList=<span class="kw">c</span>(<span class="st">&quot;PDL1+LY&quot;</span>, <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;PDL1&quot;</span>)</span>
<span id="cb164-19"><a href="collating-results-and-running-gsea.html#cb164-19"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>){</span>
<span id="cb164-20"><a href="collating-results-and-running-gsea.html#cb164-20"></a>Eres5=<span class="kw">results</span>(CDddsTreat, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, cList[i], <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb164-21"><a href="collating-results-and-running-gsea.html#cb164-21"></a>CDComp4[[i<span class="op">+</span><span class="dv">1</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb164-22"><a href="collating-results-and-running-gsea.html#cb164-22"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb164-23"><a href="collating-results-and-running-gsea.html#cb164-23"></a>CDCompSig[[i<span class="op">+</span><span class="dv">1</span>]]=g1</span>
<span id="cb164-24"><a href="collating-results-and-running-gsea.html#cb164-24"></a>CDUpDn1[ ,i<span class="op">+</span><span class="dv">1</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb164-25"><a href="collating-results-and-running-gsea.html#cb164-25"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb164-26"><a href="collating-results-and-running-gsea.html#cb164-26"></a>}</span>
<span id="cb164-27"><a href="collating-results-and-running-gsea.html#cb164-27"></a></span>
<span id="cb164-28"><a href="collating-results-and-running-gsea.html#cb164-28"></a><span class="co"># xsearchA=c(&quot;stable.Vehicle&quot;,&quot;stable.LY&quot;,  &quot;stable.PDL1+LY&quot;, &quot;stable.PDL1&quot;)</span></span>
<span id="cb164-29"><a href="collating-results-and-running-gsea.html#cb164-29"></a><span class="co"># ysearchB=c(&quot;growing.Vehicle&quot;,&quot;growing.LY&quot;,  &quot;growing.PDL1+LY&quot;, &quot;growing.PDL1&quot;)</span></span>
<span id="cb164-30"><a href="collating-results-and-running-gsea.html#cb164-30"></a><span class="co"># </span></span>
<span id="cb164-31"><a href="collating-results-and-running-gsea.html#cb164-31"></a><span class="co"># </span></span>
<span id="cb164-32"><a href="collating-results-and-running-gsea.html#cb164-32"></a><span class="co"># for (i in 1:4){</span></span>
<span id="cb164-33"><a href="collating-results-and-running-gsea.html#cb164-33"></a><span class="co"># Eres5=results(CDddsTreatG, contrast=c(&quot;Comp8&quot;, xsearchA[i], ysearchB[i]))</span></span>
<span id="cb164-34"><a href="collating-results-and-running-gsea.html#cb164-34"></a><span class="co"># CDComp4[[i+7]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb164-35"><a href="collating-results-and-running-gsea.html#cb164-35"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb164-36"><a href="collating-results-and-running-gsea.html#cb164-36"></a><span class="co"># CDCompSig[[i+7]]=g1</span></span>
<span id="cb164-37"><a href="collating-results-and-running-gsea.html#cb164-37"></a><span class="co"># CDUpDn1[ ,i+7]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb164-38"><a href="collating-results-and-running-gsea.html#cb164-38"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb164-39"><a href="collating-results-and-running-gsea.html#cb164-39"></a><span class="co"># }</span></span>
<span id="cb164-40"><a href="collating-results-and-running-gsea.html#cb164-40"></a></span>
<span id="cb164-41"><a href="collating-results-and-running-gsea.html#cb164-41"></a><span class="co">#boxplot(assay(CDdds)[&quot;Snca&quot;, ]~CDdds$treatA+CDdds$Growth)</span></span>
<span id="cb164-42"><a href="collating-results-and-running-gsea.html#cb164-42"></a></span>
<span id="cb164-43"><a href="collating-results-and-running-gsea.html#cb164-43"></a>Eres5=<span class="kw">results</span>(CDddsGrowth, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;Growth&quot;</span>, <span class="st">&quot;stable&quot;</span>, <span class="st">&quot;growing&quot;</span>))</span>
<span id="cb164-44"><a href="collating-results-and-running-gsea.html#cb164-44"></a>CDComp4[[<span class="dv">5</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb164-45"><a href="collating-results-and-running-gsea.html#cb164-45"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb164-46"><a href="collating-results-and-running-gsea.html#cb164-46"></a>CDCompSig[[<span class="dv">5</span>]]=g1</span>
<span id="cb164-47"><a href="collating-results-and-running-gsea.html#cb164-47"></a>CDUpDn1[ ,<span class="dv">5</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb164-48"><a href="collating-results-and-running-gsea.html#cb164-48"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb164-49"><a href="collating-results-and-running-gsea.html#cb164-49"></a></span>
<span id="cb164-50"><a href="collating-results-and-running-gsea.html#cb164-50"></a></span>
<span id="cb164-51"><a href="collating-results-and-running-gsea.html#cb164-51"></a>Eres5=<span class="kw">results</span>(CDddsStrMH, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;MHcut&quot;</span>, <span class="st">&quot;res&quot;</span>, <span class="st">&quot;inf&quot;</span>))</span>
<span id="cb164-52"><a href="collating-results-and-running-gsea.html#cb164-52"></a>CDComp4[[<span class="dv">6</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb164-53"><a href="collating-results-and-running-gsea.html#cb164-53"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb164-54"><a href="collating-results-and-running-gsea.html#cb164-54"></a>CDCompSig[[<span class="dv">6</span>]]=g1</span>
<span id="cb164-55"><a href="collating-results-and-running-gsea.html#cb164-55"></a>CDUpDn1[ ,<span class="dv">6</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb164-56"><a href="collating-results-and-running-gsea.html#cb164-56"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb164-57"><a href="collating-results-and-running-gsea.html#cb164-57"></a></span>
<span id="cb164-58"><a href="collating-results-and-running-gsea.html#cb164-58"></a></span>
<span id="cb164-59"><a href="collating-results-and-running-gsea.html#cb164-59"></a><span class="co"># Eres5=results(CDddsStrIF, contrast=c(&quot;IFcut&quot;, &quot;res&quot;, &quot;inf&quot;))</span></span>
<span id="cb164-60"><a href="collating-results-and-running-gsea.html#cb164-60"></a><span class="co"># CDComp4[[14]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb164-61"><a href="collating-results-and-running-gsea.html#cb164-61"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb164-62"><a href="collating-results-and-running-gsea.html#cb164-62"></a><span class="co"># CDCompSig[[14]]=g1</span></span>
<span id="cb164-63"><a href="collating-results-and-running-gsea.html#cb164-63"></a><span class="co"># CDUpDn1[ ,14]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb164-64"><a href="collating-results-and-running-gsea.html#cb164-64"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb164-65"><a href="collating-results-and-running-gsea.html#cb164-65"></a><span class="co"># </span></span>
<span id="cb164-66"><a href="collating-results-and-running-gsea.html#cb164-66"></a><span class="co"># Eres5=results(CDddsStrknn, contrast=c(&quot;knncut&quot;, &quot;res&quot;, &quot;inf&quot;))</span></span>
<span id="cb164-67"><a href="collating-results-and-running-gsea.html#cb164-67"></a><span class="co"># CDComp4[[15]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb164-68"><a href="collating-results-and-running-gsea.html#cb164-68"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb164-69"><a href="collating-results-and-running-gsea.html#cb164-69"></a><span class="co"># CDCompSig[[15]]=g1</span></span>
<span id="cb164-70"><a href="collating-results-and-running-gsea.html#cb164-70"></a><span class="co"># CDUpDn1[ ,15]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb164-71"><a href="collating-results-and-running-gsea.html#cb164-71"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb164-72"><a href="collating-results-and-running-gsea.html#cb164-72"></a></span>
<span id="cb164-73"><a href="collating-results-and-running-gsea.html#cb164-73"></a>Eres5=<span class="kw">results</span>(CDddscd8, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;CD8FracCut&quot;</span>, <span class="st">&quot;low&quot;</span>, <span class="st">&quot;high&quot;</span>))</span>
<span id="cb164-74"><a href="collating-results-and-running-gsea.html#cb164-74"></a>CDComp4[[<span class="dv">7</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb164-75"><a href="collating-results-and-running-gsea.html#cb164-75"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb164-76"><a href="collating-results-and-running-gsea.html#cb164-76"></a>CDCompSig[[<span class="dv">7</span>]]=g1</span>
<span id="cb164-77"><a href="collating-results-and-running-gsea.html#cb164-77"></a>CDUpDn1[ ,<span class="dv">7</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb164-78"><a href="collating-results-and-running-gsea.html#cb164-78"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb164-79"><a href="collating-results-and-running-gsea.html#cb164-79"></a></span>
<span id="cb164-80"><a href="collating-results-and-running-gsea.html#cb164-80"></a><span class="co"># Eres5=results(CDddsSpatMan, contrast=c(&quot;SpatialManual&quot;, &quot;Infiltrating&quot;, &quot;restricted&quot;))</span></span>
<span id="cb164-81"><a href="collating-results-and-running-gsea.html#cb164-81"></a><span class="co"># CDComp4[[17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb164-82"><a href="collating-results-and-running-gsea.html#cb164-82"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb164-83"><a href="collating-results-and-running-gsea.html#cb164-83"></a><span class="co"># CDCompSig[[17]]=g1</span></span>
<span id="cb164-84"><a href="collating-results-and-running-gsea.html#cb164-84"></a><span class="co"># CDUpDn1[ ,17]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb164-85"><a href="collating-results-and-running-gsea.html#cb164-85"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb164-86"><a href="collating-results-and-running-gsea.html#cb164-86"></a><span class="co"># </span></span>
<span id="cb164-87"><a href="collating-results-and-running-gsea.html#cb164-87"></a><span class="co"># xsearchC=c(&quot;low.inf&quot;,&quot;high.inf&quot;,  &quot;low.inf&quot;, &quot;low.res&quot;)</span></span>
<span id="cb164-88"><a href="collating-results-and-running-gsea.html#cb164-88"></a><span class="co"># ysearchD=c(&quot;low.res&quot;,&quot;high.res&quot;, &quot;high.inf&quot;, &quot;high.res&quot;)</span></span>
<span id="cb164-89"><a href="collating-results-and-running-gsea.html#cb164-89"></a></span>
<span id="cb164-90"><a href="collating-results-and-running-gsea.html#cb164-90"></a><span class="co"># for (i in 1:4){</span></span>
<span id="cb164-91"><a href="collating-results-and-running-gsea.html#cb164-91"></a><span class="co"># Eres5=results(CDddscd8MH, contrast=c(&quot;cd8MH&quot;, xsearchC[i], ysearchD[i]))</span></span>
<span id="cb164-92"><a href="collating-results-and-running-gsea.html#cb164-92"></a><span class="co"># CDComp4[[i+17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb164-93"><a href="collating-results-and-running-gsea.html#cb164-93"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb164-94"><a href="collating-results-and-running-gsea.html#cb164-94"></a><span class="co"># CDCompSig[[i+17]]=g1</span></span>
<span id="cb164-95"><a href="collating-results-and-running-gsea.html#cb164-95"></a><span class="co"># CDUpDn1[ ,i+17]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb164-96"><a href="collating-results-and-running-gsea.html#cb164-96"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb164-97"><a href="collating-results-and-running-gsea.html#cb164-97"></a><span class="co"># }</span></span>
<span id="cb164-98"><a href="collating-results-and-running-gsea.html#cb164-98"></a><span class="co"># </span></span>
<span id="cb164-99"><a href="collating-results-and-running-gsea.html#cb164-99"></a></span>
<span id="cb164-100"><a href="collating-results-and-running-gsea.html#cb164-100"></a></span>
<span id="cb164-101"><a href="collating-results-and-running-gsea.html#cb164-101"></a><span class="kw">rownames</span>(CDUpDn1)=<span class="kw">c</span>(<span class="st">&quot;down&quot;</span>, <span class="st">&quot;up&quot;</span>)</span>
<span id="cb164-102"><a href="collating-results-and-running-gsea.html#cb164-102"></a><span class="kw">colnames</span>(CDUpDn1)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb164-103"><a href="collating-results-and-running-gsea.html#cb164-103"></a><span class="kw">names</span>(CDComp4)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb164-104"><a href="collating-results-and-running-gsea.html#cb164-104"></a><span class="kw">names</span>(CDCompSig)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb164-105"><a href="collating-results-and-running-gsea.html#cb164-105"></a></span>
<span id="cb164-106"><a href="collating-results-and-running-gsea.html#cb164-106"></a></span>
<span id="cb164-107"><a href="collating-results-and-running-gsea.html#cb164-107"></a><span class="co">## write to a xls</span></span>
<span id="cb164-108"><a href="collating-results-and-running-gsea.html#cb164-108"></a><span class="kw">write_xlsx</span>(CDComp4, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/DESeq/CD_growth_treat_%s.xlsx&quot;</span>, <span class="kw">Sys.Date</span>()))</span>
<span id="cb164-109"><a href="collating-results-and-running-gsea.html#cb164-109"></a></span>
<span id="cb164-110"><a href="collating-results-and-running-gsea.html#cb164-110"></a>OutputMatrix=<span class="kw">lapply</span>( ListGSC, <span class="cf">function</span>(x) <span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">ncol=</span><span class="kw">length</span>(CDComp4)<span class="op">*</span><span class="dv">2</span>, <span class="dt">nrow=</span><span class="kw">length</span>(x)))</span>
<span id="cb164-111"><a href="collating-results-and-running-gsea.html#cb164-111"></a>CDismr3 &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(OutputMatrix), <span class="cf">function</span>(x){ <span class="kw">row.names</span>(OutputMatrix[[x]])&lt;-<span class="kw">names</span>(ListGSC[[x]]); OutputMatrix[[x]]})</span>
<span id="cb164-112"><a href="collating-results-and-running-gsea.html#cb164-112"></a>CDismrB=CDismr3</span>
<span id="cb164-113"><a href="collating-results-and-running-gsea.html#cb164-113"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(CDComp4)){</span>
<span id="cb164-114"><a href="collating-results-and-running-gsea.html#cb164-114"></a></span>
<span id="cb164-115"><a href="collating-results-and-running-gsea.html#cb164-115"></a>hits=CDCompSig[[i]]</span>
<span id="cb164-116"><a href="collating-results-and-running-gsea.html#cb164-116"></a>hits=cdGenesConv[<span class="kw">match</span>(hits, <span class="kw">rownames</span>(CDdds))]</span>
<span id="cb164-117"><a href="collating-results-and-running-gsea.html#cb164-117"></a>fcTab=CDComp4[[i]]<span class="op">$</span>log2FoldChange</span>
<span id="cb164-118"><a href="collating-results-and-running-gsea.html#cb164-118"></a><span class="kw">names</span>(fcTab)=cdGenesConv</span>
<span id="cb164-119"><a href="collating-results-and-running-gsea.html#cb164-119"></a></span>
<span id="cb164-120"><a href="collating-results-and-running-gsea.html#cb164-120"></a></span>
<span id="cb164-121"><a href="collating-results-and-running-gsea.html#cb164-121"></a><span class="cf">if</span> (<span class="kw">length</span>(hits)<span class="op">&gt;</span><span class="dv">3</span>){</span>
<span id="cb164-122"><a href="collating-results-and-running-gsea.html#cb164-122"></a>  gsca=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb164-123"><a href="collating-results-and-running-gsea.html#cb164-123"></a>gsca1 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gsca, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb164-124"><a href="collating-results-and-running-gsea.html#cb164-124"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb164-125"><a href="collating-results-and-running-gsea.html#cb164-125"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb164-126"><a href="collating-results-and-running-gsea.html#cb164-126"></a>  gsca2 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gsca1, </span>
<span id="cb164-127"><a href="collating-results-and-running-gsea.html#cb164-127"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb164-128"><a href="collating-results-and-running-gsea.html#cb164-128"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb164-129"><a href="collating-results-and-running-gsea.html#cb164-129"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb164-130"><a href="collating-results-and-running-gsea.html#cb164-130"></a>                           <span class="dt">doGSOA =</span> T)</span>
<span id="cb164-131"><a href="collating-results-and-running-gsea.html#cb164-131"></a>}<span class="cf">else</span>{</span>
<span id="cb164-132"><a href="collating-results-and-running-gsea.html#cb164-132"></a>  gsca=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab)</span>
<span id="cb164-133"><a href="collating-results-and-running-gsea.html#cb164-133"></a>gsca1 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gsca, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb164-134"><a href="collating-results-and-running-gsea.html#cb164-134"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb164-135"><a href="collating-results-and-running-gsea.html#cb164-135"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb164-136"><a href="collating-results-and-running-gsea.html#cb164-136"></a>  gsca2 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gsca1, </span>
<span id="cb164-137"><a href="collating-results-and-running-gsea.html#cb164-137"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb164-138"><a href="collating-results-and-running-gsea.html#cb164-138"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb164-139"><a href="collating-results-and-running-gsea.html#cb164-139"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb164-140"><a href="collating-results-and-running-gsea.html#cb164-140"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb164-141"><a href="collating-results-and-running-gsea.html#cb164-141"></a>}</span>
<span id="cb164-142"><a href="collating-results-and-running-gsea.html#cb164-142"></a></span>
<span id="cb164-143"><a href="collating-results-and-running-gsea.html#cb164-143"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(ListGSC)){</span>
<span id="cb164-144"><a href="collating-results-and-running-gsea.html#cb164-144"></a>  x1=gsca2<span class="op">@</span>result<span class="op">$</span>GSEA.results[[j]]</span>
<span id="cb164-145"><a href="collating-results-and-running-gsea.html#cb164-145"></a>  mx1=<span class="kw">match</span>(<span class="kw">rownames</span>(x1), <span class="kw">rownames</span>(CDismr3[[j]]))</span>
<span id="cb164-146"><a href="collating-results-and-running-gsea.html#cb164-146"></a>  CDismr3[[j]][ mx1,(<span class="dv">2</span><span class="op">*</span>i<span class="dv">-1</span>)<span class="op">:</span>(<span class="dv">2</span><span class="op">*</span>i)]=<span class="kw">data.matrix</span>(x1[, <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)])</span>
<span id="cb164-147"><a href="collating-results-and-running-gsea.html#cb164-147"></a>}</span>
<span id="cb164-148"><a href="collating-results-and-running-gsea.html#cb164-148"></a></span>
<span id="cb164-149"><a href="collating-results-and-running-gsea.html#cb164-149"></a><span class="cf">if</span> (<span class="kw">length</span>(hits)<span class="op">&gt;</span><span class="dv">3</span>){</span>
<span id="cb164-150"><a href="collating-results-and-running-gsea.html#cb164-150"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(ListGSC)){</span>
<span id="cb164-151"><a href="collating-results-and-running-gsea.html#cb164-151"></a>  x1=gsca2<span class="op">@</span>result<span class="op">$</span>HyperGeo.results[[j]]</span>
<span id="cb164-152"><a href="collating-results-and-running-gsea.html#cb164-152"></a>  mx1=<span class="kw">match</span>(<span class="kw">rownames</span>(x1), <span class="kw">rownames</span>(CDismrB[[j]]))</span>
<span id="cb164-153"><a href="collating-results-and-running-gsea.html#cb164-153"></a>  CDismrB[[j]][ mx1,(<span class="dv">2</span><span class="op">*</span>i<span class="dv">-1</span>)<span class="op">:</span>(<span class="dv">2</span><span class="op">*</span>i)]=<span class="kw">data.matrix</span>(x1[, <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">7</span>)])</span>
<span id="cb164-154"><a href="collating-results-and-running-gsea.html#cb164-154"></a>}}</span>
<span id="cb164-155"><a href="collating-results-and-running-gsea.html#cb164-155"></a></span>
<span id="cb164-156"><a href="collating-results-and-running-gsea.html#cb164-156"></a>}</span>
<span id="cb164-157"><a href="collating-results-and-running-gsea.html#cb164-157"></a></span>
<span id="cb164-158"><a href="collating-results-and-running-gsea.html#cb164-158"></a>CDismr3 &lt;-<span class="st"> </span><span class="kw">lapply</span>(CDismr3, <span class="cf">function</span>(x){ <span class="kw">colnames</span>(x)&lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">rep</span>(<span class="kw">names</span>(CDComp4), <span class="dt">each=</span><span class="dv">2</span>), <span class="kw">c</span>(<span class="st">&quot;NES&quot;</span>, <span class="st">&quot;padj&quot;</span>)); x})</span>
<span id="cb164-159"><a href="collating-results-and-running-gsea.html#cb164-159"></a><span class="kw">names</span>(CDismr3)=<span class="kw">names</span>(ListGSC)</span>
<span id="cb164-160"><a href="collating-results-and-running-gsea.html#cb164-160"></a>CDismr3&lt;-<span class="kw">lapply</span>(CDismr3, <span class="cf">function</span>(x) <span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">GeneSet=</span><span class="kw">rownames</span>(x), x)))</span>
<span id="cb164-161"><a href="collating-results-and-running-gsea.html#cb164-161"></a><span class="kw">write_xlsx</span>(CDismr3, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/DESeq/GSEA_CD_growth_treat.xlsx&quot;</span>))</span>
<span id="cb164-162"><a href="collating-results-and-running-gsea.html#cb164-162"></a></span>
<span id="cb164-163"><a href="collating-results-and-running-gsea.html#cb164-163"></a></span>
<span id="cb164-164"><a href="collating-results-and-running-gsea.html#cb164-164"></a>CDismrB &lt;-<span class="st"> </span><span class="kw">lapply</span>(CDismrB, <span class="cf">function</span>(x){ <span class="kw">colnames</span>(x)&lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">rep</span>(<span class="kw">names</span>(CDComp4), <span class="dt">each=</span><span class="dv">2</span>), <span class="kw">c</span>(<span class="st">&quot;pval&quot;</span>, <span class="st">&quot;padj&quot;</span>)); x})</span>
<span id="cb164-165"><a href="collating-results-and-running-gsea.html#cb164-165"></a><span class="kw">names</span>(CDismrB)=<span class="kw">names</span>(ListGSC)</span>
<span id="cb164-166"><a href="collating-results-and-running-gsea.html#cb164-166"></a>CDismrB&lt;-<span class="kw">lapply</span>(CDismrB, <span class="cf">function</span>(x) <span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">GeneSet=</span><span class="kw">rownames</span>(x), x)))</span>
<span id="cb164-167"><a href="collating-results-and-running-gsea.html#cb164-167"></a><span class="kw">write_xlsx</span>(CDismrB, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/DESeq/HyperGeom_CD_growth_treat.xlsx&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="collating-results-and-running-gsea.html#cb165-1"></a><span class="co">#load(&quot;rslt/DESeq/Epithelial_fraction.RData&quot;)</span></span>
<span id="cb165-2"><a href="collating-results-and-running-gsea.html#cb165-2"></a></span>
<span id="cb165-3"><a href="collating-results-and-running-gsea.html#cb165-3"></a><span class="co"># xsearch=c(&quot;stablecontrol&quot;,&quot;growingimm&quot;,  &quot;stableimm&quot;, &quot;stableimm&quot;)</span></span>
<span id="cb165-4"><a href="collating-results-and-running-gsea.html#cb165-4"></a><span class="co"># ysearch=c(&quot;growingcontrol&quot;,&quot;growingcontrol&quot;,  &quot;stablecontrol&quot;, &quot;growingimm&quot;)</span></span>
<span id="cb165-5"><a href="collating-results-and-running-gsea.html#cb165-5"></a></span>
<span id="cb165-6"><a href="collating-results-and-running-gsea.html#cb165-6"></a><span class="co">## for treatment control: big differences here which are not replicated</span></span>
<span id="cb165-7"><a href="collating-results-and-running-gsea.html#cb165-7"></a></span>
<span id="cb165-8"><a href="collating-results-and-running-gsea.html#cb165-8"></a>DNComp4=<span class="kw">list</span>()</span>
<span id="cb165-9"><a href="collating-results-and-running-gsea.html#cb165-9"></a>DNCompSig=<span class="kw">list</span>()</span>
<span id="cb165-10"><a href="collating-results-and-running-gsea.html#cb165-10"></a>DNUpDn1=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">2</span>, <span class="dt">ncol=</span><span class="dv">7</span>)</span>
<span id="cb165-11"><a href="collating-results-and-running-gsea.html#cb165-11"></a></span>
<span id="cb165-12"><a href="collating-results-and-running-gsea.html#cb165-12"></a><span class="co">#for (i in 1:4){</span></span>
<span id="cb165-13"><a href="collating-results-and-running-gsea.html#cb165-13"></a>Eres1=<span class="kw">results</span>(DNdds, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;treatA&quot;</span>, <span class="st">&quot;imm&quot;</span>, <span class="st">&quot;control&quot;</span>))</span>
<span id="cb165-14"><a href="collating-results-and-running-gsea.html#cb165-14"></a>DNComp4[[<span class="dv">1</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres1), Eres1))</span>
<span id="cb165-15"><a href="collating-results-and-running-gsea.html#cb165-15"></a>DNCompSig[[<span class="dv">1</span>]]=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres1)[<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb165-16"><a href="collating-results-and-running-gsea.html#cb165-16"></a>DNUpDn1[ ,<span class="dv">1</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres1<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb165-17"><a href="collating-results-and-running-gsea.html#cb165-17"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres1<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb165-18"><a href="collating-results-and-running-gsea.html#cb165-18"></a><span class="co">#}</span></span>
<span id="cb165-19"><a href="collating-results-and-running-gsea.html#cb165-19"></a><span class="co"># for control</span></span>
<span id="cb165-20"><a href="collating-results-and-running-gsea.html#cb165-20"></a></span>
<span id="cb165-21"><a href="collating-results-and-running-gsea.html#cb165-21"></a>cList=<span class="kw">c</span>(<span class="st">&quot;PDL1+LY&quot;</span>, <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;PDL1&quot;</span>)</span>
<span id="cb165-22"><a href="collating-results-and-running-gsea.html#cb165-22"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>){</span>
<span id="cb165-23"><a href="collating-results-and-running-gsea.html#cb165-23"></a>Eres5=<span class="kw">results</span>(DNddsTreat, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, cList[i], <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb165-24"><a href="collating-results-and-running-gsea.html#cb165-24"></a>DNComp4[[i<span class="op">+</span><span class="dv">1</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb165-25"><a href="collating-results-and-running-gsea.html#cb165-25"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb165-26"><a href="collating-results-and-running-gsea.html#cb165-26"></a>DNCompSig[[i<span class="op">+</span><span class="dv">1</span>]]=g1</span>
<span id="cb165-27"><a href="collating-results-and-running-gsea.html#cb165-27"></a>DNUpDn1[ ,i<span class="op">+</span><span class="dv">1</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb165-28"><a href="collating-results-and-running-gsea.html#cb165-28"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb165-29"><a href="collating-results-and-running-gsea.html#cb165-29"></a>}</span>
<span id="cb165-30"><a href="collating-results-and-running-gsea.html#cb165-30"></a></span>
<span id="cb165-31"><a href="collating-results-and-running-gsea.html#cb165-31"></a><span class="co"># xsearchA=c(&quot;stable.Vehicle&quot;,&quot;stable.LY&quot;,  &quot;stable.PDL1+LY&quot;, &quot;stable.PDL1&quot;)</span></span>
<span id="cb165-32"><a href="collating-results-and-running-gsea.html#cb165-32"></a><span class="co"># ysearchB=c(&quot;growing.Vehicle&quot;,&quot;growing.LY&quot;,  &quot;growing.PDL1+LY&quot;, &quot;growing.PDL1&quot;)</span></span>
<span id="cb165-33"><a href="collating-results-and-running-gsea.html#cb165-33"></a><span class="co"># </span></span>
<span id="cb165-34"><a href="collating-results-and-running-gsea.html#cb165-34"></a><span class="co"># </span></span>
<span id="cb165-35"><a href="collating-results-and-running-gsea.html#cb165-35"></a><span class="co"># for (i in 1:4){</span></span>
<span id="cb165-36"><a href="collating-results-and-running-gsea.html#cb165-36"></a><span class="co"># Eres5=results(DNddsTreatG, contrast=c(&quot;Comp8&quot;, xsearchA[i], ysearchB[i]))</span></span>
<span id="cb165-37"><a href="collating-results-and-running-gsea.html#cb165-37"></a><span class="co"># DNComp4[[i+7]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb165-38"><a href="collating-results-and-running-gsea.html#cb165-38"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb165-39"><a href="collating-results-and-running-gsea.html#cb165-39"></a><span class="co"># DNCompSig[[i+7]]=g1</span></span>
<span id="cb165-40"><a href="collating-results-and-running-gsea.html#cb165-40"></a><span class="co"># DNUpDn1[ ,i+7]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb165-41"><a href="collating-results-and-running-gsea.html#cb165-41"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb165-42"><a href="collating-results-and-running-gsea.html#cb165-42"></a><span class="co"># }</span></span>
<span id="cb165-43"><a href="collating-results-and-running-gsea.html#cb165-43"></a></span>
<span id="cb165-44"><a href="collating-results-and-running-gsea.html#cb165-44"></a><span class="co">#</span></span>
<span id="cb165-45"><a href="collating-results-and-running-gsea.html#cb165-45"></a>Eres5=<span class="kw">results</span>(DNddsGrowth, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;Growth&quot;</span>, <span class="st">&quot;stable&quot;</span>, <span class="st">&quot;growing&quot;</span>))</span>
<span id="cb165-46"><a href="collating-results-and-running-gsea.html#cb165-46"></a>DNComp4[[<span class="dv">5</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb165-47"><a href="collating-results-and-running-gsea.html#cb165-47"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb165-48"><a href="collating-results-and-running-gsea.html#cb165-48"></a>DNCompSig[[<span class="dv">5</span>]]=g1</span>
<span id="cb165-49"><a href="collating-results-and-running-gsea.html#cb165-49"></a>DNUpDn1[ ,<span class="dv">5</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb165-50"><a href="collating-results-and-running-gsea.html#cb165-50"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb165-51"><a href="collating-results-and-running-gsea.html#cb165-51"></a></span>
<span id="cb165-52"><a href="collating-results-and-running-gsea.html#cb165-52"></a></span>
<span id="cb165-53"><a href="collating-results-and-running-gsea.html#cb165-53"></a>Eres5=<span class="kw">results</span>(DNddsStrMH, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;MHcut&quot;</span>, <span class="st">&quot;res&quot;</span>, <span class="st">&quot;inf&quot;</span>))</span>
<span id="cb165-54"><a href="collating-results-and-running-gsea.html#cb165-54"></a>DNComp4[[<span class="dv">6</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb165-55"><a href="collating-results-and-running-gsea.html#cb165-55"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb165-56"><a href="collating-results-and-running-gsea.html#cb165-56"></a>DNCompSig[[<span class="dv">6</span>]]=g1</span>
<span id="cb165-57"><a href="collating-results-and-running-gsea.html#cb165-57"></a>DNUpDn1[ ,<span class="dv">6</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb165-58"><a href="collating-results-and-running-gsea.html#cb165-58"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb165-59"><a href="collating-results-and-running-gsea.html#cb165-59"></a></span>
<span id="cb165-60"><a href="collating-results-and-running-gsea.html#cb165-60"></a></span>
<span id="cb165-61"><a href="collating-results-and-running-gsea.html#cb165-61"></a><span class="co"># Eres5=results(DNddsStrIF, contrast=c(&quot;IFcut&quot;, &quot;res&quot;, &quot;inf&quot;))</span></span>
<span id="cb165-62"><a href="collating-results-and-running-gsea.html#cb165-62"></a><span class="co"># DNComp4[[14]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb165-63"><a href="collating-results-and-running-gsea.html#cb165-63"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb165-64"><a href="collating-results-and-running-gsea.html#cb165-64"></a><span class="co"># DNCompSig[[14]]=g1</span></span>
<span id="cb165-65"><a href="collating-results-and-running-gsea.html#cb165-65"></a><span class="co"># DNUpDn1[ ,14]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb165-66"><a href="collating-results-and-running-gsea.html#cb165-66"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb165-67"><a href="collating-results-and-running-gsea.html#cb165-67"></a><span class="co"># </span></span>
<span id="cb165-68"><a href="collating-results-and-running-gsea.html#cb165-68"></a><span class="co"># Eres5=results(DNddsStrknn, contrast=c(&quot;knncut&quot;, &quot;res&quot;, &quot;inf&quot;))</span></span>
<span id="cb165-69"><a href="collating-results-and-running-gsea.html#cb165-69"></a><span class="co"># DNComp4[[15]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb165-70"><a href="collating-results-and-running-gsea.html#cb165-70"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb165-71"><a href="collating-results-and-running-gsea.html#cb165-71"></a><span class="co"># DNCompSig[[15]]=g1</span></span>
<span id="cb165-72"><a href="collating-results-and-running-gsea.html#cb165-72"></a><span class="co"># DNUpDn1[ ,15]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb165-73"><a href="collating-results-and-running-gsea.html#cb165-73"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb165-74"><a href="collating-results-and-running-gsea.html#cb165-74"></a></span>
<span id="cb165-75"><a href="collating-results-and-running-gsea.html#cb165-75"></a>Eres5=<span class="kw">results</span>(DNddscd8, <span class="dt">contrast=</span><span class="kw">c</span>(<span class="st">&quot;CD8FracCut&quot;</span>, <span class="st">&quot;low&quot;</span>, <span class="st">&quot;high&quot;</span>))</span>
<span id="cb165-76"><a href="collating-results-and-running-gsea.html#cb165-76"></a>DNComp4[[<span class="dv">7</span>]]=<span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">Gene=</span><span class="kw">rownames</span>(Eres5), Eres5))</span>
<span id="cb165-77"><a href="collating-results-and-running-gsea.html#cb165-77"></a>g1=<span class="kw">sort</span>(<span class="kw">rownames</span>(Eres5)[<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span>)])</span>
<span id="cb165-78"><a href="collating-results-and-running-gsea.html#cb165-78"></a>DNCompSig[[<span class="dv">7</span>]]=g1</span>
<span id="cb165-79"><a href="collating-results-and-running-gsea.html#cb165-79"></a>DNUpDn1[ ,<span class="dv">7</span>]=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)),</span>
<span id="cb165-80"><a href="collating-results-and-running-gsea.html#cb165-80"></a>             <span class="kw">length</span>(<span class="kw">which</span>(Eres5<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>Eres5<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)))</span>
<span id="cb165-81"><a href="collating-results-and-running-gsea.html#cb165-81"></a></span>
<span id="cb165-82"><a href="collating-results-and-running-gsea.html#cb165-82"></a><span class="co"># Eres5=results(DNddsSpatMan, contrast=c(&quot;SpatialManual&quot;, &quot;Infiltrating&quot;, &quot;restricted&quot;))</span></span>
<span id="cb165-83"><a href="collating-results-and-running-gsea.html#cb165-83"></a><span class="co"># DNComp4[[17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb165-84"><a href="collating-results-and-running-gsea.html#cb165-84"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb165-85"><a href="collating-results-and-running-gsea.html#cb165-85"></a><span class="co"># DNCompSig[[17]]=g1</span></span>
<span id="cb165-86"><a href="collating-results-and-running-gsea.html#cb165-86"></a><span class="co"># DNUpDn1[ ,17]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb165-87"><a href="collating-results-and-running-gsea.html#cb165-87"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb165-88"><a href="collating-results-and-running-gsea.html#cb165-88"></a><span class="co"># </span></span>
<span id="cb165-89"><a href="collating-results-and-running-gsea.html#cb165-89"></a><span class="co"># xsearchC=c(&quot;low.inf&quot;,&quot;high.inf&quot;,  &quot;low.inf&quot;, &quot;low.res&quot;)</span></span>
<span id="cb165-90"><a href="collating-results-and-running-gsea.html#cb165-90"></a><span class="co"># ysearchD=c(&quot;low.res&quot;,&quot;high.res&quot;, &quot;high.inf&quot;, &quot;high.res&quot;)</span></span>
<span id="cb165-91"><a href="collating-results-and-running-gsea.html#cb165-91"></a><span class="co"># </span></span>
<span id="cb165-92"><a href="collating-results-and-running-gsea.html#cb165-92"></a><span class="co"># for (i in 1:4){</span></span>
<span id="cb165-93"><a href="collating-results-and-running-gsea.html#cb165-93"></a><span class="co"># Eres5=results(DNddscd8MH, contrast=c(&quot;cd8MH&quot;, xsearchC[i], ysearchD[i]))</span></span>
<span id="cb165-94"><a href="collating-results-and-running-gsea.html#cb165-94"></a><span class="co"># DNComp4[[i+17]]=as.data.frame(cbind(Gene=rownames(Eres5), Eres5))</span></span>
<span id="cb165-95"><a href="collating-results-and-running-gsea.html#cb165-95"></a><span class="co"># g1=sort(rownames(Eres5)[which(Eres5$padj&lt;0.1)])</span></span>
<span id="cb165-96"><a href="collating-results-and-running-gsea.html#cb165-96"></a><span class="co"># DNCompSig[[i+17]]=g1</span></span>
<span id="cb165-97"><a href="collating-results-and-running-gsea.html#cb165-97"></a><span class="co"># DNUpDn1[ ,i+17]=c(length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&lt;0)),</span></span>
<span id="cb165-98"><a href="collating-results-and-running-gsea.html#cb165-98"></a><span class="co">#              length(which(Eres5$padj&lt;0.1 &amp; Eres5$log2FoldChange&gt;0)))</span></span>
<span id="cb165-99"><a href="collating-results-and-running-gsea.html#cb165-99"></a><span class="co"># }</span></span>
<span id="cb165-100"><a href="collating-results-and-running-gsea.html#cb165-100"></a></span>
<span id="cb165-101"><a href="collating-results-and-running-gsea.html#cb165-101"></a></span>
<span id="cb165-102"><a href="collating-results-and-running-gsea.html#cb165-102"></a><span class="kw">rownames</span>(DNUpDn1)=<span class="kw">c</span>(<span class="st">&quot;down&quot;</span>, <span class="st">&quot;up&quot;</span>)</span>
<span id="cb165-103"><a href="collating-results-and-running-gsea.html#cb165-103"></a><span class="kw">colnames</span>(DNUpDn1)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb165-104"><a href="collating-results-and-running-gsea.html#cb165-104"></a><span class="kw">names</span>(DNComp4)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb165-105"><a href="collating-results-and-running-gsea.html#cb165-105"></a><span class="kw">names</span>(DNCompSig)=<span class="kw">c</span>(<span class="st">&quot;imm_vs_control&quot;</span>, <span class="kw">paste</span>(cList, <span class="st">&quot;_vs_Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>),<span class="st">&quot;stable_vs_growing&quot;</span>, <span class="st">&quot;res_vs_inf.MH&quot;</span>,  <span class="st">&quot;lo_vs_hi_cd8&quot;</span>)</span>
<span id="cb165-106"><a href="collating-results-and-running-gsea.html#cb165-106"></a></span>
<span id="cb165-107"><a href="collating-results-and-running-gsea.html#cb165-107"></a></span>
<span id="cb165-108"><a href="collating-results-and-running-gsea.html#cb165-108"></a><span class="co"># rownames(DNUpDn1)=c(&quot;down&quot;, &quot;up&quot;)</span></span>
<span id="cb165-109"><a href="collating-results-and-running-gsea.html#cb165-109"></a><span class="co"># colnames(DNUpDn1)=c(paste(xsearch, &quot;vs&quot;, ysearch, sep=&quot;_&quot;), paste(cList, &quot;_vs_Vehicle&quot;, sep=&quot;&quot;), paste(xsearchA, &quot;vs&quot;, ysearchB, sep=&quot;_&quot;), &quot;stable_vs_growing&quot;, &quot;res_vs_inf.MH&quot;, &quot;res_vs_inf.IF&quot;, &quot;res_vs_inf.knn&quot;, &quot;lo_vs_hi_cd8&quot;, &quot;inf_vs_res.SpatMan&quot;, &quot;low.inf_vs_low.res.cd8.mh&quot;,  &quot;hi.inf_vs_hi.res.cd8.mh&quot;,  &quot;low.inf_vs_hi.inf.cd8.mh&quot;,  &quot;low.res_vs_hi.res.cd8.mh&quot;)</span></span>
<span id="cb165-110"><a href="collating-results-and-running-gsea.html#cb165-110"></a><span class="co"># </span></span>
<span id="cb165-111"><a href="collating-results-and-running-gsea.html#cb165-111"></a><span class="co">#   #c(paste(xsearch, &quot;vs&quot;, ysearch, sep=&quot;_&quot;), paste(cList, &quot;_vs_Vehicle&quot;, sep=&quot;&quot;), paste(xsearchA, &quot;vs&quot;, ysearchB, sep=&quot;_&quot;),&quot;stable_vs_growing&quot;, &quot;res_vs_inf.MH&quot;, &quot;res_vs_inf.IF&quot;)</span></span>
<span id="cb165-112"><a href="collating-results-and-running-gsea.html#cb165-112"></a><span class="co"># </span></span>
<span id="cb165-113"><a href="collating-results-and-running-gsea.html#cb165-113"></a><span class="co"># </span></span>
<span id="cb165-114"><a href="collating-results-and-running-gsea.html#cb165-114"></a><span class="co"># names(DNComp4)=c(paste(xsearch, &quot;vs&quot;, ysearch, sep=&quot;_&quot;), paste(cList, &quot;_vs_Vehicle&quot;, sep=&quot;&quot;), paste(xsearchA, &quot;vs&quot;, ysearchB, sep=&quot;_&quot;), &quot;stable_vs_growing&quot;, &quot;res_vs_inf.MH&quot;, &quot;res_vs_inf.IF&quot;, &quot;res_vs_inf.knn&quot;, &quot;lo_vs_hi_cd8&quot;, &quot;inf_vs_res.SpatMan&quot;, &quot;low.inf_vs_low.res.cd8.mh&quot;,  &quot;hi.inf_vs_hi.res.cd8.mh&quot;,  &quot;low.inf_vs_hi.inf.cd8.mh&quot;,  &quot;low.res_vs_hi.res.cd8.mh&quot;)</span></span>
<span id="cb165-115"><a href="collating-results-and-running-gsea.html#cb165-115"></a><span class="co"># </span></span>
<span id="cb165-116"><a href="collating-results-and-running-gsea.html#cb165-116"></a><span class="co">#   #c(paste(xsearch, &quot;vs&quot;, ysearch, sep=&quot;_&quot;), paste(cList, &quot;_vs_Vehicle&quot;, sep=&quot;&quot;), paste(xsearchA, &quot;vs&quot;, ysearchB, sep=&quot;_&quot;),&quot;stable_vs_growing&quot;, &quot;res_vs_inf.MH&quot;, &quot;res_vs_inf.IF&quot;)</span></span>
<span id="cb165-117"><a href="collating-results-and-running-gsea.html#cb165-117"></a><span class="co"># names(DNCompSig)=c(paste(xsearch, &quot;vs&quot;, ysearch, sep=&quot;_&quot;), paste(cList, &quot;_vs_Vehicle&quot;, sep=&quot;&quot;), paste(xsearchA, &quot;vs&quot;, ysearchB, sep=&quot;_&quot;), &quot;stable_vs_growing&quot;, &quot;res_vs_inf.MH&quot;, &quot;res_vs_inf.IF&quot;, &quot;res_vs_inf.knn&quot;, &quot;lo_vs_hi_cd8&quot;, &quot;inf_vs_res.SpatMan&quot;, &quot;low.inf_vs_low.res.cd8.mh&quot;,  &quot;hi.inf_vs_hi.res.cd8.mh&quot;,  &quot;low.inf_vs_hi.inf.cd8.mh&quot;,  &quot;low.res_vs_hi.res.cd8.mh&quot;)</span></span>
<span id="cb165-118"><a href="collating-results-and-running-gsea.html#cb165-118"></a></span>
<span id="cb165-119"><a href="collating-results-and-running-gsea.html#cb165-119"></a><span class="co">#c(paste(xsearch, &quot;vs&quot;, ysearch, sep=&quot;_&quot;), paste(cList, &quot;_vs_Vehicle&quot;, sep=&quot;&quot;), paste(xsearchA, &quot;vs&quot;, ysearchB, sep=&quot;_&quot;),&quot;stable_vs_growing&quot;, &quot;res_vs_inf.MH&quot;, &quot;res_vs_inf.IF&quot;)</span></span>
<span id="cb165-120"><a href="collating-results-and-running-gsea.html#cb165-120"></a></span>
<span id="cb165-121"><a href="collating-results-and-running-gsea.html#cb165-121"></a><span class="co">## write to a xls</span></span>
<span id="cb165-122"><a href="collating-results-and-running-gsea.html#cb165-122"></a><span class="kw">write_xlsx</span>(DNComp4, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/DN_growth_treat_%s.xlsx&quot;</span>, <span class="kw">Sys.Date</span>()))</span>
<span id="cb165-123"><a href="collating-results-and-running-gsea.html#cb165-123"></a></span>
<span id="cb165-124"><a href="collating-results-and-running-gsea.html#cb165-124"></a>OutputMatrix=<span class="kw">lapply</span>( ListGSC, <span class="cf">function</span>(x) <span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">ncol=</span><span class="kw">length</span>(DNComp4)<span class="op">*</span><span class="dv">2</span>, <span class="dt">nrow=</span><span class="kw">length</span>(x)))</span>
<span id="cb165-125"><a href="collating-results-and-running-gsea.html#cb165-125"></a>DNismr3 &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(OutputMatrix), <span class="cf">function</span>(x){ <span class="kw">row.names</span>(OutputMatrix[[x]])&lt;-<span class="kw">names</span>(ListGSC[[x]]); OutputMatrix[[x]]})</span>
<span id="cb165-126"><a href="collating-results-and-running-gsea.html#cb165-126"></a>DNismrB=DNismr3</span>
<span id="cb165-127"><a href="collating-results-and-running-gsea.html#cb165-127"></a></span>
<span id="cb165-128"><a href="collating-results-and-running-gsea.html#cb165-128"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(DNComp4)){</span>
<span id="cb165-129"><a href="collating-results-and-running-gsea.html#cb165-129"></a></span>
<span id="cb165-130"><a href="collating-results-and-running-gsea.html#cb165-130"></a>hits=DNCompSig[[i]]</span>
<span id="cb165-131"><a href="collating-results-and-running-gsea.html#cb165-131"></a>hits=dnGenesConv[<span class="kw">match</span>(hits, <span class="kw">rownames</span>(DNdds))]</span>
<span id="cb165-132"><a href="collating-results-and-running-gsea.html#cb165-132"></a>fcTab=DNComp4[[i]]<span class="op">$</span>log2FoldChange</span>
<span id="cb165-133"><a href="collating-results-and-running-gsea.html#cb165-133"></a><span class="kw">names</span>(fcTab)=dnGenesConv</span>
<span id="cb165-134"><a href="collating-results-and-running-gsea.html#cb165-134"></a></span>
<span id="cb165-135"><a href="collating-results-and-running-gsea.html#cb165-135"></a>gsca=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb165-136"><a href="collating-results-and-running-gsea.html#cb165-136"></a>gsca1 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gsca, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb165-137"><a href="collating-results-and-running-gsea.html#cb165-137"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb165-138"><a href="collating-results-and-running-gsea.html#cb165-138"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb165-139"><a href="collating-results-and-running-gsea.html#cb165-139"></a><span class="cf">if</span> (<span class="kw">length</span>(hits)<span class="op">&gt;</span><span class="dv">0</span>){</span>
<span id="cb165-140"><a href="collating-results-and-running-gsea.html#cb165-140"></a>  gsca2 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gsca1, </span>
<span id="cb165-141"><a href="collating-results-and-running-gsea.html#cb165-141"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb165-142"><a href="collating-results-and-running-gsea.html#cb165-142"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb165-143"><a href="collating-results-and-running-gsea.html#cb165-143"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb165-144"><a href="collating-results-and-running-gsea.html#cb165-144"></a>                           <span class="dt">doGSOA =</span> T)</span>
<span id="cb165-145"><a href="collating-results-and-running-gsea.html#cb165-145"></a>}<span class="cf">else</span>{</span>
<span id="cb165-146"><a href="collating-results-and-running-gsea.html#cb165-146"></a>  gsca2 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gsca1, </span>
<span id="cb165-147"><a href="collating-results-and-running-gsea.html#cb165-147"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb165-148"><a href="collating-results-and-running-gsea.html#cb165-148"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb165-149"><a href="collating-results-and-running-gsea.html#cb165-149"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb165-150"><a href="collating-results-and-running-gsea.html#cb165-150"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb165-151"><a href="collating-results-and-running-gsea.html#cb165-151"></a>}</span>
<span id="cb165-152"><a href="collating-results-and-running-gsea.html#cb165-152"></a></span>
<span id="cb165-153"><a href="collating-results-and-running-gsea.html#cb165-153"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(ListGSC)){</span>
<span id="cb165-154"><a href="collating-results-and-running-gsea.html#cb165-154"></a>  x1=gsca2<span class="op">@</span>result<span class="op">$</span>GSEA.results[[j]]</span>
<span id="cb165-155"><a href="collating-results-and-running-gsea.html#cb165-155"></a>  mx1=<span class="kw">match</span>(<span class="kw">rownames</span>(x1), <span class="kw">rownames</span>(DNismr3[[j]]))</span>
<span id="cb165-156"><a href="collating-results-and-running-gsea.html#cb165-156"></a>  DNismr3[[j]][ mx1,(<span class="dv">2</span><span class="op">*</span>i<span class="dv">-1</span>)<span class="op">:</span>(<span class="dv">2</span><span class="op">*</span>i)]=<span class="kw">data.matrix</span>(x1[, <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)])</span>
<span id="cb165-157"><a href="collating-results-and-running-gsea.html#cb165-157"></a>}</span>
<span id="cb165-158"><a href="collating-results-and-running-gsea.html#cb165-158"></a></span>
<span id="cb165-159"><a href="collating-results-and-running-gsea.html#cb165-159"></a></span>
<span id="cb165-160"><a href="collating-results-and-running-gsea.html#cb165-160"></a><span class="cf">if</span> (<span class="kw">length</span>(hits)<span class="op">&gt;</span><span class="dv">2</span>){</span>
<span id="cb165-161"><a href="collating-results-and-running-gsea.html#cb165-161"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(ListGSC)){</span>
<span id="cb165-162"><a href="collating-results-and-running-gsea.html#cb165-162"></a>  x1=gsca2<span class="op">@</span>result<span class="op">$</span>HyperGeo.results[[j]]</span>
<span id="cb165-163"><a href="collating-results-and-running-gsea.html#cb165-163"></a>  mx1=<span class="kw">match</span>(<span class="kw">rownames</span>(x1), <span class="kw">rownames</span>(DNismrB[[j]]))</span>
<span id="cb165-164"><a href="collating-results-and-running-gsea.html#cb165-164"></a>  DNismrB[[j]][ mx1,(<span class="dv">2</span><span class="op">*</span>i<span class="dv">-1</span>)<span class="op">:</span>(<span class="dv">2</span><span class="op">*</span>i)]=<span class="kw">data.matrix</span>(x1[, <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">7</span>)])</span>
<span id="cb165-165"><a href="collating-results-and-running-gsea.html#cb165-165"></a>}}</span>
<span id="cb165-166"><a href="collating-results-and-running-gsea.html#cb165-166"></a></span>
<span id="cb165-167"><a href="collating-results-and-running-gsea.html#cb165-167"></a></span>
<span id="cb165-168"><a href="collating-results-and-running-gsea.html#cb165-168"></a>}</span>
<span id="cb165-169"><a href="collating-results-and-running-gsea.html#cb165-169"></a></span>
<span id="cb165-170"><a href="collating-results-and-running-gsea.html#cb165-170"></a>DNismr3 &lt;-<span class="st"> </span><span class="kw">lapply</span>(DNismr3, <span class="cf">function</span>(x){ <span class="kw">colnames</span>(x)&lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">rep</span>(<span class="kw">names</span>(DNComp4), <span class="dt">each=</span><span class="dv">2</span>), <span class="kw">c</span>(<span class="st">&quot;NES&quot;</span>, <span class="st">&quot;padj&quot;</span>)); x})</span>
<span id="cb165-171"><a href="collating-results-and-running-gsea.html#cb165-171"></a><span class="kw">names</span>(DNismr3)=<span class="kw">names</span>(ListGSC)</span>
<span id="cb165-172"><a href="collating-results-and-running-gsea.html#cb165-172"></a>DNismr3&lt;-<span class="kw">lapply</span>(DNismr3, <span class="cf">function</span>(x) <span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">GeneSet=</span><span class="kw">rownames</span>(x), x)))</span>
<span id="cb165-173"><a href="collating-results-and-running-gsea.html#cb165-173"></a><span class="kw">write_xlsx</span>(DNismr3, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/GSEA_DN_growth_treat.xlsx&quot;</span>))</span>
<span id="cb165-174"><a href="collating-results-and-running-gsea.html#cb165-174"></a></span>
<span id="cb165-175"><a href="collating-results-and-running-gsea.html#cb165-175"></a>DNismrB &lt;-<span class="st"> </span><span class="kw">lapply</span>(DNismrB, <span class="cf">function</span>(x){ <span class="kw">colnames</span>(x)&lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">rep</span>(<span class="kw">names</span>(DNComp4), <span class="dt">each=</span><span class="dv">2</span>), <span class="kw">c</span>(<span class="st">&quot;pval&quot;</span>, <span class="st">&quot;padj&quot;</span>)); x})</span>
<span id="cb165-176"><a href="collating-results-and-running-gsea.html#cb165-176"></a><span class="kw">names</span>(DNismrB)=<span class="kw">names</span>(ListGSC)</span>
<span id="cb165-177"><a href="collating-results-and-running-gsea.html#cb165-177"></a>DNismrB&lt;-<span class="kw">lapply</span>(DNismrB, <span class="cf">function</span>(x) <span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">GeneSet=</span><span class="kw">rownames</span>(x), x)))</span>
<span id="cb165-178"><a href="collating-results-and-running-gsea.html#cb165-178"></a><span class="kw">write_xlsx</span>(DNismrB, <span class="dt">path=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/DESeq/HyperGeom_DN_growth_treat.xlsx&quot;</span>))</span>
<span id="cb165-179"><a href="collating-results-and-running-gsea.html#cb165-179"></a></span>
<span id="cb165-180"><a href="collating-results-and-running-gsea.html#cb165-180"></a></span>
<span id="cb165-181"><a href="collating-results-and-running-gsea.html#cb165-181"></a><span class="kw">save</span>(DNComp4, EpComp4, CDComp4, DNismr3,Epismr3, CDismr3,DNismrB, EpismrHG, CDismrB,</span>
<span id="cb165-182"><a href="collating-results-and-running-gsea.html#cb165-182"></a>     DNCompSig, EpCompSig, CDCompSig, UpDn1, CDUpDn1, DNUpDn1, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/all_differential_comparisons_%s.RData&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="collating-results-and-running-gsea.html#cb166-1"></a><span class="co">## Pathway: PDL1+LY vs vehicle</span></span>
<span id="cb166-2"><a href="collating-results-and-running-gsea.html#cb166-2"></a></span>
<span id="cb166-3"><a href="collating-results-and-running-gsea.html#cb166-3"></a></span>
<span id="cb166-4"><a href="collating-results-and-running-gsea.html#cb166-4"></a>dneg1=<span class="kw">as.numeric</span>(DNismr3<span class="op">$</span>ProcessNetworks<span class="op">$</span><span class="st">`</span><span class="dt">PDL1+LY_vs_Vehicle NES</span><span class="st">`</span>[<span class="kw">which</span>(DNismr3<span class="op">$</span>ProcessNetworks<span class="op">$</span><span class="st">`</span><span class="dt">PDL1+LY_vs_Vehicle padj</span><span class="st">`</span><span class="op">&lt;</span><span class="fl">0.05</span>)])</span>
<span id="cb166-5"><a href="collating-results-and-running-gsea.html#cb166-5"></a><span class="kw">names</span>(dneg1)=<span class="kw">rownames</span>(DNismr3<span class="op">$</span>ProcessNetworks)[<span class="kw">which</span>(DNismr3<span class="op">$</span>ProcessNetworks<span class="op">$</span><span class="st">`</span><span class="dt">PDL1+LY_vs_Vehicle padj</span><span class="st">`</span><span class="op">&lt;</span><span class="fl">0.05</span>)]</span>
<span id="cb166-6"><a href="collating-results-and-running-gsea.html#cb166-6"></a></span>
<span id="cb166-7"><a href="collating-results-and-running-gsea.html#cb166-7"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb166-8"><a href="collating-results-and-running-gsea.html#cb166-8"></a><span class="kw">barplot</span>(<span class="kw">sort</span>(dneg1), <span class="dt">las=</span><span class="dv">2</span>)</span>
<span id="cb166-9"><a href="collating-results-and-running-gsea.html#cb166-9"></a></span>
<span id="cb166-10"><a href="collating-results-and-running-gsea.html#cb166-10"></a>xx2=<span class="kw">sapply</span>(<span class="kw">c</span>(<span class="st">&quot;Cell adhesion&quot;</span>, <span class="st">&quot;Proteolysis&quot;</span>, <span class="st">&quot;Autophagy&quot;</span>, <span class="st">&quot;Inflammation&quot;</span>,</span>
<span id="cb166-11"><a href="collating-results-and-running-gsea.html#cb166-11"></a>             <span class="st">&quot;Development&quot;</span>), <span class="cf">function</span>(x) <span class="kw">grep</span>(x, <span class="kw">names</span>(dneg1)))</span>
<span id="cb166-12"><a href="collating-results-and-running-gsea.html#cb166-12"></a></span>
<span id="cb166-13"><a href="collating-results-and-running-gsea.html#cb166-13"></a><span class="kw">barplot</span>(<span class="kw">sort</span>(dneg1[<span class="kw">unlist</span>(xx2)]), <span class="dt">las=</span><span class="dv">2</span>)</span></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="deseq-analysis-progressionimmunotherapy-cohort.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="erpgr-subtyping.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
