<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 DESeq analysis: Immunotherapy/Growth comparisons | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 DESeq analysis: Immunotherapy/Growth comparisons | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 DESeq analysis: Immunotherapy/Growth comparisons | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-03-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="expression-in-specific-pathways.html"/>
<link rel="next" href="summary-of-gsea-runs.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.2</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.2.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>24.3</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="deseq-analysis-immunotherapygrowth-comparisons" class="section level1">
<h1><span class="header-section-number">Chapter 12</span> DESeq analysis: Immunotherapy/Growth comparisons</h1>
<div id="summary-of-comparisons" class="section level2">
<h2><span class="header-section-number">12.1</span> Summary of comparisons</h2>
<p>Quick plot of the differences in number of differential genes:</p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb207-1"></a>Allchanges=<span class="kw">rbind</span>(UpDn1, CDUpDn1, DNUpDn1)</span>
<span id="cb207-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb207-2"></a>Allchanges=<span class="kw">cbind</span>(Allchanges, <span class="dt">Frac=</span><span class="kw">c</span>(<span class="st">&quot;Ep&quot;</span>, <span class="st">&quot;Ep&quot;</span>, <span class="st">&quot;CD&quot;</span>, <span class="st">&quot;CD&quot;</span>, <span class="st">&quot;DN&quot;</span>, <span class="st">&quot;DN&quot;</span>), <span class="dt">dir=</span><span class="kw">rownames</span>(Allchanges))</span>
<span id="cb207-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb207-3"></a></span>
<span id="cb207-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb207-4"></a>AllchangesB=<span class="kw">melt</span>(<span class="kw">as.data.frame</span>(Allchanges), <span class="dt">id.vars =</span> <span class="kw">c</span>(<span class="st">&quot;Frac&quot;</span>, <span class="st">&quot;dir&quot;</span>))</span>
<span id="cb207-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb207-5"></a></span>
<span id="cb207-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb207-6"></a><span class="kw">ggplot</span>(AllchangesB, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span><span class="kw">as.numeric</span>(value), <span class="dt">fill=</span>dir))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(Frac<span class="op">~</span>., <span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>, <span class="dt">space=</span><span class="st">&quot;free_y&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;No. differential genes&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;number of differential genes&quot;</span>)<span class="op">+</span><span class="kw">scale_y_log10</span>()</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-83-1.png" width="672" /></p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb208-1"></a><span class="co">## plot the number of samples in each comparison</span></span></code></pre></div>
<p>Notes:</p>
<table>
<thead>
<tr class="header">
<th>Comparison Type</th>
<th>Main cell type impacted</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Treatment</td>
<td>CD45+, PDL1+LY in the DN, LY may affect Ep</td>
</tr>
<tr class="even">
<td>growth</td>
<td>Ep and CD45</td>
</tr>
<tr class="odd">
<td>growth in control</td>
<td>CD and DN</td>
</tr>
<tr class="even">
<td>Spatial Patterns</td>
<td>CD45, Ep</td>
</tr>
<tr class="odd">
<td>Spatial, high cd8</td>
<td>all cases</td>
</tr>
</tbody>
</table>
</div>
<div id="deseq-any-treatment-vs-control" class="section level2">
<h2><span class="header-section-number">12.2</span> DESeq: any treatment vs control</h2>
<p>What are the differential genes if we compare any immunotherapy to the control arm?</p>
<p>First look at the epithelial fraction:</p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-1"></a><span class="co"># EpddsTreatB=EpddsTreat</span></span>
<span id="cb209-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-2"></a><span class="co"># design(EpddsTreatB)=~treatA+factor(Batch)</span></span>
<span id="cb209-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-3"></a><span class="co"># EpddsTreatB=DESeq(EpddsTreatB)</span></span>
<span id="cb209-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-4"></a></span>
<span id="cb209-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-5"></a>epr=<span class="kw">results</span>(Epdds)</span>
<span id="cb209-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-6"></a>Xa1=epr[<span class="kw">which</span>(epr<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>epr<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">60</span>), ]</span>
<span id="cb209-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-7"></a></span>
<span id="cb209-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-8"></a>epvsd=<span class="kw">vst</span>(Epdds)</span>
<span id="cb209-9"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-9"></a><span class="kw">assay</span>(epvsd)&lt;-<span class="st"> </span>limma<span class="op">::</span><span class="kw">removeBatchEffect</span>(<span class="kw">assay</span>(epvsd),<span class="kw">factor</span>(epvsd<span class="op">$</span>Batch))</span>
<span id="cb209-10"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-10"></a></span>
<span id="cb209-11"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-11"></a></span>
<span id="cb209-12"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-12"></a>Mat2=epvsd[<span class="kw">match</span>(<span class="kw">rownames</span>(Xa1), <span class="kw">rownames</span>(epvsd)), ]</span>
<span id="cb209-13"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-13"></a></span>
<span id="cb209-14"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-14"></a>colSCols=ColMerge[ ,<span class="dv">1</span>][Mat2<span class="op">$</span>Treatment]</span>
<span id="cb209-15"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-15"></a></span>
<span id="cb209-16"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb209-16"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(Mat2), <span class="dt">ColSideColors =</span> colSCols, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-84-1.png" width="672" /></p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb210-1"></a></span>
<span id="cb210-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb210-2"></a><span class="kw">write.csv</span>(epr, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp2_ep_io_vs_nonio.csv&quot;</span>)</span></code></pre></div>
<p>DN</p>
<p>List of markers from xcell:</p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-1"></a>xcellgenes=readxl<span class="op">::</span><span class="kw">read_xlsx</span>(<span class="st">&quot;../anntotations/xcell_genes.xlsx&quot;</span>, <span class="dt">sheet=</span><span class="dv">1</span>)</span>
<span id="cb211-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-2"></a>AllGenes=<span class="kw">as.vector</span>(xcellgenes[ ,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)])</span>
<span id="cb211-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-3"></a>AllGenes=<span class="kw">firstup</span>(<span class="kw">tolower</span>(<span class="kw">unique</span>(<span class="kw">as.character</span>(<span class="kw">unlist</span>(AllGenes)))))</span>
<span id="cb211-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-4"></a></span>
<span id="cb211-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-5"></a><span class="co">#rm list</span></span>
<span id="cb211-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-6"></a>rmList=<span class="kw">c</span>(<span class="st">&quot;15L_B_DN&quot;</span>, <span class="st">&quot;6R_B_DN&quot;</span>, <span class="st">&quot;10L_C_DN&quot;</span>, <span class="st">&quot;6R_D_DN&quot;</span>)</span>
<span id="cb211-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-7"></a></span>
<span id="cb211-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-8"></a>DNr=<span class="kw">results</span>(DNdds)</span>
<span id="cb211-9"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-9"></a>Xa1=DNr[<span class="kw">which</span>(DNr<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>DNr<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(DNr<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), ]</span>
<span id="cb211-10"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-10"></a></span>
<span id="cb211-11"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-11"></a><span class="co"># DNvsd=vst(DNddsTreatB)</span></span>
<span id="cb211-12"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-12"></a><span class="co"># assay(DNvsd)&lt;- limma::removeBatchEffect(assay(DNvsd),factor(DNvsd$Batch))</span></span>
<span id="cb211-13"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-13"></a>AllG2=<span class="kw">rownames</span>(Xa1)[<span class="kw">which</span>(<span class="kw">rownames</span>(Xa1)<span class="op">%in%</span><span class="kw">c</span>(AllGenes, <span class="st">&quot;Il6&quot;</span>))]</span>
<span id="cb211-14"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-14"></a></span>
<span id="cb211-15"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-15"></a>Mat2=vsdLimmaDN[<span class="kw">match</span>(AllG2, <span class="kw">rownames</span>(vsdLimmaDN)), ]</span>
<span id="cb211-16"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-16"></a><span class="kw">colnames</span>(Mat2)=Cdata<span class="op">$</span>NewID[<span class="kw">match</span>(<span class="kw">substr</span>(<span class="kw">colnames</span>(Mat2),<span class="dv">1</span> , <span class="kw">nchar</span>(<span class="kw">colnames</span>(Mat2))<span class="op">-</span><span class="dv">3</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb211-17"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-17"></a></span>
<span id="cb211-18"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-18"></a>colSCols=ColMerge[ ,<span class="dv">1</span>][Mat2<span class="op">$</span>Treatment]</span>
<span id="cb211-19"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-19"></a></span>
<span id="cb211-20"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb211-20"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(Mat2), <span class="dt">ColSideColors =</span> colSCols, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-85-1.png" width="672" /></p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-1"></a></span>
<span id="cb212-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-2"></a><span class="kw">write.csv</span>(DNr, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp2_DN_io_vs_non_io.csv&quot;</span>)</span>
<span id="cb212-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-3"></a></span>
<span id="cb212-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-4"></a>ax1=<span class="kw">results</span>(DNddsTreat, <span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb212-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-5"></a>bx1=<span class="kw">results</span>(DNddsTreat, <span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;PDL1&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb212-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-6"></a>cx1=<span class="kw">results</span>(DNddsTreat, <span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;PDL1+LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb212-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-7"></a></span>
<span id="cb212-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-8"></a>testout=<span class="kw">cbind</span>(ax1, bx1, cx1)</span>
<span id="cb212-9"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb212-9"></a><span class="kw">write.csv</span>(testout, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp3_dn_all_pairwise_treatments.csv&quot;</span>)</span></code></pre></div>
<p>CD45</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-1"></a><span class="co"># CDddsTreatB=CDddsTreat</span></span>
<span id="cb213-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-2"></a><span class="co"># design(CDddsTreatB)=~treatA</span></span>
<span id="cb213-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-3"></a><span class="co"># CDddsTreatB=DESeq(CDddsTreatB)</span></span>
<span id="cb213-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-4"></a></span>
<span id="cb213-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-5"></a>CDr=<span class="kw">results</span>(CDdds)</span>
<span id="cb213-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-6"></a>Xa1=CDr[<span class="kw">which</span>(CDr<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>CDr<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(CDr<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), ]</span>
<span id="cb213-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-7"></a></span>
<span id="cb213-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-8"></a>AllG2=<span class="kw">rownames</span>(Xa1)[<span class="kw">which</span>(<span class="kw">rownames</span>(Xa1)<span class="op">%in%</span>RatAllImm)]</span>
<span id="cb213-9"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-9"></a></span>
<span id="cb213-10"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-10"></a>Mat2=vsdLimmaCD[<span class="kw">match</span>(AllG2, <span class="kw">rownames</span>(vsdLimmaCD)), ]</span>
<span id="cb213-11"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-11"></a><span class="kw">colnames</span>(Mat2)=Cdata<span class="op">$</span>NewID[<span class="kw">match</span>(<span class="kw">substr</span>(<span class="kw">colnames</span>(Mat2),<span class="dv">1</span> , <span class="kw">nchar</span>(<span class="kw">colnames</span>(Mat2))<span class="op">-</span><span class="dv">5</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb213-12"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-12"></a></span>
<span id="cb213-13"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-13"></a>colSCols=ColMerge[ ,<span class="dv">1</span>][Mat2<span class="op">$</span>Treatment]</span>
<span id="cb213-14"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-14"></a></span>
<span id="cb213-15"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-15"></a><span class="co">#pdf(&quot;figure-outputs/FIgure3j_1_vs_all.pdf&quot;, height=7, width=6)</span></span>
<span id="cb213-16"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-16"></a></span>
<span id="cb213-17"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-17"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(Mat2), <span class="dt">ColSideColors =</span> colSCols, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>],</span>
<span id="cb213-18"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb213-18"></a>          <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<div class="figure"><span id="fig:3j"></span>
<img src="RatDCIS-bookdown_files/figure-html/3j-1.png" alt="CD45 IO vs non-IO" width="672" />
<p class="caption">
Figure 12.1: CD45 IO vs non-IO
</p>
</div>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb214-1"></a></span>
<span id="cb214-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb214-2"></a><span class="co">#dev.off()</span></span>
<span id="cb214-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb214-3"></a></span>
<span id="cb214-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb214-4"></a><span class="kw">write.csv</span>(<span class="kw">assay</span>(Mat2), <span class="dt">file=</span><span class="st">&quot;nature-tables/Figure3j_1_vs_all.csv&quot;</span>)</span>
<span id="cb214-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb214-5"></a><span class="kw">write.csv</span>(CDr, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp3_cd45_io_vs_nonio.csv&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-1"></a><span class="co">## ALSO LOOK AT the pairwise comparisons and pull out those genes</span></span>
<span id="cb215-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-2"></a>ax1=<span class="kw">results</span>(CDddsTreat, <span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb215-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-3"></a>bx1=<span class="kw">results</span>(CDddsTreat, <span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;PDL1&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb215-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-4"></a>cx1=<span class="kw">results</span>(CDddsTreat, <span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;PDL1+LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>))</span>
<span id="cb215-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-5"></a>xa=<span class="kw">rownames</span>(ax1)</span>
<span id="cb215-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-6"></a>xb=<span class="kw">rownames</span>(bx1)</span>
<span id="cb215-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-7"></a>xc=<span class="kw">rownames</span>(cx1)</span>
<span id="cb215-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-8"></a></span>
<span id="cb215-9"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-9"></a>G1=<span class="kw">rownames</span>(ax1)[<span class="kw">which</span>(ax1<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span> <span class="op">&amp;</span><span class="st"> </span>ax1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(ax1<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="dv">2</span>)]</span>
<span id="cb215-10"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-10"></a>G2=<span class="kw">rownames</span>(bx1)[<span class="kw">which</span>(bx1<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span> <span class="op">&amp;</span><span class="st"> </span>bx1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(bx1<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="dv">2</span>)]</span>
<span id="cb215-11"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-11"></a>G3=<span class="kw">rownames</span>(cx1)[<span class="kw">which</span>(cx1<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span> <span class="op">&amp;</span><span class="st"> </span>cx1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(cx1<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="dv">2</span>)]</span>
<span id="cb215-12"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-12"></a></span>
<span id="cb215-13"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-13"></a>AllG=<span class="kw">unique</span>(<span class="kw">c</span>(G1, G2, G3))</span>
<span id="cb215-14"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-14"></a></span>
<span id="cb215-15"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-15"></a>AllG2=AllG[<span class="kw">which</span>(AllG<span class="op">%in%</span>RatAllImm)]</span>
<span id="cb215-16"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-16"></a></span>
<span id="cb215-17"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-17"></a>Mat2=vsdLimmaCD[<span class="kw">match</span>(AllG2, <span class="kw">rownames</span>(vsdLimmaCD)), ]</span>
<span id="cb215-18"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-18"></a><span class="kw">colnames</span>(Mat2)=Cdata<span class="op">$</span>NewID[<span class="kw">match</span>(<span class="kw">substr</span>(<span class="kw">colnames</span>(Mat2),<span class="dv">1</span> , <span class="kw">nchar</span>(<span class="kw">colnames</span>(Mat2))<span class="op">-</span><span class="dv">5</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb215-19"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-19"></a></span>
<span id="cb215-20"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-20"></a>colSCols=ColMerge[ ,<span class="dv">1</span>][Mat2<span class="op">$</span>Treatment]</span>
<span id="cb215-21"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-21"></a><span class="co">#pdf(&quot;figure-outputs/FIgure3j_all_comparisons.pdf&quot;, height=9, width=6)</span></span>
<span id="cb215-22"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-22"></a></span>
<span id="cb215-23"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb215-23"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(Mat2), <span class="dt">ColSideColors =</span> colSCols, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<div class="figure"><span id="fig:3jb"></span>
<img src="RatDCIS-bookdown_files/figure-html/3jb-1.png" alt="IO vs any other treatment" width="672" />
<p class="caption">
Figure 12.2: IO vs any other treatment
</p>
</div>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-1"></a><span class="co">#dev.off()</span></span>
<span id="cb216-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-2"></a></span>
<span id="cb216-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-3"></a>testout=<span class="kw">cbind</span>(ax1, bx1, cx1)</span>
<span id="cb216-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-4"></a><span class="kw">write.csv</span>(testout, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp3_cd45_all_pairwise_treatments.csv&quot;</span>)</span>
<span id="cb216-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-5"></a></span>
<span id="cb216-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-6"></a>xa[<span class="dv">100</span><span class="op">:</span><span class="dv">105</span>]</span>
<span id="cb216-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-7"></a><span class="co">## [1] &quot;Rangap1&quot; &quot;Wdr4&quot;    &quot;Lrrc47&quot;  &quot;Camsap2&quot; &quot;Celf5&quot;   &quot;Med14&quot;</span></span>
<span id="cb216-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-8"></a>xb[<span class="dv">100</span><span class="op">:</span><span class="dv">105</span>]</span>
<span id="cb216-9"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-9"></a><span class="co">## [1] &quot;Rangap1&quot; &quot;Wdr4&quot;    &quot;Lrrc47&quot;  &quot;Camsap2&quot; &quot;Celf5&quot;   &quot;Med14&quot;</span></span>
<span id="cb216-10"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-10"></a>xc[<span class="dv">100</span><span class="op">:</span><span class="dv">105</span>]</span>
<span id="cb216-11"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-11"></a><span class="co">## [1] &quot;Rangap1&quot; &quot;Wdr4&quot;    &quot;Lrrc47&quot;  &quot;Camsap2&quot; &quot;Celf5&quot;   &quot;Med14&quot;</span></span>
<span id="cb216-12"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-12"></a></span>
<span id="cb216-13"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb216-13"></a><span class="co">#write.csv(assay(Mat2), file=&quot;nature-tables/Figure3j_all_comparisons.csv&quot;)</span></span></code></pre></div>
<div id="epcam-samples" class="section level3">
<h3><span class="header-section-number">12.2.1</span> EPCAM samples</h3>
<p>Note that there are a subset of samples different from the other based on CD8, growth and MH index (6RB, 3NB, 10LD, 8LD, 11ND) but these parameters don’t group the other samples well</p>
<p>There is a small subset of samples which have high expression of CCLs and based on the top variable genes, there seems to be three types of epithelial cases.</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-1"></a>SampleFrac=<span class="st">&quot;Ep&quot;</span></span>
<span id="cb217-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-2"></a></span>
<span id="cb217-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-3"></a>allEpGenes=<span class="kw">unlist</span>(EpCompSig)</span>
<span id="cb217-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-4"></a>allEpTab=<span class="kw">table</span>(allEpGenes)</span>
<span id="cb217-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-5"></a>EpGenesList=<span class="kw">names</span>(allEpTab)[<span class="kw">which</span>(allEpTab<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb217-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-6"></a><span class="co">#EpGenesList=EpGenesList[-grep(&quot;LOC&quot;, EpGenesList)]</span></span>
<span id="cb217-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-7"></a><span class="co">#EpGenesList=EpGenesList[-grep(&quot;RGD&quot;, EpGenesList)]</span></span>
<span id="cb217-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-8"></a><span class="co"># also, remove all samples which have a base mean less than 10</span></span>
<span id="cb217-9"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-9"></a>x1=EpComp4[[<span class="dv">1</span>]]<span class="op">$</span>baseMean[<span class="kw">match</span>(EpGenesList, EpComp4[[<span class="dv">1</span>]]<span class="op">$</span>Gene)]</span>
<span id="cb217-10"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-10"></a>x2=<span class="kw">which</span>(x1<span class="op">&gt;</span><span class="dv">100</span>)</span>
<span id="cb217-11"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-11"></a>vstEp=<span class="kw">vst</span>(Epdds, <span class="dt">blind=</span>T)</span>
<span id="cb217-12"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-12"></a></span>
<span id="cb217-13"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-13"></a><span class="co">#EpddsTreatG$Comp8_v2=factor(paste(EpddsTreatG$Treatment, EpddsTreatG$Growth))</span></span>
<span id="cb217-14"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-14"></a></span>
<span id="cb217-15"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-15"></a>ColsideCols=<span class="kw">hue_pal</span>()(<span class="dv">2</span>)[Epdds<span class="op">$</span>CD8FracCut]</span>
<span id="cb217-16"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-16"></a>ColsideColsB=<span class="kw">hue_pal</span>()(<span class="dv">4</span>)[Epdds<span class="op">$</span>Treatment]</span>
<span id="cb217-17"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-17"></a>ColsideColsC=<span class="kw">hue_pal</span>()(<span class="dv">2</span>)[Epdds<span class="op">$</span>Growth]</span>
<span id="cb217-18"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-18"></a>ColsideColsD=<span class="kw">hue_pal</span>()(<span class="dv">2</span>)[Epdds<span class="op">$</span>MHcut]</span>
<span id="cb217-19"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-19"></a></span>
<span id="cb217-20"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-20"></a>colTable=<span class="kw">rbind</span>(ColsideCols, ColsideColsB, ColsideColsC, ColsideColsD)</span>
<span id="cb217-21"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-21"></a><span class="kw">rownames</span>(colTable)=<span class="kw">c</span>(<span class="st">&quot;CD8&quot;</span>, <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;growth&quot;</span>, <span class="st">&quot;MH&quot;</span>)</span>
<span id="cb217-22"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-22"></a></span>
<span id="cb217-23"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-23"></a><span class="co">#pdf(sprintf(&quot;rslt/DESeq/DEseq_comparisons_%s.pdf&quot;, SampleFrac), width=7, height=10)</span></span>
<span id="cb217-24"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-24"></a></span>
<span id="cb217-25"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-25"></a>sampleDist=<span class="kw">dist</span>(<span class="kw">t</span>(<span class="kw">assay</span>(vstEp)))</span>
<span id="cb217-26"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-26"></a>sampleDistMatrix &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(sampleDist)</span>
<span id="cb217-27"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-27"></a><span class="kw">heatmap.plus</span>(sampleDistMatrix, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">9</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span> <span class="kw">t</span>(colTable), <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;Similarity between %s samples&quot;</span>, SampleFrac))</span>
<span id="cb217-28"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-28"></a></span>
<span id="cb217-29"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-29"></a><span class="co"># Take the top 150 variable genes </span></span>
<span id="cb217-30"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-30"></a><span class="co"># compute the sds for each genes</span></span>
<span id="cb217-31"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-31"></a>sds =<span class="st"> </span><span class="kw">rowSds</span>(<span class="kw">assay</span>(vstEp))</span>
<span id="cb217-32"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-32"></a>means=<span class="kw">rowMeans</span>(<span class="kw">assay</span>(vstEp))</span>
<span id="cb217-33"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-33"></a>CVs=sds<span class="op">/</span>means</span>
<span id="cb217-34"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-34"></a>CVs=CVs[<span class="op">-</span><span class="kw">which</span>(means<span class="op">&lt;</span><span class="dv">7</span>)]</span>
<span id="cb217-35"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-35"></a>sds2=<span class="kw">sort</span>(CVs, <span class="dt">decreasing =</span> T)</span>
<span id="cb217-36"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-36"></a></span>
<span id="cb217-37"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-37"></a><span class="kw">heatmap.plus</span>(<span class="kw">assay</span>(vstEp)[<span class="kw">names</span>(sds2)[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>], ], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span> <span class="kw">t</span>(colTable), <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;top 100 variable genes all %s samples&quot;</span>, SampleFrac))</span>
<span id="cb217-38"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-38"></a></span>
<span id="cb217-39"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-39"></a></span>
<span id="cb217-40"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-40"></a><span class="kw">heatmap.plus</span>(<span class="kw">assay</span>(vstEp)[EpGenesList[x2], ], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span> <span class="kw">t</span>(colTable),</span>
<span id="cb217-41"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-41"></a>          <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;DEG at least 2 %s samples&quot;</span>, SampleFrac))</span>
<span id="cb217-42"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-42"></a></span>
<span id="cb217-43"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-43"></a><span class="co"># Make a table for these genes</span></span>
<span id="cb217-44"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-44"></a>FCvalues=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="kw">length</span>(x2), <span class="dt">ncol=</span><span class="kw">length</span>(EpComp4))</span>
<span id="cb217-45"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-45"></a></span>
<span id="cb217-46"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-46"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(EpComp4)){</span>
<span id="cb217-47"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-47"></a>  FCvalues[ ,i]=EpComp4[[i]]<span class="op">$</span>log2FoldChange[<span class="kw">match</span>(EpGenesList[x2], EpComp4[[i]]<span class="op">$</span>Gene)]</span>
<span id="cb217-48"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-48"></a>}</span>
<span id="cb217-49"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-49"></a></span>
<span id="cb217-50"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-50"></a><span class="kw">colnames</span>(FCvalues)=<span class="kw">names</span>(EpComp4)</span>
<span id="cb217-51"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-51"></a><span class="kw">rownames</span>(FCvalues)=EpGenesList[x2]</span>
<span id="cb217-52"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-52"></a></span>
<span id="cb217-53"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-53"></a></span>
<span id="cb217-54"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-54"></a><span class="co"># for each list pick the top 50 diff genes FC&gt;2 and baseman &gt; 50</span></span>
<span id="cb217-55"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-55"></a>glist=<span class="ot">NA</span></span>
<span id="cb217-56"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-56"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(EpComp4)){</span>
<span id="cb217-57"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-57"></a>  glist=<span class="kw">c</span>(glist, EpComp4[[i]]<span class="op">$</span>Gene[<span class="kw">which</span>(EpComp4[[i]]<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(EpComp4[[i]]<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="dv">2</span> <span class="op">&amp;</span><span class="st"> </span>EpComp4[[i]]<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">25</span>)])</span>
<span id="cb217-58"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-58"></a>}</span>
<span id="cb217-59"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-59"></a></span>
<span id="cb217-60"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-60"></a>ColsideCols=SetCols[EpddsTreatG<span class="op">$</span>Comp8]</span>
<span id="cb217-61"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-61"></a></span>
<span id="cb217-62"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-62"></a>glistmod=<span class="kw">unique</span>(glist)</span>
<span id="cb217-63"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-63"></a>glistmod=glistmod[<span class="op">-</span><span class="kw">grep</span>(<span class="st">&quot;LOC&quot;</span>, glistmod)]</span>
<span id="cb217-64"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-64"></a>glistmod=<span class="kw">na.omit</span>(glistmod[<span class="op">-</span><span class="kw">grep</span>(<span class="st">&quot;RGD&quot;</span>, glistmod)])</span>
<span id="cb217-65"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-65"></a></span>
<span id="cb217-66"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb217-66"></a><span class="co">#heatmap.plus(assay(vstEp)[glistmod, ], trace=&quot;none&quot;, scale=&quot;row&quot;, col=RdBu[11:1], ColSideColors = t(colTable),  main=sprintf(&quot;All DEG %s samples&quot;, SampleFrac))</span></span></code></pre></div>
<p>We can also view this in the form of what is highly expressed in each comparison. Below is a heatmap of the fold changes:</p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb218-1"></a><span class="co"># distance of the samples to one another</span></span>
<span id="cb218-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb218-2"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb218-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb218-3"></a><span class="kw">heatmap.2</span>(FCvalues, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], </span>
<span id="cb218-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb218-4"></a>          <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;All comparisons %s, log2FC for each gene&quot;</span>, SampleFrac))</span>
<span id="cb218-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb218-5"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p>Below, we visualise the enrichment scores using different gene sets and</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-1"></a><span class="co">## GSEA results temp.</span></span>
<span id="cb219-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-2"></a><span class="kw">pdf</span>(<span class="st">&quot;~/Desktop/Fig5Gene_set_enrichment.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">10</span>, <span class="dt">width=</span><span class="dv">10</span>)</span>
<span id="cb219-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-3"></a></span>
<span id="cb219-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-4"></a><span class="co">#Epismr3=ismr4</span></span>
<span id="cb219-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-5"></a></span>
<span id="cb219-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-6"></a><span class="co">#pdf(sprintf(&quot;rslt/DESeq/GSEA_signature_comparisons_%s.pdf&quot;, SampleFrac), width=10, height=12)</span></span>
<span id="cb219-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-7"></a></span>
<span id="cb219-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-8"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Epismr3)){</span>
<span id="cb219-9"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-9"></a>  x1=Epismr3[[i]]</span>
<span id="cb219-10"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-10"></a>  <span class="co"># find padj&lt;0.1</span></span>
<span id="cb219-11"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-11"></a>  idx2=<span class="kw">lapply</span>(<span class="kw">seq</span>(<span class="dv">3</span>, <span class="kw">ncol</span>(x1), <span class="dt">by=</span><span class="dv">2</span>), <span class="cf">function</span>(x) <span class="kw">which</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(x1[ ,x]))<span class="op">&lt;</span><span class="fl">0.01</span>))</span>
<span id="cb219-12"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-12"></a>  unList=<span class="kw">table</span>(<span class="kw">unlist</span>(idx2))</span>
<span id="cb219-13"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-13"></a>  <span class="co">#pickThese=names(unList)#[which(unList&gt;1)]</span></span>
<span id="cb219-14"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-14"></a>  pickThese2=<span class="kw">names</span>(unList)[<span class="kw">which</span>(unList<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb219-15"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-15"></a>  </span>
<span id="cb219-16"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-16"></a>  <span class="cf">if</span> (<span class="kw">length</span>(pickThese2)<span class="op">&gt;</span><span class="dv">100</span>){</span>
<span id="cb219-17"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-17"></a>    temp=x1[<span class="kw">as.numeric</span>(pickThese2), <span class="kw">seq</span>(<span class="dv">2</span>, <span class="kw">ncol</span>(x1), <span class="dt">by=</span><span class="dv">2</span>)]</span>
<span id="cb219-18"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-18"></a>    temp2=<span class="kw">sapply</span>(temp, <span class="cf">function</span>(x) <span class="kw">as.numeric</span>(<span class="kw">as.character</span>(x)))</span>
<span id="cb219-19"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-19"></a>    temp2=<span class="kw">rowSds</span>(<span class="kw">abs</span>(temp2))</span>
<span id="cb219-20"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-20"></a>    <span class="kw">names</span>(temp2)=pickThese2</span>
<span id="cb219-21"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-21"></a>    temp2=<span class="kw">sort</span>(temp2, <span class="dt">decreasing =</span> T)</span>
<span id="cb219-22"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-22"></a>    pickThese2=<span class="kw">names</span>(temp2)[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>]</span>
<span id="cb219-23"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-23"></a>  }</span>
<span id="cb219-24"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-24"></a>  </span>
<span id="cb219-25"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-25"></a>  x23=x1[<span class="kw">as.numeric</span>(pickThese2), <span class="kw">seq</span>(<span class="dv">2</span>, <span class="kw">ncol</span>(x1), <span class="dt">by=</span><span class="dv">2</span>)]</span>
<span id="cb219-26"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-26"></a>  </span>
<span id="cb219-27"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-27"></a>  x234=<span class="kw">sapply</span>(x23, <span class="cf">function</span>(x) <span class="kw">as.numeric</span>(<span class="kw">as.character</span>(x)))</span>
<span id="cb219-28"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-28"></a>  <span class="kw">rownames</span>(x234)=<span class="kw">rownames</span>(x23)</span>
<span id="cb219-29"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-29"></a>  </span>
<span id="cb219-30"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-30"></a>  <span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">2</span>))</span>
<span id="cb219-31"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-31"></a>  <span class="kw">heatmap.2</span>(x234, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s pathways %s fraction&quot;</span>, </span>
<span id="cb219-32"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-32"></a>                                                                           <span class="kw">names</span>(Epismr3)[i], SampleFrac))</span>
<span id="cb219-33"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-33"></a>}</span>
<span id="cb219-34"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-34"></a></span>
<span id="cb219-35"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb219-35"></a><span class="kw">dev.off</span>()</span></code></pre></div>
</div>
</div>
<div id="growing-vs-stable-emphasis" class="section level2">
<h2><span class="header-section-number">12.3</span> Growing vs stable emphasis</h2>
<p>Below, we focus specifically on the growing vs stable comparison in greater depth.
Here, we look at the 3 different fractions in greater depth and look at volcano plots of DEG and heatmaps of differential genes. Samples are colored according to whether they are growing or stable</p>
<div id="dn-fraction" class="section level3">
<h3><span class="header-section-number">12.3.1</span> DN fraction</h3>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb220-1"></a></span>
<span id="cb220-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb220-2"></a><span class="co">#pdf(sprintf(&quot;rslt/DESeq/volcano_plots_differences_stable_vs_growing_%s.pdf&quot;, Sys.Date()), height=7, width=8)</span></span>
<span id="cb220-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb220-3"></a></span>
<span id="cb220-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb220-4"></a>DNa1=DNComp4<span class="op">$</span>stable_vs_growing</span>
<span id="cb220-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb220-5"></a>DNa=DNComp4<span class="op">$</span>stable_vs_growing[ <span class="kw">which</span>(DNComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>DNComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">50</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(DNComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), ]</span>
<span id="cb220-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb220-6"></a><span class="kw">with</span>(DNa1, <span class="kw">plot</span>(log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">main=</span><span class="st">&quot;Volcano plot: DN Stable (+) vs Growing (-)&quot;</span>, <span class="dt">cex=</span><span class="fl">1.0</span>, <span class="dt">xlab=</span><span class="kw">bquote</span>(<span class="op">~</span>Log[<span class="dv">2</span>]<span class="op">~</span>fold<span class="op">~</span>change), <span class="dt">ylab=</span><span class="kw">bquote</span>(<span class="op">~-</span>log[<span class="dv">10</span>]<span class="op">~</span>Q<span class="op">~</span>value)))</span>
<span id="cb220-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb220-7"></a><span class="kw">with</span>(<span class="kw">subset</span>(DNa1, padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), <span class="kw">points</span>(log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">cex=</span><span class="fl">0.5</span>))</span>
<span id="cb220-8"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb220-8"></a><span class="kw">with</span>(<span class="kw">subset</span>(DNa1, padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), <span class="kw">text</span>(log2FoldChange<span class="fl">+0.05</span>, <span class="op">-</span><span class="kw">log10</span>(padj)<span class="op">+</span><span class="fl">0.05</span>, Gene, <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">cex=</span><span class="fl">0.75</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-89-1.png" width="672" /></p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb221-1"></a></span>
<span id="cb221-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb221-2"></a></span>
<span id="cb221-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb221-3"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vstDN)[ DNa<span class="op">$</span>Gene, ], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">as.character</span>(<span class="kw">as.numeric</span>(vstDN<span class="op">$</span>Growth)), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;DN genes&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-89-2.png" width="672" /></p>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb222-1"></a></span>
<span id="cb222-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb222-2"></a><span class="kw">write.csv</span>(DNa1, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp_2_DEG_DN_growing_stable.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="cd-fraction" class="section level3">
<h3><span class="header-section-number">12.3.2</span> CD fraction</h3>
<p>Volcano plot</p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb223-1"></a></span>
<span id="cb223-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb223-2"></a><span class="co">#pdf(&quot;~/Desktop/Figure4G_CD_stable_growing.pdf&quot;, height=12, width=12)</span></span>
<span id="cb223-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb223-3"></a>CDa1=CDComp4<span class="op">$</span>stable_vs_growing</span>
<span id="cb223-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb223-4"></a>CDa=CDComp4<span class="op">$</span>stable_vs_growing[ <span class="kw">which</span>(CDComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>CDComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(CDComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="dv">2</span>), ]</span>
<span id="cb223-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb223-5"></a><span class="kw">with</span>(CDa1, <span class="kw">plot</span>(log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">main=</span><span class="st">&quot;Volcano plot: CD Stable (+) vs Growing (-)&quot;</span>, <span class="dt">cex=</span><span class="fl">1.0</span>, <span class="dt">xlab=</span><span class="kw">bquote</span>(<span class="op">~</span>Log[<span class="dv">2</span>]<span class="op">~</span>fold<span class="op">~</span>change), <span class="dt">ylab=</span><span class="kw">bquote</span>(<span class="op">~-</span>log[<span class="dv">10</span>]<span class="op">~</span>Q<span class="op">~</span>value)))</span>
<span id="cb223-6"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb223-6"></a><span class="kw">with</span>(<span class="kw">subset</span>(CDa1, padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), <span class="kw">points</span>(log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">cex=</span><span class="fl">0.5</span>))</span>
<span id="cb223-7"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb223-7"></a><span class="kw">with</span>(<span class="kw">subset</span>(CDa1, padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), <span class="kw">text</span>(log2FoldChange<span class="fl">+0.05</span>, <span class="op">-</span><span class="kw">log10</span>(padj)<span class="op">+</span><span class="fl">0.05</span>, Gene, <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">cex=</span><span class="fl">0.75</span>))</span></code></pre></div>
<div class="figure"><span id="fig:4g"></span>
<img src="RatDCIS-bookdown_files/figure-html/4g-1.png" alt="volcano plot of DEG stable vs growing CD45" width="672" />
<p class="caption">
Figure 12.3: volcano plot of DEG stable vs growing CD45
</p>
</div>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb224-1"></a></span>
<span id="cb224-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb224-2"></a><span class="kw">write.csv</span>(CDa1, <span class="dt">file=</span><span class="st">&quot;nature-tables/4g.csv&quot;</span>)</span></code></pre></div>
<p>Here are a bunch of heatmaps, which are then separated by wehther genes are downregulated (growing specific) or upregulated (stable specific)</p>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb225-1"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vstCD)[ CDa<span class="op">$</span>Gene, ], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">as.character</span>(<span class="kw">as.numeric</span>(vstCD<span class="op">$</span>Growth)), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;CD genes&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-90-1.png" width="672" /></p>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb226-1"></a></span>
<span id="cb226-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb226-2"></a><span class="co">## growing specific</span></span>
<span id="cb226-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb226-3"></a>CDg=CDa<span class="op">$</span>Gene[<span class="kw">which</span>(CDa<span class="op">$</span>log2FoldChange<span class="op">&lt;</span><span class="dv">0</span>)]</span>
<span id="cb226-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb226-4"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vstCD)[CDg, ], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">as.character</span>(<span class="kw">as.numeric</span>(vstCD<span class="op">$</span>Growth)), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;CD genes growing&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-90-2.png" width="672" /></p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb227-1"></a></span>
<span id="cb227-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb227-2"></a><span class="co">## stable specific</span></span>
<span id="cb227-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb227-3"></a>CDs=CDa<span class="op">$</span>Gene[<span class="kw">which</span>(CDa<span class="op">$</span>log2FoldChange<span class="op">&gt;</span><span class="dv">0</span>)]</span>
<span id="cb227-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb227-4"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vstCD)[CDs, ], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">as.character</span>(<span class="kw">as.numeric</span>(vstCD<span class="op">$</span>Growth)), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;CD genes stable&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-90-3.png" width="672" /></p>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb228-1"></a></span>
<span id="cb228-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb228-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
</div>
<div id="ep-fraction" class="section level3">
<h3><span class="header-section-number">12.3.3</span> Ep Fraction</h3>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb229-1"></a>Epa1=EpComp4<span class="op">$</span>stable_vs_growing</span>
<span id="cb229-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb229-2"></a>Epa=EpComp4<span class="op">$</span>stable_vs_growing[ <span class="kw">which</span>(EpComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>EpComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">50</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(EpComp4<span class="op">$</span>stable_vs_growing<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), ]</span>
<span id="cb229-3"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb229-3"></a><span class="kw">with</span>(Epa1, <span class="kw">plot</span>(log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">main=</span><span class="st">&quot;Volcano plot: Ep Stable vs Growing&quot;</span>, <span class="dt">cex=</span><span class="fl">1.0</span>, <span class="dt">xlab=</span><span class="kw">bquote</span>(<span class="op">~</span>Log[<span class="dv">2</span>]<span class="op">~</span>fold<span class="op">~</span>change), <span class="dt">ylab=</span><span class="kw">bquote</span>(<span class="op">~-</span>log[<span class="dv">10</span>]<span class="op">~</span>Q<span class="op">~</span>value)))</span>
<span id="cb229-4"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb229-4"></a><span class="kw">with</span>(<span class="kw">subset</span>(Epa1, padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), <span class="kw">points</span>(log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">cex=</span><span class="fl">0.5</span>))</span>
<span id="cb229-5"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb229-5"></a><span class="kw">with</span>(<span class="kw">subset</span>(Epa1, padj<span class="op">&lt;</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), <span class="kw">text</span>(log2FoldChange<span class="fl">+0.05</span>, <span class="op">-</span><span class="kw">log10</span>(padj)<span class="op">+</span><span class="fl">0.1</span>, Gene, <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>, <span class="dt">cex=</span><span class="fl">0.75</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-91-1.png" width="672" /></p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb230-1"></a></span>
<span id="cb230-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb230-2"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vstEp)[ Epa<span class="op">$</span>Gene, ], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">as.character</span>(<span class="kw">as.numeric</span>(vstEp<span class="op">$</span>Growth)), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;Ep genes&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-91-2.png" width="672" /></p>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb231-1"></a></span>
<span id="cb231-2"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cb231-2"></a><span class="kw">write.csv</span>(Epa1, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp_2_DEG_Ep_growing_stable.csv&quot;</span>)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="expression-in-specific-pathways.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="summary-of-gsea-runs.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
