<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 17 DESeq analysis: Characterisation cohort (big vs small) | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 17 DESeq analysis: Characterisation cohort (big vs small) | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 17 DESeq analysis: Characterisation cohort (big vs small) | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2022-02-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="read-in-er-io-data.html"/>
<link rel="next" href="signature-analysis.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.3</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html"><i class="fa fa-check"></i><b>16</b> Read in ER i/o data</a><ul>
<li class="chapter" data-level="16.1" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.1</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.2" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#do-associative-study-with-other-features"><i class="fa fa-check"></i><b>16.2</b> Do associative study with other features?</a></li>
<li class="chapter" data-level="16.3" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#use-raw-counts-and-perform-a-batch-correction-if-necessary"><i class="fa fa-check"></i><b>16.3</b> Use raw counts and perform a batch correction if necessary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="tcr-clonotype-analysis.html"><a href="tcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>21</b> TCR clonotype analysis</a><ul>
<li class="chapter" data-level="21.0.1" data-path="tcr-clonotype-analysis.html"><a href="tcr-clonotype-analysis.html#progression-cohort-2"><i class="fa fa-check"></i><b>21.0.1</b> Progression cohort</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>22</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="22.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>22.1</b> Data</a><ul>
<li class="chapter" data-level="22.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>22.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>22.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="22.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>22.3</b> Plots</a><ul>
<li class="chapter" data-level="22.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>22.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="22.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>22.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>22.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="22.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>22.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="22.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>22.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="22.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>22.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="22.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>22.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>23</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>23.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>23.2</b> Load files</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>23.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="23.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>23.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="23.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>23.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>24</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="24.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#mutational-burden"><i class="fa fa-check"></i><b>24.1</b> Mutational burden</a></li>
<li class="chapter" data-level="24.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>24.2</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="24.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>24.3</b> Look at the common mutations (cosmic ones)</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>25</b> Trichrome staining</a><ul>
<li class="chapter" data-level="25.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>25.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="25.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>25.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="25.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>25.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="25.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>25.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>26</b> CNV calling</a><ul>
<li class="chapter" data-level="26.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>26.1</b> Sample data</a></li>
<li class="chapter" data-level="26.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>26.2</b> Data Summary</a><ul>
<li class="chapter" data-level="26.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>26.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="26.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>26.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="26.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>26.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="26.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>26.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="26.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>26.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>27</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="deseq-analysis-characterisation-cohort-big-vs-small" class="section level1">
<h1><span class="header-section-number">Chapter 17</span> DESeq analysis: Characterisation cohort (big vs small)</h1>
<p>This document sets up DESeq runs to compare:</p>
<ul>
<li>CD45 samples</li>
<li>Ep samples</li>
</ul>
<p>according to size of the cohort samples</p>
<div id="cd45-samples-1" class="section level2">
<h2><span class="header-section-number">17.1</span> CD45 samples</h2>
<p>In section 6.2, we have noticed that some DN samples had expression of epithelial markers. Here, we perform a differential gene expression analysis to find genes which are different between these two fractions.</p>
<p>Below is a summary of the number of differential genes, using p value cut off of 0.05 and log2 fold change of 1.5 and base expression of 100+.</p>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-1"></a>infoTableFinal<span class="op">$</span>TumSize=Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">match</span>(infoTableFinal<span class="op">$</span>TumorID, Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb375-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-2"></a>infoTableFinal<span class="op">$</span>SizeCat=<span class="kw">factor</span>(<span class="kw">ifelse</span>(infoTableFinal<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>, <span class="kw">ifelse</span>(infoTableFinal<span class="op">$</span>TumSize<span class="op">&gt;</span>X2a, <span class="st">&quot;big&quot;</span>, <span class="st">&quot;small&quot;</span>), <span class="kw">ifelse</span>(infoTableFinal<span class="op">$</span>TumSize<span class="op">&gt;</span><span class="dv">7</span>, <span class="st">&quot;big&quot;</span>, <span class="st">&quot;small&quot;</span>)))</span>
<span id="cb375-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-3"></a>epidx=<span class="kw">as.character</span>(infoTableFinal<span class="op">$</span>SampleID[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Cohort<span class="op">!=</span><span class="st">&quot;Progression&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(infoTableFinal<span class="op">$</span>SizeCat))])</span>
<span id="cb375-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-4"></a>CD45ddsChar=<span class="kw">DESeqDataSetFromMatrix</span>(allstarFinal[ ,epidx], infoTableFinal[epidx, ], <span class="dt">design=</span><span class="op">~</span>(SizeCat)) <span class="co">## change class</span></span>
<span id="cb375-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-5"></a></span>
<span id="cb375-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-6"></a>a1x=<span class="kw">rowSums</span>(<span class="kw">counts</span>(CD45ddsChar))</span>
<span id="cb375-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-7"></a>a1b=<span class="kw">apply</span>(<span class="kw">counts</span>(CD45ddsChar), <span class="dv">1</span>, <span class="cf">function</span>(c) <span class="kw">sum</span>(c<span class="op">!=</span><span class="dv">0</span>))</span>
<span id="cb375-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-8"></a></span>
<span id="cb375-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-9"></a>sd1vals=<span class="kw">mean</span>(<span class="kw">log10</span>(a1x<span class="op">+</span><span class="dv">1</span>))<span class="op">-</span><span class="kw">sd</span>(<span class="kw">log10</span>(a1x<span class="op">+</span><span class="dv">1</span>))</span>
<span id="cb375-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-10"></a>keep=<span class="kw">which</span>(<span class="kw">rowSums</span>(<span class="kw">counts</span>(CD45ddsChar))<span class="op">&gt;</span><span class="dv">10</span><span class="op">^</span>sd1vals)</span>
<span id="cb375-11"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-11"></a>keep2=<span class="kw">which</span>(<span class="kw">apply</span>(<span class="kw">counts</span>(CD45ddsChar), <span class="dv">1</span>, <span class="cf">function</span>(c) <span class="kw">sum</span>(c<span class="op">!=</span><span class="dv">0</span>))<span class="op">&gt;</span><span class="st"> </span>(<span class="kw">ncol</span>(CD45ddsChar)<span class="op">/</span><span class="dv">2</span>))</span>
<span id="cb375-12"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-12"></a>CD45ddsChar=CD45ddsChar[<span class="kw">intersect</span>(keep, keep2), ]</span>
<span id="cb375-13"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-13"></a>CD45ddsChar=<span class="kw">DESeq</span>(CD45ddsChar)</span>
<span id="cb375-14"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-14"></a></span>
<span id="cb375-15"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-15"></a>A1=<span class="kw">results</span>(CD45ddsChar)</span>
<span id="cb375-16"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-16"></a><span class="co">#DT::datatable(as.data.frame(A1), rownames=F, class=&#39;cell-border stripe&#39;,</span></span>
<span id="cb375-17"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb375-17"></a><span class="co">#          extensions=&quot;Buttons&quot;, options=list(dom=&quot;Bfrtip&quot;, buttons=c(&#39;csv&#39;, &#39;excel&#39;), #scrollX=T))</span></span></code></pre></div>
<div id="pca-plot" class="section level3">
<h3><span class="header-section-number">17.1.1</span> PCA plot</h3>
<p>First, have a look at the samples in a PCA plot: do they separate based on size:</p>
<div class="sourceCode" id="cb376"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb376-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-1"></a></span>
<span id="cb376-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-2"></a></span>
<span id="cb376-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-3"></a>vst1=<span class="kw">vst</span>(CD45ddsChar)</span>
<span id="cb376-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-4"></a></span>
<span id="cb376-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-5"></a>ax1=<span class="kw">plotPCA</span>(vst1, <span class="st">&quot;TumSize&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">colnames</span>(vst1)))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;CD45 cells&quot;</span>)<span class="op">+</span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>ColSize)</span>
<span id="cb376-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-6"></a></span>
<span id="cb376-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-7"></a><span class="co"># nclude the control CD45 sample</span></span>
<span id="cb376-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-8"></a>vst1=vsd[, <span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">!=</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb376-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-9"></a>vst1<span class="op">$</span>SizeCat=infoTableFinal<span class="op">$</span>SizeCat[<span class="kw">match</span>(<span class="kw">colnames</span>(vst1), infoTableFinal<span class="op">$</span>SampleID)]</span>
<span id="cb376-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-10"></a></span>
<span id="cb376-11"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-11"></a></span>
<span id="cb376-12"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb376-12"></a>ax1=<span class="kw">heatmap.2</span>(<span class="kw">cor</span>((<span class="kw">assay</span>(vst1))), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSize[(vst1<span class="op">$</span>SizeCat)])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-113-1.png" width="672" /></p>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb377-1"></a></span>
<span id="cb377-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb377-2"></a>t1=ax1<span class="op">$</span>carpet</span>
<span id="cb377-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb377-3"></a>t1r=<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(t1))</span>
<span id="cb377-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb377-4"></a>t1r[<span class="dv">10</span><span class="op">:</span><span class="dv">11</span>]=<span class="kw">c</span>(<span class="dv">11</span><span class="op">:</span><span class="dv">10</span>)</span></code></pre></div>
<p>Slightly re-order this to make sure the normal mammary gland is on the outside</p>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb378-1"></a><span class="kw">heatmap.2</span>(t1, <span class="dt">Colv =</span> t1r, <span class="dt">Rowv =</span> t1r, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>])</span></code></pre></div>
<div class="figure"><span id="fig:Ext2d"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext2d-1.png" alt="correlation matrix of cd45 cells" width="672" />
<p class="caption">
Figure 17.1: correlation matrix of cd45 cells
</p>
</div>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb379-1"></a></span>
<span id="cb379-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb379-2"></a><span class="kw">rownames</span>(t1)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">rownames</span>(t1), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb379-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb379-3"></a><span class="kw">colnames</span>(t1)=<span class="kw">rownames</span>(t1)</span>
<span id="cb379-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb379-4"></a><span class="co">#write.csv(t1, file=&quot;nature-tables/Ext2d.csv&quot;)</span></span>
<span id="cb379-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb379-5"></a><span class="co">#DT::datatable(t1, rownames=F, class=&#39;cell-border stripe&#39;,</span></span>
<span id="cb379-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb379-6"></a><span class="co">#          extensions=&quot;Buttons&quot;, options=list(dom=&quot;Bfrtip&quot;, buttons=c(&#39;csv&#39;, &#39;excel&#39;), scrollX=T))</span></span></code></pre></div>
</div>
<div id="differential-gene-expression" class="section level3">
<h3><span class="header-section-number">17.1.2</span> Differential Gene Expression</h3>
<p>Below are volcano plots of the differentially expressed genes with abs(log2change)&gt;1.5, padj&lt;0.05 and baseMean&gt;100.</p>
<p>The first plot lists all differentially expressed genes, the second only lists those which are known to be immune related.</p>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-1"></a><span class="kw">print</span>(<span class="st">&#39;significant differential genes&#39;</span>)</span>
<span id="cb380-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-2"></a><span class="co">## [1] &quot;significant differential genes&quot;</span></span>
<span id="cb380-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-3"></a><span class="co"># comment this back in if we want to change to big vs small</span></span>
<span id="cb380-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-4"></a><span class="co">#CD45res=results(CD45ddsChar, contrast=c(&quot;SizeCat&quot;, &quot;big&quot;, &quot;small&quot;))</span></span>
<span id="cb380-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-5"></a><span class="co"># CD45res2=CD45res[which(CD45res$padj&lt;0.05 &amp; abs(CD45res$log2FoldChange)&gt;1.5 &amp;</span></span>
<span id="cb380-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-6"></a><span class="co">#                             CD45res$baseMean&gt;100), ]</span></span>
<span id="cb380-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-7"></a></span>
<span id="cb380-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-8"></a><span class="co"># for continuous variable</span></span>
<span id="cb380-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-9"></a>CD45res=<span class="kw">results</span>(CD45ddsChar)</span>
<span id="cb380-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-10"></a>CD45res2=CD45res[<span class="kw">which</span>(CD45res<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>CD45res<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">80</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(CD45res<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">0.2</span>), ]</span>
<span id="cb380-11"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-11"></a><span class="co">#pdf(&quot;~/Desktop/DESeq-small-vs-largeCD45-characterisation.pdf&quot;, height=6, width=6)</span></span>
<span id="cb380-12"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-12"></a>namId=<span class="kw">which</span>(<span class="kw">rownames</span>(vsd)<span class="op">%in%</span>RatAllImm <span class="op">&amp;</span><span class="st"> </span><span class="kw">rownames</span>(vsd)<span class="op">%in%</span><span class="kw">rownames</span>(CD45res2))</span>
<span id="cb380-13"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-13"></a>namIdN=<span class="kw">rownames</span>(vsd)[namId]</span>
<span id="cb380-14"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-14"></a></span>
<span id="cb380-15"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-15"></a><span class="kw">plot</span>(CD45res<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(CD45res<span class="op">$</span>padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb380-16"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-16"></a>     <span class="dt">main=</span><span class="st">&quot;small (-ve) vs large (+ve)&quot;</span>)</span>
<span id="cb380-17"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb380-17"></a><span class="kw">text</span>(CD45res2<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(CD45res2<span class="op">$</span>padj), <span class="kw">rownames</span>(CD45res2), <span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/cd450deg0char-1.png" width="672" /></p>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb381-1"></a></span>
<span id="cb381-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb381-2"></a>CD45res3=CD45res[<span class="kw">match</span>(namIdN, <span class="kw">rownames</span>(CD45res)), ]</span>
<span id="cb381-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb381-3"></a></span>
<span id="cb381-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb381-4"></a><span class="kw">plot</span>(CD45res<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(CD45res<span class="op">$</span>padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb381-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb381-5"></a>     <span class="dt">main=</span><span class="st">&quot;small (-ve) vs large (+ve)&quot;</span>)</span>
<span id="cb381-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb381-6"></a><span class="kw">text</span>(CD45res3<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(CD45res3<span class="op">$</span>padj), <span class="kw">rownames</span>(CD45res3), <span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/cd450deg0char-2.png" width="672" /></p>
<div class="sourceCode" id="cb382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb382-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb382-1"></a></span>
<span id="cb382-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb382-2"></a>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="kw">as.data.frame</span>(CD45res2), <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb382-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb382-3"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>), <span class="dt">scrollX=</span>T))</span></code></pre></div>
<div id="htmlwidget-e95faf850aaf4a5ac93f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e95faf850aaf4a5ac93f">{"x":{"filter":"none","extensions":["Buttons"],"data":[[984.046408141406,225.723587535065,486.571091288176,3620.97323319146,828.477897547184,734.720958110933,4687.81340396076,297.595136579452,219.861744420235,8774.79223724333,711.244882497177,1566.83655466653,443.662587492489,1144.4245332976,572.975310406236,738.499147210563,128.104615713183,1568.51515190562,481.470237083824,780.652819386176,334.539439232222,1466.62073347047,211.480840507607,1872.64723396084,10227.5667296068,126.844465393415,791.106365309303,272.138591039553,229.767316347247,649.424795608251,976.860535327984,247.606538845518,3457.01600532745,259.371143422054,80.8028628244363,999.461689667443,2653.86503015225,836.571879328051,964.374751088016,114.521278885635,196.269823039607,1787.73394837507,437.666255861023,862.481236891121,112.489290196333,10558.1957192604,2888.51918119894,1163.43025620198,2209.95042956181,6122.63802472064,778.197401142601,1711.86346562487,398.402940959621,2513.19637850159,148.87877462582,439.272041804075,1630.34508172958,134.625441622643,2279.7363925546,505.528826255506,1870.29288789579,767.568362329865,437.646490067142,107.299607697067,1219.74234121504,818.588258325229,95.105449111754,592.268330923036,950.845620175843,816.687422069572,699.583297292074,1951.23704413378,113.835590686489,9536.62897781148,128.083589920143,164.048144795469,214.522241955386,409.105670438815,1674.40643375932,929.074376839279,306.148873686997,4452.4352201318,850.864269476686,4394.46757602679,284.458936361642,1572.0011549797,143.993758701074,210.519151008049,2709.61725322409,1259.94043025254,1981.86392402776,1641.85618570714,430.66985287694,539.025629573324,44283.1553598051,2980.09657739742,1261.53469148221,898.299290178572,3865.21746218901,1792.77609464701,3783.36611493664,1786.20469493081,860.353268797474,431490.499943877,958.001656019441,851.041961258477,777.121304160465,164.773378596263,1627.35477614955,1414.52311792892,392.010299472031,1216.01865553972,952.182943229447,1418.96973864156,4684.50755664237,1833.3430330081,2382.68286943338,10006.0320978256,362.843197459185,4075.06388027394,7079.80963031546,140.413936456838,236.263426362387,1162.66722654659,803.498476695817,1684.35336797072,3002.54795687863,2758.29280396568,4654.85515303356,415.418688716823,7653.49171448885,3913.53980086822,515.99023164524,250.374511023028,2266.56774687775,1309.71865121032,2508.83745882713,1200.18068417036,351.90750455237,124.591599318686,2409.43492534312,174.413913834424,5005.49668812126,1972.46591360364,695.569865339814,2614.03358509034,4345.32917167233,215.187123247571,8131.20626769294,579.193289637969,413.176739482847,237.621216624736,497.762931747279,8448.2178394621,10603.7893828147,213.581359298164,955.031493495133,585.786873584886,142.405113167724,310.279818275187,1086.50088692619,221.843362243006,161.485892020694,511.941122573438,585.697819668765,750.437822735446,832.822040748986,6258.46777229102,4732.53261136431,447.138205917205,282.066261311096,543.350218041859,2402.09352735903,12387.2375619951,520.994539937908,1183.16448237503,1180.00667344145,3229.70358831302,162.915207017169,8703.96542599002,1410.95935858603,872.132530015087,554.621559921307,277.302956725312,2909.10014870773,2031.2798428402,117.942850080766,1269.31291920444,1960.59887144098,1203.55885246724,4398.51095416001,4714.44213389505,188.192105341672,2534.77864125617,311.161091486872,2730.40937428107,965.982234028389,1603.44687690441,3714.77062242633,3693.73087823867,1810.08599131723,94.4937897630929,2619.2619698753,346.645733234321,1887.30723819945,1218.1282404637,6962.36070612163,104.886805820641,185.233571941,406.681954967785,3987.06711978772,245.481948844295,3625.30952885365,3599.38839698735,479.631828074193,3831.21209573598,735.463228988693,1291.67953846706,284.306871528861,231.32689843821,197.154459431283,12946.5825540768,428.557286367945,6609.4239149648,1771.12351498478,801.580069143006,254.740375313781,400.810971567187,114.847452365423,174.838836124127,134.658837236803,638.505102570938,11735.4092967338,503.192509517402,1002.54782937021,711.367145172955,1721.73308257845,3126.32330941737,3389.22064023755,7841.28644418733,22012.4027665847,2144.0738008707,3432.13266586539,2358.30929453438,641.976379077038,553.477025412867,633.010621573242,464.797908871737,1007.87142589098,2054.65030900495,263.53689131496,214.712277305188,1483.5543257515,809.997157440518,579.304133954738,1334.67836398979,369.629050243496,4442.88751632001,2295.30248775195,330.980284612182,1544.79149003932,1891.56629177996,1250.10414392837,1456.29875842972,491.478270442716,103.723256581707,160.480241287043,1621.50103161311,432.32507398655,1663.56473560886,2167.09629625267,1658.54993891475,13741.7918637145,785.639504775433,135.475834579315,4081.00736307964,3509.40707365275,2172.93611970754,172.095791439684,2997.25475974371,312.575305391672,495.664457151064,1482.89446526694,1210.40520995004,747.94286636819,1583.02493716303,1660.91657462058,153.872193489872,621.814462013365,8399.08264328915,5048.8964560695,158.165146772053,4899.00499785492,586.86610558316,743.30350836488,879.648084693711,548.500364089319,1228.0440030334,617.578327864094,1207.745937227,8036.90631175577,387.348027986431,1087.87888269277,1084.41150904441,1816.45550844056,321.921160187567,1508.60896146492,884.361620202934,3742.06661940918,1046.72579509238,719.80024186537,3649.89605653233,2412.08688854373,81.1182514475663,8948.53820072255,7459.0287899958,748.193238249566,233.643562749897,741.271163978541,3477.39039966568,951.653006413707,1833.00627008016,375.746928068962,1898.4440704587,247.373698795911,874.016191334081,458.588824153367,554.95441995482,1014.40564107507,13093.927539316,680.625026640383,4236.19204057241,314.408730484853,9707.92203095945,1099.28573940775,1808.93261685101,526.963588170605,1210.62283067559,1034.39569022663,116.851416994363,193.15952951016,1163.71336553813,588.239733709843,7911.53330456687,604.393609116086,131.006370436418,489.271326798198,877.749662698824,6110.54739977154,2283.30441993735,4071.57641607896,661.867863965199,1247.64990374634,329.531982812062,354.711314485493,841.782562141971,264.514288580749,3495.26056573804,7144.36374830235,4059.57059712939,1654.33647372767,502.038573136342,2922.62022750888,4187.24519721418,286.515858944192,843.071418036581,149.68906462887,1848.99547827204,1832.50592513583,12382.0649275652,1624.77972127762,832.295384020282,281.409954825968,1269.23078175889,230.762021525904,113.742325522347,439.141936691601,973.223895504159,598.546471625612,708.17587003963,999.382455977088,197.188451596432,289.73919012621,214.104669853906,864.411202289042,4997.79305361858,109.70389496786,1590.29131975672,162.176060102044,560.878534398482,299.282839174513,598.303095306772,310.209873916104,8737.46263446132,262.101955706907,5101.00966808054,622.289225615428,385.980221420567,633.463375196924,504.95732435249,2840.26090087625,478.505057438366,233.579210139652,275.806552615857,124.941630439043,1173.90158155766,555.681093467731,142.185695076632,409.710694015836,300.551681134393,1935.0339519047,391.391086634963,252.591384352829,1468.10594023401,2590.00935200449,1160.12879495266,365.398343691758,2001.38778258004,462.688045634735,3857.28524525811,556.213245711416,347.475596597195,1794.81397586607,3203.39021021319,1331.15154888074,2427.53442062634,191.177107464258,1088.71584488908,3276.9714410156,502.312562310962,949.837569174208,325.738026908316,783.017533110808,1171.76695474759],[-1.44893258586331,-1.63552594259918,-2.12041165892402,-2.2217774027991,1.21172161258829,-2.76801657168303,1.49618965521964,-3.06405045424486,-1.69750118275074,-1.13138951781342,-2.13754651048689,1.73773927107301,-2.24855181966496,1.34698948627968,-1.57449338824071,-1.93792094979165,2.17813069098398,-1.24391438886999,-1.54388639626294,-1.55075551249935,1.98452495059434,-1.50169457149607,2.14109691569082,-1.54258180495585,-2.05161812926231,7.1446856175249,-1.98336082711331,-2.95414612052976,-2.6255012250966,-1.84647294864668,-2.88504440176568,1.0647759085919,-1.67631867542453,1.92486075183195,-3.47521271034075,-1.32062425346731,-1.05304799896332,-1.81468731414474,-1.06265489560585,-3.2933898005409,1.61562941049659,-1.78599067224515,1.49639144039147,1.78659602499115,-4.09381219733068,-4.4124933926187,-1.24301737471713,1.22036548147737,-1.53935413514992,1.49110331569571,-1.0878409031214,-1.70732420965561,1.41899500393537,1.56418304401522,-1.85731113601522,1.7684034570534,-2.06241073688195,2.22584639506721,-1.56457516038158,1.12691762450299,1.5996248038267,-6.69549139322083,-1.54343396949466,-2.03710107740599,-0.946396966063715,-1.34709351551588,4.15141024218258,-1.90638999275183,-1.60285783673199,-2.30497385051824,-1.19879940601141,1.94123837639243,-4.39575890328026,-2.72929352015574,-2.98570164882738,1.25510891498892,-3.36452980795328,2.8744644185975,-2.06859103792967,-2.7979679875754,1.90972099114536,1.53427233356409,-1.99166363980725,-1.9023411293958,1.88268779783873,-1.63074218075323,-5.15043939775042,-1.46822043059695,1.57396755560137,1.94471345643245,-1.36585575609488,-1.01767808416511,-2.04842177019611,1.69393125576582,1.48623826447997,-2.21065399858891,-2.25044445213956,-1.86273243335328,1.65672428577686,-1.33774521337294,1.87556166439049,-1.65851417969193,1.23320983264343,-2.17153310940382,2.11548017653359,1.60907308083309,-3.35503745774476,-1.63358595050081,-1.90423079702919,-2.57198721232061,-2.07773105514907,-2.4022840416009,1.06283216249483,-2.25880621038855,-2.18385252051857,-2.32610788330065,-2.12824207926319,-2.34083270077097,-9.18463096359957,1.75729885191929,2.02853195956777,2.05401119721293,-3.02982585200014,-2.2774658831506,2.57483724462208,1.39944426808183,-1.57492082087132,-1.86384579313214,-2.04888964747667,1.36031079865005,1.57807676804034,1.47137744466538,2.01702061687595,-2.51044629828272,1.82573321705754,-2.4925286970436,1.52336597598554,-1.20386497637601,-5.98561955898101,-4.7614269396058,-9.03525339683053,-2.47108673374707,-1.94713773269727,-1.30474025846621,-3.11927742704539,-1.78409216346063,-1.92521471045528,-1.71978977181211,-2.06544967968794,-2.57883351635307,-1.8147238428442,-1.98266469442746,-2.81126411728018,-1.56086474825003,-1.00542440976863,-2.2180861477647,3.10793130812189,-1.37730509611041,-2.17286954720293,-2.06091939958072,-1.38530062173861,-1.9771224571221,-2.32847813004329,-1.56572060585387,-1.39691316497555,2.24145016887617,-1.88537316798344,-1.65778684539312,1.17832782584003,-1.32415275244645,-2.21830263241036,-1.61014501525189,1.57851817331939,0.870105998077502,-1.56860939069064,-3.51229002372117,-2.00572968728743,-1.27524056707521,-4.83952088563613,-1.61441973732223,2.05900405208579,-2.03158628908616,-4.89663412160006,2.05331920439052,-2.57829494273128,-1.35158972424974,-3.87738423190758,-1.5563938570086,1.4787119599824,-2.02550965232356,-1.59188082556877,1.58202001297785,-2.76058662733878,2.39388961332645,-2.71163424447802,-1.65320104274178,-1.88917760079236,-1.7655257625938,1.36125335601027,-1.64537120083384,-1.87713804928016,-5.18484309566808,1.87604233516116,-2.79376938723556,-2.06603755052128,-2.10805910105839,-1.69754688778733,3.24935248996408,-3.98664917797805,-2.03593709082976,1.06331360620319,1.79398881669996,2.19596090553503,-2.35533472146301,-0.884324293453941,-1.25944754766502,2.32781238385706,-1.57356314146601,-2.09171490968424,-1.68024722477391,1.39109070510544,-1.68651426713356,-1.97676651314292,1.0275930027446,-1.83508313713876,-1.43524700421197,-2.04556136103049,1.12230336322998,-2.63374170651593,-2.62720153283382,-2.14890247145387,-2.31834052204224,1.52375658955159,-1.60139590703044,1.72892803668288,-1.33438886547065,-1.8816479654603,-1.93805200921274,-1.21840730072606,-2.20450993858981,-2.17444901845847,-1.77612215145201,-1.94786212599329,-0.867319897393809,-1.45492355066752,-2.10040378025181,-1.87909020722965,-2.27231122843841,-1.35038516955557,1.4529848943217,-1.86421521417454,2.43412478183291,-0.990716851467873,-6.30452158247934,-2.25372149625377,-5.4505647409952,-1.33310627114935,-1.91907313986519,1.06430437708407,-1.66398431726374,-1.73989399044778,-2.68904164089826,-1.65844282509784,-2.06967722530054,2.42648523780741,-5.28490840893007,-5.30851143613921,-2.36704392537473,-1.60588816633725,-2.01321235832526,2.14688011602967,-2.05382429473691,-2.17277112890082,-2.18245269168896,2.47286615409355,-2.18363815706546,-2.02644996550346,1.36969190881726,-2.03128523667887,-1.75295756274941,3.58408782425142,1.21927776240779,1.20586625168524,-2.17500435050259,1.15781728517006,-1.57389011846159,-1.01505868848164,-2.26072700357995,1.32239213283661,-1.94447458109307,-1.67203198923107,-2.85271193625953,-2.18093383279456,-1.95155859114499,-2.31220647725776,-1.15351052816864,-1.2599458419992,1.36396193224541,-1.34321051954242,-1.52487223515936,-1.96652214526589,-1.42385662081113,-2.03739012181875,-1.94730082038952,-1.41528574953291,-1.25653463679953,1.31588470454617,-2.28473894250948,-1.29252106577617,-1.43887513644765,-2.0682382983203,-2.08442107392284,-2.35988002550508,-3.02042180087798,-3.08677768858119,1.14641622155229,-1.83429471876306,-1.59614048384388,-1.66631781444711,-1.93250725001253,-2.15160861848256,-2.23193268738204,-1.31485196090887,1.32360193242084,-4.0946186420719,-1.3759828225975,-1.90648688337592,-2.14343993981507,1.51337095121968,-1.46113882929438,-1.17374928379776,-1.82460878926191,1.46856781969587,-2.74919325808338,-1.16098329317206,-2.26035249290864,-2.36157954729944,-1.47062381931864,-2.06855721081025,2.41571433496233,-2.96113825548744,-1.308291614659,-1.80737926373918,-1.87190463325161,-1.49408555347218,2.37542676014277,-1.81351310648246,-2.14511700181345,-1.54939174732352,1.8519918879083,1.40979430401094,-1.65779479166024,-1.79305035905847,-1.84773162579099,-2.83833077571584,-2.41241387374479,-5.24039536282882,-2.06470178549832,-2.46661657697293,-1.47174898251797,-2.19515206258189,-2.32882540012892,-1.59488695151453,-1.94401203165137,-1.20631441090904,2.01336323090765,-4.56096164196705,-1.67491425068637,1.93379735220602,-1.52725875686285,1.54471531836804,1.72508412621876,1.41835571920845,-1.0907086938456,-1.7824793172704,-4.06191086015894,2.14366102818419,-5.68029622544566,-1.85249267024395,1.71529568223193,-2.36847078428837,-3.37086946464578,2.18683643720594,-1.60775428701916,-2.84312225768106,-2.34659638879508,-4.65254490030995,-2.83826923491032,-2.00388954815808,-1.7167708760343,-1.89633027022967,1.64118869388904,-1.26202749309517,1.16338896152915,-1.83574702463687,2.41674457351672,-1.66761646788681,-1.33957159068804,2.05827682418699,-2.84329113939776,-1.83888505510418,-1.68113513819112,-2.07793181673425,-1.69684370030059,-1.64679077207803,-2.7918352799079,-2.36238207770191,-3.72761719797963,-4.09669448408422,-2.61595391251512,-1.6421826119419,1.6036797680928,-2.4842223672688,-1.45830431725245,-2.76213614809254,-2.31524554306563,-3.2465808712928,-2.19970603985834,-2.44859125576499,1.32915461625485,-2.17115842197443,-3.15780575004043,1.92357034602094,-1.08783180712356,1.20344435156499,-1.62062151965977,-1.96369086319616,-1.5588074795374,-1.98157098712048,-4.10288994739971,-2.0399642161599,-4.5390404461113,-1.50992661636228,-1.4019090412947],[0.37925887962899,0.439459323551506,0.589945199398653,0.658410915683418,0.385537529824973,0.810267677624354,0.409191491003716,0.808118584272048,0.503705573723807,0.359515033156027,0.633337550041624,0.472961928493833,0.649169383739718,0.347065409257392,0.466268800392798,0.465234143127832,0.597525577901066,0.399300460172721,0.493194769948593,0.501375504053598,0.558353729732753,0.459185601113308,0.634150852540134,0.463129220898356,0.655582155514246,1.86265338650953,0.520066697873644,0.874222299097634,0.617404834949395,0.537798073337168,0.663266576385699,0.34178753785513,0.529686929361304,0.492300672869235,1.10992185405017,0.387878066365796,0.280693202109256,0.558287314980468,0.30812421068206,1.05021920457956,0.510125351357031,0.488797256121653,0.452694780905053,0.432369879249097,1.32828301886222,0.810591070061893,0.403235600951096,0.383787793806015,0.479827965372846,0.414620168012086,0.33905831083359,0.543093744931608,0.416989031017552,0.393329840873385,0.563282134622082,0.578006767421079,0.544121240631696,0.570100204774692,0.46587368334154,0.317026009268656,0.437166260926062,1.73599490677342,0.479029920829081,0.654565870716708,0.297086075621839,0.402792202850646,1.3424664712972,0.559267580171409,0.512357952360182,0.617733701666988,0.33557513649208,0.402661695788559,0.989586080383384,0.625602940998716,0.736047796960971,0.366749306434751,0.793902779586974,0.806174693010235,0.642767384472857,0.72547016792365,0.457526507058495,0.433991718401471,0.434543797558149,0.594345068051353,0.540250063163174,0.485739749008293,1.06449207625381,0.453991307625412,0.465958720613374,0.432795927528277,0.370332653198941,0.329648852868426,0.572337811044197,0.439468201378254,0.46476034012386,0.587867662797175,0.654322379990206,0.531744724856147,0.371123189748388,0.360594309003481,0.426387356517786,0.446381467627387,0.366705876286433,0.639243560571535,0.599962914137892,0.49268657861538,0.735798997580878,0.506125055634643,0.534390421693489,0.643299530218342,0.542405427948474,0.693843081679304,0.343894350534129,0.647261277086737,0.612012425811953,0.493629946140437,0.628787061628987,0.702013094955768,1.14307487732829,0.327554877463293,0.521352148030296,0.589518040688544,0.984739526858249,0.535623960649252,0.36943028196653,0.360256194715732,0.516310056309685,0.510524535525758,0.64020849118493,0.427423215063246,0.371482124332904,0.36610409361008,0.542071405479324,0.782474192936752,0.449922146474958,0.649738973403739,0.422576045608796,0.382615835659232,1.94379049093433,1.35874940025074,2.29332553233546,0.798599212383748,0.552821922316784,0.421903013071016,0.698647048759214,0.559093828362238,0.627684976670177,0.507624294284249,0.505535361463451,0.782573196979649,0.535396062897764,0.510274591682095,0.903827811384951,0.423910844673396,0.310911374171836,0.697605661067797,0.824183469792147,0.414196359741245,0.577633204861982,0.492325330846958,0.439620355462685,0.637218241967233,0.736614344908026,0.495795177663853,0.446067479803094,0.730107915177779,0.436803335002826,0.39825668772966,0.375509819047236,0.398883865047131,0.64536369418213,0.513024513955324,0.372900034684454,0.285108100866828,0.50265131762837,1.05621627119649,0.635041684922835,0.386306102693499,1.10318482954115,0.478943520252474,0.66196594385841,0.583806553297426,1.51383218935065,0.638759187313669,0.604579603344587,0.423283087200897,1.22236033649687,0.481357558390582,0.469494969080107,0.621087030075483,0.518545397371509,0.433816620766709,0.690286790701372,0.618561272867754,0.861326562149707,0.511698956796595,0.533615866423836,0.55773587290429,0.319478451784738,0.505070187855913,0.596006147979177,1.12900389611583,0.38101796888699,0.871574808450361,0.622591793353006,0.64066412414728,0.449075459929163,1.02596782690141,1.04377111573045,0.574735049717594,0.289166727859481,0.574254401015219,0.423513128743806,0.635656129005544,0.285380184132092,0.411338125425913,0.543863258941956,0.488373914472138,0.670643658298533,0.544762620725473,0.453342040859844,0.54617728539048,0.486305769404416,0.31542233804439,0.513478983459581,0.390842545843441,0.574437542778747,0.361173498947059,0.789056267835423,0.846356026487193,0.453227309487256,0.693315999894657,0.466803725574415,0.443212889911626,0.55847472949245,0.367369907143706,0.569695369784597,0.57522725837634,0.353052668618721,0.421516663659601,0.579576051261632,0.507621025412658,0.58501737148064,0.270389130289498,0.476643565286878,0.60688990318297,0.594420145136629,0.613642546225021,0.405004620591605,0.283033318861803,0.515402902169578,0.574621175837928,0.296882836460622,1.68226025205846,0.577474823619294,1.73041041876912,0.39092432627866,0.521177374876589,0.335631000768393,0.427358362999089,0.522266476838543,0.773964463377923,0.522058876408812,0.55038087422822,0.564968947372986,1.31755549944422,0.768551482714105,0.665415237217938,0.484934631260389,0.569419156959594,0.616860781029013,0.515452264510552,0.655053433310923,0.568423052977971,0.559137770690297,0.608191897790933,0.59289457904127,0.415987547952918,0.496620218839605,0.561392467557432,1.13635254008651,0.394573251730127,0.358872474520366,0.556792755610929,0.375982706493019,0.501791198717444,0.32215954258672,0.684212832096279,0.413116181825785,0.60121378945868,0.469375765513581,0.810060346990049,0.629906927522353,0.478300110774611,0.576650835749733,0.351512520679072,0.3830329620374,0.429608287407215,0.433735031907835,0.447569431781864,0.583210196672647,0.452789077767587,0.579793267313069,0.600909592875541,0.393985160267348,0.393926269101718,0.3500284322639,0.50285592228156,0.386053181201713,0.465893741227232,0.634212305932739,0.577890176038527,0.587994534019672,0.859977258370099,0.795286971615775,0.340802413800656,0.467300247782167,0.500278016975222,0.504075015720713,0.585994868998674,0.637023666650933,0.630414632367408,0.366155736574141,0.401436257745407,0.847835624149306,0.40432388678932,0.607559631870883,0.675968120611294,0.403335884088376,0.3907023523079,0.378071894168939,0.514120599452805,0.454951718827637,0.637598103603091,0.371737496045947,0.616971212038375,0.592191868365734,0.421025718891294,0.586606170513729,0.775704913417531,0.892430563746007,0.364627625542869,0.539063589086383,0.598551710566113,0.441748681212041,0.567705812645662,0.500904129271279,0.663628512386036,0.506527433614247,0.465287635961142,0.413532753962885,0.511012465282593,0.576331096418226,0.412128495631513,0.696520608596591,0.73197408743187,1.22168535991602,0.522038662844899,0.714223747010058,0.435814258709519,0.586474487433841,0.608273232862487,0.514078260475354,0.561769294472923,0.389838186335539,0.569396892990321,1.3058770836349,0.527740516896431,0.452563343159088,0.497286017539721,0.404105993499956,0.557248083499446,0.450695354645806,0.333450646373609,0.370652515271143,0.842244525732691,0.443290460606969,1.76650024386944,0.528480015835569,0.376230952594744,0.611938195146713,0.952846755390273,0.596096898530856,0.524787578032152,0.729243518266621,0.638927976120337,0.981740259058448,0.637792549906197,0.642358873931705,0.454274281685896,0.582957087578833,0.406945204994537,0.411845565944868,0.351369799344317,0.596956634525125,0.560329084507139,0.467670506412305,0.358033052014481,0.395756483296519,0.669600731959762,0.501069271579935,0.525853709116021,0.512416393704238,0.544325430539127,0.534079646256249,0.728757389949991,0.582955945826175,1.20691963492457,0.88305955328054,0.840111158457861,0.455798504918474,0.433258890211066,0.698666704716427,0.46611445088917,0.700661480492073,0.681444282785029,0.885399656244588,0.670830955214906,0.691983401218186,0.426278374238014,0.483589441580501,0.75199434259485,0.406533819417778,0.356988799666409,0.378424452696651,0.476182372028098,0.574121684372903,0.391992764046668,0.572853006020662,1.02363686580178,0.621218708083357,0.957721769525031,0.471006023695161,0.415212488239477],[-3.82043154080066,-3.72167765012156,-3.59425190862712,-3.37445408312062,3.14294074856574,-3.41617547894623,3.65645348965983,-3.79158518796465,-3.37002660145572,-3.14698806300656,-3.375050966655,3.67416311204352,-3.46373670106185,3.88108250015984,-3.37679335806794,-4.16547447864153,3.64525096755711,-3.11523404789448,-3.13037868674857,-3.0930021510057,3.55424320626316,-3.27034333797744,3.37632111841294,-3.33078055831506,-3.12945999522058,3.83575692035408,-3.81366627631134,-3.37917040503201,-4.25247921051962,-3.433394502864,-4.34975092139723,3.11531519046553,-3.16473483203717,3.90992915084484,-3.13104269247379,-3.40474073680121,-3.75159779805937,-3.25045413974385,-3.44878740055372,-3.13590704319614,3.16712236747049,-3.65384758174798,3.30551953216646,4.13210103371205,-3.08203307517802,-5.44355046038415,-3.08260821163923,3.17979232579293,-3.20813759563551,3.59631159006294,-3.20841834092459,-3.14370074335991,3.40295522995577,3.97677186287713,-3.29730169280325,3.05948573049339,-3.79035145639159,3.90430730672493,-3.358367764325,3.55465353490291,3.65907652717336,-3.85686119647971,-3.221999091045,-3.11214068520788,-3.18559853094019,-3.34438826268783,3.09237536351366,-3.40872609166357,-3.12839457131174,-3.73133899655813,-3.57237254983491,4.82101574770051,-4.44201771873879,-4.36266094880992,-4.05639642038858,3.42225300216681,-4.23796199542678,3.56556022350978,-3.21825762772041,-3.85676504877298,4.17401169480483,3.53525716853607,-4.58334384473808,-3.20073511442252,3.4848451230446,-3.35723437104462,-4.83840087929635,-3.23402762549889,3.37791200372739,4.49337281785165,-3.68818613291751,-3.0871579722175,-3.57904323402797,3.85450244284646,3.19785949051479,-3.76046198573033,-3.43935118369823,-3.50305766335004,4.46408182388192,-3.70983451477607,4.39872720361082,-3.71546379043756,3.36293992649349,-3.39703556413191,3.52601823659951,3.26591620448672,-4.55972007134459,-3.22763303715998,-3.56337000014832,-3.99811765982116,-3.83058676792305,-3.46228723040182,3.09057755919534,-3.48979042984809,-3.56831402176396,-4.71225034357798,-3.38467854880728,-3.33445731652407,-8.03502127967928,5.36489905303338,3.89090553713398,3.48422110172216,-3.07677895459992,-4.25198656234496,6.96975145328058,3.88458071952403,-3.05033923245274,-3.65084469684239,-3.20034750505181,3.18258520059273,4.24805573316402,4.0190139098321,3.72095003810874,-3.20834389292842,4.05788697302803,-3.83620007275564,3.60495108943258,-3.14640656286951,-3.07935427552373,-3.50427160352537,-3.93980412699163,-3.09427644734471,-3.52217894062006,-3.09251230269501,-4.46474000367594,-3.19104249225358,-3.06716710135138,-3.38791856728806,-4.08566806030891,-3.29532563382711,-3.38949792238335,-3.88548582811403,-3.1103978898065,-3.68205901751017,-3.23379745255941,-3.17957016628788,3.7709216722166,-3.32524674280293,-3.76167701045183,-4.18609254989029,-3.15112938817547,-3.10273988864206,-3.16105455471957,-3.15799885999582,-3.13161848425307,3.07002584450874,-4.31629755750661,-4.16260893155031,3.13794144938673,-3.31964481012534,-3.43729071283073,-3.13853426386582,4.23308668945317,3.05184593293588,-3.12067100130508,-3.32535117996471,-3.15842209245075,-3.30111421534287,-4.38686315841474,-3.37079356762399,3.1104380386768,-3.47989634171021,-3.23459506017008,3.21454351682337,-4.26460788367309,-3.19311062765959,-3.17204683114938,-3.23334251198299,3.14957998991912,-3.2612332157015,-3.06989674122646,3.6467482739178,-3.99918796727062,3.87009293716041,-3.14820692132179,-3.23080792091382,-3.54033251194941,-3.16552305197836,4.26086125184892,-3.25770801840166,-3.14952799672419,-4.59240496291091,4.92376341368192,-3.20542695836151,-3.31844648866074,-3.29042788819212,-3.78009274444679,3.16710953771099,-3.819466852355,-3.54239243253071,3.67716443061837,3.12403146328245,5.1851070403615,-3.70535988561587,-3.09875857759143,-3.06183032842129,4.28014274835487,-3.22204584404719,-3.11896621074605,-3.08436585193067,3.06852349821117,-3.08785134835443,-4.06486337919431,3.25783205183136,-3.57382326492669,-3.67218722597024,-3.5609813229397,3.10738015524913,-3.33783763449587,-3.10413283608086,-4.74133492504007,-3.34384396493734,3.26423399401228,-3.61315282899318,3.09580352588944,-3.63227591461015,-3.30290198105657,-3.3691936204191,-3.45106384691255,-5.22994730374426,-3.7517923898427,-3.49891368271863,-3.32957997651143,-3.20767294330653,-3.05243510376952,-3.46093050689388,-3.16121555200277,-3.70298839677449,-3.33424632929623,5.13361783752094,-3.61700565970266,4.23605130507661,-3.33706341289042,-3.74764937515759,-3.90271818627294,-3.14986819420118,-3.41013894898697,-3.68218812322697,3.17105504154103,-3.8936509995647,-3.33142958165715,-3.47437352506095,-3.17673523052819,-3.76044539738554,4.29490018715219,-4.01114671158777,-6.90716439371431,-3.55724334668259,-3.31155595582728,-3.53555431656845,3.48033167621446,-3.98450920898973,-3.3169372426898,-3.83948659410466,4.42264193857019,-3.59037692707982,-3.41789255145543,3.29262718453362,-4.09021856062392,-3.12251706970058,3.15402808355456,3.09011763230652,3.36015252575969,-3.90630863743209,3.07944292430259,-3.13654389013673,-3.15079503879169,-3.30412833190161,3.20101751277869,-3.23424814132063,-3.56224609807361,-3.5216042198083,-3.46231123599949,-4.08019682032778,-4.00971668453674,-3.28156313163532,-3.28939273345402,3.17489669595816,-3.09684581767388,-3.40700710745267,-3.37189259804676,-3.14463552838168,-3.51399410217475,-3.24058867336611,-3.59223110985331,-3.18977112053189,3.75936519223694,-4.54352597090464,-3.34803889389744,-3.08841911603587,-3.26111347725827,-3.60695017903173,-4.01343871238457,-3.51221124917017,-3.88133818200217,3.36387353823983,-3.92530226009665,-3.19050693751137,-3.30569411789762,-3.29782281765491,-3.37759604724633,-3.54042018187367,-3.59096370634804,3.29716588096602,-4.82949586623035,-3.40316975463407,-3.13794199510128,-3.17091867864523,3.75213565398531,-3.73977484564233,-3.10456635867588,-3.54898985024895,3.22796410898329,-4.31179647892236,-3.12312668353628,-3.66362716574862,-3.9878621667276,-3.49295483228744,-3.5263134191014,3.11421816876135,-3.31806011109476,-3.58802110155853,-3.35281272994596,-3.12738999857031,-3.38220716216471,4.18425654138125,-3.62047945007118,-3.23240632639609,-3.05885060611244,3.98031614161132,3.40914786193084,-3.24413767625701,-3.1111463014956,-4.48338720902974,-4.07501334588613,-3.29576403750677,-4.28948036439516,-3.95507446564692,-3.45356281879297,-3.37700970793369,-3.74296258339716,-3.82858438332004,-3.10242053425832,-3.46051671171407,-3.09439776089752,3.53595752926223,-3.49264237739103,-3.17374580321464,4.27298715514005,-3.07118781344151,3.82254988348301,3.09572016001464,3.14703869163043,-3.27097489750714,-4.80903067922381,-4.82272159219483,4.83579327479553,-3.21556492571053,-3.50532208358912,4.55915620552239,-3.8704411704854,-3.53768268147706,3.66859220807159,-3.06362870296571,-3.89872818402148,-3.67270878173774,-4.73907925989716,-4.45014485560823,-3.11957945858647,-3.779150493977,-3.25295002091082,4.03294761492785,-3.0643221572625,3.31101012010742,-3.07517651780048,4.31308072405784,-3.56579353416958,-3.74147465757954,5.20086697517178,-4.24624855333734,-3.66992182399439,-3.19696354527415,-4.05516264168084,-3.11733313400397,-3.0834179576428,-3.83095295966671,-4.05241956037331,-3.08853803527075,-4.63920521426344,-3.11381879192875,-3.60286967644975,3.70143534114568,-3.55566159156974,-3.12864000348101,-3.94218352941666,-3.39755663310188,-3.66679707677223,-3.27907652853266,-3.53851154732097,3.11804373991702,-4.48967292354129,-4.19924136549223,4.7316367154295,-3.0472435217578,3.18014426126337,-3.40336311224083,-3.42033913131333,-3.97662309744017,-3.45912645354786,-4.00814984734461,-3.28381001669089,-4.73941450486437,-3.20574799557026,-3.37636530933563],[0.000133218379507314,0.000197903560490198,0.000325324965313063,0.000739622537111273,0.00167259693333033,0.000635073085208992,0.000255728784757923,0.00014968878496116,0.000751609275259132,0.00164961696261919,0.000738020215621086,0.000238630262009279,0.000532727673239652,0.00010399255366446,0.000733361226195134,3.1070568191017e-05,0.000267130984941697,0.00183799008971839,0.00174581090364012,0.00198142699056002,0.000379068495121749,0.00107417022257526,0.00073462124412741,0.000866028437294138,0.00175127917104714,0.000125178170543346,0.000136920330502781,0.000727049277046554,2.11416848653393e-05,0.000596074117938178,1.36292250794197e-05,0.00183748451102727,0.00155224344993238,9.23231959685866e-05,0.00174186836488545,0.000662268975498318,0.000175711186361212,0.00115220850149545,0.000563109866081463,0.00171323497558657,0.00153955520772088,0.000258339660883706,0.000948004962935646,3.59462343945891e-05,0.00205591992068662,5.22289292888216e-08,0.00205195111257199,0.00147380633432741,0.0013359756488637,0.000322761161721278,0.00133467204487114,0.00166825947825234,0.000666612026546132,6.98571253281245e-05,0.000976185824755325,0.00221717349584643,0.000150434275929629,9.44956469848611e-05,0.00078404224415401,0.000378477462745903,0.000253125749816691,0.000114852336746449,0.00127299538984847,0.00185735962783927,0.0014445490419681,0.000824642405146915,0.00198561595883254,0.000652669763369642,0.00175764054335997,0.000190464720474352,0.000353761662883235,1.42829080294995e-06,8.91192193517848e-06,1.28489993352122e-05,4.98356479336363e-05,0.000621044954970674,2.25557937970504e-05,0.000363079772283181,0.00128971919000386,0.000114897502801987,2.99282382742277e-05,0.000407378478986541,4.57598643485761e-06,0.00137077484037742,0.000492421974000241,0.000787263484568345,1.30887942540155e-06,0.00122057648569835,0.000730384489522136,7.0103896231739e-06,0.000225858408725493,0.002020801742024,0.000344854420061507,0.000115965225762882,0.00138451722954427,0.000169599847550996,0.000583110275052905,0.000459949881356957,8.04128324613104e-06,0.000207394765124345,1.08887592964464e-05,0.000202831311469774,0.000771171433936841,0.000681201131200283,0.000421857918777115,0.00109110531659841,5.12218533633466e-06,0.0012481898467786,0.000366124051094181,6.38482129101028e-05,0.000127838042954321,0.000535605217891121,0.00199767624380227,0.000483399472091032,0.000359285743135109,2.44996140175878e-06,0.000712616551717002,0.000854660162416913,9.35626638728473e-16,8.09945041401529e-08,9.9870810382513e-05,0.000493571566878574,0.00209250409027762,2.11882576856795e-05,3.17501408518181e-12,0.000102506633032138,0.00228583020894003,0.000261379260689059,0.00137261983142185,0.00145966563132691,2.15633757082679e-05,5.84422209889811e-05,0.000198474705881658,0.00133501762000248,4.95187219637573e-05,0.000124952627220843,0.000312211721999734,0.00165290065327837,0.00207449830011635,0.00045785812704136,8.1548163296739e-05,0.0019729355676485,0.000428015133498875,0.00198470006871534,8.01660406542807e-06,0.00141760429446706,0.00216098015828708,0.000704251700262734,4.39501744049744e-05,0.00098307690662596,0.000700207394821647,0.000102125450798791,0.00186835480912367,0.000231357833370148,0.00122156055835201,0.00147493656061591,0.000162645722479189,0.000883403168079507,0.00016877791153493,2.83797669537129e-05,0.00162640423586426,0.00191738081738088,0.00157199053115799,0.00158856179143075,0.0017384562263091,0.0021404024841535,1.58668061791633e-05,3.14631961222115e-05,0.00170138859286471,0.000901320450455622,0.000587564597237514,0.00169795082682578,2.30505601417338e-05,0.00227438805914208,0.0018043950118257,0.000883072292336068,0.00158625701523817,0.000963016708577425,1.14997087155112e-05,0.000749519995228843,0.00186810084112396,0.000501607818638855,0.00121815362373968,0.00130652109543946,2.00253568834801e-05,0.00140749013253707,0.00151368595339594,0.0012235077499395,0.00163505341241088,0.00110928763010468,0.00214132790211356,0.000265579866584157,6.35601860608723e-05,0.000108793871347503,0.00164275362329816,0.00123440863149437,0.00039962319581374,0.00154804395008122,2.03640632331635e-05,0.00112315905504138,0.00163534438573467,4.38167065856046e-06,8.48954672119395e-07,0.00134862270214836,0.000905196646657137,0.00100035138385597,0.000156769951724875,0.00153962313377207,0.000133740429688676,0.000396515102933653,0.000235841024514799,0.00178391380159103,2.15890833859798e-07,0.000211090796118694,0.0019433330563836,0.00219988111511817,1.8677347707647e-05,0.00127278768293664,0.0018148677255373,0.00203986579792666,0.00215119409758665,0.00201609291459885,4.80606061028972e-05,0.00112266828176843,0.00035180643405958,0.000240483389949455,0.000369471366747102,0.00188753500228603,0.000844330717572489,0.00190837628581233,2.12314609803337e-06,0.000826261956088588,0.00109760475680231,0.000302496286764041,0.00196280370715889,0.000280932503236662,0.000956898293183501,0.000753884511009518,0.000558381458471916,1.6955836482906e-07,0.000175574831623764,0.000467157787295023,0.000869770812536042,0.00133813578245651,0.00226992808005302,0.000538311753371725,0.00157112186054362,0.000213074631106871,0.000855308758727595,2.84224936535012e-07,0.000298030789636357,2.27484820712397e-05,0.000846686278879708,0.000178499536639593,9.51184283451875e-05,0.00163344137751473,0.000649297905533909,0.000231240668918196,0.00151886360887788,9.87466877660904e-05,0.000864011566873686,0.000512047690842168,0.00148942946191346,0.000169611095190348,1.74771950438794e-05,6.04245451289728e-05,4.94436860069353e-12,0.000374766958714304,0.000927786832396183,0.000406920580494454,0.000500793393944687,6.76197349694146e-05,0.00091010056911961,0.000123291861051325,9.75012536933023e-06,0.000330200115136062,0.000631080155123955,0.000992559936717221,4.30966879327573e-05,0.00179311707083839,0.00161033588617442,0.00200077237128331,0.0007789945290483,9.37167967936968e-05,0.00207388103518499,0.00170951847303085,0.00162826709738772,0.000952722094946178,0.00136943208631493,0.00121963438754371,0.000367695456795487,0.000428944021627256,0.000535557443345677,4.49975840313936e-05,6.07916445263896e-05,0.00103233411928984,0.00100403818933497,0.00149889853018309,0.00195591576061672,0.000656794166832271,0.00074653554451572,0.0016629386503224,0.000441422613621928,0.00119283166367293,0.000327858882871074,0.00142385511764023,0.000170345034915224,5.53209860063267e-06,0.000813855887159972,0.00201224461464882,0.00110975618246563,0.000309817118142191,5.98405449709956e-05,0.000444394613590518,0.000103883264093172,0.000768567515866431,8.66209015239505e-05,0.00142023431125001,0.000947414526718305,0.000974375981943757,0.000731224116170845,0.000399490453402486,0.000329457516169045,0.000976658004151125,1.36879155945528e-06,0.000666088822483274,0.00170138542529207,0.00151957676815345,0.000175334541455964,0.000184185160218099,0.00190558176842749,0.000386711985942671,0.00124674615695053,1.61933454721169e-05,0.00178940709172489,0.000248668671221656,6.66713781016901e-05,0.000477707302892156,0.000421387964662886,0.00184433060039886,0.000906449747645771,0.00033319732885803,0.000799948122955211,0.00176365804125461,0.000719058964939153,2.86100876024681e-05,0.000294057606815461,0.00122752374125858,0.00222187919660983,6.8823677954538e-05,0.000651661485449937,0.00117806789482851,0.00186362582433343,7.34674427459524e-06,4.60117082355335e-05,0.000981544191165046,1.79091645340985e-05,7.65108170790648e-05,0.000553233436834963,0.000732784637712171,0.000181863317842101,0.000128882442518683,0.00191945073354653,0.000539139768289852,0.00197212892446878,0.000406300007503696,0.000478266556500348,0.00150485420591493,1.92871571479529e-05,0.00213208992789822,0.000132078748117623,0.00196335558807167,0.00164933135056553,0.00107177423503319,1.51663930861216e-06,1.41612567540599e-06,1.32615689761313e-06,0.00130188041758217,0.000456055200409573,5.13595602658625e-06,0.000108638572207539,0.000403654845237433,0.000243889732926135,0.00218670122438708,9.66992495591641e-05,0.000239992930255227,2.14691521211408e-06,8.58123895855383e-06,0.00181109406144513,0.000157364300170597,0.0011421358392912,5.50815618266877e-05,0.00218163841556439,0.00092959846629278,0.00210378002501143,1.60995301518905e-05,0.000362756883151054,0.000182943627916213,1.98361041793889e-07,2.17379479036635e-05,0.00024262465788656,0.00138882477414779,5.00994312416697e-05,0.00182495224996003,0.002046375263184,0.000127647910261049,5.06906551161508e-05,0.00201143944053906,3.49751627343136e-06,0.00184682877066653,0.000314723342212416,0.000214383290443119,0.000377029125524457,0.00175617325055527,8.07431523864197e-05,0.000679905133544991,0.000245607545581966,0.00104147388645096,0.000402389681317412,0.001820557826944,7.13326191610984e-06,2.67810731428506e-05,2.22716738218082e-06,0.00230950522493492,0.00147201751223081,0.000665617569959836,0.000625431140669624,6.99008220608167e-05,0.000541930410016662,6.11962881919835e-05,0.00102413935664843,2.14336646352666e-06,0.00134711909019505,0.00073450324979684],[0.0107714737312762,0.0132725448769589,0.0173335657433312,0.0268453785344927,0.0419766593644356,0.0252064619503454,0.0148511126895733,0.0116893788775178,0.0270076678392176,0.0418305179088842,0.0268453785344927,0.0145796358500126,0.0223833747356232,0.00940535932076764,0.0268404331380856,0.00462886541349976,0.015193419009519,0.0438021223541096,0.0429026225184078,0.0450808363986794,0.0184257346923937,0.0330409299415969,0.0268404331380856,0.029402331621857,0.0429066499014024,0.010543633082254,0.0109476733823745,0.0268404331380856,0.00374775808076598,0.0240039482384301,0.00313301811513159,0.0438021223541096,0.0405863844231133,0.00934581783777417,0.0429012846833618,0.0257181716815036,0.0124281873737796,0.0344538444593519,0.0230124231938625,0.0424805274620724,0.0404480991858119,0.0149241875297948,0.0309717921333579,0.00515104870532333,0.0456439042351231,0.000144073501443214,0.0456439042351231,0.0395010922568832,0.0376657913867987,0.0172880905749155,0.0376657913867987,0.0419766593644356,0.0257181716815036,0.00763649178830744,0.0313268733075684,0.0477818210022842,0.0116893788775178,0.00935946947048962,0.0277279555192158,0.0184257346923937,0.0148511126895733,0.0100617384596597,0.036578726905177,0.0439561342513232,0.0392590003179213,0.0285798571269012,0.0450808363986794,0.0256283208861944,0.0429066499014024,0.0129727637389753,0.0179880469504775,0.000927044748220574,0.00234128920554189,0.00301650763116449,0.00650349558024216,0.0249185819387143,0.00379611762095359,0.0182101009426028,0.0369630169935133,0.0100617384596597,0.00453595070083221,0.0190466700726165,0.00174108394214548,0.0380539879897204,0.0214412152320401,0.0277529242451346,0.000927044748220574,0.0357147738434721,0.0268404331380856,0.00218634477728767,0.0143225384015925,0.0452282483194582,0.0177809517334517,0.0100752779611625,0.0382876268440888,0.0123055789811555,0.0237418404979105,0.0205469108943023,0.00221818798344525,0.01370295711606,0.00273060386538613,0.0134821728358885,0.0274487277485778,0.0260985183391108,0.0195579003184315,0.0334423779537412,0.00177094183741727,0.0360537349983118,0.0182754579742406,0.00741580190789552,0.0106057516237442,0.0223833747356232,0.0451615992597158,0.0211659911708431,0.0181851325218018,0.00108131496428026,0.0267637441437512,0.0292181945628492,1.032370433173e-11,0.000178738671736489,0.00940535932076764,0.0214412152320401,0.0461773802642465,0.00374775808076598,1.7516552707948e-08,0.00940535932076764,0.0488795552818687,0.0150211393877244,0.0380539879897204,0.0393788522642081,0.00374775808076598,0.00724552209429683,0.0132725448769589,0.0376657913867987,0.00650349558024216,0.010543633082254,0.0168900782059467,0.0418305179088842,0.0458717720310296,0.0205366120885137,0.00848870220581338,0.0450711616013117,0.0196074780680361,0.0450808363986794,0.00221818798344525,0.0388855220603786,0.0469375099735033,0.0267034819955292,0.00613855980233528,0.031350493028066,0.0267034819955292,0.00940535932076764,0.0439561342513232,0.0144226120531424,0.0357147738434721,0.0395010922568832,0.0123055789811555,0.029717898038382,0.0123055789811555,0.00444624938881174,0.0417694211856396,0.0445865352683897,0.0409088290584841,0.0411459878090302,0.0429012846833618,0.0467869545978633,0.00350347792037917,0.00462886541349976,0.0423772499631358,0.0302168172674424,0.0238352491394071,0.0423772499631358,0.00379611762095359,0.0487293162030558,0.0435660712483256,0.029717898038382,0.0411459878090302,0.0311610743766666,0.00281972857704336,0.0270076678392176,0.0439561342513232,0.0216200807455513,0.0357147738434721,0.0372510433257855,0.00374775808076598,0.0387287933227284,0.0401124642579072,0.0357147738434721,0.0417694211856396,0.0336402464761698,0.0467869545978633,0.0151834624242984,0.00741580190789552,0.00968089981006736,0.0418305179088842,0.035843328526076,0.0190466700726165,0.040572724335383,0.00374775808076598,0.033768220744759,0.0417694211856396,0.00172669121594843,0.000780613821013784,0.0377682814606726,0.0302168172674424,0.0317180378432953,0.012058039500572,0.0404480991858119,0.0107714737312762,0.0190466700726165,0.0145796358500126,0.0433561781646596,0.000297767432601127,0.0137821055880099,0.0448592823099092,0.0475019339025712,0.00374775808076598,0.036578726905177,0.0436279966962497,0.0455625085310178,0.0469096357169389,0.0452145715847229,0.00646708204560205,0.033768220744759,0.0179714453398769,0.0145796358500126,0.0182813769537557,0.0443128962026045,0.0291038517170053,0.0445865352683897,0.00102394020395763,0.0285798571269012,0.0335483958076362,0.0168281927605701,0.0450388057355151,0.0158964576446837,0.0310541640205493,0.0270076678392176,0.0229040186348666,0.000297767432601127,0.0124281873737796,0.0207847541331181,0.0294388071948549,0.0376657913867987,0.0487283782788035,0.0223833747356232,0.0409088290584841,0.0138297969390189,0.0292181945628492,0.000348459772191924,0.0166927499129318,0.00379611762095359,0.0291038517170053,0.0125449929126196,0.00935946947048962,0.0417694211856396,0.0256283208861944,0.0144226120531424,0.0401124642579072,0.00940535932076764,0.029402331621857,0.0219841798472859,0.039792650563567,0.0123055789811555,0.00370852634834934,0.00726064348290694,1.81853877133508e-08,0.0184257346923937,0.0307101481349537,0.0190466700726165,0.0216200807455513,0.00763649178830744,0.0302471375893548,0.010543633082254,0.00250192751919046,0.0173335657433312,0.0251384058903889,0.0315616897456421,0.00609652377756466,0.0433887143851553,0.0416122861078419,0.0451615992597158,0.027638024545077,0.00935946947048962,0.0458717720310296,0.0424805274620724,0.0417694211856396,0.0310098395151508,0.0380539879897204,0.0357147738434721,0.0182754579742406,0.0196074780680361,0.0223833747356232,0.00620629177752997,0.00726064348290694,0.0322684834907764,0.0317437174244185,0.0399489043044449,0.0450388057355151,0.0256988185703095,0.0270074531088081,0.0419766593644356,0.0200438564555734,0.0352860712519225,0.0173335657433312,0.038888161802085,0.0123055789811555,0.00184973260482972,0.0284179932244403,0.0452145715847229,0.0336402464761698,0.0168900782059467,0.00726064348290694,0.0200961072391712,0.00940535932076764,0.0274445759549197,0.00893247689173149,0.0388855220603786,0.0309717921333579,0.0313268733075684,0.0268404331380856,0.0190466700726165,0.0173335657433312,0.0313268733075684,0.000927044748220574,0.0257181716815036,0.0423772499631358,0.0401124642579072,0.0124281873737796,0.0126930140024864,0.0445865352683897,0.0187148247933835,0.0360537349983118,0.00350347792037917,0.0433887143851553,0.0146727813810682,0.00763649178830744,0.0210811060279022,0.0195579003184315,0.0438234594742677,0.0302168172674424,0.0173419779557524,0.0281102789448656,0.0429585051373143,0.0268404331380856,0.00444624938881174,0.0165542430285806,0.0357374590001244,0.0477898928955026,0.00763649178830744,0.0256283208861944,0.035037199869374,0.0439561342513232,0.00219091827907794,0.0062678171440849,0.031350493028066,0.00372848531074044,0.00813715009379463,0.0227775288881977,0.0268404331380856,0.0126930140024864,0.0106126035130683,0.0445865352683897,0.0223833747356232,0.0450711616013117,0.0190466700726165,0.0210811060279022,0.0400109911037718,0.00374775808076598,0.0467703384978707,0.0107714737312762,0.0450388057355151,0.0418305179088842,0.0330409299415969,0.000929699896179252,0.000927044748220574,0.000927044748220574,0.0372148925585536,0.0205366120885137,0.00177094183741727,0.00968089981006736,0.0190466700726165,0.0146254310494944,0.0473099241370335,0.00935946947048962,0.0145796358500126,0.00102394020395763,0.0023093997724069,0.0436279966962497,0.012058039500572,0.0342454533987475,0.00690647674085991,0.0472931203876965,0.0307101481349537,0.0462412525816257,0.00350347792037917,0.0182101009426028,0.0126930140024864,0.000297767432601127,0.00374775808076598,0.0146254310494944,0.0383107313948668,0.00650349558024216,0.0436800935489349,0.0456155649575197,0.0106057516237442,0.00650372893664661,0.0452145715847229,0.0014842920985016,0.0438234594742677,0.0169397919901063,0.0138333638991192,0.0184257346923937,0.0429066499014024,0.00848495184220719,0.0260985183391108,0.0146488305835211,0.0323707686284503,0.0190466700726165,0.0436696414402176,0.00218634477728767,0.00428675784832911,0.00102394020395763,0.0492902913963866,0.0395010922568832,0.0257181716815036,0.02500364929764,0.00763649178830744,0.0223957308768683,0.00726064348290694,0.0321947397756662,0.00102394020395763,0.0377682814606726,0.0268404331380856]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>baseMean<\/th>\n      <th>log2FoldChange<\/th>\n      <th>lfcSE<\/th>\n      <th>stat<\/th>\n      <th>pvalue<\/th>\n      <th>padj<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[0,1,2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>We can also visualise this in a heatmap below, showing immune specific differentially expressed genes:</p>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-1"></a><span class="co">#HighExprGenes=rownames(CD45res2)[which(CD45res2$baseMean&gt;100 &amp; CD45res2$log2FoldChange&lt;0) ]</span></span>
<span id="cb383-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-2"></a></span>
<span id="cb383-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-3"></a>colSide=CD45ddsChar<span class="op">$</span>SizeCat</span>
<span id="cb383-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-4"></a></span>
<span id="cb383-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-5"></a></span>
<span id="cb383-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-6"></a>t2=<span class="kw">assay</span>(vsd)[namId, <span class="kw">match</span>(<span class="kw">colnames</span>(CD45ddsChar), <span class="kw">colnames</span>(vsd))]</span>
<span id="cb383-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-7"></a></span>
<span id="cb383-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-8"></a><span class="kw">colnames</span>(t2)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(t2), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb383-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-9"></a></span>
<span id="cb383-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb383-10"></a><span class="kw">heatmap.2</span>(t2, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span> <span class="kw">palette</span>()[colSide], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;immune genes DEG&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:2d"></span>
<img src="RatDCIS-bookdown_files/figure-html/2d-1.png" alt="Differential CD45 genes big vs small" width="672" />
<p class="caption">
Figure 17.2: Differential CD45 genes big vs small
</p>
</div>
<div class="sourceCode" id="cb384"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb384-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb384-1"></a></span>
<span id="cb384-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb384-2"></a><span class="kw">write.csv</span>(t2, <span class="dt">file=</span><span class="st">&quot;nature-tables/2d.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="gsea-1" class="section level3">
<h3><span class="header-section-number">17.1.3</span> GSEA</h3>
<p>Run GSEA. Here, we will look specifically at the Process Network pathways which are enriched, focusing specifically on immune related terms</p>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-1"></a>cd45Genes=<span class="kw">rownames</span>(CD45res)</span>
<span id="cb385-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-2"></a></span>
<span id="cb385-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-3"></a>l1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(cd45Genes, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb385-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-4"></a>l2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(cd45Genes, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb385-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-5"></a>l3=Mouse2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(cd45Genes, Mouse2Hum<span class="op">$</span>MGI.symbol)]</span>
<span id="cb385-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-6"></a>cd45GenesConv=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(l1)<span class="op">==</span>F, l1, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l2)<span class="op">==</span>F, l2, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l3)<span class="op">==</span>F, l3, cd45Genes)))</span>
<span id="cb385-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-7"></a></span>
<span id="cb385-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-8"></a></span>
<span id="cb385-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-9"></a>hits=cd45GenesConv[<span class="kw">match</span>(<span class="kw">rownames</span>(CD45res2), cd45Genes)]</span>
<span id="cb385-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-10"></a><span class="co">#hits=epGenesConv[match(hits, rownames(Epdds))]</span></span>
<span id="cb385-11"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-11"></a>fcTab=CD45res<span class="op">$</span>log2FoldChange</span>
<span id="cb385-12"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-12"></a><span class="kw">names</span>(fcTab)=cd45GenesConv</span>
<span id="cb385-13"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-13"></a></span>
<span id="cb385-14"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-14"></a></span>
<span id="cb385-15"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-15"></a>  gscacd=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb385-16"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-16"></a>  gscacd &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gscacd, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb385-17"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-17"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb385-18"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-18"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb385-19"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-19"></a>  gscacd &lt;-<span class="st"> </span><span class="kw">analyze</span>(gscacd, </span>
<span id="cb385-20"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-20"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb385-21"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-21"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb385-22"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-22"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb385-23"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-23"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb385-24"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-24"></a></span>
<span id="cb385-25"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-25"></a>A1=HTSanalyzeR2<span class="op">::</span><span class="kw">summarize</span>(gscacd)  </span>
<span id="cb385-26"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-26"></a></span>
<span id="cb385-27"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-27"></a>PNresults=gscacd<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks</span>
<span id="cb385-28"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-28"></a></span>
<span id="cb385-29"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-29"></a>Ax1=<span class="kw">which</span>(PNresults<span class="op">$</span>Adjusted.Pvalue<span class="op">&lt;</span><span class="fl">0.1</span>)</span>
<span id="cb385-30"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-30"></a></span>
<span id="cb385-31"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-31"></a>Lx1=PNresults[Ax1, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</span>
<span id="cb385-32"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-32"></a>Lx1<span class="op">$</span>Group=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(Lx1), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb385-33"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-33"></a>Lx1<span class="op">$</span>Process=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(Lx1), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb385-34"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-34"></a></span>
<span id="cb385-35"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-35"></a><span class="co">## replace certain groups</span></span>
<span id="cb385-36"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-36"></a>Lx1<span class="op">$</span>Group[<span class="kw">grep</span>(<span class="st">&quot;ymphocyte&quot;</span>, Lx1<span class="op">$</span>Process)]=<span class="st">&quot;NImmune response&quot;</span></span>
<span id="cb385-37"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-37"></a></span>
<span id="cb385-38"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-38"></a><span class="co"># plot for inflammation, immune response, cell adhesion, transcription?</span></span>
<span id="cb385-39"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-39"></a></span>
<span id="cb385-40"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-40"></a>TestGrp=<span class="kw">c</span>(<span class="st">&quot;NInflammation&quot;</span>, <span class="st">&quot;NCell adhesion&quot;</span>, <span class="st">&quot;NImmune response&quot;</span>, <span class="st">&quot;NTranscription&quot;</span>)</span>
<span id="cb385-41"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-41"></a></span>
<span id="cb385-42"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-42"></a><span class="co">#pdf(&quot;~/Desktop/2E-process-networks-significant-pathways.pdf&quot;, width=6, height=6)</span></span>
<span id="cb385-43"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-43"></a></span>
<span id="cb385-44"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-44"></a><span class="co"># par(mfrow=c(2,2), oma=c(0, 2, 0, 0))</span></span>
<span id="cb385-45"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-45"></a><span class="co"># </span></span>
<span id="cb385-46"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-46"></a><span class="co"># for (i in TestGrp){</span></span>
<span id="cb385-47"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-47"></a><span class="co">#         x1=which(Lx1$Group==i)</span></span>
<span id="cb385-48"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-48"></a><span class="co">#         barplot(Lx1$Observed.score[x1], names.arg = Lx1$Process[x1], horiz = T, las=2,</span></span>
<span id="cb385-49"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-49"></a><span class="co">#                 main=i)</span></span>
<span id="cb385-50"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-50"></a><span class="co"># }</span></span>
<span id="cb385-51"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-51"></a></span>
<span id="cb385-52"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-52"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb385-53"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb385-53"></a><span class="kw">barplot</span>(Lx1<span class="op">$</span>Observed.score, <span class="dt">names.arg =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(Lx1<span class="op">$</span>Process), Lx1<span class="op">$</span>Group, Lx1<span class="op">$</span>Process), <span class="dt">horiz =</span> T, <span class="dt">las=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure"><span id="fig:char-gsea-cd45"></span>
<img src="RatDCIS-bookdown_files/figure-html/char-gsea-cd45-1.png" alt="gsea for cd45 samples" width="672" />
<p class="caption">
Figure 17.3: gsea for cd45 samples
</p>
</div>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb386-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb386-1"></a><span class="co">#dev.off()</span></span>
<span id="cb386-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb386-2"></a></span>
<span id="cb386-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb386-3"></a><span class="kw">write.csv</span>(gscacd<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp_3_cd45_big_small.csv&quot;</span>)</span>
<span id="cb386-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb386-4"></a><span class="kw">write.csv</span>(Lx1, <span class="dt">file=</span><span class="st">&quot;nature-tables/2e.csv&quot;</span>)</span></code></pre></div>
<p>In the following hairball, we can look at all the terms</p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb387-1"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(gscacd<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb387-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb387-2"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(gscacd<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb387-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb387-3"></a></span>
<span id="cb387-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb387-4"></a><span class="co">## check whether this runs:</span></span>
<span id="cb387-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb387-5"></a>gscacd<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks<span class="op">$</span>Gene.Set.Term=TermsA</span>
<span id="cb387-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb387-6"></a><span class="kw">viewEnrichMap</span>(gscacd, <span class="dt">gscs=</span><span class="kw">c</span>(<span class="st">&quot;ProcessNetworks&quot;</span>),</span>
<span id="cb387-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb387-7"></a>              <span class="dt">allSig =</span> <span class="ot">TRUE</span>, <span class="dt">gsNameType=</span><span class="st">&quot;term&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:cd45-hairball"></span>
<div id="htmlwidget-52e9c4ae859c2a7c5a0e" style="width:672px;height:480px;" class="forceGraph html-widget"></div>
<script type="application/json" data-for="htmlwidget-52e9c4ae859c2a7c5a0e">{"x":{"nodes":{"id":["ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NBlood coagulation","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NTranscription_mRNA processing"],"size":[3,4.29347826086957,4.47826086956522,5.95652173913043,6.32608695652174,6.32608695652174,6.32608695652174,7.43478260869565,8.35869565217391,9.46739130434783,9.65217391304348,9.83695652173913,9.83695652173913,10.945652173913,11.8695652173913,12.054347826087,12.4239130434783,12.4239130434783,12.6086956521739,12.7934782608696,14.8260869565217,15.5652173913043,15.5652173913043,16.4891304347826,16.4891304347826,17.2282608695652,17.2282608695652,17.5978260869565,17.7826086956522,19.2608695652174,19.8152173913043,20],"color":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"scheme":["neg","neg","pos","pos","neg","pos","neg","pos","pos","neg","neg","pos","neg","pos","neg","neg","neg","pos","neg","pos","pos","pos","pos","neg","neg","neg","neg","pos","pos","pos","pos","pos"],"label":["Complement system","Visual perception","Circadian rhythm","IL-12,15,18 signaling","ECM remodeling","Protein folding nucleus","Blood coagulation","Response to unfolded proteins","ERBB-family signaling","Iron transport","Connective tissue degradation","Endoplasmic reticulum stress pathway","Potassium transport","IL-2 signaling","Muscle contraction","Chemotaxis","Skeletal muscle development","Chromatin modification","Response to hypoxia and oxidative stress","IFN-gamma signaling","BCR pathway","G1-S Interleukin regulation","Androgen receptor nuclear signaling","Cell-matrix interactions","Calcium transport","Platelet-endothelium-leucocyte interactions","Elongation-Termination","Transcription by RNA polymerase II","T helper cell differentiation","Nuclear receptors transcriptional regulation","G1-S","mRNA processing"],"label_id":["NInflammation_Complement system","NNeurophysiological process_Visual perception","NNeurophysiological process_Circadian rhythm","NInflammation_IL-12,15,18 signaling","NProteolysis_ECM remodeling","NProtein folding_Protein folding nucleus","NBlood coagulation","NProtein folding_Response to unfolded proteins","NSignal transduction_ERBB-family signaling","NTransport_Iron transport","NProteolysis_Connective tissue degradation","NApoptosis_Endoplasmic reticulum stress pathway","NTransport_Potassium transport","NInflammation_IL-2 signaling","NMuscle contraction","NChemotaxis","NDevelopment_Skeletal muscle development","NTranscription_Chromatin modification","NResponse to hypoxia and oxidative stress","NInflammation_IFN-gamma signaling","NImmune response_BCR pathway","NCell cycle_G1-S Interleukin regulation","NSignal transduction_Androgen receptor nuclear signaling","NCell adhesion_Cell-matrix interactions","NTransport_Calcium transport","NCell adhesion_Platelet-endothelium-leucocyte interactions","NTranslation_Elongation-Termination","NTranscription_Transcription by RNA polymerase II","NImmune response_T helper cell differentiation","NTranscription_Nuclear receptors transcriptional regulation","NCell cycle_G1-S","NTranscription_mRNA processing"],"label_term":["Complement system","Visual perception","Circadian rhythm","IL-12,15,18 signaling","ECM remodeling","Protein folding nucleus","Blood coagulation","Response to unfolded proteins","ERBB-family signaling","Iron transport","Connective tissue degradation","Endoplasmic reticulum stress pathway","Potassium transport","IL-2 signaling","Muscle contraction","Chemotaxis","Skeletal muscle development","Chromatin modification","Response to hypoxia and oxidative stress","IFN-gamma signaling","BCR pathway","G1-S Interleukin regulation","Androgen receptor nuclear signaling","Cell-matrix interactions","Calcium transport","Platelet-endothelium-leucocyte interactions","Elongation-Termination","Transcription by RNA polymerase II","T helper cell differentiation","Nuclear receptors transcriptional regulation","G1-S","mRNA processing"]},"links":{"source":["ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NInflammation_Complement system","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NProtein folding_Protein folding nucleus","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NBlood coagulation","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation"],"target":["ProcessNetworks.NProteolysis_ECM remodeling","ProcessNetworks.NBlood coagulation","ProcessNetworks.NMuscle contraction","ProcessNetworks.NChemotaxis","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NNeurophysiological process_Circadian rhythm","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NInflammation_IL-12,15,18 signaling","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NSignal transduction_ERBB-family signaling","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NChemotaxis","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NBlood coagulation","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NProtein folding_Response to unfolded proteins","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NTranscription_mRNA processing","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NProteolysis_Connective tissue degradation","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NMuscle contraction","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NTransport_Iron transport","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NChemotaxis","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NApoptosis_Endoplasmic reticulum stress pathway","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NInflammation_IL-2 signaling","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NMuscle contraction","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NMuscle contraction","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NChemotaxis","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NTranscription_Chromatin modification","ProcessNetworks.NResponse to hypoxia and oxidative stress","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NTranscription_mRNA processing","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NImmune response_BCR pathway","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NCell cycle_G1-S Interleukin regulation","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NSignal transduction_Androgen receptor nuclear signaling","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranscription_Transcription by RNA polymerase II","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NTranscription_mRNA processing","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NTranslation_Elongation-Termination","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell adhesion_Platelet-endothelium-leucocyte interactions","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NImmune response_T helper cell differentiation","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NTranscription_mRNA processing","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NTranscription_mRNA processing","ProcessNetworks.NTranscription_Nuclear receptors transcriptional regulation","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NCell cycle_G1-S","ProcessNetworks.NTranscription_mRNA processing"],"weight":[0.0136986301369863,0.0277777777777778,0.00970873786407767,0.0396039603960396,0.0380952380952381,0.00840336134453781,0.015748031496063,0.0230769230769231,0.0441176470588235,0.0101010101010101,0.00952380952380952,0.018348623853211,0.0178571428571429,0.00884955752212389,0.0175438596491228,0.0241935483870968,0.0234375,0.0303030303030303,0.0214285714285714,0.00666666666666667,0.038961038961039,0.0689655172413793,0.0306122448979592,0.0808080808080808,0.0275229357798165,0.018018018018018,0.0176991150442478,0.0087719298245614,0.00869565217391304,0.0636363636363636,0.0666666666666667,0.056,0.0393700787401575,0.0300751879699248,0.0588235294117647,0.0482758620689655,0.074468085106383,0.00943396226415094,0.0686274509803922,0.127450980392157,0.0431034482758621,0.0508474576271186,0.116071428571429,0.105691056910569,0.176470588235294,0.00719424460431655,0.0275862068965517,0.235772357723577,0.0191082802547771,0.00617283950617284,0.0952380952380952,0.0103092783505155,0.549295774647887,0.025,0.016260162601626,0.016,0.00709219858156028,0.245762711864407,0.00684931506849315,0.110294117647059,0.113636363636364,0.0373831775700935,0.00862068965517241,0.00819672131147541,0.0245901639344262,0.0416666666666667,0.024390243902439,0.0072992700729927,0.0364963503649635,0.00684931506849315,0.0132450331125828,0.0065359477124183,0.0188679245283019,0.0184049079754601,0.00925925925925926,0.0679611650485437,0.00862068965517241,0.00826446280991736,0.0165289256198347,0.00806451612903226,0.00793650793650794,0.00709219858156028,0.0215827338129496,0.027972027972028,0.0208333333333333,0.170542635658915,0.0198675496688742,0.0175438596491228,0.114285714285714,0.0158730158730159,0.0153846153846154,0.0136986301369863,0.00657894736842105,0.0129032258064516,0.00598802395209581,0.00840336134453781,0.0608695652173913,0.207547169811321,0.0551181102362205,0.00740740740740741,0.0538461538461538,0.0866141732283465,0.182539682539683,0.204724409448819,0.0408163265306122,0.00636942675159236,0.0128205128205128,0.0253164556962025,0.0188679245283019,0.13013698630137,0.0484848484848485,0.0114942528735632,0.0151515151515152,0.0072463768115942,0.00719424460431655,0.0140845070422535,0.00649350649350649,0.0192307692307692,0.0123456790123457,0.0120481927710843,0.0111111111111111,0.0078125,0.0143884892086331,0.0141843971631206,0.013986013986014,0.0191082802547771,0.299212598425197,0.00609756097560976,0.0903225806451613,0.088,0.0142857142857143,0.00699300699300699,0.0507246376811594,0.0428571428571429,0.0680272108843537,0.0522875816993464,0.0125786163522013,0.0184049079754601,0.0179640718562874,0.0679012345679012,0.0225988700564972,0.00546448087431694,0.0444444444444444,0.00699300699300699,0.00606060606060606,0.0506329113924051,0.00684931506849315,0.112781954887218,0.00671140939597315,0.0413793103448276,0.178294573643411,0.347107438016529,0.31496062992126,0.0372670807453416,0.036144578313253,0.0292397660818713,0.0114942528735632,0.201342281879195,0.050561797752809,0.0382513661202186,0.0132450331125828,0.192307692307692,0.0129032258064516,0.0120481927710843,0.0117647058823529,0.0290697674418605,0.0925925925925926,0.0111731843575419,0.00546448087431694,0.00523560209424084,0.012987012987013,0.00645161290322581,0.0261437908496732,0.112676056338028,0.0833333333333333,0.101910828025478,0.0236686390532544,0.0171428571428571,0.0769230769230769,0.0167597765363128,0.0632183908045977,0.0376344086021505,0.0463576158940397,0.0127388535031847,0.0191082802547771,0.0118343195266272,0.0355029585798817,0.0526315789473684,0.0169491525423729,0.0337078651685393,0.00540540540540541,0.0108108108108108,0.0209424083769634,0.00505050505050505,0.00588235294117647,0.0542168674698795,0.021978021978022,0.0209424083769634,0.0050761421319797,0.0522875816993464,0.0487804878048781,0.0797546012269939,0.0292397660818713,0.0109289617486339,0.0502793296089385,0.0208333333333333,0.0205128205128205,0.138157894736842,0.156862745098039,0.0411764705882353,0.0167597765363128,0.0282485875706215,0.087719298245614,0.0108695652173913,0.0053475935828877,0.125,0.031413612565445,0.0362694300518135,0.220779220779221,0.0217391304347826,0.0265957446808511,0.031413612565445,0.0154639175257732,0.00505050505050505,0.149425287356322,0.0452261306532663,0.0193236714975845,0.0491803278688525,0.0307692307692308,0.0151515151515152,0.192982456140351,0.0927835051546392,0.119791666666667,0.0154639175257732,0.0154639175257732,0.0100502512562814,0.0252525252525253,0.0251256281407035,0.0927835051546392,0.0336538461538462,0.00934579439252336,0.00497512437810945,0.0899470899470899,0.0048780487804878,0.00480769230769231,0.00462962962962963,0.00980392156862745,0.0195121951219512,0.00930232558139535,0.00917431192660551,0.0390243902439024,0.0184331797235023,0.00900900900900901,0.00471698113207547,0.00454545454545455,0.00900900900900901,0.00896860986547085,0.205405405405405,0.00892857142857143,0.00888888888888889,0.0516431924882629,0.00888888888888889,0.00427350427350427,0.00425531914893617]},"options":{"title":"Enrichment Map of GSEA on ProcessNetworks","legendTitle":"-Log10(Adjusted p-values)","type":"GSEA","nodeScheme":"dual","label":{"text":"term"},"colorScaler":"log10","nPermutations":100,"colorDomain":{"neg":[0,0.05],"pos":[0,0.05]}}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 17.4: gsea hairball for all cd45 samples
</p>
</div>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb388-1"></a></span>
<span id="cb388-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb388-2"></a><span class="kw">save</span>(gscacd, <span class="dt">file=</span><span class="st">&quot;figure-outputs/2e.Rdata&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="epithelial-samples-1" class="section level2">
<h2><span class="header-section-number">17.2</span> Epithelial samples</h2>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-1"></a>infoTableFinal<span class="op">$</span>TumSize=Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">match</span>(infoTableFinal<span class="op">$</span>TumorID, Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb389-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-2"></a><span class="co">#infoTableFinal$SizeCat=factor(ifelse(infoTableFinal$Cohort==&quot;Progression&quot;, ifelse(infoTableFinal$TumSize&gt;X2a, &quot;big&quot;, &quot;small&quot;), ifelse(infoTableFinal$TumSize&gt;X1a, &quot;big&quot;, &quot;small&quot;)))</span></span>
<span id="cb389-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-3"></a></span>
<span id="cb389-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-4"></a>epidx=<span class="kw">as.character</span>(infoTableFinal<span class="op">$</span>SampleID[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Cohort<span class="op">!=</span><span class="st">&quot;Progression&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;Ep&quot;</span> <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(infoTableFinal<span class="op">$</span>SizeCat))])</span>
<span id="cb389-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-5"></a>EpddsChar=<span class="kw">DESeqDataSetFromMatrix</span>(allstarFinal[ ,epidx], infoTableFinal[epidx, ], <span class="dt">design=</span><span class="op">~</span>(SizeCat)) <span class="co">## change class</span></span>
<span id="cb389-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-6"></a></span>
<span id="cb389-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-7"></a>a1x=<span class="kw">rowSums</span>(<span class="kw">counts</span>(EpddsChar))</span>
<span id="cb389-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-8"></a>a1b=<span class="kw">apply</span>(<span class="kw">counts</span>(EpddsChar), <span class="dv">1</span>, <span class="cf">function</span>(c) <span class="kw">sum</span>(c<span class="op">!=</span><span class="dv">0</span>))</span>
<span id="cb389-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-9"></a> <span class="co"># par(mfrow=c(1,2))</span></span>
<span id="cb389-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-10"></a> <span class="co"># hist(log10(a1x+1), main=&quot;log10 total counts&quot;)</span></span>
<span id="cb389-11"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-11"></a> <span class="co"># hist((a1b+1), main=&quot;Non-zero entries&quot;)</span></span>
<span id="cb389-12"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-12"></a>sd1vals=<span class="kw">mean</span>(<span class="kw">log10</span>(a1x<span class="op">+</span><span class="dv">1</span>))<span class="op">-</span><span class="kw">sd</span>(<span class="kw">log10</span>(a1x<span class="op">+</span><span class="dv">1</span>))</span>
<span id="cb389-13"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-13"></a>keep=<span class="kw">which</span>(<span class="kw">rowSums</span>(<span class="kw">counts</span>(EpddsChar))<span class="op">&gt;</span><span class="dv">10</span><span class="op">^</span>sd1vals)</span>
<span id="cb389-14"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-14"></a>keep2=<span class="kw">which</span>(<span class="kw">apply</span>(<span class="kw">counts</span>(EpddsChar), <span class="dv">1</span>, <span class="cf">function</span>(c) <span class="kw">sum</span>(c<span class="op">!=</span><span class="dv">0</span>))<span class="op">&gt;</span><span class="st"> </span>(<span class="kw">ncol</span>(EpddsChar)<span class="op">/</span><span class="dv">2</span>))</span>
<span id="cb389-15"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-15"></a>EpddsChar=EpddsChar[<span class="kw">intersect</span>(keep, keep2), ]</span>
<span id="cb389-16"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb389-16"></a>EpddsChar=<span class="kw">DESeq</span>(EpddsChar)</span></code></pre></div>
<div id="pca-plot-1" class="section level3">
<h3><span class="header-section-number">17.2.1</span> PCA plot</h3>
<p>First, have a look at the samples in a PCA plot: do they separate based on size:</p>
<div class="sourceCode" id="cb390"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb390-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb390-1"></a></span>
<span id="cb390-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb390-2"></a><span class="co">#pdf(&quot;~/Desktop/S1CD-Ep-characterisation-PCA-outcome.pdf&quot;, width=6, height=6)</span></span>
<span id="cb390-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb390-3"></a></span>
<span id="cb390-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb390-4"></a>vst1=<span class="kw">vst</span>(EpddsChar)</span>
<span id="cb390-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb390-5"></a></span>
<span id="cb390-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb390-6"></a><span class="kw">plotPCA</span>(vst1, <span class="st">&quot;SizeCat&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">colnames</span>(vst1)))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Ep cells&quot;</span>)<span class="op">+</span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>ColSize)</span></code></pre></div>
<div class="figure"><span id="fig:Ext1f"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext1f-1.png" alt="PCA plot of epithelial samples" width="672" />
<p class="caption">
Figure 17.5: PCA plot of epithelial samples
</p>
</div>
<p>And the plot of how much samples correlate with each other</p>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb391-1"></a>ax1=<span class="kw">heatmap.2</span>(<span class="kw">cor</span>((<span class="kw">assay</span>(vst1))), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSize[(vst1<span class="op">$</span>SizeCat)])</span></code></pre></div>
<div class="figure"><span id="fig:Ext1g"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext1g-1.png" alt="correlation plot epithelial samples" width="672" />
<p class="caption">
Figure 17.6: correlation plot epithelial samples
</p>
</div>
<div class="sourceCode" id="cb392"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb392-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb392-1"></a></span>
<span id="cb392-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb392-2"></a><span class="co">#dev.off()</span></span>
<span id="cb392-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb392-3"></a></span>
<span id="cb392-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb392-4"></a><span class="kw">write.csv</span>(<span class="kw">cor</span>((<span class="kw">assay</span>(vst1))), <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext1g.csv&quot;</span>)</span></code></pre></div>
<p>Here is a volcano plot the significant differential genes showing the difference between big and small tumors in the Ep fraction:</p>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb393-1"></a>Epres=<span class="kw">results</span>(EpddsChar)<span class="co">#, contrast=c(&quot;SizeCat&quot;, &quot;big&quot;, &quot;small&quot;))</span></span>
<span id="cb393-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb393-2"></a>Epres2=Epres[<span class="kw">which</span>(Epres<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(Epres<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">0.25</span> <span class="op">&amp;</span></span>
<span id="cb393-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb393-3"></a><span class="st">                             </span>Epres<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">80</span>), ]</span>
<span id="cb393-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb393-4"></a></span>
<span id="cb393-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb393-5"></a><span class="kw">plot</span>(Epres<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(Epres<span class="op">$</span>padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb393-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb393-6"></a>     <span class="dt">main=</span><span class="st">&quot;small (-ve) vs large (+ve)&quot;</span>)</span>
<span id="cb393-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb393-7"></a><span class="kw">text</span>(Epres2<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(Epres2<span class="op">$</span>padj), <span class="kw">rownames</span>(Epres2), <span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:1g"></span>
<img src="RatDCIS-bookdown_files/figure-html/1g-1.png" alt="Ep DEG big-small" width="672" />
<p class="caption">
Figure 17.7: Ep DEG big-small
</p>
</div>
<p>And the accompanying heatmap:</p>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb394-1"></a>colSide=EpddsChar<span class="op">$</span>SizeCat</span>
<span id="cb394-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb394-2"></a></span>
<span id="cb394-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb394-3"></a>t2=<span class="kw">assay</span>(vsd)[<span class="kw">which</span>( <span class="kw">rownames</span>(vsd)<span class="op">%in%</span><span class="kw">rownames</span>(Epres2)), <span class="kw">match</span>(<span class="kw">colnames</span>(EpddsChar), <span class="kw">colnames</span>(vsd))]</span>
<span id="cb394-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb394-4"></a>a1=<span class="kw">heatmap.2</span>(t2, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span> <span class="kw">palette</span>()[colSide], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;all DEG&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:1gheatmap-1"></span>
<img src="RatDCIS-bookdown_files/figure-html/1gheatmap-1.png" alt="heatmap of big vs small" width="672" />
<p class="caption">
Figure 17.8: heatmap of big vs small
</p>
</div>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb395-1"></a></span>
<span id="cb395-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb395-2"></a><span class="kw">boxplot</span>(<span class="kw">assay</span>(vsd)[<span class="st">&quot;Creb1&quot;</span>, <span class="kw">match</span>(<span class="kw">colnames</span>(EpddsChar), <span class="kw">colnames</span>(vsd))]<span class="op">~</span>EpddsChar<span class="op">$</span>SizeCat,</span>
<span id="cb395-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb395-3"></a>        <span class="dt">main=</span><span class="st">&quot;Creb1 gene small vs big vst expression&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:1gheatmap-2"></span>
<img src="RatDCIS-bookdown_files/figure-html/1gheatmap-2.png" alt="heatmap of big vs small" width="672" />
<p class="caption">
Figure 17.9: heatmap of big vs small
</p>
</div>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb396-1"></a></span>
<span id="cb396-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb396-2"></a><span class="kw">write.csv</span>(Epres2, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;nature-tables/1gl%s.csv&quot;</span>, <span class="kw">Sys.Date</span>()))</span>
<span id="cb396-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb396-3"></a><span class="kw">write.csv</span>(Epres, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;nature-tables/Supp2_A_DEGs.csv&quot;</span>))</span></code></pre></div>
</div>
<div id="gsea-2" class="section level3">
<h3><span class="header-section-number">17.2.2</span> GSEA</h3>
<p>Run GSEA. Here, we will look specifically at the Process Network pathways which are enriched</p>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-1"></a>EpGenes=<span class="kw">rownames</span>(Epres)</span>
<span id="cb397-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-2"></a></span>
<span id="cb397-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-3"></a>l1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb397-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-4"></a>l2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb397-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-5"></a>l3=Mouse2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, Mouse2Hum<span class="op">$</span>MGI.symbol)]</span>
<span id="cb397-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-6"></a>EpGenesConv=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(l1)<span class="op">==</span>F, l1, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l2)<span class="op">==</span>F, l2, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l3)<span class="op">==</span>F, l3, EpGenes)))</span>
<span id="cb397-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-7"></a></span>
<span id="cb397-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-8"></a></span>
<span id="cb397-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-9"></a>hits=EpGenesConv[<span class="kw">match</span>(<span class="kw">rownames</span>(Epres2), EpGenes)]</span>
<span id="cb397-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-10"></a><span class="co">#hits=epGenesConv[match(hits, rownames(Epdds))]</span></span>
<span id="cb397-11"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-11"></a>fcTab=Epres<span class="op">$</span>log2FoldChange</span>
<span id="cb397-12"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-12"></a><span class="kw">names</span>(fcTab)=EpGenesConv</span>
<span id="cb397-13"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-13"></a></span>
<span id="cb397-14"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-14"></a></span>
<span id="cb397-15"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-15"></a>  gscaep=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb397-16"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-16"></a>  gscaep &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gscaep, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb397-17"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-17"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb397-18"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-18"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb397-19"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-19"></a>  gscaep &lt;-<span class="st"> </span><span class="kw">analyze</span>(gscaep, </span>
<span id="cb397-20"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-20"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb397-21"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-21"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb397-22"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-22"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb397-23"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-23"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb397-24"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-24"></a></span>
<span id="cb397-25"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-25"></a>A1=HTSanalyzeR2<span class="op">::</span><span class="kw">summarize</span>(gscaep)  </span>
<span id="cb397-26"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-26"></a></span>
<span id="cb397-27"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-27"></a><span class="kw">save</span>(gscaep, <span class="dt">file=</span><span class="st">&quot;figure-outputs/1h.Rdata&quot;</span>)</span>
<span id="cb397-28"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-28"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(gscaep<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb397-29"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-29"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(gscaep<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb397-30"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-30"></a></span>
<span id="cb397-31"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-31"></a><span class="co">## check whether this runs:</span></span>
<span id="cb397-32"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-32"></a>gscaep<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks<span class="op">$</span>Gene.Set.Term=TermsA</span>
<span id="cb397-33"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-33"></a><span class="kw">viewEnrichMap</span>(gscaep, <span class="dt">gscs=</span><span class="kw">c</span>(<span class="st">&quot;ProcessNetworks&quot;</span>),</span>
<span id="cb397-34"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-34"></a>              <span class="dt">allSig =</span> <span class="ot">TRUE</span>, <span class="dt">gsNameType=</span><span class="st">&quot;term&quot;</span>)</span>
<span id="cb397-35"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-35"></a></span>
<span id="cb397-36"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb397-36"></a><span class="kw">write.csv</span>(gscaep<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks, <span class="dt">file=</span><span class="st">&quot;nature-tables/Supp_2_B_GSEA.csv&quot;</span>)</span></code></pre></div>
<p>We can also look at the results using barplots, as shown below</p>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-1"></a>PNresultse=gscaep<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks</span>
<span id="cb398-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-2"></a></span>
<span id="cb398-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-3"></a>Ax1=<span class="kw">which</span>(PNresultse<span class="op">$</span>Adjusted.Pvalue<span class="op">&lt;</span><span class="fl">0.1</span>)</span>
<span id="cb398-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-4"></a></span>
<span id="cb398-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-5"></a>Lx1=PNresultse[Ax1, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</span>
<span id="cb398-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-6"></a>Lx1<span class="op">$</span>Group=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(Lx1), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb398-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-7"></a>Lx1<span class="op">$</span>Process=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(Lx1), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb398-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-8"></a></span>
<span id="cb398-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-9"></a><span class="co">## replace certain groups</span></span>
<span id="cb398-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-10"></a>Lx1<span class="op">$</span>Group[<span class="kw">grep</span>(<span class="st">&quot;ymphocyte&quot;</span>, Lx1<span class="op">$</span>Process)]=<span class="st">&quot;NInflammation&quot;</span></span>
<span id="cb398-11"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-11"></a>Lx1<span class="op">$</span>Group[<span class="kw">which</span>(Lx1<span class="op">$</span>Group<span class="op">==</span><span class="st">&quot;NImmune response&quot;</span>)]=<span class="st">&quot;NInflammation&quot;</span></span>
<span id="cb398-12"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-12"></a></span>
<span id="cb398-13"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-13"></a><span class="co"># plot for inflammation, immune response, cell adhesion, transcription?</span></span>
<span id="cb398-14"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-14"></a></span>
<span id="cb398-15"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-15"></a>TestGrp=<span class="kw">c</span>(<span class="st">&quot;NCell adhesion&quot;</span>,<span class="st">&quot;NCell cycle&quot;</span>,<span class="st">&quot;NDNA damage&quot;</span>, <span class="st">&quot;NDevelopment&quot;</span>, <span class="st">&quot;NTranscription&quot;</span>, <span class="st">&quot;NInflammation&quot;</span>)</span>
<span id="cb398-16"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-16"></a></span>
<span id="cb398-17"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-17"></a><span class="co">#pdf(&quot;~/Desktop/1E-process-networks-significant-pathways_ep.pdf&quot;, width=6, height=6)</span></span>
<span id="cb398-18"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-18"></a></span>
<span id="cb398-19"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-19"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>), <span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">3</span>, <span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb398-20"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-20"></a></span>
<span id="cb398-21"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-21"></a><span class="cf">for</span> (i <span class="cf">in</span> TestGrp){</span>
<span id="cb398-22"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-22"></a>        x1=<span class="kw">which</span>(Lx1<span class="op">$</span>Group<span class="op">==</span>i)</span>
<span id="cb398-23"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-23"></a>        <span class="kw">barplot</span>(Lx1<span class="op">$</span>Observed.score[x1], <span class="dt">names.arg =</span> Lx1<span class="op">$</span>Process[x1], <span class="dt">horiz =</span> T, <span class="dt">las=</span><span class="dv">2</span>,</span>
<span id="cb398-24"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-24"></a>                <span class="dt">main=</span>i)</span>
<span id="cb398-25"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb398-25"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/ep-gsea-barplots-1.png" width="672" /></p>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-1"></a></span>
<span id="cb399-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-2"></a><span class="co">#dev.off()</span></span>
<span id="cb399-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-3"></a></span>
<span id="cb399-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-4"></a><span class="kw">write.csv</span>(Lx1, <span class="dt">file=</span><span class="st">&quot;nature-tables/1h.csv&quot;</span>)</span>
<span id="cb399-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-5"></a></span>
<span id="cb399-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-6"></a></span>
<span id="cb399-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-7"></a><span class="co"># match the genes in the Tgfb pathway</span></span>
<span id="cb399-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-8"></a>Genes1=PathwayMapAllComp<span class="op">$</span><span class="st">`</span><span class="dt">Cell adhesion_Leucocyte chemotaxis</span><span class="st">`</span></span>
<span id="cb399-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-9"></a>mid=<span class="kw">match</span>(Genes1, <span class="kw">toupper</span>(<span class="kw">rownames</span>(vst1)))</span>
<span id="cb399-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-10"></a></span>
<span id="cb399-11"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-11"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vst1)[<span class="kw">na.omit</span>(mid), ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span> ColSize[vst1<span class="op">$</span>SizeCat], </span>
<span id="cb399-12"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb399-12"></a>          <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/ep-gsea-barplots-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/ep-gsea-barplots-3.png" width="672" /></p>
<p>Double check the above result by running a ssGSEA and checking the directionality</p>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb400-1"></a>Mx1=<span class="kw">assay</span>(vst1)</span>
<span id="cb400-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb400-2"></a><span class="kw">rownames</span>(Mx1)=<span class="kw">toupper</span>(<span class="kw">rownames</span>(Mx1))</span>
<span id="cb400-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb400-3"></a></span>
<span id="cb400-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb400-4"></a>testOut=<span class="kw">gsva</span>(Mx1, PathwayMapAllComp,<span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">kcdf=</span><span class="st">&quot;Gaussian&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb400-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb400-5"></a><span class="kw">boxplot</span>(testOut[<span class="st">&quot;Cell cycle_S phase&quot;</span>, ]<span class="op">~</span>vst1<span class="op">$</span>SizeCat)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-114-1.png" width="672" /></p>
<div class="sourceCode" id="cb401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb401-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb401-1"></a><span class="kw">boxplot</span>(testOut[<span class="st">&quot;Cell adhesion_Cell-matrix interactions&quot;</span>, ]<span class="op">~</span>vst1<span class="op">$</span>SizeCat)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-114-2.png" width="672" /></p>
</div>
<div id="check-expression-of-checkpoint-proteins" class="section level3">
<h3><span class="header-section-number">17.2.3</span> Check expression of checkpoint proteins</h3>
<div class="sourceCode" id="cb402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb402-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-1"></a><span class="co">#pdf(&quot;figure-outputs/Supp2_checkpoint.pdf&quot;, height=5, width=5)</span></span>
<span id="cb402-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-2"></a>ImmSuppAPCRat=<span class="kw">sapply</span>(ImmSuppAPC, <span class="cf">function</span>(x) <span class="kw">na.omit</span>(SymHum2Rat<span class="op">$</span>RGD.symbol[<span class="kw">match</span>(x, SymHum2Rat<span class="op">$</span>HGNC.symbol)]))</span>
<span id="cb402-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-3"></a></span>
<span id="cb402-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-4"></a>vstB=<span class="kw">assay</span>(vst1)[<span class="kw">match</span>(<span class="kw">unlist</span>(ImmSuppAPCRat), <span class="kw">rownames</span>(vst1)), ]</span>
<span id="cb402-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-5"></a>ColSideCols=<span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;purple&quot;</span>), <span class="dt">times=</span><span class="kw">sapply</span>(ImmSuppAPCRat, length))</span>
<span id="cb402-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-6"></a></span>
<span id="cb402-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-7"></a>rmx=<span class="kw">which</span>(<span class="kw">is.na</span>(vstB[ ,<span class="dv">1</span>]))</span>
<span id="cb402-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-8"></a></span>
<span id="cb402-9"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-9"></a><span class="kw">heatmap.2</span>(vstB[<span class="op">-</span>rmx, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, </span>
<span id="cb402-10"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb402-10"></a>          <span class="dt">RowSideColors =</span> ColSideCols[<span class="op">-</span>rmx], <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<div class="figure"><span id="fig:Ext2f-1"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext2f-1.png" alt="checkpoint proteins expressed in epithelial samples" width="672" />
<p class="caption">
Figure 17.10: checkpoint proteins expressed in epithelial samples
</p>
</div>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb403-1"></a></span>
<span id="cb403-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb403-2"></a></span>
<span id="cb403-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb403-3"></a><span class="co">## Also do this using TPM values</span></span>
<span id="cb403-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb403-4"></a>valuesB=allTPMFinal[ <span class="kw">match</span>(<span class="kw">unlist</span>(ImmSuppAPCRat), <span class="kw">rownames</span>(allTPMFinal)), <span class="kw">match</span>(<span class="kw">colnames</span>(vstB), <span class="kw">colnames</span>(allTPMFinal))]</span>
<span id="cb403-5"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb403-5"></a><span class="kw">colnames</span>(valuesB)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(valuesB), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb403-6"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb403-6"></a></span>
<span id="cb403-7"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb403-7"></a><span class="kw">heatmap.2</span>(<span class="kw">log10</span>(valuesB[ <span class="op">-</span>rmx, ]<span class="op">+</span><span class="dv">1</span>), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, </span>
<span id="cb403-8"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb403-8"></a>          <span class="dt">RowSideColors =</span> ColSideCols[<span class="op">-</span>rmx], <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<div class="figure"><span id="fig:Ext2f-2"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext2f-2.png" alt="checkpoint proteins expressed in epithelial samples" width="672" />
<p class="caption">
Figure 17.11: checkpoint proteins expressed in epithelial samples
</p>
</div>
<div class="sourceCode" id="cb404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb404-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb404-1"></a></span>
<span id="cb404-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb404-2"></a><span class="co">#dev.off()</span></span>
<span id="cb404-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb404-3"></a>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="kw">as.data.frame</span>(valuesB), <span class="dt">rownames=</span>T, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb404-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb404-4"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>), <span class="dt">scrollX=</span>T))</span></code></pre></div>
<div class="figure"><span id="fig:Ext2f-3"></span>
<div id="htmlwidget-fdb97ff2115b70b70e0a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fdb97ff2115b70b70e0a">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Cd276","Ceacam1","Cd47","Cd200","Ido1","Tdo2","Cd274","Pdcd1lg2","Cd70","Cd40","Cd48","Tnfsf15","NA.","NA..1","Tnfsf4","Tnfsf9","Cd80","Cd86","Tnfrsf14","Pvr"],[105.15,10.35,221.19,98.33,0.05,0.19,12.16,0.54,5.8,36.46,392.59,0.44,null,null,0.28,75.84,6.26,17.83,12.18,67.5],[89.16,24.76,152.11,158.77,0,0,15.15,0.18,1.14,26.16,307.44,1.38,null,null,0.33,35.51,1.34,0.98,1.1,181.64],[44.96,40.58,174.48,650.51,0,0,36.27,0,0.54,16.22,199.02,2.51,null,null,0.06,4.26,2.64,5.14,1.17,311.82],[40.16,32.74,110.66,369.05,0,0.7,32.27,0,0,8.2,112.02,0.59,null,null,0.88,4.01,10.2,22.65,2.25,120.85],[41.38,47.58,167.63,271.55,0,0,73.41,0.3,0,9.17,224.96,2,null,null,0,6.29,1.49,0,0,677.43],[19.04,16.77,96.48,392.05,0,0.42,0.42,0,1.36,0.29,3.5,0,null,null,0,0.55,0,0,0,40.89],[0,25.39,69.98,73.24,0,10.03,0,0,0,0,0,0,null,null,0,0,0,0,0.25,43.7],[20.23,19.23,93.86,113.09,0,0,14.9,0,0,0,0.05,0,null,null,0,0,0,0,0,161.06],[35.99,72.23,189.89,558.03,0,0,60.5,0,0,0,30.65,0,null,null,0,0,0,7.32,0.1,158.21],[32.04,123.13,162.63,189.71,0,0,6.33,0,0,3.95,7.64,0,null,null,0,0,0,0,0.04,229.9]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>C2T1<\/th>\n      <th>C3T1<\/th>\n      <th>C5T1<\/th>\n      <th>C9T2<\/th>\n      <th>C7T1<\/th>\n      <th>C9T1<\/th>\n      <th>C10T2<\/th>\n      <th>C10T4<\/th>\n      <th>C11T1<\/th>\n      <th>C17T4<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 17.12: checkpoint proteins expressed in epithelial samples
</p>
</div>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb405-1"></a></span>
<span id="cb405-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb405-2"></a></span>
<span id="cb405-3"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb405-3"></a></span>
<span id="cb405-4"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb405-4"></a><span class="co">#write.csv(vstB[-rmx, ], file=&quot;nature-tables/Ext2f_checkpoint_expression.csv&quot;)</span></span></code></pre></div>
<p>Write the tables to file</p>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb406-1"></a><span class="kw">write.csv</span>(infoTableFinal, <span class="dt">file=</span><span class="st">&quot;nature-tables/infoTableFinal_output.csv&quot;</span>)</span>
<span id="cb406-2"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cb406-2"></a><span class="kw">write.csv</span>(allstarFinal, <span class="dt">file=</span><span class="st">&quot;nature-tables/allstarFinal_output.csv&quot;</span>)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="read-in-er-io-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="signature-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
