<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Epcam+ Inflammatory vs non-inflammatory samples | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Epcam+ Inflammatory vs non-inflammatory samples | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Epcam+ Inflammatory vs non-inflammatory samples | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-02-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="summary-of-gsea-runs.html"/>
<link rel="next" href="tcga-associate-epcam-inflammatory-with-survival.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.2</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.2.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>24.3</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="epcam-inflammatory-vs-non-inflammatory-samples" class="section level1">
<h1><span class="header-section-number">Chapter 14</span> Epcam+ Inflammatory vs non-inflammatory samples</h1>
<div id="identification-of-inflammatory-samples" class="section level2">
<h2><span class="header-section-number">14.1</span> Identification of inflammatory samples</h2>
<p>Here, look at individual enrichment scores (ssGSEA). We notice in the Epithelial samples there are 5 samples which appear to be hyperinflammatory: They have higher enrichment for TNFA, interferon signalling for example.</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-1"></a>d1=DNismr3<span class="op">$</span>Hallmark[, <span class="kw">c</span>(<span class="st">&quot;stable_vs_growing NES&quot;</span>, <span class="st">&quot;stable_vs_growing padj&quot;</span>)]</span>
<span id="cb240-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-2"></a>e1=Epismr3<span class="op">$</span>Hallmark[, <span class="kw">c</span>(<span class="st">&quot;stable_vs_growing NES&quot;</span>, <span class="st">&quot;stable_vs_growing padj&quot;</span>)]</span>
<span id="cb240-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-3"></a>c1=CDismr3<span class="op">$</span>Hallmark[, <span class="kw">c</span>(<span class="st">&quot;stable_vs_growing NES&quot;</span>, <span class="st">&quot;stable_vs_growing padj&quot;</span>)]</span>
<span id="cb240-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-4"></a><span class="co">#</span></span>
<span id="cb240-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-5"></a>ds=<span class="kw">which</span>(d1[ ,<span class="dv">2</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb240-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-6"></a>es=<span class="kw">which</span>(e1[ ,<span class="dv">2</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb240-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-7"></a>cs=<span class="kw">which</span>(c1[ ,<span class="dv">2</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb240-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-8"></a></span>
<span id="cb240-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-9"></a>AUnique=<span class="kw">c</span>(<span class="kw">rownames</span>(d1)[ds], <span class="kw">rownames</span>(e1)[es], <span class="kw">rownames</span>(c1)[cs])</span>
<span id="cb240-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-10"></a>xalist=<span class="kw">unique</span>(AUnique)</span>
<span id="cb240-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-11"></a></span>
<span id="cb240-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-12"></a>sList=PathInH[<span class="kw">match</span>((<span class="kw">rownames</span>(e1)[es]), <span class="kw">names</span>(PathInH))]</span>
<span id="cb240-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-13"></a></span>
<span id="cb240-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-14"></a>tpmTemp=allTPMFinal[ , <span class="kw">match</span>(vstEp<span class="op">$</span>SampleID, <span class="kw">colnames</span>(allTPMFinal))]</span>
<span id="cb240-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-15"></a><span class="kw">colnames</span>(tpmTemp)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(tpmTemp), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb240-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-16"></a><span class="kw">rownames</span>(tpmTemp)=rNames2</span>
<span id="cb240-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-17"></a></span>
<span id="cb240-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-18"></a>gsva1=<span class="kw">gsva</span>(tpmTemp, sList, <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb240-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-19"></a></span>
<span id="cb240-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-20"></a>nx2=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(gsva1), <span class="cf">function</span>(x) <span class="kw">sd</span>(gsva1[x, ]))</span>
<span id="cb240-21"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-21"></a>a1=<span class="kw">which</span>(nx2<span class="op">&gt;</span><span class="fl">0.03</span>)</span>
<span id="cb240-22"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-22"></a></span>
<span id="cb240-23"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-23"></a>sList=PathInH[<span class="kw">match</span>((<span class="kw">rownames</span>(d1)[ds]), <span class="kw">names</span>(PathInH))]</span>
<span id="cb240-24"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-24"></a></span>
<span id="cb240-25"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-25"></a>tpmTemp=allTPMFinal[ , <span class="kw">match</span>(vstDN<span class="op">$</span>SampleID, <span class="kw">colnames</span>(allTPMFinal))]</span>
<span id="cb240-26"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-26"></a><span class="kw">colnames</span>(tpmTemp)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(tpmTemp), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb240-27"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-27"></a><span class="kw">rownames</span>(tpmTemp)=rNames2</span>
<span id="cb240-28"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-28"></a></span>
<span id="cb240-29"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-29"></a>gsva2=<span class="kw">gsva</span>(tpmTemp, sList, <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb240-30"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-30"></a></span>
<span id="cb240-31"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-31"></a>nx2=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(gsva2), <span class="cf">function</span>(x) <span class="kw">sd</span>(gsva2[x, ]))</span>
<span id="cb240-32"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-32"></a>a2=<span class="kw">which</span>(nx2<span class="op">&gt;</span><span class="fl">0.03</span>)</span>
<span id="cb240-33"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-33"></a></span>
<span id="cb240-34"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-34"></a>sList=PathInH[<span class="kw">match</span>((<span class="kw">rownames</span>(c1)[cs]), <span class="kw">names</span>(PathInH))]</span>
<span id="cb240-35"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-35"></a></span>
<span id="cb240-36"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-36"></a>tpmTemp=allTPMFinal[ , <span class="kw">match</span>(vstCD<span class="op">$</span>SampleID, <span class="kw">colnames</span>(allTPMFinal))]</span>
<span id="cb240-37"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-37"></a><span class="kw">rownames</span>(tpmTemp)=rNames2</span>
<span id="cb240-38"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-38"></a></span>
<span id="cb240-39"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-39"></a>gsva3=<span class="kw">gsva</span>(tpmTemp, sList, <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb240-40"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-40"></a></span>
<span id="cb240-41"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-41"></a>nx2=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(gsva3), <span class="cf">function</span>(x) <span class="kw">sd</span>(gsva3[x, ]))</span>
<span id="cb240-42"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-42"></a>a3=<span class="kw">which</span>(nx2<span class="op">&gt;</span><span class="fl">0.03</span>)</span>
<span id="cb240-43"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-43"></a></span>
<span id="cb240-44"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-44"></a><span class="co">#pdf(&quot;~/Desktop/5B-ssgsea-scores-hallmark-pathways.pdf&quot;, height=5, width=5)</span></span>
<span id="cb240-45"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-45"></a></span>
<span id="cb240-46"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-46"></a>HRstat2=Cdata<span class="op">$</span>HR_status[<span class="kw">match</span>(vstEp<span class="op">$</span>TumorID, Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb240-47"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb240-47"></a>HRstat2[<span class="kw">which</span>(HRstat2<span class="op">==</span><span class="st">&quot;Basal&quot;</span>)]=<span class="ot">NA</span></span></code></pre></div>
<p>Check which pathways are enriched in which specific samples:</p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb241-1"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">5</span>))</span>
<span id="cb241-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb241-2"></a><span class="co"># heatmap.2(gsva1[a1, ], col=RdBu[11:1], scale=&quot;none&quot;, trace=&quot;none&quot;, ColSideColors = ColSizeb[vstEp$Growth],</span></span>
<span id="cb241-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb241-3"></a><span class="co">#           main=&quot;Ep&quot;)</span></span>
<span id="cb241-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb241-4"></a><span class="kw">heatmap.2</span>(gsva1[a1, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColMerge[ vstEp<span class="op">$</span>Treatment,<span class="dv">1</span>],</span>
<span id="cb241-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb241-5"></a>          <span class="dt">main=</span><span class="st">&quot;Ep&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:5b"></span>
<img src="RatDCIS-bookdown_files/figure-html/5b-1.png" alt="ssGSEA specific pathways" width="672" />
<p class="caption">
Figure 14.1: ssGSEA specific pathways
</p>
</div>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb242-1"></a></span>
<span id="cb242-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb242-2"></a><span class="kw">write.csv</span>(gsva1[a1, ], <span class="dt">file=</span><span class="st">&quot;nature-tables/5b.csv&quot;</span>)</span></code></pre></div>
<p>Below are the pathways specific to cd45 and dn</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb243-1"></a><span class="kw">heatmap.2</span>(gsva2[a2, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSizeb[vstDN<span class="op">$</span>Growth],</span>
<span id="cb243-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb243-2"></a>          <span class="dt">main=</span><span class="st">&quot;DN&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-95-1.png" width="672" /></p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb244-1"></a></span>
<span id="cb244-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb244-2"></a><span class="kw">heatmap.2</span>(gsva3[a3, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSizeb[vstCD<span class="op">$</span>Growth],</span>
<span id="cb244-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb244-3"></a>          <span class="dt">main=</span><span class="st">&quot;CD&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-95-2.png" width="672" /></p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb245-1"></a></span>
<span id="cb245-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb245-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
</div>
<div id="deg-inflammatory-vs-non-inflammatory" class="section level2">
<h2><span class="header-section-number">14.2</span> DEG: inflammatory vs non-inflammatory</h2>
<p>What genes are different between inflammatory and non-inflammatory?</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-1"></a>Inflamm=<span class="kw">c</span>(<span class="st">&quot;11N_D_Ep&quot;</span>, <span class="st">&quot;6R_B_Ep&quot;</span>, <span class="st">&quot;8L_D_Ep&quot;</span>, <span class="st">&quot;10L_D_Ep&quot;</span>, <span class="st">&quot;3N_B_Ep&quot;</span>)</span>
<span id="cb246-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-2"></a></span>
<span id="cb246-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-3"></a>Epdds<span class="op">$</span>Inflammation=<span class="st">&quot;no&quot;</span></span>
<span id="cb246-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-4"></a>Epdds<span class="op">$</span>Inflammation[<span class="kw">which</span>(<span class="kw">colnames</span>(Epdds)<span class="op">%in%</span>Inflamm)]=<span class="st">&quot;yes&quot;</span></span>
<span id="cb246-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-5"></a>Epdds<span class="op">$</span>Inflammation=<span class="kw">factor</span>(Epdds<span class="op">$</span>Inflammation)</span>
<span id="cb246-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-6"></a></span>
<span id="cb246-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-7"></a></span>
<span id="cb246-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-8"></a>vstEpInf=Epdds</span>
<span id="cb246-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-9"></a><span class="kw">design</span>(vstEpInf)=<span class="er">~</span>Inflammation</span>
<span id="cb246-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-10"></a>vstEpInf=<span class="kw">DESeq</span>(vstEpInf)</span>
<span id="cb246-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-11"></a></span>
<span id="cb246-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-12"></a>vstEpInfRes=<span class="kw">results</span>(vstEpInf)</span>
<span id="cb246-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-13"></a></span>
<span id="cb246-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-14"></a>sigGenes=<span class="kw">rownames</span>(vstEpInfRes)[<span class="kw">which</span>(vstEpInfRes<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(vstEpInfRes<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">2.5</span> <span class="op">&amp;</span><span class="st"> </span></span>
<span id="cb246-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-15"></a><span class="st">                 </span>vstEpInfRes<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">200</span>)]</span>
<span id="cb246-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-16"></a></span>
<span id="cb246-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-17"></a><span class="kw">colnames</span>(vstEp)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(vstEp), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb246-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-18"></a></span>
<span id="cb246-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb246-19"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vstEp)[sigGenes, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale =</span> <span class="st">&quot;row&quot;</span>, <span class="dt">ColSideColors =</span> ColSizeb[vstEpInf<span class="op">$</span>Growth], <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<div class="figure"><span id="fig:Ext5c"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext5c-1.png" alt="Differential gene exp inflammatory vs non-inflammatory" width="672" />
<p class="caption">
Figure 14.2: Differential gene exp inflammatory vs non-inflammatory
</p>
</div>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb247-1"></a></span>
<span id="cb247-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb247-2"></a><span class="kw">write.csv</span>(<span class="kw">assay</span>(vstEp)[sigGenes, ], <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext5c_heatmap.csc&quot;</span>)</span></code></pre></div>
<p>There are 27 genes which differentiates these two, which includes Csf3, Ccl22, Ccl3, Pltp, Lck, Rac2, Mmp12, Rsad2, Bcl2a1, Cxcl1, Lyz2, C1s, Ccl2, Nr4a2, Plaur, Ets1, Plau, Angptl4, Il1b, Cxcl2, Hbegf, Cd74, Ccl17, Thbs1, RT1-Da, Oasl, Tyrobp.</p>
<p>We can also assess whether there is an association between immune infiltration. We can compare whether the non-inflammatory have differences in T-cell infiltration, mixing indices based on imaging data. Below ‘no’ samples are non-inflammatory and ‘yes’ samples are hyper-inflammatory.</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb248-1"></a><span class="co">## boxplots for </span></span>
<span id="cb248-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb248-2"></a>nTab=<span class="kw">data.frame</span>(<span class="dt">inf=</span>vstEpInf<span class="op">$</span>Inflammation, <span class="dt">MH=</span>vstEpInf<span class="op">$</span>MHEpCAM, <span class="dt">cd8=</span>vstEpInf<span class="op">$</span>CD8Frac,</span>
<span id="cb248-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb248-3"></a>           <span class="dt">knn=</span>vstEpInf<span class="op">$</span>knnEpCAM)</span>
<span id="cb248-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb248-4"></a>nTabmelt=<span class="kw">melt</span>(nTab, <span class="dt">measure.vars =</span> <span class="kw">c</span>(<span class="st">&quot;cd8&quot;</span>, <span class="st">&quot;knn&quot;</span>, <span class="st">&quot;MH&quot;</span>))</span>
<span id="cb248-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb248-5"></a></span>
<span id="cb248-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb248-6"></a><span class="kw">ggplot</span>(<span class="dt">data=</span>nTabmelt, <span class="kw">aes</span>(<span class="dt">x=</span>inf, <span class="dt">y=</span>value))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>variable, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Ext5d"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext5d-1.png" alt="association signature with WSI" width="672" />
<p class="caption">
Figure 14.3: association signature with WSI
</p>
</div>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-1"></a></span>
<span id="cb249-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-2"></a><span class="kw">write.csv</span>(nTabmelt, <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext5d.csv&quot;</span>)</span>
<span id="cb249-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-3"></a></span>
<span id="cb249-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-4"></a>Cdata<span class="op">$</span>Inflammation=<span class="ot">NA</span></span>
<span id="cb249-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-5"></a>Cdata<span class="op">$</span>Inflammation=vstEpInf<span class="op">$</span>Inflammation2[<span class="kw">match</span>(Cdata<span class="op">$</span>NewID, <span class="kw">colnames</span>(vstEp))]</span>
<span id="cb249-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-6"></a></span>
<span id="cb249-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-7"></a>nTab</span>
<span id="cb249-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-8"></a><span class="co">##    inf         MH        cd8       knn</span></span>
<span id="cb249-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-9"></a><span class="co">## 1  yes 0.61918860 0.06810877  34.33110</span></span>
<span id="cb249-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-10"></a><span class="co">## 2   no 0.43765215 0.02027148  44.92703</span></span>
<span id="cb249-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-11"></a><span class="co">## 3   no 0.23771466 0.15217774  69.10296</span></span>
<span id="cb249-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-12"></a><span class="co">## 4  yes 0.67778543 0.38006717  26.89773</span></span>
<span id="cb249-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-13"></a><span class="co">## 5   no 0.26460731 0.02244524  73.82461</span></span>
<span id="cb249-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-14"></a><span class="co">## 6   no 0.02617825 0.05041860 153.33513</span></span>
<span id="cb249-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-15"></a><span class="co">## 7   no 0.67710428 0.02236926  12.74758</span></span>
<span id="cb249-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-16"></a><span class="co">## 8   no 0.26412928 0.05412980  45.49485</span></span>
<span id="cb249-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-17"></a><span class="co">## 9   no         NA         NA        NA</span></span>
<span id="cb249-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-18"></a><span class="co">## 10  no         NA         NA        NA</span></span>
<span id="cb249-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-19"></a><span class="co">## 11  no         NA         NA        NA</span></span>
<span id="cb249-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-20"></a><span class="co">## 12  no 0.05741998 0.04769261  71.64542</span></span>
<span id="cb249-21"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-21"></a><span class="co">## 13  no 0.76751419 0.11152672  15.06388</span></span>
<span id="cb249-22"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-22"></a><span class="co">## 14  no 0.34918782 0.03010410  47.77260</span></span>
<span id="cb249-23"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-23"></a><span class="co">## 15 yes 0.63717941 0.27682992  32.31159</span></span>
<span id="cb249-24"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-24"></a><span class="co">## 16  no 0.38112224 0.02321393  32.11938</span></span>
<span id="cb249-25"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-25"></a><span class="co">## 17 yes 0.50296493 0.15548043  76.37955</span></span>
<span id="cb249-26"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-26"></a><span class="co">## 18  no         NA         NA        NA</span></span>
<span id="cb249-27"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-27"></a><span class="co">## 19 yes 0.64888492 0.05623363  16.36623</span></span>
<span id="cb249-28"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb249-28"></a><span class="co">## 20  no 0.36761266 0.02876680  50.00101</span></span></code></pre></div>
<p>Statistics are below:</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-1"></a><span class="kw">print</span>(<span class="st">&#39;assoc with cd8&#39;</span>)</span>
<span id="cb250-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-2"></a><span class="co">## [1] &quot;assoc with cd8&quot;</span></span>
<span id="cb250-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-3"></a><span class="co">#t.test(nTab$cd8~nTab$inf)</span></span>
<span id="cb250-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-4"></a><span class="co">#table(nTab$cd8, nTab$inf)</span></span>
<span id="cb250-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-5"></a><span class="kw">print</span>(<span class="st">&#39;assoc with MH index&#39;</span>)</span>
<span id="cb250-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-6"></a><span class="co">## [1] &quot;assoc with MH index&quot;</span></span>
<span id="cb250-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-7"></a><span class="co">#t.test(nTab$MH~nTab$inf)</span></span>
<span id="cb250-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-8"></a><span class="co">#table(nTab$MH,nTab$inf)</span></span>
<span id="cb250-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-9"></a><span class="kw">print</span>(<span class="st">&#39;assoc with knn&#39;</span>)</span>
<span id="cb250-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-10"></a><span class="co">## [1] &quot;assoc with knn&quot;</span></span>
<span id="cb250-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-11"></a><span class="co">#t.test(nTab$knn~nTab$inf)</span></span>
<span id="cb250-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-12"></a><span class="co">#table(nTab$knn~nTab$inf)</span></span>
<span id="cb250-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-13"></a></span>
<span id="cb250-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-14"></a>nTab</span>
<span id="cb250-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-15"></a><span class="co">##    inf         MH        cd8       knn</span></span>
<span id="cb250-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-16"></a><span class="co">## 1  yes 0.61918860 0.06810877  34.33110</span></span>
<span id="cb250-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-17"></a><span class="co">## 2   no 0.43765215 0.02027148  44.92703</span></span>
<span id="cb250-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-18"></a><span class="co">## 3   no 0.23771466 0.15217774  69.10296</span></span>
<span id="cb250-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-19"></a><span class="co">## 4  yes 0.67778543 0.38006717  26.89773</span></span>
<span id="cb250-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-20"></a><span class="co">## 5   no 0.26460731 0.02244524  73.82461</span></span>
<span id="cb250-21"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-21"></a><span class="co">## 6   no 0.02617825 0.05041860 153.33513</span></span>
<span id="cb250-22"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-22"></a><span class="co">## 7   no 0.67710428 0.02236926  12.74758</span></span>
<span id="cb250-23"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-23"></a><span class="co">## 8   no 0.26412928 0.05412980  45.49485</span></span>
<span id="cb250-24"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-24"></a><span class="co">## 9   no         NA         NA        NA</span></span>
<span id="cb250-25"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-25"></a><span class="co">## 10  no         NA         NA        NA</span></span>
<span id="cb250-26"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-26"></a><span class="co">## 11  no         NA         NA        NA</span></span>
<span id="cb250-27"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-27"></a><span class="co">## 12  no 0.05741998 0.04769261  71.64542</span></span>
<span id="cb250-28"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-28"></a><span class="co">## 13  no 0.76751419 0.11152672  15.06388</span></span>
<span id="cb250-29"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-29"></a><span class="co">## 14  no 0.34918782 0.03010410  47.77260</span></span>
<span id="cb250-30"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-30"></a><span class="co">## 15 yes 0.63717941 0.27682992  32.31159</span></span>
<span id="cb250-31"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-31"></a><span class="co">## 16  no 0.38112224 0.02321393  32.11938</span></span>
<span id="cb250-32"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-32"></a><span class="co">## 17 yes 0.50296493 0.15548043  76.37955</span></span>
<span id="cb250-33"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-33"></a><span class="co">## 18  no         NA         NA        NA</span></span>
<span id="cb250-34"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-34"></a><span class="co">## 19 yes 0.64888492 0.05623363  16.36623</span></span>
<span id="cb250-35"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb250-35"></a><span class="co">## 20  no 0.36761266 0.02876680  50.00101</span></span></code></pre></div>
</div>
<div id="finding-3-signatures-for-3-branches" class="section level2">
<h2><span class="header-section-number">14.3</span> Finding 3 signatures for 3 branches</h2>
<p>Below we can perform a 1 vs all analysis i.e. compare</p>
<ol style="list-style-type: decimal">
<li>growing vs the rest</li>
<li>inflammatory vs the rest</li>
<li>stable vs the rest</li>
</ol>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-1"></a><span class="co">#vstEpInf$Inflammation2=vstEpInf$Inflammation</span></span>
<span id="cb251-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-2"></a>vstEpInf<span class="op">$</span>Inflammation2=<span class="kw">factor</span>(<span class="kw">ifelse</span>(vstEpInf<span class="op">$</span>Inflammation<span class="op">==</span><span class="st">&quot;yes&quot;</span>, <span class="st">&quot;inf&quot;</span>, <span class="kw">ifelse</span>(vstEpInf<span class="op">$</span>Growth<span class="op">==</span><span class="st">&quot;growing&quot;</span>, <span class="st">&quot;grow&quot;</span>, <span class="st">&quot;stab&quot;</span>)))</span>
<span id="cb251-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-3"></a></span>
<span id="cb251-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-4"></a></span>
<span id="cb251-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-5"></a><span class="kw">design</span>(vstEpInf)=<span class="er">~</span>Inflammation2</span>
<span id="cb251-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-6"></a>vstEpInf=<span class="kw">DESeq</span>(vstEpInf)</span>
<span id="cb251-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-7"></a></span>
<span id="cb251-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-8"></a>vstEpInfRes1=<span class="kw">results</span>(vstEpInf, <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;Inflammation2&quot;</span>, <span class="st">&quot;stab&quot;</span>, <span class="st">&quot;inf&quot;</span>))</span>
<span id="cb251-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-9"></a>res1genes=<span class="kw">rownames</span>(vstEpInfRes1)[<span class="kw">which</span>(vstEpInfRes1<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(vstEpInfRes1<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span> <span class="op">&amp;</span></span>
<span id="cb251-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-10"></a><span class="st">                                         </span>vstEpInfRes1<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb251-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-11"></a></span>
<span id="cb251-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-12"></a>vstEpInfRes2=<span class="kw">results</span>(vstEpInf, <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;Inflammation2&quot;</span>, <span class="st">&quot;stab&quot;</span>, <span class="st">&quot;grow&quot;</span>))</span>
<span id="cb251-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-13"></a>res2genes=<span class="kw">rownames</span>(vstEpInfRes2)[<span class="kw">which</span>(vstEpInfRes2<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(vstEpInfRes2<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span> <span class="op">&amp;</span></span>
<span id="cb251-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-14"></a><span class="st">                                         </span>vstEpInfRes2<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb251-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-15"></a></span>
<span id="cb251-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-16"></a>vstEpInfRes3=<span class="kw">results</span>(vstEpInf, <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;Inflammation2&quot;</span>, <span class="st">&quot;grow&quot;</span>, <span class="st">&quot;inf&quot;</span>))</span>
<span id="cb251-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-17"></a>res3genes=<span class="kw">rownames</span>(vstEpInfRes3)[<span class="kw">which</span>(vstEpInfRes3<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(vstEpInfRes3<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span> <span class="op">&amp;</span></span>
<span id="cb251-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-18"></a><span class="st">                                         </span>vstEpInfRes3<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb251-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-19"></a></span>
<span id="cb251-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-20"></a>aUnique=<span class="kw">c</span>(res1genes, res2genes, res3genes)</span>
<span id="cb251-21"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-21"></a></span>
<span id="cb251-22"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-22"></a>AX1=<span class="kw">setdiff</span>(res2genes, res1genes)</span>
<span id="cb251-23"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-23"></a>ax2=<span class="kw">setdiff</span>(res3genes, res1genes)</span>
<span id="cb251-24"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-24"></a></span>
<span id="cb251-25"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb251-25"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vstEp)[AX1, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale =</span> <span class="st">&quot;row&quot;</span>, <span class="dt">ColSideColors =</span> ColSizec[vstEpInf<span class="op">$</span>Inflammation2], <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-96-1.png" width="672" /></p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb252-1"></a></span>
<span id="cb252-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb252-2"></a><span class="kw">write.csv</span>(<span class="kw">assay</span>(vstEp)[AX1, ], <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext5c.csv&quot;</span>)</span>
<span id="cb252-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb252-3"></a></span>
<span id="cb252-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb252-4"></a>nTab=<span class="kw">data.frame</span>(<span class="dt">inf=</span>vstEpInf<span class="op">$</span>Inflammation2, <span class="dt">MH=</span>vstEpInf<span class="op">$</span>MHEpCAM, <span class="dt">cd8=</span>vstEpInf<span class="op">$</span>CD8Frac,</span>
<span id="cb252-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb252-5"></a>           <span class="dt">knn=</span>vstEpInf<span class="op">$</span>knnEpCAM)</span>
<span id="cb252-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb252-6"></a>nTabmelt=<span class="kw">melt</span>(nTab, <span class="dt">measure.vars =</span> <span class="kw">c</span>(<span class="st">&quot;cd8&quot;</span>, <span class="st">&quot;knn&quot;</span>, <span class="st">&quot;MH&quot;</span>))</span>
<span id="cb252-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb252-7"></a></span>
<span id="cb252-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb252-8"></a><span class="kw">ggplot</span>(<span class="dt">data=</span>nTabmelt, <span class="kw">aes</span>(<span class="dt">x=</span>inf, <span class="dt">y=</span>value))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>variable, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-96-2.png" width="672" /></p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb253-1"></a></span>
<span id="cb253-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb253-2"></a>vstOut=<span class="kw">vst</span>(vstEpInf)</span>
<span id="cb253-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb253-3"></a></span>
<span id="cb253-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb253-4"></a><span class="kw">plotPCA</span>(vstOut, <span class="st">&quot;Inflammation2&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-96-3.png" width="672" /></p>
<p>We see that the growing vs stable samples are very similary overall, however, DEGs between growing and stable are also expressed in the inflammatory branch.</p>
<p>We can use these gene signatures to identify each branch: Below, we see the separation between the 3 groups using these genes. (note that the stable branch has no identifiable upregulated genes and is defined by the negative score of the downregulated genes:). We use ssGSEA to get a score for each sample</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-1"></a>vstEpInf<span class="op">$</span>Inflammation3=vstEpInf<span class="op">$</span>Inflammation2</span>
<span id="cb254-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-2"></a>vstEpInf<span class="op">$</span>Inflammation3[<span class="kw">which</span>(vstEpInf<span class="op">$</span>Inflammation3<span class="op">==</span><span class="st">&quot;stab&quot;</span>)]=<span class="st">&quot;inf&quot;</span></span>
<span id="cb254-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-3"></a>vstEpInf<span class="op">$</span>Inflammation3=<span class="kw">factor</span>(vstEpInf<span class="op">$</span>Inflammation3)</span>
<span id="cb254-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-4"></a><span class="kw">design</span>(vstEpInf)=<span class="er">~</span>(Inflammation3)</span>
<span id="cb254-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-5"></a>vstEpInfb=<span class="kw">DESeq</span>(vstEpInf)</span>
<span id="cb254-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-6"></a>vstEpInfRes1b=<span class="kw">results</span>(vstEpInfb)</span>
<span id="cb254-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-7"></a>res1genesGrow=<span class="kw">rownames</span>(vstEpInfRes1b)[<span class="kw">which</span>(vstEpInfRes1b<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>(vstEpInfRes1b<span class="op">$</span>log2FoldChange)<span class="op">&lt;</span>(<span class="op">-</span><span class="fl">1.5</span>) <span class="op">&amp;</span></span>
<span id="cb254-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-8"></a><span class="st">                                         </span>vstEpInfRes1b<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb254-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-9"></a>vstEpInf<span class="op">$</span>Inflammation3=vstEpInf<span class="op">$</span>Inflammation2</span>
<span id="cb254-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-10"></a>vstEpInf<span class="op">$</span>Inflammation3[<span class="kw">which</span>(vstEpInf<span class="op">$</span>Inflammation2<span class="op">==</span><span class="st">&quot;grow&quot;</span>)]=<span class="st">&quot;inf&quot;</span></span>
<span id="cb254-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-11"></a>vstEpInf<span class="op">$</span>Inflammation3=<span class="kw">factor</span>(vstEpInf<span class="op">$</span>Inflammation3)</span>
<span id="cb254-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-12"></a><span class="kw">design</span>(vstEpInf)=<span class="er">~</span>(Inflammation3)</span>
<span id="cb254-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-13"></a>vstEpInfb=<span class="kw">DESeq</span>(vstEpInf)</span>
<span id="cb254-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-14"></a>vstEpInfRes1b=<span class="kw">results</span>(vstEpInfb)</span>
<span id="cb254-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-15"></a>res1genesStab=<span class="kw">rownames</span>(vstEpInfRes1b)[<span class="kw">which</span>(vstEpInfRes1b<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>(vstEpInfRes1b<span class="op">$</span>log2FoldChange)<span class="op">&lt;</span>(<span class="op">-</span><span class="fl">1.5</span>) <span class="op">&amp;</span></span>
<span id="cb254-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-16"></a><span class="st">                                         </span>vstEpInfRes1b<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb254-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-17"></a></span>
<span id="cb254-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-18"></a>vstEpInf<span class="op">$</span>Inflammation3=vstEpInf<span class="op">$</span>Inflammation2</span>
<span id="cb254-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-19"></a>vstEpInf<span class="op">$</span>Inflammation3[<span class="kw">which</span>(vstEpInf<span class="op">$</span>Inflammation2<span class="op">==</span><span class="st">&quot;stab&quot;</span>)]=<span class="st">&quot;grow&quot;</span></span>
<span id="cb254-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-20"></a>vstEpInf<span class="op">$</span>Inflammation3=<span class="kw">factor</span>(vstEpInf<span class="op">$</span>Inflammation3)</span>
<span id="cb254-21"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-21"></a><span class="kw">design</span>(vstEpInf)=<span class="er">~</span>(Inflammation3)</span>
<span id="cb254-22"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-22"></a>vstEpInfb=<span class="kw">DESeq</span>(vstEpInf)</span>
<span id="cb254-23"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-23"></a>vstEpInfRes1b=<span class="kw">results</span>(vstEpInfb)</span>
<span id="cb254-24"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-24"></a>res1genesInf=<span class="kw">rownames</span>(vstEpInfRes1b)[<span class="kw">which</span>(vstEpInfRes1b<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>(vstEpInfRes1b<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span>(<span class="fl">1.5</span>) <span class="op">&amp;</span></span>
<span id="cb254-25"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-25"></a><span class="st">                                         </span>vstEpInfRes1b<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb254-26"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-26"></a></span>
<span id="cb254-27"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-27"></a><span class="co">## perform ssGSEA on these samples and see where they fit:</span></span>
<span id="cb254-28"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-28"></a></span>
<span id="cb254-29"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-29"></a>RatssGSEA=<span class="kw">gsva</span>(<span class="kw">assay</span>(vsdLimmaEp), </span>
<span id="cb254-30"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-30"></a>               <span class="kw">list</span>(<span class="dt">grow=</span>res1genesGrow, <span class="dt">inh=</span>res1genesInf, <span class="dt">nonstab=</span>res1genesStab), <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb254-31"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-31"></a></span>
<span id="cb254-32"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-32"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb254-33"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-33"></a><span class="kw">plot</span>(RatssGSEA[<span class="dv">1</span>, ], RatssGSEA[<span class="dv">2</span>, ], <span class="dt">col=</span><span class="kw">factor</span>(vstEpInf<span class="op">$</span>Inflammation2), <span class="dt">xlab=</span><span class="st">&quot;grow score&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;inh score&quot;</span>)</span>
<span id="cb254-34"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb254-34"></a><span class="kw">plot</span>(RatssGSEA[<span class="dv">1</span>, ], <span class="op">-</span>RatssGSEA[<span class="dv">3</span>, ], <span class="dt">col=</span><span class="kw">factor</span>(vstEpInf<span class="op">$</span>Inflammation2), <span class="dt">xlab=</span><span class="st">&quot;grow score&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;-stab score&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/wtf-is-this-1.png" width="672" /></p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb255-1"></a>RatssGSEA[<span class="dv">3</span>, ]=<span class="op">-</span>RatssGSEA[<span class="dv">3</span>, ]</span></code></pre></div>
<p>We can also overlay these signatures from ssGSEA to see how well it can predict each group:</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-1"></a><span class="co"># plot histograms for the 3 samples:</span></span>
<span id="cb256-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb256-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-3"></a></span>
<span id="cb256-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-4"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>){</span>
<span id="cb256-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-5"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="kw">c</span>(<span class="st">&quot;inf&quot;</span>, <span class="st">&quot;grow&quot;</span>, <span class="st">&quot;stab&quot;</span>)){</span>
<span id="cb256-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-6"></a>    <span class="kw">plot</span>(<span class="kw">density</span>(RatssGSEA[i, <span class="kw">which</span>(vstEpInf<span class="op">$</span>Inflammation2<span class="op">==</span>j)]), <span class="dt">main=</span><span class="kw">paste</span>(j, <span class="kw">rownames</span>(RatssGSEA)[i]), </span>
<span id="cb256-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-7"></a>         <span class="dt">xlim=</span><span class="kw">c</span>(<span class="op">-</span><span class="fl">0.6</span>, <span class="fl">0.8</span>))</span>
<span id="cb256-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-8"></a>  }</span>
<span id="cb256-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb256-9"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-97-1.png" width="672" /></p>
<p>GSEA for the above 3 groups?</p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-1"></a></span>
<span id="cb257-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-2"></a>l1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(<span class="kw">rownames</span>(vstEpInfRes1), SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb257-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-3"></a>l2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(<span class="kw">rownames</span>(vstEpInfRes1), Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb257-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-4"></a>l3=Mouse2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(<span class="kw">rownames</span>(vstEpInfRes1), Mouse2Hum<span class="op">$</span>MGI.symbol)]</span>
<span id="cb257-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-5"></a>EpGenesConv=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(l1)<span class="op">==</span>F, l1, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l2)<span class="op">==</span>F, l2, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l3)<span class="op">==</span>F, l3, <span class="kw">rownames</span>(vstEpInfRes1))))</span>
<span id="cb257-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-6"></a></span>
<span id="cb257-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-7"></a></span>
<span id="cb257-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-8"></a><span class="co">## run 1</span></span>
<span id="cb257-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-9"></a>hits=EpGenesConv[<span class="kw">match</span>(res1genes, <span class="kw">rownames</span>(vstEpInfRes1))]</span>
<span id="cb257-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-10"></a>fcTab=vstEpInfRes1<span class="op">$</span>log2FoldChange</span>
<span id="cb257-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-11"></a><span class="kw">names</span>(fcTab)=EpGenesConv</span>
<span id="cb257-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-12"></a>  gscaepInf1=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb257-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-13"></a>  gscaepInf1 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gscaepInf1, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb257-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-14"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb257-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-15"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb257-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-16"></a>  gscaepInf1 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gscaepInf1, </span>
<span id="cb257-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-17"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb257-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-18"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb257-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-19"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb257-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-20"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb257-21"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-21"></a><span class="co">## run2</span></span>
<span id="cb257-22"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-22"></a>hits=EpGenesConv[<span class="kw">match</span>(res2genes, <span class="kw">rownames</span>(vstEpInfRes2))]</span>
<span id="cb257-23"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-23"></a>fcTab=vstEpInfRes2<span class="op">$</span>log2FoldChange</span>
<span id="cb257-24"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-24"></a><span class="kw">names</span>(fcTab)=EpGenesConv</span>
<span id="cb257-25"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-25"></a>gscaepInf2=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb257-26"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-26"></a>gscaepInf2 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gscaepInf2, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb257-27"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-27"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb257-28"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-28"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb257-29"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-29"></a>gscaepInf2 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gscaepInf2, </span>
<span id="cb257-30"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-30"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb257-31"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-31"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb257-32"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-32"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb257-33"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-33"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb257-34"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-34"></a><span class="co">## run3</span></span>
<span id="cb257-35"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-35"></a>hits=EpGenesConv[<span class="kw">match</span>(res3genes, <span class="kw">rownames</span>(vstEpInfRes3))]</span>
<span id="cb257-36"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-36"></a>fcTab=vstEpInfRes3<span class="op">$</span>log2FoldChange</span>
<span id="cb257-37"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-37"></a><span class="kw">names</span>(fcTab)=EpGenesConv</span>
<span id="cb257-38"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-38"></a>gscaepInf3=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb257-39"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-39"></a>gscaepInf3 &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gscaepInf3, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb257-40"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-40"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb257-41"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-41"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb257-42"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-42"></a>gscaepInf3 &lt;-<span class="st"> </span><span class="kw">analyze</span>(gscaepInf3, </span>
<span id="cb257-43"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-43"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb257-44"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-44"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb257-45"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-45"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb257-46"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-46"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb257-47"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-47"></a></span>
<span id="cb257-48"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-48"></a></span>
<span id="cb257-49"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-49"></a>GSEA1=gscaepInf1<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks</span>
<span id="cb257-50"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-50"></a>GSEA2=gscaepInf2<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks</span>
<span id="cb257-51"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-51"></a>GSEA3=gscaepInf3<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks</span>
<span id="cb257-52"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-52"></a></span>
<span id="cb257-53"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-53"></a></span>
<span id="cb257-54"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-54"></a>lx1=<span class="kw">unique</span>(<span class="kw">c</span>(<span class="kw">rownames</span>(GSEA1)[<span class="kw">which</span>(GSEA1[ ,<span class="dv">3</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)],</span>
<span id="cb257-55"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-55"></a>      <span class="kw">rownames</span>(GSEA2)[<span class="kw">which</span>(GSEA2[ ,<span class="dv">3</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)],</span>
<span id="cb257-56"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-56"></a>      <span class="kw">rownames</span>(GSEA3)[<span class="kw">which</span>(GSEA3[ ,<span class="dv">3</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)]))</span>
<span id="cb257-57"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-57"></a></span>
<span id="cb257-58"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-58"></a></span>
<span id="cb257-59"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-59"></a>Fx=<span class="kw">cbind</span>(GSEA1[lx1, <span class="dv">1</span>], GSEA3[lx1, <span class="dv">1</span>], GSEA2[lx1, <span class="dv">1</span>])</span>
<span id="cb257-60"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-60"></a>Fx2=<span class="kw">cbind</span>(GSEA1[lx1, <span class="dv">3</span>], GSEA3[lx1, <span class="dv">3</span>], GSEA2[lx1, <span class="dv">3</span>])</span>
<span id="cb257-61"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-61"></a>Fx[<span class="kw">which</span>(Fx2<span class="op">&gt;</span><span class="fl">0.05</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">0</span></span>
<span id="cb257-62"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-62"></a></span>
<span id="cb257-63"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-63"></a><span class="kw">rownames</span>(Fx)=lx1</span>
<span id="cb257-64"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-64"></a><span class="kw">colnames</span>(Fx)=<span class="kw">c</span>(<span class="st">&quot;s/i&quot;</span>, <span class="st">&quot;g/i&quot;</span>, <span class="st">&quot;s/g&quot;</span>)</span>
<span id="cb257-65"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-65"></a></span>
<span id="cb257-66"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-66"></a></span>
<span id="cb257-67"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-67"></a>ltest=<span class="kw">which</span>(<span class="kw">rowSums</span>(<span class="kw">sign</span>(<span class="kw">abs</span>(Fx)))<span class="op">&gt;</span><span class="dv">1</span>)</span>
<span id="cb257-68"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-68"></a>ltest2=<span class="kw">which</span>(<span class="kw">abs</span>(Fx[,<span class="dv">3</span>])<span class="op">&gt;</span><span class="dv">0</span>)</span>
<span id="cb257-69"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-69"></a></span>
<span id="cb257-70"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-70"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">5</span>))</span>
<span id="cb257-71"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb257-71"></a><span class="kw">heatmap.2</span>(Fx[<span class="kw">unique</span>(<span class="kw">c</span>(ltest,ltest2)), ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/gsea-1-v-all-ep-1.png" width="672" /></p>
</div>
<div id="analyse-the-non-inflammatory-samples" class="section level2">
<h2><span class="header-section-number">14.4</span> Analyse the non-inflammatory samples</h2>
<p>We remove all the inflammtory samples and compare differences between stable and growing here.
Below is the volcano plot for the</p>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-1"></a><span class="co">#Remove inflammatory samples</span></span>
<span id="cb258-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-2"></a><span class="co">#Also remove samples which are basal-like</span></span>
<span id="cb258-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-3"></a></span>
<span id="cb258-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-4"></a>Inflamm=<span class="kw">c</span>(<span class="st">&quot;11N_D_Ep&quot;</span>, <span class="st">&quot;6R_B_Ep&quot;</span>, <span class="st">&quot;8L_D_Ep&quot;</span>, <span class="st">&quot;10L_D_Ep&quot;</span>, <span class="st">&quot;3N_B_Ep&quot;</span>) <span class="co">#,&quot;2N__Ep&quot;, &quot;15N_C_Ep&quot;, &quot;7N_A_Ep&quot;, &quot;8R_CU_Ep&quot;, &quot;12L_D_Ep&quot;, &quot;14N_D_Ep&quot;)</span></span>
<span id="cb258-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-5"></a></span>
<span id="cb258-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-6"></a>EpddsInflam=EpddsGrowth</span>
<span id="cb258-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-7"></a>EpddsInflam=EpddsInflam[, <span class="op">-</span><span class="kw">match</span>(Inflamm, <span class="kw">colnames</span>(EpddsInflam))]</span>
<span id="cb258-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-8"></a></span>
<span id="cb258-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-9"></a>EpddsInflam=<span class="kw">DESeq</span>(EpddsInflam)</span>
<span id="cb258-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-10"></a></span>
<span id="cb258-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-11"></a>ResA=<span class="kw">results</span>(EpddsInflam, <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;Growth&quot;</span>, <span class="st">&quot;stable&quot;</span>, <span class="st">&quot;growing&quot;</span>))</span>
<span id="cb258-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-12"></a>ResAb=ResA[<span class="kw">which</span>(ResA<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(ResA<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span>), ]</span>
<span id="cb258-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-13"></a></span>
<span id="cb258-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-14"></a></span>
<span id="cb258-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-15"></a><span class="co">#pdf(&quot;~/Desktop/5C-heatmap-inflamm-vs-non-inflamm-lowexpressing-included.pdf&quot;, width=8, height=12)</span></span>
<span id="cb258-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-16"></a></span>
<span id="cb258-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-17"></a><span class="kw">plot</span>(ResA<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(ResA<span class="op">$</span>padj), <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb258-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-18"></a>     <span class="dt">main=</span><span class="st">&quot;growing (-ve) vs stable (+ve)&quot;</span>)</span>
<span id="cb258-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-19"></a></span>
<span id="cb258-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb258-20"></a><span class="kw">text</span>(ResAb<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(ResAb<span class="op">$</span>padj), <span class="kw">rownames</span>(ResAb), <span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Epddsinflamm"></span>
<img src="RatDCIS-bookdown_files/figure-html/Epddsinflamm-1.png" alt="ep non-inflammatory comparison" width="672" />
<p class="caption">
Figure 14.4: ep non-inflammatory comparison
</p>
</div>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb259-1"></a></span>
<span id="cb259-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb259-2"></a><span class="kw">write.csv</span>(ResA, <span class="dt">file=</span><span class="st">&quot;nature-tables/5f.csv&quot;</span>)</span></code></pre></div>
<p>Followed by the heatmap for this analysis</p>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb260-1"></a>vst2=<span class="kw">vst</span>(EpddsInflam)</span>
<span id="cb260-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb260-2"></a></span>
<span id="cb260-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb260-3"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vst2)[<span class="kw">match</span>(<span class="kw">rownames</span>(ResAb), <span class="kw">rownames</span>(vst2)), ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], </span>
<span id="cb260-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb260-4"></a>          <span class="dt">ColSideColors =</span> ColSizeb[vst2<span class="op">$</span>Growth], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Epddsinflamm2"></span>
<img src="RatDCIS-bookdown_files/figure-html/Epddsinflamm2-1.png" alt="ep non-inflammatory comparison heatmap" width="672" />
<p class="caption">
Figure 14.5: ep non-inflammatory comparison heatmap
</p>
</div>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb261-1"></a><span class="co"># heatmap.2(assay(vstEp)[match(rownames(ResAb), rownames(vstEp)), ], col=RdBu[11:1], </span></span>
<span id="cb261-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb261-2"></a><span class="co">#           ColSideColors = ColSizeb[vstEp$Growth], trace=&quot;none&quot;, scale=&quot;row&quot;)</span></span>
<span id="cb261-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb261-3"></a></span>
<span id="cb261-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb261-4"></a><span class="co">#dev.off()</span></span>
<span id="cb261-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb261-5"></a></span>
<span id="cb261-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb261-6"></a><span class="co">#write.csv(ResAb, file=&quot;~/Desktop/5C-growing-stable-in-non-inflammatory-Ep-lowexpressing-included.csv&quot;)</span></span></code></pre></div>
<div id="gsea" class="section level3">
<h3><span class="header-section-number">14.4.1</span> GSEA</h3>
<p>Quickly run GSEA for these samples:</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-1"></a></span>
<span id="cb262-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-2"></a>EpGenes=<span class="kw">rownames</span>(ResA)</span>
<span id="cb262-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-3"></a></span>
<span id="cb262-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-4"></a>l1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb262-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-5"></a>l2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb262-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-6"></a>l3=Mouse2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, Mouse2Hum<span class="op">$</span>MGI.symbol)]</span>
<span id="cb262-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-7"></a>EpGenesConv=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(l1)<span class="op">==</span>F, l1, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l2)<span class="op">==</span>F, l2, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l3)<span class="op">==</span>F, l3, EpGenes)))</span>
<span id="cb262-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-8"></a></span>
<span id="cb262-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-9"></a></span>
<span id="cb262-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-10"></a>hits=EpGenesConv[<span class="kw">match</span>(<span class="kw">rownames</span>(ResAb), EpGenes)]</span>
<span id="cb262-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-11"></a>fcTab=ResA<span class="op">$</span>log2FoldChange</span>
<span id="cb262-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-12"></a><span class="kw">names</span>(fcTab)=EpGenesConv</span>
<span id="cb262-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-13"></a></span>
<span id="cb262-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-14"></a></span>
<span id="cb262-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-15"></a>  gscaepInf=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb262-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-16"></a>  gscaepInf &lt;-<span class="st"> </span><span class="kw">preprocess</span>(gscaepInf, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb262-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-17"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb262-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-18"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb262-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-19"></a><span class="co">## -Preprocessing for input gene list and hit list ...</span></span>
<span id="cb262-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-20"></a><span class="co">## --Removing genes without values in geneList ...</span></span>
<span id="cb262-21"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-21"></a><span class="co">## --Removing duplicated genes ...</span></span>
<span id="cb262-22"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-22"></a><span class="co">## --Converting annotations ...</span></span>
<span id="cb262-23"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-23"></a><span class="co">## -- 527 genes (out of 12341) could not be mapped to any identifier, and were removed from the data. </span></span>
<span id="cb262-24"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-24"></a><span class="co">## -- 30 genes (out of 265) could not be mapped to any identifier, and were removed from the data. </span></span>
<span id="cb262-25"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-25"></a><span class="co">## --Ordering Gene List decreasingly ...</span></span>
<span id="cb262-26"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-26"></a><span class="co">## -Preprocessing complete!</span></span>
<span id="cb262-27"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-27"></a>  gscaepInf &lt;-<span class="st"> </span><span class="kw">analyze</span>(gscaepInf, </span>
<span id="cb262-28"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-28"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb262-29"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-29"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb262-30"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-30"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb262-31"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-31"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb262-32"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-32"></a><span class="co">## --146 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named c2List!</span></span>
<span id="cb262-33"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-33"></a><span class="co">## --848 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named c5BP!</span></span>
<span id="cb262-34"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-34"></a><span class="co">## --269 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named c5MF!</span></span>
<span id="cb262-35"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-35"></a><span class="co">## --110 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named c5CC!</span></span>
<span id="cb262-36"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-36"></a><span class="co">## --377 gene sets don&#39;t have &gt;= 5 overlapped genes with universe in gene set collection named MetPathway!</span></span>
<span id="cb262-37"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-37"></a><span class="co">## -Performing gene set enrichment analysis using HTSanalyzeR2... </span></span>
<span id="cb262-38"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-38"></a><span class="co">## --Calculating the permutations ... </span></span>
<span id="cb262-39"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-39"></a><span class="co">## -Gene set enrichment analysis using HTSanalyzeR2 complete </span></span>
<span id="cb262-40"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-40"></a><span class="co">## ==============================================</span></span>
<span id="cb262-41"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-41"></a>  </span>
<span id="cb262-42"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-42"></a>A1=<span class="kw">summarize</span>(gscaepInf)  </span>
<span id="cb262-43"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-43"></a><span class="co">## </span></span>
<span id="cb262-44"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-44"></a><span class="co">## -No of genes in Gene set collections: </span></span>
<span id="cb262-45"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-45"></a><span class="co">##                 input above min size</span></span>
<span id="cb262-46"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-46"></a><span class="co">## c2List           2199           2053</span></span>
<span id="cb262-47"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-47"></a><span class="co">## c5BP             7530           6682</span></span>
<span id="cb262-48"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-48"></a><span class="co">## c5MF             1663           1394</span></span>
<span id="cb262-49"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-49"></a><span class="co">## c5CC              999            889</span></span>
<span id="cb262-50"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-50"></a><span class="co">## ProcessNetworks   158            158</span></span>
<span id="cb262-51"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-51"></a><span class="co">## MetPathway       1480           1103</span></span>
<span id="cb262-52"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-52"></a><span class="co">## Hallmark           50             50</span></span>
<span id="cb262-53"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-53"></a><span class="co">## </span></span>
<span id="cb262-54"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-54"></a><span class="co">## </span></span>
<span id="cb262-55"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-55"></a><span class="co">## -No of genes in Gene List: </span></span>
<span id="cb262-56"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-56"></a><span class="co">##           input valid duplicate removed converted to entrez</span></span>
<span id="cb262-57"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-57"></a><span class="co">## Gene List 12449 12449             12341               11814</span></span>
<span id="cb262-58"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-58"></a><span class="co">## </span></span>
<span id="cb262-59"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-59"></a><span class="co">## </span></span>
<span id="cb262-60"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-60"></a><span class="co">## -No of hits: </span></span>
<span id="cb262-61"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-61"></a><span class="co">##      input preprocessed</span></span>
<span id="cb262-62"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-62"></a><span class="co">## Hits   265          235</span></span>
<span id="cb262-63"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-63"></a><span class="co">## </span></span>
<span id="cb262-64"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-64"></a><span class="co">## </span></span>
<span id="cb262-65"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-65"></a><span class="co">## -Parameters for analysis: </span></span>
<span id="cb262-66"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-66"></a><span class="co">##               minGeneSetSize pValueCutoff pAdjustMethod</span></span>
<span id="cb262-67"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-67"></a><span class="co">## HyperGeo Test 5              0.05         BH           </span></span>
<span id="cb262-68"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-68"></a><span class="co">## </span></span>
<span id="cb262-69"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-69"></a><span class="co">##      minGeneSetSize pValueCutoff pAdjustMethod nPermutations exponent</span></span>
<span id="cb262-70"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-70"></a><span class="co">## GSEA 5              0.05         BH            100           1       </span></span>
<span id="cb262-71"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-71"></a><span class="co">## </span></span>
<span id="cb262-72"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-72"></a><span class="co">## </span></span>
<span id="cb262-73"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-73"></a><span class="co">## -Significant gene sets (adjusted p-value&lt; 0.05 ): </span></span>
<span id="cb262-74"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-74"></a><span class="co">##          c2List c5BP c5MF c5CC ProcessNetworks MetPathway Hallmark</span></span>
<span id="cb262-75"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-75"></a><span class="co">## HyperGeo     NA   NA   NA   NA              NA         NA       NA</span></span>
<span id="cb262-76"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-76"></a><span class="co">## GSEA        188  641  116  130              17         16       19</span></span>
<span id="cb262-77"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-77"></a><span class="co">## Both         NA   NA   NA   NA              NA         NA       NA</span></span>
<span id="cb262-78"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-78"></a></span>
<span id="cb262-79"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-79"></a>PNresultsef=gscaepInf<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks</span>
<span id="cb262-80"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-80"></a></span>
<span id="cb262-81"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-81"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(gscaepInf<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb262-82"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-82"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(gscaepInf<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb262-83"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-83"></a></span>
<span id="cb262-84"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-84"></a><span class="co">## check whether this runs:</span></span>
<span id="cb262-85"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-85"></a>gscaepInf<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks<span class="op">$</span>Gene.Set.Term=TermsA</span>
<span id="cb262-86"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-86"></a><span class="kw">viewEnrichMap</span>(gscaepInf, <span class="dt">gscs=</span><span class="kw">c</span>(<span class="st">&quot;ProcessNetworks&quot;</span>),</span>
<span id="cb262-87"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb262-87"></a>              <span class="dt">allSig =</span> <span class="ot">TRUE</span>, <span class="dt">gsNameType=</span><span class="st">&quot;term&quot;</span> )</span></code></pre></div>
<div class="figure"><span id="fig:gsea-non-inflammatory-branch"></span>
<div id="htmlwidget-cca7584f569b158a6d5e" style="width:672px;height:480px;" class="forceGraph html-widget"></div>
<script type="application/json" data-for="htmlwidget-cca7584f569b158a6d5e">{"x":{"nodes":{"id":["ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion"],"size":[3,5.64444444444444,8.54074074074074,8.91851851851852,10.5555555555556,11.6888888888889,12.0666666666667,12.1925925925926,12.8222222222222,13.3259259259259,14.2074074074074,14.837037037037,14.962962962963,15.2148148148148,15.4666666666667,18.1111111111111,20],"color":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"scheme":["neg","neg","pos","neg","pos","pos","neg","neg","neg","neg","neg","neg","neg","neg","neg","pos","neg"],"label":["Melatonin signaling","Visual perception","ESR2 pathway","Th17-derived cytokines","IFN-gamma signaling","Kallikrein-kinin system","Skeletal muscle development","Potassium transport","TREM1 signaling","Muscle contraction","Glycoconjugates","Neurogenesis in general","Calcium transport","Histamine signaling","Cell-matrix interactions","Ubiquitin-proteasomal proteolysis","Integrin-mediated cell-matrix adhesion"],"label_id":["NNeurophysiological process_Melatonin signaling","NNeurophysiological process_Visual perception","NSignal transduction_ESR2 pathway","NImmune response_Th17-derived cytokines","NInflammation_IFN-gamma signaling","NInflammation_Kallikrein-kinin system","NDevelopment_Skeletal muscle development","NTransport_Potassium transport","NInflammation_TREM1 signaling","NMuscle contraction","NCell adhesion_Glycoconjugates","NDevelopment_Neurogenesis in general","NTransport_Calcium transport","NInflammation_Histamine signaling","NCell adhesion_Cell-matrix interactions","NProteolysis_Ubiquitin-proteasomal proteolysis","NCell adhesion_Integrin-mediated cell-matrix adhesion"],"label_term":["Melatonin signaling","Visual perception","ESR2 pathway","Th17-derived cytokines","IFN-gamma signaling","Kallikrein-kinin system","Skeletal muscle development","Potassium transport","TREM1 signaling","Muscle contraction","Glycoconjugates","Neurogenesis in general","Calcium transport","Histamine signaling","Cell-matrix interactions","Ubiquitin-proteasomal proteolysis","Integrin-mediated cell-matrix adhesion"]},"links":{"source":["ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Melatonin signaling","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NMuscle contraction","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions"],"target":["ProcessNetworks.NNeurophysiological process_Visual perception","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NSignal transduction_ESR2 pathway","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NTransport_Potassium transport","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NImmune response_Th17-derived cytokines","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NInflammation_IFN-gamma signaling","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NInflammation_Kallikrein-kinin system","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NDevelopment_Skeletal muscle development","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NInflammation_TREM1 signaling","ProcessNetworks.NMuscle contraction","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NMuscle contraction","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NMuscle contraction","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NProteolysis_Ubiquitin-proteasomal proteolysis","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NCell adhesion_Glycoconjugates","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NDevelopment_Neurogenesis in general","ProcessNetworks.NTransport_Calcium transport","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NInflammation_Histamine signaling","ProcessNetworks.NCell adhesion_Cell-matrix interactions","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion","ProcessNetworks.NCell adhesion_Integrin-mediated cell-matrix adhesion"],"weight":[0.0535714285714286,0.0123456790123457,0.0315789473684211,0.0490196078431373,0.00917431192660551,0.018348623853211,0.00869565217391304,0.0344827586206897,0.0232558139534884,0.0152671755725191,0.0887096774193548,0.00980392156862745,0.0170940170940171,0.0078740157480315,0.0155038759689922,0.0153846153846154,0.00735294117647059,0.0291970802919708,0.00680272108843537,0.0335570469798658,0.0263157894736842,0.00518134715025907,0.075,0.0441176470588235,0.0785714285714286,0.0198675496688742,0.0666666666666667,0.00613496932515337,0.0178571428571429,0.0173410404624277,0.00568181818181818,0.0529411764705882,0.00555555555555556,0.0253807106598985,0.0382775119617225,0.169354838709677,0.124087591240876,0.00641025641025641,0.181159420289855,0.0235294117647059,0.0112994350282486,0.00558659217877095,0.130434782608696,0.0222222222222222,0.042654028436019,0.159722222222222,0.0179640718562874,0.150326797385621,0.0112359550561798,0.0564971751412429,0.00523560209424084,0.0265957446808511,0.133720930232558,0.0154639175257732,0.00460829493087558,0.0639269406392694,0.0112994350282486,0.149068322981366,0.0216216216216216,0.0315789473684211,0.0203045685279188,0.0202020202020202,0.236363636363636,0.066079295154185,0.0162162162162162,0.215189873417722,0.0310880829015544,0.02,0.0199004975124378,0.0247524752475248,0.0663265306122449,0.0425531914893617,0.0604395604395604,0.00490196078431373,0.0673575129533679,0.00478468899521531,0.0050761421319797,0.0406091370558376,0.00961538461538462,0.0242718446601942,0.189944134078212,0.0336538461538462,0.00425531914893617,0.0865800865800866,0.0295566502463054,0.014218009478673,0.0913705583756345,0.0382775119617225,0.0428571428571429,0.0493827160493827,0.00454545454545455,0.00909090909090909,0.087378640776699,0.13,0.212962962962963,0.0177777777777778,0.0131578947368421,0.0152091254752852,0.031390134529148,0.00432900432900433,0.0075187969924812,0.0843373493975904,0.133333333333333]},"options":{"title":"Enrichment Map of GSEA on ProcessNetworks","legendTitle":"-Log10(Adjusted p-values)","type":"GSEA","nodeScheme":"dual","label":{"text":"term"},"colorScaler":"log10","nPermutations":100,"colorDomain":{"neg":[0,0.05],"pos":[0,0.05]}}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 14.6: 5g: non inflammatory branch
</p>
</div>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb263-1"></a><span class="co">#plotGSEA(gscaepInf, gscs=c(&quot;ProcessNetworks&quot;), filepath=&quot;figure-outputs/&quot;, output=&quot;pdf&quot;)</span></span>
<span id="cb263-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb263-2"></a></span>
<span id="cb263-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb263-3"></a></span>
<span id="cb263-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb263-4"></a><span class="kw">save</span>(gscaepInf2, <span class="dt">file=</span><span class="st">&quot;figure-outputs/5g.Rdata&quot;</span>) <span class="co"># save this file to change the color scheme</span></span>
<span id="cb263-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb263-5"></a><span class="kw">write.csv</span>(gscaepInf<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks, <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext5g_perhaps.csv&quot;</span>)</span></code></pre></div>
<p>We’ve noticed that some of the differentially expressed genes above are splicing related or epigenetic related. Could there be an association with transcriptional diversity? Below we calculate the transcriptional diversity based on rsem values</p>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-1"></a>local.rnaseq.shannon &lt;-<span class="st"> </span><span class="cf">function</span>(exp.mat, <span class="dt">pseudoNum =</span> <span class="dv">0</span>){</span>
<span id="cb264-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-2"></a><span class="co"># calculate shannon index from transcriptome matrix</span></span>
<span id="cb264-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-3"></a><span class="kw">apply</span>(exp.mat, <span class="dv">2</span>, <span class="cf">function</span>(x){</span>
<span id="cb264-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-4"></a>x&lt;-x<span class="op">+</span>pseudoNum</span>
<span id="cb264-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-5"></a>prop&lt;-x<span class="op">/</span><span class="kw">sum</span>(x)</span>
<span id="cb264-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-6"></a><span class="co">#prop&lt;-prop[prop&gt;0]</span></span>
<span id="cb264-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-7"></a>shidx =<span class="st"> </span><span class="op">-</span><span class="kw">sum</span>(prop<span class="op">*</span><span class="kw">log</span>(prop), <span class="dt">na.rm=</span>T)<span class="op">/</span><span class="kw">log</span>(<span class="kw">length</span>(prop))</span>
<span id="cb264-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-8"></a>shidx</span>
<span id="cb264-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-9"></a>})</span>
<span id="cb264-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-10"></a>}</span>
<span id="cb264-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-11"></a></span>
<span id="cb264-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-12"></a></span>
<span id="cb264-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-13"></a>Output1=<span class="kw">local.rnaseq.shannon</span>(allrsemFinal)</span>
<span id="cb264-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-14"></a>Output2=Output1[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;Ep&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb264-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-15"></a></span>
<span id="cb264-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-16"></a>tab2=<span class="kw">data.frame</span>(<span class="dt">inf=</span>vstEpInf<span class="op">$</span>Inflammation2, <span class="dt">het=</span>Output2[<span class="op">-</span><span class="dv">3</span>])</span>
<span id="cb264-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb264-17"></a><span class="kw">ggplot</span>(tab2, <span class="kw">aes</span>(<span class="dt">x=</span>inf, <span class="dt">y=</span>het))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;transcriptional heterogeneity: rsem&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Ext5j"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext5j-1.png" alt="Transcriptional heterogeneity" width="672" />
<p class="caption">
Figure 14.7: Transcriptional heterogeneity
</p>
</div>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-1"></a><span class="kw">wilcox.test</span>(tab2<span class="op">$</span>het[tab2<span class="op">$</span>inf<span class="op">!=</span><span class="st">&quot;inf&quot;</span>]<span class="op">~</span>tab2<span class="op">$</span>inf[tab2<span class="op">$</span>inf<span class="op">!=</span><span class="st">&quot;inf&quot;</span>])</span>
<span id="cb265-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-2"></a><span class="co">## </span></span>
<span id="cb265-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-3"></a><span class="co">##  Wilcoxon rank sum exact test</span></span>
<span id="cb265-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-4"></a><span class="co">## </span></span>
<span id="cb265-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-5"></a><span class="co">## data:  tab2$het[tab2$inf != &quot;inf&quot;] by tab2$inf[tab2$inf != &quot;inf&quot;]</span></span>
<span id="cb265-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-6"></a><span class="co">## W = 9, p-value = 0.05528</span></span>
<span id="cb265-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-7"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb265-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-8"></a></span>
<span id="cb265-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb265-9"></a><span class="kw">write.csv</span>(tab2, <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext5j_transcriptional_heterigeneity.csv&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="luminal-only-non-inflammtory-samples-samples" class="section level2">
<h2><span class="header-section-number">14.5</span> Luminal-only non-inflammtory samples samples</h2>
<p>We removed all the basal samples and did the same comparison:</p>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-1"></a>Epdds<span class="op">$</span>HR=Cdata<span class="op">$</span>HR_status[<span class="kw">match</span>(<span class="kw">substr</span>(<span class="kw">colnames</span>(Epdds), <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">colnames</span>(Epdds))<span class="op">-</span><span class="dv">3</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb266-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-2"></a>vstLumOnly=Epdds[, <span class="kw">which</span>(Epdds<span class="op">$</span>HR<span class="op">==</span><span class="st">&quot;Lum&quot;</span>)]</span>
<span id="cb266-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-3"></a><span class="kw">design</span>(vstLumOnly)=<span class="er">~</span>Growth</span>
<span id="cb266-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-4"></a>vstLumOnly=<span class="kw">DESeq</span>(vstLumOnly)</span>
<span id="cb266-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-5"></a><span class="kw">colnames</span>(Epdds)</span>
<span id="cb266-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-6"></a><span class="co">##  [1] &quot;10L_D_Ep&quot;  &quot;10R_BL_Ep&quot; &quot;11L_B_Ep&quot;  &quot;11N_D_Ep&quot;  &quot;11R_D_Ep&quot;  &quot;11R_C_Ep&quot; </span></span>
<span id="cb266-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-7"></a><span class="co">##  [7] &quot;12L_D_Ep&quot;  &quot;14N_C_Ep&quot;  &quot;14N_D_Ep&quot;  &quot;14R_B_Ep&quot;  &quot;15N_C_Ep&quot;  &quot;16L_C_Ep&quot; </span></span>
<span id="cb266-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-8"></a><span class="co">## [13] &quot;17N_D_Ep&quot;  &quot;2N__Ep&quot;    &quot;3N_B_Ep&quot;   &quot;3R_C_Ep&quot;   &quot;6R_B_Ep&quot;   &quot;7N_A_Ep&quot;  </span></span>
<span id="cb266-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-9"></a><span class="co">## [19] &quot;8L_D_Ep&quot;   &quot;8R_CU_Ep&quot;</span></span>
<span id="cb266-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-10"></a></span>
<span id="cb266-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-11"></a>vstLumRes=<span class="kw">results</span>(vstLumOnly)</span>
<span id="cb266-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-12"></a>vsdLumvst=<span class="kw">vst</span>(vstLumOnly)</span>
<span id="cb266-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-13"></a></span>
<span id="cb266-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-14"></a><span class="kw">write.csv</span>(vstLumRes, <span class="dt">file=</span><span class="st">&quot;nature-tables/5j_lumonly_ep_growing_vs_stable.csv&quot;</span>)</span>
<span id="cb266-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-15"></a>genes2=<span class="kw">rownames</span>(vstLumRes)[<span class="kw">which</span>(vstLumRes<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(vstLumRes<span class="op">$</span>log2FoldChange)<span class="op">&gt;</span><span class="fl">1.5</span> <span class="op">&amp;</span><span class="st"> </span>vstLumRes<span class="op">$</span>baseMean<span class="op">&gt;</span><span class="dv">100</span>)]</span>
<span id="cb266-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-16"></a></span>
<span id="cb266-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-17"></a>ColSideCols=ColSizeb[vstLumOnly<span class="op">$</span>Growth]</span>
<span id="cb266-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-18"></a><span class="co">#pdf(&quot;figure-outputs/Figure5_XX_heatmap_lumonly_growing_vs_stable.pdf&quot;, height=9, width=5)</span></span>
<span id="cb266-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-19"></a></span>
<span id="cb266-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb266-20"></a><span class="kw">heatmap.2</span>(<span class="kw">assay</span>(vsdLumvst)[ genes2,  ], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSideCols, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">main=</span><span class="st">&quot;Ep genes&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:EpddsLum"></span>
<img src="RatDCIS-bookdown_files/figure-html/EpddsLum-1.png" alt="DEG non-inflamm Lum only" width="672" />
<p class="caption">
Figure 14.8: DEG non-inflamm Lum only
</p>
</div>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb267-1"></a><span class="co">#dev.off()</span></span>
<span id="cb267-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb267-2"></a></span>
<span id="cb267-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb267-3"></a><span class="kw">write.csv</span>(<span class="kw">assay</span>(vsdLumvst)[ genes2,  ], <span class="dt">file=</span><span class="st">&quot;nature-tables/5xx_lum_only_growing_vs_stable.csv&quot;</span>)</span></code></pre></div>
<p>Quickly run GSEA for these samples:</p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-1"></a></span>
<span id="cb268-2"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-2"></a>EpGenes=<span class="kw">rownames</span>(vstLumRes)</span>
<span id="cb268-3"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-3"></a></span>
<span id="cb268-4"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-4"></a>l1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb268-5"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-5"></a>l2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb268-6"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-6"></a>l3=Mouse2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(EpGenes, Mouse2Hum<span class="op">$</span>MGI.symbol)]</span>
<span id="cb268-7"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-7"></a>EpGenesConv=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(l1)<span class="op">==</span>F, l1, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l2)<span class="op">==</span>F, l2, <span class="kw">ifelse</span>(<span class="kw">is.na</span>(l3)<span class="op">==</span>F, l3, EpGenes)))</span>
<span id="cb268-8"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-8"></a></span>
<span id="cb268-9"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-9"></a></span>
<span id="cb268-10"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-10"></a>hits=EpGenesConv[<span class="kw">match</span>(<span class="kw">rownames</span>(vstLumRes), EpGenes)]</span>
<span id="cb268-11"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-11"></a>fcTab=vstLumRes<span class="op">$</span>log2FoldChange</span>
<span id="cb268-12"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-12"></a><span class="kw">names</span>(fcTab)=EpGenesConv</span>
<span id="cb268-13"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-13"></a></span>
<span id="cb268-14"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-14"></a></span>
<span id="cb268-15"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-15"></a>  gscaepNI_lum=<span class="kw">GSCA</span>(<span class="dt">listOfGeneSetCollections=</span>ListGSC,<span class="dt">geneList=</span>fcTab, <span class="dt">hits =</span> hits)</span>
<span id="cb268-16"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-16"></a>  gscaepNI_lum&lt;-<span class="st"> </span><span class="kw">preprocess</span>(gscaepNI_lum, <span class="dt">species=</span><span class="st">&quot;Hs&quot;</span>, <span class="dt">initialIDs=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb268-17"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-17"></a>                    <span class="dt">keepMultipleMappings=</span><span class="ot">TRUE</span>, <span class="dt">duplicateRemoverMethod=</span><span class="st">&quot;max&quot;</span>,</span>
<span id="cb268-18"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-18"></a>                    <span class="dt">orderAbsValue=</span><span class="ot">FALSE</span>)</span>
<span id="cb268-19"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-19"></a>  gscaepNI_lum &lt;-<span class="st"> </span><span class="kw">analyze</span>(gscaepNI_lum, </span>
<span id="cb268-20"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-20"></a>                 <span class="dt">para=</span><span class="kw">list</span>(<span class="dt">pValueCutoff=</span><span class="fl">0.05</span>, <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>,</span>
<span id="cb268-21"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-21"></a>                           <span class="dt">nPermutations=</span><span class="dv">100</span>, <span class="dt">minGeneSetSize=</span><span class="dv">5</span>,</span>
<span id="cb268-22"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-22"></a>                           <span class="dt">exponent=</span><span class="dv">1</span>), </span>
<span id="cb268-23"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-23"></a>                           <span class="dt">doGSOA =</span> F)</span>
<span id="cb268-24"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-24"></a>  </span>
<span id="cb268-25"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-25"></a>A1=<span class="kw">summarize</span>(gscaepNI_lum)  </span>
<span id="cb268-26"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-26"></a></span>
<span id="cb268-27"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-27"></a>PNresultsef=gscaepNI_lum<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks</span>
<span id="cb268-28"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-28"></a></span>
<span id="cb268-29"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-29"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(gscaepNI_lum<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb268-30"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-30"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(gscaepNI_lum<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb268-31"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-31"></a></span>
<span id="cb268-32"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-32"></a><span class="kw">save</span>(gscaepNI_lum, <span class="dt">file=</span><span class="st">&quot;figure-outputs/5xx_lumonly.Rdata&quot;</span>) <span class="co"># save this file to change the colors</span></span>
<span id="cb268-33"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-33"></a></span>
<span id="cb268-34"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-34"></a><span class="co">## check whether this runs:</span></span>
<span id="cb268-35"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-35"></a>gscaepNI_lum<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks<span class="op">$</span>Gene.Set.Term=TermsA</span>
<span id="cb268-36"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-36"></a><span class="kw">viewEnrichMap</span>(gscaepNI_lum, <span class="dt">gscs=</span><span class="kw">c</span>(<span class="st">&quot;ProcessNetworks&quot;</span>),</span>
<span id="cb268-37"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-37"></a>              <span class="dt">allSig =</span> <span class="ot">TRUE</span>, <span class="dt">gsNameType=</span><span class="st">&quot;term&quot;</span> )</span>
<span id="cb268-38"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-38"></a></span>
<span id="cb268-39"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#cb268-39"></a><span class="kw">write.csv</span>(gscaepNI_lum<span class="op">@</span>result<span class="op">$</span>GSEA.results<span class="op">$</span>ProcessNetworks, <span class="dt">file=</span><span class="st">&quot;nature-tables/5g_lum_gsea.csv&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="summary-of-gsea-runs.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="tcga-associate-epcam-inflammatory-with-survival.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
