<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 ER/Pgr Subtyping | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 ER/Pgr Subtyping | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 ER/Pgr Subtyping | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-03-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="collating-results-and-running-gsea.html"/>
<link rel="next" href="expression-in-specific-pathways.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.3.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.3.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>24.3</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="erpgr-subtyping" class="section level1">
<h1><span class="header-section-number">Chapter 10</span> ER/Pgr Subtyping</h1>
<p>In this section, we will look at whether there are differences in rat subtype by:</p>
<ul>
<li>creating a list of specific markers of interest</li>
<li>applying PAM clsutering (partitioning around medoids)</li>
</ul>
<p>As the DN samples do have keratin expression, we will also try to determine the subtypes in these samples too.</p>
<p>In the below analyses, we will conduct this subtyping as follows:</p>
<ul>
<li>progression specific epithelial cohort</li>
<li>characterisation specific epithelial cohort</li>
<li>dn specific cohort (progression)</li>
</ul>
<div id="gene-list" class="section level2">
<h2><span class="header-section-number">10.1</span> Gene List</h2>
<p>We will assess clustering using two different lists. A. is a more comprehensive list containing epithelial, mesenchymal markers and proliferation markers, B. has a more narrow list of subtype specific markers</p>
<ol style="list-style-type: lower-alpha">
<li>‘Ar’, ‘Cd24’, ‘Cdh1’, ‘Foxa1’, ‘Gata3’, ‘Krt8’, ‘Krt18’, ‘Krt5’, ‘Vim’, ‘Erbb2’, ‘Esr1’, ‘Pgr’, ‘Mki67’, ‘Pcna’</li>
<li>‘Ar’, ‘Foxa1’, ‘Gata3’, ‘Erbb2’, ‘Esr1’, ‘Pgr’</li>
</ol>
</div>
<div id="progression-cohort" class="section level2">
<h2><span class="header-section-number">10.2</span> Progression cohort</h2>
<p>We will be using variance stabilised counts in this section.</p>
<p>Firstly, we will look at the progression cohort. Note that the overlap between the two gene-sets are similar, and the heatmap is row-scaled.</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="erpgr-subtyping.html#cb167-1"></a>Markers1=<span class="kw">c</span>(<span class="st">&#39;Ar&#39;</span>, <span class="st">&#39;Cd24&#39;</span>, <span class="st">&#39;Cdh1&#39;</span>, <span class="st">&#39;Foxa1&#39;</span>, <span class="st">&#39;Gata3&#39;</span>, <span class="st">&#39;Krt8&#39;</span>, <span class="st">&#39;Krt18&#39;</span>, <span class="st">&#39;Krt5&#39;</span>, <span class="st">&#39;Vim&#39;</span>, <span class="st">&#39;Erbb2&#39;</span>, <span class="st">&#39;Esr1&#39;</span>, <span class="st">&#39;Pgr&#39;</span>, <span class="st">&#39;Mki67&#39;</span>, <span class="st">&#39;Pcna&#39;</span>)</span>
<span id="cb167-2"><a href="erpgr-subtyping.html#cb167-2"></a>Markers2=<span class="kw">c</span>(<span class="st">&#39;Ar&#39;</span>, <span class="st">&#39;Foxa1&#39;</span>, <span class="st">&#39;Gata3&#39;</span>, <span class="st">&#39;Erbb2&#39;</span>, <span class="st">&#39;Esr1&#39;</span>, <span class="st">&#39;Pgr&#39;</span>)</span>
<span id="cb167-3"><a href="erpgr-subtyping.html#cb167-3"></a><span class="co"># Progression cohort</span></span>
<span id="cb167-4"><a href="erpgr-subtyping.html#cb167-4"></a>Ax1a=<span class="kw">assay</span>(vstEp)[ <span class="kw">match</span>(Markers1, <span class="kw">rownames</span>(<span class="kw">assay</span>(vstEp))),]</span>
<span id="cb167-5"><a href="erpgr-subtyping.html#cb167-5"></a>Ax2a=<span class="kw">assay</span>(vstEp)[ <span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vstEp))),]</span>
<span id="cb167-6"><a href="erpgr-subtyping.html#cb167-6"></a>Pam1=<span class="kw">pam</span>(<span class="kw">t</span>(Ax1a), <span class="dv">2</span>)</span>
<span id="cb167-7"><a href="erpgr-subtyping.html#cb167-7"></a>Pam2=<span class="kw">pam</span>(<span class="kw">t</span>(Ax2a), <span class="dv">2</span>)</span>
<span id="cb167-8"><a href="erpgr-subtyping.html#cb167-8"></a></span>
<span id="cb167-9"><a href="erpgr-subtyping.html#cb167-9"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb167-10"><a href="erpgr-subtyping.html#cb167-10"></a><span class="kw">plot</span>(Pam1, <span class="dt">main=</span><span class="st">&quot;List A, progression cohort&quot;</span>)</span>
<span id="cb167-11"><a href="erpgr-subtyping.html#cb167-11"></a><span class="kw">plot</span>(Pam2, <span class="dt">main=</span><span class="st">&quot;List B, progression cohort&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-64-1.png" width="672" /></p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="erpgr-subtyping.html#cb168-1"></a></span>
<span id="cb168-2"><a href="erpgr-subtyping.html#cb168-2"></a>ColA=Pam1<span class="op">$</span>clustering</span>
<span id="cb168-3"><a href="erpgr-subtyping.html#cb168-3"></a>ColA=<span class="kw">ifelse</span>(ColA<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span>
<span id="cb168-4"><a href="erpgr-subtyping.html#cb168-4"></a>ColB=Pam2<span class="op">$</span>clustering</span>
<span id="cb168-5"><a href="erpgr-subtyping.html#cb168-5"></a>ColB=<span class="kw">ifelse</span>(ColB<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span></code></pre></div>
<p>The following shows the heatmaps using these list, using average distances in the hclustering approaches: firstly, using the longer list</p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="erpgr-subtyping.html#cb169-1"></a><span class="co">#colnames(Ax1a)=infoTableFinal$TumorIDnew[match(colnames(Ax1a), rownames(infoTableFinal))]</span></span>
<span id="cb169-2"><a href="erpgr-subtyping.html#cb169-2"></a></span>
<span id="cb169-3"><a href="erpgr-subtyping.html#cb169-3"></a><span class="co">#hclust.ave &lt;- function(x) hclust(x, method=&quot;average&quot;)</span></span>
<span id="cb169-4"><a href="erpgr-subtyping.html#cb169-4"></a>ax1=<span class="kw">heatmap.2</span>(Ax1a, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;narrow list&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(ColA)])</span></code></pre></div>
<div class="figure"><span id="fig:Ext3i"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext3i-1.png" alt="HR clustering" width="672" />
<p class="caption">
Figure 10.1: HR clustering
</p>
</div>
<p>and now using the shorter list</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="erpgr-subtyping.html#cb170-1"></a>ax1=<span class="kw">heatmap.2</span>(Ax2a, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;narrow list&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(ColB)])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-65-1.png" width="672" /></p>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="erpgr-subtyping.html#cb171-1"></a></span>
<span id="cb171-2"><a href="erpgr-subtyping.html#cb171-2"></a><span class="kw">write.csv</span>(ax1<span class="op">$</span>carpet, <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext3i.csv&quot;</span>)</span>
<span id="cb171-3"><a href="erpgr-subtyping.html#cb171-3"></a></span>
<span id="cb171-4"><a href="erpgr-subtyping.html#cb171-4"></a><span class="kw">colnames</span>(Ax2a)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(Ax2a), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb171-5"><a href="erpgr-subtyping.html#cb171-5"></a>Cdata<span class="op">$</span>HR_status=<span class="ot">NA</span></span>
<span id="cb171-6"><a href="erpgr-subtyping.html#cb171-6"></a>Cdata<span class="op">$</span>HR_status[<span class="kw">match</span>(<span class="kw">colnames</span>(Ax2a), Cdata<span class="op">$</span>NewID)]=ColB</span></code></pre></div>
</div>
<div id="charcterisation-cohort" class="section level2">
<h2><span class="header-section-number">10.3</span> Charcterisation cohort</h2>
<p>We can perform a similar analysis to the characterisation cohort: (and save to file)</p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="erpgr-subtyping.html#cb172-1"></a>c2=infoTableFinal<span class="op">$</span>SampleID[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;Ep&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">!=</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb172-2"><a href="erpgr-subtyping.html#cb172-2"></a></span>
<span id="cb172-3"><a href="erpgr-subtyping.html#cb172-3"></a>Ax1=<span class="kw">assay</span>(vsd)[ <span class="kw">match</span>(Markers1, <span class="kw">rownames</span>(<span class="kw">assay</span>(vsd))), <span class="kw">match</span>(c2, <span class="kw">colnames</span>(vsd))]</span>
<span id="cb172-4"><a href="erpgr-subtyping.html#cb172-4"></a>Ax2=<span class="kw">assay</span>(vsd)[ <span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vsd))), <span class="kw">match</span>(c2, <span class="kw">colnames</span>(vsd))]</span>
<span id="cb172-5"><a href="erpgr-subtyping.html#cb172-5"></a></span>
<span id="cb172-6"><a href="erpgr-subtyping.html#cb172-6"></a>Pam3=<span class="kw">pam</span>(<span class="kw">t</span>(Ax1), <span class="dv">2</span>)</span>
<span id="cb172-7"><a href="erpgr-subtyping.html#cb172-7"></a>Pam4=<span class="kw">pam</span>(<span class="kw">t</span>(Ax2), <span class="dv">2</span>)</span>
<span id="cb172-8"><a href="erpgr-subtyping.html#cb172-8"></a></span>
<span id="cb172-9"><a href="erpgr-subtyping.html#cb172-9"></a><span class="co">#plot(Pam3)</span></span>
<span id="cb172-10"><a href="erpgr-subtyping.html#cb172-10"></a><span class="kw">plot</span>(Pam4, <span class="dt">main=</span><span class="st">&quot;narrow list&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-66-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-66-2.png" width="672" /></p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="erpgr-subtyping.html#cb173-1"></a></span>
<span id="cb173-2"><a href="erpgr-subtyping.html#cb173-2"></a>Ax1b=<span class="kw">scale</span>(<span class="kw">t</span>(Ax1))</span>
<span id="cb173-3"><a href="erpgr-subtyping.html#cb173-3"></a></span>
<span id="cb173-4"><a href="erpgr-subtyping.html#cb173-4"></a><span class="co">#plot(Pam3, &quot;extended list&quot;)</span></span>
<span id="cb173-5"><a href="erpgr-subtyping.html#cb173-5"></a>ColC=Pam3<span class="op">$</span>clustering</span>
<span id="cb173-6"><a href="erpgr-subtyping.html#cb173-6"></a>ColC=<span class="kw">ifelse</span>(ColC<span class="op">==</span><span class="dv">2</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span>
<span id="cb173-7"><a href="erpgr-subtyping.html#cb173-7"></a>ColD=Pam4<span class="op">$</span>clustering</span>
<span id="cb173-8"><a href="erpgr-subtyping.html#cb173-8"></a>ColD=<span class="kw">ifelse</span>(ColD<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span>
<span id="cb173-9"><a href="erpgr-subtyping.html#cb173-9"></a></span>
<span id="cb173-10"><a href="erpgr-subtyping.html#cb173-10"></a><span class="co">#table(ColC, ColD)</span></span>
<span id="cb173-11"><a href="erpgr-subtyping.html#cb173-11"></a></span>
<span id="cb173-12"><a href="erpgr-subtyping.html#cb173-12"></a><span class="co">#pdf(&quot;~/Desktop/PAM50-characterisation-different-list2.pdf&quot;, width=8, height=8)</span></span>
<span id="cb173-13"><a href="erpgr-subtyping.html#cb173-13"></a></span>
<span id="cb173-14"><a href="erpgr-subtyping.html#cb173-14"></a><span class="co"># heatmap.plus(t(Ax1b), col=brewer.pal(11, &quot;RdBu&quot;)[11:1], ColSideColors =CTable,scale=&quot;none&quot;, main=&quot;Char cohort vsd marker list2&quot;, trace=&quot;none&quot;)</span></span>
<span id="cb173-15"><a href="erpgr-subtyping.html#cb173-15"></a></span>
<span id="cb173-16"><a href="erpgr-subtyping.html#cb173-16"></a><span class="kw">colnames</span>(Ax1)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">rownames</span>(Ax1b), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb173-17"><a href="erpgr-subtyping.html#cb173-17"></a><span class="kw">heatmap.2</span>((Ax1), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(ColC)])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-66-3.png" width="672" /></p>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="erpgr-subtyping.html#cb174-1"></a><span class="kw">colnames</span>(Ax2)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">rownames</span>(Ax1b), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb174-2"><a href="erpgr-subtyping.html#cb174-2"></a><span class="kw">heatmap.2</span>((Ax2), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(ColD)])</span></code></pre></div>
<div class="figure"><span id="fig:1f"></span>
<img src="RatDCIS-bookdown_files/figure-html/1f-1.png" alt="HR status in characterisation cohort" width="672" />
<p class="caption">
Figure 10.2: HR status in characterisation cohort
</p>
</div>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="erpgr-subtyping.html#cb175-1"></a></span>
<span id="cb175-2"><a href="erpgr-subtyping.html#cb175-2"></a><span class="kw">write.csv</span>(Ax2, <span class="dt">file=</span><span class="st">&quot;nature-tables/1f.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="dn-samples" class="section level2">
<h2><span class="header-section-number">10.4</span> DN samples</h2>
<p>DN samples: note there is variability in DN expression of PR, Foxa1, Esr1, Ar,
Erbb2 and Gata3 shows fairly stable expression values</p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="erpgr-subtyping.html#cb176-1"></a>Ax2=<span class="kw">assay</span>(vsdLimmaDN)[ <span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vsdLimmaDN))),]</span>
<span id="cb176-2"><a href="erpgr-subtyping.html#cb176-2"></a>Pam5=<span class="kw">pam</span>(<span class="kw">t</span>(Ax2), <span class="dv">2</span>)</span>
<span id="cb176-3"><a href="erpgr-subtyping.html#cb176-3"></a><span class="kw">plot</span>(Pam5, <span class="dt">main=</span><span class="st">&quot;extended list DN&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-67-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-67-2.png" width="672" /></p>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="erpgr-subtyping.html#cb177-1"></a>ColE=Pam5<span class="op">$</span>clustering</span>
<span id="cb177-2"><a href="erpgr-subtyping.html#cb177-2"></a>ColE=<span class="kw">ifelse</span>(ColE<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span>
<span id="cb177-3"><a href="erpgr-subtyping.html#cb177-3"></a></span>
<span id="cb177-4"><a href="erpgr-subtyping.html#cb177-4"></a></span>
<span id="cb177-5"><a href="erpgr-subtyping.html#cb177-5"></a>ColE=<span class="kw">gsub</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;red&quot;</span>, ColE)</span>
<span id="cb177-6"><a href="erpgr-subtyping.html#cb177-6"></a>ColE=<span class="kw">gsub</span>(<span class="st">&quot;Lum&quot;</span>, <span class="st">&quot;blue&quot;</span>, ColE)</span>
<span id="cb177-7"><a href="erpgr-subtyping.html#cb177-7"></a></span>
<span id="cb177-8"><a href="erpgr-subtyping.html#cb177-8"></a><span class="co">#pdf(&quot;~/Desktop/5X-DN-samples-HR-status.pdf&quot;, width=6, height=5)</span></span>
<span id="cb177-9"><a href="erpgr-subtyping.html#cb177-9"></a><span class="kw">colnames</span>(Ax2)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(Ax2), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb177-10"><a href="erpgr-subtyping.html#cb177-10"></a><span class="kw">heatmap.2</span>(Ax2, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span>ColE,<span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;DN samples vsd marker list1&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-67-3.png" width="672" /></p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="erpgr-subtyping.html#cb178-1"></a><span class="co"># Check which samples are also in the epithelial samples:</span></span>
<span id="cb178-2"><a href="erpgr-subtyping.html#cb178-2"></a></span>
<span id="cb178-3"><a href="erpgr-subtyping.html#cb178-3"></a>V2=<span class="kw">intersect</span>(<span class="kw">colnames</span>(Ax1a), <span class="kw">colnames</span>(Ax2))</span>
<span id="cb178-4"><a href="erpgr-subtyping.html#cb178-4"></a><span class="kw">colnames</span>(Ax1a)</span>
<span id="cb178-5"><a href="erpgr-subtyping.html#cb178-5"></a><span class="kw">colnames</span>(Ax2)</span>
<span id="cb178-6"><a href="erpgr-subtyping.html#cb178-6"></a><span class="kw">dim</span>(Ax2)</span></code></pre></div>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="erpgr-subtyping.html#cb179-1"></a><span class="kw">heatmap.2</span>(Ax2[ ,V2], <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span>ColE[V2],<span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;DN samples vsd marker list1&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave)</span>
<span id="cb179-2"><a href="erpgr-subtyping.html#cb179-2"></a></span>
<span id="cb179-3"><a href="erpgr-subtyping.html#cb179-3"></a><span class="kw">names</span>(ColSideColB)=<span class="kw">colnames</span>(Ax1a)</span>
<span id="cb179-4"><a href="erpgr-subtyping.html#cb179-4"></a></span>
<span id="cb179-5"><a href="erpgr-subtyping.html#cb179-5"></a><span class="kw">heatmap.2</span>(Ax1a[ ,V2], <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span>ColSideColB[V2],<span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Ep samples&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave)</span>
<span id="cb179-6"><a href="erpgr-subtyping.html#cb179-6"></a></span>
<span id="cb179-7"><a href="erpgr-subtyping.html#cb179-7"></a>Tx1=<span class="kw">cbind</span>(Ax2[ ,V2], Ax1a[ ,V2])</span>
<span id="cb179-8"><a href="erpgr-subtyping.html#cb179-8"></a>ColSide=<span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;blue&quot;</span>, <span class="kw">length</span>(V2)), <span class="kw">rep</span>(<span class="st">&quot;Green&quot;</span>, <span class="kw">length</span>(V2)))</span>
<span id="cb179-9"><a href="erpgr-subtyping.html#cb179-9"></a>ColSide=<span class="kw">c</span>(ColE[V2],ColSideColB[V2] )</span>
<span id="cb179-10"><a href="erpgr-subtyping.html#cb179-10"></a>ColSide=Cdata<span class="op">$</span>Tumor.Growth[<span class="kw">match</span>(<span class="kw">colnames</span>(Tx1), Cdata<span class="op">$</span>NewID)]</span>
<span id="cb179-11"><a href="erpgr-subtyping.html#cb179-11"></a>ColSide=<span class="kw">ifelse</span>(ColSide<span class="op">==</span><span class="st">&quot;growing&quot;</span>, <span class="st">&quot;orange&quot;</span>, <span class="st">&quot;forestgreen&quot;</span>)</span>
<span id="cb179-12"><a href="erpgr-subtyping.html#cb179-12"></a></span>
<span id="cb179-13"><a href="erpgr-subtyping.html#cb179-13"></a><span class="kw">heatmap.2</span>(Tx1, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span>ColSide,<span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Ep samples&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave)</span>
<span id="cb179-14"><a href="erpgr-subtyping.html#cb179-14"></a></span>
<span id="cb179-15"><a href="erpgr-subtyping.html#cb179-15"></a><span class="kw">heatmap.2</span>(<span class="kw">cor</span>((Tx1)), <span class="dt">trace =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">symm =</span> F, <span class="dt">ColSideColors =</span> ColSide)</span>
<span id="cb179-16"><a href="erpgr-subtyping.html#cb179-16"></a></span>
<span id="cb179-17"><a href="erpgr-subtyping.html#cb179-17"></a></span>
<span id="cb179-18"><a href="erpgr-subtyping.html#cb179-18"></a>axnew=<span class="kw">cor</span>(Tx1)</span>
<span id="cb179-19"><a href="erpgr-subtyping.html#cb179-19"></a>t2=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(axnew), <span class="cf">function</span>(x) axnew[x, <span class="kw">which</span>(<span class="kw">colnames</span>(axnew)<span class="op">==</span><span class="kw">rownames</span>(axnew)[x]) ])</span>
<span id="cb179-20"><a href="erpgr-subtyping.html#cb179-20"></a><span class="kw">colnames</span>(t2)=<span class="kw">colnames</span>(axnew)</span>
<span id="cb179-21"><a href="erpgr-subtyping.html#cb179-21"></a></span>
<span id="cb179-22"><a href="erpgr-subtyping.html#cb179-22"></a><span class="kw">boxplot</span>(t2[<span class="dv">2</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">16</span>]<span class="op">~</span>ColSide[<span class="dv">1</span><span class="op">:</span><span class="dv">16</span>])</span></code></pre></div>
<p>Do a heatmap with the Epithelial samples</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="erpgr-subtyping.html#cb180-1"></a></span>
<span id="cb180-2"><a href="erpgr-subtyping.html#cb180-2"></a>vsdProg=vsd[, <span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">!=</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb180-3"><a href="erpgr-subtyping.html#cb180-3"></a></span>
<span id="cb180-4"><a href="erpgr-subtyping.html#cb180-4"></a>CNamesList=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">ncol</span>(vsdProg))</span>
<span id="cb180-5"><a href="erpgr-subtyping.html#cb180-5"></a>CNamesList[<span class="kw">match</span>(<span class="kw">rownames</span>(CTablea), <span class="kw">colnames</span>(vsdProg))]=CTablea[ ,<span class="dv">1</span>]</span>
<span id="cb180-6"><a href="erpgr-subtyping.html#cb180-6"></a>CNamesList[<span class="kw">match</span>(<span class="kw">rownames</span>(CTabled), <span class="kw">colnames</span>(vsdProg))]=CTabled[ ,<span class="dv">1</span>]</span>
<span id="cb180-7"><a href="erpgr-subtyping.html#cb180-7"></a>CNamesList2=<span class="kw">ifelse</span>(CNamesList<span class="op">==</span><span class="st">&quot;Lum&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>)</span>
<span id="cb180-8"><a href="erpgr-subtyping.html#cb180-8"></a></span>
<span id="cb180-9"><a href="erpgr-subtyping.html#cb180-9"></a>X2=<span class="kw">rep</span>(<span class="st">&quot;darkgreen&quot;</span>, <span class="kw">ncol</span>(vsdProg))</span>
<span id="cb180-10"><a href="erpgr-subtyping.html#cb180-10"></a>X2[<span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, <span class="kw">colnames</span>(vsdProg))]=<span class="st">&quot;pink&quot;</span></span>
<span id="cb180-11"><a href="erpgr-subtyping.html#cb180-11"></a></span>
<span id="cb180-12"><a href="erpgr-subtyping.html#cb180-12"></a>X3=vsdProg<span class="op">$</span>Growth</span>
<span id="cb180-13"><a href="erpgr-subtyping.html#cb180-13"></a>X3=<span class="kw">gsub</span>(<span class="st">&quot;growing&quot;</span>, <span class="st">&quot;orange&quot;</span>, X3)</span>
<span id="cb180-14"><a href="erpgr-subtyping.html#cb180-14"></a>X3=<span class="kw">gsub</span>(<span class="st">&quot;stable&quot;</span>, <span class="st">&quot;forestgreen&quot;</span>, X3)</span>
<span id="cb180-15"><a href="erpgr-subtyping.html#cb180-15"></a></span>
<span id="cb180-16"><a href="erpgr-subtyping.html#cb180-16"></a><span class="co">#pdf(&quot;~/Desktop/5X-DN+Ep-HR-status.pdf&quot;, width=6, height=5)</span></span>
<span id="cb180-17"><a href="erpgr-subtyping.html#cb180-17"></a></span>
<span id="cb180-18"><a href="erpgr-subtyping.html#cb180-18"></a>Ax1=<span class="kw">assay</span>(vsdProg)[<span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vsdProg))), ]</span>
<span id="cb180-19"><a href="erpgr-subtyping.html#cb180-19"></a><span class="kw">heatmap.plus</span>(Ax1, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span><span class="kw">cbind</span>(CNamesList2, X2, X3),<span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">main=</span><span class="st">&quot;DN samples vsd marker list1&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb180-20"><a href="erpgr-subtyping.html#cb180-20"></a></span>
<span id="cb180-21"><a href="erpgr-subtyping.html#cb180-21"></a><span class="kw">heatmap.2</span>(Ax1, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span>CNamesList2,<span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Ep+DN markers&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb180-22"><a href="erpgr-subtyping.html#cb180-22"></a></span>
<span id="cb180-23"><a href="erpgr-subtyping.html#cb180-23"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb180-24"><a href="erpgr-subtyping.html#cb180-24"></a><span class="kw">boxplot</span>(Ax1[<span class="st">&quot;Pgr&quot;</span>, ]<span class="op">~</span>vsdProg<span class="op">$</span>Growth<span class="op">+</span>vsdProg<span class="op">$</span>Fraction, <span class="dt">ylab=</span><span class="st">&quot;Pgr&quot;</span>)</span>
<span id="cb180-25"><a href="erpgr-subtyping.html#cb180-25"></a><span class="kw">boxplot</span>(Ax1[<span class="st">&quot;Foxa1&quot;</span>, ]<span class="op">~</span>vsdProg<span class="op">$</span>Growth<span class="op">+</span>vsdProg<span class="op">$</span>Fraction, <span class="dt">ylab=</span><span class="st">&quot;Foxa1&quot;</span>)</span>
<span id="cb180-26"><a href="erpgr-subtyping.html#cb180-26"></a><span class="kw">boxplot</span>(Ax1[<span class="st">&quot;Gata3&quot;</span>, ]<span class="op">~</span>vsdProg<span class="op">$</span>Growth<span class="op">+</span>vsdProg<span class="op">$</span>Fraction, <span class="dt">ylab=</span><span class="st">&quot;Gata3&quot;</span>)</span>
<span id="cb180-27"><a href="erpgr-subtyping.html#cb180-27"></a><span class="kw">boxplot</span>(Ax1[<span class="st">&quot;Esr1&quot;</span>, ]<span class="op">~</span>vsdProg<span class="op">$</span>Growth<span class="op">+</span>vsdProg<span class="op">$</span>Fraction, <span class="dt">ylab=</span><span class="st">&quot;Esr1&quot;</span>)</span>
<span id="cb180-28"><a href="erpgr-subtyping.html#cb180-28"></a></span>
<span id="cb180-29"><a href="erpgr-subtyping.html#cb180-29"></a>Mx1=<span class="kw">table</span>(vsdProg<span class="op">$</span>Growth, CNamesList2, vsdProg<span class="op">$</span>Fraction)</span>
<span id="cb180-30"><a href="erpgr-subtyping.html#cb180-30"></a></span>
<span id="cb180-31"><a href="erpgr-subtyping.html#cb180-31"></a><span class="kw">ContTable</span>(Mx1[ , , <span class="dv">3</span>], <span class="st">&quot;HR-status-growth-Epithelial&quot;</span>, T,  <span class="st">&quot;Hr-status&quot;</span>,<span class="st">&quot;growth&quot;</span>)</span>
<span id="cb180-32"><a href="erpgr-subtyping.html#cb180-32"></a></span>
<span id="cb180-33"><a href="erpgr-subtyping.html#cb180-33"></a><span class="co">#dev.off()</span></span></code></pre></div>
</div>
<div id="comparison-with-staining" class="section level2">
<h2><span class="header-section-number">10.5</span> Comparison with staining</h2>
<p>We can collate these different scores and assess whether there are similarities with PgR and ER staining:</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="erpgr-subtyping.html#cb181-1"></a><span class="co">## match to infoTable here</span></span>
<span id="cb181-2"><a href="erpgr-subtyping.html#cb181-2"></a></span>
<span id="cb181-3"><a href="erpgr-subtyping.html#cb181-3"></a><span class="co"># Contingency Tables</span></span>
<span id="cb181-4"><a href="erpgr-subtyping.html#cb181-4"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb181-5"><a href="erpgr-subtyping.html#cb181-5"></a>a1=<span class="kw">table</span>(Cdata<span class="op">$</span>HR_status, Cdata<span class="op">$</span>PgR.IF)</span>
<span id="cb181-6"><a href="erpgr-subtyping.html#cb181-6"></a><span class="kw">ContTable</span>(a1[ ,<span class="op">-</span><span class="dv">2</span>], <span class="st">&quot;Pgr Expr&quot;</span>, T, <span class="st">&quot;Pgr IF&quot;</span>, <span class="st">&quot;compressed subtype&quot;</span>)</span>
<span id="cb181-7"><a href="erpgr-subtyping.html#cb181-7"></a></span>
<span id="cb181-8"><a href="erpgr-subtyping.html#cb181-8"></a><span class="co"># Contingency Tables</span></span>
<span id="cb181-9"><a href="erpgr-subtyping.html#cb181-9"></a>a1=<span class="kw">table</span>(Cdata<span class="op">$</span>HR_status, Cdata<span class="op">$</span>ER.IF)</span>
<span id="cb181-10"><a href="erpgr-subtyping.html#cb181-10"></a><span class="kw">ContTable</span>(a1[ ,<span class="op">-</span><span class="dv">2</span>], <span class="st">&quot;ER Expr&quot;</span>, T, <span class="st">&quot;ER IF&quot;</span>, <span class="st">&quot;compressed subtype&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-71-1.png" width="672" /></p>
</div>
<div id="summary-of-expression-markers-for-each-subtypecell-fraction" class="section level2">
<h2><span class="header-section-number">10.6</span> Summary of expression markers for each subtype/cell fraction</h2>
<p>We can pull out the main markers of interest: ER, PGR, Erbb2 and Ki67 (or PCNA).
We also compare these expression distribution of these markers to CD45, DN and Ep overall:</p>
<p>Notes:</p>
<ul>
<li>Pgr highest in luminal samples</li>
<li>Esr1 dynamic range is lower</li>
<li>Ar is higher in basal samples</li>
<li>Pcna is higher in Basal</li>
</ul>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="erpgr-subtyping.html#cb182-1"></a><span class="co"># midx=match(ExpressedaprilChar$X, colnames(allTPMFinal))</span></span>
<span id="cb182-2"><a href="erpgr-subtyping.html#cb182-2"></a><span class="co"># midx3=match(Pall$X[grep(&quot;DN&quot;,Pall$X)], colnames(allTPMFinal))</span></span>
<span id="cb182-3"><a href="erpgr-subtyping.html#cb182-3"></a><span class="co"># midx2=match(ExpressedaprilProg2$X, colnames(allTPMFinal))</span></span>
<span id="cb182-4"><a href="erpgr-subtyping.html#cb182-4"></a><span class="co"># </span></span>
<span id="cb182-5"><a href="erpgr-subtyping.html#cb182-5"></a><span class="co"># summ1=table(ExpressedaprilChar$Call)</span></span>
<span id="cb182-6"><a href="erpgr-subtyping.html#cb182-6"></a><span class="co"># summ2=table(ExpressedaprilProg2$Call)</span></span>
<span id="cb182-7"><a href="erpgr-subtyping.html#cb182-7"></a><span class="co"># summ3=table(Pall$Call[grep(&quot;DN&quot;,Pall$X)])</span></span>
<span id="cb182-8"><a href="erpgr-subtyping.html#cb182-8"></a></span>
<span id="cb182-9"><a href="erpgr-subtyping.html#cb182-9"></a></span>
<span id="cb182-10"><a href="erpgr-subtyping.html#cb182-10"></a>gList=<span class="kw">c</span>(<span class="st">&quot;Esr1&quot;</span>, <span class="st">&quot;Pgr&quot;</span>, <span class="st">&quot;Erbb2&quot;</span>, <span class="st">&quot;Ar&quot;</span>, <span class="st">&quot;Pcna&quot;</span>, <span class="st">&quot;Mki67&quot;</span>, <span class="st">&quot;Acta2&quot;</span>, <span class="st">&quot;Epcam&quot;</span>, <span class="st">&quot;Krt18&quot;</span>)</span>
<span id="cb182-11"><a href="erpgr-subtyping.html#cb182-11"></a></span>
<span id="cb182-12"><a href="erpgr-subtyping.html#cb182-12"></a><span class="co"># par(mfrow=c(2,3))</span></span>
<span id="cb182-13"><a href="erpgr-subtyping.html#cb182-13"></a><span class="co"># </span></span>
<span id="cb182-14"><a href="erpgr-subtyping.html#cb182-14"></a><span class="co"># for (i in gList){</span></span>
<span id="cb182-15"><a href="erpgr-subtyping.html#cb182-15"></a><span class="co"># boxplot(allTPMFinal[i, midx ]~ExpressedaprilChar$Call, main=&quot;Char&quot;,  names=paste(names(summ1),&quot; N=&quot;,summ1, sep=&quot;&quot;), las=2, ylab=i, xlab=&quot;&quot;)</span></span>
<span id="cb182-16"><a href="erpgr-subtyping.html#cb182-16"></a><span class="co"># boxplot(allTPMFinal[i, midx2 ]~ExpressedaprilProg2$Call, main=&quot;Prog&quot;, names=paste(names(summ2),&quot; N=&quot;,summ2, sep=&quot;&quot;), las=2,ylab=i, xlab=&quot;&quot;)</span></span>
<span id="cb182-17"><a href="erpgr-subtyping.html#cb182-17"></a><span class="co"># boxplot(allTPMFinal[i, midx3 ]~Pall$Call[grep(&quot;DN&quot;,Pall$X)], main=&quot;DN&quot;,  names=paste(names(summ3),&quot; N=&quot;,summ3, sep=&quot;&quot;), ylab=i, las=2, xlab=&quot;&quot;)</span></span>
<span id="cb182-18"><a href="erpgr-subtyping.html#cb182-18"></a><span class="co"># }</span></span>
<span id="cb182-19"><a href="erpgr-subtyping.html#cb182-19"></a><span class="co"># </span></span>
<span id="cb182-20"><a href="erpgr-subtyping.html#cb182-20"></a>summ4=<span class="kw">table</span>(infoTableFinal<span class="op">$</span>Fraction)</span>
<span id="cb182-21"><a href="erpgr-subtyping.html#cb182-21"></a></span>
<span id="cb182-22"><a href="erpgr-subtyping.html#cb182-22"></a><span class="co">#pdf(&quot;~/Desktop/FigS3-plot-gene-of-interest-average-expression.pdf&quot;, height=8, width=8)</span></span>
<span id="cb182-23"><a href="erpgr-subtyping.html#cb182-23"></a></span>
<span id="cb182-24"><a href="erpgr-subtyping.html#cb182-24"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb182-25"><a href="erpgr-subtyping.html#cb182-25"></a><span class="cf">for</span> (i <span class="cf">in</span> gList){</span>
<span id="cb182-26"><a href="erpgr-subtyping.html#cb182-26"></a><span class="kw">boxplot</span>(allTPMFinal[i, ]<span class="op">~</span>infoTableFinal<span class="op">$</span>Fraction, <span class="dt">main=</span>i,  <span class="dt">names=</span><span class="kw">paste</span>(<span class="kw">names</span>(summ4),<span class="st">&quot; N=&quot;</span>,summ4, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">ylab=</span>i, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb182-27"><a href="erpgr-subtyping.html#cb182-27"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-72-1.png" width="672" /></p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="erpgr-subtyping.html#cb183-1"></a></span>
<span id="cb183-2"><a href="erpgr-subtyping.html#cb183-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-72-2.png" width="672" /></p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="erpgr-subtyping.html#cb184-1"></a>Based on the above calls, we can condense the calls<span class="op">:</span></span>
<span id="cb184-2"><a href="erpgr-subtyping.html#cb184-2"></a></span>
<span id="cb184-3"><a href="erpgr-subtyping.html#cb184-3"></a><span class="er">*</span><span class="st"> </span>Her2, LumB are sumilar and can be combined with LumA</span>
<span id="cb184-4"><a href="erpgr-subtyping.html#cb184-4"></a><span class="op">*</span><span class="st"> </span>Normal and Basal are most likely most similar</span>
<span id="cb184-5"><a href="erpgr-subtyping.html#cb184-5"></a></span>
<span id="cb184-6"><a href="erpgr-subtyping.html#cb184-6"></a>Pall<span class="op">$</span>Call2=Pall<span class="op">$</span>Call</span>
<span id="cb184-7"><a href="erpgr-subtyping.html#cb184-7"></a>Pall<span class="op">$</span>Call2[<span class="kw">which</span>(Pall<span class="op">$</span>Call2<span class="op">==</span><span class="st">&quot;Her2&quot;</span>)]=<span class="st">&quot;LumB&quot;</span></span>
<span id="cb184-8"><a href="erpgr-subtyping.html#cb184-8"></a></span>
<span id="cb184-9"><a href="erpgr-subtyping.html#cb184-9"></a>Pall<span class="op">$</span>Call3=<span class="st">&quot;Basal&quot;</span></span>
<span id="cb184-10"><a href="erpgr-subtyping.html#cb184-10"></a>Pall<span class="op">$</span>Call3[<span class="kw">grep</span>(<span class="st">&quot;Lum&quot;</span>, Pall<span class="op">$</span>Call2)]=<span class="st">&quot;Lum&quot;</span></span>
<span id="cb184-11"><a href="erpgr-subtyping.html#cb184-11"></a></span>
<span id="cb184-12"><a href="erpgr-subtyping.html#cb184-12"></a>BLcall=<span class="kw">ifelse</span>(Pall<span class="op">$</span>Call3<span class="op">==</span><span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>)</span>
<span id="cb184-13"><a href="erpgr-subtyping.html#cb184-13"></a></span>
<span id="cb184-14"><a href="erpgr-subtyping.html#cb184-14"></a>midx4=<span class="kw">match</span>(Pall<span class="op">$</span>X, <span class="kw">colnames</span>(allTPMFinal))</span>
<span id="cb184-15"><a href="erpgr-subtyping.html#cb184-15"></a>Ax4=<span class="kw">assay</span>(vsd)[<span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vsd))),<span class="kw">na.omit</span>(midx4)]</span>
<span id="cb184-16"><a href="erpgr-subtyping.html#cb184-16"></a>Ax4b=allRSEMnorm2log[<span class="kw">na.omit</span>(<span class="kw">match</span>(<span class="kw">toupper</span>(Markers2), <span class="kw">rownames</span>(allRSEMnorm2log))), <span class="kw">na.omit</span>(midx4)]</span>
<span id="cb184-17"><a href="erpgr-subtyping.html#cb184-17"></a></span>
<span id="cb184-18"><a href="erpgr-subtyping.html#cb184-18"></a><span class="kw">heatmap.2</span>(Ax4, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span> BLcall, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;vsd marker list3&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb184-19"><a href="erpgr-subtyping.html#cb184-19"></a><span class="kw">heatmap.2</span>(Ax4b, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span> BLcall, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;RSEM marker3&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="erpgr-subtyping.html#cb185-1"></a><span class="co">## Clustering based on random genes</span></span>
<span id="cb185-2"><a href="erpgr-subtyping.html#cb185-2"></a></span>
<span id="cb185-3"><a href="erpgr-subtyping.html#cb185-3"></a>In this section we randomly select the most variable <span class="dv">200</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span> genes<span class="op">:</span><span class="st"> </span>we use both the standard deviation and coefficient of <span class="kw">variation</span> (sd normalised by the mean) to select these most variable genes.</span>
<span id="cb185-4"><a href="erpgr-subtyping.html#cb185-4"></a></span>
<span id="cb185-5"><a href="erpgr-subtyping.html#cb185-5"></a>Below is a correlation plot which visualises the concordance between the different methods.</span>
<span id="cb185-6"><a href="erpgr-subtyping.html#cb185-6"></a></span>
<span id="cb185-7"><a href="erpgr-subtyping.html#cb185-7"></a></span>
<span id="cb185-8"><a href="erpgr-subtyping.html#cb185-8"></a>ProgCoh=<span class="kw">assay</span>(vsd)[ ,<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;Ep&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb185-9"><a href="erpgr-subtyping.html#cb185-9"></a>CharCoh=<span class="kw">assay</span>(vsd)[ ,<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;Ep&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">!=</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb185-10"><a href="erpgr-subtyping.html#cb185-10"></a></span>
<span id="cb185-11"><a href="erpgr-subtyping.html#cb185-11"></a>GeneVar1=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(ProgCoh), <span class="cf">function</span>(x) <span class="kw">sd</span>(ProgCoh[x, ]))</span>
<span id="cb185-12"><a href="erpgr-subtyping.html#cb185-12"></a>GeneVar2=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(ProgCoh), <span class="cf">function</span>(x) <span class="kw">sd</span>(ProgCoh[x, ])<span class="op">/</span><span class="kw">mean</span>(ProgCoh[x, ]))</span>
<span id="cb185-13"><a href="erpgr-subtyping.html#cb185-13"></a><span class="kw">names</span>(GeneVar2)=<span class="kw">rownames</span>(ProgCoh)</span>
<span id="cb185-14"><a href="erpgr-subtyping.html#cb185-14"></a><span class="kw">names</span>(GeneVar1)=<span class="kw">rownames</span>(ProgCoh)</span>
<span id="cb185-15"><a href="erpgr-subtyping.html#cb185-15"></a>OrdGene1=<span class="kw">sort</span>(GeneVar1, <span class="dt">decreasing =</span> T)</span>
<span id="cb185-16"><a href="erpgr-subtyping.html#cb185-16"></a>OrdGene2=<span class="kw">sort</span>(GeneVar2, <span class="dt">decreasing =</span> T)</span>
<span id="cb185-17"><a href="erpgr-subtyping.html#cb185-17"></a></span>
<span id="cb185-18"><a href="erpgr-subtyping.html#cb185-18"></a>xsearch=<span class="kw">c</span>(<span class="dv">200</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>)</span>
<span id="cb185-19"><a href="erpgr-subtyping.html#cb185-19"></a></span>
<span id="cb185-20"><a href="erpgr-subtyping.html#cb185-20"></a>CallsTableProg=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">ncol=</span><span class="kw">ncol</span>(ProgCoh), <span class="dt">nrow=</span><span class="dv">10</span>)</span>
<span id="cb185-21"><a href="erpgr-subtyping.html#cb185-21"></a><span class="kw">colnames</span>(CallsTableProg)=<span class="kw">colnames</span>(ProgCoh)</span>
<span id="cb185-22"><a href="erpgr-subtyping.html#cb185-22"></a>CallsTableProg[<span class="dv">1</span>, ]=Pall<span class="op">$</span>Call3[<span class="kw">match</span>(<span class="kw">colnames</span>(CallsTableProg), Pall<span class="op">$</span>X)]</span>
<span id="cb185-23"><a href="erpgr-subtyping.html#cb185-23"></a><span class="co">#CallsTableProg[2, ]=Pall$Call2[match(colnames(CallsTableProg), Pall$X)]</span></span>
<span id="cb185-24"><a href="erpgr-subtyping.html#cb185-24"></a><span class="co">#CallsTableProg[2, ]=Pall$PamCallLong[match(colnames(CallsTableProg), Pall$X)]</span></span>
<span id="cb185-25"><a href="erpgr-subtyping.html#cb185-25"></a>CallsTableProg[<span class="dv">2</span>, ]=Pall<span class="op">$</span>PamCallshort[<span class="kw">match</span>(<span class="kw">colnames</span>(CallsTableProg), Pall<span class="op">$</span>X)]</span>
<span id="cb185-26"><a href="erpgr-subtyping.html#cb185-26"></a></span>
<span id="cb185-27"><a href="erpgr-subtyping.html#cb185-27"></a>count=<span class="dv">3</span></span>
<span id="cb185-28"><a href="erpgr-subtyping.html#cb185-28"></a></span>
<span id="cb185-29"><a href="erpgr-subtyping.html#cb185-29"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb185-30"><a href="erpgr-subtyping.html#cb185-30"></a></span>
<span id="cb185-31"><a href="erpgr-subtyping.html#cb185-31"></a><span class="cf">for</span> (i <span class="cf">in</span> xsearch){</span>
<span id="cb185-32"><a href="erpgr-subtyping.html#cb185-32"></a> <span class="co"># Int1=venn(list(SD=names(OrdGene1)[1:i], CoV=names(OrdGene2)[1:i]))</span></span>
<span id="cb185-33"><a href="erpgr-subtyping.html#cb185-33"></a>  ax1=<span class="kw">pam</span>(<span class="kw">t</span>(ProgCoh[<span class="kw">names</span>(OrdGene1)[<span class="dv">1</span><span class="op">:</span>i],  ]), <span class="dv">2</span>)</span>
<span id="cb185-34"><a href="erpgr-subtyping.html#cb185-34"></a>  bx1=<span class="kw">pam</span>(<span class="kw">t</span>(ProgCoh[<span class="kw">names</span>(OrdGene2)[<span class="dv">1</span><span class="op">:</span>i],  ]), <span class="dv">2</span>)</span>
<span id="cb185-35"><a href="erpgr-subtyping.html#cb185-35"></a>  CallsTableProg[count, ]=ax1<span class="op">$</span>clustering</span>
<span id="cb185-36"><a href="erpgr-subtyping.html#cb185-36"></a>  CallsTableProg[count<span class="op">+</span><span class="dv">1</span>, ]=bx1<span class="op">$</span>clustering</span>
<span id="cb185-37"><a href="erpgr-subtyping.html#cb185-37"></a>  count=count<span class="op">+</span><span class="dv">2</span></span>
<span id="cb185-38"><a href="erpgr-subtyping.html#cb185-38"></a>}</span>
<span id="cb185-39"><a href="erpgr-subtyping.html#cb185-39"></a></span>
<span id="cb185-40"><a href="erpgr-subtyping.html#cb185-40"></a>ConcordanceMat=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="kw">nrow</span>(CallsTableProg), <span class="dt">ncol=</span><span class="kw">nrow</span>(CallsTableProg))</span>
<span id="cb185-41"><a href="erpgr-subtyping.html#cb185-41"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(CallsTableProg)){</span>
<span id="cb185-42"><a href="erpgr-subtyping.html#cb185-42"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(CallsTableProg)){</span>
<span id="cb185-43"><a href="erpgr-subtyping.html#cb185-43"></a>    a1=<span class="kw">table</span>(CallsTableProg[i, ], CallsTableProg[j, ])</span>
<span id="cb185-44"><a href="erpgr-subtyping.html#cb185-44"></a>    ConcordanceMat[i, j]=<span class="kw">sum</span>(<span class="kw">diag</span>(a1))<span class="op">/</span><span class="kw">sum</span>(a1)</span>
<span id="cb185-45"><a href="erpgr-subtyping.html#cb185-45"></a>  }</span>
<span id="cb185-46"><a href="erpgr-subtyping.html#cb185-46"></a>}</span>
<span id="cb185-47"><a href="erpgr-subtyping.html#cb185-47"></a></span>
<span id="cb185-48"><a href="erpgr-subtyping.html#cb185-48"></a><span class="kw">colnames</span>(ConcordanceMat)=<span class="kw">c</span>(<span class="st">&quot;P50&quot;</span>, <span class="st">&quot;PamShort&quot;</span>, <span class="kw">paste</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;sd&quot;</span>, <span class="st">&quot;cov&quot;</span>), <span class="dv">4</span>), <span class="kw">rep</span>(xsearch, <span class="dt">each=</span><span class="dv">2</span>), <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>))</span>
<span id="cb185-49"><a href="erpgr-subtyping.html#cb185-49"></a><span class="kw">rownames</span>(ConcordanceMat)=<span class="kw">colnames</span>(ConcordanceMat)</span>
<span id="cb185-50"><a href="erpgr-subtyping.html#cb185-50"></a>ConcordanceMat2=ConcordanceMat</span>
<span id="cb185-51"><a href="erpgr-subtyping.html#cb185-51"></a>ConcordanceMat2[<span class="kw">which</span>(ConcordanceMat2<span class="op">&lt;</span><span class="fl">0.5</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">1</span><span class="op">-</span>ConcordanceMat2[<span class="kw">which</span>(ConcordanceMat2<span class="op">&lt;</span><span class="fl">0.5</span>, <span class="dt">arr.ind=</span>T)]</span>
<span id="cb185-52"><a href="erpgr-subtyping.html#cb185-52"></a></span>
<span id="cb185-53"><a href="erpgr-subtyping.html#cb185-53"></a><span class="co"># Example heatmap</span></span>
<span id="cb185-54"><a href="erpgr-subtyping.html#cb185-54"></a>Ax1=<span class="kw">assay</span>(vstEp)[ <span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vstEp))),]</span>
<span id="cb185-55"><a href="erpgr-subtyping.html#cb185-55"></a></span>
<span id="cb185-56"><a href="erpgr-subtyping.html#cb185-56"></a>TabOutput=<span class="kw">cbind</span>(<span class="kw">ifelse</span>(CallsTableProg[<span class="dv">4</span>, ]<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),<span class="kw">ifelse</span>(CallsTableProg[<span class="dv">5</span>, ]<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>))</span>
<span id="cb185-57"><a href="erpgr-subtyping.html#cb185-57"></a>x1=<span class="kw">match</span>(<span class="kw">colnames</span>(Ax1), <span class="kw">colnames</span>(CallsTableProg))</span>
<span id="cb185-58"><a href="erpgr-subtyping.html#cb185-58"></a></span>
<span id="cb185-59"><a href="erpgr-subtyping.html#cb185-59"></a><span class="kw">heatmap.plus</span>(Ax1, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span>(TabOutput[ x1,]),<span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Progression cohort vsd marker list1&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="erpgr-subtyping.html#cb186-1"></a></span>
<span id="cb186-2"><a href="erpgr-subtyping.html#cb186-2"></a><span class="co">## Consolidating the different results</span></span>
<span id="cb186-3"><a href="erpgr-subtyping.html#cb186-3"></a></span>
<span id="cb186-4"><a href="erpgr-subtyping.html#cb186-4"></a>Below is a concordance map of the similarity between the different metrics calculated as</span>
<span id="cb186-5"><a href="erpgr-subtyping.html#cb186-5"></a></span>
<span id="cb186-6"><a href="erpgr-subtyping.html#cb186-6"></a><span class="op">$</span><span class="er">$</span><span class="st"> </span>\frac{N_{b,b}<span class="op">+</span>N_{l,l}}{N_{total}} <span class="op">$</span><span class="er">$</span></span>
<span id="cb186-7"><a href="erpgr-subtyping.html#cb186-7"></a></span>
<span id="cb186-8"><a href="erpgr-subtyping.html#cb186-8"></a>where <span class="op">$</span>N_{b,b}<span class="op">$</span><span class="st"> </span>is the number of cases which are basal <span class="cf">in</span> two comparisons and <span class="op">$</span>N_{l,l}<span class="op">$</span><span class="st"> </span>the number of cases which are luminal <span class="cf">in</span> both comparisons.</span>
<span id="cb186-9"><a href="erpgr-subtyping.html#cb186-9"></a></span>
<span id="cb186-10"><a href="erpgr-subtyping.html#cb186-10"></a>Note that the values extend from <span class="fl">0.5</span> to <span class="dv">1</span>, which indicates that any two methods overlap more than just by chance. </span>
<span id="cb186-11"><a href="erpgr-subtyping.html#cb186-11"></a></span>
<span id="cb186-12"><a href="erpgr-subtyping.html#cb186-12"></a></span>
<span id="cb186-13"><a href="erpgr-subtyping.html#cb186-13"></a>There is good overlap between the <span class="cf">in</span><span class="op">-</span>house specific list characterisation, and some overlap between pam50 calling and <span class="cf">in</span><span class="op">-</span>house short list. </span>
<span id="cb186-14"><a href="erpgr-subtyping.html#cb186-14"></a></span>
<span id="cb186-15"><a href="erpgr-subtyping.html#cb186-15"></a></span>
<span id="cb186-16"><a href="erpgr-subtyping.html#cb186-16"></a></span>
<span id="cb186-17"><a href="erpgr-subtyping.html#cb186-17"></a></span>
<span id="cb186-18"><a href="erpgr-subtyping.html#cb186-18"></a><span class="kw">heatmap.2</span>(ConcordanceMat2, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>,<span class="dt">scale=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb186-19"><a href="erpgr-subtyping.html#cb186-19"></a></span>
<span id="cb186-20"><a href="erpgr-subtyping.html#cb186-20"></a><span class="kw">heatmap.2</span>(ConcordanceMat2[<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">7</span>], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>,<span class="dt">scale=</span><span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="erpgr-subtyping.html#cb187-1"></a></span>
<span id="cb187-2"><a href="erpgr-subtyping.html#cb187-2"></a>Below is a summary of the average expression values of the classes defined by the different methods<span class="op">:</span></span>
<span id="cb187-3"><a href="erpgr-subtyping.html#cb187-3"></a></span>
<span id="cb187-4"><a href="erpgr-subtyping.html#cb187-4"></a><span class="er">*</span><span class="st"> </span>The targeted lists show differences <span class="cf">in</span> esr1, pgr, erbb2, pcna</span>
<span id="cb187-5"><a href="erpgr-subtyping.html#cb187-5"></a><span class="op">*</span><span class="st"> </span>Ar has differing directions <span class="cf">for</span> the pam classifications</span>
<span id="cb187-6"><a href="erpgr-subtyping.html#cb187-6"></a><span class="op">*</span><span class="st"> </span>Random genes selected show differences <span class="cf">in</span> pgr, erbb2, Ar and perhaps Mki67</span>
<span id="cb187-7"><a href="erpgr-subtyping.html#cb187-7"></a></span>
<span id="cb187-8"><a href="erpgr-subtyping.html#cb187-8"></a></span>
<span id="cb187-9"><a href="erpgr-subtyping.html#cb187-9"></a></span>
<span id="cb187-10"><a href="erpgr-subtyping.html#cb187-10"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb187-11"><a href="erpgr-subtyping.html#cb187-11"></a></span>
<span id="cb187-12"><a href="erpgr-subtyping.html#cb187-12"></a>compv=<span class="kw">c</span>(<span class="st">&quot;PAM50&quot;</span>,  <span class="st">&quot;pamshort&quot;</span>, <span class="st">&quot;top200sd&quot;</span>, <span class="st">&quot;top500sd&quot;</span>, <span class="st">&quot;top1000sd&quot;</span>, <span class="st">&quot;top2000sd&quot;</span>)</span>
<span id="cb187-13"><a href="erpgr-subtyping.html#cb187-13"></a>gList=<span class="kw">c</span>(gList, <span class="st">&quot;Foxa1&quot;</span>, <span class="st">&quot;Gata3&quot;</span>, <span class="st">&quot;Vim&quot;</span>)</span>
<span id="cb187-14"><a href="erpgr-subtyping.html#cb187-14"></a></span>
<span id="cb187-15"><a href="erpgr-subtyping.html#cb187-15"></a><span class="cf">for</span> (j <span class="cf">in</span> gList){</span>
<span id="cb187-16"><a href="erpgr-subtyping.html#cb187-16"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">5</span>, <span class="dv">7</span>, <span class="dv">9</span>)){</span>
<span id="cb187-17"><a href="erpgr-subtyping.html#cb187-17"></a>  <span class="kw">boxplot</span>(ProgCoh[j, ]<span class="op">~</span>CallsTableProg[i, ], <span class="dt">main=</span><span class="kw">rownames</span>(ConcordanceMat2)[i], <span class="dt">ylab=</span>j, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb187-18"><a href="erpgr-subtyping.html#cb187-18"></a>}}</span>
<span id="cb187-19"><a href="erpgr-subtyping.html#cb187-19"></a><span class="kw">mtext</span>(<span class="st">&quot;ESR1&quot;</span>, <span class="dt">side=</span><span class="dv">2</span>, <span class="dt">outer=</span>T, <span class="dt">line=</span><span class="op">-</span><span class="dv">2</span>)</span></code></pre></div>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="erpgr-subtyping.html#cb188-1"></a></span>
<span id="cb188-2"><a href="erpgr-subtyping.html#cb188-2"></a>We can summarise the outputs <span class="cf">for</span> each sample below</span>
<span id="cb188-3"><a href="erpgr-subtyping.html#cb188-3"></a></span>
<span id="cb188-4"><a href="erpgr-subtyping.html#cb188-4"></a></span>
<span id="cb188-5"><a href="erpgr-subtyping.html#cb188-5"></a>CallsTableProg2=CallsTableProg[<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">5</span>, <span class="dv">7</span>, <span class="dv">9</span>), ]</span>
<span id="cb188-6"><a href="erpgr-subtyping.html#cb188-6"></a><span class="kw">rownames</span>(CallsTableProg2)=compv</span>
<span id="cb188-7"><a href="erpgr-subtyping.html#cb188-7"></a></span>
<span id="cb188-8"><a href="erpgr-subtyping.html#cb188-8"></a><span class="kw">scroll_box</span>(<span class="kw">kable</span>(<span class="kw">t</span>(CallsTableProg2), <span class="dt">format=</span><span class="st">&quot;html&quot;</span>),</span>
<span id="cb188-9"><a href="erpgr-subtyping.html#cb188-9"></a>         <span class="dt">height=</span><span class="st">&quot;300px&quot;</span>, <span class="dt">width=</span><span class="st">&quot;100%&quot;</span>)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="collating-results-and-running-gsea.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="expression-in-specific-pathways.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
