<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 ER/Pgr Subtyping | Rat DCIS study</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘insert study name’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 ER/Pgr Subtyping | Rat DCIS study" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘insert study name’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 ER/Pgr Subtyping | Rat DCIS study" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘insert study name’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2020-11-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="deseq-analysis-1.html"/>
<link rel="next" href="signature-analysis.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.1</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#facs-data"><i class="fa fa-check"></i><b>2.2</b> FACS data</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-table"><i class="fa fa-check"></i><b>2.3</b> Summary Table</a><ul>
<li class="chapter" data-level="2.3.1" data-path="cohort-summary.html"><a href="cohort-summary.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>2.3.1</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="2.3.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>2.3.2</b> Summary of the RNA data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>3</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="3.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>3.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="3.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>3.2</b> Cellular composition</a></li>
<li class="chapter" data-level="3.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>3.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="3.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>3.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="3.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>3.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="3.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>3.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>4</b> Spatial statistics</a><ul>
<li class="chapter" data-level="4.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>4.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="4.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification-treatment-growth"><i class="fa fa-check"></i><b>4.1.1</b> Comparison to manual classification, treatment, growth</a></li>
<li class="chapter" data-level="4.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome"><i class="fa fa-check"></i><b>4.1.2</b> Associations with outcome</a></li>
<li class="chapter" data-level="4.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>4.1.3</b> Growth</a></li>
<li class="chapter" data-level="4.1.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#association-between-distance-and-content"><i class="fa fa-check"></i><b>4.1.4</b> Association between distance and content</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>4.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="4.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>4.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="4.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>4.2.2</b> Growth</a></li>
<li class="chapter" data-level="4.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>4.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>4.3</b> M-H distances</a><ul>
<li class="chapter" data-level="4.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>4.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="4.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>4.3.2</b> Growth</a></li>
<li class="chapter" data-level="4.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>4.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>4.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="4.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#other-cell-types"><i class="fa fa-check"></i><b>4.5</b> Other cell types</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>5</b> Expression data</a><ul>
<li class="chapter" data-level="5.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>5.1</b> Running alignment</a></li>
<li class="chapter" data-level="5.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>5.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="5.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>5.3</b> Normalisation</a><ul>
<li class="chapter" data-level="5.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>5.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>5.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>6</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>6.1</b> PCA plots</a></li>
<li class="chapter" data-level="6.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>6.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="deseq-analysis.html"><a href="deseq-analysis.html"><i class="fa fa-check"></i><b>7</b> DESeq analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="deseq-analysis.html"><a href="deseq-analysis.html#dn-vs-ep"><i class="fa fa-check"></i><b>7.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="7.2" data-path="deseq-analysis.html"><a href="deseq-analysis.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>7.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="7.3" data-path="deseq-analysis.html"><a href="deseq-analysis.html#set-up-cell-type-specifoc-the-comparisons"><i class="fa fa-check"></i><b>7.3</b> Set-up cell-type specifoc the comparisons</a></li>
<li class="chapter" data-level="7.4" data-path="deseq-analysis.html"><a href="deseq-analysis.html#pca-plots-1"><i class="fa fa-check"></i><b>7.4</b> PCA plots</a><ul>
<li class="chapter" data-level="7.4.1" data-path="deseq-analysis.html"><a href="deseq-analysis.html#epcam"><i class="fa fa-check"></i><b>7.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="7.4.2" data-path="deseq-analysis.html"><a href="deseq-analysis.html#cd45"><i class="fa fa-check"></i><b>7.4.2</b> CD45</a></li>
<li class="chapter" data-level="7.4.3" data-path="deseq-analysis.html"><a href="deseq-analysis.html#dn"><i class="fa fa-check"></i><b>7.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="deseq-analysis.html"><a href="deseq-analysis.html#collating-results-and-running-gsea"><i class="fa fa-check"></i><b>7.5</b> Collating results and running GSEA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#summary-of-comparisons"><i class="fa fa-check"></i><b>8.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#summary-of-comaprisons"><i class="fa fa-check"></i><b>8.2</b> Summary of comaprisons</a><ul>
<li class="chapter" data-level="8.2.1" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#epcam-samples"><i class="fa fa-check"></i><b>8.2.1</b> EPCAM samples</a></li>
<li class="chapter" data-level="8.2.2" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#cd45-comparisons"><i class="fa fa-check"></i><b>8.2.2</b> CD45 comparisons</a></li>
<li class="chapter" data-level="8.2.3" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#dn-samples"><i class="fa fa-check"></i><b>8.2.3</b> DN samples</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>8.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="8.3.1" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#dn-fraction"><i class="fa fa-check"></i><b>8.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="8.3.2" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#cd-fraction"><i class="fa fa-check"></i><b>8.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="8.3.3" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#ep-fraction"><i class="fa fa-check"></i><b>8.3.3</b> Ep Fraction</a></li>
<li class="chapter" data-level="8.3.4" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#specific-pathways"><i class="fa fa-check"></i><b>8.3.4</b> Specific pathways</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>9</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="9.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#in-house-classifier"><i class="fa fa-check"></i><b>9.1</b> In-house classifier</a></li>
<li class="chapter" data-level="9.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#pam50-parker-scoring"><i class="fa fa-check"></i><b>9.2</b> PAM50 (Parker) scoring</a></li>
<li class="chapter" data-level="9.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#clustering-based-on-random-genes"><i class="fa fa-check"></i><b>9.3</b> Clustering based on random genes</a></li>
<li class="chapter" data-level="9.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#consolidating-the-different-results"><i class="fa fa-check"></i><b>9.4</b> Consolidating the different results</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>10</b> Signature analysis</a><ul>
<li class="chapter" data-level="10.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>10.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="10.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>10.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="10.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>10.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="10.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>10.1.3</b> MHC presentation proteins</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="signature-analysis.html"><a href="signature-analysis.html#cancer-pathways"><i class="fa fa-check"></i><b>10.2</b> Cancer pathways</a><ul>
<li class="chapter" data-level="10.2.1" data-path="signature-analysis.html"><a href="signature-analysis.html#ssgsea"><i class="fa fa-check"></i><b>10.2.1</b> ssGSEA</a></li>
<li class="chapter" data-level="10.2.2" data-path="signature-analysis.html"><a href="signature-analysis.html#regression-model-for-covariates"><i class="fa fa-check"></i><b>10.2.2</b> regression model for covariates</a></li>
<li class="chapter" data-level="10.2.3" data-path="signature-analysis.html"><a href="signature-analysis.html#taking-into-account-directionality"><i class="fa fa-check"></i><b>10.2.3</b> Taking into account directionality</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>11</b> Immune estimation</a><ul>
<li class="chapter" data-level="11.1" data-path="immune-estimation.html"><a href="immune-estimation.html#ssgsea-1"><i class="fa fa-check"></i><b>11.1</b> ssGSEA</a></li>
<li class="chapter" data-level="11.2" data-path="immune-estimation.html"><a href="immune-estimation.html#deconvolution-methods"><i class="fa fa-check"></i><b>11.2</b> Deconvolution methods</a><ul>
<li class="chapter" data-level="11.2.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>11.2.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="11.2.2" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>11.2.2</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="11.2.3" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>11.2.3</b> Associate with Treatment</a></li>
<li class="chapter" data-level="11.2.4" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>11.2.4</b> Association with Growth</a></li>
<li class="chapter" data-level="11.2.5" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-infiltration"><i class="fa fa-check"></i><b>11.2.5</b> Association with infiltration</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>12</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="12.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>12.1</b> Processing summary</a></li>
<li class="chapter" data-level="12.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>12.2</b> Summary Stats</a></li>
<li class="chapter" data-level="12.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>12.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="12.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>12.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="12.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data"><i class="fa fa-check"></i><b>12.5</b> Associate with clinicopathological data</a></li>
<li class="chapter" data-level="12.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>12.6</b> Associate with signature scores</a></li>
<li class="chapter" data-level="12.7" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#check-whether-the-clonotype-percentages-match-with-gsva-signatures-or-cd20-populations"><i class="fa fa-check"></i><b>12.7</b> Check whether the clonotype percentages match with gsva signatures or CD20 populations</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>13</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>13.1</b> Data</a><ul>
<li class="chapter" data-level="13.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>13.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="13.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>13.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>13.2</b> Plots</a><ul>
<li class="chapter" data-level="13.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>13.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="13.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>13.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>13.3</b> Coding variants</a><ul>
<li class="chapter" data-level="13.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>13.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>13.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="13.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>13.4.1</b> All mutations</a></li>
<li class="chapter" data-level="13.4.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#cosmic-only"><i class="fa fa-check"></i><b>13.4.2</b> Cosmic only:</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Rat DCIS study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="erpgr-subtyping" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> ER/Pgr Subtyping</h1>
<p>In this section, we will look at whether there are differences in rat subtype using the following approaches:</p>
<ul>
<li>Creating our own subtyping approach using specific markers of interest (e.g. using PAM clustering)
<ul>
<li>Using known gene list</li>
<li>Using random genes (top 50, 100, 200 etc variable genes)</li>
</ul></li>
<li>PAM50 subtyping using the methods published by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667820/">Parker et al</a>. This method uses rsem scores to classify samples.</li>
</ul>
<p>As the DN samples do have keratin expression, we will also try to determine the subtypes in these samples too.</p>
<p>In the below analyses, we will conduct this subtyping as follows:</p>
<ul>
<li>progression specific epithelial cohort</li>
<li>characterisation specific epithelial cohort</li>
<li>dn specific cohort (progression)</li>
</ul>
<div id="in-house-classifier" class="section level2">
<h2><span class="header-section-number">9.1</span> In-house classifier</h2>
<p>The selected gene list includes three lists: A. is a mroe comprehensive list containing epithelial, mesenchymal markers and proliferation markers, B. has a more narrow list of subtype specific markers</p>
<ol style="list-style-type: lower-alpha">
<li>‘Ar’, ‘Cd24’, ‘Cdh1’, ‘Foxa1’, ‘Gata3’, ‘Krt8’, ‘Krt18’, ‘Krt5’, ‘Vim’, ‘Erbb2’, ‘Esr1’, ‘Pgr’, ‘Mki67’, ‘Pcna’</li>
<li>‘Ar’, ‘Foxa1’, ‘Gata3’, ‘Erbb2’, ‘Esr1’, ‘Pgr’</li>
</ol>
<p>We will be using variance stabilised counts in this section.</p>
<p>Firstly, we will look at the progression cohort. Note that the overlap between the two gene-sets are similar, and the heatmap is row-scaled.</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-98-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-98-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-98-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-98-4.png" width="672" /></p>
<pre><code>##        ColB
## ColA    Basal Lum
##   Basal    10   3
##   Lum       0   7</code></pre>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-98-5.png" width="672" /></p>
<p>We can perform a similar analysis to the characterisation cohort:</p>
<pre><code>##        ColD
## ColC    Basal Lum
##   Basal     3   0
##   Lum       0   4</code></pre>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-99-1.png" width="672" /></p>
<p>DN samples:</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-100-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-100-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-100-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-100-4.png" width="672" /></p>
<pre><code>##        ColF
## ColE    Basal Lum
##   Basal     9   7
##   Lum      11   3</code></pre>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-100-5.png" width="672" /></p>
<p>We can collate these different scores and assess whether there are similarities with PgR and ER staining:</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-101-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-101-2.png" width="672" /></p>
<p>Finally, look at whether any of the above samples have similar DN and Ep calls: it appears that this call is random, and the subtype does not dictate growth rate</p>
<pre><code>## [1] &quot;extended list&quot;</code></pre>
<pre><code>## , ,  = growing
## 
##        
##         Basal Lum
##   Basal     2   3
##   Lum       2   3
## 
## , ,  = stable
## 
##        
##         Basal Lum
##   Basal     2   2
##   Lum       1   1</code></pre>
<pre><code>## [1] &quot;narrow list&quot;</code></pre>
<pre><code>## , ,  = growing
## 
##        
##         Basal Lum
##   Basal     3   1
##   Lum       3   3
## 
## , ,  = stable
## 
##        
##         Basal Lum
##   Basal     3   0
##   Lum       1   2</code></pre>
</div>
<div id="pam50-parker-scoring" class="section level2">
<h2><span class="header-section-number">9.2</span> PAM50 (Parker) scoring</h2>
<p>Here, we have used rsem counts in PAM50 subtyping.</p>
<p>Plot all samples together and compare the confidence and scores in different subtypes:
There is overlap between the LumB and Her2 scores, similarly Normal and LumA scores are similar</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-104-1.png" width="672" /></p>
<p>We can pull out the main markers of interest: ER, PGR, Erbb2 and Ki67 (or PCNA).
We also compare these expression distribution of these markers to CD45, DN and Ep overall:</p>
<p>Notes:</p>
<ul>
<li>Pgr highest in luminal samples</li>
<li>Esr1 dynamic range is lower</li>
<li>Ar is higher in basal samples</li>
<li>Pcna is higher in Basal</li>
</ul>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-105-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-105-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-105-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-105-4.png" width="672" /></p>
<p>heatmap based on the previous markers:</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-106-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-106-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-106-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-106-4.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-106-5.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-106-6.png" width="672" /></p>
<p>Based on the above calls, we can condense the calls:</p>
<ul>
<li>Her2, LumB are sumilar and can be combined with LumA</li>
<li>Normal and Basal are most likely most similar</li>
</ul>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-107-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-107-2.png" width="672" /></p>
</div>
<div id="clustering-based-on-random-genes" class="section level2">
<h2><span class="header-section-number">9.3</span> Clustering based on random genes</h2>
<p>In this section we randomly select the most variable 50, 100, 200, 500 genes: we use both the standard deviation and coefficient of variation (sd normalised by the mean) to select these most variable genes.</p>
<p>Below is a correlation plot which visualises the concordance between the different methods.</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-108-1.png" width="672" /></p>
</div>
<div id="consolidating-the-different-results" class="section level2">
<h2><span class="header-section-number">9.4</span> Consolidating the different results</h2>
<p>Below is a concordance map of the similarity between the different metrics calculated as</p>
<p><span class="math display">\[ \frac{N_{b,b}+N_{l,l}}{N_{total}} \]</span></p>
<p>where <span class="math inline">\(N_{b,b}\)</span> is the number of cases which are basal in two comparisons and <span class="math inline">\(N_{l,l}\)</span> the number of cases which are luminal in both comparisons.</p>
<p>Note that the values extend from 0.5 to 1, which indicates that any two methods overlap more than just by chance.</p>
<p>There is good overlap between the in-house specific list characterisation, and some overlap between pam50 calling and in-house short list.</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-109-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-109-2.png" width="672" /></p>
<p>Below is a summary of the average expression values of the classes defined by the different methods:</p>
<ul>
<li>The targeted lists show differences in esr1, pgr, erbb2, pcna</li>
<li>Ar has differing directions for the pam classifications</li>
<li>Random genes selected show differences in pgr, erbb2, Ar and perhaps Mki67</li>
</ul>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-4.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-5.png" width="672" /></p>
<p>We can summarise the outputs for each sample below</p>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:300px; overflow-x: scroll; width:100%; ">
<table>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PAM50
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
pamlong
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
pamshort
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
top50cov
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
top50sd
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
10L_D_Ep
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
10R_BL_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
10R_BU_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
11L_B_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
11N_D_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
11R_D_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
11R_C_Ep
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
12L_D_Ep
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
14N_C_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
14N_D_Ep
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
14R_B_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
15N_C_Ep
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
16L_C_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
17N_D_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2N__Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
3N_B_Ep
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
3R_C_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
6R_B_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
7N_A_Ep
</td>
<td style="text-align:left;">
Lum
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
8L_D_Ep
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
8R_CU_Ep
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="deseq-analysis-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="signature-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
