<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 ER/Pgr Subtyping | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 ER/Pgr Subtyping | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 ER/Pgr Subtyping | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="collating-results-and-running-gsea.html"/>
<link rel="next" href="expression-in-specific-pathways.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redo-the-survival-plots-based-on-luminal-cases-only"><i class="fa fa-check"></i><b>15.4.3</b> Redo the survival plots based on Luminal cases only</a></li>
<li class="chapter" data-level="15.4.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.4</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>16</b> METABRIC: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="16.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="16.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="16.2.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#all-samples-1"><i class="fa fa-check"></i><b>16.2.1</b> All samples</a></li>
<li class="chapter" data-level="16.2.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases-1"><i class="fa fa-check"></i><b>16.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.4" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint-1"><i class="fa fa-check"></i><b>16.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="16.4.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#overall-survival-1"><i class="fa fa-check"></i><b>16.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="16.4.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#met-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>16.4.2</b> MET: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>21</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="21.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>21.1</b> Data</a><ul>
<li class="chapter" data-level="21.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>21.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>21.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="21.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>21.3</b> Plots</a><ul>
<li class="chapter" data-level="21.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>21.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="21.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>21.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>21.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="21.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>21.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>21.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="21.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>21.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>21.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>22</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>22.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>22.2</b> Load files</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>22.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="22.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>22.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="22.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>22.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>23</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>23.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>23.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>23.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>24</b> Trichrome staining</a><ul>
<li class="chapter" data-level="24.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>24.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="24.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>24.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="24.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>24.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="24.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>24.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>25</b> CNV calling</a><ul>
<li class="chapter" data-level="25.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>25.1</b> Sample data</a></li>
<li class="chapter" data-level="25.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>25.2</b> Data Summary</a><ul>
<li class="chapter" data-level="25.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>25.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="25.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>25.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="25.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>25.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>25.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="25.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>25.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>26</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="erpgr-subtyping" class="section level1">
<h1><span class="header-section-number">Chapter 10</span> ER/Pgr Subtyping</h1>
<p>In this section, we will look at whether there are differences in rat subtype by:</p>
<ul>
<li>creating a list of specific markers of interest</li>
<li>applying PAM clsutering (partitioning around medoids)</li>
</ul>
<p>As the DN samples do have keratin expression, we will also try to determine the subtypes in these samples too.</p>
<p>In the below analyses, we will conduct this subtyping as follows:</p>
<ul>
<li>progression specific epithelial cohort</li>
<li>characterisation specific epithelial cohort</li>
<li>dn specific cohort (progression)</li>
</ul>
<div id="gene-list" class="section level2">
<h2><span class="header-section-number">10.1</span> Gene List</h2>
<p>We will assess clustering using two different lists. A. is a more comprehensive list containing epithelial, mesenchymal markers and proliferation markers, B. has a more narrow list of subtype specific markers</p>
<ol style="list-style-type: lower-alpha">
<li>‘Ar’, ‘Cd24’, ‘Cdh1’, ‘Foxa1’, ‘Gata3’, ‘Krt8’, ‘Krt18’, ‘Krt5’, ‘Vim’, ‘Erbb2’, ‘Esr1’, ‘Pgr’, ‘Mki67’, ‘Pcna’</li>
<li>‘Ar’, ‘Foxa1’, ‘Gata3’, ‘Erbb2’, ‘Esr1’, ‘Pgr’</li>
</ol>
</div>
<div id="progression-cohort" class="section level2">
<h2><span class="header-section-number">10.2</span> Progression cohort</h2>
<p>We will be using variance stabilised counts in this section.</p>
<p>Firstly, we will look at the progression cohort. Note that the overlap between the two gene-sets are similar, and the heatmap is row-scaled.</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="erpgr-subtyping.html#cb187-1"></a>Markers1=<span class="kw">c</span>(<span class="st">&#39;Ar&#39;</span>, <span class="st">&#39;Cd24&#39;</span>, <span class="st">&#39;Cdh1&#39;</span>, <span class="st">&#39;Foxa1&#39;</span>, <span class="st">&#39;Gata3&#39;</span>, <span class="st">&#39;Krt8&#39;</span>, <span class="st">&#39;Krt18&#39;</span>, <span class="st">&#39;Krt5&#39;</span>, <span class="st">&#39;Vim&#39;</span>, <span class="st">&#39;Erbb2&#39;</span>, <span class="st">&#39;Esr1&#39;</span>, <span class="st">&#39;Pgr&#39;</span>, <span class="st">&#39;Mki67&#39;</span>, <span class="st">&#39;Pcna&#39;</span>)</span>
<span id="cb187-2"><a href="erpgr-subtyping.html#cb187-2"></a>Markers2=<span class="kw">c</span>(<span class="st">&#39;Ar&#39;</span>, <span class="st">&#39;Foxa1&#39;</span>, <span class="st">&#39;Gata3&#39;</span>, <span class="st">&#39;Erbb2&#39;</span>, <span class="st">&#39;Esr1&#39;</span>, <span class="st">&#39;Pgr&#39;</span>)</span>
<span id="cb187-3"><a href="erpgr-subtyping.html#cb187-3"></a><span class="co"># Progression cohort</span></span>
<span id="cb187-4"><a href="erpgr-subtyping.html#cb187-4"></a>Ax1a=<span class="kw">assay</span>(vstEp)[ <span class="kw">match</span>(Markers1, <span class="kw">rownames</span>(<span class="kw">assay</span>(vstEp))),]</span>
<span id="cb187-5"><a href="erpgr-subtyping.html#cb187-5"></a>Ax2a=<span class="kw">assay</span>(vstEp)[ <span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vstEp))),]</span>
<span id="cb187-6"><a href="erpgr-subtyping.html#cb187-6"></a>Pam1=<span class="kw">pam</span>(<span class="kw">t</span>(Ax1a), <span class="dv">2</span>)</span>
<span id="cb187-7"><a href="erpgr-subtyping.html#cb187-7"></a>Pam2=<span class="kw">pam</span>(<span class="kw">t</span>(Ax2a), <span class="dv">2</span>)</span>
<span id="cb187-8"><a href="erpgr-subtyping.html#cb187-8"></a></span>
<span id="cb187-9"><a href="erpgr-subtyping.html#cb187-9"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb187-10"><a href="erpgr-subtyping.html#cb187-10"></a><span class="kw">plot</span>(Pam1, <span class="dt">main=</span><span class="st">&quot;List A, progression cohort&quot;</span>)</span>
<span id="cb187-11"><a href="erpgr-subtyping.html#cb187-11"></a><span class="kw">plot</span>(Pam2, <span class="dt">main=</span><span class="st">&quot;List B, progression cohort&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-66-1.png" width="672" /></p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="erpgr-subtyping.html#cb188-1"></a></span>
<span id="cb188-2"><a href="erpgr-subtyping.html#cb188-2"></a>ColA=Pam1<span class="op">$</span>clustering</span>
<span id="cb188-3"><a href="erpgr-subtyping.html#cb188-3"></a>ColA=<span class="kw">ifelse</span>(ColA<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span>
<span id="cb188-4"><a href="erpgr-subtyping.html#cb188-4"></a>ColB=Pam2<span class="op">$</span>clustering</span>
<span id="cb188-5"><a href="erpgr-subtyping.html#cb188-5"></a>ColB=<span class="kw">ifelse</span>(ColB<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span></code></pre></div>
<p>The following shows the heatmaps using these list, using average distances in the hclustering approaches: firstly, using the longer list</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="erpgr-subtyping.html#cb189-1"></a><span class="co">#colnames(Ax1a)=infoTableFinal$TumorIDnew[match(colnames(Ax1a), rownames(infoTableFinal))]</span></span>
<span id="cb189-2"><a href="erpgr-subtyping.html#cb189-2"></a></span>
<span id="cb189-3"><a href="erpgr-subtyping.html#cb189-3"></a><span class="co">#hclust.ave &lt;- function(x) hclust(x, method=&quot;average&quot;)</span></span>
<span id="cb189-4"><a href="erpgr-subtyping.html#cb189-4"></a>ax1=<span class="kw">heatmap.2</span>(Ax1a, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;narrow list&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(ColA)])</span></code></pre></div>
<div class="figure"><span id="fig:Ext3i"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext3i-1.png" alt="HR clustering" width="672" />
<p class="caption">
Figure 10.1: HR clustering
</p>
</div>
<p>and now using the shorter list</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="erpgr-subtyping.html#cb190-1"></a>ax1=<span class="kw">heatmap.2</span>(Ax2a, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;narrow list&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(ColB)])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-67-1.png" width="672" /></p>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="erpgr-subtyping.html#cb191-1"></a></span>
<span id="cb191-2"><a href="erpgr-subtyping.html#cb191-2"></a>DT<span class="op">::</span><span class="kw">datatable</span>(ax1<span class="op">$</span>carpet, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb191-3"><a href="erpgr-subtyping.html#cb191-3"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div id="htmlwidget-e7a2b2b66ebfc8b1fd4e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e7a2b2b66ebfc8b1fd4e">{"x":{"filter":"none","extensions":["Buttons"],"data":[[-2.37071985592226,2.70845994610738,-0.30166629861151,-0.755052144771171,-0.884688956343299,-0.922708251861021,-0.476659520185623,-0.657696925293398,-0.111866156592946,0.323735068149399,0.643212226230347,-0.452151412283149,-0.315154229391736,0.177065804651109,0.321985020761525,0.462103425260041,0.977769478333374,0.718380231510046,0.373243904201227,0.542408646051676],[1.37688274942631,-1.54513030238915,1.25983955633338,-0.339321527397364,0.122658649764387,-0.399422524878372,-1.54463625748855,-0.431740995894515,0.778697609288362,-0.17434269738648,-0.513248050379905,1.85683241055702,-0.219189495488363,-0.995595165040859,0.154959802636187,0.0751307220136637,1.1605884602356,1.20528667263219,-1.3498369708681,-0.478412645675442],[-1.5661671882917,-0.347550007638775,-0.00322063272642297,-0.815733684531144,-1.1713360491523,-0.497943166024642,-0.744379384262309,-0.793390400164487,-0.0643654032002818,-0.96948242144453,-1.09319049930129,1.69613194023453,1.57014300646836,0.401039737891359,1.30389116672761,1.54835679217159,0.989323029407433,0.393962137485797,0.269388685580394,-0.105477659229192],[-1.78679496759196,0.0586200599262163,-0.942239034821197,-1.5248731487053,-0.897668697910503,-0.193634517616094,0.0149879528433826,0.200030294256105,-0.246292657473421,-0.549691912884928,-0.58051817200022,0.258088592313993,-0.940156083755917,-0.18492387211281,0.302342012584606,1.39947695574835,1.61659024068283,1.47085603126656,1.2070031158207,1.31879780942961],[-2.48138164526347,1.4247243974121,0.637512572624767,-0.673369621427188,-0.228002069299017,-0.917658877482884,-0.29767936322599,-0.0197867710968111,-0.60789660611979,-0.585364199981966,-0.109492748129565,0.672746184990649,1.48073909690538,0.857620326804339,-0.0698975590509796,0.680450447475287,1.68352444181812,-1.1872672824977,0.114852256597231,-0.374372981052502],[-0.965523400904802,0.99013899141964,0.178741743573361,-1.33320558464341,-1.14615368401537,-1.65613563394803,-1.10298916892654,-0.755458876885742,-0.686528506898349,-0.830145292535395,0.225627689228789,1.0403595225885,1.2459952732309,1.43960900110747,1.12334794032219,1.20294239592519,0.878729246471747,-0.0652948339350463,0.166387988260787,0.0495551905640945]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>Esr1<\/th>\n      <th>Ar<\/th>\n      <th>Gata3<\/th>\n      <th>Pgr<\/th>\n      <th>Erbb2<\/th>\n      <th>Foxa1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[0,1,2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="erpgr-subtyping.html#cb192-1"></a><span class="co">#write.csv(ax1$carpet, file=&quot;nature-tables/Ext3i.csv&quot;)</span></span>
<span id="cb192-2"><a href="erpgr-subtyping.html#cb192-2"></a></span>
<span id="cb192-3"><a href="erpgr-subtyping.html#cb192-3"></a><span class="kw">colnames</span>(Ax2a)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(Ax2a), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb192-4"><a href="erpgr-subtyping.html#cb192-4"></a>Cdata<span class="op">$</span>HR_status=<span class="ot">NA</span></span>
<span id="cb192-5"><a href="erpgr-subtyping.html#cb192-5"></a>Cdata<span class="op">$</span>HR_status[<span class="kw">match</span>(<span class="kw">colnames</span>(Ax2a), Cdata<span class="op">$</span>NewID)]=ColB</span></code></pre></div>
</div>
<div id="charcterisation-cohort" class="section level2">
<h2><span class="header-section-number">10.3</span> Charcterisation cohort</h2>
<p>We can perform a similar analysis to the characterisation cohort: (and save to file)</p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="erpgr-subtyping.html#cb193-1"></a>c2=infoTableFinal<span class="op">$</span>SampleID[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;Ep&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">!=</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb193-2"><a href="erpgr-subtyping.html#cb193-2"></a></span>
<span id="cb193-3"><a href="erpgr-subtyping.html#cb193-3"></a>Ax1=<span class="kw">assay</span>(vsd)[ <span class="kw">match</span>(Markers1, <span class="kw">rownames</span>(<span class="kw">assay</span>(vsd))), <span class="kw">match</span>(c2, <span class="kw">colnames</span>(vsd))]</span>
<span id="cb193-4"><a href="erpgr-subtyping.html#cb193-4"></a>Ax2=<span class="kw">assay</span>(vsd)[ <span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vsd))), <span class="kw">match</span>(c2, <span class="kw">colnames</span>(vsd))]</span>
<span id="cb193-5"><a href="erpgr-subtyping.html#cb193-5"></a></span>
<span id="cb193-6"><a href="erpgr-subtyping.html#cb193-6"></a>Pam3=<span class="kw">pam</span>(<span class="kw">t</span>(Ax1), <span class="dv">2</span>)</span>
<span id="cb193-7"><a href="erpgr-subtyping.html#cb193-7"></a>Pam4=<span class="kw">pam</span>(<span class="kw">t</span>(Ax2), <span class="dv">2</span>)</span>
<span id="cb193-8"><a href="erpgr-subtyping.html#cb193-8"></a></span>
<span id="cb193-9"><a href="erpgr-subtyping.html#cb193-9"></a><span class="co">#plot(Pam3)</span></span>
<span id="cb193-10"><a href="erpgr-subtyping.html#cb193-10"></a><span class="kw">plot</span>(Pam4, <span class="dt">main=</span><span class="st">&quot;narrow list&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-68-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-68-2.png" width="672" /></p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="erpgr-subtyping.html#cb194-1"></a></span>
<span id="cb194-2"><a href="erpgr-subtyping.html#cb194-2"></a>Ax1b=<span class="kw">scale</span>(<span class="kw">t</span>(Ax1))</span>
<span id="cb194-3"><a href="erpgr-subtyping.html#cb194-3"></a></span>
<span id="cb194-4"><a href="erpgr-subtyping.html#cb194-4"></a><span class="co">#plot(Pam3, &quot;extended list&quot;)</span></span>
<span id="cb194-5"><a href="erpgr-subtyping.html#cb194-5"></a>ColC=Pam3<span class="op">$</span>clustering</span>
<span id="cb194-6"><a href="erpgr-subtyping.html#cb194-6"></a>ColC=<span class="kw">ifelse</span>(ColC<span class="op">==</span><span class="dv">2</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span>
<span id="cb194-7"><a href="erpgr-subtyping.html#cb194-7"></a>ColD=Pam4<span class="op">$</span>clustering</span>
<span id="cb194-8"><a href="erpgr-subtyping.html#cb194-8"></a>ColD=<span class="kw">ifelse</span>(ColD<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span>
<span id="cb194-9"><a href="erpgr-subtyping.html#cb194-9"></a></span>
<span id="cb194-10"><a href="erpgr-subtyping.html#cb194-10"></a><span class="kw">colnames</span>(Ax1)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">rownames</span>(Ax1b), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb194-11"><a href="erpgr-subtyping.html#cb194-11"></a><span class="kw">heatmap.2</span>((Ax1), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(ColC)])</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-68-3.png" width="672" /></p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="erpgr-subtyping.html#cb195-1"></a><span class="kw">colnames</span>(Ax2)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">rownames</span>(Ax1b), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb195-2"><a href="erpgr-subtyping.html#cb195-2"></a><span class="kw">heatmap.2</span>((Ax2), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">ColSideColors =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(ColD)])</span></code></pre></div>
<div class="figure"><span id="fig:1f"></span>
<img src="RatDCIS-bookdown_files/figure-html/1f-1.png" alt="HR status in characterisation cohort" width="672" />
<p class="caption">
Figure 10.2: HR status in characterisation cohort
</p>
</div>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="erpgr-subtyping.html#cb196-1"></a></span>
<span id="cb196-2"><a href="erpgr-subtyping.html#cb196-2"></a>DT<span class="op">::</span><span class="kw">datatable</span>(Ax2, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb196-3"><a href="erpgr-subtyping.html#cb196-3"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:1f"></span>
<div id="htmlwidget-2f21399cc02b52d10193" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2f21399cc02b52d10193">{"x":{"filter":"none","extensions":["Buttons"],"data":[[6.48360450117851,7.90963097512506,9.95889593775928,8.44805315855177,5.80015583847111,7.04515663256397],[7.50767408519707,9.25385781431872,10.0904454647446,8.62680246794623,5.70846697731827,7.53312153444781],[6.23626286027911,10.543671121836,10.7585684271846,8.62710452260842,7.18975706038384,8.58778524670972],[7.72141286293054,10.1289445330561,10.8941452075278,9.64713465670414,8.26552232468635,10.0513122360454],[6.6274387035706,8.75050056951538,10.2942428984437,7.40440246639605,6.86739632733047,8.52782130260375],[9.91086356415845,10.5402320250603,10.8080164619882,9.52237392991308,6.39098964004859,8.84104923292186],[8.25103893197021,4.4068174184481,10.0607931765081,8.60561833738462,3.48748816779447,3.48748816779447],[9.25168947573516,5.89822488446483,10.0862825029536,8.66702433577038,3.48748816779447,8.04593927151905],[8.8268542887157,10.2449156480424,10.7677957511341,8.77889585560645,7.49186671968341,11.0370633332014],[7.78023995181119,8.9717309984166,8.87611558820805,8.4168112805742,3.48748816779447,6.76443751619616]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>C2T1<\/th>\n      <th>C3T1<\/th>\n      <th>C5T1<\/th>\n      <th>C9T2<\/th>\n      <th>C7T1<\/th>\n      <th>C9T1<\/th>\n      <th>C10T2<\/th>\n      <th>C10T4<\/th>\n      <th>C11T1<\/th>\n      <th>C17T4<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[0,1,2,3,4,5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 10.2: HR status in characterisation cohort
</p>
</div>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="erpgr-subtyping.html#cb197-1"></a></span>
<span id="cb197-2"><a href="erpgr-subtyping.html#cb197-2"></a><span class="co">#write.csv(Ax2, file=&quot;nature-tables/1f.csv&quot;)</span></span></code></pre></div>
</div>
<div id="dn-samples" class="section level2">
<h2><span class="header-section-number">10.4</span> DN samples</h2>
<p>DN samples: note there is variability in DN expression of PR, Foxa1, Esr1, Ar,
Erbb2 and Gata3 shows fairly stable expression values</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="erpgr-subtyping.html#cb198-1"></a>Ax2=<span class="kw">assay</span>(vsdLimmaDN)[ <span class="kw">match</span>(Markers2, <span class="kw">rownames</span>(<span class="kw">assay</span>(vsdLimmaDN))),]</span>
<span id="cb198-2"><a href="erpgr-subtyping.html#cb198-2"></a>Pam5=<span class="kw">pam</span>(<span class="kw">t</span>(Ax2), <span class="dv">2</span>)</span>
<span id="cb198-3"><a href="erpgr-subtyping.html#cb198-3"></a><span class="kw">plot</span>(Pam5, <span class="dt">main=</span><span class="st">&quot;extended list DN&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-69-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-69-2.png" width="672" /></p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="erpgr-subtyping.html#cb199-1"></a>ColE=Pam5<span class="op">$</span>clustering</span>
<span id="cb199-2"><a href="erpgr-subtyping.html#cb199-2"></a>ColE=<span class="kw">ifelse</span>(ColE<span class="op">==</span><span class="dv">1</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Lum&quot;</span>)</span>
<span id="cb199-3"><a href="erpgr-subtyping.html#cb199-3"></a></span>
<span id="cb199-4"><a href="erpgr-subtyping.html#cb199-4"></a></span>
<span id="cb199-5"><a href="erpgr-subtyping.html#cb199-5"></a>ColE=<span class="kw">gsub</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;red&quot;</span>, ColE)</span>
<span id="cb199-6"><a href="erpgr-subtyping.html#cb199-6"></a>ColE=<span class="kw">gsub</span>(<span class="st">&quot;Lum&quot;</span>, <span class="st">&quot;blue&quot;</span>, ColE)</span>
<span id="cb199-7"><a href="erpgr-subtyping.html#cb199-7"></a></span>
<span id="cb199-8"><a href="erpgr-subtyping.html#cb199-8"></a><span class="co">#pdf(&quot;~/Desktop/5X-DN-samples-HR-status.pdf&quot;, width=6, height=5)</span></span>
<span id="cb199-9"><a href="erpgr-subtyping.html#cb199-9"></a><span class="kw">colnames</span>(Ax2)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(Ax2), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb199-10"><a href="erpgr-subtyping.html#cb199-10"></a><span class="kw">heatmap.2</span>(Ax2, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>)[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">ColSideColors =</span>ColE,<span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, <span class="dt">main=</span><span class="st">&quot;DN samples vsd marker list1&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">hclustfun =</span> hclust.ave)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-69-3.png" width="672" /></p>
</div>
<div id="comparison-with-staining" class="section level2">
<h2><span class="header-section-number">10.5</span> Comparison with staining</h2>
<p>We can collate these different scores and assess whether there are similarities with PgR and ER staining:</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="erpgr-subtyping.html#cb200-1"></a><span class="co">## match to infoTable here</span></span>
<span id="cb200-2"><a href="erpgr-subtyping.html#cb200-2"></a></span>
<span id="cb200-3"><a href="erpgr-subtyping.html#cb200-3"></a><span class="co"># Contingency Tables</span></span>
<span id="cb200-4"><a href="erpgr-subtyping.html#cb200-4"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb200-5"><a href="erpgr-subtyping.html#cb200-5"></a>a1=<span class="kw">table</span>(Cdata<span class="op">$</span>HR_status, Cdata<span class="op">$</span>PgR.IF)</span>
<span id="cb200-6"><a href="erpgr-subtyping.html#cb200-6"></a><span class="kw">ContTable</span>(a1[ ,<span class="op">-</span><span class="dv">2</span>], <span class="st">&quot;Pgr Expr&quot;</span>, T, <span class="st">&quot;Pgr IF&quot;</span>, <span class="st">&quot;compressed subtype&quot;</span>)</span>
<span id="cb200-7"><a href="erpgr-subtyping.html#cb200-7"></a></span>
<span id="cb200-8"><a href="erpgr-subtyping.html#cb200-8"></a><span class="co"># Contingency Tables</span></span>
<span id="cb200-9"><a href="erpgr-subtyping.html#cb200-9"></a>a1=<span class="kw">table</span>(Cdata<span class="op">$</span>HR_status, Cdata<span class="op">$</span>ER.IF)</span>
<span id="cb200-10"><a href="erpgr-subtyping.html#cb200-10"></a><span class="kw">ContTable</span>(a1[ ,<span class="op">-</span><span class="dv">2</span>], <span class="st">&quot;ER Expr&quot;</span>, T, <span class="st">&quot;ER IF&quot;</span>, <span class="st">&quot;compressed subtype&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-70-1.png" width="672" /></p>
</div>
<div id="summary-of-expression-markers-for-each-subtypecell-fraction" class="section level2">
<h2><span class="header-section-number">10.6</span> Summary of expression markers for each subtype/cell fraction</h2>
<p>We can pull out the main markers of interest: ER, PGR, Erbb2 and Ki67 (or PCNA).
We also compare these expression distribution of these markers to CD45, DN and Ep overall:</p>
<p>Notes:</p>
<ul>
<li>Pgr highest in luminal samples</li>
<li>Esr1 dynamic range is lower</li>
<li>Ar is higher in basal samples</li>
<li>Pcna is higher in Basal</li>
</ul>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="erpgr-subtyping.html#cb201-1"></a>gList=<span class="kw">c</span>(<span class="st">&quot;Esr1&quot;</span>, <span class="st">&quot;Pgr&quot;</span>, <span class="st">&quot;Erbb2&quot;</span>, <span class="st">&quot;Ar&quot;</span>, <span class="st">&quot;Pcna&quot;</span>, <span class="st">&quot;Mki67&quot;</span>, <span class="st">&quot;Acta2&quot;</span>, <span class="st">&quot;Epcam&quot;</span>, <span class="st">&quot;Krt18&quot;</span>)</span>
<span id="cb201-2"><a href="erpgr-subtyping.html#cb201-2"></a></span>
<span id="cb201-3"><a href="erpgr-subtyping.html#cb201-3"></a></span>
<span id="cb201-4"><a href="erpgr-subtyping.html#cb201-4"></a>summ4=<span class="kw">table</span>(infoTableFinal<span class="op">$</span>Fraction)</span>
<span id="cb201-5"><a href="erpgr-subtyping.html#cb201-5"></a></span>
<span id="cb201-6"><a href="erpgr-subtyping.html#cb201-6"></a><span class="co">#pdf(&quot;~/Desktop/FigS3-plot-gene-of-interest-average-expression.pdf&quot;, height=8, width=8)</span></span>
<span id="cb201-7"><a href="erpgr-subtyping.html#cb201-7"></a></span>
<span id="cb201-8"><a href="erpgr-subtyping.html#cb201-8"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb201-9"><a href="erpgr-subtyping.html#cb201-9"></a><span class="cf">for</span> (i <span class="cf">in</span> gList){</span>
<span id="cb201-10"><a href="erpgr-subtyping.html#cb201-10"></a><span class="kw">boxplot</span>(allTPMFinal[i, ]<span class="op">~</span>infoTableFinal<span class="op">$</span>Fraction, <span class="dt">main=</span>i,  <span class="dt">names=</span><span class="kw">paste</span>(<span class="kw">names</span>(summ4),<span class="st">&quot; N=&quot;</span>,summ4, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">ylab=</span>i, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb201-11"><a href="erpgr-subtyping.html#cb201-11"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-71-1.png" width="672" /></p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="erpgr-subtyping.html#cb202-1"></a></span>
<span id="cb202-2"><a href="erpgr-subtyping.html#cb202-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-71-2.png" width="672" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="collating-results-and-running-gsea.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="expression-in-specific-pathways.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
