<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Expression data | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Expression data | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Expression data | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-02-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="spatial-statistics.html"/>
<link rel="next" href="rna-data-preliminary-plots.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.2</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.2.1" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.2.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="expression-data" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Expression data</h1>
<p>This file looks at loading and pre-processing data for:</p>
<ul>
<li>differential gene expression analysis</li>
<li>uploading into CIBERSORT/TIMER</li>
</ul>
<div id="running-alignment" class="section level2">
<h2><span class="header-section-number">6.1</span> Running alignment</h2>
<p>Samples were mapped in star using the following parameters. Note that the first two batches of samples run had shorter read lengths (~75 bp) whereas batch 3 had lengths of ~150bp</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="expression-data.html#cb104-1"></a><span class="co">## Not run here</span></span>
<span id="cb104-2"><a href="expression-data.html#cb104-2"></a>STAR \</span>
<span id="cb104-3"><a href="expression-data.html#cb104-3"></a>     <span class="op">--</span>readFilesCommand zcat \</span>
<span id="cb104-4"><a href="expression-data.html#cb104-4"></a>     <span class="op">--</span>genomeDir <span class="op">/</span>n<span class="op">/</span>scratch2<span class="op">/</span>at268<span class="op">/</span>rn6_v2 \</span>
<span id="cb104-5"><a href="expression-data.html#cb104-5"></a>     <span class="op">--</span>sjdbGTFfile <span class="op">/</span>n<span class="op">/</span>scratch2<span class="op">/</span>at268<span class="op">/</span>rn6_v2<span class="op">/</span>rn6.refGene.gtf \</span>
<span id="cb104-6"><a href="expression-data.html#cb104-6"></a>     <span class="op">--</span>runThreadN <span class="dv">10</span> \</span>
<span id="cb104-7"><a href="expression-data.html#cb104-7"></a>     <span class="op">--</span>runMode alignReads \</span>
<span id="cb104-8"><a href="expression-data.html#cb104-8"></a>     <span class="op">--</span>genomeLoad NoSharedMemory\</span>
<span id="cb104-9"><a href="expression-data.html#cb104-9"></a>     <span class="op">--</span>outSAMattributes NH HI AS nM NM\</span>
<span id="cb104-10"><a href="expression-data.html#cb104-10"></a>     <span class="op">--</span>outSAMstrandField intronMotif\</span>
<span id="cb104-11"><a href="expression-data.html#cb104-11"></a>     <span class="op">--</span>outFilterMultimapNmax <span class="dv">20</span>\</span>
<span id="cb104-12"><a href="expression-data.html#cb104-12"></a>     <span class="op">--</span>alignSJoverhangMin <span class="dv">8</span>\</span>
<span id="cb104-13"><a href="expression-data.html#cb104-13"></a>     <span class="op">--</span>readFilesIn <span class="op">$</span><span class="dv">1</span> <span class="op">$</span><span class="dv">2</span> \</span>
<span id="cb104-14"><a href="expression-data.html#cb104-14"></a>     <span class="op">--</span>alignSJDBoverhangMin <span class="dv">1</span>\</span>
<span id="cb104-15"><a href="expression-data.html#cb104-15"></a>     <span class="op">--</span>outFilterMismatchNmax <span class="dv">999</span>\</span>
<span id="cb104-16"><a href="expression-data.html#cb104-16"></a>     <span class="op">--</span>outFilterMismatchNoverLmax <span class="fl">0.1</span>\</span>
<span id="cb104-17"><a href="expression-data.html#cb104-17"></a>     <span class="op">--</span>alignIntronMin <span class="dv">20</span>\</span>
<span id="cb104-18"><a href="expression-data.html#cb104-18"></a>     <span class="op">--</span>alignIntronMax <span class="dv">1000000</span>\</span>
<span id="cb104-19"><a href="expression-data.html#cb104-19"></a>     <span class="op">--</span>alignMatesGapMax <span class="dv">1000000</span>\</span>
<span id="cb104-20"><a href="expression-data.html#cb104-20"></a>     <span class="op">--</span>outFilterType BySJout\</span>
<span id="cb104-21"><a href="expression-data.html#cb104-21"></a>     <span class="op">--</span>outFilterScoreMinOverLread <span class="fl">0.33</span> \</span>
<span id="cb104-22"><a href="expression-data.html#cb104-22"></a>     <span class="op">--</span>outFilterMatchNminOverLread <span class="fl">0.33</span> \</span>
<span id="cb104-23"><a href="expression-data.html#cb104-23"></a>     <span class="op">--</span>limitSjdbInsertNsj <span class="dv">1200000</span> \</span>
<span id="cb104-24"><a href="expression-data.html#cb104-24"></a>     <span class="op">--</span>outFilterIntronMotifs None \</span>
<span id="cb104-25"><a href="expression-data.html#cb104-25"></a>     <span class="op">--</span>alignSoftClipAtReferenceEnds Yes\</span>
<span id="cb104-26"><a href="expression-data.html#cb104-26"></a>     <span class="op">--</span>outSAMattrRGline ID<span class="op">:</span><span class="er">$</span><span class="dv">4</span> SM<span class="op">:</span><span class="er">$</span><span class="dv">4</span> \</span>
<span id="cb104-27"><a href="expression-data.html#cb104-27"></a>     <span class="op">--</span>chimSegmentMin <span class="dv">15</span> \</span>
<span id="cb104-28"><a href="expression-data.html#cb104-28"></a>     <span class="op">--</span>chimJunctionOverhangMin <span class="dv">15</span>\</span>
<span id="cb104-29"><a href="expression-data.html#cb104-29"></a>     <span class="op">--</span>limitBAMsortRAM <span class="dv">0</span>\</span>
<span id="cb104-30"><a href="expression-data.html#cb104-30"></a>     <span class="op">--</span>outSAMtype BAM SortedByCoordinate\</span>
<span id="cb104-31"><a href="expression-data.html#cb104-31"></a>     <span class="op">--</span>outSAMunmapped Within \</span>
<span id="cb104-32"><a href="expression-data.html#cb104-32"></a>     <span class="op">--</span>quantMode GeneCounts transcriptomeSAM \</span>
<span id="cb104-33"><a href="expression-data.html#cb104-33"></a>     <span class="op">--</span>quantTranscriptomeBan IndelSoftclipSingleend \</span>
<span id="cb104-34"><a href="expression-data.html#cb104-34"></a>     <span class="op">--</span>outFileNamePrefix <span class="op">$</span><span class="dv">3</span> \</span>
<span id="cb104-35"><a href="expression-data.html#cb104-35"></a>     <span class="op">--</span>twopassMode Basic</span></code></pre></div>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="expression-data.html#cb105-1"></a><span class="co"># Upload infoTable</span></span>
<span id="cb105-2"><a href="expression-data.html#cb105-2"></a>infoTable=<span class="kw">read.csv</span>(<span class="st">&quot;../metadata/AllRNA_samples_jan.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="rna-initial-qc" class="section level2">
<h2><span class="header-section-number">6.2</span> RNA Initial QC</h2>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="expression-data.html#cb106-1"></a>BatchNo=<span class="st">&quot;april&quot;</span></span>
<span id="cb106-2"><a href="expression-data.html#cb106-2"></a></span>
<span id="cb106-3"><a href="expression-data.html#cb106-3"></a>rsemFiles=<span class="kw">dir</span>(<span class="st">&quot;../data/RNA_expression/rsem/&quot;</span>, <span class="st">&quot;.results&quot;</span>)</span>
<span id="cb106-4"><a href="expression-data.html#cb106-4"></a>allrsem=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">17455</span>, <span class="dt">ncol=</span><span class="kw">length</span>(rsemFiles)) <span class="co">#31038</span></span>
<span id="cb106-5"><a href="expression-data.html#cb106-5"></a>allTPM=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">17455</span>, <span class="dt">ncol=</span><span class="kw">length</span>(rsemFiles))</span>
<span id="cb106-6"><a href="expression-data.html#cb106-6"></a>allFPKM=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">17455</span>, <span class="dt">ncol=</span><span class="kw">length</span>(rsemFiles))</span>
<span id="cb106-7"><a href="expression-data.html#cb106-7"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(rsemFiles)){</span>
<span id="cb106-8"><a href="expression-data.html#cb106-8"></a>  a1=<span class="kw">read.delim</span>(<span class="kw">file.path</span>(<span class="st">&quot;../data/RNA_expression/rsem/&quot;</span>, rsemFiles[i]))</span>
<span id="cb106-9"><a href="expression-data.html#cb106-9"></a>  allrsem[ ,i]=a1<span class="op">$</span>expected_count</span>
<span id="cb106-10"><a href="expression-data.html#cb106-10"></a>  allTPM[ ,i]=a1<span class="op">$</span>TPM</span>
<span id="cb106-11"><a href="expression-data.html#cb106-11"></a>  allFPKM[ ,i]=a1<span class="op">$</span>FPKM</span>
<span id="cb106-12"><a href="expression-data.html#cb106-12"></a>}</span>
<span id="cb106-13"><a href="expression-data.html#cb106-13"></a></span>
<span id="cb106-14"><a href="expression-data.html#cb106-14"></a>cNames=<span class="kw">unlist</span>(<span class="kw">strsplit</span>(rsemFiles, <span class="st">&quot;.genes.results&quot;</span>))</span>
<span id="cb106-15"><a href="expression-data.html#cb106-15"></a>cNames=<span class="kw">unlist</span>(<span class="kw">strsplit</span>(cNames, <span class="st">&quot;_0.33_v2&quot;</span>))</span>
<span id="cb106-16"><a href="expression-data.html#cb106-16"></a>sAnnot=<span class="kw">match</span>(cNames, infoTable<span class="op">$</span>starSampleName)</span>
<span id="cb106-17"><a href="expression-data.html#cb106-17"></a><span class="kw">colnames</span>(allrsem)=<span class="kw">paste</span>(infoTable<span class="op">$</span>Rat_ID[sAnnot],infoTable<span class="op">$</span>Location[sAnnot], infoTable<span class="op">$</span>Fraction[sAnnot],  <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb106-18"><a href="expression-data.html#cb106-18"></a><span class="kw">rownames</span>(allrsem)=a1<span class="op">$</span>gene_id</span>
<span id="cb106-19"><a href="expression-data.html#cb106-19"></a></span>
<span id="cb106-20"><a href="expression-data.html#cb106-20"></a><span class="kw">colnames</span>(allTPM)=<span class="kw">colnames</span>(allrsem)</span>
<span id="cb106-21"><a href="expression-data.html#cb106-21"></a><span class="kw">rownames</span>(allTPM)=<span class="kw">rownames</span>(allrsem)</span>
<span id="cb106-22"><a href="expression-data.html#cb106-22"></a></span>
<span id="cb106-23"><a href="expression-data.html#cb106-23"></a><span class="kw">colnames</span>(allFPKM)=<span class="kw">colnames</span>(allrsem)</span>
<span id="cb106-24"><a href="expression-data.html#cb106-24"></a><span class="kw">rownames</span>(allFPKM)=<span class="kw">rownames</span>(allrsem)</span>
<span id="cb106-25"><a href="expression-data.html#cb106-25"></a></span>
<span id="cb106-26"><a href="expression-data.html#cb106-26"></a>starFiles=<span class="kw">dir</span>(<span class="st">&quot;../data/RNA_expression/star_april/&quot;</span>, <span class="st">&quot;.tab&quot;</span>)</span>
<span id="cb106-27"><a href="expression-data.html#cb106-27"></a>allstar=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">17455</span>, <span class="dt">ncol=</span><span class="kw">length</span>(starFiles))</span>
<span id="cb106-28"><a href="expression-data.html#cb106-28"></a>allmapp=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">4</span>, <span class="dt">ncol=</span><span class="kw">length</span>(starFiles))</span>
<span id="cb106-29"><a href="expression-data.html#cb106-29"></a></span>
<span id="cb106-30"><a href="expression-data.html#cb106-30"></a></span>
<span id="cb106-31"><a href="expression-data.html#cb106-31"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(starFiles)){</span>
<span id="cb106-32"><a href="expression-data.html#cb106-32"></a>  a1=<span class="kw">read.delim</span>(<span class="kw">file.path</span>(<span class="st">&quot;../data/RNA_expression/star_april/&quot;</span>, starFiles[i]), <span class="dt">header=</span>F)</span>
<span id="cb106-33"><a href="expression-data.html#cb106-33"></a>  allstar[ ,i]=a1[ <span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>),<span class="dv">2</span>]</span>
<span id="cb106-34"><a href="expression-data.html#cb106-34"></a>  allmapp[ ,i]=a1[ <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>),<span class="dv">2</span>]</span>
<span id="cb106-35"><a href="expression-data.html#cb106-35"></a>}</span>
<span id="cb106-36"><a href="expression-data.html#cb106-36"></a></span>
<span id="cb106-37"><a href="expression-data.html#cb106-37"></a>cNames=<span class="kw">unlist</span>(<span class="kw">strsplit</span>(starFiles, <span class="st">&quot;ReadsPerGene.out.tab&quot;</span>))</span>
<span id="cb106-38"><a href="expression-data.html#cb106-38"></a>cNames=<span class="kw">unlist</span>(<span class="kw">strsplit</span>(cNames, <span class="st">&quot;_0.33_v2&quot;</span>))</span>
<span id="cb106-39"><a href="expression-data.html#cb106-39"></a>sAnnot=<span class="kw">match</span>(cNames, infoTable<span class="op">$</span>starSampleName)</span>
<span id="cb106-40"><a href="expression-data.html#cb106-40"></a><span class="kw">colnames</span>(allstar)=<span class="kw">paste</span>(infoTable<span class="op">$</span>Rat_ID[sAnnot], infoTable<span class="op">$</span>Location[sAnnot],infoTable<span class="op">$</span>Fraction[sAnnot], <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb106-41"><a href="expression-data.html#cb106-41"></a><span class="kw">rownames</span>(allstar)=a1[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>) ,<span class="dv">1</span>]</span>
<span id="cb106-42"><a href="expression-data.html#cb106-42"></a></span>
<span id="cb106-43"><a href="expression-data.html#cb106-43"></a><span class="kw">colnames</span>(allmapp)=<span class="kw">colnames</span>(allstar)</span>
<span id="cb106-44"><a href="expression-data.html#cb106-44"></a>allmapp=<span class="kw">rbind</span>(allmapp, <span class="kw">colSums</span>(allstar))</span>
<span id="cb106-45"><a href="expression-data.html#cb106-45"></a></span>
<span id="cb106-46"><a href="expression-data.html#cb106-46"></a><span class="kw">rownames</span>(allmapp)=<span class="kw">c</span>(<span class="kw">as.character</span>(a1[<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dv">1</span>]), <span class="st">&quot;UniqueReads&quot;</span>)</span>
<span id="cb106-47"><a href="expression-data.html#cb106-47"></a></span>
<span id="cb106-48"><a href="expression-data.html#cb106-48"></a></span>
<span id="cb106-49"><a href="expression-data.html#cb106-49"></a><span class="co">## in all cases, remove the files </span></span>
<span id="cb106-50"><a href="expression-data.html#cb106-50"></a>id=<span class="kw">match</span>(cNames, infoTable<span class="op">$</span>starSampleName)</span>
<span id="cb106-51"><a href="expression-data.html#cb106-51"></a>infoTable=infoTable[ <span class="kw">na.omit</span>(id), ]</span>
<span id="cb106-52"><a href="expression-data.html#cb106-52"></a>id2=<span class="kw">which</span>(<span class="kw">is.na</span>(id))</span>
<span id="cb106-53"><a href="expression-data.html#cb106-53"></a></span>
<span id="cb106-54"><a href="expression-data.html#cb106-54"></a><span class="cf">if</span> (<span class="kw">length</span>(id2)<span class="op">&gt;</span><span class="dv">0</span>){</span>
<span id="cb106-55"><a href="expression-data.html#cb106-55"></a></span>
<span id="cb106-56"><a href="expression-data.html#cb106-56"></a>allmapp=allmapp[ , <span class="op">-</span><span class="kw">grep</span>(<span class="st">&quot;NA_NA&quot;</span>, <span class="kw">colnames</span>(allmapp))]</span>
<span id="cb106-57"><a href="expression-data.html#cb106-57"></a></span>
<span id="cb106-58"><a href="expression-data.html#cb106-58"></a>allTPM=allTPM[ , <span class="op">-</span>id2]</span>
<span id="cb106-59"><a href="expression-data.html#cb106-59"></a>allFPKM=allFPKM[ , <span class="op">-</span>id2]</span>
<span id="cb106-60"><a href="expression-data.html#cb106-60"></a>allrsem=allrsem[ , <span class="op">-</span>id2]</span>
<span id="cb106-61"><a href="expression-data.html#cb106-61"></a>allstar=allstar[, <span class="op">-</span>id2]</span>
<span id="cb106-62"><a href="expression-data.html#cb106-62"></a>}</span></code></pre></div>
<p>Default output from R showing the number of unique reads compared to multimapped, unmapped etc. This is shown for each batch.
Note that batch 3 has differences (high percentage of unmapped) compared to the other batches, possibly due to DNA contamination.</p>
<p>Below we check for three measures:</p>
<ul>
<li>mapped million reads (ideally, 10M+ reads)</li>
<li>Gene Sparsity: This is a measurement of the number of genes which have non-zero values. Ideally, would be greater than 10K, but values which are too high may also suggest contamination from DNA (unexpressed genes are also counted)</li>
<li>Varability: standard deviation of the transcriptomic counts. If this value is too low, would suggest that high DNA contamination, non-representative transcriptome.</li>
</ul>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="expression-data.html#cb107-1"></a><span class="co"># number of mapped reads</span></span>
<span id="cb107-2"><a href="expression-data.html#cb107-2"></a>UnMappedNorm=<span class="kw">t</span>(allmapp)<span class="op">/</span><span class="kw">colSums</span>(allmapp)</span>
<span id="cb107-3"><a href="expression-data.html#cb107-3"></a></span>
<span id="cb107-4"><a href="expression-data.html#cb107-4"></a>mUnMap=<span class="kw">melt</span>(UnMappedNorm)</span>
<span id="cb107-5"><a href="expression-data.html#cb107-5"></a>mUnMap<span class="op">$</span>Batch=infoTable<span class="op">$</span>Batch[<span class="kw">match</span>(mUnMap<span class="op">$</span>Var1, infoTable<span class="op">$</span>SampleID)]</span>
<span id="cb107-6"><a href="expression-data.html#cb107-6"></a></span>
<span id="cb107-7"><a href="expression-data.html#cb107-7"></a><span class="co"># how many genes represented</span></span>
<span id="cb107-8"><a href="expression-data.html#cb107-8"></a>Sparsity=<span class="kw">colSums</span>(<span class="kw">sign</span>(allstar))</span>
<span id="cb107-9"><a href="expression-data.html#cb107-9"></a><span class="co"># check how skewed the data is</span></span>
<span id="cb107-10"><a href="expression-data.html#cb107-10"></a></span>
<span id="cb107-11"><a href="expression-data.html#cb107-11"></a>cSDs=<span class="kw">colSds</span>(allstar)</span>
<span id="cb107-12"><a href="expression-data.html#cb107-12"></a></span>
<span id="cb107-13"><a href="expression-data.html#cb107-13"></a>TVals=<span class="kw">data.frame</span>(<span class="dt">MappedReadsM=</span>allmapp[<span class="dv">5</span>, ]<span class="op">/</span><span class="fl">1E6</span>, <span class="dt">GeneSparsityK=</span>Sparsity<span class="op">/</span><span class="fl">1E3</span>,<span class="dt">Batch=</span>infoTable<span class="op">$</span>Batch, <span class="dt">GeneVariabilityCounts=</span>cSDs, <span class="dt">Type=</span><span class="kw">as.character</span>(infoTable<span class="op">$</span>Fraction), <span class="dt">names=</span><span class="kw">colnames</span>(allstar))</span>
<span id="cb107-14"><a href="expression-data.html#cb107-14"></a></span>
<span id="cb107-15"><a href="expression-data.html#cb107-15"></a>mTV=<span class="kw">melt</span>(TVals, <span class="dt">measure.vars =</span> <span class="kw">c</span>(<span class="st">&quot;MappedReadsM&quot;</span>, <span class="st">&quot;GeneSparsityK&quot;</span>, <span class="st">&quot;GeneVariabilityCounts&quot;</span>))</span>
<span id="cb107-16"><a href="expression-data.html#cb107-16"></a></span>
<span id="cb107-17"><a href="expression-data.html#cb107-17"></a><span class="co">#pdf(sprintf(&quot;../rslt/DESeq/GE_preprocessing_%s_%s.pdf&quot;, BatchNo,Sys.Date()), height=5, width=8)</span></span>
<span id="cb107-18"><a href="expression-data.html#cb107-18"></a></span>
<span id="cb107-19"><a href="expression-data.html#cb107-19"></a><span class="kw">ggplot</span>(mUnMap, <span class="kw">aes</span>(<span class="dt">x=</span>Var1, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var2))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>Batch, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;proportion of reads&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;mapping summary&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/summary-plots-rna-quality-1.png" width="672" /></p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="expression-data.html#cb108-1"></a></span>
<span id="cb108-2"><a href="expression-data.html#cb108-2"></a><span class="kw">ggplot</span>(mTV, <span class="kw">aes</span>(<span class="dt">x=</span>names, <span class="dt">y=</span>value,<span class="dt">fill=</span>Type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span></span>
<span id="cb108-3"><a href="expression-data.html#cb108-3"></a><span class="st">  </span><span class="kw">facet_grid</span>(variable<span class="op">~</span>Batch, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;#Mapped Reads, #Unique Gebes, #Variability&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/summary-plots-rna-quality-2.png" width="672" /></p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="expression-data.html#cb109-1"></a></span>
<span id="cb109-2"><a href="expression-data.html#cb109-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb109-3"><a href="expression-data.html#cb109-3"></a></span>
<span id="cb109-4"><a href="expression-data.html#cb109-4"></a><span class="kw">plot</span>(<span class="kw">density</span>(TVals<span class="op">$</span>MappedReadsM), <span class="dt">main=</span><span class="st">&quot;mapped reads&quot;</span>)</span>
<span id="cb109-5"><a href="expression-data.html#cb109-5"></a>x1=<span class="kw">mean</span>(TVals<span class="op">$</span>MappedReadsM)</span>
<span id="cb109-6"><a href="expression-data.html#cb109-6"></a>sdv=<span class="kw">sd</span>(TVals<span class="op">$</span>MappedReadsM)</span>
<span id="cb109-7"><a href="expression-data.html#cb109-7"></a><span class="kw">abline</span>(<span class="dt">v=</span><span class="kw">c</span>(x1, x1<span class="fl">-1.5</span><span class="op">*</span>sdv, x1<span class="fl">+1.5</span><span class="op">*</span>sdv), <span class="dt">col=</span><span class="st">&quot;grey&quot;</span>, <span class="dt">lty=</span><span class="dv">2</span>)</span>
<span id="cb109-8"><a href="expression-data.html#cb109-8"></a><span class="kw">text</span>( <span class="kw">c</span>(x1, x1<span class="fl">-1.5</span><span class="op">*</span>sdv, x1<span class="fl">+1.5</span><span class="op">*</span>sdv), <span class="fl">0.07</span>, <span class="kw">c</span>(x1, x1<span class="dv">-2</span><span class="op">*</span>sdv, x1<span class="op">+</span><span class="dv">2</span><span class="op">*</span>sdv), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex =</span> <span class="fl">0.75</span>, <span class="dt">srt=</span><span class="dv">90</span>)</span>
<span id="cb109-9"><a href="expression-data.html#cb109-9"></a></span>
<span id="cb109-10"><a href="expression-data.html#cb109-10"></a><span class="kw">plot</span>(<span class="kw">density</span>(TVals<span class="op">$</span>GeneSparsityK), <span class="dt">main=</span><span class="st">&quot;gene sparsity&quot;</span>)</span>
<span id="cb109-11"><a href="expression-data.html#cb109-11"></a>x1=<span class="kw">mean</span>(TVals<span class="op">$</span>GeneSparsityK)</span>
<span id="cb109-12"><a href="expression-data.html#cb109-12"></a>sdv=<span class="kw">sd</span>(TVals<span class="op">$</span>GeneSparsityK)</span>
<span id="cb109-13"><a href="expression-data.html#cb109-13"></a><span class="kw">abline</span>(<span class="dt">v=</span><span class="kw">c</span>(x1, x1<span class="fl">-1.5</span><span class="op">*</span>sdv, x1<span class="fl">+1.5</span><span class="op">*</span>sdv), <span class="dt">col=</span><span class="st">&quot;grey&quot;</span>, <span class="dt">lty=</span><span class="dv">2</span>)</span>
<span id="cb109-14"><a href="expression-data.html#cb109-14"></a><span class="kw">text</span>( <span class="kw">c</span>(x1, x1<span class="fl">-1.5</span><span class="op">*</span>sdv, x1<span class="fl">+1.5</span><span class="op">*</span>sdv), <span class="fl">0.07</span>, <span class="kw">c</span>(x1, x1<span class="dv">-2</span><span class="op">*</span>sdv, x1<span class="op">+</span><span class="dv">2</span><span class="op">*</span>sdv), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex =</span> <span class="fl">0.75</span>, <span class="dt">srt=</span><span class="dv">90</span>)</span>
<span id="cb109-15"><a href="expression-data.html#cb109-15"></a></span>
<span id="cb109-16"><a href="expression-data.html#cb109-16"></a><span class="kw">plot</span>(<span class="kw">density</span>(TVals<span class="op">$</span>GeneVariabilityCounts), <span class="dt">main=</span><span class="st">&quot;Gene Variability&quot;</span>)</span>
<span id="cb109-17"><a href="expression-data.html#cb109-17"></a>x1=<span class="kw">mean</span>(TVals<span class="op">$</span>GeneVariabilityCounts)</span>
<span id="cb109-18"><a href="expression-data.html#cb109-18"></a>sdv=<span class="kw">sd</span>(TVals<span class="op">$</span>GeneVariabilityCounts)</span>
<span id="cb109-19"><a href="expression-data.html#cb109-19"></a><span class="kw">abline</span>(<span class="dt">v=</span><span class="kw">c</span>(x1, x1<span class="fl">-1.5</span><span class="op">*</span>sdv, x1<span class="fl">+1.5</span><span class="op">*</span>sdv), <span class="dt">col=</span><span class="st">&quot;grey&quot;</span>, <span class="dt">lty=</span><span class="dv">2</span>)</span>
<span id="cb109-20"><a href="expression-data.html#cb109-20"></a><span class="kw">text</span>( <span class="kw">c</span>(x1, x1<span class="fl">-1.5</span><span class="op">*</span>sdv, x1<span class="fl">+1.5</span><span class="op">*</span>sdv), <span class="fl">0.00015</span>, <span class="kw">c</span>(x1, x1<span class="dv">-2</span><span class="op">*</span>sdv, x1<span class="op">+</span><span class="dv">2</span><span class="op">*</span>sdv), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex =</span> <span class="fl">0.75</span>, <span class="dt">srt=</span><span class="dv">90</span>)</span>
<span id="cb109-21"><a href="expression-data.html#cb109-21"></a></span>
<span id="cb109-22"><a href="expression-data.html#cb109-22"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/summary-plots-rna-quality-3.png" width="672" /></p>
<p>Samples to remove from analysis:</p>
<p>The thresholds indicated below are based on the above density plots, and removes cases which are &lt;1.5 SD of the mean</p>
<ul>
<li>low total number of mapped reads (under 1.5M)</li>
<li>sparsity: less than 8K genes</li>
<li>variability : threshold under 500</li>
</ul>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="expression-data.html#cb110-1"></a>rmSamples=<span class="kw">which</span>(TVals<span class="op">$</span>MappedReadsM<span class="op">&lt;</span><span class="fl">1.5</span> <span class="op">|</span><span class="st"> </span>TVals<span class="op">$</span>GeneSparsityK<span class="op">&lt;</span><span class="dv">8</span> <span class="op">|</span><span class="st"> </span>TVals<span class="op">$</span>GeneVariabilityCounts<span class="op">&lt;</span><span class="dv">500</span>)</span>
<span id="cb110-2"><a href="expression-data.html#cb110-2"></a>allstarFinal=allstar[ ,<span class="op">-</span>rmSamples]</span>
<span id="cb110-3"><a href="expression-data.html#cb110-3"></a>allrsemFinal=allrsem[ ,<span class="op">-</span>rmSamples]</span>
<span id="cb110-4"><a href="expression-data.html#cb110-4"></a>allTPMFinal=allTPM[ ,<span class="op">-</span>rmSamples]</span>
<span id="cb110-5"><a href="expression-data.html#cb110-5"></a>allFPKMFinal=allFPKM[ ,<span class="op">-</span>rmSamples]</span>
<span id="cb110-6"><a href="expression-data.html#cb110-6"></a>infoTableFinal=infoTable[<span class="op">-</span>rmSamples, ]</span></code></pre></div>
<p>The omitted samples are:</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="expression-data.html#cb111-1"></a><span class="kw">kable</span>(TVals[rmSamples, ])</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
MappedReadsM
</th>
<th style="text-align:right;">
GeneSparsityK
</th>
<th style="text-align:right;">
Batch
</th>
<th style="text-align:right;">
GeneVariabilityCounts
</th>
<th style="text-align:left;">
Type
</th>
<th style="text-align:left;">
names
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
6R_C_CD45
</td>
<td style="text-align:right;">
2.287110
</td>
<td style="text-align:right;">
5.585
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1063.38925
</td>
<td style="text-align:left;">
CD45
</td>
<td style="text-align:left;">
6R_C_CD45
</td>
</tr>
<tr>
<td style="text-align:left;">
10L_B_DN
</td>
<td style="text-align:right;">
0.864838
</td>
<td style="text-align:right;">
3.308
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
559.30659
</td>
<td style="text-align:left;">
DN
</td>
<td style="text-align:left;">
10L_B_DN
</td>
</tr>
<tr>
<td style="text-align:left;">
NMU1_LL_Ep
</td>
<td style="text-align:right;">
1.032440
</td>
<td style="text-align:right;">
16.708
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
69.82645
</td>
<td style="text-align:left;">
Ep
</td>
<td style="text-align:left;">
NMU1_LL_Ep
</td>
</tr>
<tr>
<td style="text-align:left;">
NMU5_LA_Ep
</td>
<td style="text-align:right;">
0.500429
</td>
<td style="text-align:right;">
15.707
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
35.56084
</td>
<td style="text-align:left;">
Ep
</td>
<td style="text-align:left;">
NMU5_LA_Ep
</td>
</tr>
<tr>
<td style="text-align:left;">
Control1__Ep
</td>
<td style="text-align:right;">
8.224593
</td>
<td style="text-align:right;">
5.800
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3164.00515
</td>
<td style="text-align:left;">
Ep
</td>
<td style="text-align:left;">
Control1__Ep
</td>
</tr>
<tr>
<td style="text-align:left;">
NMU13_RAU_Ep
</td>
<td style="text-align:right;">
3.049733
</td>
<td style="text-align:right;">
7.242
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1069.99912
</td>
<td style="text-align:left;">
Ep
</td>
<td style="text-align:left;">
NMU13_RAU_Ep
</td>
</tr>
</tbody>
</table>
<p>We are left with 110 samples.</p>
<p>There are 47, 32, 31 samples in the CD45, Ep, DN fractions.</p>
<p>There are 20, 49, 41 samples from batches 1, 2 and 3 respectively.</p>
</div>
<div id="normalisation" class="section level2">
<h2><span class="header-section-number">6.3</span> Normalisation</h2>
<p>Run through DESEq and normalise the library. Using all samples, we run the model:</p>
<p>expression ~ Celltype + factor (Batch)</p>
<p>and keep the genes which have a total count of at least half the number of samples. ie.
<span class="math display">\[ sum(gene_i) &gt; N_{samples}/2 \]</span></p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="expression-data.html#cb112-1"></a><span class="co"># remove rows where counts are low</span></span>
<span id="cb112-2"><a href="expression-data.html#cb112-2"></a><span class="kw">rownames</span>(infoTableFinal)=infoTableFinal<span class="op">$</span>SampleID</span>
<span id="cb112-3"><a href="expression-data.html#cb112-3"></a>infoTableFinal<span class="op">$</span>Batch=<span class="kw">factor</span>(infoTableFinal<span class="op">$</span>Batch)</span>
<span id="cb112-4"><a href="expression-data.html#cb112-4"></a>dds=<span class="kw">DESeqDataSetFromMatrix</span>(allstarFinal, infoTableFinal, <span class="dt">design=</span><span class="op">~</span>Fraction<span class="op">+</span><span class="kw">factor</span>(Batch)) <span class="co">## change class</span></span>
<span id="cb112-5"><a href="expression-data.html#cb112-5"></a>keep=<span class="kw">rowSums</span>(<span class="kw">counts</span>(dds))<span class="op">&gt;</span>(<span class="kw">ncol</span>(dds)<span class="op">/</span><span class="dv">2</span>)</span>
<span id="cb112-6"><a href="expression-data.html#cb112-6"></a>dds=dds[keep, ]</span>
<span id="cb112-7"><a href="expression-data.html#cb112-7"></a>dds=<span class="kw">DESeq</span>(dds)</span>
<span id="cb112-8"><a href="expression-data.html#cb112-8"></a></span>
<span id="cb112-9"><a href="expression-data.html#cb112-9"></a>vsd &lt;-<span class="st"> </span><span class="kw">varianceStabilizingTransformation</span>(dds)</span>
<span id="cb112-10"><a href="expression-data.html#cb112-10"></a>normalizedTableVSD &lt;-<span class="st"> </span><span class="kw">assay</span>(vsd)</span>
<span id="cb112-11"><a href="expression-data.html#cb112-11"></a></span>
<span id="cb112-12"><a href="expression-data.html#cb112-12"></a>infoTableFinal<span class="op">$</span>TumorIDnew=Cdata<span class="op">$</span>NewID[<span class="kw">match</span>(infoTableFinal<span class="op">$</span>TumorID, Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb112-13"><a href="expression-data.html#cb112-13"></a></span>
<span id="cb112-14"><a href="expression-data.html#cb112-14"></a><span class="kw">save</span>(dds, vsd,allstarFinal, allrsemFinal,allTPMFinal, normalizedTableVSD,infoTableFinal,  <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;../dds_normalised_data_newstar_RNAseq%s_%s.RData&quot;</span>, BatchNo, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
<div id="preliminary-visualisation-to-remove-outliers" class="section level3">
<h3><span class="header-section-number">6.3.1</span> preliminary visualisation (to remove outliers)</h3>
<p>Below are PCA plots based on:</p>
<ul>
<li>Batch</li>
<li>CellType</li>
</ul>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="expression-data.html#cb113-1"></a>vsd2 &lt;-<span class="st"> </span><span class="kw">vst</span>(dds)</span>
<span id="cb113-2"><a href="expression-data.html#cb113-2"></a></span>
<span id="cb113-3"><a href="expression-data.html#cb113-3"></a><span class="co">#pdf(sprintf(&quot;../rslt/DESeq/PCA_preliminary_%s_%s.pdf&quot;,BatchNo, Sys.Date() ), width=8, height=6)</span></span>
<span id="cb113-4"><a href="expression-data.html#cb113-4"></a></span>
<span id="cb113-5"><a href="expression-data.html#cb113-5"></a><span class="kw">plotPCA</span>(vsd2, <span class="st">&quot;Batch&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Batch&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-1.png" width="672" /></p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="expression-data.html#cb114-1"></a><span class="kw">plotPCA</span>(vsd2, <span class="st">&quot;Fraction&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Fraction&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-2.png" width="672" /></p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="expression-data.html#cb115-1"></a><span class="kw">plotPCA</span>(vsd2, <span class="kw">c</span>(<span class="st">&quot;Fraction&quot;</span>))<span class="op">+</span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">label =</span> name)) </span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/vsd-transform-3.png" width="672" /></p>
<p>Batches in general separate out well, however, some samples appear to be outliers in comparison to the main group. We look in closer detail the CD45, DN and EpCAM populations.</p>
<p>In the CD45 population, narrow down to only immune related genes to see if there is a difference.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="expression-data.html#cb116-1"></a><span class="kw">plotPCA</span>(vsd2[<span class="kw">rownames</span>(<span class="kw">assay</span>(vsd2))<span class="op">%in%</span>RatAllImm  , <span class="kw">grep</span>(<span class="st">&quot;CD45&quot;</span>, <span class="kw">colnames</span>(vsd2))], <span class="kw">c</span>(<span class="st">&quot;Growth&quot;</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;CD45 only Growth&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-51-1.png" width="672" /></p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="expression-data.html#cb117-1"></a><span class="kw">plotPCA</span>(vsd2[ , <span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, <span class="kw">colnames</span>(vsd2))], <span class="st">&quot;Growth&quot;</span>)<span class="op">+</span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">label =</span> name))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Ep only Growth&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-51-2.png" width="672" /></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="expression-data.html#cb118-1"></a><span class="kw">plotPCA</span>(vsd2[ , <span class="kw">grep</span>(<span class="st">&quot;DN&quot;</span>, <span class="kw">colnames</span>(vsd2))], <span class="st">&quot;Growth&quot;</span>)<span class="op">+</span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">label =</span> name))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;DN only Growth&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-51-3.png" width="672" /></p>
<p>Based on the above plots, we remove the following outliers and re-run the normalisation:</p>
<ul>
<li>2R_D_DN</li>
<li>4L_B_CD45</li>
</ul>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="expression-data.html#cb119-1"></a>rmThese=<span class="kw">c</span>(<span class="st">&quot;2R_D_DN&quot;</span>, <span class="st">&quot;4L_B_CD45&quot;</span>)</span>
<span id="cb119-2"><a href="expression-data.html#cb119-2"></a></span>
<span id="cb119-3"><a href="expression-data.html#cb119-3"></a><span class="co">## Note based on the above plots, sample &quot;2R_D_DN&quot; is misclassified as a cd45 sample. Need to remove this sample and re-run the preprocessing:</span></span>
<span id="cb119-4"><a href="expression-data.html#cb119-4"></a>allstarFinal=allstarFinal[ ,<span class="op">-</span><span class="kw">match</span>(rmThese, <span class="kw">colnames</span>(allstarFinal))]</span>
<span id="cb119-5"><a href="expression-data.html#cb119-5"></a>infoTableFinal=infoTableFinal[<span class="op">-</span><span class="kw">match</span>(rmThese, <span class="kw">rownames</span>(infoTableFinal)), ]</span>
<span id="cb119-6"><a href="expression-data.html#cb119-6"></a>allrsemFinal=allrsemFinal[ ,<span class="op">-</span><span class="kw">match</span>(rmThese, <span class="kw">colnames</span>(allrsemFinal))]</span>
<span id="cb119-7"><a href="expression-data.html#cb119-7"></a>allTPMFinal=allTPMFinal[ ,<span class="op">-</span><span class="kw">match</span>(rmThese, <span class="kw">colnames</span>(allTPMFinal))]</span>
<span id="cb119-8"><a href="expression-data.html#cb119-8"></a></span>
<span id="cb119-9"><a href="expression-data.html#cb119-9"></a>dds=<span class="kw">DESeqDataSetFromMatrix</span>(allstarFinal, infoTableFinal, <span class="dt">design=</span><span class="op">~</span>Fraction<span class="op">+</span><span class="kw">factor</span>(Batch)) <span class="co">## change class</span></span>
<span id="cb119-10"><a href="expression-data.html#cb119-10"></a>keep=<span class="kw">rowSums</span>(<span class="kw">counts</span>(dds))<span class="op">&gt;</span>(<span class="kw">ncol</span>(dds)<span class="op">/</span><span class="dv">2</span>)</span>
<span id="cb119-11"><a href="expression-data.html#cb119-11"></a>dds=dds[keep, ]</span>
<span id="cb119-12"><a href="expression-data.html#cb119-12"></a>dds=<span class="kw">DESeq</span>(dds)</span>
<span id="cb119-13"><a href="expression-data.html#cb119-13"></a></span>
<span id="cb119-14"><a href="expression-data.html#cb119-14"></a>vsd &lt;-<span class="st"> </span><span class="kw">varianceStabilizingTransformation</span>(dds)</span>
<span id="cb119-15"><a href="expression-data.html#cb119-15"></a>normalizedTableVSD &lt;-<span class="st"> </span><span class="kw">assay</span>(vsd)</span>
<span id="cb119-16"><a href="expression-data.html#cb119-16"></a></span>
<span id="cb119-17"><a href="expression-data.html#cb119-17"></a>infoTableFinal<span class="op">$</span>MHEpCAM=df.Spatial<span class="op">$</span>MH.EpCAM[<span class="kw">match</span>(<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, infoTableFinal<span class="op">$</span>TumorID), <span class="kw">rownames</span>(df.Spatial))]</span>
<span id="cb119-18"><a href="expression-data.html#cb119-18"></a>infoTableFinal<span class="op">$</span>IFEpCAM=df.Spatial<span class="op">$</span>IF.EpCAM[<span class="kw">match</span>(<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, infoTableFinal<span class="op">$</span>TumorID), <span class="kw">rownames</span>(df.Spatial))]</span>
<span id="cb119-19"><a href="expression-data.html#cb119-19"></a>infoTableFinal<span class="op">$</span>knnEpCAM=df.Spatial<span class="op">$</span>knn.EpCAM[<span class="kw">match</span>(<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, infoTableFinal<span class="op">$</span>TumorID), <span class="kw">rownames</span>(df.Spatial))]</span>
<span id="cb119-20"><a href="expression-data.html#cb119-20"></a>infoTableFinal<span class="op">$</span>CD8Frac=df.Spatial<span class="op">$</span>CD8frac[<span class="kw">match</span>(<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, infoTableFinal<span class="op">$</span>TumorID),<span class="kw">rownames</span>(df.Spatial))]</span>
<span id="cb119-21"><a href="expression-data.html#cb119-21"></a></span>
<span id="cb119-22"><a href="expression-data.html#cb119-22"></a><span class="co"># infoTableFinal$Treatment=RNADNAsamples$Treatment[match(infoTableFinal$FqFile, RNADNAsamples$FqFile.CD45)]</span></span>
<span id="cb119-23"><a href="expression-data.html#cb119-23"></a><span class="co"># idx=match(infoTableFinal$FqFile, RNADNAsamples$FqFile.Ep)</span></span>
<span id="cb119-24"><a href="expression-data.html#cb119-24"></a><span class="co"># infoTableFinal$Treatment[-which(is.na(idx))]=RNADNAsamples$Treatment[na.omit(idx)]</span></span>
<span id="cb119-25"><a href="expression-data.html#cb119-25"></a><span class="co"># idx=match(infoTableFinal$FqFile, RNADNAsamples$FqFile.DN)</span></span>
<span id="cb119-26"><a href="expression-data.html#cb119-26"></a><span class="co"># infoTableFinal$Treatment[-which(is.na(idx))]=RNADNAsamples$Treatment[na.omit(idx)]</span></span>
<span id="cb119-27"><a href="expression-data.html#cb119-27"></a></span>
<span id="cb119-28"><a href="expression-data.html#cb119-28"></a><span class="kw">save</span>(dds, vsd,allstarFinal, allrsemFinal,allTPMFinal, normalizedTableVSD,infoTableFinal,  <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/dds_normalised_data_newstar_RNAseq_%s_rm_outliers.RData&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
</div>
</div>
<div id="processing-files-for-external-software" class="section level2">
<h2><span class="header-section-number">6.4</span> Processing files for external software</h2>
<p>We also process these files for external software (TIMER) - which can also run cibersort</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="expression-data.html#cb120-1"></a>Rnames=<span class="kw">rownames</span>(allrsemFinal)</span>
<span id="cb120-2"><a href="expression-data.html#cb120-2"></a>mNames1=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(Rnames, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb120-3"><a href="expression-data.html#cb120-3"></a>mNames2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(Rnames, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb120-4"><a href="expression-data.html#cb120-4"></a></span>
<span id="cb120-5"><a href="expression-data.html#cb120-5"></a>HumNameFinal=<span class="kw">ifelse</span>(<span class="kw">is.na</span>(mNames1), mNames2, mNames1)</span>
<span id="cb120-6"><a href="expression-data.html#cb120-6"></a>x1=<span class="kw">which</span>(<span class="kw">is.na</span>(HumNameFinal)<span class="op">==</span>T)</span>
<span id="cb120-7"><a href="expression-data.html#cb120-7"></a></span>
<span id="cb120-8"><a href="expression-data.html#cb120-8"></a><span class="co">## save rsem for xcell</span></span>
<span id="cb120-9"><a href="expression-data.html#cb120-9"></a>allrsemSave=allrsemFinal[<span class="op">-</span>x1, ]</span>
<span id="cb120-10"><a href="expression-data.html#cb120-10"></a><span class="kw">rownames</span>(allrsemSave)=<span class="kw">na.omit</span>(HumNameFinal)</span>
<span id="cb120-11"><a href="expression-data.html#cb120-11"></a></span>
<span id="cb120-12"><a href="expression-data.html#cb120-12"></a><span class="kw">write.table</span>(allrsemSave, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;../output4external/RSEM_for_xcell%s__%s.txt&quot;</span>,BatchNo, <span class="kw">Sys.Date</span>()), <span class="dt">col.names =</span> <span class="ot">NA</span>)</span>
<span id="cb120-13"><a href="expression-data.html#cb120-13"></a></span>
<span id="cb120-14"><a href="expression-data.html#cb120-14"></a><span class="co">## save row names for cibersort</span></span>
<span id="cb120-15"><a href="expression-data.html#cb120-15"></a>alltpmSave=allTPMFinal[<span class="op">-</span>x1, ]</span>
<span id="cb120-16"><a href="expression-data.html#cb120-16"></a><span class="kw">rownames</span>(alltpmSave)=<span class="kw">na.omit</span>(HumNameFinal)</span>
<span id="cb120-17"><a href="expression-data.html#cb120-17"></a><span class="kw">write.table</span>(alltpmSave, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;../output4external/TPM_for_cibersort%s_%s.txt&quot;</span>,BatchNo, <span class="kw">Sys.Date</span>()), <span class="dt">col.names =</span> <span class="ot">NA</span>)</span></code></pre></div>
<p>Also split up the immunotherapy and the characterisation cohorts.
Save the mouse names for TIMER cistrome: check that this is actually required for TIMER</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="expression-data.html#cb121-1"></a></span>
<span id="cb121-2"><a href="expression-data.html#cb121-2"></a><span class="co"># Write files for CIBERSORT using rgd. Use cohort specific values</span></span>
<span id="cb121-3"><a href="expression-data.html#cb121-3"></a>allTPMCD45=allTPMFinal[ , <span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">==</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb121-4"><a href="expression-data.html#cb121-4"></a></span>
<span id="cb121-5"><a href="expression-data.html#cb121-5"></a><span class="kw">write.csv</span>(allTPMCD45, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;../output4external/CD45_TPM_rgd_names_prog_%s.csv&quot;</span>, <span class="kw">Sys.Date</span>()))</span>
<span id="cb121-6"><a href="expression-data.html#cb121-6"></a></span>
<span id="cb121-7"><a href="expression-data.html#cb121-7"></a>allTPMCD45=allTPMFinal[ , <span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span> <span class="op">&amp;</span><span class="st"> </span>infoTableFinal<span class="op">$</span>Cohort<span class="op">!=</span><span class="st">&quot;Progression&quot;</span>)]</span>
<span id="cb121-8"><a href="expression-data.html#cb121-8"></a></span>
<span id="cb121-9"><a href="expression-data.html#cb121-9"></a><span class="kw">write.csv</span>(allTPMCD45, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;../output4external/CD45_TPM_rgd_names_char_%s.csv&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="spatial-statistics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="rna-data-preliminary-plots.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
