<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 18 Immune estimation | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 18 Immune estimation | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 18 Immune estimation | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-02-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="signature-analysis.html"/>
<link rel="next" href="bcr-clonotype-analysis.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.2</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.2.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="24.3" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>24.3</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="immune-estimation" class="section level1">
<h1><span class="header-section-number">Chapter 18</span> Immune estimation</h1>
<p>In this section, we will look at Deconvolution methods (CIBERSORT, TIMER etc) for estimating immune fractions and cell types</p>
<p>Deconvolution was performed using the TIMER website, which lists results from TIMER,
XCELL,
CIBERSORT,
EPIC, MMPCOUNTER</p>
<p>TPM counts were used for this analysis (using Rat gene names) on the TIMER website ()</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="immune-estimation.html#cb337-1"></a>ProgSpecCD45=<span class="kw">read.csv</span>(<span class="st">&quot;../data/RNA_expression/CD45_TPM_rgd_names_prog_12-08_estimation_matrix.csv&quot;</span>)</span>
<span id="cb337-2"><a href="immune-estimation.html#cb337-2"></a><span class="kw">colnames</span>(ProgSpecCD45)=<span class="kw">gsub</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">colnames</span>(ProgSpecCD45))</span>
<span id="cb337-3"><a href="immune-estimation.html#cb337-3"></a>CharSpecCD45=<span class="kw">read.csv</span>(<span class="st">&quot;../data/RNA_expression/CD45-tpm-rgdnames-char-2020-11-22-estimation_matrix.csv&quot;</span>)</span>
<span id="cb337-4"><a href="immune-estimation.html#cb337-4"></a><span class="kw">colnames</span>(CharSpecCD45)=<span class="kw">gsub</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">colnames</span>(CharSpecCD45))</span>
<span id="cb337-5"><a href="immune-estimation.html#cb337-5"></a></span>
<span id="cb337-6"><a href="immune-estimation.html#cb337-6"></a><span class="co">## merge the two together</span></span>
<span id="cb337-7"><a href="immune-estimation.html#cb337-7"></a></span>
<span id="cb337-8"><a href="immune-estimation.html#cb337-8"></a>output1=<span class="kw">merge</span>(ProgSpecCD45, CharSpecCD45, <span class="dt">by.x=</span><span class="st">&quot;cell_type&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;cell_type&quot;</span>, <span class="dt">all=</span>T)</span></code></pre></div>
<div id="overview-of-the-cell-types" class="section level2">
<h2><span class="header-section-number">18.1</span> Overview of the cell types</h2>
<p>Below, we will look at the enrichment scores of specific cell types compared to others using these different methods. It appears that most methods have scores which skews towards high representation of T cells:</p>
<p>TIMER for example shows an enrichment of dendritic and CD8 T cells.
EPIC in contrast shows enrichment for CD4+ and to a lesser extend CD8 T cells
MMPCOUNTER puts an unusually large weighting to T cells and this does not fit our FACS analysis
XCELL enriches for T cells</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="immune-estimation.html#cb338-1"></a>RowNames=<span class="kw">c</span>(<span class="st">&quot;TIMER&quot;</span>, <span class="st">&quot;CIBERSORT$&quot;</span>, <span class="st">&quot;CIBERSORT-ABS&quot;</span>, <span class="st">&quot;EPIC&quot;</span>, <span class="st">&quot;MMCPCOUNTER&quot;</span>, <span class="st">&quot;XCELL&quot;</span>)</span>
<span id="cb338-2"><a href="immune-estimation.html#cb338-2"></a>Type=<span class="kw">c</span>(<span class="st">&quot;enrichment&quot;</span>, <span class="st">&quot;fraction&quot;</span>, <span class="st">&quot;enrichment&quot;</span>, <span class="st">&quot;fraction&quot;</span>, <span class="st">&quot;enrichment&quot;</span>, <span class="st">&quot;enrichment&quot;</span>)</span>
<span id="cb338-3"><a href="immune-estimation.html#cb338-3"></a></span>
<span id="cb338-4"><a href="immune-estimation.html#cb338-4"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">2</span>))</span>
<span id="cb338-5"><a href="immune-estimation.html#cb338-5"></a></span>
<span id="cb338-6"><a href="immune-estimation.html#cb338-6"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(RowNames)){</span>
<span id="cb338-7"><a href="immune-estimation.html#cb338-7"></a>timSamples=output1[<span class="kw">grep</span>(RowNames[i], output1<span class="op">$</span>cell_type), ]</span>
<span id="cb338-8"><a href="immune-estimation.html#cb338-8"></a><span class="kw">rownames</span>(timSamples)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(timSamples<span class="op">$</span>cell_type), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb338-9"><a href="immune-estimation.html#cb338-9"></a><span class="kw">boxplot</span>(<span class="kw">t</span>(timSamples[ ,<span class="op">-</span><span class="dv">1</span>]), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">main=</span>RowNames[i], <span class="dt">ylab=</span>Type[i])</span>
<span id="cb338-10"><a href="immune-estimation.html#cb338-10"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-112-1.png" width="672" /></p>
</div>
<div id="characterisation-cohort-assoc-with-size" class="section level2">
<h2><span class="header-section-number">18.2</span> characterisation cohort: assoc with size</h2>
<p>Associations with size? Perform a correlation test between all of the information above and tumor size.</p>
<p>We obtain a matrix which is colored with a coefficient correlation (red is associative, blue is negatively associated). Correlations which are significant are marked with an asterisk:</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="immune-estimation.html#cb339-1"></a>sizeInfo=infoTableFinal<span class="op">$</span>TumSize[<span class="kw">match</span>(<span class="kw">colnames</span>(CharSpecCD45)[<span class="op">-</span><span class="dv">1</span>], <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb339-2"><a href="immune-estimation.html#cb339-2"></a></span>
<span id="cb339-3"><a href="immune-estimation.html#cb339-3"></a>CorVals=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(CharSpecCD45))</span>
<span id="cb339-4"><a href="immune-estimation.html#cb339-4"></a><span class="kw">names</span>(CorVals)=CharSpecCD45<span class="op">$</span>cell_type</span>
<span id="cb339-5"><a href="immune-estimation.html#cb339-5"></a>CorValsP=CorVals</span>
<span id="cb339-6"><a href="immune-estimation.html#cb339-6"></a></span>
<span id="cb339-7"><a href="immune-estimation.html#cb339-7"></a>CorVals=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(CharSpecCD45), <span class="cf">function</span>(x) <span class="kw">cor</span>(<span class="kw">t</span>(CharSpecCD45[x, <span class="dv">-1</span>]), sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>))</span>
<span id="cb339-8"><a href="immune-estimation.html#cb339-8"></a>CorValsP=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(CharSpecCD45), <span class="cf">function</span>(x) <span class="kw">cor.test</span>(<span class="kw">t</span>(CharSpecCD45[x, <span class="dv">-1</span>]), sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb339-9"><a href="immune-estimation.html#cb339-9"></a></span>
<span id="cb339-10"><a href="immune-estimation.html#cb339-10"></a>naidx=<span class="kw">which</span>(CorValsP<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb339-11"><a href="immune-estimation.html#cb339-11"></a></span>
<span id="cb339-12"><a href="immune-estimation.html#cb339-12"></a>RNames1=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(CharSpecCD45<span class="op">$</span>cell_type, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb339-13"><a href="immune-estimation.html#cb339-13"></a>RNamesMethod=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(CharSpecCD45<span class="op">$</span>cell_type, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb339-14"><a href="immune-estimation.html#cb339-14"></a></span>
<span id="cb339-15"><a href="immune-estimation.html#cb339-15"></a></span>
<span id="cb339-16"><a href="immune-estimation.html#cb339-16"></a>df2=<span class="kw">data.frame</span>(<span class="dt">cell=</span>RNames1, <span class="dt">method=</span>RNamesMethod, <span class="dt">cor=</span>CorVals, <span class="dt">p=</span>CorValsP)</span>
<span id="cb339-17"><a href="immune-estimation.html#cb339-17"></a>ax1=<span class="kw">acast</span>(df2[ ,<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)], RNames1<span class="op">~</span>RNamesMethod)</span>
<span id="cb339-18"><a href="immune-estimation.html#cb339-18"></a></span>
<span id="cb339-19"><a href="immune-estimation.html#cb339-19"></a><span class="co">#df2=data.frame(cell=RNames1, method=RNamesMethod, cor=CorValsP)</span></span>
<span id="cb339-20"><a href="immune-estimation.html#cb339-20"></a>ax2=<span class="kw">acast</span>(df2[ ,<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, <span class="dv">4</span>)], RNames1<span class="op">~</span>RNamesMethod)</span>
<span id="cb339-21"><a href="immune-estimation.html#cb339-21"></a>ax2[<span class="kw">which</span>(ax2<span class="op">&gt;</span><span class="fl">0.1</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">0</span></span>
<span id="cb339-22"><a href="immune-estimation.html#cb339-22"></a>ax2[<span class="kw">which</span>(<span class="kw">is.na</span>(ax2), <span class="dt">arr.ind =</span> T)]=<span class="dv">0</span></span>
<span id="cb339-23"><a href="immune-estimation.html#cb339-23"></a></span>
<span id="cb339-24"><a href="immune-estimation.html#cb339-24"></a><span class="co">#par(oma=c(3, 5, 2, 2))</span></span>
<span id="cb339-25"><a href="immune-estimation.html#cb339-25"></a><span class="co">#OutputplotFun(ax1, scaleR=&quot;none&quot;, main=&quot;cell type correlation&quot;, classN=&quot;no&quot;, sigMat=ax2)</span></span>
<span id="cb339-26"><a href="immune-estimation.html#cb339-26"></a></span>
<span id="cb339-27"><a href="immune-estimation.html#cb339-27"></a><span class="co">## Also plot these all separately</span></span>
<span id="cb339-28"><a href="immune-estimation.html#cb339-28"></a>sizeInfoCut=infoTableFinal<span class="op">$</span>SizeCat[<span class="kw">match</span>(<span class="kw">colnames</span>(CharSpecCD45)[<span class="op">-</span><span class="dv">1</span>], <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb339-29"><a href="immune-estimation.html#cb339-29"></a></span>
<span id="cb339-30"><a href="immune-estimation.html#cb339-30"></a>ttestVal=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(CharSpecCD45), <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(<span class="kw">t</span>(CharSpecCD45[x, <span class="dv">-1</span>])<span class="op">~</span>sizeInfoCut)<span class="op">$</span>p.value)</span>
<span id="cb339-31"><a href="immune-estimation.html#cb339-31"></a>df3=<span class="kw">data.frame</span>(<span class="dt">cell=</span>RNames1, <span class="dt">method=</span>RNamesMethod, <span class="dt">cor=</span>ttestVal)</span>
<span id="cb339-32"><a href="immune-estimation.html#cb339-32"></a>ax3=<span class="kw">acast</span>(df3, RNames1<span class="op">~</span>RNamesMethod)</span>
<span id="cb339-33"><a href="immune-estimation.html#cb339-33"></a>ax3[<span class="kw">which</span>(ax3<span class="op">&gt;</span><span class="fl">0.1</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">0</span></span>
<span id="cb339-34"><a href="immune-estimation.html#cb339-34"></a>ax3[<span class="kw">which</span>(<span class="kw">is.na</span>(ax3), <span class="dt">arr.ind =</span> T)]=<span class="dv">0</span></span></code></pre></div>
<p>The above matrices are sparse, and we can zoom in on specific methods to see whether there is an association</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="immune-estimation.html#cb340-1"></a><span class="co">#pdf(&quot;~/Desktop/2F_characterisation_association_size_immune_types.pdf&quot;, width=8, height=3)</span></span>
<span id="cb340-2"><a href="immune-estimation.html#cb340-2"></a></span>
<span id="cb340-3"><a href="immune-estimation.html#cb340-3"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb340-4"><a href="immune-estimation.html#cb340-4"></a></span>
<span id="cb340-5"><a href="immune-estimation.html#cb340-5"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(ax1)){</span>
<span id="cb340-6"><a href="immune-estimation.html#cb340-6"></a>  t1=ax1[, i]</span>
<span id="cb340-7"><a href="immune-estimation.html#cb340-7"></a>  t2=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(t1))</span>
<span id="cb340-8"><a href="immune-estimation.html#cb340-8"></a>  <span class="kw">image</span>(<span class="kw">cbind</span>(ax1[t2, i],ax1[t2, i]), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">xaxt=</span><span class="st">&quot;none&quot;</span>, <span class="dt">yaxt=</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb340-9"><a href="immune-estimation.html#cb340-9"></a>        <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;correlation %s (* sig p&lt;0.1)&quot;</span>, <span class="kw">colnames</span>(ax1)[i]))</span>
<span id="cb340-10"><a href="immune-estimation.html#cb340-10"></a>  <span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">length</span>(t2)), <span class="kw">names</span>(t1)[t2], <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex=</span><span class="fl">0.7</span>)</span>
<span id="cb340-11"><a href="immune-estimation.html#cb340-11"></a>  <span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="kw">c</span>(<span class="st">&quot;cor p&quot;</span>, <span class="st">&quot;wilcox p&quot;</span>), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex=</span><span class="fl">0.7</span>)</span>
<span id="cb340-12"><a href="immune-estimation.html#cb340-12"></a>  mx=<span class="kw">which</span>(ax2[ t2,i]<span class="op">&gt;</span><span class="dv">0</span>)</span>
<span id="cb340-13"><a href="immune-estimation.html#cb340-13"></a>  <span class="kw">text</span>((mx<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">length</span>(t2)<span class="op">-</span><span class="dv">1</span>), <span class="dv">0</span> , <span class="st">&quot;*&quot;</span>)</span>
<span id="cb340-14"><a href="immune-estimation.html#cb340-14"></a>  mx=<span class="kw">which</span>(ax3[ t2,i]<span class="op">&gt;</span><span class="dv">0</span>)</span>
<span id="cb340-15"><a href="immune-estimation.html#cb340-15"></a>  <span class="kw">text</span>((mx<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">length</span>(t2)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span> , <span class="st">&quot;*&quot;</span>)</span>
<span id="cb340-16"><a href="immune-estimation.html#cb340-16"></a> }</span></code></pre></div>
<div class="figure"><span id="fig:2f-1"></span>
<img src="RatDCIS-bookdown_files/figure-html/2f-1.png" alt="correlation coefficient values" width="672" />
<p class="caption">
Figure 18.1: correlation coefficient values
</p>
</div>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="immune-estimation.html#cb341-1"></a></span>
<span id="cb341-2"><a href="immune-estimation.html#cb341-2"></a><span class="kw">write.csv</span>(df2, <span class="dt">file=</span><span class="st">&quot;nature-tables/2f_correlation_coefficients_pvalues.csv&quot;</span>)</span>
<span id="cb341-3"><a href="immune-estimation.html#cb341-3"></a><span class="co">#write.csv(ax1, file=&quot;nature-tables/2f_correlation_coefficients.csv&quot;)</span></span>
<span id="cb341-4"><a href="immune-estimation.html#cb341-4"></a></span>
<span id="cb341-5"><a href="immune-estimation.html#cb341-5"></a><span class="co">#dev.off()</span></span></code></pre></div>
<div class="figure"><span id="fig:2f-2"></span>
<img src="RatDCIS-bookdown_files/figure-html/2f-2.png" alt="correlation coefficient values" width="672" />
<p class="caption">
Figure 18.2: correlation coefficient values
</p>
</div>
<p>These accompany the following plots with individual samples:</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="immune-estimation.html#cb342-1"></a></span>
<span id="cb342-2"><a href="immune-estimation.html#cb342-2"></a><span class="co">#pdf(&quot;~/Desktop/2F-sample-celltypes.pdf&quot;, height=6, width=7)</span></span>
<span id="cb342-3"><a href="immune-estimation.html#cb342-3"></a></span>
<span id="cb342-4"><a href="immune-estimation.html#cb342-4"></a>x2=<span class="kw">c</span>(<span class="kw">colnames</span>(CharSpecCD45)[<span class="kw">order</span>(sizeInfo)<span class="op">+</span><span class="dv">1</span>])</span>
<span id="cb342-5"><a href="immune-estimation.html#cb342-5"></a></span>
<span id="cb342-6"><a href="immune-estimation.html#cb342-6"></a>ax1=<span class="kw">grep</span>(<span class="st">&quot;TIMER&quot;</span>, CharSpecCD45<span class="op">$</span>cell_type)</span>
<span id="cb342-7"><a href="immune-estimation.html#cb342-7"></a>T1melt=<span class="kw">melt</span>(CharSpecCD45[ax1, ])</span>
<span id="cb342-8"><a href="immune-estimation.html#cb342-8"></a>T1melt<span class="op">$</span>variable=<span class="kw">factor</span>(T1melt<span class="op">$</span>variable, x2)</span>
<span id="cb342-9"><a href="immune-estimation.html#cb342-9"></a>T1melt<span class="op">$</span>NewID=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(T1melt<span class="op">$</span>variable, <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb342-10"><a href="immune-estimation.html#cb342-10"></a></span>
<span id="cb342-11"><a href="immune-estimation.html#cb342-11"></a></span>
<span id="cb342-12"><a href="immune-estimation.html#cb342-12"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>NewID, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span></span>
<span id="cb342-13"><a href="immune-estimation.html#cb342-13"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;TIMER:char cohort separately&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span></span>
<span id="cb342-14"><a href="immune-estimation.html#cb342-14"></a><span class="st">   </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#66C2A5&quot;</span>, <span class="st">&quot;#FED976&quot;</span>, <span class="st">&quot;#FEEBE2&quot;</span> ,<span class="st">&quot;#AE017E&quot;</span>, <span class="st">&quot;#2171B5&quot;</span>,   <span class="st">&quot;#BDD7E7&quot;</span>, <span class="st">&quot;#EFF3FF&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<div class="figure"><span id="fig:2fplot"></span>
<img src="RatDCIS-bookdown_files/figure-html/2fplot-1.png" alt="TIMER" width="672" />
<p class="caption">
Figure 18.3: TIMER
</p>
</div>
<p>And below is the result for CIBERSORT</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="immune-estimation.html#cb343-1"></a>ax1=<span class="kw">grep</span>(<span class="st">&quot;CIBERSORT$&quot;</span>, CharSpecCD45<span class="op">$</span>cell_type)</span>
<span id="cb343-2"><a href="immune-estimation.html#cb343-2"></a>T1melt=<span class="kw">melt</span>(CharSpecCD45[ax1, ])</span>
<span id="cb343-3"><a href="immune-estimation.html#cb343-3"></a></span>
<span id="cb343-4"><a href="immune-estimation.html#cb343-4"></a>T1melt<span class="op">$</span>variable=<span class="kw">factor</span>(T1melt<span class="op">$</span>variable, x2)</span>
<span id="cb343-5"><a href="immune-estimation.html#cb343-5"></a></span>
<span id="cb343-6"><a href="immune-estimation.html#cb343-6"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span></span>
<span id="cb343-7"><a href="immune-estimation.html#cb343-7"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;CIBSERT:char cohort separately&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span></span>
<span id="cb343-8"><a href="immune-estimation.html#cb343-8"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/2fagain-1.png" width="672" /></p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="immune-estimation.html#cb344-1"></a></span>
<span id="cb344-2"><a href="immune-estimation.html#cb344-2"></a><span class="co"># Also do a version where the Bcells, Macrophages,NK, Mast Cells, CD4, CD8 Cells are merged together, NK</span></span>
<span id="cb344-3"><a href="immune-estimation.html#cb344-3"></a></span>
<span id="cb344-4"><a href="immune-estimation.html#cb344-4"></a>T1melt<span class="op">$</span>cell_type2=<span class="kw">substr</span>(T1melt<span class="op">$</span>cell_type, <span class="dv">1</span>, <span class="dv">4</span>)</span>
<span id="cb344-5"><a href="immune-estimation.html#cb344-5"></a>T1melt<span class="op">$</span>cell_type2[<span class="kw">grep</span>(<span class="st">&quot;CD8&quot;</span>, T1melt<span class="op">$</span>cell_type)]=<span class="st">&quot;T CD8&quot;</span></span>
<span id="cb344-6"><a href="immune-estimation.html#cb344-6"></a>T1melt<span class="op">$</span>cell_type2[<span class="kw">grep</span>(<span class="st">&quot;CD4&quot;</span>, T1melt<span class="op">$</span>cell_type)]=<span class="st">&quot;T CD4&quot;</span></span>
<span id="cb344-7"><a href="immune-estimation.html#cb344-7"></a></span>
<span id="cb344-8"><a href="immune-estimation.html#cb344-8"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type2))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span></span>
<span id="cb344-9"><a href="immune-estimation.html#cb344-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;CIBSERT:char cohort separately&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span></span>
<span id="cb344-10"><a href="immune-estimation.html#cb344-10"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#66C2A5&quot;</span>, <span class="st">&quot;#E6F598&quot;</span>, <span class="st">&quot;#FED976&quot;</span>, <span class="st">&quot;#FC4E2A&quot;</span>, <span class="st">&quot;#bdbdbd&quot;</span>, <span class="st">&quot;#FEEBE2&quot;</span> ,<span class="st">&quot;#AE017E&quot;</span>, <span class="st">&quot;#5E4FA2&quot;</span>, <span class="st">&quot;#2171B5&quot;</span>, <span class="st">&quot;#BDD7E7&quot;</span>, <span class="st">&quot;#EFF3FF&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/2fagain-2.png" width="672" /></p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="immune-estimation.html#cb345-1"></a></span>
<span id="cb345-2"><a href="immune-estimation.html#cb345-2"></a>T1melt<span class="op">$</span>sdat=<span class="kw">paste</span>(T1melt<span class="op">$</span>variable, T1melt<span class="op">$</span>cell_type2, <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb345-3"><a href="immune-estimation.html#cb345-3"></a>T2=<span class="kw">by</span>(T1melt<span class="op">$</span>value, T1melt<span class="op">$</span>sdat, sum)</span>
<span id="cb345-4"><a href="immune-estimation.html#cb345-4"></a>T2m=<span class="kw">stack</span>(T2)</span>
<span id="cb345-5"><a href="immune-estimation.html#cb345-5"></a>T2m<span class="op">$</span>ind1=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(T2m<span class="op">$</span>ind), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb345-6"><a href="immune-estimation.html#cb345-6"></a>T2m<span class="op">$</span>ind2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(T2m<span class="op">$</span>ind), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb345-7"><a href="immune-estimation.html#cb345-7"></a></span>
<span id="cb345-8"><a href="immune-estimation.html#cb345-8"></a>T2mTab=<span class="kw">acast</span>(T2m[ ,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], ind2<span class="op">~</span>ind1, <span class="dt">value.var=</span><span class="st">&quot;values&quot;</span>)</span>
<span id="cb345-9"><a href="immune-estimation.html#cb345-9"></a>CorV=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(T2mTab), <span class="cf">function</span>(x) <span class="kw">cor</span>(T2mTab[x, ], sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>))</span>
<span id="cb345-10"><a href="immune-estimation.html#cb345-10"></a>CorVP=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(T2mTab), <span class="cf">function</span>(x) <span class="kw">cor.test</span>(T2mTab[x, ], sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb345-11"><a href="immune-estimation.html#cb345-11"></a>wilP=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(T2mTab), <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(T2mTab[x, ]<span class="op">~</span><span class="st"> </span>sizeInfoCut)<span class="op">$</span>p.value)</span>
<span id="cb345-12"><a href="immune-estimation.html#cb345-12"></a></span>
<span id="cb345-13"><a href="immune-estimation.html#cb345-13"></a><span class="co"># dev.off()</span></span>
<span id="cb345-14"><a href="immune-estimation.html#cb345-14"></a></span>
<span id="cb345-15"><a href="immune-estimation.html#cb345-15"></a><span class="kw">write.csv</span>(T1melt, <span class="dt">file=</span><span class="st">&quot;nature-tables/2f.csv&quot;</span>)  </span>
<span id="cb345-16"><a href="immune-estimation.html#cb345-16"></a>  </span></code></pre></div>
</div>
<div id="comparison-with-facs-data" class="section level2">
<h2><span class="header-section-number">18.3</span> Comparison with FACS data</h2>
<p>In this section, we compare how well the estimates from RNAseq deconvolution methods associate with FACS data. Below is a heatmap showing the correlation coefficient of each cell type (by FACS) and the method,</p>
<p>Note that there are twice as manay samples with reliable myeloid derived cell information than for leukocytes.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="immune-estimation.html#cb346-1"></a><span class="co">## annotate the FACS data here</span></span>
<span id="cb346-2"><a href="immune-estimation.html#cb346-2"></a></span>
<span id="cb346-3"><a href="immune-estimation.html#cb346-3"></a>infoTableFinal<span class="op">$</span>Name2=<span class="ot">NA</span></span>
<span id="cb346-4"><a href="immune-estimation.html#cb346-4"></a>infoTableFinal<span class="op">$</span>Name2[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span>)]=<span class="kw">paste</span>(infoTableFinal<span class="op">$</span>Rat_ID[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span>)], infoTableFinal<span class="op">$</span>Location[<span class="kw">which</span>(infoTableFinal<span class="op">$</span>Fraction<span class="op">==</span><span class="st">&quot;CD45&quot;</span>)], <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb346-5"><a href="immune-estimation.html#cb346-5"></a>m2=<span class="kw">match</span>(<span class="kw">colnames</span>(Fdata), infoTableFinal<span class="op">$</span>Name2)</span>
<span id="cb346-6"><a href="immune-estimation.html#cb346-6"></a></span>
<span id="cb346-7"><a href="immune-estimation.html#cb346-7"></a><span class="co">## get rid of the NA samples</span></span>
<span id="cb346-8"><a href="immune-estimation.html#cb346-8"></a>naom=infoTableFinal<span class="op">$</span>SampleID[m2[<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(m2))]]</span>
<span id="cb346-9"><a href="immune-estimation.html#cb346-9"></a>mid=<span class="kw">colnames</span>(Fdata)[<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(m2))]</span>
<span id="cb346-10"><a href="immune-estimation.html#cb346-10"></a></span>
<span id="cb346-11"><a href="immune-estimation.html#cb346-11"></a>lx1=Fdata[, <span class="kw">c</span>(<span class="dv">1</span>, <span class="kw">match</span>(mid, <span class="kw">colnames</span>(Fdata)))]</span>
<span id="cb346-12"><a href="immune-estimation.html#cb346-12"></a>out2=output1[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="kw">match</span>(naom, <span class="kw">colnames</span>(output1)))]</span>
<span id="cb346-13"><a href="immune-estimation.html#cb346-13"></a></span>
<span id="cb346-14"><a href="immune-estimation.html#cb346-14"></a><span class="co">#head(out2)</span></span>
<span id="cb346-15"><a href="immune-estimation.html#cb346-15"></a><span class="kw">colnames</span>(lx1)=<span class="kw">colnames</span>(out2)</span>
<span id="cb346-16"><a href="immune-estimation.html#cb346-16"></a></span>
<span id="cb346-17"><a href="immune-estimation.html#cb346-17"></a><span class="co"># Run all the association tests here</span></span>
<span id="cb346-18"><a href="immune-estimation.html#cb346-18"></a><span class="co">## New Table</span></span>
<span id="cb346-19"><a href="immune-estimation.html#cb346-19"></a><span class="co"># -cd8</span></span>
<span id="cb346-20"><a href="immune-estimation.html#cb346-20"></a><span class="co"># Th</span></span>
<span id="cb346-21"><a href="immune-estimation.html#cb346-21"></a><span class="co"># Tregs</span></span>
<span id="cb346-22"><a href="immune-estimation.html#cb346-22"></a><span class="co"># B cells</span></span>
<span id="cb346-23"><a href="immune-estimation.html#cb346-23"></a><span class="co"># Macrophage</span></span>
<span id="cb346-24"><a href="immune-estimation.html#cb346-24"></a></span>
<span id="cb346-25"><a href="immune-estimation.html#cb346-25"></a><span class="co"># </span></span>
<span id="cb346-26"><a href="immune-estimation.html#cb346-26"></a>MergedTable=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">ncol=</span><span class="dv">21</span>, <span class="dt">nrow=</span><span class="dv">1</span>)</span>
<span id="cb346-27"><a href="immune-estimation.html#cb346-27"></a><span class="kw">colnames</span>(MergedTable)=<span class="kw">colnames</span>(out2)</span>
<span id="cb346-28"><a href="immune-estimation.html#cb346-28"></a></span>
<span id="cb346-29"><a href="immune-estimation.html#cb346-29"></a>tx1=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(out2<span class="op">$</span>cell_type), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb346-30"><a href="immune-estimation.html#cb346-30"></a>MethodSumm=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(out2<span class="op">$</span>cell_type), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb346-31"><a href="immune-estimation.html#cb346-31"></a>tabtx1=<span class="kw">table</span>(tx1)</span>
<span id="cb346-32"><a href="immune-estimation.html#cb346-32"></a>TheseFracs=<span class="kw">names</span>(tabtx1)[<span class="kw">which</span>(tabtx1<span class="op">&gt;</span><span class="dv">3</span>)]</span>
<span id="cb346-33"><a href="immune-estimation.html#cb346-33"></a></span>
<span id="cb346-34"><a href="immune-estimation.html#cb346-34"></a></span>
<span id="cb346-35"><a href="immune-estimation.html#cb346-35"></a></span>
<span id="cb346-36"><a href="immune-estimation.html#cb346-36"></a>testSetB=<span class="kw">c</span>(<span class="st">&quot;B cell&quot;</span>, <span class="st">&quot;MHCII-hi&quot;</span>, <span class="st">&quot;MHCII-lo&quot;</span>, <span class="st">&quot;Monocyte&quot;</span>, <span class="st">&quot;Neutrophil&quot;</span>, <span class="st">&quot;NK cells&quot;</span>, <span class="st">&quot;CD8&quot;</span>, <span class="st">&quot;Treg&quot;</span>)</span>
<span id="cb346-37"><a href="immune-estimation.html#cb346-37"></a><span class="co">#  &quot;CD8&quot;, &quot;Th&quot;, &quot;Treg&quot;, &quot;B cells&quot;, &quot;NK cells&quot;, &quot;DC&quot;, &quot;Neutrophil&quot;, &quot;Monocyte&quot;, &quot;gd T&quot;, &quot;MHCII-hi&quot;, &quot;MHCII-lo&quot;)</span></span>
<span id="cb346-38"><a href="immune-estimation.html#cb346-38"></a>testSet=TheseFracs[<span class="op">-</span><span class="dv">2</span>]<span class="co">#c(&quot;CD8&quot;, &quot;CD4&quot;, &quot;Treg&quot;, &quot;B cell&quot;, &quot;NK&quot;, &quot;dendritic&quot;, &quot;Neutrophil&quot;, &quot;Monocyte&quot;, &quot;gamma delta&quot;, &quot;M1&quot;, &quot;M2&quot;)</span></span>
<span id="cb346-39"><a href="immune-estimation.html#cb346-39"></a> </span>
<span id="cb346-40"><a href="immune-estimation.html#cb346-40"></a>CMat=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="kw">length</span>(testSet), <span class="dt">ncol=</span><span class="kw">length</span>(<span class="kw">unique</span>(MethodSumm)))</span>
<span id="cb346-41"><a href="immune-estimation.html#cb346-41"></a><span class="kw">rownames</span>(CMat)=testSet</span>
<span id="cb346-42"><a href="immune-estimation.html#cb346-42"></a><span class="kw">colnames</span>(CMat)=<span class="kw">unique</span>(MethodSumm)</span>
<span id="cb346-43"><a href="immune-estimation.html#cb346-43"></a> </span>
<span id="cb346-44"><a href="immune-estimation.html#cb346-44"></a>PMat=CMat</span>
<span id="cb346-45"><a href="immune-estimation.html#cb346-45"></a><span class="co">#testSet=&quot;M2&quot;</span></span>
<span id="cb346-46"><a href="immune-estimation.html#cb346-46"></a><span class="co">#testSetB=&quot;MHCII-lo&quot;</span></span>
<span id="cb346-47"><a href="immune-estimation.html#cb346-47"></a></span>
<span id="cb346-48"><a href="immune-estimation.html#cb346-48"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(testSet)){</span>
<span id="cb346-49"><a href="immune-estimation.html#cb346-49"></a>CD8Table=<span class="kw">rbind</span>(lx1[<span class="kw">grep</span>(testSetB[j], lx1<span class="op">$</span>cell_type), ],</span>
<span id="cb346-50"><a href="immune-estimation.html#cb346-50"></a>               out2[<span class="kw">which</span>(tx1<span class="op">==</span>testSet[j]), ])</span>
<span id="cb346-51"><a href="immune-estimation.html#cb346-51"></a>CD8Table[<span class="dv">1</span>,<span class="dv">1</span>]=<span class="kw">paste</span>(testSetB[j], <span class="st">&quot;facs&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb346-52"><a href="immune-estimation.html#cb346-52"></a>ms=MethodSumm[<span class="kw">which</span>(tx1<span class="op">==</span>testSet[j])]</span>
<span id="cb346-53"><a href="immune-estimation.html#cb346-53"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb346-54"><a href="immune-estimation.html#cb346-54"></a></span>
<span id="cb346-55"><a href="immune-estimation.html#cb346-55"></a></span>
<span id="cb346-56"><a href="immune-estimation.html#cb346-56"></a>cVals=<span class="kw">sapply</span>(<span class="dv">2</span><span class="op">:</span><span class="kw">nrow</span>(CD8Table), <span class="cf">function</span>(x) <span class="kw">cor</span>(<span class="kw">t</span>(CD8Table[<span class="dv">1</span>, <span class="dv">-1</span>]), <span class="kw">t</span>(CD8Table[x, <span class="dv">-1</span>]), <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>))</span>
<span id="cb346-57"><a href="immune-estimation.html#cb346-57"></a>cVals2=<span class="kw">sapply</span>(<span class="dv">2</span><span class="op">:</span><span class="kw">nrow</span>(CD8Table), <span class="cf">function</span>(x) <span class="kw">cor.test</span>(<span class="kw">t</span>(CD8Table[<span class="dv">1</span>, <span class="dv">-1</span>]), <span class="kw">t</span>(CD8Table[x, <span class="dv">-1</span>]), <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb346-58"><a href="immune-estimation.html#cb346-58"></a></span>
<span id="cb346-59"><a href="immune-estimation.html#cb346-59"></a>CMat[j, <span class="kw">match</span>(ms, <span class="kw">colnames</span>(CMat))]=cVals</span>
<span id="cb346-60"><a href="immune-estimation.html#cb346-60"></a>PMat[j, <span class="kw">match</span>(ms, <span class="kw">colnames</span>(PMat))]=cVals2</span>
<span id="cb346-61"><a href="immune-estimation.html#cb346-61"></a></span>
<span id="cb346-62"><a href="immune-estimation.html#cb346-62"></a>MergedTable=<span class="kw">rbind</span>(MergedTable, CD8Table)</span>
<span id="cb346-63"><a href="immune-estimation.html#cb346-63"></a></span>
<span id="cb346-64"><a href="immune-estimation.html#cb346-64"></a><span class="co">#for (i in 2:nrow(CD8Table)){</span></span>
<span id="cb346-65"><a href="immune-estimation.html#cb346-65"></a>  <span class="co">#a1=cor(t(CD8Table[1, -1]), t(CD8Table[i, -1]), use=&quot;complete&quot;)</span></span>
<span id="cb346-66"><a href="immune-estimation.html#cb346-66"></a>  <span class="co">#plot(t(CD8Table[1, -1]), t(CD8Table[i, -1]), main=CD8Table[i,1], ylab=&quot;method&quot;, xlab=&quot;FACS&quot;)</span></span>
<span id="cb346-67"><a href="immune-estimation.html#cb346-67"></a>  <span class="co">#text(min(t(CD8Table[1, -1]), na.rm=T)*2, max(t(CD8Table[i, -1]), na.rm=T), paste(&quot;r=&quot;, round(a1, digits=2), sep=&quot;&quot;))</span></span>
<span id="cb346-68"><a href="immune-estimation.html#cb346-68"></a>  </span>
<span id="cb346-69"><a href="immune-estimation.html#cb346-69"></a>}</span>
<span id="cb346-70"><a href="immune-estimation.html#cb346-70"></a></span>
<span id="cb346-71"><a href="immune-estimation.html#cb346-71"></a></span>
<span id="cb346-72"><a href="immune-estimation.html#cb346-72"></a><span class="co">##for the following, find the terms and calculate the sum</span></span>
<span id="cb346-73"><a href="immune-estimation.html#cb346-73"></a></span>
<span id="cb346-74"><a href="immune-estimation.html#cb346-74"></a>testSetB=<span class="kw">c</span>(<span class="st">&quot;Macro&quot;</span>, <span class="st">&quot;Th&quot;</span>, <span class="st">&quot;B cell&quot;</span>)</span>
<span id="cb346-75"><a href="immune-estimation.html#cb346-75"></a>testSet=<span class="kw">c</span>(<span class="st">&quot;Macro&quot;</span>, <span class="st">&quot;CD4&quot;</span>, <span class="st">&quot;B cell&quot;</span>)</span>
<span id="cb346-76"><a href="immune-estimation.html#cb346-76"></a>rmThese=<span class="kw">c</span>(<span class="st">&quot;M0&quot;</span>, <span class="st">&quot;naive&quot;</span>, <span class="ot">NULL</span>)</span>
<span id="cb346-77"><a href="immune-estimation.html#cb346-77"></a></span>
<span id="cb346-78"><a href="immune-estimation.html#cb346-78"></a>savTemp=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">3</span>, <span class="dt">ncol=</span><span class="kw">ncol</span>(CMat))</span>
<span id="cb346-79"><a href="immune-estimation.html#cb346-79"></a><span class="kw">rownames</span>(savTemp)=<span class="kw">paste</span>(<span class="st">&quot;all&quot;</span>, testSet, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb346-80"><a href="immune-estimation.html#cb346-80"></a>savTempP=savTemp</span>
<span id="cb346-81"><a href="immune-estimation.html#cb346-81"></a></span>
<span id="cb346-82"><a href="immune-estimation.html#cb346-82"></a></span>
<span id="cb346-83"><a href="immune-estimation.html#cb346-83"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(testSetB)){</span>
<span id="cb346-84"><a href="immune-estimation.html#cb346-84"></a>  </span>
<span id="cb346-85"><a href="immune-estimation.html#cb346-85"></a>  CD8Table=lx1[<span class="kw">grep</span>(testSetB[j], lx1<span class="op">$</span>cell_type), ]</span>
<span id="cb346-86"><a href="immune-estimation.html#cb346-86"></a>  CD8Table=<span class="kw">colSums</span>(CD8Table[, <span class="dv">-1</span>])</span>
<span id="cb346-87"><a href="immune-estimation.html#cb346-87"></a>  <span class="co">#rownames(CD8Table)=&quot;facs&quot;</span></span>
<span id="cb346-88"><a href="immune-estimation.html#cb346-88"></a>  </span>
<span id="cb346-89"><a href="immune-estimation.html#cb346-89"></a>  outB=out2[<span class="kw">grep</span>(testSet[j], out2<span class="op">$</span>cell_type), ]</span>
<span id="cb346-90"><a href="immune-estimation.html#cb346-90"></a>  rm2=<span class="kw">grep</span>(rmThese[j], outB<span class="op">$</span>cell_type)</span>
<span id="cb346-91"><a href="immune-estimation.html#cb346-91"></a>  </span>
<span id="cb346-92"><a href="immune-estimation.html#cb346-92"></a>  <span class="cf">if</span> (<span class="kw">length</span>(<span class="kw">na.omit</span>(rm2))<span class="op">&gt;</span><span class="dv">0</span>){</span>
<span id="cb346-93"><a href="immune-estimation.html#cb346-93"></a>    outB=outB[<span class="op">-</span>rm2, ]</span>
<span id="cb346-94"><a href="immune-estimation.html#cb346-94"></a>  }</span>
<span id="cb346-95"><a href="immune-estimation.html#cb346-95"></a>  </span>
<span id="cb346-96"><a href="immune-estimation.html#cb346-96"></a>  namOut=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(outB<span class="op">$</span>cell_type), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb346-97"><a href="immune-estimation.html#cb346-97"></a>  nam2=<span class="kw">unique</span>(namOut[<span class="kw">which</span>(<span class="kw">duplicated</span>(namOut))])</span>
<span id="cb346-98"><a href="immune-estimation.html#cb346-98"></a>  </span>
<span id="cb346-99"><a href="immune-estimation.html#cb346-99"></a>  outC=<span class="kw">sapply</span>(nam2, <span class="cf">function</span>(x) <span class="kw">colSums</span>(outB[<span class="kw">which</span>(namOut<span class="op">==</span>x), <span class="dv">-1</span> ]))</span>
<span id="cb346-100"><a href="immune-estimation.html#cb346-100"></a>  outB=outB[<span class="op">-</span><span class="kw">which</span>(namOut<span class="op">%in%</span>nam2), ]</span>
<span id="cb346-101"><a href="immune-estimation.html#cb346-101"></a>  <span class="kw">rownames</span>(outB)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(outB<span class="op">$</span>cell_type), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb346-102"><a href="immune-estimation.html#cb346-102"></a>  </span>
<span id="cb346-103"><a href="immune-estimation.html#cb346-103"></a>  allD=<span class="kw">rbind</span>(CD8Table, outB[, <span class="dv">-1</span>], <span class="kw">t</span>(outC))</span>
<span id="cb346-104"><a href="immune-estimation.html#cb346-104"></a>  <span class="kw">rownames</span>(allD)[<span class="dv">1</span>]=<span class="st">&quot;facs&quot;</span></span>
<span id="cb346-105"><a href="immune-estimation.html#cb346-105"></a>  allD<span class="op">$</span>method=<span class="kw">rownames</span>(allD)</span>
<span id="cb346-106"><a href="immune-estimation.html#cb346-106"></a>  </span>
<span id="cb346-107"><a href="immune-estimation.html#cb346-107"></a> <span class="co"># allD=data.matrix(allD)</span></span>
<span id="cb346-108"><a href="immune-estimation.html#cb346-108"></a>  </span>
<span id="cb346-109"><a href="immune-estimation.html#cb346-109"></a><span class="co">#pdf(sprintf(&quot;rslt/Immune decomposition/correlations_combined_%s.pdf&quot;, testSetB[j]), width=10, height=10)</span></span>
<span id="cb346-110"><a href="immune-estimation.html#cb346-110"></a></span>
<span id="cb346-111"><a href="immune-estimation.html#cb346-111"></a><span class="co">#ggplot(temp8, aes(x=method, y=value, col=variable))+geom_bar(stat=&quot;identity&quot;)+facet_wrap(~variable, scale=&quot;free_y&quot;)+theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</span></span>
<span id="cb346-112"><a href="immune-estimation.html#cb346-112"></a></span>
<span id="cb346-113"><a href="immune-estimation.html#cb346-113"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb346-114"><a href="immune-estimation.html#cb346-114"></a></span>
<span id="cb346-115"><a href="immune-estimation.html#cb346-115"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">nrow</span>(allD)){</span>
<span id="cb346-116"><a href="immune-estimation.html#cb346-116"></a>  a1=<span class="kw">cor</span>(<span class="kw">t</span>(allD[<span class="dv">1</span>, <span class="op">-</span><span class="kw">ncol</span>(allD)]), <span class="kw">t</span>(allD[i, <span class="op">-</span><span class="kw">ncol</span>(allD)]), <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb346-117"><a href="immune-estimation.html#cb346-117"></a>  a2=<span class="kw">cor.test</span>(<span class="kw">t</span>(allD[<span class="dv">1</span>, <span class="op">-</span><span class="kw">ncol</span>(allD)]), <span class="kw">t</span>(allD[i, <span class="op">-</span><span class="kw">ncol</span>(allD)]), <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)<span class="op">$</span>p.value</span>
<span id="cb346-118"><a href="immune-estimation.html#cb346-118"></a>  savTemp[j, <span class="kw">match</span>(<span class="kw">rownames</span>(allD)[i], <span class="kw">colnames</span>(CMat))]=a1</span>
<span id="cb346-119"><a href="immune-estimation.html#cb346-119"></a>  savTempP[j, <span class="kw">match</span>(<span class="kw">rownames</span>(allD)[i], <span class="kw">colnames</span>(CMat))]=a2</span>
<span id="cb346-120"><a href="immune-estimation.html#cb346-120"></a><span class="co"># plot(t(allD[1, -ncol(allD)]), t(allD[i,-ncol(allD)]), main=allD[i,ncol(allD)], ylab=&quot;method&quot;, xlab=&quot;FACS&quot;)</span></span>
<span id="cb346-121"><a href="immune-estimation.html#cb346-121"></a><span class="co">#  text(min(t(allD[1, -ncol(allD)]), na.rm=T)*2, max(t(allD[i, -ncol(allD)]), na.rm=T), paste(&quot;r=&quot;, round(a1, digits=2), sep=&quot;&quot;))</span></span>
<span id="cb346-122"><a href="immune-estimation.html#cb346-122"></a>  </span>
<span id="cb346-123"><a href="immune-estimation.html#cb346-123"></a>}</span>
<span id="cb346-124"><a href="immune-estimation.html#cb346-124"></a></span>
<span id="cb346-125"><a href="immune-estimation.html#cb346-125"></a>cell_type=<span class="kw">paste</span>(testSetB[j], allD<span class="op">$</span>method, <span class="dt">sep=</span><span class="st">&quot;_all_&quot;</span>)</span>
<span id="cb346-126"><a href="immune-estimation.html#cb346-126"></a></span>
<span id="cb346-127"><a href="immune-estimation.html#cb346-127"></a>MergedTable=<span class="kw">rbind</span>(MergedTable, <span class="kw">cbind</span>(cell_type, allD[ ,<span class="op">-</span><span class="dv">21</span>]))</span>
<span id="cb346-128"><a href="immune-estimation.html#cb346-128"></a></span>
<span id="cb346-129"><a href="immune-estimation.html#cb346-129"></a>}</span>
<span id="cb346-130"><a href="immune-estimation.html#cb346-130"></a></span>
<span id="cb346-131"><a href="immune-estimation.html#cb346-131"></a>CorMatAll=<span class="kw">rbind</span>(CMat, savTemp)</span>
<span id="cb346-132"><a href="immune-estimation.html#cb346-132"></a>CorMatP=<span class="kw">rbind</span>(PMat, savTempP)</span>
<span id="cb346-133"><a href="immune-estimation.html#cb346-133"></a></span>
<span id="cb346-134"><a href="immune-estimation.html#cb346-134"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">5</span>))</span>
<span id="cb346-135"><a href="immune-estimation.html#cb346-135"></a><span class="kw">heatmap.2</span>(CorMatAll, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">main=</span><span class="st">&quot;correlation FACS and GE&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/imm-assoc-facs-1.png" width="672" /></p>
<p>We can plot associations between the different cell types below, here selecting:</p>
<ul>
<li>Bcells</li>
<li>CD8 T cells</li>
<li>M2 macophage</li>
<li>M1 macrophage</li>
</ul>
<p>with each method. The correlation coefficient is indicated.</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="immune-estimation.html#cb347-1"></a>testSetB=<span class="kw">c</span>(<span class="st">&quot;CD8&quot;</span>,  <span class="st">&quot;B cells&quot;</span>, <span class="st">&quot;MHCII-hi&quot;</span>, <span class="st">&quot;MHCII-lo&quot;</span>)</span>
<span id="cb347-2"><a href="immune-estimation.html#cb347-2"></a>testSet=<span class="kw">c</span>(<span class="st">&quot;CD8&quot;</span>, <span class="st">&quot;B cell&quot;</span>,<span class="st">&quot;M1&quot;</span>, <span class="st">&quot;M2&quot;</span>)</span>
<span id="cb347-3"><a href="immune-estimation.html#cb347-3"></a></span>
<span id="cb347-4"><a href="immune-estimation.html#cb347-4"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb347-5"><a href="immune-estimation.html#cb347-5"></a></span>
<span id="cb347-6"><a href="immune-estimation.html#cb347-6"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(testSetB)){ </span>
<span id="cb347-7"><a href="immune-estimation.html#cb347-7"></a>CMat=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">ncol=</span><span class="kw">length</span>(testSet), <span class="dt">nrow=</span><span class="dv">10</span>) </span>
<span id="cb347-8"><a href="immune-estimation.html#cb347-8"></a>CD8Table=<span class="kw">rbind</span>(lx1[<span class="kw">grep</span>(testSetB[j], lx1<span class="op">$</span>cell_type), ],</span>
<span id="cb347-9"><a href="immune-estimation.html#cb347-9"></a>               out2[<span class="kw">grep</span>(testSet[j], out2<span class="op">$</span>cell_type), ])</span>
<span id="cb347-10"><a href="immune-estimation.html#cb347-10"></a></span>
<span id="cb347-11"><a href="immune-estimation.html#cb347-11"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">nrow</span>(CD8Table)){</span>
<span id="cb347-12"><a href="immune-estimation.html#cb347-12"></a>  a1=<span class="kw">cor</span>(<span class="kw">t</span>(CD8Table[<span class="dv">1</span>, <span class="dv">-1</span>]), <span class="kw">t</span>(CD8Table[i, <span class="dv">-1</span>]), <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb347-13"><a href="immune-estimation.html#cb347-13"></a>  <span class="kw">plot</span>(<span class="kw">t</span>(CD8Table[<span class="dv">1</span>, <span class="dv">-1</span>]), <span class="kw">t</span>(CD8Table[i, <span class="dv">-1</span>]), <span class="dt">main=</span>CD8Table[i,<span class="dv">1</span>], <span class="dt">ylab=</span><span class="st">&quot;method&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;FACS&quot;</span>)</span>
<span id="cb347-14"><a href="immune-estimation.html#cb347-14"></a>  <span class="kw">text</span>(<span class="kw">min</span>(<span class="kw">t</span>(CD8Table[<span class="dv">1</span>, <span class="dv">-1</span>]), <span class="dt">na.rm=</span>T)<span class="op">*</span><span class="dv">2</span>, <span class="kw">max</span>(<span class="kw">t</span>(CD8Table[i, <span class="dv">-1</span>]), <span class="dt">na.rm=</span>T), <span class="kw">paste</span>(<span class="st">&quot;r=&quot;</span>, <span class="kw">round</span>(a1, <span class="dt">digits=</span><span class="dv">2</span>), <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb347-15"><a href="immune-estimation.html#cb347-15"></a>}</span>
<span id="cb347-16"><a href="immune-estimation.html#cb347-16"></a><span class="kw">mtext</span>(testSetB[j], <span class="dt">side=</span><span class="dv">3</span>, <span class="dt">line=</span><span class="op">-</span><span class="dv">2</span>, <span class="dt">outer=</span>T)</span>
<span id="cb347-17"><a href="immune-estimation.html#cb347-17"></a>  </span>
<span id="cb347-18"><a href="immune-estimation.html#cb347-18"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-114-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-114-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-114-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-114-4.png" width="672" /></p>
</div>
<div id="progression-cohort-1" class="section level2">
<h2><span class="header-section-number">18.4</span> Progression cohort</h2>
<p>We perform the same sort of analysis for the progression cohort:</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="immune-estimation.html#cb348-1"></a><span class="co"># drop samples</span></span>
<span id="cb348-2"><a href="immune-estimation.html#cb348-2"></a><span class="co">#dsamp=c(&quot;2R_D_CD45&quot;, &quot;3L_D_CD45&quot;)</span></span>
<span id="cb348-3"><a href="immune-estimation.html#cb348-3"></a></span>
<span id="cb348-4"><a href="immune-estimation.html#cb348-4"></a></span>
<span id="cb348-5"><a href="immune-estimation.html#cb348-5"></a>sizeInfo=infoTableFinal<span class="op">$</span>GrowthRate[<span class="kw">match</span>(<span class="kw">colnames</span>(ProgSpecCD45)[<span class="op">-</span><span class="dv">1</span>], <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb348-6"><a href="immune-estimation.html#cb348-6"></a></span>
<span id="cb348-7"><a href="immune-estimation.html#cb348-7"></a>CorVals=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(ProgSpecCD45))</span>
<span id="cb348-8"><a href="immune-estimation.html#cb348-8"></a><span class="kw">names</span>(CorVals)=ProgSpecCD45<span class="op">$</span>cell_type</span>
<span id="cb348-9"><a href="immune-estimation.html#cb348-9"></a>CorValsP=CorVals</span>
<span id="cb348-10"><a href="immune-estimation.html#cb348-10"></a></span>
<span id="cb348-11"><a href="immune-estimation.html#cb348-11"></a>CorVals=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(ProgSpecCD45), <span class="cf">function</span>(x) <span class="kw">cor</span>(<span class="kw">t</span>(ProgSpecCD45[x, <span class="dv">-1</span>]), sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>))</span>
<span id="cb348-12"><a href="immune-estimation.html#cb348-12"></a>CorValsP=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(ProgSpecCD45), <span class="cf">function</span>(x) <span class="kw">cor.test</span>(<span class="kw">t</span>(ProgSpecCD45[x, <span class="dv">-1</span>]), sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb348-13"><a href="immune-estimation.html#cb348-13"></a></span>
<span id="cb348-14"><a href="immune-estimation.html#cb348-14"></a>naidx=<span class="kw">which</span>(CorValsP<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb348-15"><a href="immune-estimation.html#cb348-15"></a></span>
<span id="cb348-16"><a href="immune-estimation.html#cb348-16"></a>RNames1=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(ProgSpecCD45<span class="op">$</span>cell_type, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb348-17"><a href="immune-estimation.html#cb348-17"></a>RNamesMethod=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(ProgSpecCD45<span class="op">$</span>cell_type, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb348-18"><a href="immune-estimation.html#cb348-18"></a></span>
<span id="cb348-19"><a href="immune-estimation.html#cb348-19"></a></span>
<span id="cb348-20"><a href="immune-estimation.html#cb348-20"></a>df2=<span class="kw">data.frame</span>(<span class="dt">cell=</span>RNames1, <span class="dt">method=</span>RNamesMethod, <span class="dt">cor=</span>CorVals, <span class="dt">p=</span>CorValsP)</span>
<span id="cb348-21"><a href="immune-estimation.html#cb348-21"></a>ax1=<span class="kw">acast</span>(df2[ ,<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)], RNames1<span class="op">~</span>RNamesMethod)</span>
<span id="cb348-22"><a href="immune-estimation.html#cb348-22"></a></span>
<span id="cb348-23"><a href="immune-estimation.html#cb348-23"></a><span class="co">#df2=data.frame(cell=RNames1, method=RNamesMethod, cor=CorValsP)</span></span>
<span id="cb348-24"><a href="immune-estimation.html#cb348-24"></a>ax2=<span class="kw">acast</span>(df2[ ,<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>,<span class="dv">4</span>)], RNames1<span class="op">~</span>RNamesMethod)</span>
<span id="cb348-25"><a href="immune-estimation.html#cb348-25"></a>ax2[<span class="kw">which</span>(ax2<span class="op">&gt;</span><span class="fl">0.1</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">0</span></span>
<span id="cb348-26"><a href="immune-estimation.html#cb348-26"></a>ax2[<span class="kw">which</span>(<span class="kw">is.na</span>(ax2), <span class="dt">arr.ind =</span> T)]=<span class="dv">0</span></span>
<span id="cb348-27"><a href="immune-estimation.html#cb348-27"></a></span>
<span id="cb348-28"><a href="immune-estimation.html#cb348-28"></a><span class="kw">pdf</span>(<span class="st">&quot;~/Desktop/4F_progression_association_growthrate_all_methods.pdf&quot;</span>, <span class="dt">width=</span><span class="dv">8</span>, <span class="dt">height=</span><span class="dv">6</span>)</span>
<span id="cb348-29"><a href="immune-estimation.html#cb348-29"></a></span>
<span id="cb348-30"><a href="immune-estimation.html#cb348-30"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb348-31"><a href="immune-estimation.html#cb348-31"></a><span class="kw">OutputplotFun</span>(ax1, <span class="dt">scaleR=</span><span class="st">&quot;none&quot;</span>, <span class="dt">main=</span><span class="st">&quot;cell type correlation&quot;</span>, <span class="dt">classN=</span><span class="st">&quot;no&quot;</span>, <span class="dt">sigMat=</span>ax2)</span>
<span id="cb348-32"><a href="immune-estimation.html#cb348-32"></a></span>
<span id="cb348-33"><a href="immune-estimation.html#cb348-33"></a><span class="kw">dev.off</span>()</span>
<span id="cb348-34"><a href="immune-estimation.html#cb348-34"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb348-35"><a href="immune-estimation.html#cb348-35"></a><span class="co">##                 2</span></span>
<span id="cb348-36"><a href="immune-estimation.html#cb348-36"></a></span>
<span id="cb348-37"><a href="immune-estimation.html#cb348-37"></a><span class="co">## Also plot these all separately</span></span>
<span id="cb348-38"><a href="immune-estimation.html#cb348-38"></a>sizeInfoCut=infoTableFinal<span class="op">$</span>Growth[<span class="kw">match</span>(<span class="kw">colnames</span>(ProgSpecCD45)[<span class="op">-</span><span class="dv">1</span>], <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb348-39"><a href="immune-estimation.html#cb348-39"></a></span>
<span id="cb348-40"><a href="immune-estimation.html#cb348-40"></a>ttestVal=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(ProgSpecCD45), <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(<span class="kw">t</span>(ProgSpecCD45[x, <span class="dv">-1</span>])<span class="op">~</span>sizeInfoCut)<span class="op">$</span>p.value)</span>
<span id="cb348-41"><a href="immune-estimation.html#cb348-41"></a>df3=<span class="kw">data.frame</span>(<span class="dt">cell=</span>RNames1, <span class="dt">method=</span>RNamesMethod, <span class="dt">cor=</span>ttestVal)</span>
<span id="cb348-42"><a href="immune-estimation.html#cb348-42"></a>ax3=<span class="kw">acast</span>(df3, RNames1<span class="op">~</span>RNamesMethod)</span>
<span id="cb348-43"><a href="immune-estimation.html#cb348-43"></a>ax3[<span class="kw">which</span>(ax3<span class="op">&gt;</span><span class="fl">0.1</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">0</span></span>
<span id="cb348-44"><a href="immune-estimation.html#cb348-44"></a>ax3[<span class="kw">which</span>(<span class="kw">is.na</span>(ax3), <span class="dt">arr.ind =</span> T)]=<span class="dv">0</span></span>
<span id="cb348-45"><a href="immune-estimation.html#cb348-45"></a></span>
<span id="cb348-46"><a href="immune-estimation.html#cb348-46"></a><span class="co">#pdf(&quot;~/Desktop/4F_progression_association_growth_types.pdf&quot;, width=8, height=3)</span></span>
<span id="cb348-47"><a href="immune-estimation.html#cb348-47"></a></span>
<span id="cb348-48"><a href="immune-estimation.html#cb348-48"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb348-49"><a href="immune-estimation.html#cb348-49"></a></span>
<span id="cb348-50"><a href="immune-estimation.html#cb348-50"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(ax1)){</span>
<span id="cb348-51"><a href="immune-estimation.html#cb348-51"></a>  t1=ax1[, i]</span>
<span id="cb348-52"><a href="immune-estimation.html#cb348-52"></a>  t2=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(t1))</span>
<span id="cb348-53"><a href="immune-estimation.html#cb348-53"></a>  <span class="kw">image</span>(<span class="kw">cbind</span>(ax1[t2, i],ax1[t2, i]), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">xaxt=</span><span class="st">&quot;none&quot;</span>, <span class="dt">yaxt=</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb348-54"><a href="immune-estimation.html#cb348-54"></a>        <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;correlation %s (* sig p&lt;0.1)&quot;</span>, <span class="kw">colnames</span>(ax1)[i]))</span>
<span id="cb348-55"><a href="immune-estimation.html#cb348-55"></a>  <span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">length</span>(t2)), <span class="kw">names</span>(t1)[t2], <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex=</span><span class="fl">0.7</span>)</span>
<span id="cb348-56"><a href="immune-estimation.html#cb348-56"></a>  <span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="kw">c</span>(<span class="st">&quot;cor p&quot;</span>, <span class="st">&quot;wilcox p&quot;</span>), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex=</span><span class="fl">0.7</span>)</span>
<span id="cb348-57"><a href="immune-estimation.html#cb348-57"></a>  mx=<span class="kw">which</span>(ax2[ t2,i]<span class="op">&gt;</span><span class="dv">0</span>)</span>
<span id="cb348-58"><a href="immune-estimation.html#cb348-58"></a>  <span class="kw">text</span>((mx<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">length</span>(t2)<span class="op">-</span><span class="dv">1</span>), <span class="dv">0</span> , <span class="st">&quot;*&quot;</span>)</span>
<span id="cb348-59"><a href="immune-estimation.html#cb348-59"></a>  mx=<span class="kw">which</span>(ax3[ t2,i]<span class="op">&gt;</span><span class="dv">0</span>)</span>
<span id="cb348-60"><a href="immune-estimation.html#cb348-60"></a>  <span class="kw">text</span>((mx<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">length</span>(t2)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span> , <span class="st">&quot;*&quot;</span>)</span>
<span id="cb348-61"><a href="immune-estimation.html#cb348-61"></a> }</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-115-1.png" width="672" /></p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="immune-estimation.html#cb349-1"></a></span>
<span id="cb349-2"><a href="immune-estimation.html#cb349-2"></a><span class="kw">write.csv</span>(df2, <span class="dt">file=</span><span class="st">&quot;nature-tables/4f.csv&quot;</span>)</span>
<span id="cb349-3"><a href="immune-estimation.html#cb349-3"></a></span>
<span id="cb349-4"><a href="immune-estimation.html#cb349-4"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-115-2.png" width="672" /></p>
<p>Make the plots with individual samples. Firstly TIMER</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="immune-estimation.html#cb350-1"></a></span>
<span id="cb350-2"><a href="immune-estimation.html#cb350-2"></a><span class="co">#pdf(&quot;~/Desktop/4F-sample-celltypes-arranged_by_growth_rate.pdf&quot;, height=6, width=10)</span></span>
<span id="cb350-3"><a href="immune-estimation.html#cb350-3"></a></span>
<span id="cb350-4"><a href="immune-estimation.html#cb350-4"></a>x2=<span class="kw">c</span>(<span class="kw">colnames</span>(ProgSpecCD45)[<span class="kw">order</span>(sizeInfo)<span class="op">+</span><span class="dv">1</span>])</span>
<span id="cb350-5"><a href="immune-estimation.html#cb350-5"></a></span>
<span id="cb350-6"><a href="immune-estimation.html#cb350-6"></a>ax1=<span class="kw">grep</span>(<span class="st">&quot;TIMER&quot;</span>, ProgSpecCD45<span class="op">$</span>cell_type)</span>
<span id="cb350-7"><a href="immune-estimation.html#cb350-7"></a>T1melt=<span class="kw">melt</span>(ProgSpecCD45[ax1, ])</span>
<span id="cb350-8"><a href="immune-estimation.html#cb350-8"></a>T1melt<span class="op">$</span>growth=infoTableFinal<span class="op">$</span>Growth[<span class="kw">match</span>(T1melt<span class="op">$</span>variable, infoTableFinal<span class="op">$</span>SampleID)]</span>
<span id="cb350-9"><a href="immune-estimation.html#cb350-9"></a>T1melt<span class="op">$</span>treatment=infoTableFinal<span class="op">$</span>Treatment[<span class="kw">match</span>(T1melt<span class="op">$</span>variable, infoTableFinal<span class="op">$</span>SampleID)]</span>
<span id="cb350-10"><a href="immune-estimation.html#cb350-10"></a></span>
<span id="cb350-11"><a href="immune-estimation.html#cb350-11"></a>T1melt<span class="op">$</span>variable=<span class="kw">factor</span>(T1melt<span class="op">$</span>variable, x2)</span>
<span id="cb350-12"><a href="immune-estimation.html#cb350-12"></a></span>
<span id="cb350-13"><a href="immune-estimation.html#cb350-13"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>growth, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>, <span class="dt">drop=</span>T)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;TIMER:prog growth&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span></span>
<span id="cb350-14"><a href="immune-estimation.html#cb350-14"></a><span class="st">   </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#66C2A5&quot;</span>, <span class="st">&quot;#FED976&quot;</span>, <span class="st">&quot;#FEEBE2&quot;</span> ,<span class="st">&quot;#AE017E&quot;</span>, <span class="st">&quot;#2171B5&quot;</span>,   <span class="st">&quot;#BDD7E7&quot;</span>, <span class="st">&quot;#EFF3FF&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<div class="figure"><span id="fig:4f"></span>
<img src="RatDCIS-bookdown_files/figure-html/4f-1.png" alt="Progression CIBERSORT TIMER" width="672" />
<p class="caption">
Figure 18.4: Progression CIBERSORT TIMER
</p>
</div>
<p>then CIBERSORT</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="immune-estimation.html#cb351-1"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>treatment, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>, <span class="dt">drop=</span>T)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;TIMER:prog treatment&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span></span>
<span id="cb351-2"><a href="immune-estimation.html#cb351-2"></a><span class="st">   </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#66C2A5&quot;</span>, <span class="st">&quot;#FED976&quot;</span>, <span class="st">&quot;#FEEBE2&quot;</span> ,<span class="st">&quot;#AE017E&quot;</span>, <span class="st">&quot;#2171B5&quot;</span>,   <span class="st">&quot;#BDD7E7&quot;</span>, <span class="st">&quot;#EFF3FF&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-116-1.png" width="672" /></p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="immune-estimation.html#cb352-1"></a></span>
<span id="cb352-2"><a href="immune-estimation.html#cb352-2"></a></span>
<span id="cb352-3"><a href="immune-estimation.html#cb352-3"></a>ax1=<span class="kw">grep</span>(<span class="st">&quot;CIBERSORT$&quot;</span>, ProgSpecCD45<span class="op">$</span>cell_type)</span>
<span id="cb352-4"><a href="immune-estimation.html#cb352-4"></a>T1melt=<span class="kw">melt</span>(ProgSpecCD45[ax1, ])</span>
<span id="cb352-5"><a href="immune-estimation.html#cb352-5"></a></span>
<span id="cb352-6"><a href="immune-estimation.html#cb352-6"></a>T1melt<span class="op">$</span>variable=<span class="kw">factor</span>(T1melt<span class="op">$</span>variable, x2)</span>
<span id="cb352-7"><a href="immune-estimation.html#cb352-7"></a>T1melt<span class="op">$</span>growth=infoTableFinal<span class="op">$</span>Growth[<span class="kw">match</span>(T1melt<span class="op">$</span>variable, infoTableFinal<span class="op">$</span>SampleID)]</span>
<span id="cb352-8"><a href="immune-estimation.html#cb352-8"></a>T1melt<span class="op">$</span>treatment=infoTableFinal<span class="op">$</span>Treatment[<span class="kw">match</span>(T1melt<span class="op">$</span>variable, infoTableFinal<span class="op">$</span>SampleID)]</span>
<span id="cb352-9"><a href="immune-estimation.html#cb352-9"></a></span>
<span id="cb352-10"><a href="immune-estimation.html#cb352-10"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>growth, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span></span>
<span id="cb352-11"><a href="immune-estimation.html#cb352-11"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;CIBSERT:char cohort separately&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span></span>
<span id="cb352-12"><a href="immune-estimation.html#cb352-12"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-116-2.png" width="672" /></p>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="immune-estimation.html#cb353-1"></a></span>
<span id="cb353-2"><a href="immune-estimation.html#cb353-2"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>treatment, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span></span>
<span id="cb353-3"><a href="immune-estimation.html#cb353-3"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;CIBSERT:char cohort separately&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span></span>
<span id="cb353-4"><a href="immune-estimation.html#cb353-4"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-116-3.png" width="672" /></p>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="immune-estimation.html#cb354-1"></a></span>
<span id="cb354-2"><a href="immune-estimation.html#cb354-2"></a><span class="co"># Also do a version where the Bcells, Macrophages,NK, Mast Cells, CD4, CD8 Cells are merged together, NK</span></span>
<span id="cb354-3"><a href="immune-estimation.html#cb354-3"></a></span>
<span id="cb354-4"><a href="immune-estimation.html#cb354-4"></a>T1melt<span class="op">$</span>cell_type2=<span class="kw">substr</span>(T1melt<span class="op">$</span>cell_type, <span class="dv">1</span>, <span class="dv">4</span>)</span>
<span id="cb354-5"><a href="immune-estimation.html#cb354-5"></a>T1melt<span class="op">$</span>cell_type2[<span class="kw">grep</span>(<span class="st">&quot;CD8&quot;</span>, T1melt<span class="op">$</span>cell_type)]=<span class="st">&quot;T CD8&quot;</span></span>
<span id="cb354-6"><a href="immune-estimation.html#cb354-6"></a>T1melt<span class="op">$</span>cell_type2[<span class="kw">grep</span>(<span class="st">&quot;CD4&quot;</span>, T1melt<span class="op">$</span>cell_type)]=<span class="st">&quot;T CD4&quot;</span></span>
<span id="cb354-7"><a href="immune-estimation.html#cb354-7"></a></span>
<span id="cb354-8"><a href="immune-estimation.html#cb354-8"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type2))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>growth, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span></span>
<span id="cb354-9"><a href="immune-estimation.html#cb354-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;CIBSERT:prog growth&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span></span>
<span id="cb354-10"><a href="immune-estimation.html#cb354-10"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#66C2A5&quot;</span>, <span class="st">&quot;#E6F598&quot;</span>, <span class="st">&quot;#FED976&quot;</span>, <span class="st">&quot;#FC4E2A&quot;</span>, <span class="st">&quot;#bdbdbd&quot;</span>, <span class="st">&quot;#FEEBE2&quot;</span> ,<span class="st">&quot;#AE017E&quot;</span>, <span class="st">&quot;#5E4FA2&quot;</span>, <span class="st">&quot;#2171B5&quot;</span>, <span class="st">&quot;#BDD7E7&quot;</span>, <span class="st">&quot;#EFF3FF&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-116-4.png" width="672" /></p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="immune-estimation.html#cb355-1"></a></span>
<span id="cb355-2"><a href="immune-estimation.html#cb355-2"></a><span class="kw">ggplot</span>(T1melt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">fill=</span>cell_type2))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>treatment, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span></span>
<span id="cb355-3"><a href="immune-estimation.html#cb355-3"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;CIBSERT:prog treatment&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))<span class="op">+</span></span>
<span id="cb355-4"><a href="immune-estimation.html#cb355-4"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#66C2A5&quot;</span>, <span class="st">&quot;#E6F598&quot;</span>, <span class="st">&quot;#FED976&quot;</span>, <span class="st">&quot;#FC4E2A&quot;</span>, <span class="st">&quot;#bdbdbd&quot;</span>, <span class="st">&quot;#FEEBE2&quot;</span> ,<span class="st">&quot;#AE017E&quot;</span>, <span class="st">&quot;#5E4FA2&quot;</span>, <span class="st">&quot;#2171B5&quot;</span>, <span class="st">&quot;#BDD7E7&quot;</span>, <span class="st">&quot;#EFF3FF&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-116-5.png" width="672" /></p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="immune-estimation.html#cb356-1"></a></span>
<span id="cb356-2"><a href="immune-estimation.html#cb356-2"></a></span>
<span id="cb356-3"><a href="immune-estimation.html#cb356-3"></a></span>
<span id="cb356-4"><a href="immune-estimation.html#cb356-4"></a>T1melt<span class="op">$</span>sdat=<span class="kw">paste</span>(T1melt<span class="op">$</span>variable, T1melt<span class="op">$</span>cell_type2, <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb356-5"><a href="immune-estimation.html#cb356-5"></a>T2=<span class="kw">by</span>(T1melt<span class="op">$</span>value, T1melt<span class="op">$</span>sdat, sum)</span>
<span id="cb356-6"><a href="immune-estimation.html#cb356-6"></a>T2m=<span class="kw">stack</span>(T2)</span>
<span id="cb356-7"><a href="immune-estimation.html#cb356-7"></a>T2m<span class="op">$</span>ind1=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(T2m<span class="op">$</span>ind), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb356-8"><a href="immune-estimation.html#cb356-8"></a>T2m<span class="op">$</span>ind2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">as.character</span>(T2m<span class="op">$</span>ind), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb356-9"><a href="immune-estimation.html#cb356-9"></a></span>
<span id="cb356-10"><a href="immune-estimation.html#cb356-10"></a>T2mTab=<span class="kw">acast</span>(T2m[ ,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], ind2<span class="op">~</span>ind1, <span class="dt">value.var=</span><span class="st">&quot;values&quot;</span>)</span>
<span id="cb356-11"><a href="immune-estimation.html#cb356-11"></a>CorV=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(T2mTab), <span class="cf">function</span>(x) <span class="kw">cor</span>(T2mTab[x, ], sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>))</span>
<span id="cb356-12"><a href="immune-estimation.html#cb356-12"></a>CorVP=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(T2mTab), <span class="cf">function</span>(x) <span class="kw">cor.test</span>(T2mTab[x, ], sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb356-13"><a href="immune-estimation.html#cb356-13"></a>wilP=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(T2mTab), <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(T2mTab[x, ]<span class="op">~</span><span class="st"> </span>sizeInfoCut)<span class="op">$</span>p.value)</span>
<span id="cb356-14"><a href="immune-estimation.html#cb356-14"></a></span>
<span id="cb356-15"><a href="immune-estimation.html#cb356-15"></a><span class="co"># image(cbind(CorV,CorV), col=RdBu[11:1], xaxt=&quot;none&quot;, yaxt=&quot;none&quot;,</span></span>
<span id="cb356-16"><a href="immune-estimation.html#cb356-16"></a><span class="co">#         main=&quot;correlation CIBERSORT growth rate (* sig p&lt;0.1)&quot;)</span></span>
<span id="cb356-17"><a href="immune-estimation.html#cb356-17"></a><span class="co">#   axis(1, at=seq(0, 1, length=length(CorV)), rownames(T2mTab), las=2, cex=0.7)</span></span>
<span id="cb356-18"><a href="immune-estimation.html#cb356-18"></a><span class="co">#   axis(2, at=c(0,1),c(&quot;cor p&quot;, &quot;wilcox p&quot;), las=2, cex=0.7)</span></span>
<span id="cb356-19"><a href="immune-estimation.html#cb356-19"></a><span class="co">#   mx=which(CorVP&lt;0.1)</span></span>
<span id="cb356-20"><a href="immune-estimation.html#cb356-20"></a><span class="co">#   text((mx-1)/(CorVP-1), 0 , &quot;*&quot;)</span></span>
<span id="cb356-21"><a href="immune-estimation.html#cb356-21"></a><span class="co">#   mx=which(wilP&lt;0.1)</span></span>
<span id="cb356-22"><a href="immune-estimation.html#cb356-22"></a><span class="co">#   text((mx-1)/(wilP-1), 1 , &quot;*&quot;)</span></span>
<span id="cb356-23"><a href="immune-estimation.html#cb356-23"></a></span>
<span id="cb356-24"><a href="immune-estimation.html#cb356-24"></a>  <span class="co">#dev.off()</span></span>
<span id="cb356-25"><a href="immune-estimation.html#cb356-25"></a></span>
<span id="cb356-26"><a href="immune-estimation.html#cb356-26"></a><span class="kw">write.csv</span>(T1melt, <span class="dt">file=</span><span class="st">&quot;nature-tables/4f_image.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="clinical-associations" class="section level2">
<h2><span class="header-section-number">18.5</span> Clinical associations</h2>
<div id="associate-with-treatment" class="section level3">
<h3><span class="header-section-number">18.5.1</span> Associate with Treatment</h3>
<p>Look at association with treatment, growth and spatial infiltration for each method.</p>
<p>Associations with treatment:</p>
<ul>
<li>higher CD4, CD8 in most treatments</li>
<li>growth: stable associated with higher CD8</li>
<li>infiltration: more neutrophils and CD8? maybe CD4 cells</li>
</ul>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="immune-estimation.html#cb357-1"></a><span class="co">#sizeCutOff=7</span></span>
<span id="cb357-2"><a href="immune-estimation.html#cb357-2"></a></span>
<span id="cb357-3"><a href="immune-estimation.html#cb357-3"></a>MergedTablemelt=<span class="kw">melt</span>(MergedTable[<span class="op">-</span><span class="dv">1</span>, ])</span>
<span id="cb357-4"><a href="immune-estimation.html#cb357-4"></a>MergedTablemelt<span class="op">$</span>Treatment=infoTableFinal<span class="op">$</span>Treatment[<span class="kw">match</span>(MergedTablemelt<span class="op">$</span>variable, infoTableFinal<span class="op">$</span>SampleID)]</span>
<span id="cb357-5"><a href="immune-estimation.html#cb357-5"></a>MergedTablemelt<span class="op">$</span>Growth=infoTableFinal<span class="op">$</span>Growth[<span class="kw">match</span>(MergedTablemelt<span class="op">$</span>variable, infoTableFinal<span class="op">$</span>SampleID)]</span>
<span id="cb357-6"><a href="immune-estimation.html#cb357-6"></a>MergedTablemelt<span class="op">$</span>InfRes=infoTableFinal<span class="op">$</span>MHcut[<span class="kw">match</span>(MergedTablemelt<span class="op">$</span>variable, infoTableFinal<span class="op">$</span>SampleID)]</span>
<span id="cb357-7"><a href="immune-estimation.html#cb357-7"></a>MergedTablemelt<span class="op">$</span>cell_type2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(MergedTablemelt<span class="op">$</span>cell_type, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb357-8"><a href="immune-estimation.html#cb357-8"></a>MergedTablemelt<span class="op">$</span>method=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(MergedTablemelt<span class="op">$</span>cell_type, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="kw">length</span>(x)])</span>
<span id="cb357-9"><a href="immune-estimation.html#cb357-9"></a></span>
<span id="cb357-10"><a href="immune-estimation.html#cb357-10"></a>unValues=<span class="kw">unique</span>(MergedTablemelt<span class="op">$</span>cell_type2)</span>
<span id="cb357-11"><a href="immune-estimation.html#cb357-11"></a></span>
<span id="cb357-12"><a href="immune-estimation.html#cb357-12"></a>CompTest=<span class="kw">list</span>()</span>
<span id="cb357-13"><a href="immune-estimation.html#cb357-13"></a></span>
<span id="cb357-14"><a href="immune-estimation.html#cb357-14"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(unValues)){</span>
<span id="cb357-15"><a href="immune-estimation.html#cb357-15"></a>ax=MergedTablemelt[MergedTablemelt<span class="op">$</span>cell_type2<span class="op">==</span>unValues[i], ]</span>
<span id="cb357-16"><a href="immune-estimation.html#cb357-16"></a>p&lt;-<span class="kw">ggplot</span>(ax, <span class="kw">aes</span>(<span class="dt">x=</span>Treatment, <span class="dt">y=</span><span class="kw">as.numeric</span>(value), <span class="dt">col=</span>Treatment))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>method, <span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(unValues[i], <span class="st">&quot;Treatment&quot;</span>))</span>
<span id="cb357-17"><a href="immune-estimation.html#cb357-17"></a><span class="kw">print</span>(p)</span>
<span id="cb357-18"><a href="immune-estimation.html#cb357-18"></a><span class="co">#p&lt;-ggplot(ax, aes(x=Growth, y=as.numeric(value), col=Growth))+geom_boxplot()+facet_wrap(~method, scale=&quot;free_y&quot;)+ggtitle(paste(unValues[i], &quot;Growth&quot;))</span></span>
<span id="cb357-19"><a href="immune-estimation.html#cb357-19"></a><span class="co">#print(p)</span></span>
<span id="cb357-20"><a href="immune-estimation.html#cb357-20"></a><span class="co">#p&lt;-ggplot(ax, aes(x=InfRes, y=as.numeric(value), col=InfRes))+geom_boxplot()+facet_wrap(~method, #scale=&quot;free_y&quot;)+ggtitle(paste(unValues[i], &quot;Infiltration&quot;))</span></span>
<span id="cb357-21"><a href="immune-estimation.html#cb357-21"></a><span class="co">#print(p)</span></span>
<span id="cb357-22"><a href="immune-estimation.html#cb357-22"></a></span>
<span id="cb357-23"><a href="immune-estimation.html#cb357-23"></a>unT=<span class="kw">sort</span>(<span class="kw">unique</span>(ax<span class="op">$</span>Treatment))</span>
<span id="cb357-24"><a href="immune-estimation.html#cb357-24"></a>unB=<span class="kw">sort</span>(<span class="kw">unique</span>(ax<span class="op">$</span>method))</span>
<span id="cb357-25"><a href="immune-estimation.html#cb357-25"></a>Outcome1=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">5</span>, <span class="dt">ncol=</span><span class="kw">length</span>(unB))</span>
<span id="cb357-26"><a href="immune-estimation.html#cb357-26"></a></span>
<span id="cb357-27"><a href="immune-estimation.html#cb357-27"></a><span class="kw">rownames</span>(Outcome1)=<span class="kw">c</span>(<span class="kw">paste</span>(unT[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], <span class="st">&quot;vs.Vehicle&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="st">&quot;Grow.Stable&quot;</span>, <span class="st">&quot;Inf.res&quot;</span>)</span>
<span id="cb357-28"><a href="immune-estimation.html#cb357-28"></a><span class="kw">colnames</span>(Outcome1)=unB</span>
<span id="cb357-29"><a href="immune-estimation.html#cb357-29"></a></span>
<span id="cb357-30"><a href="immune-estimation.html#cb357-30"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(Outcome1)){</span>
<span id="cb357-31"><a href="immune-estimation.html#cb357-31"></a>Outcome1[<span class="dv">1</span> , ]=<span class="kw">sapply</span>(unB, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(ax<span class="op">$</span>value[<span class="kw">which</span>(ax<span class="op">$</span>Treatment<span class="op">==</span><span class="st">&quot;LY&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x)], </span>
<span id="cb357-32"><a href="immune-estimation.html#cb357-32"></a>                  ax<span class="op">$</span>value[ax<span class="op">$</span>Treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb357-33"><a href="immune-estimation.html#cb357-33"></a>Outcome1[<span class="dv">2</span> , ]=<span class="kw">sapply</span>(unB, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(ax<span class="op">$</span>value[ax<span class="op">$</span>Treatment<span class="op">==</span><span class="st">&quot;PDL1&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x], </span>
<span id="cb357-34"><a href="immune-estimation.html#cb357-34"></a>                  ax<span class="op">$</span>value[ax<span class="op">$</span>Treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb357-35"><a href="immune-estimation.html#cb357-35"></a>Outcome1[<span class="dv">3</span> , ]=<span class="kw">sapply</span>(unB, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(ax<span class="op">$</span>value[ax<span class="op">$</span>Treatment<span class="op">==</span><span class="st">&quot;PDL1+LY&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x], </span>
<span id="cb357-36"><a href="immune-estimation.html#cb357-36"></a>                  ax<span class="op">$</span>value[ax<span class="op">$</span>Treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb357-37"><a href="immune-estimation.html#cb357-37"></a>Outcome1[<span class="dv">4</span> , ]=<span class="kw">sapply</span>(unB, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(ax<span class="op">$</span>value[ax<span class="op">$</span>Growth<span class="op">==</span><span class="st">&quot;growing&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x], </span>
<span id="cb357-38"><a href="immune-estimation.html#cb357-38"></a>                  ax<span class="op">$</span>value[ax<span class="op">$</span>Growth<span class="op">==</span><span class="st">&quot;stable&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb357-39"><a href="immune-estimation.html#cb357-39"></a>Outcome1[<span class="dv">5</span> , ]=<span class="kw">sapply</span>(unB, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(ax<span class="op">$</span>value[ax<span class="op">$</span>InfRes<span class="op">==</span><span class="st">&quot;inf&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x], </span>
<span id="cb357-40"><a href="immune-estimation.html#cb357-40"></a>                  ax<span class="op">$</span>value[ax<span class="op">$</span>InfRes<span class="op">==</span><span class="st">&quot;res&quot;</span> <span class="op">&amp;</span><span class="st"> </span>ax<span class="op">$</span>method<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb357-41"><a href="immune-estimation.html#cb357-41"></a></span>
<span id="cb357-42"><a href="immune-estimation.html#cb357-42"></a>}</span>
<span id="cb357-43"><a href="immune-estimation.html#cb357-43"></a>CompTest[[i]]=<span class="kw">t</span>(Outcome1)</span>
<span id="cb357-44"><a href="immune-estimation.html#cb357-44"></a>o1=Outcome1</span>
<span id="cb357-45"><a href="immune-estimation.html#cb357-45"></a>o1[<span class="kw">which</span>(o1<span class="op">&lt;</span><span class="fl">0.05</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">3</span></span>
<span id="cb357-46"><a href="immune-estimation.html#cb357-46"></a>o1[<span class="kw">which</span>(o1<span class="op">&lt;</span><span class="fl">0.1</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">2</span></span>
<span id="cb357-47"><a href="immune-estimation.html#cb357-47"></a>o1[<span class="kw">which</span>(o1<span class="op">&lt;=</span><span class="dv">1</span>, <span class="dt">arr.ind=</span>T)]=<span class="dv">0</span></span>
<span id="cb357-48"><a href="immune-estimation.html#cb357-48"></a><span class="co">#heatmap.2(o1, col=brewer.pal(9, &quot;Blues&quot;), scale=&quot;none&quot;, trace=&quot;none&quot;, main=paste(&quot;pvalue summary&quot;, unValues[i]), Colv = NA, Rowv = NA)</span></span>
<span id="cb357-49"><a href="immune-estimation.html#cb357-49"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-4.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-5.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-6.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-7.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-8.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-9.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-10.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-11.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-12.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-13.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-14.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/immune-assoc-outcome-15.png" width="672" /></p>
<p>Using wilcox tests for significance, we can make the above comparisons and see if there is an association with outcome:</p>
<ul>
<li>There are differences in B-cell content between LY vs V comaprisons using xcell and cibersort</li>
<li>Macrophages are different in PDL1+LY</li>
</ul>
</div>
<div id="association-with-growth" class="section level3">
<h3><span class="header-section-number">18.5.2</span> Association with Growth</h3>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="immune-estimation.html#cb358-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(unValues)){</span>
<span id="cb358-2"><a href="immune-estimation.html#cb358-2"></a>ax=MergedTablemelt[MergedTablemelt<span class="op">$</span>cell_type2<span class="op">==</span>unValues[i], ]</span>
<span id="cb358-3"><a href="immune-estimation.html#cb358-3"></a>p&lt;-<span class="kw">ggplot</span>(ax, <span class="kw">aes</span>(<span class="dt">x=</span>Growth, <span class="dt">y=</span><span class="kw">as.numeric</span>(value), <span class="dt">col=</span>Growth))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>method, <span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(unValues[i], <span class="st">&quot;Growth&quot;</span>))</span>
<span id="cb358-4"><a href="immune-estimation.html#cb358-4"></a><span class="kw">print</span>(p)</span>
<span id="cb358-5"><a href="immune-estimation.html#cb358-5"></a>}</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-3.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-4.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-5.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-6.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-7.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-8.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-9.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-10.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-11.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-12.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-13.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-14.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-117-15.png" width="672" /></p>
<div class="sourceCode" id="cb359"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb359-1"><a href="immune-estimation.html#cb359-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(unValues)){</span>
<span id="cb359-2"><a href="immune-estimation.html#cb359-2"></a>ax=MergedTablemelt[MergedTablemelt<span class="op">$</span>cell_type2<span class="op">==</span>unValues[i], ]</span>
<span id="cb359-3"><a href="immune-estimation.html#cb359-3"></a>p&lt;-<span class="kw">ggplot</span>(ax, <span class="kw">aes</span>(<span class="dt">x=</span>InfRes, <span class="dt">y=</span><span class="kw">as.numeric</span>(value), <span class="dt">col=</span>InfRes))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>method, <span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(unValues[i], <span class="st">&quot;Infiltration&quot;</span>))</span>
<span id="cb359-4"><a href="immune-estimation.html#cb359-4"></a><span class="kw">print</span>(p)</span>
<span id="cb359-5"><a href="immune-estimation.html#cb359-5"></a>}</span></code></pre></div>
</div>
</div>
<div id="summary-of-the-outcome" class="section level2">
<h2><span class="header-section-number">18.6</span> Summary of the outcome</h2>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="immune-estimation.html#cb360-1"></a>CompTest2=<span class="kw">do.call</span>(rbind, CompTest)</span>
<span id="cb360-2"><a href="immune-estimation.html#cb360-2"></a> </span>
<span id="cb360-3"><a href="immune-estimation.html#cb360-3"></a><span class="kw">write.csv</span>(CompTest2, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/p_values_differences_treatment_growth_infiltration_%s.csv&quot;</span>, <span class="kw">Sys.Date</span>()))</span>
<span id="cb360-4"><a href="immune-estimation.html#cb360-4"></a></span>
<span id="cb360-5"><a href="immune-estimation.html#cb360-5"></a><span class="kw">scroll_box</span>(<span class="kw">kable</span>(CompTest2, <span class="dt">format=</span><span class="st">&quot;html&quot;</span>),</span>
<span id="cb360-6"><a href="immune-estimation.html#cb360-6"></a>         <span class="dt">height=</span><span class="st">&quot;300px&quot;</span>, <span class="dt">width=</span><span class="st">&quot;100%&quot;</span>)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:300px; overflow-x: scroll; width:100%; ">
<table>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
LYvs.Vehicle
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
PDL1vs.Vehicle
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
PDL1+LYvs.Vehicle
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Grow.Stable
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Inf.res
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
0.4126984
</td>
<td style="text-align:right;">
0.5476190
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.5880289
</td>
<td style="text-align:right;">
0.1892774
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
0.4126984
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.6623377
</td>
<td style="text-align:right;">
0.6426471
</td>
<td style="text-align:right;">
0.5358197
</td>
</tr>
<tr>
<td style="text-align:left;">
EPIC
</td>
<td style="text-align:right;">
0.5031709
</td>
<td style="text-align:right;">
0.6764350
</td>
<td style="text-align:right;">
0.3052553
</td>
<td style="text-align:right;">
0.2388414
</td>
<td style="text-align:right;">
0.8841264
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.0265187
</td>
<td style="text-align:right;">
0.1979593
</td>
<td style="text-align:right;">
0.0265187
</td>
<td style="text-align:right;">
0.1028077
</td>
<td style="text-align:right;">
0.5031709
</td>
</tr>
<tr>
<td style="text-align:left;">
MMCPCOUNTER
</td>
<td style="text-align:right;">
0.0004218
</td>
<td style="text-align:right;">
0.0110240
</td>
<td style="text-align:right;">
0.0372030
</td>
<td style="text-align:right;">
0.0027499
</td>
<td style="text-align:right;">
0.8841264
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.8735909
</td>
<td style="text-align:right;">
0.1566299
</td>
<td style="text-align:right;">
0.3301798
</td>
<td style="text-align:right;">
0.4282170
</td>
<td style="text-align:right;">
0.0562203
</td>
</tr>
<tr>
<td style="text-align:left;">
TIMER
</td>
<td style="text-align:right;">
0.7506528
</td>
<td style="text-align:right;">
0.9082682
</td>
<td style="text-align:right;">
0.1930218
</td>
<td style="text-align:right;">
0.3922756
</td>
<td style="text-align:right;">
0.4516905
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.0912068
</td>
<td style="text-align:right;">
0.0288056
</td>
<td style="text-align:right;">
0.0802331
</td>
<td style="text-align:right;">
0.2334498
</td>
<td style="text-align:right;">
0.6325687
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.9047619
</td>
<td style="text-align:right;">
0.6904762
</td>
<td style="text-align:right;">
0.5367965
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8665113
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
0.3710934
</td>
<td style="text-align:right;">
0.4237108
</td>
<td style="text-align:right;">
NaN
</td>
<td style="text-align:right;">
0.0572674
</td>
<td style="text-align:right;">
0.4226781
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
0.3710934
</td>
<td style="text-align:right;">
0.4237108
</td>
<td style="text-align:right;">
NaN
</td>
<td style="text-align:right;">
0.0572674
</td>
<td style="text-align:right;">
0.4226781
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.3095238
</td>
<td style="text-align:right;">
0.1610557
</td>
<td style="text-align:right;">
0.0707781
</td>
<td style="text-align:right;">
0.6843748
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.1111111
</td>
<td style="text-align:right;">
0.0952381
</td>
<td style="text-align:right;">
0.0519481
</td>
<td style="text-align:right;">
0.0033561
</td>
<td style="text-align:right;">
0.7282210
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.2222222
</td>
<td style="text-align:right;">
0.5367965
</td>
<td style="text-align:right;">
0.2749226
</td>
<td style="text-align:right;">
0.1205905
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.0317460
</td>
<td style="text-align:right;">
0.2467532
</td>
<td style="text-align:right;">
0.1574303
</td>
<td style="text-align:right;">
0.4634033
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
0.1904762
</td>
<td style="text-align:right;">
0.0317460
</td>
<td style="text-align:right;">
0.0519481
</td>
<td style="text-align:right;">
0.0968008
</td>
<td style="text-align:right;">
0.2809635
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.2622393
</td>
<td style="text-align:right;">
0.4633439
</td>
<td style="text-align:right;">
0.1908413
</td>
<td style="text-align:right;">
0.0288101
</td>
<td style="text-align:right;">
0.9048611
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.5555556
</td>
<td style="text-align:right;">
0.4633439
</td>
<td style="text-align:right;">
0.5830249
</td>
<td style="text-align:right;">
0.0088199
</td>
<td style="text-align:right;">
0.6843748
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
0.1904762
</td>
<td style="text-align:right;">
0.6904762
</td>
<td style="text-align:right;">
0.5367965
</td>
<td style="text-align:right;">
0.0144737
</td>
<td style="text-align:right;">
0.3968920
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
0.1904762
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.2467532
</td>
<td style="text-align:right;">
0.0144737
</td>
<td style="text-align:right;">
0.5358197
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.1904762
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.3290043
</td>
<td style="text-align:right;">
0.4377967
</td>
<td style="text-align:right;">
0.1519814
</td>
</tr>
<tr>
<td style="text-align:left;">
MMCPCOUNTER
</td>
<td style="text-align:right;">
0.1312104
</td>
<td style="text-align:right;">
0.0720057
</td>
<td style="text-align:right;">
0.4652088
</td>
<td style="text-align:right;">
0.3530684
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.5023350
</td>
<td style="text-align:right;">
0.4237108
</td>
<td style="text-align:right;">
0.3613104
</td>
<td style="text-align:right;">
0.5293681
</td>
<td style="text-align:right;">
NaN
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.1399825
</td>
<td style="text-align:right;">
0.1612376
</td>
<td style="text-align:right;">
0.0995755
</td>
<td style="text-align:right;">
0.0080464
</td>
<td style="text-align:right;">
0.6827511
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
NaN
</td>
<td style="text-align:right;">
NaN
</td>
<td style="text-align:right;">
0.4652088
</td>
<td style="text-align:right;">
0.2084128
</td>
<td style="text-align:right;">
0.3495748
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
NaN
</td>
<td style="text-align:right;">
NaN
</td>
<td style="text-align:right;">
0.4652088
</td>
<td style="text-align:right;">
0.2084128
</td>
<td style="text-align:right;">
0.3495748
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.7301587
</td>
<td style="text-align:right;">
0.5476190
</td>
<td style="text-align:right;">
0.6623377
</td>
<td style="text-align:right;">
0.8167957
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.2404208
</td>
<td style="text-align:right;">
0.1797125
</td>
<td style="text-align:right;">
0.1363802
</td>
<td style="text-align:right;">
0.3228308
</td>
<td style="text-align:right;">
0.4226781
</td>
</tr>
<tr>
<td style="text-align:left;">
TIMER
</td>
<td style="text-align:right;">
0.5555556
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.9307359
</td>
<td style="text-align:right;">
0.1574303
</td>
<td style="text-align:right;">
0.2318570
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.9008654
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.6321424
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.5500973
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.6666667
</td>
<td style="text-align:right;">
0.5333333
</td>
<td style="text-align:right;">
0.3333333
</td>
<td style="text-align:right;">
0.4206349
</td>
<td style="text-align:right;">
0.1904762
</td>
</tr>
<tr>
<td style="text-align:left;">
EPIC
</td>
<td style="text-align:right;">
0.9047619
</td>
<td style="text-align:right;">
0.2222222
</td>
<td style="text-align:right;">
0.6623377
</td>
<td style="text-align:right;">
0.4854231
</td>
<td style="text-align:right;">
0.6943279
</td>
</tr>
<tr>
<td style="text-align:left;">
MMCPCOUNTER
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.4206349
</td>
<td style="text-align:right;">
0.3290043
</td>
<td style="text-align:right;">
0.0004902
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.9047619
</td>
<td style="text-align:right;">
0.4206349
</td>
<td style="text-align:right;">
0.9307359
</td>
<td style="text-align:right;">
0.0264435
</td>
<td style="text-align:right;">
0.9550894
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.4126984
</td>
<td style="text-align:right;">
0.5476190
</td>
<td style="text-align:right;">
0.9307359
</td>
<td style="text-align:right;">
0.0296956
</td>
<td style="text-align:right;">
0.0400932
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.3333333
</td>
<td style="text-align:right;">
0.2666667
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0317460
</td>
<td style="text-align:right;">
0.0634921
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.3095238
</td>
<td style="text-align:right;">
0.1255411
</td>
<td style="text-align:right;">
0.3928793
</td>
<td style="text-align:right;">
0.5358197
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
0.1904762
</td>
<td style="text-align:right;">
0.6904762
</td>
<td style="text-align:right;">
0.1255411
</td>
<td style="text-align:right;">
0.5880289
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
EPIC
</td>
<td style="text-align:right;">
0.9047619
</td>
<td style="text-align:right;">
0.8412698
</td>
<td style="text-align:right;">
0.0519481
</td>
<td style="text-align:right;">
0.3506966
</td>
<td style="text-align:right;">
0.2809635
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.5385943
</td>
<td style="text-align:right;">
0.0317460
</td>
<td style="text-align:right;">
0.0519481
</td>
<td style="text-align:right;">
0.7807063
</td>
<td style="text-align:right;">
0.5207033
</td>
</tr>
<tr>
<td style="text-align:left;">
TIMER
</td>
<td style="text-align:right;">
0.4126984
</td>
<td style="text-align:right;">
0.8412698
</td>
<td style="text-align:right;">
0.6623377
</td>
<td style="text-align:right;">
0.0455624
</td>
<td style="text-align:right;">
0.0721057
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8412698
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0011610
</td>
<td style="text-align:right;">
0.2809635
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.2666667
</td>
<td style="text-align:right;">
0.3333333
</td>
<td style="text-align:right;">
0.3095238
</td>
<td style="text-align:right;">
0.1111111
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
0.9047619
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.7922078
</td>
<td style="text-align:right;">
0.0455624
</td>
<td style="text-align:right;">
0.7788656
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
0.9047619
</td>
<td style="text-align:right;">
0.4206349
</td>
<td style="text-align:right;">
0.5367965
</td>
<td style="text-align:right;">
0.0186017
</td>
<td style="text-align:right;">
0.8665113
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.5555556
</td>
<td style="text-align:right;">
0.4206349
</td>
<td style="text-align:right;">
0.5367965
</td>
<td style="text-align:right;">
0.9384675
</td>
<td style="text-align:right;">
0.5358197
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.7301587
</td>
<td style="text-align:right;">
0.1507937
</td>
<td style="text-align:right;">
0.1255411
</td>
<td style="text-align:right;">
0.1826883
</td>
<td style="text-align:right;">
0.6943279
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.0555556
</td>
<td style="text-align:right;">
0.2467532
</td>
<td style="text-align:right;">
0.1574303
</td>
<td style="text-align:right;">
0.4634033
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
0.1904762
</td>
<td style="text-align:right;">
0.0317460
</td>
<td style="text-align:right;">
0.0519481
</td>
<td style="text-align:right;">
0.1146285
</td>
<td style="text-align:right;">
0.2809635
</td>
</tr>
<tr>
<td style="text-align:left;">
EPIC
</td>
<td style="text-align:right;">
0.1111111
</td>
<td style="text-align:right;">
0.0952381
</td>
<td style="text-align:right;">
0.1774892
</td>
<td style="text-align:right;">
0.0034056
</td>
<td style="text-align:right;">
0.8665113
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.1904762
</td>
<td style="text-align:right;">
0.2222222
</td>
<td style="text-align:right;">
0.7922078
</td>
<td style="text-align:right;">
0.4854231
</td>
<td style="text-align:right;">
0.2809635
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.4126984
</td>
<td style="text-align:right;">
0.5476190
</td>
<td style="text-align:right;">
0.1669941
</td>
<td style="text-align:right;">
0.0054748
</td>
<td style="text-align:right;">
0.7721395
</td>
</tr>
<tr>
<td style="text-align:left;">
TIMER
</td>
<td style="text-align:right;">
0.2622393
</td>
<td style="text-align:right;">
0.4633439
</td>
<td style="text-align:right;">
0.2621926
</td>
<td style="text-align:right;">
0.6271128
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.1111111
</td>
<td style="text-align:right;">
0.2222222
</td>
<td style="text-align:right;">
0.1774892
</td>
<td style="text-align:right;">
0.0034056
</td>
<td style="text-align:right;">
0.7788656
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.2222222
</td>
<td style="text-align:right;">
0.2722293
</td>
<td style="text-align:right;">
0.1653588
</td>
<td style="text-align:right;">
0.9538159
</td>
</tr>
<tr>
<td style="text-align:left;">
CIBERSORT-ABS
</td>
<td style="text-align:right;">
0.4126984
</td>
<td style="text-align:right;">
0.2222222
</td>
<td style="text-align:right;">
0.2722293
</td>
<td style="text-align:right;">
0.2845377
</td>
<td style="text-align:right;">
0.8620647
</td>
</tr>
<tr>
<td style="text-align:left;">
EPIC
</td>
<td style="text-align:right;">
0.4126984
</td>
<td style="text-align:right;">
0.1507937
</td>
<td style="text-align:right;">
0.0173160
</td>
<td style="text-align:right;">
0.2749226
</td>
<td style="text-align:right;">
0.4634033
</td>
</tr>
<tr>
<td style="text-align:left;">
facs
</td>
<td style="text-align:right;">
0.3333333
</td>
<td style="text-align:right;">
0.5333333
</td>
<td style="text-align:right;">
0.3333333
</td>
<td style="text-align:right;">
0.0555556
</td>
<td style="text-align:right;">
0.1904762
</td>
</tr>
<tr>
<td style="text-align:left;">
QUANTISEQ
</td>
<td style="text-align:right;">
0.5023350
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.3413145
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.1582563
</td>
</tr>
<tr>
<td style="text-align:left;">
TIMER
</td>
<td style="text-align:right;">
0.2008251
</td>
<td style="text-align:right;">
0.1059975
</td>
<td style="text-align:right;">
0.0616637
</td>
<td style="text-align:right;">
0.4698277
</td>
<td style="text-align:right;">
0.5207033
</td>
</tr>
<tr>
<td style="text-align:left;">
XCELL
</td>
<td style="text-align:right;">
0.2857143
</td>
<td style="text-align:right;">
0.0952381
</td>
<td style="text-align:right;">
0.1774892
</td>
<td style="text-align:right;">
0.7573013
</td>
<td style="text-align:right;">
0.5358197
</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="immune-estimation.html#cb361-1"></a></span>
<span id="cb361-2"><a href="immune-estimation.html#cb361-2"></a>CompTest2=<span class="kw">data.frame</span>(CompTest2)</span>
<span id="cb361-3"><a href="immune-estimation.html#cb361-3"></a><span class="co"># CompTest2$Method=sapply(strsplit(rownames(CompTest2), &quot;_&quot;), function(x) x[2])</span></span>
<span id="cb361-4"><a href="immune-estimation.html#cb361-4"></a><span class="co"># CompTest2$CellType=sapply(strsplit(rownames(CompTest2), &quot;_&quot;), function(x) x[1])</span></span>
<span id="cb361-5"><a href="immune-estimation.html#cb361-5"></a><span class="co"># </span></span>
<span id="cb361-6"><a href="immune-estimation.html#cb361-6"></a><span class="co"># x2=unique(CompTest2$CellType[which(duplicated(CompTest2$CellType))])</span></span>
<span id="cb361-7"><a href="immune-estimation.html#cb361-7"></a><span class="co"># </span></span>
<span id="cb361-8"><a href="immune-estimation.html#cb361-8"></a><span class="co"># CompTest3=CompTest2[which(CompTest2$CellType%in%x2), ]</span></span>
<span id="cb361-9"><a href="immune-estimation.html#cb361-9"></a><span class="co"># CompTest3melt=melt(CompTest3)</span></span>
<span id="cb361-10"><a href="immune-estimation.html#cb361-10"></a><span class="co"># </span></span>
<span id="cb361-11"><a href="immune-estimation.html#cb361-11"></a><span class="co"># CompTest3melt$P2=ifelse(CompTest3melt$value&lt;0.05, &quot;**&quot;, ifelse(CompTest3melt$value&lt;0.1, &quot;*&quot;, &quot;ns&quot;))</span></span>
<span id="cb361-12"><a href="immune-estimation.html#cb361-12"></a><span class="co"># </span></span>
<span id="cb361-13"><a href="immune-estimation.html#cb361-13"></a><span class="co"># </span></span>
<span id="cb361-14"><a href="immune-estimation.html#cb361-14"></a><span class="co"># </span></span>
<span id="cb361-15"><a href="immune-estimation.html#cb361-15"></a><span class="co"># p=ggplot(CompTest3melt, aes(x=CellType, y=Method, fill=P2))+geom_tile()+facet_wrap(~variable)+theme(axis.text.x = element_text(angle = 90, hjust = 1))+ylab(&quot;No. differential genes&quot;)+scale_fill_manual(values=c(&quot;#084594&quot;, &quot;#2171b5&quot;, &quot;#deebf7&quot;))+ggtitle(&quot;p values of comparisons&quot;)</span></span>
<span id="cb361-16"><a href="immune-estimation.html#cb361-16"></a><span class="co"># print(p)</span></span>
<span id="cb361-17"><a href="immune-estimation.html#cb361-17"></a></span>
<span id="cb361-18"><a href="immune-estimation.html#cb361-18"></a><span class="co">##dev.off()</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="signature-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="bcr-clonotype-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
