<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 23 Mutations in RNA | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 23 Mutations in RNA | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 23 Mutations in RNA | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2022-02-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="whole-genome-sequencing-mutations.html"/>
<link rel="next" href="mutations-in-progression-cohort.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.3</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html"><i class="fa fa-check"></i><b>16</b> Read in ER i/o data</a><ul>
<li class="chapter" data-level="16.1" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.1</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.2" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#do-associative-study-with-other-features"><i class="fa fa-check"></i><b>16.2</b> Do associative study with other features?</a></li>
<li class="chapter" data-level="16.3" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#use-raw-counts-and-perform-a-batch-correction-if-necessary"><i class="fa fa-check"></i><b>16.3</b> Use raw counts and perform a batch correction if necessary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="tcr-clonotype-analysis.html"><a href="tcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>21</b> TCR clonotype analysis</a><ul>
<li class="chapter" data-level="21.0.1" data-path="tcr-clonotype-analysis.html"><a href="tcr-clonotype-analysis.html#progression-cohort-2"><i class="fa fa-check"></i><b>21.0.1</b> Progression cohort</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>22</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="22.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>22.1</b> Data</a><ul>
<li class="chapter" data-level="22.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>22.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>22.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="22.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>22.3</b> Plots</a><ul>
<li class="chapter" data-level="22.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>22.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="22.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>22.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>22.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="22.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>22.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="22.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>22.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="22.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>22.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="22.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>22.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>23</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>23.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>23.2</b> Load files</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>23.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="23.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>23.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="23.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>23.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>24</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="24.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#mutational-burden"><i class="fa fa-check"></i><b>24.1</b> Mutational burden</a></li>
<li class="chapter" data-level="24.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>24.2</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="24.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>24.3</b> Look at the common mutations (cosmic ones)</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>25</b> Trichrome staining</a><ul>
<li class="chapter" data-level="25.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>25.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="25.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>25.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="25.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>25.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="25.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>25.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>26</b> CNV calling</a><ul>
<li class="chapter" data-level="26.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>26.1</b> Sample data</a></li>
<li class="chapter" data-level="26.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>26.2</b> Data Summary</a><ul>
<li class="chapter" data-level="26.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>26.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="26.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>26.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="26.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>26.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="26.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>26.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="26.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>26.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>27</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mutations-in-rna" class="section level1">
<h1><span class="header-section-number">Chapter 23</span> Mutations in RNA</h1>
<p>In this section, we will check RNA data for the common mutations identified in the previous section. Some of these mutations could be uncharacterised polymoprphisms, and if this is the case, we will see in both CD45 and Epithelial fractions</p>
<div id="haplotype-caller" class="section level2">
<h2><span class="header-section-number">23.1</span> Haplotype caller</h2>
<p>Mutational calling from RNA data was performed using haplotype caller. The key steps are:</p>
<ol style="list-style-type: decimal">
<li>run haplotype caller</li>
<li>run snpeff to annotate</li>
<li>from snpsift to convert to a text file that can be read</li>
</ol>
<p>Note that a dbSNP file is needed to filter for polymorphisms. We have used the <a href="https://www.biorxiv.org/content/10.1101/412924v1.full">harlan variants</a> to perform filtering due to similarity in population frequencies in the previous section</p>
<p>An example of the code is shown here:</p>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="mutations-in-rna.html#cb546-1"></a> <span class="co"># pipeline</span></span>
<span id="cb546-2"><a href="mutations-in-rna.html#cb546-2"></a></span>
<span id="cb546-3"><a href="mutations-in-rna.html#cb546-3"></a><span class="co"># 1. run haplotyper</span></span>
<span id="cb546-4"><a href="mutations-in-rna.html#cb546-4"></a><span class="co"># A. Assemble BAM files</span></span>
<span id="cb546-5"><a href="mutations-in-rna.html#cb546-5"></a>Inputs<span class="op">:</span><span class="st"> </span></span>
<span id="cb546-6"><a href="mutations-in-rna.html#cb546-6"></a><span class="st">  </span><span class="er">$</span><span class="dv">1</span> fastq <span class="kw">file</span> (<span class="cf">if</span> paired end, <span class="op">$</span><span class="dv">2</span> will be the other)</span>
<span id="cb546-7"><a href="mutations-in-rna.html#cb546-7"></a>  <span class="op">$</span><span class="dv">2</span> aligned bam</span>
<span id="cb546-8"><a href="mutations-in-rna.html#cb546-8"></a><span class="co"># fastq to unaligned BAM</span></span>
<span id="cb546-9"><a href="mutations-in-rna.html#cb546-9"></a>java <span class="op">-</span>jar <span class="op">$</span>PICARD<span class="op">/</span>picard<span class="dv">-2</span>.<span class="dv">8</span>.<span class="fl">0.</span>jar FastqToSam F1=<span class="er">$</span><span class="dv">1</span> O=<span class="er">$</span>uBAM\</span>
<span id="cb546-10"><a href="mutations-in-rna.html#cb546-10"></a><span class="co">## for paired end:</span></span>
<span id="cb546-11"><a href="mutations-in-rna.html#cb546-11"></a><span class="co"># java -jar $PICARD/picard-2.8.0.jar FastqToSam F1=$1 F2=$2 O=$uBAM \</span></span>
<span id="cb546-12"><a href="mutations-in-rna.html#cb546-12"></a></span>
<span id="cb546-13"><a href="mutations-in-rna.html#cb546-13"></a><span class="co"># change read groups in aligned BAM file to ensure uniqueness</span></span>
<span id="cb546-14"><a href="mutations-in-rna.html#cb546-14"></a>java <span class="op">-</span>jar <span class="op">$</span>PICARD<span class="op">/</span>picard<span class="dv">-2</span>.<span class="dv">8</span>.<span class="fl">0.</span>jar AddOrReplaceReadGroups \</span>
<span id="cb546-15"><a href="mutations-in-rna.html#cb546-15"></a>     I=<span class="st"> </span><span class="er">$</span><span class="dv">2</span> \</span>
<span id="cb546-16"><a href="mutations-in-rna.html#cb546-16"></a>     O=<span class="st"> &quot;${2/.bam/ARG.bam}&quot;</span> \</span>
<span id="cb546-17"><a href="mutations-in-rna.html#cb546-17"></a>     ID=<span class="er">$</span>baseName2 \</span>
<span id="cb546-18"><a href="mutations-in-rna.html#cb546-18"></a>     LB=library \</span>
<span id="cb546-19"><a href="mutations-in-rna.html#cb546-19"></a>     PL=illumina \</span>
<span id="cb546-20"><a href="mutations-in-rna.html#cb546-20"></a>     PU=hiseq2000 \</span>
<span id="cb546-21"><a href="mutations-in-rna.html#cb546-21"></a>     SM=<span class="er">$</span>baseName2 \</span>
<span id="cb546-22"><a href="mutations-in-rna.html#cb546-22"></a>     CN=MFBC \</span>
<span id="cb546-23"><a href="mutations-in-rna.html#cb546-23"></a>     DT=<span class="dv">2018-08</span><span class="op">-</span>20T00<span class="op">:</span><span class="dv">00</span><span class="op">:</span><span class="dv">00-0400</span></span>
<span id="cb546-24"><a href="mutations-in-rna.html#cb546-24"></a><span class="co"># merge files</span></span>
<span id="cb546-25"><a href="mutations-in-rna.html#cb546-25"></a>gatk MergeBamAlignment \</span>
<span id="cb546-26"><a href="mutations-in-rna.html#cb546-26"></a><span class="op">--</span>REFERENCE_SEQUENCE ..<span class="op">/</span>Alignment<span class="op">/</span>rn6.fa \</span>
<span id="cb546-27"><a href="mutations-in-rna.html#cb546-27"></a><span class="op">--</span>UNMAPPED_BAM <span class="op">$</span>uBAM \</span>
<span id="cb546-28"><a href="mutations-in-rna.html#cb546-28"></a><span class="op">--</span>ALIGNED_BAM <span class="st">&quot;${2/.bam/ARG.bam}&quot;</span> \</span>
<span id="cb546-29"><a href="mutations-in-rna.html#cb546-29"></a><span class="op">--</span>OUTPUT <span class="op">$</span>mBAM \</span>
<span id="cb546-30"><a href="mutations-in-rna.html#cb546-30"></a><span class="op">--</span>INCLUDE_SECONDARY_ALIGNMENTS false \</span>
<span id="cb546-31"><a href="mutations-in-rna.html#cb546-31"></a><span class="op">--</span>PAIRED_RUN true \</span>
<span id="cb546-32"><a href="mutations-in-rna.html#cb546-32"></a><span class="op">--</span>VALIDATION_STRINGENCY SILENT</span>
<span id="cb546-33"><a href="mutations-in-rna.html#cb546-33"></a></span>
<span id="cb546-34"><a href="mutations-in-rna.html#cb546-34"></a><span class="co">#B. Mark duplicates</span></span>
<span id="cb546-35"><a href="mutations-in-rna.html#cb546-35"></a>gatk MarkDuplicates \</span>
<span id="cb546-36"><a href="mutations-in-rna.html#cb546-36"></a>        <span class="op">--</span>INPUT <span class="op">$</span>mBAM \</span>
<span id="cb546-37"><a href="mutations-in-rna.html#cb546-37"></a>        <span class="op">--</span>OUTPUT <span class="op">$</span>dBAM \</span>
<span id="cb546-38"><a href="mutations-in-rna.html#cb546-38"></a>        <span class="op">--</span>CREATE_INDEX true \</span>
<span id="cb546-39"><a href="mutations-in-rna.html#cb546-39"></a>        <span class="op">--</span>VALIDATION_STRINGENCY SILENT \</span>
<span id="cb546-40"><a href="mutations-in-rna.html#cb546-40"></a>        <span class="op">--</span>METRICS_FILE <span class="op">$</span>{outName}.metrics</span>
<span id="cb546-41"><a href="mutations-in-rna.html#cb546-41"></a><span class="co">#C. Split sigar reads</span></span>
<span id="cb546-42"><a href="mutations-in-rna.html#cb546-42"></a>gatk SplitNCigarReads \</span>
<span id="cb546-43"><a href="mutations-in-rna.html#cb546-43"></a>     <span class="op">-</span>R ..<span class="op">/</span>Alignment<span class="op">/</span>rn6.fa \</span>
<span id="cb546-44"><a href="mutations-in-rna.html#cb546-44"></a>     <span class="op">-</span>I <span class="op">$</span>dBAM \</span>
<span id="cb546-45"><a href="mutations-in-rna.html#cb546-45"></a>     <span class="op">-</span>O <span class="op">$</span>cBAM \</span>
<span id="cb546-46"><a href="mutations-in-rna.html#cb546-46"></a></span>
<span id="cb546-47"><a href="mutations-in-rna.html#cb546-47"></a><span class="co">#D. Base recalibration</span></span>
<span id="cb546-48"><a href="mutations-in-rna.html#cb546-48"></a>gatk BaseRecalibrator \</span>
<span id="cb546-49"><a href="mutations-in-rna.html#cb546-49"></a>             <span class="op">-</span>R ..<span class="op">/</span>Alignment<span class="op">/</span>rn6.fa \</span>
<span id="cb546-50"><a href="mutations-in-rna.html#cb546-50"></a>             <span class="op">-</span>I <span class="op">$</span>cBAM \</span>
<span id="cb546-51"><a href="mutations-in-rna.html#cb546-51"></a>             <span class="op">-</span>O <span class="op">$</span>{outName}_recalibration \</span>
<span id="cb546-52"><a href="mutations-in-rna.html#cb546-52"></a>             <span class="op">-</span>known<span class="op">-</span>sites harlan_snps_mod_<span class="dv">1-5</span><span class="fl">-2020.</span>SDsort.vcf.gz </span>
<span id="cb546-53"><a href="mutations-in-rna.html#cb546-53"></a></span>
<span id="cb546-54"><a href="mutations-in-rna.html#cb546-54"></a>gatk ApplyBQSR \</span>
<span id="cb546-55"><a href="mutations-in-rna.html#cb546-55"></a>             <span class="op">--</span>add<span class="op">-</span>output<span class="op">-</span>sam<span class="op">-</span>program<span class="op">-</span>record \</span>
<span id="cb546-56"><a href="mutations-in-rna.html#cb546-56"></a>             <span class="op">-</span>R ..<span class="op">/</span>Alignment<span class="op">/</span>rn6.fa \</span>
<span id="cb546-57"><a href="mutations-in-rna.html#cb546-57"></a>             <span class="op">-</span>I <span class="op">$</span>cBAM \</span>
<span id="cb546-58"><a href="mutations-in-rna.html#cb546-58"></a>             <span class="op">--</span>use<span class="op">-</span>original<span class="op">-</span>qualities \</span>
<span id="cb546-59"><a href="mutations-in-rna.html#cb546-59"></a>             <span class="op">-</span>O <span class="op">$</span>bBAM \</span>
<span id="cb546-60"><a href="mutations-in-rna.html#cb546-60"></a>             <span class="op">--</span>bqsr<span class="op">-</span>recal<span class="op">-</span>file <span class="op">$</span>{outName}_recalibration</span>
<span id="cb546-61"><a href="mutations-in-rna.html#cb546-61"></a></span>
<span id="cb546-62"><a href="mutations-in-rna.html#cb546-62"></a><span class="co">#E. Haplotype calling</span></span>
<span id="cb546-63"><a href="mutations-in-rna.html#cb546-63"></a>gatk HaplotypeCaller \</span>
<span id="cb546-64"><a href="mutations-in-rna.html#cb546-64"></a> <span class="op">-</span>R ..<span class="op">/</span>Alignment<span class="op">/</span>rn6.fa \</span>
<span id="cb546-65"><a href="mutations-in-rna.html#cb546-65"></a> <span class="op">-</span>I <span class="op">$</span>bBAM \</span>
<span id="cb546-66"><a href="mutations-in-rna.html#cb546-66"></a> <span class="op">-</span>L ..<span class="op">/</span>Alignment<span class="op">/</span>rn6_refGene_intervals.bed \</span>
<span id="cb546-67"><a href="mutations-in-rna.html#cb546-67"></a> <span class="op">-</span>O <span class="op">$</span>{outName}HCR.vcf.gz  \</span>
<span id="cb546-68"><a href="mutations-in-rna.html#cb546-68"></a> <span class="op">--</span>dont<span class="op">-</span>use<span class="op">-</span>soft<span class="op">-</span>clipped<span class="op">-</span>bases true \</span>
<span id="cb546-69"><a href="mutations-in-rna.html#cb546-69"></a> <span class="op">--</span>standard<span class="op">-</span>min<span class="op">-</span>confidence<span class="op">-</span>threshold<span class="op">-</span><span class="cf">for</span><span class="op">-</span>calling <span class="dv">20</span> \</span>
<span id="cb546-70"><a href="mutations-in-rna.html#cb546-70"></a> <span class="op">--</span>dbsnp all_rat_snps_<span class="fl">0504.</span>SD.vcf.gz</span>
<span id="cb546-71"><a href="mutations-in-rna.html#cb546-71"></a></span>
<span id="cb546-72"><a href="mutations-in-rna.html#cb546-72"></a><span class="co"># F. Filtering</span></span>
<span id="cb546-73"><a href="mutations-in-rna.html#cb546-73"></a>gatk VariantFiltration \</span>
<span id="cb546-74"><a href="mutations-in-rna.html#cb546-74"></a> <span class="op">--</span>R ..<span class="op">/</span>Alignment<span class="op">/</span>rn6.fa \</span>
<span id="cb546-75"><a href="mutations-in-rna.html#cb546-75"></a> <span class="op">--</span>V <span class="op">$</span>{outName}HCR.vcf.gz \</span>
<span id="cb546-76"><a href="mutations-in-rna.html#cb546-76"></a> <span class="op">--</span>window <span class="dv">35</span> \</span>
<span id="cb546-77"><a href="mutations-in-rna.html#cb546-77"></a> <span class="op">--</span>cluster <span class="dv">3</span> \</span>
<span id="cb546-78"><a href="mutations-in-rna.html#cb546-78"></a> <span class="op">--</span>filter<span class="op">-</span>name <span class="st">&quot;FS&quot;</span> \</span>
<span id="cb546-79"><a href="mutations-in-rna.html#cb546-79"></a> <span class="op">--</span>filter <span class="st">&quot;FS &gt; 30.0&quot;</span> \</span>
<span id="cb546-80"><a href="mutations-in-rna.html#cb546-80"></a> <span class="op">--</span>filter<span class="op">-</span>name <span class="st">&quot;QD&quot;</span> \</span>
<span id="cb546-81"><a href="mutations-in-rna.html#cb546-81"></a> <span class="op">--</span>filter <span class="st">&quot;QD &lt; 2.0&quot;</span> \</span>
<span id="cb546-82"><a href="mutations-in-rna.html#cb546-82"></a> <span class="op">-</span>O <span class="op">$</span>{outName}_filt_HCR</span>
<span id="cb546-83"><a href="mutations-in-rna.html#cb546-83"></a></span>
<span id="cb546-84"><a href="mutations-in-rna.html#cb546-84"></a><span class="co"># 2. run SnpEff to annotate</span></span>
<span id="cb546-85"><a href="mutations-in-rna.html#cb546-85"></a>java <span class="op">-</span>jar <span class="op">$</span>SNPEFF<span class="op">/</span>snpEff.jar <span class="op">-</span>v Rnor_<span class="dv">6</span>.<span class="fl">0.86</span> <span class="op">/</span>n<span class="op">/</span>scratch2<span class="op">/</span>at268<span class="op">/</span>carlos_data<span class="op">/</span><span class="dv">20191030</span>_Ep_NMU1_CGDA7428_S1_R1_<span class="dv">001</span>_filt <span class="op">&gt;</span><span class="st"> </span><span class="er">/</span>n<span class="op">/</span>scratch2<span class="op">/</span>at268<span class="op">/</span>carlos_data<span class="op">/</span><span class="dv">20191030</span>_Ep_NMU1_CGDA7428_S1_R1_<span class="dv">001</span>_filt_ann</span>
<span id="cb546-86"><a href="mutations-in-rna.html#cb546-86"></a></span>
<span id="cb546-87"><a href="mutations-in-rna.html#cb546-87"></a><span class="co"># 3. run SnpSift to save as a .txt file that can be easily read</span></span>
<span id="cb546-88"><a href="mutations-in-rna.html#cb546-88"></a>cat <span class="op">$</span>f <span class="op">|</span><span class="st"> </span><span class="er">$</span>SNPEFF<span class="op">/</span>scripts<span class="op">/</span>vcfEffOnePerLine.pl <span class="op">|</span><span class="st"> </span>java <span class="op">-</span>jar <span class="op">$</span>SNPEFF<span class="op">/</span>SnpSift.jar extractFields <span class="op">-</span>e <span class="st">&quot;&quot;</span> <span class="op">-</span><span class="st"> </span>CHROM POS ID REF ALT QUAL FILTER AF AC DP MQ  <span class="st">&quot;ANN[*].ANNOTATION&quot;</span> <span class="st">&quot;ANN[*].GENE&quot;</span> <span class="st">&quot;ANN[*].GENEID&quot;</span> <span class="st">&quot;ANN[*].BIOTYPE&quot;</span> <span class="st">&quot;ANN[*].HGVS_C&quot;</span> <span class="st">&quot;ANN[*].HGVS_P&quot;</span> <span class="st">&quot;ANN[*].CDNA_POS&quot;</span> <span class="st">&quot;ANN[*].CDNA_LEN&quot;</span> <span class="st">&quot;ANN[*].CDS_POS&quot;</span> <span class="st">&quot;ANN[*].CDS_LEN&quot;</span> <span class="st">&quot;ANN[*].AA_POS&quot;</span> <span class="st">&quot;ANN[*].AA_LEN&quot;</span> <span class="st">&quot;NMD[*].NUMTR&quot;</span> <span class="st">&quot;NMD[*].PERC&quot;</span> <span class="st">&quot;GEN[*]&quot;</span> <span class="op">&gt;</span><span class="st"> </span><span class="er">$</span>ofile</span></code></pre></div>
<p>From processing the mutational information:</p>
<p>Header information - Genotypes:
GT: genotype
AD: Allelic depths for the ref and alt alleles in the order listed
DP: Approximate read depth (reads with MQ=255 or with bad mates are filtered)
GQ: Genotype Quality
PL: Normalized, Phred-scaled likelihoods for genotypes</p>
<p>Header info: Info
AC: Allele count in genotypes, for each ALT allele, in the same order as listed
AF: Allele Frequency, for each ALT allele, in the same order as listed
AN: Total number of alleles in called genotypes
DB: dbSNP Membership
DP: Approximate read depth; some reads may have been filtered
ExcessHet: Phred-scaled p-value for exact test of excess heterozygosity
FS: Phred-scaled p-value using Fisher’s exact test to detect strand bias
MLEAC: Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC)
MLEAF: Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF)
MQ: RMS Mapping Quality
QD:Variant Confidence/Quality by Depth
SOR: Symmetric Odds Ratio of 2x2 contingency table to detect strand bias</p>
<p>SnpEff Annotations:
Allele : C
Annotation: missense_variant
Annotation_Impact MODERATE
Gene_Name Raet1l
Gene_ID ENSRNOG00000040300
Feature_Type transcript
Feature_ID ENSRNOT00000062027.4
Transcript_BioType protein_coding
HGVS.c c.1017C&gt;G
HGVS.p p.Cys339Trp
cDNA.pos / cDNA.length 1017/1185
CDS.pos / CDS.length 1017/1185
AA.pos / AA.length 339/394</p>
<p>missense_variant|MODERATE|Raet1l|ENSRNOG00000040300|transcript|ENSRNOT00000062027.4|protein_coding|6/7|c.1017C&gt;G|p.Cys339Trp|1017/1185|1017/1185|339/394||</p>
<p>AC=2;AF=1.00;AN=2;DP=49;ExcessHet=3.0103;FS=0.000;MLEAC=2;MLEAF=1.00;MQ=60.00;QD=34.99;SOR=0.874;ANN=GC|frameshift_variant&amp;splice_region_variant|HIGH|Reps1|ENSRNOG00000059224|transcript|ENSRNOT00000089607.1|protein_coding|8/29|c.119dupC|p.Gln41fs|142/2608|120/1842|40/613||INFO_REALIGN_3_PRIME;LOF=(Reps1|ENSRNOG00000059224|2|0.50)</p>
<p>AC=2;AF=1.00;AN=2;BaseQRankSum=1.593;ClippingRankSum=0.000;DP=9;ExcessHet=3.0103;FS=0.000;MLEAC=2;MLEAF=1.00;MQ=60.00;MQRankSum=0.000;QD=33.20;ReadPosRankSum=0.765;SOR=1.609;ANN=AT|frameshift_variant&amp;stop_gained|HIGH|Pacs1|ENSRNOG00000020350|transcript|ENSRNOT00000027632.5|protein_coding|3/24|c.464dupA|p.Tyr155fs|464/2886|464/2886|155/961||;LOF=(Pacs1|ENSRNOG00000020350|1|1.00);NMD=(Pacs1|ENSRNOG00000020350|1|1.00)</p>
</div>
<div id="load-files" class="section level2">
<h2><span class="header-section-number">23.2</span> Load files</h2>
<div class="sourceCode" id="cb547"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb547-1"><a href="mutations-in-rna.html#cb547-1"></a>hfiles=<span class="kw">dir</span>(<span class="st">&quot;../data/haplotypeCaller/output_HaplotypeCaller_May/characterisation/&quot;</span>,</span>
<span id="cb547-2"><a href="mutations-in-rna.html#cb547-2"></a>           <span class="dt">pattern=</span><span class="st">&quot;.txt&quot;</span>, <span class="dt">full.names =</span> T)</span>
<span id="cb547-3"><a href="mutations-in-rna.html#cb547-3"></a></span>
<span id="cb547-4"><a href="mutations-in-rna.html#cb547-4"></a>allHfiles=<span class="kw">list</span>()</span>
<span id="cb547-5"><a href="mutations-in-rna.html#cb547-5"></a>allCodingHaplo=<span class="kw">list</span>()</span>
<span id="cb547-6"><a href="mutations-in-rna.html#cb547-6"></a>allCosmicHaplo=<span class="kw">list</span>()</span>
<span id="cb547-7"><a href="mutations-in-rna.html#cb547-7"></a></span>
<span id="cb547-8"><a href="mutations-in-rna.html#cb547-8"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(hfiles)){</span>
<span id="cb547-9"><a href="mutations-in-rna.html#cb547-9"></a>A1=<span class="kw">read.delim</span>(hfiles[i], <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">stringsAsFactors =</span> F)</span>
<span id="cb547-10"><a href="mutations-in-rna.html#cb547-10"></a>A1[<span class="kw">which</span>(A1<span class="op">==</span>(<span class="op">-</span><span class="dv">1</span>), <span class="dt">arr.ind=</span>T)]=<span class="st">&quot;&quot;</span></span>
<span id="cb547-11"><a href="mutations-in-rna.html#cb547-11"></a></span>
<span id="cb547-12"><a href="mutations-in-rna.html#cb547-12"></a>cNames=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">colnames</span>(A1)[<span class="dv">12</span><span class="op">:</span><span class="dv">27</span>], <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb547-13"><a href="mutations-in-rna.html#cb547-13"></a>cNames[<span class="dv">13</span><span class="op">:</span><span class="dv">16</span>]=<span class="kw">paste</span>(<span class="kw">c</span>(<span class="st">&quot;LOF&quot;</span>, <span class="st">&quot;LOF&quot;</span>, <span class="st">&quot;NMD&quot;</span>, <span class="st">&quot;NMD&quot;</span>), cNames[<span class="dv">13</span><span class="op">:</span><span class="dv">16</span>], <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb547-14"><a href="mutations-in-rna.html#cb547-14"></a><span class="kw">colnames</span>(A1)[<span class="dv">12</span><span class="op">:</span><span class="dv">28</span>]=<span class="kw">c</span>(cNames, <span class="st">&quot;GEN&quot;</span>)</span>
<span id="cb547-15"><a href="mutations-in-rna.html#cb547-15"></a></span>
<span id="cb547-16"><a href="mutations-in-rna.html#cb547-16"></a>GEN2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(A1<span class="op">$</span>GEN, <span class="st">&quot;:&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb547-17"><a href="mutations-in-rna.html#cb547-17"></a>gRef=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(GEN2, <span class="st">&quot;,&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb547-18"><a href="mutations-in-rna.html#cb547-18"></a>gAlt=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(GEN2, <span class="st">&quot;,&quot;</span>), <span class="cf">function</span>(x) x[<span class="kw">length</span>(x)])</span>
<span id="cb547-19"><a href="mutations-in-rna.html#cb547-19"></a></span>
<span id="cb547-20"><a href="mutations-in-rna.html#cb547-20"></a>A1<span class="op">$</span>n_ref=<span class="kw">as.numeric</span>(gRef)</span>
<span id="cb547-21"><a href="mutations-in-rna.html#cb547-21"></a>A1<span class="op">$</span>n_alt=<span class="kw">as.numeric</span>(gAlt)</span>
<span id="cb547-22"><a href="mutations-in-rna.html#cb547-22"></a>A1<span class="op">$</span>VAF=A1<span class="op">$</span>n_alt<span class="op">/</span>(A1<span class="op">$</span>n_alt<span class="op">+</span>A1<span class="op">$</span>n_ref)</span>
<span id="cb547-23"><a href="mutations-in-rna.html#cb547-23"></a></span>
<span id="cb547-24"><a href="mutations-in-rna.html#cb547-24"></a>allHfiles[[i]]=A1[<span class="kw">which</span>(A1<span class="op">$</span>n_ref<span class="op">+</span>A1<span class="op">$</span>n_alt<span class="op">&gt;</span><span class="dv">10</span>), ]</span>
<span id="cb547-25"><a href="mutations-in-rna.html#cb547-25"></a>allCodingHaplo[[i]]=A1[<span class="kw">which</span>(A1<span class="op">$</span>HGVS_P<span class="op">!=</span><span class="st">&quot;&quot;</span> <span class="op">&amp;</span><span class="st"> </span>A1<span class="op">$</span>n_ref<span class="op">+</span>A1<span class="op">$</span>n_alt<span class="op">&gt;</span><span class="dv">10</span>), ]</span>
<span id="cb547-26"><a href="mutations-in-rna.html#cb547-26"></a>allCosmicHaplo[[i]]=A1[<span class="kw">which</span>(A1<span class="op">$</span>GENE<span class="op">%in%</span>RatCosmic <span class="op">&amp;</span><span class="st"> </span>A1<span class="op">$</span>n_ref<span class="op">+</span>A1<span class="op">$</span>n_alt<span class="op">&gt;</span><span class="dv">10</span>), ]</span>
<span id="cb547-27"><a href="mutations-in-rna.html#cb547-27"></a>}</span>
<span id="cb547-28"><a href="mutations-in-rna.html#cb547-28"></a></span>
<span id="cb547-29"><a href="mutations-in-rna.html#cb547-29"></a><span class="co">## names of the files</span></span>
<span id="cb547-30"><a href="mutations-in-rna.html#cb547-30"></a></span>
<span id="cb547-31"><a href="mutations-in-rna.html#cb547-31"></a>fNam=<span class="kw">unlist</span>(<span class="kw">strsplit</span>(hfiles, <span class="st">&quot;_filt_HCR_ann_siftB.txt&quot;</span>)) <span class="co">#remove HCR where not needed</span></span>
<span id="cb547-32"><a href="mutations-in-rna.html#cb547-32"></a>fNam=<span class="kw">unlist</span>(<span class="kw">strsplit</span>(fNam, <span class="st">&quot;.vcf.g&quot;</span>))</span>
<span id="cb547-33"><a href="mutations-in-rna.html#cb547-33"></a>fNam2=<span class="kw">paste</span>(fNam, <span class="st">&quot;.fastq.gz&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb547-34"><a href="mutations-in-rna.html#cb547-34"></a>fNam2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(fNam2, <span class="st">&quot;/&quot;</span>), <span class="cf">function</span>(x) x[<span class="kw">length</span>(x)])</span>
<span id="cb547-35"><a href="mutations-in-rna.html#cb547-35"></a>idx1=infoTableFinal<span class="op">$</span>SampleID[<span class="kw">match</span>(fNam2, infoTableFinal<span class="op">$</span>FqFile)]</span>
<span id="cb547-36"><a href="mutations-in-rna.html#cb547-36"></a>idx1[<span class="kw">which</span>(<span class="kw">is.na</span>(idx1))]=fNam2[<span class="kw">which</span>(<span class="kw">is.na</span>(idx1))]</span>
<span id="cb547-37"><a href="mutations-in-rna.html#cb547-37"></a>idx1[<span class="kw">grep</span>(<span class="st">&quot;CD45plus&quot;</span>, idx1)]=<span class="kw">substr</span>(idx1[<span class="kw">grep</span>(<span class="st">&quot;CD45plus&quot;</span>, idx1)],<span class="dv">10</span>, <span class="dv">21</span>) </span>
<span id="cb547-38"><a href="mutations-in-rna.html#cb547-38"></a><span class="co"># sNames=paste(tempAnnot$Rat_ID[idx1],tempAnnot$Location[idx1],  tempAnnot$Fraction[idx1], sep=&quot;_&quot;)</span></span>
<span id="cb547-39"><a href="mutations-in-rna.html#cb547-39"></a></span>
<span id="cb547-40"><a href="mutations-in-rna.html#cb547-40"></a><span class="kw">names</span>(allHfiles)=idx1</span>
<span id="cb547-41"><a href="mutations-in-rna.html#cb547-41"></a><span class="kw">names</span>(allCodingHaplo)=idx1</span>
<span id="cb547-42"><a href="mutations-in-rna.html#cb547-42"></a><span class="kw">names</span>(allCosmicHaplo)=idx1</span>
<span id="cb547-43"><a href="mutations-in-rna.html#cb547-43"></a><span class="co"># remove the samples which do not pass QC</span></span>
<span id="cb547-44"><a href="mutations-in-rna.html#cb547-44"></a>idx2rm=<span class="kw">setdiff</span>(<span class="kw">grep</span>(<span class="st">&quot;fastq.gz&quot;</span>, idx1), <span class="kw">grep</span>(<span class="st">&quot;CD45plus&quot;</span>, idx1))</span>
<span id="cb547-45"><a href="mutations-in-rna.html#cb547-45"></a></span>
<span id="cb547-46"><a href="mutations-in-rna.html#cb547-46"></a><span class="co">## write all these lists to file</span></span>
<span id="cb547-47"><a href="mutations-in-rna.html#cb547-47"></a>allHfiles=allHfiles[<span class="op">-</span>idx2rm]</span>
<span id="cb547-48"><a href="mutations-in-rna.html#cb547-48"></a>allCodingHaplo=allCodingHaplo[<span class="op">-</span>idx2rm]</span>
<span id="cb547-49"><a href="mutations-in-rna.html#cb547-49"></a>allCosmicHaplo=allCosmicHaplo[<span class="op">-</span>idx2rm]</span>
<span id="cb547-50"><a href="mutations-in-rna.html#cb547-50"></a></span>
<span id="cb547-51"><a href="mutations-in-rna.html#cb547-51"></a><span class="kw">save</span>(allHfiles,allCodingHaplo, allCodingHaplo,  <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/HaplotypeVariants_characterisation_all_%s.Rdata&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
</div>
<div id="identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data" class="section level2">
<h2><span class="header-section-number">23.3</span> Identifying polymorphisms: compare frequencies in CD45 and Ep data</h2>
<p>Search for whether mutations in the cosmic set is present in the RNA data. We will firstly look at the 90 most common genes, and then repeat the same for the top cosmic genes.</p>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="mutations-in-rna.html#cb548-1"></a><span class="co"># replace topN with topCOSMIC</span></span>
<span id="cb548-2"><a href="mutations-in-rna.html#cb548-2"></a><span class="co"># templace tempxb with tempz2</span></span>
<span id="cb548-3"><a href="mutations-in-rna.html#cb548-3"></a></span>
<span id="cb548-4"><a href="mutations-in-rna.html#cb548-4"></a>MeltWGSCoding<span class="op">$</span>mutPos=<span class="kw">paste</span>(MeltWGSCoding<span class="op">$</span>Gene_Symbol, MeltWGSCoding<span class="op">$</span>Start_Position)</span>
<span id="cb548-5"><a href="mutations-in-rna.html#cb548-5"></a>MeltWGSCodingB<span class="op">$</span>mutPos=<span class="kw">paste</span>(MeltWGSCodingB<span class="op">$</span>Gene_Symbol, MeltWGSCodingB<span class="op">$</span>Start_Position)</span>
<span id="cb548-6"><a href="mutations-in-rna.html#cb548-6"></a>MallCodingHaplo=<span class="kw">melt</span>(allCodingHaplo,  <span class="dt">measure.var=</span><span class="st">&quot;POS&quot;</span>)</span>
<span id="cb548-7"><a href="mutations-in-rna.html#cb548-7"></a>MallCodingHaplo<span class="op">$</span>mutPos=<span class="kw">paste</span>(MallCodingHaplo<span class="op">$</span>GENE, MallCodingHaplo<span class="op">$</span>value)</span></code></pre></div>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="mutations-in-rna.html#cb549-1"></a>xb=<span class="kw">sort</span>(<span class="kw">table</span>(MeltWGSCodingB<span class="op">$</span>mutPos[MeltWGSCodingB<span class="op">$</span>Gene_Symbol<span class="op">%in%</span>topCOSMIC]), <span class="dt">decreasing =</span> T)</span>
<span id="cb549-2"><a href="mutations-in-rna.html#cb549-2"></a></span>
<span id="cb549-3"><a href="mutations-in-rna.html#cb549-3"></a>allCoding_top90=<span class="kw">lapply</span>(allCodingHaplo, <span class="cf">function</span>(x) x[<span class="kw">which</span>(<span class="kw">paste</span>(x<span class="op">$</span>GENE, x<span class="op">$</span>POS)<span class="op">%in%</span><span class="kw">names</span>(xb)), ])</span>
<span id="cb549-4"><a href="mutations-in-rna.html#cb549-4"></a></span>
<span id="cb549-5"><a href="mutations-in-rna.html#cb549-5"></a>mallCoding90=<span class="kw">melt</span>(allCoding_top90, <span class="dt">measure.var=</span><span class="st">&quot;POS&quot;</span>)</span>
<span id="cb549-6"><a href="mutations-in-rna.html#cb549-6"></a>mallCoding90<span class="op">$</span>mutUnique=<span class="kw">paste</span>(mallCoding90<span class="op">$</span>GENE, mallCoding90<span class="op">$</span>value)</span>
<span id="cb549-7"><a href="mutations-in-rna.html#cb549-7"></a></span>
<span id="cb549-8"><a href="mutations-in-rna.html#cb549-8"></a>x1=<span class="kw">table</span>(mallCoding90<span class="op">$</span>mutUnique)</span>
<span id="cb549-9"><a href="mutations-in-rna.html#cb549-9"></a></span>
<span id="cb549-10"><a href="mutations-in-rna.html#cb549-10"></a>tabN=<span class="kw">acast</span>(mallCoding90[ , <span class="kw">c</span>(<span class="st">&quot;L1&quot;</span>, <span class="st">&quot;GENE&quot;</span>)],L1<span class="op">~</span>GENE)</span>
<span id="cb549-11"><a href="mutations-in-rna.html#cb549-11"></a><span class="co">#heatmap.2(tabN, Colv = NA, trace=&quot;none&quot;, scale=&quot;none&quot;, col=c(&quot;white&quot;, brewer.pal(5, &quot;Blues&quot;)))</span></span>
<span id="cb549-12"><a href="mutations-in-rna.html#cb549-12"></a>cNamesMatch=<span class="kw">match</span>(<span class="kw">colnames</span>(tabN), <span class="kw">colnames</span>(tempz2))</span>
<span id="cb549-13"><a href="mutations-in-rna.html#cb549-13"></a></span>
<span id="cb549-14"><a href="mutations-in-rna.html#cb549-14"></a>tabN2=<span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow=</span><span class="kw">nrow</span>(tabN), <span class="dt">ncol=</span><span class="kw">ncol</span>(tempz2))</span>
<span id="cb549-15"><a href="mutations-in-rna.html#cb549-15"></a>tabN2[ ,cNamesMatch]=tabN</span>
<span id="cb549-16"><a href="mutations-in-rna.html#cb549-16"></a><span class="kw">colnames</span>(tabN2)=<span class="kw">colnames</span>(tempz2)</span>
<span id="cb549-17"><a href="mutations-in-rna.html#cb549-17"></a><span class="kw">rownames</span>(tabN2)=<span class="kw">rownames</span>(tabN)</span>
<span id="cb549-18"><a href="mutations-in-rna.html#cb549-18"></a>id2=<span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, <span class="kw">rownames</span>(tabN))</span>
<span id="cb549-19"><a href="mutations-in-rna.html#cb549-19"></a>id3=<span class="kw">grep</span>(<span class="st">&quot;CD45$&quot;</span>, <span class="kw">rownames</span>(tabN))</span>
<span id="cb549-20"><a href="mutations-in-rna.html#cb549-20"></a>id4=<span class="kw">grep</span>(<span class="st">&quot;CD45plus&quot;</span>, <span class="kw">rownames</span>(tabN))</span>
<span id="cb549-21"><a href="mutations-in-rna.html#cb549-21"></a>tabN2[id2, ]=<span class="kw">sign</span>(tabN2[id2, ])</span>
<span id="cb549-22"><a href="mutations-in-rna.html#cb549-22"></a>tabN2[id3, ]=<span class="kw">sign</span>(tabN2[id3, ])<span class="op">*</span><span class="dv">2</span></span>
<span id="cb549-23"><a href="mutations-in-rna.html#cb549-23"></a>tabN2[id4, ]=<span class="kw">sign</span>(tabN2[id4, ])<span class="op">*</span><span class="dv">3</span></span>
<span id="cb549-24"><a href="mutations-in-rna.html#cb549-24"></a>tabN2=tabN2[<span class="kw">c</span>(id2, id3, id4), ]</span>
<span id="cb549-25"><a href="mutations-in-rna.html#cb549-25"></a></span>
<span id="cb549-26"><a href="mutations-in-rna.html#cb549-26"></a></span>
<span id="cb549-27"><a href="mutations-in-rna.html#cb549-27"></a><span class="co">#heatmap.2(tabN2, Colv = NA, Rowv = NA, trace=&quot;none&quot;, scale=&quot;none&quot;, col=c(&quot;white&quot;, brewer.pal(5, &quot;Blues&quot;)))</span></span>
<span id="cb549-28"><a href="mutations-in-rna.html#cb549-28"></a></span>
<span id="cb549-29"><a href="mutations-in-rna.html#cb549-29"></a><span class="kw">image</span>(<span class="kw">t</span>((tabN2)), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>), <span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)</span>
<span id="cb549-30"><a href="mutations-in-rna.html#cb549-30"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(tempz2)), <span class="kw">colnames</span>(tempz2), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb549-31"><a href="mutations-in-rna.html#cb549-31"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(tabN2)), <span class="kw">rownames</span>(tabN2), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-160-1.png" width="672" /></p>
<div class="sourceCode" id="cb550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb550-1"><a href="mutations-in-rna.html#cb550-1"></a></span>
<span id="cb550-2"><a href="mutations-in-rna.html#cb550-2"></a>lx2=<span class="kw">sapply</span>(allCoding_top90, nrow)</span>
<span id="cb550-3"><a href="mutations-in-rna.html#cb550-3"></a></span>
<span id="cb550-4"><a href="mutations-in-rna.html#cb550-4"></a><span class="kw">barplot</span>(lx2, <span class="dt">ylab=</span><span class="st">&quot;no# mutations in top 90 genes&quot;</span>, <span class="dt">las=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-160-2.png" width="672" /></p>
</div>
<div id="find-coding-mutations-which-are-specific-to-ep-cells" class="section level2">
<h2><span class="header-section-number">23.4</span> Find coding mutations which are specific to ep cells</h2>
<p>Here, look for mutations from RNA which are present in the epithelial cells specific</p>
<div class="sourceCode" id="cb551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb551-1"><a href="mutations-in-rna.html#cb551-1"></a>VarTypes=<span class="kw">c</span>(<span class="st">&quot;frameshift&quot;</span>, <span class="st">&quot;missense&quot;</span>, <span class="st">&quot;start_lost&quot;</span>, <span class="st">&quot;stop_gained&quot;</span>, <span class="st">&quot;nonsense&quot;</span>, <span class="st">&quot;stop_retained&quot;</span>)</span>
<span id="cb551-2"><a href="mutations-in-rna.html#cb551-2"></a>n2=<span class="kw">unique</span>(<span class="kw">unlist</span>(<span class="kw">sapply</span>(VarTypes, <span class="cf">function</span>(x) <span class="kw">grep</span>(x, MallCodingHaplo<span class="op">$</span>ANNOTATION))))</span>
<span id="cb551-3"><a href="mutations-in-rna.html#cb551-3"></a>MallCodingHaploFunc=MallCodingHaplo[ n2, ]</span>
<span id="cb551-4"><a href="mutations-in-rna.html#cb551-4"></a><span class="co">## Look at epithealial samples</span></span>
<span id="cb551-5"><a href="mutations-in-rna.html#cb551-5"></a></span>
<span id="cb551-6"><a href="mutations-in-rna.html#cb551-6"></a>xa=<span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, MallCodingHaploFunc<span class="op">$</span>L1)</span>
<span id="cb551-7"><a href="mutations-in-rna.html#cb551-7"></a>MallEpi=MallCodingHaploFunc[xa, ]</span>
<span id="cb551-8"><a href="mutations-in-rna.html#cb551-8"></a>EpVarList=<span class="kw">table</span>(MallCodingHaploFunc<span class="op">$</span>GENE[xa])</span>
<span id="cb551-9"><a href="mutations-in-rna.html#cb551-9"></a>EpVarNames=<span class="kw">names</span>(EpVarList)</span>
<span id="cb551-10"><a href="mutations-in-rna.html#cb551-10"></a>EpVarN2=<span class="kw">names</span>(EpVarList[<span class="kw">which</span>(EpVarList<span class="op">&gt;</span><span class="dv">1</span>)])</span>
<span id="cb551-11"><a href="mutations-in-rna.html#cb551-11"></a>EpVarListB=<span class="kw">table</span>(MallCodingHaploFunc<span class="op">$</span>mutPos[xa])</span>
<span id="cb551-12"><a href="mutations-in-rna.html#cb551-12"></a>EpLoc=<span class="kw">names</span>(EpVarListB)[<span class="kw">which</span>(EpVarListB<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb551-13"><a href="mutations-in-rna.html#cb551-13"></a></span>
<span id="cb551-14"><a href="mutations-in-rna.html#cb551-14"></a><span class="co">## Look at CD45 samples</span></span>
<span id="cb551-15"><a href="mutations-in-rna.html#cb551-15"></a>xa=<span class="kw">grep</span>(<span class="st">&quot;CD45$&quot;</span>, MallCodingHaploFunc<span class="op">$</span>L1)</span>
<span id="cb551-16"><a href="mutations-in-rna.html#cb551-16"></a>CDVarList=<span class="kw">table</span>(MallCodingHaploFunc<span class="op">$</span>GENE[xa])</span>
<span id="cb551-17"><a href="mutations-in-rna.html#cb551-17"></a>CDVarNames=<span class="kw">names</span>(CDVarList)</span>
<span id="cb551-18"><a href="mutations-in-rna.html#cb551-18"></a>CDVarN2=<span class="kw">names</span>(CDVarList[<span class="kw">which</span>(CDVarList<span class="op">&gt;=</span><span class="dv">1</span>)])</span>
<span id="cb551-19"><a href="mutations-in-rna.html#cb551-19"></a>CDVarListB=<span class="kw">table</span>(MallCodingHaploFunc<span class="op">$</span>mutPos[xa])</span>
<span id="cb551-20"><a href="mutations-in-rna.html#cb551-20"></a>CDLoc=<span class="kw">names</span>(CDVarListB)[<span class="kw">which</span>(CDVarListB<span class="op">&gt;=</span><span class="dv">1</span>)]</span>
<span id="cb551-21"><a href="mutations-in-rna.html#cb551-21"></a></span>
<span id="cb551-22"><a href="mutations-in-rna.html#cb551-22"></a></span>
<span id="cb551-23"><a href="mutations-in-rna.html#cb551-23"></a><span class="co">## Look at CD45 normal</span></span>
<span id="cb551-24"><a href="mutations-in-rna.html#cb551-24"></a>xa=<span class="kw">grep</span>(<span class="st">&quot;CD45plus&quot;</span>, MallCodingHaploFunc<span class="op">$</span>L1)</span>
<span id="cb551-25"><a href="mutations-in-rna.html#cb551-25"></a>nCD45VarList=<span class="kw">table</span>(MallCodingHaploFunc<span class="op">$</span>GENE[xa])</span>
<span id="cb551-26"><a href="mutations-in-rna.html#cb551-26"></a>nCD45VarNames=<span class="kw">names</span>(nCD45VarList)</span>
<span id="cb551-27"><a href="mutations-in-rna.html#cb551-27"></a>nCD45VarN2=<span class="kw">names</span>(nCD45VarList[<span class="kw">which</span>(nCD45VarList<span class="op">&gt;=</span><span class="dv">1</span>)])</span>
<span id="cb551-28"><a href="mutations-in-rna.html#cb551-28"></a>nCD45VarListB=<span class="kw">table</span>(MallCodingHaploFunc<span class="op">$</span>mutPos[xa])</span>
<span id="cb551-29"><a href="mutations-in-rna.html#cb551-29"></a>nCDLoc=<span class="kw">names</span>(nCD45VarListB)[<span class="kw">which</span>(nCD45VarListB<span class="op">&gt;=</span><span class="dv">1</span>)]</span>
<span id="cb551-30"><a href="mutations-in-rna.html#cb551-30"></a></span>
<span id="cb551-31"><a href="mutations-in-rna.html#cb551-31"></a><span class="co">## List of ep-spec variants</span></span>
<span id="cb551-32"><a href="mutations-in-rna.html#cb551-32"></a>eplist1=<span class="kw">setdiff</span>(<span class="kw">setdiff</span>(EpVarN2, CDVarN2), nCD45VarN2)</span>
<span id="cb551-33"><a href="mutations-in-rna.html#cb551-33"></a><span class="co">## Obtain the list of Ep-specific variants and cross-reference with genomic data</span></span>
<span id="cb551-34"><a href="mutations-in-rna.html#cb551-34"></a>WGSListgenes=<span class="kw">table</span>(MeltWGSCodingB<span class="op">$</span>Gene_Symbol)</span>
<span id="cb551-35"><a href="mutations-in-rna.html#cb551-35"></a>WGSListgenes=<span class="kw">names</span>(WGSListgenes)[<span class="kw">which</span>(WGSListgenes<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb551-36"><a href="mutations-in-rna.html#cb551-36"></a></span>
<span id="cb551-37"><a href="mutations-in-rna.html#cb551-37"></a>RList=MallEpi[<span class="kw">which</span>(MallEpi<span class="op">$</span>GENE<span class="op">%in%</span>eplist1), ]</span>
<span id="cb551-38"><a href="mutations-in-rna.html#cb551-38"></a>Rlist2=<span class="kw">table</span>(RList<span class="op">$</span>mutPos)</span>
<span id="cb551-39"><a href="mutations-in-rna.html#cb551-39"></a>Rlist2=<span class="kw">names</span>(Rlist2)[<span class="kw">which</span>(Rlist2<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb551-40"><a href="mutations-in-rna.html#cb551-40"></a></span>
<span id="cb551-41"><a href="mutations-in-rna.html#cb551-41"></a>WGSList=<span class="kw">paste</span>(MeltWGSCodingB<span class="op">$</span>Gene_Symbol, MeltWGSCodingB<span class="op">$</span>Start_Position)</span>
<span id="cb551-42"><a href="mutations-in-rna.html#cb551-42"></a>t2=<span class="kw">table</span>(WGSList)</span>
<span id="cb551-43"><a href="mutations-in-rna.html#cb551-43"></a>WGSList=<span class="kw">names</span>(t2)[<span class="kw">which</span>(t2<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb551-44"><a href="mutations-in-rna.html#cb551-44"></a>mutlist=<span class="kw">intersect</span>(Rlist2, WGSList)</span>
<span id="cb551-45"><a href="mutations-in-rna.html#cb551-45"></a></span>
<span id="cb551-46"><a href="mutations-in-rna.html#cb551-46"></a><span class="co">#pdf(&quot;~/Desktop/RNA-specific-mutations-gene-level.pdf&quot;, height=6, width=6)</span></span>
<span id="cb551-47"><a href="mutations-in-rna.html#cb551-47"></a></span>
<span id="cb551-48"><a href="mutations-in-rna.html#cb551-48"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">Ep=</span>EpVarNames, <span class="dt">CD45=</span>CDVarNames, <span class="dt">nCD45=</span>nCD45VarNames))</span>
<span id="cb551-49"><a href="mutations-in-rna.html#cb551-49"></a><span class="kw">mtext</span>(<span class="st">&quot;no genes w mutations overlapping in different fractions&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-161-1.png" width="672" /></p>
<div class="sourceCode" id="cb552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb552-1"><a href="mutations-in-rna.html#cb552-1"></a></span>
<span id="cb552-2"><a href="mutations-in-rna.html#cb552-2"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">Ep=</span>EpVarN2, <span class="dt">CD45=</span>CDVarN2, <span class="dt">nCD45=</span>nCD45VarN2))</span>
<span id="cb552-3"><a href="mutations-in-rna.html#cb552-3"></a><span class="kw">mtext</span>(<span class="st">&quot;no genes with overlapping variants in 2+ samples&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-161-2.png" width="672" /></p>
<div class="sourceCode" id="cb553"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb553-1"><a href="mutations-in-rna.html#cb553-1"></a></span>
<span id="cb553-2"><a href="mutations-in-rna.html#cb553-2"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">Ep_RNA=</span>eplist1, <span class="dt">Ep_WGS=</span>WGSListgenes))</span>
<span id="cb553-3"><a href="mutations-in-rna.html#cb553-3"></a><span class="kw">mtext</span>(<span class="st">&quot;No genes with overlaps in 2+ samples&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-161-3.png" width="672" /></p>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="mutations-in-rna.html#cb554-1"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p>Location specific mutations:</p>
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="mutations-in-rna.html#cb555-1"></a><span class="co">#pdf(&quot;~/Desktop/RNA-specific-mutations-location-level.pdf&quot;, height=7, width=10)</span></span>
<span id="cb555-2"><a href="mutations-in-rna.html#cb555-2"></a></span>
<span id="cb555-3"><a href="mutations-in-rna.html#cb555-3"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">Ep=</span>EpLoc, <span class="dt">CD45=</span>CDLoc, <span class="dt">nCD45=</span>nCDLoc))</span>
<span id="cb555-4"><a href="mutations-in-rna.html#cb555-4"></a><span class="kw">mtext</span>(<span class="st">&quot;no locations with overlapping variants in 2+ samples&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-162-1.png" width="672" /></p>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="mutations-in-rna.html#cb556-1"></a></span>
<span id="cb556-2"><a href="mutations-in-rna.html#cb556-2"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">Ep_RNA=</span>Rlist2, <span class="dt">Ep_WGS=</span>WGSList))</span>
<span id="cb556-3"><a href="mutations-in-rna.html#cb556-3"></a><span class="kw">mtext</span>(<span class="st">&quot;No ep. specifc locations with overlaps in 2+ samples&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-162-2.png" width="672" /></p>
<div class="sourceCode" id="cb557"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb557-1"><a href="mutations-in-rna.html#cb557-1"></a></span>
<span id="cb557-2"><a href="mutations-in-rna.html#cb557-2"></a><span class="co"># plots of mutational frequencies of the locations in A1</span></span>
<span id="cb557-3"><a href="mutations-in-rna.html#cb557-3"></a>Ttemp=<span class="kw">acast</span>(MallCodingHaploFunc[<span class="kw">which</span>(MallCodingHaploFunc<span class="op">$</span>mutPos<span class="op">%in%</span>mutlist), <span class="kw">c</span>(<span class="st">&quot;L1&quot;</span>, <span class="st">&quot;mutPos&quot;</span>)], mutPos<span class="op">~</span>L1)</span>
<span id="cb557-4"><a href="mutations-in-rna.html#cb557-4"></a>Ttemp=<span class="kw">cbind</span>(Ttemp, <span class="dv">0</span>)</span>
<span id="cb557-5"><a href="mutations-in-rna.html#cb557-5"></a><span class="kw">colnames</span>(Ttemp)[<span class="kw">ncol</span>(Ttemp)]=<span class="st">&quot;NMU13_LAU_Ep&quot;</span></span>
<span id="cb557-6"><a href="mutations-in-rna.html#cb557-6"></a></span>
<span id="cb557-7"><a href="mutations-in-rna.html#cb557-7"></a>ax2=<span class="kw">setdiff</span>(<span class="kw">unique</span>(MallEpi<span class="op">$</span>L1), <span class="kw">colnames</span>(Ttemp))</span>
<span id="cb557-8"><a href="mutations-in-rna.html#cb557-8"></a></span>
<span id="cb557-9"><a href="mutations-in-rna.html#cb557-9"></a>Wtemp=<span class="kw">acast</span>(MeltWGSCoding[<span class="kw">which</span>(MeltWGSCodingB<span class="op">$</span>mutPos<span class="op">%in%</span>mutlist), <span class="kw">c</span>(<span class="st">&quot;L1&quot;</span>, <span class="st">&quot;mutPos&quot;</span>, <span class="st">&quot;Start_Position&quot;</span>)],mutPos<span class="op">~</span>L1)</span>
<span id="cb557-10"><a href="mutations-in-rna.html#cb557-10"></a>Wtemp[<span class="kw">which</span>(<span class="kw">is.na</span>(Wtemp), <span class="dt">arr.ind =</span> T)]=<span class="dv">0</span></span>
<span id="cb557-11"><a href="mutations-in-rna.html#cb557-11"></a></span>
<span id="cb557-12"><a href="mutations-in-rna.html#cb557-12"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb557-13"><a href="mutations-in-rna.html#cb557-13"></a><span class="kw">image</span>(<span class="kw">sign</span>((Ttemp)), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>,<span class="st">&quot;darkblue&quot;</span>), <span class="dt">xaxt=</span><span class="st">&quot;none&quot;</span>, <span class="dt">yaxt=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb557-14"><a href="mutations-in-rna.html#cb557-14"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(Ttemp)),<span class="kw">rownames</span>(Ttemp), <span class="dt">las=</span><span class="dv">2</span>)</span>
<span id="cb557-15"><a href="mutations-in-rna.html#cb557-15"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(Ttemp)),<span class="kw">colnames</span>(Ttemp), <span class="dt">las=</span><span class="dv">2</span>)</span>
<span id="cb557-16"><a href="mutations-in-rna.html#cb557-16"></a><span class="kw">mtext</span>(<span class="st">&quot;Mutations in RNA &amp; in WGS data&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-162-3.png" width="672" /></p>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="mutations-in-rna.html#cb558-1"></a></span>
<span id="cb558-2"><a href="mutations-in-rna.html#cb558-2"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb558-3"><a href="mutations-in-rna.html#cb558-3"></a><span class="kw">image</span>(<span class="kw">sign</span>(Wtemp), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>,<span class="st">&quot;darkblue&quot;</span>), <span class="dt">xaxt=</span><span class="st">&quot;none&quot;</span>, <span class="dt">yaxt=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb558-4"><a href="mutations-in-rna.html#cb558-4"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(Wtemp)),<span class="kw">rownames</span>(Wtemp), <span class="dt">las=</span><span class="dv">2</span>)</span>
<span id="cb558-5"><a href="mutations-in-rna.html#cb558-5"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(Wtemp)),<span class="kw">colnames</span>(Wtemp), <span class="dt">las=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-162-4.png" width="672" /></p>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="mutations-in-rna.html#cb559-1"></a></span>
<span id="cb559-2"><a href="mutations-in-rna.html#cb559-2"></a><span class="co"># mtext(&quot;Mutations in WGS data&quot;)</span></span>
<span id="cb559-3"><a href="mutations-in-rna.html#cb559-3"></a><span class="co"># par(oma=c(2,2,2,5))</span></span>
<span id="cb559-4"><a href="mutations-in-rna.html#cb559-4"></a><span class="co"># heatmap.2(t(ExprVSD), trace=&quot;none&quot;, scale=&quot;none&quot;, Rowv = NA, Colv = NA, col=RdBu[11:1],</span></span>
<span id="cb559-5"><a href="mutations-in-rna.html#cb559-5"></a><span class="co">#           main=&quot;gene expression, vst transformed&quot;)</span></span>
<span id="cb559-6"><a href="mutations-in-rna.html#cb559-6"></a><span class="co"># heatmap.2(t(ExprVSD), trace=&quot;none&quot;, scale=&quot;col&quot;, Rowv = NA, Colv = NA, col=RdBu[11:1],</span></span>
<span id="cb559-7"><a href="mutations-in-rna.html#cb559-7"></a><span class="co">#           main=&quot;gene expression, col scaled&quot;)</span></span>
<span id="cb559-8"><a href="mutations-in-rna.html#cb559-8"></a><span class="co"># </span></span>
<span id="cb559-9"><a href="mutations-in-rna.html#cb559-9"></a><span class="co"># # calculate the correlation between mutation and expression</span></span>
<span id="cb559-10"><a href="mutations-in-rna.html#cb559-10"></a><span class="co"># </span></span>
<span id="cb559-11"><a href="mutations-in-rna.html#cb559-11"></a><span class="co"># mvals=rep(NA, nrow(ExprVSD))</span></span>
<span id="cb559-12"><a href="mutations-in-rna.html#cb559-12"></a><span class="co"># names(mvals)=rownames(ExprVSD)</span></span>
<span id="cb559-13"><a href="mutations-in-rna.html#cb559-13"></a><span class="co"># mvalsP=mvals</span></span>
<span id="cb559-14"><a href="mutations-in-rna.html#cb559-14"></a><span class="co"># </span></span>
<span id="cb559-15"><a href="mutations-in-rna.html#cb559-15"></a><span class="co"># x1=match(colnames(ExprVSD), colnames(Ttemp))</span></span>
<span id="cb559-16"><a href="mutations-in-rna.html#cb559-16"></a><span class="co"># for (i in 1:length(mvals)){</span></span>
<span id="cb559-17"><a href="mutations-in-rna.html#cb559-17"></a><span class="co">#   ax1=glm(sign(Ttemp[i, x1])~ExprVSD[i, ], family =binomial(link=&quot;logit&quot;))</span></span>
<span id="cb559-18"><a href="mutations-in-rna.html#cb559-18"></a><span class="co">#   mvals[i]=coefficients(ax1)[2]</span></span>
<span id="cb559-19"><a href="mutations-in-rna.html#cb559-19"></a><span class="co">#   mvalsP[i]=summary(ax1)$coefficients[2,4]</span></span>
<span id="cb559-20"><a href="mutations-in-rna.html#cb559-20"></a><span class="co"># }</span></span>
<span id="cb559-21"><a href="mutations-in-rna.html#cb559-21"></a><span class="co"># </span></span>
<span id="cb559-22"><a href="mutations-in-rna.html#cb559-22"></a><span class="co"># barplot(mvalsP,las=2, ylab = &quot;P value of association bw mutation and expression&quot;)</span></span>
<span id="cb559-23"><a href="mutations-in-rna.html#cb559-23"></a></span>
<span id="cb559-24"><a href="mutations-in-rna.html#cb559-24"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p>There are 6 mutations which are found in both the RNA and WGS data (2+ samples, not found in RNAseq)</p>
</div>
<div id="filtering-wgs-data-based-on-cd45-data" class="section level2">
<h2><span class="header-section-number">23.5</span> Filtering WGS data based on CD45 data</h2>
<ul>
<li>Use the entire list for WGS</li>
<li>Find mutations in CD45 (either) and remove</li>
</ul>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="mutations-in-rna.html#cb560-1"></a><span class="co">#head(MeltWGSCoding)</span></span>
<span id="cb560-2"><a href="mutations-in-rna.html#cb560-2"></a><span class="co">#head(MallCodingHaploFunc)</span></span>
<span id="cb560-3"><a href="mutations-in-rna.html#cb560-3"></a></span>
<span id="cb560-4"><a href="mutations-in-rna.html#cb560-4"></a>lx1=<span class="kw">grep</span>(<span class="st">&quot;CD45&quot;</span>, MallCodingHaploFunc<span class="op">$</span>L1)</span>
<span id="cb560-5"><a href="mutations-in-rna.html#cb560-5"></a></span>
<span id="cb560-6"><a href="mutations-in-rna.html#cb560-6"></a>mutDiff=<span class="kw">setdiff</span>(MeltWGSCodingB<span class="op">$</span>mutPos, MallCodingHaploFunc<span class="op">$</span>mutPos[lx1])</span>
<span id="cb560-7"><a href="mutations-in-rna.html#cb560-7"></a></span>
<span id="cb560-8"><a href="mutations-in-rna.html#cb560-8"></a>NUnique=<span class="kw">table</span>(MeltWGSCodingB<span class="op">$</span>mutPos[MeltWGSCodingB<span class="op">$</span>mutPos<span class="op">%in%</span>mutDiff])</span>
<span id="cb560-9"><a href="mutations-in-rna.html#cb560-9"></a>NUniqueB=<span class="kw">table</span>(MeltWGSCodingB<span class="op">$</span>Gene_Symbol[MeltWGSCodingB<span class="op">$</span>mutPos<span class="op">%in%</span>mutDiff])</span>
<span id="cb560-10"><a href="mutations-in-rna.html#cb560-10"></a></span>
<span id="cb560-11"><a href="mutations-in-rna.html#cb560-11"></a></span>
<span id="cb560-12"><a href="mutations-in-rna.html#cb560-12"></a>rGeneOut=<span class="kw">acast</span>(MeltWGSCodingB[ MeltWGSCodingB<span class="op">$</span>mutPos<span class="op">%in%</span><span class="kw">names</span>(NUnique),<span class="kw">c</span>(<span class="st">&quot;Gene_Symbol&quot;</span>, <span class="st">&quot;L1&quot;</span>, <span class="st">&quot;Variant_Classification&quot;</span>)], L1<span class="op">~</span>Gene_Symbol, </span>
<span id="cb560-13"><a href="mutations-in-rna.html#cb560-13"></a>               <span class="dt">value.var=</span><span class="st">&quot;Variant_Classification&quot;</span>, <span class="dt">fun.aggregate=</span><span class="cf">function</span>(x) <span class="kw">paste</span>(x, <span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>))</span>
<span id="cb560-14"><a href="mutations-in-rna.html#cb560-14"></a></span>
<span id="cb560-15"><a href="mutations-in-rna.html#cb560-15"></a>mxl=<span class="kw">unique</span>(<span class="kw">as.vector</span>(rGeneOut))</span>
<span id="cb560-16"><a href="mutations-in-rna.html#cb560-16"></a>rGeneOut2b=rGeneOut</span>
<span id="cb560-17"><a href="mutations-in-rna.html#cb560-17"></a>rGeneOut2b[<span class="kw">grep</span>(<span class="st">&quot;Nonsense&quot;</span>,rGeneOut2b)]=<span class="dv">3</span></span>
<span id="cb560-18"><a href="mutations-in-rna.html#cb560-18"></a>rGeneOut2b[<span class="kw">grep</span>(<span class="st">&quot;Frame_Shift&quot;</span>,rGeneOut2b)]=<span class="dv">2</span></span>
<span id="cb560-19"><a href="mutations-in-rna.html#cb560-19"></a>rGeneOut2b[<span class="kw">grep</span>(<span class="st">&quot;Missense_&quot;</span>, rGeneOut2b)]=<span class="dv">1</span></span>
<span id="cb560-20"><a href="mutations-in-rna.html#cb560-20"></a>rGeneOut2b[<span class="kw">grep</span>(<span class="st">&quot;Splice_&quot;</span>, rGeneOut2b)]=<span class="dv">4</span></span>
<span id="cb560-21"><a href="mutations-in-rna.html#cb560-21"></a>rGeneOut2b[<span class="kw">which</span>(rGeneOut2b<span class="op">==</span><span class="st">&quot;&quot;</span>)]=<span class="dv">0</span></span>
<span id="cb560-22"><a href="mutations-in-rna.html#cb560-22"></a></span>
<span id="cb560-23"><a href="mutations-in-rna.html#cb560-23"></a>rGeneOut2b=<span class="kw">apply</span>(rGeneOut2b, <span class="dv">2</span>, as.numeric)</span>
<span id="cb560-24"><a href="mutations-in-rna.html#cb560-24"></a><span class="kw">rownames</span>(rGeneOut2b)=<span class="kw">rownames</span>(rGeneOut)</span>
<span id="cb560-25"><a href="mutations-in-rna.html#cb560-25"></a></span>
<span id="cb560-26"><a href="mutations-in-rna.html#cb560-26"></a>rmThese=<span class="kw">c</span>(<span class="st">&quot;^AABR&quot;</span>, <span class="st">&quot;^Olr&quot;</span>, <span class="st">&quot;^LOC&quot;</span>, <span class="st">&quot;^AC[0-9]+&quot;</span>, <span class="st">&quot;^RGD&quot;</span>, <span class="st">&quot;^Rpl&quot;</span>, <span class="st">&quot;^1700&quot;</span>, <span class="st">&quot;^RT1&quot;</span>, <span class="st">&quot;^ENSRNOG&quot;</span>, <span class="st">&quot;^Vom2&quot;</span>)</span>
<span id="cb560-27"><a href="mutations-in-rna.html#cb560-27"></a>rmList=<span class="kw">unlist</span>(<span class="kw">sapply</span>(rmThese, <span class="cf">function</span>(x) <span class="kw">grep</span>(x, <span class="kw">colnames</span>(rGeneOut2b))))</span>
<span id="cb560-28"><a href="mutations-in-rna.html#cb560-28"></a>rGeneOut2b=rGeneOut2b[ ,<span class="op">-</span>rmList]</span>
<span id="cb560-29"><a href="mutations-in-rna.html#cb560-29"></a></span>
<span id="cb560-30"><a href="mutations-in-rna.html#cb560-30"></a></span>
<span id="cb560-31"><a href="mutations-in-rna.html#cb560-31"></a>NtopA=<span class="kw">names</span>(NUnique)[<span class="kw">which</span>(NUnique<span class="op">&gt;</span><span class="dv">6</span>)]</span>
<span id="cb560-32"><a href="mutations-in-rna.html#cb560-32"></a>NtopAb=<span class="kw">unique</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(NtopA, <span class="st">&quot; &quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]))</span>
<span id="cb560-33"><a href="mutations-in-rna.html#cb560-33"></a>tempAnew=rGeneOut2b[, <span class="kw">na.omit</span>(<span class="kw">match</span>(NtopAb, <span class="kw">colnames</span>(rGeneOut2b)))]</span>
<span id="cb560-34"><a href="mutations-in-rna.html#cb560-34"></a></span>
<span id="cb560-35"><a href="mutations-in-rna.html#cb560-35"></a>tempA=<span class="kw">heatmap.2</span>(<span class="kw">sign</span>(tempAnew))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-163-1.png" width="672" /></p>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="mutations-in-rna.html#cb561-1"></a>tempA=tempAnew[tempA<span class="op">$</span>rowInd, tempA<span class="op">$</span>colInd]</span>
<span id="cb561-2"><a href="mutations-in-rna.html#cb561-2"></a></span>
<span id="cb561-3"><a href="mutations-in-rna.html#cb561-3"></a><span class="co">#HSAnew=hotspotLocExtract(tempA)</span></span>
<span id="cb561-4"><a href="mutations-in-rna.html#cb561-4"></a></span>
<span id="cb561-5"><a href="mutations-in-rna.html#cb561-5"></a></span>
<span id="cb561-6"><a href="mutations-in-rna.html#cb561-6"></a><span class="co">## NtopB=which(NUnique&gt;10)</span></span>
<span id="cb561-7"><a href="mutations-in-rna.html#cb561-7"></a>NtopB=<span class="kw">names</span>(NUniqueB)[<span class="kw">which</span>(NUniqueB<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb561-8"><a href="mutations-in-rna.html#cb561-8"></a>NtopB2=NtopB[NtopB<span class="op">%in%</span>RatCosmic]</span>
<span id="cb561-9"><a href="mutations-in-rna.html#cb561-9"></a></span>
<span id="cb561-10"><a href="mutations-in-rna.html#cb561-10"></a>tempBnew=rGeneOut2b[, <span class="kw">na.omit</span>(<span class="kw">match</span>(NtopB2, <span class="kw">colnames</span>(rGeneOut2b)))]</span>
<span id="cb561-11"><a href="mutations-in-rna.html#cb561-11"></a></span>
<span id="cb561-12"><a href="mutations-in-rna.html#cb561-12"></a>tempB=<span class="kw">heatmap.2</span>(<span class="kw">sign</span>(tempBnew))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-163-2.png" width="672" /></p>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="mutations-in-rna.html#cb562-1"></a>tempB=tempBnew[tempB<span class="op">$</span>rowInd, tempB<span class="op">$</span>colInd]</span>
<span id="cb562-2"><a href="mutations-in-rna.html#cb562-2"></a></span>
<span id="cb562-3"><a href="mutations-in-rna.html#cb562-3"></a><span class="co">#HSBnew=hotspotLocExtract(tempB)</span></span>
<span id="cb562-4"><a href="mutations-in-rna.html#cb562-4"></a></span>
<span id="cb562-5"><a href="mutations-in-rna.html#cb562-5"></a><span class="co">#pdf(&quot;~/Desktop/WGS-coding-mutations-not-in-CD45-RNA.pdf&quot;, height=7, width=13)</span></span>
<span id="cb562-6"><a href="mutations-in-rna.html#cb562-6"></a><span class="co">## Commonly mutated Genes:</span></span>
<span id="cb562-7"><a href="mutations-in-rna.html#cb562-7"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb562-8"><a href="mutations-in-rna.html#cb562-8"></a><span class="kw">image</span>(<span class="kw">t</span>(tempA), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)</span>
<span id="cb562-9"><a href="mutations-in-rna.html#cb562-9"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(tempA)), <span class="kw">colnames</span>(tempA), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.5</span>)</span>
<span id="cb562-10"><a href="mutations-in-rna.html#cb562-10"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(tempA)), <span class="kw">rownames</span>(tempA), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.5</span>)</span>
<span id="cb562-11"><a href="mutations-in-rna.html#cb562-11"></a><span class="co">#text(na.omit(HSAnew$Y-1)/(ncol(tempA)-1), na.omit(HSAnew$X-1)/(nrow(tempA)-1) , &quot;*&quot;)</span></span>
<span id="cb562-12"><a href="mutations-in-rna.html#cb562-12"></a><span class="kw">legend</span>(<span class="st">&quot;top&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">-0.15</span>),    <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;missense&quot;</span>,<span class="st">&quot;frameshift&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;splice&quot;</span>, <span class="st">&quot;hotspot human&quot;</span>),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>,<span class="dv">19</span>, <span class="dv">8</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="st">&quot;black&quot;</span>), <span class="dt">horiz =</span> T)</span>
<span id="cb562-13"><a href="mutations-in-rna.html#cb562-13"></a><span class="kw">mtext</span>(<span class="st">&quot;top 118 mutations&quot;</span>, <span class="dt">side=</span><span class="dv">3</span>, <span class="dt">line=</span><span class="op">-</span><span class="dv">1</span>, <span class="dt">outer=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-163-3.png" width="672" /></p>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="mutations-in-rna.html#cb563-1"></a><span class="co">#dev.off()</span></span>
<span id="cb563-2"><a href="mutations-in-rna.html#cb563-2"></a></span>
<span id="cb563-3"><a href="mutations-in-rna.html#cb563-3"></a><span class="co">#pdf(&quot;~/Desktop/WGS-COSMIC-mutations-not-in-CD45-RNA.pdf&quot;, height=7, width=13)</span></span>
<span id="cb563-4"><a href="mutations-in-rna.html#cb563-4"></a><span class="co">## Commonly mutated Genes:</span></span>
<span id="cb563-5"><a href="mutations-in-rna.html#cb563-5"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb563-6"><a href="mutations-in-rna.html#cb563-6"></a><span class="kw">image</span>(<span class="kw">t</span>(tempB), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)</span>
<span id="cb563-7"><a href="mutations-in-rna.html#cb563-7"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(tempB)), <span class="kw">colnames</span>(tempB), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb563-8"><a href="mutations-in-rna.html#cb563-8"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(tempB)), <span class="kw">rownames</span>(tempB), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb563-9"><a href="mutations-in-rna.html#cb563-9"></a><span class="co">#text(na.omit(HSBnew$Y-1)/(ncol(tempB)-1), na.omit(HSBnew$X-1)/(nrow(tempB)-1) , &quot;*&quot;)</span></span>
<span id="cb563-10"><a href="mutations-in-rna.html#cb563-10"></a><span class="kw">legend</span>(<span class="st">&quot;top&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">-0.15</span>),    <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;missense&quot;</span>,<span class="st">&quot;frameshift&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;splice&quot;</span>, <span class="st">&quot;hotspot human&quot;</span>),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>,<span class="dv">19</span>, <span class="dv">8</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="st">&quot;black&quot;</span>), <span class="dt">horiz =</span> T)</span>
<span id="cb563-11"><a href="mutations-in-rna.html#cb563-11"></a><span class="kw">mtext</span>(<span class="st">&quot;Common COSMIC mutations&quot;</span>, <span class="dt">side=</span><span class="dv">3</span>, <span class="dt">line=</span><span class="op">-</span><span class="dv">1</span>, <span class="dt">outer=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-163-4.png" width="672" /></p>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="mutations-in-rna.html#cb564-1"></a><span class="co">#dev.off()</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="whole-genome-sequencing-mutations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mutations-in-progression-cohort.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
