<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 23 Mutations in progression cohort | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 23 Mutations in progression cohort | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 23 Mutations in progression cohort | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-06-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutations-in-rna.html"/>
<link rel="next" href="trichrome-staining.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.3</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html"><i class="fa fa-check"></i><b>16</b> Read in ER i/o data</a><ul>
<li class="chapter" data-level="16.1" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.1</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.2" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#do-associative-study-with-other-features"><i class="fa fa-check"></i><b>16.2</b> Do associative study with other features?</a></li>
<li class="chapter" data-level="16.3" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#use-raw-counts-and-perform-a-batch-correction-if-necessary"><i class="fa fa-check"></i><b>16.3</b> Use raw counts and perform a batch correction if necessary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>21</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="21.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>21.1</b> Data</a><ul>
<li class="chapter" data-level="21.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>21.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>21.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="21.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>21.3</b> Plots</a><ul>
<li class="chapter" data-level="21.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>21.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="21.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>21.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>21.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="21.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>21.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>21.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="21.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>21.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>21.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>22</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>22.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>22.2</b> Load files</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>22.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="22.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>22.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="22.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>22.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>23</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>23.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>23.2</b> Look at the common mutations (cosmic ones)</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>24</b> Trichrome staining</a><ul>
<li class="chapter" data-level="24.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>24.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="24.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>24.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="24.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>24.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="24.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>24.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>25</b> CNV calling</a><ul>
<li class="chapter" data-level="25.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>25.1</b> Sample data</a></li>
<li class="chapter" data-level="25.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>25.2</b> Data Summary</a><ul>
<li class="chapter" data-level="25.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>25.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="25.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>25.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="25.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>25.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>25.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="25.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>25.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>26</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mutations-in-progression-cohort" class="section level1">
<h1><span class="header-section-number">Chapter 23</span> Mutations in progression cohort</h1>
<div class="sourceCode" id="cb553"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb553-1"><a href="mutations-in-progression-cohort.html#cb553-1"></a>hfiles=<span class="kw">dir</span>(<span class="st">&quot;../data/haplotypeCaller/output_HaplotypeCaller_May/progression/&quot;</span>,</span>
<span id="cb553-2"><a href="mutations-in-progression-cohort.html#cb553-2"></a>           <span class="dt">pattern=</span><span class="st">&quot;.txt&quot;</span>, <span class="dt">full.names =</span> T)</span>
<span id="cb553-3"><a href="mutations-in-progression-cohort.html#cb553-3"></a></span>
<span id="cb553-4"><a href="mutations-in-progression-cohort.html#cb553-4"></a>allHfilesP=<span class="kw">list</span>()</span>
<span id="cb553-5"><a href="mutations-in-progression-cohort.html#cb553-5"></a>allCodingHaploP=<span class="kw">list</span>()</span>
<span id="cb553-6"><a href="mutations-in-progression-cohort.html#cb553-6"></a>allCosmicHaploP=<span class="kw">list</span>()</span>
<span id="cb553-7"><a href="mutations-in-progression-cohort.html#cb553-7"></a></span>
<span id="cb553-8"><a href="mutations-in-progression-cohort.html#cb553-8"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(hfiles)){</span>
<span id="cb553-9"><a href="mutations-in-progression-cohort.html#cb553-9"></a>A1=<span class="kw">read.delim</span>(hfiles[i], <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">stringsAsFactors =</span> F)</span>
<span id="cb553-10"><a href="mutations-in-progression-cohort.html#cb553-10"></a>A1[<span class="kw">which</span>(A1<span class="op">==</span>(<span class="op">-</span><span class="dv">1</span>), <span class="dt">arr.ind=</span>T)]=<span class="st">&quot;&quot;</span></span>
<span id="cb553-11"><a href="mutations-in-progression-cohort.html#cb553-11"></a></span>
<span id="cb553-12"><a href="mutations-in-progression-cohort.html#cb553-12"></a>cNames=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">colnames</span>(A1)[<span class="dv">12</span><span class="op">:</span><span class="dv">27</span>], <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb553-13"><a href="mutations-in-progression-cohort.html#cb553-13"></a>cNames[<span class="dv">13</span><span class="op">:</span><span class="dv">16</span>]=<span class="kw">paste</span>(<span class="kw">c</span>(<span class="st">&quot;LOF&quot;</span>, <span class="st">&quot;LOF&quot;</span>, <span class="st">&quot;NMD&quot;</span>, <span class="st">&quot;NMD&quot;</span>), cNames[<span class="dv">13</span><span class="op">:</span><span class="dv">16</span>], <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb553-14"><a href="mutations-in-progression-cohort.html#cb553-14"></a><span class="kw">colnames</span>(A1)[<span class="dv">12</span><span class="op">:</span><span class="dv">28</span>]=<span class="kw">c</span>(cNames, <span class="st">&quot;GEN&quot;</span>)</span>
<span id="cb553-15"><a href="mutations-in-progression-cohort.html#cb553-15"></a></span>
<span id="cb553-16"><a href="mutations-in-progression-cohort.html#cb553-16"></a>GEN2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(A1<span class="op">$</span>GEN, <span class="st">&quot;:&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb553-17"><a href="mutations-in-progression-cohort.html#cb553-17"></a>gRef=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(GEN2, <span class="st">&quot;,&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb553-18"><a href="mutations-in-progression-cohort.html#cb553-18"></a>gAlt=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(GEN2, <span class="st">&quot;,&quot;</span>), <span class="cf">function</span>(x) x[<span class="kw">length</span>(x)])</span>
<span id="cb553-19"><a href="mutations-in-progression-cohort.html#cb553-19"></a></span>
<span id="cb553-20"><a href="mutations-in-progression-cohort.html#cb553-20"></a>A1<span class="op">$</span>n_ref=<span class="kw">as.numeric</span>(gRef)</span>
<span id="cb553-21"><a href="mutations-in-progression-cohort.html#cb553-21"></a>A1<span class="op">$</span>n_alt=<span class="kw">as.numeric</span>(gAlt)</span>
<span id="cb553-22"><a href="mutations-in-progression-cohort.html#cb553-22"></a>A1<span class="op">$</span>VAF=A1<span class="op">$</span>n_alt<span class="op">/</span>(A1<span class="op">$</span>n_alt<span class="op">+</span>A1<span class="op">$</span>n_ref)</span>
<span id="cb553-23"><a href="mutations-in-progression-cohort.html#cb553-23"></a></span>
<span id="cb553-24"><a href="mutations-in-progression-cohort.html#cb553-24"></a>allHfilesP[[i]]=A1[<span class="kw">which</span>(A1<span class="op">$</span>n_ref<span class="op">+</span>A1<span class="op">$</span>n_alt<span class="op">&gt;</span><span class="dv">10</span>), ]</span>
<span id="cb553-25"><a href="mutations-in-progression-cohort.html#cb553-25"></a>allCodingHaploP[[i]]=A1[<span class="kw">which</span>(A1<span class="op">$</span>HGVS_P<span class="op">!=</span><span class="st">&quot;&quot;</span> <span class="op">&amp;</span><span class="st"> </span>A1<span class="op">$</span>n_ref<span class="op">+</span>A1<span class="op">$</span>n_alt<span class="op">&gt;</span><span class="dv">10</span>), ]</span>
<span id="cb553-26"><a href="mutations-in-progression-cohort.html#cb553-26"></a>allCosmicHaploP[[i]]=A1[<span class="kw">which</span>(A1<span class="op">$</span>GENE<span class="op">%in%</span>RatCosmic <span class="op">&amp;</span><span class="st"> </span>A1<span class="op">$</span>n_ref<span class="op">+</span>A1<span class="op">$</span>n_alt<span class="op">&gt;</span><span class="dv">10</span>), ]</span>
<span id="cb553-27"><a href="mutations-in-progression-cohort.html#cb553-27"></a>}</span>
<span id="cb553-28"><a href="mutations-in-progression-cohort.html#cb553-28"></a></span>
<span id="cb553-29"><a href="mutations-in-progression-cohort.html#cb553-29"></a><span class="co">## names of the files</span></span>
<span id="cb553-30"><a href="mutations-in-progression-cohort.html#cb553-30"></a></span>
<span id="cb553-31"><a href="mutations-in-progression-cohort.html#cb553-31"></a>fNam=<span class="kw">unlist</span>(<span class="kw">strsplit</span>(hfiles, <span class="st">&quot;_filt_HCR_ann_siftB.txt&quot;</span>)) <span class="co">#remove HCR where not needed</span></span>
<span id="cb553-32"><a href="mutations-in-progression-cohort.html#cb553-32"></a>fNam=<span class="kw">unlist</span>(<span class="kw">strsplit</span>(fNam, <span class="st">&quot;.vcf.g&quot;</span>))</span>
<span id="cb553-33"><a href="mutations-in-progression-cohort.html#cb553-33"></a>fNam2=<span class="kw">paste</span>(fNam, <span class="st">&quot;.fastq.gz&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb553-34"><a href="mutations-in-progression-cohort.html#cb553-34"></a>fNam2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(fNam2, <span class="st">&quot;/&quot;</span>), <span class="cf">function</span>(x) x[<span class="kw">length</span>(x)])</span>
<span id="cb553-35"><a href="mutations-in-progression-cohort.html#cb553-35"></a>idx1=infoTableFinal<span class="op">$</span>SampleID[<span class="kw">match</span>(fNam2, infoTableFinal<span class="op">$</span>FqFile)]</span>
<span id="cb553-36"><a href="mutations-in-progression-cohort.html#cb553-36"></a>idx1[<span class="kw">which</span>(<span class="kw">is.na</span>(idx1))]=fNam2[<span class="kw">which</span>(<span class="kw">is.na</span>(idx1))]</span>
<span id="cb553-37"><a href="mutations-in-progression-cohort.html#cb553-37"></a>idx1[<span class="kw">grep</span>(<span class="st">&quot;CD45plus&quot;</span>, idx1)]=<span class="kw">substr</span>(idx1[<span class="kw">grep</span>(<span class="st">&quot;CD45plus&quot;</span>, idx1)],<span class="dv">10</span>, <span class="dv">21</span>) </span>
<span id="cb553-38"><a href="mutations-in-progression-cohort.html#cb553-38"></a><span class="co"># sNames=paste(tempAnnot$Rat_ID[idx1],tempAnnot$Location[idx1],  tempAnnot$Fraction[idx1], sep=&quot;_&quot;)</span></span>
<span id="cb553-39"><a href="mutations-in-progression-cohort.html#cb553-39"></a></span>
<span id="cb553-40"><a href="mutations-in-progression-cohort.html#cb553-40"></a><span class="kw">names</span>(allHfilesP)=idx1</span>
<span id="cb553-41"><a href="mutations-in-progression-cohort.html#cb553-41"></a><span class="kw">names</span>(allCodingHaploP)=idx1</span>
<span id="cb553-42"><a href="mutations-in-progression-cohort.html#cb553-42"></a><span class="kw">names</span>(allCosmicHaploP)=idx1</span>
<span id="cb553-43"><a href="mutations-in-progression-cohort.html#cb553-43"></a><span class="co"># remove the samples which do not pass QC</span></span>
<span id="cb553-44"><a href="mutations-in-progression-cohort.html#cb553-44"></a>idx2rm=<span class="kw">setdiff</span>(<span class="kw">grep</span>(<span class="st">&quot;fastq.gz&quot;</span>, idx1), <span class="kw">grep</span>(<span class="st">&quot;CD45plus&quot;</span>, idx1))</span>
<span id="cb553-45"><a href="mutations-in-progression-cohort.html#cb553-45"></a></span>
<span id="cb553-46"><a href="mutations-in-progression-cohort.html#cb553-46"></a><span class="co">## write all these lists to file</span></span>
<span id="cb553-47"><a href="mutations-in-progression-cohort.html#cb553-47"></a>allHfilesP=allHfilesP[<span class="op">-</span>idx2rm]</span>
<span id="cb553-48"><a href="mutations-in-progression-cohort.html#cb553-48"></a>allCodingHaploP=allCodingHaploP[<span class="op">-</span>idx2rm]</span>
<span id="cb553-49"><a href="mutations-in-progression-cohort.html#cb553-49"></a>allCosmicHaploP=allCosmicHaploP[<span class="op">-</span>idx2rm]</span>
<span id="cb553-50"><a href="mutations-in-progression-cohort.html#cb553-50"></a></span>
<span id="cb553-51"><a href="mutations-in-progression-cohort.html#cb553-51"></a><span class="kw">save</span>(allHfilesP,allCodingHaploP, allCodingHaploP,  <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/HaplotypeVariants_progression_all_%s.Rdata&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
<div id="summary-of-common-mutations" class="section level2">
<h2><span class="header-section-number">23.1</span> Summary of common mutations:</h2>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="mutations-in-progression-cohort.html#cb554-1"></a>MallCodingHaploP=<span class="kw">melt</span>(allCodingHaploP,  <span class="dt">measure.var=</span><span class="st">&quot;POS&quot;</span>)</span>
<span id="cb554-2"><a href="mutations-in-progression-cohort.html#cb554-2"></a>MallCodingHaploP<span class="op">$</span>mutPos=<span class="kw">paste</span>(MallCodingHaploP<span class="op">$</span>GENE, MallCodingHaploP<span class="op">$</span>value)</span>
<span id="cb554-3"><a href="mutations-in-progression-cohort.html#cb554-3"></a></span>
<span id="cb554-4"><a href="mutations-in-progression-cohort.html#cb554-4"></a>VarTypes=<span class="kw">c</span>(<span class="st">&quot;frameshift&quot;</span>, <span class="st">&quot;missense&quot;</span>, <span class="st">&quot;start_lost&quot;</span>, <span class="st">&quot;stop_gained&quot;</span>, <span class="st">&quot;nonsense&quot;</span>, <span class="st">&quot;stop_retained&quot;</span>, <span class="st">&quot;splice&quot;</span>)</span>
<span id="cb554-5"><a href="mutations-in-progression-cohort.html#cb554-5"></a>n2=<span class="kw">unique</span>(<span class="kw">unlist</span>(<span class="kw">sapply</span>(VarTypes, <span class="cf">function</span>(x) <span class="kw">grep</span>(x, MallCodingHaploP<span class="op">$</span>ANNOTATION))))</span>
<span id="cb554-6"><a href="mutations-in-progression-cohort.html#cb554-6"></a>MallCodingHaploFuncP=MallCodingHaploP[ n2, ]</span>
<span id="cb554-7"><a href="mutations-in-progression-cohort.html#cb554-7"></a><span class="co">## Look at epithealial samples</span></span></code></pre></div>
<p>look at topCOSMIC:</p>
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="mutations-in-progression-cohort.html#cb555-1"></a></span>
<span id="cb555-2"><a href="mutations-in-progression-cohort.html#cb555-2"></a>xa=<span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, MallCodingHaploFuncP<span class="op">$</span>L1)</span>
<span id="cb555-3"><a href="mutations-in-progression-cohort.html#cb555-3"></a>MallEpi=MallCodingHaploFuncP[xa, ]</span>
<span id="cb555-4"><a href="mutations-in-progression-cohort.html#cb555-4"></a>EpVarList=<span class="kw">table</span>(MallCodingHaploFuncP<span class="op">$</span>GENE[xa])</span>
<span id="cb555-5"><a href="mutations-in-progression-cohort.html#cb555-5"></a>EpVarNames=<span class="kw">names</span>(EpVarList)</span>
<span id="cb555-6"><a href="mutations-in-progression-cohort.html#cb555-6"></a>EpVarN2=<span class="kw">names</span>(EpVarList[<span class="kw">which</span>(EpVarList<span class="op">&gt;</span><span class="dv">1</span>)])</span>
<span id="cb555-7"><a href="mutations-in-progression-cohort.html#cb555-7"></a>EpVarListB=<span class="kw">table</span>(MallCodingHaploFuncP<span class="op">$</span>mutPos[xa])</span>
<span id="cb555-8"><a href="mutations-in-progression-cohort.html#cb555-8"></a>EpLoc=<span class="kw">names</span>(EpVarListB)[<span class="kw">which</span>(EpVarListB<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb555-9"><a href="mutations-in-progression-cohort.html#cb555-9"></a></span>
<span id="cb555-10"><a href="mutations-in-progression-cohort.html#cb555-10"></a><span class="co">## Look at CD45 samples</span></span>
<span id="cb555-11"><a href="mutations-in-progression-cohort.html#cb555-11"></a>xa=<span class="kw">grep</span>(<span class="st">&quot;CD45$&quot;</span>, MallCodingHaploFuncP<span class="op">$</span>L1)</span>
<span id="cb555-12"><a href="mutations-in-progression-cohort.html#cb555-12"></a>CDVarList=<span class="kw">table</span>(MallCodingHaploFuncP<span class="op">$</span>GENE[xa])</span>
<span id="cb555-13"><a href="mutations-in-progression-cohort.html#cb555-13"></a>CDVarNames=<span class="kw">names</span>(CDVarList)</span>
<span id="cb555-14"><a href="mutations-in-progression-cohort.html#cb555-14"></a>CDVarN2=<span class="kw">names</span>(CDVarList[<span class="kw">which</span>(CDVarList<span class="op">&gt;=</span><span class="dv">1</span>)])</span>
<span id="cb555-15"><a href="mutations-in-progression-cohort.html#cb555-15"></a>CDVarListB=<span class="kw">table</span>(MallCodingHaploFuncP<span class="op">$</span>mutPos[xa])</span>
<span id="cb555-16"><a href="mutations-in-progression-cohort.html#cb555-16"></a>CDLoc=<span class="kw">names</span>(CDVarListB)[<span class="kw">which</span>(CDVarListB<span class="op">&gt;=</span><span class="dv">1</span>)]</span>
<span id="cb555-17"><a href="mutations-in-progression-cohort.html#cb555-17"></a></span>
<span id="cb555-18"><a href="mutations-in-progression-cohort.html#cb555-18"></a><span class="co">## Look at DN samples</span></span>
<span id="cb555-19"><a href="mutations-in-progression-cohort.html#cb555-19"></a>xa=<span class="kw">grep</span>(<span class="st">&quot;DN&quot;</span>, MallCodingHaploFuncP<span class="op">$</span>L1)</span>
<span id="cb555-20"><a href="mutations-in-progression-cohort.html#cb555-20"></a>DNVarList=<span class="kw">table</span>(MallCodingHaploFuncP<span class="op">$</span>GENE[xa])</span>
<span id="cb555-21"><a href="mutations-in-progression-cohort.html#cb555-21"></a>DNVarNames=<span class="kw">names</span>(DNVarList)</span>
<span id="cb555-22"><a href="mutations-in-progression-cohort.html#cb555-22"></a>DNVarN2=<span class="kw">names</span>(DNVarList[<span class="kw">which</span>(DNVarList<span class="op">&gt;</span><span class="dv">1</span>)])</span>
<span id="cb555-23"><a href="mutations-in-progression-cohort.html#cb555-23"></a>DNVarListB=<span class="kw">table</span>(MallCodingHaploFuncP<span class="op">$</span>mutPos[xa])</span>
<span id="cb555-24"><a href="mutations-in-progression-cohort.html#cb555-24"></a>DNLoc=<span class="kw">names</span>(DNVarListB)[<span class="kw">which</span>(DNVarListB<span class="op">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb555-25"><a href="mutations-in-progression-cohort.html#cb555-25"></a></span>
<span id="cb555-26"><a href="mutations-in-progression-cohort.html#cb555-26"></a><span class="co">#pdf(&quot;~/Desktop/Supp5-progression-RNA-specific-mutations-gene-level.pdf&quot;, height=6, width=6)</span></span>
<span id="cb555-27"><a href="mutations-in-progression-cohort.html#cb555-27"></a></span>
<span id="cb555-28"><a href="mutations-in-progression-cohort.html#cb555-28"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">Ep=</span>EpVarNames, <span class="dt">CD45=</span>CDVarNames, <span class="dt">DN=</span>DNVarNames))</span>
<span id="cb555-29"><a href="mutations-in-progression-cohort.html#cb555-29"></a><span class="kw">mtext</span>(<span class="st">&quot;no genes w mutations overlapping in different fractions&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-159-1.png" width="672" /></p>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="mutations-in-progression-cohort.html#cb556-1"></a></span>
<span id="cb556-2"><a href="mutations-in-progression-cohort.html#cb556-2"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">Ep=</span>EpVarN2, <span class="dt">CD45=</span>CDVarN2, <span class="dt">DN=</span>DNVarN2))</span>
<span id="cb556-3"><a href="mutations-in-progression-cohort.html#cb556-3"></a><span class="kw">mtext</span>(<span class="st">&quot;no genes with overlapping variants in 2+ samples&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-159-2.png" width="672" /></p>
<div class="sourceCode" id="cb557"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb557-1"><a href="mutations-in-progression-cohort.html#cb557-1"></a></span>
<span id="cb557-2"><a href="mutations-in-progression-cohort.html#cb557-2"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">Ep=</span>EpLoc, <span class="dt">CD45=</span>CDLoc, <span class="dt">DN=</span>DNLoc))</span>
<span id="cb557-3"><a href="mutations-in-progression-cohort.html#cb557-3"></a><span class="kw">mtext</span>(<span class="st">&quot;no locations with overlapping variants in 2+ samples&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-159-3.png" width="672" /></p>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="mutations-in-progression-cohort.html#cb558-1"></a></span>
<span id="cb558-2"><a href="mutations-in-progression-cohort.html#cb558-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
</div>
<div id="look-at-the-common-mutations-cosmic-ones" class="section level2">
<h2><span class="header-section-number">23.2</span> Look at the common mutations (cosmic ones)</h2>
<p>color coding:
1. splice
2. frameshift
3. missense
4. stop-gained</p>
<p>Look for genes which we reported to be common in this cohort previously, as well as the breast-cancer specific ones</p>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="mutations-in-progression-cohort.html#cb559-1"></a>GeneList=<span class="kw">c</span>(<span class="st">&quot;Atm&quot;</span>, <span class="st">&quot;Tert&quot;</span>, <span class="st">&quot;Ncor2&quot;</span>, <span class="st">&quot;Epha3&quot;</span>, <span class="st">&quot;Arid1a&quot;</span>, <span class="st">&quot;Tp53&quot;</span>, <span class="st">&quot;Pten&quot;</span>, <span class="st">&quot;Pik3ca&quot;</span>, <span class="st">&quot;Pik3r1&quot;</span>, <span class="st">&quot;Esr1&quot;</span>, <span class="st">&quot;Ctcf&quot;</span>, <span class="st">&quot;Brca1&quot;</span>, <span class="st">&quot;Brca2&quot;</span>, <span class="st">&quot;Notch1&quot;</span>, <span class="st">&quot;Kmt2d&quot;</span>, <span class="st">&quot;Foxa1&quot;</span>, <span class="st">&quot;Gata3&quot;</span>, <span class="st">&quot;Mdm4&quot;</span>, <span class="st">&quot;Ccne1&quot;</span>)</span>
<span id="cb559-2"><a href="mutations-in-progression-cohort.html#cb559-2"></a></span>
<span id="cb559-3"><a href="mutations-in-progression-cohort.html#cb559-3"></a></span>
<span id="cb559-4"><a href="mutations-in-progression-cohort.html#cb559-4"></a>MallCodingHaploFuncPC=MallCodingHaploFuncP[<span class="kw">which</span>(MallCodingHaploFuncP<span class="op">$</span>GENE<span class="op">%in%</span><span class="kw">c</span>(topCOSMIC, GeneList)), ]</span>
<span id="cb559-5"><a href="mutations-in-progression-cohort.html#cb559-5"></a></span>
<span id="cb559-6"><a href="mutations-in-progression-cohort.html#cb559-6"></a>MallCodingHaploFuncPC=MallCodingHaploFuncPC[<span class="kw">which</span>(MallCodingHaploFuncPC<span class="op">$</span>FILTER<span class="op">==</span><span class="st">&quot;PASS&quot;</span>), ]</span>
<span id="cb559-7"><a href="mutations-in-progression-cohort.html#cb559-7"></a></span>
<span id="cb559-8"><a href="mutations-in-progression-cohort.html#cb559-8"></a>xab=<span class="kw">unique</span>(MallCodingHaploFuncPC<span class="op">$</span>ANNOTATION)</span>
<span id="cb559-9"><a href="mutations-in-progression-cohort.html#cb559-9"></a></span>
<span id="cb559-10"><a href="mutations-in-progression-cohort.html#cb559-10"></a>MallCodingHaploFuncPC<span class="op">$</span>ANNOTATION[<span class="kw">grep</span>(<span class="st">&quot;frameshift_variant&quot;</span>,MallCodingHaploFuncPC<span class="op">$</span>ANNOTATION )]=<span class="st">&quot;frameshift_variant&quot;</span></span>
<span id="cb559-11"><a href="mutations-in-progression-cohort.html#cb559-11"></a></span>
<span id="cb559-12"><a href="mutations-in-progression-cohort.html#cb559-12"></a>MallCodingHaploFuncPC<span class="op">$</span>ANNOTATION[<span class="kw">grep</span>(<span class="st">&quot;missense_variant&quot;</span>,MallCodingHaploFuncPC<span class="op">$</span>ANNOTATION )]=<span class="st">&quot;missense_variant&quot;</span></span>
<span id="cb559-13"><a href="mutations-in-progression-cohort.html#cb559-13"></a></span>
<span id="cb559-14"><a href="mutations-in-progression-cohort.html#cb559-14"></a>MallCodingHaploFuncPC<span class="op">$</span>ANNOTATION=<span class="kw">as.numeric</span>(<span class="kw">factor</span>(MallCodingHaploFuncPC<span class="op">$</span>ANNOTATION, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;splice_region_variant&amp;synonymous_variant&quot;</span>, <span class="st">&quot;frameshift_variant&quot;</span>, <span class="st">&quot;missense_variant&quot;</span>, <span class="st">&quot;stop_gained&quot;</span>)))</span>
<span id="cb559-15"><a href="mutations-in-progression-cohort.html#cb559-15"></a></span>
<span id="cb559-16"><a href="mutations-in-progression-cohort.html#cb559-16"></a>AcastTab=<span class="kw">acast</span>(MallCodingHaploFuncPC[ ,<span class="kw">c</span>(<span class="st">&quot;L1&quot;</span>, <span class="st">&quot;GENE&quot;</span>, <span class="st">&quot;ANNOTATION&quot;</span>)], </span>
<span id="cb559-17"><a href="mutations-in-progression-cohort.html#cb559-17"></a>               L1<span class="op">~</span>GENE, <span class="dt">fun.aggregate =</span> max)</span>
<span id="cb559-18"><a href="mutations-in-progression-cohort.html#cb559-18"></a>AcastTab[<span class="kw">which</span>(AcastTab<span class="op">==</span><span class="st">&quot;-Inf&quot;</span>, <span class="dt">arr.ind =</span> T)]=<span class="dv">0</span></span>
<span id="cb559-19"><a href="mutations-in-progression-cohort.html#cb559-19"></a></span>
<span id="cb559-20"><a href="mutations-in-progression-cohort.html#cb559-20"></a>indx=<span class="kw">rownames</span>(AcastTab)[<span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, <span class="kw">rownames</span>(AcastTab))]</span>
<span id="cb559-21"><a href="mutations-in-progression-cohort.html#cb559-21"></a>indx=<span class="kw">substr</span>(indx, <span class="dv">1</span>, <span class="kw">nchar</span>(indx)<span class="op">-</span><span class="dv">3</span>)</span>
<span id="cb559-22"><a href="mutations-in-progression-cohort.html#cb559-22"></a></span>
<span id="cb559-23"><a href="mutations-in-progression-cohort.html#cb559-23"></a>n2=<span class="kw">unlist</span>(<span class="kw">sapply</span>(indx, <span class="cf">function</span>(x) <span class="kw">grep</span>(x, <span class="kw">rownames</span>(AcastTab))))</span>
<span id="cb559-24"><a href="mutations-in-progression-cohort.html#cb559-24"></a>AcastTab2=AcastTab</span>
<span id="cb559-25"><a href="mutations-in-progression-cohort.html#cb559-25"></a>AcastTab[<span class="kw">grep</span>(<span class="st">&quot;CD45&quot;</span>, <span class="kw">rownames</span>(AcastTab)), ]=<span class="kw">sign</span>(AcastTab[<span class="kw">grep</span>(<span class="st">&quot;CD45&quot;</span>, <span class="kw">rownames</span>(AcastTab)), ])</span>
<span id="cb559-26"><a href="mutations-in-progression-cohort.html#cb559-26"></a>AcastTab[<span class="kw">grep</span>(<span class="st">&quot;DN&quot;</span>, <span class="kw">rownames</span>(AcastTab)), ]=<span class="kw">sign</span>(AcastTab[<span class="kw">grep</span>(<span class="st">&quot;DN&quot;</span>, <span class="kw">rownames</span>(AcastTab)), ])<span class="op">*</span><span class="dv">2</span></span>
<span id="cb559-27"><a href="mutations-in-progression-cohort.html#cb559-27"></a>AcastTab[<span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, <span class="kw">rownames</span>(AcastTab)), ]=<span class="kw">sign</span>(AcastTab[<span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, <span class="kw">rownames</span>(AcastTab)), ])<span class="op">*</span><span class="dv">3</span></span>
<span id="cb559-28"><a href="mutations-in-progression-cohort.html#cb559-28"></a></span>
<span id="cb559-29"><a href="mutations-in-progression-cohort.html#cb559-29"></a><span class="co">#pdf(&quot;~/Desktop/FigureS3_biphasic_mutations_again.pdf&quot;, height=12, width=15)</span></span>
<span id="cb559-30"><a href="mutations-in-progression-cohort.html#cb559-30"></a></span>
<span id="cb559-31"><a href="mutations-in-progression-cohort.html#cb559-31"></a><span class="kw">image</span>(AcastTab[n2, ], <span class="dt">xaxt=</span><span class="st">&quot;none&quot;</span>, <span class="dt">yaxt=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;forestgreen&quot;</span>))</span>
<span id="cb559-32"><a href="mutations-in-progression-cohort.html#cb559-32"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(AcastTab[n2, ])), <span class="kw">rownames</span>(AcastTab[n2, ]), <span class="dt">las=</span><span class="dv">2</span>)</span>
<span id="cb559-33"><a href="mutations-in-progression-cohort.html#cb559-33"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(AcastTab)), <span class="kw">colnames</span>(AcastTab), <span class="dt">las=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-160-1.png" width="672" /></p>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="mutations-in-progression-cohort.html#cb560-1"></a></span>
<span id="cb560-2"><a href="mutations-in-progression-cohort.html#cb560-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p>View the above using only the Epcam samples:</p>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="mutations-in-progression-cohort.html#cb561-1"></a>Acast2=AcastTab2[<span class="kw">grep</span>(<span class="st">&quot;Ep&quot;</span>, <span class="kw">rownames</span>(AcastTab)), ]</span>
<span id="cb561-2"><a href="mutations-in-progression-cohort.html#cb561-2"></a>rS=<span class="kw">colSums</span>(Acast2)</span>
<span id="cb561-3"><a href="mutations-in-progression-cohort.html#cb561-3"></a>Acast2=Acast2[ ,<span class="op">-</span><span class="kw">which</span>(rS<span class="op">==</span><span class="dv">0</span>)]</span>
<span id="cb561-4"><a href="mutations-in-progression-cohort.html#cb561-4"></a></span>
<span id="cb561-5"><a href="mutations-in-progression-cohort.html#cb561-5"></a><span class="co">#pdf(&quot;~/Desktop/FigureS3_common_mutations_again.pdf&quot;, height=12, width=12)</span></span>
<span id="cb561-6"><a href="mutations-in-progression-cohort.html#cb561-6"></a><span class="co"># also group this according to growing vs stable</span></span>
<span id="cb561-7"><a href="mutations-in-progression-cohort.html#cb561-7"></a>m2=<span class="kw">melt</span>(Acast2)</span>
<span id="cb561-8"><a href="mutations-in-progression-cohort.html#cb561-8"></a><span class="kw">head</span>(m2)</span>
<span id="cb561-9"><a href="mutations-in-progression-cohort.html#cb561-9"></a><span class="co">##        Var1  Var2 value</span></span>
<span id="cb561-10"><a href="mutations-in-progression-cohort.html#cb561-10"></a><span class="co">## 1  10L_D_Ep Akap9     0</span></span>
<span id="cb561-11"><a href="mutations-in-progression-cohort.html#cb561-11"></a><span class="co">## 2 10R_BL_Ep Akap9     0</span></span>
<span id="cb561-12"><a href="mutations-in-progression-cohort.html#cb561-12"></a><span class="co">## 3 10R_BU_Ep Akap9     0</span></span>
<span id="cb561-13"><a href="mutations-in-progression-cohort.html#cb561-13"></a><span class="co">## 4  11L_B_Ep Akap9     0</span></span>
<span id="cb561-14"><a href="mutations-in-progression-cohort.html#cb561-14"></a><span class="co">## 5  11N_D_Ep Akap9     0</span></span>
<span id="cb561-15"><a href="mutations-in-progression-cohort.html#cb561-15"></a><span class="co">## 6  11R_C_Ep Akap9     0</span></span>
<span id="cb561-16"><a href="mutations-in-progression-cohort.html#cb561-16"></a>m2<span class="op">$</span>growth=Cdata<span class="op">$</span>Tumor.Growth[<span class="kw">match</span>(<span class="kw">substr</span>(m2<span class="op">$</span>Var1, <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">as.character</span>(m2<span class="op">$</span>Var1))<span class="op">-</span><span class="dv">3</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb561-17"><a href="mutations-in-progression-cohort.html#cb561-17"></a>m2<span class="op">$</span>Treatment=Cdata<span class="op">$</span>Treatment[<span class="kw">match</span>(<span class="kw">substr</span>(m2<span class="op">$</span>Var1, <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">as.character</span>(m2<span class="op">$</span>Var1))<span class="op">-</span><span class="dv">3</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb561-18"><a href="mutations-in-progression-cohort.html#cb561-18"></a>m2<span class="op">$</span>newid=Cdata<span class="op">$</span>NewID[<span class="kw">match</span>(<span class="kw">substr</span>(m2<span class="op">$</span>Var1, <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">as.character</span>(m2<span class="op">$</span>Var1))<span class="op">-</span><span class="dv">3</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb561-19"><a href="mutations-in-progression-cohort.html#cb561-19"></a></span>
<span id="cb561-20"><a href="mutations-in-progression-cohort.html#cb561-20"></a><span class="kw">ggplot</span>(m2, <span class="kw">aes</span>(<span class="dt">x=</span>newid, <span class="dt">y=</span>Var2, <span class="dt">fill=</span><span class="kw">factor</span>(value)))<span class="op">+</span><span class="kw">geom_tile</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>growth, <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))<span class="op">+</span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>,<span class="st">&quot;red&quot;</span>, <span class="st">&quot;forestgreen&quot;</span>, <span class="st">&quot;orange&quot;</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/summary-mutations-rna-Ep-1.png" width="672" /></p>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="mutations-in-progression-cohort.html#cb562-1"></a></span>
<span id="cb562-2"><a href="mutations-in-progression-cohort.html#cb562-2"></a><span class="kw">pdf</span>(<span class="st">&quot;~/Desktop/FigureS3_common_mutations_ep_again.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">12</span>, <span class="dt">width=</span><span class="dv">12</span>)</span>
<span id="cb562-3"><a href="mutations-in-progression-cohort.html#cb562-3"></a><span class="kw">ggplot</span>(m2, <span class="kw">aes</span>(<span class="dt">x=</span>newid, <span class="dt">y=</span>Var2, <span class="dt">fill=</span><span class="kw">factor</span>(value)))<span class="op">+</span><span class="kw">geom_tile</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>Treatment<span class="op">+</span>growth, <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))<span class="op">+</span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>,<span class="st">&quot;red&quot;</span>, <span class="st">&quot;forestgreen&quot;</span>, <span class="st">&quot;orange&quot;</span>))</span>
<span id="cb562-4"><a href="mutations-in-progression-cohort.html#cb562-4"></a><span class="kw">dev.off</span>()</span>
<span id="cb562-5"><a href="mutations-in-progression-cohort.html#cb562-5"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb562-6"><a href="mutations-in-progression-cohort.html#cb562-6"></a><span class="co">##                 2</span></span>
<span id="cb562-7"><a href="mutations-in-progression-cohort.html#cb562-7"></a></span>
<span id="cb562-8"><a href="mutations-in-progression-cohort.html#cb562-8"></a>Candidates=<span class="kw">c</span>(<span class="st">&quot;Polg&quot;</span>, <span class="st">&quot;Foxa1&quot;</span>, <span class="st">&quot;Ncor2&quot;</span>, <span class="st">&quot;Atm&quot;</span>)</span>
<span id="cb562-9"><a href="mutations-in-progression-cohort.html#cb562-9"></a>pvals=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(Candidates))</span>
<span id="cb562-10"><a href="mutations-in-progression-cohort.html#cb562-10"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Candidates)){</span>
<span id="cb562-11"><a href="mutations-in-progression-cohort.html#cb562-11"></a>  xa=<span class="kw">table</span>(m2<span class="op">$</span>value[<span class="kw">which</span>(m2<span class="op">$</span>Var2<span class="op">==</span>Candidates[i])], m2<span class="op">$</span>growth[<span class="kw">which</span>(m2<span class="op">$</span>Var2<span class="op">==</span>Candidates[i])])</span>
<span id="cb562-12"><a href="mutations-in-progression-cohort.html#cb562-12"></a>   pvals[i]=<span class="kw">fisher.test</span>(xa)<span class="op">$</span>p.value</span>
<span id="cb562-13"><a href="mutations-in-progression-cohort.html#cb562-13"></a>}</span></code></pre></div>
<p>Repeat in the DN samples</p>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="mutations-in-progression-cohort.html#cb563-1"></a>Acast2=AcastTab2[<span class="kw">grep</span>(<span class="st">&quot;DN&quot;</span>, <span class="kw">rownames</span>(AcastTab)), ]</span>
<span id="cb563-2"><a href="mutations-in-progression-cohort.html#cb563-2"></a>rS=<span class="kw">colSums</span>(Acast2)</span>
<span id="cb563-3"><a href="mutations-in-progression-cohort.html#cb563-3"></a>Acast2=Acast2[ ,<span class="op">-</span><span class="kw">which</span>(rS<span class="op">==</span><span class="dv">0</span>)]</span>
<span id="cb563-4"><a href="mutations-in-progression-cohort.html#cb563-4"></a></span>
<span id="cb563-5"><a href="mutations-in-progression-cohort.html#cb563-5"></a><span class="co">#pdf(&quot;~/Desktop/FigureS3_common_mutations_again.pdf&quot;, height=12, width=12)</span></span>
<span id="cb563-6"><a href="mutations-in-progression-cohort.html#cb563-6"></a><span class="co"># also group this according to growing vs stable</span></span>
<span id="cb563-7"><a href="mutations-in-progression-cohort.html#cb563-7"></a>m3=<span class="kw">melt</span>(Acast2)</span>
<span id="cb563-8"><a href="mutations-in-progression-cohort.html#cb563-8"></a><span class="kw">head</span>(m3)</span>
<span id="cb563-9"><a href="mutations-in-progression-cohort.html#cb563-9"></a><span class="co">##        Var1 Var2 value</span></span>
<span id="cb563-10"><a href="mutations-in-progression-cohort.html#cb563-10"></a><span class="co">## 1  10L_C_DN  Atm     0</span></span>
<span id="cb563-11"><a href="mutations-in-progression-cohort.html#cb563-11"></a><span class="co">## 2 10R_BU_DN  Atm     0</span></span>
<span id="cb563-12"><a href="mutations-in-progression-cohort.html#cb563-12"></a><span class="co">## 3  11L_B_DN  Atm     0</span></span>
<span id="cb563-13"><a href="mutations-in-progression-cohort.html#cb563-13"></a><span class="co">## 4  11R_C_DN  Atm     0</span></span>
<span id="cb563-14"><a href="mutations-in-progression-cohort.html#cb563-14"></a><span class="co">## 5  11R_D_DN  Atm     0</span></span>
<span id="cb563-15"><a href="mutations-in-progression-cohort.html#cb563-15"></a><span class="co">## 6  12L_D_DN  Atm     0</span></span>
<span id="cb563-16"><a href="mutations-in-progression-cohort.html#cb563-16"></a>m3<span class="op">$</span>growth=Cdata<span class="op">$</span>Tumor.Growth[<span class="kw">match</span>(<span class="kw">substr</span>(m3<span class="op">$</span>Var1, <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">as.character</span>(m3<span class="op">$</span>Var1))<span class="op">-</span><span class="dv">3</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb563-17"><a href="mutations-in-progression-cohort.html#cb563-17"></a>m3<span class="op">$</span>Treatment=Cdata<span class="op">$</span>Treatment[<span class="kw">match</span>(<span class="kw">substr</span>(m3<span class="op">$</span>Var1, <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">as.character</span>(m3<span class="op">$</span>Var1))<span class="op">-</span><span class="dv">3</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb563-18"><a href="mutations-in-progression-cohort.html#cb563-18"></a>m3<span class="op">$</span>newid=Cdata<span class="op">$</span>NewID[<span class="kw">match</span>(<span class="kw">substr</span>(m3<span class="op">$</span>Var1, <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">as.character</span>(m3<span class="op">$</span>Var1))<span class="op">-</span><span class="dv">3</span>), Cdata<span class="op">$</span>TumorID)]</span>
<span id="cb563-19"><a href="mutations-in-progression-cohort.html#cb563-19"></a></span>
<span id="cb563-20"><a href="mutations-in-progression-cohort.html#cb563-20"></a><span class="kw">ggplot</span>(m3, <span class="kw">aes</span>(<span class="dt">x=</span>newid, <span class="dt">y=</span>Var2, <span class="dt">fill=</span><span class="kw">factor</span>(value)))<span class="op">+</span><span class="kw">geom_tile</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>growth, <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))<span class="op">+</span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>,<span class="st">&quot;red&quot;</span>, <span class="st">&quot;forestgreen&quot;</span>, <span class="st">&quot;orange&quot;</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/summary-mutations-rna-DN-1.png" width="672" /></p>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="mutations-in-progression-cohort.html#cb564-1"></a></span>
<span id="cb564-2"><a href="mutations-in-progression-cohort.html#cb564-2"></a><span class="kw">pdf</span>(<span class="st">&quot;~/Desktop/FigureS3_common_mutations_dn_again.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">12</span>, <span class="dt">width=</span><span class="dv">12</span>)</span>
<span id="cb564-3"><a href="mutations-in-progression-cohort.html#cb564-3"></a><span class="kw">ggplot</span>(m3, <span class="kw">aes</span>(<span class="dt">x=</span>newid, <span class="dt">y=</span>Var2, <span class="dt">fill=</span><span class="kw">factor</span>(value)))<span class="op">+</span><span class="kw">geom_tile</span>()<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>Treatment<span class="op">+</span>growth, <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))<span class="op">+</span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>,<span class="st">&quot;red&quot;</span>, <span class="st">&quot;forestgreen&quot;</span>, <span class="st">&quot;orange&quot;</span>))</span>
<span id="cb564-4"><a href="mutations-in-progression-cohort.html#cb564-4"></a><span class="kw">dev.off</span>()</span>
<span id="cb564-5"><a href="mutations-in-progression-cohort.html#cb564-5"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb564-6"><a href="mutations-in-progression-cohort.html#cb564-6"></a><span class="co">##                 2</span></span>
<span id="cb564-7"><a href="mutations-in-progression-cohort.html#cb564-7"></a></span>
<span id="cb564-8"><a href="mutations-in-progression-cohort.html#cb564-8"></a>Candidates=<span class="kw">c</span>(<span class="st">&quot;Casp3&quot;</span>, <span class="st">&quot;Hras&quot;</span>, <span class="st">&quot;Fat1&quot;</span>, <span class="st">&quot;Setd2&quot;</span>, <span class="st">&quot;Ranbp2&quot;</span>, <span class="st">&quot;Pold1&quot;</span>, <span class="st">&quot;Pms2&quot;</span>)</span>
<span id="cb564-9"><a href="mutations-in-progression-cohort.html#cb564-9"></a>pvals=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(Candidates))</span>
<span id="cb564-10"><a href="mutations-in-progression-cohort.html#cb564-10"></a>x1=<span class="kw">table</span>(m3<span class="op">$</span>growth[<span class="kw">which</span>(m3<span class="op">$</span>Var2<span class="op">==</span>Candidates[<span class="dv">1</span>])])</span>
<span id="cb564-11"><a href="mutations-in-progression-cohort.html#cb564-11"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Candidates)){</span>
<span id="cb564-12"><a href="mutations-in-progression-cohort.html#cb564-12"></a>  xa=<span class="kw">prop.test</span>(<span class="kw">c</span>(</span>
<span id="cb564-13"><a href="mutations-in-progression-cohort.html#cb564-13"></a>    <span class="kw">length</span>(<span class="kw">which</span>(m3<span class="op">$</span>value[<span class="kw">which</span>(m3<span class="op">$</span>growth<span class="op">==</span><span class="st">&quot;growing&quot;</span> <span class="op">&amp;</span><span class="st"> </span>m3<span class="op">$</span>Var2<span class="op">==</span>Candidates[i])]<span class="op">&gt;</span><span class="dv">0</span>)), </span>
<span id="cb564-14"><a href="mutations-in-progression-cohort.html#cb564-14"></a>    <span class="kw">length</span>(<span class="kw">which</span>(m3<span class="op">$</span>value[<span class="kw">which</span>(m3<span class="op">$</span>growth<span class="op">==</span><span class="st">&quot;stable&quot;</span> <span class="op">&amp;</span><span class="st"> </span>m3<span class="op">$</span>Var2<span class="op">==</span>Candidates[i])]<span class="op">&gt;</span><span class="dv">0</span>))), x1)</span>
<span id="cb564-15"><a href="mutations-in-progression-cohort.html#cb564-15"></a>   pvals[i]=xa<span class="op">$</span>p.value</span>
<span id="cb564-16"><a href="mutations-in-progression-cohort.html#cb564-16"></a>}</span>
<span id="cb564-17"><a href="mutations-in-progression-cohort.html#cb564-17"></a></span>
<span id="cb564-18"><a href="mutations-in-progression-cohort.html#cb564-18"></a>pvals</span>
<span id="cb564-19"><a href="mutations-in-progression-cohort.html#cb564-19"></a><span class="co">## [1] 0.3774645 0.7197694 0.9561687 0.3774645 1.0000000 0.3881479 0.1133864</span></span>
<span id="cb564-20"><a href="mutations-in-progression-cohort.html#cb564-20"></a></span>
<span id="cb564-21"><a href="mutations-in-progression-cohort.html#cb564-21"></a>Candidates=<span class="kw">c</span>(<span class="st">&quot;Casp3&quot;</span>, <span class="st">&quot;Hras&quot;</span>, <span class="st">&quot;Fat1&quot;</span>, <span class="st">&quot;Setd2&quot;</span>, <span class="st">&quot;Ranbp2&quot;</span>, <span class="st">&quot;Pold1&quot;</span>, <span class="st">&quot;Pms2&quot;</span>)</span>
<span id="cb564-22"><a href="mutations-in-progression-cohort.html#cb564-22"></a>pvals=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(Candidates))</span>
<span id="cb564-23"><a href="mutations-in-progression-cohort.html#cb564-23"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Candidates)){</span>
<span id="cb564-24"><a href="mutations-in-progression-cohort.html#cb564-24"></a>  xa=<span class="kw">table</span>(m3<span class="op">$</span>value[<span class="kw">which</span>(m3<span class="op">$</span>Var2<span class="op">==</span>Candidates[i])], m3<span class="op">$</span>growth[<span class="kw">which</span>(m3<span class="op">$</span>Var2<span class="op">==</span>Candidates[i])])</span>
<span id="cb564-25"><a href="mutations-in-progression-cohort.html#cb564-25"></a>   pvals[i]=<span class="kw">fisher.test</span>(xa)<span class="op">$</span>p.value</span>
<span id="cb564-26"><a href="mutations-in-progression-cohort.html#cb564-26"></a>}</span></code></pre></div>
<p>Also compare here if there is a difference in frequency between DN and Ep samples:</p>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="mutations-in-progression-cohort.html#cb565-1"></a></span>
<span id="cb565-2"><a href="mutations-in-progression-cohort.html#cb565-2"></a>Candidates=<span class="kw">c</span>(<span class="st">&quot;Casp3&quot;</span>, <span class="st">&quot;Hras&quot;</span>, <span class="st">&quot;Fat1&quot;</span>, <span class="st">&quot;Setd2&quot;</span>, <span class="st">&quot;Ranbp2&quot;</span>, <span class="st">&quot;Pold1&quot;</span>, <span class="st">&quot;Pms2&quot;</span>, <span class="st">&quot;Bcl9l&quot;</span>,<span class="st">&quot;Atm&quot;</span> )</span>
<span id="cb565-3"><a href="mutations-in-progression-cohort.html#cb565-3"></a>pvals=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(Candidates))</span>
<span id="cb565-4"><a href="mutations-in-progression-cohort.html#cb565-4"></a>x1=<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">which</span>(m2<span class="op">$</span>Var2<span class="op">==</span>Candidates[<span class="dv">1</span>])), <span class="kw">length</span>(<span class="kw">which</span>(m3<span class="op">$</span>Var2<span class="op">==</span>Candidates[<span class="dv">1</span>])))</span>
<span id="cb565-5"><a href="mutations-in-progression-cohort.html#cb565-5"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Candidates)){</span>
<span id="cb565-6"><a href="mutations-in-progression-cohort.html#cb565-6"></a>   xa=<span class="kw">prop.test</span>(<span class="kw">c</span>(</span>
<span id="cb565-7"><a href="mutations-in-progression-cohort.html#cb565-7"></a>    <span class="kw">length</span>(<span class="kw">which</span>(m2<span class="op">$</span>value[<span class="kw">which</span>(  m2<span class="op">$</span>Var2<span class="op">==</span>Candidates[i])]<span class="op">&gt;</span><span class="dv">0</span>)), </span>
<span id="cb565-8"><a href="mutations-in-progression-cohort.html#cb565-8"></a>    <span class="kw">length</span>(<span class="kw">which</span>(m3<span class="op">$</span>value[<span class="kw">which</span>( m3<span class="op">$</span>Var2<span class="op">==</span>Candidates[i])]<span class="op">&gt;</span><span class="dv">0</span>))), x1)</span>
<span id="cb565-9"><a href="mutations-in-progression-cohort.html#cb565-9"></a>   pvals[i]=xa<span class="op">$</span>p.value</span>
<span id="cb565-10"><a href="mutations-in-progression-cohort.html#cb565-10"></a>}</span>
<span id="cb565-11"><a href="mutations-in-progression-cohort.html#cb565-11"></a></span>
<span id="cb565-12"><a href="mutations-in-progression-cohort.html#cb565-12"></a>pvals</span>
<span id="cb565-13"><a href="mutations-in-progression-cohort.html#cb565-13"></a><span class="co">## [1] 0.31627188 0.30894961 0.16343819 0.21479916 0.26039249 1.00000000 0.28667561</span></span>
<span id="cb565-14"><a href="mutations-in-progression-cohort.html#cb565-14"></a><span class="co">## [8] 0.03536095 0.87669802</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutations-in-rna.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="trichrome-staining.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
