% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Rat DCIS study},
  pdfauthor={Anne Trinh},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{booktabs}
\usepackage[]{natbib}
\bibliographystyle{apalike}

\title{Rat DCIS study}
\author{Anne Trinh}
\date{2021-02-18}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\hypertarget{preface}{%
\chapter*{Preface}\label{preface}}
\addcontentsline{toc}{chapter}{Preface}

Placeholder

\hypertarget{packages-and-software}{%
\section{Packages and Software}\label{packages-and-software}}

\hypertarget{external-software}{%
\section{External software}\label{external-software}}

\hypertarget{annotations}{%
\section{Annotations}\label{annotations}}

\hypertarget{genomic-properties}{%
\subsection{Genomic properties}\label{genomic-properties}}

\hypertarget{gene-name-homologs-between-organisms}{%
\subsection{Gene name homologs between organisms}\label{gene-name-homologs-between-organisms}}

\hypertarget{gene-signatures-and-data-bases}{%
\subsection{Gene signatures and data-bases}\label{gene-signatures-and-data-bases}}

\hypertarget{human-gene-homologs}{%
\subsubsection{Human gene homologs}\label{human-gene-homologs}}

\hypertarget{gsea-compendiums}{%
\subsubsection{GSEA compendiums}\label{gsea-compendiums}}

\hypertarget{cohort-summary}{%
\chapter{Cohort summary}\label{cohort-summary}}

Placeholder

\hypertarget{size-information}{%
\section{Size information}\label{size-information}}

\hypertarget{calculating-growth-rates}{%
\section{Calculating growth rates}\label{calculating-growth-rates}}

\hypertarget{facs-data-dncd45epcam}{%
\section{FACS data (DN/CD45/EpCAM)}\label{facs-data-dncd45epcam}}

\hypertarget{facs-data}{%
\section{FACS data}\label{facs-data}}

\hypertarget{summary-table}{%
\section{Summary Table}\label{summary-table}}

\hypertarget{compare-the-characterisation-vs-progression-cohort}{%
\subsection{Compare the characterisation vs progression cohort}\label{compare-the-characterisation-vs-progression-cohort}}

\hypertarget{summary-of-the-rna-data}{%
\subsection{Summary of the RNA data}\label{summary-of-the-rna-data}}

\hypertarget{whole-slide-imaging}{%
\chapter{Whole-slide imaging}\label{whole-slide-imaging}}

Placeholder

\hypertarget{associate-the-frequencies-with-other-data-types}{%
\section{Associate the frequencies with other data types}\label{associate-the-frequencies-with-other-data-types}}

\hypertarget{cellular-composition}{%
\section{Cellular composition}\label{cellular-composition}}

\hypertarget{associate-composition-with-other-covariates}{%
\section{Associate composition with other covariates}\label{associate-composition-with-other-covariates}}

\hypertarget{estimate-tumor-size}{%
\section{Estimate tumor size}\label{estimate-tumor-size}}

\hypertarget{correlations-between-different-subpopulations}{%
\section{Correlations between different subpopulations}\label{correlations-between-different-subpopulations}}

\hypertarget{associations-between-cd8-counts-with-other-clinical-variables}{%
\section{Associations between CD8 counts with other clinical variables}\label{associations-between-cd8-counts-with-other-clinical-variables}}

\hypertarget{spatial-statistics}{%
\chapter{Spatial statistics}\label{spatial-statistics}}

Placeholder

\hypertarget{knn-distances}{%
\section{knn-Distances:}\label{knn-distances}}

\hypertarget{comparison-to-manual-classification}{%
\subsection{Comparison to manual classification}\label{comparison-to-manual-classification}}

\hypertarget{associations-with-outcome-to-treatment}{%
\subsection{Associations with outcome to treatment}\label{associations-with-outcome-to-treatment}}

\hypertarget{growth}{%
\subsection{Growth}\label{growth}}

\hypertarget{the-interacting-fraction}{%
\section{The interacting fraction}\label{the-interacting-fraction}}

\hypertarget{comparison-to-manual-select-optimal-r}{%
\subsection{Comparison to manual \& select optimal r}\label{comparison-to-manual-select-optimal-r}}

\hypertarget{growth-1}{%
\subsection{Growth}\label{growth-1}}

\hypertarget{treatment}{%
\subsection{Treatment}\label{treatment}}

\hypertarget{m-h-distances}{%
\section{M-H distances}\label{m-h-distances}}

\hypertarget{comparison-to-manual-scoring}{%
\subsection{Comparison to Manual Scoring}\label{comparison-to-manual-scoring}}

\hypertarget{growth-2}{%
\subsection{Growth}\label{growth-2}}

\hypertarget{treatment-1}{%
\subsection{Treatment}\label{treatment-1}}

\hypertarget{comparison-between-metrics}{%
\section{Comparison between metrics}\label{comparison-between-metrics}}

\hypertarget{other-cell-types}{%
\section{Other cell types}\label{other-cell-types}}

\hypertarget{expression-data}{%
\chapter{Expression data}\label{expression-data}}

Placeholder

\hypertarget{running-alignment}{%
\section{Running alignment}\label{running-alignment}}

\hypertarget{rna-initial-qc}{%
\section{RNA Initial QC}\label{rna-initial-qc}}

\hypertarget{normalisation}{%
\section{Normalisation}\label{normalisation}}

\hypertarget{preliminary-visualisation-to-remove-outliers}{%
\subsection{preliminary visualisation (to remove outliers)}\label{preliminary-visualisation-to-remove-outliers}}

\hypertarget{processing-files-for-external-software}{%
\section{Processing files for external software}\label{processing-files-for-external-software}}

\hypertarget{rna-data-preliminary-plots}{%
\chapter{RNA data: preliminary plots}\label{rna-data-preliminary-plots}}

Placeholder

\hypertarget{pca-plots}{%
\section{PCA plots}\label{pca-plots}}

\hypertarget{expression-patterns-by-cell-type}{%
\section{Expression patterns by cell type}\label{expression-patterns-by-cell-type}}

\hypertarget{deseq-analysis-progressionimmunotherapy-cohort}{%
\chapter{DESeq analysis: Progression/Immunotherapy cohort}\label{deseq-analysis-progressionimmunotherapy-cohort}}

Placeholder

\hypertarget{dn-vs-ep}{%
\section{DN vs Ep}\label{dn-vs-ep}}

\hypertarget{no.-samples-in-comparisons}{%
\section{No.~samples in comparisons}\label{no.-samples-in-comparisons}}

\hypertarget{set-up-cell-type-specific-the-comparisons}{%
\section{Set-up cell-type specific the comparisons}\label{set-up-cell-type-specific-the-comparisons}}

\hypertarget{pca-plots-1}{%
\section{PCA plots}\label{pca-plots-1}}

\hypertarget{epcam}{%
\subsection{EpCAM}\label{epcam}}

\hypertarget{cd45}{%
\subsection{CD45}\label{cd45}}

\hypertarget{dn}{%
\subsection{DN}\label{dn}}

\hypertarget{pearson-correlation-plots-of-samples}{%
\section{Pearson correlation plots of samples}\label{pearson-correlation-plots-of-samples}}

\hypertarget{collating-results-and-running-gsea}{%
\chapter{Collating results and running GSEA}\label{collating-results-and-running-gsea}}

Placeholder

\hypertarget{erpgr-subtyping}{%
\chapter{ER/Pgr Subtyping}\label{erpgr-subtyping}}

Placeholder

\hypertarget{gene-list}{%
\section{Gene List}\label{gene-list}}

\hypertarget{progression-cohort}{%
\section{Progression cohort}\label{progression-cohort}}

\hypertarget{charcterisation-cohort}{%
\section{Charcterisation cohort}\label{charcterisation-cohort}}

\hypertarget{dn-samples}{%
\section{DN samples}\label{dn-samples}}

\hypertarget{comparison-with-staining}{%
\section{Comparison with staining}\label{comparison-with-staining}}

\hypertarget{summary-of-expression-markers-for-each-subtypecell-fraction}{%
\section{Summary of expression markers for each subtype/cell fraction}\label{summary-of-expression-markers-for-each-subtypecell-fraction}}

\hypertarget{dn-samples-1}{%
\subsubsection{DN samples:}\label{dn-samples-1}}

\hypertarget{cd45-samples}{%
\subsubsection{CD45 samples:}\label{cd45-samples}}

\hypertarget{epithelial-samples}{%
\subsubsection{Epithelial samples:}\label{epithelial-samples}}

\hypertarget{deseq-analysis-immunotherapygrowth-comparisons}{%
\chapter{DESeq analysis: Immunotherapy/Growth comparisons}\label{deseq-analysis-immunotherapygrowth-comparisons}}

Placeholder

\hypertarget{summary-of-comparisons}{%
\section{Summary of comparisons}\label{summary-of-comparisons}}

\hypertarget{deseq-any-treatment-vs-control}{%
\section{DESeq: any treatment vs control}\label{deseq-any-treatment-vs-control}}

\hypertarget{epcam-samples}{%
\subsection{EPCAM samples}\label{epcam-samples}}

\hypertarget{growing-vs-stable-emphasis}{%
\section{Growing vs stable emphasis}\label{growing-vs-stable-emphasis}}

\hypertarget{dn-fraction}{%
\subsection{DN fraction}\label{dn-fraction}}

\hypertarget{cd-fraction}{%
\subsection{CD fraction}\label{cd-fraction}}

\hypertarget{ep-fraction}{%
\subsection{Ep Fraction}\label{ep-fraction}}

\hypertarget{summary-of-gsea-runs}{%
\chapter{Summary of GSEA runs}\label{summary-of-gsea-runs}}

Placeholder

\hypertarget{stable-vs-growing-all-samples}{%
\section{Stable vs growing: all samples}\label{stable-vs-growing-all-samples}}

\hypertarget{barplots-of-enriched-pathways}{%
\subsection{barplots of enriched pathways}\label{barplots-of-enriched-pathways}}

\hypertarget{comparisons-based-on-treatment}{%
\section{Comparisons based on treatment}\label{comparisons-based-on-treatment}}

\hypertarget{pathways-of-interest-2}{%
\section{Pathways of Interest 2}\label{pathways-of-interest-2}}

\hypertarget{epcam-inflammatory-vs-non-inflammatory-samples}{%
\chapter{Epcam+ Inflammatory vs non-inflammatory samples}\label{epcam-inflammatory-vs-non-inflammatory-samples}}

Placeholder

\hypertarget{comparison-of-enriched-pathways-across-samples}{%
\section{Comparison of enriched pathways across samples}\label{comparison-of-enriched-pathways-across-samples}}

\hypertarget{deseq-comparison}{%
\section{DeSeq comparison}\label{deseq-comparison}}

\hypertarget{compare-inflammatory-vs-non-inflammatory}{%
\subsection{Compare inflammatory vs non-inflammatory}\label{compare-inflammatory-vs-non-inflammatory}}

\hypertarget{finding-a-signature-for-each-branch}{%
\subsection{Finding a signature for each branch}\label{finding-a-signature-for-each-branch}}

\hypertarget{analyse-the-non-inflammatory-samples}{%
\section{Analyse the non-inflammatory samples}\label{analyse-the-non-inflammatory-samples}}

\hypertarget{non-inflammatory-branch-ep-only-samples}{%
\subsection{Non-inflammatory branch: Ep only samples}\label{non-inflammatory-branch-ep-only-samples}}

\hypertarget{gsea}{%
\subsection{GSEA}\label{gsea}}

\hypertarget{same-analysis-with-only-luminal-samples}{%
\section{Same analysis with only luminal samples}\label{same-analysis-with-only-luminal-samples}}

\hypertarget{associate-epcam-inflammatory-with-survival}{%
\chapter{Associate Epcam+ inflammatory with survival}\label{associate-epcam-inflammatory-with-survival}}

Placeholder

\hypertarget{associating-cd74-with-phenotype-and-outcome}{%
\section{Associating CD74 with phenotype and outcome}\label{associating-cd74-with-phenotype-and-outcome}}

\hypertarget{signature-lum-cases-non-inflammatory-growing-vs-stable}{%
\section{Signature: Lum cases non-inflammatory: growing vs stable}\label{signature-lum-cases-non-inflammatory-growing-vs-stable}}

\hypertarget{deseq-analysis-characterisation-cohort-big-vs-small}{%
\chapter{DESeq analysis: Characterisation cohort (big vs small)}\label{deseq-analysis-characterisation-cohort-big-vs-small}}

Placeholder

\hypertarget{cd45-samples-1}{%
\section{CD45 samples}\label{cd45-samples-1}}

\hypertarget{pca-plot}{%
\subsection{PCA plot}\label{pca-plot}}

\hypertarget{differential-gene-expression}{%
\subsection{Differential Gene Expression}\label{differential-gene-expression}}

\hypertarget{gsea-1}{%
\subsection{GSEA}\label{gsea-1}}

\hypertarget{epithelial-samples-1}{%
\section{Epithelial samples}\label{epithelial-samples-1}}

\hypertarget{pca-plot-1}{%
\subsection{PCA plot}\label{pca-plot-1}}

\hypertarget{gsea-2}{%
\subsection{GSEA}\label{gsea-2}}

\hypertarget{check-expression-of-checkpoint-proteins}{%
\section{Check expression of checkpoint proteins}\label{check-expression-of-checkpoint-proteins}}

\hypertarget{signature-analysis}{%
\chapter{Signature analysis}\label{signature-analysis}}

Placeholder

\hypertarget{mhc-signature-analysis}{%
\section{MHC signature analysis}\label{mhc-signature-analysis}}

\hypertarget{mhc-i}{%
\subsection{MHC-I}\label{mhc-i}}

\hypertarget{mhc-ii}{%
\subsection{MHC-II}\label{mhc-ii}}

\hypertarget{mhc-presentation-proteins}{%
\subsection{MHC presentation proteins}\label{mhc-presentation-proteins}}

\hypertarget{inflammation-related-genes-il6-jak-stat-and-tnfnfkb}{%
\subsection{inflammation related genes: IL6-JAK-STAT and TNF/NFKB}\label{inflammation-related-genes-il6-jak-stat-and-tnfnfkb}}

\hypertarget{immune-estimation}{%
\chapter{Immune estimation}\label{immune-estimation}}

Placeholder

\hypertarget{overview-of-the-cell-types}{%
\section{Overview of the cell types}\label{overview-of-the-cell-types}}

\hypertarget{characterisation-cohort-assoc-with-size}{%
\section{characterisation cohort: assoc with size}\label{characterisation-cohort-assoc-with-size}}

\hypertarget{comparison-with-facs-data}{%
\section{Comparison with FACS data}\label{comparison-with-facs-data}}

\hypertarget{progression-cohort-1}{%
\section{Progression cohort}\label{progression-cohort-1}}

\hypertarget{clinical-associations}{%
\section{Clinical associations}\label{clinical-associations}}

\hypertarget{associate-with-treatment}{%
\subsection{Associate with Treatment}\label{associate-with-treatment}}

\hypertarget{association-with-growth}{%
\subsection{Association with Growth}\label{association-with-growth}}

\hypertarget{summary-of-the-outcome}{%
\section{Summary of the outcome}\label{summary-of-the-outcome}}

\hypertarget{bcr-clonotype-analysis}{%
\chapter{BCR clonotype analysis}\label{bcr-clonotype-analysis}}

Placeholder

\hypertarget{processing-summary}{%
\section{Processing summary}\label{processing-summary}}

\hypertarget{summary-stats}{%
\section{Summary Stats}\label{summary-stats}}

\hypertarget{diversity-metrics}{%
\section{Diversity metrics}\label{diversity-metrics}}

\hypertarget{compare-the-characterisation-cohort}{%
\section{Compare the characterisation cohort}\label{compare-the-characterisation-cohort}}

\hypertarget{associate-with-clinicopathological-data-progression}{%
\section{Associate with clinicopathological data (progression)}\label{associate-with-clinicopathological-data-progression}}

\hypertarget{associate-with-signature-scores}{%
\section{Associate with signature scores}\label{associate-with-signature-scores}}

\hypertarget{whole-genome-sequencing-mutations}{%
\chapter{Whole Genome Sequencing Mutations}\label{whole-genome-sequencing-mutations}}

Placeholder

\hypertarget{data}{%
\section{Data}\label{data}}

\hypertarget{extract-mutation-signatures}{%
\subsection{Extract mutation signatures}\label{extract-mutation-signatures}}

\hypertarget{annotate-the-data-with-human-common-variants}{%
\subsection{Annotate the data with human common variants}\label{annotate-the-data-with-human-common-variants}}

\hypertarget{plots}{%
\section{Plots}\label{plots}}

\hypertarget{quick-overview}{%
\subsection{Quick overview}\label{quick-overview}}

\hypertarget{coding-variants-tmb}{%
\subsection{Coding variants \& TMB}\label{coding-variants-tmb}}

\hypertarget{coding-variants}{%
\section{Coding variants}\label{coding-variants}}

\hypertarget{sites-which-are-commonly-mutated}{%
\subsection{Sites which are commonly mutated?}\label{sites-which-are-commonly-mutated}}

\hypertarget{overview-of-the-mutations}{%
\section{Overview of the mutations}\label{overview-of-the-mutations}}

\hypertarget{all-mutations}{%
\subsection{All mutations}\label{all-mutations}}

\hypertarget{metacore-analysis-of-commonly-mutated-pathways}{%
\section{Metacore analysis of commonly mutated pathways}\label{metacore-analysis-of-commonly-mutated-pathways}}

\hypertarget{mutations-in-rna}{%
\chapter{Mutations in RNA}\label{mutations-in-rna}}

Placeholder

\hypertarget{haplotype-caller}{%
\section{Haplotype caller}\label{haplotype-caller}}

\hypertarget{load-files}{%
\section{Load files}\label{load-files}}

\hypertarget{identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data}{%
\section{Identifying polymorphisms: compare frequencies in CD45 and Ep data}\label{identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data}}

\hypertarget{find-coding-mutations-which-are-specific-to-ep-cells}{%
\section{Find coding mutations which are specific to ep cells}\label{find-coding-mutations-which-are-specific-to-ep-cells}}

\hypertarget{filtering-wgs-data-based-on-cd45-data}{%
\section{Filtering WGS data based on CD45 data}\label{filtering-wgs-data-based-on-cd45-data}}

\hypertarget{mutations-in-progression-cohort}{%
\chapter{Mutations in progression cohort}\label{mutations-in-progression-cohort}}

Placeholder

\hypertarget{summary-of-common-mutations}{%
\section{Summary of common mutations:}\label{summary-of-common-mutations}}

\hypertarget{look-at-the-common-mutations-cosmic-ones}{%
\section{Look at the common mutations (cosmic ones)}\label{look-at-the-common-mutations-cosmic-ones}}

\hypertarget{also-look-at-the-top-rna-mutations-with-are-epdn-specific}{%
\section{Also look at the top RNA-mutations with are Ep/DN specific}\label{also-look-at-the-top-rna-mutations-with-are-epdn-specific}}

\hypertarget{mutation-load}{%
\section{Mutation load?}\label{mutation-load}}

\hypertarget{trichrome-staining}{%
\chapter{Trichrome staining}\label{trichrome-staining}}

Placeholder

\hypertarget{associations-with-cellular-fraction-wsi}{%
\section{Associations with cellular fraction (wsi)}\label{associations-with-cellular-fraction-wsi}}

\hypertarget{associations-with-growth-and-spatial-patterns}{%
\section{Associations with growth and spatial patterns}\label{associations-with-growth-and-spatial-patterns}}

\hypertarget{appendix-appendix}{%
\appendix}


Here we have a links to the list of the figures in this study

\hypertarget{main-figures}{%
\chapter{Main Figures}\label{main-figures}}

\hypertarget{figure-1}{%
\section*{Figure 1}\label{figure-1}}
\addcontentsline{toc}{section}{Figure 1}

\hypertarget{figure-2-immune-system-of-nmu-rat}{%
\section*{Figure 2: Immune system of NMU-rat}\label{figure-2-immune-system-of-nmu-rat}}
\addcontentsline{toc}{section}{Figure 2: Immune system of NMU-rat}

Fig\ref{char-gsea-cd45} : GSEA of \label{fig:char-gsea-cd45}

\hypertarget{figure-3-immunotherapy-applied-to-nmu-rats}{%
\section*{Figure 3: Immunotherapy applied to NMU rats}\label{figure-3-immunotherapy-applied-to-nmu-rats}}
\addcontentsline{toc}{section}{Figure 3: Immunotherapy applied to NMU rats}

This figure concerns analyses run in @ref(\#gPreface)

Fig\ref{3b} : Growth rates of \protect\hyperlink{fig:3b}{tumors}
Fig\ref{3c} : Growth rates of tumors w.r.t. treatment
Fig\ref{3d} : Histogram of growth rates of tumors
Fig\ref{3e} : Contingency table growth rate and treatment

\hypertarget{extended-figures}{%
\chapter{Extended Figures}\label{extended-figures}}

Ext Fig\ref{fig:5xta}

  \bibliography{book.bib,packages.bib}

\end{document}
