<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 16 Read in ER i/o data | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 16 Read in ER i/o data | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 16 Read in ER i/o data | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2022-02-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tcga-associate-epcam-inflammatory-with-survival.html"/>
<link rel="next" href="deseq-analysis-characterisation-cohort-big-vs-small.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.3</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html"><i class="fa fa-check"></i><b>16</b> Read in ER i/o data</a><ul>
<li class="chapter" data-level="16.1" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.1</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.2" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#do-associative-study-with-other-features"><i class="fa fa-check"></i><b>16.2</b> Do associative study with other features?</a></li>
<li class="chapter" data-level="16.3" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#use-raw-counts-and-perform-a-batch-correction-if-necessary"><i class="fa fa-check"></i><b>16.3</b> Use raw counts and perform a batch correction if necessary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="tcr-clonotype-analysis.html"><a href="tcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>21</b> TCR clonotype analysis</a><ul>
<li class="chapter" data-level="21.0.1" data-path="tcr-clonotype-analysis.html"><a href="tcr-clonotype-analysis.html#progression-cohort-2"><i class="fa fa-check"></i><b>21.0.1</b> Progression cohort</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>22</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="22.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>22.1</b> Data</a><ul>
<li class="chapter" data-level="22.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>22.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>22.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="22.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>22.3</b> Plots</a><ul>
<li class="chapter" data-level="22.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>22.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="22.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>22.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>22.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="22.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>22.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="22.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>22.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="22.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>22.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="22.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>22.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>23</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>23.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>23.2</b> Load files</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>23.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="23.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>23.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="23.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>23.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>24</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="24.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#mutational-burden"><i class="fa fa-check"></i><b>24.1</b> Mutational burden</a></li>
<li class="chapter" data-level="24.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>24.2</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="24.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>24.3</b> Look at the common mutations (cosmic ones)</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>25</b> Trichrome staining</a><ul>
<li class="chapter" data-level="25.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>25.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="25.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>25.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="25.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>25.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="25.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>25.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>26</b> CNV calling</a><ul>
<li class="chapter" data-level="26.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>26.1</b> Sample data</a></li>
<li class="chapter" data-level="26.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>26.2</b> Data Summary</a><ul>
<li class="chapter" data-level="26.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>26.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="26.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>26.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="26.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>26.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="26.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>26.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="26.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>26.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>27</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="read-in-er-io-data" class="section level1">
<h1><span class="header-section-number">Chapter 16</span> Read in ER i/o data</h1>
<p>Read in data from cohort of samples which have ER</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="read-in-er-io-data.html#cb345-1"></a>ERTPM=<span class="kw">read.delim</span>(<span class="st">&quot;../data/Tolaney/rnaseq_tpm_counts.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">skip=</span><span class="dv">1</span>)</span>
<span id="cb345-2"><a href="read-in-er-io-data.html#cb345-2"></a><span class="kw">rownames</span>(ERTPM)=ERTPM[ ,<span class="dv">1</span>]</span>
<span id="cb345-3"><a href="read-in-er-io-data.html#cb345-3"></a></span>
<span id="cb345-4"><a href="read-in-er-io-data.html#cb345-4"></a>ERcounts=<span class="kw">read.delim</span>(<span class="st">&quot;../data/Tolaney/raw_rnaseq_counts.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb345-5"><a href="read-in-er-io-data.html#cb345-5"></a>ERcounts=ERcounts[<span class="op">-</span><span class="dv">25800</span>, ]</span>
<span id="cb345-6"><a href="read-in-er-io-data.html#cb345-6"></a><span class="kw">rownames</span>(ERcounts)=ERcounts[ ,<span class="dv">1</span>]</span>
<span id="cb345-7"><a href="read-in-er-io-data.html#cb345-7"></a></span>
<span id="cb345-8"><a href="read-in-er-io-data.html#cb345-8"></a>ERClin=<span class="kw">read.delim</span>(<span class="st">&quot;../data/Tolaney/clin_data.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">skip=</span><span class="dv">1</span>)</span>
<span id="cb345-9"><a href="read-in-er-io-data.html#cb345-9"></a></span>
<span id="cb345-10"><a href="read-in-er-io-data.html#cb345-10"></a>ERcibersort=<span class="kw">read.csv</span>(<span class="st">&quot;../data/Tolaney/CIBERSORT.csv&quot;</span>, <span class="dt">skip=</span><span class="dv">1</span>)</span>
<span id="cb345-11"><a href="read-in-er-io-data.html#cb345-11"></a>ERcytokines=<span class="kw">read.csv</span>(<span class="st">&quot;../data/Tolaney/cytokines.csv&quot;</span>, <span class="dt">skip=</span><span class="dv">1</span>)</span>
<span id="cb345-12"><a href="read-in-er-io-data.html#cb345-12"></a></span>
<span id="cb345-13"><a href="read-in-er-io-data.html#cb345-13"></a><span class="co">## Append data where necessary:</span></span>
<span id="cb345-14"><a href="read-in-er-io-data.html#cb345-14"></a>ERClin<span class="op">$</span>CD8_Tcell=ERcibersort<span class="op">$</span>T.cells.CD8[<span class="kw">match</span>(<span class="kw">paste</span>(<span class="st">&quot;Patient&quot;</span>, ERClin<span class="op">$</span>Subject.ID, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), ERcibersort<span class="op">$</span>Patient)]</span>
<span id="cb345-15"><a href="read-in-er-io-data.html#cb345-15"></a>ERClin<span class="op">$</span>DCs=ERcibersort<span class="op">$</span>Dendritic.cells.activated[<span class="kw">match</span>(<span class="kw">paste</span>(<span class="st">&quot;Patient&quot;</span>, ERClin<span class="op">$</span>Subject.ID, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), ERcibersort<span class="op">$</span>Patient)]</span>
<span id="cb345-16"><a href="read-in-er-io-data.html#cb345-16"></a>ERClin<span class="op">$</span>MICB=ERcytokines<span class="op">$</span>MICB[<span class="kw">match</span>(ERClin<span class="op">$</span>Subject.ID, ERcytokines<span class="op">$</span>Subject.ID)]</span>
<span id="cb345-17"><a href="read-in-er-io-data.html#cb345-17"></a>ERClin<span class="op">$</span>MICA=ERcytokines<span class="op">$</span>MICA[<span class="kw">match</span>(ERClin<span class="op">$</span>Subject.ID, ERcytokines<span class="op">$</span>Subject.ID)]</span></code></pre></div>
<p>There are 30 patients in this cohort</p>
<div id="signature-lum-cases-non-inflammatory-growing-vs-stable-1" class="section level2">
<h2><span class="header-section-number">16.1</span> Signature: Lum cases non-inflammatory: growing vs stable</h2>
<p>Here, use the signature which was applied before to see the difference between the non-inflammatory growing vs stable samples. We compare this to the oncotypedx list of genes and mammaprint</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="read-in-er-io-data.html#cb346-1"></a><span class="co">#GlistGrowing=read.csv(&quot;figure-outputs/lumsig_posFC.csv&quot;)</span></span>
<span id="cb346-2"><a href="read-in-er-io-data.html#cb346-2"></a><span class="co">#GlistGrowing=GlistGrowing$x</span></span>
<span id="cb346-3"><a href="read-in-er-io-data.html#cb346-3"></a><span class="co">#Glistnon=read.csv(&quot;figure-outputs/lumsig_negFC.csv&quot;)</span></span>
<span id="cb346-4"><a href="read-in-er-io-data.html#cb346-4"></a><span class="co">#Glistnon=Glistnon$x</span></span></code></pre></div>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="read-in-er-io-data.html#cb347-1"></a></span>
<span id="cb347-2"><a href="read-in-er-io-data.html#cb347-2"></a>HumGeneList=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(GlistGrowing, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb347-3"><a href="read-in-er-io-data.html#cb347-3"></a>HumGeneList2=Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(GlistGrowing, Rat2Hum<span class="op">$</span>RGD.symbol)]</span>
<span id="cb347-4"><a href="read-in-er-io-data.html#cb347-4"></a></span>
<span id="cb347-5"><a href="read-in-er-io-data.html#cb347-5"></a>TCGAssgsea=<span class="kw">gsva</span>(<span class="kw">data.matrix</span>(ERTPM[ ,<span class="op">-</span><span class="dv">1</span>]), <span class="kw">list</span>(<span class="dt">grow=</span><span class="kw">na.omit</span>(HumGeneList2),  <span class="dt">stab=</span>Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(Glistnon, Rat2Hum<span class="op">$</span>RGD.symbol)],<span class="dt">Onco=</span>Oncoprint[[<span class="dv">1</span>]], <span class="dt">Mamma=</span>Oncoprint[[<span class="dv">2</span>]]), <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>F)</span>
<span id="cb347-6"><a href="read-in-er-io-data.html#cb347-6"></a></span>
<span id="cb347-7"><a href="read-in-er-io-data.html#cb347-7"></a>TCGAssgsea=<span class="kw">t</span>(<span class="kw">scale</span>(<span class="kw">t</span>(TCGAssgsea)))</span>
<span id="cb347-8"><a href="read-in-er-io-data.html#cb347-8"></a></span>
<span id="cb347-9"><a href="read-in-er-io-data.html#cb347-9"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb347-10"><a href="read-in-er-io-data.html#cb347-10"></a><span class="kw">hist</span>(TCGAssgsea[<span class="dv">1</span>, ], <span class="dt">main=</span><span class="st">&quot;growing&quot;</span>)</span>
<span id="cb347-11"><a href="read-in-er-io-data.html#cb347-11"></a><span class="kw">hist</span>(TCGAssgsea[<span class="dv">2</span>, ], <span class="dt">main=</span><span class="st">&quot;stable&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-105-1.png" width="672" /></p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="read-in-er-io-data.html#cb348-1"></a><span class="kw">plot</span>(TCGAssgsea[<span class="dv">1</span>, ], TCGAssgsea[<span class="dv">2</span>, ], <span class="dt">ylab=</span><span class="st">&quot;stable&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;growing&quot;</span>)</span>
<span id="cb348-2"><a href="read-in-er-io-data.html#cb348-2"></a></span>
<span id="cb348-3"><a href="read-in-er-io-data.html#cb348-3"></a><span class="co">## also compare with onco or mamma print</span></span>
<span id="cb348-4"><a href="read-in-er-io-data.html#cb348-4"></a></span>
<span id="cb348-5"><a href="read-in-er-io-data.html#cb348-5"></a><span class="kw">plot</span>(TCGAssgsea[<span class="dv">1</span>, ], TCGAssgsea[<span class="dv">3</span>, ], <span class="dt">xlab=</span><span class="st">&quot;growing&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;oncotype&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-105-2.png" width="672" /></p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="read-in-er-io-data.html#cb349-1"></a><span class="kw">plot</span>(TCGAssgsea[<span class="dv">1</span>, ], TCGAssgsea[<span class="dv">4</span>, ], <span class="dt">xlab=</span><span class="st">&quot;growing&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;mammaprint&quot;</span>)</span>
<span id="cb349-2"><a href="read-in-er-io-data.html#cb349-2"></a></span>
<span id="cb349-3"><a href="read-in-er-io-data.html#cb349-3"></a><span class="co">## double check the </span></span>
<span id="cb349-4"><a href="read-in-er-io-data.html#cb349-4"></a></span>
<span id="cb349-5"><a href="read-in-er-io-data.html#cb349-5"></a><span class="kw">plot</span>(TCGAssgsea[<span class="dv">1</span>, ], ERTPM[<span class="st">&quot;ADAMTS10&quot;</span> ,<span class="op">-</span><span class="dv">1</span>], <span class="dt">xlab=</span><span class="st">&quot;growing sig&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;ADAMST10&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-105-3.png" width="672" /></p>
<p>The Growing signature genes (positive coefficient) are Setd7, Hspa1l, Hbb, Kifc2, Kmt5c, Pik3c2b, Lmntd2, Nsmf, Nrbp2, LOC100134871, Hsp90aa1, Coro6, Chst8, Slc4a3, Fhod1, Rpl8, Spp1, Catsperg, Col5a1, LOC100911498, Noxa1, F8, Rps19, Rps6, Irs1, Leng8, Plekhh1, Lmbr1l, Ikbke, Pnisr, Adamts10, Zfp692, Trim41, Sema6d, Mgp, Cdc42bpg, Slc16a13, Dmpk and stable (negative coefficient) are Fgfr2, Mx1, Endou, Grhl3, Slpi, Gbp2, Plg, Aldh1a3, Upk3a.</p>
<p>Below are the survival curves for the growing signature:</p>
<p>We can compare high and low, as well as with treatment</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="read-in-er-io-data.html#cb350-1"></a>ax1=<span class="kw">match</span>(<span class="kw">colnames</span>(TCGAssgsea), <span class="kw">paste</span>(<span class="st">&quot;Patient&quot;</span>, ERClin<span class="op">$</span>Subject.ID, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb350-2"><a href="read-in-er-io-data.html#cb350-2"></a>ax3=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(ax1))</span>
<span id="cb350-3"><a href="read-in-er-io-data.html#cb350-3"></a>ssPS=<span class="kw">Surv</span>( ERClin<span class="op">$</span>PFS.Months[<span class="kw">na.omit</span>(ax1)], ERClin<span class="op">$</span>Progression[<span class="kw">na.omit</span>(ax1)])</span>
<span id="cb350-4"><a href="read-in-er-io-data.html#cb350-4"></a>ssOS=<span class="kw">Surv</span>(ERClin<span class="op">$</span>OS.Months[<span class="kw">na.omit</span>(ax1)], ERClin<span class="op">$</span>Death[<span class="kw">na.omit</span>(ax1)] )</span>
<span id="cb350-5"><a href="read-in-er-io-data.html#cb350-5"></a></span>
<span id="cb350-6"><a href="read-in-er-io-data.html#cb350-6"></a></span>
<span id="cb350-7"><a href="read-in-er-io-data.html#cb350-7"></a>newtab=<span class="kw">data.frame</span>(<span class="dt">PFSMo=</span>ERClin<span class="op">$</span>PFS.Months[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-8"><a href="read-in-er-io-data.html#cb350-8"></a>                  <span class="dt">PFS=</span>ERClin<span class="op">$</span>Progression[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-9"><a href="read-in-er-io-data.html#cb350-9"></a>                  <span class="dt">OSMo=</span>ERClin<span class="op">$</span>OS.Months[<span class="kw">na.omit</span>(ax1)], </span>
<span id="cb350-10"><a href="read-in-er-io-data.html#cb350-10"></a>                  <span class="dt">OS=</span>ERClin<span class="op">$</span>Death[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-11"><a href="read-in-er-io-data.html#cb350-11"></a>                  <span class="dt">Response=</span>ERClin<span class="op">$</span>Response[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-12"><a href="read-in-er-io-data.html#cb350-12"></a>                  <span class="dt">grow=</span>TCGAssgsea[<span class="dv">1</span>,ax3 ],</span>
<span id="cb350-13"><a href="read-in-er-io-data.html#cb350-13"></a>                  <span class="dt">stab=</span>TCGAssgsea[<span class="dv">2</span>,ax3 ],</span>
<span id="cb350-14"><a href="read-in-er-io-data.html#cb350-14"></a>                  <span class="dt">Arm=</span>ERClin<span class="op">$</span>Arm[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-15"><a href="read-in-er-io-data.html#cb350-15"></a>                  <span class="dt">PDL1=</span>ERClin<span class="op">$</span>PD.L1[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-16"><a href="read-in-er-io-data.html#cb350-16"></a>                  <span class="dt">ClinBen=</span>ERClin<span class="op">$</span>Clinical.Benefit[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-17"><a href="read-in-er-io-data.html#cb350-17"></a>                  <span class="dt">TMB=</span>ERClin<span class="op">$</span>TMB[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-18"><a href="read-in-er-io-data.html#cb350-18"></a>                  <span class="dt">Site=</span>ERClin<span class="op">$</span>Biopsy[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-19"><a href="read-in-er-io-data.html#cb350-19"></a>                  <span class="dt">CD8=</span>ERClin<span class="op">$</span>CD8_Tcell[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-20"><a href="read-in-er-io-data.html#cb350-20"></a>                  <span class="dt">DC=</span>ERClin<span class="op">$</span>DCs[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-21"><a href="read-in-er-io-data.html#cb350-21"></a>                  <span class="dt">MICB=</span>ERClin<span class="op">$</span>MICB[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-22"><a href="read-in-er-io-data.html#cb350-22"></a>                  <span class="dt">MICA=</span>ERClin<span class="op">$</span>MICA[<span class="kw">na.omit</span>(ax1)],</span>
<span id="cb350-23"><a href="read-in-er-io-data.html#cb350-23"></a>                  <span class="dt">Purity=</span>ERClin<span class="op">$</span>Purity[<span class="kw">na.omit</span>(ax1)]</span>
<span id="cb350-24"><a href="read-in-er-io-data.html#cb350-24"></a>                  )</span>
<span id="cb350-25"><a href="read-in-er-io-data.html#cb350-25"></a></span>
<span id="cb350-26"><a href="read-in-er-io-data.html#cb350-26"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>stab, <span class="kw">quantile</span>(newtab<span class="op">$</span>stab, <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb350-27"><a href="read-in-er-io-data.html#cb350-27"></a></span>
<span id="cb350-28"><a href="read-in-er-io-data.html#cb350-28"></a></span>
<span id="cb350-29"><a href="read-in-er-io-data.html#cb350-29"></a>dfs1=<span class="kw">coxph</span>(ssPS<span class="op">~</span>TCGAvalCut<span class="op">+</span>Arm, <span class="dt">data=</span>newtab)</span>
<span id="cb350-30"><a href="read-in-er-io-data.html#cb350-30"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb350-31"><a href="read-in-er-io-data.html#cb350-31"></a><span class="co">## Call:</span></span>
<span id="cb350-32"><a href="read-in-er-io-data.html#cb350-32"></a><span class="co">## coxph(formula = ssPS ~ TCGAvalCut + Arm, data = newtab)</span></span>
<span id="cb350-33"><a href="read-in-er-io-data.html#cb350-33"></a><span class="co">## </span></span>
<span id="cb350-34"><a href="read-in-er-io-data.html#cb350-34"></a><span class="co">##   n= 29, number of events= 27 </span></span>
<span id="cb350-35"><a href="read-in-er-io-data.html#cb350-35"></a><span class="co">##    (1 observation deleted due to missingness)</span></span>
<span id="cb350-36"><a href="read-in-er-io-data.html#cb350-36"></a><span class="co">## </span></span>
<span id="cb350-37"><a href="read-in-er-io-data.html#cb350-37"></a><span class="co">##                coef exp(coef) se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb350-38"><a href="read-in-er-io-data.html#cb350-38"></a><span class="co">## TCGAvalCutP -0.6499    0.5221   0.4256 -1.527    0.127</span></span>
<span id="cb350-39"><a href="read-in-er-io-data.html#cb350-39"></a><span class="co">## ArmB         0.1130    1.1196   0.4254  0.266    0.791</span></span>
<span id="cb350-40"><a href="read-in-er-io-data.html#cb350-40"></a><span class="co">## </span></span>
<span id="cb350-41"><a href="read-in-er-io-data.html#cb350-41"></a><span class="co">##             exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb350-42"><a href="read-in-er-io-data.html#cb350-42"></a><span class="co">## TCGAvalCutP    0.5221     1.9153    0.2267     1.202</span></span>
<span id="cb350-43"><a href="read-in-er-io-data.html#cb350-43"></a><span class="co">## ArmB           1.1196     0.8932    0.4864     2.577</span></span>
<span id="cb350-44"><a href="read-in-er-io-data.html#cb350-44"></a><span class="co">## </span></span>
<span id="cb350-45"><a href="read-in-er-io-data.html#cb350-45"></a><span class="co">## Concordance= 0.586  (se = 0.068 )</span></span>
<span id="cb350-46"><a href="read-in-er-io-data.html#cb350-46"></a><span class="co">## Likelihood ratio test= 2.8  on 2 df,   p=0.2</span></span>
<span id="cb350-47"><a href="read-in-er-io-data.html#cb350-47"></a><span class="co">## Wald test            = 2.82  on 2 df,   p=0.2</span></span>
<span id="cb350-48"><a href="read-in-er-io-data.html#cb350-48"></a><span class="co">## Score (logrank) test = 2.92  on 2 df,   p=0.2</span></span>
<span id="cb350-49"><a href="read-in-er-io-data.html#cb350-49"></a>adfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb350-50"><a href="read-in-er-io-data.html#cb350-50"></a>os1=<span class="kw">coxph</span>(ssOS<span class="op">~</span>TCGAvalCut<span class="op">+</span>Arm, <span class="dt">data=</span>newtab)</span>
<span id="cb350-51"><a href="read-in-er-io-data.html#cb350-51"></a><span class="kw">summary</span>(os1)</span>
<span id="cb350-52"><a href="read-in-er-io-data.html#cb350-52"></a><span class="co">## Call:</span></span>
<span id="cb350-53"><a href="read-in-er-io-data.html#cb350-53"></a><span class="co">## coxph(formula = ssOS ~ TCGAvalCut + Arm, data = newtab)</span></span>
<span id="cb350-54"><a href="read-in-er-io-data.html#cb350-54"></a><span class="co">## </span></span>
<span id="cb350-55"><a href="read-in-er-io-data.html#cb350-55"></a><span class="co">##   n= 29, number of events= 22 </span></span>
<span id="cb350-56"><a href="read-in-er-io-data.html#cb350-56"></a><span class="co">##    (1 observation deleted due to missingness)</span></span>
<span id="cb350-57"><a href="read-in-er-io-data.html#cb350-57"></a><span class="co">## </span></span>
<span id="cb350-58"><a href="read-in-er-io-data.html#cb350-58"></a><span class="co">##                coef exp(coef) se(coef)      z Pr(&gt;|z|)  </span></span>
<span id="cb350-59"><a href="read-in-er-io-data.html#cb350-59"></a><span class="co">## TCGAvalCutP -0.9841    0.3738   0.4424 -2.224   0.0261 *</span></span>
<span id="cb350-60"><a href="read-in-er-io-data.html#cb350-60"></a><span class="co">## ArmB         0.0176    1.0178   0.4311  0.041   0.9674  </span></span>
<span id="cb350-61"><a href="read-in-er-io-data.html#cb350-61"></a><span class="co">## ---</span></span>
<span id="cb350-62"><a href="read-in-er-io-data.html#cb350-62"></a><span class="co">## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb350-63"><a href="read-in-er-io-data.html#cb350-63"></a><span class="co">## </span></span>
<span id="cb350-64"><a href="read-in-er-io-data.html#cb350-64"></a><span class="co">##             exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb350-65"><a href="read-in-er-io-data.html#cb350-65"></a><span class="co">## TCGAvalCutP    0.3738     2.6754    0.1571    0.8896</span></span>
<span id="cb350-66"><a href="read-in-er-io-data.html#cb350-66"></a><span class="co">## ArmB           1.0178     0.9826    0.4372    2.3690</span></span>
<span id="cb350-67"><a href="read-in-er-io-data.html#cb350-67"></a><span class="co">## </span></span>
<span id="cb350-68"><a href="read-in-er-io-data.html#cb350-68"></a><span class="co">## Concordance= 0.621  (se = 0.066 )</span></span>
<span id="cb350-69"><a href="read-in-er-io-data.html#cb350-69"></a><span class="co">## Likelihood ratio test= 5.06  on 2 df,   p=0.08</span></span>
<span id="cb350-70"><a href="read-in-er-io-data.html#cb350-70"></a><span class="co">## Wald test            = 4.95  on 2 df,   p=0.08</span></span>
<span id="cb350-71"><a href="read-in-er-io-data.html#cb350-71"></a><span class="co">## Score (logrank) test = 5.32  on 2 df,   p=0.07</span></span>
<span id="cb350-72"><a href="read-in-er-io-data.html#cb350-72"></a>aos=<span class="kw">summary</span>(os1)</span>
<span id="cb350-73"><a href="read-in-er-io-data.html#cb350-73"></a></span>
<span id="cb350-74"><a href="read-in-er-io-data.html#cb350-74"></a>ns=<span class="kw">table</span>(TCGAvalCut)</span>
<span id="cb350-75"><a href="read-in-er-io-data.html#cb350-75"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb350-76"><a href="read-in-er-io-data.html#cb350-76"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(adfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], adfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb350-77"><a href="read-in-er-io-data.html#cb350-77"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;stab sig: all&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PFS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb350-78"><a href="read-in-er-io-data.html#cb350-78"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))              </span>
<span id="cb350-79"><a href="read-in-er-io-data.html#cb350-79"></a><span class="kw">text</span>(<span class="dv">12</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb350-80"><a href="read-in-er-io-data.html#cb350-80"></a>      </span>
<span id="cb350-81"><a href="read-in-er-io-data.html#cb350-81"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(aos<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], aos<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb350-82"><a href="read-in-er-io-data.html#cb350-82"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;stab sig: all&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb350-83"><a href="read-in-er-io-data.html#cb350-83"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb350-84"><a href="read-in-er-io-data.html#cb350-84"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-106-1.png" width="672" /></p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="read-in-er-io-data.html#cb351-1"></a></span>
<span id="cb351-2"><a href="read-in-er-io-data.html#cb351-2"></a><span class="co">## survival outcome: cts variable</span></span>
<span id="cb351-3"><a href="read-in-er-io-data.html#cb351-3"></a>dfs1=<span class="kw">coxph</span>(ssPS<span class="op">~</span>stab<span class="op">+</span>Arm, <span class="dt">data=</span>newtab)</span>
<span id="cb351-4"><a href="read-in-er-io-data.html#cb351-4"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb351-5"><a href="read-in-er-io-data.html#cb351-5"></a><span class="co">## Call:</span></span>
<span id="cb351-6"><a href="read-in-er-io-data.html#cb351-6"></a><span class="co">## coxph(formula = ssPS ~ stab + Arm, data = newtab)</span></span>
<span id="cb351-7"><a href="read-in-er-io-data.html#cb351-7"></a><span class="co">## </span></span>
<span id="cb351-8"><a href="read-in-er-io-data.html#cb351-8"></a><span class="co">##   n= 30, number of events= 28 </span></span>
<span id="cb351-9"><a href="read-in-er-io-data.html#cb351-9"></a><span class="co">## </span></span>
<span id="cb351-10"><a href="read-in-er-io-data.html#cb351-10"></a><span class="co">##         coef exp(coef) se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb351-11"><a href="read-in-er-io-data.html#cb351-11"></a><span class="co">## stab -0.2647    0.7675   0.1864 -1.420    0.156</span></span>
<span id="cb351-12"><a href="read-in-er-io-data.html#cb351-12"></a><span class="co">## ArmB  0.4682    1.5971   0.4140  1.131    0.258</span></span>
<span id="cb351-13"><a href="read-in-er-io-data.html#cb351-13"></a><span class="co">## </span></span>
<span id="cb351-14"><a href="read-in-er-io-data.html#cb351-14"></a><span class="co">##      exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb351-15"><a href="read-in-er-io-data.html#cb351-15"></a><span class="co">## stab    0.7675     1.3030    0.5326     1.106</span></span>
<span id="cb351-16"><a href="read-in-er-io-data.html#cb351-16"></a><span class="co">## ArmB    1.5971     0.6261    0.7095     3.595</span></span>
<span id="cb351-17"><a href="read-in-er-io-data.html#cb351-17"></a><span class="co">## </span></span>
<span id="cb351-18"><a href="read-in-er-io-data.html#cb351-18"></a><span class="co">## Concordance= 0.592  (se = 0.071 )</span></span>
<span id="cb351-19"><a href="read-in-er-io-data.html#cb351-19"></a><span class="co">## Likelihood ratio test= 2.77  on 2 df,   p=0.3</span></span>
<span id="cb351-20"><a href="read-in-er-io-data.html#cb351-20"></a><span class="co">## Wald test            = 2.71  on 2 df,   p=0.3</span></span>
<span id="cb351-21"><a href="read-in-er-io-data.html#cb351-21"></a><span class="co">## Score (logrank) test = 2.78  on 2 df,   p=0.2</span></span>
<span id="cb351-22"><a href="read-in-er-io-data.html#cb351-22"></a>os1=<span class="kw">coxph</span>(ssOS<span class="op">~</span>stab<span class="op">+</span>Arm, <span class="dt">data=</span>newtab)</span>
<span id="cb351-23"><a href="read-in-er-io-data.html#cb351-23"></a><span class="kw">summary</span>(os1)</span>
<span id="cb351-24"><a href="read-in-er-io-data.html#cb351-24"></a><span class="co">## Call:</span></span>
<span id="cb351-25"><a href="read-in-er-io-data.html#cb351-25"></a><span class="co">## coxph(formula = ssOS ~ stab + Arm, data = newtab)</span></span>
<span id="cb351-26"><a href="read-in-er-io-data.html#cb351-26"></a><span class="co">## </span></span>
<span id="cb351-27"><a href="read-in-er-io-data.html#cb351-27"></a><span class="co">##   n= 30, number of events= 22 </span></span>
<span id="cb351-28"><a href="read-in-er-io-data.html#cb351-28"></a><span class="co">## </span></span>
<span id="cb351-29"><a href="read-in-er-io-data.html#cb351-29"></a><span class="co">##         coef exp(coef) se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb351-30"><a href="read-in-er-io-data.html#cb351-30"></a><span class="co">## stab -0.1572    0.8546   0.2208 -0.712    0.477</span></span>
<span id="cb351-31"><a href="read-in-er-io-data.html#cb351-31"></a><span class="co">## ArmB  0.2415    1.2731   0.4558  0.530    0.596</span></span>
<span id="cb351-32"><a href="read-in-er-io-data.html#cb351-32"></a><span class="co">## </span></span>
<span id="cb351-33"><a href="read-in-er-io-data.html#cb351-33"></a><span class="co">##      exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb351-34"><a href="read-in-er-io-data.html#cb351-34"></a><span class="co">## stab    0.8546     1.1702    0.5544     1.317</span></span>
<span id="cb351-35"><a href="read-in-er-io-data.html#cb351-35"></a><span class="co">## ArmB    1.2731     0.7855    0.5211     3.111</span></span>
<span id="cb351-36"><a href="read-in-er-io-data.html#cb351-36"></a><span class="co">## </span></span>
<span id="cb351-37"><a href="read-in-er-io-data.html#cb351-37"></a><span class="co">## Concordance= 0.557  (se = 0.075 )</span></span>
<span id="cb351-38"><a href="read-in-er-io-data.html#cb351-38"></a><span class="co">## Likelihood ratio test= 0.61  on 2 df,   p=0.7</span></span>
<span id="cb351-39"><a href="read-in-er-io-data.html#cb351-39"></a><span class="co">## Wald test            = 0.6  on 2 df,   p=0.7</span></span>
<span id="cb351-40"><a href="read-in-er-io-data.html#cb351-40"></a><span class="co">## Score (logrank) test = 0.61  on 2 df,   p=0.7</span></span></code></pre></div>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="read-in-er-io-data.html#cb352-1"></a></span>
<span id="cb352-2"><a href="read-in-er-io-data.html#cb352-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb352-3"><a href="read-in-er-io-data.html#cb352-3"></a></span>
<span id="cb352-4"><a href="read-in-er-io-data.html#cb352-4"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>grow, <span class="kw">quantile</span>(newtab<span class="op">$</span>grow, <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;P&quot;</span>))</span>
<span id="cb352-5"><a href="read-in-er-io-data.html#cb352-5"></a></span>
<span id="cb352-6"><a href="read-in-er-io-data.html#cb352-6"></a>dfs1=<span class="kw">coxph</span>(ssPS<span class="op">~</span>TCGAvalCut<span class="op">+</span>Arm, <span class="dt">data=</span>newtab)</span>
<span id="cb352-7"><a href="read-in-er-io-data.html#cb352-7"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb352-8"><a href="read-in-er-io-data.html#cb352-8"></a><span class="co">## Call:</span></span>
<span id="cb352-9"><a href="read-in-er-io-data.html#cb352-9"></a><span class="co">## coxph(formula = ssPS ~ TCGAvalCut + Arm, data = newtab)</span></span>
<span id="cb352-10"><a href="read-in-er-io-data.html#cb352-10"></a><span class="co">## </span></span>
<span id="cb352-11"><a href="read-in-er-io-data.html#cb352-11"></a><span class="co">##   n= 29, number of events= 27 </span></span>
<span id="cb352-12"><a href="read-in-er-io-data.html#cb352-12"></a><span class="co">##    (1 observation deleted due to missingness)</span></span>
<span id="cb352-13"><a href="read-in-er-io-data.html#cb352-13"></a><span class="co">## </span></span>
<span id="cb352-14"><a href="read-in-er-io-data.html#cb352-14"></a><span class="co">##               coef exp(coef) se(coef)     z Pr(&gt;|z|)</span></span>
<span id="cb352-15"><a href="read-in-er-io-data.html#cb352-15"></a><span class="co">## TCGAvalCutP 0.4793    1.6150   0.4293 1.117    0.264</span></span>
<span id="cb352-16"><a href="read-in-er-io-data.html#cb352-16"></a><span class="co">## ArmB        0.5381    1.7127   0.4380 1.228    0.219</span></span>
<span id="cb352-17"><a href="read-in-er-io-data.html#cb352-17"></a><span class="co">## </span></span>
<span id="cb352-18"><a href="read-in-er-io-data.html#cb352-18"></a><span class="co">##             exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb352-19"><a href="read-in-er-io-data.html#cb352-19"></a><span class="co">## TCGAvalCutP     1.615     0.6192    0.6963     3.746</span></span>
<span id="cb352-20"><a href="read-in-er-io-data.html#cb352-20"></a><span class="co">## ArmB            1.713     0.5839    0.7258     4.041</span></span>
<span id="cb352-21"><a href="read-in-er-io-data.html#cb352-21"></a><span class="co">## </span></span>
<span id="cb352-22"><a href="read-in-er-io-data.html#cb352-22"></a><span class="co">## Concordance= 0.544  (se = 0.079 )</span></span>
<span id="cb352-23"><a href="read-in-er-io-data.html#cb352-23"></a><span class="co">## Likelihood ratio test= 2.07  on 2 df,   p=0.4</span></span>
<span id="cb352-24"><a href="read-in-er-io-data.html#cb352-24"></a><span class="co">## Wald test            = 2.06  on 2 df,   p=0.4</span></span>
<span id="cb352-25"><a href="read-in-er-io-data.html#cb352-25"></a><span class="co">## Score (logrank) test = 2.06  on 2 df,   p=0.4</span></span>
<span id="cb352-26"><a href="read-in-er-io-data.html#cb352-26"></a>adfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb352-27"><a href="read-in-er-io-data.html#cb352-27"></a>os1=<span class="kw">coxph</span>(ssOS<span class="op">~</span>TCGAvalCut<span class="op">+</span>Arm, <span class="dt">data=</span>newtab)</span>
<span id="cb352-28"><a href="read-in-er-io-data.html#cb352-28"></a><span class="kw">summary</span>(os1)</span>
<span id="cb352-29"><a href="read-in-er-io-data.html#cb352-29"></a><span class="co">## Call:</span></span>
<span id="cb352-30"><a href="read-in-er-io-data.html#cb352-30"></a><span class="co">## coxph(formula = ssOS ~ TCGAvalCut + Arm, data = newtab)</span></span>
<span id="cb352-31"><a href="read-in-er-io-data.html#cb352-31"></a><span class="co">## </span></span>
<span id="cb352-32"><a href="read-in-er-io-data.html#cb352-32"></a><span class="co">##   n= 29, number of events= 21 </span></span>
<span id="cb352-33"><a href="read-in-er-io-data.html#cb352-33"></a><span class="co">##    (1 observation deleted due to missingness)</span></span>
<span id="cb352-34"><a href="read-in-er-io-data.html#cb352-34"></a><span class="co">## </span></span>
<span id="cb352-35"><a href="read-in-er-io-data.html#cb352-35"></a><span class="co">##               coef exp(coef) se(coef)     z Pr(&gt;|z|)</span></span>
<span id="cb352-36"><a href="read-in-er-io-data.html#cb352-36"></a><span class="co">## TCGAvalCutP 0.2460    1.2789   0.5407 0.455    0.649</span></span>
<span id="cb352-37"><a href="read-in-er-io-data.html#cb352-37"></a><span class="co">## ArmB        0.2555    1.2911   0.5356 0.477    0.633</span></span>
<span id="cb352-38"><a href="read-in-er-io-data.html#cb352-38"></a><span class="co">## </span></span>
<span id="cb352-39"><a href="read-in-er-io-data.html#cb352-39"></a><span class="co">##             exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb352-40"><a href="read-in-er-io-data.html#cb352-40"></a><span class="co">## TCGAvalCutP     1.279     0.7819    0.4432     3.690</span></span>
<span id="cb352-41"><a href="read-in-er-io-data.html#cb352-41"></a><span class="co">## ArmB            1.291     0.7745    0.4519     3.689</span></span>
<span id="cb352-42"><a href="read-in-er-io-data.html#cb352-42"></a><span class="co">## </span></span>
<span id="cb352-43"><a href="read-in-er-io-data.html#cb352-43"></a><span class="co">## Concordance= 0.511  (se = 0.068 )</span></span>
<span id="cb352-44"><a href="read-in-er-io-data.html#cb352-44"></a><span class="co">## Likelihood ratio test= 0.27  on 2 df,   p=0.9</span></span>
<span id="cb352-45"><a href="read-in-er-io-data.html#cb352-45"></a><span class="co">## Wald test            = 0.28  on 2 df,   p=0.9</span></span>
<span id="cb352-46"><a href="read-in-er-io-data.html#cb352-46"></a><span class="co">## Score (logrank) test = 0.28  on 2 df,   p=0.9</span></span>
<span id="cb352-47"><a href="read-in-er-io-data.html#cb352-47"></a>aos=<span class="kw">summary</span>(os1)</span>
<span id="cb352-48"><a href="read-in-er-io-data.html#cb352-48"></a></span>
<span id="cb352-49"><a href="read-in-er-io-data.html#cb352-49"></a>ns=<span class="kw">table</span>(TCGAvalCut)</span>
<span id="cb352-50"><a href="read-in-er-io-data.html#cb352-50"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(adfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], adfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb352-51"><a href="read-in-er-io-data.html#cb352-51"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;growing sig&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PFS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb352-52"><a href="read-in-er-io-data.html#cb352-52"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))                    </span>
<span id="cb352-53"><a href="read-in-er-io-data.html#cb352-53"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb352-54"><a href="read-in-er-io-data.html#cb352-54"></a></span>
<span id="cb352-55"><a href="read-in-er-io-data.html#cb352-55"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(aos<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], aos<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb352-56"><a href="read-in-er-io-data.html#cb352-56"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(<span class="st">&quot;growing sig&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb352-57"><a href="read-in-er-io-data.html#cb352-57"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb352-58"><a href="read-in-er-io-data.html#cb352-58"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/os-growing-tol-1.png" width="672" /></p>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="read-in-er-io-data.html#cb353-1"></a></span>
<span id="cb353-2"><a href="read-in-er-io-data.html#cb353-2"></a>dfs1=<span class="kw">coxph</span>(ssPS<span class="op">~</span>grow<span class="op">+</span>Arm, <span class="dt">data=</span>newtab)</span>
<span id="cb353-3"><a href="read-in-er-io-data.html#cb353-3"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb353-4"><a href="read-in-er-io-data.html#cb353-4"></a><span class="co">## Call:</span></span>
<span id="cb353-5"><a href="read-in-er-io-data.html#cb353-5"></a><span class="co">## coxph(formula = ssPS ~ grow + Arm, data = newtab)</span></span>
<span id="cb353-6"><a href="read-in-er-io-data.html#cb353-6"></a><span class="co">## </span></span>
<span id="cb353-7"><a href="read-in-er-io-data.html#cb353-7"></a><span class="co">##   n= 30, number of events= 28 </span></span>
<span id="cb353-8"><a href="read-in-er-io-data.html#cb353-8"></a><span class="co">## </span></span>
<span id="cb353-9"><a href="read-in-er-io-data.html#cb353-9"></a><span class="co">##        coef exp(coef) se(coef)     z Pr(&gt;|z|)</span></span>
<span id="cb353-10"><a href="read-in-er-io-data.html#cb353-10"></a><span class="co">## grow 0.1363    1.1461   0.1824 0.747    0.455</span></span>
<span id="cb353-11"><a href="read-in-er-io-data.html#cb353-11"></a><span class="co">## ArmB 0.4490    1.5667   0.4150 1.082    0.279</span></span>
<span id="cb353-12"><a href="read-in-er-io-data.html#cb353-12"></a><span class="co">## </span></span>
<span id="cb353-13"><a href="read-in-er-io-data.html#cb353-13"></a><span class="co">##      exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb353-14"><a href="read-in-er-io-data.html#cb353-14"></a><span class="co">## grow     1.146     0.8726    0.8015     1.639</span></span>
<span id="cb353-15"><a href="read-in-er-io-data.html#cb353-15"></a><span class="co">## ArmB     1.567     0.6383    0.6946     3.534</span></span>
<span id="cb353-16"><a href="read-in-er-io-data.html#cb353-16"></a><span class="co">## </span></span>
<span id="cb353-17"><a href="read-in-er-io-data.html#cb353-17"></a><span class="co">## Concordance= 0.568  (se = 0.081 )</span></span>
<span id="cb353-18"><a href="read-in-er-io-data.html#cb353-18"></a><span class="co">## Likelihood ratio test= 1.39  on 2 df,   p=0.5</span></span>
<span id="cb353-19"><a href="read-in-er-io-data.html#cb353-19"></a><span class="co">## Wald test            = 1.44  on 2 df,   p=0.5</span></span>
<span id="cb353-20"><a href="read-in-er-io-data.html#cb353-20"></a><span class="co">## Score (logrank) test = 1.45  on 2 df,   p=0.5</span></span>
<span id="cb353-21"><a href="read-in-er-io-data.html#cb353-21"></a>os1=<span class="kw">coxph</span>(ssOS<span class="op">~</span>grow<span class="op">+</span>Arm, <span class="dt">data=</span>newtab)</span>
<span id="cb353-22"><a href="read-in-er-io-data.html#cb353-22"></a><span class="kw">summary</span>(os1)</span>
<span id="cb353-23"><a href="read-in-er-io-data.html#cb353-23"></a><span class="co">## Call:</span></span>
<span id="cb353-24"><a href="read-in-er-io-data.html#cb353-24"></a><span class="co">## coxph(formula = ssOS ~ grow + Arm, data = newtab)</span></span>
<span id="cb353-25"><a href="read-in-er-io-data.html#cb353-25"></a><span class="co">## </span></span>
<span id="cb353-26"><a href="read-in-er-io-data.html#cb353-26"></a><span class="co">##   n= 30, number of events= 22 </span></span>
<span id="cb353-27"><a href="read-in-er-io-data.html#cb353-27"></a><span class="co">## </span></span>
<span id="cb353-28"><a href="read-in-er-io-data.html#cb353-28"></a><span class="co">##        coef exp(coef) se(coef)     z Pr(&gt;|z|)</span></span>
<span id="cb353-29"><a href="read-in-er-io-data.html#cb353-29"></a><span class="co">## grow 0.2072    1.2303   0.2565 0.808    0.419</span></span>
<span id="cb353-30"><a href="read-in-er-io-data.html#cb353-30"></a><span class="co">## ArmB 0.3770    1.4578   0.5117 0.737    0.461</span></span>
<span id="cb353-31"><a href="read-in-er-io-data.html#cb353-31"></a><span class="co">## </span></span>
<span id="cb353-32"><a href="read-in-er-io-data.html#cb353-32"></a><span class="co">##      exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb353-33"><a href="read-in-er-io-data.html#cb353-33"></a><span class="co">## grow     1.230     0.8128    0.7441     2.034</span></span>
<span id="cb353-34"><a href="read-in-er-io-data.html#cb353-34"></a><span class="co">## ArmB     1.458     0.6859    0.5348     3.974</span></span>
<span id="cb353-35"><a href="read-in-er-io-data.html#cb353-35"></a><span class="co">## </span></span>
<span id="cb353-36"><a href="read-in-er-io-data.html#cb353-36"></a><span class="co">## Concordance= 0.511  (se = 0.074 )</span></span>
<span id="cb353-37"><a href="read-in-er-io-data.html#cb353-37"></a><span class="co">## Likelihood ratio test= 0.76  on 2 df,   p=0.7</span></span>
<span id="cb353-38"><a href="read-in-er-io-data.html#cb353-38"></a><span class="co">## Wald test            = 0.78  on 2 df,   p=0.7</span></span>
<span id="cb353-39"><a href="read-in-er-io-data.html#cb353-39"></a><span class="co">## Score (logrank) test = 0.77  on 2 df,   p=0.7</span></span></code></pre></div>
<p>Check what happens when you have treament arm:</p>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="read-in-er-io-data.html#cb354-1"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>stab, <span class="kw">quantile</span>(newtab<span class="op">$</span>stab, <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb354-2"><a href="read-in-er-io-data.html#cb354-2"></a></span>
<span id="cb354-3"><a href="read-in-er-io-data.html#cb354-3"></a>tx=newtab<span class="op">$</span>Arm</span>
<span id="cb354-4"><a href="read-in-er-io-data.html#cb354-4"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb354-5"><a href="read-in-er-io-data.html#cb354-5"></a><span class="co"># ax2=plot(survfit(ssPS~TCGAvalCut+ERClin$Arm[na.omit(ax1)]), main=paste(i, &quot;lum growing&quot;), col=brewer.pal(4, &quot;Blues&quot;),lwd=2, xlab=&quot;Time (months)&quot;, ylab=&quot;PS&quot;, mark.time=T)</span></span>
<span id="cb354-6"><a href="read-in-er-io-data.html#cb354-6"></a></span>
<span id="cb354-7"><a href="read-in-er-io-data.html#cb354-7"></a>dfs1=<span class="kw">coxph</span>(ssPS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>]<span class="op">~</span>TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>])</span>
<span id="cb354-8"><a href="read-in-er-io-data.html#cb354-8"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb354-9"><a href="read-in-er-io-data.html#cb354-9"></a><span class="co">## Call:</span></span>
<span id="cb354-10"><a href="read-in-er-io-data.html#cb354-10"></a><span class="co">## coxph(formula = ssPS[newtab$Arm == &quot;A&quot;] ~ TCGAvalCut[newtab$Arm == </span></span>
<span id="cb354-11"><a href="read-in-er-io-data.html#cb354-11"></a><span class="co">##     &quot;A&quot;])</span></span>
<span id="cb354-12"><a href="read-in-er-io-data.html#cb354-12"></a><span class="co">## </span></span>
<span id="cb354-13"><a href="read-in-er-io-data.html#cb354-13"></a><span class="co">##   n= 13, number of events= 13 </span></span>
<span id="cb354-14"><a href="read-in-er-io-data.html#cb354-14"></a><span class="co">##    (1 observation deleted due to missingness)</span></span>
<span id="cb354-15"><a href="read-in-er-io-data.html#cb354-15"></a><span class="co">## </span></span>
<span id="cb354-16"><a href="read-in-er-io-data.html#cb354-16"></a><span class="co">##                                   coef exp(coef) se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb354-17"><a href="read-in-er-io-data.html#cb354-17"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;A&quot;]P -0.9221    0.3977   0.6532 -1.412    0.158</span></span>
<span id="cb354-18"><a href="read-in-er-io-data.html#cb354-18"></a><span class="co">## </span></span>
<span id="cb354-19"><a href="read-in-er-io-data.html#cb354-19"></a><span class="co">##                                exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb354-20"><a href="read-in-er-io-data.html#cb354-20"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;A&quot;]P    0.3977      2.515    0.1105     1.431</span></span>
<span id="cb354-21"><a href="read-in-er-io-data.html#cb354-21"></a><span class="co">## </span></span>
<span id="cb354-22"><a href="read-in-er-io-data.html#cb354-22"></a><span class="co">## Concordance= 0.597  (se = 0.079 )</span></span>
<span id="cb354-23"><a href="read-in-er-io-data.html#cb354-23"></a><span class="co">## Likelihood ratio test= 1.94  on 1 df,   p=0.2</span></span>
<span id="cb354-24"><a href="read-in-er-io-data.html#cb354-24"></a><span class="co">## Wald test            = 1.99  on 1 df,   p=0.2</span></span>
<span id="cb354-25"><a href="read-in-er-io-data.html#cb354-25"></a><span class="co">## Score (logrank) test = 2.12  on 1 df,   p=0.1</span></span>
<span id="cb354-26"><a href="read-in-er-io-data.html#cb354-26"></a>adfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb354-27"><a href="read-in-er-io-data.html#cb354-27"></a>dfs1=<span class="kw">coxph</span>(ssPS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>])</span>
<span id="cb354-28"><a href="read-in-er-io-data.html#cb354-28"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb354-29"><a href="read-in-er-io-data.html#cb354-29"></a><span class="co">## Call:</span></span>
<span id="cb354-30"><a href="read-in-er-io-data.html#cb354-30"></a><span class="co">## coxph(formula = ssPS[newtab$Arm == &quot;B&quot;] ~ TCGAvalCut[newtab$Arm == </span></span>
<span id="cb354-31"><a href="read-in-er-io-data.html#cb354-31"></a><span class="co">##     &quot;B&quot;])</span></span>
<span id="cb354-32"><a href="read-in-er-io-data.html#cb354-32"></a><span class="co">## </span></span>
<span id="cb354-33"><a href="read-in-er-io-data.html#cb354-33"></a><span class="co">##   n= 16, number of events= 14 </span></span>
<span id="cb354-34"><a href="read-in-er-io-data.html#cb354-34"></a><span class="co">## </span></span>
<span id="cb354-35"><a href="read-in-er-io-data.html#cb354-35"></a><span class="co">##                                   coef exp(coef) se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb354-36"><a href="read-in-er-io-data.html#cb354-36"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;B&quot;]P -0.5066    0.6026   0.5771 -0.878     0.38</span></span>
<span id="cb354-37"><a href="read-in-er-io-data.html#cb354-37"></a><span class="co">## </span></span>
<span id="cb354-38"><a href="read-in-er-io-data.html#cb354-38"></a><span class="co">##                                exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb354-39"><a href="read-in-er-io-data.html#cb354-39"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;B&quot;]P    0.6026       1.66    0.1944     1.867</span></span>
<span id="cb354-40"><a href="read-in-er-io-data.html#cb354-40"></a><span class="co">## </span></span>
<span id="cb354-41"><a href="read-in-er-io-data.html#cb354-41"></a><span class="co">## Concordance= 0.626  (se = 0.08 )</span></span>
<span id="cb354-42"><a href="read-in-er-io-data.html#cb354-42"></a><span class="co">## Likelihood ratio test= 0.77  on 1 df,   p=0.4</span></span>
<span id="cb354-43"><a href="read-in-er-io-data.html#cb354-43"></a><span class="co">## Wald test            = 0.77  on 1 df,   p=0.4</span></span>
<span id="cb354-44"><a href="read-in-er-io-data.html#cb354-44"></a><span class="co">## Score (logrank) test = 0.78  on 1 df,   p=0.4</span></span>
<span id="cb354-45"><a href="read-in-er-io-data.html#cb354-45"></a>bdfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb354-46"><a href="read-in-er-io-data.html#cb354-46"></a></span>
<span id="cb354-47"><a href="read-in-er-io-data.html#cb354-47"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb354-48"><a href="read-in-er-io-data.html#cb354-48"></a><span class="co">## plot separately the different arms</span></span>
<span id="cb354-49"><a href="read-in-er-io-data.html#cb354-49"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(adfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], adfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb354-50"><a href="read-in-er-io-data.html#cb354-50"></a>ns=<span class="kw">table</span>(TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>])</span>
<span id="cb354-51"><a href="read-in-er-io-data.html#cb354-51"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[tx<span class="op">==</span><span class="st">&quot;A&quot;</span>]<span class="op">~</span>TCGAvalCut[tx<span class="op">==</span><span class="st">&quot;A&quot;</span>]), <span class="dt">main=</span><span class="st">&quot;stable ArmA&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">12</span>))</span>
<span id="cb354-52"><a href="read-in-er-io-data.html#cb354-52"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>])) </span>
<span id="cb354-53"><a href="read-in-er-io-data.html#cb354-53"></a><span class="kw">text</span>(<span class="dv">10</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb354-54"><a href="read-in-er-io-data.html#cb354-54"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(bdfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], bdfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb354-55"><a href="read-in-er-io-data.html#cb354-55"></a>ns=<span class="kw">table</span>(TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>])</span>
<span id="cb354-56"><a href="read-in-er-io-data.html#cb354-56"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[tx<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut[tx<span class="op">==</span><span class="st">&quot;B&quot;</span>]), <span class="dt">main=</span><span class="st">&quot;stable ArmB&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">12</span>))</span>
<span id="cb354-57"><a href="read-in-er-io-data.html#cb354-57"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))  </span>
<span id="cb354-58"><a href="read-in-er-io-data.html#cb354-58"></a><span class="kw">text</span>(<span class="dv">10</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb354-59"><a href="read-in-er-io-data.html#cb354-59"></a></span>
<span id="cb354-60"><a href="read-in-er-io-data.html#cb354-60"></a><span class="co">## plot the OS result</span></span>
<span id="cb354-61"><a href="read-in-er-io-data.html#cb354-61"></a><span class="co"># OS based on arm: stable</span></span>
<span id="cb354-62"><a href="read-in-er-io-data.html#cb354-62"></a></span>
<span id="cb354-63"><a href="read-in-er-io-data.html#cb354-63"></a>dfs1=<span class="kw">coxph</span>(ssOS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>]<span class="op">~</span>TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>])</span>
<span id="cb354-64"><a href="read-in-er-io-data.html#cb354-64"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb354-65"><a href="read-in-er-io-data.html#cb354-65"></a><span class="co">## Call:</span></span>
<span id="cb354-66"><a href="read-in-er-io-data.html#cb354-66"></a><span class="co">## coxph(formula = ssOS[newtab$Arm == &quot;A&quot;] ~ TCGAvalCut[newtab$Arm == </span></span>
<span id="cb354-67"><a href="read-in-er-io-data.html#cb354-67"></a><span class="co">##     &quot;A&quot;])</span></span>
<span id="cb354-68"><a href="read-in-er-io-data.html#cb354-68"></a><span class="co">## </span></span>
<span id="cb354-69"><a href="read-in-er-io-data.html#cb354-69"></a><span class="co">##   n= 13, number of events= 10 </span></span>
<span id="cb354-70"><a href="read-in-er-io-data.html#cb354-70"></a><span class="co">##    (1 observation deleted due to missingness)</span></span>
<span id="cb354-71"><a href="read-in-er-io-data.html#cb354-71"></a><span class="co">## </span></span>
<span id="cb354-72"><a href="read-in-er-io-data.html#cb354-72"></a><span class="co">##                                   coef exp(coef) se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb354-73"><a href="read-in-er-io-data.html#cb354-73"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;A&quot;]P -0.5760    0.5622   0.6418 -0.897     0.37</span></span>
<span id="cb354-74"><a href="read-in-er-io-data.html#cb354-74"></a><span class="co">## </span></span>
<span id="cb354-75"><a href="read-in-er-io-data.html#cb354-75"></a><span class="co">##                                exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb354-76"><a href="read-in-er-io-data.html#cb354-76"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;A&quot;]P    0.5622      1.779    0.1598     1.978</span></span>
<span id="cb354-77"><a href="read-in-er-io-data.html#cb354-77"></a><span class="co">## </span></span>
<span id="cb354-78"><a href="read-in-er-io-data.html#cb354-78"></a><span class="co">## Concordance= 0.555  (se = 0.088 )</span></span>
<span id="cb354-79"><a href="read-in-er-io-data.html#cb354-79"></a><span class="co">## Likelihood ratio test= 0.8  on 1 df,   p=0.4</span></span>
<span id="cb354-80"><a href="read-in-er-io-data.html#cb354-80"></a><span class="co">## Wald test            = 0.81  on 1 df,   p=0.4</span></span>
<span id="cb354-81"><a href="read-in-er-io-data.html#cb354-81"></a><span class="co">## Score (logrank) test = 0.83  on 1 df,   p=0.4</span></span>
<span id="cb354-82"><a href="read-in-er-io-data.html#cb354-82"></a>adfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb354-83"><a href="read-in-er-io-data.html#cb354-83"></a>dfs1=<span class="kw">coxph</span>(ssOS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>])</span>
<span id="cb354-84"><a href="read-in-er-io-data.html#cb354-84"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb354-85"><a href="read-in-er-io-data.html#cb354-85"></a><span class="co">## Call:</span></span>
<span id="cb354-86"><a href="read-in-er-io-data.html#cb354-86"></a><span class="co">## coxph(formula = ssOS[newtab$Arm == &quot;B&quot;] ~ TCGAvalCut[newtab$Arm == </span></span>
<span id="cb354-87"><a href="read-in-er-io-data.html#cb354-87"></a><span class="co">##     &quot;B&quot;])</span></span>
<span id="cb354-88"><a href="read-in-er-io-data.html#cb354-88"></a><span class="co">## </span></span>
<span id="cb354-89"><a href="read-in-er-io-data.html#cb354-89"></a><span class="co">##   n= 16, number of events= 12 </span></span>
<span id="cb354-90"><a href="read-in-er-io-data.html#cb354-90"></a><span class="co">## </span></span>
<span id="cb354-91"><a href="read-in-er-io-data.html#cb354-91"></a><span class="co">##                                   coef exp(coef) se(coef)      z Pr(&gt;|z|)  </span></span>
<span id="cb354-92"><a href="read-in-er-io-data.html#cb354-92"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;B&quot;]P -1.5044    0.2222   0.6941 -2.167   0.0302 *</span></span>
<span id="cb354-93"><a href="read-in-er-io-data.html#cb354-93"></a><span class="co">## ---</span></span>
<span id="cb354-94"><a href="read-in-er-io-data.html#cb354-94"></a><span class="co">## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb354-95"><a href="read-in-er-io-data.html#cb354-95"></a><span class="co">## </span></span>
<span id="cb354-96"><a href="read-in-er-io-data.html#cb354-96"></a><span class="co">##                                exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb354-97"><a href="read-in-er-io-data.html#cb354-97"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;B&quot;]P    0.2222      4.501     0.057    0.8659</span></span>
<span id="cb354-98"><a href="read-in-er-io-data.html#cb354-98"></a><span class="co">## </span></span>
<span id="cb354-99"><a href="read-in-er-io-data.html#cb354-99"></a><span class="co">## Concordance= 0.694  (se = 0.054 )</span></span>
<span id="cb354-100"><a href="read-in-er-io-data.html#cb354-100"></a><span class="co">## Likelihood ratio test= 5.39  on 1 df,   p=0.02</span></span>
<span id="cb354-101"><a href="read-in-er-io-data.html#cb354-101"></a><span class="co">## Wald test            = 4.7  on 1 df,   p=0.03</span></span>
<span id="cb354-102"><a href="read-in-er-io-data.html#cb354-102"></a><span class="co">## Score (logrank) test = 5.5  on 1 df,   p=0.02</span></span>
<span id="cb354-103"><a href="read-in-er-io-data.html#cb354-103"></a>bdfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb354-104"><a href="read-in-er-io-data.html#cb354-104"></a></span>
<span id="cb354-105"><a href="read-in-er-io-data.html#cb354-105"></a><span class="co">## plot separately the different arms</span></span>
<span id="cb354-106"><a href="read-in-er-io-data.html#cb354-106"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(adfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], adfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb354-107"><a href="read-in-er-io-data.html#cb354-107"></a>ns=<span class="kw">table</span>(TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>])</span>
<span id="cb354-108"><a href="read-in-er-io-data.html#cb354-108"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[tx<span class="op">==</span><span class="st">&quot;A&quot;</span>]<span class="op">~</span>TCGAvalCut[tx<span class="op">==</span><span class="st">&quot;A&quot;</span>]), <span class="dt">main=</span><span class="st">&quot;stable ArmA&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">35</span>))</span>
<span id="cb354-109"><a href="read-in-er-io-data.html#cb354-109"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>])) </span>
<span id="cb354-110"><a href="read-in-er-io-data.html#cb354-110"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb354-111"><a href="read-in-er-io-data.html#cb354-111"></a></span>
<span id="cb354-112"><a href="read-in-er-io-data.html#cb354-112"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(bdfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], bdfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb354-113"><a href="read-in-er-io-data.html#cb354-113"></a>ns=<span class="kw">table</span>(TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>])</span>
<span id="cb354-114"><a href="read-in-er-io-data.html#cb354-114"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[tx<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut[tx<span class="op">==</span><span class="st">&quot;B&quot;</span>]), <span class="dt">main=</span><span class="st">&quot;stable ArmB&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">35</span>))</span>
<span id="cb354-115"><a href="read-in-er-io-data.html#cb354-115"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))  </span>
<span id="cb354-116"><a href="read-in-er-io-data.html#cb354-116"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/stable-arm-1.png" width="672" /></p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="read-in-er-io-data.html#cb355-1"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>grow, <span class="kw">c</span>(<span class="op">-</span><span class="dv">10</span>, <span class="kw">median</span>(newtab<span class="op">$</span>grow), <span class="dv">10</span>), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb355-2"><a href="read-in-er-io-data.html#cb355-2"></a></span>
<span id="cb355-3"><a href="read-in-er-io-data.html#cb355-3"></a><span class="co"># ax2=plot(survfit(ssPS~TCGAvalCut+ERClin$Arm[na.omit(ax1)]), main=paste(i, &quot;lum growing&quot;), col=brewer.pal(4, &quot;Blues&quot;),lwd=2, xlab=&quot;Time (months)&quot;, ylab=&quot;PS&quot;, mark.time=T)</span></span>
<span id="cb355-4"><a href="read-in-er-io-data.html#cb355-4"></a></span>
<span id="cb355-5"><a href="read-in-er-io-data.html#cb355-5"></a>dfs1=<span class="kw">coxph</span>(ssPS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>]<span class="op">~</span>TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>])</span>
<span id="cb355-6"><a href="read-in-er-io-data.html#cb355-6"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb355-7"><a href="read-in-er-io-data.html#cb355-7"></a><span class="co">## Call:</span></span>
<span id="cb355-8"><a href="read-in-er-io-data.html#cb355-8"></a><span class="co">## coxph(formula = ssPS[newtab$Arm == &quot;A&quot;] ~ TCGAvalCut[newtab$Arm == </span></span>
<span id="cb355-9"><a href="read-in-er-io-data.html#cb355-9"></a><span class="co">##     &quot;A&quot;])</span></span>
<span id="cb355-10"><a href="read-in-er-io-data.html#cb355-10"></a><span class="co">## </span></span>
<span id="cb355-11"><a href="read-in-er-io-data.html#cb355-11"></a><span class="co">##   n= 14, number of events= 14 </span></span>
<span id="cb355-12"><a href="read-in-er-io-data.html#cb355-12"></a><span class="co">## </span></span>
<span id="cb355-13"><a href="read-in-er-io-data.html#cb355-13"></a><span class="co">##                                  coef exp(coef) se(coef)     z Pr(&gt;|z|)</span></span>
<span id="cb355-14"><a href="read-in-er-io-data.html#cb355-14"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;A&quot;]P 0.6467    1.9092   0.6795 0.952    0.341</span></span>
<span id="cb355-15"><a href="read-in-er-io-data.html#cb355-15"></a><span class="co">## </span></span>
<span id="cb355-16"><a href="read-in-er-io-data.html#cb355-16"></a><span class="co">##                                exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb355-17"><a href="read-in-er-io-data.html#cb355-17"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;A&quot;]P     1.909     0.5238     0.504     7.232</span></span>
<span id="cb355-18"><a href="read-in-er-io-data.html#cb355-18"></a><span class="co">## </span></span>
<span id="cb355-19"><a href="read-in-er-io-data.html#cb355-19"></a><span class="co">## Concordance= 0.533  (se = 0.093 )</span></span>
<span id="cb355-20"><a href="read-in-er-io-data.html#cb355-20"></a><span class="co">## Likelihood ratio test= 1  on 1 df,   p=0.3</span></span>
<span id="cb355-21"><a href="read-in-er-io-data.html#cb355-21"></a><span class="co">## Wald test            = 0.91  on 1 df,   p=0.3</span></span>
<span id="cb355-22"><a href="read-in-er-io-data.html#cb355-22"></a><span class="co">## Score (logrank) test = 0.93  on 1 df,   p=0.3</span></span>
<span id="cb355-23"><a href="read-in-er-io-data.html#cb355-23"></a>adfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb355-24"><a href="read-in-er-io-data.html#cb355-24"></a>dfs1=<span class="kw">coxph</span>(ssPS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>])</span>
<span id="cb355-25"><a href="read-in-er-io-data.html#cb355-25"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb355-26"><a href="read-in-er-io-data.html#cb355-26"></a><span class="co">## Call:</span></span>
<span id="cb355-27"><a href="read-in-er-io-data.html#cb355-27"></a><span class="co">## coxph(formula = ssPS[newtab$Arm == &quot;B&quot;] ~ TCGAvalCut[newtab$Arm == </span></span>
<span id="cb355-28"><a href="read-in-er-io-data.html#cb355-28"></a><span class="co">##     &quot;B&quot;])</span></span>
<span id="cb355-29"><a href="read-in-er-io-data.html#cb355-29"></a><span class="co">## </span></span>
<span id="cb355-30"><a href="read-in-er-io-data.html#cb355-30"></a><span class="co">##   n= 16, number of events= 14 </span></span>
<span id="cb355-31"><a href="read-in-er-io-data.html#cb355-31"></a><span class="co">## </span></span>
<span id="cb355-32"><a href="read-in-er-io-data.html#cb355-32"></a><span class="co">##                                  coef exp(coef) se(coef)     z Pr(&gt;|z|)</span></span>
<span id="cb355-33"><a href="read-in-er-io-data.html#cb355-33"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;B&quot;]P 0.1065    1.1123   0.5883 0.181    0.856</span></span>
<span id="cb355-34"><a href="read-in-er-io-data.html#cb355-34"></a><span class="co">## </span></span>
<span id="cb355-35"><a href="read-in-er-io-data.html#cb355-35"></a><span class="co">##                                exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb355-36"><a href="read-in-er-io-data.html#cb355-36"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;B&quot;]P     1.112      0.899    0.3511     3.524</span></span>
<span id="cb355-37"><a href="read-in-er-io-data.html#cb355-37"></a><span class="co">## </span></span>
<span id="cb355-38"><a href="read-in-er-io-data.html#cb355-38"></a><span class="co">## Concordance= 0.549  (se = 0.098 )</span></span>
<span id="cb355-39"><a href="read-in-er-io-data.html#cb355-39"></a><span class="co">## Likelihood ratio test= 0.03  on 1 df,   p=0.9</span></span>
<span id="cb355-40"><a href="read-in-er-io-data.html#cb355-40"></a><span class="co">## Wald test            = 0.03  on 1 df,   p=0.9</span></span>
<span id="cb355-41"><a href="read-in-er-io-data.html#cb355-41"></a><span class="co">## Score (logrank) test = 0.03  on 1 df,   p=0.9</span></span>
<span id="cb355-42"><a href="read-in-er-io-data.html#cb355-42"></a>bdfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb355-43"><a href="read-in-er-io-data.html#cb355-43"></a></span>
<span id="cb355-44"><a href="read-in-er-io-data.html#cb355-44"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb355-45"><a href="read-in-er-io-data.html#cb355-45"></a><span class="co">## plot separately the different arms</span></span>
<span id="cb355-46"><a href="read-in-er-io-data.html#cb355-46"></a>ns=<span class="kw">table</span>(TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>])</span>
<span id="cb355-47"><a href="read-in-er-io-data.html#cb355-47"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(adfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], adfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb355-48"><a href="read-in-er-io-data.html#cb355-48"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[tx<span class="op">==</span><span class="st">&quot;A&quot;</span>]<span class="op">~</span>TCGAvalCut[tx<span class="op">==</span><span class="st">&quot;A&quot;</span>]), <span class="dt">main=</span><span class="st">&quot;grow ArmA&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb355-49"><a href="read-in-er-io-data.html#cb355-49"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb355-50"><a href="read-in-er-io-data.html#cb355-50"></a><span class="kw">text</span>(<span class="dv">12</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb355-51"><a href="read-in-er-io-data.html#cb355-51"></a>ns=<span class="kw">table</span>(TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>])</span>
<span id="cb355-52"><a href="read-in-er-io-data.html#cb355-52"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(bdfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], bdfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb355-53"><a href="read-in-er-io-data.html#cb355-53"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[tx<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut[tx<span class="op">==</span><span class="st">&quot;B&quot;</span>]), <span class="dt">main=</span><span class="st">&quot;grow ArmB&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb355-54"><a href="read-in-er-io-data.html#cb355-54"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb355-55"><a href="read-in-er-io-data.html#cb355-55"></a><span class="kw">text</span>(<span class="dv">12</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb355-56"><a href="read-in-er-io-data.html#cb355-56"></a></span>
<span id="cb355-57"><a href="read-in-er-io-data.html#cb355-57"></a></span>
<span id="cb355-58"><a href="read-in-er-io-data.html#cb355-58"></a><span class="co">## plot the OS result</span></span>
<span id="cb355-59"><a href="read-in-er-io-data.html#cb355-59"></a><span class="co"># OS based on arm: stable</span></span>
<span id="cb355-60"><a href="read-in-er-io-data.html#cb355-60"></a></span>
<span id="cb355-61"><a href="read-in-er-io-data.html#cb355-61"></a>dfs1=<span class="kw">coxph</span>(ssOS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>]<span class="op">~</span>TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>])</span>
<span id="cb355-62"><a href="read-in-er-io-data.html#cb355-62"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb355-63"><a href="read-in-er-io-data.html#cb355-63"></a><span class="co">## Call:</span></span>
<span id="cb355-64"><a href="read-in-er-io-data.html#cb355-64"></a><span class="co">## coxph(formula = ssOS[newtab$Arm == &quot;A&quot;] ~ TCGAvalCut[newtab$Arm == </span></span>
<span id="cb355-65"><a href="read-in-er-io-data.html#cb355-65"></a><span class="co">##     &quot;A&quot;])</span></span>
<span id="cb355-66"><a href="read-in-er-io-data.html#cb355-66"></a><span class="co">## </span></span>
<span id="cb355-67"><a href="read-in-er-io-data.html#cb355-67"></a><span class="co">##   n= 14, number of events= 10 </span></span>
<span id="cb355-68"><a href="read-in-er-io-data.html#cb355-68"></a><span class="co">## </span></span>
<span id="cb355-69"><a href="read-in-er-io-data.html#cb355-69"></a><span class="co">##                                   coef exp(coef) se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb355-70"><a href="read-in-er-io-data.html#cb355-70"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;A&quot;]P -0.7614    0.4670   0.7141 -1.066    0.286</span></span>
<span id="cb355-71"><a href="read-in-er-io-data.html#cb355-71"></a><span class="co">## </span></span>
<span id="cb355-72"><a href="read-in-er-io-data.html#cb355-72"></a><span class="co">##                                exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb355-73"><a href="read-in-er-io-data.html#cb355-73"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;A&quot;]P     0.467      2.141    0.1152     1.893</span></span>
<span id="cb355-74"><a href="read-in-er-io-data.html#cb355-74"></a><span class="co">## </span></span>
<span id="cb355-75"><a href="read-in-er-io-data.html#cb355-75"></a><span class="co">## Concordance= 0.608  (se = 0.084 )</span></span>
<span id="cb355-76"><a href="read-in-er-io-data.html#cb355-76"></a><span class="co">## Likelihood ratio test= 1.03  on 1 df,   p=0.3</span></span>
<span id="cb355-77"><a href="read-in-er-io-data.html#cb355-77"></a><span class="co">## Wald test            = 1.14  on 1 df,   p=0.3</span></span>
<span id="cb355-78"><a href="read-in-er-io-data.html#cb355-78"></a><span class="co">## Score (logrank) test = 1.19  on 1 df,   p=0.3</span></span>
<span id="cb355-79"><a href="read-in-er-io-data.html#cb355-79"></a>adfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb355-80"><a href="read-in-er-io-data.html#cb355-80"></a>dfs1=<span class="kw">coxph</span>(ssOS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>])</span>
<span id="cb355-81"><a href="read-in-er-io-data.html#cb355-81"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb355-82"><a href="read-in-er-io-data.html#cb355-82"></a><span class="co">## Call:</span></span>
<span id="cb355-83"><a href="read-in-er-io-data.html#cb355-83"></a><span class="co">## coxph(formula = ssOS[newtab$Arm == &quot;B&quot;] ~ TCGAvalCut[newtab$Arm == </span></span>
<span id="cb355-84"><a href="read-in-er-io-data.html#cb355-84"></a><span class="co">##     &quot;B&quot;])</span></span>
<span id="cb355-85"><a href="read-in-er-io-data.html#cb355-85"></a><span class="co">## </span></span>
<span id="cb355-86"><a href="read-in-er-io-data.html#cb355-86"></a><span class="co">##   n= 16, number of events= 12 </span></span>
<span id="cb355-87"><a href="read-in-er-io-data.html#cb355-87"></a><span class="co">## </span></span>
<span id="cb355-88"><a href="read-in-er-io-data.html#cb355-88"></a><span class="co">##                                  coef exp(coef) se(coef)     z Pr(&gt;|z|)</span></span>
<span id="cb355-89"><a href="read-in-er-io-data.html#cb355-89"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;B&quot;]P 0.8206    2.2718   0.6166 1.331    0.183</span></span>
<span id="cb355-90"><a href="read-in-er-io-data.html#cb355-90"></a><span class="co">## </span></span>
<span id="cb355-91"><a href="read-in-er-io-data.html#cb355-91"></a><span class="co">##                                exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb355-92"><a href="read-in-er-io-data.html#cb355-92"></a><span class="co">## TCGAvalCut[newtab$Arm == &quot;B&quot;]P     2.272     0.4402    0.6785     7.607</span></span>
<span id="cb355-93"><a href="read-in-er-io-data.html#cb355-93"></a><span class="co">## </span></span>
<span id="cb355-94"><a href="read-in-er-io-data.html#cb355-94"></a><span class="co">## Concordance= 0.563  (se = 0.071 )</span></span>
<span id="cb355-95"><a href="read-in-er-io-data.html#cb355-95"></a><span class="co">## Likelihood ratio test= 1.69  on 1 df,   p=0.2</span></span>
<span id="cb355-96"><a href="read-in-er-io-data.html#cb355-96"></a><span class="co">## Wald test            = 1.77  on 1 df,   p=0.2</span></span>
<span id="cb355-97"><a href="read-in-er-io-data.html#cb355-97"></a><span class="co">## Score (logrank) test = 1.86  on 1 df,   p=0.2</span></span>
<span id="cb355-98"><a href="read-in-er-io-data.html#cb355-98"></a>bdfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb355-99"><a href="read-in-er-io-data.html#cb355-99"></a></span>
<span id="cb355-100"><a href="read-in-er-io-data.html#cb355-100"></a><span class="co">## plot separately the different arms</span></span>
<span id="cb355-101"><a href="read-in-er-io-data.html#cb355-101"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(adfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], adfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb355-102"><a href="read-in-er-io-data.html#cb355-102"></a>ns=<span class="kw">table</span>(TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>])</span>
<span id="cb355-103"><a href="read-in-er-io-data.html#cb355-103"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[tx<span class="op">==</span><span class="st">&quot;A&quot;</span>]<span class="op">~</span>TCGAvalCut[tx<span class="op">==</span><span class="st">&quot;A&quot;</span>]), <span class="dt">main=</span><span class="st">&quot;grow ArmA&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb355-104"><a href="read-in-er-io-data.html#cb355-104"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb355-105"><a href="read-in-er-io-data.html#cb355-105"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb355-106"><a href="read-in-er-io-data.html#cb355-106"></a>ns=<span class="kw">table</span>(TCGAvalCut[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>])</span>
<span id="cb355-107"><a href="read-in-er-io-data.html#cb355-107"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(bdfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], bdfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb355-108"><a href="read-in-er-io-data.html#cb355-108"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[tx<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut[tx<span class="op">==</span><span class="st">&quot;B&quot;</span>]), <span class="dt">main=</span><span class="st">&quot;grow ArmB&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb355-109"><a href="read-in-er-io-data.html#cb355-109"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb355-110"><a href="read-in-er-io-data.html#cb355-110"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/growing-arms-1.png" width="672" /></p>
<p>Note that the ams are imbalanced, do this again after splitting accord to median in both groups</p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="read-in-er-io-data.html#cb356-1"></a></span>
<span id="cb356-2"><a href="read-in-er-io-data.html#cb356-2"></a>tx=<span class="kw">which</span>(newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>)</span>
<span id="cb356-3"><a href="read-in-er-io-data.html#cb356-3"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>grow[tx], <span class="kw">c</span>(<span class="op">-</span><span class="dv">10</span>, <span class="kw">median</span>(newtab<span class="op">$</span>grow[tx]), <span class="dv">10</span>), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb356-4"><a href="read-in-er-io-data.html#cb356-4"></a></span>
<span id="cb356-5"><a href="read-in-er-io-data.html#cb356-5"></a><span class="co"># ax2=plot(survfit(ssPS~TCGAvalCut+ERClin$Arm[na.omit(ax1)]), main=paste(i, &quot;lum growing&quot;), col=brewer.pal(4, &quot;Blues&quot;),lwd=2, xlab=&quot;Time (months)&quot;, ylab=&quot;PS&quot;, mark.time=T)</span></span>
<span id="cb356-6"><a href="read-in-er-io-data.html#cb356-6"></a></span>
<span id="cb356-7"><a href="read-in-er-io-data.html#cb356-7"></a>dfs1=<span class="kw">coxph</span>(ssPS[tx]<span class="op">~</span>TCGAvalCut)</span>
<span id="cb356-8"><a href="read-in-er-io-data.html#cb356-8"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb356-9"><a href="read-in-er-io-data.html#cb356-9"></a>adfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb356-10"><a href="read-in-er-io-data.html#cb356-10"></a></span>
<span id="cb356-11"><a href="read-in-er-io-data.html#cb356-11"></a>dfs1=<span class="kw">coxph</span>(ssOS[tx]<span class="op">~</span>TCGAvalCut)</span>
<span id="cb356-12"><a href="read-in-er-io-data.html#cb356-12"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb356-13"><a href="read-in-er-io-data.html#cb356-13"></a>aos=<span class="kw">summary</span>(dfs1)</span>
<span id="cb356-14"><a href="read-in-er-io-data.html#cb356-14"></a></span>
<span id="cb356-15"><a href="read-in-er-io-data.html#cb356-15"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb356-16"><a href="read-in-er-io-data.html#cb356-16"></a><span class="co">## plot separately the different arms</span></span>
<span id="cb356-17"><a href="read-in-er-io-data.html#cb356-17"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(adfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], adfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb356-18"><a href="read-in-er-io-data.html#cb356-18"></a>ns=<span class="kw">table</span>(TCGAvalCut)</span>
<span id="cb356-19"><a href="read-in-er-io-data.html#cb356-19"></a></span>
<span id="cb356-20"><a href="read-in-er-io-data.html#cb356-20"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[tx]<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="st">&quot;grow ArmA&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb356-21"><a href="read-in-er-io-data.html#cb356-21"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb356-22"><a href="read-in-er-io-data.html#cb356-22"></a><span class="kw">text</span>(<span class="dv">12</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb356-23"><a href="read-in-er-io-data.html#cb356-23"></a></span>
<span id="cb356-24"><a href="read-in-er-io-data.html#cb356-24"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(aos<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], aos<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb356-25"><a href="read-in-er-io-data.html#cb356-25"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[tx]<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="st">&quot;grow ArmA&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb356-26"><a href="read-in-er-io-data.html#cb356-26"></a><span class="kw">text</span>(<span class="dv">15</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb356-27"><a href="read-in-er-io-data.html#cb356-27"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb356-28"><a href="read-in-er-io-data.html#cb356-28"></a></span>
<span id="cb356-29"><a href="read-in-er-io-data.html#cb356-29"></a><span class="co">## plot the OS result</span></span>
<span id="cb356-30"><a href="read-in-er-io-data.html#cb356-30"></a><span class="co"># OS based on arm: stable</span></span>
<span id="cb356-31"><a href="read-in-er-io-data.html#cb356-31"></a></span>
<span id="cb356-32"><a href="read-in-er-io-data.html#cb356-32"></a>tx=<span class="kw">which</span>(newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>)</span>
<span id="cb356-33"><a href="read-in-er-io-data.html#cb356-33"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>grow[tx], <span class="kw">c</span>(<span class="op">-</span><span class="dv">10</span>, <span class="kw">median</span>(newtab<span class="op">$</span>grow[tx]), <span class="dv">10</span>), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb356-34"><a href="read-in-er-io-data.html#cb356-34"></a></span>
<span id="cb356-35"><a href="read-in-er-io-data.html#cb356-35"></a></span>
<span id="cb356-36"><a href="read-in-er-io-data.html#cb356-36"></a>dfs1=<span class="kw">coxph</span>(ssOS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut)</span>
<span id="cb356-37"><a href="read-in-er-io-data.html#cb356-37"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb356-38"><a href="read-in-er-io-data.html#cb356-38"></a>bos=<span class="kw">summary</span>(dfs1)</span>
<span id="cb356-39"><a href="read-in-er-io-data.html#cb356-39"></a>dfs1=<span class="kw">coxph</span>(ssPS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut)</span>
<span id="cb356-40"><a href="read-in-er-io-data.html#cb356-40"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb356-41"><a href="read-in-er-io-data.html#cb356-41"></a>bdfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb356-42"><a href="read-in-er-io-data.html#cb356-42"></a>ns=<span class="kw">table</span>(TCGAvalCut)</span>
<span id="cb356-43"><a href="read-in-er-io-data.html#cb356-43"></a><span class="co">## plot separately the different arms</span></span>
<span id="cb356-44"><a href="read-in-er-io-data.html#cb356-44"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(bdfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], bdfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb356-45"><a href="read-in-er-io-data.html#cb356-45"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[tx]<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="st">&quot;grow ArmB&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">12</span>))</span>
<span id="cb356-46"><a href="read-in-er-io-data.html#cb356-46"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb356-47"><a href="read-in-er-io-data.html#cb356-47"></a><span class="kw">text</span>(<span class="dv">12</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb356-48"><a href="read-in-er-io-data.html#cb356-48"></a></span>
<span id="cb356-49"><a href="read-in-er-io-data.html#cb356-49"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(bos<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], bos<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb356-50"><a href="read-in-er-io-data.html#cb356-50"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[tx]<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="st">&quot;grow ArmB&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">35</span>))</span>
<span id="cb356-51"><a href="read-in-er-io-data.html#cb356-51"></a><span class="kw">text</span>(<span class="dv">15</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb356-52"><a href="read-in-er-io-data.html#cb356-52"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span></code></pre></div>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="read-in-er-io-data.html#cb357-1"></a></span>
<span id="cb357-2"><a href="read-in-er-io-data.html#cb357-2"></a>tx=<span class="kw">which</span>(newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>)</span>
<span id="cb357-3"><a href="read-in-er-io-data.html#cb357-3"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>stab[tx], <span class="kw">c</span>(<span class="op">-</span><span class="dv">10</span>, <span class="kw">median</span>(newtab<span class="op">$</span>stab[tx]), <span class="dv">10</span>), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb357-4"><a href="read-in-er-io-data.html#cb357-4"></a></span>
<span id="cb357-5"><a href="read-in-er-io-data.html#cb357-5"></a><span class="co"># ax2=plot(survfit(ssPS~TCGAvalCut+ERClin$Arm[na.omit(ax1)]), main=paste(i, &quot;lum growing&quot;), col=brewer.pal(4, &quot;Blues&quot;),lwd=2, xlab=&quot;Time (months)&quot;, ylab=&quot;PS&quot;, mark.time=T)</span></span>
<span id="cb357-6"><a href="read-in-er-io-data.html#cb357-6"></a></span>
<span id="cb357-7"><a href="read-in-er-io-data.html#cb357-7"></a>dfs1=<span class="kw">coxph</span>(ssPS[tx]<span class="op">~</span>TCGAvalCut)</span>
<span id="cb357-8"><a href="read-in-er-io-data.html#cb357-8"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb357-9"><a href="read-in-er-io-data.html#cb357-9"></a>adfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb357-10"><a href="read-in-er-io-data.html#cb357-10"></a></span>
<span id="cb357-11"><a href="read-in-er-io-data.html#cb357-11"></a>dfs1=<span class="kw">coxph</span>(ssOS[tx]<span class="op">~</span>TCGAvalCut)</span>
<span id="cb357-12"><a href="read-in-er-io-data.html#cb357-12"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb357-13"><a href="read-in-er-io-data.html#cb357-13"></a>aos=<span class="kw">summary</span>(dfs1)</span>
<span id="cb357-14"><a href="read-in-er-io-data.html#cb357-14"></a></span>
<span id="cb357-15"><a href="read-in-er-io-data.html#cb357-15"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb357-16"><a href="read-in-er-io-data.html#cb357-16"></a><span class="co">## plot separately the different arms</span></span>
<span id="cb357-17"><a href="read-in-er-io-data.html#cb357-17"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(adfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], adfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb357-18"><a href="read-in-er-io-data.html#cb357-18"></a>ns=<span class="kw">table</span>(TCGAvalCut)</span>
<span id="cb357-19"><a href="read-in-er-io-data.html#cb357-19"></a></span>
<span id="cb357-20"><a href="read-in-er-io-data.html#cb357-20"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[tx]<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="st">&quot;stab ArmA&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb357-21"><a href="read-in-er-io-data.html#cb357-21"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb357-22"><a href="read-in-er-io-data.html#cb357-22"></a><span class="kw">text</span>(<span class="dv">12</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb357-23"><a href="read-in-er-io-data.html#cb357-23"></a></span>
<span id="cb357-24"><a href="read-in-er-io-data.html#cb357-24"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(aos<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], aos<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb357-25"><a href="read-in-er-io-data.html#cb357-25"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[tx]<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="st">&quot;stab ArmA&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb357-26"><a href="read-in-er-io-data.html#cb357-26"></a><span class="kw">text</span>(<span class="dv">15</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb357-27"><a href="read-in-er-io-data.html#cb357-27"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb357-28"><a href="read-in-er-io-data.html#cb357-28"></a></span>
<span id="cb357-29"><a href="read-in-er-io-data.html#cb357-29"></a><span class="co">## plot the OS result</span></span>
<span id="cb357-30"><a href="read-in-er-io-data.html#cb357-30"></a><span class="co"># OS based on arm: stable</span></span>
<span id="cb357-31"><a href="read-in-er-io-data.html#cb357-31"></a></span>
<span id="cb357-32"><a href="read-in-er-io-data.html#cb357-32"></a>tx=<span class="kw">which</span>(newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>)</span>
<span id="cb357-33"><a href="read-in-er-io-data.html#cb357-33"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>stab[tx], <span class="kw">c</span>(<span class="op">-</span><span class="dv">10</span>, <span class="kw">median</span>(newtab<span class="op">$</span>stab[tx]), <span class="dv">10</span>), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb357-34"><a href="read-in-er-io-data.html#cb357-34"></a></span>
<span id="cb357-35"><a href="read-in-er-io-data.html#cb357-35"></a></span>
<span id="cb357-36"><a href="read-in-er-io-data.html#cb357-36"></a>dfs1=<span class="kw">coxph</span>(ssOS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut)</span>
<span id="cb357-37"><a href="read-in-er-io-data.html#cb357-37"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb357-38"><a href="read-in-er-io-data.html#cb357-38"></a>bos=<span class="kw">summary</span>(dfs1)</span>
<span id="cb357-39"><a href="read-in-er-io-data.html#cb357-39"></a>dfs1=<span class="kw">coxph</span>(ssPS[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>]<span class="op">~</span>TCGAvalCut)</span>
<span id="cb357-40"><a href="read-in-er-io-data.html#cb357-40"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb357-41"><a href="read-in-er-io-data.html#cb357-41"></a>bdfs=<span class="kw">summary</span>(dfs1)</span>
<span id="cb357-42"><a href="read-in-er-io-data.html#cb357-42"></a>ns=<span class="kw">table</span>(TCGAvalCut)</span>
<span id="cb357-43"><a href="read-in-er-io-data.html#cb357-43"></a><span class="co">## plot separately the different arms</span></span>
<span id="cb357-44"><a href="read-in-er-io-data.html#cb357-44"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(bdfs<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], bdfs<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb357-45"><a href="read-in-er-io-data.html#cb357-45"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[tx]<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="st">&quot;stab ArmB&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">12</span>))</span>
<span id="cb357-46"><a href="read-in-er-io-data.html#cb357-46"></a><span class="kw">text</span>(<span class="dv">4</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb357-47"><a href="read-in-er-io-data.html#cb357-47"></a><span class="kw">text</span>(<span class="dv">12</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span>
<span id="cb357-48"><a href="read-in-er-io-data.html#cb357-48"></a></span>
<span id="cb357-49"><a href="read-in-er-io-data.html#cb357-49"></a>vals=<span class="kw">round</span>(<span class="kw">c</span>(bos<span class="op">$</span>conf.int[ <span class="dv">1</span>,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)], bos<span class="op">$</span>logtest[<span class="dv">3</span>]), <span class="dv">2</span>)</span>
<span id="cb357-50"><a href="read-in-er-io-data.html#cb357-50"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[tx]<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="st">&quot;stab ArmB&quot;</span>, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>)[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">35</span>))</span>
<span id="cb357-51"><a href="read-in-er-io-data.html#cb357-51"></a><span class="kw">text</span>(<span class="dv">15</span>, <span class="fl">0.1</span>, <span class="kw">sprintf</span>(<span class="st">&quot;HR =%s (%s-%s), log-rank=%s&quot;</span>,vals[<span class="dv">1</span>], vals[<span class="dv">2</span>], vals[<span class="dv">3</span>], vals[<span class="dv">4</span>]))   </span>
<span id="cb357-52"><a href="read-in-er-io-data.html#cb357-52"></a><span class="kw">text</span>(<span class="dv">30</span>, <span class="fl">0.8</span>, <span class="kw">sprintf</span>(<span class="st">&quot;L=%s, H=%s&quot;</span>,ns[<span class="dv">1</span>], ns[<span class="dv">2</span>]))</span></code></pre></div>
</div>
<div id="do-associative-study-with-other-features" class="section level2">
<h2><span class="header-section-number">16.2</span> Do associative study with other features?</h2>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="read-in-er-io-data.html#cb358-1"></a><span class="co"># growing signature</span></span>
<span id="cb358-2"><a href="read-in-er-io-data.html#cb358-2"></a>TCGAvalCut=<span class="kw">cut</span>(newtab<span class="op">$</span>grow, <span class="kw">c</span>(<span class="op">-</span><span class="dv">10</span>, <span class="kw">median</span>(newtab<span class="op">$</span>grow), <span class="dv">10</span>), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;H&quot;</span>))</span>
<span id="cb358-3"><a href="read-in-er-io-data.html#cb358-3"></a></span>
<span id="cb358-4"><a href="read-in-er-io-data.html#cb358-4"></a><span class="kw">table</span>(TCGAvalCut, newtab<span class="op">$</span>PDL1)</span>
<span id="cb358-5"><a href="read-in-er-io-data.html#cb358-5"></a><span class="co">##           </span></span>
<span id="cb358-6"><a href="read-in-er-io-data.html#cb358-6"></a><span class="co">## TCGAvalCut    Negative Positive</span></span>
<span id="cb358-7"><a href="read-in-er-io-data.html#cb358-7"></a><span class="co">##          L  1        6        8</span></span>
<span id="cb358-8"><a href="read-in-er-io-data.html#cb358-8"></a><span class="co">##          H  0       11        4</span></span>
<span id="cb358-9"><a href="read-in-er-io-data.html#cb358-9"></a>m1=<span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">8</span>, <span class="dv">11</span>, <span class="dv">4</span>), <span class="dt">ncol=</span><span class="dv">2</span>)</span>
<span id="cb358-10"><a href="read-in-er-io-data.html#cb358-10"></a><span class="kw">chisq.test</span>(m1)</span>
<span id="cb358-11"><a href="read-in-er-io-data.html#cb358-11"></a><span class="co">## </span></span>
<span id="cb358-12"><a href="read-in-er-io-data.html#cb358-12"></a><span class="co">##  Pearson&#39;s Chi-squared test with Yates&#39; continuity correction</span></span>
<span id="cb358-13"><a href="read-in-er-io-data.html#cb358-13"></a><span class="co">## </span></span>
<span id="cb358-14"><a href="read-in-er-io-data.html#cb358-14"></a><span class="co">## data:  m1</span></span>
<span id="cb358-15"><a href="read-in-er-io-data.html#cb358-15"></a><span class="co">## X-squared = 1.6587, df = 1, p-value = 0.1978</span></span>
<span id="cb358-16"><a href="read-in-er-io-data.html#cb358-16"></a></span>
<span id="cb358-17"><a href="read-in-er-io-data.html#cb358-17"></a><span class="kw">table</span>(TCGAvalCut, newtab<span class="op">$</span>Arm)</span>
<span id="cb358-18"><a href="read-in-er-io-data.html#cb358-18"></a><span class="co">##           </span></span>
<span id="cb358-19"><a href="read-in-er-io-data.html#cb358-19"></a><span class="co">## TCGAvalCut  A  B</span></span>
<span id="cb358-20"><a href="read-in-er-io-data.html#cb358-20"></a><span class="co">##          L  4 11</span></span>
<span id="cb358-21"><a href="read-in-er-io-data.html#cb358-21"></a><span class="co">##          H 10  5</span></span>
<span id="cb358-22"><a href="read-in-er-io-data.html#cb358-22"></a><span class="kw">chisq.test</span>(<span class="kw">table</span>(TCGAvalCut, newtab<span class="op">$</span>Arm))</span>
<span id="cb358-23"><a href="read-in-er-io-data.html#cb358-23"></a><span class="co">## </span></span>
<span id="cb358-24"><a href="read-in-er-io-data.html#cb358-24"></a><span class="co">##  Pearson&#39;s Chi-squared test with Yates&#39; continuity correction</span></span>
<span id="cb358-25"><a href="read-in-er-io-data.html#cb358-25"></a><span class="co">## </span></span>
<span id="cb358-26"><a href="read-in-er-io-data.html#cb358-26"></a><span class="co">## data:  table(TCGAvalCut, newtab$Arm)</span></span>
<span id="cb358-27"><a href="read-in-er-io-data.html#cb358-27"></a><span class="co">## X-squared = 3.3482, df = 1, p-value = 0.06728</span></span>
<span id="cb358-28"><a href="read-in-er-io-data.html#cb358-28"></a></span>
<span id="cb358-29"><a href="read-in-er-io-data.html#cb358-29"></a><span class="kw">table</span>(TCGAvalCut,newtab<span class="op">$</span>ClinBen)</span>
<span id="cb358-30"><a href="read-in-er-io-data.html#cb358-30"></a><span class="co">##           </span></span>
<span id="cb358-31"><a href="read-in-er-io-data.html#cb358-31"></a><span class="co">## TCGAvalCut  0  1</span></span>
<span id="cb358-32"><a href="read-in-er-io-data.html#cb358-32"></a><span class="co">##          L 10  5</span></span>
<span id="cb358-33"><a href="read-in-er-io-data.html#cb358-33"></a><span class="co">##          H  9  6</span></span>
<span id="cb358-34"><a href="read-in-er-io-data.html#cb358-34"></a><span class="kw">chisq.test</span>(<span class="kw">table</span>(TCGAvalCut,newtab<span class="op">$</span>ClinBen))</span>
<span id="cb358-35"><a href="read-in-er-io-data.html#cb358-35"></a><span class="co">## </span></span>
<span id="cb358-36"><a href="read-in-er-io-data.html#cb358-36"></a><span class="co">##  Pearson&#39;s Chi-squared test with Yates&#39; continuity correction</span></span>
<span id="cb358-37"><a href="read-in-er-io-data.html#cb358-37"></a><span class="co">## </span></span>
<span id="cb358-38"><a href="read-in-er-io-data.html#cb358-38"></a><span class="co">## data:  table(TCGAvalCut, newtab$ClinBen)</span></span>
<span id="cb358-39"><a href="read-in-er-io-data.html#cb358-39"></a><span class="co">## X-squared = 0, df = 1, p-value = 1</span></span>
<span id="cb358-40"><a href="read-in-er-io-data.html#cb358-40"></a></span>
<span id="cb358-41"><a href="read-in-er-io-data.html#cb358-41"></a><span class="kw">table</span>(TCGAvalCut,newtab<span class="op">$</span>Response)</span>
<span id="cb358-42"><a href="read-in-er-io-data.html#cb358-42"></a><span class="co">##           </span></span>
<span id="cb358-43"><a href="read-in-er-io-data.html#cb358-43"></a><span class="co">## TCGAvalCut NE PD PR SD</span></span>
<span id="cb358-44"><a href="read-in-er-io-data.html#cb358-44"></a><span class="co">##          L  2  3  4  6</span></span>
<span id="cb358-45"><a href="read-in-er-io-data.html#cb358-45"></a><span class="co">##          H  1  5  4  5</span></span>
<span id="cb358-46"><a href="read-in-er-io-data.html#cb358-46"></a></span>
<span id="cb358-47"><a href="read-in-er-io-data.html#cb358-47"></a><span class="kw">chisq.test</span>(<span class="kw">table</span>(TCGAvalCut,newtab<span class="op">$</span>Response))</span>
<span id="cb358-48"><a href="read-in-er-io-data.html#cb358-48"></a><span class="co">## </span></span>
<span id="cb358-49"><a href="read-in-er-io-data.html#cb358-49"></a><span class="co">##  Pearson&#39;s Chi-squared test</span></span>
<span id="cb358-50"><a href="read-in-er-io-data.html#cb358-50"></a><span class="co">## </span></span>
<span id="cb358-51"><a href="read-in-er-io-data.html#cb358-51"></a><span class="co">## data:  table(TCGAvalCut, newtab$Response)</span></span>
<span id="cb358-52"><a href="read-in-er-io-data.html#cb358-52"></a><span class="co">## X-squared = 0.92424, df = 3, p-value = 0.8196</span></span></code></pre></div>
<p>boxplot the growing scores with other covariates:</p>
<div class="sourceCode" id="cb359"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb359-1"><a href="read-in-er-io-data.html#cb359-1"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb359-2"><a href="read-in-er-io-data.html#cb359-2"></a><span class="kw">plot</span>(newtab<span class="op">$</span>grow, newtab<span class="op">$</span>Purity, <span class="dt">ylab=</span><span class="st">&quot;Purity&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;signature&quot;</span>)</span>
<span id="cb359-3"><a href="read-in-er-io-data.html#cb359-3"></a><span class="kw">plot</span>(newtab<span class="op">$</span>grow, newtab<span class="op">$</span>CD8, <span class="dt">ylab=</span><span class="st">&quot;CIBERSORT CD8&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;signature&quot;</span>)</span>
<span id="cb359-4"><a href="read-in-er-io-data.html#cb359-4"></a><span class="kw">plot</span>(newtab<span class="op">$</span>grow, newtab<span class="op">$</span>DC, <span class="dt">ylab=</span><span class="st">&quot;CIBERSORT DC&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;signature&quot;</span>)</span>
<span id="cb359-5"><a href="read-in-er-io-data.html#cb359-5"></a><span class="kw">plot</span>(newtab<span class="op">$</span>grow,newtab<span class="op">$</span>TMB, <span class="dt">ylab=</span><span class="st">&quot;TMB&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;signature&quot;</span>)</span>
<span id="cb359-6"><a href="read-in-er-io-data.html#cb359-6"></a><span class="kw">boxplot</span>(newtab<span class="op">$</span>grow[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>] <span class="op">~</span><span class="st"> </span>newtab<span class="op">$</span>Response[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;A&quot;</span>], <span class="dt">main=</span><span class="st">&quot;Arm A&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;signature&quot;</span>)</span>
<span id="cb359-7"><a href="read-in-er-io-data.html#cb359-7"></a><span class="kw">boxplot</span>(newtab<span class="op">$</span>grow[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>] <span class="op">~</span><span class="st"> </span>newtab<span class="op">$</span>Response[newtab<span class="op">$</span>Arm<span class="op">==</span><span class="st">&quot;B&quot;</span>], <span class="dt">main=</span><span class="st">&quot;Arm B&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;signature&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-107-1.png" width="672" /></p>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="read-in-er-io-data.html#cb360-1"></a></span>
<span id="cb360-2"><a href="read-in-er-io-data.html#cb360-2"></a><span class="co">#plot(newtab$grow, newtab$MICB)</span></span>
<span id="cb360-3"><a href="read-in-er-io-data.html#cb360-3"></a><span class="co">#plot(newtab$grow, newtab$MICA, xlab=&quot;Signature&quot;, ylab=&quot;MICA&quot;, pch=19)</span></span>
<span id="cb360-4"><a href="read-in-er-io-data.html#cb360-4"></a><span class="kw">boxplot</span>(newtab<span class="op">$</span>TMB<span class="op">~</span>TCGAvalCut)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-107-2.png" width="672" /></p>
<p>Check the association between expression of the growing signatrue and the cytokines profiled</p>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="read-in-er-io-data.html#cb361-1"></a>ERcyt2=ERcytokines[<span class="kw">which</span>(ERcytokines<span class="op">$</span>Time<span class="op">==</span><span class="st">&quot;Pre&quot;</span>), ]</span>
<span id="cb361-2"><a href="read-in-er-io-data.html#cb361-2"></a>ERcyt2<span class="op">$</span>grow=newtab<span class="op">$</span>grow[<span class="kw">match</span>(<span class="kw">paste</span>(<span class="st">&quot;Patient&quot;</span>, ERcyt2<span class="op">$</span>Subject.ID, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="kw">rownames</span>(newtab))]</span>
<span id="cb361-3"><a href="read-in-er-io-data.html#cb361-3"></a>ERcyt3=ERcytokines[<span class="kw">which</span>(ERcytokines<span class="op">$</span>Time<span class="op">==</span><span class="st">&quot;Post&quot;</span>), ]</span>
<span id="cb361-4"><a href="read-in-er-io-data.html#cb361-4"></a>ERcyt2<span class="op">$</span>MICA_post=ERcyt3<span class="op">$</span>MICA[<span class="kw">match</span>(ERcyt2<span class="op">$</span>Subject.ID, ERcyt3<span class="op">$</span>Subject.ID)]</span>
<span id="cb361-5"><a href="read-in-er-io-data.html#cb361-5"></a>ERcyt2<span class="op">$</span>MICB_post=ERcyt3<span class="op">$</span>MICB[<span class="kw">match</span>(ERcyt2<span class="op">$</span>Subject.ID, ERcyt3<span class="op">$</span>Subject.ID)]</span>
<span id="cb361-6"><a href="read-in-er-io-data.html#cb361-6"></a></span>
<span id="cb361-7"><a href="read-in-er-io-data.html#cb361-7"></a>CorVals=<span class="kw">sapply</span>(<span class="kw">c</span>(<span class="dv">6</span><span class="op">:</span><span class="dv">19</span>, <span class="dv">21</span><span class="op">:</span><span class="dv">22</span>), <span class="cf">function</span>(x) <span class="kw">cor</span>(ERcyt2[ ,x], ERcyt2<span class="op">$</span>grow, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span> ))</span>
<span id="cb361-8"><a href="read-in-er-io-data.html#cb361-8"></a>PVals=<span class="kw">sapply</span>(<span class="kw">c</span>(<span class="dv">6</span><span class="op">:</span><span class="dv">19</span>, <span class="dv">21</span><span class="op">:</span><span class="dv">22</span>), <span class="cf">function</span>(x) <span class="kw">cor.test</span>(ERcyt2[ ,x], ERcyt2<span class="op">$</span>grow, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>,<span class="dt">method=</span><span class="st">&quot;spearman&quot;</span> )<span class="op">$</span>p.value)</span>
<span id="cb361-9"><a href="read-in-er-io-data.html#cb361-9"></a></span>
<span id="cb361-10"><a href="read-in-er-io-data.html#cb361-10"></a><span class="kw">names</span>(CorVals)=<span class="kw">colnames</span>(ERcyt2)[<span class="kw">c</span>(<span class="dv">6</span><span class="op">:</span><span class="dv">19</span>, <span class="dv">21</span><span class="op">:</span><span class="dv">22</span>)]</span>
<span id="cb361-11"><a href="read-in-er-io-data.html#cb361-11"></a><span class="kw">names</span>(PVals)=<span class="kw">colnames</span>(ERcyt2)[<span class="kw">c</span>(<span class="dv">6</span><span class="op">:</span><span class="dv">19</span>, <span class="dv">21</span><span class="op">:</span><span class="dv">22</span>)]</span>
<span id="cb361-12"><a href="read-in-er-io-data.html#cb361-12"></a></span>
<span id="cb361-13"><a href="read-in-er-io-data.html#cb361-13"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb361-14"><a href="read-in-er-io-data.html#cb361-14"></a><span class="kw">plot</span>(ERcyt2<span class="op">$</span>grow, ERcyt2<span class="op">$</span>MICA, <span class="dt">xlab=</span><span class="st">&quot;Signature&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;MICA pre&quot;</span>, <span class="dt">pch=</span><span class="dv">19</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">300</span>))</span>
<span id="cb361-15"><a href="read-in-er-io-data.html#cb361-15"></a><span class="kw">text</span>(<span class="dv">1</span>,<span class="dv">250</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(CorVals[<span class="dv">9</span>],<span class="dv">2</span>), <span class="kw">round</span>(PVals[<span class="dv">9</span>],<span class="dv">3</span>) ))</span>
<span id="cb361-16"><a href="read-in-er-io-data.html#cb361-16"></a><span class="kw">plot</span>(ERcyt2<span class="op">$</span>grow, (ERcyt2<span class="op">$</span>MICA_post), <span class="dt">xlab=</span><span class="st">&quot;Signature&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;MICA post&quot;</span>, <span class="dt">pch=</span><span class="dv">19</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">300</span>))</span>
<span id="cb361-17"><a href="read-in-er-io-data.html#cb361-17"></a><span class="kw">text</span>(<span class="dv">1</span>,<span class="dv">250</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(CorVals[<span class="dv">15</span>],<span class="dv">2</span>), <span class="kw">round</span>(PVals[<span class="dv">15</span>],<span class="dv">3</span>) ))</span>
<span id="cb361-18"><a href="read-in-er-io-data.html#cb361-18"></a><span class="kw">plot</span>(ERcyt2<span class="op">$</span>grow, ERcyt2<span class="op">$</span>MICB, <span class="dt">xlab=</span><span class="st">&quot;Signature&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;MICB pre&quot;</span>, <span class="dt">pch=</span><span class="dv">19</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">300</span>))</span>
<span id="cb361-19"><a href="read-in-er-io-data.html#cb361-19"></a><span class="kw">text</span>(<span class="dv">1</span>,<span class="dv">250</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(CorVals[<span class="dv">10</span>],<span class="dv">2</span>), <span class="kw">round</span>(PVals[<span class="dv">10</span>],<span class="dv">3</span>) ))</span>
<span id="cb361-20"><a href="read-in-er-io-data.html#cb361-20"></a><span class="kw">plot</span>(ERcyt2<span class="op">$</span>grow, (ERcyt2<span class="op">$</span>MICB_post), <span class="dt">xlab=</span><span class="st">&quot;Signature&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;MICB post&quot;</span>, <span class="dt">pch=</span><span class="dv">19</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">300</span>))</span>
<span id="cb361-21"><a href="read-in-er-io-data.html#cb361-21"></a><span class="kw">text</span>(<span class="dv">1</span>,<span class="dv">250</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(CorVals[<span class="dv">16</span>],<span class="dv">2</span>), <span class="kw">round</span>(PVals[<span class="dv">16</span>],<span class="dv">3</span>) ))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-108-1.png" width="672" /></p>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb362-1"><a href="read-in-er-io-data.html#cb362-1"></a></span>
<span id="cb362-2"><a href="read-in-er-io-data.html#cb362-2"></a><span class="co">#text(1,250, sprintf(&quot;cor=%s, p=%s&quot;, round(CorVals[9],2), round(PVals[9],3) ))</span></span>
<span id="cb362-3"><a href="read-in-er-io-data.html#cb362-3"></a></span>
<span id="cb362-4"><a href="read-in-er-io-data.html#cb362-4"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb362-5"><a href="read-in-er-io-data.html#cb362-5"></a><span class="kw">plot</span>(ERcyt2<span class="op">$</span>grow, (ERcyt2<span class="op">$</span>MICB<span class="op">+</span>ERcyt2<span class="op">$</span>MICA), <span class="dt">xlab=</span><span class="st">&quot;Signature&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;MICA+MICB_pre&quot;</span>, <span class="dt">pch=</span><span class="dv">19</span>,</span>
<span id="cb362-6"><a href="read-in-er-io-data.html#cb362-6"></a>     <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">400</span>))</span>
<span id="cb362-7"><a href="read-in-er-io-data.html#cb362-7"></a><span class="kw">plot</span>(ERcyt2<span class="op">$</span>grow, (ERcyt2<span class="op">$</span>MICB_post<span class="op">+</span>ERcyt2<span class="op">$</span>MICA_post), <span class="dt">xlab=</span><span class="st">&quot;Signature&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;MICA+MICB_post&quot;</span>, <span class="dt">pch=</span><span class="dv">19</span>,</span>
<span id="cb362-8"><a href="read-in-er-io-data.html#cb362-8"></a>     <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">400</span>))</span>
<span id="cb362-9"><a href="read-in-er-io-data.html#cb362-9"></a><span class="kw">text</span>(<span class="dv">1</span>,<span class="dv">250</span>, <span class="kw">sprintf</span>(<span class="st">&quot;cor=%s, p=%s&quot;</span>, <span class="kw">round</span>(CorVals[<span class="dv">10</span>],<span class="dv">2</span>), <span class="kw">round</span>(PVals[<span class="dv">10</span>],<span class="dv">3</span>) ))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-108-2.png" width="672" /></p>
<div class="sourceCode" id="cb363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb363-1"><a href="read-in-er-io-data.html#cb363-1"></a></span>
<span id="cb363-2"><a href="read-in-er-io-data.html#cb363-2"></a><span class="co">#text(1,250, sprintf(&quot;cor=%s, p=%s&quot;, round(CorVals[9],2), round(PVals[9],3) ))</span></span>
<span id="cb363-3"><a href="read-in-er-io-data.html#cb363-3"></a></span>
<span id="cb363-4"><a href="read-in-er-io-data.html#cb363-4"></a></span>
<span id="cb363-5"><a href="read-in-er-io-data.html#cb363-5"></a><span class="co">## check for changes over time according to whether they are high or low?</span></span></code></pre></div>
</div>
<div id="use-raw-counts-and-perform-a-batch-correction-if-necessary" class="section level2">
<h2><span class="header-section-number">16.3</span> Use raw counts and perform a batch correction if necessary</h2>
<p>Note there are 4 samples whose profiles are out of the ordinary:</p>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="read-in-er-io-data.html#cb364-1"></a>ax1=<span class="kw">match</span>(<span class="kw">colnames</span>(ERcounts)[<span class="op">-</span><span class="dv">1</span>], <span class="kw">paste</span>(<span class="st">&quot;Patient&quot;</span>, ERClin<span class="op">$</span>Subject.ID, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb364-2"><a href="read-in-er-io-data.html#cb364-2"></a></span>
<span id="cb364-3"><a href="read-in-er-io-data.html#cb364-3"></a>colDat=ERClin[ax1, ]</span>
<span id="cb364-4"><a href="read-in-er-io-data.html#cb364-4"></a>colDat<span class="op">$</span>Sample=<span class="kw">paste</span>(<span class="st">&quot;Patient&quot;</span>, colDat<span class="op">$</span>Subject.ID, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb364-5"><a href="read-in-er-io-data.html#cb364-5"></a>ddER=<span class="kw">DESeqDataSetFromMatrix</span>(<span class="dt">countData=</span><span class="kw">round</span>(ERcounts[ ,<span class="op">-</span><span class="dv">1</span>]), <span class="dt">colDat=</span>colDat, <span class="dt">design=</span><span class="op">~</span>Arm)</span>
<span id="cb364-6"><a href="read-in-er-io-data.html#cb364-6"></a>ddER=<span class="kw">DESeq</span>(ddER)</span>
<span id="cb364-7"><a href="read-in-er-io-data.html#cb364-7"></a></span>
<span id="cb364-8"><a href="read-in-er-io-data.html#cb364-8"></a>ddERvst=<span class="kw">vst</span>(ddER)</span>
<span id="cb364-9"><a href="read-in-er-io-data.html#cb364-9"></a><span class="kw">plotPCA</span>(ddERvst,<span class="st">&quot;Arm&quot;</span>)<span class="op">+</span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">label =</span> name))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/tol-rawcounts-1.png" width="672" /></p>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="read-in-er-io-data.html#cb365-1"></a></span>
<span id="cb365-2"><a href="read-in-er-io-data.html#cb365-2"></a><span class="co">## Do a batch correction based on these 4 patients:</span></span>
<span id="cb365-3"><a href="read-in-er-io-data.html#cb365-3"></a></span>
<span id="cb365-4"><a href="read-in-er-io-data.html#cb365-4"></a>ddERvst<span class="op">$</span>Batch=<span class="dv">1</span></span>
<span id="cb365-5"><a href="read-in-er-io-data.html#cb365-5"></a>ddERvst<span class="op">$</span>Batch[<span class="kw">which</span>(ddERvst<span class="op">$</span>Subject.ID<span class="op">%in%</span><span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">82</span>, <span class="dv">41</span>, <span class="dv">38</span>))]=<span class="dv">2</span></span>
<span id="cb365-6"><a href="read-in-er-io-data.html#cb365-6"></a><span class="co">#ddERvst$Batch=factor(ddERvst$Batch)</span></span>
<span id="cb365-7"><a href="read-in-er-io-data.html#cb365-7"></a></span>
<span id="cb365-8"><a href="read-in-er-io-data.html#cb365-8"></a><span class="kw">assay</span>(ddERvst)=limma<span class="op">::</span><span class="kw">removeBatchEffect</span>(<span class="kw">assay</span>(ddERvst),ddERvst<span class="op">$</span>Batch)</span>
<span id="cb365-9"><a href="read-in-er-io-data.html#cb365-9"></a></span>
<span id="cb365-10"><a href="read-in-er-io-data.html#cb365-10"></a><span class="kw">plotPCA</span>(ddERvst,<span class="st">&quot;Arm&quot;</span>)<span class="op">+</span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">label =</span> name))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/tol-rawcounts-2.png" width="672" /></p>
<p>After correction, can we use these values in a ssGSEA</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="read-in-er-io-data.html#cb366-1"></a>ERssgsea=GSVA<span class="op">::</span><span class="kw">gsva</span>(<span class="kw">assay</span>(ddERvst), <span class="kw">list</span>(<span class="dt">grow=</span><span class="kw">na.omit</span>(HumGeneList2), </span>
<span id="cb366-2"><a href="read-in-er-io-data.html#cb366-2"></a>                                                <span class="dt">stab=</span>Rat2Hum<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(Glistnon, Rat2Hum<span class="op">$</span>RGD.symbol)],<span class="dt">Onco=</span>Oncoprint[[<span class="dv">1</span>]], <span class="dt">Mamma=</span>Oncoprint[[<span class="dv">2</span>]]), <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>F)</span></code></pre></div>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="read-in-er-io-data.html#cb367-1"></a>ax1=<span class="kw">match</span>(<span class="kw">colnames</span>(ERssgsea), <span class="kw">paste</span>(<span class="st">&quot;Patient&quot;</span>, ERClin<span class="op">$</span>Subject.ID, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb367-2"><a href="read-in-er-io-data.html#cb367-2"></a>ax3=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(ax1))</span>
<span id="cb367-3"><a href="read-in-er-io-data.html#cb367-3"></a>ssPS=<span class="kw">Surv</span>(ERClin<span class="op">$</span>PFS.Months[<span class="kw">na.omit</span>(ax1)], ERClin<span class="op">$</span>Progression[<span class="kw">na.omit</span>(ax1)])</span>
<span id="cb367-4"><a href="read-in-er-io-data.html#cb367-4"></a>ssOS=<span class="kw">Surv</span>(ERClin<span class="op">$</span>OS.Months[<span class="kw">na.omit</span>(ax1)], ERClin<span class="op">$</span>Death[<span class="kw">na.omit</span>(ax1)] )</span>
<span id="cb367-5"><a href="read-in-er-io-data.html#cb367-5"></a>ArmInfo=ERClin<span class="op">$</span>Arm[<span class="kw">na.omit</span>(ax1)]</span>
<span id="cb367-6"><a href="read-in-er-io-data.html#cb367-6"></a>PDL1info=ERClin<span class="op">$</span>PD.L1[<span class="kw">na.omit</span>(ax1)]</span>
<span id="cb367-7"><a href="read-in-er-io-data.html#cb367-7"></a>Purity=ERClin<span class="op">$</span>Purity[<span class="kw">na.omit</span>(ax1)]</span>
<span id="cb367-8"><a href="read-in-er-io-data.html#cb367-8"></a><span class="kw">plot</span>(ERssgsea[<span class="dv">2</span>, ax3]<span class="op">~</span>Purity)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-1.png" width="672" /></p>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="read-in-er-io-data.html#cb368-1"></a></span>
<span id="cb368-2"><a href="read-in-er-io-data.html#cb368-2"></a>TCGAvalCut=<span class="kw">cut</span>(ERssgsea[<span class="dv">1</span>,ax3 ], <span class="kw">quantile</span>(ERssgsea[<span class="dv">1</span>,ax3], <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>,<span class="fl">0.67</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;M&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb368-3"><a href="read-in-er-io-data.html#cb368-3"></a></span>
<span id="cb368-4"><a href="read-in-er-io-data.html#cb368-4"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum growing&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-2.png" width="672" /></p>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="read-in-er-io-data.html#cb369-1"></a></span>
<span id="cb369-2"><a href="read-in-er-io-data.html#cb369-2"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum growing&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-3.png" width="672" /></p>
<div class="sourceCode" id="cb370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb370-1"><a href="read-in-er-io-data.html#cb370-1"></a></span>
<span id="cb370-2"><a href="read-in-er-io-data.html#cb370-2"></a></span>
<span id="cb370-3"><a href="read-in-er-io-data.html#cb370-3"></a></span>
<span id="cb370-4"><a href="read-in-er-io-data.html#cb370-4"></a>TCGAvalCut=<span class="kw">cut</span>(ERssgsea[<span class="dv">2</span>,ax3 ], <span class="kw">quantile</span>(ERssgsea[<span class="dv">2</span>,ax3], <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.33</span>,<span class="fl">0.67</span>, <span class="dv">1</span>)), <span class="kw">c</span>(<span class="st">&quot;L&quot;</span>,<span class="st">&quot;M&quot;</span>, <span class="st">&quot;P&quot;</span>))</span>
<span id="cb370-5"><a href="read-in-er-io-data.html#cb370-5"></a></span>
<span id="cb370-6"><a href="read-in-er-io-data.html#cb370-6"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum stable&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-4.png" width="672" /></p>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="read-in-er-io-data.html#cb371-1"></a></span>
<span id="cb371-2"><a href="read-in-er-io-data.html#cb371-2"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS<span class="op">~</span>TCGAvalCut), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum stable&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-110-5.png" width="672" /></p>
<p>Run the above with Arm Stratification</p>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb372-1"><a href="read-in-er-io-data.html#cb372-1"></a>aarm=<span class="kw">which</span>(ArmInfo<span class="op">==</span><span class="st">&quot;A&quot;</span>)</span>
<span id="cb372-2"><a href="read-in-er-io-data.html#cb372-2"></a>barm=<span class="kw">which</span>(ArmInfo<span class="op">==</span><span class="st">&quot;B&quot;</span>)</span>
<span id="cb372-3"><a href="read-in-er-io-data.html#cb372-3"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb372-4"><a href="read-in-er-io-data.html#cb372-4"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[aarm]<span class="op">~</span>TCGAvalCut[aarm]), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum growing Arm A&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb372-5"><a href="read-in-er-io-data.html#cb372-5"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssPS[barm]<span class="op">~</span>TCGAvalCut[barm]), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum growing Arm B&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PS&quot;</span>, <span class="dt">mark.time=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-111-1.png" width="672" /></p>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="read-in-er-io-data.html#cb373-1"></a></span>
<span id="cb373-2"><a href="read-in-er-io-data.html#cb373-2"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[aarm]<span class="op">~</span>TCGAvalCut[aarm]), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum growing&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span>
<span id="cb373-3"><a href="read-in-er-io-data.html#cb373-3"></a>ax2=<span class="kw">plot</span>(<span class="kw">survfit</span>(ssOS[barm]<span class="op">~</span>TCGAvalCut[barm]), <span class="dt">main=</span><span class="kw">paste</span>(i, <span class="st">&quot;lum growing&quot;</span>), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>),<span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;Time (months)&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;OS&quot;</span>, <span class="dt">mark.time=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-111-2.png" width="672" /></p>
<p>Also run the above as a continuous variable</p>
<div class="sourceCode" id="cb374"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb374-1"><a href="read-in-er-io-data.html#cb374-1"></a><span class="co">## stable signature</span></span>
<span id="cb374-2"><a href="read-in-er-io-data.html#cb374-2"></a>ssPS=<span class="kw">Surv</span>( colDat<span class="op">$</span>PFS.Months, colDat<span class="op">$</span>Progression)</span>
<span id="cb374-3"><a href="read-in-er-io-data.html#cb374-3"></a>ssOS=<span class="kw">Surv</span>(colDat<span class="op">$</span>OS.Months, colDat<span class="op">$</span>Death )</span>
<span id="cb374-4"><a href="read-in-er-io-data.html#cb374-4"></a></span>
<span id="cb374-5"><a href="read-in-er-io-data.html#cb374-5"></a></span>
<span id="cb374-6"><a href="read-in-er-io-data.html#cb374-6"></a>dfs1=<span class="kw">coxph</span>(ssPS<span class="op">~</span><span class="st"> </span>ERssgsea[<span class="dv">1</span>, ax3])</span>
<span id="cb374-7"><a href="read-in-er-io-data.html#cb374-7"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb374-8"><a href="read-in-er-io-data.html#cb374-8"></a><span class="co">## Call:</span></span>
<span id="cb374-9"><a href="read-in-er-io-data.html#cb374-9"></a><span class="co">## coxph(formula = ssPS ~ ERssgsea[1, ax3])</span></span>
<span id="cb374-10"><a href="read-in-er-io-data.html#cb374-10"></a><span class="co">## </span></span>
<span id="cb374-11"><a href="read-in-er-io-data.html#cb374-11"></a><span class="co">##   n= 30, number of events= 28 </span></span>
<span id="cb374-12"><a href="read-in-er-io-data.html#cb374-12"></a><span class="co">## </span></span>
<span id="cb374-13"><a href="read-in-er-io-data.html#cb374-13"></a><span class="co">##                       coef exp(coef)  se(coef)     z Pr(&gt;|z|)</span></span>
<span id="cb374-14"><a href="read-in-er-io-data.html#cb374-14"></a><span class="co">## ERssgsea[1, ax3] 0.0003460 1.0003461 0.0004729 0.732    0.464</span></span>
<span id="cb374-15"><a href="read-in-er-io-data.html#cb374-15"></a><span class="co">## </span></span>
<span id="cb374-16"><a href="read-in-er-io-data.html#cb374-16"></a><span class="co">##                  exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb374-17"><a href="read-in-er-io-data.html#cb374-17"></a><span class="co">## ERssgsea[1, ax3]         1     0.9997    0.9994     1.001</span></span>
<span id="cb374-18"><a href="read-in-er-io-data.html#cb374-18"></a><span class="co">## </span></span>
<span id="cb374-19"><a href="read-in-er-io-data.html#cb374-19"></a><span class="co">## Concordance= 0.532  (se = 0.06 )</span></span>
<span id="cb374-20"><a href="read-in-er-io-data.html#cb374-20"></a><span class="co">## Likelihood ratio test= 0.54  on 1 df,   p=0.5</span></span>
<span id="cb374-21"><a href="read-in-er-io-data.html#cb374-21"></a><span class="co">## Wald test            = 0.54  on 1 df,   p=0.5</span></span>
<span id="cb374-22"><a href="read-in-er-io-data.html#cb374-22"></a><span class="co">## Score (logrank) test = 0.54  on 1 df,   p=0.5</span></span>
<span id="cb374-23"><a href="read-in-er-io-data.html#cb374-23"></a>dfs1=<span class="kw">coxph</span>(ssOS<span class="op">~</span><span class="st"> </span>ERssgsea[<span class="dv">1</span>, ax3])</span>
<span id="cb374-24"><a href="read-in-er-io-data.html#cb374-24"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb374-25"><a href="read-in-er-io-data.html#cb374-25"></a><span class="co">## Call:</span></span>
<span id="cb374-26"><a href="read-in-er-io-data.html#cb374-26"></a><span class="co">## coxph(formula = ssOS ~ ERssgsea[1, ax3])</span></span>
<span id="cb374-27"><a href="read-in-er-io-data.html#cb374-27"></a><span class="co">## </span></span>
<span id="cb374-28"><a href="read-in-er-io-data.html#cb374-28"></a><span class="co">##   n= 30, number of events= 22 </span></span>
<span id="cb374-29"><a href="read-in-er-io-data.html#cb374-29"></a><span class="co">## </span></span>
<span id="cb374-30"><a href="read-in-er-io-data.html#cb374-30"></a><span class="co">##                       coef exp(coef)  se(coef)    z Pr(&gt;|z|)</span></span>
<span id="cb374-31"><a href="read-in-er-io-data.html#cb374-31"></a><span class="co">## ERssgsea[1, ax3] 0.0004201 1.0004202 0.0005525 0.76    0.447</span></span>
<span id="cb374-32"><a href="read-in-er-io-data.html#cb374-32"></a><span class="co">## </span></span>
<span id="cb374-33"><a href="read-in-er-io-data.html#cb374-33"></a><span class="co">##                  exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb374-34"><a href="read-in-er-io-data.html#cb374-34"></a><span class="co">## ERssgsea[1, ax3]         1     0.9996    0.9993     1.002</span></span>
<span id="cb374-35"><a href="read-in-er-io-data.html#cb374-35"></a><span class="co">## </span></span>
<span id="cb374-36"><a href="read-in-er-io-data.html#cb374-36"></a><span class="co">## Concordance= 0.511  (se = 0.07 )</span></span>
<span id="cb374-37"><a href="read-in-er-io-data.html#cb374-37"></a><span class="co">## Likelihood ratio test= 0.59  on 1 df,   p=0.4</span></span>
<span id="cb374-38"><a href="read-in-er-io-data.html#cb374-38"></a><span class="co">## Wald test            = 0.58  on 1 df,   p=0.4</span></span>
<span id="cb374-39"><a href="read-in-er-io-data.html#cb374-39"></a><span class="co">## Score (logrank) test = 0.58  on 1 df,   p=0.4</span></span>
<span id="cb374-40"><a href="read-in-er-io-data.html#cb374-40"></a></span>
<span id="cb374-41"><a href="read-in-er-io-data.html#cb374-41"></a><span class="co">## growing signature</span></span>
<span id="cb374-42"><a href="read-in-er-io-data.html#cb374-42"></a>dfs1=<span class="kw">coxph</span>(ssPS<span class="op">~</span><span class="st"> </span>ERssgsea[<span class="dv">2</span>, ax3])</span>
<span id="cb374-43"><a href="read-in-er-io-data.html#cb374-43"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb374-44"><a href="read-in-er-io-data.html#cb374-44"></a><span class="co">## Call:</span></span>
<span id="cb374-45"><a href="read-in-er-io-data.html#cb374-45"></a><span class="co">## coxph(formula = ssPS ~ ERssgsea[2, ax3])</span></span>
<span id="cb374-46"><a href="read-in-er-io-data.html#cb374-46"></a><span class="co">## </span></span>
<span id="cb374-47"><a href="read-in-er-io-data.html#cb374-47"></a><span class="co">##   n= 30, number of events= 28 </span></span>
<span id="cb374-48"><a href="read-in-er-io-data.html#cb374-48"></a><span class="co">## </span></span>
<span id="cb374-49"><a href="read-in-er-io-data.html#cb374-49"></a><span class="co">##                        coef  exp(coef)   se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb374-50"><a href="read-in-er-io-data.html#cb374-50"></a><span class="co">## ERssgsea[2, ax3] -0.0002148  0.9997852  0.0001496 -1.436    0.151</span></span>
<span id="cb374-51"><a href="read-in-er-io-data.html#cb374-51"></a><span class="co">## </span></span>
<span id="cb374-52"><a href="read-in-er-io-data.html#cb374-52"></a><span class="co">##                  exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb374-53"><a href="read-in-er-io-data.html#cb374-53"></a><span class="co">## ERssgsea[2, ax3]    0.9998          1    0.9995         1</span></span>
<span id="cb374-54"><a href="read-in-er-io-data.html#cb374-54"></a><span class="co">## </span></span>
<span id="cb374-55"><a href="read-in-er-io-data.html#cb374-55"></a><span class="co">## Concordance= 0.597  (se = 0.07 )</span></span>
<span id="cb374-56"><a href="read-in-er-io-data.html#cb374-56"></a><span class="co">## Likelihood ratio test= 1.94  on 1 df,   p=0.2</span></span>
<span id="cb374-57"><a href="read-in-er-io-data.html#cb374-57"></a><span class="co">## Wald test            = 2.06  on 1 df,   p=0.2</span></span>
<span id="cb374-58"><a href="read-in-er-io-data.html#cb374-58"></a><span class="co">## Score (logrank) test = 2.09  on 1 df,   p=0.1</span></span>
<span id="cb374-59"><a href="read-in-er-io-data.html#cb374-59"></a>dfs1=<span class="kw">coxph</span>(ssOS<span class="op">~</span><span class="st"> </span>ERssgsea[<span class="dv">2</span>, ax3])</span>
<span id="cb374-60"><a href="read-in-er-io-data.html#cb374-60"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb374-61"><a href="read-in-er-io-data.html#cb374-61"></a><span class="co">## Call:</span></span>
<span id="cb374-62"><a href="read-in-er-io-data.html#cb374-62"></a><span class="co">## coxph(formula = ssOS ~ ERssgsea[2, ax3])</span></span>
<span id="cb374-63"><a href="read-in-er-io-data.html#cb374-63"></a><span class="co">## </span></span>
<span id="cb374-64"><a href="read-in-er-io-data.html#cb374-64"></a><span class="co">##   n= 30, number of events= 22 </span></span>
<span id="cb374-65"><a href="read-in-er-io-data.html#cb374-65"></a><span class="co">## </span></span>
<span id="cb374-66"><a href="read-in-er-io-data.html#cb374-66"></a><span class="co">##                        coef  exp(coef)   se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb374-67"><a href="read-in-er-io-data.html#cb374-67"></a><span class="co">## ERssgsea[2, ax3] -0.0001429  0.9998571  0.0001779 -0.803    0.422</span></span>
<span id="cb374-68"><a href="read-in-er-io-data.html#cb374-68"></a><span class="co">## </span></span>
<span id="cb374-69"><a href="read-in-er-io-data.html#cb374-69"></a><span class="co">##                  exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb374-70"><a href="read-in-er-io-data.html#cb374-70"></a><span class="co">## ERssgsea[2, ax3]    0.9999          1    0.9995         1</span></span>
<span id="cb374-71"><a href="read-in-er-io-data.html#cb374-71"></a><span class="co">## </span></span>
<span id="cb374-72"><a href="read-in-er-io-data.html#cb374-72"></a><span class="co">## Concordance= 0.58  (se = 0.074 )</span></span>
<span id="cb374-73"><a href="read-in-er-io-data.html#cb374-73"></a><span class="co">## Likelihood ratio test= 0.61  on 1 df,   p=0.4</span></span>
<span id="cb374-74"><a href="read-in-er-io-data.html#cb374-74"></a><span class="co">## Wald test            = 0.65  on 1 df,   p=0.4</span></span>
<span id="cb374-75"><a href="read-in-er-io-data.html#cb374-75"></a><span class="co">## Score (logrank) test = 0.65  on 1 df,   p=0.4</span></span>
<span id="cb374-76"><a href="read-in-er-io-data.html#cb374-76"></a></span>
<span id="cb374-77"><a href="read-in-er-io-data.html#cb374-77"></a><span class="co">## combined signature</span></span>
<span id="cb374-78"><a href="read-in-er-io-data.html#cb374-78"></a>dfs1=<span class="kw">coxph</span>(ssPS<span class="op">~</span>(ERssgsea[<span class="dv">2</span>, ax3]<span class="op">-</span>ERssgsea[<span class="dv">1</span>, ax3] ))</span>
<span id="cb374-79"><a href="read-in-er-io-data.html#cb374-79"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb374-80"><a href="read-in-er-io-data.html#cb374-80"></a><span class="co">## Call:</span></span>
<span id="cb374-81"><a href="read-in-er-io-data.html#cb374-81"></a><span class="co">## coxph(formula = ssPS ~ (ERssgsea[2, ax3] - ERssgsea[1, ax3]))</span></span>
<span id="cb374-82"><a href="read-in-er-io-data.html#cb374-82"></a><span class="co">## </span></span>
<span id="cb374-83"><a href="read-in-er-io-data.html#cb374-83"></a><span class="co">##   n= 30, number of events= 28 </span></span>
<span id="cb374-84"><a href="read-in-er-io-data.html#cb374-84"></a><span class="co">## </span></span>
<span id="cb374-85"><a href="read-in-er-io-data.html#cb374-85"></a><span class="co">##                        coef  exp(coef)   se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb374-86"><a href="read-in-er-io-data.html#cb374-86"></a><span class="co">## ERssgsea[2, ax3] -0.0002148  0.9997852  0.0001496 -1.436    0.151</span></span>
<span id="cb374-87"><a href="read-in-er-io-data.html#cb374-87"></a><span class="co">## </span></span>
<span id="cb374-88"><a href="read-in-er-io-data.html#cb374-88"></a><span class="co">##                  exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb374-89"><a href="read-in-er-io-data.html#cb374-89"></a><span class="co">## ERssgsea[2, ax3]    0.9998          1    0.9995         1</span></span>
<span id="cb374-90"><a href="read-in-er-io-data.html#cb374-90"></a><span class="co">## </span></span>
<span id="cb374-91"><a href="read-in-er-io-data.html#cb374-91"></a><span class="co">## Concordance= 0.597  (se = 0.07 )</span></span>
<span id="cb374-92"><a href="read-in-er-io-data.html#cb374-92"></a><span class="co">## Likelihood ratio test= 1.94  on 1 df,   p=0.2</span></span>
<span id="cb374-93"><a href="read-in-er-io-data.html#cb374-93"></a><span class="co">## Wald test            = 2.06  on 1 df,   p=0.2</span></span>
<span id="cb374-94"><a href="read-in-er-io-data.html#cb374-94"></a><span class="co">## Score (logrank) test = 2.09  on 1 df,   p=0.1</span></span>
<span id="cb374-95"><a href="read-in-er-io-data.html#cb374-95"></a>dfs1=<span class="kw">coxph</span>(ssOS<span class="op">~</span><span class="st"> </span>(ERssgsea[<span class="dv">2</span>, ax3]<span class="op">-</span>ERssgsea[<span class="dv">1</span>, ax3] ))</span>
<span id="cb374-96"><a href="read-in-er-io-data.html#cb374-96"></a><span class="kw">summary</span>(dfs1)</span>
<span id="cb374-97"><a href="read-in-er-io-data.html#cb374-97"></a><span class="co">## Call:</span></span>
<span id="cb374-98"><a href="read-in-er-io-data.html#cb374-98"></a><span class="co">## coxph(formula = ssOS ~ (ERssgsea[2, ax3] - ERssgsea[1, ax3]))</span></span>
<span id="cb374-99"><a href="read-in-er-io-data.html#cb374-99"></a><span class="co">## </span></span>
<span id="cb374-100"><a href="read-in-er-io-data.html#cb374-100"></a><span class="co">##   n= 30, number of events= 22 </span></span>
<span id="cb374-101"><a href="read-in-er-io-data.html#cb374-101"></a><span class="co">## </span></span>
<span id="cb374-102"><a href="read-in-er-io-data.html#cb374-102"></a><span class="co">##                        coef  exp(coef)   se(coef)      z Pr(&gt;|z|)</span></span>
<span id="cb374-103"><a href="read-in-er-io-data.html#cb374-103"></a><span class="co">## ERssgsea[2, ax3] -0.0001429  0.9998571  0.0001779 -0.803    0.422</span></span>
<span id="cb374-104"><a href="read-in-er-io-data.html#cb374-104"></a><span class="co">## </span></span>
<span id="cb374-105"><a href="read-in-er-io-data.html#cb374-105"></a><span class="co">##                  exp(coef) exp(-coef) lower .95 upper .95</span></span>
<span id="cb374-106"><a href="read-in-er-io-data.html#cb374-106"></a><span class="co">## ERssgsea[2, ax3]    0.9999          1    0.9995         1</span></span>
<span id="cb374-107"><a href="read-in-er-io-data.html#cb374-107"></a><span class="co">## </span></span>
<span id="cb374-108"><a href="read-in-er-io-data.html#cb374-108"></a><span class="co">## Concordance= 0.58  (se = 0.074 )</span></span>
<span id="cb374-109"><a href="read-in-er-io-data.html#cb374-109"></a><span class="co">## Likelihood ratio test= 0.61  on 1 df,   p=0.4</span></span>
<span id="cb374-110"><a href="read-in-er-io-data.html#cb374-110"></a><span class="co">## Wald test            = 0.65  on 1 df,   p=0.4</span></span>
<span id="cb374-111"><a href="read-in-er-io-data.html#cb374-111"></a><span class="co">## Score (logrank) test = 0.65  on 1 df,   p=0.4</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tcga-associate-epcam-inflammatory-with-survival.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="deseq-analysis-characterisation-cohort-big-vs-small.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
