<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Summary of GSEA runs | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Summary of GSEA runs | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Summary of GSEA runs | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="deseq-analysis-immunotherapygrowth-comparisons.html"/>
<link rel="next" href="epcam-inflammatory-vs-non-inflammatory-samples.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redo-the-survival-plots-based-on-luminal-cases-only"><i class="fa fa-check"></i><b>15.4.3</b> Redo the survival plots based on Luminal cases only</a></li>
<li class="chapter" data-level="15.4.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.4</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>16</b> METABRIC: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="16.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="16.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="16.2.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#all-samples-1"><i class="fa fa-check"></i><b>16.2.1</b> All samples</a></li>
<li class="chapter" data-level="16.2.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases-1"><i class="fa fa-check"></i><b>16.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.4" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint-1"><i class="fa fa-check"></i><b>16.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="16.4.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#overall-survival-1"><i class="fa fa-check"></i><b>16.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="16.4.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#met-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>16.4.2</b> MET: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>21</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="21.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>21.1</b> Data</a><ul>
<li class="chapter" data-level="21.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>21.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>21.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="21.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>21.3</b> Plots</a><ul>
<li class="chapter" data-level="21.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>21.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="21.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>21.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>21.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="21.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>21.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>21.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="21.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>21.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>21.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>22</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>22.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>22.2</b> Load files</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>22.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="22.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>22.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="22.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>22.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>23</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>23.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>23.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>23.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>24</b> Trichrome staining</a><ul>
<li class="chapter" data-level="24.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>24.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="24.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>24.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="24.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>24.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="24.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>24.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>25</b> CNV calling</a><ul>
<li class="chapter" data-level="25.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>25.1</b> Sample data</a></li>
<li class="chapter" data-level="25.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>25.2</b> Data Summary</a><ul>
<li class="chapter" data-level="25.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>25.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="25.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>25.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="25.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>25.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>25.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="25.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>25.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>26</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="summary-of-gsea-runs" class="section level1">
<h1><span class="header-section-number">Chapter 13</span> Summary of GSEA runs</h1>
<div id="stable-vs-growing-all-samples" class="section level2">
<h2><span class="header-section-number">13.1</span> Stable vs growing: all samples</h2>
<p>Below is a summary of growing vs stable comparisons using Process Networks:</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="summary-of-gsea-runs.html#cb247-1"></a>d1=DNismr3<span class="op">$</span>ProcessNetworks[, <span class="kw">c</span>(<span class="st">&quot;stable_vs_growing NES&quot;</span>, <span class="st">&quot;stable_vs_growing padj&quot;</span>)]</span>
<span id="cb247-2"><a href="summary-of-gsea-runs.html#cb247-2"></a>e1=Epismr3<span class="op">$</span>ProcessNetworks[, <span class="kw">c</span>(<span class="st">&quot;stable_vs_growing NES&quot;</span>, <span class="st">&quot;stable_vs_growing padj&quot;</span>)]</span>
<span id="cb247-3"><a href="summary-of-gsea-runs.html#cb247-3"></a>c1=CDismr3<span class="op">$</span>ProcessNetworks[, <span class="kw">c</span>(<span class="st">&quot;stable_vs_growing NES&quot;</span>, <span class="st">&quot;stable_vs_growing padj&quot;</span>)]</span>
<span id="cb247-4"><a href="summary-of-gsea-runs.html#cb247-4"></a><span class="co">#</span></span>
<span id="cb247-5"><a href="summary-of-gsea-runs.html#cb247-5"></a>ds=<span class="kw">which</span>(d1[ ,<span class="dv">2</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb247-6"><a href="summary-of-gsea-runs.html#cb247-6"></a>es=<span class="kw">which</span>(e1[ ,<span class="dv">2</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb247-7"><a href="summary-of-gsea-runs.html#cb247-7"></a>cs=<span class="kw">which</span>(c1[ ,<span class="dv">2</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb247-8"><a href="summary-of-gsea-runs.html#cb247-8"></a></span>
<span id="cb247-9"><a href="summary-of-gsea-runs.html#cb247-9"></a>AUnique=<span class="kw">c</span>(<span class="kw">rownames</span>(d1)[ds], <span class="kw">rownames</span>(e1)[es], <span class="kw">rownames</span>(c1)[cs])</span>
<span id="cb247-10"><a href="summary-of-gsea-runs.html#cb247-10"></a>xalist=<span class="kw">unique</span>(AUnique)</span>
<span id="cb247-11"><a href="summary-of-gsea-runs.html#cb247-11"></a></span>
<span id="cb247-12"><a href="summary-of-gsea-runs.html#cb247-12"></a>tx=<span class="kw">cbind</span>(<span class="kw">as.numeric</span>(d1[<span class="kw">match</span>(xalist, <span class="kw">rownames</span>(d1)), <span class="dv">1</span>]),<span class="kw">as.numeric</span>(e1[<span class="kw">match</span>(xalist, <span class="kw">rownames</span>(e1)), <span class="dv">1</span>]),</span>
<span id="cb247-13"><a href="summary-of-gsea-runs.html#cb247-13"></a>         <span class="kw">as.numeric</span>(c1[<span class="kw">match</span>(xalist, <span class="kw">rownames</span>(c1)), <span class="dv">1</span>]))</span>
<span id="cb247-14"><a href="summary-of-gsea-runs.html#cb247-14"></a>tx2=<span class="kw">cbind</span>(<span class="kw">as.numeric</span>(d1[<span class="kw">match</span>(xalist, <span class="kw">rownames</span>(d1)), <span class="dv">2</span>]),<span class="kw">as.numeric</span>(e1[<span class="kw">match</span>(xalist, <span class="kw">rownames</span>(e1)), <span class="dv">2</span>]),</span>
<span id="cb247-15"><a href="summary-of-gsea-runs.html#cb247-15"></a>         <span class="kw">as.numeric</span>(c1[<span class="kw">match</span>(xalist, <span class="kw">rownames</span>(c1)), <span class="dv">2</span>]))</span>
<span id="cb247-16"><a href="summary-of-gsea-runs.html#cb247-16"></a>tx[<span class="kw">which</span>(tx2<span class="op">&gt;</span><span class="fl">0.1</span>, <span class="dt">arr.ind =</span> T)]=<span class="dv">0</span></span>
<span id="cb247-17"><a href="summary-of-gsea-runs.html#cb247-17"></a><span class="kw">rownames</span>(tx)=<span class="kw">substr</span>(xalist, <span class="dv">2</span>, <span class="dv">100</span>)</span>
<span id="cb247-18"><a href="summary-of-gsea-runs.html#cb247-18"></a>txb=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(tx), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb247-19"><a href="summary-of-gsea-runs.html#cb247-19"></a>rCol=<span class="kw">c</span>(<span class="kw">brewer.pal</span>(<span class="dv">12</span>, <span class="st">&quot;Set3&quot;</span>), <span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Set2&quot;</span>))[<span class="kw">factor</span>(txb)]</span>
<span id="cb247-20"><a href="summary-of-gsea-runs.html#cb247-20"></a><span class="kw">colnames</span>(tx)=<span class="kw">c</span>(<span class="st">&quot;DN&quot;</span>, <span class="st">&quot;Ep&quot;</span>, <span class="st">&quot;CD&quot;</span>)</span>
<span id="cb247-21"><a href="summary-of-gsea-runs.html#cb247-21"></a></span>
<span id="cb247-22"><a href="summary-of-gsea-runs.html#cb247-22"></a></span>
<span id="cb247-23"><a href="summary-of-gsea-runs.html#cb247-23"></a><span class="co">#pdf(&quot;~/Desktop/5A-summary-pathways-process-networks.pdf&quot;, height=9, width=4)</span></span>
<span id="cb247-24"><a href="summary-of-gsea-runs.html#cb247-24"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">5</span>))</span>
<span id="cb247-25"><a href="summary-of-gsea-runs.html#cb247-25"></a>ax=<span class="kw">heatmap.2</span>(tx, <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">scale =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">RowSideColors =</span> rCol)</span></code></pre></div>
<div class="figure"><span id="fig:5a"></span>
<img src="RatDCIS-bookdown_files/figure-html/5a-1.png" alt="GSEA summary stable growing" width="672" />
<p class="caption">
Figure 13.1: GSEA summary stable growing
</p>
</div>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="summary-of-gsea-runs.html#cb248-1"></a><span class="co">## re-order this?</span></span>
<span id="cb248-2"><a href="summary-of-gsea-runs.html#cb248-2"></a>axb=ax<span class="op">$</span>carpet</span>
<span id="cb248-3"><a href="summary-of-gsea-runs.html#cb248-3"></a>an2=<span class="kw">c</span>(<span class="kw">brewer.pal</span>(<span class="dv">12</span>, <span class="st">&quot;Set3&quot;</span>), <span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Set2&quot;</span>))[<span class="kw">factor</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">colnames</span>(axb), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]))]</span>
<span id="cb248-4"><a href="summary-of-gsea-runs.html#cb248-4"></a><span class="kw">colnames</span>(axb)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">colnames</span>(axb), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="kw">length</span>(x)])</span>
<span id="cb248-5"><a href="summary-of-gsea-runs.html#cb248-5"></a>rOrd=<span class="kw">order</span>(an2)</span></code></pre></div>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="summary-of-gsea-runs.html#cb249-1"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">3</span>))</span>
<span id="cb249-2"><a href="summary-of-gsea-runs.html#cb249-2"></a><span class="kw">heatmap.2</span>(<span class="kw">t</span>(axb[, rOrd]), <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">trace =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">scale =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">RowSideColors =</span> an2[rOrd],</span>
<span id="cb249-3"><a href="summary-of-gsea-runs.html#cb249-3"></a>             <span class="dt">Rowv =</span> <span class="ot">NA</span>, <span class="dt">main=</span><span class="st">&quot;stable red, growing blue&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/5a-part2-1.png" width="672" /></p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="summary-of-gsea-runs.html#cb250-1"></a><span class="co">#dev.off()</span></span>
<span id="cb250-2"><a href="summary-of-gsea-runs.html#cb250-2"></a></span>
<span id="cb250-3"><a href="summary-of-gsea-runs.html#cb250-3"></a>d1<span class="op">$</span>names=<span class="kw">rownames</span>(d1)</span>
<span id="cb250-4"><a href="summary-of-gsea-runs.html#cb250-4"></a>e1<span class="op">$</span>names=<span class="kw">rownames</span>(e1)</span>
<span id="cb250-5"><a href="summary-of-gsea-runs.html#cb250-5"></a>c1<span class="op">$</span>names=<span class="kw">rownames</span>(c1)</span>
<span id="cb250-6"><a href="summary-of-gsea-runs.html#cb250-6"></a></span>
<span id="cb250-7"><a href="summary-of-gsea-runs.html#cb250-7"></a>All1=<span class="kw">merge</span>(d1, e1, <span class="dt">by.x=</span><span class="st">&quot;names&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;names&quot;</span>, <span class="dt">all=</span>T)</span>
<span id="cb250-8"><a href="summary-of-gsea-runs.html#cb250-8"></a>All2=<span class="kw">merge</span>(All1, c1, <span class="dt">by.x=</span><span class="st">&quot;names&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;names&quot;</span>, <span class="dt">all=</span>T)</span>
<span id="cb250-9"><a href="summary-of-gsea-runs.html#cb250-9"></a></span>
<span id="cb250-10"><a href="summary-of-gsea-runs.html#cb250-10"></a><span class="co">#write.csv(tx, file=&quot;nature-tables/5a.csv&quot;)</span></span>
<span id="cb250-11"><a href="summary-of-gsea-runs.html#cb250-11"></a><span class="co">#write.csv(t(axb[, rOrd]), file=&quot;nature-tables/5a_version2.csv&quot;)</span></span>
<span id="cb250-12"><a href="summary-of-gsea-runs.html#cb250-12"></a></span>
<span id="cb250-13"><a href="summary-of-gsea-runs.html#cb250-13"></a>DT<span class="op">::</span><span class="kw">datatable</span>(All2, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb250-14"><a href="summary-of-gsea-runs.html#cb250-14"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div id="htmlwidget-dc09926437696db6522e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-dc09926437696db6522e">{"x":{"filter":"none","extensions":["Buttons"],"data":[["NApoptosis_Anti-Apoptosis mediated by external signals by Estrogen","NApoptosis_Anti-Apoptosis mediated by external signals via MAPK and JAK/STAT","NApoptosis_Anti-apoptosis mediated by external signals via NF-kB","NApoptosis_Anti-Apoptosis mediated by external signals via PI3K/AKT","NApoptosis_Apoptosis stimulation by external signals","NApoptosis_Apoptotic mitochondria","NApoptosis_Apoptotic nucleus","NApoptosis_Death Domain receptors &amp; caspases in apoptosis","NApoptosis_Endoplasmic reticulum stress pathway","NAutophagy_Autophagy","NBlood coagulation","NCardiac development_BMP_TGF_beta_signaling","NCardiac development_FGF_ErbB signaling","NCardiac development_Role of NADPH oxidase and ROS","NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","NCell adhesion_Amyloid proteins","NCell adhesion_Attractive and repulsive receptors","NCell adhesion_Cadherins","NCell adhesion_Cell junctions","NCell adhesion_Cell-matrix interactions","NCell adhesion_Glycoconjugates","NCell adhesion_Integrin priming","NCell adhesion_Integrin-mediated cell-matrix adhesion","NCell adhesion_Leucocyte chemotaxis","NCell adhesion_Platelet aggregation","NCell adhesion_Platelet-endothelium-leucocyte interactions","NCell adhesion_Synaptic contact","NCell cycle_Core","NCell cycle_G0-G1","NCell cycle_G1-S","NCell cycle_G1-S Growth factor regulation","NCell cycle_G1-S Interleukin regulation","NCell cycle_G2-M","NCell cycle_Meiosis","NCell cycle_Mitosis","NCell cycle_S phase","NChemotaxis","NCytoskeleton_Actin filaments","NCytoskeleton_Cytoplasmic microtubules","NCytoskeleton_Intermediate filaments","NCytoskeleton_Macropinocytosis and its regulation","NCytoskeleton_Regulation of cytoskeleton rearrangement","NCytoskeleton_Spindle microtubules","NDevelopment_Blood vessel morphogenesis","NDevelopment_Cartilage development","NDevelopment_EMT_Regulation of epithelial-to-mesenchymal transition","NDevelopment_ERK5 in cell proliferation and neuronal survival","NDevelopment_Hedgehog signaling","NDevelopment_Hemopoiesis, Erythropoietin pathway","NDevelopment_Keratinocyte differentiation","NDevelopment_Melanocyte development and pigmentation","NDevelopment_Neurogenesis in general","NDevelopment_Neurogenesis_Axonal guidance","NDevelopment_Neurogenesis_Synaptogenesis","NDevelopment_Neuromuscular junction","NDevelopment_Ossification and bone remodeling","NDevelopment_Regulation of angiogenesis","NDevelopment_Regulation of telomere length","NDevelopment_Skeletal muscle development","NDNA damage_BER-NER repair","NDNA damage_Checkpoint","NDNA damage_Core","NDNA damage_DBS repair","NDNA damage_MMR repair","NImmune response_Antigen presentation","NImmune response_BCR pathway","NImmune response_IL-5 signalling","NImmune response_Innate immune response to RNA viral infection ","NImmune response_Phagocytosis","NImmune response_Phagosome in antigen presentation","NImmune response_T helper cell differentiation","NImmune response_TCR signaling","NImmune response_Th17-derived cytokines","NInflammation_Amphoterin signaling","NInflammation_Complement system","NInflammation_Histamine signaling","NInflammation_IFN-gamma signaling","NInflammation_IgE signaling","NInflammation_IL-10 anti-inflammatory response","NInflammation_IL-12,15,18 signaling","NInflammation_IL-13 signaling pathway","NInflammation_IL-2 signaling","NInflammation_IL-4 signaling","NInflammation_IL-6 signaling","NInflammation_Inflammasome","NInflammation_Innate inflammatory response","NInflammation_Interferon signaling","NInflammation_Jak-STAT Pathway","NInflammation_Kallikrein-kinin system","NInflammation_MIF signaling","NInflammation_Neutrophil activation","NInflammation_NK cell cytotoxicity","NInflammation_Protein C signaling","NInflammation_TREM1 signaling","NMuscle contraction","NMuscle contraction_Nitric oxide signaling in the cardiovascular system","NMuscle contraction_Relaxin signaling","NNeurophysiological process_Circadian rhythm","NNeurophysiological process_Corticoliberin signaling","NNeurophysiological process_GABAergic neurotransmission","NNeurophysiological process_Long-term potentiation","NNeurophysiological process_Melatonin signaling","NNeurophysiological process_Olfactory transduction","NNeurophysiological process_Taste signaling","NNeurophysiological process_Transmission of nerve impulse","NNeurophysiological process_Visual perception","NProliferation_Lymphocyte proliferation","NProliferation_Negative regulation of cell proliferation","NProliferation_Positive regulation cell proliferation","NProtein folding_ER and cytoplasm","NProtein folding_Folding in normal condition","NProtein folding_Protein folding nucleus","NProtein folding_Response to unfolded proteins","NProteolysis_Connective tissue degradation","NProteolysis_ECM remodeling","NProteolysis_Proteolysis in cell cycle and apoptosis","NProteolysis_Ubiquitin-proteasomal proteolysis","NRegulation of metabolism_Bile acid regulation of lipid metabolism and negative FXR-dependent regulation of bile acids concentration","NReproduction_Feeding and Neurohormone signaling ","NReproduction_FSH-beta signaling pathway","NReproduction_GnRH signaling pathway","NReproduction_Gonadotropin regulation","NReproduction_Male sex differentiation","NReproduction_Progesterone signaling","NReproduction_Spermatogenesis, motility and copulation","NResponse to hypoxia and oxidative stress","NSignal transduction_Androgen receptor nuclear signaling","NSignal transduction_Androgen receptor signaling cross-talk","NSignal Transduction_BMP and GDF signaling","NSignal Transduction_Cholecystokinin signaling","NSignal transduction_CREM pathway","NSignal transduction_ERBB-family signaling","NSignal transduction_ESR1-membrane pathway","NSignal transduction_ESR1-nuclear pathway","NSignal transduction_ESR2 pathway","NSignal transduction_Leptin signaling","NSignal transduction_Neuropeptide signaling pathways","NSignal transduction_Nitric oxide signaling","NSignal transduction_NOTCH signaling","NSignal transduction_Oxytocin signaling","NSignal Transduction_TGF-beta, GDF and Activin signaling","NSignal transduction_WNT signaling","NTranscription_Chromatin modification","NTranscription_mRNA processing","NTranscription_Nuclear receptors transcriptional regulation","NTranscription_Transcription by RNA polymerase II","NTranslation_Elongation-Termination","NTranslation_Regulation of initiation","NTranslation_Selenium pathway","NTranslation_Translation in mitochondria","NTranslation_Translation initiation","NTransport_Bile acids transport and its regulation","NTransport_Calcium transport","NTransport_Iron transport","NTransport_Manganese transport","NTransport_Potassium transport","NTransport_Sodium transport","NTransport_Synaptic vesicle exocytosis"],["0.342701370816744","0.244768171398732","-0.277260126901354","0.342474305086012","-0.20100526210413","-0.322069060486035","0.214930435383964","-0.206983748921366","-0.407801877274196","-0.272441506021126","-0.382540765458028","0.337325477639144","0.35071212257729","0.284501931271123","0.484306635960086","0.423961685549047","0.415322979744444","0.338846792056112","0.247577478212875","0.591614930275162","0.440964520874551","0.482105963197433","0.452439661113409","0.408206805133742","0.508495932402177","0.400442564615767","0.396968204939023","0.508006659962784","0.341265690818258","0.292198173506561","0.224331603960863","-0.212074187329427","0.378732728653489","0.286987306026065","0.413306133496529","0.383516201626154","0.323931877429014","0.326537582217484","0.340142283456951","0.42226355535511","0.286995835307007","0.336947733708031","0.516577548198649","0.368492230839879","0.562653167648982","0.364056542749382","-0.298643902229295","0.379384707635492","0.219324470337693","-0.31502382720926","0.290888981435832","0.442732934662483","0.440814177171832","0.484432171274152","0.317231935045806","0.510868519523954","0.316857497342947","0.251523287610626","0.422414946515266","0.22819193577727","0.367482080928676","0.462242658119237","0.297265752940861","0.44018031385048","-0.316628744215021","0.304461724436753","-0.272991926697561","-0.395257984886608","0.369696708969038","0.261112470957872","0.3092839269121","0.29723972148332","-0.367273053986391","-0.282148472675152","-0.399781632085438","0.348530751316429","-0.323352391915133","0.248009317601086","0.265340667139783","-0.356200241996444","0.358627280840961","0.35394960287804","0.254859777454004","-0.353501522813405","0.224642918797878","-0.411220081326443","-0.445574323669613","0.314875483067819","-0.265686196980944","-0.256763081290742","0.319260341024646","-0.26269548251201","-0.415080031307766","0.266343391541865","0.469518062943953","0.377641786597668","0.361648894932412","0.416063924635079","0.376462420438417","0.539805316522534","0.539233565277572","0.525588449560341","0.720573995978247","0.310908656354208","0.593825903390097","0.504304152145973","0.317097728367522","0.248402515371018","0.348584026607013","-0.271801625123427","-0.22547471862837","-0.237285318579998","-0.218321294796416","0.473959470776795","0.476401756566381","0.280677275597585","-0.195752481715909","0.353440923568465","0.33999545590761","0.355596835753907","0.357268585693488","0.344093686701366","0.278699646891003","0.308640440744193","0.23210881373334","0.220502088460749","-0.202364813620883","-0.248535981462751","0.448108249034398","0.262865846976022","0.256013508238258","-0.354937414701889","0.313083462696232","0.304122685323166","-0.207080660093666","-0.269848276084348","0.557358859371123","0.402343398386583","0.32210512819928","0.339411126459212","0.273271303388246","0.353366950449233","0.183449780456405","0.14374560563998","0.253291515900373","-0.0801324800070686","-0.556728277868432","-0.193830060114297","0.454493692801296","-0.262994225746303","-0.56135504809729","0.603200407618248","0.577324560429211","0.345548374247149","0.331037693972814","0.501943882440293","0.451142162545184","0.434508695290337"],["1","0.422099262240108","0","0.957348094312516","0.489766536964981","0.260196382428941","0.422099262240108","0.422099262240108","0","0.422099262240108","0.164106910039114","0.945481701458643","0.933635459068295","0.708811499703616","0","0.566854312091871","0.540859393650036","0.965405585660692","0.489766536964981","0","0.518383526383526","0.260196382428941","0.260196382428941","0.576784003332639","0","0.760344057780696","0.78501933391543","0","0.895962985560301","0.717821499857428","0.457847832676569","0.317452711223203","0.717821499857428","0.832530505188733","0.566854312091871","0.77260358056266","0.903283165553673","0.932197741159045","0.9364166150031","0.665868738634485","0.760344057780696","0.972288602185882","0","0.746298342541436","0.164106910039114","0.855132588209511","0.489766536964981","0.728574972436604","0.377279880104909","0.377279880104909","0.746298342541436","0.377279880104909","0","0.164106910039114","0.869192921881744","0","0.933635459068295","0.717821499857428","0.649595733700327","0.540859393650036","0.882922733552981","0.78501933391543","0.844009834599911","0.717821499857428","0","0.89092953055064","0.540859393650036","0","0.855132588209511","0.317452711223203","0.869192921881744","0.78501933391543","0","0.164106910039114","0.260196382428941","0.920793146597363","0","0.566854312091871","0.728574972436604","0.260196382428941","0.972288602185882","0.972288602185882","0.708811499703616","0","0.566854312091871","0","0","0.920793146597363","0","0.317452711223203","0.895962985560301","0.164106910039114","0","0.717821499857428","0.317452711223203","0.895962985560301","0.98634735327043","0.903283165553673","0.957348094312516","0","0.317452711223203","0.67972998729352","0.422099262240108","0.832530505188733","0","0.566854312091871","0.882922733552981","0.422099262240108","0.980248235580433","0.540859393650036","0.260196382428941","0.596069455406472","0.566854312091871","0.489766536964981","0.457847832676569","0.821912313432836","0.164106910039114","0.965405585660692","0.972288602185882","0.920793146597363","0.933635459068295","0.945481701458643","0.649595733700327","0.78501933391543","0.377279880104909","0.576784003332639","0.317452711223203","0.518383526383526","0.377279880104909","0.566854312091871","0.540859393650036","0.164106910039114","0.797033609352168","0.832530505188733","0.489766536964981","0","0.317452711223203","0.855132588209511","0.915058453679143","0.965405585660692","0.708811499703616","0.9364166150031","0.540859393650036","0.317452711223203","0.518383526383526","0.317452711223203","0","0.317452711223203","0.635477821853588","0.260196382428941","0","0.457847832676569","0","0.957348094312516","0.980248235580433","0","0.422099262240108","0.596069455406472"],["-0.374399316227365","-0.38031380506899","-0.459314591994637","-0.439704426819946","-0.332404743064469","-0.351804562386705","-0.228967205350103","-0.370235793909332","-0.47493938973049","-0.301692674103381","-0.544537246143589","-0.256681768430063","0.20336024638004","-0.253878100272956","0.299968984467324","-0.256115953699279","-0.336842077468743","-0.202083975830769","-0.207567037538786","-0.288378725746302","-0.391656317753942","-0.234132128882176","-0.309356088889694","-0.636251323652007","-0.364099265254971","-0.516684890863345","-0.239600394070808","0.426965656420817","0.251144981685086","0.139632668986477","-0.303226132189443","-0.355119861334711","0.258531859032321","0.175264404441031","0.272254465341549","0.378874710535532","-0.652373212172324","0.211628961599915","0.236991554983505","0.268852363632407","-0.392042577595253","-0.260351398321027","0.32619447075985","-0.337201285270924","-0.355571435926554","-0.313553697868285","-0.389158026321165","-0.228414648310209","-0.280425010271949","-0.284491920750904","-0.408205404565124","0.273486728077857","-0.211436167559781","0.299021511460313","0.258393544567449","-0.249306009849955","-0.425349982313245","0.311207682286501","0.359790325038881","0.299615041475118","0.347253321273874","0.437581755580504","0.35837121823182","0.30926842801443","-0.635838374213216","-0.476465445036621","-0.499504395905393","-0.43622021719289","-0.471871907997477","-0.408733303695207","-0.566024082866224","-0.576007348997439","-0.501184840878011","-0.491434228831937","-0.701071073390931","-0.410991826895193","-0.509617287933754","-0.335858307845243","-0.412297395161556","-0.470595670699412","-0.34974652531427","-0.36235683767651","-0.655472523740724","-0.292162500960726","-0.371670560020933","-0.465695065338363","-0.634846775793164","-0.585368573722467","-0.470720375121043","-0.350369611626511","-0.469920620547141","-0.560334882420046","-0.536714423468605","-0.486571783041969","0.393129348129591","-0.269766618599271","-0.407641595941718","0.232638726402751","-0.359366480709357","0.303422409263745","0.37157504203319","0.328089422010473","0.350500300063681","0.620350970287194","0.258776778064746","0.488295821920071","-0.58515137992335","-0.39208555027691","-0.406544848510004","-0.323945738902414","0.231432032936301","0.179300251853571","0.211095576585718","-0.369507771458287","-0.436952634458893","-0.301954774180341","-0.128933006173044","-0.273963391411919","-0.263251602312031","-0.181053035178952","-0.370510975181783","-0.352475745280138","-0.233679371282223","-0.252617027421288","0.172490579355971","-0.246875562687827","0.205268759864952","-0.302111416321096","0.234642175190612","-0.167636264179991","-0.250751354052029","-0.379317424828321","0.238913452622402","0.181732079696151","-0.276050272652264","-0.389550560332163","0.225542653702482","-0.220682880248606","-0.252512057297522","-0.440079733369622","-0.260467775933053","-0.219554986518706","0.37696573164159","0.331453063766358","0.183621211838654","0.340579043375023","0.452811704436146","-0.156521985083071","-0.373801029335518","0.486907677282816","0.446147058846501","-0.337251674336304","0.34703556349274","-0.261443725655118","-0.250932734593445","0.408703534512452","-0.373249954111633","0.266292608065294"],["0.419863780102165","0","0","0","0.34661231374754","0.38528125","0.870003207526997","0.256053997923157","0","0.792078082530269","0","0.801355320173327","0.832103487064117","0.88036543106164","0.38528125","0.682836923076923","0","0.947029296177206","0.940803052325581","0.38528125","0","0.90610579247655","0.201454248366013","0","0.121288735858338","0","0.88036543106164","0.121288735858338","0.857389981527763","0.987448512585812","0.514136780650542","0.256053997923157","0.601938971229294","0.889905135079398","0.419863780102165","0","0","0.857389981527763","0.772171189979123","0.698110564744449","0.201454248366013","0.698110564744449","0.301996325780772","0","0.482648977816442","0.121288735858338","0.644548276425421","0.921794392523364","0.627749490835031","0.843427121262275","0.627749490835031","0.38528125","0.967392750533049","0.38528125","0.801355320173327","0.744545704090036","0","0.627749490835031","0.256053997923157","0.419863780102165","0.201454248366013","0.564148713060057","0.121288735858338","0.682836923076923","0","0","0.121288735858338","0","0","0","0","0","0","0","0","0","0","0.121288735858338","0.121288735858338","0.201454248366013","0.514136780650542","0.34661231374754","0","0.58882072005878","0.121288735858338","0","0","0","0","0.121288735858338","0","0","0","0","0","0.698110564744449","0.301996325780772","0.930918968692449","0.514136780650542","0.564148713060057","0.301996325780772","0.792078082530269","0.889905135079398","0.301996325780772","0.644548276425421","0.256053997923157","0","0","0","0.832103487064117","0.801355320173327","0.930918968692449","0.95397168732964","0.256053997923157","0.121288735858338","0.514136780650542","0.947029296177206","0.930918968692449","0.627749490835031","0.911960170697013","0.121288735858338","0.34661231374754","0.857389981527763","0.698110564744449","0.974788459920935","0.898332020480504","0.898332020480504","0.663164056320203","0.843427121262275","0.980681083768332","0.832103487064117","0.34661231374754","0.822689829728486","0.921794392523364","0.730349788434415","0.121288735858338","0.898332020480504","0.911960170697013","0.759317177667693","0.34661231374754","0.744545704090036","0.987448512585812","0.121288735858338","0.121288735858338","0.940803052325581","0.256053997923157","0","0.947029296177206","0.482648977816442","0","0","0.744545704090036","0.201454248366013","0.889905135079398","0.843427121262275","0","0","0.541717595366487"],["0.273152042710106","-0.186774717375572","0.217751179738003","-0.242651059877777","-0.173318225352886","0.331580657257232","-0.238430938800026","0.192051073777702","-0.285498832532368","0.236760775654154","0.505028371777109","-0.301657295814902","0.30876136043813","0.19179149733185","-0.223151836622505","-0.253572169925781","-0.297572271997599","-0.323442632296983","0.290061345619443","-0.307654780234483","-0.258839102167479","-0.245748422988895","0.260698544597352","0.266498337080347","0.343116831570985","0.340329931863947","-0.380359652803175","0.315538359026554","0.212210115132119","0.132988658908179","-0.200931057199172","-0.246878376527588","0.269166741348827","-0.248321385210275","0.246874418766659","0.186039022993659","0.398375796285583","0.214177502039818","0.204995761945659","0.357387915686557","0.257473928324319","-0.219360721245428","0.270850429133416","-0.325358592417609","-0.452706609798234","0.280184405150434","-0.525809187580189","-0.274407408639972","-0.220734589981602","0.224500898200329","-0.279865668037243","-0.30110719548983","-0.248849278845265","-0.327478932673988","-0.212591046928758","0.292647709966624","-0.279322835130041","-0.191982110961706","-0.360442177145789","0.140035822864498","-0.145743291675215","0.200958575830209","-0.154314796858993","-0.126907608441674","0.255629861395644","-0.282700829556085","-0.207001180025128","0.252422971156661","0.448957736959684","0.354466606606613","-0.313987044916534","-0.369176077873089","0.29693350940394","-0.195279133620342","0.698543749111893","0.304689020318336","-0.239912131091777","0.199866200927937","-0.218055857982301","-0.359907885819906","-0.300054271333541","-0.284804218820822","-0.220832328915464","-0.187812089988221","0.275289785020069","0.33260152890138","0.290312663421536","-0.310566857682617","0.378218559964479","0.294929116390093","0.319632689866756","-0.26528459526788","-0.257171509850313","0.300580014282393","-0.440380975460353","-0.450815680667951","-0.329274878454016","-0.433424390599795","-0.312778986947914","-0.304618505922093","-0.308614928464702","-0.395553074157692","-0.388283165819849","0.380880490564595","-0.502090355017561","-0.506601247909127","-0.324277569346851","0.291513161755498","0.259328009292836","-0.213518459599633","-0.145099234114229","-0.22203849500863","-0.301146303423849","0.382808879166988","0.412649004175353","0.217812044690142","0.196371480109379","0.40472466156186","0.216420130117687","-0.227722089507925","0.248040895284055","-0.279178604475797","-0.250490695664748","0.249515103201815","-0.33294921358048","0.356893200389636","-0.215230450230639","-0.244412275877814","0.334255872818485","-0.298502309526937","-0.310377111921185","-0.301753951359308","0.351309315393631","0.222047486185759","-0.185993837859004","-0.386515952389725","-0.456688564413801","-0.245186587562044","-0.259651575099212","0.36255832922778","-0.224812761863042","-0.217212864711803","-0.315271356029947","-0.211743448934176","-0.255174404834401","-0.237735930346109","0.136738467433281","-0.182691194476246","0.383876421123762","0.589875869698421","-0.13749367107557","0.542329978815811","-0.410515604360266","0.329986245996615","0.358457354886283","-0.375789633932309","-0.425879025477733","-0.260053939421056"],["0.869186046511628","0.988954560421468","0.934437843291111","0.926628085508445","0.988954560421468","0.486468253968254","0.934437843291111","1","0.849395993372496","0.981677734375","0","0.813470471134705","0.602038060159607","0.988954560421468","0.957330097087379","0.754690265486726","0.602038060159607","0.486468253968254","0.542233689642463","0.64768972142171","0.934437843291111","0.987870299309286","0.673725669796199","0.774732342007435","0.30571072319202","0.437235844850647","0.186590563165906","0.602038060159607","0.988954560421468","0.988954560421468","1","0.882797396482482","0.754690265486726","0.981677734375","0.741023574450937","0.963478800413651","0","0.969680559780955","0.987870299309286","0.30571072319202","0.859526730937774","0.930206944125086","0.882797396482482","0.486468253968254","0.186590563165906","0.718701842546064","0.378558929490479","0.741023574450937","0.988954560421468","0.988954560421468","0.957330097087379","0.572086666666667","0.869186046511628","0.542233689642463","0.988954560421468","0.718701842546064","0.64768972142171","0.988954560421468","0.186590563165906","0.981677734375","0.988954560421468","0.988954560421468","0.988954560421468","0.988954560421468","0.813470471134705","0.813470471134705","0.973805977559329","0.957330097087379","0","0.186590563165906","0.618967741935484","0","0.782489361702128","0.988954560421468","0","0.642336913806654","0.969680559780955","0.988954560421468","0.988954560421468","0.618967741935484","0.869186046511628","0.832401234567901","0.934437843291111","0.988954560421468","0.882797396482482","0.378558929490479","0.799317089910776","0.642336913806654","0.378558929490479","0.618967741935484","0.486468253968254","0.799317089910776","0.934437843291111","0.486468253968254","0","0.186590563165906","0.799317089910776","0.572086666666667","0.882797396482482","0.799317089910776","0.869186046511628","0.572086666666667","0.903500798427712","0.947250482936252","0","0.186590563165906","0.30571072319202","0.718701842546064","0.64768972142171","0.973805977559329","0.988954560421468","0.988954560421468","0.859526730937774","0.378558929490479","0.378558929490479","0.987870299309286","0.988954560421468","0.673725669796199","0.947250482936252","0.988954560421468","0.930206944125086","0.718701842546064","0.859526730937774","0.699745334796926","0.486468253968254","0.186590563165906","0.988954560421468","0.973805977559329","0.602038060159607","0.826582115909456","0.754690265486726","0.832401234567901","0.30571072319202","0.942378651438612","0.988954560421468","0.378558929490479","0","0.973805977559329","0.782489361702128","0.64768972142171","0.988954560421468","0.973805977559329","0.618967741935484","0.988954560421468","0.826582115909456","0.957330097087379","0.988954560421468","0.988954560421468","0.572086666666667","0","0.988954560421468","0.378558929490479","0","0.486468253968254","0.30571072319202","0","0","0.930206944125086"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>names<\/th>\n      <th>stable_vs_growing NES.x<\/th>\n      <th>stable_vs_growing padj.x<\/th>\n      <th>stable_vs_growing NES.y<\/th>\n      <th>stable_vs_growing padj.y<\/th>\n      <th>stable_vs_growing NES<\/th>\n      <th>stable_vs_growing padj<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="barplots-of-enriched-pathways" class="section level3">
<h3><span class="header-section-number">13.1.1</span> barplots of enriched pathways</h3>
<p>Below are the bar-plots for the enriched pathways in CD45 cells:</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="summary-of-gsea-runs.html#cb251-1"></a>termA=c1[cs, ]</span>
<span id="cb251-2"><a href="summary-of-gsea-runs.html#cb251-2"></a></span>
<span id="cb251-3"><a href="summary-of-gsea-runs.html#cb251-3"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb251-4"><a href="summary-of-gsea-runs.html#cb251-4"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(termA)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb251-5"><a href="summary-of-gsea-runs.html#cb251-5"></a></span>
<span id="cb251-6"><a href="summary-of-gsea-runs.html#cb251-6"></a>sigTerms=TermsA</span>
<span id="cb251-7"><a href="summary-of-gsea-runs.html#cb251-7"></a>TermType=<span class="kw">substr</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]), <span class="dv">2</span>, <span class="dv">40</span>)</span>
<span id="cb251-8"><a href="summary-of-gsea-runs.html#cb251-8"></a></span>
<span id="cb251-9"><a href="summary-of-gsea-runs.html#cb251-9"></a><span class="co">## check whether this runs:</span></span>
<span id="cb251-10"><a href="summary-of-gsea-runs.html#cb251-10"></a></span>
<span id="cb251-11"><a href="summary-of-gsea-runs.html#cb251-11"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Apoptosis&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb251-12"><a href="summary-of-gsea-runs.html#cb251-12"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Proliferation&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb251-13"><a href="summary-of-gsea-runs.html#cb251-13"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Translation&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb251-14"><a href="summary-of-gsea-runs.html#cb251-14"></a>testType=<span class="kw">c</span>(<span class="st">&quot;Inflammation&quot;</span>, <span class="st">&quot;Immune response&quot;</span>, <span class="st">&quot;Transcription&quot;</span>, <span class="st">&quot;Transport&quot;</span>)</span>
<span id="cb251-15"><a href="summary-of-gsea-runs.html#cb251-15"></a></span>
<span id="cb251-16"><a href="summary-of-gsea-runs.html#cb251-16"></a><span class="kw">pdf</span>(<span class="st">&quot;~/Desktop/4H-growing(-ve)-vs-stable(+ve)-CD45.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">8</span>, <span class="dt">width=</span><span class="dv">8</span>)</span>
<span id="cb251-17"><a href="summary-of-gsea-runs.html#cb251-17"></a><span class="co">#par(oma=c(2, 3, 0,0), mfrow=c(2,2))</span></span>
<span id="cb251-18"><a href="summary-of-gsea-runs.html#cb251-18"></a><span class="co">#for (i in testType){</span></span>
<span id="cb251-19"><a href="summary-of-gsea-runs.html#cb251-19"></a><span class="co">#  barplot(as.numeric(termA[which(TermType==i), 1]), names.arg = sigTerms[which(TermType==i)], horiz = T, las=2,</span></span>
<span id="cb251-20"><a href="summary-of-gsea-runs.html#cb251-20"></a><span class="co">#          xlab=&quot;NES&quot;, main=i, col=factor(sign(as.numeric(termA[which(TermType==i), 1]))))</span></span>
<span id="cb251-21"><a href="summary-of-gsea-runs.html#cb251-21"></a><span class="co">#}</span></span>
<span id="cb251-22"><a href="summary-of-gsea-runs.html#cb251-22"></a></span>
<span id="cb251-23"><a href="summary-of-gsea-runs.html#cb251-23"></a>a1=<span class="kw">order</span>(termA[ ,<span class="dv">1</span>])</span>
<span id="cb251-24"><a href="summary-of-gsea-runs.html#cb251-24"></a><span class="kw">barplot</span>(<span class="kw">as.numeric</span>(termA[a1, <span class="dv">1</span>]), <span class="dt">names.arg =</span> sigTerms[a1], <span class="dt">horiz =</span> T, <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;NES&quot;</span>, <span class="dt">main=</span><span class="st">&quot;GSEA rslt grow-stable&quot;</span>, <span class="dt">col=</span><span class="kw">factor</span>(<span class="kw">sign</span>(<span class="kw">as.numeric</span>(termA[a1, <span class="dv">1</span>]))))</span>
<span id="cb251-25"><a href="summary-of-gsea-runs.html#cb251-25"></a><span class="kw">dev.off</span>()</span>
<span id="cb251-26"><a href="summary-of-gsea-runs.html#cb251-26"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb251-27"><a href="summary-of-gsea-runs.html#cb251-27"></a><span class="co">##                 2</span></span>
<span id="cb251-28"><a href="summary-of-gsea-runs.html#cb251-28"></a>Ntab=<span class="kw">data.frame</span>(<span class="dt">TermType=</span>TermType, <span class="dt">sigTerm=</span>sigTerms, termA)</span>
<span id="cb251-29"><a href="summary-of-gsea-runs.html#cb251-29"></a><span class="kw">write.csv</span>(Ntab, <span class="dt">file=</span><span class="st">&quot;nature-tables/4h.csv&quot;</span>)</span>
<span id="cb251-30"><a href="summary-of-gsea-runs.html#cb251-30"></a></span>
<span id="cb251-31"><a href="summary-of-gsea-runs.html#cb251-31"></a>DT<span class="op">::</span><span class="kw">datatable</span>(Ntab, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb251-32"><a href="summary-of-gsea-runs.html#cb251-32"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:4h"></span>
<div id="htmlwidget-c41570729ea20f24ea59" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c41570729ea20f24ea59">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Chemotaxis","Immune response","Signal transduction","Blood coagulation","Muscle contraction","Transport","Inflammation","Immune response","Transport","Neurophysiological process","Transport","Transcription"],["Chemotaxis","TCR signaling","Neuropeptide signaling pathways","Blood coagulation","Muscle contraction","Calcium transport","Complement system","Phagocytosis","Potassium transport","Transmission of nerve impulse","Sodium transport","Translation in mitochondria"],["0.398375796285583","-0.369176077873089","-0.456688564413801","0.505028371777109","-0.440380975460353","-0.410515604360266","0.698543749111893","0.448957736959684","-0.375789633932309","-0.502090355017561","-0.425879025477733","0.589875869698421"],["0","0","0","0","0","0","0","0","0","0","0","0"],["NChemotaxis","NImmune response_TCR signaling","NSignal transduction_Neuropeptide signaling pathways","NBlood coagulation","NMuscle contraction","NTransport_Calcium transport","NInflammation_Complement system","NImmune response_Phagocytosis","NTransport_Potassium transport","NNeurophysiological process_Transmission of nerve impulse","NTransport_Sodium transport","NTranslation_Translation in mitochondria"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>TermType<\/th>\n      <th>sigTerm<\/th>\n      <th>stable_vs_growing.NES<\/th>\n      <th>stable_vs_growing.padj<\/th>\n      <th>names<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 13.2: CD45 cells enriched pathways
</p>
</div>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="summary-of-gsea-runs.html#cb252-1"></a>termA=d1[ds, ]</span>
<span id="cb252-2"><a href="summary-of-gsea-runs.html#cb252-2"></a></span>
<span id="cb252-3"><a href="summary-of-gsea-runs.html#cb252-3"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb252-4"><a href="summary-of-gsea-runs.html#cb252-4"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(termA)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb252-5"><a href="summary-of-gsea-runs.html#cb252-5"></a></span>
<span id="cb252-6"><a href="summary-of-gsea-runs.html#cb252-6"></a>sigTerms=TermsA</span>
<span id="cb252-7"><a href="summary-of-gsea-runs.html#cb252-7"></a>TermType=<span class="kw">substr</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]), <span class="dv">2</span>, <span class="dv">40</span>)</span>
<span id="cb252-8"><a href="summary-of-gsea-runs.html#cb252-8"></a></span>
<span id="cb252-9"><a href="summary-of-gsea-runs.html#cb252-9"></a><span class="co">## check whether this runs:</span></span>
<span id="cb252-10"><a href="summary-of-gsea-runs.html#cb252-10"></a></span>
<span id="cb252-11"><a href="summary-of-gsea-runs.html#cb252-11"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Apoptosis&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb252-12"><a href="summary-of-gsea-runs.html#cb252-12"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Proliferation&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb252-13"><a href="summary-of-gsea-runs.html#cb252-13"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Translation&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb252-14"><a href="summary-of-gsea-runs.html#cb252-14"></a>testType=<span class="kw">c</span>(<span class="st">&quot;Inflammation&quot;</span>, <span class="st">&quot;Immune response&quot;</span>, <span class="st">&quot;Transcription&quot;</span>, <span class="st">&quot;Transport&quot;</span>)</span>
<span id="cb252-15"><a href="summary-of-gsea-runs.html#cb252-15"></a></span>
<span id="cb252-16"><a href="summary-of-gsea-runs.html#cb252-16"></a><span class="kw">pdf</span>(<span class="st">&quot;~/Desktop/4H-growing(-ve)-vs-stable(+ve)-DN.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">8</span>, <span class="dt">width=</span><span class="dv">8</span>)</span>
<span id="cb252-17"><a href="summary-of-gsea-runs.html#cb252-17"></a><span class="co">#par(oma=c(2, 3, 0,0), mfrow=c(2,2))</span></span>
<span id="cb252-18"><a href="summary-of-gsea-runs.html#cb252-18"></a><span class="co">#for (i in testType){</span></span>
<span id="cb252-19"><a href="summary-of-gsea-runs.html#cb252-19"></a><span class="co">#  barplot(as.numeric(termA[which(TermType==i), 1]), names.arg = sigTerms[which(TermType==i)], horiz = T, las=2,</span></span>
<span id="cb252-20"><a href="summary-of-gsea-runs.html#cb252-20"></a><span class="co">#          xlab=&quot;NES&quot;, main=i, col=factor(sign(as.numeric(termA[which(TermType==i), 1]))))</span></span>
<span id="cb252-21"><a href="summary-of-gsea-runs.html#cb252-21"></a><span class="co">#}</span></span>
<span id="cb252-22"><a href="summary-of-gsea-runs.html#cb252-22"></a></span>
<span id="cb252-23"><a href="summary-of-gsea-runs.html#cb252-23"></a>a1=<span class="kw">order</span>(termA[ ,<span class="dv">1</span>])</span>
<span id="cb252-24"><a href="summary-of-gsea-runs.html#cb252-24"></a><span class="kw">barplot</span>(<span class="kw">as.numeric</span>(termA[a1, <span class="dv">1</span>]), <span class="dt">names.arg =</span> sigTerms[a1], <span class="dt">horiz =</span> T, <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;NES&quot;</span>, <span class="dt">main=</span><span class="st">&quot;GSEA rslt grow-stable&quot;</span>, <span class="dt">col=</span><span class="kw">factor</span>(<span class="kw">sign</span>(<span class="kw">as.numeric</span>(termA[a1, <span class="dv">1</span>]))))</span>
<span id="cb252-25"><a href="summary-of-gsea-runs.html#cb252-25"></a><span class="kw">dev.off</span>()</span>
<span id="cb252-26"><a href="summary-of-gsea-runs.html#cb252-26"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb252-27"><a href="summary-of-gsea-runs.html#cb252-27"></a><span class="co">##                 2</span></span>
<span id="cb252-28"><a href="summary-of-gsea-runs.html#cb252-28"></a>Ntab=<span class="kw">data.frame</span>(<span class="dt">TermType=</span>TermType, <span class="dt">sigTerm=</span>sigTerms, termA)</span>
<span id="cb252-29"><a href="summary-of-gsea-runs.html#cb252-29"></a><span class="kw">write.csv</span>(Ntab, <span class="dt">file=</span><span class="st">&quot;nature-tables/4hDN.csv&quot;</span>)</span>
<span id="cb252-30"><a href="summary-of-gsea-runs.html#cb252-30"></a></span>
<span id="cb252-31"><a href="summary-of-gsea-runs.html#cb252-31"></a>DT<span class="op">::</span><span class="kw">datatable</span>(Ntab, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb252-32"><a href="summary-of-gsea-runs.html#cb252-32"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:4hDN"></span>
<div id="htmlwidget-053ae7f1e47d591f6d69" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-053ae7f1e47d591f6d69">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Cell adhesion","Immune response","Inflammation","Development","Cytoskeleton","Transcription","Development","Inflammation","Cell cycle","Cardiac development","Transcription","Inflammation","Transport","Transcription","Immune response","Inflammation","Transport","Neurophysiological process","Inflammation","Immune response","Signal transduction","Cell adhesion","Inflammation","Neurophysiological process","Transcription"],["Cell-matrix interactions","Antigen presentation","Interferon signaling","Ossification and bone remodeling","Spindle microtubules","Translation initiation","Neurogenesis","IL-6 signaling","Core","Wnt","Anti-apoptosis mediated by external signals via NF-kB","IFN-gamma signaling","Calcium transport","Endoplasmic reticulum stress pathway","Th17-derived cytokines","Protein C signaling","Potassium transport","Transmission of nerve impulse","Innate inflammatory response","Innate immune response to RNA viral infection ","Leptin signaling","Platelet aggregation","Kallikrein-kinin system","GABAergic neurotransmission","Elongation-Termination"],["0.591614930275162","-0.316628744215021","-0.445574323669613","0.510868519523954","0.516577548198649","-0.56135504809729","0.440814177171832","-0.353501522813405","0.508006659962784","0.484306635960086","-0.277260126901354","-0.323352391915133","0.577324560429211","-0.407801877274196","-0.367273053986391","-0.415080031307766","0.501943882440293","0.593825903390097","-0.411220081326443","-0.395257984886608","-0.269848276084348","0.508495932402177","-0.265686196980944","0.539805316522534","-0.556728277868432"],["0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0"],["NCell adhesion_Cell-matrix interactions","NImmune response_Antigen presentation","NInflammation_Interferon signaling","NDevelopment_Ossification and bone remodeling","NCytoskeleton_Spindle microtubules","NTranslation_Translation initiation","NDevelopment_Neurogenesis_Axonal guidance","NInflammation_IL-6 signaling","NCell cycle_Core","NCardiac development_Wnt_beta-catenin, Notch, VEGF, IP3 and integrin signaling","NApoptosis_Anti-apoptosis mediated by external signals via NF-kB","NInflammation_IFN-gamma signaling","NTransport_Calcium transport","NApoptosis_Endoplasmic reticulum stress pathway","NImmune response_Th17-derived cytokines","NInflammation_Protein C signaling","NTransport_Potassium transport","NNeurophysiological process_Transmission of nerve impulse","NInflammation_Innate inflammatory response","NImmune response_Innate immune response to RNA viral infection ","NSignal transduction_Leptin signaling","NCell adhesion_Platelet aggregation","NInflammation_Kallikrein-kinin system","NNeurophysiological process_GABAergic neurotransmission","NTranslation_Elongation-Termination"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>TermType<\/th>\n      <th>sigTerm<\/th>\n      <th>stable_vs_growing.NES<\/th>\n      <th>stable_vs_growing.padj<\/th>\n      <th>names<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 13.3: DN cells enriched pathways
</p>
</div>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="summary-of-gsea-runs.html#cb253-1"></a>termA=e1[es, ]</span>
<span id="cb253-2"><a href="summary-of-gsea-runs.html#cb253-2"></a></span>
<span id="cb253-3"><a href="summary-of-gsea-runs.html#cb253-3"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb253-4"><a href="summary-of-gsea-runs.html#cb253-4"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(termA)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb253-5"><a href="summary-of-gsea-runs.html#cb253-5"></a></span>
<span id="cb253-6"><a href="summary-of-gsea-runs.html#cb253-6"></a>sigTerms=TermsA</span>
<span id="cb253-7"><a href="summary-of-gsea-runs.html#cb253-7"></a>TermType=<span class="kw">substr</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]), <span class="dv">2</span>, <span class="dv">40</span>)</span>
<span id="cb253-8"><a href="summary-of-gsea-runs.html#cb253-8"></a></span>
<span id="cb253-9"><a href="summary-of-gsea-runs.html#cb253-9"></a><span class="co">## check whether this runs:</span></span>
<span id="cb253-10"><a href="summary-of-gsea-runs.html#cb253-10"></a></span>
<span id="cb253-11"><a href="summary-of-gsea-runs.html#cb253-11"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Apoptosis&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb253-12"><a href="summary-of-gsea-runs.html#cb253-12"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Proliferation&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb253-13"><a href="summary-of-gsea-runs.html#cb253-13"></a>TermType[<span class="kw">which</span>(TermType<span class="op">==</span><span class="st">&quot;Translation&quot;</span>)]=<span class="st">&quot;Transcription&quot;</span></span>
<span id="cb253-14"><a href="summary-of-gsea-runs.html#cb253-14"></a>testType=<span class="kw">c</span>(<span class="st">&quot;Inflammation&quot;</span>, <span class="st">&quot;Immune response&quot;</span>, <span class="st">&quot;Transcription&quot;</span>, <span class="st">&quot;Transport&quot;</span>)</span>
<span id="cb253-15"><a href="summary-of-gsea-runs.html#cb253-15"></a></span>
<span id="cb253-16"><a href="summary-of-gsea-runs.html#cb253-16"></a><span class="kw">pdf</span>(<span class="st">&quot;~/Desktop/4H-growing(-ve)-vs-stable(+ve)-EP.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">8</span>, <span class="dt">width=</span><span class="dv">8</span>)</span>
<span id="cb253-17"><a href="summary-of-gsea-runs.html#cb253-17"></a><span class="co">#par(oma=c(2, 3, 0,0), mfrow=c(2,2))</span></span>
<span id="cb253-18"><a href="summary-of-gsea-runs.html#cb253-18"></a><span class="co">#for (i in testType){</span></span>
<span id="cb253-19"><a href="summary-of-gsea-runs.html#cb253-19"></a><span class="co">#  barplot(as.numeric(termA[which(TermType==i), 1]), names.arg = sigTerms[which(TermType==i)], horiz = T, las=2,</span></span>
<span id="cb253-20"><a href="summary-of-gsea-runs.html#cb253-20"></a><span class="co">#          xlab=&quot;NES&quot;, main=i, col=factor(sign(as.numeric(termA[which(TermType==i), 1]))))</span></span>
<span id="cb253-21"><a href="summary-of-gsea-runs.html#cb253-21"></a><span class="co">#}</span></span>
<span id="cb253-22"><a href="summary-of-gsea-runs.html#cb253-22"></a></span>
<span id="cb253-23"><a href="summary-of-gsea-runs.html#cb253-23"></a>a1=<span class="kw">order</span>(termA[ ,<span class="dv">1</span>])</span>
<span id="cb253-24"><a href="summary-of-gsea-runs.html#cb253-24"></a><span class="kw">barplot</span>(<span class="kw">as.numeric</span>(termA[a1, <span class="dv">1</span>]), <span class="dt">names.arg =</span> sigTerms[a1], <span class="dt">horiz =</span> T, <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">xlab=</span><span class="st">&quot;NES&quot;</span>, <span class="dt">main=</span><span class="st">&quot;GSEA rslt grow-stable&quot;</span>, <span class="dt">col=</span><span class="kw">factor</span>(<span class="kw">sign</span>(<span class="kw">as.numeric</span>(termA[a1, <span class="dv">1</span>]))))</span>
<span id="cb253-25"><a href="summary-of-gsea-runs.html#cb253-25"></a><span class="kw">dev.off</span>()</span>
<span id="cb253-26"><a href="summary-of-gsea-runs.html#cb253-26"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb253-27"><a href="summary-of-gsea-runs.html#cb253-27"></a><span class="co">##                 2</span></span>
<span id="cb253-28"><a href="summary-of-gsea-runs.html#cb253-28"></a>Ntab=<span class="kw">data.frame</span>(<span class="dt">TermType=</span>TermType, <span class="dt">sigTerm=</span>sigTerms, termA)</span>
<span id="cb253-29"><a href="summary-of-gsea-runs.html#cb253-29"></a><span class="kw">write.csv</span>(Ntab, <span class="dt">file=</span><span class="st">&quot;nature-tables/4hEP.csv&quot;</span>)</span>
<span id="cb253-30"><a href="summary-of-gsea-runs.html#cb253-30"></a></span>
<span id="cb253-31"><a href="summary-of-gsea-runs.html#cb253-31"></a>DT<span class="op">::</span><span class="kw">datatable</span>(Ntab, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb253-32"><a href="summary-of-gsea-runs.html#cb253-32"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:4hEp"></span>
<div id="htmlwidget-410ded99ed2daf2b1da3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-410ded99ed2daf2b1da3">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Transcription","Inflammation","Immune response","Cell adhesion","Inflammation","Cell cycle","Chemotaxis","Cell adhesion","Transcription","Transcription","Inflammation","Transcription","Development","Development","Immune response","Inflammation","Transcription","Immune response","Cell adhesion","Blood coagulation","Transcription","Muscle contraction","Inflammation","Inflammation","Transcription","Transcription","Immune response","Inflammation","Inflammation","Inflammation","Immune response","Inflammation","Immune response","Immune response","Transport","Inflammation","Immune response","Cell adhesion","Inflammation","Inflammation","Transport","Transcription","Transcription"],["Positive regulation cell proliferation","Jak-STAT Pathway","Antigen presentation","Platelet-endothelium-leucocyte interactions","Interferon signaling","S phase","Chemotaxis","Attractive and repulsive receptors","Negative regulation of cell proliferation","Translation initiation","Amphoterin signaling","Lymphocyte proliferation","Regulation of angiogenesis","Blood vessel morphogenesis","TCR signaling","NK cell cytotoxicity","Anti-Apoptosis mediated by external signals via PI3K/AKT","Phagosome in antigen presentation","Leucocyte chemotaxis","Blood coagulation","Anti-apoptosis mediated by external signals via NF-kB","Muscle contraction","IFN-gamma signaling","IL-4 signaling","Endoplasmic reticulum stress pathway","Anti-Apoptosis mediated by external signals via MAPK and JAK/STAT","Th17-derived cytokines","Protein C signaling","TREM1 signaling","Complement system","Phagocytosis","Neutrophil activation","T helper cell differentiation","BCR pathway","Potassium transport","Innate inflammatory response","Innate immune response to RNA viral infection ","Glycoconjugates","Histamine signaling","Kallikrein-kinin system","Sodium transport","Translation in mitochondria","Elongation-Termination"],["-0.406544848510004","-0.585368573722467","-0.635838374213216","-0.516684890863345","-0.634846775793164","0.378874710535532","-0.652373212172324","-0.336842077468743","-0.39208555027691","0.446147058846501","-0.491434228831937","-0.58515137992335","-0.425349982313245","-0.337201285270924","-0.576007348997439","-0.560334882420046","-0.439704426819946","-0.408733303695207","-0.636251323652007","-0.544537246143589","-0.459314591994637","0.393129348129591","-0.509617287933754","-0.655472523740724","-0.47493938973049","-0.38031380506899","-0.501184840878011","-0.536714423468605","-0.486571783041969","-0.701071073390931","-0.471871907997477","-0.469920620547141","-0.566024082866224","-0.476465445036621","0.408703534512452","-0.465695065338363","-0.43622021719289","-0.391656317753942","-0.410991826895193","-0.470720375121043","-0.373249954111633","0.486907677282816","0.452811704436146"],["0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0"],["NProliferation_Positive regulation cell proliferation","NInflammation_Jak-STAT Pathway","NImmune response_Antigen presentation","NCell adhesion_Platelet-endothelium-leucocyte interactions","NInflammation_Interferon signaling","NCell cycle_S phase","NChemotaxis","NCell adhesion_Attractive and repulsive receptors","NProliferation_Negative regulation of cell proliferation","NTranslation_Translation initiation","NInflammation_Amphoterin signaling","NProliferation_Lymphocyte proliferation","NDevelopment_Regulation of angiogenesis","NDevelopment_Blood vessel morphogenesis","NImmune response_TCR signaling","NInflammation_NK cell cytotoxicity","NApoptosis_Anti-Apoptosis mediated by external signals via PI3K/AKT","NImmune response_Phagosome in antigen presentation","NCell adhesion_Leucocyte chemotaxis","NBlood coagulation","NApoptosis_Anti-apoptosis mediated by external signals via NF-kB","NMuscle contraction","NInflammation_IFN-gamma signaling","NInflammation_IL-4 signaling","NApoptosis_Endoplasmic reticulum stress pathway","NApoptosis_Anti-Apoptosis mediated by external signals via MAPK and JAK/STAT","NImmune response_Th17-derived cytokines","NInflammation_Protein C signaling","NInflammation_TREM1 signaling","NInflammation_Complement system","NImmune response_Phagocytosis","NInflammation_Neutrophil activation","NImmune response_T helper cell differentiation","NImmune response_BCR pathway","NTransport_Potassium transport","NInflammation_Innate inflammatory response","NImmune response_Innate immune response to RNA viral infection ","NCell adhesion_Glycoconjugates","NInflammation_Histamine signaling","NInflammation_Kallikrein-kinin system","NTransport_Sodium transport","NTranslation_Translation in mitochondria","NTranslation_Elongation-Termination"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>TermType<\/th>\n      <th>sigTerm<\/th>\n      <th>stable_vs_growing.NES<\/th>\n      <th>stable_vs_growing.padj<\/th>\n      <th>names<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 13.4: Ep cells enriched pathways
</p>
</div>
</div>
</div>
<div id="comparisons-based-on-treatment" class="section level2">
<h2><span class="header-section-number">13.2</span> Comparisons based on treatment</h2>
<p>We can do the same analysis for the DN compartment, but for comparing treatment effect:</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="summary-of-gsea-runs.html#cb254-1"></a></span>
<span id="cb254-2"><a href="summary-of-gsea-runs.html#cb254-2"></a>d1=DNismr3<span class="op">$</span>ProcessNetworks[, <span class="kw">c</span>(<span class="st">&quot;imm_vs_control NES&quot;</span>, <span class="st">&quot;imm_vs_control padj&quot;</span>)]</span>
<span id="cb254-3"><a href="summary-of-gsea-runs.html#cb254-3"></a>ds=<span class="kw">which</span>(d1[ ,<span class="dv">2</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb254-4"><a href="summary-of-gsea-runs.html#cb254-4"></a></span>
<span id="cb254-5"><a href="summary-of-gsea-runs.html#cb254-5"></a>termA=d1[ds, ]</span>
<span id="cb254-6"><a href="summary-of-gsea-runs.html#cb254-6"></a></span>
<span id="cb254-7"><a href="summary-of-gsea-runs.html#cb254-7"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb254-8"><a href="summary-of-gsea-runs.html#cb254-8"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(termA)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb254-9"><a href="summary-of-gsea-runs.html#cb254-9"></a></span>
<span id="cb254-10"><a href="summary-of-gsea-runs.html#cb254-10"></a>sigTerms=TermsA</span>
<span id="cb254-11"><a href="summary-of-gsea-runs.html#cb254-11"></a>TermType=<span class="kw">substr</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]), <span class="dv">2</span>, <span class="dv">40</span>)</span>
<span id="cb254-12"><a href="summary-of-gsea-runs.html#cb254-12"></a></span>
<span id="cb254-13"><a href="summary-of-gsea-runs.html#cb254-13"></a><span class="co">## check whether this runs:</span></span>
<span id="cb254-14"><a href="summary-of-gsea-runs.html#cb254-14"></a><span class="kw">pdf</span>(<span class="st">&quot;figure-outputs/Ext3h.pdf&quot;</span>, <span class="dt">height=</span><span class="dv">7</span>, <span class="dt">width=</span><span class="dv">7</span>)</span>
<span id="cb254-15"><a href="summary-of-gsea-runs.html#cb254-15"></a><span class="kw">barplot</span>(<span class="kw">as.numeric</span>(termA[, <span class="dv">1</span>]), <span class="dt">names.arg =</span> sigTerms, <span class="dt">horiz =</span> T, <span class="dt">las=</span><span class="dv">2</span>,</span>
<span id="cb254-16"><a href="summary-of-gsea-runs.html#cb254-16"></a>         <span class="dt">xlab=</span><span class="st">&quot;NES&quot;</span>, <span class="dt">main=</span>testType, <span class="dt">col=</span><span class="kw">factor</span>(<span class="kw">sign</span>(<span class="kw">as.numeric</span>(termA[, <span class="dv">1</span>]))))</span>
<span id="cb254-17"><a href="summary-of-gsea-runs.html#cb254-17"></a><span class="kw">dev.off</span>()</span>
<span id="cb254-18"><a href="summary-of-gsea-runs.html#cb254-18"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb254-19"><a href="summary-of-gsea-runs.html#cb254-19"></a><span class="co">##                 2</span></span>
<span id="cb254-20"><a href="summary-of-gsea-runs.html#cb254-20"></a>Ntab=<span class="kw">data.frame</span>(<span class="dt">TermType=</span>TermType, <span class="dt">sigTerm=</span>sigTerms, termA)</span>
<span id="cb254-21"><a href="summary-of-gsea-runs.html#cb254-21"></a></span>
<span id="cb254-22"><a href="summary-of-gsea-runs.html#cb254-22"></a><span class="co">#write.csv(Ntab, file=&quot;nature-tables/Ext3h.csv&quot;)</span></span>
<span id="cb254-23"><a href="summary-of-gsea-runs.html#cb254-23"></a>DT<span class="op">::</span><span class="kw">datatable</span>(Ntab, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb254-24"><a href="summary-of-gsea-runs.html#cb254-24"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:Ext3h"></span>
<div id="htmlwidget-61f21c48354a4044c254" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-61f21c48354a4044c254">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Development","Development","Cell adhesion","Cell adhesion","Proteolysis","Transcription","Cell adhesion","Development","Cell adhesion","Proteolysis","Translation","Development","Development","Protein folding","Reproduction","Development","Inflammation","Cell adhesion","Cardiac development","Cell cycle","Muscle contraction","Development","Transport","Development","Cell adhesion","Immune response","Cell adhesion","Translation","Translation"],["Neurogenesis","Skeletal muscle development","Integrin-mediated cell-matrix adhesion","Cell-matrix interactions","Connective tissue degradation","Transcription by RNA polymerase II","Platelet-endothelium-leucocyte interactions","Ossification and bone remodeling","Attractive and repulsive receptors","ECM remodeling","Translation initiation","Neurogenesis","EMT","Protein folding nucleus","Spermatogenesis, motility and copulation","Blood vessel morphogenesis","NK cell cytotoxicity","Amyloid proteins","Wnt","Meiosis","Muscle contraction","Hemopoiesis, Erythropoietin pathway","Calcium transport","Regulation of telomere length","Integrin priming","T helper cell differentiation","Glycoconjugates","Translation in mitochondria","Elongation-Termination"],["-0.531477360014668","-0.524598947731966","-0.513422184476964","-0.6644008968927","-0.592615519348458","0.342863904971003","-0.50595143692371","-0.511050647317339","-0.50418433401643","-0.566495126517865","0.668248720413524","-0.500791799008836","-0.471966695514757","0.325858371688988","0.31729344650432","-0.481641274100002","0.268864782672753","-0.504328277098405","-0.597426397728821","0.305914015738161","-0.515399292061276","0.297360248830548","-0.474842931762632","0.480264746753393","-0.598125714659555","0.318942100007748","-0.445117281878897","0.471764826191514","0.664188159622963"],["0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>TermType<\/th>\n      <th>sigTerm<\/th>\n      <th>imm_vs_control.NES<\/th>\n      <th>imm_vs_control.padj<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 13.5: DN assoc treatment GSEA
</p>
</div>
<p>Also look at the output for combo treatment vs vehcile</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="summary-of-gsea-runs.html#cb255-1"></a>d1=DNismr3<span class="op">$</span>ProcessNetworks[, <span class="kw">c</span>(<span class="st">&quot;PDL1+LY_vs_Vehicle NES&quot;</span>, <span class="st">&quot;PDL1+LY_vs_Vehicle padj&quot;</span>)]</span>
<span id="cb255-2"><a href="summary-of-gsea-runs.html#cb255-2"></a>ds=<span class="kw">which</span>(d1[ ,<span class="dv">2</span>]<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb255-3"><a href="summary-of-gsea-runs.html#cb255-3"></a></span>
<span id="cb255-4"><a href="summary-of-gsea-runs.html#cb255-4"></a>termA=d1[ds, ]</span>
<span id="cb255-5"><a href="summary-of-gsea-runs.html#cb255-5"></a></span>
<span id="cb255-6"><a href="summary-of-gsea-runs.html#cb255-6"></a>TermsA=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(termA), <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb255-7"><a href="summary-of-gsea-runs.html#cb255-7"></a>TermsA[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))]=<span class="kw">substr</span>(<span class="kw">rownames</span>(termA)[<span class="kw">which</span>(<span class="kw">is.na</span>(TermsA))], <span class="dv">2</span>, <span class="dv">50</span>)</span>
<span id="cb255-8"><a href="summary-of-gsea-runs.html#cb255-8"></a></span>
<span id="cb255-9"><a href="summary-of-gsea-runs.html#cb255-9"></a><span class="kw">barplot</span>(<span class="kw">as.numeric</span>(termA[, <span class="dv">1</span>]), <span class="dt">names.arg =</span> TermsA, <span class="dt">horiz =</span> T, <span class="dt">las=</span><span class="dv">2</span>,</span>
<span id="cb255-10"><a href="summary-of-gsea-runs.html#cb255-10"></a>         <span class="dt">xlab=</span><span class="st">&quot;NES&quot;</span>, <span class="dt">main=</span>testType, <span class="dt">col=</span><span class="kw">factor</span>(<span class="kw">sign</span>(<span class="kw">as.numeric</span>(termA[, <span class="dv">1</span>]))))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-81-1.png" width="672" /></p>
</div>
<div id="pathways-of-interest-2" class="section level2">
<h2><span class="header-section-number">13.3</span> Pathways of Interest 2</h2>
<p>Focus on:</p>
<ul>
<li>JAK-STAT signalling</li>
<li>interferon-gamma signalling</li>
<li>IL-6 signalling</li>
<li>BCR pathway</li>
<li>TCR pathway</li>
<li>antigen presentation</li>
<li>KRAS signalling?</li>
<li>T helper differentiation</li>
</ul>
<p>In DN samples:</p>
<ul>
<li>androgen receptor signalling?</li>
<li>esr1 signalling?</li>
<li>myc targets</li>
<li>EMT</li>
</ul>
<p>Perform GSVA on these pathways</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="summary-of-gsea-runs.html#cb256-1"></a><span class="kw">load</span>(<span class="st">&quot;../anntotations/Metacore_extracted_Process_networks_nov2020.RData&quot;</span>)</span>
<span id="cb256-2"><a href="summary-of-gsea-runs.html#cb256-2"></a></span>
<span id="cb256-3"><a href="summary-of-gsea-runs.html#cb256-3"></a>sList=PathwayMapAllComp[<span class="kw">match</span>(<span class="kw">substr</span>(<span class="kw">rownames</span>(e1)[es], <span class="dv">2</span>, <span class="dv">100</span>), <span class="kw">names</span>(PathwayMapAllComp))]</span>
<span id="cb256-4"><a href="summary-of-gsea-runs.html#cb256-4"></a></span>
<span id="cb256-5"><a href="summary-of-gsea-runs.html#cb256-5"></a>rNames2=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(<span class="kw">rownames</span>(allTPMFinal), SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb256-6"><a href="summary-of-gsea-runs.html#cb256-6"></a>rNames2[<span class="kw">which</span>(<span class="kw">is.na</span>(rNames2))]=<span class="kw">toupper</span>(<span class="kw">rownames</span>(allTPMFinal)[<span class="kw">which</span>(<span class="kw">is.na</span>(rNames2))])</span>
<span id="cb256-7"><a href="summary-of-gsea-runs.html#cb256-7"></a></span>
<span id="cb256-8"><a href="summary-of-gsea-runs.html#cb256-8"></a></span>
<span id="cb256-9"><a href="summary-of-gsea-runs.html#cb256-9"></a>tpmTemp=allTPMFinal[ , <span class="kw">match</span>(vstEp<span class="op">$</span>SampleID, <span class="kw">colnames</span>(allTPMFinal))]</span>
<span id="cb256-10"><a href="summary-of-gsea-runs.html#cb256-10"></a><span class="kw">rownames</span>(tpmTemp)=rNames2</span>
<span id="cb256-11"><a href="summary-of-gsea-runs.html#cb256-11"></a><span class="kw">colnames</span>(tpmTemp)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(tpmTemp), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb256-12"><a href="summary-of-gsea-runs.html#cb256-12"></a></span>
<span id="cb256-13"><a href="summary-of-gsea-runs.html#cb256-13"></a></span>
<span id="cb256-14"><a href="summary-of-gsea-runs.html#cb256-14"></a>gsva1=<span class="kw">gsva</span>(tpmTemp, sList, <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb256-15"><a href="summary-of-gsea-runs.html#cb256-15"></a></span>
<span id="cb256-16"><a href="summary-of-gsea-runs.html#cb256-16"></a>nx2=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(gsva1), <span class="cf">function</span>(x) <span class="kw">sd</span>(gsva1[x, ]))</span>
<span id="cb256-17"><a href="summary-of-gsea-runs.html#cb256-17"></a>a1=<span class="kw">which</span>(nx2<span class="op">&gt;</span><span class="fl">0.03</span>)</span>
<span id="cb256-18"><a href="summary-of-gsea-runs.html#cb256-18"></a></span>
<span id="cb256-19"><a href="summary-of-gsea-runs.html#cb256-19"></a>sList=PathwayMapAllComp[<span class="kw">match</span>(<span class="kw">substr</span>(<span class="kw">rownames</span>(d1)[ds], <span class="dv">2</span>, <span class="dv">100</span>), <span class="kw">names</span>(PathwayMapAllComp))]</span>
<span id="cb256-20"><a href="summary-of-gsea-runs.html#cb256-20"></a></span>
<span id="cb256-21"><a href="summary-of-gsea-runs.html#cb256-21"></a>tpmTemp=allTPMFinal[ , <span class="kw">match</span>(vstDN<span class="op">$</span>SampleID, <span class="kw">colnames</span>(allTPMFinal))]</span>
<span id="cb256-22"><a href="summary-of-gsea-runs.html#cb256-22"></a><span class="kw">colnames</span>(tpmTemp)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(tpmTemp), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb256-23"><a href="summary-of-gsea-runs.html#cb256-23"></a><span class="kw">rownames</span>(tpmTemp)=rNames2</span>
<span id="cb256-24"><a href="summary-of-gsea-runs.html#cb256-24"></a></span>
<span id="cb256-25"><a href="summary-of-gsea-runs.html#cb256-25"></a>gsva2=<span class="kw">gsva</span>(tpmTemp, sList, <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb256-26"><a href="summary-of-gsea-runs.html#cb256-26"></a></span>
<span id="cb256-27"><a href="summary-of-gsea-runs.html#cb256-27"></a>nx2=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(gsva2), <span class="cf">function</span>(x) <span class="kw">sd</span>(gsva2[x, ]))</span>
<span id="cb256-28"><a href="summary-of-gsea-runs.html#cb256-28"></a>a2=<span class="kw">which</span>(nx2<span class="op">&gt;</span><span class="fl">0.03</span>)</span>
<span id="cb256-29"><a href="summary-of-gsea-runs.html#cb256-29"></a></span>
<span id="cb256-30"><a href="summary-of-gsea-runs.html#cb256-30"></a>sList=PathwayMapAllComp[<span class="kw">match</span>(<span class="kw">substr</span>(<span class="kw">rownames</span>(c1)[cs], <span class="dv">2</span>, <span class="dv">100</span>), <span class="kw">names</span>(PathwayMapAllComp))]</span>
<span id="cb256-31"><a href="summary-of-gsea-runs.html#cb256-31"></a></span>
<span id="cb256-32"><a href="summary-of-gsea-runs.html#cb256-32"></a>tpmTemp=allTPMFinal[ , <span class="kw">match</span>(vstCD<span class="op">$</span>SampleID, <span class="kw">colnames</span>(allTPMFinal))]</span>
<span id="cb256-33"><a href="summary-of-gsea-runs.html#cb256-33"></a><span class="kw">colnames</span>(tpmTemp)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(tpmTemp), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb256-34"><a href="summary-of-gsea-runs.html#cb256-34"></a><span class="kw">rownames</span>(tpmTemp)=rNames2</span>
<span id="cb256-35"><a href="summary-of-gsea-runs.html#cb256-35"></a></span>
<span id="cb256-36"><a href="summary-of-gsea-runs.html#cb256-36"></a>gsva3=<span class="kw">gsva</span>(tpmTemp, sList, <span class="dt">method=</span><span class="st">&quot;ssgsea&quot;</span>, <span class="dt">ssgsea.norm=</span>T)</span>
<span id="cb256-37"><a href="summary-of-gsea-runs.html#cb256-37"></a></span>
<span id="cb256-38"><a href="summary-of-gsea-runs.html#cb256-38"></a>nx2=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(gsva3), <span class="cf">function</span>(x) <span class="kw">sd</span>(gsva3[x, ]))</span>
<span id="cb256-39"><a href="summary-of-gsea-runs.html#cb256-39"></a>a3=<span class="kw">which</span>(nx2<span class="op">&gt;</span><span class="fl">0.03</span>)</span>
<span id="cb256-40"><a href="summary-of-gsea-runs.html#cb256-40"></a></span>
<span id="cb256-41"><a href="summary-of-gsea-runs.html#cb256-41"></a><span class="co">#pdf(&quot;~/Desktop/5B-ssgsea-scores.pdf&quot;, height=5, width=5)</span></span>
<span id="cb256-42"><a href="summary-of-gsea-runs.html#cb256-42"></a></span>
<span id="cb256-43"><a href="summary-of-gsea-runs.html#cb256-43"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">5</span>))</span>
<span id="cb256-44"><a href="summary-of-gsea-runs.html#cb256-44"></a><span class="kw">heatmap.2</span>(gsva1[a1, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSizeb[vstEp<span class="op">$</span>Growth],</span>
<span id="cb256-45"><a href="summary-of-gsea-runs.html#cb256-45"></a>          <span class="dt">main=</span><span class="st">&quot;Ep&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-82-1.png" width="672" /></p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="summary-of-gsea-runs.html#cb257-1"></a></span>
<span id="cb257-2"><a href="summary-of-gsea-runs.html#cb257-2"></a><span class="kw">heatmap.2</span>(gsva2[a2, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSizeb[vstDN<span class="op">$</span>Growth],</span>
<span id="cb257-3"><a href="summary-of-gsea-runs.html#cb257-3"></a>          <span class="dt">main=</span><span class="st">&quot;DN&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-82-2.png" width="672" /></p>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="summary-of-gsea-runs.html#cb258-1"></a></span>
<span id="cb258-2"><a href="summary-of-gsea-runs.html#cb258-2"></a><span class="kw">heatmap.2</span>(gsva3[a3, ], <span class="dt">col=</span>RdBu[<span class="dv">11</span><span class="op">:</span><span class="dv">1</span>], <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">ColSideColors =</span> ColSizeb[vstCD<span class="op">$</span>Growth],</span>
<span id="cb258-3"><a href="summary-of-gsea-runs.html#cb258-3"></a>          <span class="dt">main=</span><span class="st">&quot;CD&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-82-3.png" width="672" /></p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="summary-of-gsea-runs.html#cb259-1"></a></span>
<span id="cb259-2"><a href="summary-of-gsea-runs.html#cb259-2"></a><span class="co">#dev.off()</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="deseq-analysis-immunotherapygrowth-comparisons.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="epcam-inflammatory-vs-non-inflammatory-samples.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
