<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 25 Trichrome staining | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 25 Trichrome staining | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 25 Trichrome staining | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2022-02-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutations-in-progression-cohort.html"/>
<link rel="next" href="cnv-calling.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.3</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html"><i class="fa fa-check"></i><b>16</b> Read in ER i/o data</a><ul>
<li class="chapter" data-level="16.1" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.1</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.2" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#do-associative-study-with-other-features"><i class="fa fa-check"></i><b>16.2</b> Do associative study with other features?</a></li>
<li class="chapter" data-level="16.3" data-path="read-in-er-io-data.html"><a href="read-in-er-io-data.html#use-raw-counts-and-perform-a-batch-correction-if-necessary"><i class="fa fa-check"></i><b>16.3</b> Use raw counts and perform a batch correction if necessary</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="tcr-clonotype-analysis.html"><a href="tcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>21</b> TCR clonotype analysis</a><ul>
<li class="chapter" data-level="21.0.1" data-path="tcr-clonotype-analysis.html"><a href="tcr-clonotype-analysis.html#progression-cohort-2"><i class="fa fa-check"></i><b>21.0.1</b> Progression cohort</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>22</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="22.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>22.1</b> Data</a><ul>
<li class="chapter" data-level="22.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>22.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>22.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="22.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>22.3</b> Plots</a><ul>
<li class="chapter" data-level="22.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>22.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="22.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>22.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>22.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="22.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>22.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="22.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>22.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="22.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>22.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="22.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>22.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>23</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>23.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>23.2</b> Load files</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>23.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="23.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>23.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="23.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>23.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>24</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="24.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#mutational-burden"><i class="fa fa-check"></i><b>24.1</b> Mutational burden</a></li>
<li class="chapter" data-level="24.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>24.2</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="24.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>24.3</b> Look at the common mutations (cosmic ones)</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>25</b> Trichrome staining</a><ul>
<li class="chapter" data-level="25.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>25.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="25.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>25.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="25.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>25.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="25.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>25.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>26</b> CNV calling</a><ul>
<li class="chapter" data-level="26.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>26.1</b> Sample data</a></li>
<li class="chapter" data-level="26.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>26.2</b> Data Summary</a><ul>
<li class="chapter" data-level="26.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>26.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="26.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>26.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="26.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>26.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="26.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>26.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="26.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>26.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>27</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="trichrome-staining" class="section level1">
<h1><span class="header-section-number">Chapter 25</span> Trichrome staining</h1>
<p>Quantification of trichrome staining was performed in Qupath using the following steps:</p>
<ul>
<li>image is loaded as a “DAB” image</li>
<li>using “gold standard” trichrome-stained images with good stroma and epithelial content, estimate the stain vectors</li>
<li>Perform color deconvolution</li>
<li>A pixel classifier was used to estimate trichrome content</li>
<li>A pixel classifier was used to estimate tumor content</li>
</ul>
<p>A snippet of the qupath script is shown below:</p>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="trichrome-staining.html#cb580-1"></a><span class="co"># insert some text here</span></span>
<span id="cb580-2"><a href="trichrome-staining.html#cb580-2"></a></span>
<span id="cb580-3"><a href="trichrome-staining.html#cb580-3"></a><span class="kw">setImageType</span>(<span class="st">&#39;BRIGHTFIELD_H_DAB&#39;</span>);</span>
<span id="cb580-4"><a href="trichrome-staining.html#cb580-4"></a><span class="kw">setColorDeconvolutionStains</span>(<span class="st">&#39;{&quot;Name&quot; : &quot;trichrome&quot;, &quot;Stain 1&quot; : &quot;Hematoxylin&quot;, &quot;Values 1&quot; : &quot;0.71695 0.66336 0.21432 &quot;, &quot;Stain 2&quot; : &quot;DAB&quot;, &quot;Values 2&quot; : &quot;0.46299 0.76212 0.45257 &quot;, &quot;Background&quot; : &quot; 255 255 255 &quot;}&#39;</span>);</span>
<span id="cb580-5"><a href="trichrome-staining.html#cb580-5"></a><span class="kw">selectAnnotations</span>();</span>
<span id="cb580-6"><a href="trichrome-staining.html#cb580-6"></a><span class="kw">addPixelClassifierMeasurements</span>(<span class="st">&quot;stroma_classifier_again&quot;</span>, <span class="st">&quot;stroma_classifier_again&quot;</span>)</span>
<span id="cb580-7"><a href="trichrome-staining.html#cb580-7"></a><span class="kw">selectAnnotations</span>();</span>
<span id="cb580-8"><a href="trichrome-staining.html#cb580-8"></a><span class="kw">addPixelClassifierMeasurements</span>(<span class="st">&quot;test_tumor&quot;</span>, <span class="st">&quot;test_tumor&quot;</span>)</span>
<span id="cb580-9"><a href="trichrome-staining.html#cb580-9"></a></span>
<span id="cb580-10"><a href="trichrome-staining.html#cb580-10"></a>def entry =<span class="st"> </span><span class="kw">getProjectEntry</span>()</span>
<span id="cb580-11"><a href="trichrome-staining.html#cb580-11"></a>def name =<span class="st"> </span><span class="kw">entry.getImageName</span>() <span class="op">+</span><span class="st"> &#39;.txt&#39;</span></span>
<span id="cb580-12"><a href="trichrome-staining.html#cb580-12"></a>def path =<span class="st"> </span><span class="kw">buildFilePath</span>(PROJECT_BASE_DIR, <span class="st">&#39;annotation results&#39;</span>)</span>
<span id="cb580-13"><a href="trichrome-staining.html#cb580-13"></a><span class="kw">mkdirs</span>(path)</span>
<span id="cb580-14"><a href="trichrome-staining.html#cb580-14"></a>path =<span class="st"> </span><span class="kw">buildFilePath</span>(path, name)</span>
<span id="cb580-15"><a href="trichrome-staining.html#cb580-15"></a><span class="kw">saveAnnotationMeasurements</span>(path)</span></code></pre></div>
<div id="associations-with-cellular-fraction-wsi" class="section level2">
<h2><span class="header-section-number">25.1</span> Associations with cellular fraction (wsi)</h2>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="trichrome-staining.html#cb581-1"></a><span class="co"># load the data</span></span>
<span id="cb581-2"><a href="trichrome-staining.html#cb581-2"></a>TrichromeData=<span class="kw">read.xlsx</span>(<span class="st">&quot;../data/trichrome-staining-result.xls&quot;</span>, <span class="dv">1</span>)</span>
<span id="cb581-3"><a href="trichrome-staining.html#cb581-3"></a>midx=<span class="kw">match</span>(TrichromeData<span class="op">$</span>SampleID, <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))</span>
<span id="cb581-4"><a href="trichrome-staining.html#cb581-4"></a></span>
<span id="cb581-5"><a href="trichrome-staining.html#cb581-5"></a>Cdata<span class="op">$</span>Trichrome=<span class="ot">NA</span></span>
<span id="cb581-6"><a href="trichrome-staining.html#cb581-6"></a>Cdata<span class="op">$</span>Trichrome[<span class="kw">na.omit</span>(midx)]=TrichromeData<span class="op">$</span>Percentage.Stroma[<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(midx))]</span>
<span id="cb581-7"><a href="trichrome-staining.html#cb581-7"></a></span>
<span id="cb581-8"><a href="trichrome-staining.html#cb581-8"></a>t2=WSIvalFracs[, <span class="kw">match</span>(<span class="kw">rownames</span>(df.Spatial), <span class="kw">colnames</span>(WSIvalFracs))]</span>
<span id="cb581-9"><a href="trichrome-staining.html#cb581-9"></a></span>
<span id="cb581-10"><a href="trichrome-staining.html#cb581-10"></a>df.Spatial=<span class="kw">cbind</span>(df.Spatial, <span class="kw">t</span>(t2))</span>
<span id="cb581-11"><a href="trichrome-staining.html#cb581-11"></a></span>
<span id="cb581-12"><a href="trichrome-staining.html#cb581-12"></a></span>
<span id="cb581-13"><a href="trichrome-staining.html#cb581-13"></a>df.Spatial<span class="op">$</span>Trichrome=<span class="ot">NA</span></span>
<span id="cb581-14"><a href="trichrome-staining.html#cb581-14"></a>midx=<span class="kw">match</span>(TrichromeData<span class="op">$</span>SampleID, <span class="kw">rownames</span>(df.Spatial))</span>
<span id="cb581-15"><a href="trichrome-staining.html#cb581-15"></a>df.Spatial<span class="op">$</span>Trichrome[<span class="kw">na.omit</span>(midx)]=TrichromeData<span class="op">$</span>Percentage.Stroma[<span class="op">-</span><span class="kw">which</span>(<span class="kw">is.na</span>(midx))]</span>
<span id="cb581-16"><a href="trichrome-staining.html#cb581-16"></a><span class="co"># plot associations</span></span>
<span id="cb581-17"><a href="trichrome-staining.html#cb581-17"></a></span>
<span id="cb581-18"><a href="trichrome-staining.html#cb581-18"></a>n2=<span class="kw">c</span>(<span class="st">&quot;CD8&quot;</span>, <span class="st">&quot;EpCAM&quot;</span>, <span class="st">&quot;SMA&quot;</span>, <span class="st">&quot;Unclass&quot;</span>, <span class="st">&quot;EpCAM: SMA&quot;</span>)</span>
<span id="cb581-19"><a href="trichrome-staining.html#cb581-19"></a></span>
<span id="cb581-20"><a href="trichrome-staining.html#cb581-20"></a><span class="co">#pdf(&quot;~/Desktop/richrome-association-WSI-data-Calc2.pdf&quot;, width=6, height=5)</span></span>
<span id="cb581-21"><a href="trichrome-staining.html#cb581-21"></a></span>
<span id="cb581-22"><a href="trichrome-staining.html#cb581-22"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb581-23"><a href="trichrome-staining.html#cb581-23"></a><span class="cf">for</span> (i <span class="cf">in</span> n2){</span>
<span id="cb581-24"><a href="trichrome-staining.html#cb581-24"></a>   a1=<span class="kw">cor.test</span>(df.Spatial<span class="op">$</span>Trichrome, df.Spatial[ ,<span class="kw">match</span>(i, <span class="kw">colnames</span>(df.Spatial))], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb581-25"><a href="trichrome-staining.html#cb581-25"></a>   n1=<span class="kw">paste</span>(i, <span class="st">&quot;cor:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate,<span class="dv">1</span>), <span class="st">&quot;p:&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>p.value,<span class="dv">2</span>))</span>
<span id="cb581-26"><a href="trichrome-staining.html#cb581-26"></a>   <span class="kw">plot</span>(df.Spatial<span class="op">$</span>Trichrome, df.Spatial[ ,<span class="kw">match</span>(i, <span class="kw">colnames</span>(df.Spatial))], <span class="dt">main=</span>n1, <span class="dt">xlab=</span><span class="st">&quot;Trichrome&quot;</span>, <span class="dt">ylab=</span>i)</span>
<span id="cb581-27"><a href="trichrome-staining.html#cb581-27"></a>}</span>
<span id="cb581-28"><a href="trichrome-staining.html#cb581-28"></a></span>
<span id="cb581-29"><a href="trichrome-staining.html#cb581-29"></a><span class="co">#dev.off()</span></span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-170-1.png" width="672" /></p>
</div>
<div id="associations-with-cd8-content" class="section level2">
<h2><span class="header-section-number">25.2</span> Associations with CD8 content</h2>
<div class="sourceCode" id="cb582"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb582-1"><a href="trichrome-staining.html#cb582-1"></a></span>
<span id="cb582-2"><a href="trichrome-staining.html#cb582-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb582-3"><a href="trichrome-staining.html#cb582-3"></a></span>
<span id="cb582-4"><a href="trichrome-staining.html#cb582-4"></a>a1=<span class="kw">wilcox.test</span>(df.Spatial<span class="op">$</span>Trichrome<span class="op">~</span>df.Spatial<span class="op">$</span>IF.EpCAMcut)<span class="op">$</span>p.val</span>
<span id="cb582-5"><a href="trichrome-staining.html#cb582-5"></a><span class="kw">boxplot</span>(df.Spatial<span class="op">$</span>Trichrome<span class="op">~</span>df.Spatial<span class="op">$</span>IF.EpCAMcut, <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;Ep int. fraction %s&quot;</span>, <span class="kw">round</span>(a1,<span class="dv">2</span>)))</span>
<span id="cb582-6"><a href="trichrome-staining.html#cb582-6"></a>a1=<span class="kw">wilcox.test</span>(df.Spatial<span class="op">$</span>Trichrome<span class="op">~</span>df.Spatial<span class="op">$</span>CD8Fraccut)<span class="op">$</span>p.val</span>
<span id="cb582-7"><a href="trichrome-staining.html#cb582-7"></a><span class="kw">boxplot</span>(df.Spatial<span class="op">$</span>Trichrome<span class="op">~</span>df.Spatial<span class="op">$</span>CD8Fraccut, <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;CD8 fraction vs Trichrome %s&quot;</span>, <span class="kw">round</span>(a1,<span class="dv">2</span>)))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/3gpre-1.png" width="672" /></p>
</div>
<div id="associations-with-growth-and-treatment" class="section level2">
<h2><span class="header-section-number">25.3</span> Associations with growth and treatment</h2>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="trichrome-staining.html#cb583-1"></a>df.Spatial<span class="op">$</span>Growth=Cdata<span class="op">$</span>Tumor.Growth[<span class="kw">match</span>(<span class="kw">rownames</span>(df.Spatial), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb583-2"><a href="trichrome-staining.html#cb583-2"></a>df.Spatial<span class="op">$</span>Treatment=Cdata<span class="op">$</span>Treatment[<span class="kw">match</span>(<span class="kw">rownames</span>(df.Spatial), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb583-3"><a href="trichrome-staining.html#cb583-3"></a>df.Spatial<span class="op">$</span>NewID=Cdata<span class="op">$</span>Treatment[<span class="kw">match</span>(<span class="kw">rownames</span>(df.Spatial), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>NewID))]</span>
<span id="cb583-4"><a href="trichrome-staining.html#cb583-4"></a>df.Spatial<span class="op">$</span>trichrome_pc=Cdata<span class="op">$</span>Trichrome_encapsulation.[<span class="kw">match</span>(<span class="kw">rownames</span>(df.Spatial), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb583-5"><a href="trichrome-staining.html#cb583-5"></a></span>
<span id="cb583-6"><a href="trichrome-staining.html#cb583-6"></a>pv1=<span class="kw">sapply</span>(<span class="kw">c</span>(<span class="st">&quot;PDL1&quot;</span>, <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;PDL1+LY&quot;</span>), <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(df.Spatial<span class="op">$</span>Trichrome[<span class="kw">which</span>(df.Spatial<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>, x))]<span class="op">~</span></span>
<span id="cb583-7"><a href="trichrome-staining.html#cb583-7"></a><span class="st">           </span>df.Spatial<span class="op">$</span>Treatment[<span class="kw">which</span>(df.Spatial<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Vehicle&quot;</span>, x))])<span class="op">$</span>p.value)</span>
<span id="cb583-8"><a href="trichrome-staining.html#cb583-8"></a><span class="kw">names</span>(pv1)=<span class="kw">c</span>(<span class="st">&quot;PDL1&quot;</span>, <span class="st">&quot;LY&quot;</span>, <span class="st">&quot;PDL1+LY&quot;</span>)</span>
<span id="cb583-9"><a href="trichrome-staining.html#cb583-9"></a></span>
<span id="cb583-10"><a href="trichrome-staining.html#cb583-10"></a><span class="kw">ggplot</span>(df.Spatial[ ,<span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;Trichrome&quot;</span>)], <span class="kw">aes</span>(<span class="dt">x=</span>Treatment, <span class="dt">y=</span>Trichrome, <span class="dt">col=</span>Treatment))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_jitter</span>()<span class="op">+</span></span>
<span id="cb583-11"><a href="trichrome-staining.html#cb583-11"></a><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>ColMerge[ ,<span class="dv">1</span>])<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="kw">paste</span>(<span class="kw">names</span>(pv1), <span class="kw">round</span>(pv1, <span class="dv">2</span>)), <span class="dt">collapse=</span><span class="st">&quot;;&quot;</span>))</span></code></pre></div>
<div class="figure"><span id="fig:3g-1"></span>
<img src="RatDCIS-bookdown_files/figure-html/3g-1.png" alt="Trichrome staining with treatment" width="672" />
<p class="caption">
Figure 25.1: Trichrome staining with treatment
</p>
</div>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="trichrome-staining.html#cb584-1"></a></span>
<span id="cb584-2"><a href="trichrome-staining.html#cb584-2"></a><span class="kw">ggplot</span>(df.Spatial[ ,<span class="kw">c</span>(<span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;trichrome_pc&quot;</span>)], <span class="kw">aes</span>(<span class="dt">x=</span>Treatment, <span class="dt">y=</span>trichrome_pc, <span class="dt">col=</span>Treatment))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_jitter</span>()<span class="op">+</span></span>
<span id="cb584-3"><a href="trichrome-staining.html#cb584-3"></a><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>ColMerge[ ,<span class="dv">1</span>])<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="kw">paste</span>(<span class="kw">names</span>(pv1), <span class="kw">round</span>(pv1, <span class="dv">2</span>)), <span class="dt">collapse=</span><span class="st">&quot;;&quot;</span>))</span></code></pre></div>
<div class="figure"><span id="fig:3g-2"></span>
<img src="RatDCIS-bookdown_files/figure-html/3g-2.png" alt="Trichrome staining with treatment" width="672" />
<p class="caption">
Figure 25.2: Trichrome staining with treatment
</p>
</div>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="trichrome-staining.html#cb585-1"></a></span>
<span id="cb585-2"><a href="trichrome-staining.html#cb585-2"></a>a1=df.Spatial[ ,<span class="kw">match</span>(<span class="kw">c</span>(<span class="st">&quot;Trichrome&quot;</span>, <span class="st">&quot;Growth&quot;</span>, <span class="st">&quot;trichrome_pc&quot;</span>), <span class="kw">colnames</span>(df.Spatial))]</span>
<span id="cb585-3"><a href="trichrome-staining.html#cb585-3"></a>a1a=<span class="kw">wilcox.test</span>(df.Spatial<span class="op">$</span>Trichrome[<span class="kw">which</span>(df.Spatial<span class="op">$</span>Growth<span class="op">==</span><span class="st">&quot;growing&quot;</span>)],</span>
<span id="cb585-4"><a href="trichrome-staining.html#cb585-4"></a>           df.Spatial<span class="op">$</span>Trichrome[<span class="kw">which</span>(df.Spatial<span class="op">$</span>Growth<span class="op">==</span><span class="st">&quot;stable&quot;</span>)])</span>
<span id="cb585-5"><a href="trichrome-staining.html#cb585-5"></a>a1a</span>
<span id="cb585-6"><a href="trichrome-staining.html#cb585-6"></a><span class="co">## </span></span>
<span id="cb585-7"><a href="trichrome-staining.html#cb585-7"></a><span class="co">##  Wilcoxon rank sum exact test</span></span>
<span id="cb585-8"><a href="trichrome-staining.html#cb585-8"></a><span class="co">## </span></span>
<span id="cb585-9"><a href="trichrome-staining.html#cb585-9"></a><span class="co">## data:  df.Spatial$Trichrome[which(df.Spatial$Growth == &quot;growing&quot;)] and df.Spatial$Trichrome[which(df.Spatial$Growth == &quot;stable&quot;)]</span></span>
<span id="cb585-10"><a href="trichrome-staining.html#cb585-10"></a><span class="co">## W = 79, p-value = 0.04521</span></span>
<span id="cb585-11"><a href="trichrome-staining.html#cb585-11"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb585-12"><a href="trichrome-staining.html#cb585-12"></a></span>
<span id="cb585-13"><a href="trichrome-staining.html#cb585-13"></a><span class="kw">ggplot</span>(a1, <span class="kw">aes</span>(<span class="dt">x=</span>Growth, <span class="dt">y=</span>Trichrome, <span class="dt">col=</span>Growth))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_jitter</span>()<span class="op">+</span></span>
<span id="cb585-14"><a href="trichrome-staining.html#cb585-14"></a><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(ColSize, <span class="st">&quot;black&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;growth&quot;</span>) <span class="co"># round(a1a, 2)))</span></span></code></pre></div>
<div class="figure"><span id="fig:3gb-1"></span>
<img src="RatDCIS-bookdown_files/figure-html/3gb-1.png" alt="Trichrome with growth" width="672" />
<p class="caption">
Figure 25.3: Trichrome with growth
</p>
</div>
<div class="sourceCode" id="cb586"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb586-1"><a href="trichrome-staining.html#cb586-1"></a></span>
<span id="cb586-2"><a href="trichrome-staining.html#cb586-2"></a><span class="kw">ggplot</span>(a1, <span class="kw">aes</span>(<span class="dt">x=</span>Growth, <span class="dt">y=</span>trichrome_pc, <span class="dt">col=</span>Growth))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_jitter</span>()<span class="op">+</span></span>
<span id="cb586-3"><a href="trichrome-staining.html#cb586-3"></a><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(ColSize, <span class="st">&quot;black&quot;</span>))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;growth&quot;</span>) <span class="co"># round(a1a, 2)))</span></span></code></pre></div>
<div class="figure"><span id="fig:3gb-2"></span>
<img src="RatDCIS-bookdown_files/figure-html/3gb-2.png" alt="Trichrome with growth" width="672" />
<p class="caption">
Figure 25.4: Trichrome with growth
</p>
</div>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="trichrome-staining.html#cb587-1"></a></span>
<span id="cb587-2"><a href="trichrome-staining.html#cb587-2"></a><span class="co">#print(p)</span></span>
<span id="cb587-3"><a href="trichrome-staining.html#cb587-3"></a></span>
<span id="cb587-4"><a href="trichrome-staining.html#cb587-4"></a><span class="co">#write.csv(df.Spatial[ ,c(&quot;Trichrome&quot;, &quot;Growth&quot;, &quot;Treatment&quot;)], file=&quot;nature-tables/3g_trichrome.csv&quot;)</span></span>
<span id="cb587-5"><a href="trichrome-staining.html#cb587-5"></a>DT<span class="op">::</span><span class="kw">datatable</span>(df.Spatial[ ,<span class="kw">c</span>(<span class="st">&quot;Trichrome&quot;</span>, <span class="st">&quot;Growth&quot;</span>, <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;trichrome_pc&quot;</span>)], <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>, <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>), <span class="dt">scrollX=</span>T))</span></code></pre></div>
<div class="figure"><span id="fig:3gb-3"></span>
<div id="htmlwidget-19c9127315f8ed40c136" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-19c9127315f8ed40c136">{"x":{"filter":"none","extensions":["Buttons"],"data":[[0.12664383309166,0.150025705049749,0.176618842694733,null,null,null,0.145513854031148,null,0.133181297679202,0.374043581170937,0.124598668056126,0.161619227345989,null,0.276814784264216,0.166215509004265,0.142755297464136,0.15485032053922,0.0217913475951452,0.00295475085300382,null,0.126578024868847,0.140691073282721,null,null,0.1935505629915,0.359852126250554,null,0.299769371626505,0.350950798461984,0.34230818181261,0.126077261157979,0.19008060786476,null,0.0974472884287867,0.433978944576645,0.12018569287281,0.154296069680264,0.185753281488712,0.294572883370248,null,0.244984442858644,null,null,null,null,0.256966970951017,null,null,0.114344669757582,0.108067761591177,0.238777705969805,0.238762369875553,null,null,null,0.445971508497105,0.415612058953682,null],["growing","growing","stable",null,"stable","growing",null,null,"growing","stable","growing","growing",null,"stable","growing","growing","stable","stable","growing","growing","stable","growing",null,"stable","stable","stable","stable","growing","stable","growing","growing","growing","growing","growing","stable","growing","growing","growing","growing","growing","growing","stable","growing","growing","stable","growing","growing","growing","growing","growing","growing","growing","stable","growing","stable","stable","growing","growing"],["PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY"],["0.75","0.5","0",null,null,null,"1",null,"1","0","1","1",null,"0","1","0.75","0.25","0","0",null,"0.5","1",null,null,"1","0.5",null,"1","0","1","0.5","0.75",null,"1","0","1","0","1","1",null,"1",null,null,null,null,"1","1",null,"0.25","1","1","1",null,null,null,"0.25","1","lots"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>Trichrome<\/th>\n      <th>Growth<\/th>\n      <th>Treatment<\/th>\n      <th>trichrome_pc<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"scrollX":true,"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 25.5: Trichrome with growth
</p>
</div>
<p>Trichrome encapsulation</p>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="trichrome-staining.html#cb588-1"></a>df.Spatial<span class="op">$</span>trichrome_pc[<span class="kw">which</span>(df.Spatial<span class="op">$</span>trichrome_pc<span class="op">==</span><span class="st">&quot;lots&quot;</span>)]=<span class="dv">1</span></span>
<span id="cb588-2"><a href="trichrome-staining.html#cb588-2"></a>df.Spatial<span class="op">$</span>trichrome_pc[<span class="kw">which</span>(df.Spatial<span class="op">$</span>trichrome_pc<span class="op">==</span><span class="st">&quot;0% ignore&quot;</span>)]=<span class="ot">NA</span></span>
<span id="cb588-3"><a href="trichrome-staining.html#cb588-3"></a>df.Spatial<span class="op">$</span>trichrome_pc</span>
<span id="cb588-4"><a href="trichrome-staining.html#cb588-4"></a><span class="co">##  [1] &quot;0.75&quot; &quot;0.5&quot;  &quot;0&quot;    NA     NA     NA     &quot;1&quot;    NA     &quot;1&quot;    &quot;0&quot;   </span></span>
<span id="cb588-5"><a href="trichrome-staining.html#cb588-5"></a><span class="co">## [11] &quot;1&quot;    &quot;1&quot;    NA     &quot;0&quot;    &quot;1&quot;    &quot;0.75&quot; &quot;0.25&quot; &quot;0&quot;    &quot;0&quot;    NA    </span></span>
<span id="cb588-6"><a href="trichrome-staining.html#cb588-6"></a><span class="co">## [21] &quot;0.5&quot;  &quot;1&quot;    NA     NA     &quot;1&quot;    &quot;0.5&quot;  NA     &quot;1&quot;    &quot;0&quot;    &quot;1&quot;   </span></span>
<span id="cb588-7"><a href="trichrome-staining.html#cb588-7"></a><span class="co">## [31] &quot;0.5&quot;  &quot;0.75&quot; NA     &quot;1&quot;    &quot;0&quot;    &quot;1&quot;    &quot;0&quot;    &quot;1&quot;    &quot;1&quot;    NA    </span></span>
<span id="cb588-8"><a href="trichrome-staining.html#cb588-8"></a><span class="co">## [41] &quot;1&quot;    NA     NA     NA     NA     &quot;1&quot;    &quot;1&quot;    NA     &quot;0.25&quot; &quot;1&quot;   </span></span>
<span id="cb588-9"><a href="trichrome-staining.html#cb588-9"></a><span class="co">## [51] &quot;1&quot;    &quot;1&quot;    NA     NA     NA     &quot;0.25&quot; &quot;1&quot;    &quot;1&quot;</span></span>
<span id="cb588-10"><a href="trichrome-staining.html#cb588-10"></a><span class="co">#wilcox.test(factor(as.numeric(df.Spatial$trichrome_pc))~df.Spatial$Growth)</span></span></code></pre></div>
</div>
<div id="association-with-hyperinflammatory-status" class="section level2">
<h2><span class="header-section-number">25.4</span> Association with hyperinflammatory status</h2>
<p>We can also do a boxplot for trichrome status and whether it associates with hyperinflammation in our rat samples</p>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="trichrome-staining.html#cb589-1"></a>vstEpInf<span class="op">$</span>Inflammation3</span>
<span id="cb589-2"><a href="trichrome-staining.html#cb589-2"></a><span class="co">##  [1] inf  grow grow inf  grow grow grow grow grow grow grow grow grow grow inf </span></span>
<span id="cb589-3"><a href="trichrome-staining.html#cb589-3"></a><span class="co">## [16] grow inf  grow inf  grow</span></span>
<span id="cb589-4"><a href="trichrome-staining.html#cb589-4"></a><span class="co">## Levels: grow inf</span></span>
<span id="cb589-5"><a href="trichrome-staining.html#cb589-5"></a>vstEpInf<span class="op">$</span>Inflammation2</span>
<span id="cb589-6"><a href="trichrome-staining.html#cb589-6"></a><span class="co">##  [1] inf  grow grow inf  grow grow stab stab stab grow grow stab stab grow inf </span></span>
<span id="cb589-7"><a href="trichrome-staining.html#cb589-7"></a><span class="co">## [16] grow inf  grow inf  grow</span></span>
<span id="cb589-8"><a href="trichrome-staining.html#cb589-8"></a><span class="co">## Levels: grow inf stab</span></span>
<span id="cb589-9"><a href="trichrome-staining.html#cb589-9"></a>vstEpInf<span class="op">$</span>Inflammation</span>
<span id="cb589-10"><a href="trichrome-staining.html#cb589-10"></a><span class="co">##  [1] yes no  no  yes no  no  no  no  no  no  no  no  no  no  yes no  yes no  yes</span></span>
<span id="cb589-11"><a href="trichrome-staining.html#cb589-11"></a><span class="co">## [20] no </span></span>
<span id="cb589-12"><a href="trichrome-staining.html#cb589-12"></a><span class="co">## Levels: no yes</span></span>
<span id="cb589-13"><a href="trichrome-staining.html#cb589-13"></a><span class="kw">save</span>(vstEpInf, <span class="dt">file=</span><span class="st">&quot;figure-outputs/temp_test.RData&quot;</span>)</span>
<span id="cb589-14"><a href="trichrome-staining.html#cb589-14"></a></span>
<span id="cb589-15"><a href="trichrome-staining.html#cb589-15"></a>ax1=<span class="kw">colnames</span>(vstEpInf)</span>
<span id="cb589-16"><a href="trichrome-staining.html#cb589-16"></a></span>
<span id="cb589-17"><a href="trichrome-staining.html#cb589-17"></a>t2=<span class="kw">match</span>(<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">substr</span>( ax1, <span class="dv">1</span>, <span class="kw">nchar</span>(ax1)<span class="op">-</span><span class="dv">2</span>)), <span class="kw">rownames</span>(df.Spatial))</span>
<span id="cb589-18"><a href="trichrome-staining.html#cb589-18"></a></span>
<span id="cb589-19"><a href="trichrome-staining.html#cb589-19"></a>newTab=<span class="kw">data.frame</span>(<span class="dt">trichrome=</span>df.Spatial[<span class="kw">na.omit</span>(t2), <span class="st">&quot;Trichrome&quot;</span>], </span>
<span id="cb589-20"><a href="trichrome-staining.html#cb589-20"></a>             <span class="dt">inf=</span>vstEpInf<span class="op">$</span>Inflammation2[<span class="op">-</span><span class="kw">which</span>(<span class="kw">is.na</span>(t2))],</span>
<span id="cb589-21"><a href="trichrome-staining.html#cb589-21"></a>             <span class="dt">names=</span>ax1[<span class="op">-</span><span class="kw">which</span>(<span class="kw">is.na</span>(t2))],</span>
<span id="cb589-22"><a href="trichrome-staining.html#cb589-22"></a>             <span class="dt">names2=</span><span class="kw">rownames</span>(df.Spatial)[<span class="kw">na.omit</span>(t2)])</span>
<span id="cb589-23"><a href="trichrome-staining.html#cb589-23"></a><span class="kw">wilcox.test</span>(newTab<span class="op">$</span>trichrome[newTab<span class="op">$</span>inf<span class="op">==</span><span class="st">&quot;grow&quot;</span>], newTab<span class="op">$</span>trichrome[newTab<span class="op">$</span>inf<span class="op">!=</span><span class="st">&quot;grow&quot;</span>])</span>
<span id="cb589-24"><a href="trichrome-staining.html#cb589-24"></a><span class="co">## </span></span>
<span id="cb589-25"><a href="trichrome-staining.html#cb589-25"></a><span class="co">##  Wilcoxon rank sum exact test</span></span>
<span id="cb589-26"><a href="trichrome-staining.html#cb589-26"></a><span class="co">## </span></span>
<span id="cb589-27"><a href="trichrome-staining.html#cb589-27"></a><span class="co">## data:  newTab$trichrome[newTab$inf == &quot;grow&quot;] and newTab$trichrome[newTab$inf != &quot;grow&quot;]</span></span>
<span id="cb589-28"><a href="trichrome-staining.html#cb589-28"></a><span class="co">## W = 6, p-value = 0.04507</span></span>
<span id="cb589-29"><a href="trichrome-staining.html#cb589-29"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb589-30"><a href="trichrome-staining.html#cb589-30"></a><span class="co">#pdf(&quot;figure-outputs/EXT5I_trichrome.pdf&quot;, height=5, width=4)</span></span>
<span id="cb589-31"><a href="trichrome-staining.html#cb589-31"></a><span class="kw">ggplot</span>(newTab, <span class="kw">aes</span>(<span class="dt">y=</span>trichrome,<span class="dt">x=</span>inf))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">theme_bw</span>()</span></code></pre></div>
<div class="figure"><span id="fig:Ext5i"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext5i-1.png" alt="association with hyperinflammation" width="672" />
<p class="caption">
Figure 25.6: association with hyperinflammation
</p>
</div>
<p>Also check the encapsulated status?</p>
<div class="sourceCode" id="cb590"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb590-1"><a href="trichrome-staining.html#cb590-1"></a>newTab<span class="op">$</span>trichrome_pc=Cdata<span class="op">$</span>Trichrome_encapsulation.[<span class="kw">match</span>(newTab<span class="op">$</span>names2, <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb590-2"><a href="trichrome-staining.html#cb590-2"></a>newTab<span class="op">$</span>trichrome_pc[<span class="kw">which</span>(newTab<span class="op">$</span>trichrome_pc<span class="op">==</span><span class="st">&quot;lots&quot;</span>)]=<span class="dv">1</span></span>
<span id="cb590-3"><a href="trichrome-staining.html#cb590-3"></a>newTab<span class="op">$</span>trichrome_pc=<span class="kw">as.numeric</span>(newTab<span class="op">$</span>trichrome_pc)</span>
<span id="cb590-4"><a href="trichrome-staining.html#cb590-4"></a><span class="co">#pdf(&quot;figure-outputs/EXT5I_trichrome_pt2.pdf&quot;, height=5, width=4)</span></span>
<span id="cb590-5"><a href="trichrome-staining.html#cb590-5"></a><span class="kw">ggplot</span>(newTab, <span class="kw">aes</span>(<span class="dt">y=</span>trichrome_pc,<span class="dt">x=</span>inf))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_jitter</span>(<span class="dt">col=</span><span class="st">&quot;grey&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-172-1.png" width="672" /></p>
<div class="sourceCode" id="cb591"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb591-1"><a href="trichrome-staining.html#cb591-1"></a></span>
<span id="cb591-2"><a href="trichrome-staining.html#cb591-2"></a><span class="kw">wilcox.test</span>(newTab<span class="op">$</span>trichrome_pc[newTab<span class="op">$</span>inf<span class="op">==</span><span class="st">&quot;grow&quot;</span>], newTab<span class="op">$</span>trichrome_pc[newTab<span class="op">$</span>inf<span class="op">!=</span><span class="st">&quot;grow&quot;</span>])</span>
<span id="cb591-3"><a href="trichrome-staining.html#cb591-3"></a><span class="co">## </span></span>
<span id="cb591-4"><a href="trichrome-staining.html#cb591-4"></a><span class="co">##  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb591-5"><a href="trichrome-staining.html#cb591-5"></a><span class="co">## </span></span>
<span id="cb591-6"><a href="trichrome-staining.html#cb591-6"></a><span class="co">## data:  newTab$trichrome_pc[newTab$inf == &quot;grow&quot;] and newTab$trichrome_pc[newTab$inf != &quot;grow&quot;]</span></span>
<span id="cb591-7"><a href="trichrome-staining.html#cb591-7"></a><span class="co">## W = 42.5, p-value = 0.01389</span></span>
<span id="cb591-8"><a href="trichrome-staining.html#cb591-8"></a><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb591-9"><a href="trichrome-staining.html#cb591-9"></a>DT<span class="op">::</span><span class="kw">datatable</span>(newTab, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb591-10"><a href="trichrome-staining.html#cb591-10"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div id="htmlwidget-9b2bd51e6cf0b33441f0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9b2bd51e6cf0b33441f0">{"x":{"filter":"none","extensions":["Buttons"],"data":[[0.176618842694733,null,0.133181297679202,0.374043581170937,0.161619227345989,0.124598668056126,0.276814784264216,0.15485032053922,null,0.350950798461984,0.126077261157979,0.433978944576645,0.154296069680264,0.114344669757582,0.445971508497105,null],["inf","grow","grow","inf","grow","grow","stab","stab","stab","stab","grow","inf","grow","inf","inf","grow"],["10L_D_Ep","10R_BL_Ep","11L_B_Ep","11N_D_Ep","11R_D_Ep","11R_C_Ep","12L_D_Ep","14N_C_Ep","16L_C_Ep","17N_D_Ep","2N__Ep","3N_B_Ep","3R_C_Ep","6R_B_Ep","8L_D_Ep","8R_CU_Ep"],["10LD","10RBL","11LB","11ND","11RD","11RC","12LD","14NC","16LC","17ND","2N","3NB","3RC","6RB","8LD","8RCU"],[0,null,1,0,1,1,0,0.25,null,0,0.5,0,0,0.25,0.25,1]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>trichrome<\/th>\n      <th>inf<\/th>\n      <th>names<\/th>\n      <th>names2<\/th>\n      <th>trichrome_pc<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[0,4]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutations-in-progression-cohort.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cnv-calling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
