<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 20 Whole Genome Sequencing Mutations | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 20 Whole Genome Sequencing Mutations | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 20 Whole Genome Sequencing Mutations | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-02-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="bcr-clonotype-analysis.html"/>
<link rel="next" href="mutations-in-rna.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.3</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.4</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-any-treatment-vs-control"><i class="fa fa-check"></i><b>12.2</b> DESeq: any treatment vs control</a><ul>
<li class="chapter" data-level="12.2.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#epcam-samples"><i class="fa fa-check"></i><b>12.2.1</b> EPCAM samples</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.2</b> Signature: Lum cases non-inflammatory: growing vs stable</a><ul>
<li class="chapter" data-level="15.2.1" data-path="associate-epcam-inflammatory-with-survival.html"><a href="associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.2.1</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>16</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="16.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>16.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="16.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>16.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>16.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="16.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>16.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>16.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>16.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="16.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>16.2.2</b> GSEA</a></li>
<li class="chapter" data-level="16.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>16.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>17</b> Signature analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>17.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="17.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>17.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="17.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>17.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="17.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>17.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="17.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>17.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>18</b> Immune estimation</a><ul>
<li class="chapter" data-level="18.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>18.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="18.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>18.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="18.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>18.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="18.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>18.4</b> Progression cohort</a></li>
<li class="chapter" data-level="18.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>18.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="18.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>18.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="18.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>18.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>18.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>19</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="19.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>19.1</b> Processing summary</a></li>
<li class="chapter" data-level="19.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>19.2</b> Summary Stats</a></li>
<li class="chapter" data-level="19.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>19.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="19.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>19.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="19.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>19.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="19.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>19.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>20</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="20.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>20.1</b> Data</a><ul>
<li class="chapter" data-level="20.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>20.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="20.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>20.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>20.2</b> Plots</a><ul>
<li class="chapter" data-level="20.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>20.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="20.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>20.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>20.3</b> Coding variants</a><ul>
<li class="chapter" data-level="20.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>20.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>20.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="20.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>20.4.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>20.5</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>21</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="21.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>21.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="21.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>21.2</b> Load files</a></li>
<li class="chapter" data-level="21.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>21.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="21.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>21.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="21.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>21.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>22</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>22.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>22.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>22.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>23</b> Trichrome staining</a><ul>
<li class="chapter" data-level="23.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>23.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="23.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>23.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="23.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>23.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="23.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>23.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>24</b> CNV calling</a><ul>
<li class="chapter" data-level="24.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>24.1</b> Sample data</a></li>
<li class="chapter" data-level="24.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>24.2</b> Data Summary</a><ul>
<li class="chapter" data-level="24.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>24.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="24.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>24.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="24.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>24.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>25</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="whole-genome-sequencing-mutations" class="section level1">
<h1><span class="header-section-number">Chapter 20</span> Whole Genome Sequencing Mutations</h1>
<p>Whole genome sequencing was performed on XX samples by the new york genome centre.
The pipeline involves:</p>
<p><span style="color:red"> Insert information on the NYGC pipeline here </span></p>
<div id="data" class="section level2">
<h2><span class="header-section-number">20.1</span> Data</h2>
<p>In this section, we load the maf output files and create lists which</p>
<ul>
<li>filter out coding mutations</li>
<li>Determine mutational signatures by:
<ul>
<li>Creating bed files with genomic locations of the SNP +- 1 base location</li>
<li>Extracting the nucleotide sequence using bedtools</li>
<li>Convert all mutations to C &gt; N and T &gt; N to obtain 96 mutational signatures</li>
<li>run with single sample SigProfiler to obtain the signature scores</li>
</ul></li>
<li>Annotate data with common human variants</li>
</ul>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="whole-genome-sequencing-mutations.html#cb391-1"></a><span class="co"># Load and save for the first time here</span></span>
<span id="cb391-2"><a href="whole-genome-sequencing-mutations.html#cb391-2"></a><span class="co"># redo</span></span>
<span id="cb391-3"><a href="whole-genome-sequencing-mutations.html#cb391-3"></a></span>
<span id="cb391-4"><a href="whole-genome-sequencing-mutations.html#cb391-4"></a>mafdir=<span class="st">&#39;../data/wholegenome_mutations/maf/&#39;</span></span>
<span id="cb391-5"><a href="whole-genome-sequencing-mutations.html#cb391-5"></a>x1=<span class="kw">dir</span>(mafdir, <span class="st">&#39;annotated.maf&#39;</span>)</span>
<span id="cb391-6"><a href="whole-genome-sequencing-mutations.html#cb391-6"></a><span class="co">#names2=sapply(strsplit( x1,&quot;--&quot;), function(x) x[1])</span></span>
<span id="cb391-7"><a href="whole-genome-sequencing-mutations.html#cb391-7"></a>AllMutWGS=<span class="kw">lapply</span>(x1, <span class="cf">function</span>(x) <span class="kw">read.delim</span>(<span class="kw">paste</span>(mafdir, x, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">skip =</span> <span class="dv">1</span>))</span>
<span id="cb391-8"><a href="whole-genome-sequencing-mutations.html#cb391-8"></a><span class="kw">names</span>(AllMutWGS)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>( x1,<span class="st">&quot;--&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb391-9"><a href="whole-genome-sequencing-mutations.html#cb391-9"></a><span class="co"># # ## summary of the number of mutations?</span></span>
<span id="cb391-10"><a href="whole-genome-sequencing-mutations.html#cb391-10"></a><span class="kw">print</span>(<span class="st">&#39;no of mutations&#39;</span>)</span>
<span id="cb391-11"><a href="whole-genome-sequencing-mutations.html#cb391-11"></a><span class="co">## [1] &quot;no of mutations&quot;</span></span>
<span id="cb391-12"><a href="whole-genome-sequencing-mutations.html#cb391-12"></a>TotCount=<span class="kw">lapply</span>(AllMutWGS, nrow)</span>
<span id="cb391-13"><a href="whole-genome-sequencing-mutations.html#cb391-13"></a></span>
<span id="cb391-14"><a href="whole-genome-sequencing-mutations.html#cb391-14"></a>SearchVars=<span class="kw">c</span>(<span class="st">&quot;Splice_Site&quot;</span>, <span class="st">&quot;Nonstop_Mutation&quot;</span>, <span class="st">&quot;Nonsense_Mutation&quot;</span>, <span class="st">&quot;Missense_Mutation&quot;</span>, <span class="st">&quot;Frame_Shift_Ins&quot;</span>, <span class="st">&quot;Frame_Shift_Del&quot;</span>)</span>
<span id="cb391-15"><a href="whole-genome-sequencing-mutations.html#cb391-15"></a></span>
<span id="cb391-16"><a href="whole-genome-sequencing-mutations.html#cb391-16"></a>AllMutWGS=<span class="kw">lapply</span>(AllMutWGS, <span class="cf">function</span>(x) {x<span class="op">$</span>VAF&lt;-x<span class="op">$</span>t_alt_count<span class="op">/</span>(x<span class="op">$</span>t_alt_count<span class="op">+</span>x<span class="op">$</span>t_ref_count); x})</span>
<span id="cb391-17"><a href="whole-genome-sequencing-mutations.html#cb391-17"></a></span>
<span id="cb391-18"><a href="whole-genome-sequencing-mutations.html#cb391-18"></a>AllMutWGScoding=<span class="kw">lapply</span>(AllMutWGS, <span class="cf">function</span>(x) x[ x<span class="op">$</span>Variant_Classification<span class="op">%in%</span>SearchVars, ])</span>
<span id="cb391-19"><a href="whole-genome-sequencing-mutations.html#cb391-19"></a>A2=<span class="kw">sapply</span>(AllMutWGScoding, nrow)</span>
<span id="cb391-20"><a href="whole-genome-sequencing-mutations.html#cb391-20"></a></span>
<span id="cb391-21"><a href="whole-genome-sequencing-mutations.html#cb391-21"></a><span class="co"># Save the combined mutation lists</span></span>
<span id="cb391-22"><a href="whole-genome-sequencing-mutations.html#cb391-22"></a><span class="kw">save</span>(AllMutWGS, AllMutWGScoding, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/WGS_mutations_coding_%s.RData&quot;</span>, <span class="kw">Sys.Date</span>()))</span>
<span id="cb391-23"><a href="whole-genome-sequencing-mutations.html#cb391-23"></a></span>
<span id="cb391-24"><a href="whole-genome-sequencing-mutations.html#cb391-24"></a><span class="co">## filters on VAF and depth?</span></span>
<span id="cb391-25"><a href="whole-genome-sequencing-mutations.html#cb391-25"></a>AllMutWGScoding=<span class="kw">lapply</span>(AllMutWGScoding, <span class="cf">function</span>(x) {x&lt;-x[<span class="kw">which</span>(x<span class="op">$</span>VAF<span class="op">&gt;=</span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>x<span class="op">$</span>n_ref_count<span class="op">&gt;</span><span class="dv">8</span> <span class="op">&amp;</span></span>
<span id="cb391-26"><a href="whole-genome-sequencing-mutations.html#cb391-26"></a><span class="st">                                                                    </span>(x<span class="op">$</span>t_alt_count<span class="op">+</span>x<span class="op">$</span>t_ref_count)<span class="op">&gt;</span><span class="dv">14</span>), ]; x})</span>
<span id="cb391-27"><a href="whole-genome-sequencing-mutations.html#cb391-27"></a></span>
<span id="cb391-28"><a href="whole-genome-sequencing-mutations.html#cb391-28"></a><span class="co">#load(&#39;data/wholegenome_mutations/WGS_mutations_coding_2020-10-14.RData&#39;)</span></span>
<span id="cb391-29"><a href="whole-genome-sequencing-mutations.html#cb391-29"></a>MeltWGSCoding=<span class="kw">melt</span>(AllMutWGScoding, <span class="dt">measure.vars=</span><span class="kw">c</span>(<span class="st">&quot;Tumor_Sample_Barcode&quot;</span>))</span>
<span id="cb391-30"><a href="whole-genome-sequencing-mutations.html#cb391-30"></a>MeltWGSCoding<span class="op">$</span>Hum_Gene=SymHum2Rat<span class="op">$</span>HGNC.symbol[<span class="kw">match</span>(MeltWGSCoding<span class="op">$</span>Gene_Symbol, SymHum2Rat<span class="op">$</span>RGD.symbol)]</span>
<span id="cb391-31"><a href="whole-genome-sequencing-mutations.html#cb391-31"></a>MeltWGSCoding<span class="op">$</span>Hum_Gene[<span class="kw">which</span>(<span class="kw">is.na</span>(MeltWGSCoding<span class="op">$</span>Hum_Gene))]=<span class="kw">toupper</span>(MeltWGSCoding<span class="op">$</span>Gene_Symbol[<span class="kw">which</span>(<span class="kw">is.na</span>(MeltWGSCoding<span class="op">$</span>Hum_Gene))])</span>
<span id="cb391-32"><a href="whole-genome-sequencing-mutations.html#cb391-32"></a>MutAllCosmic=<span class="kw">lapply</span>(AllMutWGScoding, <span class="cf">function</span>(x) x[<span class="kw">which</span>(x<span class="op">$</span>Gene_Symbol<span class="op">%in%</span>RatCosmic), ])</span>
<span id="cb391-33"><a href="whole-genome-sequencing-mutations.html#cb391-33"></a>MeltCosmic=<span class="kw">melt</span>(MutAllCosmic, <span class="dt">measure.vars=</span><span class="kw">c</span>(<span class="st">&quot;Tumor_Sample_Barcode&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb392"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb392-1"><a href="whole-genome-sequencing-mutations.html#cb392-1"></a><span class="co"># Find complete numbers here, so we don&#39;t have to go back </span></span>
<span id="cb392-2"><a href="whole-genome-sequencing-mutations.html#cb392-2"></a>TotCount=<span class="kw">sapply</span>(AllMutWGS, nrow)</span>
<span id="cb392-3"><a href="whole-genome-sequencing-mutations.html#cb392-3"></a>SearchVars=<span class="kw">c</span>(<span class="st">&quot;Splice_Site&quot;</span>, <span class="st">&quot;Nonstop_Mutation&quot;</span>, <span class="st">&quot;Nonsense_Mutation&quot;</span>, <span class="st">&quot;Missense_Mutation&quot;</span>, <span class="st">&quot;Frame_Shift_Ins&quot;</span>, <span class="st">&quot;Frame_Shift_Del&quot;</span>)</span>
<span id="cb392-4"><a href="whole-genome-sequencing-mutations.html#cb392-4"></a></span>
<span id="cb392-5"><a href="whole-genome-sequencing-mutations.html#cb392-5"></a><span class="kw">head</span>(TotCount)</span>
<span id="cb392-6"><a href="whole-genome-sequencing-mutations.html#cb392-6"></a><span class="co">##  C1R-tumor  C2N-tumor C2R1-tumor C2R2-tumor C2R3-tumor  C4P-tumor </span></span>
<span id="cb392-7"><a href="whole-genome-sequencing-mutations.html#cb392-7"></a><span class="co">##     240153     262425     235677     222717     238324     258677</span></span></code></pre></div>
<div id="extract-mutation-signatures" class="section level3">
<h3><span class="header-section-number">20.1.1</span> Extract mutation signatures</h3>
<p>Annotate the files so that the nucleotide codon can be extracted based on the genome file.
For this, use all mutations (not just coding ones)
See the bash file extract-trinucleotide-forSigProfiler.sh to see the output of this.
The output table $fout3 should have 6 columns with the genomic location, the reference and alt allele and codon context.</p>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="whole-genome-sequencing-mutations.html#cb393-1"></a>TBed=<span class="kw">lapply</span>(AllMutWGScoding, <span class="cf">function</span>(x) <span class="kw">cbind</span>(<span class="kw">as.character</span>(x<span class="op">$</span>Chromosome), x<span class="op">$</span>Start_Position<span class="dv">-2</span>, x<span class="op">$</span>End_Position<span class="op">+</span><span class="dv">1</span>, <span class="kw">as.character</span>(x<span class="op">$</span>Reference_Allele), <span class="kw">as.character</span>(x<span class="op">$</span>Tumor_Seq_Allele2)))</span>
<span id="cb393-2"><a href="whole-genome-sequencing-mutations.html#cb393-2"></a>TBed=<span class="kw">lapply</span>(TBed, <span class="cf">function</span>(x) x[<span class="kw">which</span>(<span class="kw">nchar</span>(x[ ,<span class="dv">4</span>])<span class="op">==</span><span class="dv">1</span>), ])</span>
<span id="cb393-3"><a href="whole-genome-sequencing-mutations.html#cb393-3"></a><span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(TBed), <span class="cf">function</span>(x) <span class="kw">write.table</span>(TBed[[x]], <span class="dt">file=</span><span class="kw">paste</span>(<span class="kw">names</span>(TBed)[x], <span class="st">&quot;coding_tricodon.bed&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="dt">col.names=</span>F, <span class="dt">row.names =</span> F, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> F))</span>
<span id="cb393-4"><a href="whole-genome-sequencing-mutations.html#cb393-4"></a></span>
<span id="cb393-5"><a href="whole-genome-sequencing-mutations.html#cb393-5"></a><span class="co">## bedtools command</span></span>
<span id="cb393-6"><a href="whole-genome-sequencing-mutations.html#cb393-6"></a>bedtools getfasta <span class="op">-</span>fi ..<span class="op">/</span>Genomic_Files<span class="op">/</span>rn6.fa <span class="op">-</span>bed <span class="op">$</span>f <span class="op">-</span>tab <span class="op">-</span>fo <span class="op">$</span>fout</span>
<span id="cb393-7"><a href="whole-genome-sequencing-mutations.html#cb393-7"></a><span class="kw">f3</span>() { awk <span class="st">&#39;{print $1, $2, $3, $4 ,$5}&#39;</span> <span class="st">&quot;$1&quot;</span>; }; paste <span class="op">&lt;</span>(f3 <span class="op">$</span>f) <span class="op">&lt;</span>(f3 <span class="op">$</span>fout) <span class="op">&gt;</span><span class="st"> </span><span class="er">$</span>fout2</span>
<span id="cb393-8"><a href="whole-genome-sequencing-mutations.html#cb393-8"></a>awk <span class="st">&#39;{if (length($7)==3) print $1, $2, $3, $4, $5, toupper($7)}&#39;</span> <span class="op">$</span>fout2 <span class="op">&gt;</span><span class="st"> </span><span class="er">$</span>fout3</span>
<span id="cb393-9"><a href="whole-genome-sequencing-mutations.html#cb393-9"></a><span class="co">## see the bash file extract-trinucleotide-forSigProfiler.sh to see the output of this.</span></span></code></pre></div>
<p>Following running the above bash script, we can run the following extract the tri-nucleotide sequence</p>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="whole-genome-sequencing-mutations.html#cb394-1"></a>FFiles=<span class="kw">dir</span>(<span class="st">&quot;../data/tricodon_mut_sigs/extract_context/&quot;</span>, <span class="st">&quot;*.txt&quot;</span>,<span class="dt">full.names=</span>T)</span>
<span id="cb394-2"><a href="whole-genome-sequencing-mutations.html#cb394-2"></a>OutputList=<span class="kw">sapply</span>(FFiles, <span class="cf">function</span>(x) <span class="kw">FindTriNucleotideContext</span>(x))</span>
<span id="cb394-3"><a href="whole-genome-sequencing-mutations.html#cb394-3"></a>TableOut=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(OutputList), <span class="cf">function</span>(x) <span class="kw">unlist</span>(OutputList[<span class="dv">3</span>, x]))</span>
<span id="cb394-4"><a href="whole-genome-sequencing-mutations.html#cb394-4"></a><span class="co">## make a plot here</span></span>
<span id="cb394-5"><a href="whole-genome-sequencing-mutations.html#cb394-5"></a><span class="co"># condense the type of mutations here</span></span>
<span id="cb394-6"><a href="whole-genome-sequencing-mutations.html#cb394-6"></a>a1=<span class="kw">basename</span>(FFiles)</span>
<span id="cb394-7"><a href="whole-genome-sequencing-mutations.html#cb394-7"></a>a2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(a1, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb394-8"><a href="whole-genome-sequencing-mutations.html#cb394-8"></a>xall=<span class="kw">as.character</span>(<span class="kw">unlist</span>(OutputList[<span class="dv">1</span>, <span class="dv">1</span>]))</span>
<span id="cb394-9"><a href="whole-genome-sequencing-mutations.html#cb394-9"></a>TestAll=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(TableOut), <span class="cf">function</span>(x) <span class="kw">unlist</span>(<span class="kw">by</span>(TableOut[ ,x], xall, sum)))</span>
<span id="cb394-10"><a href="whole-genome-sequencing-mutations.html#cb394-10"></a>TestAll2=<span class="kw">t</span>(TestAll)<span class="op">/</span><span class="kw">colSums</span>(TestAll)</span>
<span id="cb394-11"><a href="whole-genome-sequencing-mutations.html#cb394-11"></a><span class="kw">rownames</span>(TestAll2)=<span class="kw">sapply</span>(a2, <span class="cf">function</span>(x) Cdata<span class="op">$</span>NewID[<span class="kw">grep</span>(x,  Cdata<span class="op">$</span>WGS)])</span>
<span id="cb394-12"><a href="whole-genome-sequencing-mutations.html#cb394-12"></a>TestAll2melt=<span class="kw">melt</span>(TestAll2)</span>
<span id="cb394-13"><a href="whole-genome-sequencing-mutations.html#cb394-13"></a></span>
<span id="cb394-14"><a href="whole-genome-sequencing-mutations.html#cb394-14"></a><span class="co">#pdf(&quot;~/Desktop/Frequency_mutations_6types.pdf&quot;, height=5, width=6)</span></span>
<span id="cb394-15"><a href="whole-genome-sequencing-mutations.html#cb394-15"></a><span class="kw">ggplot</span>(TestAll2melt, <span class="kw">aes</span>(<span class="dt">x=</span>Var1, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var2))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Frequency of Mutation&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:1j"></span>
<img src="RatDCIS-bookdown_files/figure-html/1j-1.png" alt="Frequency of different mutational codons" width="672" />
<p class="caption">
Figure 20.1: Frequency of different mutational codons
</p>
</div>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="whole-genome-sequencing-mutations.html#cb395-1"></a><span class="co">#dev.off()</span></span>
<span id="cb395-2"><a href="whole-genome-sequencing-mutations.html#cb395-2"></a></span>
<span id="cb395-3"><a href="whole-genome-sequencing-mutations.html#cb395-3"></a></span>
<span id="cb395-4"><a href="whole-genome-sequencing-mutations.html#cb395-4"></a><span class="kw">write.csv</span>(TestAll2melt, <span class="dt">file=</span><span class="st">&quot;nature-tables/1j.csv&quot;</span>)</span>
<span id="cb395-5"><a href="whole-genome-sequencing-mutations.html#cb395-5"></a></span>
<span id="cb395-6"><a href="whole-genome-sequencing-mutations.html#cb395-6"></a>TableOut=<span class="kw">cbind</span>(<span class="kw">as.character</span>(<span class="kw">unlist</span>(OutputList[<span class="dv">1</span>, <span class="dv">1</span>])), <span class="kw">as.character</span>(<span class="kw">unlist</span>(OutputList[<span class="dv">2</span>, <span class="dv">1</span>])), TableOut)</span>
<span id="cb395-7"><a href="whole-genome-sequencing-mutations.html#cb395-7"></a><span class="kw">colnames</span>(TableOut)=<span class="kw">c</span>(<span class="st">&quot;MutationType&quot;</span>,<span class="st">&quot;Nutrinucleotide&quot;</span>, <span class="kw">paste</span>(<span class="st">&quot;BRCA&quot;</span>, a2, <span class="dt">sep=</span><span class="st">&quot;::&quot;</span>))</span>
<span id="cb395-8"><a href="whole-genome-sequencing-mutations.html#cb395-8"></a><span class="co">#write.csv(TableOut, file=&quot;../data/tricodon_mut_sigs/output4SigProfiler-jan2021.csv&quot;, sep = &quot;,&quot;, row.names = F)</span></span></code></pre></div>
<p>This is then run in python (sigprofiler) with the following code:</p>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="whole-genome-sequencing-mutations.html#cb396-1"></a>module load gcc</span>
<span id="cb396-2"><a href="whole-genome-sequencing-mutations.html#cb396-2"></a>module load python<span class="op">/</span><span class="dv">3</span>.<span class="fl">7.4</span></span>
<span id="cb396-3"><a href="whole-genome-sequencing-mutations.html#cb396-3"></a>python3</span>
<span id="cb396-4"><a href="whole-genome-sequencing-mutations.html#cb396-4"></a>import SigProfilerExtractor</span>
<span id="cb396-5"><a href="whole-genome-sequencing-mutations.html#cb396-5"></a>from sigproSS import spss_pcwag <span class="co">## (spss uses only vcf files)</span></span>
<span id="cb396-6"><a href="whole-genome-sequencing-mutations.html#cb396-6"></a><span class="kw">spss_pcwag.single_sample_pcwag</span>(<span class="st">&quot;output4SigProfiler-jan2021.csv&quot;</span>, <span class="dt">output=</span><span class="st">&quot;output-jan2021&quot;</span>)</span></code></pre></div>
<p>The results from this analysis are shown below</p>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="whole-genome-sequencing-mutations.html#cb397-1"></a>SBSprof=<span class="kw">read.delim</span>(<span class="st">&quot;../data/tricodon_mut_sigs/output-jan2021-allsamples/sig_activities.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb397-2"><a href="whole-genome-sequencing-mutations.html#cb397-2"></a>SBSprofb=SBSprof[ ,<span class="op">-</span><span class="dv">1</span>]<span class="op">/</span><span class="kw">rowSums</span>(SBSprof[ ,<span class="op">-</span><span class="dv">1</span>])</span>
<span id="cb397-3"><a href="whole-genome-sequencing-mutations.html#cb397-3"></a>SBSprofb=SBSprofb[ ,<span class="op">-</span><span class="kw">which</span>(<span class="kw">colSums</span>(SBSprofb)<span class="op">==</span><span class="dv">0</span>)]</span>
<span id="cb397-4"><a href="whole-genome-sequencing-mutations.html#cb397-4"></a><span class="kw">rownames</span>(SBSprofb)=SBSprof[ ,<span class="dv">1</span>]</span>
<span id="cb397-5"><a href="whole-genome-sequencing-mutations.html#cb397-5"></a></span>
<span id="cb397-6"><a href="whole-genome-sequencing-mutations.html#cb397-6"></a>SBSmelt=<span class="kw">melt</span>(<span class="kw">cbind</span>(SBSprofb[ , <span class="dv">-1</span>], <span class="dt">id=</span><span class="kw">rownames</span>(SBSprofb)))</span>
<span id="cb397-7"><a href="whole-genome-sequencing-mutations.html#cb397-7"></a>SBSsummary &lt;-<span class="st"> </span><span class="kw">aggregate</span>(. <span class="op">~</span><span class="st"> </span>variable, mean, <span class="dt">data=</span>SBSmelt)</span>
<span id="cb397-8"><a href="whole-genome-sequencing-mutations.html#cb397-8"></a></span>
<span id="cb397-9"><a href="whole-genome-sequencing-mutations.html#cb397-9"></a><span class="kw">ggplot</span>(SBSmelt, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>value, <span class="dt">col=</span>variable))<span class="op">+</span><span class="kw">geom_jitter</span>()<span class="op">+</span><span class="kw">geom_crossbar</span>(<span class="dt">data=</span>SBSsummary , <span class="kw">aes</span>(<span class="dt">ymin =</span> value, <span class="dt">ymax =</span> value),<span class="dt">size=</span><span class="fl">0.5</span>,<span class="dt">col=</span><span class="st">&quot;black&quot;</span>, <span class="dt">width =</span> <span class="fl">.7</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Percentage of mutations&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Ext1j"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext1j-1.png" alt="Alexandrov mutational signatures" width="672" />
<p class="caption">
Figure 20.2: Alexandrov mutational signatures
</p>
</div>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="whole-genome-sequencing-mutations.html#cb398-1"></a></span>
<span id="cb398-2"><a href="whole-genome-sequencing-mutations.html#cb398-2"></a><span class="kw">write.csv</span>(SBSmelt, <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext1j.csv&quot;</span>)</span></code></pre></div>
<p>We can see whether any of these associate with tumor size for example:</p>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="whole-genome-sequencing-mutations.html#cb399-1"></a><span class="co">## Assign with growth data or tumor size?</span></span>
<span id="cb399-2"><a href="whole-genome-sequencing-mutations.html#cb399-2"></a>tname=<span class="kw">substr</span>(SBSprof<span class="op">$</span>Cancer.Types, <span class="dv">7</span>, <span class="dv">20</span>)</span>
<span id="cb399-3"><a href="whole-genome-sequencing-mutations.html#cb399-3"></a>idx=<span class="kw">sapply</span>(tname, <span class="cf">function</span>(x) <span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS))</span>
<span id="cb399-4"><a href="whole-genome-sequencing-mutations.html#cb399-4"></a>sizeInfo=Cdata<span class="op">$</span>Tumor.diameter.sac.mm[idx]</span>
<span id="cb399-5"><a href="whole-genome-sequencing-mutations.html#cb399-5"></a></span>
<span id="cb399-6"><a href="whole-genome-sequencing-mutations.html#cb399-6"></a><span class="co">#par(mfrow=c(2,1))</span></span>
<span id="cb399-7"><a href="whole-genome-sequencing-mutations.html#cb399-7"></a>cor1=<span class="kw">cor.test</span>(SBSprofb<span class="op">$</span>SBS32<span class="op">+</span>SBSprofb<span class="op">$</span>SBS11, sizeInfo, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb399-8"><a href="whole-genome-sequencing-mutations.html#cb399-8"></a>cor2=<span class="kw">cor.test</span>(SBSprofb<span class="op">$</span>SBS32[<span class="kw">which</span>(SBSprofb<span class="op">$</span>SBS32<span class="op">&gt;</span><span class="dv">0</span>)], sizeInfo[<span class="kw">which</span>(SBSprofb<span class="op">$</span>SBS32<span class="op">&gt;</span><span class="dv">0</span>)], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb399-9"><a href="whole-genome-sequencing-mutations.html#cb399-9"></a>cor3=<span class="kw">cor.test</span>(SBSprofb<span class="op">$</span>SBS11[<span class="kw">which</span>(SBSprofb<span class="op">$</span>SBS11<span class="op">&gt;</span><span class="dv">0</span>)], sizeInfo[<span class="kw">which</span>(SBSprofb<span class="op">$</span>SBS11<span class="op">&gt;</span><span class="dv">0</span>)], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb399-10"><a href="whole-genome-sequencing-mutations.html#cb399-10"></a><span class="kw">plot</span>(SBSprofb<span class="op">$</span>SBS32<span class="op">+</span>SBSprofb<span class="op">$</span>SBS11, sizeInfo, <span class="dt">xlab=</span><span class="st">&quot;SBS 32 or 11 score&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;tumor size&quot;</span>,</span>
<span id="cb399-11"><a href="whole-genome-sequencing-mutations.html#cb399-11"></a>     <span class="dt">col=</span><span class="kw">ifelse</span>(SBSprofb<span class="op">$</span>SBS11<span class="op">==</span><span class="dv">0</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;association between signature score and size, p=%s&quot;</span>, <span class="kw">round</span>(cor1<span class="op">$</span>p.value, <span class="dv">2</span>)))</span>
<span id="cb399-12"><a href="whole-genome-sequencing-mutations.html#cb399-12"></a><span class="kw">legend</span>(<span class="st">&quot;topleft&quot;</span>,<span class="kw">c</span>(<span class="kw">paste</span>(<span class="st">&quot;SB32&quot;</span>, <span class="kw">round</span>(cor2<span class="op">$</span>p.value,<span class="dv">2</span>)), <span class="kw">paste</span>(<span class="st">&quot;SB11&quot;</span>,<span class="kw">round</span>(cor3<span class="op">$</span>p.value,<span class="dv">2</span>))), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">pch=</span><span class="dv">19</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-132-1.png" width="672" /></p>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="whole-genome-sequencing-mutations.html#cb400-1"></a></span>
<span id="cb400-2"><a href="whole-genome-sequencing-mutations.html#cb400-2"></a><span class="co">## Associations version 2</span></span>
<span id="cb400-3"><a href="whole-genome-sequencing-mutations.html#cb400-3"></a>SBSprofb<span class="op">$</span>Age=Cdata<span class="op">$</span>Time.NMU2Sac.days[<span class="kw">sapply</span>(<span class="kw">substr</span>(<span class="kw">rownames</span>(SBSprofb), <span class="dv">7</span>, <span class="dv">30</span>), <span class="cf">function</span>(x) <span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS))]</span>
<span id="cb400-4"><a href="whole-genome-sequencing-mutations.html#cb400-4"></a>SBSprofb<span class="op">$</span>Cohort=Cdata<span class="op">$</span>Char.Cohort.Batch[<span class="kw">sapply</span>(<span class="kw">substr</span>(<span class="kw">rownames</span>(SBSprofb), <span class="dv">7</span>, <span class="dv">30</span>), <span class="cf">function</span>(x) <span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS))]</span>
<span id="cb400-5"><a href="whole-genome-sequencing-mutations.html#cb400-5"></a></span>
<span id="cb400-6"><a href="whole-genome-sequencing-mutations.html#cb400-6"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb400-7"><a href="whole-genome-sequencing-mutations.html#cb400-7"></a><span class="kw">plot</span>(SBSprofb<span class="op">$</span>Age,SBSprofb<span class="op">$</span>SBS1,  <span class="dt">ylab=</span><span class="st">&quot;Frequency Mut in SBS1&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Time NMU2Sac&quot;</span>, <span class="dt">main=</span><span class="st">&quot;SBS1&quot;</span>, <span class="dt">col=</span><span class="kw">factor</span>(SBSprofb<span class="op">$</span>Cohort))</span>
<span id="cb400-8"><a href="whole-genome-sequencing-mutations.html#cb400-8"></a><span class="kw">plot</span>( SBSprofb<span class="op">$</span>Age, SBSprofb<span class="op">$</span>SBS5, <span class="dt">ylab=</span><span class="st">&quot;Frequency Mut in SBS5&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Time NMU2Sac&quot;</span>, <span class="dt">main=</span><span class="st">&quot;SBS5&quot;</span>, <span class="dt">col=</span><span class="kw">factor</span>(SBSprofb<span class="op">$</span>Cohort))</span>
<span id="cb400-9"><a href="whole-genome-sequencing-mutations.html#cb400-9"></a><span class="kw">plot</span>(SBSprofb<span class="op">$</span>Age, SBSprofb<span class="op">$</span>SBS32<span class="op">+</span>SBSprofb<span class="op">$</span>SBS11, <span class="dt">ylab=</span><span class="st">&quot;SBS 32 or 11 score&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;tumor size&quot;</span>,</span>
<span id="cb400-10"><a href="whole-genome-sequencing-mutations.html#cb400-10"></a>     <span class="dt">col=</span><span class="kw">ifelse</span>(SBSprofb<span class="op">$</span>SBS11<span class="op">==</span><span class="dv">0</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="dt">main=</span><span class="st">&quot;SBS11/32&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-132-2.png" width="672" /></p>
</div>
<div id="annotate-the-data-with-human-common-variants" class="section level3">
<h3><span class="header-section-number">20.1.2</span> Annotate the data with human common variants</h3>
<div class="sourceCode" id="cb401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb401-1"><a href="whole-genome-sequencing-mutations.html#cb401-1"></a><span class="co">## The following is run independently and saved to fle</span></span>
<span id="cb401-2"><a href="whole-genome-sequencing-mutations.html#cb401-2"></a><span class="co">## issues with the RnvorV87 pointer when cached</span></span>
<span id="cb401-3"><a href="whole-genome-sequencing-mutations.html#cb401-3"></a><span class="co"># remove sourcing CreateRnor87</span></span>
<span id="cb401-4"><a href="whole-genome-sequencing-mutations.html#cb401-4"></a><span class="co">## Append to MeltCosmic?</span></span>
<span id="cb401-5"><a href="whole-genome-sequencing-mutations.html#cb401-5"></a><span class="kw">source</span>(<span class="st">&quot;../rscript/CreateRnor87db.R&quot;</span>)</span>
<span id="cb401-6"><a href="whole-genome-sequencing-mutations.html#cb401-6"></a>AAout=<span class="kw">strsplit</span>(<span class="kw">as.character</span>(MeltCosmic<span class="op">$</span>HGVSp_Short), <span class="st">&quot;[0-9]+&quot;</span>)</span>
<span id="cb401-7"><a href="whole-genome-sequencing-mutations.html#cb401-7"></a>AAwt=<span class="kw">sapply</span>(AAout, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb401-8"><a href="whole-genome-sequencing-mutations.html#cb401-8"></a>AAmt=<span class="kw">sapply</span>(AAout, <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb401-9"><a href="whole-genome-sequencing-mutations.html#cb401-9"></a>AAno=<span class="kw">regmatches</span>(MeltCosmic<span class="op">$</span>HGVSp_Short, <span class="kw">gregexpr</span>(<span class="st">&quot;[[:digit:]]+&quot;</span>, MeltCosmic<span class="op">$</span>HGVSp_Short))</span>
<span id="cb401-10"><a href="whole-genome-sequencing-mutations.html#cb401-10"></a></span>
<span id="cb401-11"><a href="whole-genome-sequencing-mutations.html#cb401-11"></a>MeltCosmic<span class="op">$</span>AAno=<span class="ot">NA</span></span>
<span id="cb401-12"><a href="whole-genome-sequencing-mutations.html#cb401-12"></a>MeltCosmic<span class="op">$</span>AAno[<span class="kw">which</span>(<span class="kw">sapply</span>(AAno, length)<span class="op">==</span><span class="dv">1</span>)]=<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(<span class="kw">unlist</span>(AAno)))</span>
<span id="cb401-13"><a href="whole-genome-sequencing-mutations.html#cb401-13"></a>FindVarTest=<span class="kw">FindHumanAAHomolog</span>(MeltCosmic<span class="op">$</span>Gene_Symbol, AAwt, MeltCosmic<span class="op">$</span>AAno, AAmt)</span>
<span id="cb401-14"><a href="whole-genome-sequencing-mutations.html#cb401-14"></a>MeltCosmic=<span class="kw">cbind</span>(MeltCosmic, FindVarTest)</span>
<span id="cb401-15"><a href="whole-genome-sequencing-mutations.html#cb401-15"></a></span>
<span id="cb401-16"><a href="whole-genome-sequencing-mutations.html#cb401-16"></a></span>
<span id="cb401-17"><a href="whole-genome-sequencing-mutations.html#cb401-17"></a></span>
<span id="cb401-18"><a href="whole-genome-sequencing-mutations.html#cb401-18"></a><span class="kw">save</span>(MeltCosmic, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/MeltCosmic_coding_genes_WGS_rat_VAFncount%s.RData&quot;</span>, <span class="kw">Sys.Date</span>())) <span class="co">#, sep=&quot;\t&quot;, row.names=F)</span></span>
<span id="cb401-19"><a href="whole-genome-sequencing-mutations.html#cb401-19"></a><span class="kw">write.csv</span>(MeltCosmic, <span class="dt">file=</span><span class="st">&quot;outputs/SupplementalTable_AllRatMutations_annotated_w_human_homolog_VAFncount.csv&quot;</span>, <span class="dt">row.names =</span> F, <span class="dt">quote=</span>F)</span></code></pre></div>
<p>From this analysis, below are the genes with mutations in known human hotpot mutations in common genes</p>
<div class="sourceCode" id="cb402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb402-1"><a href="whole-genome-sequencing-mutations.html#cb402-1"></a><span class="kw">load</span>(<span class="st">&quot;outputs/MeltCosmic_coding_genes_WGS_rat_VAFncount2021-01-11.RData&quot;</span>)</span>
<span id="cb402-2"><a href="whole-genome-sequencing-mutations.html#cb402-2"></a></span>
<span id="cb402-3"><a href="whole-genome-sequencing-mutations.html#cb402-3"></a>Xa=<span class="kw">which</span>(MeltCosmic<span class="op">$</span>Gene_Symbol <span class="op">%in%</span><span class="st"> </span>BList<span class="op">$</span>RatGene)</span>
<span id="cb402-4"><a href="whole-genome-sequencing-mutations.html#cb402-4"></a>Xb=<span class="kw">match</span>(<span class="kw">paste</span>(MeltCosmic<span class="op">$</span>Gene_Symbol, MeltCosmic<span class="op">$</span>AAno), <span class="kw">paste</span>(BList<span class="op">$</span>RatGene,BList<span class="op">$</span>RatAAno))</span>
<span id="cb402-5"><a href="whole-genome-sequencing-mutations.html#cb402-5"></a>Xc=<span class="kw">match</span>(<span class="kw">paste</span>(MeltCosmic<span class="op">$</span>Gene_Symbol, MeltCosmic<span class="op">$</span>AAno), <span class="kw">paste</span>(ChangList2<span class="op">$</span>RatGene, ChangList2<span class="op">$</span>RatAAno))</span>
<span id="cb402-6"><a href="whole-genome-sequencing-mutations.html#cb402-6"></a>x2=<span class="kw">c</span>(<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(Xb)), <span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(Xc)))</span>
<span id="cb402-7"><a href="whole-genome-sequencing-mutations.html#cb402-7"></a></span>
<span id="cb402-8"><a href="whole-genome-sequencing-mutations.html#cb402-8"></a><span class="co">#head(MeltCosmic)</span></span>
<span id="cb402-9"><a href="whole-genome-sequencing-mutations.html#cb402-9"></a></span>
<span id="cb402-10"><a href="whole-genome-sequencing-mutations.html#cb402-10"></a>Xb=<span class="kw">match</span>(<span class="kw">paste</span>(MeltCosmic<span class="op">$</span>HumGene, MeltCosmic<span class="op">$</span>HumAAno), <span class="kw">paste</span>(BList<span class="op">$</span>Gene,BList<span class="op">$</span>AAno))</span>
<span id="cb402-11"><a href="whole-genome-sequencing-mutations.html#cb402-11"></a>Xc=<span class="kw">match</span>(<span class="kw">paste</span>(MeltCosmic<span class="op">$</span>HumGene, MeltCosmic<span class="op">$</span>HumAAno), <span class="kw">paste</span>(ChangList2<span class="op">$</span>Gene, ChangList2<span class="op">$</span>AAno))</span>
<span id="cb402-12"><a href="whole-genome-sequencing-mutations.html#cb402-12"></a>x3=<span class="kw">c</span>(<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(Xb)), <span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(Xc)))</span>
<span id="cb402-13"><a href="whole-genome-sequencing-mutations.html#cb402-13"></a>x4=<span class="kw">unique</span>(<span class="kw">c</span>(x2, x3))</span>
<span id="cb402-14"><a href="whole-genome-sequencing-mutations.html#cb402-14"></a><span class="kw">kable</span>(<span class="kw">head</span>(MeltCosmic))</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
Gene_Symbol
</th>
<th style="text-align:right;">
Entrez_Gene_Id
</th>
<th style="text-align:left;">
Center
</th>
<th style="text-align:left;">
NCBI_Build
</th>
<th style="text-align:left;">
Chromosome
</th>
<th style="text-align:right;">
Start_Position
</th>
<th style="text-align:right;">
End_Position
</th>
<th style="text-align:left;">
Strand
</th>
<th style="text-align:left;">
Variant_Classification
</th>
<th style="text-align:left;">
Variant_Type
</th>
<th style="text-align:left;">
Reference_Allele
</th>
<th style="text-align:left;">
Tumor_Seq_Allele1
</th>
<th style="text-align:left;">
Tumor_Seq_Allele2
</th>
<th style="text-align:left;">
dbSNP_RS
</th>
<th style="text-align:left;">
dbSNP_Val_Status
</th>
<th style="text-align:left;">
Matched_Norm_Sample_Barcode
</th>
<th style="text-align:left;">
Match_Norm_Seq_Allele1
</th>
<th style="text-align:left;">
Match_Norm_Seq_Allele2
</th>
<th style="text-align:left;">
Tumor_Validation_Allele1
</th>
<th style="text-align:left;">
Tumor_Validation_Allele2
</th>
<th style="text-align:left;">
Match_Norm_Validation_Allele1
</th>
<th style="text-align:left;">
Match_Norm_Validation_Allele2
</th>
<th style="text-align:left;">
Verification_Status
</th>
<th style="text-align:left;">
Validation_Status
</th>
<th style="text-align:left;">
Mutation_Status
</th>
<th style="text-align:left;">
Sequencing_Phase
</th>
<th style="text-align:left;">
Sequence_Source
</th>
<th style="text-align:left;">
Validation_Method
</th>
<th style="text-align:left;">
Score
</th>
<th style="text-align:left;">
BAM_file
</th>
<th style="text-align:left;">
Sequencer
</th>
<th style="text-align:right;">
t_alt_count
</th>
<th style="text-align:right;">
t_ref_count
</th>
<th style="text-align:right;">
n_alt_count
</th>
<th style="text-align:right;">
n_ref_count
</th>
<th style="text-align:left;">
HGVSp_Short
</th>
<th style="text-align:right;">
VAF
</th>
<th style="text-align:left;">
variable
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
L1
</th>
<th style="text-align:right;">
AAno
</th>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
AAno
</th>
<th style="text-align:left;">
AA1
</th>
<th style="text-align:left;">
Variant
</th>
<th style="text-align:left;">
HumGene
</th>
<th style="text-align:left;">
RatSequence
</th>
<th style="text-align:left;">
RatProt
</th>
<th style="text-align:left;">
HumSequence
</th>
<th style="text-align:left;">
HumProt
</th>
<th style="text-align:right;">
HumAAno
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Ncor2
</td>
<td style="text-align:right;">
360801
</td>
<td style="text-align:left;">
nygenome.org
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:left;">
chr12
</td>
<td style="text-align:right;">
37001310
</td>
<td style="text-align:right;">
37001310
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:left;">
SNP
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
novel
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
control4-liver
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Unknown
</td>
<td style="text-align:left;">
Untested
</td>
<td style="text-align:left;">
Somatic
</td>
<td style="text-align:left;">
Phase_I
</td>
<td style="text-align:left;">
WGS
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Illumina_HiSeq
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:left;">
A804T
</td>
<td style="text-align:right;">
0.4545455
</td>
<td style="text-align:left;">
Tumor_Sample_Barcode
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:right;">
804
</td>
<td style="text-align:left;">
Ncor2
</td>
<td style="text-align:right;">
804
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
NCOR2
</td>
<td style="text-align:left;">
PPSPVASPPPA
</td>
<td style="text-align:left;">
ENSRNOP00000001334
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Ncor2
</td>
<td style="text-align:right;">
360801
</td>
<td style="text-align:left;">
nygenome.org
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:left;">
chr12
</td>
<td style="text-align:right;">
37024167
</td>
<td style="text-align:right;">
37024167
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:left;">
SNP
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
novel
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
control4-liver
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Unknown
</td>
<td style="text-align:left;">
Untested
</td>
<td style="text-align:left;">
Somatic
</td>
<td style="text-align:left;">
Phase_I
</td>
<td style="text-align:left;">
WGS
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Illumina_HiSeq
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:left;">
A1959V
</td>
<td style="text-align:right;">
0.4146341
</td>
<td style="text-align:left;">
Tumor_Sample_Barcode
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:right;">
1959
</td>
<td style="text-align:left;">
Ncor2
</td>
<td style="text-align:right;">
1959
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
V
</td>
<td style="text-align:left;">
NCOR2
</td>
<td style="text-align:left;">
PAKSLAPHHAS
</td>
<td style="text-align:left;">
ENSRNOP00000001334
</td>
<td style="text-align:left;">
PAKNLAPHHAS
</td>
<td style="text-align:left;">
ENSP00000384202
</td>
<td style="text-align:right;">
1990
</td>
</tr>
<tr>
<td style="text-align:left;">
Clip1
</td>
<td style="text-align:right;">
65201
</td>
<td style="text-align:left;">
nygenome.org
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:left;">
chr12
</td>
<td style="text-align:right;">
38377153
</td>
<td style="text-align:right;">
38377153
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:left;">
SNP
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
novel
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
control4-liver
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Unknown
</td>
<td style="text-align:left;">
Untested
</td>
<td style="text-align:left;">
Somatic
</td>
<td style="text-align:left;">
Phase_I
</td>
<td style="text-align:left;">
WGS
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Illumina_HiSeq
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:left;">
I5V
</td>
<td style="text-align:right;">
0.4871795
</td>
<td style="text-align:left;">
Tumor_Sample_Barcode
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
Clip1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
I
</td>
<td style="text-align:left;">
V
</td>
<td style="text-align:left;">
CLIP1
</td>
<td style="text-align:left;">
MSDSILSAAA
</td>
<td style="text-align:left;">
ENSRNOP00000073711
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Clip1
</td>
<td style="text-align:right;">
65201
</td>
<td style="text-align:left;">
nygenome.org
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:left;">
chr12
</td>
<td style="text-align:right;">
38407184
</td>
<td style="text-align:right;">
38407184
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:left;">
SNP
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
novel
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
control4-liver
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Unknown
</td>
<td style="text-align:left;">
Untested
</td>
<td style="text-align:left;">
Somatic
</td>
<td style="text-align:left;">
Phase_I
</td>
<td style="text-align:left;">
WGS
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Illumina_HiSeq
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
43
</td>
<td style="text-align:left;">
T848A
</td>
<td style="text-align:right;">
0.5769231
</td>
<td style="text-align:left;">
Tumor_Sample_Barcode
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:right;">
848
</td>
<td style="text-align:left;">
Clip1
</td>
<td style="text-align:right;">
848
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
CLIP1
</td>
<td style="text-align:left;">
EEAVSTQTSMQ
</td>
<td style="text-align:left;">
ENSRNOP00000073711
</td>
<td style="text-align:left;">
EEAVSVQRSMQ
</td>
<td style="text-align:left;">
ENSP00000303585
</td>
<td style="text-align:right;">
859
</td>
</tr>
<tr>
<td style="text-align:left;">
Cxcr4
</td>
<td style="text-align:right;">
60628
</td>
<td style="text-align:left;">
nygenome.org
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:left;">
chr13
</td>
<td style="text-align:right;">
45318778
</td>
<td style="text-align:right;">
45318778
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:left;">
SNP
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
novel
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
control4-liver
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Unknown
</td>
<td style="text-align:left;">
Untested
</td>
<td style="text-align:left;">
Somatic
</td>
<td style="text-align:left;">
Phase_I
</td>
<td style="text-align:left;">
WGS
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Illumina_HiSeq
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:left;">
E2K
</td>
<td style="text-align:right;">
0.1250000
</td>
<td style="text-align:left;">
Tumor_Sample_Barcode
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Cxcr4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
K
</td>
<td style="text-align:left;">
CXCR4
</td>
<td style="text-align:left;">
MEIYTSD
</td>
<td style="text-align:left;">
ENSRNOP00000005143
</td>
<td style="text-align:left;">
ISIYTSD
</td>
<td style="text-align:left;">
ENSP00000241393
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
Nrg1
</td>
<td style="text-align:right;">
112400
</td>
<td style="text-align:left;">
nygenome.org
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:left;">
chr16
</td>
<td style="text-align:right;">
63959893
</td>
<td style="text-align:right;">
63959893
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:left;">
SNP
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
novel
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
control4-liver
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Unknown
</td>
<td style="text-align:left;">
Untested
</td>
<td style="text-align:left;">
Somatic
</td>
<td style="text-align:left;">
Phase_I
</td>
<td style="text-align:left;">
WGS
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Illumina_HiSeq
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:left;">
P210S
</td>
<td style="text-align:right;">
0.1754386
</td>
<td style="text-align:left;">
Tumor_Sample_Barcode
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:left;">
C1R-tumor
</td>
<td style="text-align:right;">
210
</td>
<td style="text-align:left;">
Nrg1
</td>
<td style="text-align:right;">
210
</td>
<td style="text-align:left;">
P
</td>
<td style="text-align:left;">
S
</td>
<td style="text-align:left;">
NRG1
</td>
<td style="text-align:left;">
KSGTQPQTTET
</td>
<td style="text-align:left;">
ENSRNOP00000072731
</td>
<td style="text-align:left;">
KSATQPQTTET
</td>
<td style="text-align:left;">
ENSP00000433289
</td>
<td style="text-align:right;">
209
</td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="whole-genome-sequencing-mutations.html#cb403-1"></a></span>
<span id="cb403-2"><a href="whole-genome-sequencing-mutations.html#cb403-2"></a></span>
<span id="cb403-3"><a href="whole-genome-sequencing-mutations.html#cb403-3"></a>x2</span>
<span id="cb403-4"><a href="whole-genome-sequencing-mutations.html#cb403-4"></a><span class="co">##  [1] 114 191 211 293 319 412 423 484 486 114 191 211 259 293 319 412 423 455 484</span></span>
<span id="cb403-5"><a href="whole-genome-sequencing-mutations.html#cb403-5"></a><span class="co">## [20] 486 499</span></span>
<span id="cb403-6"><a href="whole-genome-sequencing-mutations.html#cb403-6"></a>x3</span>
<span id="cb403-7"><a href="whole-genome-sequencing-mutations.html#cb403-7"></a><span class="co">##  [1] 114 191 211 293 319 412 423 484 486 114 191 211 259 293 319 412 423 455 476</span></span>
<span id="cb403-8"><a href="whole-genome-sequencing-mutations.html#cb403-8"></a><span class="co">## [20] 484 486 499</span></span>
<span id="cb403-9"><a href="whole-genome-sequencing-mutations.html#cb403-9"></a>x4</span>
<span id="cb403-10"><a href="whole-genome-sequencing-mutations.html#cb403-10"></a><span class="co">##  [1] 114 191 211 293 319 412 423 484 486 259 455 499 476</span></span></code></pre></div>
<p>And here, we plot the frequency of these different mutations</p>
<div class="sourceCode" id="cb404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb404-1"><a href="whole-genome-sequencing-mutations.html#cb404-1"></a>idx=<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(MeltCosmic<span class="op">$</span>HumAAno))</span>
<span id="cb404-2"><a href="whole-genome-sequencing-mutations.html#cb404-2"></a><span class="kw">length</span>(idx)</span>
<span id="cb404-3"><a href="whole-genome-sequencing-mutations.html#cb404-3"></a><span class="co">## [1] 383</span></span>
<span id="cb404-4"><a href="whole-genome-sequencing-mutations.html#cb404-4"></a>Nx=<span class="kw">paste</span>(MeltCosmic<span class="op">$</span>Gene[idx], MeltCosmic<span class="op">$</span>AAno[idx])</span>
<span id="cb404-5"><a href="whole-genome-sequencing-mutations.html#cb404-5"></a>T2=<span class="kw">sort</span>(<span class="kw">table</span>(Nx), <span class="dt">decreasing =</span> T)</span>
<span id="cb404-6"><a href="whole-genome-sequencing-mutations.html#cb404-6"></a><span class="co">#head(T2)</span></span>
<span id="cb404-7"><a href="whole-genome-sequencing-mutations.html#cb404-7"></a></span>
<span id="cb404-8"><a href="whole-genome-sequencing-mutations.html#cb404-8"></a>ind2=<span class="kw">which</span>(T2<span class="op">&gt;</span><span class="dv">1</span>)</span>
<span id="cb404-9"><a href="whole-genome-sequencing-mutations.html#cb404-9"></a><span class="co">#pdf(&quot;~/Desktop/Frequency_of_mutations.pdf&quot;, height=6, width=9)</span></span>
<span id="cb404-10"><a href="whole-genome-sequencing-mutations.html#cb404-10"></a><span class="kw">barplot</span>(T2[ind2]<span class="op">/</span><span class="dv">31</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="dt">main=</span><span class="st">&quot;most frequent coding mutations with human homolog&quot;</span>, <span class="dt">las=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Ext1k"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext1k-1.png" alt="most frequent mutations" width="672" />
<p class="caption">
Figure 20.3: most frequent mutations
</p>
</div>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="whole-genome-sequencing-mutations.html#cb405-1"></a><span class="co">#dev.off()</span></span>
<span id="cb405-2"><a href="whole-genome-sequencing-mutations.html#cb405-2"></a></span>
<span id="cb405-3"><a href="whole-genome-sequencing-mutations.html#cb405-3"></a>ttemp=<span class="kw">names</span>(T2[ind2])</span></code></pre></div>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="whole-genome-sequencing-mutations.html#cb406-1"></a>Nx=<span class="kw">paste</span>(MeltCosmic<span class="op">$</span>Gene, MeltCosmic<span class="op">$</span>AAno)</span>
<span id="cb406-2"><a href="whole-genome-sequencing-mutations.html#cb406-2"></a>T2=<span class="kw">sort</span>(<span class="kw">table</span>(Nx), <span class="dt">decreasing =</span> T)</span>
<span id="cb406-3"><a href="whole-genome-sequencing-mutations.html#cb406-3"></a>ind2=<span class="kw">which</span>(T2<span class="op">&gt;</span><span class="dv">1</span>)</span>
<span id="cb406-4"><a href="whole-genome-sequencing-mutations.html#cb406-4"></a><span class="co">#pdf(&quot;~/Desktop/Frequency_of_mutations_2.pdf&quot;, height=6, width=9)</span></span>
<span id="cb406-5"><a href="whole-genome-sequencing-mutations.html#cb406-5"></a><span class="kw">barplot</span>(T2[ind2]<span class="op">/</span><span class="dv">31</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="dt">main=</span><span class="st">&quot;most frequent coding mutations &quot;</span>, <span class="dt">las=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-134-1.png" width="672" /></p>
<div class="sourceCode" id="cb407"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb407-1"><a href="whole-genome-sequencing-mutations.html#cb407-1"></a><span class="co">#dev.off()</span></span>
<span id="cb407-2"><a href="whole-genome-sequencing-mutations.html#cb407-2"></a></span>
<span id="cb407-3"><a href="whole-genome-sequencing-mutations.html#cb407-3"></a>tableout=<span class="kw">data.frame</span>(<span class="dt">gene=</span><span class="kw">names</span>(T2[ind2]), <span class="dt">ncount=</span>T2[ind2], <span class="dt">freq=</span>T2[ind2]<span class="op">/</span><span class="dv">31</span>,</span>
<span id="cb407-4"><a href="whole-genome-sequencing-mutations.html#cb407-4"></a>                    <span class="dt">humanHom=</span><span class="kw">ifelse</span>(<span class="kw">names</span>(T2[ind2])<span class="op">%in%</span>ttemp, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb407-5"><a href="whole-genome-sequencing-mutations.html#cb407-5"></a></span>
<span id="cb407-6"><a href="whole-genome-sequencing-mutations.html#cb407-6"></a><span class="kw">write.csv</span>(tableout, <span class="dt">file=</span><span class="st">&quot;nature-tables/Ext1k_maybe.csv&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="plots" class="section level2">
<h2><span class="header-section-number">20.2</span> Plots</h2>
<div id="quick-overview" class="section level3">
<h3><span class="header-section-number">20.2.1</span> Quick overview</h3>
<p>WGS all mutations summary: Look at the overall mutation burden, and the tumor mutational burden, defined as # coding mutations per MB.</p>
<p>The following plot lists tumors in increasing size</p>
<p>Make plots of:</p>
<ul>
<li>overall types of mutations per sample (ranked by size)</li>
<li>correlation between mutation type and size</li>
<li>type of mutation?</li>
</ul>
<div class="sourceCode" id="cb408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb408-1"><a href="whole-genome-sequencing-mutations.html#cb408-1"></a><span class="co">#load(&quot;data/wholegenome_mutations/WGS_mutations_coding.RData&quot;)</span></span>
<span id="cb408-2"><a href="whole-genome-sequencing-mutations.html#cb408-2"></a></span>
<span id="cb408-3"><a href="whole-genome-sequencing-mutations.html#cb408-3"></a>TabSamples=<span class="kw">lapply</span>(AllMutWGScoding, <span class="cf">function</span>(x) <span class="kw">table</span>(<span class="kw">factor</span>(x<span class="op">$</span>Variant_Classification, <span class="dt">levels=</span>SearchVars)))</span>
<span id="cb408-4"><a href="whole-genome-sequencing-mutations.html#cb408-4"></a>TabSamples2=<span class="kw">unlist</span>(TabSamples)</span>
<span id="cb408-5"><a href="whole-genome-sequencing-mutations.html#cb408-5"></a>namidx=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">names</span>(TabSamples2), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb408-6"><a href="whole-genome-sequencing-mutations.html#cb408-6"></a>namidx2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">names</span>(TabSamples2), <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb408-7"><a href="whole-genome-sequencing-mutations.html#cb408-7"></a>namidx3=<span class="kw">sapply</span>(namidx, <span class="cf">function</span>(x) Cdata<span class="op">$</span>NewID[<span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS)])</span>
<span id="cb408-8"><a href="whole-genome-sequencing-mutations.html#cb408-8"></a>TotTabl=<span class="kw">data.frame</span>(<span class="dt">val=</span>TabSamples2, <span class="dt">nam=</span>namidx, <span class="dt">variant=</span>namidx2, <span class="dt">nam2=</span>namidx3)</span>
<span id="cb408-9"><a href="whole-genome-sequencing-mutations.html#cb408-9"></a></span>
<span id="cb408-10"><a href="whole-genome-sequencing-mutations.html#cb408-10"></a><span class="co">## remove 0 samples</span></span>
<span id="cb408-11"><a href="whole-genome-sequencing-mutations.html#cb408-11"></a><span class="co"># searchNo=stack(by(TotTabl$val, TotTabl$variant, sum))</span></span>
<span id="cb408-12"><a href="whole-genome-sequencing-mutations.html#cb408-12"></a><span class="co"># TotTabl=TotTabl[ which(TotTabl$variant %in% searchNo$ind[which(searchNo$values!=0)]), ]</span></span>
<span id="cb408-13"><a href="whole-genome-sequencing-mutations.html#cb408-13"></a></span>
<span id="cb408-14"><a href="whole-genome-sequencing-mutations.html#cb408-14"></a>lx1=<span class="kw">match</span>(TotTabl<span class="op">$</span>nam2, Cdata<span class="op">$</span>NewID, <span class="cf">function</span>(x) <span class="kw">tryCatch</span>({x[<span class="dv">1</span>]}, <span class="dt">error=</span><span class="cf">function</span>(e){<span class="ot">NA</span>}))</span>
<span id="cb408-15"><a href="whole-genome-sequencing-mutations.html#cb408-15"></a>TotTabl<span class="op">$</span>Size=Cdata<span class="op">$</span>Tumor.diameter.sac.mm[lx1]</span>
<span id="cb408-16"><a href="whole-genome-sequencing-mutations.html#cb408-16"></a></span>
<span id="cb408-17"><a href="whole-genome-sequencing-mutations.html#cb408-17"></a>TotTabl=TotTabl[<span class="kw">order</span>(TotTabl<span class="op">$</span>Size, TotTabl<span class="op">$</span>val), ]</span>
<span id="cb408-18"><a href="whole-genome-sequencing-mutations.html#cb408-18"></a></span>
<span id="cb408-19"><a href="whole-genome-sequencing-mutations.html#cb408-19"></a>levelNam=<span class="kw">unique</span>(TotTabl<span class="op">$</span>nam2)</span>
<span id="cb408-20"><a href="whole-genome-sequencing-mutations.html#cb408-20"></a></span>
<span id="cb408-21"><a href="whole-genome-sequencing-mutations.html#cb408-21"></a>TotTabl<span class="op">$</span>nam2=<span class="kw">factor</span>(TotTabl<span class="op">$</span>nam2, <span class="dt">levels=</span>levelNam)</span>
<span id="cb408-22"><a href="whole-genome-sequencing-mutations.html#cb408-22"></a><span class="co">##pdf(sprintf(&quot;rslt/WGS/coding_mutation_types_%s.pdf&quot;, Sys.Date()), height=6, width=9)</span></span>
<span id="cb408-23"><a href="whole-genome-sequencing-mutations.html#cb408-23"></a><span class="co">#pdf(&quot;~/Desktop/No_mutations_VAFncount_adjusted_TMB.pdf&quot;, height = 5, width=7)</span></span>
<span id="cb408-24"><a href="whole-genome-sequencing-mutations.html#cb408-24"></a><span class="kw">ggplot</span>(TotTabl, <span class="kw">aes</span>(<span class="dt">x=</span>nam2, <span class="dt">y=</span>val<span class="op">/</span><span class="fl">43.423804</span>, <span class="dt">fill=</span>variant))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Set1&quot;</span>)))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Number of coding mutations/Mb&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;sorted by increasing tumor size at sac&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:1i"></span>
<img src="RatDCIS-bookdown_files/figure-html/1i-1.png" alt="summary of mutation burden" width="672" />
<p class="caption">
Figure 20.4: summary of mutation burden
</p>
</div>
<div class="sourceCode" id="cb409"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb409-1"><a href="whole-genome-sequencing-mutations.html#cb409-1"></a><span class="co">#dev.off()</span></span>
<span id="cb409-2"><a href="whole-genome-sequencing-mutations.html#cb409-2"></a></span>
<span id="cb409-3"><a href="whole-genome-sequencing-mutations.html#cb409-3"></a><span class="kw">write.csv</span>(TotTabl, <span class="dt">file=</span><span class="st">&quot;nature-tables/1i.csv&quot;</span>)</span></code></pre></div>
<p>We can also see whether any of these correlate with size, but didn’t find associations as shown below:</p>
<div class="sourceCode" id="cb410"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb410-1"><a href="whole-genome-sequencing-mutations.html#cb410-1"></a>ax1=<span class="kw">lapply</span>(<span class="kw">sort</span>(SearchVars), <span class="cf">function</span>(x) <span class="kw">cor.test</span>(TotTabl[TotTabl<span class="op">$</span>variant<span class="op">==</span>x , <span class="st">&quot;val&quot;</span>], TotTabl[TotTabl<span class="op">$</span>variant<span class="op">==</span>x ,  <span class="st">&quot;Size&quot;</span>], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)) </span>
<span id="cb410-2"><a href="whole-genome-sequencing-mutations.html#cb410-2"></a>pvals=<span class="kw">sapply</span>(ax1, <span class="cf">function</span>(x) x<span class="op">$</span>p.value)</span>
<span id="cb410-3"><a href="whole-genome-sequencing-mutations.html#cb410-3"></a>cors=<span class="kw">sapply</span>(ax1, <span class="cf">function</span>(x) x<span class="op">$</span>estimate)</span>
<span id="cb410-4"><a href="whole-genome-sequencing-mutations.html#cb410-4"></a></span>
<span id="cb410-5"><a href="whole-genome-sequencing-mutations.html#cb410-5"></a>ann_text2 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">lab=</span><span class="kw">paste</span>(<span class="st">&quot;cor=&quot;</span>, <span class="kw">round</span>(cors, <span class="dv">2</span>),<span class="st">&quot;p=&quot;</span>, <span class="kw">round</span>(pvals,<span class="dv">2</span>),<span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">variant=</span><span class="kw">sort</span>(SearchVars),  <span class="dt">val=</span><span class="dv">1</span>, <span class="dt">Size=</span><span class="dv">5</span>)</span>
<span id="cb410-6"><a href="whole-genome-sequencing-mutations.html#cb410-6"></a></span>
<span id="cb410-7"><a href="whole-genome-sequencing-mutations.html#cb410-7"></a><span class="kw">ggplot</span>(TotTabl, <span class="kw">aes</span>(<span class="dt">x=</span>Size, <span class="dt">y=</span>val, <span class="dt">col=</span>variant))<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>variant, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>)<span class="op">+</span><span class="kw">geom_text</span>(<span class="dt">data=</span>ann_text2, <span class="kw">aes</span>(<span class="dt">label=</span>lab))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-135-1.png" width="672" /></p>
<div class="sourceCode" id="cb411"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb411-1"><a href="whole-genome-sequencing-mutations.html#cb411-1"></a></span>
<span id="cb411-2"><a href="whole-genome-sequencing-mutations.html#cb411-2"></a><span class="co">#dev.off()</span></span>
<span id="cb411-3"><a href="whole-genome-sequencing-mutations.html#cb411-3"></a><span class="co">#MeltWGSCoding=melt(AllMutWGScoding, measure.vars=c(&quot;Tumor_Sample_Barcode&quot;))</span></span></code></pre></div>
</div>
<div id="coding-variants-tmb" class="section level3">
<h3><span class="header-section-number">20.2.2</span> Coding variants &amp; TMB</h3>
<p>There are a few ways of figuring out the TMB. We can</p>
<ul>
<li>report the number of mutations (missense, insertion/deletion, frameshifts, nonsense)</li>
<li>normalise the above value to the known length of coding regions in the rat genome (UCSC exon length 43,423,804)</li>
<li>plot of TMB vs objective response rate: there is no true relationship between final tumor size and mutational burden. Could also be confounded by subclonality</li>
</ul>
<div class="sourceCode" id="cb412"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb412-1"><a href="whole-genome-sequencing-mutations.html#cb412-1"></a>searchThese=<span class="kw">c</span>(<span class="st">&quot;Missense_Mutation&quot;</span>, <span class="st">&quot;Nonsense_Mutation&quot;</span>, <span class="st">&quot;Frame_Shift_Del&quot;</span>, <span class="st">&quot;Frame_Shift_Ins&quot;</span>, <span class="st">&quot;Nonstop_Mutation&quot;</span>)</span>
<span id="cb412-2"><a href="whole-genome-sequencing-mutations.html#cb412-2"></a>NVar2=TotTabl[TotTabl<span class="op">$</span>variant<span class="op">%in%</span>searchThese, ]</span>
<span id="cb412-3"><a href="whole-genome-sequencing-mutations.html#cb412-3"></a>Ntot=<span class="kw">by</span>(NVar2<span class="op">$</span>val, NVar2<span class="op">$</span>nam, sum)</span>
<span id="cb412-4"><a href="whole-genome-sequencing-mutations.html#cb412-4"></a><span class="co"># check this is unique based on the location</span></span>
<span id="cb412-5"><a href="whole-genome-sequencing-mutations.html#cb412-5"></a>Ntot2=<span class="kw">stack</span>(Ntot)</span>
<span id="cb412-6"><a href="whole-genome-sequencing-mutations.html#cb412-6"></a><span class="kw">colnames</span>(Ntot2)[<span class="dv">1</span>]=<span class="st">&quot;NCodingMut&quot;</span></span>
<span id="cb412-7"><a href="whole-genome-sequencing-mutations.html#cb412-7"></a>Ntot2<span class="op">$</span>TMB=Ntot2<span class="op">$</span>NCodingMut<span class="op">/</span><span class="fl">43.423804</span></span>
<span id="cb412-8"><a href="whole-genome-sequencing-mutations.html#cb412-8"></a>TotMut=<span class="kw">sapply</span>(AllMutWGS, nrow)</span>
<span id="cb412-9"><a href="whole-genome-sequencing-mutations.html#cb412-9"></a>Ntot2<span class="op">$</span>TotalMut=TotMut[<span class="kw">match</span>(Ntot2<span class="op">$</span>ind, <span class="kw">names</span>(TotMut))]</span>
<span id="cb412-10"><a href="whole-genome-sequencing-mutations.html#cb412-10"></a>Ntot2<span class="op">$</span>Tum.size=TotTabl<span class="op">$</span>Size[<span class="kw">match</span>(Ntot2<span class="op">$</span>ind, TotTabl<span class="op">$</span>nam)]</span>
<span id="cb412-11"><a href="whole-genome-sequencing-mutations.html#cb412-11"></a>Ntot2=Ntot2[<span class="kw">order</span>(Ntot2<span class="op">$</span>Tum.size), ]</span>
<span id="cb412-12"><a href="whole-genome-sequencing-mutations.html#cb412-12"></a><span class="co">#Ntot2$normCov=Ntot2$values/43.423804</span></span>
<span id="cb412-13"><a href="whole-genome-sequencing-mutations.html#cb412-13"></a></span>
<span id="cb412-14"><a href="whole-genome-sequencing-mutations.html#cb412-14"></a>Ntot2<span class="op">$</span>NsynFreq=Ntot2<span class="op">$</span>NCodingMut<span class="op">/</span>Ntot2<span class="op">$</span>TotalMut</span>
<span id="cb412-15"><a href="whole-genome-sequencing-mutations.html#cb412-15"></a>Ntot2<span class="op">$</span>ID2=<span class="kw">sapply</span>(Ntot2<span class="op">$</span>ind, <span class="cf">function</span>(x) Cdata<span class="op">$</span>NewID[<span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS)])</span>
<span id="cb412-16"><a href="whole-genome-sequencing-mutations.html#cb412-16"></a>Ntot2=Ntot2[<span class="kw">order</span>(Ntot2<span class="op">$</span>Tum.size), ]</span>
<span id="cb412-17"><a href="whole-genome-sequencing-mutations.html#cb412-17"></a></span>
<span id="cb412-18"><a href="whole-genome-sequencing-mutations.html#cb412-18"></a>IDfact=Ntot2<span class="op">$</span>ID2</span>
<span id="cb412-19"><a href="whole-genome-sequencing-mutations.html#cb412-19"></a>Ntot2<span class="op">$</span>cohort=Cdata<span class="op">$</span>Char.Cohort.Batch[<span class="kw">match</span>(Ntot2<span class="op">$</span>ID2, Cdata<span class="op">$</span>NewID)]</span>
<span id="cb412-20"><a href="whole-genome-sequencing-mutations.html#cb412-20"></a></span>
<span id="cb412-21"><a href="whole-genome-sequencing-mutations.html#cb412-21"></a></span>
<span id="cb412-22"><a href="whole-genome-sequencing-mutations.html#cb412-22"></a>Ntot2melt=<span class="kw">melt</span>(Ntot2)</span>
<span id="cb412-23"><a href="whole-genome-sequencing-mutations.html#cb412-23"></a>Ntot2melt<span class="op">$</span>ID2=<span class="kw">factor</span>(Ntot2melt<span class="op">$</span>ID2, <span class="dt">levels=</span>IDfact)</span>
<span id="cb412-24"><a href="whole-genome-sequencing-mutations.html#cb412-24"></a></span>
<span id="cb412-25"><a href="whole-genome-sequencing-mutations.html#cb412-25"></a><span class="co">#pdf(sprintf(&quot;~/Desktop/frequency_mutational_types_VAF_ncount#2_%s.pdf&quot;, Sys.Date()), height=10, width=6)</span></span>
<span id="cb412-26"><a href="whole-genome-sequencing-mutations.html#cb412-26"></a><span class="kw">ggplot</span>(Ntot2melt, <span class="kw">aes</span>(<span class="dt">x=</span>ID2, <span class="dt">y=</span>value, <span class="dt">col=</span>variable))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(variable<span class="op">~</span>., <span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>)<span class="op">+</span></span>
<span id="cb412-27"><a href="whole-genome-sequencing-mutations.html#cb412-27"></a><span class="st">  </span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))</span></code></pre></div>
<div class="figure"><span id="fig:1ib"></span>
<img src="RatDCIS-bookdown_files/figure-html/1ib-1.png" alt="summary of mutational burden?" width="672" />
<p class="caption">
Figure 20.5: summary of mutational burden?
</p>
</div>
<div class="sourceCode" id="cb413"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb413-1"><a href="whole-genome-sequencing-mutations.html#cb413-1"></a><span class="co">#dev.off()</span></span>
<span id="cb413-2"><a href="whole-genome-sequencing-mutations.html#cb413-2"></a><span class="kw">write.csv</span>(Ntot2melt, <span class="dt">file=</span><span class="st">&quot;nature-tables/1i_maybe.csv&quot;</span>)</span>
<span id="cb413-3"><a href="whole-genome-sequencing-mutations.html#cb413-3"></a><span class="co">## As a metric for TMB, comapre the number of mutations to the covered genome</span></span></code></pre></div>
<p>Calculate whether there is an association of TMB with cohort type:</p>
<div class="sourceCode" id="cb414"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb414-1"><a href="whole-genome-sequencing-mutations.html#cb414-1"></a>Ntot2melt<span class="op">$</span>size2=Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">match</span>(Ntot2melt<span class="op">$</span>ID2, Cdata<span class="op">$</span>NewID)]</span>
<span id="cb414-2"><a href="whole-genome-sequencing-mutations.html#cb414-2"></a></span>
<span id="cb414-3"><a href="whole-genome-sequencing-mutations.html#cb414-3"></a>ax1=<span class="kw">lapply</span>(<span class="kw">sort</span>(SearchVars), <span class="cf">function</span>(x) <span class="kw">cor.test</span>(TotTabl[TotTabl<span class="op">$</span>variant<span class="op">==</span>x , <span class="st">&quot;val&quot;</span>], TotTabl[TotTabl<span class="op">$</span>variant<span class="op">==</span>x ,  <span class="st">&quot;Size&quot;</span>], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>, <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>)) </span>
<span id="cb414-4"><a href="whole-genome-sequencing-mutations.html#cb414-4"></a>pvals=<span class="kw">sapply</span>(ax1, <span class="cf">function</span>(x) x<span class="op">$</span>p.value)</span>
<span id="cb414-5"><a href="whole-genome-sequencing-mutations.html#cb414-5"></a>cors=<span class="kw">sapply</span>(ax1, <span class="cf">function</span>(x) x<span class="op">$</span>estimate)</span>
<span id="cb414-6"><a href="whole-genome-sequencing-mutations.html#cb414-6"></a></span>
<span id="cb414-7"><a href="whole-genome-sequencing-mutations.html#cb414-7"></a></span>
<span id="cb414-8"><a href="whole-genome-sequencing-mutations.html#cb414-8"></a><span class="co"># p value calculation: figure out the latency and penetrance</span></span>
<span id="cb414-9"><a href="whole-genome-sequencing-mutations.html#cb414-9"></a></span>
<span id="cb414-10"><a href="whole-genome-sequencing-mutations.html#cb414-10"></a>pvalOutputs=<span class="kw">sapply</span>(<span class="kw">c</span>(<span class="st">&quot;NCodingMut&quot;</span>, <span class="st">&quot;TMB&quot;</span>, <span class="st">&quot;TotalMut&quot;</span>,<span class="st">&quot;Tum.size&quot;</span>,</span>
<span id="cb414-11"><a href="whole-genome-sequencing-mutations.html#cb414-11"></a>                     <span class="st">&quot;NsynFreq&quot;</span>) , <span class="cf">function</span>(x) <span class="kw">t.test</span>(Ntot2[ Ntot2<span class="op">$</span>cohort<span class="op">!=</span><span class="st">&quot;cohort-2&quot;</span>,x]<span class="op">~</span><span class="st"> </span>Ntot2<span class="op">$</span>cohort[Ntot2<span class="op">$</span>cohort<span class="op">!=</span><span class="st">&quot;cohort-2&quot;</span>])<span class="op">$</span>p.value)</span>
<span id="cb414-12"><a href="whole-genome-sequencing-mutations.html#cb414-12"></a></span>
<span id="cb414-13"><a href="whole-genome-sequencing-mutations.html#cb414-13"></a><span class="co"># cohort size calculation</span></span>
<span id="cb414-14"><a href="whole-genome-sequencing-mutations.html#cb414-14"></a>corOutputs=<span class="kw">sapply</span>(<span class="kw">c</span>(<span class="st">&quot;NCodingMut&quot;</span>, <span class="st">&quot;TMB&quot;</span>, <span class="st">&quot;TotalMut&quot;</span>,</span>
<span id="cb414-15"><a href="whole-genome-sequencing-mutations.html#cb414-15"></a>                     <span class="st">&quot;NsynFreq&quot;</span>) , <span class="cf">function</span>(x) <span class="kw">cor.test</span>(Ntot2[ ,x], Ntot2<span class="op">$</span>Tum.size)<span class="op">$</span>p.value)</span>
<span id="cb414-16"><a href="whole-genome-sequencing-mutations.html#cb414-16"></a></span>
<span id="cb414-17"><a href="whole-genome-sequencing-mutations.html#cb414-17"></a></span>
<span id="cb414-18"><a href="whole-genome-sequencing-mutations.html#cb414-18"></a><span class="co">#pdf(&quot;~/Desktop/Supp_1G-H_WGS_mutations_assoc_clinico.pdf&quot;, width=8, height=5)</span></span>
<span id="cb414-19"><a href="whole-genome-sequencing-mutations.html#cb414-19"></a></span>
<span id="cb414-20"><a href="whole-genome-sequencing-mutations.html#cb414-20"></a>ann_text2 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">lab=</span><span class="kw">paste</span>(<span class="st">&quot;p=&quot;</span>, <span class="kw">round</span>(pvalOutputs, <span class="dv">2</span>)), <span class="dt">val=</span><span class="dv">1</span>, <span class="dt">Size=</span><span class="dv">5</span>, <span class="dt">cohort=</span><span class="st">&quot;latency-4&quot;</span>,</span>
<span id="cb414-21"><a href="whole-genome-sequencing-mutations.html#cb414-21"></a>                        <span class="dt">variable=</span><span class="kw">c</span>(<span class="st">&quot;NCodingMut&quot;</span>, <span class="st">&quot;TMB&quot;</span>, <span class="st">&quot;TotalMut&quot;</span>, <span class="st">&quot;Tum.size&quot;</span>,<span class="st">&quot;NsynFreq&quot;</span>))</span>
<span id="cb414-22"><a href="whole-genome-sequencing-mutations.html#cb414-22"></a></span>
<span id="cb414-23"><a href="whole-genome-sequencing-mutations.html#cb414-23"></a><span class="kw">ggplot</span>(Ntot2melt, <span class="kw">aes</span>(<span class="dt">x=</span>cohort, <span class="dt">y=</span>value, <span class="dt">col=</span>cohort))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>variable, <span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))<span class="op">+</span><span class="kw">geom_text</span>(<span class="dt">data=</span>ann_text2, <span class="kw">aes</span>(<span class="dt">label=</span>lab, <span class="dt">x=</span><span class="dv">1</span>, <span class="dt">y=</span><span class="dv">0</span>))</span></code></pre></div>
<div class="figure"><span id="fig:Ext1h"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext1h-1.png" alt="assoc of mutations with cohort" width="672" />
<p class="caption">
Figure 20.6: assoc of mutations with cohort
</p>
</div>
<div class="sourceCode" id="cb415"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb415-1"><a href="whole-genome-sequencing-mutations.html#cb415-1"></a>ann_text2 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">lab=</span><span class="kw">paste</span>(<span class="st">&quot;p=&quot;</span>, <span class="kw">round</span>(corOutputs, <span class="dv">2</span>)), <span class="dt">val=</span><span class="dv">1</span>, <span class="dt">Size=</span><span class="dv">5</span>, <span class="dt">cohort=</span><span class="st">&quot;latency-4&quot;</span>,</span>
<span id="cb415-2"><a href="whole-genome-sequencing-mutations.html#cb415-2"></a>                        <span class="dt">variable=</span><span class="kw">c</span>(<span class="st">&quot;NCodingMut&quot;</span>, <span class="st">&quot;TMB&quot;</span>, <span class="st">&quot;TotalMut&quot;</span>,</span>
<span id="cb415-3"><a href="whole-genome-sequencing-mutations.html#cb415-3"></a>                     <span class="st">&quot;NsynFreq&quot;</span>))</span>
<span id="cb415-4"><a href="whole-genome-sequencing-mutations.html#cb415-4"></a></span>
<span id="cb415-5"><a href="whole-genome-sequencing-mutations.html#cb415-5"></a><span class="kw">ggplot</span>(Ntot2melt, <span class="kw">aes</span>(<span class="dt">x=</span>size2, <span class="dt">y=</span>value, <span class="dt">col=</span>cohort))<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>variable, <span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>))<span class="op">+</span><span class="kw">geom_text</span>(<span class="dt">data=</span>ann_text2, <span class="kw">aes</span>(<span class="dt">label=</span>lab, <span class="dt">x=</span><span class="dv">10</span>, <span class="dt">y=</span><span class="dv">0</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-136-1.png" width="672" /></p>
<div class="sourceCode" id="cb416"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb416-1"><a href="whole-genome-sequencing-mutations.html#cb416-1"></a><span class="co">#dev.off()</span></span>
<span id="cb416-2"><a href="whole-genome-sequencing-mutations.html#cb416-2"></a></span>
<span id="cb416-3"><a href="whole-genome-sequencing-mutations.html#cb416-3"></a><span class="kw">write.csv</span>(Ntot2melt, <span class="dt">file =</span> <span class="st">&quot;nature-tables/Ext1h-i.csv&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb417-1"><a href="whole-genome-sequencing-mutations.html#cb417-1"></a><span class="co"># To obtain an idea of the coverage:</span></span>
<span id="cb417-2"><a href="whole-genome-sequencing-mutations.html#cb417-2"></a>Using bedtools and samtools</span>
<span id="cb417-3"><a href="whole-genome-sequencing-mutations.html#cb417-3"></a>here, counting only <span class="st">&quot;propoet pairs&quot;</span> with maping quality greater than <span class="dv">30</span></span>
<span id="cb417-4"><a href="whole-genome-sequencing-mutations.html#cb417-4"></a><span class="kw">see</span> (https<span class="op">:</span><span class="er">//</span>www.biostars.org<span class="op">/</span>p<span class="op">/</span><span class="dv">11105</span><span class="op">/</span>)</span>
<span id="cb417-5"><a href="whole-genome-sequencing-mutations.html#cb417-5"></a>samtools view <span class="op">-</span>u <span class="op">-</span>q <span class="dv">30</span> <span class="op">-</span>f <span class="dv">0x2</span> aln.bam <span class="op">|</span><span class="st"> </span>coverageBed <span class="op">-</span>abam stdin <span class="op">-</span>b exons.bed <span class="op">&gt;</span><span class="st"> </span>exons.bed.proper.q30.coverage</span></code></pre></div>
</div>
</div>
<div id="coding-variants" class="section level2">
<h2><span class="header-section-number">20.3</span> Coding variants</h2>
<div id="sites-which-are-commonly-mutated" class="section level3">
<h3><span class="header-section-number">20.3.1</span> Sites which are commonly mutated?</h3>
<p>Could this be a variant (SD strain “rsSNP”) or is it actually a mutation?</p>
<p>Variants were annotated with snpeff. Raw data from this was obtained from <a href="https://www.biorxiv.org/content/10.1101/412924v1.full" class="uri">https://www.biorxiv.org/content/10.1101/412924v1.full</a></p>
<div class="sourceCode" id="cb418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb418-1"><a href="whole-genome-sequencing-mutations.html#cb418-1"></a></span>
<span id="cb418-2"><a href="whole-genome-sequencing-mutations.html#cb418-2"></a>harlanRats=<span class="kw">read.vcfR</span>(<span class="st">&quot;~/Desktop/carlos_study_files/allChr.allSamps.90DR2.maf01.hweE7.noIBD.HarlanOnly.vcf.gz&quot;</span>)</span>
<span id="cb418-3"><a href="whole-genome-sequencing-mutations.html#cb418-3"></a>harlanRats<span class="op">@</span>fix[ ,<span class="dv">1</span>]=<span class="kw">paste</span>(<span class="st">&quot;chr&quot;</span>, harlanRats<span class="op">@</span>fix[ ,<span class="dv">1</span>], <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb418-4"><a href="whole-genome-sequencing-mutations.html#cb418-4"></a>harlanRats<span class="op">@</span>fix[ ,<span class="dv">3</span>]=<span class="kw">paste</span>(<span class="st">&quot;H&quot;</span>, harlanRats<span class="op">@</span>fix[ ,<span class="dv">3</span>], <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb418-5"><a href="whole-genome-sequencing-mutations.html#cb418-5"></a>harlanRats<span class="op">@</span>meta=harlanRats<span class="op">@</span>meta[<span class="op">-</span><span class="kw">grep</span>(<span class="st">&quot;##contig&quot;</span>,harlanRats<span class="op">@</span>meta )]</span>
<span id="cb418-6"><a href="whole-genome-sequencing-mutations.html#cb418-6"></a></span>
<span id="cb418-7"><a href="whole-genome-sequencing-mutations.html#cb418-7"></a><span class="co"># figure out the genotype distribution</span></span>
<span id="cb418-8"><a href="whole-genome-sequencing-mutations.html#cb418-8"></a>GT=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(harlanRats<span class="op">@</span>gt), <span class="cf">function</span>(x) </span>
<span id="cb418-9"><a href="whole-genome-sequencing-mutations.html#cb418-9"></a>  <span class="kw">table</span>(<span class="kw">factor</span>(harlanRats<span class="op">@</span>gt[x, <span class="dv">-1</span>],<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;0/0&quot;</span>,<span class="st">&quot;0/1&quot;</span>,<span class="st">&quot;1/1&quot;</span>))))</span>
<span id="cb418-10"><a href="whole-genome-sequencing-mutations.html#cb418-10"></a>GT2=<span class="kw">paste</span>(GT[<span class="dv">1</span>, ], GT[<span class="dv">2</span>, ], GT[<span class="dv">3</span>, ], <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb418-11"><a href="whole-genome-sequencing-mutations.html#cb418-11"></a>GT3=<span class="kw">paste</span>(GT[<span class="dv">1</span>, ]<span class="op">/</span><span class="dv">2281</span>, GT[<span class="dv">2</span>, ]<span class="op">/</span><span class="dv">2281</span>, GT[<span class="dv">3</span>, ]<span class="op">/</span><span class="dv">2281</span>, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb418-12"><a href="whole-genome-sequencing-mutations.html#cb418-12"></a></span>
<span id="cb418-13"><a href="whole-genome-sequencing-mutations.html#cb418-13"></a>GenotypeS=<span class="kw">cbind</span>(<span class="dt">FORMAT=</span><span class="st">&quot;GT&quot;</span>, <span class="dt">Annot=</span><span class="kw">c</span>(<span class="st">&quot;0/0,0/1,1/1&quot;</span>), GT2, GT3 )</span>
<span id="cb418-14"><a href="whole-genome-sequencing-mutations.html#cb418-14"></a>harlanRats<span class="op">@</span>gt=GenotypeS</span>
<span id="cb418-15"><a href="whole-genome-sequencing-mutations.html#cb418-15"></a><span class="co">## save this to file or something</span></span>
<span id="cb418-16"><a href="whole-genome-sequencing-mutations.html#cb418-16"></a><span class="kw">write.vcf</span>(harlanRats, <span class="dt">file=</span><span class="st">&quot;~/Desktop/dbSNP_info/harlan_snps_mod_1-5-2020.vcf.gz&quot;</span>)</span>
<span id="cb418-17"><a href="whole-genome-sequencing-mutations.html#cb418-17"></a></span>
<span id="cb418-18"><a href="whole-genome-sequencing-mutations.html#cb418-18"></a><span class="co"># loading the charles river rat snp sites</span></span>
<span id="cb418-19"><a href="whole-genome-sequencing-mutations.html#cb418-19"></a>CRRats=<span class="kw">read.vcfR</span>(<span class="st">&quot;~/Desktop/dbSNP_info/charles_river_200Ksnps.gz&quot;</span>)</span>
<span id="cb418-20"><a href="whole-genome-sequencing-mutations.html#cb418-20"></a>CRRats<span class="op">@</span>fix[ ,<span class="dv">1</span>]=<span class="kw">paste</span>(<span class="st">&quot;chr&quot;</span>, CRRats<span class="op">@</span>fix[ ,<span class="dv">1</span>], <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb418-21"><a href="whole-genome-sequencing-mutations.html#cb418-21"></a>CRRats<span class="op">@</span>fix[ ,<span class="dv">3</span>]=<span class="kw">paste</span>(<span class="st">&quot;CR&quot;</span>, CRRats<span class="op">@</span>fix[ ,<span class="dv">3</span>], <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb418-22"><a href="whole-genome-sequencing-mutations.html#cb418-22"></a>CRRats<span class="op">@</span>meta=CRRats<span class="op">@</span>meta[<span class="op">-</span><span class="kw">grep</span>(<span class="st">&quot;##contig&quot;</span>,CRRats<span class="op">@</span>meta )]</span>
<span id="cb418-23"><a href="whole-genome-sequencing-mutations.html#cb418-23"></a></span>
<span id="cb418-24"><a href="whole-genome-sequencing-mutations.html#cb418-24"></a>crGT=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(CRRats<span class="op">@</span>gt), <span class="cf">function</span>(x) </span>
<span id="cb418-25"><a href="whole-genome-sequencing-mutations.html#cb418-25"></a>  <span class="kw">table</span>(<span class="kw">factor</span>(CRRats<span class="op">@</span>gt[x, <span class="dv">-1</span>],<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;0/0&quot;</span>,<span class="st">&quot;0/1&quot;</span>,<span class="st">&quot;1/1&quot;</span>))))</span>
<span id="cb418-26"><a href="whole-genome-sequencing-mutations.html#cb418-26"></a>crGT2=<span class="kw">paste</span>(crGT[<span class="dv">1</span>, ], crGT[<span class="dv">2</span>, ], crGT[<span class="dv">3</span>, ], <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb418-27"><a href="whole-genome-sequencing-mutations.html#cb418-27"></a>crGT3=<span class="kw">paste</span>(crGT[<span class="dv">1</span>, ]<span class="op">/</span><span class="dv">1780</span>, crGT[<span class="dv">2</span>, ]<span class="op">/</span><span class="dv">1780</span>, crGT[<span class="dv">3</span>, ]<span class="op">/</span><span class="dv">1780</span>, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb418-28"><a href="whole-genome-sequencing-mutations.html#cb418-28"></a>GenotypeS=<span class="kw">cbind</span>(<span class="dt">FORMAT=</span><span class="st">&quot;GT&quot;</span>, <span class="dt">Annot=</span><span class="kw">c</span>(<span class="st">&quot;0/0,0/1,1/1&quot;</span>), crGT2, crGT3 )</span>
<span id="cb418-29"><a href="whole-genome-sequencing-mutations.html#cb418-29"></a>CRRats<span class="op">@</span>gt=GenotypeS</span>
<span id="cb418-30"><a href="whole-genome-sequencing-mutations.html#cb418-30"></a><span class="kw">write.vcf</span>(CRRats, <span class="dt">file=</span><span class="st">&quot;~/Desktop/dbSNP_info/charlesRiver_snps_mod_1-5-2020.vcf.gz&quot;</span>)</span></code></pre></div>
<p>Note that the nihRats package may not be required here. We probably just want the harlan snps</p>
<div class="sourceCode" id="cb419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb419-1"><a href="whole-genome-sequencing-mutations.html#cb419-1"></a><span class="co"># loading known list of rat snp sites</span></span>
<span id="cb419-2"><a href="whole-genome-sequencing-mutations.html#cb419-2"></a>nihRats=<span class="kw">read.vcfR</span>(<span class="st">&quot;../anntotations/dbSNP/rat_10116_00-All.wchr.SDsort.vcf.gz&quot;</span>)</span>
<span id="cb419-3"><a href="whole-genome-sequencing-mutations.html#cb419-3"></a><span class="co">#nihRats@fix[ ,1]=paste(&quot;chr&quot;, nihRats@fix[ ,1], sep=&quot;&quot;)</span></span>
<span id="cb419-4"><a href="whole-genome-sequencing-mutations.html#cb419-4"></a>r=<span class="kw">regexpr</span>(<span class="st">&quot;[a-z]+_[a-z]+&quot;</span>,nihRats<span class="op">@</span>fix[ ,<span class="dv">8</span>]) <span class="co">#ANN=[A-Z]\\|</span></span>
<span id="cb419-5"><a href="whole-genome-sequencing-mutations.html#cb419-5"></a>ModType=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(nihRats<span class="op">@</span>fix[ ,<span class="dv">8</span>]))</span>
<span id="cb419-6"><a href="whole-genome-sequencing-mutations.html#cb419-6"></a>ModType[r<span class="op">!=-</span><span class="dv">1</span>]=<span class="kw">regmatches</span>(nihRats<span class="op">@</span>fix[ ,<span class="dv">8</span>], r)</span>
<span id="cb419-7"><a href="whole-genome-sequencing-mutations.html#cb419-7"></a><span class="co">#ModType=substr(ModType, 7, 25)</span></span>
<span id="cb419-8"><a href="whole-genome-sequencing-mutations.html#cb419-8"></a><span class="co"># Harlan samples</span></span>
<span id="cb419-9"><a href="whole-genome-sequencing-mutations.html#cb419-9"></a>harlanRats=<span class="kw">read.vcfR</span>(<span class="st">&quot;../anntotations/dbSNP/harlan_snps_mod_1-5-2020.SDsort.vcf.gz&quot;</span>)</span>
<span id="cb419-10"><a href="whole-genome-sequencing-mutations.html#cb419-10"></a></span>
<span id="cb419-11"><a href="whole-genome-sequencing-mutations.html#cb419-11"></a>GT3=crGT3=<span class="dv">1</span><span class="op">-</span><span class="kw">as.numeric</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(harlanRats<span class="op">@</span>gt[ ,<span class="dv">4</span>], <span class="st">&quot;,&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]))</span>
<span id="cb419-12"><a href="whole-genome-sequencing-mutations.html#cb419-12"></a></span>
<span id="cb419-13"><a href="whole-genome-sequencing-mutations.html#cb419-13"></a>r=<span class="kw">regexpr</span>(<span class="st">&quot;ANN=[A-Z]</span><span class="ch">\\</span><span class="st">|[a-z]+_[a-z]+&quot;</span>, harlanRats<span class="op">@</span>fix[ ,<span class="dv">8</span>])</span>
<span id="cb419-14"><a href="whole-genome-sequencing-mutations.html#cb419-14"></a>HarType=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(harlanRats<span class="op">@</span>fix[ ,<span class="dv">8</span>]))</span>
<span id="cb419-15"><a href="whole-genome-sequencing-mutations.html#cb419-15"></a>HarType[r<span class="op">!=-</span><span class="dv">1</span>]=<span class="kw">regmatches</span>(harlanRats<span class="op">@</span>fix[ ,<span class="dv">8</span>], r)</span>
<span id="cb419-16"><a href="whole-genome-sequencing-mutations.html#cb419-16"></a>HarType=<span class="kw">substr</span>(HarType, <span class="dv">7</span>, <span class="dv">25</span>)</span>
<span id="cb419-17"><a href="whole-genome-sequencing-mutations.html#cb419-17"></a><span class="co"># charles river samples</span></span>
<span id="cb419-18"><a href="whole-genome-sequencing-mutations.html#cb419-18"></a>CRRats=<span class="kw">read.vcfR</span>(<span class="st">&quot;../anntotations/dbSNP/charlesRiver_snps_mod_1-5-2020.SDsort.vcf.gz&quot;</span>)</span>
<span id="cb419-19"><a href="whole-genome-sequencing-mutations.html#cb419-19"></a>crGT3=<span class="dv">1</span><span class="op">-</span><span class="kw">as.numeric</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(CRRats<span class="op">@</span>gt[ ,<span class="dv">4</span>], <span class="st">&quot;,&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>]))</span>
<span id="cb419-20"><a href="whole-genome-sequencing-mutations.html#cb419-20"></a></span>
<span id="cb419-21"><a href="whole-genome-sequencing-mutations.html#cb419-21"></a>r=<span class="kw">regexpr</span>(<span class="st">&quot;ANN=[A-Z]</span><span class="ch">\\</span><span class="st">|[a-z]+_[a-z]+&quot;</span>, CRRats<span class="op">@</span>fix[ ,<span class="dv">8</span>])</span>
<span id="cb419-22"><a href="whole-genome-sequencing-mutations.html#cb419-22"></a>CRType=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(CRRats<span class="op">@</span>fix[ ,<span class="dv">8</span>]))</span>
<span id="cb419-23"><a href="whole-genome-sequencing-mutations.html#cb419-23"></a>CRType[r<span class="op">!=-</span><span class="dv">1</span>]=<span class="kw">regmatches</span>(CRRats<span class="op">@</span>fix[ ,<span class="dv">8</span>], r)</span>
<span id="cb419-24"><a href="whole-genome-sequencing-mutations.html#cb419-24"></a></span>
<span id="cb419-25"><a href="whole-genome-sequencing-mutations.html#cb419-25"></a>CRType=<span class="kw">substr</span>(CRType, <span class="dv">7</span>, <span class="dv">25</span>)</span></code></pre></div>
<div class="sourceCode" id="cb420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb420-1"><a href="whole-genome-sequencing-mutations.html#cb420-1"></a><span class="co"># This is just a plot that compares the different databases.</span></span>
<span id="cb420-2"><a href="whole-genome-sequencing-mutations.html#cb420-2"></a><span class="co">## Look at the overlap between all the databases:</span></span>
<span id="cb420-3"><a href="whole-genome-sequencing-mutations.html#cb420-3"></a>NIHr=<span class="kw">paste</span>(nihRats<span class="op">@</span>fix[ ,<span class="dv">1</span>], nihRats<span class="op">@</span>fix[ ,<span class="dv">2</span>])</span>
<span id="cb420-4"><a href="whole-genome-sequencing-mutations.html#cb420-4"></a>harr=<span class="kw">paste</span>(harlanRats<span class="op">@</span>fix[ ,<span class="dv">1</span>], harlanRats<span class="op">@</span>fix[ ,<span class="dv">2</span>])</span>
<span id="cb420-5"><a href="whole-genome-sequencing-mutations.html#cb420-5"></a>crr=<span class="kw">paste</span>(CRRats<span class="op">@</span>fix[ ,<span class="dv">1</span>], CRRats<span class="op">@</span>fix[ ,<span class="dv">2</span>])</span>
<span id="cb420-6"><a href="whole-genome-sequencing-mutations.html#cb420-6"></a></span>
<span id="cb420-7"><a href="whole-genome-sequencing-mutations.html#cb420-7"></a><span class="co"># overview of the overlaps</span></span>
<span id="cb420-8"><a href="whole-genome-sequencing-mutations.html#cb420-8"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">NIH=</span>NIHr, <span class="dt">harlan=</span>harr, <span class="dt">CR=</span>crr))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-138-1.png" width="672" /></p>
<div class="sourceCode" id="cb421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb421-1"><a href="whole-genome-sequencing-mutations.html#cb421-1"></a></span>
<span id="cb421-2"><a href="whole-genome-sequencing-mutations.html#cb421-2"></a><span class="co">## Annotate the above variants for protein coding regions first, then narrow down to known </span></span>
<span id="cb421-3"><a href="whole-genome-sequencing-mutations.html#cb421-3"></a>AllLevels=<span class="kw">unique</span>(<span class="kw">c</span>(HarType, CRType, ModType))</span>
<span id="cb421-4"><a href="whole-genome-sequencing-mutations.html#cb421-4"></a>allTypes=<span class="kw">rbind</span>(<span class="kw">table</span>(<span class="kw">factor</span>(ModType, <span class="dt">levels=</span>AllLevels)), <span class="kw">table</span>(<span class="kw">factor</span>(HarType, <span class="dt">levels=</span>AllLevels)),</span>
<span id="cb421-5"><a href="whole-genome-sequencing-mutations.html#cb421-5"></a>               <span class="kw">table</span>(<span class="kw">factor</span>(CRType, <span class="dt">levels =</span> AllLevels)))</span>
<span id="cb421-6"><a href="whole-genome-sequencing-mutations.html#cb421-6"></a><span class="kw">rownames</span>(allTypes)=<span class="kw">c</span>(<span class="st">&quot;NIH&quot;</span>, <span class="st">&quot;harlan&quot;</span>, <span class="st">&quot;CR&quot;</span>)</span>
<span id="cb421-7"><a href="whole-genome-sequencing-mutations.html#cb421-7"></a>allTypesNorm=allTypes<span class="op">/</span><span class="kw">rowSums</span>(allTypes)</span>
<span id="cb421-8"><a href="whole-genome-sequencing-mutations.html#cb421-8"></a></span>
<span id="cb421-9"><a href="whole-genome-sequencing-mutations.html#cb421-9"></a>t2=<span class="kw">melt</span>(allTypesNorm)</span>
<span id="cb421-10"><a href="whole-genome-sequencing-mutations.html#cb421-10"></a><span class="kw">ggplot</span>(t2[<span class="kw">which</span>(t2<span class="op">$</span>value<span class="op">&lt;</span><span class="fl">0.04</span>),  ], <span class="kw">aes</span>(<span class="dt">x=</span>Var1, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var2))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;frequency snp type&quot;</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;dataset&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-138-2.png" width="672" /></p>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="whole-genome-sequencing-mutations.html#cb422-1"></a>SearchVars=<span class="kw">c</span>(<span class="st">&quot;missense_variant&quot;</span>,<span class="st">&quot;stop_lost&quot;</span>, <span class="st">&quot;stop_gained&quot;</span>, <span class="st">&quot;initiator_codon&quot;</span>, <span class="st">&quot;start_lost&quot;</span>, <span class="st">&quot;frameshift_variant&quot;</span>)</span>
<span id="cb422-2"><a href="whole-genome-sequencing-mutations.html#cb422-2"></a>NIHr2=NIHr[<span class="kw">which</span>(ModType<span class="op">%in%</span>SearchVars)]</span>
<span id="cb422-3"><a href="whole-genome-sequencing-mutations.html#cb422-3"></a>harr2=harr[<span class="kw">which</span>(HarType<span class="op">%in%</span>SearchVars)]</span>
<span id="cb422-4"><a href="whole-genome-sequencing-mutations.html#cb422-4"></a>crr2=crr[<span class="kw">which</span>(CRType<span class="op">%in%</span>SearchVars)]</span>
<span id="cb422-5"><a href="whole-genome-sequencing-mutations.html#cb422-5"></a></span>
<span id="cb422-6"><a href="whole-genome-sequencing-mutations.html#cb422-6"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">NIH=</span>NIHr2, <span class="dt">harlan=</span>harr2, <span class="dt">CR=</span>crr2))</span>
<span id="cb422-7"><a href="whole-genome-sequencing-mutations.html#cb422-7"></a><span class="kw">mtext</span>(<span class="st">&quot;any coding variant&quot;</span>, <span class="dt">side=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-139-1.png" width="672" /></p>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="whole-genome-sequencing-mutations.html#cb423-1"></a></span>
<span id="cb423-2"><a href="whole-genome-sequencing-mutations.html#cb423-2"></a><span class="co">## Genes affected</span></span>
<span id="cb423-3"><a href="whole-genome-sequencing-mutations.html#cb423-3"></a>r=<span class="kw">regexpr</span>(<span class="st">&quot;[A-Za-z0-9</span><span class="ch">\\</span><span class="st">.]+</span><span class="ch">\\</span><span class="st">|ENSRNOG&quot;</span>, harlanRats<span class="op">@</span>fix[<span class="kw">which</span>(HarType<span class="op">%in%</span>SearchVars),<span class="dv">8</span>])</span>
<span id="cb423-4"><a href="whole-genome-sequencing-mutations.html#cb423-4"></a>codhrgrp=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(<span class="kw">which</span>(HarType<span class="op">%in%</span>SearchVars)))</span>
<span id="cb423-5"><a href="whole-genome-sequencing-mutations.html#cb423-5"></a>codhrgrp[r<span class="op">!=-</span><span class="dv">1</span>]=<span class="kw">regmatches</span>(harlanRats<span class="op">@</span>fix[<span class="kw">which</span>(HarType<span class="op">%in%</span>SearchVars),<span class="dv">8</span>], r)</span>
<span id="cb423-6"><a href="whole-genome-sequencing-mutations.html#cb423-6"></a>codhrgrp=<span class="kw">substr</span>(codhrgrp, <span class="dv">1</span>, <span class="kw">nchar</span>(codhrgrp)<span class="op">-</span><span class="dv">8</span>)</span>
<span id="cb423-7"><a href="whole-genome-sequencing-mutations.html#cb423-7"></a></span>
<span id="cb423-8"><a href="whole-genome-sequencing-mutations.html#cb423-8"></a>r=<span class="kw">regexpr</span>(<span class="st">&quot;[A-Za-z0-9</span><span class="ch">\\</span><span class="st">.]+</span><span class="ch">\\</span><span class="st">|ENSRNOG&quot;</span>, CRRats<span class="op">@</span>fix[<span class="kw">which</span>(CRType<span class="op">%in%</span>SearchVars),<span class="dv">8</span>])</span>
<span id="cb423-9"><a href="whole-genome-sequencing-mutations.html#cb423-9"></a>codcrgrp=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(<span class="kw">which</span>(CRType<span class="op">%in%</span>SearchVars)))</span>
<span id="cb423-10"><a href="whole-genome-sequencing-mutations.html#cb423-10"></a>codcrgrp[r<span class="op">!=-</span><span class="dv">1</span>]=<span class="kw">regmatches</span>(CRRats<span class="op">@</span>fix[<span class="kw">which</span>(CRType<span class="op">%in%</span>SearchVars),<span class="dv">8</span>], r)</span>
<span id="cb423-11"><a href="whole-genome-sequencing-mutations.html#cb423-11"></a>codcrgrp=<span class="kw">substr</span>(codhrgrp, <span class="dv">1</span>, <span class="kw">nchar</span>(codcrgrp)<span class="op">-</span><span class="dv">8</span>)</span>
<span id="cb423-12"><a href="whole-genome-sequencing-mutations.html#cb423-12"></a></span>
<span id="cb423-13"><a href="whole-genome-sequencing-mutations.html#cb423-13"></a>r=<span class="kw">regexpr</span>(<span class="st">&quot;[A-Za-z0-9</span><span class="ch">\\</span><span class="st">.]+</span><span class="ch">\\</span><span class="st">|ENSRNOG&quot;</span>, nihRats<span class="op">@</span>fix[<span class="kw">which</span>(ModType<span class="op">%in%</span>SearchVars),<span class="dv">8</span>])</span>
<span id="cb423-14"><a href="whole-genome-sequencing-mutations.html#cb423-14"></a>codnihgrp=<span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(<span class="kw">which</span>(ModType<span class="op">%in%</span>SearchVars)))</span>
<span id="cb423-15"><a href="whole-genome-sequencing-mutations.html#cb423-15"></a>codnihgrp[r<span class="op">!=-</span><span class="dv">1</span>]=<span class="kw">regmatches</span>(nihRats<span class="op">@</span>fix[<span class="kw">which</span>(ModType<span class="op">%in%</span>SearchVars),<span class="dv">8</span>], r)</span>
<span id="cb423-16"><a href="whole-genome-sequencing-mutations.html#cb423-16"></a>codnihgrp=<span class="kw">substr</span>(codnihgrp, <span class="dv">1</span>, <span class="kw">nchar</span>(codnihgrp)<span class="op">-</span><span class="dv">8</span>)</span>
<span id="cb423-17"><a href="whole-genome-sequencing-mutations.html#cb423-17"></a></span>
<span id="cb423-18"><a href="whole-genome-sequencing-mutations.html#cb423-18"></a><span class="kw">venn</span>(<span class="kw">list</span>(<span class="dt">NIH=</span>codnihgrp, <span class="dt">harlan=</span>codhrgrp, <span class="dt">CR=</span>codcrgrp))</span>
<span id="cb423-19"><a href="whole-genome-sequencing-mutations.html#cb423-19"></a><span class="kw">mtext</span>(<span class="st">&quot;any cosmic coding variant&quot;</span>, <span class="dt">side=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-139-2.png" width="672" /></p>
<div class="sourceCode" id="cb424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb424-1"><a href="whole-genome-sequencing-mutations.html#cb424-1"></a></span>
<span id="cb424-2"><a href="whole-genome-sequencing-mutations.html#cb424-2"></a><span class="co">## Look for Cosmic specific genes</span></span>
<span id="cb424-3"><a href="whole-genome-sequencing-mutations.html#cb424-3"></a>codcrgrp2=codcrgrp[codcrgrp<span class="op">%in%</span>RatCosmic]</span>
<span id="cb424-4"><a href="whole-genome-sequencing-mutations.html#cb424-4"></a>codhrgrp2=codhrgrp[codhrgrp<span class="op">%in%</span>RatCosmic]</span>
<span id="cb424-5"><a href="whole-genome-sequencing-mutations.html#cb424-5"></a>codnihgrp2=codnihgrp[codnihgrp<span class="op">%in%</span>RatCosmic] <span class="co">#c(RatBreastCosmic, codhrgrp2, &quot;Tert&quot;, &quot;Atm&quot;)]</span></span></code></pre></div>
<p>Look at the distribution of mutations:</p>
<p>Whole genome</p>
<ul>
<li>1188 genes in more than 2 animals (1435 in at least 2)</li>
<li>59 variants were identified as population variant in Harlan</li>
<li>47 in charles river (6 are novel in this cohort only)</li>
</ul>
<p>Cosmic genes:</p>
<ul>
<li>42 sites which have mutations in more than 2 animals</li>
<li>50 sites in more than 1 animal</li>
<li>5 sites which have been identified in GWAS studies of harlan rats. On average, the variant is found at a frequency of ~15%. The CR set, these variants appear in 50% of the population</li>
<li>4 of the 5 rats from above were identified in</li>
<li>0 sites in the nih dbSNP database for rats</li>
</ul>
<div class="sourceCode" id="cb425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb425-1"><a href="whole-genome-sequencing-mutations.html#cb425-1"></a><span class="co"># Check TGFBr polymorphisms:</span></span>
<span id="cb425-2"><a href="whole-genome-sequencing-mutations.html#cb425-2"></a></span>
<span id="cb425-3"><a href="whole-genome-sequencing-mutations.html#cb425-3"></a>r=<span class="kw">grep</span>(<span class="st">&quot;Tgfb&quot;</span>, harlanRats<span class="op">@</span>fix[,<span class="dv">8</span>])</span>
<span id="cb425-4"><a href="whole-genome-sequencing-mutations.html#cb425-4"></a>harlanRatsTgfb=harlanRats[r]</span>
<span id="cb425-5"><a href="whole-genome-sequencing-mutations.html#cb425-5"></a><span class="kw">write.vcf</span>(harlanRatsTgfb, <span class="dt">file=</span><span class="st">&quot;~/Desktop/harlanRats_Tgfb.vcf.gz&quot;</span>)</span>
<span id="cb425-6"><a href="whole-genome-sequencing-mutations.html#cb425-6"></a></span>
<span id="cb425-7"><a href="whole-genome-sequencing-mutations.html#cb425-7"></a>r=<span class="kw">grep</span>(<span class="st">&quot;Tgfb&quot;</span>, CRRats<span class="op">@</span>fix[,<span class="dv">8</span>])</span>
<span id="cb425-8"><a href="whole-genome-sequencing-mutations.html#cb425-8"></a>CRRatsTgfb=CRRats[r]</span>
<span id="cb425-9"><a href="whole-genome-sequencing-mutations.html#cb425-9"></a><span class="kw">write.vcf</span>(CRRatsTgfb, <span class="dt">file=</span><span class="st">&quot;~/Desktop/CRRats_Tgfb.vcf.gz&quot;</span>)</span>
<span id="cb425-10"><a href="whole-genome-sequencing-mutations.html#cb425-10"></a></span>
<span id="cb425-11"><a href="whole-genome-sequencing-mutations.html#cb425-11"></a>r=<span class="kw">grep</span>(<span class="st">&quot;Tgfb&quot;</span>, nihRats<span class="op">@</span>fix[,<span class="dv">8</span>])</span>
<span id="cb425-12"><a href="whole-genome-sequencing-mutations.html#cb425-12"></a>nihRatsTgfb=nihRats[r]</span>
<span id="cb425-13"><a href="whole-genome-sequencing-mutations.html#cb425-13"></a><span class="kw">write.vcf</span>(nihRatsTgfb, <span class="dt">file=</span><span class="st">&quot;~/Desktop/nih_Tgfb.vcf.gz&quot;</span>)</span></code></pre></div>
<p>Here, compare the frequencies of the most common mutations in our population with those reported in the harlan and charles river data-bases:</p>
<ul>
<li>filter out mutatations occuring in 2+ samples</li>
</ul>
<div class="sourceCode" id="cb426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb426-1"><a href="whole-genome-sequencing-mutations.html#cb426-1"></a>Nsamp=<span class="dv">1</span></span>
<span id="cb426-2"><a href="whole-genome-sequencing-mutations.html#cb426-2"></a><span class="co">## Firstly whole genome</span></span>
<span id="cb426-3"><a href="whole-genome-sequencing-mutations.html#cb426-3"></a>lx1=<span class="kw">table</span>(<span class="kw">paste</span>(MeltWGSCoding<span class="op">$</span>Chromosome, MeltWGSCoding<span class="op">$</span>Start_Position))</span>
<span id="cb426-4"><a href="whole-genome-sequencing-mutations.html#cb426-4"></a>lx1=<span class="kw">sort</span>(lx1, <span class="dt">decreasing=</span>T)</span>
<span id="cb426-5"><a href="whole-genome-sequencing-mutations.html#cb426-5"></a><span class="kw">hist</span>(lx1)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/pop-gen-plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="whole-genome-sequencing-mutations.html#cb427-1"></a><span class="kw">length</span>(<span class="kw">which</span>(lx1<span class="op">&gt;</span>Nsamp))</span>
<span id="cb427-2"><a href="whole-genome-sequencing-mutations.html#cb427-2"></a><span class="co">## [1] 1346</span></span>
<span id="cb427-3"><a href="whole-genome-sequencing-mutations.html#cb427-3"></a>WGSList=<span class="kw">names</span>(lx1)[<span class="kw">which</span>(lx1<span class="op">&gt;</span>Nsamp)]</span>
<span id="cb427-4"><a href="whole-genome-sequencing-mutations.html#cb427-4"></a>WGSList=<span class="kw">strsplit</span>(WGSList, <span class="st">&quot; &quot;</span>)</span>
<span id="cb427-5"><a href="whole-genome-sequencing-mutations.html#cb427-5"></a>WGSList=<span class="kw">data.frame</span>(<span class="dt">chr=</span><span class="kw">sapply</span>(WGSList, <span class="cf">function</span>(x) x[<span class="dv">1</span>]), <span class="dt">pos=</span><span class="kw">sapply</span>(WGSList, <span class="cf">function</span>(x) x[<span class="dv">2</span>]), <span class="dt">freq=</span>lx1[<span class="kw">which</span>(lx1<span class="op">&gt;</span>Nsamp)])</span>
<span id="cb427-6"><a href="whole-genome-sequencing-mutations.html#cb427-6"></a>WGSList<span class="op">$</span>gene=<span class="kw">as.character</span>(MeltWGSCoding<span class="op">$</span>Gene_Symbol[<span class="kw">match</span>(WGSList[ ,<span class="dv">2</span>], MeltWGSCoding<span class="op">$</span>Start_Position)])</span>
<span id="cb427-7"><a href="whole-genome-sequencing-mutations.html#cb427-7"></a>WGSList<span class="op">$</span>AA=<span class="kw">as.character</span>(MeltWGSCoding<span class="op">$</span>HGVSp_Short[<span class="kw">match</span>(WGSList[ ,<span class="dv">2</span>], MeltWGSCoding<span class="op">$</span>Start_Position)])</span>
<span id="cb427-8"><a href="whole-genome-sequencing-mutations.html#cb427-8"></a>WGSList<span class="op">$</span>VariantClass=<span class="kw">as.character</span>(MeltWGSCoding<span class="op">$</span>Variant_Classification[<span class="kw">match</span>(WGSList[ ,<span class="dv">2</span>], MeltWGSCoding<span class="op">$</span>Start_Position)])</span>
<span id="cb427-9"><a href="whole-genome-sequencing-mutations.html#cb427-9"></a>WGSList<span class="op">$</span>Cosmic=<span class="kw">ifelse</span>(WGSList<span class="op">$</span>gene<span class="op">%in%</span>RatCosmic, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb427-10"><a href="whole-genome-sequencing-mutations.html#cb427-10"></a></span>
<span id="cb427-11"><a href="whole-genome-sequencing-mutations.html#cb427-11"></a><span class="co">## Secondly, cosmic specific genes</span></span>
<span id="cb427-12"><a href="whole-genome-sequencing-mutations.html#cb427-12"></a></span>
<span id="cb427-13"><a href="whole-genome-sequencing-mutations.html#cb427-13"></a><span class="co">## WGSList</span></span>
<span id="cb427-14"><a href="whole-genome-sequencing-mutations.html#cb427-14"></a>midx=<span class="kw">match</span>(WGSList<span class="op">$</span>freq.Var1, <span class="kw">paste</span>(harlanRats<span class="op">@</span>fix[ ,<span class="dv">1</span>], harlanRats<span class="op">@</span>fix[ ,<span class="dv">2</span>], <span class="dt">sep=</span><span class="st">&quot; &quot;</span>))</span>
<span id="cb427-15"><a href="whole-genome-sequencing-mutations.html#cb427-15"></a>WGSList<span class="op">$</span>HarlanFreq=GT3[midx]</span>
<span id="cb427-16"><a href="whole-genome-sequencing-mutations.html#cb427-16"></a>midx=<span class="kw">match</span>(WGSList<span class="op">$</span>freq.Var1, <span class="kw">paste</span>(nihRats<span class="op">@</span>fix[ ,<span class="dv">1</span>], nihRats<span class="op">@</span>fix[ ,<span class="dv">2</span>], <span class="dt">sep=</span><span class="st">&quot; &quot;</span>))</span>
<span id="cb427-17"><a href="whole-genome-sequencing-mutations.html#cb427-17"></a>WGSList<span class="op">$</span>rsID=nihRats<span class="op">@</span>fix[midx, <span class="dv">3</span>]</span>
<span id="cb427-18"><a href="whole-genome-sequencing-mutations.html#cb427-18"></a><span class="co">## CR</span></span>
<span id="cb427-19"><a href="whole-genome-sequencing-mutations.html#cb427-19"></a>midx=<span class="kw">match</span>(WGSList<span class="op">$</span>freq.Var1, <span class="kw">paste</span>(CRRats<span class="op">@</span>fix[ ,<span class="dv">1</span>], CRRats<span class="op">@</span>fix[ ,<span class="dv">2</span>], <span class="dt">sep=</span><span class="st">&quot; &quot;</span>))</span>
<span id="cb427-20"><a href="whole-genome-sequencing-mutations.html#cb427-20"></a>WGSList<span class="op">$</span>CRFreq=crGT3[midx]</span>
<span id="cb427-21"><a href="whole-genome-sequencing-mutations.html#cb427-21"></a></span>
<span id="cb427-22"><a href="whole-genome-sequencing-mutations.html#cb427-22"></a>WGSplot=<span class="kw">melt</span>(WGSList[ ,<span class="kw">c</span>(<span class="st">&quot;freq.Freq&quot;</span>, <span class="st">&quot;gene&quot;</span>, <span class="st">&quot;VariantClass&quot;</span>, <span class="st">&quot;HarlanFreq&quot;</span>, <span class="st">&quot;CRFreq&quot;</span>, <span class="st">&quot;Cosmic&quot;</span>)], <span class="dt">measure.vars =</span> <span class="kw">c</span>(<span class="st">&quot;HarlanFreq&quot;</span>, <span class="st">&quot;CRFreq&quot;</span>))</span>
<span id="cb427-23"><a href="whole-genome-sequencing-mutations.html#cb427-23"></a>WGSplot=WGSplot[<span class="op">-</span><span class="kw">which</span>(<span class="kw">is.na</span>(WGSplot<span class="op">$</span>value)), ]</span>
<span id="cb427-24"><a href="whole-genome-sequencing-mutations.html#cb427-24"></a></span>
<span id="cb427-25"><a href="whole-genome-sequencing-mutations.html#cb427-25"></a><span class="kw">ggplot</span>(WGSplot, <span class="kw">aes</span>(<span class="dt">x=</span>freq.Freq<span class="op">/</span><span class="dv">16</span>, <span class="dt">y=</span>value, <span class="dt">col=</span>VariantClass, <span class="dt">label=</span>WGSplot<span class="op">$</span>gene))<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>variable)<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span></span>
<span id="cb427-26"><a href="whole-genome-sequencing-mutations.html#cb427-26"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">size=</span>(WGSplot<span class="op">$</span>Cosmic<span class="op">+</span><span class="dv">3</span>), <span class="dt">col=</span><span class="st">&quot;black&quot;</span>, <span class="dt">nudge_y=</span><span class="fl">0.01</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;Cohort frequency&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Database frequency&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;all genes compared to Harlan Frequencies&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/pop-gen-plot-2.png" width="672" /></p>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="whole-genome-sequencing-mutations.html#cb428-1"></a></span>
<span id="cb428-2"><a href="whole-genome-sequencing-mutations.html#cb428-2"></a> <span class="kw">kable</span>(WGSplot[WGSplot<span class="op">$</span>Cosmic<span class="op">==</span><span class="dv">1</span>, ])</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
freq.Freq
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:left;">
VariantClass
</th>
<th style="text-align:right;">
Cosmic
</th>
<th style="text-align:left;">
variable
</th>
<th style="text-align:right;">
value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
142
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:left;">
Prf1
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HarlanFreq
</td>
<td style="text-align:right;">
0.4528715
</td>
</tr>
<tr>
<td style="text-align:left;">
530
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
Tcf3
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HarlanFreq
</td>
<td style="text-align:right;">
0.0679527
</td>
</tr>
<tr>
<td style="text-align:left;">
676
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
Rnf213
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HarlanFreq
</td>
<td style="text-align:right;">
0.1402893
</td>
</tr>
<tr>
<td style="text-align:left;">
677
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
Rnf213
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HarlanFreq
</td>
<td style="text-align:right;">
0.1402893
</td>
</tr>
<tr>
<td style="text-align:left;">
678
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
Rnf213
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HarlanFreq
</td>
<td style="text-align:right;">
0.1411662
</td>
</tr>
<tr>
<td style="text-align:left;">
1195
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Ptpn13
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HarlanFreq
</td>
<td style="text-align:right;">
0.0271811
</td>
</tr>
<tr>
<td style="text-align:left;">
1876
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
Tcf3
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
CRFreq
</td>
<td style="text-align:right;">
0.1488764
</td>
</tr>
<tr>
<td style="text-align:left;">
2022
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
Rnf213
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
CRFreq
</td>
<td style="text-align:right;">
0.5264045
</td>
</tr>
<tr>
<td style="text-align:left;">
2023
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
Rnf213
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
CRFreq
</td>
<td style="text-align:right;">
0.5264045
</td>
</tr>
<tr>
<td style="text-align:left;">
2024
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
Rnf213
</td>
<td style="text-align:left;">
Missense_Mutation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
CRFreq
</td>
<td style="text-align:right;">
0.5365169
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="overview-of-the-mutations" class="section level2">
<h2><span class="header-section-number">20.4</span> Overview of the mutations</h2>
<div id="all-mutations" class="section level3">
<h3><span class="header-section-number">20.4.1</span> All mutations</h3>
<p>SNPs are removed from this analysis: based on frequencies in the harlan cohort</p>
<p>Color codes:</p>
<ul>
<li>black:</li>
<li>pink:</li>
<li>green:</li>
<li>blue:</li>
</ul>
<div class="sourceCode" id="cb429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb429-1"><a href="whole-genome-sequencing-mutations.html#cb429-1"></a><span class="kw">palette</span>(<span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Set1&quot;</span>))</span>
<span id="cb429-2"><a href="whole-genome-sequencing-mutations.html#cb429-2"></a></span>
<span id="cb429-3"><a href="whole-genome-sequencing-mutations.html#cb429-3"></a>Nmut=<span class="dv">86</span> <span class="co">## was previously 92: this appears at least 17 times</span></span>
<span id="cb429-4"><a href="whole-genome-sequencing-mutations.html#cb429-4"></a></span>
<span id="cb429-5"><a href="whole-genome-sequencing-mutations.html#cb429-5"></a>rmHarlanSNP=WGSList[<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(WGSList<span class="op">$</span>HarlanFreq)), ]</span>
<span id="cb429-6"><a href="whole-genome-sequencing-mutations.html#cb429-6"></a>x1a=<span class="kw">which</span>(MeltWGSCoding<span class="op">$</span>Gene_Symbol<span class="op">%in%</span>rmHarlanSNP<span class="op">$</span>gene <span class="op">&amp;</span><span class="st">  </span>MeltWGSCoding<span class="op">$</span>Start_Position<span class="op">%in%</span>rmHarlanSNP<span class="op">$</span>pos)</span>
<span id="cb429-7"><a href="whole-genome-sequencing-mutations.html#cb429-7"></a>MeltWGSCodingB=MeltWGSCoding[<span class="op">-</span>x1a, ]</span>
<span id="cb429-8"><a href="whole-genome-sequencing-mutations.html#cb429-8"></a></span>
<span id="cb429-9"><a href="whole-genome-sequencing-mutations.html#cb429-9"></a>rGeneOut=<span class="kw">acast</span>(MeltWGSCodingB[ ,<span class="kw">c</span>(<span class="st">&quot;Gene_Symbol&quot;</span>, <span class="st">&quot;L1&quot;</span>, <span class="st">&quot;Variant_Classification&quot;</span>)], L1<span class="op">~</span>Gene_Symbol, </span>
<span id="cb429-10"><a href="whole-genome-sequencing-mutations.html#cb429-10"></a>               <span class="dt">value.var=</span><span class="st">&quot;Variant_Classification&quot;</span>, <span class="dt">fun.aggregate=</span><span class="cf">function</span>(x) <span class="kw">paste</span>(x, <span class="dt">collapse =</span> <span class="st">&quot;, &quot;</span>))</span>
<span id="cb429-11"><a href="whole-genome-sequencing-mutations.html#cb429-11"></a></span>
<span id="cb429-12"><a href="whole-genome-sequencing-mutations.html#cb429-12"></a>mxl=<span class="kw">unique</span>(<span class="kw">as.vector</span>(rGeneOut))</span>
<span id="cb429-13"><a href="whole-genome-sequencing-mutations.html#cb429-13"></a>rGeneOut2=rGeneOut</span>
<span id="cb429-14"><a href="whole-genome-sequencing-mutations.html#cb429-14"></a>rGeneOut2[<span class="kw">grep</span>(<span class="st">&quot;Nonsense&quot;</span>,rGeneOut2)]=<span class="dv">3</span></span>
<span id="cb429-15"><a href="whole-genome-sequencing-mutations.html#cb429-15"></a>rGeneOut2[<span class="kw">grep</span>(<span class="st">&quot;Frame_Shift&quot;</span>,rGeneOut2)]=<span class="dv">2</span></span>
<span id="cb429-16"><a href="whole-genome-sequencing-mutations.html#cb429-16"></a>rGeneOut2[<span class="kw">grep</span>(<span class="st">&quot;Missense_&quot;</span>, rGeneOut2)]=<span class="dv">1</span></span>
<span id="cb429-17"><a href="whole-genome-sequencing-mutations.html#cb429-17"></a>rGeneOut2[<span class="kw">grep</span>(<span class="st">&quot;Splice_&quot;</span>, rGeneOut2)]=<span class="dv">4</span></span>
<span id="cb429-18"><a href="whole-genome-sequencing-mutations.html#cb429-18"></a>rGeneOut2[<span class="kw">which</span>(rGeneOut2<span class="op">==</span><span class="st">&quot;&quot;</span>)]=<span class="dv">0</span></span>
<span id="cb429-19"><a href="whole-genome-sequencing-mutations.html#cb429-19"></a></span>
<span id="cb429-20"><a href="whole-genome-sequencing-mutations.html#cb429-20"></a>rGeneOut2=<span class="kw">apply</span>(rGeneOut2, <span class="dv">2</span>, as.numeric)</span>
<span id="cb429-21"><a href="whole-genome-sequencing-mutations.html#cb429-21"></a><span class="kw">rownames</span>(rGeneOut2)=<span class="kw">rownames</span>(rGeneOut) </span>
<span id="cb429-22"><a href="whole-genome-sequencing-mutations.html#cb429-22"></a></span>
<span id="cb429-23"><a href="whole-genome-sequencing-mutations.html#cb429-23"></a><span class="co">#get rid of genes which are associated with Olfactory receptors, AABR etc mutations</span></span>
<span id="cb429-24"><a href="whole-genome-sequencing-mutations.html#cb429-24"></a>rmThese=<span class="kw">c</span>(<span class="st">&quot;^AABR&quot;</span>, <span class="st">&quot;^Olr&quot;</span>, <span class="st">&quot;^LOC&quot;</span>, <span class="st">&quot;^AC[0-9]+&quot;</span>, <span class="st">&quot;^RGD&quot;</span>, <span class="st">&quot;^Rpl&quot;</span>, <span class="st">&quot;^1700&quot;</span>, <span class="st">&quot;^RT1&quot;</span>, <span class="st">&quot;^ENSRNOG&quot;</span>, <span class="st">&quot;^Vom2&quot;</span>)</span>
<span id="cb429-25"><a href="whole-genome-sequencing-mutations.html#cb429-25"></a>rmList=<span class="kw">unlist</span>(<span class="kw">sapply</span>(rmThese, <span class="cf">function</span>(x) <span class="kw">grep</span>(x, <span class="kw">colnames</span>(rGeneOut2))))</span>
<span id="cb429-26"><a href="whole-genome-sequencing-mutations.html#cb429-26"></a>rGeneOut2=rGeneOut2[ ,<span class="op">-</span>rmList]</span>
<span id="cb429-27"><a href="whole-genome-sequencing-mutations.html#cb429-27"></a></span>
<span id="cb429-28"><a href="whole-genome-sequencing-mutations.html#cb429-28"></a>GeneList=<span class="kw">colSums</span>(<span class="kw">sign</span>(rGeneOut2), <span class="dt">na.rm=</span>T)</span>
<span id="cb429-29"><a href="whole-genome-sequencing-mutations.html#cb429-29"></a>samp2=<span class="kw">rowSums</span>(<span class="kw">sign</span>(rGeneOut2), <span class="dt">na.rm=</span>T)</span>
<span id="cb429-30"><a href="whole-genome-sequencing-mutations.html#cb429-30"></a>rGeneOut2=rGeneOut2[ <span class="kw">order</span>(samp2, <span class="dt">decreasing=</span>T),]</span>
<span id="cb429-31"><a href="whole-genome-sequencing-mutations.html#cb429-31"></a></span>
<span id="cb429-32"><a href="whole-genome-sequencing-mutations.html#cb429-32"></a>topN=<span class="kw">names</span>(<span class="kw">sort</span>(GeneList, <span class="dt">decreasing=</span>T)[<span class="dv">1</span><span class="op">:</span>Nmut])</span>
<span id="cb429-33"><a href="whole-genome-sequencing-mutations.html#cb429-33"></a></span>
<span id="cb429-34"><a href="whole-genome-sequencing-mutations.html#cb429-34"></a><span class="co"># do a plot of the top 30 most common mutations</span></span>
<span id="cb429-35"><a href="whole-genome-sequencing-mutations.html#cb429-35"></a>tempx=rGeneOut2[ ,topN]</span>
<span id="cb429-36"><a href="whole-genome-sequencing-mutations.html#cb429-36"></a>a2=<span class="kw">heatmap</span>(tempx, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">9</span>), <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">main=</span><span class="st">&quot;all mutations in cosmic&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-141-1.png" width="672" /></p>
<div class="sourceCode" id="cb430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb430-1"><a href="whole-genome-sequencing-mutations.html#cb430-1"></a>RowSC=<span class="kw">ifelse</span>(<span class="kw">colnames</span>(tempx)<span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(RatCosmic, <span class="kw">as.character</span>(AllCancerPathwayGenes)), <span class="st">&quot;black&quot;</span>, <span class="st">&quot;white&quot;</span>)</span>
<span id="cb430-2"><a href="whole-genome-sequencing-mutations.html#cb430-2"></a></span>
<span id="cb430-3"><a href="whole-genome-sequencing-mutations.html#cb430-3"></a><span class="kw">heatmap.2</span>(tempx, <span class="dt">Rowv =</span> <span class="ot">NA</span>, <span class="dt">Colv =</span> <span class="ot">NA</span>,<span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">ColSideColors=</span>RowSC)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-141-2.png" width="672" /></p>
<div class="sourceCode" id="cb431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb431-1"><a href="whole-genome-sequencing-mutations.html#cb431-1"></a><span class="co"># check if any of these are hotspot mutations</span></span>
<span id="cb431-2"><a href="whole-genome-sequencing-mutations.html#cb431-2"></a>Hotspots=MeltCosmic[<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(MeltCosmic<span class="op">$</span>HumSequence)<span class="op">==</span>T), ]</span>
<span id="cb431-3"><a href="whole-genome-sequencing-mutations.html#cb431-3"></a><span class="co">#head(Hotspots)</span></span>
<span id="cb431-4"><a href="whole-genome-sequencing-mutations.html#cb431-4"></a></span>
<span id="cb431-5"><a href="whole-genome-sequencing-mutations.html#cb431-5"></a>hotspotLocExtract=<span class="cf">function</span>(amat){</span>
<span id="cb431-6"><a href="whole-genome-sequencing-mutations.html#cb431-6"></a>HList=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(Hotspots), <span class="cf">function</span>(i) <span class="kw">ifelse</span>(amat[<span class="kw">match</span>(Hotspots<span class="op">$</span>L1[i], <span class="kw">rownames</span>(amat)),<span class="kw">match</span>(Hotspots<span class="op">$</span>Gene_Symbol[i], <span class="kw">colnames</span>(amat))]<span class="op">&gt;</span><span class="dv">0</span>, </span>
<span id="cb431-7"><a href="whole-genome-sequencing-mutations.html#cb431-7"></a>                                                 <span class="kw">paste</span>(<span class="kw">match</span>(Hotspots<span class="op">$</span>L1[i], <span class="kw">rownames</span>(amat)),<span class="kw">match</span>(Hotspots<span class="op">$</span>Gene_Symbol[i], <span class="kw">colnames</span>(amat))), <span class="st">&quot;0 0&quot;</span>))</span>
<span id="cb431-8"><a href="whole-genome-sequencing-mutations.html#cb431-8"></a>List2=<span class="kw">strsplit</span>(<span class="kw">as.character</span>(HList), <span class="st">&quot; &quot;</span>)</span>
<span id="cb431-9"><a href="whole-genome-sequencing-mutations.html#cb431-9"></a>Xcord=<span class="kw">na.omit</span>(<span class="kw">as.numeric</span>(<span class="kw">sapply</span>(List2, <span class="cf">function</span>(x) x[<span class="dv">1</span>])))</span>
<span id="cb431-10"><a href="whole-genome-sequencing-mutations.html#cb431-10"></a>Ycord=<span class="kw">na.omit</span>(<span class="kw">as.numeric</span>(<span class="kw">sapply</span>(List2, <span class="cf">function</span>(x) x[<span class="dv">2</span>])))</span>
<span id="cb431-11"><a href="whole-genome-sequencing-mutations.html#cb431-11"></a><span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">X=</span>Xcord, <span class="dt">Y=</span>Ycord))</span>
<span id="cb431-12"><a href="whole-genome-sequencing-mutations.html#cb431-12"></a>}</span>
<span id="cb431-13"><a href="whole-genome-sequencing-mutations.html#cb431-13"></a></span>
<span id="cb431-14"><a href="whole-genome-sequencing-mutations.html#cb431-14"></a><span class="co">## top 90 mutations</span></span>
<span id="cb431-15"><a href="whole-genome-sequencing-mutations.html#cb431-15"></a>a2=<span class="kw">heatmap.2</span>(<span class="kw">sign</span>(tempx), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;grey&quot;</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-141-3.png" width="672" /></p>
<div class="sourceCode" id="cb432"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb432-1"><a href="whole-genome-sequencing-mutations.html#cb432-1"></a>tempxb=tempx[a2<span class="op">$</span>rowInd, a2<span class="op">$</span>colInd]</span>
<span id="cb432-2"><a href="whole-genome-sequencing-mutations.html#cb432-2"></a>hotspot1b=<span class="kw">hotspotLocExtract</span>(tempxb)</span>
<span id="cb432-3"><a href="whole-genome-sequencing-mutations.html#cb432-3"></a><span class="kw">rownames</span>(tempxb)=Cdata<span class="op">$</span>NewID[<span class="kw">sapply</span>(<span class="kw">rownames</span>(tempxb), <span class="cf">function</span>(x) <span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS))]</span>
<span id="cb432-4"><a href="whole-genome-sequencing-mutations.html#cb432-4"></a></span>
<span id="cb432-5"><a href="whole-genome-sequencing-mutations.html#cb432-5"></a>GeneList=<span class="kw">unique</span>(Hotspots<span class="op">$</span>Gene_Symbol)</span>
<span id="cb432-6"><a href="whole-genome-sequencing-mutations.html#cb432-6"></a>Hotspots<span class="op">$</span>Gene_Symbol</span>
<span id="cb432-7"><a href="whole-genome-sequencing-mutations.html#cb432-7"></a><span class="co">##   [1] &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Cxcr4&quot;    &quot;Nrg1&quot;     &quot;Atp1a1&quot;   &quot;Bcl10&quot;   </span></span>
<span id="cb432-8"><a href="whole-genome-sequencing-mutations.html#cb432-8"></a><span class="co">##   [7] &quot;Plcg1&quot;    &quot;Ddx6&quot;     &quot;Col3a1&quot;   &quot;Tert&quot;     &quot;Mecom&quot;    &quot;Tert&quot;    </span></span>
<span id="cb432-9"><a href="whole-genome-sequencing-mutations.html#cb432-9"></a><span class="co">##  [13] &quot;Polg&quot;     &quot;Ciita&quot;    &quot;Rnf213&quot;   &quot;Epha3&quot;    &quot;Pik3r1&quot;   &quot;Ros1&quot;    </span></span>
<span id="cb432-10"><a href="whole-genome-sequencing-mutations.html#cb432-10"></a><span class="co">##  [19] &quot;Braf&quot;     &quot;Tert&quot;     &quot;Tlx1&quot;     &quot;Rnf213&quot;   &quot;Epha3&quot;    &quot;Fat1&quot;    </span></span>
<span id="cb432-11"><a href="whole-genome-sequencing-mutations.html#cb432-11"></a><span class="co">##  [25] &quot;Ros1&quot;     &quot;Jun&quot;      &quot;Tert&quot;     &quot;Esr1&quot;     &quot;Rnf213&quot;   &quot;Aspscr1&quot; </span></span>
<span id="cb432-12"><a href="whole-genome-sequencing-mutations.html#cb432-12"></a><span class="co">##  [31] &quot;Epha3&quot;    &quot;Cacna1d&quot;  &quot;Tbl1xr1&quot;  &quot;Ros1&quot;     &quot;Ros1&quot;     &quot;Arid1a&quot;  </span></span>
<span id="cb432-13"><a href="whole-genome-sequencing-mutations.html#cb432-13"></a><span class="co">##  [37] &quot;Brd4&quot;     &quot;Col3a1&quot;   &quot;Tert&quot;     &quot;Pold1&quot;    &quot;Ncor2&quot;    &quot;Clip1&quot;   </span></span>
<span id="cb432-14"><a href="whole-genome-sequencing-mutations.html#cb432-14"></a><span class="co">##  [43] &quot;Ros1&quot;     &quot;Tcf3&quot;     &quot;Ptprb&quot;    &quot;Bcl9l&quot;    &quot;Tert&quot;     &quot;Pold1&quot;   </span></span>
<span id="cb432-15"><a href="whole-genome-sequencing-mutations.html#cb432-15"></a><span class="co">##  [49] &quot;Hras&quot;     &quot;Robo2&quot;    &quot;Robo2&quot;    &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Mecom&quot;   </span></span>
<span id="cb432-16"><a href="whole-genome-sequencing-mutations.html#cb432-16"></a><span class="co">##  [55] &quot;Prcc&quot;     &quot;Ros1&quot;     &quot;Src&quot;      &quot;Stag1&quot;    &quot;Bcorl1&quot;   &quot;Tert&quot;    </span></span>
<span id="cb432-17"><a href="whole-genome-sequencing-mutations.html#cb432-17"></a><span class="co">##  [61] &quot;Il21r&quot;    &quot;Aff4&quot;     &quot;Robo2&quot;    &quot;Pms2&quot;     &quot;Akt3&quot;     &quot;Mecom&quot;   </span></span>
<span id="cb432-18"><a href="whole-genome-sequencing-mutations.html#cb432-18"></a><span class="co">##  [67] &quot;Ros1&quot;     &quot;Acvr1&quot;    &quot;Tal2&quot;     &quot;Smarca4&quot;  &quot;Tert&quot;     &quot;Polg&quot;    </span></span>
<span id="cb432-19"><a href="whole-genome-sequencing-mutations.html#cb432-19"></a><span class="co">##  [73] &quot;Palb2&quot;    &quot;Cpeb3&quot;    &quot;Rnf43&quot;    &quot;Prkar1a&quot;  &quot;Epha3&quot;    &quot;Eif3el1&quot; </span></span>
<span id="cb432-20"><a href="whole-genome-sequencing-mutations.html#cb432-20"></a><span class="co">##  [79] &quot;N4bp2&quot;    &quot;Fgfr3&quot;    &quot;Fgfr3&quot;    &quot;Crtc1&quot;    &quot;Apc&quot;      &quot;Mecom&quot;   </span></span>
<span id="cb432-21"><a href="whole-genome-sequencing-mutations.html#cb432-21"></a><span class="co">##  [85] &quot;Trim33&quot;   &quot;Ros1&quot;     &quot;Zfp384&quot;   &quot;Hif1a&quot;    &quot;Tcf3&quot;     &quot;Ptprb&quot;   </span></span>
<span id="cb432-22"><a href="whole-genome-sequencing-mutations.html#cb432-22"></a><span class="co">##  [91] &quot;Ep300&quot;    &quot;Fat3&quot;     &quot;Fat3&quot;     &quot;Rhoa&quot;     &quot;Tert&quot;     &quot;Hras&quot;    </span></span>
<span id="cb432-23"><a href="whole-genome-sequencing-mutations.html#cb432-23"></a><span class="co">##  [97] &quot;Men1&quot;     &quot;Eif3el1&quot;  &quot;Ktn1&quot;     &quot;Mecom&quot;    &quot;Ros1&quot;     &quot;Ss18l1&quot;  </span></span>
<span id="cb432-24"><a href="whole-genome-sequencing-mutations.html#cb432-24"></a><span class="co">## [103] &quot;Fkbp9&quot;    &quot;Tcf3&quot;     &quot;Ptprb&quot;    &quot;Ddx6&quot;     &quot;Sf3b1&quot;    &quot;Tert&quot;    </span></span>
<span id="cb432-25"><a href="whole-genome-sequencing-mutations.html#cb432-25"></a><span class="co">## [109] &quot;Hras&quot;     &quot;Robo2&quot;    &quot;Robo2&quot;    &quot;Eif3el1&quot;  &quot;Patz1&quot;    &quot;Mecom&quot;   </span></span>
<span id="cb432-26"><a href="whole-genome-sequencing-mutations.html#cb432-26"></a><span class="co">## [115] &quot;Pde4dip&quot;  &quot;Ros1&quot;     &quot;Msh2&quot;     &quot;Tcf3&quot;     &quot;Ptprb&quot;    &quot;Bcorl1&quot;  </span></span>
<span id="cb432-27"><a href="whole-genome-sequencing-mutations.html#cb432-27"></a><span class="co">## [121] &quot;Crebbp&quot;   &quot;Rnf213&quot;   &quot;Epha3&quot;    &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Mecom&quot;   </span></span>
<span id="cb432-28"><a href="whole-genome-sequencing-mutations.html#cb432-28"></a><span class="co">## [127] &quot;Tert&quot;     &quot;Pold1&quot;    &quot;Robo2&quot;    &quot;Robo2&quot;    &quot;Ncor2&quot;    &quot;Nf2&quot;     </span></span>
<span id="cb432-29"><a href="whole-genome-sequencing-mutations.html#cb432-29"></a><span class="co">## [133] &quot;Arhgef10&quot; &quot;Mecom&quot;    &quot;Pik3ca&quot;   &quot;Arid1a&quot;   &quot;Tcf3&quot;     &quot;Ebf1&quot;    </span></span>
<span id="cb432-30"><a href="whole-genome-sequencing-mutations.html#cb432-30"></a><span class="co">## [139] &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Cdc73&quot;    &quot;Fat1&quot;     &quot;Ss18&quot;     &quot;Calr&quot;    </span></span>
<span id="cb432-31"><a href="whole-genome-sequencing-mutations.html#cb432-31"></a><span class="co">## [145] &quot;Lmna&quot;     &quot;Trim33&quot;   &quot;Mafb&quot;     &quot;Tnc&quot;      &quot;Fat3&quot;     &quot;Smarca4&quot; </span></span>
<span id="cb432-32"><a href="whole-genome-sequencing-mutations.html#cb432-32"></a><span class="co">## [151] &quot;Foxo4&quot;    &quot;Tert&quot;     &quot;Picalm&quot;   &quot;Hras&quot;     &quot;Ncor2&quot;    &quot;Clip1&quot;   </span></span>
<span id="cb432-33"><a href="whole-genome-sequencing-mutations.html#cb432-33"></a><span class="co">## [157] &quot;Abl2&quot;     &quot;Arhgap26&quot; &quot;Mecom&quot;    &quot;Ctnnd1&quot;   &quot;Plcg1&quot;    &quot;Ret&quot;     </span></span>
<span id="cb432-34"><a href="whole-genome-sequencing-mutations.html#cb432-34"></a><span class="co">## [163] &quot;Prex2&quot;    &quot;Tnc&quot;      &quot;Arid1a&quot;   &quot;Fbxo11&quot;   &quot;Erbb3&quot;    &quot;Ptprb&quot;   </span></span>
<span id="cb432-35"><a href="whole-genome-sequencing-mutations.html#cb432-35"></a><span class="co">## [169] &quot;Med12&quot;    &quot;Tert&quot;     &quot;Hras&quot;     &quot;Rnf213&quot;   &quot;Ncor2&quot;    &quot;Clip1&quot;   </span></span>
<span id="cb432-36"><a href="whole-genome-sequencing-mutations.html#cb432-36"></a><span class="co">## [175] &quot;Kat6b&quot;    &quot;Fat1&quot;     &quot;Ctcf&quot;     &quot;Mecom&quot;    &quot;Nras&quot;     &quot;Nras&quot;    </span></span>
<span id="cb432-37"><a href="whole-genome-sequencing-mutations.html#cb432-37"></a><span class="co">## [181] &quot;Mutyh&quot;    &quot;Dicer1&quot;   &quot;Ptprb&quot;    &quot;Arid2&quot;    &quot;Tert&quot;     &quot;Shtn1&quot;   </span></span>
<span id="cb432-38"><a href="whole-genome-sequencing-mutations.html#cb432-38"></a><span class="co">## [187] &quot;Rnf213&quot;   &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Mlf1&quot;     &quot;Creb3l2&quot;  &quot;Lck&quot;     </span></span>
<span id="cb432-39"><a href="whole-genome-sequencing-mutations.html#cb432-39"></a><span class="co">## [193] &quot;Arid1a&quot;   &quot;Ptprb&quot;    &quot;Fat3&quot;     &quot;Fat3&quot;     &quot;Fat3&quot;     &quot;Muc16&quot;   </span></span>
<span id="cb432-40"><a href="whole-genome-sequencing-mutations.html#cb432-40"></a><span class="co">## [199] &quot;Aff3&quot;     &quot;Tert&quot;     &quot;Rnf213&quot;   &quot;Mecom&quot;    &quot;Snd1&quot;     &quot;Tcf3&quot;    </span></span>
<span id="cb432-41"><a href="whole-genome-sequencing-mutations.html#cb432-41"></a><span class="co">## [205] &quot;Tert&quot;     &quot;Tfpt&quot;     &quot;Ros1&quot;     &quot;Notch1&quot;   &quot;Akap9&quot;    &quot;Ptprd&quot;   </span></span>
<span id="cb432-42"><a href="whole-genome-sequencing-mutations.html#cb432-42"></a><span class="co">## [211] &quot;Etv1&quot;     &quot;Foxa1&quot;    &quot;Ctnnb1&quot;   &quot;Zmym3&quot;    &quot;Tert&quot;     &quot;Hras&quot;    </span></span>
<span id="cb432-43"><a href="whole-genome-sequencing-mutations.html#cb432-43"></a><span class="co">## [217] &quot;Gnaq&quot;     &quot;Cltc&quot;     &quot;Pde4dip&quot;  &quot;Ros1&quot;     &quot;Akap9&quot;    &quot;Braf&quot;    </span></span>
<span id="cb432-44"><a href="whole-genome-sequencing-mutations.html#cb432-44"></a><span class="co">## [223] &quot;Foxa1&quot;    &quot;Tert&quot;     &quot;Pold1&quot;    &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Mecom&quot;   </span></span>
<span id="cb432-45"><a href="whole-genome-sequencing-mutations.html#cb432-45"></a><span class="co">## [229] &quot;Sdha&quot;     &quot;Tert&quot;     &quot;Men1&quot;     &quot;Pten&quot;     &quot;Snx29&quot;    &quot;Usp32&quot;   </span></span>
<span id="cb432-46"><a href="whole-genome-sequencing-mutations.html#cb432-46"></a><span class="co">## [235] &quot;Zcchc8&quot;   &quot;Tbx3&quot;     &quot;Mdm4&quot;     &quot;Kat6b&quot;    &quot;Mecom&quot;    &quot;Pik3ca&quot;  </span></span>
<span id="cb432-47"><a href="whole-genome-sequencing-mutations.html#cb432-47"></a><span class="co">## [241] &quot;Bcl9&quot;     &quot;Pde4dip&quot;  &quot;Gopc&quot;     &quot;Gopc&quot;     &quot;Set&quot;      &quot;Creb3l2&quot; </span></span>
<span id="cb432-48"><a href="whole-genome-sequencing-mutations.html#cb432-48"></a><span class="co">## [247] &quot;Erc1&quot;     &quot;Sfpq&quot;     &quot;Arid1a&quot;   &quot;Arhgap5&quot;  &quot;Cbl&quot;      &quot;Atm&quot;     </span></span>
<span id="cb432-49"><a href="whole-genome-sequencing-mutations.html#cb432-49"></a><span class="co">## [253] &quot;Atr&quot;      &quot;Mllt1&quot;    &quot;Flna&quot;     &quot;Tert&quot;     &quot;Hras&quot;     &quot;Tp53&quot;    </span></span>
<span id="cb432-50"><a href="whole-genome-sequencing-mutations.html#cb432-50"></a><span class="co">## [259] &quot;Epha3&quot;    &quot;Cblb&quot;     &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Ddr2&quot;     &quot;Pdgfra&quot;  </span></span>
<span id="cb432-51"><a href="whole-genome-sequencing-mutations.html#cb432-51"></a><span class="co">## [265] &quot;Pdgfra&quot;   &quot;Dcc&quot;      &quot;Mecom&quot;    &quot;Pik3ca&quot;   &quot;Fat4&quot;     &quot;Sdc4&quot;    </span></span>
<span id="cb432-52"><a href="whole-genome-sequencing-mutations.html#cb432-52"></a><span class="co">## [271] &quot;Erc1&quot;     &quot;Epha7&quot;    &quot;Prdm2&quot;    &quot;Ptprb&quot;    &quot;Atm&quot;      &quot;Tert&quot;    </span></span>
<span id="cb432-53"><a href="whole-genome-sequencing-mutations.html#cb432-53"></a><span class="co">## [277] &quot;Smarce1&quot;  &quot;Dgcr8&quot;    &quot;Ncor2&quot;    &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Mn1&quot;     </span></span>
<span id="cb432-54"><a href="whole-genome-sequencing-mutations.html#cb432-54"></a><span class="co">## [283] &quot;Trim27&quot;   &quot;Arhgap26&quot; &quot;Mecom&quot;    &quot;Ppp6c&quot;    &quot;Ext2&quot;     &quot;Mitf&quot;    </span></span>
<span id="cb432-55"><a href="whole-genome-sequencing-mutations.html#cb432-55"></a><span class="co">## [289] &quot;Ret&quot;      &quot;Usp44&quot;    &quot;Ptprb&quot;    &quot;Med12&quot;    &quot;Flna&quot;     &quot;Tert&quot;    </span></span>
<span id="cb432-56"><a href="whole-genome-sequencing-mutations.html#cb432-56"></a><span class="co">## [295] &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Casp3&quot;    &quot;Fat1&quot;     &quot;Fat1&quot;     &quot;Cyld&quot;    </span></span>
<span id="cb432-57"><a href="whole-genome-sequencing-mutations.html#cb432-57"></a><span class="co">## [301] &quot;Mecom&quot;    &quot;Foxa1&quot;    &quot;Dnm2&quot;     &quot;Smarca4&quot;  &quot;Tert&quot;     &quot;Pold1&quot;   </span></span>
<span id="cb432-58"><a href="whole-genome-sequencing-mutations.html#cb432-58"></a><span class="co">## [307] &quot;Crebbp&quot;   &quot;Rara&quot;     &quot;Robo2&quot;    &quot;Cxcr4&quot;    &quot;Ank1&quot;     &quot;Syk&quot;     </span></span>
<span id="cb432-59"><a href="whole-genome-sequencing-mutations.html#cb432-59"></a><span class="co">## [313] &quot;Csf1r&quot;    &quot;Cyld&quot;     &quot;Pde4dip&quot;  &quot;Ros1&quot;     &quot;Snd1&quot;     &quot;Ncoa1&quot;   </span></span>
<span id="cb432-60"><a href="whole-genome-sequencing-mutations.html#cb432-60"></a><span class="co">## [319] &quot;Foxa1&quot;    &quot;Foxo4&quot;    &quot;Flna&quot;     &quot;Tert&quot;     &quot;Cltc&quot;     &quot;Stat5b&quot;  </span></span>
<span id="cb432-61"><a href="whole-genome-sequencing-mutations.html#cb432-61"></a><span class="co">## [325] &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Cacna1d&quot;  &quot;Casp3&quot;    &quot;Fat1&quot;     &quot;Fat1&quot;    </span></span>
<span id="cb432-62"><a href="whole-genome-sequencing-mutations.html#cb432-62"></a><span class="co">## [331] &quot;Mecom&quot;    &quot;Ranbp2&quot;   &quot;Thrap3&quot;   &quot;Foxa1&quot;    &quot;Vav1&quot;     &quot;Rbm10&quot;   </span></span>
<span id="cb432-63"><a href="whole-genome-sequencing-mutations.html#cb432-63"></a><span class="co">## [337] &quot;Tert&quot;     &quot;Ercc2&quot;    &quot;Chd2&quot;     &quot;Ncor2&quot;    &quot;Clip1&quot;    &quot;Hook3&quot;   </span></span>
<span id="cb432-64"><a href="whole-genome-sequencing-mutations.html#cb432-64"></a><span class="co">## [343] &quot;Ros1&quot;     &quot;Plcg1&quot;    &quot;Arid1a&quot;   &quot;Ptprb&quot;    &quot;Fat3&quot;     &quot;Fat3&quot;    </span></span>
<span id="cb432-65"><a href="whole-genome-sequencing-mutations.html#cb432-65"></a><span class="co">## [349] &quot;Fat3&quot;     &quot;Sf3b1&quot;    &quot;Ncor2&quot;    &quot;Mecom&quot;    &quot;Akap9&quot;    &quot;Gli1&quot;    </span></span>
<span id="cb432-66"><a href="whole-genome-sequencing-mutations.html#cb432-66"></a><span class="co">## [355] &quot;Tert&quot;     &quot;Men1&quot;     &quot;Myh11&quot;    &quot;Rabep1&quot;   &quot;Ncor2&quot;    &quot;Clip1&quot;   </span></span>
<span id="cb432-67"><a href="whole-genome-sequencing-mutations.html#cb432-67"></a><span class="co">## [361] &quot;Pbrm1&quot;    &quot;Itgav&quot;    &quot;Gata2&quot;    &quot;Zfp384&quot;   &quot;Klf4&quot;     &quot;Strn&quot;    </span></span>
<span id="cb432-68"><a href="whole-genome-sequencing-mutations.html#cb432-68"></a><span class="co">## [367] &quot;Tcf3&quot;     &quot;Fat3&quot;     &quot;Fat3&quot;     &quot;Fat3&quot;     &quot;Cbl&quot;      &quot;Med12&quot;   </span></span>
<span id="cb432-69"><a href="whole-genome-sequencing-mutations.html#cb432-69"></a><span class="co">## [373] &quot;Tert&quot;     &quot;Tp53&quot;     &quot;Cdk12&quot;    &quot;Epha3&quot;    &quot;Ncor2&quot;    &quot;Fat1&quot;    </span></span>
<span id="cb432-70"><a href="whole-genome-sequencing-mutations.html#cb432-70"></a><span class="co">## [379] &quot;Fat1&quot;     &quot;Pcm1&quot;     &quot;Ros1&quot;     &quot;Tshr&quot;     &quot;Tcf3&quot;</span></span>
<span id="cb432-71"><a href="whole-genome-sequencing-mutations.html#cb432-71"></a></span>
<span id="cb432-72"><a href="whole-genome-sequencing-mutations.html#cb432-72"></a>tempy=rGeneOut2[ ,<span class="kw">match</span>(GeneList, <span class="kw">colnames</span>(rGeneOut2))]</span>
<span id="cb432-73"><a href="whole-genome-sequencing-mutations.html#cb432-73"></a><span class="co">#a2=heatmap.2(sign(tempy), trace=&quot;none&quot;, col=c(&quot;white&quot;, &quot;grey&quot;))</span></span>
<span id="cb432-74"><a href="whole-genome-sequencing-mutations.html#cb432-74"></a><span class="co">#tempyb=tempy[a2$rowInd, a2$colInd]</span></span>
<span id="cb432-75"><a href="whole-genome-sequencing-mutations.html#cb432-75"></a><span class="co">#hotspot2=hotspotLocExtract(tempyb)</span></span>
<span id="cb432-76"><a href="whole-genome-sequencing-mutations.html#cb432-76"></a><span class="co">#rownames(tempyb)=Cdata$NewID[sapply(rownames(tempyb), function(x) grep(x, Cdata$WGS))]</span></span>
<span id="cb432-77"><a href="whole-genome-sequencing-mutations.html#cb432-77"></a></span>
<span id="cb432-78"><a href="whole-genome-sequencing-mutations.html#cb432-78"></a><span class="co">##cosmic genes</span></span>
<span id="cb432-79"><a href="whole-genome-sequencing-mutations.html#cb432-79"></a>x1=<span class="kw">which</span>(<span class="kw">colnames</span>(rGeneOut2)<span class="op">%in%</span>RatCosmic)</span>
<span id="cb432-80"><a href="whole-genome-sequencing-mutations.html#cb432-80"></a>tempz=rGeneOut2[ ,x1]</span>
<span id="cb432-81"><a href="whole-genome-sequencing-mutations.html#cb432-81"></a>a2=<span class="kw">heatmap.2</span>(<span class="kw">sign</span>(tempz), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-141-4.png" width="672" /></p>
<div class="sourceCode" id="cb433"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb433-1"><a href="whole-genome-sequencing-mutations.html#cb433-1"></a>tempzb=tempz[a2<span class="op">$</span>rowInd, a2<span class="op">$</span>colInd]</span>
<span id="cb433-2"><a href="whole-genome-sequencing-mutations.html#cb433-2"></a>hotspotzb=<span class="kw">hotspotLocExtract</span>(tempzb)</span>
<span id="cb433-3"><a href="whole-genome-sequencing-mutations.html#cb433-3"></a></span>
<span id="cb433-4"><a href="whole-genome-sequencing-mutations.html#cb433-4"></a></span>
<span id="cb433-5"><a href="whole-genome-sequencing-mutations.html#cb433-5"></a>x2=<span class="kw">which</span>(<span class="kw">colSums</span>(<span class="kw">sign</span>(tempzb))<span class="op">==</span><span class="dv">1</span>)</span>
<span id="cb433-6"><a href="whole-genome-sequencing-mutations.html#cb433-6"></a>tempz2=tempzb[ , <span class="op">-</span>x2]</span>
<span id="cb433-7"><a href="whole-genome-sequencing-mutations.html#cb433-7"></a></span>
<span id="cb433-8"><a href="whole-genome-sequencing-mutations.html#cb433-8"></a>hotspotzb2=<span class="kw">hotspotLocExtract</span>(tempz2)</span>
<span id="cb433-9"><a href="whole-genome-sequencing-mutations.html#cb433-9"></a></span>
<span id="cb433-10"><a href="whole-genome-sequencing-mutations.html#cb433-10"></a><span class="kw">rownames</span>(tempzb)=Cdata<span class="op">$</span>NewID[<span class="kw">sapply</span>(<span class="kw">rownames</span>(tempzb), <span class="cf">function</span>(x) <span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS))]</span>
<span id="cb433-11"><a href="whole-genome-sequencing-mutations.html#cb433-11"></a><span class="kw">rownames</span>(tempz2)=Cdata<span class="op">$</span>NewID[<span class="kw">sapply</span>(<span class="kw">rownames</span>(tempz2), <span class="cf">function</span>(x) <span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS))]</span>
<span id="cb433-12"><a href="whole-genome-sequencing-mutations.html#cb433-12"></a></span>
<span id="cb433-13"><a href="whole-genome-sequencing-mutations.html#cb433-13"></a></span>
<span id="cb433-14"><a href="whole-genome-sequencing-mutations.html#cb433-14"></a>topCOSMIC=<span class="kw">colnames</span>(tempz2)</span>
<span id="cb433-15"><a href="whole-genome-sequencing-mutations.html#cb433-15"></a></span>
<span id="cb433-16"><a href="whole-genome-sequencing-mutations.html#cb433-16"></a><span class="kw">save</span>(topN, topCOSMIC, <span class="dt">file=</span><span class="st">&quot;outputs/genes-selected-common-WGX-mutations.RData&quot;</span>)</span>
<span id="cb433-17"><a href="whole-genome-sequencing-mutations.html#cb433-17"></a></span>
<span id="cb433-18"><a href="whole-genome-sequencing-mutations.html#cb433-18"></a><span class="co">#x3=table(colSums(rGeneOut2))</span></span>
<span id="cb433-19"><a href="whole-genome-sequencing-mutations.html#cb433-19"></a><span class="kw">write.csv</span>(rGeneOut[ , x1], <span class="dt">file=</span><span class="st">&quot;nature-tables/1k-mutations.csv&quot;</span>)</span></code></pre></div>
<p>Note that in this table, the distribution of samples with mutations is as follows:</p>
<p>2695, 591, 289, 282, 131, 70, 45, 39, 40, 23, 15, 20, 19, 8, 8, 18, 10, 14, 5, 4, 14, 8, 6, 6, 1, 4, 8, 3, 2, 4, 1, 1, 1, 1, 1, 1</p>
<p>The number of genes which have shared mutations in 10 or more samples is:</p>
<p>4622, which is why we have use these values here.</p>
<p>The number of genes which are cosmic associated are:</p>
<p>221 and the number which appear in more than 1 sample is 82.</p>
<p>These genes are tabulated below: (nice figures generated here).</p>
<div class="sourceCode" id="cb434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb434-1"><a href="whole-genome-sequencing-mutations.html#cb434-1"></a><span class="co">### keep these values</span></span>
<span id="cb434-2"><a href="whole-genome-sequencing-mutations.html#cb434-2"></a>top90genes=<span class="kw">colnames</span>(rGeneOut2)</span>
<span id="cb434-3"><a href="whole-genome-sequencing-mutations.html#cb434-3"></a></span>
<span id="cb434-4"><a href="whole-genome-sequencing-mutations.html#cb434-4"></a><span class="co">#pdf(sprintf(&quot;~/Desktop/summary_mutations_top90_human_homology_%s.pdf&quot;, Sys.Date()), width=14, height=7)</span></span>
<span id="cb434-5"><a href="whole-genome-sequencing-mutations.html#cb434-5"></a></span>
<span id="cb434-6"><a href="whole-genome-sequencing-mutations.html#cb434-6"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb434-7"><a href="whole-genome-sequencing-mutations.html#cb434-7"></a><span class="kw">image</span>(<span class="kw">t</span>(tempxb), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>), <span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)</span>
<span id="cb434-8"><a href="whole-genome-sequencing-mutations.html#cb434-8"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(tempxb)), <span class="kw">colnames</span>(tempxb), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb434-9"><a href="whole-genome-sequencing-mutations.html#cb434-9"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(tempxb)), <span class="kw">rownames</span>(tempxb), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb434-10"><a href="whole-genome-sequencing-mutations.html#cb434-10"></a><span class="kw">text</span>(<span class="kw">na.omit</span>(hotspot1b<span class="op">$</span>Y<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">ncol</span>(tempxb)<span class="op">-</span><span class="dv">1</span>), <span class="kw">na.omit</span>(hotspot1b<span class="op">$</span>X<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">nrow</span>(tempxb)<span class="op">-</span><span class="dv">1</span>) , <span class="st">&quot;*&quot;</span>)</span>
<span id="cb434-11"><a href="whole-genome-sequencing-mutations.html#cb434-11"></a><span class="kw">legend</span>(<span class="st">&quot;top&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">-0.15</span>),    <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;missense&quot;</span>,<span class="st">&quot;frameshift&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;splice&quot;</span>, <span class="st">&quot;hotspot human&quot;</span>),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>,<span class="dv">19</span>, <span class="dv">8</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="st">&quot;black&quot;</span>), <span class="dt">horiz =</span> T)</span>
<span id="cb434-12"><a href="whole-genome-sequencing-mutations.html#cb434-12"></a><span class="kw">mtext</span>(<span class="st">&quot;top 90 mutations&quot;</span>, <span class="dt">side=</span><span class="dv">3</span>, <span class="dt">line=</span><span class="op">-</span><span class="dv">1</span>, <span class="dt">outer=</span>T)</span></code></pre></div>
<div class="figure"><span id="fig:1k"></span>
<img src="RatDCIS-bookdown_files/figure-html/1k-1.png" alt="summary of mutations acccording to frequency cosmic" width="672" />
<p class="caption">
Figure 20.7: summary of mutations acccording to frequency cosmic
</p>
</div>
<p>Below is the above list refined according to cosmic specifc genes, and listing</p>
<div class="sourceCode" id="cb435"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb435-1"><a href="whole-genome-sequencing-mutations.html#cb435-1"></a><span class="co"># par(mar = c(4, 8, 4, 2),  xpd = TRUE)</span></span>
<span id="cb435-2"><a href="whole-genome-sequencing-mutations.html#cb435-2"></a><span class="co"># image(t(tempyb), col=c(&quot;white&quot;, 1:4), xaxt=&#39;n&#39;, yaxt=&#39;n&#39;)</span></span>
<span id="cb435-3"><a href="whole-genome-sequencing-mutations.html#cb435-3"></a><span class="co"># axis(1, at=seq(0, 1, length=ncol(tempyb)), colnames(tempyb), las=2, cex.axis=0.5)</span></span>
<span id="cb435-4"><a href="whole-genome-sequencing-mutations.html#cb435-4"></a><span class="co"># axis(2, at=seq(0, 1, length=nrow(tempyb)), rownames(tempyb), las=2, cex.axis=0.5)</span></span>
<span id="cb435-5"><a href="whole-genome-sequencing-mutations.html#cb435-5"></a><span class="co"># text(na.omit(hotspot2$Y-1)/(ncol(tempyb)-1), na.omit(hotspot2$X-1)/(nrow(tempyb)-1) , &quot;*&quot;)</span></span>
<span id="cb435-6"><a href="whole-genome-sequencing-mutations.html#cb435-6"></a><span class="co"># legend(&quot;top&quot;, inset = c(-0.5, -0.15),    legend = c(&quot;missense&quot;,&quot;frameshift&quot;, &quot;Nonsense&quot;, &quot;splice&quot;, &quot;hotspot human&quot;),  pch = c(19, 19, 19,19, 8),  col = c(1:4, &quot;black&quot;), horiz = T)</span></span>
<span id="cb435-7"><a href="whole-genome-sequencing-mutations.html#cb435-7"></a><span class="co"># mtext(&quot; &#39;Hotspot&#39; mutations only&quot;, side=3, line=-1, outer=T)</span></span>
<span id="cb435-8"><a href="whole-genome-sequencing-mutations.html#cb435-8"></a></span>
<span id="cb435-9"><a href="whole-genome-sequencing-mutations.html#cb435-9"></a><span class="co">## Plot by picking out cancer related genes</span></span>
<span id="cb435-10"><a href="whole-genome-sequencing-mutations.html#cb435-10"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb435-11"><a href="whole-genome-sequencing-mutations.html#cb435-11"></a><span class="kw">image</span>(<span class="kw">t</span>(tempzb), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)</span>
<span id="cb435-12"><a href="whole-genome-sequencing-mutations.html#cb435-12"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(tempzb)), <span class="kw">colnames</span>(tempzb), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb435-13"><a href="whole-genome-sequencing-mutations.html#cb435-13"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(tempzb)), <span class="kw">rownames</span>(tempzb), <span class="dt">las=</span><span class="dv">2</span>,  <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb435-14"><a href="whole-genome-sequencing-mutations.html#cb435-14"></a><span class="kw">text</span>(<span class="kw">na.omit</span>(hotspotzb<span class="op">$</span>Y<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">ncol</span>(tempzb)<span class="op">-</span><span class="dv">1</span>), <span class="kw">na.omit</span>(hotspotzb<span class="op">$</span>X<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">nrow</span>(tempzb)<span class="op">-</span><span class="dv">1</span>) , <span class="st">&quot;*&quot;</span>)</span>
<span id="cb435-15"><a href="whole-genome-sequencing-mutations.html#cb435-15"></a><span class="kw">legend</span>(<span class="st">&quot;top&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">-0.15</span>),    <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;missense&quot;</span>,<span class="st">&quot;frameshift&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;splice&quot;</span>, <span class="st">&quot;hotspot human&quot;</span>),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>,<span class="dv">19</span>, <span class="dv">8</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="st">&quot;black&quot;</span>), <span class="dt">horiz =</span> T)</span>
<span id="cb435-16"><a href="whole-genome-sequencing-mutations.html#cb435-16"></a><span class="kw">mtext</span>(<span class="st">&quot;Cancer related genes&quot;</span>, <span class="dt">side=</span><span class="dv">3</span>, <span class="dt">line=</span><span class="op">-</span><span class="dv">1</span>, <span class="dt">outer=</span>T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-142-1.png" width="672" /></p>
<div class="sourceCode" id="cb436"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb436-1"><a href="whole-genome-sequencing-mutations.html#cb436-1"></a></span>
<span id="cb436-2"><a href="whole-genome-sequencing-mutations.html#cb436-2"></a></span>
<span id="cb436-3"><a href="whole-genome-sequencing-mutations.html#cb436-3"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">4</span>, <span class="dv">2</span>),  <span class="dt">xpd =</span> <span class="ot">TRUE</span>)</span>
<span id="cb436-4"><a href="whole-genome-sequencing-mutations.html#cb436-4"></a><span class="kw">image</span>(<span class="kw">t</span>(tempz2), <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)</span>
<span id="cb436-5"><a href="whole-genome-sequencing-mutations.html#cb436-5"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(tempz2)), <span class="kw">colnames</span>(tempz2), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb436-6"><a href="whole-genome-sequencing-mutations.html#cb436-6"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(tempz2)), <span class="kw">rownames</span>(tempz2), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb436-7"><a href="whole-genome-sequencing-mutations.html#cb436-7"></a><span class="kw">text</span>(<span class="kw">na.omit</span>(hotspotzb2<span class="op">$</span>Y<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">ncol</span>(tempz2)<span class="op">-</span><span class="dv">1</span>), <span class="kw">na.omit</span>(hotspotzb2<span class="op">$</span>X<span class="dv">-1</span>)<span class="op">/</span>(<span class="kw">nrow</span>(tempz2)<span class="op">-</span><span class="dv">1</span>) , <span class="st">&quot;*&quot;</span>)</span>
<span id="cb436-8"><a href="whole-genome-sequencing-mutations.html#cb436-8"></a><span class="kw">legend</span>(<span class="st">&quot;top&quot;</span>, <span class="dt">inset =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.5</span>, <span class="fl">-0.15</span>),    <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;missense&quot;</span>,<span class="st">&quot;frameshift&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;splice&quot;</span>, <span class="st">&quot;hotspot human&quot;</span>),  <span class="dt">pch =</span> <span class="kw">c</span>(<span class="dv">19</span>, <span class="dv">19</span>, <span class="dv">19</span>,<span class="dv">19</span>, <span class="dv">8</span>),  <span class="dt">col =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="st">&quot;black&quot;</span>), <span class="dt">horiz =</span> T)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-142-2.png" width="672" /></p>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="whole-genome-sequencing-mutations.html#cb437-1"></a></span>
<span id="cb437-2"><a href="whole-genome-sequencing-mutations.html#cb437-2"></a><span class="co">#dev.off()</span></span></code></pre></div>
</div>
</div>
<div id="metacore-analysis-of-commonly-mutated-pathways" class="section level2">
<h2><span class="header-section-number">20.5</span> Metacore analysis of commonly mutated pathways</h2>
<p>Save these lists for metacore</p>
<div class="sourceCode" id="cb438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb438-1"><a href="whole-genome-sequencing-mutations.html#cb438-1"></a>allMut=<span class="kw">unique</span>(MeltWGSCodingB<span class="op">$</span>Gene_Symbol)</span>
<span id="cb438-2"><a href="whole-genome-sequencing-mutations.html#cb438-2"></a><span class="kw">write.csv</span>(allMut, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/all_mut_rat_WGS_coding_metacore_%s.csv&quot;</span>, <span class="kw">Sys.Date</span>()), <span class="dt">row.names =</span> F, <span class="dt">quote=</span>F)</span>
<span id="cb438-3"><a href="whole-genome-sequencing-mutations.html#cb438-3"></a></span>
<span id="cb438-4"><a href="whole-genome-sequencing-mutations.html#cb438-4"></a>allMut=<span class="kw">unique</span>(MeltCosmic<span class="op">$</span>Gene_Symbol)</span>
<span id="cb438-5"><a href="whole-genome-sequencing-mutations.html#cb438-5"></a><span class="kw">write.csv</span>(allMut, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/all_mut_rat_WES_cosmic_metacore_%s.csv&quot;</span>, <span class="kw">Sys.Date</span>()), <span class="dt">row.names =</span> F, <span class="dt">quote=</span>F)</span></code></pre></div>
<p>Load the metacore results:</p>
<div class="sourceCode" id="cb439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb439-1"><a href="whole-genome-sequencing-mutations.html#cb439-1"></a>Process_Networks=readxl<span class="op">::</span><span class="kw">read_excel</span>(<span class="st">&quot;../data/wholegenome_mutations/all rat WES coding process networks-jan2021.xls&quot;</span>, <span class="dv">1</span>, <span class="dt">skip=</span><span class="dv">2</span>)</span>
<span id="cb439-2"><a href="whole-genome-sequencing-mutations.html#cb439-2"></a><span class="co">#Pathway_Maps=readxl::read_excel(&quot;../data/wholegenome_mutations/all rat WES pathway maps coding-cosmic.xls&quot;, 1, skip = 2)</span></span>
<span id="cb439-3"><a href="whole-genome-sequencing-mutations.html#cb439-3"></a></span>
<span id="cb439-4"><a href="whole-genome-sequencing-mutations.html#cb439-4"></a>Terms_PN=<span class="kw">strsplit</span>(Process_Networks<span class="op">$</span>Networks, <span class="st">&quot;_&quot;</span>)</span>
<span id="cb439-5"><a href="whole-genome-sequencing-mutations.html#cb439-5"></a>TermsPN1=<span class="kw">sapply</span>(Terms_PN, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb439-6"><a href="whole-genome-sequencing-mutations.html#cb439-6"></a>TermsPN2=<span class="kw">sapply</span>(Terms_PN, <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb439-7"><a href="whole-genome-sequencing-mutations.html#cb439-7"></a></span>
<span id="cb439-8"><a href="whole-genome-sequencing-mutations.html#cb439-8"></a></span>
<span id="cb439-9"><a href="whole-genome-sequencing-mutations.html#cb439-9"></a><span class="kw">load</span>(<span class="st">&quot;../anntotations/Metacore_extracted_Process_networks_nov2020.RData&quot;</span>)</span>
<span id="cb439-10"><a href="whole-genome-sequencing-mutations.html#cb439-10"></a>n2=<span class="kw">which</span>(Process_Networks<span class="op">$</span>FDR<span class="op">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb439-11"><a href="whole-genome-sequencing-mutations.html#cb439-11"></a>m1=<span class="kw">match</span>(Process_Networks<span class="op">$</span>Networks[n2], <span class="kw">names</span>(PathwayMapSet))</span>
<span id="cb439-12"><a href="whole-genome-sequencing-mutations.html#cb439-12"></a><span class="co">#figure out which is significant</span></span>
<span id="cb439-13"><a href="whole-genome-sequencing-mutations.html#cb439-13"></a></span>
<span id="cb439-14"><a href="whole-genome-sequencing-mutations.html#cb439-14"></a></span>
<span id="cb439-15"><a href="whole-genome-sequencing-mutations.html#cb439-15"></a>N2=<span class="kw">lapply</span>(m1, <span class="cf">function</span>(x) <span class="kw">which</span>(MeltWGSCodingB<span class="op">$</span>Hum_Gene <span class="op">%in%</span><span class="st"> </span>PathwayMapSet[[x]]))</span>
<span id="cb439-16"><a href="whole-genome-sequencing-mutations.html#cb439-16"></a>GList2=<span class="kw">lapply</span>(N2, <span class="cf">function</span>(x) MeltWGSCodingB[x, <span class="kw">c</span>(<span class="st">&quot;Hum_Gene&quot;</span>, <span class="st">&quot;L1&quot;</span>)])</span>
<span id="cb439-17"><a href="whole-genome-sequencing-mutations.html#cb439-17"></a>SumTable2=<span class="kw">sapply</span>(GList2, <span class="cf">function</span>(x) <span class="kw">summary</span>(<span class="kw">factor</span>(x[ ,<span class="st">&quot;L1&quot;</span>], <span class="dt">levels=</span><span class="kw">unique</span>(MeltWGSCodingB<span class="op">$</span>L1))))</span>
<span id="cb439-18"><a href="whole-genome-sequencing-mutations.html#cb439-18"></a><span class="kw">colnames</span>(SumTable2)=Process_Networks<span class="op">$</span>Networks[n2]</span>
<span id="cb439-19"><a href="whole-genome-sequencing-mutations.html#cb439-19"></a><span class="co">#listGenes=strsplit(Process_Networks$Network.Objects.from.Active.Data[1:15], &quot;,&quot;)</span></span>
<span id="cb439-20"><a href="whole-genome-sequencing-mutations.html#cb439-20"></a></span>
<span id="cb439-21"><a href="whole-genome-sequencing-mutations.html#cb439-21"></a><span class="co">## do this for ERS1 networks</span></span>
<span id="cb439-22"><a href="whole-genome-sequencing-mutations.html#cb439-22"></a>er1=GList2[[<span class="dv">8</span>]] <span class="co">#6,8, 35</span></span>
<span id="cb439-23"><a href="whole-genome-sequencing-mutations.html#cb439-23"></a>axtemp=<span class="kw">table</span>(er1[ ,<span class="dv">1</span>], er1[,<span class="dv">2</span>])</span>
<span id="cb439-24"><a href="whole-genome-sequencing-mutations.html#cb439-24"></a><span class="kw">colnames</span>(axtemp)=Cdata<span class="op">$</span>NewID[<span class="kw">sapply</span>(<span class="kw">colnames</span>(axtemp), <span class="cf">function</span>(x) <span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS) ) ]</span>
<span id="cb439-25"><a href="whole-genome-sequencing-mutations.html#cb439-25"></a>a2=<span class="kw">heatmap.2</span>(<span class="kw">sign</span>(axtemp), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;Blues&quot;</span>), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:1l-1"></span>
<img src="RatDCIS-bookdown_files/figure-html/1l-1.png" alt="summary of metacore common mutated pathways" width="672" />
<p class="caption">
Figure 20.8: summary of metacore common mutated pathways
</p>
</div>
<div class="sourceCode" id="cb440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb440-1"><a href="whole-genome-sequencing-mutations.html#cb440-1"></a></span>
<span id="cb440-2"><a href="whole-genome-sequencing-mutations.html#cb440-2"></a><span class="kw">image</span>(<span class="kw">t</span>(a2<span class="op">$</span>carpet), <span class="dt">xaxt=</span><span class="st">&quot;n&quot;</span>, <span class="dt">yaxt=</span><span class="st">&quot;n&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;skyblue&quot;</span>))</span>
<span id="cb440-3"><a href="whole-genome-sequencing-mutations.html#cb440-3"></a><span class="kw">axis</span>(<span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">nrow</span>(axtemp)), <span class="kw">colnames</span>(a2<span class="op">$</span>carpet), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span>
<span id="cb440-4"><a href="whole-genome-sequencing-mutations.html#cb440-4"></a><span class="kw">axis</span>(<span class="dv">2</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(axtemp)), <span class="kw">rownames</span>(a2<span class="op">$</span>carpet), <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">cex.axis=</span><span class="fl">0.7</span>)</span></code></pre></div>
<div class="figure"><span id="fig:1l-2"></span>
<img src="RatDCIS-bookdown_files/figure-html/1l-2.png" alt="summary of metacore common mutated pathways" width="672" />
<p class="caption">
Figure 20.9: summary of metacore common mutated pathways
</p>
</div>
<div class="sourceCode" id="cb441"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb441-1"><a href="whole-genome-sequencing-mutations.html#cb441-1"></a></span>
<span id="cb441-2"><a href="whole-genome-sequencing-mutations.html#cb441-2"></a><span class="co"># load(&quot;../anntotations/Metacore_extracted_pathway_maps_nov2020.RData&quot;)</span></span>
<span id="cb441-3"><a href="whole-genome-sequencing-mutations.html#cb441-3"></a><span class="co"># n2=which(Pathway_Maps$FDR...7&lt;0.001)</span></span>
<span id="cb441-4"><a href="whole-genome-sequencing-mutations.html#cb441-4"></a><span class="co"># m1=match(Pathway_Maps$Maps[n2], names(PathwayMapSet))</span></span>
<span id="cb441-5"><a href="whole-genome-sequencing-mutations.html#cb441-5"></a></span>
<span id="cb441-6"><a href="whole-genome-sequencing-mutations.html#cb441-6"></a></span>
<span id="cb441-7"><a href="whole-genome-sequencing-mutations.html#cb441-7"></a><span class="co"># N2=lapply(m1, function(x) which(MeltWGSCodingB$Hum_Gene %in% PathwayMapSet[[x]]))</span></span>
<span id="cb441-8"><a href="whole-genome-sequencing-mutations.html#cb441-8"></a><span class="co"># GList2=lapply(N2, function(x) MeltWGSCodingB[x, c(&quot;Hum_Gene&quot;, &quot;L1&quot;)])</span></span>
<span id="cb441-9"><a href="whole-genome-sequencing-mutations.html#cb441-9"></a><span class="co"># SumTable3=sapply(GList2, function(x) summary(factor(x[ ,&quot;L1&quot;], levels=unique(MeltWGSCodingB$L1))))</span></span>
<span id="cb441-10"><a href="whole-genome-sequencing-mutations.html#cb441-10"></a><span class="co"># colnames(SumTable3)=Pathway_Maps$Maps[n2]</span></span>
<span id="cb441-11"><a href="whole-genome-sequencing-mutations.html#cb441-11"></a><span class="co"># #listGenes=strsplit(Process_Networks$Network.Objects.from.Active.Data[1:15], &quot;,&quot;)</span></span>
<span id="cb441-12"><a href="whole-genome-sequencing-mutations.html#cb441-12"></a><span class="co"># head(SumTable3)</span></span>
<span id="cb441-13"><a href="whole-genome-sequencing-mutations.html#cb441-13"></a></span>
<span id="cb441-14"><a href="whole-genome-sequencing-mutations.html#cb441-14"></a><span class="co">#pdf(&quot;rslt/WGS/Metacore_analysis_coding_mutations.pdf&quot;, height=13, width=13)</span></span>
<span id="cb441-15"><a href="whole-genome-sequencing-mutations.html#cb441-15"></a></span>
<span id="cb441-16"><a href="whole-genome-sequencing-mutations.html#cb441-16"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>, <span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb441-17"><a href="whole-genome-sequencing-mutations.html#cb441-17"></a>n2=<span class="kw">order</span>(Process_Networks<span class="op">$</span>FDR)</span>
<span id="cb441-18"><a href="whole-genome-sequencing-mutations.html#cb441-18"></a><span class="kw">barplot</span>(<span class="op">-</span><span class="kw">log10</span>(<span class="kw">as.numeric</span>(Process_Networks<span class="op">$</span>FDR[n2[<span class="dv">1</span><span class="op">:</span><span class="dv">15</span>]])), <span class="dt">names.arg =</span> Process_Networks<span class="op">$</span>Networks[n2[<span class="dv">1</span><span class="op">:</span><span class="dv">15</span>]], <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">main=</span><span class="st">&quot;Process networks all coding&quot;</span>, <span class="dt">horiz =</span> T, <span class="dt">cex.names =</span> <span class="fl">0.5</span>)</span>
<span id="cb441-19"><a href="whole-genome-sequencing-mutations.html#cb441-19"></a>n2=<span class="kw">order</span>(Process_Networks<span class="op">$</span>FDR)</span>
<span id="cb441-20"><a href="whole-genome-sequencing-mutations.html#cb441-20"></a><span class="co"># barplot(-log10(as.numeric(Process_Networks$FDR...11[n2[1:15]])), names.arg = Process_Networks$Networks[n2[1:15]], las=2, main=&quot;Process networks cosmic only&quot;, horiz = T, cex.names = 0.5)</span></span>
<span id="cb441-21"><a href="whole-genome-sequencing-mutations.html#cb441-21"></a><span class="co"># barplot(-log10(as.numeric(Pathway_Maps$FDR[1:15])), names.arg = Pathway_Maps$Maps[1:15], las=2, main=&quot;Pathway Maps&quot;, horiz = T, cex.names = 0.5)</span></span></code></pre></div>
<div class="figure"><span id="fig:1l-3"></span>
<img src="RatDCIS-bookdown_files/figure-html/1l-3.png" alt="summary of metacore common mutated pathways" width="672" />
<p class="caption">
Figure 20.10: summary of metacore common mutated pathways
</p>
</div>
<div class="sourceCode" id="cb442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb442-1"><a href="whole-genome-sequencing-mutations.html#cb442-1"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">3</span>))</span>
<span id="cb442-2"><a href="whole-genome-sequencing-mutations.html#cb442-2"></a>rNames=<span class="kw">strsplit</span>(<span class="kw">colnames</span>(SumTable2), <span class="st">&quot;_&quot;</span>)</span>
<span id="cb442-3"><a href="whole-genome-sequencing-mutations.html#cb442-3"></a>rNames2=<span class="kw">sapply</span>(rNames, <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb442-4"><a href="whole-genome-sequencing-mutations.html#cb442-4"></a>cCols=<span class="kw">brewer.pal</span>(<span class="dv">12</span>,<span class="st">&quot;Set3&quot;</span>)[<span class="kw">factor</span>(rNames2)]</span>
<span id="cb442-5"><a href="whole-genome-sequencing-mutations.html#cb442-5"></a>lx=<span class="kw">order</span>(cCols)</span>
<span id="cb442-6"><a href="whole-genome-sequencing-mutations.html#cb442-6"></a><span class="kw">colnames</span>(SumTable2)=<span class="kw">sapply</span>(rNames, <span class="cf">function</span>(x) x[<span class="dv">2</span>])</span>
<span id="cb442-7"><a href="whole-genome-sequencing-mutations.html#cb442-7"></a><span class="kw">rownames</span>(SumTable2)=Cdata<span class="op">$</span>NewID[<span class="kw">sapply</span>(<span class="kw">rownames</span>(SumTable2), <span class="cf">function</span>(x) <span class="kw">grep</span>(x, Cdata<span class="op">$</span>WGS))]</span>
<span id="cb442-8"><a href="whole-genome-sequencing-mutations.html#cb442-8"></a><span class="kw">heatmap.2</span>(<span class="kw">sign</span>(SumTable2[, lx]), <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;skyblue&quot;</span>), <span class="dt">main=</span><span class="st">&quot;Process Networks: scaled No mutations affecting each pathway&quot;</span>,</span>
<span id="cb442-9"><a href="whole-genome-sequencing-mutations.html#cb442-9"></a>          <span class="dt">ColSideColors =</span> cCols[lx], <span class="dt">Colv =</span> <span class="ot">NA</span>)</span></code></pre></div>
<div class="figure"><span id="fig:1lb"></span>
<img src="RatDCIS-bookdown_files/figure-html/1lb-1.png" alt="summary of metacore common mutated pathways" width="672" />
<p class="caption">
Figure 20.11: summary of metacore common mutated pathways
</p>
</div>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="whole-genome-sequencing-mutations.html#cb443-1"></a></span>
<span id="cb443-2"><a href="whole-genome-sequencing-mutations.html#cb443-2"></a></span>
<span id="cb443-3"><a href="whole-genome-sequencing-mutations.html#cb443-3"></a><span class="kw">write.csv</span>(SumTable2[, lx], <span class="dt">file=</span><span class="st">&quot;nature-tables/1l_samples_with_muttions.csv&quot;</span>)</span>
<span id="cb443-4"><a href="whole-genome-sequencing-mutations.html#cb443-4"></a></span>
<span id="cb443-5"><a href="whole-genome-sequencing-mutations.html#cb443-5"></a><span class="co"># par(oma=c(2, 0,0,3))</span></span>
<span id="cb443-6"><a href="whole-genome-sequencing-mutations.html#cb443-6"></a><span class="co"># # rNames=strsplit(colnames(SumTable3), &quot;_&quot;)</span></span>
<span id="cb443-7"><a href="whole-genome-sequencing-mutations.html#cb443-7"></a><span class="co"># # rNames2=sapply(rNames, function(x) x[1])</span></span>
<span id="cb443-8"><a href="whole-genome-sequencing-mutations.html#cb443-8"></a><span class="co"># # cCols=brewer.pal(12,&quot;Set3&quot;)[factor(rNames2)]</span></span>
<span id="cb443-9"><a href="whole-genome-sequencing-mutations.html#cb443-9"></a><span class="co"># # lx=order(cCols)r</span></span>
<span id="cb443-10"><a href="whole-genome-sequencing-mutations.html#cb443-10"></a><span class="co"># heatmap.2(SumTable3, scale=&quot;col&quot;, trace=&quot;none&quot;, col=brewer.pal(9, &quot;Blues&quot;), main=&quot;Pathway Maps&quot;)</span></span>
<span id="cb443-11"><a href="whole-genome-sequencing-mutations.html#cb443-11"></a><span class="co"># write.csv(SumTable3, file=&quot;outputs/no mutations pathway maps all coding.csv&quot;)</span></span>
<span id="cb443-12"><a href="whole-genome-sequencing-mutations.html#cb443-12"></a><span class="co"># write.csv(SumTable2, file=&quot;outputs/no mutations process networks all coding.csv&quot;)</span></span>
<span id="cb443-13"><a href="whole-genome-sequencing-mutations.html#cb443-13"></a><span class="co">#dev.off()</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bcr-clonotype-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mutations-in-rna.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
