<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Whole Genome Sequencing Mutations | Rat DCIS study</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘insert study name’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Whole Genome Sequencing Mutations | Rat DCIS study" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘insert study name’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Whole Genome Sequencing Mutations | Rat DCIS study" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘insert study name’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2020-11-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="bcr-clonotype-analysis.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-summary.html"><a href="cohort-summary.html"><i class="fa fa-check"></i><b>2</b> Cohort summary</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-summary.html"><a href="cohort-summary.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.1</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-summary.html"><a href="cohort-summary.html#facs-data"><i class="fa fa-check"></i><b>2.2</b> FACS data</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-table"><i class="fa fa-check"></i><b>2.3</b> Summary Table</a><ul>
<li class="chapter" data-level="2.3.1" data-path="cohort-summary.html"><a href="cohort-summary.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>2.3.1</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="2.3.2" data-path="cohort-summary.html"><a href="cohort-summary.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>2.3.2</b> Summary of the RNA data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>3</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="3.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>3.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="3.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>3.2</b> Cellular composition</a></li>
<li class="chapter" data-level="3.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>3.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="3.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>3.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="3.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>3.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="3.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>3.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>4</b> Spatial statistics</a><ul>
<li class="chapter" data-level="4.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>4.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="4.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification-treatment-growth"><i class="fa fa-check"></i><b>4.1.1</b> Comparison to manual classification, treatment, growth</a></li>
<li class="chapter" data-level="4.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome"><i class="fa fa-check"></i><b>4.1.2</b> Associations with outcome</a></li>
<li class="chapter" data-level="4.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>4.1.3</b> Growth</a></li>
<li class="chapter" data-level="4.1.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#association-between-distance-and-content"><i class="fa fa-check"></i><b>4.1.4</b> Association between distance and content</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>4.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="4.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>4.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="4.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>4.2.2</b> Growth</a></li>
<li class="chapter" data-level="4.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>4.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>4.3</b> M-H distances</a><ul>
<li class="chapter" data-level="4.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>4.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="4.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>4.3.2</b> Growth</a></li>
<li class="chapter" data-level="4.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>4.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>4.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="4.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#other-cell-types"><i class="fa fa-check"></i><b>4.5</b> Other cell types</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>5</b> Expression data</a><ul>
<li class="chapter" data-level="5.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>5.1</b> Running alignment</a></li>
<li class="chapter" data-level="5.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>5.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="5.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>5.3</b> Normalisation</a><ul>
<li class="chapter" data-level="5.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>5.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>5.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>6</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>6.1</b> PCA plots</a></li>
<li class="chapter" data-level="6.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>6.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="deseq-analysis.html"><a href="deseq-analysis.html"><i class="fa fa-check"></i><b>7</b> DESeq analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="deseq-analysis.html"><a href="deseq-analysis.html#dn-vs-ep"><i class="fa fa-check"></i><b>7.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="7.2" data-path="deseq-analysis.html"><a href="deseq-analysis.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>7.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="7.3" data-path="deseq-analysis.html"><a href="deseq-analysis.html#set-up-cell-type-specifoc-the-comparisons"><i class="fa fa-check"></i><b>7.3</b> Set-up cell-type specifoc the comparisons</a></li>
<li class="chapter" data-level="7.4" data-path="deseq-analysis.html"><a href="deseq-analysis.html#pca-plots-1"><i class="fa fa-check"></i><b>7.4</b> PCA plots</a><ul>
<li class="chapter" data-level="7.4.1" data-path="deseq-analysis.html"><a href="deseq-analysis.html#epcam"><i class="fa fa-check"></i><b>7.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="7.4.2" data-path="deseq-analysis.html"><a href="deseq-analysis.html#cd45"><i class="fa fa-check"></i><b>7.4.2</b> CD45</a></li>
<li class="chapter" data-level="7.4.3" data-path="deseq-analysis.html"><a href="deseq-analysis.html#dn"><i class="fa fa-check"></i><b>7.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="deseq-analysis.html"><a href="deseq-analysis.html#collating-results-and-running-gsea"><i class="fa fa-check"></i><b>7.5</b> Collating results and running GSEA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#summary-of-comparisons"><i class="fa fa-check"></i><b>8.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#summary-of-comaprisons"><i class="fa fa-check"></i><b>8.2</b> Summary of comaprisons</a><ul>
<li class="chapter" data-level="8.2.1" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#epcam-samples"><i class="fa fa-check"></i><b>8.2.1</b> EPCAM samples</a></li>
<li class="chapter" data-level="8.2.2" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#cd45-comparisons"><i class="fa fa-check"></i><b>8.2.2</b> CD45 comparisons</a></li>
<li class="chapter" data-level="8.2.3" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#dn-samples"><i class="fa fa-check"></i><b>8.2.3</b> DN samples</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>8.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="8.3.1" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#dn-fraction"><i class="fa fa-check"></i><b>8.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="8.3.2" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#cd-fraction"><i class="fa fa-check"></i><b>8.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="8.3.3" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#ep-fraction"><i class="fa fa-check"></i><b>8.3.3</b> Ep Fraction</a></li>
<li class="chapter" data-level="8.3.4" data-path="deseq-analysis-1.html"><a href="deseq-analysis-1.html#specific-pathways"><i class="fa fa-check"></i><b>8.3.4</b> Specific pathways</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>9</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="9.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#in-house-classifier"><i class="fa fa-check"></i><b>9.1</b> In-house classifier</a></li>
<li class="chapter" data-level="9.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#pam50-parker-scoring"><i class="fa fa-check"></i><b>9.2</b> PAM50 (Parker) scoring</a></li>
<li class="chapter" data-level="9.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#clustering-based-on-random-genes"><i class="fa fa-check"></i><b>9.3</b> Clustering based on random genes</a></li>
<li class="chapter" data-level="9.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#consolidating-the-different-results"><i class="fa fa-check"></i><b>9.4</b> Consolidating the different results</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>10</b> Signature analysis</a><ul>
<li class="chapter" data-level="10.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>10.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="10.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>10.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="10.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>10.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="10.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>10.1.3</b> MHC presentation proteins</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="signature-analysis.html"><a href="signature-analysis.html#cancer-pathways"><i class="fa fa-check"></i><b>10.2</b> Cancer pathways</a><ul>
<li class="chapter" data-level="10.2.1" data-path="signature-analysis.html"><a href="signature-analysis.html#ssgsea"><i class="fa fa-check"></i><b>10.2.1</b> ssGSEA</a></li>
<li class="chapter" data-level="10.2.2" data-path="signature-analysis.html"><a href="signature-analysis.html#regression-model-for-covariates"><i class="fa fa-check"></i><b>10.2.2</b> regression model for covariates</a></li>
<li class="chapter" data-level="10.2.3" data-path="signature-analysis.html"><a href="signature-analysis.html#taking-into-account-directionality"><i class="fa fa-check"></i><b>10.2.3</b> Taking into account directionality</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>11</b> Immune estimation</a><ul>
<li class="chapter" data-level="11.1" data-path="immune-estimation.html"><a href="immune-estimation.html#ssgsea-1"><i class="fa fa-check"></i><b>11.1</b> ssGSEA</a></li>
<li class="chapter" data-level="11.2" data-path="immune-estimation.html"><a href="immune-estimation.html#deconvolution-methods"><i class="fa fa-check"></i><b>11.2</b> Deconvolution methods</a><ul>
<li class="chapter" data-level="11.2.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>11.2.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="11.2.2" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>11.2.2</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="11.2.3" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>11.2.3</b> Associate with Treatment</a></li>
<li class="chapter" data-level="11.2.4" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>11.2.4</b> Association with Growth</a></li>
<li class="chapter" data-level="11.2.5" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-infiltration"><i class="fa fa-check"></i><b>11.2.5</b> Association with infiltration</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>12</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="12.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>12.1</b> Processing summary</a></li>
<li class="chapter" data-level="12.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>12.2</b> Summary Stats</a></li>
<li class="chapter" data-level="12.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>12.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="12.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>12.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="12.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data"><i class="fa fa-check"></i><b>12.5</b> Associate with clinicopathological data</a></li>
<li class="chapter" data-level="12.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>12.6</b> Associate with signature scores</a></li>
<li class="chapter" data-level="12.7" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#check-whether-the-clonotype-percentages-match-with-gsva-signatures-or-cd20-populations"><i class="fa fa-check"></i><b>12.7</b> Check whether the clonotype percentages match with gsva signatures or CD20 populations</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>13</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="13.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>13.1</b> Data</a><ul>
<li class="chapter" data-level="13.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>13.1.1</b> Extract mutation signatures</a></li>
<li class="chapter" data-level="13.1.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>13.1.2</b> Annotate the data with human common variants</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>13.2</b> Plots</a><ul>
<li class="chapter" data-level="13.2.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>13.2.1</b> Quick overview</a></li>
<li class="chapter" data-level="13.2.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>13.2.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants"><i class="fa fa-check"></i><b>13.3</b> Coding variants</a><ul>
<li class="chapter" data-level="13.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>13.3.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>13.4</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="13.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>13.4.1</b> All mutations</a></li>
<li class="chapter" data-level="13.4.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#cosmic-only"><i class="fa fa-check"></i><b>13.4.2</b> Cosmic only:</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Rat DCIS study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="whole-genome-sequencing-mutations" class="section level1">
<h1><span class="header-section-number">Chapter 13</span> Whole Genome Sequencing Mutations</h1>
<p>Whole genome sequencing was performed on XX samples by the new york genome centre.
The pipeline involves:</p>
<p><span style="color:red"> Insert information on the NYGC pipeline here </span></p>
<div id="data" class="section level2">
<h2><span class="header-section-number">13.1</span> Data</h2>
<p>In this section, we load the maf output files and create lists which</p>
<ul>
<li>filter out coding mutations</li>
<li>Determine mutational signatures by:
<ul>
<li>Creating bed files with genomic locations of the SNP +- 1 base location</li>
<li>Extracting the nucleotide sequence using bedtools</li>
<li>Convert all mutations to C &gt; N and T &gt; N to obtain 96 mutational signatures</li>
<li>run with single sample SigProfiler to obtain the signature scores</li>
</ul></li>
<li>Annotate data with common human variants</li>
</ul>
<pre><code>## [1] &quot;no of mutations&quot;</code></pre>
<pre><code>##  C1R-tumor  C2N-tumor C2R1-tumor C2R2-tumor C2R3-tumor  C4P-tumor 
##     240153     262425     235677     222717     238324     258677</code></pre>
<div id="extract-mutation-signatures" class="section level3">
<h3><span class="header-section-number">13.1.1</span> Extract mutation signatures</h3>
<p>Annotate the files so that the nucleotide codon can be extracted based on the genome file.
See the bash file extract-trinucleotide-forSigProfiler.sh to see the output of this.
The output table $fout3 should have 6 columns with the genomic location, the reference and alt allele and codon context.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="whole-genome-sequencing-mutations.html#cb45-1"></a>TBed=<span class="kw">lapply</span>(AllMutWGS, <span class="cf">function</span>(x) <span class="kw">cbind</span>(<span class="kw">as.character</span>(x<span class="op">$</span>Chromosome), x<span class="op">$</span>Start_Position<span class="dv">-2</span>, x<span class="op">$</span>End_Position<span class="op">+</span><span class="dv">1</span>, <span class="kw">as.character</span>(x<span class="op">$</span>Reference_Allele), <span class="kw">as.character</span>(x<span class="op">$</span>Tumor_Seq_Allele2)))</span>
<span id="cb45-2"><a href="whole-genome-sequencing-mutations.html#cb45-2"></a>TBed=<span class="kw">lapply</span>(TBed, <span class="cf">function</span>(x) x[<span class="kw">which</span>(<span class="kw">nchar</span>(x[ ,<span class="dv">4</span>])<span class="op">==</span><span class="dv">1</span>), ])</span>
<span id="cb45-3"><a href="whole-genome-sequencing-mutations.html#cb45-3"></a><span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(TBed), <span class="cf">function</span>(x) <span class="kw">write.table</span>(TBed[[x]], <span class="dt">file=</span><span class="kw">paste</span>(<span class="kw">names</span>(TBed)[x], <span class="st">&quot;tricodon.bed&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="dt">col.names=</span>F, <span class="dt">row.names =</span> F, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> F))</span>
<span id="cb45-4"><a href="whole-genome-sequencing-mutations.html#cb45-4"></a></span>
<span id="cb45-5"><a href="whole-genome-sequencing-mutations.html#cb45-5"></a><span class="co">## bedtools command</span></span>
<span id="cb45-6"><a href="whole-genome-sequencing-mutations.html#cb45-6"></a>bedtools getfasta <span class="op">-</span>fi ..<span class="op">/</span>Genomic_Files<span class="op">/</span>rn6.fa <span class="op">-</span>bed <span class="op">$</span>f <span class="op">-</span>tab <span class="op">-</span>fo <span class="op">$</span>fout</span>
<span id="cb45-7"><a href="whole-genome-sequencing-mutations.html#cb45-7"></a><span class="kw">f3</span>() { awk <span class="st">&#39;{print $1, $2, $3, $4 ,$5}&#39;</span> <span class="st">&quot;$1&quot;</span>; }; paste <span class="op">&lt;</span>(f3 <span class="op">$</span>f) <span class="op">&lt;</span>(f3 <span class="op">$</span>fout) <span class="op">&gt;</span><span class="st"> </span><span class="er">$</span>fout2</span>
<span id="cb45-8"><a href="whole-genome-sequencing-mutations.html#cb45-8"></a>awk <span class="st">&#39;{if (length($7)==3) print $1, $2, $3, $4, $5, toupper($7)}&#39;</span> <span class="op">$</span>fout2 <span class="op">&gt;</span><span class="st"> </span><span class="er">$</span>fout3</span>
<span id="cb45-9"><a href="whole-genome-sequencing-mutations.html#cb45-9"></a><span class="co">## see the bash file extract-trinucleotide-forSigProfiler.sh to see the output of this.</span></span></code></pre></div>
<p>Following running the above bash script, we can run the following extract the tri-nucleotide sequence</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="whole-genome-sequencing-mutations.html#cb46-1"></a>FFiles=<span class="kw">dir</span>(<span class="st">&quot;../data/tricodon_mut_sigs/&quot;</span>, <span class="st">&quot;*.txt&quot;</span>,<span class="dt">full.names=</span>T)</span>
<span id="cb46-2"><a href="whole-genome-sequencing-mutations.html#cb46-2"></a>OutputList=<span class="kw">sapply</span>(FFiles, <span class="cf">function</span>(x) <span class="kw">FindTriNucleotideContext</span>(x))</span>
<span id="cb46-3"><a href="whole-genome-sequencing-mutations.html#cb46-3"></a>TableOut=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(OutputList), <span class="cf">function</span>(x) <span class="kw">unlist</span>(OutputList[<span class="dv">3</span>, x]))</span>
<span id="cb46-4"><a href="whole-genome-sequencing-mutations.html#cb46-4"></a>a1=<span class="kw">basename</span>(FFiles)</span>
<span id="cb46-5"><a href="whole-genome-sequencing-mutations.html#cb46-5"></a>a2=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(a1, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb46-6"><a href="whole-genome-sequencing-mutations.html#cb46-6"></a>TableOut=<span class="kw">cbind</span>(<span class="kw">as.character</span>(<span class="kw">unlist</span>(OutputList[<span class="dv">1</span>, <span class="dv">1</span>])), <span class="kw">as.character</span>(<span class="kw">unlist</span>(OutputList[<span class="dv">2</span>, <span class="dv">1</span>])), TableOut)</span>
<span id="cb46-7"><a href="whole-genome-sequencing-mutations.html#cb46-7"></a><span class="kw">colnames</span>(TableOut)=<span class="kw">c</span>(<span class="st">&quot;MutationType&quot;</span>,<span class="st">&quot;Nutrinucleotide&quot;</span>, <span class="kw">paste</span>(<span class="st">&quot;BRCA&quot;</span>, a2, <span class="dt">sep=</span><span class="st">&quot;::&quot;</span>))</span>
<span id="cb46-8"><a href="whole-genome-sequencing-mutations.html#cb46-8"></a><span class="kw">write.csv</span>(TableOut, <span class="dt">file=</span><span class="st">&quot;../data/tricodon_mut_sigs/output4SigProfiler.csv&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="dt">row.names =</span> F)</span></code></pre></div>
<p>This is then run in python (sigprofiler) with the following code:</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="whole-genome-sequencing-mutations.html#cb47-1"></a>module load gcc</span>
<span id="cb47-2"><a href="whole-genome-sequencing-mutations.html#cb47-2"></a>module load python<span class="op">/</span><span class="dv">3</span>.<span class="fl">7.4</span></span>
<span id="cb47-3"><a href="whole-genome-sequencing-mutations.html#cb47-3"></a>from SigProfilerExtractor import <span class="kw">spss_pcwag</span> (spss uses only vcf files)</span>
<span id="cb47-4"><a href="whole-genome-sequencing-mutations.html#cb47-4"></a><span class="kw">spss_pcwag.single_sample_pcwag</span>(<span class="st">&quot;output4SigProfiler.csv&quot;</span>, <span class="dt">output=</span><span class="st">&quot;example_output&quot;</span>)</span></code></pre></div>
<p>The results from this analysis are shown below</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-144-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-144-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-144-3.png" width="672" /></p>
</div>
<div id="annotate-the-data-with-human-common-variants" class="section level3">
<h3><span class="header-section-number">13.1.2</span> Annotate the data with human common variants</h3>
<p>From this analysis, below are the genes with mutations in known human hotpot mutations in common genes</p>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
L1
</th>
<th style="text-align:right;">
AAno
</th>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
AAno.1
</th>
<th style="text-align:left;">
AA1
</th>
<th style="text-align:left;">
Variant
</th>
<th style="text-align:left;">
HumGene
</th>
<th style="text-align:left;">
RatSequence
</th>
<th style="text-align:left;">
RatProt
</th>
<th style="text-align:left;">
HumSequence
</th>
<th style="text-align:left;">
HumProt
</th>
<th style="text-align:right;">
HumAAno
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
166
</td>
<td style="text-align:left;">
NMU12-tumor-LA
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:left;">
Hras
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
HRAS
</td>
<td style="text-align:left;">
VVVGAGGVGKS
</td>
<td style="text-align:left;">
ENSRNOP00000022363
</td>
<td style="text-align:left;">
VVVGAGGVGKS
</td>
<td style="text-align:left;">
ENSP00000380722
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
168
</td>
<td style="text-align:left;">
NMU12-tumor-LA
</td>
<td style="text-align:right;">
264
</td>
<td style="text-align:left;">
Tp53
</td>
<td style="text-align:right;">
264
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
R
</td>
<td style="text-align:left;">
TP53
</td>
<td style="text-align:left;">
SGNLLGRDSFE
</td>
<td style="text-align:left;">
ENSRNOP00000047840
</td>
<td style="text-align:left;">
SGNLLGRNSFE
</td>
<td style="text-align:left;">
ENSP00000352610
</td>
<td style="text-align:right;">
266
</td>
</tr>
<tr>
<td style="text-align:left;">
131
</td>
<td style="text-align:left;">
NMU1-tumor
</td>
<td style="text-align:right;">
118
</td>
<td style="text-align:left;">
Pik3ca
</td>
<td style="text-align:right;">
118
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
D
</td>
<td style="text-align:left;">
PIK3CA
</td>
<td style="text-align:left;">
LNREIGFVIGM
</td>
<td style="text-align:left;">
ENSRNOP00000072496
</td>
<td style="text-align:left;">
LNREIGFAIGM
</td>
<td style="text-align:left;">
ENSP00000263967
</td>
<td style="text-align:right;">
118
</td>
</tr>
<tr>
<td style="text-align:left;">
181
</td>
<td style="text-align:left;">
NMU12-tumor-LA
</td>
<td style="text-align:right;">
542
</td>
<td style="text-align:left;">
Pik3ca
</td>
<td style="text-align:right;">
542
</td>
<td style="text-align:left;">
E
</td>
<td style="text-align:left;">
K
</td>
<td style="text-align:left;">
PIK3CA
</td>
<td style="text-align:left;">
RDPLSEITEQE
</td>
<td style="text-align:left;">
ENSRNOP00000072496
</td>
<td style="text-align:left;">
RDPLSEITEQE
</td>
<td style="text-align:left;">
ENSP00000263967
</td>
<td style="text-align:right;">
542
</td>
</tr>
<tr>
<td style="text-align:left;">
155
</td>
<td style="text-align:left;">
NMU1-tumor
</td>
<td style="text-align:right;">
3016
</td>
<td style="text-align:left;">
Atm
</td>
<td style="text-align:right;">
3016
</td>
<td style="text-align:left;">
R
</td>
<td style="text-align:left;">
H
</td>
<td style="text-align:left;">
ATM
</td>
<td style="text-align:left;">
NKVAERVLMRL
</td>
<td style="text-align:left;">
ENSRNOP00000045529
</td>
<td style="text-align:left;">
NKVAERVLMRL
</td>
<td style="text-align:left;">
ENSP00000388058
</td>
<td style="text-align:right;">
3008
</td>
</tr>
</tbody>
</table>
<p>And here, we plot the frequency of these different mutations</p>
<pre><code>## [1] 377</code></pre>
<pre><code>## Nx
## Tert 241 Tert 356 Muc1 173  Atm 915 Prf1 292 Prf1 363 
##       14       14       12       11       11       11</code></pre>
</div>
</div>
<div id="plots" class="section level2">
<h2><span class="header-section-number">13.2</span> Plots</h2>
<div id="quick-overview" class="section level3">
<h3><span class="header-section-number">13.2.1</span> Quick overview</h3>
<p>WGS all mutations summary: Look at the overall mutation burden, and the tumor mutational burden, defined as # coding mutations per MB.</p>
<p>The following plot lists tumors in increasing size</p>
<p>Make plots of:
* overall types of mutations per sample (ranked by size)
* correlation between mutation type and size
* type of mutation?</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-147-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-147-2.png" width="672" /></p>
</div>
<div id="coding-variants-tmb" class="section level3">
<h3><span class="header-section-number">13.2.2</span> Coding variants &amp; TMB</h3>
<p>There are a few ways of figuring out the TMB. We can</p>
<ul>
<li>report the number of mutations (missense, insertion/deletion, frameshifts, nonsense)</li>
<li>normalise the above value to the known length of coding regions in the rat genome (UCSC exon length 43,423,804)</li>
<li>plot of TMB vs objective response rate: there is no true relationship between final tumor size and mutational burden. Could also be confounded by subclonality</li>
</ul>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-148-1.png" width="672" /></p>
</div>
</div>
<div id="coding-variants" class="section level2">
<h2><span class="header-section-number">13.3</span> Coding variants</h2>
<div id="sites-which-are-commonly-mutated" class="section level3">
<h3><span class="header-section-number">13.3.1</span> Sites which are commonly mutated?</h3>
<p>Could this be a variant (SD strain “rsSNP”) or is it actually a mutation?</p>
<p>Variants were annotated with snpeff. Raw data from this was obtained from <a href="https://www.biorxiv.org/content/10.1101/412924v1.full" class="uri">https://www.biorxiv.org/content/10.1101/412924v1.full</a></p>
<p>Note that the nihRats package may not be required here. We probably just want the harlan snps</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-150-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-150-2.png" width="672" /></p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-151-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-151-2.png" width="672" /></p>
<p>Look at the distribution of mutations:</p>
<p>Whole genome</p>
<ul>
<li>1188 genes in more than 2 animals (1435 in at least 2)</li>
<li>59 variants were identified as population variant in Harlan</li>
<li>47 in charles river (6 are novel in this cohort only)</li>
</ul>
<p>Cosmic genes:</p>
<ul>
<li>42 sites which have mutations in more than 2 animals</li>
<li>50 sites in more than 1 animal</li>
<li>5 sites which have been identified in GWAS studies of harlan rats. On average, the variant is found at a frequency of ~15%. The CR set, these variants appear in 50% of the population</li>
<li>4 of the 5 rats from above were identified in</li>
<li>0 sites in the nih dbSNP database for rats</li>
</ul>
<p>Here, compare the frequencies of the most common mutations in our population with those reported in the harlan and charles river data-bases:</p>
<ul>
<li>filter out mutatations occuring in 2+ samples</li>
</ul>
<p><img src="RatDCIS-bookdown_files/figure-html/pop-gen-plot-1.png" width="672" /></p>
<pre><code>## [1] 1093</code></pre>
<p><img src="RatDCIS-bookdown_files/figure-html/pop-gen-plot-2.png" width="672" /></p>
<pre><code>##      freq.Freq   gene      VariantClass Cosmic   variable      value
## 140         11   Prf1 Missense_Mutation      1 HarlanFreq 0.45287155
## 699          3 Rnf213 Missense_Mutation      1 HarlanFreq 0.14028935
## 700          3 Rnf213 Missense_Mutation      1 HarlanFreq 0.14028935
## 701          3 Rnf213 Missense_Mutation      1 HarlanFreq 0.14116616
## 805          3   Tcf3 Missense_Mutation      1 HarlanFreq 0.06795265
## 953          2 Ptpn13 Missense_Mutation      1 HarlanFreq 0.02718106
## 1792         3 Rnf213 Missense_Mutation      1     CRFreq 0.52640449
## 1793         3 Rnf213 Missense_Mutation      1     CRFreq 0.52640449
## 1794         3 Rnf213 Missense_Mutation      1     CRFreq 0.53651685
## 1898         3   Tcf3 Missense_Mutation      1     CRFreq 0.14887640</code></pre>
</div>
</div>
<div id="overview-of-the-mutations" class="section level2">
<h2><span class="header-section-number">13.4</span> Overview of the mutations</h2>
<div id="all-mutations" class="section level3">
<h3><span class="header-section-number">13.4.1</span> All mutations</h3>
<p>SNPs are removed from this analysis: based on frequencies in the harlan cohort</p>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-153-1.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-153-2.png" width="672" /><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-153-3.png" width="672" /></p>
<p>test break here</p>
<p>Save these lists for metacore</p>
<p>Load the metacore results:</p>
</div>
<div id="cosmic-only" class="section level3">
<h3><span class="header-section-number">13.4.2</span> Cosmic only:</h3>
<p>Probably don’t need this section?</p>
<p>Note the cut-offs:</p>
<table>
<thead>
<tr class="header">
<th>Mut in N samples</th>
<th>Genes</th>
<th>NO genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>27</td>
<td>Atm Tert</td>
<td>2</td>
</tr>
<tr class="even">
<td>19</td>
<td>Prf1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>18</td>
<td>Muc1 Ncor2 Clip1</td>
<td>3</td>
</tr>
<tr class="even">
<td>3</td>
<td>Pik3ca, Brca2</td>
<td>17</td>
</tr>
<tr class="odd">
<td>2</td>
<td>Braf, Nras, Bcl9l Atr Erbb3 Gata2</td>
<td>49</td>
</tr>
</tbody>
</table>
<p>Based on this data, check whether the mutation sites are listed in the above db Sets</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bcr-clonotype-analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
