<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Whole-slide imaging | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Whole-slide imaging | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Whole-slide imaging | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cohort-summary-table.html"/>
<link rel="next" href="spatial-statistics.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redo-the-survival-plots-based-on-luminal-cases-only"><i class="fa fa-check"></i><b>15.4.3</b> Redo the survival plots based on Luminal cases only</a></li>
<li class="chapter" data-level="15.4.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.4</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>16</b> METABRIC: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="16.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="16.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="16.2.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#all-samples-1"><i class="fa fa-check"></i><b>16.2.1</b> All samples</a></li>
<li class="chapter" data-level="16.2.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases-1"><i class="fa fa-check"></i><b>16.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.4" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint-1"><i class="fa fa-check"></i><b>16.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="16.4.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#overall-survival-1"><i class="fa fa-check"></i><b>16.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="16.4.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#met-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>16.4.2</b> MET: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>21</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="21.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>21.1</b> Data</a><ul>
<li class="chapter" data-level="21.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>21.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>21.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="21.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>21.3</b> Plots</a><ul>
<li class="chapter" data-level="21.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>21.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="21.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>21.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>21.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="21.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>21.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>21.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="21.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>21.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>21.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>22</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>22.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>22.2</b> Load files</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>22.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="22.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>22.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="22.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>22.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>23</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>23.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>23.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-epdn-specific"><i class="fa fa-check"></i><b>23.3</b> Also look at the top RNA-mutations with are Ep/DN specific</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>24</b> Trichrome staining</a><ul>
<li class="chapter" data-level="24.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>24.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="24.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>24.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="24.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>24.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="24.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>24.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>25</b> CNV calling</a><ul>
<li class="chapter" data-level="25.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>25.1</b> Sample data</a></li>
<li class="chapter" data-level="25.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>25.2</b> Data Summary</a><ul>
<li class="chapter" data-level="25.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>25.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="25.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>25.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="25.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>25.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>25.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="25.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>25.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>26</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="whole-slide-imaging" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Whole-slide imaging</h1>
<p>In this section, we will be looking at the composition and spatial distribution of cells in whole slide images. These sections have previously been assessed using an external script: SpatialStatisticsWSI.R</p>
<p>The following markers have been used:</p>
<ul>
<li>EpCAM (tumor cells)</li>
<li>SMA (fibroblasts or myeopithelial cells)</li>
<li>CD8 (T cells)</li>
</ul>
<p>Note that in some images a double positive EpCAM+/SMA+ population exists. Some CD8 cells have Epcam+ or SMA+ staining, however, we consider all of these to be simply CD8+</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="whole-slide-imaging.html#cb50-1"></a>WSIpath=<span class="st">&quot;../data/WSI-data/locationData/&quot;</span></span>
<span id="cb50-2"><a href="whole-slide-imaging.html#cb50-2"></a>WSIfiles=<span class="kw">dir</span>(WSIpath, <span class="st">&quot;*.csv&quot;</span>)</span>
<span id="cb50-3"><a href="whole-slide-imaging.html#cb50-3"></a>WSIsummary=<span class="kw">lapply</span>(WSIfiles, <span class="cf">function</span>(x) <span class="kw">read.csv</span>(<span class="kw">paste</span>(WSIpath, x, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)))</span>
<span id="cb50-4"><a href="whole-slide-imaging.html#cb50-4"></a><span class="kw">names</span>(WSIsummary)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(WSIfiles, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb50-5"><a href="whole-slide-imaging.html#cb50-5"></a></span>
<span id="cb50-6"><a href="whole-slide-imaging.html#cb50-6"></a><span class="co">## knn data</span></span>
<span id="cb50-7"><a href="whole-slide-imaging.html#cb50-7"></a>WSIpath=<span class="st">&quot;../data/WSI-data/knn-values/&quot;</span></span>
<span id="cb50-8"><a href="whole-slide-imaging.html#cb50-8"></a>WSIfiles=<span class="kw">dir</span>(WSIpath, <span class="st">&quot;*.csv&quot;</span>)</span>
<span id="cb50-9"><a href="whole-slide-imaging.html#cb50-9"></a>WSIknn=<span class="kw">lapply</span>(WSIfiles, <span class="cf">function</span>(x) <span class="kw">read.csv</span>(<span class="kw">paste</span>(WSIpath, x, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)))</span>
<span id="cb50-10"><a href="whole-slide-imaging.html#cb50-10"></a><span class="kw">names</span>(WSIknn)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(WSIfiles, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb50-11"><a href="whole-slide-imaging.html#cb50-11"></a></span>
<span id="cb50-12"><a href="whole-slide-imaging.html#cb50-12"></a><span class="co">## knn data</span></span>
<span id="cb50-13"><a href="whole-slide-imaging.html#cb50-13"></a>WSIpath=<span class="st">&quot;../data/WSI-data/interactingFraction/&quot;</span></span>
<span id="cb50-14"><a href="whole-slide-imaging.html#cb50-14"></a>WSIfiles=<span class="kw">dir</span>(WSIpath, <span class="st">&quot;*.csv&quot;</span>)</span>
<span id="cb50-15"><a href="whole-slide-imaging.html#cb50-15"></a>WSIIF=<span class="kw">lapply</span>(WSIfiles, <span class="cf">function</span>(x) <span class="kw">read.csv</span>(<span class="kw">paste</span>(WSIpath, x, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)))</span>
<span id="cb50-16"><a href="whole-slide-imaging.html#cb50-16"></a>WSIIF=<span class="kw">lapply</span>(WSIIF, <span class="cf">function</span>(x) {<span class="kw">colnames</span>(x)&lt;-<span class="kw">c</span>(<span class="st">&quot;RN&quot;</span>,<span class="st">&quot;Grid&quot;</span>, <span class="st">&quot;NearestNeighbor&quot;</span>, <span class="st">&quot;IF&quot;</span>, <span class="st">&quot;Reference&quot;</span>); x})</span>
<span id="cb50-17"><a href="whole-slide-imaging.html#cb50-17"></a><span class="kw">names</span>(WSIIF)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(WSIfiles, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb50-18"><a href="whole-slide-imaging.html#cb50-18"></a></span>
<span id="cb50-19"><a href="whole-slide-imaging.html#cb50-19"></a><span class="co"># MH data</span></span>
<span id="cb50-20"><a href="whole-slide-imaging.html#cb50-20"></a>WSIpath=<span class="st">&quot;../data/WSI-data/mh-values/&quot;</span></span>
<span id="cb50-21"><a href="whole-slide-imaging.html#cb50-21"></a>WSIfiles=<span class="kw">dir</span>(WSIpath, <span class="st">&quot;*.csv&quot;</span>)</span>
<span id="cb50-22"><a href="whole-slide-imaging.html#cb50-22"></a>WSIMH=<span class="kw">lapply</span>(WSIfiles, <span class="cf">function</span>(x) <span class="kw">read.csv</span>(<span class="kw">paste</span>(WSIpath, x, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)))</span>
<span id="cb50-23"><a href="whole-slide-imaging.html#cb50-23"></a><span class="kw">names</span>(WSIMH)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(WSIfiles, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb50-24"><a href="whole-slide-imaging.html#cb50-24"></a></span>
<span id="cb50-25"><a href="whole-slide-imaging.html#cb50-25"></a><span class="co"># MH set-up summary</span></span>
<span id="cb50-26"><a href="whole-slide-imaging.html#cb50-26"></a>WSIpath=<span class="st">&quot;../data/WSI-data/MHset-upSummary/&quot;</span></span>
<span id="cb50-27"><a href="whole-slide-imaging.html#cb50-27"></a>WSIfiles=<span class="kw">dir</span>(WSIpath, <span class="st">&quot;*.csv&quot;</span>)</span>
<span id="cb50-28"><a href="whole-slide-imaging.html#cb50-28"></a>WSIMHsetup=<span class="kw">lapply</span>(WSIfiles, <span class="cf">function</span>(x) <span class="kw">read.csv</span>(<span class="kw">paste</span>(WSIpath, x, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)))</span>
<span id="cb50-29"><a href="whole-slide-imaging.html#cb50-29"></a><span class="kw">names</span>(WSIMHsetup)=<span class="kw">sapply</span>(<span class="kw">strsplit</span>(WSIfiles, <span class="st">&quot;_&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])</span>
<span id="cb50-30"><a href="whole-slide-imaging.html#cb50-30"></a></span>
<span id="cb50-31"><a href="whole-slide-imaging.html#cb50-31"></a><span class="kw">save</span>(WSIsummary, WSIknn, WSIIF, WSIMH, WSIMHsetup, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;outputs/WSI_raw_data%s.RData&quot;</span>, <span class="kw">Sys.Date</span>()))</span></code></pre></div>
<div id="associate-the-frequencies-with-other-data-types" class="section level2">
<h2><span class="header-section-number">4.1</span> Associate the frequencies with other data types</h2>
<p><span style="color:red">
UPDATE THE CIBERSORT INFORMATION. </span></p>
<p>Note there are 47 samples with imaging data. 33 of these samples have FACS data, manual counts and TIMER scores</p>
<p>Correlate the following information:</p>
<ul>
<li>“CD8.WSI”: CD8 total counts</li>
<li>“CD8Frac.WSI”: CD8 fraction (normalised by cell count)</li>
<li>“CD8_EPorSMARatio.WSI”: CD8/EP+SMA ratio (any EpCAM or SMA + cell)</li>
<li>“CD8_AnySMARatio.WSI”: CD8/Any EPcam+ cell</li>
<li>“CD8_EPRatio.WSI”: CD8 to EpCAM+SMA- ratio</li>
<li>“CD8normTumSize”: normalised CD8 counts per mm of tumor size at sac</li>
<li>“CD8.EpBoundingBox”: approx area per CD8 cell (density)</li>
</ul>
<p><span style="color:red">
UPDATE THE CIBERSORT INFORMATION. </span></p>
<p>Below are heatmaps which show the correlation between two variables (red is correlated and blue is anti-correlated), and the p.value is indicated in the middle of the square.</p>
<p>It appears that CD8 whole-slide imaging associates well with:</p>
<ul>
<li>FACS data (both CD8 and CD45)</li>
<li>Manual scoring (Fig 4) of CD8 cells</li>
<li>Some CD8 gene signature scores (mainly in EPC, TIMER)</li>
</ul>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="whole-slide-imaging.html#cb51-1"></a>SummaryData=<span class="kw">read.csv</span>(<span class="st">&quot;../metadata/summary_final_210330.csv&quot;</span>, <span class="dt">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb51-2"><a href="whole-slide-imaging.html#cb51-2"></a></span>
<span id="cb51-3"><a href="whole-slide-imaging.html#cb51-3"></a><span class="co">## CD summary</span></span>
<span id="cb51-4"><a href="whole-slide-imaging.html#cb51-4"></a>NAidx=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(SummaryData), <span class="cf">function</span>(x) <span class="kw">length</span>(<span class="kw">which</span>(<span class="op">!</span><span class="kw">is.na</span>(SummaryData[, x]))))</span>
<span id="cb51-5"><a href="whole-slide-imaging.html#cb51-5"></a><span class="kw">names</span>(NAidx)=<span class="kw">colnames</span>(SummaryData)</span>
<span id="cb51-6"><a href="whole-slide-imaging.html#cb51-6"></a></span>
<span id="cb51-7"><a href="whole-slide-imaging.html#cb51-7"></a>colTestCD8=<span class="kw">c</span>(<span class="st">&quot;CD8.EpBoundingBox&quot;</span>, <span class="st">&quot;CD45.FACS&quot;</span>, <span class="st">&quot;CD8.FACS&quot;</span>, <span class="st">&quot;OverallCD8_Fig4c.Manual&quot;</span>, <span class="st">&quot;TIL_Fig4d.Manual&quot;</span>, <span class="st">&quot;T.cell.CD8._TIMER&quot;</span>, <span class="st">&quot;T.cell.CD8._CIBERSORT&quot;</span>,<span class="st">&quot;T.cell.CD8._CIBERSORT.ABS&quot;</span>, <span class="st">&quot;T.cell.CD8._EPIC&quot;</span>)</span>
<span id="cb51-8"><a href="whole-slide-imaging.html#cb51-8"></a></span>
<span id="cb51-9"><a href="whole-slide-imaging.html#cb51-9"></a><span class="co"># colTestCD8=c(&quot;CD8.EpDomTiles&quot;,&quot;CD8.EpBoundingBox&quot;, &quot;CD8.TumSize&quot;, &quot;CD45.FACS&quot;, &quot;CD8.FACS&quot;, &quot;OverallCD8_Fig4c.Manual&quot;, &quot;TIL_Fig4d.Manual&quot;, )</span></span>
<span id="cb51-10"><a href="whole-slide-imaging.html#cb51-10"></a>indx1=<span class="kw">c</span>(<span class="st">&quot;CD8.WSI&quot;</span>, <span class="st">&quot;CD8Frac.WSI&quot;</span>, <span class="st">&quot;CD8_EPorSMARatio.WSI&quot;</span>,<span class="st">&quot;CD8_AnySMARatio.WSI&quot;</span>, <span class="st">&quot;CD8_EPRatio.WSI&quot;</span>,<span class="st">&quot;CD8normTumSize&quot;</span>, <span class="st">&quot;CD8.EpBoundingBox&quot;</span>) <span class="co">#, &quot;log2CD8_EPorSMARatio.WSI&quot;, &quot;log2CD8_EPRatio.WSI&quot;)</span></span>
<span id="cb51-11"><a href="whole-slide-imaging.html#cb51-11"></a></span>
<span id="cb51-12"><a href="whole-slide-imaging.html#cb51-12"></a>CDsummary=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="kw">length</span>(indx1), <span class="dt">ncol=</span><span class="kw">length</span>(colTestCD8))</span>
<span id="cb51-13"><a href="whole-slide-imaging.html#cb51-13"></a><span class="kw">rownames</span>(CDsummary)=<span class="kw">paste</span>(indx1, <span class="st">&quot; N=&quot;</span>, NAidx[<span class="kw">match</span>(indx1, <span class="kw">names</span>(NAidx))], <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb51-14"><a href="whole-slide-imaging.html#cb51-14"></a><span class="kw">colnames</span>(CDsummary)=<span class="kw">paste</span>(colTestCD8, <span class="st">&quot; N=&quot;</span>, NAidx[<span class="kw">match</span>(colTestCD8, <span class="kw">names</span>(NAidx))], <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb51-15"><a href="whole-slide-imaging.html#cb51-15"></a>CDsummaryP=CDsummary</span>
<span id="cb51-16"><a href="whole-slide-imaging.html#cb51-16"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(indx1)){</span>
<span id="cb51-17"><a href="whole-slide-imaging.html#cb51-17"></a>  CDsummary[i, ]=<span class="kw">sapply</span>(colTestCD8, <span class="cf">function</span>(x) <span class="kw">cor</span>(SummaryData[, indx1[i]], SummaryData[,x], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>))</span>
<span id="cb51-18"><a href="whole-slide-imaging.html#cb51-18"></a>  CDsummaryP[i, ]=<span class="kw">sapply</span>(colTestCD8, <span class="cf">function</span>(x) <span class="kw">cor.test</span>(SummaryData[, indx1[i]], SummaryData[,x], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb51-19"><a href="whole-slide-imaging.html#cb51-19"></a>}</span>
<span id="cb51-20"><a href="whole-slide-imaging.html#cb51-20"></a></span>
<span id="cb51-21"><a href="whole-slide-imaging.html#cb51-21"></a><span class="co">## Do the same with EPCAM/FACS data</span></span>
<span id="cb51-22"><a href="whole-slide-imaging.html#cb51-22"></a></span>
<span id="cb51-23"><a href="whole-slide-imaging.html#cb51-23"></a>colTestFACS=<span class="kw">grep</span>(<span class="st">&quot;FACS&quot;</span>, <span class="kw">colnames</span>(SummaryData), <span class="dt">value =</span> T)</span>
<span id="cb51-24"><a href="whole-slide-imaging.html#cb51-24"></a>colTestWSI=<span class="kw">grep</span>(<span class="st">&quot;WSI&quot;</span>, <span class="kw">colnames</span>(SummaryData), <span class="dt">value =</span> T)</span>
<span id="cb51-25"><a href="whole-slide-imaging.html#cb51-25"></a>FACSsummary=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="kw">length</span>(colTestFACS), <span class="dt">ncol=</span><span class="kw">length</span>(colTestWSI))</span>
<span id="cb51-26"><a href="whole-slide-imaging.html#cb51-26"></a><span class="kw">rownames</span>(FACSsummary)=<span class="kw">paste</span>(colTestFACS, <span class="st">&quot; N=&quot;</span>, NAidx[<span class="kw">match</span>(colTestFACS, <span class="kw">names</span>(NAidx))], <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb51-27"><a href="whole-slide-imaging.html#cb51-27"></a><span class="kw">colnames</span>(FACSsummary)=<span class="kw">paste</span>(colTestWSI, <span class="st">&quot; N=&quot;</span>, NAidx[<span class="kw">match</span>(colTestWSI, <span class="kw">names</span>(NAidx))], <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb51-28"><a href="whole-slide-imaging.html#cb51-28"></a>FACSsummaryP=FACSsummary</span>
<span id="cb51-29"><a href="whole-slide-imaging.html#cb51-29"></a></span>
<span id="cb51-30"><a href="whole-slide-imaging.html#cb51-30"></a></span>
<span id="cb51-31"><a href="whole-slide-imaging.html#cb51-31"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(colTestFACS)){</span>
<span id="cb51-32"><a href="whole-slide-imaging.html#cb51-32"></a>  FACSsummary[i, ]=<span class="kw">sapply</span>(colTestWSI, <span class="cf">function</span>(x) <span class="kw">cor</span>(SummaryData[, colTestFACS[i]], SummaryData[,x], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>))</span>
<span id="cb51-33"><a href="whole-slide-imaging.html#cb51-33"></a>  FACSsummaryP[i, ]=<span class="kw">sapply</span>(colTestWSI, <span class="cf">function</span>(x) <span class="kw">cor.test</span>(SummaryData[, colTestFACS[i]], SummaryData[,x], <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)<span class="op">$</span>p.value)</span>
<span id="cb51-34"><a href="whole-slide-imaging.html#cb51-34"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="whole-slide-imaging.html#cb52-1"></a><span class="kw">PvalHM</span>(CDsummary, <span class="kw">round</span>(CDsummaryP,<span class="dv">2</span>), <span class="st">&quot;CD8 WSI correlation with Pvalues&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Ext4b"></span>
<img src="RatDCIS-bookdown_files/figure-html/Ext4b-1.png" alt="association with facs" width="672" />
<p class="caption">
Figure 4.1: association with facs
</p>
</div>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="whole-slide-imaging.html#cb53-1"></a><span class="kw">PvalHM</span>(FACSsummary, <span class="kw">round</span>(FACSsummaryP, <span class="dv">2</span>), <span class="st">&quot;WSI vs FACS w Pvalues&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
</div>
<div id="cellular-composition" class="section level2">
<h2><span class="header-section-number">4.2</span> Cellular composition</h2>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="whole-slide-imaging.html#cb54-1"></a>WSIvals=<span class="kw">sapply</span>(WSIsummary, <span class="cf">function</span>(x) <span class="kw">table</span>(<span class="kw">factor</span>(x<span class="op">$</span>Class2, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;CD8&quot;</span>, <span class="st">&quot;EpCAM&quot;</span>, <span class="st">&quot;EpCAM: SMA&quot;</span>, <span class="st">&quot;SMA&quot;</span>, <span class="st">&quot;Unclass&quot;</span>))))</span>
<span id="cb54-2"><a href="whole-slide-imaging.html#cb54-2"></a>WSIvalFracs=<span class="kw">t</span>(<span class="kw">t</span>(WSIvals)<span class="op">/</span><span class="kw">colSums</span>(WSIvals))</span>
<span id="cb54-3"><a href="whole-slide-imaging.html#cb54-3"></a></span>
<span id="cb54-4"><a href="whole-slide-imaging.html#cb54-4"></a>lxmatch=<span class="kw">match</span>(<span class="kw">colnames</span>(WSIvalFracs), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))</span>
<span id="cb54-5"><a href="whole-slide-imaging.html#cb54-5"></a></span>
<span id="cb54-6"><a href="whole-slide-imaging.html#cb54-6"></a>Cdata<span class="op">$</span>CD8Fraction=<span class="ot">NA</span></span>
<span id="cb54-7"><a href="whole-slide-imaging.html#cb54-7"></a>Cdata<span class="op">$</span>CD8Fraction[<span class="kw">na.omit</span>(lxmatch)]=WSIvalFracs[<span class="dv">1</span>, ]</span>
<span id="cb54-8"><a href="whole-slide-imaging.html#cb54-8"></a></span>
<span id="cb54-9"><a href="whole-slide-imaging.html#cb54-9"></a>Cdata<span class="op">$</span>EpCAMFraction=<span class="ot">NA</span></span>
<span id="cb54-10"><a href="whole-slide-imaging.html#cb54-10"></a>Cdata<span class="op">$</span>EpCAMFraction[<span class="kw">na.omit</span>(lxmatch)]=WSIvalFracs[<span class="dv">2</span>, ]</span>
<span id="cb54-11"><a href="whole-slide-imaging.html#cb54-11"></a></span>
<span id="cb54-12"><a href="whole-slide-imaging.html#cb54-12"></a>Cdata<span class="op">$</span>DPFraction=<span class="ot">NA</span></span>
<span id="cb54-13"><a href="whole-slide-imaging.html#cb54-13"></a>Cdata<span class="op">$</span>DPFraction[<span class="kw">na.omit</span>(lxmatch)]=WSIvalFracs[<span class="dv">3</span>, ]</span>
<span id="cb54-14"><a href="whole-slide-imaging.html#cb54-14"></a></span>
<span id="cb54-15"><a href="whole-slide-imaging.html#cb54-15"></a>Cdata<span class="op">$</span>SMAFraction=<span class="ot">NA</span></span>
<span id="cb54-16"><a href="whole-slide-imaging.html#cb54-16"></a>Cdata<span class="op">$</span>SMAFraction[<span class="kw">na.omit</span>(lxmatch)]=WSIvalFracs[<span class="dv">4</span>, ]</span>
<span id="cb54-17"><a href="whole-slide-imaging.html#cb54-17"></a></span>
<span id="cb54-18"><a href="whole-slide-imaging.html#cb54-18"></a>Cdata<span class="op">$</span>UnclassFraction=<span class="ot">NA</span></span>
<span id="cb54-19"><a href="whole-slide-imaging.html#cb54-19"></a>Cdata<span class="op">$</span>UnclassFraction[<span class="kw">na.omit</span>(lxmatch)]=WSIvalFracs[<span class="dv">5</span>, ]</span>
<span id="cb54-20"><a href="whole-slide-imaging.html#cb54-20"></a></span>
<span id="cb54-21"><a href="whole-slide-imaging.html#cb54-21"></a><span class="co">#write.csv(WSIvals, file=&quot;nature-tables/raw_WSI_values.csv&quot;)</span></span>
<span id="cb54-22"><a href="whole-slide-imaging.html#cb54-22"></a><span class="co">#write.csv(WSIvalFracs, file=&quot;nature-tables/raw_WSI_values_fractions.csv&quot;)</span></span>
<span id="cb54-23"><a href="whole-slide-imaging.html#cb54-23"></a></span>
<span id="cb54-24"><a href="whole-slide-imaging.html#cb54-24"></a>df.Spatial=<span class="kw">cbind</span>(<span class="kw">t</span>(WSIvals), <span class="kw">t</span>(WSIvalFracs))</span>
<span id="cb54-25"><a href="whole-slide-imaging.html#cb54-25"></a><span class="kw">colnames</span>(df.Spatial)=<span class="kw">c</span>(<span class="st">&quot;CD8&quot;</span>, <span class="st">&quot;EpCAM&quot;</span>, <span class="st">&quot;EpCAM:SMA&quot;</span>, <span class="st">&quot;SMA&quot;</span>, <span class="st">&quot;Unclass&quot;</span>, <span class="kw">paste</span>(<span class="kw">c</span>(<span class="st">&quot;CD8&quot;</span>, <span class="st">&quot;EpCAM&quot;</span>, <span class="st">&quot;EpCAM:SMA&quot;</span>, <span class="st">&quot;SMA&quot;</span>, <span class="st">&quot;Unclass&quot;</span>), <span class="st">&quot;frac&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb54-26"><a href="whole-slide-imaging.html#cb54-26"></a></span>
<span id="cb54-27"><a href="whole-slide-imaging.html#cb54-27"></a>DT<span class="op">::</span><span class="kw">datatable</span>(df.Spatial, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb54-28"><a href="whole-slide-imaging.html#cb54-28"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div id="htmlwidget-2a5d57a6420fbb80c8c0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2a5d57a6420fbb80c8c0">{"x":{"filter":"none","extensions":["Buttons"],"data":[[65507,3481,67342,27239,22618,3393,178767,1823,196293,42097,40084,29817,2692,3560,121602,97911,11415,2194,6350,3993,1678,24041,12242,22907,69599,137787,22697,114939,13298,85774,109694,24008,31061,83808,98748,43810,11691,37888,4456,4010,69477,30482,7634,93441,25942,187348,78416,170266,79608,157455,32396,106074,72918,23710,50862,7860,4518,83679],[761887,102702,192699,324933,463070,58223,437050,15806,587504,46872,247458,352081,14614,117731,1156717,340247,115152,107760,105525,83111,64379,109825,303229,229322,779658,687621,179849,444216,62937,1378423,1143016,120410,111306,623759,168767,717597,195683,542224,396164,198640,752029,234195,139335,516380,176986,1078667,377308,980499,90700,495313,417604,310952,530505,284804,403704,82972,141557,750075],[343389,58017,204061,0,0,25198,895393,1209,13,2054,137553,198409,289,1764,0,59025,26633,28128,20305,0,8785,0,0,0,0,242704,0,0,203,250877,335772,473,51141,232788,0,0,58422,163326,0,0,407898,0,0,0,7127,0,0,223461,0,238235,0,0,441179,735372,0,259,0,0],[1126173,106919,402092,167028,282453,42642,763776,5729,47,1551,165687,375106,125,316,1221416,167505,17181,18909,130687,99191,16699,0,267100,115205,905754,572319,372829,1252644,1332,112621,714690,10966,76195,157121,15223,625176,76220,231943,96651,265301,556391,320467,0,833223,54197,1452387,912277,528928,247319,782142,543360,1628191,772709,167172,886443,508,63579,1475968],[563894,75799,122548,65796,227392,37922,390628,19118,506036,18188,204242,373020,7425,35776,917869,257678,40501,55255,227056,183857,68695,598394,131130,112871,446321,188661,165181,619810,41466,258890,1340651,123894,148799,480097,73972,256989,161604,574396,50231,57984,694672,86337,149118,371152,480977,421479,41718,183692,94386,324989,573677,167163,466302,188017,240078,48175,46047,599152],[0.0228977401821137,0.0100340714520434,0.0681087685159526,0.046562711539908,0.0227194879526846,0.0202714813177359,0.0670640985529038,0.0417305711342566,0.152177738773681,0.380067171051444,0.050418603715108,0.0224452418752018,0.107059057466693,0.0223692560965648,0.0355810679060535,0.106152004735647,0.0541297977067744,0.0103370617114103,0.0129612204366809,0.010787460286585,0.0104720537207619,0.0328312348073089,0.0171528413159012,0.0476926119861338,0.0316167665758732,0.0753308198822148,0.0306485937592836,0.0472687015058753,0.111526720118085,0.0411073596330847,0.0301040967138085,0.0858191749091156,0.0742194780431157,0.0531246414587471,0.276829917860447,0.0266553579642389,0.0232139311385568,0.0244473882371464,0.00813878305467377,0.00762451633757023,0.0280096449579857,0.0453951787168959,0.0257829624400936,0.0515054602700039,0.0348107762848735,0.0596672294268477,0.0556252700006171,0.0815901125430434,0.155480427254777,0.0788010213529223,0.0206734110298608,0.0479456512895615,0.0319309795486363,0.0169469113521434,0.0321690077775606,0.0562336342953625,0.0176690744267719,0.0287668011746126],[0.266314906408934,0.296041139404701,0.194893106594036,0.555444823554349,0.465147815290904,0.347853361851617,0.163958472607062,0.361817557514021,0.455467236429688,0.423177624094906,0.311258528044436,0.265034819219336,0.581189103201432,0.739762609411425,0.338458463882884,0.368885019612605,0.546049449455146,0.507712748414576,0.215390990012716,0.224532084116795,0.401776130207943,0.149980881107803,0.424868397270005,0.477450786479425,0.354175562795616,0.375935710177509,0.242856718465585,0.182683975918826,0.5278355530209,0.660611956857736,0.313685928213308,0.430418479290512,0.265962886676766,0.395391528632906,0.473121022679488,0.436608192400455,0.388552877169294,0.349872271946222,0.723584571380561,0.377689258178292,0.30318040917295,0.348773829788185,0.470588036624371,0.2846329724021,0.237492099743837,0.343537541709383,0.267647665953286,0.469847319830979,0.177143939704656,0.247887779298085,0.26649275033072,0.140550899935816,0.232309502529544,0.203565927487804,0.255333197983413,0.593615407729621,0.553603622981529,0.257857507750422],[0.120030410542321,0.167235485042575,0.206384476435713,0,0,0.150545471925821,0.335904973488285,0.027675403456564,1.00783553364504e-05,0.018544266084036,0.173017418342088,0.149355669424051,0.0114933386359117,0.0110840920658259,0,0.063993035302689,0.126293377338986,0.132525465733159,0.0414452883412291,0,0.0548253825607229,0,0,0,0,0.132690974538186,0,0,0.00170250595457748,0.12023329986557,0.092148273941956,0.00169078930906413,0.122200132854801,0.147560841875463,0,0,0.11600413009809,0.105386775000532,0,0,0.164444034127445,0,0,0,0.00956350329898595,0,0,0.107080733317169,0,0.119228740414807,0,0,0.193193417623739,0.525612994299805,0,0.00185299125731538,0,0],[0.393649789398256,0.308196749664186,0.406670294171786,0.285519900990776,0.283720378932692,0.254764664412288,0.286529107365132,0.131143413070848,3.64371308317822e-05,0.014002997417887,0.208405029282135,0.282367270310208,0.00497116723006562,0.0019855856535153,0.35738956298038,0.181603615050858,0.0814721028821806,0.089090018186444,0.266750081135199,0.267973697291923,0.104215032826581,0,0.374246358068715,0.239858006891454,0.411457244977132,0.312897875011208,0.503444709110452,0.515150256476267,0.0111711228152571,0.0539738376342205,0.196137408430651,0.0391991449539054,0.182066035526712,0.0995966589184779,0.0426761234616355,0.380376399695298,0.151344267503276,0.149662177203559,0.176530861987719,0.504436860068259,0.224308970851053,0.477254010165589,0,0.459279482481496,0.0727252965195933,0.462561160757366,0.647133932365244,0.25345808938465,0.483032657373934,0.391436209983915,0.346743567637522,0.735945452408718,0.338371256425673,0.119487518539035,0.560654157551102,0.00363443845064175,0.248645879366917,0.507401833149184],[0.197107153468375,0.218492554436495,0.123943354282513,0.112472563914967,0.228412317823719,0.226565020492538,0.146543347986618,0.43763305482431,0.392308509310462,0.164207941351727,0.256900420616233,0.280796999171204,0.295287333465898,0.224798456772669,0.268570905230682,0.2793663252982,0.192055272616914,0.260334705954411,0.463452420074175,0.496706758304696,0.428711400683991,0.817187884084888,0.183732403345379,0.234998594642987,0.202750425651378,0.103144620390882,0.223049978664679,0.254897066099032,0.347764098091181,0.124073546009389,0.367924292700277,0.442872411537403,0.355551466898605,0.304326329114405,0.20737293599843,0.156360049940009,0.320884794090783,0.37063138761254,0.0917457835770463,0.110249365415878,0.280056940890566,0.12857698132933,0.503629000935536,0.2045820848464,0.64540832415271,0.134234068106403,0.0295931316808527,0.0880237449241583,0.184342975666633,0.162646248950271,0.366090271001897,0.0755579963659046,0.204194843872407,0.134386648321212,0.151843636687924,0.34466352826706,0.180081423224782,0.205973857925782]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>CD8<\/th>\n      <th>EpCAM<\/th>\n      <th>EpCAM:SMA<\/th>\n      <th>SMA<\/th>\n      <th>Unclass<\/th>\n      <th>CD8frac<\/th>\n      <th>EpCAMfrac<\/th>\n      <th>EpCAM:SMAfrac<\/th>\n      <th>SMAfrac<\/th>\n      <th>Unclassfrac<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[0,1,2,3,4,5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Here, we look at the raw distributions of the different cell types and see if there are associations with:</p>
<ul>
<li>tumor size</li>
<li>growth rate</li>
<li>growth rate (categorical)</li>
<li>treatment</li>
<li>stromal restricted or infiltrating</li>
</ul>
<p>Below are the total cell counts:</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="whole-slide-imaging.html#cb55-1"></a>WSIvals=WSIvals[ , <span class="kw">order</span>(WSIvals[<span class="dv">1</span>, ])]</span>
<span id="cb55-2"><a href="whole-slide-imaging.html#cb55-2"></a>ordV=<span class="kw">colnames</span>(WSIvals)</span>
<span id="cb55-3"><a href="whole-slide-imaging.html#cb55-3"></a></span>
<span id="cb55-4"><a href="whole-slide-imaging.html#cb55-4"></a>WSIvalFracs=WSIvalFracs[ , <span class="kw">order</span>(WSIvalFracs[<span class="dv">1</span>, ])]</span>
<span id="cb55-5"><a href="whole-slide-imaging.html#cb55-5"></a>ordV2=<span class="kw">colnames</span>(WSIvalFracs)</span>
<span id="cb55-6"><a href="whole-slide-imaging.html#cb55-6"></a></span>
<span id="cb55-7"><a href="whole-slide-imaging.html#cb55-7"></a></span>
<span id="cb55-8"><a href="whole-slide-imaging.html#cb55-8"></a><span class="co">## match with meta data# get the sample Names</span></span>
<span id="cb55-9"><a href="whole-slide-imaging.html#cb55-9"></a>CdataTID=<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID)</span>
<span id="cb55-10"><a href="whole-slide-imaging.html#cb55-10"></a>l1=<span class="kw">match</span>(<span class="kw">colnames</span>(WSIvals), CdataTID)</span>
<span id="cb55-11"><a href="whole-slide-imaging.html#cb55-11"></a>namesA=Cdata<span class="op">$</span>NewID[l1]</span>
<span id="cb55-12"><a href="whole-slide-imaging.html#cb55-12"></a><span class="co">###</span></span>
<span id="cb55-13"><a href="whole-slide-imaging.html#cb55-13"></a>Treat=Cdata<span class="op">$</span>Treatment[l1]</span>
<span id="cb55-14"><a href="whole-slide-imaging.html#cb55-14"></a>Growth=Cdata<span class="op">$</span>Growth2[l1]</span>
<span id="cb55-15"><a href="whole-slide-imaging.html#cb55-15"></a>WSImelt=<span class="kw">melt</span>(WSIvals)</span>
<span id="cb55-16"><a href="whole-slide-imaging.html#cb55-16"></a><span class="co">#WSImelt$Var2=factor(WSImelt$Var2, levels=ordV)</span></span>
<span id="cb55-17"><a href="whole-slide-imaging.html#cb55-17"></a>WSImelt<span class="op">$</span>treatment=Treat[<span class="kw">match</span>(WSImelt<span class="op">$</span>Var2, <span class="kw">colnames</span>(WSIvals))]</span>
<span id="cb55-18"><a href="whole-slide-imaging.html#cb55-18"></a>WSImelt<span class="op">$</span>growth=Growth[<span class="kw">match</span>(WSImelt<span class="op">$</span>Var2, <span class="kw">colnames</span>(WSIvals))]</span>
<span id="cb55-19"><a href="whole-slide-imaging.html#cb55-19"></a>WSImelt<span class="op">$</span>Name=namesA[<span class="kw">match</span>(WSImelt<span class="op">$</span>Var2, <span class="kw">colnames</span>(WSIvals))]</span>
<span id="cb55-20"><a href="whole-slide-imaging.html#cb55-20"></a>WSImelt<span class="op">$</span>Name=<span class="kw">factor</span>(WSImelt<span class="op">$</span>Name, <span class="dt">levels=</span>namesA)</span>
<span id="cb55-21"><a href="whole-slide-imaging.html#cb55-21"></a></span>
<span id="cb55-22"><a href="whole-slide-imaging.html#cb55-22"></a>l2=<span class="kw">match</span>(<span class="kw">colnames</span>(WSIvalFracs), CdataTID)</span>
<span id="cb55-23"><a href="whole-slide-imaging.html#cb55-23"></a>namesA=Cdata<span class="op">$</span>NewID[l2]</span>
<span id="cb55-24"><a href="whole-slide-imaging.html#cb55-24"></a>Treat=<span class="kw">factor</span>(Cdata<span class="op">$</span>Treatment[l2])</span>
<span id="cb55-25"><a href="whole-slide-imaging.html#cb55-25"></a>Growth=Cdata<span class="op">$</span>Growth2[l2]</span>
<span id="cb55-26"><a href="whole-slide-imaging.html#cb55-26"></a>WSIfracMelt=<span class="kw">melt</span>(WSIvalFracs)</span>
<span id="cb55-27"><a href="whole-slide-imaging.html#cb55-27"></a><span class="co">#WSIfracMelt$Var2=factor(WSIfracMelt$Var2, levels=ordV2)</span></span>
<span id="cb55-28"><a href="whole-slide-imaging.html#cb55-28"></a>WSIfracMelt<span class="op">$</span>treatment=Treat[<span class="kw">match</span>(WSIfracMelt<span class="op">$</span>Var2, <span class="kw">colnames</span>(WSIvalFracs))]</span>
<span id="cb55-29"><a href="whole-slide-imaging.html#cb55-29"></a>WSIfracMelt<span class="op">$</span>growth=Growth[<span class="kw">match</span>(WSIfracMelt<span class="op">$</span>Var2, <span class="kw">colnames</span>(WSIvalFracs))]</span>
<span id="cb55-30"><a href="whole-slide-imaging.html#cb55-30"></a>WSIfracMelt<span class="op">$</span>Name=namesA[<span class="kw">match</span>(WSIfracMelt<span class="op">$</span>Var2, <span class="kw">colnames</span>(WSIvalFracs))]</span>
<span id="cb55-31"><a href="whole-slide-imaging.html#cb55-31"></a>WSIfracMelt<span class="op">$</span>Name=<span class="kw">factor</span>(WSIfracMelt<span class="op">$</span>Name, <span class="dt">levels =</span> namesA)</span>
<span id="cb55-32"><a href="whole-slide-imaging.html#cb55-32"></a></span>
<span id="cb55-33"><a href="whole-slide-imaging.html#cb55-33"></a></span>
<span id="cb55-34"><a href="whole-slide-imaging.html#cb55-34"></a><span class="co">#pdf(sprintf(&quot;rslt/WSI-analysis/summary_distributions_norm_cell_count_%s.pdf&quot;, Sys.Date()), width=9, height=5)</span></span>
<span id="cb55-35"><a href="whole-slide-imaging.html#cb55-35"></a><span class="kw">ggplot</span>(WSImelt, <span class="kw">aes</span>(<span class="dt">x=</span>Name, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var1))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;ordered by CD8&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="whole-slide-imaging.html#cb56-1"></a><span class="kw">ggplot</span>(WSImelt, <span class="kw">aes</span>(<span class="dt">x=</span>Name, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var1))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>treatment, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;ordered by treatment&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-29-2.png" width="672" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="whole-slide-imaging.html#cb57-1"></a><span class="kw">ggplot</span>(WSImelt, <span class="kw">aes</span>(<span class="dt">x=</span>Name, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var1))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>growth, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;ordered by growth&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-29-3.png" width="672" /></p>
<p>Here, the same data is shown and normalised according to total cell count:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="whole-slide-imaging.html#cb58-1"></a><span class="kw">pdf</span>(<span class="st">&quot;figure-outputs/Fig4_WSI-normalised-all-samples.pdf&quot;</span>, <span class="dt">width=</span><span class="dv">8</span>, <span class="dt">height=</span><span class="dv">5</span>)</span>
<span id="cb58-2"><a href="whole-slide-imaging.html#cb58-2"></a><span class="kw">ggplot</span>(WSIfracMelt, <span class="kw">aes</span>(<span class="dt">x=</span>Name, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var1))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;ordered by CD8&quot;</span>)</span>
<span id="cb58-3"><a href="whole-slide-imaging.html#cb58-3"></a><span class="kw">dev.off</span>()</span>
<span id="cb58-4"><a href="whole-slide-imaging.html#cb58-4"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb58-5"><a href="whole-slide-imaging.html#cb58-5"></a><span class="co">##                 2</span></span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="whole-slide-imaging.html#cb59-1"></a><span class="kw">pdf</span>(<span class="st">&quot;figure-outputs/Fig4_WSI-normalised-all-samples-treatment.pdf&quot;</span>, <span class="dt">width=</span><span class="dv">8</span>, <span class="dt">height=</span><span class="dv">5</span>)</span>
<span id="cb59-2"><a href="whole-slide-imaging.html#cb59-2"></a><span class="kw">ggplot</span>(WSIfracMelt, <span class="kw">aes</span>(<span class="dt">x=</span>Name, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var1))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>treatment, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;ordered by treatment&quot;</span>)</span>
<span id="cb59-3"><a href="whole-slide-imaging.html#cb59-3"></a><span class="kw">dev.off</span>()</span>
<span id="cb59-4"><a href="whole-slide-imaging.html#cb59-4"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb59-5"><a href="whole-slide-imaging.html#cb59-5"></a><span class="co">##                 2</span></span></code></pre></div>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="whole-slide-imaging.html#cb60-1"></a><span class="kw">pdf</span>(<span class="st">&quot;figure-outputs/Fig4_WSI-normalised-all-samples-growth2.pdf&quot;</span>, <span class="dt">width=</span><span class="dv">8</span>, <span class="dt">height=</span><span class="dv">5</span>)</span>
<span id="cb60-2"><a href="whole-slide-imaging.html#cb60-2"></a><span class="kw">ggplot</span>(WSIfracMelt, <span class="kw">aes</span>(<span class="dt">x=</span>Name, <span class="dt">y=</span>value, <span class="dt">fill=</span>Var1))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>growth, <span class="dt">space=</span><span class="st">&quot;free_x&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;ordered by growth&quot;</span>)</span>
<span id="cb60-3"><a href="whole-slide-imaging.html#cb60-3"></a><span class="kw">dev.off</span>()</span>
<span id="cb60-4"><a href="whole-slide-imaging.html#cb60-4"></a><span class="co">## quartz_off_screen </span></span>
<span id="cb60-5"><a href="whole-slide-imaging.html#cb60-5"></a><span class="co">##                 2</span></span>
<span id="cb60-6"><a href="whole-slide-imaging.html#cb60-6"></a></span>
<span id="cb60-7"><a href="whole-slide-imaging.html#cb60-7"></a>DT<span class="op">::</span><span class="kw">datatable</span>(WSIfracMelt, <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb60-8"><a href="whole-slide-imaging.html#cb60-8"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure">
<div id="htmlwidget-950d2d831bd71b5db789" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-950d2d831bd71b5db789">{"x":{"filter":"none","extensions":["Buttons"],"data":[["CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass"],["4ND","4ND","4ND","4ND","4ND","4NC","4NC","4NC","4NC","4NC","10LC","10LC","10LC","10LC","10LC","14RD","14RD","14RD","14RD","14RD","15ND","15ND","15ND","15ND","15ND","15NC-D","15NC-D","15NC-D","15NC-D","15NC-D","15LB","15LB","15LB","15LB","15LB","7RB","7RB","7RB","7RB","7RB","16LA","16LA","16LA","16LA","16LA","8RCL","8RCL","8RCL","8RCL","8RCL","10RBL","10RBL","10RBL","10RBL","10RBL","7NB","7NB","7NB","7NB","7NB","12LD","12LD","12LD","12LD","12LD","11RD","11RD","11RD","11RD","11RD","10ND","10ND","10ND","10ND","10ND","10LB","10LB","10LB","10LB","10LB","3RC","3RC","3RC","3RC","3RC","4LB","4LB","4LB","4LB","4LB","5LB","5LB","5LB","5LB","5LB","3RB","3RB","3RB","3RB","3RB","4RB","4RB","4RB","4RB","4RB","8RCU","8RCU","8RCU","8RCU","8RCU","2N","2N","2N","2N","2N","16RD","16RD","16RD","16RD","16RD","16LD","16LD","16LD","16LD","16LD","7RA","7RA","7RA","7RA","7RA","7RD","7RD","7RD","7RD","7RD","15RD","15RD","15RD","15RD","15RD","5LD","5LD","5LD","5LD","5LD","13NA","13NA","13NA","13NA","13NA","17RB","17RB","17RB","17RB","17RB","10RC","10RC","10RC","10RC","10RC","5LA","5LA","5LA","5LA","5LA","10NA","10NA","10NA","10NA","10NA","17NA","17NA","17NA","17NA","17NA","16LC","16LC","16LC","16LC","16LC","7NC","7NC","7NC","7NC","7NC","11RC","11RC","11RC","11RC","11RC","5LC","5LC","5LC","5LC","5LC","3LA","3LA","3LA","3LA","3LA","14NC","14NC","14NC","14NC","14NC","6LDU","6LDU","6LDU","6LDU","6LDU","8LD","8LD","8LD","8LD","8LD","5RB","5RB","5RB","5RB","5RB","10RBU","10RBU","10RBU","10RBU","10RBU","10LD","10LD","10LD","10LD","10LD","2RD","2RD","2RD","2RD","2RD","16ND","16ND","16ND","16ND","16ND","6RC","6RC","6RC","6RC","6RC","6ND","6ND","6ND","6ND","6ND","2RA","2RA","2RA","2RA","2RA","14NB","14NB","14NB","14NB","14NB","12LC","12LC","12LC","12LC","12LC","17ND","17ND","17ND","17ND","17ND","11LB","11LB","11LB","11LB","11LB","6RB","6RB","6RB","6RB","6RB","3NB","3NB","3NB","3NB","3NB","11ND","11ND","11ND","11ND","11ND"],[0.00762451633757023,0.377689258178292,0,0.504436860068259,0.110249365415878,0.00813878305467377,0.723584571380561,0,0.176530861987719,0.0917457835770463,0.0100340714520434,0.296041139404701,0.167235485042575,0.308196749664186,0.218492554436495,0.0103370617114103,0.507712748414576,0.132525465733159,0.089090018186444,0.260334705954411,0.0104720537207619,0.401776130207943,0.0548253825607229,0.104215032826581,0.428711400683991,0.010787460286585,0.224532084116795,0,0.267973697291923,0.496706758304696,0.0129612204366809,0.215390990012716,0.0414452883412291,0.266750081135199,0.463452420074175,0.0169469113521434,0.203565927487804,0.525612994299805,0.119487518539035,0.134386648321212,0.0171528413159012,0.424868397270005,0,0.374246358068715,0.183732403345379,0.0176690744267719,0.553603622981529,0,0.248645879366917,0.180081423224782,0.0202714813177359,0.347853361851617,0.150545471925821,0.254764664412288,0.226565020492538,0.0206734110298608,0.26649275033072,0,0.346743567637522,0.366090271001897,0.0223692560965648,0.739762609411425,0.0110840920658259,0.0019855856535153,0.224798456772669,0.0224452418752018,0.265034819219336,0.149355669424051,0.282367270310208,0.280796999171204,0.0227194879526846,0.465147815290904,0,0.283720378932692,0.228412317823719,0.0228977401821137,0.266314906408934,0.120030410542321,0.393649789398256,0.197107153468375,0.0232139311385568,0.388552877169294,0.11600413009809,0.151344267503276,0.320884794090783,0.0244473882371464,0.349872271946222,0.105386775000532,0.149662177203559,0.37063138761254,0.0257829624400936,0.470588036624371,0,0,0.503629000935536,0.0266553579642389,0.436608192400455,0,0.380376399695298,0.156360049940009,0.0280096449579857,0.30318040917295,0.164444034127445,0.224308970851053,0.280056940890566,0.0287668011746126,0.257857507750422,0,0.507401833149184,0.205973857925782,0.0301040967138085,0.313685928213308,0.092148273941956,0.196137408430651,0.367924292700277,0.0306485937592836,0.242856718465585,0,0.503444709110452,0.223049978664679,0.0316167665758732,0.354175562795616,0,0.411457244977132,0.202750425651378,0.0319309795486363,0.232309502529544,0.193193417623739,0.338371256425673,0.204194843872407,0.0321690077775606,0.255333197983413,0,0.560654157551102,0.151843636687924,0.0328312348073089,0.149980881107803,0,0,0.817187884084888,0.0348107762848735,0.237492099743837,0.00956350329898595,0.0727252965195933,0.64540832415271,0.0355810679060535,0.338458463882884,0,0.35738956298038,0.268570905230682,0.0411073596330847,0.660611956857736,0.12023329986557,0.0539738376342205,0.124073546009389,0.0417305711342566,0.361817557514021,0.027675403456564,0.131143413070848,0.43763305482431,0.0453951787168959,0.348773829788185,0,0.477254010165589,0.12857698132933,0.046562711539908,0.555444823554349,0,0.285519900990776,0.112472563914967,0.0472687015058753,0.182683975918826,0,0.515150256476267,0.254897066099032,0.0476926119861338,0.477450786479425,0,0.239858006891454,0.234998594642987,0.0479456512895615,0.140550899935816,0,0.735945452408718,0.0755579963659046,0.050418603715108,0.311258528044436,0.173017418342088,0.208405029282135,0.256900420616233,0.0515054602700039,0.2846329724021,0,0.459279482481496,0.2045820848464,0.0531246414587471,0.395391528632906,0.147560841875463,0.0995966589184779,0.304326329114405,0.0541297977067744,0.546049449455146,0.126293377338986,0.0814721028821806,0.192055272616914,0.0556252700006171,0.267647665953286,0,0.647133932365244,0.0295931316808527,0.0562336342953625,0.593615407729621,0.00185299125731538,0.00363443845064175,0.34466352826706,0.0596672294268477,0.343537541709383,0,0.462561160757366,0.134234068106403,0.0670640985529038,0.163958472607062,0.335904973488285,0.286529107365132,0.146543347986618,0.0681087685159526,0.194893106594036,0.206384476435713,0.406670294171786,0.123943354282513,0.0742194780431157,0.265962886676766,0.122200132854801,0.182066035526712,0.355551466898605,0.0753308198822148,0.375935710177509,0.132690974538186,0.312897875011208,0.103144620390882,0.0788010213529223,0.247887779298085,0.119228740414807,0.391436209983915,0.162646248950271,0.0815901125430434,0.469847319830979,0.107080733317169,0.25345808938465,0.0880237449241583,0.0858191749091156,0.430418479290512,0.00169078930906413,0.0391991449539054,0.442872411537403,0.106152004735647,0.368885019612605,0.063993035302689,0.181603615050858,0.2793663252982,0.107059057466693,0.581189103201432,0.0114933386359117,0.00497116723006562,0.295287333465898,0.111526720118085,0.5278355530209,0.00170250595457748,0.0111711228152571,0.347764098091181,0.152177738773681,0.455467236429688,1.00783553364504e-05,3.64371308317822e-05,0.392308509310462,0.155480427254777,0.177143939704656,0,0.483032657373934,0.184342975666633,0.276829917860447,0.473121022679488,0,0.0426761234616355,0.20737293599843,0.380067171051444,0.423177624094906,0.018544266084036,0.014002997417887,0.164207941351727],["Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","PDL1","PDL1","PDL1","PDL1","PDL1","LY","LY","LY","LY","LY","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","LY","LY","LY","LY","LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","LY","LY","LY","LY","LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1"],["growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing",null,null,null,null,null,"growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing",null,null,null,null,null,"stable","stable","stable","stable","stable",null,null,null,null,null,"growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing",null,null,null,null,null,"stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing",null,null,null,null,null,"stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable"],["P10T2","P10T2","P10T2","P10T2","P10T2","P10T1","P10T1","P10T1","P10T1","P10T1","P24T2","P24T2","P24T2","P24T2","P24T2","P32T2","P32T2","P32T2","P32T2","P32T2","P34T3","P34T3","P34T3","P34T3","P34T3","P34T2","P34T2","P34T2","P34T2","P34T2","P33T1","P33T1","P33T1","P33T1","P33T1","P20T2","P20T2","P20T2","P20T2","P20T2","P36T1","P36T1","P36T1","P36T1","P36T1","P22T1","P22T1","P22T1","P22T1","P22T1","P26T1","P26T1","P26T1","P26T1","P26T1","P19T2","P19T2","P19T2","P19T2","P19T2","P30T2","P30T2","P30T2","P30T2","P30T2","P29T2","P29T2","P29T2","P29T2","P29T2","P25T2","P25T2","P25T2","P25T2","P25T2","P24T1","P24T1","P24T1","P24T1","P24T1","P8T2","P8T2","P8T2","P8T2","P8T2","P9T1","P9T1","P9T1","P9T1","P9T1","P12T2","P12T2","P12T2","P12T2","P12T2","P8T1","P8T1","P8T1","P8T1","P8T1","P11T1","P11T1","P11T1","P11T1","P11T1","P22T2","P22T2","P22T2","P22T2","P22T2","P4T1","P4T1","P4T1","P4T1","P4T1","P38T2","P38T2","P38T2","P38T2","P38T2","P36T3","P36T3","P36T3","P36T3","P36T3","P20T1","P20T1","P20T1","P20T1","P20T1","P20T3","P20T3","P20T3","P20T3","P20T3","P35T1","P35T1","P35T1","P35T1","P35T1","P12T4","P12T4","P12T4","P12T4","P12T4","P41T1","P41T1","P41T1","P41T1","P41T1","P40T1","P40T1","P40T1","P40T1","P40T1","P26T3","P26T3","P26T3","P26T3","P26T3","P12T1","P12T1","P12T1","P12T1","P12T1","P25T1","P25T1","P25T1","P25T1","P25T1","P39T1","P39T1","P39T1","P39T1","P39T1","P36T2","P36T2","P36T2","P36T2","P36T2","P19T3","P19T3","P19T3","P19T3","P19T3","P29T1","P29T1","P29T1","P29T1","P29T1","P12T3","P12T3","P12T3","P12T3","P12T3","P6T1","P6T1","P6T1","P6T1","P6T1","P31T2","P31T2","P31T2","P31T2","P31T2","P15T2","P15T2","P15T2","P15T2","P15T2","P21T1","P21T1","P21T1","P21T1","P21T1","P14T1","P14T1","P14T1","P14T1","P14T1","P26T2","P26T2","P26T2","P26T2","P26T2","P24T3","P24T3","P24T3","P24T3","P24T3","P5T3","P5T3","P5T3","P5T3","P5T3","P37T1","P37T1","P37T1","P37T1","P37T1","P17T2","P17T2","P17T2","P17T2","P17T2","P16T1","P16T1","P16T1","P16T1","P16T1","P5T1","P5T1","P5T1","P5T1","P5T1","P31T1","P31T1","P31T1","P31T1","P31T1","P30T1","P30T1","P30T1","P30T1","P30T1","P39T3","P39T3","P39T3","P39T3","P39T3","P27T1","P27T1","P27T1","P27T1","P27T1","P17T1","P17T1","P17T1","P17T1","P17T1","P7T1","P7T1","P7T1","P7T1","P7T1","P28T1","P28T1","P28T1","P28T1","P28T1"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>Var1<\/th>\n      <th>Var2<\/th>\n      <th>value<\/th>\n      <th>treatment<\/th>\n      <th>growth<\/th>\n      <th>Name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
(#fig:Ext4c_pt2)association with growth
</p>
</div>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="whole-slide-imaging.html#cb61-1"></a></span>
<span id="cb61-2"><a href="whole-slide-imaging.html#cb61-2"></a><span class="co">#write.csv(WSIfracMelt, file=&quot;nature-tables/Ext4c-d.csv&quot;)</span></span></code></pre></div>
</div>
<div id="associate-composition-with-other-covariates" class="section level2">
<h2><span class="header-section-number">4.3</span> Associate composition with other covariates</h2>
<p>Make Boxplots of the above data, and index samples according to growth or with treatment</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="whole-slide-imaging.html#cb62-1"></a><span class="co">#pdf(&quot;~/Desktop/Fig4B-summary-WSI-growth-treatment.pdf&quot;, height=7, width=7)</span></span>
<span id="cb62-2"><a href="whole-slide-imaging.html#cb62-2"></a></span>
<span id="cb62-3"><a href="whole-slide-imaging.html#cb62-3"></a>nx1=<span class="kw">levels</span>(WSIfracMelt<span class="op">$</span>Var1)</span>
<span id="cb62-4"><a href="whole-slide-imaging.html#cb62-4"></a></span>
<span id="cb62-5"><a href="whole-slide-imaging.html#cb62-5"></a>nOut=<span class="kw">sapply</span>(nx1, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(WSIfracMelt<span class="op">$</span>value[WSIfracMelt<span class="op">$</span>growth<span class="op">==</span><span class="st">&quot;growing&quot;</span> <span class="op">&amp;</span><span class="st"> </span>WSIfracMelt<span class="op">$</span>Var1<span class="op">==</span>x],WSIfracMelt<span class="op">$</span>value[WSIfracMelt<span class="op">$</span>growth<span class="op">==</span><span class="st">&quot;stable&quot;</span> <span class="op">&amp;</span><span class="st"> </span>WSIfracMelt<span class="op">$</span>Var1<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb62-6"><a href="whole-slide-imaging.html#cb62-6"></a></span>
<span id="cb62-7"><a href="whole-slide-imaging.html#cb62-7"></a><span class="kw">ggplot</span>(WSIfracMelt[<span class="op">-</span><span class="kw">which</span>(<span class="kw">is.na</span>(WSIfracMelt<span class="op">$</span>growth)), ], <span class="kw">aes</span>(<span class="dt">x=</span>Var1, <span class="dt">y=</span>value, <span class="dt">col=</span><span class="kw">factor</span>(growth)))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>(<span class="dt">position=</span><span class="kw">position_jitterdodge</span>(),<span class="dt">alpha=</span><span class="fl">0.3</span>)<span class="op">+</span></span>
<span id="cb62-8"><a href="whole-slide-imaging.html#cb62-8"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(ColSize, <span class="st">&quot;black&quot;</span>))<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;ordered by CD8&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="kw">names</span>(nOut), <span class="kw">round</span>(nOut,<span class="dv">2</span>), <span class="dt">collapse=</span><span class="st">&quot; ;&quot;</span>))</span></code></pre></div>
<div class="figure"><span id="fig:4b"></span>
<img src="RatDCIS-bookdown_files/figure-html/4b-1.png" alt="association of frequency with treatment and growth" width="672" />
<p class="caption">
Figure 4.2: association of frequency with treatment and growth
</p>
</div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="whole-slide-imaging.html#cb63-1"></a>nOutA=<span class="kw">sapply</span>(nx1, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(WSIfracMelt<span class="op">$</span>value[WSIfracMelt<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span> <span class="op">&amp;</span><span class="st"> </span>WSIfracMelt<span class="op">$</span>Var1<span class="op">==</span>x],WSIfracMelt<span class="op">$</span>value[WSIfracMelt<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;LY&quot;</span> <span class="op">&amp;</span><span class="st"> </span>WSIfracMelt<span class="op">$</span>Var1<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb63-2"><a href="whole-slide-imaging.html#cb63-2"></a></span>
<span id="cb63-3"><a href="whole-slide-imaging.html#cb63-3"></a>nOutB=<span class="kw">sapply</span>(nx1, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(WSIfracMelt<span class="op">$</span>value[WSIfracMelt<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span> <span class="op">&amp;</span><span class="st"> </span>WSIfracMelt<span class="op">$</span>Var1<span class="op">==</span>x],WSIfracMelt<span class="op">$</span>value[WSIfracMelt<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;PDL1&quot;</span> <span class="op">&amp;</span><span class="st"> </span>WSIfracMelt<span class="op">$</span>Var1<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb63-4"><a href="whole-slide-imaging.html#cb63-4"></a></span>
<span id="cb63-5"><a href="whole-slide-imaging.html#cb63-5"></a>nOutC=<span class="kw">sapply</span>(nx1, <span class="cf">function</span>(x) <span class="kw">wilcox.test</span>(WSIfracMelt<span class="op">$</span>value[WSIfracMelt<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;Vehicle&quot;</span> <span class="op">&amp;</span><span class="st"> </span>WSIfracMelt<span class="op">$</span>Var1<span class="op">==</span>x],WSIfracMelt<span class="op">$</span>value[WSIfracMelt<span class="op">$</span>treatment<span class="op">==</span><span class="st">&quot;PDL1+LY&quot;</span> <span class="op">&amp;</span><span class="st"> </span>WSIfracMelt<span class="op">$</span>Var1<span class="op">==</span>x])<span class="op">$</span>p.value)</span>
<span id="cb63-6"><a href="whole-slide-imaging.html#cb63-6"></a></span>
<span id="cb63-7"><a href="whole-slide-imaging.html#cb63-7"></a></span>
<span id="cb63-8"><a href="whole-slide-imaging.html#cb63-8"></a><span class="kw">ggplot</span>(WSIfracMelt, <span class="kw">aes</span>(<span class="dt">x=</span>Var1, <span class="dt">y=</span>value, <span class="dt">col=</span>(treatment)))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">geom_point</span>(<span class="dt">position=</span><span class="kw">position_jitterdodge</span>(),<span class="dt">alpha=</span><span class="fl">0.3</span>)<span class="op">+</span></span>
<span id="cb63-9"><a href="whole-slide-imaging.html#cb63-9"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>ColMerge[ ,<span class="dv">1</span>])<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;ordered by CD8&quot;</span>)<span class="op">+</span><span class="kw">theme_bw</span>()<span class="op">+</span></span>
<span id="cb63-10"><a href="whole-slide-imaging.html#cb63-10"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="kw">paste</span>(<span class="kw">names</span>(nOutA), <span class="kw">round</span>(nOutA,<span class="dv">2</span>), <span class="dt">collapse=</span><span class="st">&quot; ;&quot;</span>),<span class="kw">paste</span>(<span class="kw">names</span>(nOutB), <span class="kw">round</span>(nOutB,<span class="dv">2</span>), <span class="dt">collapse=</span><span class="st">&quot; ;&quot;</span>),<span class="kw">paste</span>(<span class="kw">names</span>(nOutC), <span class="kw">round</span>(nOutC,<span class="dv">2</span>), <span class="dt">collapse=</span><span class="st">&quot; ;&quot;</span>) ) )</span></code></pre></div>
<div class="figure"><span id="fig:4b2"></span>
<img src="RatDCIS-bookdown_files/figure-html/4b2-1.png" alt="association with treatment" width="672" />
<p class="caption">
Figure 4.3: association with treatment
</p>
</div>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="whole-slide-imaging.html#cb64-1"></a></span>
<span id="cb64-2"><a href="whole-slide-imaging.html#cb64-2"></a><span class="co">#dev.off()</span></span>
<span id="cb64-3"><a href="whole-slide-imaging.html#cb64-3"></a>DT<span class="op">::</span><span class="kw">datatable</span>(WSIfracMelt[<span class="op">-</span><span class="kw">which</span>(<span class="kw">is.na</span>(WSIfracMelt<span class="op">$</span>growth)), ], <span class="dt">rownames=</span>F, <span class="dt">class=</span><span class="st">&#39;cell-border stripe&#39;</span>,</span>
<span id="cb64-4"><a href="whole-slide-imaging.html#cb64-4"></a>          <span class="dt">extensions=</span><span class="st">&quot;Buttons&quot;</span>, <span class="dt">options=</span><span class="kw">list</span>(<span class="dt">dom=</span><span class="st">&quot;Bfrtip&quot;</span>, <span class="dt">buttons=</span><span class="kw">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:4b2"></span>
<div id="htmlwidget-1fa37557c6a2861fd08c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1fa37557c6a2861fd08c">{"x":{"filter":"none","extensions":["Buttons"],"data":[["CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass","CD8","EpCAM","EpCAM: SMA","SMA","Unclass"],["4ND","4ND","4ND","4ND","4ND","4NC","4NC","4NC","4NC","4NC","10LC","10LC","10LC","10LC","10LC","14RD","14RD","14RD","14RD","14RD","15ND","15ND","15ND","15ND","15ND","15NC-D","15NC-D","15NC-D","15NC-D","15NC-D","15LB","15LB","15LB","15LB","15LB","7RB","7RB","7RB","7RB","7RB","8RCL","8RCL","8RCL","8RCL","8RCL","10RBL","10RBL","10RBL","10RBL","10RBL","7NB","7NB","7NB","7NB","7NB","12LD","12LD","12LD","12LD","12LD","11RD","11RD","11RD","11RD","11RD","10ND","10ND","10ND","10ND","10ND","10LB","10LB","10LB","10LB","10LB","3RC","3RC","3RC","3RC","3RC","4LB","4LB","4LB","4LB","4LB","5LB","5LB","5LB","5LB","5LB","3RB","3RB","3RB","3RB","3RB","4RB","4RB","4RB","4RB","4RB","8RCU","8RCU","8RCU","8RCU","8RCU","2N","2N","2N","2N","2N","16RD","16RD","16RD","16RD","16RD","16LD","16LD","16LD","16LD","16LD","7RA","7RA","7RA","7RA","7RA","7RD","7RD","7RD","7RD","7RD","15RD","15RD","15RD","15RD","15RD","5LD","5LD","5LD","5LD","5LD","13NA","13NA","13NA","13NA","13NA","17RB","17RB","17RB","17RB","17RB","5LA","5LA","5LA","5LA","5LA","17NA","17NA","17NA","17NA","17NA","16LC","16LC","16LC","16LC","16LC","7NC","7NC","7NC","7NC","7NC","11RC","11RC","11RC","11RC","11RC","5LC","5LC","5LC","5LC","5LC","3LA","3LA","3LA","3LA","3LA","14NC","14NC","14NC","14NC","14NC","6LDU","6LDU","6LDU","6LDU","6LDU","8LD","8LD","8LD","8LD","8LD","5RB","5RB","5RB","5RB","5RB","10LD","10LD","10LD","10LD","10LD","2RD","2RD","2RD","2RD","2RD","16ND","16ND","16ND","16ND","16ND","6RC","6RC","6RC","6RC","6RC","6ND","6ND","6ND","6ND","6ND","2RA","2RA","2RA","2RA","2RA","14NB","14NB","14NB","14NB","14NB","17ND","17ND","17ND","17ND","17ND","11LB","11LB","11LB","11LB","11LB","6RB","6RB","6RB","6RB","6RB","3NB","3NB","3NB","3NB","3NB","11ND","11ND","11ND","11ND","11ND"],[0.00762451633757023,0.377689258178292,0,0.504436860068259,0.110249365415878,0.00813878305467377,0.723584571380561,0,0.176530861987719,0.0917457835770463,0.0100340714520434,0.296041139404701,0.167235485042575,0.308196749664186,0.218492554436495,0.0103370617114103,0.507712748414576,0.132525465733159,0.089090018186444,0.260334705954411,0.0104720537207619,0.401776130207943,0.0548253825607229,0.104215032826581,0.428711400683991,0.010787460286585,0.224532084116795,0,0.267973697291923,0.496706758304696,0.0129612204366809,0.215390990012716,0.0414452883412291,0.266750081135199,0.463452420074175,0.0169469113521434,0.203565927487804,0.525612994299805,0.119487518539035,0.134386648321212,0.0176690744267719,0.553603622981529,0,0.248645879366917,0.180081423224782,0.0202714813177359,0.347853361851617,0.150545471925821,0.254764664412288,0.226565020492538,0.0206734110298608,0.26649275033072,0,0.346743567637522,0.366090271001897,0.0223692560965648,0.739762609411425,0.0110840920658259,0.0019855856535153,0.224798456772669,0.0224452418752018,0.265034819219336,0.149355669424051,0.282367270310208,0.280796999171204,0.0227194879526846,0.465147815290904,0,0.283720378932692,0.228412317823719,0.0228977401821137,0.266314906408934,0.120030410542321,0.393649789398256,0.197107153468375,0.0232139311385568,0.388552877169294,0.11600413009809,0.151344267503276,0.320884794090783,0.0244473882371464,0.349872271946222,0.105386775000532,0.149662177203559,0.37063138761254,0.0257829624400936,0.470588036624371,0,0,0.503629000935536,0.0266553579642389,0.436608192400455,0,0.380376399695298,0.156360049940009,0.0280096449579857,0.30318040917295,0.164444034127445,0.224308970851053,0.280056940890566,0.0287668011746126,0.257857507750422,0,0.507401833149184,0.205973857925782,0.0301040967138085,0.313685928213308,0.092148273941956,0.196137408430651,0.367924292700277,0.0306485937592836,0.242856718465585,0,0.503444709110452,0.223049978664679,0.0316167665758732,0.354175562795616,0,0.411457244977132,0.202750425651378,0.0319309795486363,0.232309502529544,0.193193417623739,0.338371256425673,0.204194843872407,0.0321690077775606,0.255333197983413,0,0.560654157551102,0.151843636687924,0.0328312348073089,0.149980881107803,0,0,0.817187884084888,0.0348107762848735,0.237492099743837,0.00956350329898595,0.0727252965195933,0.64540832415271,0.0355810679060535,0.338458463882884,0,0.35738956298038,0.268570905230682,0.0411073596330847,0.660611956857736,0.12023329986557,0.0539738376342205,0.124073546009389,0.0453951787168959,0.348773829788185,0,0.477254010165589,0.12857698132933,0.0472687015058753,0.182683975918826,0,0.515150256476267,0.254897066099032,0.0476926119861338,0.477450786479425,0,0.239858006891454,0.234998594642987,0.0479456512895615,0.140550899935816,0,0.735945452408718,0.0755579963659046,0.050418603715108,0.311258528044436,0.173017418342088,0.208405029282135,0.256900420616233,0.0515054602700039,0.2846329724021,0,0.459279482481496,0.2045820848464,0.0531246414587471,0.395391528632906,0.147560841875463,0.0995966589184779,0.304326329114405,0.0541297977067744,0.546049449455146,0.126293377338986,0.0814721028821806,0.192055272616914,0.0556252700006171,0.267647665953286,0,0.647133932365244,0.0295931316808527,0.0562336342953625,0.593615407729621,0.00185299125731538,0.00363443845064175,0.34466352826706,0.0596672294268477,0.343537541709383,0,0.462561160757366,0.134234068106403,0.0681087685159526,0.194893106594036,0.206384476435713,0.406670294171786,0.123943354282513,0.0742194780431157,0.265962886676766,0.122200132854801,0.182066035526712,0.355551466898605,0.0753308198822148,0.375935710177509,0.132690974538186,0.312897875011208,0.103144620390882,0.0788010213529223,0.247887779298085,0.119228740414807,0.391436209983915,0.162646248950271,0.0815901125430434,0.469847319830979,0.107080733317169,0.25345808938465,0.0880237449241583,0.0858191749091156,0.430418479290512,0.00169078930906413,0.0391991449539054,0.442872411537403,0.106152004735647,0.368885019612605,0.063993035302689,0.181603615050858,0.2793663252982,0.111526720118085,0.5278355530209,0.00170250595457748,0.0111711228152571,0.347764098091181,0.152177738773681,0.455467236429688,1.00783553364504e-05,3.64371308317822e-05,0.392308509310462,0.155480427254777,0.177143939704656,0,0.483032657373934,0.184342975666633,0.276829917860447,0.473121022679488,0,0.0426761234616355,0.20737293599843,0.380067171051444,0.423177624094906,0.018544266084036,0.014002997417887,0.164207941351727],["Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","PDL1","PDL1","PDL1","PDL1","PDL1","LY","LY","LY","LY","LY","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","LY","LY","LY","LY","LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","LY","LY","LY","LY","LY","LY","LY","LY","LY","LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","PDL1+LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1","LY","LY","LY","LY","LY","Vehicle","Vehicle","Vehicle","Vehicle","Vehicle","PDL1","PDL1","PDL1","PDL1","PDL1"],["growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","growing","growing","growing","growing","growing","growing","growing","growing","growing","growing","stable","stable","stable","stable","stable","stable","stable","stable","stable","stable"],["P10T2","P10T2","P10T2","P10T2","P10T2","P10T1","P10T1","P10T1","P10T1","P10T1","P24T2","P24T2","P24T2","P24T2","P24T2","P32T2","P32T2","P32T2","P32T2","P32T2","P34T3","P34T3","P34T3","P34T3","P34T3","P34T2","P34T2","P34T2","P34T2","P34T2","P33T1","P33T1","P33T1","P33T1","P33T1","P20T2","P20T2","P20T2","P20T2","P20T2","P22T1","P22T1","P22T1","P22T1","P22T1","P26T1","P26T1","P26T1","P26T1","P26T1","P19T2","P19T2","P19T2","P19T2","P19T2","P30T2","P30T2","P30T2","P30T2","P30T2","P29T2","P29T2","P29T2","P29T2","P29T2","P25T2","P25T2","P25T2","P25T2","P25T2","P24T1","P24T1","P24T1","P24T1","P24T1","P8T2","P8T2","P8T2","P8T2","P8T2","P9T1","P9T1","P9T1","P9T1","P9T1","P12T2","P12T2","P12T2","P12T2","P12T2","P8T1","P8T1","P8T1","P8T1","P8T1","P11T1","P11T1","P11T1","P11T1","P11T1","P22T2","P22T2","P22T2","P22T2","P22T2","P4T1","P4T1","P4T1","P4T1","P4T1","P38T2","P38T2","P38T2","P38T2","P38T2","P36T3","P36T3","P36T3","P36T3","P36T3","P20T1","P20T1","P20T1","P20T1","P20T1","P20T3","P20T3","P20T3","P20T3","P20T3","P35T1","P35T1","P35T1","P35T1","P35T1","P12T4","P12T4","P12T4","P12T4","P12T4","P41T1","P41T1","P41T1","P41T1","P41T1","P40T1","P40T1","P40T1","P40T1","P40T1","P12T1","P12T1","P12T1","P12T1","P12T1","P39T1","P39T1","P39T1","P39T1","P39T1","P36T2","P36T2","P36T2","P36T2","P36T2","P19T3","P19T3","P19T3","P19T3","P19T3","P29T1","P29T1","P29T1","P29T1","P29T1","P12T3","P12T3","P12T3","P12T3","P12T3","P6T1","P6T1","P6T1","P6T1","P6T1","P31T2","P31T2","P31T2","P31T2","P31T2","P15T2","P15T2","P15T2","P15T2","P15T2","P21T1","P21T1","P21T1","P21T1","P21T1","P14T1","P14T1","P14T1","P14T1","P14T1","P24T3","P24T3","P24T3","P24T3","P24T3","P5T3","P5T3","P5T3","P5T3","P5T3","P37T1","P37T1","P37T1","P37T1","P37T1","P17T2","P17T2","P17T2","P17T2","P17T2","P16T1","P16T1","P16T1","P16T1","P16T1","P5T1","P5T1","P5T1","P5T1","P5T1","P31T1","P31T1","P31T1","P31T1","P31T1","P39T3","P39T3","P39T3","P39T3","P39T3","P27T1","P27T1","P27T1","P27T1","P27T1","P17T1","P17T1","P17T1","P17T1","P17T1","P7T1","P7T1","P7T1","P7T1","P7T1","P28T1","P28T1","P28T1","P28T1","P28T1"]],"container":"<table class=\"cell-border stripe\">\n  <thead>\n    <tr>\n      <th>Var1<\/th>\n      <th>Var2<\/th>\n      <th>value<\/th>\n      <th>treatment<\/th>\n      <th>growth<\/th>\n      <th>Name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 4.3: association with treatment
</p>
</div>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="whole-slide-imaging.html#cb65-1"></a></span>
<span id="cb65-2"><a href="whole-slide-imaging.html#cb65-2"></a><span class="co">#write.csv(WSIfracMelt[-which(is.na(WSIfracMelt$growth)), ], file=&quot;nature-tables/4b.csv&quot;)</span></span></code></pre></div>
<p>P values when comparing to the vehicle for each comparison:</p>
<table>
<thead>
<tr class="header">
<th>Treatment</th>
<th>values</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LY</td>
<td>0.8630663, 0.0323942, 0.9495167, 0.0437861, 0.04824</td>
</tr>
<tr class="even">
<td>PDL1</td>
<td>0.9624018, 0.6258328, 0.0255031, 0.9373764, 0.3049106</td>
</tr>
<tr class="odd">
<td>PDL1+LY</td>
<td>0.2307253, 1, 1, 0.9298782, 0.4479729</td>
</tr>
</tbody>
</table>
</div>
<div id="estimate-tumor-size" class="section level2">
<h2><span class="header-section-number">4.4</span> Estimate tumor size</h2>
<p>Using WSI data, we can estimate a tumor size for each tissue sample and compare to the final tumor sizes. This will be based on the distribution of EpCAM+ cells.
This estimate is can be used to normalise CD8 counts.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="whole-slide-imaging.html#cb66-1"></a><span class="co">#estimate the tumor areas</span></span>
<span id="cb66-2"><a href="whole-slide-imaging.html#cb66-2"></a>Tarea=<span class="kw">lapply</span>(WSIsummary, <span class="cf">function</span>(x) <span class="kw">ripras</span>(x<span class="op">$</span>Centroid.X.µm[x<span class="op">$</span>Class2<span class="op">==</span><span class="st">&quot;EpCAM&quot;</span>], x<span class="op">$</span>Centroid.Y.µm[x<span class="op">$</span>Class2<span class="op">==</span><span class="st">&quot;EpCAM&quot;</span>], <span class="st">&quot;convex&quot;</span>))</span>
<span id="cb66-3"><a href="whole-slide-imaging.html#cb66-3"></a>TareaSum=<span class="kw">sapply</span>(Tarea, area)</span>
<span id="cb66-4"><a href="whole-slide-imaging.html#cb66-4"></a></span>
<span id="cb66-5"><a href="whole-slide-imaging.html#cb66-5"></a><span class="co"># Tumor diameter</span></span>
<span id="cb66-6"><a href="whole-slide-imaging.html#cb66-6"></a>Tdiameter=Cdata<span class="op">$</span>Tumor.diameter.sac.mm[<span class="kw">match</span>(<span class="kw">colnames</span>(WSIvals), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))]</span>
<span id="cb66-7"><a href="whole-slide-imaging.html#cb66-7"></a></span>
<span id="cb66-8"><a href="whole-slide-imaging.html#cb66-8"></a><span class="kw">plot</span>(TareaSum, Tdiameter, <span class="dt">ylab=</span><span class="st">&quot;Tumor Diameter&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Epcam cells bounding area&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="whole-slide-imaging.html#cb67-1"></a></span>
<span id="cb67-2"><a href="whole-slide-imaging.html#cb67-2"></a><span class="kw">cor.test</span>(TareaSum, Tdiameter, <span class="dt">use=</span><span class="st">&quot;complete&quot;</span>)</span>
<span id="cb67-3"><a href="whole-slide-imaging.html#cb67-3"></a><span class="co">## </span></span>
<span id="cb67-4"><a href="whole-slide-imaging.html#cb67-4"></a><span class="co">##  Pearson&#39;s product-moment correlation</span></span>
<span id="cb67-5"><a href="whole-slide-imaging.html#cb67-5"></a><span class="co">## </span></span>
<span id="cb67-6"><a href="whole-slide-imaging.html#cb67-6"></a><span class="co">## data:  TareaSum and Tdiameter</span></span>
<span id="cb67-7"><a href="whole-slide-imaging.html#cb67-7"></a><span class="co">## t = 1.8547, df = 56, p-value = 0.06891</span></span>
<span id="cb67-8"><a href="whole-slide-imaging.html#cb67-8"></a><span class="co">## alternative hypothesis: true correlation is not equal to 0</span></span>
<span id="cb67-9"><a href="whole-slide-imaging.html#cb67-9"></a><span class="co">## 95 percent confidence interval:</span></span>
<span id="cb67-10"><a href="whole-slide-imaging.html#cb67-10"></a><span class="co">##  -0.01890939  0.46967349</span></span>
<span id="cb67-11"><a href="whole-slide-imaging.html#cb67-11"></a><span class="co">## sample estimates:</span></span>
<span id="cb67-12"><a href="whole-slide-imaging.html#cb67-12"></a><span class="co">##       cor </span></span>
<span id="cb67-13"><a href="whole-slide-imaging.html#cb67-13"></a><span class="co">## 0.2405613</span></span>
<span id="cb67-14"><a href="whole-slide-imaging.html#cb67-14"></a></span>
<span id="cb67-15"><a href="whole-slide-imaging.html#cb67-15"></a><span class="co">## append to Cdata</span></span>
<span id="cb67-16"><a href="whole-slide-imaging.html#cb67-16"></a>Cdata<span class="op">$</span>TumorAreaWSI=<span class="ot">NA</span></span>
<span id="cb67-17"><a href="whole-slide-imaging.html#cb67-17"></a>lxmatch=<span class="kw">match</span>(<span class="kw">colnames</span>(WSIvalFracs), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID)) <span class="co"># probably don&#39;t need this</span></span>
<span id="cb67-18"><a href="whole-slide-imaging.html#cb67-18"></a>Cdata<span class="op">$</span>TumorAreaWSI[(lxmatch)]=TareaSum <span class="co">#[-which(is.na(lxmatch))]</span></span></code></pre></div>
</div>
<div id="correlations-between-different-subpopulations" class="section level2">
<h2><span class="header-section-number">4.5</span> Correlations between different subpopulations</h2>
<p>Look for correlates between different subpopulations: Naturally, we would expect a negative correlation since this should sum to 1. Below are heatmaps showing correlations between different cell types, and significant associations are linearly shown.</p>
<p>Note the following negative correlations:</p>
<ul>
<li>epcam and SMA</li>
<li>SMA+ and Unclass</li>
</ul>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="whole-slide-imaging.html#cb68-1"></a>Ax1=<span class="kw">cor</span>(<span class="kw">t</span>(WSIvalFracs))</span>
<span id="cb68-2"><a href="whole-slide-imaging.html#cb68-2"></a></span>
<span id="cb68-3"><a href="whole-slide-imaging.html#cb68-3"></a><span class="co">#pdf(sprintf(&quot;rslt/WSI-analysis/summary-associations-between-subgroups_%s.pdf&quot;, Sys.Date()), height=8, width=8)</span></span>
<span id="cb68-4"><a href="whole-slide-imaging.html#cb68-4"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">0</span>,<span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb68-5"><a href="whole-slide-imaging.html#cb68-5"></a><span class="kw">heatmap.2</span>(Ax1, <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;RdBu&quot;</span>), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>,<span class="dt">scale=</span><span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-31-1.png" width="672" /></p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="whole-slide-imaging.html#cb69-1"></a></span>
<span id="cb69-2"><a href="whole-slide-imaging.html#cb69-2"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb69-3"><a href="whole-slide-imaging.html#cb69-3"></a></span>
<span id="cb69-4"><a href="whole-slide-imaging.html#cb69-4"></a>a1=<span class="kw">cor.test</span>(WSIvalFracs[<span class="dv">2</span>, ], WSIvalFracs[<span class="dv">4</span>, ])</span>
<span id="cb69-5"><a href="whole-slide-imaging.html#cb69-5"></a><span class="kw">plot</span>(WSIvalFracs[<span class="dv">2</span>, ], WSIvalFracs[<span class="dv">4</span>, ], <span class="dt">xlab=</span><span class="st">&quot;Epcam+&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;SMA+&quot;</span>, <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;rho=%s,p=%s&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(a1<span class="op">$</span>p.value, <span class="dv">2</span>)))</span>
<span id="cb69-6"><a href="whole-slide-imaging.html#cb69-6"></a></span>
<span id="cb69-7"><a href="whole-slide-imaging.html#cb69-7"></a>a1=<span class="kw">cor.test</span>(WSIvalFracs[<span class="dv">4</span>, ], WSIvalFracs[<span class="dv">5</span>, ])</span>
<span id="cb69-8"><a href="whole-slide-imaging.html#cb69-8"></a><span class="kw">plot</span>(WSIvalFracs[<span class="dv">4</span>, ], WSIvalFracs[<span class="dv">5</span>, ], <span class="dt">xlab=</span><span class="st">&quot;SMA+&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Unclass&quot;</span>, <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;rho=%s,p=%s&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(a1<span class="op">$</span>p.value, <span class="dv">2</span>)))</span>
<span id="cb69-9"><a href="whole-slide-imaging.html#cb69-9"></a></span>
<span id="cb69-10"><a href="whole-slide-imaging.html#cb69-10"></a>a1=<span class="kw">cor.test</span>(WSIvalFracs[<span class="dv">3</span>, ], WSIvalFracs[<span class="dv">4</span>, ])</span>
<span id="cb69-11"><a href="whole-slide-imaging.html#cb69-11"></a><span class="kw">plot</span>(WSIvalFracs[<span class="dv">3</span>, ], WSIvalFracs[<span class="dv">4</span>, ], <span class="dt">xlab=</span><span class="st">&quot;Epcam+SMA+&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;SMA&quot;</span>, <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;rho=%s,p=%s&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(a1<span class="op">$</span>p.value, <span class="dv">2</span>)))</span>
<span id="cb69-12"><a href="whole-slide-imaging.html#cb69-12"></a></span>
<span id="cb69-13"><a href="whole-slide-imaging.html#cb69-13"></a>a1=<span class="kw">cor.test</span>(WSIvalFracs[<span class="dv">2</span>, ], WSIvalFracs[<span class="dv">5</span>, ])</span>
<span id="cb69-14"><a href="whole-slide-imaging.html#cb69-14"></a><span class="kw">plot</span>(WSIvalFracs[<span class="dv">2</span>, ], WSIvalFracs[<span class="dv">5</span>, ], <span class="dt">xlab=</span><span class="st">&quot;Epcam+&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;nclass&quot;</span>, <span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&quot;rho=%s,p=%s&quot;</span>, <span class="kw">round</span>(a1<span class="op">$</span>estimate, <span class="dv">2</span>), <span class="kw">round</span>(a1<span class="op">$</span>p.value, <span class="dv">2</span>)))</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-31-2.png" width="672" /></p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="whole-slide-imaging.html#cb70-1"></a></span>
<span id="cb70-2"><a href="whole-slide-imaging.html#cb70-2"></a><span class="co">#devoff()</span></span>
<span id="cb70-3"><a href="whole-slide-imaging.html#cb70-3"></a></span>
<span id="cb70-4"><a href="whole-slide-imaging.html#cb70-4"></a><span class="co"># correlations</span></span>
<span id="cb70-5"><a href="whole-slide-imaging.html#cb70-5"></a></span>
<span id="cb70-6"><a href="whole-slide-imaging.html#cb70-6"></a>ctest=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow=</span><span class="dv">5</span>, <span class="dt">ncol=</span><span class="dv">5</span>)</span>
<span id="cb70-7"><a href="whole-slide-imaging.html#cb70-7"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>){</span>
<span id="cb70-8"><a href="whole-slide-imaging.html#cb70-8"></a>  ctest[i, ]=<span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="cf">function</span>(x) <span class="kw">cor.test</span>(WSIvalFracs[i, ], WSIvalFracs[x, ])<span class="op">$</span>p.value)</span>
<span id="cb70-9"><a href="whole-slide-imaging.html#cb70-9"></a>}</span></code></pre></div>
<p>All associations determined using a correlation test</p>
</div>
<div id="associations-between-cd8-counts-with-other-clinical-variables" class="section level2">
<h2><span class="header-section-number">4.6</span> Associations between CD8 counts with other clinical variables</h2>
<p>Below we assess whether any of the CD8-variables described in section 3.1 is associated with</p>
<ul>
<li>treatment</li>
<li>growth</li>
<li>spatial pattern</li>
</ul>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="whole-slide-imaging.html#cb71-1"></a>indx1=<span class="kw">c</span>(indx1, <span class="st">&quot;log2CD8_EPorSMARatio.WSI&quot;</span>, <span class="st">&quot;log2CD8_EPRatio.WSI&quot;</span>)</span>
<span id="cb71-2"><a href="whole-slide-imaging.html#cb71-2"></a></span>
<span id="cb71-3"><a href="whole-slide-imaging.html#cb71-3"></a>a1=<span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">ncol=</span><span class="kw">length</span>(indx1), <span class="dt">nrow=</span><span class="dv">5</span>)</span>
<span id="cb71-4"><a href="whole-slide-imaging.html#cb71-4"></a></span>
<span id="cb71-5"><a href="whole-slide-imaging.html#cb71-5"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(indx1)){</span>
<span id="cb71-6"><a href="whole-slide-imaging.html#cb71-6"></a>a1[<span class="dv">1</span>, i]=<span class="kw">wilcox.test</span>(SummaryData[SummaryData<span class="op">$</span>Growth<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;stable&quot;</span>, <span class="st">&quot;growing&quot;</span>) ,indx1[i] ]<span class="op">~</span><span class="st"> </span>SummaryData<span class="op">$</span>Growth[SummaryData<span class="op">$</span>Growth<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;stable&quot;</span>, <span class="st">&quot;growing&quot;</span>)])<span class="op">$</span>p.value </span>
<span id="cb71-7"><a href="whole-slide-imaging.html#cb71-7"></a>a1[<span class="dv">2</span>, i]=<span class="kw">wilcox.test</span>(SummaryData[SummaryData<span class="op">$</span>SpatialManual<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Infiltrating&quot;</span>, <span class="st">&quot;restricted&quot;</span>) ,indx1[i] ]<span class="op">~</span><span class="st"> </span>SummaryData<span class="op">$</span>SpatialManual[SummaryData<span class="op">$</span>SpatialManual<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Infiltrating&quot;</span>, <span class="st">&quot;restricted&quot;</span>)])<span class="op">$</span>p.value</span>
<span id="cb71-8"><a href="whole-slide-imaging.html#cb71-8"></a>a1[<span class="dv">3</span>, i]=<span class="kw">wilcox.test</span>(SummaryData[SummaryData<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;PDL1&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>) ,indx1[i] ]<span class="op">~</span><span class="st"> </span>SummaryData<span class="op">$</span>Treatment[SummaryData<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;PDL1&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>)])<span class="op">$</span>p.value</span>
<span id="cb71-9"><a href="whole-slide-imaging.html#cb71-9"></a>a1[<span class="dv">4</span>, i]=<span class="kw">wilcox.test</span>(SummaryData[SummaryData<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;PDL1+LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>) ,indx1[i] ]<span class="op">~</span><span class="st"> </span>SummaryData<span class="op">$</span>Treatment[SummaryData<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;PDL1+LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>)])<span class="op">$</span>p.value</span>
<span id="cb71-10"><a href="whole-slide-imaging.html#cb71-10"></a>a1[<span class="dv">5</span>, i]=<span class="kw">wilcox.test</span>(SummaryData[SummaryData<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>) ,indx1[i] ]<span class="op">~</span><span class="st"> </span>SummaryData<span class="op">$</span>Treatment[SummaryData<span class="op">$</span>Treatment<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;LY&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>)])<span class="op">$</span>p.value</span>
<span id="cb71-11"><a href="whole-slide-imaging.html#cb71-11"></a>}</span>
<span id="cb71-12"><a href="whole-slide-imaging.html#cb71-12"></a></span>
<span id="cb71-13"><a href="whole-slide-imaging.html#cb71-13"></a><span class="kw">colnames</span>(a1)=indx1</span>
<span id="cb71-14"><a href="whole-slide-imaging.html#cb71-14"></a><span class="kw">rownames</span>(a1)=<span class="kw">c</span>(<span class="st">&quot;growth&quot;</span>, <span class="st">&quot;spatial&quot;</span>, <span class="st">&quot;pdl1&quot;</span>, <span class="st">&quot;pdl1+ly&quot;</span>, <span class="st">&quot;ly&quot;</span>)</span>
<span id="cb71-15"><a href="whole-slide-imaging.html#cb71-15"></a></span>
<span id="cb71-16"><a href="whole-slide-imaging.html#cb71-16"></a><span class="kw">par</span>(<span class="dt">oma=</span><span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">4</span>))</span>
<span id="cb71-17"><a href="whole-slide-imaging.html#cb71-17"></a><span class="kw">heatmap.2</span>(<span class="op">-</span><span class="kw">log10</span>(a1), <span class="dt">col=</span><span class="kw">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Blues&quot;</span>), <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="RatDCIS-bookdown_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
<p>Note that CD8 normalised by tumor size is associated with growth (but this could a reflection of the size of the tumor), and there is a borderline difference once normalised by epithelial content. In addition the CD8 total count is associated with pdl1+ly treatment.</p>
<p>Note that p=0.05 is designated by a value of 1.3</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cohort-summary-table.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spatial-statistics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
