<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 26 Writing files to file | NMU-induced rat tumor models for I/O</title>
  <meta name="description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 26 Writing files to file | NMU-induced rat tumor models for I/O" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://polyak-lab.github.io/RatDCISbookdown/" />
  
  <meta property="og:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  <meta name="github-repo" content="polyak-lab/RatDCISbookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 26 Writing files to file | NMU-induced rat tumor models for I/O" />
  
  <meta name="twitter:description" content="This is a summary of the code required in the study: ‘NMU-induced rat tumor models for I/O’" />
  

<meta name="author" content="Anne Trinh" />


<meta name="date" content="2021-06-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cnv-calling.html"/>
<link rel="next" href="list-of-figures.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.17/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.10.20/jszip.min.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/forceGraph-0.1.0.9/common.css" rel="stylesheet" />
<script src="libs/forceGraph-0.1.0.9/common.js"></script>
<script src="libs/forceGraph-0.1.0.9/jquery.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/settings.js"></script>
<script src="libs/forceGraph-0.1.0.9/sigma.min.js"></script>
<script src="libs/forceGraph-0.1.0.9/plugins.min.js"></script>
<script src="libs/forceGraph-binding-0.99.19/forceGraph.js"></script>
<script type="text/javascript">
  
  // toggle visibility of R source blocks in R Markdown output
function toggle_R() {
  var x = document.getElementsByClassName('r');
  if (x.length == 0) return;
  function toggle_vis(o) {
    var d = o.style.display;
    o.style.display = (d == 'block' || d == '') ? 'none':'block';
  }
  
  for (i = 0; i < x.length; i++) {
    var y = x[i];
    if (y.tagName.toLowerCase() === 'pre') toggle_vis(y);
  }
  
  var elem = document.getElementById("myButton1");
  if (elem.value === "Hide Global") elem.value = "Show Global";
  else elem.value = "Hide Global";
}

document.write('<input onclick="toggle_R();" type="button" value="Hide Global" id="myButton1" style="position: absolute; top: 10%; right: 2%; z-index: 200"></input>')

</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rat prevention study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="prerequisites.html"><a href="prerequisites.html#packages-and-software"><i class="fa fa-check"></i><b>1.1</b> Packages and Software</a></li>
<li class="chapter" data-level="1.2" data-path="prerequisites.html"><a href="prerequisites.html#external-software"><i class="fa fa-check"></i><b>1.2</b> External software</a></li>
<li class="chapter" data-level="1.3" data-path="prerequisites.html"><a href="prerequisites.html#annotations"><i class="fa fa-check"></i><b>1.3</b> Annotations</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prerequisites.html"><a href="prerequisites.html#genomic-properties"><i class="fa fa-check"></i><b>1.3.1</b> Genomic properties</a></li>
<li class="chapter" data-level="1.3.2" data-path="prerequisites.html"><a href="prerequisites.html#gene-name-homologs-between-organisms"><i class="fa fa-check"></i><b>1.3.2</b> Gene name homologs between organisms</a></li>
<li class="chapter" data-level="1.3.3" data-path="prerequisites.html"><a href="prerequisites.html#gene-signatures-and-data-bases"><i class="fa fa-check"></i><b>1.3.3</b> Gene signatures and data-bases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html"><i class="fa fa-check"></i><b>2</b> Cohort characteristics</a><ul>
<li class="chapter" data-level="2.1" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#size-information"><i class="fa fa-check"></i><b>2.1</b> Size information</a></li>
<li class="chapter" data-level="2.2" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#calculating-growth-rates"><i class="fa fa-check"></i><b>2.2</b> Calculating growth rates</a></li>
<li class="chapter" data-level="2.3" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#km-curves-for-survival"><i class="fa fa-check"></i><b>2.3</b> KM curves for survival</a></li>
<li class="chapter" data-level="2.4" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data-dncd45epcam"><i class="fa fa-check"></i><b>2.4</b> FACS data (DN/CD45/EpCAM)</a></li>
<li class="chapter" data-level="2.5" data-path="cohort-characteristics.html"><a href="cohort-characteristics.html#facs-data"><i class="fa fa-check"></i><b>2.5</b> FACS data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html"><i class="fa fa-check"></i><b>3</b> Cohort Summary Table</a><ul>
<li class="chapter" data-level="3.1" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#total-number-of-samples"><i class="fa fa-check"></i><b>3.1</b> Total number of samples</a></li>
<li class="chapter" data-level="3.2" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#compare-the-characterisation-vs-progression-cohort"><i class="fa fa-check"></i><b>3.2</b> Compare the characterisation vs progression cohort</a></li>
<li class="chapter" data-level="3.3" data-path="cohort-summary-table.html"><a href="cohort-summary-table.html#summary-of-the-rna-data"><i class="fa fa-check"></i><b>3.3</b> Summary of the RNA data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html"><i class="fa fa-check"></i><b>4</b> Whole-slide imaging</a><ul>
<li class="chapter" data-level="4.1" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-the-frequencies-with-other-data-types"><i class="fa fa-check"></i><b>4.1</b> Associate the frequencies with other data types</a></li>
<li class="chapter" data-level="4.2" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#cellular-composition"><i class="fa fa-check"></i><b>4.2</b> Cellular composition</a></li>
<li class="chapter" data-level="4.3" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associate-composition-with-other-covariates"><i class="fa fa-check"></i><b>4.3</b> Associate composition with other covariates</a></li>
<li class="chapter" data-level="4.4" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#estimate-tumor-size"><i class="fa fa-check"></i><b>4.4</b> Estimate tumor size</a></li>
<li class="chapter" data-level="4.5" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#correlations-between-different-subpopulations"><i class="fa fa-check"></i><b>4.5</b> Correlations between different subpopulations</a></li>
<li class="chapter" data-level="4.6" data-path="whole-slide-imaging.html"><a href="whole-slide-imaging.html#associations-between-cd8-counts-with-other-clinical-variables"><i class="fa fa-check"></i><b>4.6</b> Associations between CD8 counts with other clinical variables</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>5</b> Spatial statistics</a><ul>
<li class="chapter" data-level="5.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#knn-distances"><i class="fa fa-check"></i><b>5.1</b> knn-Distances:</a><ul>
<li class="chapter" data-level="5.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-classification"><i class="fa fa-check"></i><b>5.1.1</b> Comparison to manual classification</a></li>
<li class="chapter" data-level="5.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#associations-with-outcome-to-treatment"><i class="fa fa-check"></i><b>5.1.2</b> Associations with outcome to treatment</a></li>
<li class="chapter" data-level="5.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth"><i class="fa fa-check"></i><b>5.1.3</b> Growth</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#the-interacting-fraction"><i class="fa fa-check"></i><b>5.2</b> The interacting fraction</a><ul>
<li class="chapter" data-level="5.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-select-optimal-r"><i class="fa fa-check"></i><b>5.2.1</b> Comparison to manual &amp; select optimal r</a></li>
<li class="chapter" data-level="5.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-1"><i class="fa fa-check"></i><b>5.2.2</b> Growth</a></li>
<li class="chapter" data-level="5.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment"><i class="fa fa-check"></i><b>5.2.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#m-h-distances"><i class="fa fa-check"></i><b>5.3</b> M-H distances</a><ul>
<li class="chapter" data-level="5.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-to-manual-scoring"><i class="fa fa-check"></i><b>5.3.1</b> Comparison to Manual Scoring</a></li>
<li class="chapter" data-level="5.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#growth-2"><i class="fa fa-check"></i><b>5.3.2</b> Growth</a></li>
<li class="chapter" data-level="5.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#treatment-1"><i class="fa fa-check"></i><b>5.3.3</b> Treatment</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#comparison-between-metrics"><i class="fa fa-check"></i><b>5.4</b> Comparison between metrics</a></li>
<li class="chapter" data-level="5.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#distances-to-unclassified-cells"><i class="fa fa-check"></i><b>5.5</b> Distances to “unclassified cells”</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-data.html"><a href="expression-data.html"><i class="fa fa-check"></i><b>6</b> Expression data</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-data.html"><a href="expression-data.html#running-alignment"><i class="fa fa-check"></i><b>6.1</b> Running alignment</a></li>
<li class="chapter" data-level="6.2" data-path="expression-data.html"><a href="expression-data.html#rna-initial-qc"><i class="fa fa-check"></i><b>6.2</b> RNA Initial QC</a></li>
<li class="chapter" data-level="6.3" data-path="expression-data.html"><a href="expression-data.html#normalisation"><i class="fa fa-check"></i><b>6.3</b> Normalisation</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-data.html"><a href="expression-data.html#preliminary-visualisation-to-remove-outliers"><i class="fa fa-check"></i><b>6.3.1</b> preliminary visualisation (to remove outliers)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-data.html"><a href="expression-data.html#processing-files-for-external-software"><i class="fa fa-check"></i><b>6.4</b> Processing files for external software</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html"><i class="fa fa-check"></i><b>7</b> RNA data: preliminary plots</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#pca-plots"><i class="fa fa-check"></i><b>7.1</b> PCA plots</a></li>
<li class="chapter" data-level="7.2" data-path="rna-data-preliminary-plots.html"><a href="rna-data-preliminary-plots.html#expression-patterns-by-cell-type"><i class="fa fa-check"></i><b>7.2</b> Expression patterns by cell type</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html"><i class="fa fa-check"></i><b>8</b> DESeq analysis: Progression/Immunotherapy cohort</a><ul>
<li class="chapter" data-level="8.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn-vs-ep"><i class="fa fa-check"></i><b>8.1</b> DN vs Ep</a></li>
<li class="chapter" data-level="8.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#no.-samples-in-comparisons"><i class="fa fa-check"></i><b>8.2</b> No. samples in comparisons</a></li>
<li class="chapter" data-level="8.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#set-up-cell-type-specific-the-comparisons"><i class="fa fa-check"></i><b>8.3</b> Set-up cell-type specific the comparisons</a></li>
<li class="chapter" data-level="8.4" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pca-plots-1"><i class="fa fa-check"></i><b>8.4</b> PCA plots</a><ul>
<li class="chapter" data-level="8.4.1" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#epcam"><i class="fa fa-check"></i><b>8.4.1</b> EpCAM</a></li>
<li class="chapter" data-level="8.4.2" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#cd45"><i class="fa fa-check"></i><b>8.4.2</b> CD45</a></li>
<li class="chapter" data-level="8.4.3" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#dn"><i class="fa fa-check"></i><b>8.4.3</b> DN</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="deseq-analysis-progressionimmunotherapy-cohort.html"><a href="deseq-analysis-progressionimmunotherapy-cohort.html#pearson-correlation-plots-of-samples"><i class="fa fa-check"></i><b>8.5</b> Pearson correlation plots of samples</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="collating-results-and-running-gsea.html"><a href="collating-results-and-running-gsea.html"><i class="fa fa-check"></i><b>9</b> Collating results and running GSEA</a></li>
<li class="chapter" data-level="10" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html"><i class="fa fa-check"></i><b>10</b> ER/Pgr Subtyping</a><ul>
<li class="chapter" data-level="10.1" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#gene-list"><i class="fa fa-check"></i><b>10.1</b> Gene List</a></li>
<li class="chapter" data-level="10.2" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#progression-cohort"><i class="fa fa-check"></i><b>10.2</b> Progression cohort</a></li>
<li class="chapter" data-level="10.3" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#charcterisation-cohort"><i class="fa fa-check"></i><b>10.3</b> Charcterisation cohort</a></li>
<li class="chapter" data-level="10.4" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#dn-samples"><i class="fa fa-check"></i><b>10.4</b> DN samples</a></li>
<li class="chapter" data-level="10.5" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#comparison-with-staining"><i class="fa fa-check"></i><b>10.5</b> Comparison with staining</a></li>
<li class="chapter" data-level="10.6" data-path="erpgr-subtyping.html"><a href="erpgr-subtyping.html#summary-of-expression-markers-for-each-subtypecell-fraction"><i class="fa fa-check"></i><b>10.6</b> Summary of expression markers for each subtype/cell fraction</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="expression-in-specific-pathways.html"><a href="expression-in-specific-pathways.html"><i class="fa fa-check"></i><b>11</b> Expression in Specific pathways</a></li>
<li class="chapter" data-level="12" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html"><i class="fa fa-check"></i><b>12</b> DESeq analysis: Immunotherapy/Growth comparisons</a><ul>
<li class="chapter" data-level="12.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#summary-of-comparisons"><i class="fa fa-check"></i><b>12.1</b> Summary of comparisons</a></li>
<li class="chapter" data-level="12.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#deseq-immunotherapy"><i class="fa fa-check"></i><b>12.2</b> DESeq: immunotherapy</a></li>
<li class="chapter" data-level="12.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#growing-vs-stable-emphasis"><i class="fa fa-check"></i><b>12.3</b> Growing vs stable emphasis</a><ul>
<li class="chapter" data-level="12.3.1" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#dn-fraction"><i class="fa fa-check"></i><b>12.3.1</b> DN fraction</a></li>
<li class="chapter" data-level="12.3.2" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#cd-fraction"><i class="fa fa-check"></i><b>12.3.2</b> CD fraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="deseq-analysis-immunotherapygrowth-comparisons.html"><a href="deseq-analysis-immunotherapygrowth-comparisons.html#ep-fraction"><i class="fa fa-check"></i><b>12.3.3</b> Ep Fraction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html"><i class="fa fa-check"></i><b>13</b> Summary of GSEA runs</a><ul>
<li class="chapter" data-level="13.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#stable-vs-growing-all-samples"><i class="fa fa-check"></i><b>13.1</b> Stable vs growing: all samples</a><ul>
<li class="chapter" data-level="13.1.1" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#barplots-of-enriched-pathways"><i class="fa fa-check"></i><b>13.1.1</b> barplots of enriched pathways</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#comparisons-based-on-treatment"><i class="fa fa-check"></i><b>13.2</b> Comparisons based on treatment</a></li>
<li class="chapter" data-level="13.3" data-path="summary-of-gsea-runs.html"><a href="summary-of-gsea-runs.html#pathways-of-interest-2"><i class="fa fa-check"></i><b>13.3</b> Pathways of Interest 2</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html"><i class="fa fa-check"></i><b>14</b> Epcam+ Inflammatory vs non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#identification-of-inflammatory-samples"><i class="fa fa-check"></i><b>14.1</b> Identification of inflammatory samples</a></li>
<li class="chapter" data-level="14.2" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#deg-inflammatory-vs-non-inflammatory"><i class="fa fa-check"></i><b>14.2</b> DEG: inflammatory vs non-inflammatory</a></li>
<li class="chapter" data-level="14.3" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#finding-3-signatures-for-3-branches"><i class="fa fa-check"></i><b>14.3</b> Finding 3 signatures for 3 branches</a></li>
<li class="chapter" data-level="14.4" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#analyse-the-non-inflammatory-samples"><i class="fa fa-check"></i><b>14.4</b> Analyse the non-inflammatory samples</a><ul>
<li class="chapter" data-level="14.4.1" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#gsea"><i class="fa fa-check"></i><b>14.4.1</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="epcam-inflammatory-vs-non-inflammatory-samples.html"><a href="epcam-inflammatory-vs-non-inflammatory-samples.html#luminal-only-non-inflammtory-samples-samples"><i class="fa fa-check"></i><b>14.5</b> Luminal-only non-inflammtory samples samples</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>15</b> TCGA: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="15.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="15.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome"><i class="fa fa-check"></i><b>15.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="15.2.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#all-samples"><i class="fa fa-check"></i><b>15.2.1</b> All samples</a></li>
<li class="chapter" data-level="15.2.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases"><i class="fa fa-check"></i><b>15.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable"><i class="fa fa-check"></i><b>15.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="15.4" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint"><i class="fa fa-check"></i><b>15.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="15.4.1" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#overall-survival"><i class="fa fa-check"></i><b>15.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="15.4.2" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#dfs-survival"><i class="fa fa-check"></i><b>15.4.2</b> DFS survival</a></li>
<li class="chapter" data-level="15.4.3" data-path="tcga-associate-epcam-inflammatory-with-survival.html"><a href="tcga-associate-epcam-inflammatory-with-survival.html#tcga-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>15.4.3</b> TCGA: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html"><i class="fa fa-check"></i><b>16</b> METABRIC: Associate Epcam+ inflammatory with survival</a><ul>
<li class="chapter" data-level="16.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-cd74-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.1</b> Associating CD74 with phenotype and outcome</a></li>
<li class="chapter" data-level="16.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#associating-admats10-with-phenotype-and-outcome-1"><i class="fa fa-check"></i><b>16.2</b> Associating ADMATS10 with phenotype and outcome</a><ul>
<li class="chapter" data-level="16.2.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#all-samples-1"><i class="fa fa-check"></i><b>16.2.1</b> All samples</a></li>
<li class="chapter" data-level="16.2.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#redone-using-only-er-cases-1"><i class="fa fa-check"></i><b>16.2.2</b> Redone using only ER cases</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#signature-lum-cases-non-inflammatory-growing-vs-stable-1"><i class="fa fa-check"></i><b>16.3</b> Signature: Lum cases non-inflammatory: growing vs stable</a></li>
<li class="chapter" data-level="16.4" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#comparison-to-oncotype-and-mammaprint-1"><i class="fa fa-check"></i><b>16.4</b> Comparison to oncotype and mammaprint</a><ul>
<li class="chapter" data-level="16.4.1" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#overall-survival-1"><i class="fa fa-check"></i><b>16.4.1</b> Overall survival</a></li>
<li class="chapter" data-level="16.4.2" data-path="metabric-associate-epcam-inflammatory-with-survival.html"><a href="metabric-associate-epcam-inflammatory-with-survival.html#met-luma-high-vs-low-signatures"><i class="fa fa-check"></i><b>16.4.2</b> MET: LumA high vs low signatures</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html"><i class="fa fa-check"></i><b>17</b> DESeq analysis: Characterisation cohort (big vs small)</a><ul>
<li class="chapter" data-level="17.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#cd45-samples-1"><i class="fa fa-check"></i><b>17.1</b> CD45 samples</a><ul>
<li class="chapter" data-level="17.1.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot"><i class="fa fa-check"></i><b>17.1.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.1.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#differential-gene-expression"><i class="fa fa-check"></i><b>17.1.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="17.1.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-1"><i class="fa fa-check"></i><b>17.1.3</b> GSEA</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#epithelial-samples-1"><i class="fa fa-check"></i><b>17.2</b> Epithelial samples</a><ul>
<li class="chapter" data-level="17.2.1" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#pca-plot-1"><i class="fa fa-check"></i><b>17.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="17.2.2" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#gsea-2"><i class="fa fa-check"></i><b>17.2.2</b> GSEA</a></li>
<li class="chapter" data-level="17.2.3" data-path="deseq-analysis-characterisation-cohort-big-vs-small.html"><a href="deseq-analysis-characterisation-cohort-big-vs-small.html#check-expression-of-checkpoint-proteins"><i class="fa fa-check"></i><b>17.2.3</b> Check expression of checkpoint proteins</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="signature-analysis.html"><a href="signature-analysis.html"><i class="fa fa-check"></i><b>18</b> Signature analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-signature-analysis"><i class="fa fa-check"></i><b>18.1</b> MHC signature analysis</a><ul>
<li class="chapter" data-level="18.1.1" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-i"><i class="fa fa-check"></i><b>18.1.1</b> MHC-I</a></li>
<li class="chapter" data-level="18.1.2" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-ii"><i class="fa fa-check"></i><b>18.1.2</b> MHC-II</a></li>
<li class="chapter" data-level="18.1.3" data-path="signature-analysis.html"><a href="signature-analysis.html#mhc-presentation-proteins"><i class="fa fa-check"></i><b>18.1.3</b> MHC presentation proteins</a></li>
<li class="chapter" data-level="18.1.4" data-path="signature-analysis.html"><a href="signature-analysis.html#inflammation-related-genes-il6-jak-stat-and-tnfnfkb"><i class="fa fa-check"></i><b>18.1.4</b> inflammation related genes: IL6-JAK-STAT and TNF/NFKB</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="immune-estimation.html"><a href="immune-estimation.html"><i class="fa fa-check"></i><b>19</b> Immune estimation</a><ul>
<li class="chapter" data-level="19.1" data-path="immune-estimation.html"><a href="immune-estimation.html#overview-of-the-cell-types"><i class="fa fa-check"></i><b>19.1</b> Overview of the cell types</a></li>
<li class="chapter" data-level="19.2" data-path="immune-estimation.html"><a href="immune-estimation.html#characterisation-cohort-assoc-with-size"><i class="fa fa-check"></i><b>19.2</b> characterisation cohort: assoc with size</a></li>
<li class="chapter" data-level="19.3" data-path="immune-estimation.html"><a href="immune-estimation.html#comparison-with-facs-data"><i class="fa fa-check"></i><b>19.3</b> Comparison with FACS data</a></li>
<li class="chapter" data-level="19.4" data-path="immune-estimation.html"><a href="immune-estimation.html#progression-cohort-1"><i class="fa fa-check"></i><b>19.4</b> Progression cohort</a></li>
<li class="chapter" data-level="19.5" data-path="immune-estimation.html"><a href="immune-estimation.html#clinical-associations"><i class="fa fa-check"></i><b>19.5</b> Clinical associations</a><ul>
<li class="chapter" data-level="19.5.1" data-path="immune-estimation.html"><a href="immune-estimation.html#associate-with-treatment"><i class="fa fa-check"></i><b>19.5.1</b> Associate with Treatment</a></li>
<li class="chapter" data-level="19.5.2" data-path="immune-estimation.html"><a href="immune-estimation.html#association-with-growth"><i class="fa fa-check"></i><b>19.5.2</b> Association with Growth</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="immune-estimation.html"><a href="immune-estimation.html#summary-of-the-outcome"><i class="fa fa-check"></i><b>19.6</b> Summary of the outcome</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html"><i class="fa fa-check"></i><b>20</b> BCR clonotype analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#processing-summary"><i class="fa fa-check"></i><b>20.1</b> Processing summary</a></li>
<li class="chapter" data-level="20.2" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#summary-stats"><i class="fa fa-check"></i><b>20.2</b> Summary Stats</a></li>
<li class="chapter" data-level="20.3" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#diversity-metrics"><i class="fa fa-check"></i><b>20.3</b> Diversity metrics</a></li>
<li class="chapter" data-level="20.4" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#compare-the-characterisation-cohort"><i class="fa fa-check"></i><b>20.4</b> Compare the characterisation cohort</a></li>
<li class="chapter" data-level="20.5" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-clinicopathological-data-progression"><i class="fa fa-check"></i><b>20.5</b> Associate with clinicopathological data (progression)</a></li>
<li class="chapter" data-level="20.6" data-path="bcr-clonotype-analysis.html"><a href="bcr-clonotype-analysis.html#associate-with-signature-scores"><i class="fa fa-check"></i><b>20.6</b> Associate with signature scores</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html"><i class="fa fa-check"></i><b>21</b> Whole Genome Sequencing Mutations</a><ul>
<li class="chapter" data-level="21.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#data"><i class="fa fa-check"></i><b>21.1</b> Data</a><ul>
<li class="chapter" data-level="21.1.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#extract-mutation-signatures"><i class="fa fa-check"></i><b>21.1.1</b> Extract mutation signatures</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#annotate-the-data-with-human-common-variants"><i class="fa fa-check"></i><b>21.2</b> Annotate the data with human common variants</a></li>
<li class="chapter" data-level="21.3" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#plots"><i class="fa fa-check"></i><b>21.3</b> Plots</a><ul>
<li class="chapter" data-level="21.3.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#quick-overview"><i class="fa fa-check"></i><b>21.3.1</b> Quick overview</a></li>
<li class="chapter" data-level="21.3.2" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#coding-variants-tmb"><i class="fa fa-check"></i><b>21.3.2</b> Coding variants &amp; TMB</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#comparison-with-rat-mutational-datasets"><i class="fa fa-check"></i><b>21.4</b> Comparison with rat mutational datasets</a><ul>
<li class="chapter" data-level="21.4.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#sites-which-are-commonly-mutated"><i class="fa fa-check"></i><b>21.4.1</b> Sites which are commonly mutated?</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#overview-of-the-mutations"><i class="fa fa-check"></i><b>21.5</b> Overview of the mutations</a><ul>
<li class="chapter" data-level="21.5.1" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#all-mutations"><i class="fa fa-check"></i><b>21.5.1</b> All mutations</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="whole-genome-sequencing-mutations.html"><a href="whole-genome-sequencing-mutations.html#metacore-analysis-of-commonly-mutated-pathways"><i class="fa fa-check"></i><b>21.6</b> Metacore analysis of commonly mutated pathways</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html"><i class="fa fa-check"></i><b>22</b> Mutations in RNA</a><ul>
<li class="chapter" data-level="22.1" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#haplotype-caller"><i class="fa fa-check"></i><b>22.1</b> Haplotype caller</a></li>
<li class="chapter" data-level="22.2" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#load-files"><i class="fa fa-check"></i><b>22.2</b> Load files</a></li>
<li class="chapter" data-level="22.3" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#identifying-polymorphisms-compare-frequencies-in-cd45-and-ep-data"><i class="fa fa-check"></i><b>22.3</b> Identifying polymorphisms: compare frequencies in CD45 and Ep data</a></li>
<li class="chapter" data-level="22.4" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#find-coding-mutations-which-are-specific-to-ep-cells"><i class="fa fa-check"></i><b>22.4</b> Find coding mutations which are specific to ep cells</a></li>
<li class="chapter" data-level="22.5" data-path="mutations-in-rna.html"><a href="mutations-in-rna.html#filtering-wgs-data-based-on-cd45-data"><i class="fa fa-check"></i><b>22.5</b> Filtering WGS data based on CD45 data</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html"><i class="fa fa-check"></i><b>23</b> Mutations in progression cohort</a><ul>
<li class="chapter" data-level="23.1" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#summary-of-common-mutations"><i class="fa fa-check"></i><b>23.1</b> Summary of common mutations:</a></li>
<li class="chapter" data-level="23.2" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#look-at-the-common-mutations-cosmic-ones"><i class="fa fa-check"></i><b>23.2</b> Look at the common mutations (cosmic ones)</a></li>
<li class="chapter" data-level="23.3" data-path="mutations-in-progression-cohort.html"><a href="mutations-in-progression-cohort.html#also-look-at-the-top-rna-mutations-with-are-ep-specific"><i class="fa fa-check"></i><b>23.3</b> Also look at the top RNA-mutations with are Ep specific</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="trichrome-staining.html"><a href="trichrome-staining.html"><i class="fa fa-check"></i><b>24</b> Trichrome staining</a><ul>
<li class="chapter" data-level="24.1" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cellular-fraction-wsi"><i class="fa fa-check"></i><b>24.1</b> Associations with cellular fraction (wsi)</a></li>
<li class="chapter" data-level="24.2" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-cd8-content"><i class="fa fa-check"></i><b>24.2</b> Associations with CD8 content</a></li>
<li class="chapter" data-level="24.3" data-path="trichrome-staining.html"><a href="trichrome-staining.html#associations-with-growth-and-treatment"><i class="fa fa-check"></i><b>24.3</b> Associations with growth and treatment</a></li>
<li class="chapter" data-level="24.4" data-path="trichrome-staining.html"><a href="trichrome-staining.html#association-with-hyperinflammatory-status"><i class="fa fa-check"></i><b>24.4</b> Association with hyperinflammatory status</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="cnv-calling.html"><a href="cnv-calling.html"><i class="fa fa-check"></i><b>25</b> CNV calling</a><ul>
<li class="chapter" data-level="25.1" data-path="cnv-calling.html"><a href="cnv-calling.html#sample-data"><i class="fa fa-check"></i><b>25.1</b> Sample data</a></li>
<li class="chapter" data-level="25.2" data-path="cnv-calling.html"><a href="cnv-calling.html#data-summary"><i class="fa fa-check"></i><b>25.2</b> Data Summary</a><ul>
<li class="chapter" data-level="25.2.1" data-path="cnv-calling.html"><a href="cnv-calling.html#frequency-of-gains-and-losses-across-the-genome"><i class="fa fa-check"></i><b>25.2.1</b> Frequency of gains and losses across the genome</a></li>
<li class="chapter" data-level="25.2.2" data-path="cnv-calling.html"><a href="cnv-calling.html#loci-which-have-a-hit-in-at-least-30-of-samples"><i class="fa fa-check"></i><b>25.2.2</b> Loci which have a hit in at least 30% of samples</a></li>
<li class="chapter" data-level="25.2.3" data-path="cnv-calling.html"><a href="cnv-calling.html#annotated-plot-of-genome-and-locations-of-genes"><i class="fa fa-check"></i><b>25.2.3</b> Annotated plot of genome and locations of genes</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="cnv-calling.html"><a href="cnv-calling.html#samples-with-cnvs-in-breast-related-genes"><i class="fa fa-check"></i><b>25.3</b> Samples with CNVs in breast-related genes</a></li>
<li class="chapter" data-level="25.4" data-path="cnv-calling.html"><a href="cnv-calling.html#summary-of-the-sequencing-depth"><i class="fa fa-check"></i><b>25.4</b> Summary of the sequencing depth</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="writing-files-to-file.html"><a href="writing-files-to-file.html"><i class="fa fa-check"></i><b>26</b> Writing files to file</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="list-of-figures.html"><a href="list-of-figures.html"><i class="fa fa-check"></i><b>A</b> List of Figures</a><ul>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#main-figures"><i class="fa fa-check"></i>Main Figures</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-1-nmu-induced-mammary-tumors"><i class="fa fa-check"></i>Figure 1: NMU-induced mammary tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-2-immune-system-of-nmu-rat"><i class="fa fa-check"></i>Figure 2: Immune system of NMU-rat</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-3-immunotherapy-applied-to-nmu-rats"><i class="fa fa-check"></i>Figure 3: Immunotherapy applied to NMU rats</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-4-characterizing-cd45-cells-in-tumors"><i class="fa fa-check"></i>Figure 4: Characterizing CD45 cells in tumors</a></li>
<li class="chapter" data-level="" data-path="list-of-figures.html"><a href="list-of-figures.html#figure-5-epithelial-cells-in-growing-vs-stable-rats"><i class="fa fa-check"></i>Figure 5 : Epithelial cells in growing vs stable rats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html"><i class="fa fa-check"></i>Extended Figures</a><ul>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-1"><i class="fa fa-check"></i>Extended data 1</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-2"><i class="fa fa-check"></i>Extended data 2</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-3"><i class="fa fa-check"></i>Extended data 3</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-4"><i class="fa fa-check"></i>Extended data 4</a></li>
<li class="chapter" data-level="" data-path="extended-figures.html"><a href="extended-figures.html#extended-data-5"><i class="fa fa-check"></i>Extended data 5</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NMU-induced rat tumor models for I/O</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="writing-files-to-file" class="section level1">
<h1><span class="header-section-number">Chapter 26</span> Writing files to file</h1>
<p>Update the metadata file:
write to file:</p>
<ul>
<li>updated ext data table</li>
<li>spatial information summary</li>
<li>star aligned table</li>
<li>colData information</li>
<li>collapsed information on wsi/facs</li>
</ul>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="writing-files-to-file.html#cb584-1"></a><span class="kw">write.xlsx</span>(Cdata, <span class="dt">file=</span><span class="kw">sprintf</span>(<span class="st">&quot;../metadata/Ext_Data_Table_updated_%s.xlsx&quot;</span>, <span class="kw">Sys.Date</span>()))</span>
<span id="cb584-2"><a href="writing-files-to-file.html#cb584-2"></a></span>
<span id="cb584-3"><a href="writing-files-to-file.html#cb584-3"></a><span class="co">#colnames(Cdata)</span></span>
<span id="cb584-4"><a href="writing-files-to-file.html#cb584-4"></a></span>
<span id="cb584-5"><a href="writing-files-to-file.html#cb584-5"></a>x1=<span class="kw">match</span>(<span class="kw">rownames</span>(df.Spatial), <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;&quot;</span>, Cdata<span class="op">$</span>TumorID))</span>
<span id="cb584-6"><a href="writing-files-to-file.html#cb584-6"></a>df.Spatial<span class="op">$</span>NewID=Cdata<span class="op">$</span>NewID[x1]</span>
<span id="cb584-7"><a href="writing-files-to-file.html#cb584-7"></a>dfB=Cdata[x1, <span class="kw">c</span>(<span class="st">&quot;NewID&quot;</span>, <span class="st">&quot;TumorID&quot;</span>,<span class="st">&quot;TumorAreaWSI&quot;</span>, <span class="st">&quot;Tumor.diameter.sac.mm&quot;</span>, <span class="st">&quot;CD45.Frac.FACS&quot;</span>,   </span>
<span id="cb584-8"><a href="writing-files-to-file.html#cb584-8"></a><span class="st">&quot;DN.Frac.FACS&quot;</span>, <span class="st">&quot;EpCAM.Frac.FACS&quot;</span>)]</span>
<span id="cb584-9"><a href="writing-files-to-file.html#cb584-9"></a></span>
<span id="cb584-10"><a href="writing-files-to-file.html#cb584-10"></a><span class="kw">write.csv</span>(df.Spatial, <span class="dt">file=</span><span class="st">&quot;nature-tables/summary_spatial_information.csv&quot;</span>)</span>
<span id="cb584-11"><a href="writing-files-to-file.html#cb584-11"></a></span>
<span id="cb584-12"><a href="writing-files-to-file.html#cb584-12"></a><span class="kw">colnames</span>(allstarFinal)=infoTableFinal<span class="op">$</span>TumorIDnew[<span class="kw">match</span>(<span class="kw">colnames</span>(allstarFinal), <span class="kw">rownames</span>(infoTableFinal))]</span>
<span id="cb584-13"><a href="writing-files-to-file.html#cb584-13"></a>infoTableFinal2=infoTableFinal[ ,<span class="kw">c</span>(<span class="st">&quot;TumorIDnew&quot;</span>,<span class="st">&quot;Growth&quot;</span>, <span class="st">&quot;CD8Frac&quot;</span>, <span class="st">&quot;TumSize&quot;</span>, <span class="st">&quot;Time.NMU2Sac&quot;</span>, </span>
<span id="cb584-14"><a href="writing-files-to-file.html#cb584-14"></a>                                   <span class="st">&quot;Age.Injection&quot;</span>,<span class="st">&quot;Treatment&quot;</span>,<span class="st">&quot;Cohort&quot;</span>, <span class="st">&quot;SizeCat&quot;</span>)]</span>
<span id="cb584-15"><a href="writing-files-to-file.html#cb584-15"></a><span class="kw">write.csv</span>(infoTableFinal2, <span class="dt">file=</span><span class="st">&quot;nature-tables/infoTableFinal2_output.csv&quot;</span>)</span>
<span id="cb584-16"><a href="writing-files-to-file.html#cb584-16"></a><span class="kw">write.csv</span>(allstarFinal, <span class="dt">file=</span><span class="st">&quot;nature-tables/allstarFinal_output.csv&quot;</span>)</span>
<span id="cb584-17"><a href="writing-files-to-file.html#cb584-17"></a></span>
<span id="cb584-18"><a href="writing-files-to-file.html#cb584-18"></a></span>
<span id="cb584-19"><a href="writing-files-to-file.html#cb584-19"></a>colTestCD8=<span class="kw">c</span>(<span class="st">&quot;CD8.EpBoundingBox&quot;</span>, <span class="st">&quot;CD8.TumSize&quot;</span>, <span class="st">&quot;CD45.FACS&quot;</span>, <span class="st">&quot;CD8.FACS&quot;</span>, <span class="st">&quot;OverallCD8_Fig4c.Manual&quot;</span>, <span class="st">&quot;TIL_Fig4d.Manual&quot;</span>, <span class="st">&quot;T.cell.CD8._TIMER&quot;</span>, <span class="st">&quot;T.cell.CD8._CIBERSORT&quot;</span>,<span class="st">&quot;T.cell.CD8._CIBERSORT.ABS&quot;</span>, <span class="st">&quot;T.cell.CD8._EPIC&quot;</span>)</span>
<span id="cb584-20"><a href="writing-files-to-file.html#cb584-20"></a></span>
<span id="cb584-21"><a href="writing-files-to-file.html#cb584-21"></a><span class="co"># colTestCD8=c(&quot;CD8.EpDomTiles&quot;,&quot;CD8.EpBoundingBox&quot;, &quot;CD8.TumSize&quot;, &quot;CD45.FACS&quot;, &quot;CD8.FACS&quot;, &quot;OverallCD8_Fig4c.Manual&quot;, &quot;TIL_Fig4d.Manual&quot;, )</span></span>
<span id="cb584-22"><a href="writing-files-to-file.html#cb584-22"></a>indx1=<span class="kw">c</span>(<span class="st">&quot;CD8.WSI&quot;</span>, <span class="st">&quot;CD8Frac.WSI&quot;</span>, <span class="st">&quot;CD8_EPorSMARatio.WSI&quot;</span>,<span class="st">&quot;CD8_AnyEPRatio.WSI&quot;</span>, <span class="st">&quot;CD8_EPRatio.WSI&quot;</span>,<span class="st">&quot;CD8normTumSize&quot;</span>, <span class="st">&quot;CD8.EpBoundingBox&quot;</span>)</span>
<span id="cb584-23"><a href="writing-files-to-file.html#cb584-23"></a></span>
<span id="cb584-24"><a href="writing-files-to-file.html#cb584-24"></a>df.Spatial2=<span class="kw">merge</span>(df.Spatial, dfB, <span class="dt">by.x=</span><span class="st">&quot;NewID&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;NewID&quot;</span>, <span class="dt">all=</span>T)</span>
<span id="cb584-25"><a href="writing-files-to-file.html#cb584-25"></a><span class="co">#head(df.Spatial2)</span></span>
<span id="cb584-26"><a href="writing-files-to-file.html#cb584-26"></a></span>
<span id="cb584-27"><a href="writing-files-to-file.html#cb584-27"></a><span class="kw">colnames</span>(ProgSpecCD45)=<span class="kw">gsub</span>(<span class="st">&quot;_CD45&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">colnames</span>(ProgSpecCD45))</span>
<span id="cb584-28"><a href="writing-files-to-file.html#cb584-28"></a><span class="co">#colnames(ProgSpecCD45)=gsub(&quot;_&quot;, &quot;&quot;, colnames(ProgSpecCD45))</span></span>
<span id="cb584-29"><a href="writing-files-to-file.html#cb584-29"></a><span class="kw">rownames</span>(ProgSpecCD45)=ProgSpecCD45[ ,<span class="dv">1</span>]</span>
<span id="cb584-30"><a href="writing-files-to-file.html#cb584-30"></a>ProgSpecCD45B=<span class="kw">t</span>(ProgSpecCD45[<span class="kw">grep</span>(<span class="st">&quot;CD8&quot;</span>,<span class="kw">rownames</span>(ProgSpecCD45)), <span class="dv">-1</span> ])</span>
<span id="cb584-31"><a href="writing-files-to-file.html#cb584-31"></a>ProgSpecCD45B=<span class="kw">data.frame</span>(ProgSpecCD45B, <span class="dt">sample=</span><span class="kw">rownames</span>(ProgSpecCD45B))</span>
<span id="cb584-32"><a href="writing-files-to-file.html#cb584-32"></a></span>
<span id="cb584-33"><a href="writing-files-to-file.html#cb584-33"></a>df.Spatial3=<span class="kw">merge</span>(df.Spatial2, ProgSpecCD45B, <span class="dt">by.x=</span><span class="st">&quot;TumorID&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;sample&quot;</span>, <span class="dt">all=</span>T)</span>
<span id="cb584-34"><a href="writing-files-to-file.html#cb584-34"></a>df.Spatial3<span class="op">$</span>TumorID2=<span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot;&quot;</span>, df.Spatial3<span class="op">$</span>TumorID)</span>
<span id="cb584-35"><a href="writing-files-to-file.html#cb584-35"></a></span>
<span id="cb584-36"><a href="writing-files-to-file.html#cb584-36"></a>SummaryData<span class="op">$</span>TumorID2=<span class="kw">rownames</span>(SummaryData)</span>
<span id="cb584-37"><a href="writing-files-to-file.html#cb584-37"></a></span>
<span id="cb584-38"><a href="writing-files-to-file.html#cb584-38"></a>df.Spatial4=<span class="kw">merge</span>(df.Spatial3, SummaryData[ ,<span class="kw">c</span>(<span class="dv">6</span><span class="op">:</span><span class="dv">8</span>,<span class="dv">17</span><span class="op">:</span><span class="dv">19</span>, <span class="kw">ncol</span>(SummaryData))], <span class="dt">by.x=</span><span class="st">&quot;TumorID2&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;TumorID2&quot;</span>, <span class="dt">all=</span>T)</span>
<span id="cb584-39"><a href="writing-files-to-file.html#cb584-39"></a></span>
<span id="cb584-40"><a href="writing-files-to-file.html#cb584-40"></a><span class="kw">write.csv</span>(df.Spatial4, <span class="dt">file=</span><span class="st">&quot;../metadata/summary_new_210330.csv&quot;</span>)</span>
<span id="cb584-41"><a href="writing-files-to-file.html#cb584-41"></a></span>
<span id="cb584-42"><a href="writing-files-to-file.html#cb584-42"></a><span class="co">## merge he data from df.Spatial with the data from </span></span>
<span id="cb584-43"><a href="writing-files-to-file.html#cb584-43"></a></span>
<span id="cb584-44"><a href="writing-files-to-file.html#cb584-44"></a><span class="co">#df1=data.frame(CD8.WSI=df.Spatial$)</span></span>
<span id="cb584-45"><a href="writing-files-to-file.html#cb584-45"></a></span>
<span id="cb584-46"><a href="writing-files-to-file.html#cb584-46"></a> <span class="co">#  df2=data.frame(CD8.WSI=t(WSIvals[ 1, ]), </span></span>
<span id="cb584-47"><a href="writing-files-to-file.html#cb584-47"></a> <span class="co">#                 CD8Frac.WSI=t(WSIvalFracs[ ,2]), </span></span>
<span id="cb584-48"><a href="writing-files-to-file.html#cb584-48"></a> <span class="co">#                 CD8_EPorSMARatio.WSI=(WSIvals[1,  ]/colSums(WSIvals[2:4, ])),</span></span>
<span id="cb584-49"><a href="writing-files-to-file.html#cb584-49"></a> <span class="co">#                 CD8_AnyEPRatio.WSI=(WSIvals[1,  ]/colSums(WSIvals[2:3, ])),</span></span>
<span id="cb584-50"><a href="writing-files-to-file.html#cb584-50"></a> <span class="co">#                 CD8_EPRatio.WSI=(WSIvals[1,  ]/(WSIvals[2, ])),</span></span>
<span id="cb584-51"><a href="writing-files-to-file.html#cb584-51"></a> <span class="co">#                 CD8normTumSize=(WSIvals[1,  ]/df.Spatial))</span></span>
<span id="cb584-52"><a href="writing-files-to-file.html#cb584-52"></a> <span class="co"># #                )</span></span></code></pre></div>
<p>Below is the session info</p>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="writing-files-to-file.html#cb585-1"></a><span class="kw">sessionInfo</span>()</span>
<span id="cb585-2"><a href="writing-files-to-file.html#cb585-2"></a><span class="co">## R version 4.0.2 (2020-06-22)</span></span>
<span id="cb585-3"><a href="writing-files-to-file.html#cb585-3"></a><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span id="cb585-4"><a href="writing-files-to-file.html#cb585-4"></a><span class="co">## Running under: macOS  10.16</span></span>
<span id="cb585-5"><a href="writing-files-to-file.html#cb585-5"></a><span class="co">## </span></span>
<span id="cb585-6"><a href="writing-files-to-file.html#cb585-6"></a><span class="co">## Matrix products: default</span></span>
<span id="cb585-7"><a href="writing-files-to-file.html#cb585-7"></a><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span></span>
<span id="cb585-8"><a href="writing-files-to-file.html#cb585-8"></a><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span></span>
<span id="cb585-9"><a href="writing-files-to-file.html#cb585-9"></a><span class="co">## </span></span>
<span id="cb585-10"><a href="writing-files-to-file.html#cb585-10"></a><span class="co">## locale:</span></span>
<span id="cb585-11"><a href="writing-files-to-file.html#cb585-11"></a><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span id="cb585-12"><a href="writing-files-to-file.html#cb585-12"></a><span class="co">## </span></span>
<span id="cb585-13"><a href="writing-files-to-file.html#cb585-13"></a><span class="co">## attached base packages:</span></span>
<span id="cb585-14"><a href="writing-files-to-file.html#cb585-14"></a><span class="co">##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    </span></span>
<span id="cb585-15"><a href="writing-files-to-file.html#cb585-15"></a><span class="co">##  [8] datasets  methods   base     </span></span>
<span id="cb585-16"><a href="writing-files-to-file.html#cb585-16"></a><span class="co">## </span></span>
<span id="cb585-17"><a href="writing-files-to-file.html#cb585-17"></a><span class="co">## other attached packages:</span></span>
<span id="cb585-18"><a href="writing-files-to-file.html#cb585-18"></a><span class="co">##  [1] org.Hs.eg.db_3.11.4         pander_0.6.3               </span></span>
<span id="cb585-19"><a href="writing-files-to-file.html#cb585-19"></a><span class="co">##  [3] writexl_1.3.1               xlsx_0.6.4.2               </span></span>
<span id="cb585-20"><a href="writing-files-to-file.html#cb585-20"></a><span class="co">##  [5] tcR_2.3.2                   igraph_1.2.6               </span></span>
<span id="cb585-21"><a href="writing-files-to-file.html#cb585-21"></a><span class="co">##  [7] gridExtra_2.3               spatstat_1.64-1            </span></span>
<span id="cb585-22"><a href="writing-files-to-file.html#cb585-22"></a><span class="co">##  [9] rpart_4.1-15                nlme_3.1-149               </span></span>
<span id="cb585-23"><a href="writing-files-to-file.html#cb585-23"></a><span class="co">## [11] spatstat.data_1.4-3         scales_1.1.1               </span></span>
<span id="cb585-24"><a href="writing-files-to-file.html#cb585-24"></a><span class="co">## [13] RColorBrewer_1.1-2          rmarkdown_2.6              </span></span>
<span id="cb585-25"><a href="writing-files-to-file.html#cb585-25"></a><span class="co">## [15] reshape2_1.4.4              pamr_1.56.1                </span></span>
<span id="cb585-26"><a href="writing-files-to-file.html#cb585-26"></a><span class="co">## [17] survival_3.2-7              cluster_2.1.0              </span></span>
<span id="cb585-27"><a href="writing-files-to-file.html#cb585-27"></a><span class="co">## [19] limma_3.44.3                knitr_1.31                 </span></span>
<span id="cb585-28"><a href="writing-files-to-file.html#cb585-28"></a><span class="co">## [21] kableExtra_1.2.1            heatmap.plus_1.3           </span></span>
<span id="cb585-29"><a href="writing-files-to-file.html#cb585-29"></a><span class="co">## [23] GSVA_1.36.3                 gplots_3.1.0               </span></span>
<span id="cb585-30"><a href="writing-files-to-file.html#cb585-30"></a><span class="co">## [25] ggrepel_0.8.2               ggplot2_3.3.2              </span></span>
<span id="cb585-31"><a href="writing-files-to-file.html#cb585-31"></a><span class="co">## [27] GenVisR_1.20.0              forestplot_1.10.1          </span></span>
<span id="cb585-32"><a href="writing-files-to-file.html#cb585-32"></a><span class="co">## [29] checkmate_2.0.0             magrittr_1.5               </span></span>
<span id="cb585-33"><a href="writing-files-to-file.html#cb585-33"></a><span class="co">## [31] EnsDb.Hsapiens.v86_2.99.0   DT_0.17                    </span></span>
<span id="cb585-34"><a href="writing-files-to-file.html#cb585-34"></a><span class="co">## [33] dplyr_1.0.2                 DESeq2_1.28.1              </span></span>
<span id="cb585-35"><a href="writing-files-to-file.html#cb585-35"></a><span class="co">## [35] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        </span></span>
<span id="cb585-36"><a href="writing-files-to-file.html#cb585-36"></a><span class="co">## [37] matrixStats_0.57.0          colorspace_1.4-1           </span></span>
<span id="cb585-37"><a href="writing-files-to-file.html#cb585-37"></a><span class="co">## [39] Biostrings_2.56.0           XVector_0.28.0             </span></span>
<span id="cb585-38"><a href="writing-files-to-file.html#cb585-38"></a><span class="co">## [41] biomaRt_2.44.4              beeswarm_0.2.3             </span></span>
<span id="cb585-39"><a href="writing-files-to-file.html#cb585-39"></a><span class="co">## [43] AnnotationHub_2.20.2        BiocFileCache_1.12.1       </span></span>
<span id="cb585-40"><a href="writing-files-to-file.html#cb585-40"></a><span class="co">## [45] dbplyr_1.4.4                ensembldb_2.12.1           </span></span>
<span id="cb585-41"><a href="writing-files-to-file.html#cb585-41"></a><span class="co">## [47] AnnotationFilter_1.12.0     GenomicFeatures_1.40.1     </span></span>
<span id="cb585-42"><a href="writing-files-to-file.html#cb585-42"></a><span class="co">## [49] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        </span></span>
<span id="cb585-43"><a href="writing-files-to-file.html#cb585-43"></a><span class="co">## [51] HTSanalyzeR2_0.99.19        vcfR_1.12.0                </span></span>
<span id="cb585-44"><a href="writing-files-to-file.html#cb585-44"></a><span class="co">## [53] GSEABase_1.50.1             graph_1.66.0               </span></span>
<span id="cb585-45"><a href="writing-files-to-file.html#cb585-45"></a><span class="co">## [55] annotate_1.66.0             XML_3.99-0.5               </span></span>
<span id="cb585-46"><a href="writing-files-to-file.html#cb585-46"></a><span class="co">## [57] AnnotationDbi_1.50.3        IRanges_2.22.2             </span></span>
<span id="cb585-47"><a href="writing-files-to-file.html#cb585-47"></a><span class="co">## [59] S4Vectors_0.26.1            Biobase_2.48.0             </span></span>
<span id="cb585-48"><a href="writing-files-to-file.html#cb585-48"></a><span class="co">## [61] BiocGenerics_0.34.0        </span></span>
<span id="cb585-49"><a href="writing-files-to-file.html#cb585-49"></a><span class="co">## </span></span>
<span id="cb585-50"><a href="writing-files-to-file.html#cb585-50"></a><span class="co">## loaded via a namespace (and not attached):</span></span>
<span id="cb585-51"><a href="writing-files-to-file.html#cb585-51"></a><span class="co">##   [1] shinydashboard_0.7.1          tidyselect_1.1.0             </span></span>
<span id="cb585-52"><a href="writing-files-to-file.html#cb585-52"></a><span class="co">##   [3] RSQLite_2.2.1                 htmlwidgets_1.5.2            </span></span>
<span id="cb585-53"><a href="writing-files-to-file.html#cb585-53"></a><span class="co">##   [5] BiocParallel_1.22.0           gmp_0.6-1                    </span></span>
<span id="cb585-54"><a href="writing-files-to-file.html#cb585-54"></a><span class="co">##   [7] munsell_0.5.0                 codetools_0.2-16             </span></span>
<span id="cb585-55"><a href="writing-files-to-file.html#cb585-55"></a><span class="co">##   [9] preprocessCore_1.50.0         miniUI_0.1.1.1               </span></span>
<span id="cb585-56"><a href="writing-files-to-file.html#cb585-56"></a><span class="co">##  [11] withr_2.3.0                   Category_2.54.0              </span></span>
<span id="cb585-57"><a href="writing-files-to-file.html#cb585-57"></a><span class="co">##  [13] highr_0.8                     rstudioapi_0.11              </span></span>
<span id="cb585-58"><a href="writing-files-to-file.html#cb585-58"></a><span class="co">##  [15] robustbase_0.93-6             splots_1.54.0                </span></span>
<span id="cb585-59"><a href="writing-files-to-file.html#cb585-59"></a><span class="co">##  [17] tensor_1.5                    rJava_0.9-13                 </span></span>
<span id="cb585-60"><a href="writing-files-to-file.html#cb585-60"></a><span class="co">##  [19] labeling_0.4.2                GenomeInfoDbData_1.2.3       </span></span>
<span id="cb585-61"><a href="writing-files-to-file.html#cb585-61"></a><span class="co">##  [21] polyclip_1.10-0               hwriter_1.3.2                </span></span>
<span id="cb585-62"><a href="writing-files-to-file.html#cb585-62"></a><span class="co">##  [23] farver_2.0.3                  bit64_4.0.5                  </span></span>
<span id="cb585-63"><a href="writing-files-to-file.html#cb585-63"></a><span class="co">##  [25] vctrs_0.3.4                   generics_0.0.2               </span></span>
<span id="cb585-64"><a href="writing-files-to-file.html#cb585-64"></a><span class="co">##  [27] xfun_0.21                     R6_2.4.1                     </span></span>
<span id="cb585-65"><a href="writing-files-to-file.html#cb585-65"></a><span class="co">##  [29] locfit_1.5-9.4                msigdbr_7.2.1                </span></span>
<span id="cb585-66"><a href="writing-files-to-file.html#cb585-66"></a><span class="co">##  [31] spatstat.utils_1.17-0         bitops_1.0-6                 </span></span>
<span id="cb585-67"><a href="writing-files-to-file.html#cb585-67"></a><span class="co">##  [33] fgsea_1.14.0                  assertthat_0.2.1             </span></span>
<span id="cb585-68"><a href="writing-files-to-file.html#cb585-68"></a><span class="co">##  [35] promises_1.1.1                pinfsc50_1.2.0               </span></span>
<span id="cb585-69"><a href="writing-files-to-file.html#cb585-69"></a><span class="co">##  [37] gtable_0.3.0                  affy_1.66.0                  </span></span>
<span id="cb585-70"><a href="writing-files-to-file.html#cb585-70"></a><span class="co">##  [39] goftest_1.2-2                 cellHTS2_2.52.0              </span></span>
<span id="cb585-71"><a href="writing-files-to-file.html#cb585-71"></a><span class="co">##  [41] rlang_0.4.8                   genefilter_1.70.0            </span></span>
<span id="cb585-72"><a href="writing-files-to-file.html#cb585-72"></a><span class="co">##  [43] splines_4.0.2                 rtracklayer_1.48.0           </span></span>
<span id="cb585-73"><a href="writing-files-to-file.html#cb585-73"></a><span class="co">##  [45] lazyeval_0.2.2                abind_1.4-5                  </span></span>
<span id="cb585-74"><a href="writing-files-to-file.html#cb585-74"></a><span class="co">##  [47] BiocManager_1.30.10           yaml_2.2.1                   </span></span>
<span id="cb585-75"><a href="writing-files-to-file.html#cb585-75"></a><span class="co">##  [49] crosstalk_1.1.0.1             backports_1.1.10             </span></span>
<span id="cb585-76"><a href="writing-files-to-file.html#cb585-76"></a><span class="co">##  [51] httpuv_1.5.4                  RBGL_1.64.0                  </span></span>
<span id="cb585-77"><a href="writing-files-to-file.html#cb585-77"></a><span class="co">##  [53] tools_4.0.2                   bookdown_0.21                </span></span>
<span id="cb585-78"><a href="writing-files-to-file.html#cb585-78"></a><span class="co">##  [55] affyio_1.58.0                 ellipsis_0.3.1               </span></span>
<span id="cb585-79"><a href="writing-files-to-file.html#cb585-79"></a><span class="co">##  [57] Rcpp_1.0.5                    plyr_1.8.6                   </span></span>
<span id="cb585-80"><a href="writing-files-to-file.html#cb585-80"></a><span class="co">##  [59] progress_1.2.2                zlibbioc_1.34.0              </span></span>
<span id="cb585-81"><a href="writing-files-to-file.html#cb585-81"></a><span class="co">##  [61] purrr_0.3.4                   RCurl_1.98-1.2               </span></span>
<span id="cb585-82"><a href="writing-files-to-file.html#cb585-82"></a><span class="co">##  [63] prettyunits_1.1.1             deldir_0.1-29                </span></span>
<span id="cb585-83"><a href="writing-files-to-file.html#cb585-83"></a><span class="co">##  [65] openssl_1.4.3                 viridis_0.5.1                </span></span>
<span id="cb585-84"><a href="writing-files-to-file.html#cb585-84"></a><span class="co">##  [67] data.table_1.13.2             colourpicker_1.1.0           </span></span>
<span id="cb585-85"><a href="writing-files-to-file.html#cb585-85"></a><span class="co">##  [69] mvtnorm_1.1-1                 ProtGenerics_1.20.0          </span></span>
<span id="cb585-86"><a href="writing-files-to-file.html#cb585-86"></a><span class="co">##  [71] xlsxjars_0.6.1                hms_0.5.3                    </span></span>
<span id="cb585-87"><a href="writing-files-to-file.html#cb585-87"></a><span class="co">##  [73] mime_0.9                      evaluate_0.14                </span></span>
<span id="cb585-88"><a href="writing-files-to-file.html#cb585-88"></a><span class="co">##  [75] xtable_1.8-4                  compiler_4.0.2               </span></span>
<span id="cb585-89"><a href="writing-files-to-file.html#cb585-89"></a><span class="co">##  [77] tibble_3.0.4                  KernSmooth_2.23-17           </span></span>
<span id="cb585-90"><a href="writing-files-to-file.html#cb585-90"></a><span class="co">##  [79] crayon_1.3.4                  htmltools_0.5.0              </span></span>
<span id="cb585-91"><a href="writing-files-to-file.html#cb585-91"></a><span class="co">##  [81] mgcv_1.8-33                   pcaPP_1.9-73                 </span></span>
<span id="cb585-92"><a href="writing-files-to-file.html#cb585-92"></a><span class="co">##  [83] later_1.1.0.1                 geneplotter_1.66.0           </span></span>
<span id="cb585-93"><a href="writing-files-to-file.html#cb585-93"></a><span class="co">##  [85] rrcov_1.5-5                   DBI_1.1.0                    </span></span>
<span id="cb585-94"><a href="writing-files-to-file.html#cb585-94"></a><span class="co">##  [87] MASS_7.3-53                   rappdirs_0.3.1               </span></span>
<span id="cb585-95"><a href="writing-files-to-file.html#cb585-95"></a><span class="co">##  [89] Matrix_1.2-18                 permute_0.9-5                </span></span>
<span id="cb585-96"><a href="writing-files-to-file.html#cb585-96"></a><span class="co">##  [91] vsn_3.56.0                    RankProd_3.14.0              </span></span>
<span id="cb585-97"><a href="writing-files-to-file.html#cb585-97"></a><span class="co">##  [93] pkgconfig_2.0.3               GenomicAlignments_1.24.0     </span></span>
<span id="cb585-98"><a href="writing-files-to-file.html#cb585-98"></a><span class="co">##  [95] xml2_1.3.2                    foreach_1.5.1                </span></span>
<span id="cb585-99"><a href="writing-files-to-file.html#cb585-99"></a><span class="co">##  [97] prada_1.63.0                  stringdist_0.9.6.3           </span></span>
<span id="cb585-100"><a href="writing-files-to-file.html#cb585-100"></a><span class="co">##  [99] webshot_0.5.2                 rvest_0.3.6                  </span></span>
<span id="cb585-101"><a href="writing-files-to-file.html#cb585-101"></a><span class="co">## [101] BioNet_1.47.0                 stringr_1.4.0                </span></span>
<span id="cb585-102"><a href="writing-files-to-file.html#cb585-102"></a><span class="co">## [103] VariantAnnotation_1.34.0      digest_0.6.26                </span></span>
<span id="cb585-103"><a href="writing-files-to-file.html#cb585-103"></a><span class="co">## [105] vegan_2.5-6                   fastmatch_1.1-0              </span></span>
<span id="cb585-104"><a href="writing-files-to-file.html#cb585-104"></a><span class="co">## [107] curl_4.3                      shiny_1.5.0                  </span></span>
<span id="cb585-105"><a href="writing-files-to-file.html#cb585-105"></a><span class="co">## [109] Rsamtools_2.4.0               gtools_3.8.2                 </span></span>
<span id="cb585-106"><a href="writing-files-to-file.html#cb585-106"></a><span class="co">## [111] FField_0.1.0                  jsonlite_1.7.1               </span></span>
<span id="cb585-107"><a href="writing-files-to-file.html#cb585-107"></a><span class="co">## [113] lifecycle_0.2.0               viridisLite_0.3.0            </span></span>
<span id="cb585-108"><a href="writing-files-to-file.html#cb585-108"></a><span class="co">## [115] askpass_1.1                   BSgenome_1.56.0              </span></span>
<span id="cb585-109"><a href="writing-files-to-file.html#cb585-109"></a><span class="co">## [117] pillar_1.4.6                  lattice_0.20-41              </span></span>
<span id="cb585-110"><a href="writing-files-to-file.html#cb585-110"></a><span class="co">## [119] KEGGREST_1.28.0               fastmap_1.0.1                </span></span>
<span id="cb585-111"><a href="writing-files-to-file.html#cb585-111"></a><span class="co">## [121] httr_1.4.2                    DEoptimR_1.0-8               </span></span>
<span id="cb585-112"><a href="writing-files-to-file.html#cb585-112"></a><span class="co">## [123] GO.db_3.11.4                  interactiveDisplayBase_1.26.3</span></span>
<span id="cb585-113"><a href="writing-files-to-file.html#cb585-113"></a><span class="co">## [125] glue_1.4.2                    shinythemes_1.1.2            </span></span>
<span id="cb585-114"><a href="writing-files-to-file.html#cb585-114"></a><span class="co">## [127] png_0.1-7                     iterators_1.0.13             </span></span>
<span id="cb585-115"><a href="writing-files-to-file.html#cb585-115"></a><span class="co">## [129] BiocVersion_3.11.1            bit_4.0.4                    </span></span>
<span id="cb585-116"><a href="writing-files-to-file.html#cb585-116"></a><span class="co">## [131] stringi_1.5.3                 blob_1.2.1                   </span></span>
<span id="cb585-117"><a href="writing-files-to-file.html#cb585-117"></a><span class="co">## [133] caTools_1.18.0                memoise_1.1.0                </span></span>
<span id="cb585-118"><a href="writing-files-to-file.html#cb585-118"></a><span class="co">## [135] Rmpfr_0.8-1                   ape_5.4-1</span></span></code></pre></div>

</div>



</div>
            </section>

          </div>
        </div>
      </div>
<a href="cnv-calling.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="list-of-figures.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RatDCIS-bookdown.pdf", "RatDCIS-bookdown.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
